
Author: Robert Millan

An ugly kludge.  Should this be merged upstream?

Index: grub-pc_2.00-SprezzOS2/grub-core/kern/emu/hostdisk.c
===================================================================
--- grub-pc_2.00-SprezzOS2.orig/grub-core/kern/emu/hostdisk.c	2012-06-25 04:32:04.000000000 -0400
+++ grub-pc_2.00-SprezzOS2/grub-core/kern/emu/hostdisk.c	2012-09-30 11:08:55.947601676 -0400
@@ -1331,6 +1331,18 @@
 	  continue;
 	}
 
+      if (! strncmp (p, "/dev/fd", sizeof ("/dev/fd") - 1))
+	{
+	  char *q = p + sizeof ("/dev/fd") - 1;
+	  if (*q >= '0' && *q <= '9')
+	    {
+	      free (map[drive].drive);
+	      map[drive].drive = NULL;
+	      grub_util_info ("`%s' looks like a floppy drive, skipping", p);
+	      continue;
+	    }
+	}
+
 #ifdef __linux__
       /* On Linux, the devfs uses symbolic links horribly, and that
 	 confuses the interface very much, so use realpath to expand
