Index: grub-pc_2.00-SprezzOS2/util/grub-mkconfig.in
===================================================================
--- grub-pc_2.00-SprezzOS2.orig/util/grub-mkconfig.in	2012-04-05 13:41:07.000000000 -0400
+++ grub-pc_2.00-SprezzOS2/util/grub-mkconfig.in	2012-09-30 11:08:55.960935009 -0400
@@ -253,6 +253,10 @@
   esac
 done
 
+if [ "x${grub_cfg}" != "x" ] && ! grep "^password " ${grub_cfg}.new >/dev/null; then
+  chmod 444 ${grub_cfg}.new || true
+fi
+
 if test "x${grub_cfg}" != "x" ; then
   if ! ${grub_script_check} ${grub_cfg}.new; then
     # TRANSLATORS: %s is replaced by filename
