Description: Add support for GRUB_CMDLINE_GNUMACH
Author: Samuel Thibault <samuel.thibault@ens-lyon.org>
Origin: upstream, http://bazaar.launchpad.net/~vcs-imports/grub/grub2-bzr/revision/3942
Last-Update: 2012-03-05

Index: b/docs/grub.texi
===================================================================
--- a/docs/grub.texi
+++ b/docs/grub.texi
@@ -1167,6 +1167,9 @@
 As @samp{GRUB_CMDLINE_LINUX} and @samp{GRUB_CMDLINE_LINUX_DEFAULT}, but for
 NetBSD.
 
+@item GRUB_CMDLINE_GNUMACH
+As @samp{GRUB_CMDLINE_LINUX}, but for GNU Mach.
+
 @item GRUB_CMDLINE_XEN
 @itemx GRUB_CMDLINE_XEN_DEFAULT
 The values of these options are appended to the values of
Index: b/util/grub-mkconfig.in
===================================================================
--- a/util/grub-mkconfig.in
+++ b/util/grub-mkconfig.in
@@ -244,6 +244,7 @@
   GRUB_CMDLINE_LINUX_XEN_REPLACE_DEFAULT \
   GRUB_CMDLINE_NETBSD \
   GRUB_CMDLINE_NETBSD_DEFAULT \
+  GRUB_CMDLINE_GNUMACH \
   GRUB_TERMINAL_INPUT \
   GRUB_TERMINAL_OUTPUT \
   GRUB_SERIAL_COMMAND \
Index: b/util/grub.d/10_hurd.in
===================================================================
--- a/util/grub.d/10_hurd.in
+++ b/util/grub.d/10_hurd.in
@@ -84,7 +84,7 @@
   message="$(gettext_printf "Loading GNU Mach ...")"
   cat << EOF
 	echo		'$message'
-	multiboot	${kernel} root=device:${GRUB_DEVICE#/dev/}
+	multiboot	${kernel} root=device:${GRUB_DEVICE#/dev/} ${GRUB_CMDLINE_GNUMACH}
 EOF
   save_default_entry | sed -e "s/^/\t/"
   prepare_grub_to_access_device ${GRUB_DEVICE} | sed -e "s/^/\t/"
@@ -108,7 +108,7 @@
   message="$(gettext_printf "Loading GNU Mach ...")"
   cat << EOF
 	echo		'$message'
-	multiboot	${kernel} root=device:${GRUB_DEVICE#/dev/} -s
+	multiboot	${kernel} root=device:${GRUB_DEVICE#/dev/} -s ${GRUB_CMDLINE_GNUMACH}
 EOF
   prepare_grub_to_access_device ${GRUB_DEVICE} | sed -e "s/^/\t/"
   message="$(gettext_printf "Loading the Hurd ...")"
