Index: grub-pc_2.00-SprezzOS2/util/getroot.c
===================================================================
--- grub-pc_2.00-SprezzOS2.orig/util/getroot.c	2012-09-30 11:10:26.860933370 -0400
+++ grub-pc_2.00-SprezzOS2/util/getroot.c	2012-09-30 11:11:24.477598998 -0400
@@ -269,7 +269,7 @@
   zpool_handle_t *zpool;
   libzfs_handle_t *libzfs;
   nvlist_t *config, *vdev_tree;
-  nvlist_t **children, **path;
+  nvlist_t **children;
   unsigned int nvlist_count;
   unsigned int i;
   char *device = 0;
