---
 rtkit-daemon.c |    3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

--- rtkit.orig/rtkit-daemon.c
+++ rtkit/rtkit-daemon.c
@@ -1766,8 +1766,7 @@ static int drop_privileges(void) {
                 cap_t caps;
                 const cap_value_t cap_values[] = {
                         CAP_SYS_NICE,             /* Needed for obvious reasons */
-                        CAP_DAC_READ_SEARCH,      /* Needed so that we can verify resource limits */
-                        CAP_SYS_PTRACE            /* Needed so that we can read /proc/$$/exe. Linux is weird. */
+                        CAP_DAC_READ_SEARCH      /* Needed so that we can verify resource limits */
                 };
 
                 /* Third, say that we want to keep caps */
