Index: scorched3d-43.2a.dfsg/src/client/client/SecureID.cpp
===================================================================
--- scorched3d-43.2a.dfsg.orig/src/client/client/SecureID.cpp	2012-05-22 11:05:46.000000000 +0000
+++ scorched3d-43.2a.dfsg/src/client/client/SecureID.cpp	2012-05-22 11:07:03.532930546 +0000
@@ -1,6 +1,7 @@
 #include <client/SecureID.h>
 #include <net/NetInterface.h>
 #include <common/Defines.h>
+#include <unistd.h>
 
 #ifdef _WIN32
 
