Index: scorched3d/src/client/client/SecureID.cpp
===================================================================
--- scorched3d.orig/src/client/client/SecureID.cpp	2011-02-06 22:23:50.280579227 +0100
+++ scorched3d/src/client/client/SecureID.cpp	2011-02-06 22:24:48.612082583 +0100
@@ -27,7 +27,7 @@
     return Key;
 }
 
-#elif defined(__DARWIN__)
+#elif defined(__DARWIN__) || defined(__FreeBSD_kernel__)
 
 #include <stdio.h>
 #include <sys/types.h>
