# see #325341
# Thanks to Christophe Rhodes <csr21@cam.ac.uk>
--- netpbm-free-10.0/pnm/pstopnm.c	2004-01-05 10:39:56.000000000 +0000
+++ netpbm-free-10.0.new/pnm/pstopnm.c	2005-08-27 20:45:03.000000000 +0100
@@ -39,6 +39,7 @@
     char *input_filespec;  /* Filespecs of input files */
     unsigned int forceplain;
     struct box extract_box;
+    int extract_box_supplied;
     unsigned int nocrop;
     unsigned int format_type;
     unsigned int verbose;
@@ -145,8 +146,9 @@
         else cmdline_p->extract_box.urx = urx * 72;
         if (!urySpec) cmdline_p->extract_box.ury = 720;
         else cmdline_p->extract_box.ury = ury * 72;
+        cmdline_p->extract_box_supplied = TRUE;
     } else {
-        cmdline_p->extract_box.llx = -1;
+        cmdline_p->extract_box_supplied = FALSE;
     }
 
 }
@@ -284,26 +286,28 @@
 
     struct box retval;
 
-    if (cmdline.extract_box.llx != -1)
+    if (cmdline.extract_box_supplied == TRUE)
         /* User told us what box to extract, so that's what we'll do */
         retval = cmdline_extract_box;
     else {
         /* Try to get the bounding box from the DSC %%BoundingBox
            statement (A Postscript comment) in the input.
         */
-        struct box ps_bb;  /* Box described by %%BoundingBox stmt in input */
+        struct box ps_bb;  /* Box described by %%BoundingBox stmt in
+                              input */
+        int found_BB; /* logical */
 
+        found_BB = FALSE;
         if (strcmp(input_filespec, "-") == 0)
             /* Can't read stdin, because we need it positioned for the 
                Ghostscript interpreter to read it.
             */
-            ps_bb.llx = -1;
+            ;
         else {
             FILE *infile;
-            int found_BB, eof;  /* logical */
+            int eof;  /* logical */
             infile = pm_openr(input_filespec);
             
-            found_BB = FALSE;
             eof = FALSE;
             while (!eof && !found_BB) {
                 char line[200];
@@ -322,13 +326,12 @@
             fclose(infile);
 
             if (!found_BB) {
-                ps_bb.llx = -1;
                 pm_message("Warning: no %%%%BoundingBox statement "
                            "in the input or command line.\n"
                            "Will use defaults");
             }
         }
-        if (ps_bb.llx != -1) {
+        if (found_BB) {
             if (verbose)
                 pm_message("Using %%%%BoundingBox statement from input.");
             retval = ps_bb;
