# see #405048
# Thanks to Håkon Stordahl <hakon@stordahl.org>
--- a/pbm/pbmtextps.c	2004-01-05 10:39:56.000000000 +0000
+++ b/pbm/pbmtextps.c	2010-05-05 23:49:57.000000000 +0000
@@ -265,7 +265,7 @@
 
   if(fwrite(ps, 1, strlen(ps), psfile) != strlen(ps))
     pm_error("Can't write postscript to temp file");
-  fclose(psfile);
+  fflush(psfile);
 
   if (cmdl->verbose)
     pm_message("Running Postscript interpreter '%s'", com);
@@ -275,7 +275,7 @@
 
   free(com);
 
-  pbmfile = pm_openw(pbmfname);
+  pbmfile = pm_openr(pbmfname);
 
   com = crop_command();
   if(com) {
