# see #483921
# Thanks to Rik Snel <rsnel@cube.dyndns.org>
--- a/pnm/giftopnm.c.old      2008-06-01 10:41:52.000000000 +0200
+++ b/pnm/giftopnm.c  2008-06-01 10:35:26.000000000 +0200
@@ -391,7 +391,7 @@
         done = FALSE;
         retval = 0;
     } else {
-        if ( (curbit+codeSize) >= lastbit) {
+        while ( (curbit+codeSize) >= lastbit) {
             unsigned int count;
             unsigned int assumed_count;
             bool eof;
