# see #495716
# Thanks to Anders Boström <anders@bostrom.dyndns.org>
--- a/pnm/pnmindex.orig       2008-06-01 14:55:13.000000000 +0200
+++ b/pnm/pnmindex    2008-08-19 23:32:30.000000000 +0200
@@ -108,7 +108,8 @@
 
 for i in "$@"; do
 
-    description=(`pnmfile $i`)
+    cp "$i" $tmpfile
+    description=(`pnmfile $tmpfile`)
 
     format=${description[1]}
     width=${description[3]}
@@ -119,10 +120,8 @@
         exit $?
     fi
 
-    if [ $width -le $size ] && \
-       [ $height -le $size ]; then
-        cat $i > $tmpfile
-    else
+    if [ $width -gt $size ] || \
+       [ $height -gt $size ]; then
         case $format in
 
         PBM) 
