# see #531626
# Thanks to Vladimir Rutsky <altsysrq@gmail.com>
--- netpbm-free-10.0/ppm/pcxtoppm.c.orig	2003-08-17 00:10:10.000000000 +0400
+++ netpbm-free-10.0/ppm/pcxtoppm.c	2009-06-02 22:34:09.000000000 +0400
@@ -481,14 +481,14 @@
     }
 
     redrow   = (unsigned char *)
-        pm_allocrow(BytesPerLine, sizeof(unsigned char*));
+        pm_allocrow(BytesPerLine, sizeof(unsigned char));
     greenrow = (unsigned char *)
-        pm_allocrow(BytesPerLine, sizeof(unsigned char*));
+        pm_allocrow(BytesPerLine, sizeof(unsigned char));
     bluerow  = (unsigned char *)
-        pm_allocrow(BytesPerLine, sizeof(unsigned char*));
+        pm_allocrow(BytesPerLine, sizeof(unsigned char));
     if( Planes == 4 )
         intensityrow = (unsigned char *)
-            pm_allocrow(BytesPerLine, sizeof(unsigned char*));
+            pm_allocrow(BytesPerLine, sizeof(unsigned char));
     else
         intensityrow = (unsigned char *)NULL;
 
