# see #531630
# Thanks to Vladimir Rutsky <altsysrq@gmail.com>
--- netpbm-free-10.0/ppm/pcxtoppm.c.orig	2003-08-17 00:10:10.000000000 +0400
+++ netpbm-free-10.0/ppm/pcxtoppm.c	2009-06-03 00:01:33.000000000 +0400
@@ -668,9 +668,11 @@
    raster byte that follows.  If the upper two bits are not 11, the byte 
    _is_ a raster byte, with repetition count 1.
 
-   A run can't span rows, but it can span planes within a row.  That's
+   A run can span planes within a row and can span rows.  That's
    why 'repetitionsLeft' and 'c' are static variables in this
    subroutine.
+   Note than spanning of rows is against ZSoft PCX file format technical 
+   manual.
 -----------------------------------------------------------------------------*/
     static int repetitionsLeft = 0;
     static int c;
