# see #602360
# Thanks to Bart Massey <bart@po8.org>
--- a/pnm/pnmquant.dist	2010-11-03 10:27:30.000000000 -0700
+++ b/pnm/pnmquant	2010-11-03 21:58:20.000000000 -0700
@@ -9,6 +9,7 @@
 use Getopt::Long;
 use File::Temp "tempfile";
 use File::Spec;
+use File::Copy;
 use Fcntl ":seek";
 
 my ($TRUE, $FALSE) = (1,0);
 
@@ -39,7 +40,13 @@
 if (@ARGV > 1) {
     $infile = $ARGV[1];
 } else {
-    $infile = "-";
+    my ($savefileFh, $savefileSpec) =
+	tempfile("pnmquantSaveXXXX", 
+		 SUFFIX => ".pnm",
+		 UNLINK => $TRUE,
+		 DIR => File::Spec->tmpdir());
+    copy(\*STDIN, $savefileFh);
+    $infile = $savefileSpec;
 }
 
 my $averageOpt;
