--- db-5.1.25.orig/dist/configure
+++ db-5.1.25/dist/configure
@@ -17473,7 +17473,7 @@ $as_echo "$_cur" >&6; }
 done
 _ACJNI_FOLLOWED="$_cur"
 
-_JTOPDIR=`echo "$_ACJNI_FOLLOWED" | sed -e 's://*:/:g' -e 's:/[^/]*$::'`
+_JTOPDIR=`echo "$_ACJNI_JAVAC" | sed -e 's://*:/:g' -e 's:/[^/]*$::'`
 case "$host_os" in
 	darwin*)	_JTOPDIR=`echo "$_JTOPDIR" | sed -e 's:/[^/]*$::'`
 			_JINC="$_JTOPDIR/Headers";;
--- db-5.1.25.orig/dist/aclocal_java/ac_jni_include_dirs.m4
+++ db-5.1.25/dist/aclocal_java/ac_jni_include_dirs.m4
@@ -44,7 +44,7 @@ case "$_ACJNI_JAVAC" in
 esac
 
 _ACJNI_FOLLOW_SYMLINKS("$_ACJNI_JAVAC")
-_JTOPDIR=`echo "$_ACJNI_FOLLOWED" | sed -e 's://*:/:g' -e 's:/[[^/]]*$::'`
+_JTOPDIR=`echo "$_ACJNI_JAVAC" | sed -e 's://*:/:g' -e 's:/[[^/]]*$::'`
 case "$host_os" in
 	darwin*)	_JTOPDIR=`echo "$_JTOPDIR" | sed -e 's:/[[^/]]*$::'`
 			_JINC="$_JTOPDIR/Headers";;
