commit e34a6d141b30eb02e2e2df7d96161dd1877d24fd
Author: Hendrik Sattler <post@hendrik-sattler.de>
Date:   Sun Feb 8 23:31:13 2009 +0100

    change version for cmake build

diff --git a/CMakeLists.txt b/CMakeLists.txt
index ba01757..d2e84da 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -6,7 +6,7 @@ project ( openobex C )
 # The project version
 #
 set ( VERSION_MAJOR 1 )
-set ( VERSION_MINOR 4 )
+set ( VERSION_MINOR 5 )
 set ( VERSION_PATCH 0 )
 
 set ( SHORT_VERSION "${VERSION_MAJOR}.${VERSION_MINOR}" )
