commit 50214c540be7e2d6b10222338a58290d6398fa76
Author: Hendrik Sattler <post@hendrik-sattler.de>
Date:   Sun Feb 8 23:39:26 2009 +0100

    fix undefined use of sprintf

diff --git a/ircp/dirtraverse.c b/ircp/dirtraverse.c
index 7f95287..f6c1cf3 100644
--- a/ircp/dirtraverse.c
+++ b/ircp/dirtraverse.c
@@ -59,7 +59,8 @@ int visit_dir(char *path, visit_cb cb, void *userdata)
 				if( ret < 0)
 					goto out;
 				len = strlen(path);
-				sprintf(path, "%s%s/", path, dirent->d_name);
+				strcat(path, "/");
+				strcat(path, dirent->d_name);
 				ret = visit_dir(t, cb, userdata);
 				if(ret < 0)
 					goto out;
