commit 7b35b38c83470fb50deb117249801b00af78902e
Author: Hendrik Sattler <post@hendrik-sattler.de>
Date:   Wed Feb 11 21:13:40 2009 +0100

    fix finding bluetooth on Linux

diff --git a/CMakeModules/FindBluetooth.cmake b/CMakeModules/FindBluetooth.cmake
index cbd3666..4c36282 100644
--- a/CMakeModules/FindBluetooth.cmake
+++ b/CMakeModules/FindBluetooth.cmake
@@ -16,6 +16,14 @@ if ( CMAKE_SYSTEM_NAME STREQUAL "Linux" )
   if ( PKG_CONFIG_FOUND )
     pkg_check_modules ( PKGCONFIG_BLUEZ bluez )
   endif ( PKG_CONFIG_FOUND )
+  find_path ( Bluetooth_INCLUDE_DIRS
+    NAMES
+      bluetooth/bluetooth.h
+    PATH_SUFFIXES
+      include
+  )
+  mark_as_advanced ( Bluetooth_INCLUDE_DIRS )
+
   if ( PKGCONFIG_BLUEZ_FOUND )
     foreach ( i ${PKGCONFIG_BLUEZ_LIBRARIES} )
       find_library ( ${i}_LIBRARY
@@ -32,14 +40,6 @@ if ( CMAKE_SYSTEM_NAME STREQUAL "Linux" )
     add_definitions ( -DHAVE_SDPLIB )
 
   else ( PKGCONFIG_BLUEZ_FOUND )
-    find_path ( Bluetooth_INCLUDE_DIRS
-      NAMES
-        bluetooth/bluetooth.h
-      PATH_SUFFIXES
-        include
-    )
-    mark_as_advanced ( Bluetooth_INCLUDE_DIRS )
-
     find_library ( bluetooth_LIBRARY
       NAMES
         bluetooth
