Author: Matthias Klose <doko@ubuntu.com>
Description: Explicitely use bash for the ld testsuite.
--- a/ld/testsuite/config/default.exp
+++ b/ld/testsuite/config/default.exp
@@ -120,10 +120,10 @@
 	#makefile rules, with embedded shell variable expansions.
 	#make wants $$shell_var, we want $shell_var ...
 	set cmd "host='$target_triplet' && . $srcdir/../configure.host && sed -e 's,\\\$\\\$,\$,g' <<EOF\n\$$varname\nEOF"
-	set status [catch "exec sh -c [list $cmd]" result]
+	set status [catch "exec bash -c [list $cmd]" result]
 	if $status { error "Error getting native link files: $result" }
 	set cmd "CC='$CC' && eval echo \"$result\""
-	set status [catch "exec sh -c [list $cmd]" result]
+	set status [catch "exec bash -c [list $cmd]" result]
 	if $status { error "Error getting native link files: $result" }
 	set $varname $result
 	send_log "$varname = $result\n"
@@ -133,7 +133,7 @@
 proc get_target_emul {} {
     global target_triplet
     global srcdir
-    set status [catch "exec sh -c \"targ='$target_triplet' && . $srcdir/../configure.tgt && echo \\\$targ_emul\"" result]
+    set status [catch "exec bash -c \"targ='$target_triplet' && . $srcdir/../configure.tgt && echo \\\$targ_emul\"" result]
     if $status { error "Error getting emulation name: $result" }
     return $result
 }
