--- a/ld/configure.in
+++ b/ld/configure.in
@@ -39,7 +39,9 @@
  *) TARGET_SYSTEM_ROOT=$with_sysroot ;;
  esac
 
+ if test "x$TARGET_SYSTEM_ROOT" != x/; then
  TARGET_SYSTEM_ROOT_DEFINE='-DTARGET_SYSTEM_ROOT=\"$(TARGET_SYSTEM_ROOT)\"'
+ fi
  use_sysroot=yes
 
  if test "x$prefix" = xNONE; then
--- a/ld/configure
+++ b/ld/configure
@@ -4282,7 +4282,9 @@
  *) TARGET_SYSTEM_ROOT=$with_sysroot ;;
  esac
 
+ if test "x$TARGET_SYSTEM_ROOT" != x/; then
  TARGET_SYSTEM_ROOT_DEFINE='-DTARGET_SYSTEM_ROOT=\"$(TARGET_SYSTEM_ROOT)\"'
+ fi
  use_sysroot=yes
 
  if test "x$prefix" = xNONE; then
--- a/ld/ldmain.c
+++ b/ld/ldmain.c
@@ -225,8 +225,8 @@
     {
       if (*TARGET_SYSTEM_ROOT == 0)
 	{
-	  einfo ("%P%F: this linker was not configured to use sysroots\n");
 	  ld_sysroot = "";
+	  ld_canon_sysroot = "";
 	}
       else
 	ld_canon_sysroot = lrealpath (ld_sysroot);
