From 40a429d2b408e93956cd41061a30a545d31be860 Mon Sep 17 00:00:00 2001
From: Sam Hartman <hartmans@debian.org>
Date: Mon, 26 Dec 2011 18:19:19 -0500
Subject: Debian: set AI_ADDRCONFIG for kprop slave

---
 src/slave/kprop.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/src/slave/kprop.c b/src/slave/kprop.c
index 473f9f23a637d15680b928294a1463793f20d34c..6fbfc1e 100644
--- a/src/slave/kprop.c
+++ b/src/slave/kprop.c
@@ -321,6 +321,7 @@ open_connection(krb5_context context, char *host, int *fd)
     int error;
 
     memset(&hints, 0, sizeof(hints));
+    hints.ai_flags = AI_ADDRCONFIG;
     hints.ai_family = PF_UNSPEC;
     hints.ai_socktype = SOCK_STREAM;
     hints.ai_flags = AI_ADDRCONFIG;
