Index: shared-mime-info-1.0/freedesktop.org.xml.in
===================================================================
--- shared-mime-info-1.0.orig/freedesktop.org.xml.in	2012-01-09 17:03:17.000000000 +0100
+++ shared-mime-info-1.0/freedesktop.org.xml.in	2012-06-07 18:04:45.863226901 +0200
@@ -184,7 +184,7 @@
     <acronym>PDF</acronym>
     <expanded-acronym>Portable Document Format</expanded-acronym>
     <generic-icon name="x-office-document"/>
-    <magic priority="50">
+    <magic priority="60">
       <match type="string" value="%PDF-" offset="0:1024"/>
     </magic>
     <glob pattern="*.pdf"/>
