Index: shared-mime-info-1.0/freedesktop.org.xml.in
===================================================================
--- shared-mime-info-1.0.orig/freedesktop.org.xml.in	2012-06-07 18:04:34.395226725 +0200
+++ shared-mime-info-1.0/freedesktop.org.xml.in	2012-06-07 18:04:45.343226892 +0200
@@ -5250,7 +5250,13 @@
       <match type="string" value="%" offset="0"/>
     </magic>
     <magic priority="50">
-      <match type="string" value="documentclass" offset="1"/>
+      <match value="\documentclass" type="string" offset="0:100"/>
+    </magic>
+    <magic priority="50">
+      <match value="\chapter" type="string" offset="0:100"/>
+    </magic>
+    <magic priority="50">
+      <match value="\section" type="string" offset="0:100"/>
     </magic>
   </mime-type>
   <mime-type type="text/x-texinfo">
