Index: shared-mime-info-1.0/freedesktop.org.xml.in
===================================================================
--- shared-mime-info-1.0.orig/freedesktop.org.xml.in	2012-06-07 18:04:35.079226738 +0200
+++ shared-mime-info-1.0/freedesktop.org.xml.in	2012-06-07 18:04:44.747226884 +0200
@@ -2787,6 +2787,10 @@
     </magic>
     <glob pattern="*.psw"/>
   </mime-type>
+  <mime-type type="application/x-mps">
+    <_comment>Linear and integer program expression format</_comment>
+    <glob pattern="*.mps"/>
+  </mime-type>
   <mime-type type="application/x-profile">
     <_comment>profiler results</_comment>
     <sub-class-of type="text/plain"/>
