Index: shared-mime-info-1.0/freedesktop.org.xml.in
===================================================================
--- shared-mime-info-1.0.orig/freedesktop.org.xml.in	2012-06-07 18:04:35.963226751 +0200
+++ shared-mime-info-1.0/freedesktop.org.xml.in	2012-06-07 18:04:36.475226760 +0200
@@ -4480,7 +4480,6 @@
   </mime-type>
   <mime-type type="inode/directory">
     <_comment>folder</_comment>
-    <alias type="x-directory/normal"/>
   </mime-type>
   <mime-type type="inode/fifo">
     <_comment>pipe</_comment>
