# upstream http://bugzilla.gnome.org/show_bug.cgi?id=300547

Index: gnome-vfs-2.24.2/libgnomevfs/gnome-vfs-volume-monitor-daemon.c
===================================================================
--- gnome-vfs-2.24.2.orig/libgnomevfs/gnome-vfs-volume-monitor-daemon.c	2009-04-20 12:46:46.000000000 +0200
+++ gnome-vfs-2.24.2/libgnomevfs/gnome-vfs-volume-monitor-daemon.c	2010-02-06 11:55:44.829288390 +0100
@@ -743,6 +743,9 @@
 			mount = l->data;
 			uri = gnome_vfs_get_uri_from_local_path (mount->mount_path);
 
+			if (uri == NULL)
+			    continue;
+
 			drive = _gnome_vfs_volume_monitor_find_fstab_drive_by_activation_uri (volume_monitor, uri);
 
 			if (drive != NULL) {
