Index: gnome-vfs-2.24.2/libgnomevfs/gnome-vfs-hal-mounts.c
===================================================================
--- gnome-vfs-2.24.2.orig/libgnomevfs/gnome-vfs-hal-mounts.c	2009-04-20 12:46:46.000000000 +0200
+++ gnome-vfs-2.24.2/libgnomevfs/gnome-vfs-hal-mounts.c	2010-02-06 11:55:45.209277862 +0100
@@ -1038,17 +1038,19 @@
 			vol->priv->volume_type = GNOME_VFS_VOLUME_TYPE_VFS_MOUNT;
 			vol->priv->activation_uri = g_strdup_printf ("cdda://%s", 
 								     libhal_volume_get_device_file (hal_volume));
+			vol->priv->is_read_only = TRUE;
 		} else if (libhal_volume_disc_is_blank (hal_volume)) {
 			vol->priv->volume_type = GNOME_VFS_VOLUME_TYPE_VFS_MOUNT;
 			vol->priv->activation_uri = g_strdup ("burn:///");
+			vol->priv->is_read_only = FALSE;
 		} else {
 			vol->priv->activation_uri = gnome_vfs_get_uri_from_local_path (
 				libhal_volume_get_mount_point (hal_volume));
+			vol->priv->is_read_only = libhal_device_get_property_bool (volume_monitor_daemon->hal_ctx,
+										   libhal_volume_get_udi (hal_volume),
+										   "volume.is_mounted_read_only", NULL);
 		}
 		vol->priv->filesystem_type = g_strdup (libhal_volume_get_fstype (hal_volume));
-		vol->priv->is_read_only = libhal_device_get_property_bool (volume_monitor_daemon->hal_ctx,
-									   libhal_volume_get_udi (hal_volume),
-									   "volume.is_mounted_read_only", NULL);
 		vol->priv->is_mounted = TRUE;
 		
 		vol->priv->device_type = _hal_get_gnome_vfs_device_type (hal_drive);
