Index: gnome-vfs-2.24.2/modules/sftp-method.c
===================================================================
--- gnome-vfs-2.24.2.orig/modules/sftp-method.c	2009-04-20 12:46:46.000000000 +0200
+++ gnome-vfs-2.24.2/modules/sftp-method.c	2010-02-06 11:55:45.785283080 +0100
@@ -1315,6 +1315,8 @@
 			if (g_str_has_suffix (buffer, "password: ") ||
 			    g_str_has_suffix (buffer, "Password: ") ||
 			    g_str_has_suffix (buffer, "Password:")  ||
+			    g_str_has_prefix (buffer, "Password for ") ||
+			    g_str_has_prefix (buffer, "Enter Kerberos password") ||
 			    g_str_has_prefix (buffer, "Enter passphrase for key")) {
 				if (!done_auth && password != NULL) {
 					g_io_channel_write_chars (tty_channel, password, -1, &len, NULL);
