Index: gnome-vfs-2.24.2/modules/default-modules.conf
===================================================================
--- gnome-vfs-2.24.2.orig/modules/default-modules.conf	2009-04-20 12:46:46.000000000 +0200
+++ gnome-vfs-2.24.2/modules/default-modules.conf	2010-02-06 11:55:46.073282906 +0100
@@ -6,8 +6,6 @@
 # <method 1> <method 2> ... <method N> : <module>
 #
 
-bzip2: bzip2
-
 computer: computer
 
 dns-sd: dns-sd
@@ -16,20 +14,12 @@
 
 test: vfs-test
 
-ftp: ftp
-
 network: network
 
-nntp: nntp
+#nntp: nntp
 
 gzip ugzip: gzip
 
-http: http
-
-dav: http
-
-davs: http
-
 #efs: vefs
 
 #nfs: nfs
