unchanged:
Index: gnome-vfs-2.24.2/libgnomevfs/gnome-vfs-hal-mounts.c
===================================================================
--- gnome-vfs-2.24.2.orig/libgnomevfs/gnome-vfs-hal-mounts.c	2010-02-06 11:55:45.209277862 +0100
+++ gnome-vfs-2.24.2/libgnomevfs/gnome-vfs-hal-mounts.c	2010-02-06 11:55:46.357284162 +0100
@@ -645,6 +645,19 @@
 		"/var",
 		"/proc",
 		"/sbin",
+		"/usr/local",
+		"/var/local",
+		"/var/crash",
+		"/var/mail",
+		"/var/run",
+		"/var/tmp",
+ 		NULL
+ 	};
+
+	const char *fhs23_recursive_toplevel_mount_points[] = {
+ 		"/live/cow",
+ 		"/live/image",
+ 		"/live/installer",
 		NULL
 	};
 
@@ -694,6 +707,12 @@
 				if (strcmp (mount_point, fhs23_toplevel_mount_points[i]) == 0)
 					goto out;
 			}
+
+			for (i = 0; fhs23_recursive_toplevel_mount_points[i] != NULL; i++) {
+				if (strncmp (mount_point, fhs23_recursive_toplevel_mount_points[i],
+					    strlen(fhs23_recursive_toplevel_mount_points[i])) == 0)
+					goto out;
+			}
 		}
 	}
 #if 0
