? libgnomevfs/.gnome-vfs-volume-ops.c.swp
Index: gnome-vfs-2.24.2/libgnomevfs/gnome-vfs-volume-ops.c
===================================================================
--- gnome-vfs-2.24.2.orig/libgnomevfs/gnome-vfs-volume-ops.c	2009-04-20 12:46:46.000000000 +0200
+++ gnome-vfs-2.24.2/libgnomevfs/gnome-vfs-volume-ops.c	2010-02-06 11:55:47.237281162 +0100
@@ -823,8 +823,6 @@
 # if defined(HAL_MOUNT)
 		       if (hal_udi != NULL && g_file_test (HAL_MOUNT, G_FILE_TEST_IS_EXECUTABLE))
 			       command = HAL_MOUNT;
-		       else
-			       command = find_command (MOUNT_COMMAND);
 # else
 		       ;
 # endif
@@ -847,8 +845,6 @@
 # if defined(HAL_UMOUNT)
 		       if (hal_udi != NULL && g_file_test (HAL_UMOUNT, G_FILE_TEST_IS_EXECUTABLE))
 			       command = HAL_UMOUNT;
-		       else
-			       command = find_command (UMOUNT_COMMAND);
 # else
 		       ;
 #endif
