Index: gnome-vfs-2.24.2/libgnomevfs/gnome-vfs-configuration.c
===================================================================
--- gnome-vfs-2.24.2.orig/libgnomevfs/gnome-vfs-configuration.c	2009-04-20 12:46:46.000000000 +0200
+++ gnome-vfs-2.24.2/libgnomevfs/gnome-vfs-configuration.c	2010-02-06 11:55:47.525283222 +0100
@@ -358,6 +358,8 @@
 	guint line_buffer_size;
 	guint line_number;
 
+	if (!g_file_test (file_name, G_FILE_TEST_IS_REGULAR))
+		return;
 	f = g_fopen (file_name, "r");
 	if (f == NULL) {
 		g_warning (_("Configuration file `%s' was not found: %s"),
