Index: gnome-vfs-2.24.2/schemas/desktop_gnome_url_handlers.schemas.in
===================================================================
--- gnome-vfs-2.24.2.orig/schemas/desktop_gnome_url_handlers.schemas.in	2009-04-20 12:46:46.000000000 +0200
+++ gnome-vfs-2.24.2/schemas/desktop_gnome_url_handlers.schemas.in	2010-02-06 11:55:48.669281725 +0100
@@ -172,6 +172,40 @@
     </schema>
 
     <schema>
+      <key>/schemas/desktop/gnome/url-handlers/cdda/enabled</key>
+      <applyto>/desktop/gnome/url-handlers/cdda/enabled</applyto>
+      <owner>gnome</owner>
+      <type>bool</type>
+      <default>true</default>
+      <locale name="C">
+        <short>Whether the specified command should handle "cdda" URLs</short>
+        <long>True if the command specified in the "command" key should handle "cdda" URLs.</long>
+      </locale>
+    </schema>
+    <schema>
+      <key>/schemas/desktop/gnome/url-handlers/cdda/command</key>
+      <applyto>/desktop/gnome/url-handlers/cdda/command</applyto>
+      <owner>gnome</owner>
+      <type>string</type>
+      <default>sound-juicer %s</default>
+      <locale name="C">
+        <short>The handler for "cdda" URLs</short>
+        <long>The command used to handle "cdda" URLs, if enabled.</long>
+      </locale>
+    </schema>
+    <schema>
+      <key>/schemas/desktop/gnome/url-handlers/cdda/needs_terminal</key>
+      <applyto>/desktop/gnome/url-handlers/cdda/needs_terminal</applyto>
+      <owner>gnome</owner>
+      <type>bool</type>
+      <default>false</default>
+      <locale name="C">
+        <short>Run the command in a terminal</short>
+        <long>True if the command used to handle this type of URL should be run in a terminal.</long>
+      </locale>
+    </schema>
+
+    <schema>
       <key>/schemas/desktop/gnome/url-handlers/https/enabled</key>
       <applyto>/desktop/gnome/url-handlers/https/enabled</applyto>
       <owner>gnome</owner>
