Index: gnome-vfs-2.24.2/libgnomevfs/gnome-vfs-volume-monitor-client.c
===================================================================
--- gnome-vfs-2.24.2.orig/libgnomevfs/gnome-vfs-volume-monitor-client.c	2009-04-20 12:46:46.000000000 +0200
+++ gnome-vfs-2.24.2/libgnomevfs/gnome-vfs-volume-monitor-client.c	2010-02-06 11:55:49.249283807 +0100
@@ -212,6 +212,9 @@
 	if (volume_monitor_client->dbus_conn == NULL)
 		return;
 
+	if (volume_monitor_client->dbus_conn == NULL)
+		return;
+
 	list = get_drives (volume_monitor_client->dbus_conn, volume_monitor);
 	for (l = list; l; l = l->next) {
 		drive = l->data;
@@ -235,6 +238,9 @@
 	if (volume_monitor_client->dbus_conn == NULL)
 		return;
 
+	if (volume_monitor_client->dbus_conn == NULL)
+		return;
+
 	list = get_volumes (volume_monitor_client->dbus_conn, volume_monitor);
 	for (l = list; l; l = l->next) {
 		volume = l->data;
