Index: gnome-vfs-2.24.2/schemas/desktop_gnome_url_handlers.schemas.in
===================================================================
--- gnome-vfs-2.24.2.orig/schemas/desktop_gnome_url_handlers.schemas.in	2010-02-06 11:55:48.669281725 +0100
+++ gnome-vfs-2.24.2/schemas/desktop_gnome_url_handlers.schemas.in	2010-02-06 11:55:49.837281614 +0100
@@ -274,40 +274,6 @@
     </schema>
 
     <schema>
-      <key>/schemas/desktop/gnome/url-handlers/aim/enabled</key>
-      <applyto>/desktop/gnome/url-handlers/aim/enabled</applyto>
-      <owner>gnome</owner>
-      <type>bool</type>
-      <default>true</default>
-      <locale name="C">
-        <short>Whether the specified command should handle "aim" URLs</short>
-        <long>True if the command specified in the "command" key should handle "aim" URLs.</long>
-      </locale>
-    </schema>
-    <schema>
-      <key>/schemas/desktop/gnome/url-handlers/aim/command</key>
-      <applyto>/desktop/gnome/url-handlers/aim/command</applyto>
-      <owner>gnome</owner>
-      <type>string</type>
-      <default>gaim-remote uri "%s"</default>
-      <locale name="C">
-        <short>The handler for "aim" URLs</short>
-        <long>The command used to handle "aim" URLs, if enabled.</long>
-      </locale>
-    </schema>
-    <schema>
-      <key>/schemas/desktop/gnome/url-handlers/aim/needs_terminal</key>
-      <applyto>/desktop/gnome/url-handlers/aim/needs_terminal</applyto>
-      <owner>gnome</owner>
-      <type>bool</type>
-      <default>false</default>
-      <locale name="C">
-        <short>Run the command in a terminal</short>
-        <long>True if the command used to handle this type of URL should be run in a terminal.</long>
-      </locale>
-    </schema>
-
-    <schema>
       <key>/schemas/desktop/gnome/url-handlers/callto/enabled</key>
       <applyto>/desktop/gnome/url-handlers/callto/enabled</applyto>
       <owner>gnome</owner>
