Index: gnome-vfs-2.24.2/libgnomevfs/gnome-vfs-filesystem-type.c
===================================================================
--- gnome-vfs-2.24.2.orig/libgnomevfs/gnome-vfs-filesystem-type.c	2009-10-08 16:03:36.000000000 +0200
+++ gnome-vfs-2.24.2/libgnomevfs/gnome-vfs-filesystem-type.c	2010-02-06 11:55:51.089283831 +0100
@@ -68,6 +68,7 @@
 	{ "msdos"    , N_("MSDOS Volume"), 0 },
 	{ "msdosfs"  , N_("MSDOS Volume"), 0 },
 	{ "nfs"      , N_("NFS Network Volume"), 1 },
+	{ "nfs4"     , N_("NFS Network Volume"), 1 },
 	{ "ntfs"     , N_("Windows NT Volume"), 0 },
 	{ "ntfs-3g"  , N_("Windows NT Volume"), 1 },
 	{ "nilfs2"   , N_("NILFS Linux Volume"), 1 },
Index: gnome-vfs-2.24.2/libgnomevfs/gnome-vfs-volume-monitor-daemon.c
===================================================================
--- gnome-vfs-2.24.2.orig/libgnomevfs/gnome-vfs-volume-monitor-daemon.c	2010-02-06 11:55:50.133278538 +0100
+++ gnome-vfs-2.24.2/libgnomevfs/gnome-vfs-volume-monitor-daemon.c	2010-02-06 11:55:51.093281826 +0100
@@ -895,7 +895,8 @@
 		if (vol->priv->device_type == GNOME_VFS_DEVICE_TYPE_UNKNOWN) {
 			vol->priv->device_type = GNOME_VFS_DEVICE_TYPE_CDROM;
 		}
-	} else if (strcmp (mount->filesystem_type, "nfs") == 0) {
+	} else if ((strcmp (mount->filesystem_type, "nfs") == 0) ||
+		   (strcmp (mount->filesystem_type, "nfs4") == 0)) {
 		if (strstr (vol->priv->device_path, "(pid") != NULL) {
 			vol->priv->device_type = GNOME_VFS_DEVICE_TYPE_AUTOFS;
 		} else {
Index: gnome-vfs-2.24.2/modules/file-method.c
===================================================================
--- gnome-vfs-2.24.2.orig/modules/file-method.c	2010-02-06 11:55:50.729281262 +0100
+++ gnome-vfs-2.24.2/modules/file-method.c	2010-02-06 11:55:51.097278145 +0100
@@ -1279,6 +1279,7 @@
 	G_LOCK (fstype);
 	type = filesystem_type (path, path, &statbuf);
 	is_local = ((strcmp (type, "nfs") != 0) && 
+		    (strcmp (type, "nfs4") != 0) &&
 		    (strcmp (type, "afs") != 0) &&
 		    (strcmp (type, "autofs") != 0) &&
 		    (strcmp (type, "unknown") != 0) &&
