Index: libdvb-0.5.5.1/Makefile
===================================================================
--- libdvb-0.5.5.1.orig/Makefile	2005-03-22 14:23:25.000000000 +0100
+++ libdvb-0.5.5.1/Makefile	2007-03-16 17:44:28.000000000 +0100
@@ -8,6 +8,7 @@
 
 RESOURCES=README
 include config.mk
+include config.deb
 
 
 
Index: libdvb-0.5.5.1/config.mk
===================================================================
--- libdvb-0.5.5.1.orig/config.mk	2007-03-16 17:44:57.000000000 +0100
+++ libdvb-0.5.5.1/config.mk	2007-03-16 17:45:14.000000000 +0100
@@ -3,7 +3,7 @@
 
 PREFIX = /usr/local
 VERSION = 0.5.5.1
-CFLAGS=-O2 -Wall -g
+#CFLAGS=-O2 -Wall -g
 INCLUDES=-I. -I../include 
 TEST_PROGS= merge_dvb conv satscan cam_set cam_test quickscan cam_menu dump_TS
 #CXX = g++-3.2
Index: libdvb-0.5.5.1/dvb-mpegtools/Makefile
===================================================================
--- libdvb-0.5.5.1.orig/dvb-mpegtools/Makefile	2007-03-16 17:44:12.000000000 +0100
+++ libdvb-0.5.5.1/dvb-mpegtools/Makefile	2007-03-16 17:44:39.000000000 +0100
@@ -1,5 +1,6 @@
 
 include ../config.mk
+include ../config.deb
 
 DCFLAGS =  $(CFLAGS) -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE
 LIBS   = -L../ -ldvbmpegtools
Index: libdvb-0.5.5.1/libdvb/Makefile
===================================================================
--- libdvb-0.5.5.1.orig/libdvb/Makefile	2007-03-16 17:44:12.000000000 +0100
+++ libdvb-0.5.5.1/libdvb/Makefile	2007-03-16 17:44:42.000000000 +0100
@@ -1,5 +1,6 @@
 
 include ../config.mk
+include ../config.deb
 
 OBJS=devices.o DVB.o OSD.o 
 LIBS=-L./ -lpthread
Index: libdvb-0.5.5.1/libdvbci/Makefile
===================================================================
--- libdvb-0.5.5.1.orig/libdvbci/Makefile	2007-03-16 17:44:12.000000000 +0100
+++ libdvb-0.5.5.1/libdvbci/Makefile	2007-03-16 17:44:41.000000000 +0100
@@ -1,5 +1,6 @@
 
 include ../config.mk
+include ../config.deb
 
 CI_OBJS=ci.o ci2.o
 LIBS=-L./ -lpthread
Index: libdvb-0.5.5.1/libdvbmpeg/Makefile
===================================================================
--- libdvb-0.5.5.1.orig/libdvbmpeg/Makefile	2007-03-16 17:44:12.000000000 +0100
+++ libdvb-0.5.5.1/libdvbmpeg/Makefile	2007-03-16 17:44:44.000000000 +0100
@@ -1,5 +1,6 @@
 
 include ../config.mk
+include ../config.deb
 
 DCFLAGS =  $(CFLAGS) -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE
 MFLAG  = -M
Index: libdvb-0.5.5.1/sample_progs/Makefile
===================================================================
--- libdvb-0.5.5.1.orig/sample_progs/Makefile	2007-03-16 17:44:13.000000000 +0100
+++ libdvb-0.5.5.1/sample_progs/Makefile	2007-03-16 17:44:46.000000000 +0100
@@ -1,5 +1,6 @@
 
 include ../config.mk
+include ../config.deb
 
 LIBS=-L../ 
 DVB_LIBS=libdvb.a libdvbci.a
