Index: libdvb-0.5.5.1/Makefile
===================================================================
--- libdvb-0.5.5.1.orig/Makefile	2005-03-22 14:23:25.000000000 +0100
+++ libdvb-0.5.5.1/Makefile	2007-03-16 17:44:28.000000000 +0100
@@ -1,5 +1,5 @@
 
-SOURCEDIRS=include libdvb libdvbci libdvbmpeg sample_progs dvb-mpegtools samplerc 
+SOURCEDIRS=include libdvb libdvbci libdvbmpeg sample_progs samplerc 
 DVB-MPEGTOOLS= dvb-mpegtools_main dvb-mplex audiofilter dvbaudio
 DVB_LIBS=libdvb.a libdvbci.a libdvbmpegtools.a
 HEADERS=include/DVB.hh include/ci.hh include/devices.hh include/channel.h include/ctools.h \
@@ -8,19 +8,19 @@
 
 
 
-main: $(DVB_LIBS) $(TEST_PROGS) $(DVB-MPEGTOOLS)
+main: $(DVB_LIBS) $(TEST_PROGS) #$(DVB-MPEGTOOLS)
 
 install: $(DVB_LIBS) $(HEADERS)
 	mkdir -p $(DESTDIR)$(PREFIX)/lib
 	install -m 644 $(DVB_LIBS) $(DESTDIR)$(PREFIX)/lib/
-	mkdir -p $(DESTDIR)$(PREFIX)/include
-	install -m 644 $(HEADERS) $(DESTDIR)$(PREFIX)/include/
-	make -C dvb-mpegtools install
+	mkdir -p $(DESTDIR)$(PREFIX)/include/libdvb
+	install -m 644 $(HEADERS) $(DESTDIR)$(PREFIX)/include/libdvb
+	#make -C dvb-mpegtools install
 
 uninstall: 
 	for i in $(DVB_LIBS); do rm -f $(DESTDIR)$(PREFIX)/lib/$$i;done
 	for i in $(HEADERS); do rm -f $(DESTDIR)$(PREFIX)/$$i;done
-	make -C dvb-mpegtools uninstall
+	#make -C dvb-mpegtools uninstall
 
 libdvb.a: libdvb/libdvb.a
 	cp libdvb/libdvb.a .
@@ -69,8 +70,8 @@
 	make -C libdvbci clean
 	make -C libdvbmpeg clean
 	make -C sample_progs clean
-	make -C dvb-mpegtools clean
-	rm include/*~
+	#make -C dvb-mpegtools clean
+	rm -f include/*~
 
 dist:	
 	mkdir libdvb-$(VERSION)
