--- libdvb-0.5.5.1.orig/libdvbmpeg/transform.c
+++ libdvb-0.5.5.1/libdvbmpeg/transform.c
@@ -2458,9 +2458,9 @@
 	ipack *p;
 	uint8_t *sb;
 	int64_t apts=0;
-	int64_t apos=0;
+	intptr_t apos=0;
 	int64_t vpts=0;
-	int64_t vpos=0;
+	intptr_t vpos=0;
 	int verb = 0;
 	uint64_t length =0;
 	uint64_t l=0;
@@ -2556,7 +2556,7 @@
 					      get_vinfo( pay, l,&p->vi,1)+1) >0
 						){
 						vpts = trans_pts_dts(sb+9);
-						vpos = (int)(l-count+pay);
+						vpos = (intptr_t)(l-count+pay);
 						printf("vpts : %fs\n",
 						       vpts/90000.);
 					}
@@ -2573,7 +2573,7 @@
 					      get_ac3info( pay, l,&p->ai,1)+1) >0
 						){
 						apts = trans_pts_dts(sb+9);
-						apos = (int)(l-count+pay);
+						apos = (intptr_t)(l-count+pay);
 						printf("apts : %fs\n",
 						       apts/90000.);
 					}
--- libdvb-0.5.5.1.orig/sample_progs/cam_menu.cc
+++ libdvb-0.5.5.1/sample_progs/cam_menu.cc
@@ -46,7 +46,7 @@
 
 void byebye(int i, void *f)
 {
-	int fd = (int) f;
+	int fd = (intptr_t) f;
 
 	CamPacket cp(fd, CMENU_CLOSE, 0, NULL);
 	cp.psend();
