Description: Stupid typo in clx 
Author: Peter Van Eynde
Forwarded: not-needed

--- clisp.orig/modules/clx/new-clx/clx.f
+++ clisp/modules/clx/new-clx/clx.f
@@ -3109,7 +3109,7 @@
     }
   } else {
     pushSTACK(TheSubr (subr_self)->name);
-    error(error_condition,"~S: At least :DRAWABLE should be specifed.");
+    error(error_condition,"~S: At least :DRAWABLE should be specified.");
   }
   skipSTACK(26);
 }
