Index: clisp/src/build-aux/config.sub
===================================================================
--- clisp.orig/src/build-aux/config.sub	2011-04-04 16:47:54.000000000 +0200
+++ clisp/src/build-aux/config.sub	2011-04-08 23:32:27.254284905 +0200
@@ -987,7 +987,7 @@
 		basic_machine=s390-ibm
 		;;
 	s390x | s390x-*)
-		basic_machine=s390x-ibm
+		basic_machine=s390-ibm
 		;;
 	sa29200)
 		basic_machine=a29k-amd
