Description: Fix FTBFS due to missing __aligned_u64 definition
Bug-Debian: http://bugs.debian.org/661449
From: Cyril Brulebois <kibi@debian.org>
---

--- a/extensions/ebt_pkttype.c
+++ b/extensions/ebt_pkttype.c
@@ -12,6 +12,11 @@
 #include <getopt.h>
 #include <netdb.h>
 #include "../include/ebtables_u.h"
+/* BEGIN: Workaround */
+#ifndef __aligned_u64
+#define __aligned_u64 __u64 __attribute__((aligned(8)))
+#endif
+/* END: Workaround */
 #include <linux/if_packet.h>
 #include <linux/netfilter_bridge/ebt_pkttype.h>
 
