Description: Fixes manpage-has-errors-from-man
Origin: vendor
Forwarded: no
Author: Alessandro Ghedini <ghedo@debian.org>
Reviewed-by: Alessandro Ghedini <ghedo@debian.org>
Last-Update: 2012-04-30

--- a/lib/Moose/Manual/Attributes.pod
+++ b/lib/Moose/Manual/Attributes.pod
@@ -624,7 +624,7 @@
 However, we recommend that you use L<Moose::Meta::Attribute::Native> traits
 for these types of attributes, which gives you much more control over how
 they are accessed and manipulated. See also
-L<Moose::Manual::BestPractices#Use_Moose::Meta::Attribute::Native_traits_instead_of_auto_deref>.
+L<Moose::Manual::BestPractices/"Use Moose::Meta::Attribute::Native traits instead of auto deref">.
 
 =head2 Initializer
 
--- a/lib/Moose/Cookbook/Meta/Labeled_AttributeTrait.pod
+++ b/lib/Moose/Cookbook/Meta/Labeled_AttributeTrait.pod
@@ -171,8 +171,8 @@
 When you pass a metaclass to C<has>, it will take the name you provide and
 prefix it with C<Moose::Meta::Attribute::Custom::Trait::>. Then it calls
 C<register_implementation> in the package. In this case, that means Moose ends
-up calling
-C<Moose::Meta::Attribute::Custom::Trait::Labeled::register_implementation>.
+up calling C<register_implementation> from
+C<Moose::Meta::Attribute::Custom::Trait::Labeled>.
 
 If this function exists, it should return the I<real> trait's package
 name. This is exactly what our code does, returning
