--- a/ffmpeg/libavdevice/v4l.c
+++ b/ffmpeg/libavdevice/v4l.c
@@ -26,7 +26,7 @@
 #include <sys/mman.h>
 #include <sys/time.h>
 #define _LINUX_TIME_H 1
-#include <linux/videodev.h>
+#include <libv4l1-videodev.h>
 #include <time.h>
 
 typedef struct {
--- a/src/v4l.h
+++ b/src/v4l.h
@@ -40,7 +40,7 @@
 
 #define _DEVICE_H_
 #define _LINUX_TIME_H
-#include <linux/videodev.h>
+#include <libv4l1-videodev.h>
 
 #include "displayer.h"
 
