--- a/Makefile.flags
+++ b/Makefile.flags
@@ -4,6 +4,11 @@
 
 BB_VER = $(VERSION).$(PATCHLEVEL).$(SUBLEVEL)$(EXTRAVERSION)
 export BB_VER
+ifndef BB_EXTRA_VERSION
+BB_BT = AUTOCONF_TIMESTAMP
+else
+BB_BT = KBUILD_STR($(BB_EXTRA_VERSION))
+endif
 SKIP_STRIP ?= n
 
 # -std=gnu99 needed for [U]LLONG_MAX on some systems
@@ -15,7 +20,7 @@ CPPFLAGS += \
 	-include include/autoconf.h \
 	-D_GNU_SOURCE -DNDEBUG \
 	$(if $(CONFIG_LFS),-D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64) \
-	-D"BB_VER=KBUILD_STR($(BB_VER))" -DBB_BT=AUTOCONF_TIMESTAMP
+	-D"BB_VER=KBUILD_STR($(BB_VER))" -D"BB_BT=$(BB_BT)"
 
 CFLAGS += $(call cc-option,-Wall,)
 CFLAGS += $(call cc-option,-Wshadow,)
