Description: Spelling error
Author: Xavier Guimard <x.guimard@free.fr>
Bug: https://rt.cpan.org/Ticket/Display.html?id=82322
Forwarded: https://rt.cpan.org/Ticket/Display.html?id=82322
Last-Update: 2012-12-29

--- a/Approx.pm
+++ b/Approx.pm
@@ -499,7 +499,7 @@
 =head1 DESCRIPTION
 
 String::Approx lets you match and substitute strings approximately.
-With this you can emulate errors: typing errorrs, speling errors,
+With this you can emulate errors: typing errors, spelling errors,
 closely related vocabularies (colour color), genetic mutations (GAG
 ACT), abbreviations (McScot, MacScot).
 
