Author: Ashish Shukla <wahjava@members.fsf.org>
Subject: use 'ref' rather than 'UNIVERSAL::isa'
Reviewed-by: Nicholas Bamber <nicholas@periapt.co.uk>
Last-Update: 2010-11-03
Bug: http://rt.cpan.org/Ticket/Display.html?id=62667
Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=602056
--- a/lib/SOAP/Lite.pm
+++ b/lib/SOAP/Lite.pm
@@ -3764,13 +3764,13 @@
                 # fillup parameters
                 UNIVERSAL::isa($_[$param] => 'SOAP::Data')
                     ? $_[$param]->SOAP::Data::value($value)
-                    : UNIVERSAL::isa($_[$param] => 'ARRAY')
+                    : (ref($_[$param] eq 'ARRAY'))
                         ? (@{$_[$param]} = @$value)
-                        : UNIVERSAL::isa($_[$param] => 'HASH')
+                        : (ref($_[$param]) eq 'HASH')
                             ? (%{$_[$param]} = %$value)
-                            : UNIVERSAL::isa($_[$param] => 'SCALAR')
+                            : (ref($_[$param]) eq 'SCALAR')
                                 ? (${$_[$param]} = $$value)
-                                : ($_[$param] = $value)
+                                : ($_[$param] = $value);
             }
         }
     }
