From: =?UTF-8?q?Guido=20G=C3=BCnther?= <agx@sigxcpu.org>
Date: Sat, 24 Nov 2012 11:37:02 +0100
Subject: Move profile

---
 mozilla/build/unix/mozilla.in |   16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/mozilla/build/unix/mozilla.in b/mozilla/build/unix/mozilla.in
index d2251ea..505396b 100644
--- a/mozilla/build/unix/mozilla.in
+++ b/mozilla/build/unix/mozilla.in
@@ -65,6 +65,22 @@ if [ $found = 0 ]; then
   fi
 fi
 
+# On 1st run, create a profile based on application.ini configuration
+# by moving an existing ~/.mozilla-thunderbird profile
+if [ -e "$moz_libdir/application.ini" ]; then
+  pfname=`grep ^Name $moz_libdir/application.ini | cut -d= -f2 | tr 'A-Z' 'a-z'`
+  if [ -f "$HOME/.$pfname" ]; then
+    # should never happen
+    mv $HOME/.$pfname $HOME/.$pfname.moved_out_by_mozilla_startup_script
+  fi
+  if [ "$HOME/.$pfname" != "$HOME/.mozilla-thunderbird" ] && \
+     [ ! -d "$HOME/.$pfname" ] && \
+     [   -d "$HOME/.mozilla-thunderbird" ] ; then
+    echo "*INFO* No $HOME/.$pfname detected. Moving $HOME/.mozilla-thunderbird into this place"
+    mv $HOME/.mozilla-thunderbird $HOME/.$pfname
+  fi
+fi
+
 script_args=""
 debugging=0
 MOZILLA_BIN="${progbase}-bin"
