From: =?UTF-8?q?Guido=20G=C3=BCnther?= <agx@sigxcpu.org>
Date: Thu, 25 Oct 2012 19:37:44 +0200
Subject: stop configure if '--with-system-bz2' was passed, but no libbz2 is
 around

---
 mozilla/configure.in |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/mozilla/configure.in b/mozilla/configure.in
index f81e251..77fb0f7 100644
--- a/mozilla/configure.in
+++ b/mozilla/configure.in
@@ -4020,7 +4020,8 @@ if test -z "$BZ2_DIR" -o "$BZ2_DIR" = no; then
     MOZ_NATIVE_BZ2=
 else
     AC_CHECK_LIB(bz2, BZ2_bzread, [MOZ_NATIVE_BZ2=1 MOZ_BZ2_LIBS="-lbz2"],
-	[MOZ_NATIVE_BZ2= MOZ_BZ2_CFLAGS= MOZ_BZ2_LIBS=])
+    [echo "You have run configure with --with-system-bz2, but there is no libbz2 around."
+     exit 1], $MOZ_BZ2_LIBS)
 fi
 CFLAGS=$_SAVE_CFLAGS
 LDFLAGS=$_SAVE_LDFLAGS
