From: Chow Loong Jin <hyperair@debian.org>
Date: Fri, 20 Jan 2012 23:14:40 +0800
Subject: Fix FTBFS involving bad cast (libgpod forwardport)

---
 .../AppleDeviceTrackInfo.cs                        |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Dap/Banshee.Dap.AppleDevice/Banshee.Dap.AppleDevice/AppleDeviceTrackInfo.cs b/src/Dap/Banshee.Dap.AppleDevice/Banshee.Dap.AppleDevice/AppleDeviceTrackInfo.cs
index 178266d..b8549f7 100644
--- a/src/Dap/Banshee.Dap.AppleDevice/Banshee.Dap.AppleDevice/AppleDeviceTrackInfo.cs
+++ b/src/Dap/Banshee.Dap.AppleDevice/Banshee.Dap.AppleDevice/AppleDeviceTrackInfo.cs
@@ -215,7 +215,7 @@ namespace Banshee.Dap.AppleDevice
             track.CDs = DiscCount;
             track.CDNumber = DiscNumber;
             track.TrackLength = (int) Duration.TotalMilliseconds;
-            track.Size = (int)FileSize;
+            track.Size = (uint)FileSize;
             track.Grouping = Grouping;
             try {
                 track.TimePlayed = LastPlayed;
-- 
