--- a/SConstruct
+++ b/SConstruct
@@ -515,6 +515,12 @@
     libraries['aubio'] = LibraryInfo()
     libraries['aubio'].ParseConfig('pkg-config --cflags --libs aubio')
 
+libraries['vamp'] = LibraryInfo ()
+libraries['vamp'].ParseConfig('pkg-config --cflags --libs vamp-sdk')
+
+libraries['vamphost'] = LibraryInfo ()
+libraries['vamphost'].ParseConfig('pkg-config --cflags --libs vamp-hostsdk')
+
 env = conf.Finish ()
 
 if env['FFT_ANALYSIS']:
@@ -920,13 +926,6 @@
 # these are part of the Ardour source tree because they are C++
 # 
 
-libraries['vamp'] = LibraryInfo (LIBS='vampsdk',
-                                 LIBPATH='#libs/vamp-sdk',
-                                 CPPPATH='#libs/vamp-sdk')
-libraries['vamphost'] = LibraryInfo (LIBS='vamphostsdk',
-                                 LIBPATH='#libs/vamp-sdk',
-                                 CPPPATH='#libs/vamp-sdk')
-
 env['RUBBERBAND'] = False
 
 conf = Configure (env)
@@ -1130,7 +1129,6 @@
         'libs/pbd',
         'libs/midi++2',
         'libs/ardour',
-        'libs/vamp-sdk',
         'libs/vamp-plugins/',
     # these are unconditionally included but have
     # tests internally to avoid compilation etc
@@ -1193,7 +1191,6 @@
         'libs/pbd',
         'libs/midi++2',
         'libs/ardour',
-        'libs/vamp-sdk',
         'libs/vamp-plugins/',
     # these are unconditionally included but have
     # tests internally to avoid compilation etc
