Description: Fix for broken symbol resolving in libardourvampplugins.so
Forwarded: not-needed
Origin: vendor, http://bugs.debian.org/cgi-bin/bugreport.cgi?msg=5;bug=560052
Bug-Debian: http://bugs.debian.org/560052
Acked-by: Klaumi Klingsporn <klaumikli@gmx.de>
Author: Adrian Knoth <adi@drcomp.erfurt.thur.de>

--- a/libs/vamp-plugins/SConscript
+++ b/libs/vamp-plugins/SConscript
@@ -16,7 +16,6 @@
 Import('env install_prefix libraries')
 vampplugs = env.Clone()
 
-vampplugs.Append (CPPATH='#libs/vamp-sdk/vamp', CXXFLAGS="-Ilibs/vamp-sdk")
 vampplugs.Merge ([libraries['vamp'],
                   libraries['vamphost'],
                   libraries['aubio'],
