--- a/src/main.cpp
+++ b/src/main.cpp
@@ -122,6 +122,7 @@ void fillCommandLineOptions(KCmdLineOpti
     // TODO - Document this option more clearly
     options.add("p <property=value>",ki18n("Change the value of a profile property."));
     options.add("list-profile-properties", ki18n("List all the profile properties names and their type (for use with -p)"));
+    options.add("T <value>", ki18n("Debian policy compatibility, not used"));
     options.add("!e <cmd>",ki18n("Command to execute"));
     options.add("+[args]",ki18n("Arguments passed to command"));
     options.add("", ki18n("Use --nofork to run in the foreground (helpful with the -e option)."));
