Description: honour Debian's CFLAGS and LDFLAGS
Author: Evgeni Golov <evgeni@debian.org>
Forwarded: not-needed
Last-Update: 2011-11-30

--- a/Makefile
+++ b/Makefile
@@ -1,4 +1,4 @@
-CFLAGS = -O3 -Wall $(shell gimptool-2.0 --cflags) -fopenmp
+CFLAGS := $(CFLAGS) $(shell gimptool-2.0 --cflags) -fopenmp
 LIBS = $(shell gimptool-2.0 --libs) -llensfun -lexiv2
 PLUGIN = gimplensfun
 SOURCES = src/gimplensfun.c
@@ -12,7 +12,7 @@ all: $(PLUGIN)
 OBJECTS = $(subst .c,.o,$(SOURCES))
 
 $(PLUGIN): $(OBJECTS)
-	$(CC) $(CFLAGS) -o $@ $^ $(LIBS)
+	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $^ $(LIBS)
 
 %.o: %.c $(HEADERS)
 	$(CC) $(CFLAGS) -c -o $@ $*.c
