Index: colord-0.1.21/src/cd-main.c
===================================================================
--- colord-0.1.21.orig/src/cd-main.c	2012-10-10 16:30:32.979090557 +1100
+++ colord-0.1.21/src/cd-main.c	2012-10-10 16:32:42.779083627 +1100
@@ -1285,6 +1285,8 @@
 		NULL
 	};
 
+	connection = g_object_ref (connection_);
+
 	registration_id = g_dbus_connection_register_object (connection_,
 							     COLORD_DBUS_PATH,
 							     introspection_daemon->interfaces[0],
@@ -1577,7 +1579,6 @@
 	CdSensor *sensor = NULL;
 
 	g_debug ("CdMain: acquired name: %s", name);
-	connection = g_object_ref (connection_);
 
 	/* add system profiles */
 	profile_store = cd_profile_store_new ();
