--- a/doc/grep.in.1	2012-03-02 17:44:06.000000000 +1100
+++ b/doc/grep.in.1	2012-04-13 09:33:25.000000000 +1000
@@ -114,7 +114,7 @@ This is the default.
 .BR \-P ", " \-\^\-perl\-regexp
 Interpret
 .I PATTERN
-as a Perl regular expression.
+as a Perl regular expression (PCRE, see below).
 This is highly experimental and
 .B "grep \-P"
 may warn of unimplemented features.
@@ -598,7 +598,7 @@ expressions, by using various operators
 .PP
 .B grep
 understands three different versions of regular expression syntax:
-\*(lqbasic,\*(rq \*(lqextended\*(rq and \*(lqperl.\*(rq In
+\*(lqbasic\*(rq (BRE), \*(lqextended\*(rq (ERE) and \*(lqperl\*(rq (PRCE). In
 .RB "\s-1GNU\s0\ " grep ,
 there is no difference in available functionality between basic and
 extended syntaxes.
--- a/doc/grep.info	2012-04-24 00:08:39.000000000 +1000
+++ b/doc/grep.info	2012-05-08 11:49:59.000000000 +1000
@@ -778,7 +778,7 @@ options.
 
 `-P'
 `--perl-regexp'
-     Interpret the pattern as a Perl regular expression.  This is
+     Interpret the pattern as a Perl regular expression (PCRE). This is
      highly experimental and `grep -P' may warn of unimplemented
      features.
 
@@ -799,7 +799,7 @@ A "regular expression" is a pattern that
 Regular expressions are constructed analogously to arithmetic
 expressions, by using various operators to combine smaller expressions.
 `grep' understands three different versions of regular expression
-syntax: "basic," (BRE) "extended" (ERE) and "perl".  In GNU `grep',
+syntax: "basic," (BRE) "extended" (ERE) and "perl" (PERL).  In GNU `grep',
 there is no difference in available functionality between the basic and
 extended syntaxes.  In other implementations, basic regular expressions
 are less powerful.  The following description applies to extended
