From: Teemu Ikonen <tpikonen@gmail.com>
Subject: Debian changes to Imakefile.
X-git-branch: p/imakefile

 src/Imakefile |   28 ++++++++++++++--------------
 1 files changed, 14 insertions(+), 14 deletions(-)

diff --git a/src/Imakefile b/src/Imakefile
index b5a88dd..83dddea 100755
--- a/src/Imakefile
+++ b/src/Imakefile
@@ -1,8 +1,8 @@
-#define CBFLIB_LOCAL
-#define CQRLIB_LOCAL
-#define CVECTOR_LOCAL
-#define NEARTREE_LOCAL
-#define XFORMSLIB_LOCAL
+# #define CBFLIB_LOCAL
+# #define CQRLIB_LOCAL
+# #define CVECTOR_LOCAL
+# #define NEARTREE_LOCAL
+# #define XFORMSLIB_LOCAL
 ###########################################################################
 #                               RasMol 2.7.5                              #
 #                                                                         #
@@ -124,7 +124,7 @@ DEPTHDEF = -DTHIRTYTWOBIT
   LOCALEDEF =
 #endif
 
-PKGDIR = $(HOME)
+PKGDIR = /usr
 
 #ifndef GTKWIN
 #define X11WIN
@@ -197,7 +197,7 @@ CBFLIB_OBJDEP = $(CBFLIB_DIR)/lib/libcbf.a
 ifeq ($(strip $(CBFLIB_DIR)),)
 CBFLIB_DIR = $(PKGDIR)
 endif
-CBFLIB_INCLUDE_DIR = $(CBFLIB_DIR)/include/cbflib
+CBFLIB_INCLUDE_DIR = $(CBFLIB_DIR)/include/cbf
 CBFLIB_LDLIB = -L$(CBFLIB_DIR)/lib -lcbf
 CBFLIB_OBJDEP =
 #define NO_CBFLIB_BUILD
@@ -276,7 +276,7 @@ NEARTREE_OBJDEP =
 
 # RasMol's on-line help direcory
 #   e.g. /usr/local/lib/rasmol/rasmol.hlp
-RASMOLDIR = $(USRLIBDIR)/rasmol/
+RASMOLDIR = /usr/share/rasmol/
 
 
 #ifndef USE_XFORMSLIB
@@ -332,23 +332,23 @@ LDLIBS = -lm -lXi $(XLIB) $(EXTRA_LIBRARIES) $(XFORMSLIB) $(GTKLIBS) \
 #
 
 #ifdef HPArchitecture
-CCOPTIONS = -Ae
-CDEBUGFLAGS = +O2 +Onolimit
+#CCOPTIONS = -Ae
+#CDEBUGFLAGS = -O2 # -Onolimit
 
 #ifdef HPUX_PARISC_32
-CCOPTIONS = -Ae +DA2.0 +DS2.0
+#CCOPTIONS = -Ae -DA2.0 -DS2.0
 #endif
 
 #ifdef HPUX_PARISC_64
-CCOPTIONS = -Ae +DA2.0W +DS2.0 -D_LONGLONG
+#CCOPTIONS = -Ae -DA2.0W -DS2.0 -D_LONGLONG
 #endif
 
 #ifdef HPUX_IA64_32
-CCOPTIONS = -Ae +DD32 +DSmckinley
+#CCOPTIONS = -Ae -DD32 -DSmckinley
 #endif
 
 #ifdef HPUX_IA64_64
-CCOPTIONS = -Ae +DD64 +DSmckinley -D_LONGLONG
+#CCOPTIONS = -Ae -DD64 -DSmckinley -D_LONGLONG
 #endif
 #endif
 
