From: Teemu Ikonen <tpikonen@gmail.com>
Subject: rasmol.h: Use C99 exact-width ints in typedefs.
X-git-branch: p/intwidth

 src/rasmol.h |   18 ++++++++----------
 1 files changed, 8 insertions(+), 10 deletions(-)

diff --git a/src/rasmol.h b/src/rasmol.h
index 94e3640..b6cde5e 100755
--- a/src/rasmol.h
+++ b/src/rasmol.h
@@ -78,6 +78,7 @@
 /* rasmol.h
  */
 
+#include <stdint.h>
 
 /*===========================*/
 /*  User Definable Options!  */
@@ -202,7 +203,7 @@
 
 typedef double Real;
 #ifndef APPLEMAC
-typedef unsigned char Byte;
+typedef uint8_t Byte;
 #endif
 
 #ifdef __STDC__
@@ -211,21 +212,18 @@ typedef signed char Char;
 typedef char Char;
 #endif
 
-#ifdef _LONGLONG
-typedef unsigned int Card;
-typedef int Long;
-#else
-typedef unsigned long Card;
+typedef uint32_t Card;
+/* Long has to be defined as long, instead of int32_t because of
+   multiple printf and scanf calls using %ld in the format. */
 typedef long Long;
-#endif
 
 #ifdef EIGHTBIT
-typedef unsigned char Pixel;
+typedef uint8_t Pixel;
 #else
 #ifdef THIRTYTWOBIT
-typedef Long Pixel;
+typedef int32_t Pixel;
 #else
-typedef short Pixel;
+typedef int16_t Pixel;
 #endif
 #endif
 
