Resolve patch overlaps.
--- a/src/gtkwin.c
+++ b/src/gtkwin.c
@@ -567,103 +567,119 @@
 		
 }
 
-void export_cb(GtkAction *action, gpointer user_data)
+
+void export_cb(GtkAction * action, gpointer user_data)
 {
-	static char *fname = NULL;
+    static char *fname = NULL;
 
-	if(!exportdialog) {
-		build_exportdialog();
-	}
-	if(fname) {
-		gtk_file_chooser_set_filename(GTK_FILE_CHOOSER (exportdialog), fname);
-		gtk_file_chooser_unselect_all(GTK_FILE_CHOOSER (exportdialog));
-	}
-	if(gtk_toggle_button_get_active(
-		GTK_TOGGLE_BUTTON(gtk_builder_get_object(sizebuilder, "size_current")))) 
-	{
-		GtkLabel *header = NULL;
-		char header_str[128];
-	
-		export_x = XRange;
-		export_y = YRange;
-		header = GTK_LABEL(gtk_builder_get_object(sizebuilder, "size_header"));
-		snprintf(header_str, 127, "Size: %d x %d", export_x, export_y);
-		gtk_label_set_text(header, header_str);
-	}
-	while(TRUE) {
-		GtkWidget *question;
-		GdkPixbuf *pbuf;
-		GError *err = NULL;
-		gboolean success = TRUE;
-		guint format = 0;
-		gchar *formatname = NULL;
-		GSList *tmplist = NULL;
-	    guchar *tmpbuf;
-	   
-		if (gtk_dialog_run (GTK_DIALOG (exportdialog)) != GTK_RESPONSE_ACCEPT) {
-			break;
-		}
-		
-		if(fname) {
-			g_free (fname);
-		}
-	    fname = gtk_file_chooser_get_filename (GTK_FILE_CHOOSER (exportdialog));
-		if(g_file_test(fname, G_FILE_TEST_EXISTS)) {
-	    	gint resp;
-			question = gtk_message_dialog_new(GTK_WINDOW(exportdialog), 
-				GTK_DIALOG_MODAL | GTK_DIALOG_DESTROY_WITH_PARENT, 
-				GTK_MESSAGE_QUESTION, GTK_BUTTONS_YES_NO, 
-				"A file named \"%s\" already exists. Do you want to replace it?", 
-				fname);
-			resp = gtk_dialog_run(GTK_DIALOG(question));
-			gtk_widget_destroy (question);
-			if(resp != GTK_RESPONSE_YES) {
-				continue;
-			}
-		}
-		
-		format = egg_file_format_chooser_get_format (
-			EGG_FILE_FORMAT_CHOOSER(format_chooser), fname);
-		if(!format) {
-			char *tmp;
-			question = gtk_message_dialog_new(GTK_WINDOW(exportdialog), 
-				GTK_DIALOG_MODAL | GTK_DIALOG_DESTROY_WITH_PARENT, 
-				GTK_MESSAGE_WARNING, GTK_BUTTONS_OK, 
-				"Could not recognise format of output file \"%s\", please choose another format.", 
-				((tmp = rindex(fname, '/')) != NULL) ? (tmp+1) : (fname));
-			gtk_dialog_run(GTK_DIALOG(question));			   
-			gtk_widget_destroy (question);
-			continue;
-		}
-		formatname = (gchar *) egg_file_format_chooser_get_format_data(
-			EGG_FILE_FORMAT_CHOOSER(format_chooser), format);
-		
-	    tmpbuf = g_new(guchar, 4*export_x*export_y);
-	    render_buffer((Pixel *) tmpbuf, export_x, export_y); 
-	    pbuf = gdk_pixbuf_new_from_data(tmpbuf,
-			GDK_COLORSPACE_RGB,
-			TRUE, 8,
-			export_x, export_y, 4*export_x,
-			NULL, NULL);
+    if (!exportdialog) {
+        build_exportdialog();
+    }
+    if (fname) {
+        gtk_file_chooser_set_filename(GTK_FILE_CHOOSER(exportdialog), fname);
+        gtk_file_chooser_unselect_all(GTK_FILE_CHOOSER(exportdialog));
+    }
+    if (gtk_toggle_button_get_active
+        (GTK_TOGGLE_BUTTON
+         (gtk_builder_get_object(sizebuilder, "size_current")))) {
+        GtkLabel *header = NULL;
+        char header_str[128];
+
+        export_x = XRange;
+        export_y = YRange;
+        header = GTK_LABEL(gtk_builder_get_object(sizebuilder, "size_header"));
+        snprintf(header_str, 127, "Size: %d x %d", export_x, export_y);
+        gtk_label_set_text(header, header_str);
+    }
+    while (TRUE) {
+        GtkWidget *question;
+        GdkPixbuf *pbuf;
+        GError *err = NULL;
+        gboolean success = TRUE;
+        guint format = 0;
+        gchar *formatname = NULL;
+        GSList *tmplist = NULL;
+        Pixel *tmpbuf;
+
+        if (gtk_dialog_run(GTK_DIALOG(exportdialog)) != GTK_RESPONSE_ACCEPT) {
+            break;
+        }
+
+        if (fname) {
+            g_free(fname);
+        }
+        fname = gtk_file_chooser_get_filename(GTK_FILE_CHOOSER(exportdialog));
+        if (g_file_test(fname, G_FILE_TEST_EXISTS)) {
+            gint resp;
+            question = gtk_message_dialog_new(GTK_WINDOW(exportdialog),
+                                              GTK_DIALOG_MODAL |
+                                              GTK_DIALOG_DESTROY_WITH_PARENT,
+                                              GTK_MESSAGE_QUESTION,
+                                              GTK_BUTTONS_YES_NO,
+                                              "A file named \"%s\" already exists. Do you want to replace it?",
+                                              fname);
+            resp = gtk_dialog_run(GTK_DIALOG(question));
+            gtk_widget_destroy(question);
+            if (resp != GTK_RESPONSE_YES) {
+                continue;
+            }
+        }
+
+        format =
+            egg_file_format_chooser_get_format(EGG_FILE_FORMAT_CHOOSER
+                                               (format_chooser), fname);
+        if (!format) {
+            char *tmp;
+            question = gtk_message_dialog_new(GTK_WINDOW(exportdialog),
+                                              GTK_DIALOG_MODAL |
+                                              GTK_DIALOG_DESTROY_WITH_PARENT,
+                                              GTK_MESSAGE_WARNING,
+                                              GTK_BUTTONS_OK,
+                                              "Could not recognise format of output file \"%s\", please choose another format.",
+                                              ((tmp =
+                                                rindex(fname,
+                                                       '/')) !=
+                                               NULL) ? (tmp + 1) : (fname));
+            gtk_dialog_run(GTK_DIALOG(question));
+            gtk_widget_destroy(question);
+            continue;
+        }
+        formatname =
+            (gchar *)
+            egg_file_format_chooser_get_format_data(EGG_FILE_FORMAT_CHOOSER
+                                                    (format_chooser), format);
+
+        tmpbuf = g_new(Pixel, export_x * export_y);
+        render_buffer(tmpbuf, export_x, export_y);
+        pbuf = gdk_pixbuf_new_from_data((guchar *) tmpbuf,
+                                        GDK_COLORSPACE_RGB,
+                                        TRUE, 8,
+                                        export_x, export_y,
+                                        sizeof(Pixel) * export_x,
+                                        NULL, NULL);
         // Use TIFF compression 5 = LZW
         success = gdk_pixbuf_save(pbuf, fname, formatname, &err,
                                   "compression", "5", NULL);
-	  	
-		if(!success && (err != NULL) ) {
-			question = gtk_message_dialog_new(GTK_WINDOW(exportdialog), 
-				GTK_DIALOG_MODAL | GTK_DIALOG_DESTROY_WITH_PARENT, 
-				GTK_MESSAGE_WARNING, GTK_BUTTONS_OK, "Error saving image: %s", 
-				err->message);
-			gtk_dialog_run(GTK_DIALOG(question));			   
-			gtk_widget_destroy (question);				   
-		}
-		g_object_unref(pbuf);
-	    g_free(tmpbuf);
-	    break;
-	}
-	gtk_widget_hide (exportdialog);   
+
+        if (!success && (err != NULL)) {
+            question = gtk_message_dialog_new(GTK_WINDOW(exportdialog),
+                                              GTK_DIALOG_MODAL |
+                                              GTK_DIALOG_DESTROY_WITH_PARENT,
+                                              GTK_MESSAGE_WARNING,
+                                              GTK_BUTTONS_OK,
+                                              "Error saving image: %s",
+                                              err->message);
+            gtk_dialog_run(GTK_DIALOG(question));
+            gtk_widget_destroy(question);
+        }
+        g_object_unref(pbuf);
+        g_free(tmpbuf);
+        break;
+    }
+    gtk_widget_hide(exportdialog);
 }
 
+
 void pagesetup_cb(GtkAction *action, gpointer user_data)
 {
 	if (print_pagesetup == NULL) {
