Index: net-snmp-5.4.3/local/Makefile.in
===================================================================
--- net-snmp-5.4.3.orig/local/Makefile.in	2008-03-21 18:22:57.000000000 +0100
+++ net-snmp-5.4.3/local/Makefile.in	2010-06-04 17:10:51.000000000 +0200
@@ -100,7 +100,7 @@
 
 mib2c.made: $(srcdir)/mib2c
 	if test "x$(PERL)" != "x" ; then \
-	  $(PERL) -p -e 's%^#!.*/perl.*%#!$(PERL)%;s#/usr/local/share/snmp#$(snmplibdir)#;' ${srcdir}/mib2c > mib2c.made; \
+	  $(PERL) -p -e 's%^#!.*/perl.*%#!$(PERL)%;s#/usr/local/share/snmp#$(snmplibdir)#;s#/usr/local/etc/snmp#$(SNMPCONFPATH)#;' ${srcdir}/mib2c > mib2c.made; \
 	else \
 	  touch mib2c.made; \
         fi
Index: net-snmp-5.4.3/Makefile.top
===================================================================
--- net-snmp-5.4.3.orig/Makefile.top	2009-10-29 10:55:42.000000000 +0100
+++ net-snmp-5.4.3/Makefile.top	2010-06-04 17:10:51.000000000 +0200
@@ -24,8 +24,9 @@
 man5dir		= $(mandir)/man5
 man8dir		= $(mandir)/man8
 snmplibdir	= $(datadir)/snmp
-mibdir		= $(snmplibdir)/mibs
+mibdir		= $(datadir)/mibs/netsnmp
 persistentdir	= @PERSISTENT_DIRECTORY@
+sysconfdir	= @sysconfdir@
 DESTDIR         = @INSTALL_PREFIX@
 INSTALL_PREFIX  = $(DESTDIR)
 
Index: net-snmp-5.4.3/mibs/Makefile.in
===================================================================
--- net-snmp-5.4.3.orig/mibs/Makefile.in	2009-05-29 11:08:28.000000000 +0200
+++ net-snmp-5.4.3/mibs/Makefile.in	2010-06-04 17:13:21.000000000 +0200
@@ -13,53 +13,24 @@
 OTHERUNINSTALL=mibsuninstall
 
 
-V1MIBS	= RFC1155-SMI.txt RFC1213-MIB.txt RFC-1215.txt
+NETSNMPMIBS = NET-SNMP-TC NET-SNMP-MIB NET-SNMP-AGENT-MIB \
+	NET-SNMP-EXAMPLES-MIB NET-SNMP-EXTEND-MIB NET-SNMP-PASS-MIB \
+	NET-SNMP-MONITOR-MIB NET-SNMP-SYSTEM-MIB \
+	NET-SNMP-VACM-MIB
 
-V2MIBS = SNMPv2-CONF.txt SNMPv2-SMI.txt SNMPv2-TC.txt SNMPv2-TM.txt \
-	SNMPv2-MIB.txt
+UCDMIBS = UCD-SNMP-MIB UCD-DEMO-MIB UCD-IPFWACC-MIB \
+	UCD-DLMOD-MIB UCD-DISKIO-MIB
 
-V3MIBS	= SNMP-FRAMEWORK-MIB.txt SNMP-MPD-MIB.txt SNMP-TARGET-MIB.txt \
-	SNMP-NOTIFICATION-MIB.txt SNMP-PROXY-MIB.txt \
-	SNMP-USER-BASED-SM-MIB.txt SNMP-VIEW-BASED-ACM-MIB.txt \
-	SNMP-COMMUNITY-MIB.txt TRANSPORT-ADDRESS-MIB.txt
-
-AGENTMIBS = AGENTX-MIB.txt SMUX-MIB.txt
-
-IANAMIBS = IANAifType-MIB.txt IANA-LANGUAGE-MIB.txt \
-	IANA-ADDRESS-FAMILY-NUMBERS-MIB.txt
-
-RFCMIBS	= IF-MIB.txt IF-INVERTED-STACK-MIB.txt \
-	EtherLike-MIB.txt \
-	IP-MIB.txt IP-FORWARD-MIB.txt IANA-RTPROTO-MIB.txt \
-	TCP-MIB.txt UDP-MIB.txt \
-	INET-ADDRESS-MIB.txt HCNUM-TC.txt \
-	HOST-RESOURCES-MIB.txt HOST-RESOURCES-TYPES.txt \
-	RMON-MIB.txt \
-	IPV6-TC.txt IPV6-MIB.txt IPV6-ICMP-MIB.txt IPV6-TCP-MIB.txt \
-	IPV6-UDP-MIB.txt \
-	DISMAN-EVENT-MIB.txt DISMAN-SCRIPT-MIB.txt DISMAN-SCHEDULE-MIB.txt \
-	NOTIFICATION-LOG-MIB.txt SNMP-USM-AES-MIB.txt \
-	SNMP-USM-DH-OBJECTS-MIB.txt \
-	SCTP-MIB.txt
-
-NETSNMPMIBS = NET-SNMP-TC.txt NET-SNMP-MIB.txt NET-SNMP-AGENT-MIB.txt \
-	NET-SNMP-EXAMPLES-MIB.txt NET-SNMP-EXTEND-MIB.txt NET-SNMP-PASS-MIB.txt
-
-UCDMIBS = UCD-SNMP-MIB.txt UCD-DEMO-MIB.txt UCD-IPFWACC-MIB.txt \
-	UCD-DLMOD-MIB.txt UCD-DISKIO-MIB.txt
-
-DEFAULTMIBS = @default_mibs_install@
-
-MIBS	= $(V1MIBS) $(V2MIBS) $(V3MIBS) $(RFCMIBS) \
-	$(AGENTMIBS) $(IANAMIBS) \
-	$(NETSNMPMIBS) $(UCDMIBS) $(DEFAULTMIBS)
+EXTRAMIBS = GNOME-SMI LM-SENSORS-MIB
+
+MIBS	= $(NETSNMPMIBS) $(UCDMIBS) $(EXTRAMIBS)
 
 all: standardall
 
 mibsinstall: installdirs
-	@for i in $(MIBS) ; do \
-		$(INSTALL_DATA) $(srcdir)/$$i $(INSTALL_PREFIX)$(mibdir) ; \
-		echo "install: installed $$i in $(INSTALL_PREFIX)$(mibdir)" ; \
+	@for i in $(MIBS); do \
+		$(INSTALL_DATA) $(srcdir)/$$i.txt $(INSTALL_PREFIX)$(mibdir)/$$i ; \
+		echo "install: installed $$i.txt in $(INSTALL_PREFIX)$(mibdir)/$$i" ; \
 	done
 
 mibsuninstall: installdirs
Index: net-snmp-5.4.3/Makefile.in
===================================================================
--- net-snmp-5.4.3.orig/Makefile.in	2007-06-08 22:32:56.000000000 +0200
+++ net-snmp-5.4.3/Makefile.in	2010-06-04 17:10:51.000000000 +0200
@@ -156,7 +156,7 @@
 #
 # override LD_RUN_PATH to avoid dependencies on the build directory
 perlmodules: perlmakefiles subdirs
-	@(cd perl ; $(MAKE) LD_RUN_PATH="$(libdir):`$(PERL) -e 'use Config; print qq($$Config{archlibexp}/CORE);'`") ; \
+	@(cd perl ; $(MAKE) LD_RUN_PATH="") ; \
         if test $$? != 0 ; then \
            exit 1 ; \
         fi
