Index: net-snmp-5.4.3/local/Makefile.in
===================================================================
--- net-snmp-5.4.3.orig/local/Makefile.in	2008-03-21 18:22:57.000000000 +0100
+++ net-snmp-5.4.3/local/Makefile.in	2010-06-04 17:10:51.000000000 +0200
@@ -100,7 +100,7 @@
 
 mib2c.made: $(srcdir)/mib2c
 	if test "x$(PERL)" != "x" ; then \
-	  $(PERL) -p -e 's%^#!.*/perl.*%#!$(PERL)%;s#/usr/local/share/snmp#$(snmplibdir)#;' ${srcdir}/mib2c > mib2c.made; \
+	  $(PERL) -p -e 's%^#!.*/perl.*%#!$(PERL)%;s#/usr/local/share/snmp#$(snmplibdir)#;s#/usr/local/etc/snmp#$(SNMPCONFPATH)#;' ${srcdir}/mib2c > mib2c.made; \
 	else \
 	  touch mib2c.made; \
         fi
Index: net-snmp-5.4.3/Makefile.top
===================================================================
--- net-snmp-5.4.3.orig/Makefile.top	2009-10-29 10:55:42.000000000 +0100
+++ net-snmp-5.4.3/Makefile.top	2010-06-04 17:10:51.000000000 +0200
@@ -24,8 +24,9 @@
 man5dir		= $(mandir)/man5
 man8dir		= $(mandir)/man8
 snmplibdir	= $(datadir)/snmp
-mibdir		= $(snmplibdir)/mibs
+mibdir		= $(datadir)/mibs/netsnmp
 persistentdir	= @PERSISTENT_DIRECTORY@
+sysconfdir	= @sysconfdir@
 DESTDIR         = @INSTALL_PREFIX@
 INSTALL_PREFIX  = $(DESTDIR)
 
Index: net-snmp-5.4.3/Makefile.in
===================================================================
--- net-snmp-5.4.3.orig/Makefile.in	2007-06-08 22:32:56.000000000 +0200
+++ net-snmp-5.4.3/Makefile.in	2010-06-04 17:10:51.000000000 +0200
@@ -156,7 +156,7 @@
 #
 # override LD_RUN_PATH to avoid dependencies on the build directory
 perlmodules: perlmakefiles subdirs
-	@(cd perl ; $(MAKE) LD_RUN_PATH="$(libdir):`$(PERL) -e 'use Config; print qq($$Config{archlibexp}/CORE);'`") ; \
+	@(cd perl ; $(MAKE) LD_RUN_PATH="") ; \
         if test $$? != 0 ; then \
            exit 1 ; \
         fi
