only in patch2:
unchanged:
Index: net-snmp-5.4.2.1/Makefile.in
===================================================================
--- net-snmp-5.4.2.1.orig/Makefile.in	2009-11-27 15:13:25.000000000 +0100
+++ net-snmp-5.4.2.1/Makefile.in	2009-11-27 15:13:35.000000000 +0100
@@ -17,8 +17,8 @@
 INCLUDESUBDIR=system
 INCLUDESUBDIRHEADERS= aix.h bsd.h bsdi3.h bsdi4.h bsdi.h cygwin.h \
 	darwin.h darwin7.h darwin8.h darwin9.h freebsd5.h freebsd6.h \
-	dynix.h freebsd2.h freebsd3.h freebsd4.h freebsd.h generic.h hpux.h \
-	irix.h linux.h mingw32.h mips.h netbsd.h openbsd.h osf5.h \
+	dynix.h freebsd2.h freebsd3.h freebsd4.h freebsd.h generic.h kfreebsd.h \
+	hpux.h irix.h linux.h mingw32.h mips.h netbsd.h openbsd.h osf5.h \
 	solaris2.3.h solaris2.4.h solaris2.5.h solaris2.6.h \
 	solaris.h sunos.h svr5.h sysv.h ultrix4.h dragonfly.h
 INCLUDESUBDIR2=machine
Index: net-snmp-5.4.2.1/include/net-snmp/system/kfreebsd.h
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ net-snmp-5.4.2.1/include/net-snmp/system/kfreebsd.h	2009-11-27 15:13:35.000000000 +0100
@@ -0,0 +1,65 @@
+#include "freebsd6.h"
+#define freebsd6 freebsd6
+
+#include <osreldate.h>
+#define __FreeBSD_version __FreeBSD_kernel_version
+
+#include <sys/queue.h>
+#include <sys/_types.h>
+
+typedef unsigned int tcp_seq;
+typedef unsigned int tcp_cc;        
+#define TCPOPT_SACK_HDR         (TCPOPT_NOP<<24|TCPOPT_NOP<<16|TCPOPT_SACK<<8)
+#define MAX_SACK_BLKS   6       /* Max # SACK blocks stored at sender side */
+#define TCP_MAX_SACK    3       /* MAX # SACKs sent in any segment */
+
+#include <netinet/tcp_var.h>  
+
+#include <net/ethernet.h>
+#include <net/if_arp.h>
+#include <netinet/in.h>
+
+/*
+ * Externalized form of struct socket used by the sysctl(3) interface.
+ */
+struct xsocket {
+        size_t  xso_len;        /* length of this structure */
+        struct  socket *xso_so; /* makes a convenient handle sometimes */
+        short   so_type;
+        short   so_options;
+        short   so_linger;
+        short   so_state;
+        caddr_t so_pcb;         /* another convenient handle */
+        int     xso_protocol;
+        int     xso_family;
+        short   so_qlen;
+        short   so_incqlen;
+        short   so_qlimit;
+        short   so_timeo;
+        u_short so_error;
+        pid_t   so_pgid;
+        u_long  so_oobmark;
+        struct xsockbuf {
+                u_int   sb_cc;
+                u_int   sb_hiwat;
+                u_int   sb_mbcnt;
+                u_int   sb_mbmax;
+                int     sb_lowat;
+                int     sb_timeo;
+                short   sb_flags;
+        } so_rcv, so_snd;
+        uid_t   so_uid;         /* XXX */
+};
+typedef     u_quad_t so_gen_t;
+
+#define _SYS_SOCKETVAR_H_
+#include <netinet/in_pcb.h>
+
+struct  xtcpcb {
+        size_t  xt_len;
+        struct  inpcb   xt_inp;
+        struct  tcpcb   xt_tp;
+        struct  xsocket xt_socket;
+        u_quad_t        xt_alignment_hack;
+};
+
Index: net-snmp-5.4.2.1/agent/mibgroup/hardware/cpu/cpu_sysctl.c
===================================================================
--- net-snmp-5.4.2.1.orig/agent/mibgroup/hardware/cpu/cpu_sysctl.c	2009-11-27 15:15:37.000000000 +0100
+++ net-snmp-5.4.2.1/agent/mibgroup/hardware/cpu/cpu_sysctl.c	2009-11-27 15:19:01.000000000 +0100
@@ -11,7 +11,7 @@
 #include <unistd.h>
 #include <sys/types.h>
 
-#if defined(__FreeBSD__)
+#if defined(__FreeBSD__) || defined(__FreeBSD_kernel__)
 #include <sys/resource.h>
 #if !defined(CPUSTATES)
 #include <sys/dkstat.h>
@@ -73,7 +73,7 @@
 #define NETSNMP_CPU_STATS uint64_t
 #endif
 
-#elif defined(__FreeBSD__)
+#elif defined(__FreeBSD__) || defined(__FreeBSD_kernel__)
 #define NETSNMP_KERN_CPU  0    /* dummy value - sysctlnametomib(2) should be used */
 
 #else
@@ -107,7 +107,7 @@
 #define NETSNMP_VM_STATS_TYPE  struct uvmexp
 #endif  /* VM_UVMEXP2 || VM_UVMEXP */
 
-#elif defined(__FreeBSD__)                /* FreeBSD */
+#elif defined(__FreeBSD__) || defined(__FreeBSD_kernel__) /* FreeBSD */
 #define NETSNMP_VM_STATS       VM_METER
 #define NETSNMP_VM_STATS_TYPE  struct vmmeter
     #define NS_VM_INTR		v_intr
@@ -159,7 +159,7 @@
     size_t         mem_size  = sizeof(NETSNMP_VM_STATS_TYPE);
     netsnmp_cpu_info *cpu = netsnmp_cpu_get_byIdx( -1, 0 );
 
-#if defined(__FreeBSD__)
+#if defined(__FreeBSD__) || defined(__FreeBSD_kernel__)
     sysctlbyname("kern.cp_time", cpu_stats, &cpu_size, NULL, 0);
 #else
     sysctl(cpu_mib, 2,  cpu_stats, &cpu_size, NULL, 0);
