Index: net-snmp-5.4.3/configure.in
===================================================================
--- net-snmp-5.4.3.orig/configure.in	2010-05-23 22:39:44.000000000 +0200
+++ net-snmp-5.4.3/configure.in	2010-06-04 17:27:58.000000000 +0200
@@ -2609,13 +2609,19 @@
 
 AC_CACHE_CHECK(for mount table location,ac_cv_ETC_MNTTAB,
 [ac_cv_ETC_MNTTAB="unknown"
-for i in /etc/mnttab /etc/mtab /etc/filesystems /dev/mnttab
-  do
-  if test -f $i -o -c $i; then
-    ac_cv_ETC_MNTTAB="$i"
-    break;
-  fi
-done
+AC_ARG_WITH(mnttab,
+[  --with-mnttab=\"/etc/mnttab\"  Mount table location.
+                                 The default is to autodetect this.],
+	[ac_cv_ETC_MNTTAB="$with_mnttab"])
+if test "$ac_cv_ETC_MNTTAB" = "unknown"; then
+  for i in /etc/mnttab /etc/mtab /etc/filesystems /dev/mnttab
+    do
+    if test -f $i -o -c $i; then
+      ac_cv_ETC_MNTTAB="$i"
+      break;
+    fi
+  done
+fi
 ])
 
 AC_DEFINE_UNQUOTED(ETC_MNTTAB,"$ac_cv_ETC_MNTTAB")
