--- net-snmp-5.4.3~dfsg.orig/configure.in
+++ net-snmp-5.4.3~dfsg/configure.in
@@ -2890,6 +2890,9 @@
 	AC_DEFINE(HAVE_LIBKVM, 1, [Define to 1 if you have the `kvm' library (-lkvm).])
 	LAGENTLIBS="$LAGENTLIBS ${_libkvm}"
 fi
+AC_CHECK_FUNC(nlist, AC_DEFINE(HAVE_NLIST, 1, [Define to 1 if you have the `nlist' function.]),
+        AC_CHECK_LIB(bsd, nlist, AC_DEFINE(HAVE_NLIST, 1, [Define to 1 if you have the `nlist' function.])
+        LAGENTLIBS="$LAGENTLIBS bsd"))
 
 # DYNAMIC MODULE SUPPORT
 AC_CHECK_FUNC(dlopen, AC_DEFINE(HAVE_DLOPEN, 1, [Define to 1 if you have the `dlopen' function.]),
