Index: net-snmp-5.4.3~dfsg/agent/agent_read_config.c
===================================================================
--- net-snmp-5.4.3~dfsg.orig/agent/agent_read_config.c	2008-07-24 08:53:02.000000000 +0200
+++ net-snmp-5.4.3~dfsg/agent/agent_read_config.c	2011-01-05 12:45:24.000000000 +0100
@@ -282,7 +282,9 @@
     snmp_call_callbacks(SNMP_CALLBACK_APPLICATION,
                         SNMPD_CALLBACK_PRE_UPDATE_CONFIG, NULL);
     free_config();
+#ifdef USING_MIBII_VACM_CONF_MODULE
     vacm_standard_views(0,0,NULL,NULL);
+#endif
     read_configs();
 }
 
Index: net-snmp-5.4.3~dfsg/apps/snmptrapd.c
===================================================================
--- net-snmp-5.4.3~dfsg.orig/apps/snmptrapd.c	2010-04-30 21:10:17.000000000 +0200
+++ net-snmp-5.4.3~dfsg/apps/snmptrapd.c	2011-01-05 12:45:24.000000000 +0100
@@ -1477,7 +1477,9 @@
 trapd_update_config(void)
 {
     free_config();
+#ifdef USING_MIBII_VACM_CONF_MODULE
     vacm_standard_views(0,0,NULL,NULL);
+#endif
     read_configs();
 }
 
Index: net-snmp-5.4.3~dfsg/apps/snmptrapd_auth.c
===================================================================
--- net-snmp-5.4.3~dfsg.orig/apps/snmptrapd_auth.c	2006-09-15 02:48:50.000000000 +0200
+++ net-snmp-5.4.3~dfsg/apps/snmptrapd_auth.c	2011-01-05 12:47:02.000000000 +0100
@@ -18,6 +18,7 @@
 #include "snmptrapd_ds.h"
 #include "mibII/vacm_conf.h"
 #include <net-snmp/agent/agent_trap.h>
+#include <net-snmp/agent/net-snmp-agent-includes.h>
 
 /**
  * initializes the snmptrapd authorization code registering needed
@@ -32,8 +33,10 @@
                                            netsnmp_trapd_auth);
     traph->authtypes = TRAP_AUTH_NONE;
 
+#ifdef USING_MIBII_VACM_CONF_MODULE
     /* register our configuration tokens for VACM configs */
     init_vacm_config_tokens();
+#endif
 
     /* register a config token for turning off the authorization entirely */
     netsnmp_ds_register_config(ASN_BOOLEAN, "snmptrapd", "disableAuthorization",
@@ -109,6 +112,7 @@
         return NETSNMPTRAPD_HANDLER_FINISH;
     }
 
+#ifdef USING_MIBII_VACM_CONF_MODULE
     /* check the pdu against each typo of VACM access we may want to
        check up on later.  We cache the results for future lookup on
        each call to netsnmp_trapd_check_auth */
@@ -125,6 +129,7 @@
         }
     }
     DEBUGMSGTL(("snmptrapd:auth", "Final bitmask auth: %x\n", ret));
+#endif
 
     if (ret) {
         /* we have policy to at least do "something".  Remember and continue. */
