Index: net-snmp-5.4.2.1/configure.in
===================================================================
--- net-snmp-5.4.2.1.orig/configure.in	2009-12-01 17:30:39.000000000 +0100
+++ net-snmp-5.4.2.1/configure.in	2009-12-01 17:30:40.000000000 +0100
@@ -3050,6 +3050,10 @@
   fi
 fi
 
+# Save CFLAGS for net-snmp-config
+LIBCFLAGS="$CFLAGS"
+AC_SUBST(LIBCFLAGS)
+
 ####################
 # on to embedding...
 
Index: net-snmp-5.4.2.1/net-snmp-config.in
===================================================================
--- net-snmp-5.4.2.1.orig/net-snmp-config.in	2009-12-01 17:30:12.000000000 +0100
+++ net-snmp-5.4.2.1/net-snmp-config.in	2009-12-01 17:31:16.000000000 +0100
@@ -107,6 +107,9 @@
     --base-cflags)
       echo @CFLAGS@ @CPPFLAGS@ -I${NSC_INCLUDEDIR}
       ;;
+    --base-lib-cflags)
+      echo @LIBCFLAGS@ @CPPFLAGS@ -I${NSC_INCLUDEDIR}
+      ;;
     --cflags|--cf*)
       echo @CFLAGS@ @DEVFLAGS@ @CPPFLAGS@ -I. -I${NSC_INCLUDEDIR}
       ;;
@@ -684,6 +687,8 @@
   echo "  These options produce the various compilation flags needed when"
   echo "  building external SNMP applications:"
   echo ""
+  echo "    --base-lib-cflags lists additional compilation flags needed for linking"
+  echo "                      against libsnmp"
   echo "    --base-cflags     lists additional compilation flags needed"
   echo "    --cflags          lists additional compilation flags needed"
   echo "                      (includes -I. and extra developer warning flags)"
