Index: net-snmp-5.4.3~dfsg/configure.in
===================================================================
--- net-snmp-5.4.3~dfsg.orig/configure.in	2011-05-07 10:22:19.000000000 +0000
+++ net-snmp-5.4.3~dfsg/configure.in	2011-05-07 10:24:48.425604190 +0000
@@ -2727,6 +2727,7 @@
     AC_CHECK_LIB(pkcs11, C_Initialize, AC_DEFINE(HAVE_LIBPKCS11)
             LIBPKCS11="-lpkcs11")
 fi
+AC_SUBST(LIBCRYPTO)
 
 LIBS=$NONAGENTLIBS
 
@@ -2779,10 +2780,12 @@
 			AC_MSG_RESULT(no)
 			AC_MSG_ERROR(Asked to use libwrap but I couldn't find it.))])
     LAGENTLIBS="$_wraplibs"
+    LIBWRAP="$_wraplibs"
     CPPFLAGS=${_cppflags}
     LDFLAGS=${_ldflags}
     LIBS=${_libs}
   fi
+  AC_SUBST(LIBWRAP)
 ])
 
 AC_CHECK_HEADERS(rpm/rpmlib.h rpm/header.h)
Index: net-snmp-5.4.3~dfsg/apps/Makefile.in
===================================================================
--- net-snmp-5.4.3~dfsg.orig/apps/Makefile.in	2007-07-04 23:26:56.000000000 +0000
+++ net-snmp-5.4.3~dfsg/apps/Makefile.in	2011-05-07 10:27:52.989604084 +0000
@@ -50,6 +50,8 @@
 AGENTLIB        = ../agent/libnetsnmpagent.$(LIB_EXTENSION)$(LIB_VERSION)
 MIBLIB          = ../agent/libnetsnmpmibs.$(LIB_EXTENSION)$(LIB_VERSION)
 USEAGENTLIBS	= $(MIBLIB) $(AGENTLIB) $(HELPERLIB) $(USELIBS)
+LIBCRYPTO       = @LIBCRYPTO@
+LIBWRAP         = @LIBWRAP@
 
 #
 # link path in src dir
@@ -115,7 +117,7 @@
 	$(LINK) ${CFLAGS} -o $@ snmptest.$(OSUFFIX) $(LOCAL_LIBS) ${LDFLAGS} ${LIBS} 
 
 snmptrapd$(EXEEXT):    $(TRAPD_OBJECTS) $(USETRAPLIBS) $(INSTALLLIBS)
-	$(LINK) ${CFLAGS} -o $@ $(TRAPD_OBJECTS) $(INSTALLLIBS) $(LOCAL_LIBS) ${LDFLAGS} ${TRAPLIBS}
+	$(LINK) ${CFLAGS} -o $@ $(TRAPD_OBJECTS) $(INSTALLLIBS) $(LOCAL_LIBS) ${LDFLAGS} ${TRAPLIBS} $(LIBWRAP)
 
 snmptrap$(EXEEXT):    snmptrap.$(OSUFFIX) $(USELIBS)
 	$(LINK) ${CFLAGS} -o $@ snmptrap.$(OSUFFIX) $(LOCAL_LIBS) ${LDFLAGS} ${LIBS} 
@@ -128,7 +130,7 @@
 	$(LINK) ${CFLAGS} -o $@ snmpset.$(OSUFFIX) $(LOCAL_LIBS) ${LDFLAGS} ${LIBS} 
 
 snmpusm$(EXEEXT):    snmpusm.$(OSUFFIX) $(USELIBS)
-	$(LINK) ${CFLAGS} -o $@ snmpusm.$(OSUFFIX) $(LOCAL_LIBS) ${LDFLAGS} ${LIBS} 
+	$(LINK) ${CFLAGS} -o $@ snmpusm.$(OSUFFIX) $(LOCAL_LIBS) ${LDFLAGS} ${LIBS} $(LIBCRYPTO)
 
 snmpvacm$(EXEEXT):    snmpvacm.$(OSUFFIX) $(USELIBS)
 	$(LINK) ${CFLAGS} -o $@ snmpvacm.$(OSUFFIX) $(LOCAL_LIBS) ${LDFLAGS} ${LIBS} 
