Description: Avoid a regression test which requires X11
Author: Jonas Smedegaard <dr@jones.dk>
Forwarded: no
Last-Update: 2012-04-29

--- a/tests/Makefile.am
+++ b/tests/Makefile.am
@@ -1,4 +1,4 @@
-TESTS = check_looper check_classes check_threads check_bool check_symlinks check_containers check_statesaving check_signals check_imageimporter check_properties
+TESTS = check_looper check_classes check_threads check_bool check_symlinks check_containers check_statesaving check_signals check_properties
 # check_midi check_moviesaver 
 
 noinst_PROGRAMS = $(TESTS)
