Subject: fix format string in Ruby binding
Author: Johannes Brandstätter <jbrandstaetter@gmail.com>
Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=676081
Forwarded: no
Last-Update: 2012-07-01

--- rrdtool-1.4.7.orig/bindings/ruby/main.c
+++ rrdtool-1.4.7/bindings/ruby/main.c
@@ -27,7 +27,7 @@
 
 #define RRD_CHECK_ERROR  \
     if (rrd_test_error()) \
-      rb_raise(rb_eRRDError, rrd_get_error()); \
+      rb_raise(rb_eRRDError, "%s", rrd_get_error()); \
     rrd_clear_error();
 
 string_arr string_arr_new(
@@ -56,8 +56,8 @@
             break;
         default:
             rb_raise(rb_eTypeError,
-                     "invalid argument - %s, expected T_STRING or T_FIXNUM on index %ld",
-                     (long)rb_class2name(CLASS_OF(v)), i);
+                     "invalid argument - %s, expected T_STRING or T_FIXNUM on index %d",
+                     rb_class2name(CLASS_OF(v)), i);
             break;
         }
     }
