# Remove shabangs from .py files.

diff -Nur fusion-icon-0.0.0+git20071028/FusionIcon/data.py fusion-icon-0.0.0+git20071028.new/FusionIcon/data.py
--- fusion-icon-0.0.0+git20071028/FusionIcon/data.py	2007-09-18 10:53:04.000000000 +0200
+++ fusion-icon-0.0.0+git20071028.new/FusionIcon/data.py	2007-12-11 21:40:36.000000000 +0100
@@ -1,4 +1,3 @@
-#!/usr/bin/env python
 # This file is part of Fusion-icon.
 
 # Fusion-icon is free software; you can redistribute it and/or modify
diff -Nur fusion-icon-0.0.0+git20071028/FusionIcon/environment.py fusion-icon-0.0.0+git20071028.new/FusionIcon/environment.py
--- fusion-icon-0.0.0+git20071028/FusionIcon/environment.py	2007-09-18 10:53:04.000000000 +0200
+++ fusion-icon-0.0.0+git20071028.new/FusionIcon/environment.py	2007-12-11 21:40:31.000000000 +0100
@@ -1,4 +1,3 @@
-#!/usr/bin/env python
 # This file is part of Fusion-icon.
 
 # Fusion-icon is free software; you can redistribute it and/or modify
diff -Nur fusion-icon-0.0.0+git20071028/FusionIcon/execute.py fusion-icon-0.0.0+git20071028.new/FusionIcon/execute.py
--- fusion-icon-0.0.0+git20071028/FusionIcon/execute.py	2007-09-18 10:53:04.000000000 +0200
+++ fusion-icon-0.0.0+git20071028.new/FusionIcon/execute.py	2007-12-11 21:40:20.000000000 +0100
@@ -1,4 +1,3 @@
-#!/usr/bin/env python
 # This file is part of Fusion-icon.
 
 # Fusion-icon is free software; you can redistribute it and/or modify
diff -Nur fusion-icon-0.0.0+git20071028/FusionIcon/__init__.py fusion-icon-0.0.0+git20071028.new/FusionIcon/__init__.py
--- fusion-icon-0.0.0+git20071028/FusionIcon/__init__.py	2007-09-18 10:53:04.000000000 +0200
+++ fusion-icon-0.0.0+git20071028.new/FusionIcon/__init__.py	2007-12-11 21:40:14.000000000 +0100
@@ -1,4 +1 @@
-#!/usr/bin/env python
-
-#
-
+# Fusion Icon
diff -Nur fusion-icon-0.0.0+git20071028/FusionIcon/interface_gtk/__init__.py fusion-icon-0.0.0+git20071028.new/FusionIcon/interface_gtk/__init__.py
--- fusion-icon-0.0.0+git20071028/FusionIcon/interface_gtk/__init__.py	2007-09-18 10:53:04.000000000 +0200
+++ fusion-icon-0.0.0+git20071028.new/FusionIcon/interface_gtk/__init__.py	2007-12-11 21:41:46.000000000 +0100
@@ -1,3 +1,3 @@
-#!/usr/bin/env python
+# Fusion Icon
 
 import main
diff -Nur fusion-icon-0.0.0+git20071028/FusionIcon/interface_gtk/main.py fusion-icon-0.0.0+git20071028.new/FusionIcon/interface_gtk/main.py
--- fusion-icon-0.0.0+git20071028/FusionIcon/interface_gtk/main.py	2007-09-18 10:53:04.000000000 +0200
+++ fusion-icon-0.0.0+git20071028.new/FusionIcon/interface_gtk/main.py	2007-12-11 21:41:41.000000000 +0100
@@ -1,4 +1,3 @@
-#!/usr/bin/env python
 # This file is part of Fusion-icon.
 
 # Fusion-icon is free software; you can redistribute it and/or modify
diff -Nur fusion-icon-0.0.0+git20071028/FusionIcon/interface.py fusion-icon-0.0.0+git20071028.new/FusionIcon/interface.py
--- fusion-icon-0.0.0+git20071028/FusionIcon/interface.py	2007-09-18 10:53:04.000000000 +0200
+++ fusion-icon-0.0.0+git20071028.new/FusionIcon/interface.py	2007-12-11 21:40:43.000000000 +0100
@@ -1,4 +1,3 @@
-#!/usr/bin/env python
 # This file is part of Fusion-icon.
 
 # Fusion-icon is free software; you can redistribute it and/or modify
diff -Nur fusion-icon-0.0.0+git20071028/FusionIcon/interface_qt4/__init__.py fusion-icon-0.0.0+git20071028.new/FusionIcon/interface_qt4/__init__.py
--- fusion-icon-0.0.0+git20071028/FusionIcon/interface_qt4/__init__.py	2007-09-18 10:53:04.000000000 +0200
+++ fusion-icon-0.0.0+git20071028.new/FusionIcon/interface_qt4/__init__.py	2007-12-11 21:41:35.000000000 +0100
@@ -1,3 +1,3 @@
-#!/usr/bin/env python
+# Fusion Icon
 
 import main
diff -Nur fusion-icon-0.0.0+git20071028/FusionIcon/interface_qt4/main.py fusion-icon-0.0.0+git20071028.new/FusionIcon/interface_qt4/main.py
--- fusion-icon-0.0.0+git20071028/FusionIcon/interface_qt4/main.py	2007-09-18 10:53:04.000000000 +0200
+++ fusion-icon-0.0.0+git20071028.new/FusionIcon/interface_qt4/main.py	2007-12-11 21:41:24.000000000 +0100
@@ -1,4 +1,3 @@
-#!/usr/bin/env python
 # This file is part of Fusion-icon.
 
 # Fusion-icon is free software; you can redistribute it and/or modify
diff -Nur fusion-icon-0.0.0+git20071028/FusionIcon/parser.py fusion-icon-0.0.0+git20071028.new/FusionIcon/parser.py
--- fusion-icon-0.0.0+git20071028/FusionIcon/parser.py	2007-09-18 10:53:04.000000000 +0200
+++ fusion-icon-0.0.0+git20071028.new/FusionIcon/parser.py	2007-12-11 21:40:52.000000000 +0100
@@ -1,4 +1,3 @@
-#!/usr/bin/env python
 # This file is part of Fusion-icon.
 
 # Fusion-icon is free software; you can redistribute it and/or modify
diff -Nur fusion-icon-0.0.0+git20071028/FusionIcon/start.py fusion-icon-0.0.0+git20071028.new/FusionIcon/start.py
--- fusion-icon-0.0.0+git20071028/FusionIcon/start.py	2007-09-18 10:53:04.000000000 +0200
+++ fusion-icon-0.0.0+git20071028.new/FusionIcon/start.py	2007-12-11 21:40:58.000000000 +0100
@@ -1,4 +1,3 @@
-#!/usr/bin/env python
 # This file is part of Fusion-icon.
 
 # Fusion-icon is free software; you can redistribute it and/or modify
diff -Nur fusion-icon-0.0.0+git20071028/FusionIcon/util.py fusion-icon-0.0.0+git20071028.new/FusionIcon/util.py
--- fusion-icon-0.0.0+git20071028/FusionIcon/util.py	2007-09-18 10:53:04.000000000 +0200
+++ fusion-icon-0.0.0+git20071028.new/FusionIcon/util.py	2007-12-11 21:41:04.000000000 +0100
@@ -1,5 +1,3 @@
-#!/usr/bin/env python
-
 # This file is part of Fusion-icon.
 
 # Fusion-icon is free software; you can redistribute it and/or modify
