diff -ruN slang-pre2.2.2-old/slsh/Makefile.in slang-2.1.4/slsh/Makefile.in
--- slang-2.2.2-old/slsh/Makefile.in	2008-08-25 00:57:48.000000000 +0100
+++ slang-2.2.2/slsh/Makefile.in	2009-04-22 13:40:00.000000000 +0100
@@ -14,7 +14,7 @@
 SLANG_SRCINC	= -I@SRCDIR@
 SLANG_SRCLIB	= -L@ELFDIR@
 #---------------------------------------------------------------------------
-OTHER_LIBS	= @TERMCAP@ @DYNAMIC_LINK_LIB@ @LIBS@ @M_LIB@
+OTHER_LIBS	= @TERMCAP@ @LIBS@ 
 RPATH		= @RPATH@
 #----------------------------------------------------------------------------
 INSTALL		= @INSTALL@
