From: David Paleino <dapal@debian.org>
Subject: fix FTBFS
Origin: vendor
Forwarded: no

---
 gthumb/Makefile.am |    2 ++
 1 file changed, 2 insertions(+)

--- gthumb.orig/gthumb/Makefile.am
+++ gthumb/gthumb/Makefile.am
@@ -276,6 +276,7 @@ gthumb_LDADD =						\
 	$(LIBSOUP_LIBS)					\
 	$(LIBCHAMPLAIN_LIBS)				\
 	$(LIBRSVG_LIBS)					\
+	-lICE                           \
 	$(NULL)
 
 if RUN_IN_PLACE
@@ -308,6 +309,7 @@ gthumb_CFLAGS =							\
 	$(LIBSOUP_CFLAGS)					\
 	$(LIBCHAMPLAIN_CFLAGS)					\
 	$(SMCLIENT_CFLAGS)					\
+	-lICE								\
 	-I$(top_srcdir)/copy-n-paste/				\
 	-DGTHUMB_LOCALEDIR=\"$(datadir)/locale\"		\
 	-DGTHUMB_PREFIX=\"$(prefix)\"           		\
