From: David Paleino <dapal@debian.org>
Subject: add missing included header
Origin: vendor
Forwarded: no

---
 extensions/exiv2_tools/exiv2-utils.cpp |    1 +
 1 file changed, 1 insertion(+)

--- gthumb.orig/extensions/exiv2_tools/exiv2-utils.cpp
+++ gthumb/extensions/exiv2_tools/exiv2-utils.cpp
@@ -29,6 +29,7 @@
 #include <exiv2/image.hpp>
 #include <exiv2/exif.hpp>
 #include <iostream>
+#include <iomanip>
 #include <string>
 #include <sstream>
 #include <vector>
