Index: epiphany-3.4.1/configure.ac
===================================================================
--- epiphany-3.4.1.orig/configure.ac	2012-04-18 16:30:21.000000000 +0200
+++ epiphany-3.4.1/configure.ac	2012-04-30 18:05:10.686231369 +0200
@@ -20,7 +20,7 @@
 m4_define([epiphany_version_micro],[1])
 m4_define([epiphany_version],[epiphany_version_major.epiphany_version_minor.epiphany_version_micro])
 
-AC_INIT([GNOME Web Browser],[epiphany_version],[http://bugzilla.gnome.org/enter_bug.cgi?product=epiphany],[epiphany])
+AC_INIT([GNOME Web Browser],[epiphany_version],[http://bugzilla.gnome.org/enter_bug.cgi?product=epiphany],[epiphany-browser])
 
 AC_PREREQ([2.59])
 
Index: epiphany-3.4.1/data/epiphany.desktop.in.in
===================================================================
--- epiphany-3.4.1.orig/data/epiphany.desktop.in.in	2012-03-26 17:59:35.000000000 +0200
+++ epiphany-3.4.1/data/epiphany.desktop.in.in	2012-04-30 18:05:10.686231369 +0200
@@ -3,7 +3,7 @@
 _GenericName=Web Browser
 _X-GNOME-FullName=Epiphany Web Browser
 _Comment=Browse the web
-Exec=epiphany %U
+Exec=epiphany-browser %U
 StartupNotify=true
 Terminal=false
 Type=Application
Index: epiphany-3.4.1/src/ephy-main.c
===================================================================
--- epiphany-3.4.1.orig/src/ephy-main.c	2012-04-18 16:28:36.000000000 +0200
+++ epiphany-3.4.1/src/ephy-main.c	2012-04-30 18:05:10.694231364 +0200
@@ -317,7 +317,7 @@
   user_time = get_startup_id ();
 
   option_context = g_option_context_new ("");
-  option_group = g_option_group_new ("epiphany",
+  option_group = g_option_group_new ("epiphany-browser",
                                      N_("Web"),
                                      N_("Web options"),
                                      NULL, NULL);
@@ -449,7 +449,7 @@
       /* Skip the 'app-' part */
       app_name += strlen (EPHY_WEB_APP_PREFIX);
 
-      g_set_prgname (app_name);
+      g_set_prgname ("epiphany-browser");
       g_set_application_name (app_name);
 
       gtk_window_set_default_icon_from_file (app_icon, NULL);
@@ -463,7 +463,7 @@
   } else {
     mode = EPHY_EMBED_SHELL_MODE_BROWSER;
 
-    g_set_prgname ("epiphany");
+    g_set_prgname ("epiphany-browser");
     g_set_application_name (_("Web"));
 
     gtk_window_set_default_icon_name ("web-browser");
