Index: epiphany-browser-3.4.0.1/data/default-bookmarks.rdf.in
===================================================================
--- epiphany-browser-3.4.0.1.orig/data/default-bookmarks.rdf.in	2012-03-16 00:53:14.000000000 +0100
+++ epiphany-browser-3.4.0.1/data/default-bookmarks.rdf.in	2012-03-28 08:57:44.984358139 +0200
@@ -20,4 +20,21 @@
 	 "http://www.google.nl" and "http://www.google.nl/search?q=%s" -->
     <_ephy:smartlink>http://www.google.com/search?q=%s&amp;ie=UTF-8&amp;oe=UTF-8</_ephy:smartlink>
   </item>
+  <item rdf:about="http://www.debian.org/">
+    <_title>Debian</_title>
+    <_link>http://www.debian.org/</_link>
+  </item>
+  <item rdf:about="http://www.gnome.org/">
+    <_title>GNOME</_title>
+    <_link>http://www.gnome.org/</_link>
+  </item>
+  <item rdf:about="http://planet.debian.org/">
+    <_title>Planet Debian</_title>
+    <_link>http://planet.debian.org/</_link>
+  </item>
+  <item rdf:about="http://bugs.debian.org/">
+    <_title>Debian Bug Tracking System</_title>
+    <_link>http://bugs.debian.org/</_link>
+    <_ephy:smartlink>http://bugs.debian.org/%s</_ephy:smartlink>
+  </item>
 </rdf:RDF>
