GNOME #571794

Index: epiphany-browser-2.91.92/src/bookmarks/ephy-bookmark-action.c
===================================================================
--- epiphany-browser-2.91.92.orig/src/bookmarks/ephy-bookmark-action.c	2010-12-16 09:02:19.000000000 -0200
+++ epiphany-browser-2.91.92/src/bookmarks/ephy-bookmark-action.c	2011-03-31 23:48:20.204655880 -0300
@@ -398,6 +398,18 @@
 		gdk_event_free (event);
 	}
 
+	/* If we are clicking on the button, activate the entry instead */
+	if (!GTK_IS_EDITABLE (widget))
+	{
+		GtkWidget *otherwidget = gtk_widget_get_parent (widget);
+		if (otherwidget)
+			otherwidget = gtk_widget_get_parent (otherwidget);
+		if (otherwidget)
+			otherwidget = GTK_WIDGET (g_object_get_data (G_OBJECT (otherwidget), "entry"));
+		if (otherwidget)
+			widget = otherwidget;
+	}
+
 	ephy_bookmark_action_activate
 	  (action, widget, (control || ephy_gui_is_middle_click ()) ? EPHY_LINK_NEW_TAB : 0);
 }
