From: Unknown
Subject: Debian local changes

--- /dev/null
+++ b/Xcb.ad
@@ -0,0 +1,65 @@
+
+! Xcb application defaults resource file.
+! ======================================================================
+
+
+! The xcb widget hierarchy consists of a number of custom buffer
+! widgets contained within a single Athena form widget.
+! The widget names and classes are as follows:
+!
+! buffer widgets    name = "buffer<NN>"    class = "Buffer"
+! form widget       name = "container"     class = "Form"
+
+
+
+! Buffer widgets are numbered from 0... onwards, and are named accordingly.
+! As well as the standard core resources, each buffer widget supports
+! resources for "foreground" and "fontSet".
+
+! Application wide resources are as follows:
+!
+! "bufferCount" (default value 8)
+!    This is the number of buffer widgets to create.
+!    Any number of widgets (greater than zero) can be created.
+!
+! "layout" (default value "h")
+!    Only the first character of the resource value is significant.
+!    This is the geometry arrangement to apply in the container widget.
+!    The layout can be "h" (horizontal), "v" (vertical), or some
+!    other value to disable the inbuilt geometry code and specify
+!    the layout via your X resources.
+
+Xcb.bufferCount:		8
+Xcb.layout:			horizontal
+Xcb*fontSet:			fixed
+Xcb*borderWidth:		1
+Xcb*Buffer.width:		60
+Xcb*Buffer.height:		60
+
+
+Xcb*container.defaultDistance:	0
+
+
+
+
+
+! Below is an example of a layout scheme which arranges
+! 10 cut buffer widgets in two rows of 5 windows.
+! Refer to the "Athena Widget Set" manual for full details.
+!
+!Xcb.bufferCount:		10
+!Xcb.layout:			custom
+!Xcb*buffer1.fromHoriz:		buffer0
+!Xcb*buffer2.fromHoriz:		buffer1
+!Xcb*buffer3.fromHoriz:		buffer2
+!Xcb*buffer4.fromHoriz:		buffer3
+!Xcb*buffer5.fromVert:		buffer0
+!Xcb*buffer6.fromHoriz:		buffer5
+!Xcb*buffer6.fromVert:		buffer1
+!Xcb*buffer7.fromHoriz:		buffer6
+!Xcb*buffer7.fromVert:		buffer2
+!Xcb*buffer8.fromHoriz:		buffer7
+!Xcb*buffer8.fromVert:		buffer3
+!Xcb*buffer9.fromHoriz:		buffer8
+!Xcb*buffer9.fromVert:		buffer4
+
