From: Jari Aalto <jari.aalto@cante.net>
Subject: Fix hypens in manual page.

---
 xcb.man |   10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

--- a/xcb.man
+++ b/xcb.man
@@ -186,11 +186,11 @@
 <Btn1Down>:		cut() \\n\\
 Shift <Btn2Down>:	clear() \\n\\
 <Btn2Down>:		paste() \\n\\
-Shift <Btn3Down>:	rotate(-1) \\n\\
+Shift <Btn3Down>:	rotate(\-1) \\n\\
 <Btn3Down>:		rotate(1) \\n\\
-<Key>Left:		rotate(-1) \\n\\
+<Key>Left:		rotate(\-1) \\n\\
 <Key>Right:		rotate(1) \\n\\
-<Key>Up:		rotate(-1) \\n\\
+<Key>Up:		rotate(\-1) \\n\\
 <Key>Down:		rotate(1) \\n\\
 <Key>q:			quit() \\n\\
 <SelReq>:		selreq() \\n\\
@@ -210,7 +210,7 @@
 This loads the string "G'day." into four of the cut buffers
 on the display "bigears".
 .sp 1
-xsendevent -win buffer5 '<Btn1Down>'
+xsendevent \-win buffer5 '<Btn1Down>'
 .br
 This uses the program xsendevent to send a synthetic mouse click event
 to an xcb subwindow, thereby making that window the owner of
@@ -227,7 +227,7 @@
 This concatenates the values in the first 8 cut buffers, and places
 the result back in cut buffer zero.
 .sp 1
-xcb -S 0 && xcb -p 0
+xcb \-S 0 && xcb \-p 0
 .br
 The first command copies the current PRIMARY selection
 into the first cut buffer.  If the copy succeeds, then
