Author: Michael Piefel <piefel@debian.org>
Description: n/a

diff -Naurp file.orig/src/Makefile.in file/src/Makefile.in
--- file.orig/src/Makefile.in	2011-09-08 17:51:11.289392945 +0200
+++ file/src/Makefile.in	2011-09-08 17:58:24.985388126 +0200
@@ -368,7 +368,7 @@ clean-binPROGRAMS:
 	rm -f $$list
 file$(EXEEXT): $(file_OBJECTS) $(file_DEPENDENCIES) 
 	@rm -f file$(EXEEXT)
-	$(AM_V_CCLD)$(LINK) $(file_OBJECTS) $(file_LDADD) $(LIBS)
+	$(AM_V_CCLD)$(LINK) $(file_OBJECTS) $(file_LDADD)
 
 mostlyclean-compile:
 	-rm -f *.$(OBJEXT)
