From: Tim Weippert <weiti@weiti.org>
Date: Sun, 13 Feb 2011 14:24:08 +0100
Subject: [PATCH] additional manpage fix

---
 docs/man/c-icap-mkbdb.8.in |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/docs/man/c-icap-mkbdb.8.in b/docs/man/c-icap-mkbdb.8.in
index c50bc1a..031a899 100644
--- a/docs/man/c-icap-mkbdb.8.in
+++ b/docs/man/c-icap-mkbdb.8.in
@@ -52,7 +52,7 @@ It builds the
 .B keys 
 Berkeley DB database, using string type for keys and values.
 .TP
-c-icap-mkbdb \-o keys --dump
+c-icap-mkbdb \-o keys \-\-dump
 Dump the contents of the keys database
 .SH SEE ALSO
 .BR c-icap "(8)"
-- 
