Author: Daniel Baumann <daniel.baumann@progress-technologies.net>
Description: Don't exclude fioc.h from examples (Closes: #626522).

diff -Naurp fuse.orig/example/Makefile.am fuse/example/Makefile.am
--- fuse.orig/example/Makefile.am	2012-05-18 08:02:47.735592281 +0200
+++ fuse/example/Makefile.am	2012-05-18 08:13:59.963995428 +0200
@@ -1,7 +1,6 @@
 ## Process this file with automake to produce Makefile.in
 
 AM_CPPFLAGS = -I$(top_srcdir)/include -D_FILE_OFFSET_BITS=64 -D_REENTRANT
-noinst_HEADERS = fioc.h
 noinst_PROGRAMS = fusexmp fusexmp_fh null hello hello_ll fioc fioclient \
 		  fsel fselclient cusexmp
 
diff -Naurp fuse.orig/example/Makefile.in fuse/example/Makefile.in
--- fuse.orig/example/Makefile.in	2012-05-18 08:02:47.735592281 +0200
+++ fuse/example/Makefile.in	2012-05-18 08:14:05.156029876 +0200
@@ -262,7 +262,6 @@ top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 AM_CPPFLAGS = -I$(top_srcdir)/include -D_FILE_OFFSET_BITS=64 -D_REENTRANT
-noinst_HEADERS = fioc.h
 LDADD = ../lib/libfuse.la
 fusexmp_fh_LDADD = ../lib/libfuse.la ../lib/libulockmgr.la
 fioclient_CPPFLAGS = 
