Description: Show GNOME Documents in menus everywhere
Author: Jeremy Bicha <jbicha@ubuntu.com>
Index: gnome-boxes-3.3.3/data/gnome-boxes.desktop.in
===================================================================
--- gnome-boxes-3.3.3.orig/data/gnome-boxes.desktop.in	2011-12-19 14:26:07.000000000 -0500
+++ gnome-boxes-3.3.3/data/gnome-boxes.desktop.in	2012-01-14 11:30:21.941544882 -0500
@@ -7,6 +7,5 @@
 Terminal=false
 Type=Application
 StartupNotify=true
-OnlyShowIn=GNOME;
 Categories=GNOME;GTK;Network;RemoteAccess;
 MimeType=x-content/bootable-media;
Index: gnome-boxes-3.3.3/data/gnome-boxes.desktop.in.in
===================================================================
--- gnome-boxes-3.3.3.orig/data/gnome-boxes.desktop.in.in	2011-12-02 09:28:14.000000000 -0500
+++ gnome-boxes-3.3.3/data/gnome-boxes.desktop.in.in	2012-01-14 11:30:35.609612658 -0500
@@ -7,6 +7,5 @@
 Terminal=false
 Type=Application
 StartupNotify=true
-OnlyShowIn=GNOME;
 Categories=GNOME;GTK;Network;RemoteAccess;
 MimeType=x-content/bootable-media;
