Description: Trigger build failure when a compiler is not found
Origin: vendor
Forwarded: not-needed
Author: Alessandro Ghedini <ghedo@debian.org>
Reviewed-by: Alessandro Ghedini <ghedo@debian.org>
Last-Update: 2012-08-07

--- a/Makefile.PL
+++ b/Makefile.PL
@@ -18,7 +18,7 @@
 }
 
 # Check that we have a C compiler
-check_lib_or_exit();
+assert_lib();
 
 my %mm_args = (
   ( MM->can( 'signature_target' ) ? ( SIGN => 1 ) : () ),
