Description: Include ALSA audio backend only on Linux.
Origin: http://anonscm.debian.org/gitweb/?p=pkg-games/bsnes.git;a=tree;f=debian/patches

--- a/bsnes/target-ui/Makefile
+++ b/bsnes/target-ui/Makefile
@@ -15,8 +15,12 @@
 # platform
 ifeq ($(platform),x)
   ruby := video.glx video.xv video.sdl
-  ruby += audio.alsa audio.openal audio.oss audio.pulseaudio audio.pulseaudiosimple audio.ao
+  ruby += audio.openal audio.oss audio.pulseaudio audio.pulseaudiosimple audio.ao
   ruby += input.sdl input.x
+
+  ifneq ($(findstring Linux,$(uname)),)
+    ruby += audio.alsa
+  endif
 else ifeq ($(platform),osx)
   ruby :=
   ruby += audio.openal
--- a/bsnes/target-debugger/Makefile
+++ b/bsnes/target-debugger/Makefile
@@ -13,7 +13,11 @@
 
 # platform
 ifeq ($(platform),x)
-  ruby := audio.alsa
+  ifneq ($(findstring Linux,$(uname)),)
+    ruby := audio.alsa
+  else
+    ruby := audio.openal
+  endif
 else ifeq ($(platform),osx)
   ruby := audio.openal
 else ifeq ($(platform),win)
