Index: b/gparted.desktop.in.in
===================================================================
--- a/gparted.desktop.in.in
+++ b/gparted.desktop.in.in
@@ -3,9 +3,9 @@
 _GenericName=Partition Editor
 _X-GNOME-FullName=GParted Partition Editor
 _Comment=Create, reorganize, and delete partitions
-Exec=@gksuprog@ @installdir@/gparted %f
+Exec=gparted-pkexec
 Icon=gparted
 Terminal=false
 Type=Application
-Categories=GNOME;System;Filesystem;
+Categories=GNOME;System;Filesystem;Settings;
 StartupNotify=true
