Index: rsyslog/plugins/ommysql/createDB.sql
===================================================================
--- rsyslog.orig/plugins/ommysql/createDB.sql	2007-12-12 18:18:29.000000000 +0100
+++ rsyslog/plugins/ommysql/createDB.sql	2008-02-12 16:03:04.000000000 +0100
@@ -1,6 +1,4 @@
-CREATE DATABASE Syslog;
-USE Syslog;
-CREATE TABLE SystemEvents
+CREATE TABLE IF NOT EXISTS SystemEvents
 (
         ID int unsigned not null auto_increment primary key,
         CustomerID bigint,
@@ -28,7 +26,7 @@
         SystemID int NULL
 );
 
-CREATE TABLE SystemEventsProperties
+CREATE TABLE IF NOT EXISTS SystemEventsProperties
 (
         ID int unsigned not null auto_increment primary key,
         SystemEventID int NULL ,
Index: rsyslog/plugins/ompgsql/createDB.sql
===================================================================
--- rsyslog.orig/plugins/ompgsql/createDB.sql	2007-12-12 18:18:29.000000000 +0100
+++ rsyslog/plugins/ompgsql/createDB.sql	2008-02-12 16:03:04.000000000 +0100
@@ -1,5 +1,3 @@
-CREATE DATABASE 'Syslog' WITH ENCODING 'SQL_ASCII';
-\c Syslog;
 CREATE TABLE SystemEvents
 (
         ID serial not null primary key,
