From: Etienne Millon <etienne.millon@gmail.com>
Date: Mon, 17 Oct 2011 14:24:54 +0200
Subject: Fix typo : Psychadelic -> Psychedelic

---
 gmpc-jamendo/src/jamendo.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/gmpc-jamendo/src/jamendo.c b/gmpc-jamendo/src/jamendo.c
index 2dbb4f1..6dcc296 100644
--- a/gmpc-jamendo/src/jamendo.c
+++ b/gmpc-jamendo/src/jamendo.c
@@ -46,7 +46,7 @@ char *GENRE_LIST[] =
 "Bass ", "Soul ", "Punk ", "Space ", "Meditative ", "Instrumental Pop ", "Instrumental Rock ",
 "Ethnic ", "Gothic ", "Darkwave ", "Techno-Industrial ", "Electronic ", "Pop-Folk ", "Eurodance ",
 "Dream ", "Southern Rock ", "Comedy ", "Cult ", "Gangsta ", "Top 40 ", "Christian Rap ",
-"Pop/Funk ", "Jungle ", "Native US ", "Cabaret ", "New Wave ", "Psychadelic ", "Rave ",
+"Pop/Funk ", "Jungle ", "Native US ", "Cabaret ", "New Wave ", "Psychedelic ", "Rave ",
 "Showtunes ", "Trailer ", "Lo-Fi ", "Tribal ", "Acid Punk ", "Acid Jazz ", "Polka ",
 "Retro ", "Musical ", "Rock & Roll ", "Hard Rock ", "Folk ", "Folk-Rock ", "National Folk ",
 "Swing ", "Fast Fusion ", "Bebob ", "Latin ", "Revival ", "Celtic ", "Bluegrass ",
-- 
