=== modified file 'configure.ac'
--- old/configure.ac	2012-08-13 12:50:00 +0000
+++ new/configure.ac	2012-08-21 14:16:58 +0000
@@ -72,7 +72,7 @@
 if test "x$enable_python" = "xyes"; then
     PKG_CHECK_MODULES(PYTHON, [pygobject-3.0 >= $PYGOBJECT_REQUIRED])
 
-    pyoverridesdir=`$PYTHON -c "import gi;print gi._overridesdir"`
+    pyoverridesdir=`$PYTHON -c "import gi;print (gi._overridesdir)" 2>/dev/null`
     AC_SUBST(pyoverridesdir)
 fi
 

