---
 pm/sleep.d/55NetworkManager |    8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

Index: pm-utils/pm/sleep.d/55NetworkManager
===================================================================
--- pm-utils.orig/pm/sleep.d/55NetworkManager	2010-06-18 15:19:18.000000000 +0200
+++ pm-utils/pm/sleep.d/55NetworkManager	2010-06-18 16:16:09.056074013 +0200
@@ -12,7 +12,7 @@
 {
 	# Tell NetworkManager to shut down networking
         printf "Having NetworkManager put all interaces to sleep..."
-	dbus_send --system                         \
+	dbus_send --print-reply --system                         \
 		--dest=org.freedesktop.NetworkManager  \
 		/org/freedesktop/NetworkManager        \
 		org.freedesktop.NetworkManager.sleep && \
@@ -23,7 +23,7 @@
 {
 	# Wake up NetworkManager and make it do a new connection
 	printf "Having NetworkManager wake interfaces back up..."
-        dbus_send --system                        \
+        dbus_send --print-reply --system                        \
 		--dest=org.freedesktop.NetworkManager \
 		/org/freedesktop/NetworkManager       \
 		org.freedesktop.NetworkManager.wake && \
