Index: gnome-scan-0.6.2/modules/Makefile.am
===================================================================
--- gnome-scan-0.6.2.orig/modules/Makefile.am	2010-04-29 23:05:26.250381445 +0200
+++ gnome-scan-0.6.2/modules/Makefile.am	2010-04-29 23:13:55.209884583 +0200
@@ -24,6 +24,10 @@ libgsfiles_la_SOURCES = \
 
 libgsfiles_la_LDFLAGS = -avoid-version
 
+libgsfiles_la_LIBADD = \
+	-L$(top_builddir)/lib -l@SONAME@ \
+	$(GNOME_SCAN_LIBS)
+
 libgsane_la_SOURCES = \
 	gsane-backend.h     \
 	gsane-backend.c     \
@@ -39,7 +43,8 @@ libgsane_la_LDFLAGS = -avoid-version
 
 libgsane_la_LIBADD = \
 	-L$(top_builddir)/lib -l@SONAME@ \
-	-lsane
+	-lsane \
+	$(GNOME_SCAN_LIBS)
 
 #
 ## File created by the gnome-build tools
