From: Loic Minier <lool@dooz.org>
Subject: Add Maintainer Mode

Index: alsa-lib-1.0.23/configure.in
===================================================================
--- alsa-lib-1.0.23.orig/configure.in	2010-05-30 20:56:46.000000000 +0200
+++ alsa-lib-1.0.23/configure.in	2010-05-30 20:56:55.000000000 +0200
@@ -22,6 +22,8 @@
 
 AC_PREFIX_DEFAULT(/usr)
 
+AM_MAINTAINER_MODE([enable])
+
 dnl Checks for programs.
 
 dnl try to gues cross-compiler if not set
