Index: clang-3.2~rc1/tools/clang/tools/libclang/Makefile
===================================================================
--- clang-3.2~rc1.orig/tools/clang/tools/libclang/Makefile	2012-11-13 21:13:57.000000000 +0100
+++ clang-3.2~rc1/tools/clang/tools/libclang/Makefile	2012-11-17 15:42:01.000000000 +0100
@@ -14,6 +14,7 @@
 
 LINK_LIBS_IN_SHARED = 1
 SHARED_LIBRARY = 1
+SONAME_MAJOR=1
 
 include $(CLANG_LEVEL)/../../Makefile.config
 LINK_COMPONENTS := $(TARGETS_TO_BUILD) asmparser support mc
@@ -27,7 +28,7 @@
 
 # Add soname to the library.
 ifeq ($(HOST_OS), $(filter $(HOST_OS), Linux FreeBSD GNU))
-        LDFLAGS += -Wl,-soname,lib$(LIBRARYNAME)$(SHLIBEXT)
+        LDFLAGS += -Wl,-soname,lib$(LIBRARYNAME)$(SHLIBEXT).$(SONAME_MAJOR)
 endif
 
 ##===----------------------------------------------------------------------===##
