Index: clang-3.2~rc1/tools/clang/lib/Basic/Version.cpp
===================================================================
--- clang-3.2~rc1.orig/tools/clang/lib/Basic/Version.cpp	2012-11-13 21:13:57.000000000 +0100
+++ clang-3.2~rc1/tools/clang/lib/Basic/Version.cpp	2012-11-17 15:43:03.000000000 +0100
@@ -15,6 +15,7 @@
 #include "clang/Basic/LLVM.h"
 #include "llvm/Support/raw_ostream.h"
 #include "llvm/Config/config.h"
+#include "clang/Debian/debian_path.h"
 #include <cstring>
 #include <cstdlib>
 
@@ -117,7 +118,8 @@
 #ifdef CLANG_VENDOR
   OS << CLANG_VENDOR;
 #endif
-  OS << "clang version " CLANG_VERSION_STRING " "
+  OS << "clang version " CLANG_VERSION_STRING "-" DEB_PATCHSETVERSION " "
+
      << getClangFullRepositoryVersion();
 
   // If vendor supplied, include the base LLVM version as well.
