Index: clang-3.2~rc1/tools/clang/Makefile
===================================================================
--- clang-3.2~rc1.orig/tools/clang/Makefile	2012-11-13 21:13:57.000000000 +0100
+++ clang-3.2~rc1/tools/clang/Makefile	2012-11-17 15:43:15.000000000 +0100
@@ -47,9 +47,7 @@
 
 # Set common Clang build flags.
 CPP.Flags += -I$(PROJ_SRC_DIR)/$(CLANG_LEVEL)/include -I$(PROJ_OBJ_DIR)/$(CLANG_LEVEL)/include
-ifdef CLANG_VENDOR
-CPP.Flags += -DCLANG_VENDOR='"$(CLANG_VENDOR) "'
-endif
+CPP.Flags += -DCLANG_VENDOR='"Debian "'
 ifdef CLANG_REPOSITORY_STRING
 CPP.Flags += -DCLANG_REPOSITORY_STRING='"$(CLANG_REPOSITORY_STRING)"'
 endif
