Index: clang-3.2~rc1/tools/clang/lib/Driver/ToolChains.cpp
===================================================================
--- clang-3.2~rc1.orig/tools/clang/lib/Driver/ToolChains.cpp	2012-11-13 21:13:57.000000000 +0100
+++ clang-3.2~rc1/tools/clang/lib/Driver/ToolChains.cpp	2012-11-17 15:43:18.000000000 +0100
@@ -2174,6 +2174,11 @@
     addSystemInclude(DriverArgs, CC1Args, P.str());
   }
 
+  // Under Debian, clang headers are installed into
+  // '/usr/include/clang/VERSION/include/'
+  addSystemInclude(DriverArgs, CC1Args, "/usr/include/clang/" + std::string(CLANG_VERSION_STRING) + "/include/");
+  
+
   if (DriverArgs.hasArg(options::OPT_nostdlibinc))
     return;
 
