Index: clang-3.2~rc1/tools/clang/unittests/Tooling/CompilationDatabaseTest.cpp
===================================================================
--- clang-3.2~rc1.orig/tools/clang/unittests/Tooling/CompilationDatabaseTest.cpp	2012-11-13 21:13:57.000000000 +0100
+++ clang-3.2~rc1/tools/clang/unittests/Tooling/CompilationDatabaseTest.cpp	2012-11-17 15:43:28.000000000 +0100
@@ -262,19 +262,6 @@
   return FoundCommand.CommandLine;
 }
 
-TEST(unescapeJsonCommandLine, ReturnsEmptyArrayOnEmptyString) {
-  std::vector<std::string> Result = unescapeJsonCommandLine("");
-  EXPECT_TRUE(Result.empty());
-}
-
-TEST(unescapeJsonCommandLine, SplitsOnSpaces) {
-  std::vector<std::string> Result = unescapeJsonCommandLine("a b c");
-  ASSERT_EQ(3ul, Result.size());
-  EXPECT_EQ("a", Result[0]);
-  EXPECT_EQ("b", Result[1]);
-  EXPECT_EQ("c", Result[2]);
-}
-
 TEST(unescapeJsonCommandLine, MungesMultipleSpaces) {
   std::vector<std::string> Result = unescapeJsonCommandLine("   a   b   ");
   ASSERT_EQ(2ul, Result.size());
