Index: clang-3.2/tools/clang/lib/Driver/ToolChains.cpp
===================================================================
--- clang-3.2~rc1.orig/tools/clang/lib/Driver/ToolChains.cpp	2012-11-27 12:21:42.000000000 +0100
+++ clang-3.2~rc1/tools/clang/lib/Driver/ToolChains.cpp	2012-11-29 09:47:06.000000000 +0100
@@ -36,6 +36,9 @@
 #include <cstdlib> // ::getenv

 #include "clang/Config/config.h" // for GCC_INSTALL_PREFIX
+#include "clang/Debian/debian_path.h"
+
+#include <iostream>
 
 using namespace clang::driver;
 using namespace clang::driver::toolchains;
@@ -2316,6 +2318,10 @@
 
   const std::string IncludePathCandidates[] = {
     LibDir.str() + "/../include/c++/" + Version.str(),
+    // Once more, Debian headers locations changed (since gcc version 4.7.2-5)
+    // See Debian bug tracker #693240
+    LibDir.str() + "/../include/"+ TripleStr.str() +"/c++/" + Version.str(),
+    LibDir.str() + "/../include/" + DEB_HOST_MULTIARCH_TRIPLET + "/c++/" + Version.str(),
     // Gentoo is weird and places its headers inside the GCC install, so if the
     // first attempt to find the headers fails, try this pattern.
     InstallDir.str() + "/include/g++-v4",
@@ -2390,10 +2397,10 @@
   };
 
   for (unsigned i = 0; i < llvm::array_lengthof(IncludePathCandidates); ++i) {
-    if (addLibStdCXXIncludePaths(IncludePathCandidates[i], (TripleStr +
+      addLibStdCXXIncludePaths(IncludePathCandidates[i], (TripleStr +
                 GCCInstallation.getMultiarchSuffix()),
-            DriverArgs, CC1Args))
-      break;
+                               DriverArgs, CC1Args);
+
   }
 }
 
