Index: clang-3.2~rc3/tools/clang/lib/Lex/HeaderSearch.cpp
===================================================================
--- clang-3.2~rc3.orig/tools/clang/lib/Lex/HeaderSearch.cpp	2012-11-06 20:39:40.000000000 +0100
+++ clang-3.2~rc3/tools/clang/lib/Lex/HeaderSearch.cpp	2012-12-11 18:49:49.000000000 +0100
@@ -25,6 +25,11 @@
 #include <cstdio>
 using namespace clang;
 
+// For GNU Hurd
+#if defined(__GNU__) && !defined(PATH_MAX)
+# define PATH_MAX 4096
+#endif
+
 const IdentifierInfo *
 HeaderFileInfo::getControllingMacro(ExternalIdentifierLookup *External) {
   if (ControllingMacro)
Index: clang-3.2~rc3/tools/clang/lib/Lex/ModuleMap.cpp
===================================================================
--- clang-3.2~rc3.orig/tools/clang/lib/Lex/ModuleMap.cpp	2012-11-06 20:41:11.000000000 +0100
+++ clang-3.2~rc3/tools/clang/lib/Lex/ModuleMap.cpp	2012-12-11 18:50:00.000000000 +0100
@@ -30,6 +30,11 @@
 #include <stdlib.h>
 using namespace clang;
 
+// For GNU Hurd
+#if defined(__GNU__) && !defined(PATH_MAX)
+# define PATH_MAX 4096
+#endif
+
 Module::ExportDecl 
 ModuleMap::resolveExport(Module *Mod, 
                          const Module::UnresolvedExportDecl &Unresolved,
