Index: clang-3.2/tools/clang/lib/Driver/ToolChains.h
===================================================================
--- clang-3.2.orig/tools/clang/lib/Driver/ToolChains.h	2012-11-21 09:33:40.000000000 +0100
+++ clang-3.2/tools/clang/lib/Driver/ToolChains.h	2012-12-25 11:47:02.000000000 +0100
@@ -131,6 +131,8 @@
   virtual bool IsUnwindTablesDefault() const;
   virtual bool isPICDefault() const;
   virtual bool isPICDefaultForced() const;
+  virtual void AddClangSystemIncludeArgs(const ArgList &DriverArgs,
+                                        ArgStringList &CC1Args) const;
 
 protected:
   /// \name ToolChain Implementation Helper Functions
Index: clang-3.2/tools/clang/lib/Driver/ToolChains.cpp
===================================================================
--- clang-3.2.orig/tools/clang/lib/Driver/ToolChains.cpp	2012-12-25 11:45:53.000000000 +0100
+++ clang-3.2/tools/clang/lib/Driver/ToolChains.cpp	2012-12-25 11:46:29.000000000 +0100
@@ -943,6 +943,13 @@
   return GoodVersion;
 }
 
+void Generic_GCC::AddClangSystemIncludeArgs(const ArgList &DriverArgs,
+                                      ArgStringList &CC1Args) const {
+  // Under Debian, clang headers are installed into
+  // '/usr/include/clang/VERSION/include/'
+  addSystemInclude(DriverArgs, CC1Args, "/usr/include/clang/" + std::string(CLANG_VERSION_STRING) + "/include/");
+}
+
 /// \brief Less-than for GCCVersion, implementing a Strict Weak Ordering.
 bool Generic_GCC::GCCVersion::operator<(const GCCVersion &RHS) const {
   if (Major < RHS.Major) return true; if (Major > RHS.Major) return false;
