Index: clang-3.2/tools/clang/lib/Driver/ToolChains.cpp
===================================================================
--- clang-3.2.orig/tools/clang/lib/Driver/ToolChains.cpp	2013-03-15 14:54:26.000000000 +0100
+++ clang-3.2/tools/clang/lib/Driver/ToolChains.cpp	2013-03-15 14:55:53.000000000 +0100
@@ -1103,6 +1103,7 @@
   static const char *const PPCTriples[] = {
     "powerpc-linux-gnu",
     "powerpc-unknown-linux-gnu",
+    "powerpc-linux-gnuspe",
     "powerpc-suse-linux",
     "powerpc-montavista-linuxspe"
   };
@@ -2003,6 +2004,8 @@
       return "mipsel-linux-gnu";
     return TargetTriple.str();
   case llvm::Triple::ppc:
+    if (llvm::sys::fs::exists(SysRoot + "/lib/powerpc-linux-gnuspe"))
+      return "powerpc-linux-gnuspe";
     if (llvm::sys::fs::exists(SysRoot + "/lib/powerpc-linux-gnu"))
       return "powerpc-linux-gnu";
     return TargetTriple.str();
Index: clang-3.2/tools/clang/lib/Lex/Makefile
===================================================================
--- clang-3.2.orig/tools/clang/lib/Lex/Makefile	2013-03-15 14:54:26.000000000 +0100
+++ clang-3.2/tools/clang/lib/Lex/Makefile	2013-03-15 14:54:55.000000000 +0100
@@ -17,8 +17,10 @@
 LIBRARYNAME := clangLex
 
 ifeq ($(ARCH),PowerPC)
+ifneq ($(shell dpkg-architecture -qDEB_HOST_ARCH),powerpcspe)
 CXX.Flags += -maltivec
 endif
+endif
 
 include $(CLANG_LEVEL)/Makefile
 
