Index: clang-3.2/tools/clang/tools/scan-build/scan-build
===================================================================
--- clang-3.2.orig/tools/clang/tools/scan-build/scan-build	2012-09-15 07:51:10.000000000 +0200
+++ clang-3.2/tools/clang/tools/scan-build/scan-build	2013-03-15 15:00:05.000000000 +0100
@@ -1459,7 +1459,7 @@
 
 # Find 'clang'
 if (!defined $AnalyzerDiscoveryMethod) {
-  $Clang = Cwd::realpath("$RealBin/bin/clang");
+  $Clang = Cwd::realpath("/usr/bin/clang");
   if (!defined $Clang || ! -x $Clang) {
     $Clang = Cwd::realpath("$RealBin/clang");
   }
