Index: clang-3.2~rc1/tools/clang/lib/Driver/Tools.cpp
===================================================================
--- clang-3.2~rc1.orig/tools/clang/lib/Driver/Tools.cpp	2012-11-27 12:21:42.000000000 +0100
+++ clang-3.2~rc1/tools/clang/lib/Driver/Tools.cpp	2012-11-27 18:22:13.000000000 +0100
@@ -32,10 +32,13 @@
 #include "llvm/Support/Process.h"
 #include "llvm/Support/ErrorHandling.h"
 
+#include "llvm/Config/config.h"
+
 #include "InputInfo.h"
 #include "SanitizerArgs.h"
 #include "ToolChains.h"
 
+
 using namespace clang::driver;
 using namespace clang::driver::tools;
 using namespace clang;
@@ -218,7 +221,7 @@
   // libprofile_rt.so. We used to use the -l:libprofile_rt.a syntax, but that is
   // not supported by old linkers.
   std::string ProfileRT =
-    std::string(TC.getDriver().Dir) + "/../lib/libprofile_rt.a";
+    std::string(TC.getDriver().Dir) + "/../lib/llvm-"+PACKAGE_VERSION+"/lib/libprofile_rt.a";
 
   CmdArgs.push_back(Args.MakeArgString(ProfileRT));
 }
