Description: Allow nvidia-settings to work with dark backgrounds
Author: Albert Milone <albertomilone@alice.it>

--- a/src/gtk+-2.x/ctkglx.c
+++ b/src/gtk+-2.x/ctkglx.c
@@ -267,8 +267,8 @@ GtkWidget* ctk_glx_new(NvCtrlAttributeHa
     event = gtk_event_box_new();
     gtk_scrolled_window_set_policy(GTK_SCROLLED_WINDOW(scrollWin),
                                    GTK_POLICY_NEVER, GTK_POLICY_ALWAYS);
-    gtk_widget_modify_fg(event, GTK_STATE_NORMAL, &(event->style->text[GTK_STATE_NORMAL]));
-    gtk_widget_modify_bg(event, GTK_STATE_NORMAL, &(event->style->base[GTK_STATE_NORMAL]));
+    //gtk_widget_modify_fg(event, GTK_STATE_NORMAL, &(event->style->text[GTK_STATE_NORMAL]));
+    //gtk_widget_modify_bg(event, GTK_STATE_NORMAL, &(event->style->base[GTK_STATE_NORMAL]));
     gtk_container_add(GTK_CONTAINER(event), hbox);
     gtk_scrolled_window_add_with_viewport(GTK_SCROLLED_WINDOW(scrollWin),
                                           event);
@@ -351,15 +351,15 @@ GtkWidget* ctk_glx_new(NvCtrlAttributeHa
     data_table = gtk_table_new(num_fbconfigs, NUM_FBCONFIG_ATTRIBS, FALSE);
     event = gtk_event_box_new();
     
-    gtk_widget_modify_fg(data_table, GTK_STATE_NORMAL, 
-                         &(data_table->style->text[GTK_STATE_NORMAL]));
-    gtk_widget_modify_bg(data_table, GTK_STATE_NORMAL, 
-                         &(data_table->style->base[GTK_STATE_NORMAL]));
+    //gtk_widget_modify_fg(data_table, GTK_STATE_NORMAL, 
+    //                     &(data_table->style->text[GTK_STATE_NORMAL]));
+    //gtk_widget_modify_bg(data_table, GTK_STATE_NORMAL, 
+    //                     &(data_table->style->base[GTK_STATE_NORMAL]));
     gtk_container_add (GTK_CONTAINER(event), data_table);
-    gtk_widget_modify_fg(event, GTK_STATE_NORMAL, 
-                         &(event->style->text[GTK_STATE_NORMAL]));
-    gtk_widget_modify_bg(event, GTK_STATE_NORMAL, 
-                         &(event->style->base[GTK_STATE_NORMAL]));
+    //gtk_widget_modify_fg(event, GTK_STATE_NORMAL, 
+    //                     &(event->style->text[GTK_STATE_NORMAL]));
+    //gtk_widget_modify_bg(event, GTK_STATE_NORMAL, 
+    //                     &(event->style->base[GTK_STATE_NORMAL]));
     gtk_container_add(GTK_CONTAINER(data_viewport), event);
 
     /* Pack the fbconfig header and data tables */
