Description: Fix build failure with LDFLAGS="-Wl,-z,defs"
Bug: https://bugzilla.gnome.org/show_bug.cgi?id=674292
Index: gnome-online-accounts-3.4.1/src/goabackend/Makefile.am
===================================================================
--- gnome-online-accounts-3.4.1.orig/src/goabackend/Makefile.am	2012-04-18 00:59:55.555429577 +0200
+++ gnome-online-accounts-3.4.1/src/goabackend/Makefile.am	2012-04-18 00:59:58.439429696 +0200
@@ -92,6 +92,7 @@
 	$(NULL)
 
 libgoa_backend_1_0_la_LIBADD = 					\
+	$(top_builddir)/src/goa/libgoa-1.0.la			\
 	$(WEBKIT_GTK_LIBS)					\
 	$(JSON_GLIB_LIBS)					\
 	$(GLIB_LIBS) 						\
