Description: install regexp-assemble file
Author: Xavier Guimard <x.guimard@free.fr>
Forwarded: not-needed
Last-Update: 2013-04-01

--- a/Makefile.PL
+++ b/Makefile.PL
@@ -10,7 +10,7 @@
     print "Adding testcover target\n";
 }
 
-my $file = [];
+my $file = ['debian/regexp-assemble'];
 #if( my $response = prompt( 'Install eg/assemble as a program (y/N)', 'n' )) {
 #    push @$file, "eg/assemble" if $response =~ /^\s*y/im,
 #}
