Description: Spelling errors
Author: Xavier Guimard <x.guimard@free.fr>
Bug: https://rt.cpan.org/Ticket/Display.html?id=84336
Forwarded: https://rt.cpan.org/Ticket/Display.html?id=84336
Last-Update: 2013-04-01

--- a/Assemble.pm
+++ b/Assemble.pm
@@ -49,7 +49,7 @@
 trie. Note that Perl's own regular expression engine will implement
 trie optimisations in perl 5.10 (they are already available in
 perl 5.9.3 if you want to try them out). C<Regexp::Assemble> will
-do the right thing when it knows it's running on a a trie'd perl.
+do the right thing when it knows it's running on a trie'd perl.
 (At least in some version after this one).
 
 Some more examples of usage appear in the accompanying README. If
@@ -3211,7 +3211,7 @@
 =item Regexp::Trie
 
 Funnily enough, this was my working name for C<Regexp::Assemble>
-during its developement. I changed the name because I thought it
+during its development. I changed the name because I thought it
 was too obscure. Anyway, C<Regexp::Trie> does much the same as
 C<Regexp::Optimizer> and C<Regexp::Assemble> except that it runs
 much faster (according to the author). It does not recognise
