Description: Spelling fixes
Author: Thibaut GRIDEL <tgridel@free.fr>

--- giflib-4.1.6.orig/doc/gifrotat.html
+++ giflib-4.1.6/doc/gifrotat.html
@@ -38,7 +38,7 @@ Screen (of source image).<P>
 <DT> [-s Width Height]
 <DD> Since the rotated image will have the same image size as
    the original, some parts of the image will by clipped out and lost. By
-   specifing a (bigger) size explicitly using the `-s' option, these parts
+   specifying a (bigger) size explicitly using the `-s' option, these parts
    may be saved.<P>
 
 <DT> [-h]
--- giflib-4.1.6.orig/util/gifbg.c
+++ giflib-4.1.6/util/gifbg.c
@@ -195,7 +195,7 @@ int main(int argc, char **argv)
 	    break;
     }
     if (Direction == DIR_NONE)
-	GIF_EXIT("Direction requested (-d option) is wierd!");
+	GIF_EXIT("Direction requested (-d option) is weird!");
 
     /* We are going to handle only TOP, TOP_RIGHT, RIGHT, BOT_RIGHT  so flip */
     /* the complement cases (TOP <-> BOT for example) by flipping the	     */
