Index: libplist/libplist++.pc.in
===================================================================
--- libplist.orig/libplist++.pc.in	2010-01-13 23:36:20.000000000 +0100
+++ libplist/libplist++.pc.in	2010-01-13 23:37:08.000000000 +0100
@@ -6,7 +6,7 @@
 Name: libplist++
 Description: C++ binding of libplist
 Version: ${LIBPLIST_VERSION}
-Requires: libplist >= ${LIBPLIST_VERSION}
+Requires.private: libplist >= ${LIBPLIST_VERSION}
 Libs: -L${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR} -lplist++
 Cflags: -I${CMAKE_INSTALL_PREFIX}/include
 
