Index: libplist/swig/CMakeLists.txt
===================================================================
--- libplist.orig/swig/CMakeLists.txt	2010-01-18 00:42:55.000000000 +0100
+++ libplist/swig/CMakeLists.txt	2010-01-18 00:43:03.000000000 +0100
@@ -1,5 +1,7 @@
 INCLUDE( ${SWIG_USE_FILE} )
 
+SET(CMAKE_SKIP_RPATH ON)
+
 SET(CMAKE_SWIG_FLAGS -Werror -Wall -modern)
 
 INCLUDE_DIRECTORIES( ${PYTHON_INCLUDE_PATH} )
