Index: libtirpc-0.2.2/src/svc_auth_gss.c
===================================================================
--- libtirpc-0.2.2.orig/src/svc_auth_gss.c
+++ libtirpc-0.2.2/src/svc_auth_gss.c
@@ -386,6 +386,7 @@ _svcauth_gss(struct svc_req *rqst, struc
 			return (AUTH_FAILED);
 		}
 		if ((gd = calloc(sizeof(*gd), 1)) == NULL) {
+			free(auth);
 			fprintf(stderr, "svcauth_gss: out_of_memory\n");
 			return (AUTH_FAILED);
 		}
