Index: libnl-1.1/include/netlink-local.h
===================================================================
--- libnl-1.1.orig/include/netlink-local.h	2008-10-08 19:58:47.000000000 +0200
+++ libnl-1.1/include/netlink-local.h	2008-10-08 19:58:51.000000000 +0200
@@ -26,6 +26,7 @@
 #include <sys/socket.h>
 #include <inttypes.h>
 #include <assert.h>
+#include <limits.h>
 
 #include <arpa/inet.h>
 #include <netdb.h>
