Description: Problems with glyphs Ѝ (U+040D) for Bulgarian language.
Origin: https://admin.fedoraproject.org/updates/FEDORA-2012-7730/liberation-fonts-1.07.2-4.fc17
Bug-Redhat: https://bugzilla.redhat.com/show_bug.cgi?id=799384

--- trunk.orig/src/LiberationMono-Bold.sfd
+++ trunk/src/LiberationMono-Bold.sfd
@@ -13,13 +13,14 @@ Descent: 410
 LayerCount: 2
 Layer: 0 1 "Back"  1
 Layer: 1 1 "Fore"  0
+NeedsXUIDChange: 1
 XUID: [1021 131 222397055 13598569]
 FSType: 0
 OS2Version: 3
 OS2_WeightWidthSlopeOnly: 0
 OS2_UseTypoMetrics: 1
 CreationTime: 1095817380
-ModificationTime: 1328170964
+ModificationTime: 1331115791
 PfmFamily: 17
 TTFWeight: 700
 TTFWidth: 5
@@ -3229,7 +3230,7 @@ NameList: Adobe Glyph List
 DisplaySize: -24
 AntiAlias: 1
 FitToEm: 1
-WinInfo: 1113629 29 18
+WinInfo: 812 29 18
 BeginPrivate: 0
 EndPrivate
 TeXData: 1 0 0 346030 173015 115343 553984 -1048576 115343 783286 444596 497025 792723 393216 433062 380633 303038 157286 324010 404750 52429 2506097 1059062 262144
@@ -64564,11 +64565,11 @@ EndChar
 StartChar: uni040D
 Encoding: 1037 1037 670
 Width: 1229
-Flags: W
+Flags: HW
 LayerCount: 2
 Fore
+Refer: 430 1048 N 1 0 0 1 0 0 2
 Refer: 67 96 N 1 0 0 1 6 243 2
-Refer: 49 78 N 1 0 0 1 0 0 2
 EndChar
 
 StartChar: uni0450
@@ -64595,7 +64596,6 @@ StartChar: S_BE
 Encoding: 1114109 -1 673
 Width: 1229
 VWidth: 0
-Flags: H
 LayerCount: 2
 Fore
 SplineSet
--- trunk.orig/src/LiberationMono-BoldItalic.sfd
+++ trunk/src/LiberationMono-BoldItalic.sfd
@@ -13,13 +13,14 @@ Descent: 410
 LayerCount: 2
 Layer: 0 1 "Back"  1
 Layer: 1 1 "Fore"  0
+NeedsXUIDChange: 1
 XUID: [1021 131 222397055 8302692]
 FSType: 0
 OS2Version: 3
 OS2_WeightWidthSlopeOnly: 0
 OS2_UseTypoMetrics: 1
 CreationTime: 1095817380
-ModificationTime: 1328172358
+ModificationTime: 1331115858
 PfmFamily: 17
 TTFWeight: 700
 TTFWidth: 5
@@ -2841,7 +2842,7 @@ NameList: Adobe Glyph List
 DisplaySize: -24
 AntiAlias: 1
 FitToEm: 1
-WinInfo: 1113629 29 18
+WinInfo: 783 29 18
 BeginPrivate: 0
 EndPrivate
 BeginChars: 1114127 675
@@ -75038,11 +75039,11 @@ EndChar
 StartChar: uni040D
 Encoding: 1037 1037 670
 Width: 1229
-Flags: W
+Flags: HW
 LayerCount: 2
 Fore
+Refer: 430 1048 N 1 0 0 1 0 0 2
 Refer: 67 96 N 1 0 0 1 -15 241 2
-Refer: 49 78 N 1 0 0 1 0 0 2
 EndChar
 
 StartChar: uni0450
@@ -75069,7 +75070,6 @@ StartChar: S_BE
 Encoding: 1114109 -1 673
 Width: 1229
 VWidth: 0
-Flags: H
 LayerCount: 2
 Fore
 SplineSet
@@ -75117,7 +75117,7 @@ EndChar
 StartChar: S_TE
 Encoding: 1114108 -1 674
 Width: 1229
-Flags: HW
+Flags: W
 LayerCount: 2
 Fore
 Refer: 329 713 N 1 0 0 1 0 0 2
--- trunk.orig/src/LiberationMono-Italic.sfd
+++ trunk/src/LiberationMono-Italic.sfd
@@ -13,13 +13,14 @@ Descent: 410
 LayerCount: 2
 Layer: 0 1 "Back"  1
 Layer: 1 1 "Fore"  0
+NeedsXUIDChange: 1
 XUID: [1021 131 222397055 1812088]
 FSType: 0
 OS2Version: 3
 OS2_WeightWidthSlopeOnly: 0
 OS2_UseTypoMetrics: 1
 CreationTime: 1095817380
-ModificationTime: 1328171470
+ModificationTime: 1331115721
 PfmFamily: 17
 TTFWeight: 400
 TTFWidth: 5
@@ -3415,7 +3416,7 @@ NameList: Adobe Glyph List
 DisplaySize: -24
 AntiAlias: 1
 FitToEm: 1
-WinInfo: 1113629 29 18
+WinInfo: 783 29 18
 BeginPrivate: 0
 EndPrivate
 BeginChars: 1114127 675
@@ -80346,11 +80347,11 @@ EndChar
 StartChar: uni040D
 Encoding: 1037 1037 670
 Width: 1229
-Flags: W
+Flags: HW
 LayerCount: 2
 Fore
+Refer: 430 1048 N 1 0 0 1 0 0 2
 Refer: 67 96 N 1 0 0 1 36.2126 241 2
-Refer: 49 78 N 1 0 0 1 0 0 2
 EndChar
 
 StartChar: uni0450
@@ -80377,7 +80378,7 @@ StartChar: S_BE
 Encoding: 1114109 -1 673
 Width: 1229
 VWidth: 0
-Flags: HW
+Flags: W
 LayerCount: 2
 Fore
 SplineSet
@@ -80427,7 +80428,7 @@ EndChar
 StartChar: S_TE
 Encoding: 1114108 -1 674
 Width: 1229
-Flags: HW
+Flags: W
 LayerCount: 2
 Fore
 Refer: 329 713 N 1 0 0 1 0 0 2
--- trunk.orig/src/LiberationSans-Bold.sfd
+++ trunk/src/LiberationSans-Bold.sfd
@@ -13,13 +13,14 @@ Descent: 410
 LayerCount: 2
 Layer: 0 1 "Back"  1
 Layer: 1 1 "Fore"  0
+NeedsXUIDChange: 1
 XUID: [1021 131 222397055 3251725]
 FSType: 0
 OS2Version: 3
 OS2_WeightWidthSlopeOnly: 0
 OS2_UseTypoMetrics: 1
 CreationTime: 1123372800
-ModificationTime: 1327994263
+ModificationTime: 1331115041
 PfmFamily: 17
 TTFWeight: 700
 TTFWidth: 5
@@ -3103,7 +3104,7 @@ NameList: Adobe Glyph List
 DisplaySize: -36
 AntiAlias: 1
 FitToEm: 1
-WinInfo: 1113400 38 14
+WinInfo: 616 44 17
 BeginPrivate: 0
 EndPrivate
 BeginChars: 1114132 681
@@ -83954,11 +83955,11 @@ EndChar
 StartChar: uni040D
 Encoding: 1037 1037 675
 Width: 1479
-Flags: W
+Flags: HW
 LayerCount: 2
 Fore
+Refer: 430 1048 N 1 0 0 1 0 0 2
 Refer: 67 96 N 1 0 0 1 332 312 2
-Refer: 49 78 N 1 0 0 1 0 0 2
 EndChar
 
 StartChar: uni0450
--- trunk.orig/src/LiberationSans-BoldItalic.sfd
+++ trunk/src/LiberationSans-BoldItalic.sfd
@@ -13,13 +13,14 @@ Descent: 410
 LayerCount: 2
 Layer: 0 1 "Back"  1
 Layer: 1 1 "Fore"  0
+NeedsXUIDChange: 1
 XUID: [1021 131 222397055 951244]
 FSType: 0
 OS2Version: 3
 OS2_WeightWidthSlopeOnly: 0
 OS2_UseTypoMetrics: 1
 CreationTime: 1123372800
-ModificationTime: 1327998476
+ModificationTime: 1331115542
 PfmFamily: 17
 TTFWeight: 700
 TTFWidth: 5
@@ -3000,7 +3001,7 @@ NameList: Adobe Glyph List
 DisplaySize: -36
 AntiAlias: 1
 FitToEm: 1
-WinInfo: 44 44 17
+WinInfo: 616 44 17
 BeginPrivate: 0
 EndPrivate
 BeginChars: 1114132 682
@@ -80984,12 +80985,12 @@ EndChar
 
 StartChar: uni040D
 Encoding: 1037 1037 675
-Width: 1479
-Flags: W
+Width: 1473
+Flags: HWO
 LayerCount: 2
 Fore
+Refer: 429 1048 N 1 0 0 1 0 0 2
 Refer: 67 96 N 1 0 0 1 336 306 2
-Refer: 49 78 N 1 0 0 1 0 0 2
 EndChar
 
 StartChar: uni045D
--- trunk.orig/src/LiberationSans-Italic.sfd
+++ trunk/src/LiberationSans-Italic.sfd
@@ -13,13 +13,14 @@ Descent: 410
 LayerCount: 2
 Layer: 0 1 "Back"  1
 Layer: 1 1 "Fore"  0
+NeedsXUIDChange: 1
 XUID: [1021 131 222397055 2032059]
 FSType: 0
 OS2Version: 3
 OS2_WeightWidthSlopeOnly: 0
 OS2_UseTypoMetrics: 1
 CreationTime: 1123372800
-ModificationTime: 1327997434
+ModificationTime: 1331115461
 PfmFamily: 17
 TTFWeight: 400
 TTFWidth: 5
@@ -3144,7 +3145,7 @@ NameList: Adobe Glyph List
 DisplaySize: -36
 AntiAlias: 1
 FitToEm: 1
-WinInfo: 1113838 22 14
+WinInfo: 836 22 14
 BeginPrivate: 0
 EndPrivate
 BeginChars: 1114132 682
@@ -105529,12 +105530,12 @@ EndChar
 
 StartChar: uni040D
 Encoding: 1037 1037 675
-Width: 1479
-Flags: W
+Width: 1465
+Flags: HW
 LayerCount: 2
 Fore
+Refer: 430 1048 N 1 0 0 1 0 0 2
 Refer: 67 96 N 1 0 0 1 513 267 2
-Refer: 49 78 N 1 0 0 1 0 0 2
 EndChar
 
 StartChar: uni0450
@@ -105600,7 +105601,7 @@ EndChar
 StartChar: S_BE
 Encoding: 1114109 -1 680
 Width: 1140
-Flags: HW
+Flags: W
 LayerCount: 2
 Fore
 SplineSet
@@ -105651,7 +105652,7 @@ EndChar
 StartChar: S_TE
 Encoding: 1114108 -1 681
 Width: 1700
-Flags: HW
+Flags: W
 LayerCount: 2
 Fore
 Refer: 329 713 S 1 0 0 1 582 0 2
--- trunk.orig/src/LiberationSans-Regular.sfd
+++ trunk/src/LiberationSans-Regular.sfd
@@ -13,13 +13,14 @@ Descent: 410
 LayerCount: 2
 Layer: 0 1 "Back"  1
 Layer: 1 1 "Fore"  0
+NeedsXUIDChange: 1
 XUID: [1021 131 222397055 16587066]
 FSType: 0
 OS2Version: 3
 OS2_WeightWidthSlopeOnly: 0
 OS2_UseTypoMetrics: 1
 CreationTime: 1123372800
-ModificationTime: 1327993790
+ModificationTime: 1331114858
 PfmFamily: 17
 TTFWeight: 400
 TTFWidth: 5
@@ -2842,7 +2843,7 @@ NameList: Adobe Glyph List
 DisplaySize: -36
 AntiAlias: 1
 FitToEm: 1
-WinInfo: 0 44 17
+WinInfo: 748 44 17
 BeginPrivate: 0
 EndPrivate
 BeginChars: 1114132 681
@@ -86833,11 +86834,11 @@ EndChar
 StartChar: uni040D
 Encoding: 1037 1037 675
 Width: 1479
-Flags: W
+Flags: HW
 LayerCount: 2
 Fore
-Refer: 67 96 N 1 0 0 1 405 269 2
-Refer: 49 78 N 1 0 0 1 0 0 2
+Refer: 430 1048 N 1 0 0 1 0 0 2
+Refer: 67 96 N 1 0 0 1 385 269 2
 EndChar
 
 StartChar: uni0450
--- trunk.orig/src/LiberationSansNarrow-Bold.sfd
+++ trunk/src/LiberationSansNarrow-Bold.sfd
@@ -19,7 +19,7 @@ OS2Version: 1
 OS2_WeightWidthSlopeOnly: 0
 OS2_UseTypoMetrics: 1
 CreationTime: 1123372800
-ModificationTime: 1328007340
+ModificationTime: 1331118512
 PfmFamily: 17
 TTFWeight: 700
 TTFWidth: 3
@@ -800,7 +800,7 @@ NameList: Adobe Glyph List
 DisplaySize: -36
 AntiAlias: 1
 FitToEm: 1
-WinInfo: 924 22 14
+WinInfo: 880 44 17
 BeginPrivate: 0
 EndPrivate
 BeginChars: 65556 681
@@ -59702,12 +59702,12 @@ EndChar
 
 StartChar: uni040D
 Encoding: 1037 1037 675
-Width: 1212
-Flags: W
+Width: 1208
+Flags: HW
 LayerCount: 2
 Fore
+Refer: 430 1048 N 1 0 0 1 0 0 2
 Refer: 67 96 N 1 0 0 1 317 316 2
-Refer: 49 78 N 1 0 0 1 0 0 2
 EndChar
 
 StartChar: uni0450
@@ -59773,7 +59773,7 @@ EndChar
 StartChar: S_BE
 Encoding: 65522 -1 680
 Width: 1018
-Flags: HW
+Flags: W
 LayerCount: 2
 Fore
 SplineSet
--- trunk.orig/src/LiberationSansNarrow-BoldItalic.sfd
+++ trunk/src/LiberationSansNarrow-BoldItalic.sfd
@@ -19,7 +19,7 @@ OS2Version: 1
 OS2_WeightWidthSlopeOnly: 0
 OS2_UseTypoMetrics: 1
 CreationTime: 1123372800
-ModificationTime: 1328009085
+ModificationTime: 1331118589
 PfmFamily: 17
 TTFWeight: 700
 TTFWidth: 3
@@ -805,7 +805,7 @@ NameList: Adobe Glyph List
 DisplaySize: -36
 AntiAlias: 1
 FitToEm: 1
-WinInfo: 65252 22 14
+WinInfo: 660 44 17
 BeginPrivate: 0
 EndPrivate
 BeginChars: 65558 682
@@ -75122,12 +75122,12 @@ EndChar
 
 StartChar: uni040D
 Encoding: 1037 1037 675
-Width: 1212
-Flags: W
+Width: 1208
+Flags: HW
 LayerCount: 2
 Fore
+Refer: 430 1048 N 1 0 0 1 0 0 2
 Refer: 67 96 N 1 0 0 1 244 303 2
-Refer: 49 78 N 1 0 0 1 0 0 2
 EndChar
 
 StartChar: uni045D
@@ -75193,7 +75193,7 @@ EndChar
 StartChar: S_BE
 Encoding: 65556 -1 680
 Width: 1018
-Flags: HW
+Flags: W
 LayerCount: 2
 Fore
 SplineSet
@@ -75249,7 +75249,7 @@ EndChar
 StartChar: S_TE
 Encoding: 65557 -1 681
 Width: 1487
-Flags: HW
+Flags: W
 LayerCount: 2
 Fore
 Refer: 329 713 N 1 0 0 1 170 0 2
--- trunk.orig/src/LiberationSansNarrow-Italic.sfd
+++ trunk/src/LiberationSansNarrow-Italic.sfd
@@ -19,7 +19,7 @@ OS2Version: 1
 OS2_WeightWidthSlopeOnly: 0
 OS2_UseTypoMetrics: 1
 CreationTime: 1123372800
-ModificationTime: 1328010105
+ModificationTime: 1331118431
 PfmFamily: 17
 TTFWeight: 400
 TTFWidth: 3
@@ -790,7 +790,7 @@ NameList: Adobe Glyph List
 DisplaySize: -36
 AntiAlias: 1
 FitToEm: 1
-WinInfo: 65280 20 14
+WinInfo: 616 44 17
 BeginPrivate: 0
 EndPrivate
 BeginChars: 65558 682
@@ -78424,12 +78424,12 @@ EndChar
 
 StartChar: uni040D
 Encoding: 1037 1037 675
-Width: 1212
-Flags: W
+Width: 1202
+Flags: HW
 LayerCount: 2
 Fore
+Refer: 430 1048 N 1 0 0 1 0 0 2
 Refer: 67 96 N 1 0 0 1 474 270 2
-Refer: 49 78 N 1 0 0 1 0 0 2
 EndChar
 
 StartChar: uni0450
@@ -78495,7 +78495,7 @@ EndChar
 StartChar: S_BE
 Encoding: 65556 -1 680
 Width: 936
-Flags: HW
+Flags: W
 LayerCount: 2
 Fore
 SplineSet
@@ -78546,7 +78546,7 @@ EndChar
 StartChar: S_TE
 Encoding: 65557 -1 681
 Width: 1395
-Flags: HW
+Flags: W
 LayerCount: 2
 Fore
 Refer: 329 713 S 1 0 0 1 220 0 2
--- trunk.orig/src/LiberationSansNarrow-Regular.sfd
+++ trunk/src/LiberationSansNarrow-Regular.sfd
@@ -19,7 +19,7 @@ OS2Version: 1
 OS2_WeightWidthSlopeOnly: 0
 OS2_UseTypoMetrics: 1
 CreationTime: 1123372800
-ModificationTime: 1328004541
+ModificationTime: 1331118372
 PfmFamily: 17
 TTFWeight: 400
 TTFWidth: 3
@@ -785,7 +785,7 @@ NameList: Adobe Glyph List
 DisplaySize: -36
 AntiAlias: 1
 FitToEm: 1
-WinInfo: 0 44 17
+WinInfo: 528 44 17
 BeginPrivate: 0
 EndPrivate
 BeginChars: 65556 681
@@ -35767,7 +35767,7 @@ StartChar: afii10026
 Encoding: 1048 1048 430
 Width: 1207
 GlyphClass: 2
-Flags: W
+Flags: WO
 TtInstrs:
 SVTCA[y-axis]
 PUSHB_3
@@ -61982,12 +61982,12 @@ EndChar
 
 StartChar: uni040D
 Encoding: 1037 1037 675
-Width: 1212
-Flags: W
+Width: 1207
+Flags: HWO
 LayerCount: 2
 Fore
+Refer: 430 1048 N 1 0 0 1 0 0 2
 Refer: 67 96 N 1 0 0 1 317 264 2
-Refer: 49 78 N 1 0 0 1 0 0 2
 EndChar
 
 StartChar: uni0450
@@ -62053,7 +62053,7 @@ EndChar
 StartChar: S_BE
 Encoding: 65522 -1 680
 Width: 934
-Flags: HW
+Flags: W
 LayerCount: 2
 Fore
 SplineSet
--- trunk.orig/src/LiberationSerif-Bold.sfd
+++ trunk/src/LiberationSerif-Bold.sfd
@@ -13,13 +13,14 @@ Descent: 410
 LayerCount: 2
 Layer: 0 1 "Back"  1
 Layer: 1 1 "Fore"  0
+NeedsXUIDChange: 1
 XUID: [1021 131 222397055 11337752]
 FSType: 0
 OS2Version: 3
 OS2_WeightWidthSlopeOnly: 0
 OS2_UseTypoMetrics: 1
 CreationTime: 1123398000
-ModificationTime: 1327999755
+ModificationTime: 1331117181
 PfmFamily: 17
 TTFWeight: 700
 TTFWidth: 5
@@ -2704,7 +2705,7 @@ NameList: Adobe Glyph List
 DisplaySize: -24
 AntiAlias: 1
 FitToEm: 1
-WinInfo: 66 33 18
+WinInfo: 627 33 18
 BeginPrivate: 0
 EndPrivate
 BeginChars: 1114124 671
@@ -93433,12 +93434,12 @@ EndChar
 
 StartChar: uni040D
 Encoding: 1037 1037 667
-Width: 1479
-Flags: W
+Width: 1593
+Flags: HW
 LayerCount: 2
 Fore
+Refer: 430 1048 N 1 0 0 1 0 0 2
 Refer: 67 96 N 1 0 0 1 339 334 2
-Refer: 49 78 N 1 0 0 1 0 0 2
 EndChar
 
 StartChar: uni0450
@@ -93464,7 +93465,7 @@ EndChar
 StartChar: S_BE
 Encoding: 1114109 -1 670
 Width: 1063
-Flags: HW
+Flags: W
 LayerCount: 2
 Fore
 SplineSet
--- trunk.orig/src/LiberationSerif-BoldItalic.sfd
+++ trunk/src/LiberationSerif-BoldItalic.sfd
@@ -20,7 +20,7 @@ OS2Version: 3
 OS2_WeightWidthSlopeOnly: 0
 OS2_UseTypoMetrics: 1
 CreationTime: 1123398000
-ModificationTime: 1328001460
+ModificationTime: 1331117345
 PfmFamily: 17
 TTFWeight: 700
 TTFWidth: 5
@@ -3180,7 +3180,7 @@ NameList: Adobe Glyph List
 DisplaySize: -24
 AntiAlias: 1
 FitToEm: 1
-WinInfo: 682 31 18
+WinInfo: 868 31 18
 BeginPrivate: 0
 EndPrivate
 BeginChars: 1114124 672
@@ -94874,12 +94874,12 @@ EndChar
 
 StartChar: uni040D
 Encoding: 1037 1037 667
-Width: 1479
-Flags: W
+Width: 1593
+Flags: HW
 LayerCount: 2
 Fore
+Refer: 430 1048 N 1 0 0 1 0 0 2
 Refer: 67 96 N 1 0 0 1 295 333 2
-Refer: 49 78 N 1 0 0 1 0 0 2
 EndChar
 
 StartChar: uni0450
@@ -94965,7 +94965,7 @@ EndChar
 StartChar: S_TE
 Encoding: 1114108 -1 671
 Width: 1706
-Flags: HWO
+Flags: W
 LayerCount: 2
 Fore
 Refer: 329 713 N 1 0 0 1 560 0 2
--- trunk.orig/src/LiberationSerif-Italic.sfd
+++ trunk/src/LiberationSerif-Italic.sfd
@@ -13,13 +13,14 @@ Descent: 410
 LayerCount: 2
 Layer: 0 1 "Back"  1
 Layer: 1 1 "Fore"  0
+NeedsXUIDChange: 1
 XUID: [1021 131 222397055 13298560]
 FSType: 0
 OS2Version: 3
 OS2_WeightWidthSlopeOnly: 0
 OS2_UseTypoMetrics: 1
 CreationTime: 1123398000
-ModificationTime: 1328000087
+ModificationTime: 1331117272
 PfmFamily: 17
 TTFWeight: 400
 TTFWidth: 5
@@ -3100,7 +3101,7 @@ NameList: Adobe Glyph List
 DisplaySize: -24
 AntiAlias: 1
 FitToEm: 1
-WinInfo: 60 30 18
+WinInfo: 690 30 18
 BeginPrivate: 0
 EndPrivate
 BeginChars: 1114124 673
@@ -88694,12 +88695,12 @@ EndChar
 
 StartChar: uni040D
 Encoding: 1037 1037 668
-Width: 1366
-Flags: W
+Width: 1479
+Flags: HW
 LayerCount: 2
 Fore
+Refer: 430 1048 N 1 0 0 1 0 0 2
 Refer: 67 96 N 1 0 0 1 254 332 2
-Refer: 49 78 N 1 0 0 1 0 0 2
 EndChar
 
 StartChar: uni0450
@@ -88725,7 +88726,7 @@ EndChar
 StartChar: S_BE
 Encoding: 1114109 -1 671
 Width: 965
-Flags: HW
+Flags: W
 LayerCount: 2
 Fore
 SplineSet
--- trunk.orig/src/LiberationSerif-Regular.sfd
+++ trunk/src/LiberationSerif-Regular.sfd
@@ -13,13 +13,14 @@ Descent: 410
 LayerCount: 2
 Layer: 0 1 "Back"  1
 Layer: 1 1 "Fore"  0
+NeedsXUIDChange: 1
 XUID: [1021 131 222397055 16509670]
 FSType: 0
 OS2Version: 3
 OS2_WeightWidthSlopeOnly: 0
 OS2_UseTypoMetrics: 1
 CreationTime: 1123398000
-ModificationTime: 1327999557
+ModificationTime: 1331117005
 PfmFamily: 17
 TTFWeight: 400
 TTFWidth: 5
@@ -2742,7 +2743,7 @@ NameList: Adobe Glyph List
 DisplaySize: -24
 AntiAlias: 1
 FitToEm: 1
-WinInfo: 1113629 29 18
+WinInfo: 696 29 18
 BeginPrivate: 0
 EndPrivate
 BeginChars: 1114124 672
@@ -97065,11 +97066,11 @@ EndChar
 StartChar: uni040D
 Encoding: 1037 1037 668
 Width: 1479
-Flags: W
+Flags: HWO
 LayerCount: 2
 Fore
+Refer: 430 1048 N 1 0 0 1 0 0 2
 Refer: 67 96 N 1 0 0 1 279 330 2
-Refer: 49 78 N 1 0 0 1 0 0 2
 EndChar
 
 StartChar: uni0450
@@ -97095,7 +97096,7 @@ EndChar
 StartChar: S_BE
 Encoding: 1114109 -1 671
 Width: 965
-Flags: HW
+Flags: W
 LayerCount: 2
 Fore
 SplineSet
