Last-Update: 2013-03-04
Author: Dmitry Smirnov <onlyjob@member.fsf.org>
Forwarded: not-needed
Description: don't install non-executable files to /usr/bin

--- a/configure.ac
+++ b/configure.ac
@@ -403,9 +403,9 @@
 if test "x$prefix" = xNONE; then
 	prefix=/usr/local
 fi
 
-AC_DEFINE_UNQUOTED([CGMINER_PREFIX], ["$prefix/bin"], [Path to cgminer install])
+AC_DEFINE_UNQUOTED([CGMINER_PREFIX], ["$bindir"], [Path to cgminer install])
 
 AC_DEFINE_UNQUOTED([PHATK_KERNNAME], ["phatk121016"], [Filename for phatk kernel])
 AC_DEFINE_UNQUOTED([POCLBM_KERNNAME], ["poclbm130302"], [Filename for poclbm kernel])
 AC_DEFINE_UNQUOTED([DIAKGCN_KERNNAME], ["diakgcn121016"], [Filename for diakgcn kernel])
