Index: gdm-3.0.4/configure.ac
===================================================================
--- gdm-3.0.4.orig/configure.ac	2011-05-31 18:24:23.000000000 +0200
+++ gdm-3.0.4/configure.ac	2011-10-14 14:31:00.163703427 +0200
@@ -1169,28 +1169,10 @@
 dnl - Check for Xnest / Xephyr support
 dnl ---------------------------------------------------------------------------
 
-# Use Xephyr if it is available.  It works better than Xnest since Xephyr
-# supports the Xserver extensions, even if on a remote machine.
-#
+# Use the wrapper script for the nested X server
+X_XNEST_CMD=/usr/share/gdm/gdmXnestWrapper
+X_XNEST_CONFIG_OPTIONS="-audit 0"
 X_XNEST_UNSCALED_FONTPATH="true"
-if test -x $X_SERVER_PATH/Xephyr; then
-    X_XNEST_CMD="$X_SERVER_PATH/Xephyr"
-    X_XNEST_CONFIG_OPTIONS="-audit 0"
-else
-    if test -x /usr/openwin/bin/Xnest; then
-        # If on Solaris, Xnest is only shipped with the Xsun Xserver, so
-        # use this version.
-        #
-        X_XNEST_CMD="/usr/openwin/bin/Xnest"
-        X_XNEST_CONFIG_OPTIONS="-audit 0 -name Xnest -pn"
-        X_XNEST_UNSCALED_FONTPATH="false"
-    else
-        if test -x $X_SERVER_PATH/Xnest; then
-            X_XNEST_CMD="$X_SERVER_PATH/Xnest"
-            X_XNEST_CONFIG_OPTIONS="-audit 0 -name Xnest"
-        fi
-    fi
-fi
 
 dnl ---------------------------------------------------------------------------
 dnl - Expand vars
