GNOME #645918

--- a/gui/simple-chooser/gdm-chooser-session.c
+++ b/gui/simple-chooser/gdm-chooser-session.c
@@ -133,7 +133,7 @@
         ret = FALSE;
 
         error = NULL;
-        g_spawn_command_line_async (LIBEXECDIR "/gnome-settings-daemon", &error);
+        g_spawn_command_line_async (GSD_BINARY, &error);
         if (error != NULL) {
                 g_warning ("Error starting settings daemon: %s", error->message);
                 g_error_free (error);
--- a/configure.ac
+++ b/configure.ac
@@ -74,6 +74,7 @@
 FONTCONFIG_REQUIRED_VERSION=2.5.0
 UPOWER_REQUIRED_VERSION=0.9.0
 ACCOUNTS_SERVICE_REQUIRED_VERSION=0.6.12
+GSD_REQUIRED_VERSION=2.91.2
 NSS_REQUIRED_VERSION=3.11.1
 
 EXTRA_COMPILE_WARNINGS(yes)
@@ -185,6 +186,16 @@
 AC_SUBST(LIBGDM_GIR_INCLUDES)
 
 dnl ---------------------------------------------------------------------------
+dnl - Paths for dependencies
+dnl ---------------------------------------------------------------------------
+
+PKG_CHECK_MODULES(GSD,
+        gnome-settings-daemon >= $GSD_REQUIRED_VERSION
+)
+GSD_BINARY=`pkg-config --variable=binary gnome-settings-daemon`
+AC_SUBST(GSD_BINARY)
+
+dnl ---------------------------------------------------------------------------
 dnl - Configuration file stuff
 dnl ---------------------------------------------------------------------------
 
--- a/gui/simple-chooser/Makefile.am
+++ b/gui/simple-chooser/Makefile.am
@@ -14,6 +14,7 @@
 	-DSBINDIR=\""$(sbindir)"\"		 	\
 	-DPIXMAPDIR=\""$(pixmapdir)"\"		 	\
 	-DAT_SPI_REGISTRYD_DIR="\"$(AT_SPI_REGISTRYD_DIR)\""	\
+	-DGSD_BINARY="\"$(GSD_BINARY)\""		\
 	$(GTK_CFLAGS)					\
 	$(XLIB_CFLAGS)					\
 	$(XDMCP_CFLAGS)					\
