--- a/daemon/gdm-simple-slave.c
+++ b/daemon/gdm-simple-slave.c
@@ -506,7 +506,8 @@
         }
 
         g_debug ("GdmSimpleSlave: Starting automatic login conversation");
-        gdm_session_start_conversation (slave->priv->session, "gdm-autologin");
+        gdm_session_start_conversation (slave->priv->session,
+          GDM_PAM_DOMAIN"-autologin");
 }
 
 static void
--- a/configure.ac
+++ b/configure.ac
@@ -532,6 +532,21 @@
 fi
 AC_SUBST(PAM_PREFIX)
 
+AC_ARG_WITH(pam-domain,
+            AS_HELP_STRING([--with-pam-domain=<domain>],
+                           [specify the naming scheme of pam files]),
+            [if test x$withval != x; then
+               AC_MSG_RESULT("PAM files will be named ${withval} and ${withval}-autologin.")
+             fi])
+
+if test x$withval != x; then
+	GDM_PAM_DOMAIN="$withval"
+else
+	GDM_PAM_DOMAIN=gdm
+fi
+AC_SUBST(GDM_PAM_DOMAIN)
+AC_DEFINE_UNQUOTED([GDM_PAM_DOMAIN], "$GDM_PAM_DOMAIN", [Naming scheme for PAM files])
+
 have_pam=no
 AC_CHECK_LIB(pam, pam_start, have_pam=yes)
 
--- a/gui/simple-greeter/extensions/unified/Makefile.am
+++ b/gui/simple-greeter/extensions/unified/Makefile.am
@@ -1,5 +1,5 @@
 NULL =
-PAM_SERVICE_NAME = gdm
+PAM_SERVICE_NAME = $(GDM_PAM_DOMAIN)
 
 extensiondir = $(GDM_SIMPLE_GREETER_EXTENSIONS_DATA_DIR)/unified
 extension_DATA = page.ui
--- a/daemon/Makefile.am
+++ b/daemon/Makefile.am
@@ -20,6 +20,7 @@
 	-DGDM_SCREENSHOT_DIR=\"$(GDM_SCREENSHOT_DIR)\"		\
 	-DGDM_CACHE_DIR=\""$(localstatedir)/cache/gdm"\"	\
 	-DGDM_SESSION_DEFAULT_PATH=\"$(GDM_SESSION_DEFAULT_PATH)\" \
+	-DGDM_PAM_DOMAIN="\"$(GDM_PAM_DOMAIN)\"" \
 	-DCONSOLEKIT_DIR=\"$(CONSOLEKIT_DIR)\" \
 	$(DISABLE_DEPRECATED_CFLAGS)			\
 	$(DAEMON_CFLAGS)				\
--- a/gui/simple-greeter/Makefile.am
+++ b/gui/simple-greeter/Makefile.am
@@ -21,6 +21,7 @@
 	-DSBINDIR=\""$(sbindir)"\"		 	\
 	-DGDM_CACHE_DIR=\""$(localstatedir)/cache/gdm"\"	\
 	-DAT_SPI_REGISTRYD_DIR="\"$(AT_SPI_REGISTRYD_DIR)\""	\
+	-DGDM_PAM_DOMAIN="\"$(GDM_PAM_DOMAIN)\""	\
 	$(UPOWER_CFLAGS)				\
 	-DGDM_SIMPLE_GREETER_PLUGINS_DIR="\"$(GDM_SIMPLE_GREETER_PLUGINS_DIR)\""\
 	$(DISABLE_DEPRECATED_CFLAGS)			\
--- a/gui/simple-greeter/gdm-greeter-login-window.c
+++ b/gui/simple-greeter/gdm-greeter-login-window.c
@@ -753,7 +753,7 @@
         if (find_extension_with_service_name (login_window, "gdm-password") != NULL) {
                 begin_single_service_verification (login_window, "gdm-password");
         } else {
-                begin_single_service_verification (login_window, "gdm");
+                begin_single_service_verification (login_window, GDM_PAM_DOMAIN);
         }
 }
 
@@ -985,7 +985,7 @@
          * FIXME: we need to get this policy out of the source code
          */
         if (strcmp (service_name, "gdm-password") == 0 ||
-            strcmp (service_name, "gdm") == 0) {
+            strcmp (service_name, GDM_PAM_DOMAIN) == 0) {
                 g_debug ("GdmGreeterLoginWindow: main conversation failed, starting over");
                 restart_conversations (login_window);
                 reset_dialog_after_messages (login_window, MODE_SELECTION);
