From: Jeremy Bicha <jbicha@ubuntu.com>
Description: Don't mix intltool and gettext macros.

Index: gdm-3.5.91/configure.ac
===================================================================
--- gdm-3.5.91.orig/configure.ac	2012-09-05 12:41:58.770013245 -0400
+++ gdm-3.5.91/configure.ac	2012-09-05 12:43:49.646015603 -0400
@@ -52,9 +52,6 @@
 # i18n stuff
 IT_PROG_INTLTOOL([0.40.0])
 
-AM_GNU_GETTEXT_VERSION([0.17])
-AM_GNU_GETTEXT([external])
-
 GETTEXT_PACKAGE=gdm
 AC_SUBST(GETTEXT_PACKAGE)
 AC_DEFINE_UNQUOTED(GETTEXT_PACKAGE, "$GETTEXT_PACKAGE", [gettext package])
