Index: gdm-3.4.1/daemon/gdm-local-display-factory.c
===================================================================
--- gdm-3.4.1.orig/daemon/gdm-local-display-factory.c	2012-06-09 17:20:47.000000000 +0200
+++ gdm-3.4.1/daemon/gdm-local-display-factory.c	2012-06-09 17:32:13.684235405 +0200
@@ -331,8 +331,6 @@ on_display_status_changed (GdmDisplay
 
                 if (switch_on_finish) {
                 	switch_to_login_session (factory);
-                 } else if (type == GDM_TYPE_STATIC_DISPLAY) {
-                 	create_display (factory, seat_id, type);
                  }
                 break;
         case GDM_DISPLAY_FAILED:
Index: gdm-3.4.1/daemon/gdm-static-display.c
===================================================================
--- gdm-3.4.1.orig/daemon/gdm-static-display.c	2011-10-21 06:13:08.000000000 +0200
+++ gdm-3.4.1/daemon/gdm-static-display.c	2012-06-09 17:31:32.568034385 +0200
@@ -93,21 +93,13 @@ gdm_static_display_manage (GdmDisplay *d
 static gboolean
 gdm_static_display_finish (GdmDisplay *display)
 {
-        int status;
-
         g_return_val_if_fail (GDM_IS_DISPLAY (display), FALSE);
 
-        /* Don't call parent's finish since we don't ever
-           want to be put in the FINISHED state */
+        GDM_DISPLAY_CLASS (gdm_static_display_parent_class)->finish (display);
 
-        /* restart static displays */
+        /* We don't restart static displays unless switch_on_finish is set */
         gdm_display_unmanage (display);
 
-        status = gdm_display_get_status (display);
-        if (status != GDM_DISPLAY_FAILED) {
-                gdm_display_manage (display);
-        }
-
         return TRUE;
 }
 
