GNOME #678718

Index: gdm-3.4.1/data/gdm-fallback.session
===================================================================
--- gdm-3.4.1.orig/data/gdm-fallback.session	2011-10-21 06:34:00.000000000 +0200
+++ gdm-3.4.1/data/gdm-fallback.session	2012-06-21 00:16:50.139568722 +0200
@@ -1,5 +1,5 @@
 [GNOME Session]
 Name=Display Manager
-RequiredComponents=gdm-simple-greeter;gnome-settings-daemon;
+RequiredComponents=gdm-simple-greeter;gnome-settings-daemon;polkit-gnome-authentication-agent-1
 RequiredProviders=windowmanager;
 DefaultProvider-windowmanager=metacity
