Index: gdm-3.4.1/configure.ac
===================================================================
--- gdm-3.4.1.orig/configure.ac	2012-06-20 21:36:05.936420624 +0200
+++ gdm-3.4.1/configure.ac	2012-06-20 21:36:06.004420954 +0200
@@ -37,6 +37,8 @@ AC_SUBST(LT_AGE)
 AC_HEADER_STDC
 
 AC_SUBST(VERSION)
+MAJOR_VERSION=`echo $VERSION | cut -d. -f -2`
+AC_SUBST(MAJOR_VERSION)
 
 AM_CONFIG_HEADER(config.h)
 AC_CONFIG_MACRO_DIR([m4])
Index: gdm-3.4.1/data/Makefile.am
===================================================================
--- gdm-3.4.1.orig/data/Makefile.am	2012-06-20 21:36:05.800419964 +0200
+++ gdm-3.4.1/data/Makefile.am	2012-06-20 21:36:06.008420978 +0200
@@ -80,6 +80,7 @@ gdm.schemas.in: $(srcdir)/gdm.schemas.in
 
 gdm-shell.session: $(srcdir)/gdm-shell.session.in
 	sed	-e 's,[@]SESSION_LIBEXECDIR[@],$(SESSION_LIBEXECDIR),g' \
+		-e 's,[@]MAJOR_VERSION[@],$(MAJOR_VERSION),g' \
 		< $(srcdir)/gdm-shell.session.in > gdm-shell.session
 
 localealiasdir = $(datadir)/gdm
Index: gdm-3.4.1/data/gdm-shell.session.in
===================================================================
--- gdm-3.4.1.orig/data/gdm-shell.session.in	2012-06-20 21:36:05.800419964 +0200
+++ gdm-3.4.1/data/gdm-shell.session.in	2012-06-20 21:53:17.285462627 +0200
@@ -1,5 +1,5 @@
 [GNOME Session]
 Name=Display Manager
 RequiredComponents=gnome-shell;gnome-settings-daemon;
-IsRunnableHelper=bash -c 'gnome-shell --help | grep -q gdm-mode && @SESSION_LIBEXECDIR@/gnome-session-check-accelerated'
+IsRunnableHelper=bash -c 'gnome-shell --version | cut -d\  -f 3 | cut -d. -f -2 | xargs test @MAJOR_VERSION@ = && @SESSION_LIBEXECDIR@/gnome-session-check-accelerated'
 FallbackSession=gdm-fallback
