--- a/gui/simple-greeter/libgdmsimplegreeter/Makefile.am
+++ b/gui/simple-greeter/libgdmsimplegreeter/Makefile.am
@@ -14,7 +14,7 @@
 	$(GTK_CFLAGS)				\
 	$(NULL)
 
-lib_LTLIBRARIES = 			\
+pkglib_LTLIBRARIES = 			\
 	libgdmsimplegreeter.la		\
 	$(NULL)
 
--- a/gui/libgdm/Makefile.am
+++ b/gui/libgdm/Makefile.am
@@ -40,7 +40,7 @@
         -DDATADIR=\""$(datadir)"\"                                            \
         $(END_OF_LIST)
 
-lib_LTLIBRARIES =                                                             \
+pkglib_LTLIBRARIES =                                                   \
         libgdm.la                                                      \
         $(END_OF_LIST)
 
@@ -87,9 +87,9 @@
 -include $(INTROSPECTION_MAKEFILE)
 
 if HAVE_INTROSPECTION
-girdir = $(datadir)/gir-1.0
+girdir = $(pkglibdir)
 gir_DATA = Gdm-1.0.gir
-typelibsdir = $(libdir)/girepository-1.0
+typelibsdir = $(pkglibdir)
 typelibs_DATA = $(gir_DATA:.gir=.typelib)
 INTROSPECTION_GIRS = $(gir_DATA)
 
