--- a/daemon/gdm-manager.c
+++ b/daemon/gdm-manager.c
@@ -444,7 +444,7 @@
                                                      "org.freedesktop.ConsoleKit",
                                                      sessions[i],
                                                      "org.freedesktop.ConsoleKit.Session",
-                                                     "GetX11DisplayDevice",
+                                                     "GetX11Display",
                                                      NULL,
                                                      G_VARIANT_TYPE ("(s)"),
                                                      G_DBUS_CALL_FLAGS_NONE,
