--- /dev/null
+++ b/Makefile
@@ -0,0 +1,18 @@
+DESTDIR =
+
+all:
+	cd src; \
+	qmake -config release; \
+	make
+	# preparing the installation for debian.
+	# See debian/zegrapher.install
+	cp -a src/ZeGrapher src/zegrapher
+	cp src/icons/logoLogiciel.png src/icons/zegrapher.png
+clean:
+	cd src; \
+	[ ! -f Makefile ] || make distclean
+	rm -f src/zegrapher src/icons/zegrapher.png
+
+install:
+
+.PHONY: all clean install
