--- /dev/null
+++ b/zegrapher.desktop
@@ -0,0 +1,13 @@
+[Desktop Entry]
+Encoding=UTF-8
+Version=1.0
+Type=Application
+Exec=/usr/bin/zegrapher
+Icon=zegrapher.png
+Terminal=false
+Name=ZeGrapher
+GenericName=Plots of functions and series
+GenericName[fr_FR]=Graphiques de fonctions et de suites
+Comment=Interactive tool plot mathematic objects
+Comment[fr_FR]=Outil interactif pour tracer des courbes mathématiques
+Categories=GNOME;Application;Education;Math;
