From 629055acc3d2070c051aa8bd7b85f59c13b4d879 Mon Sep 17 00:00:00 2001
From: Nobuhiro Iwamatsu <iwamatsu@debian.org>
Date: Wed, 24 Aug 2011 12:13:20 +0900
Subject: Optimize of i386 changed from 686 to 486.

Signed-off-by: Nobuhiro Iwamatsu <iwamatsu@debian.org>
---
 CMakeLists.txt |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0709624..8aa2c7e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1175,7 +1175,7 @@ if(CMAKE_COMPILER_IS_GNUCXX)
     endif()
 
     if(X86 AND NOT MINGW64 AND NOT X86_64 AND NOT APPLE)
-        set(EXTRA_C_FLAGS "${EXTRA_C_FLAGS} -march=i686")
+        set(EXTRA_C_FLAGS "${EXTRA_C_FLAGS} -march=i486")
     endif()
 
     # Other optimizations
-- 
1.7.5.4

