Description: FTBFS with libav 0.8
Forwarded: not yet
Author: Nobuhiro Iwamatsu <iwamatsu@debian.org>
Bug-Debian: http://bugs.debian.org/654220
Last-Update: 2012-01-27

diff --git a/modules/highgui/src/cap_ffmpeg_impl.hpp b/modules/highgui/src/cap_ffmpeg_impl.hpp
index 14ed8b7..f1e0fcb 100644
--- a/modules/highgui/src/cap_ffmpeg_impl.hpp
+++ b/modules/highgui/src/cap_ffmpeg_impl.hpp
@@ -108,6 +108,7 @@ extern "C" {
 #if defined(HAVE_LIBSWSCALE_SWSCALE_H)
   #include <libswscale/swscale.h>
 #endif
+#include <libavutil/mathematics.h>
 
 #endif
 
