Subject: Spelling errors
Bug: https://rt.cpan.org/Ticket/Display.html?id=74481
Forwarded: https://rt.cpan.org/Ticket/Display.html?id=74481
Author: Nicholas Bamber <nicholas@periapt.co.uk>
Reviewed-by: gregor herrmann <gregoa@debian.org>
Last-Update: 2012-01-28

--- a/lib/Class/Inspector.pm
+++ b/lib/Class/Inspector.pm
@@ -151,7 +151,7 @@
   print Class->filename( 'Foo::Bar' );
   > Foo/Bar.pm
 
-This filename will be returned with the right seperator for the local
+This filename will be returned with the right separator for the local
 platform, and should work on all platforms.
 
 Returns the filename on success or C<undef> if the class name is invalid.
@@ -355,7 +355,7 @@
 The C<expanded> option will cause a lot more information about method to be 
 returned. Instead of just the method name, you will instead get an array
 reference containing the method name as a single combined name, ala C<full>,
-the seperate class and method, and a CODE ref to the actual function ( if
+the separate class and method, and a CODE ref to the actual function ( if
 available ). Please note that the function reference is not guarenteed to 
 be available. C<Class::Inspector> is intended at some later time, work 
 with modules that have some some of common run-time loader in place ( e.g
