Index: gnome-js-common-0.1.1/gnome-js-common.pc.in
===================================================================
--- gnome-js-common-0.1.1.orig/gnome-js-common.pc.in	2009-08-26 22:14:51.161301547 +0200
+++ gnome-js-common-0.1.1/gnome-js-common.pc.in	2009-08-26 22:14:56.249295958 +0200
@@ -3,7 +3,7 @@ exec_prefix=@exec_prefix@
 libdir=@libdir@
 bindir=@bindir@
 includedir=@includedir@
-moduledir=@libdir@/gnome-js
+moduledir=@datadir@/gnome-js
 
 Name: GNOME JS Common
 Description: Common modules for GNOME JavaScript bindings.
Index: gnome-js-common-0.1.1/modules/Makefile.am
===================================================================
--- gnome-js-common-0.1.1.orig/modules/Makefile.am	2009-08-26 22:15:30.337296112 +0200
+++ gnome-js-common-0.1.1/modules/Makefile.am	2009-08-26 22:15:34.009296043 +0200
@@ -1,6 +1,6 @@
 SUBDIRS=tweener
 
-moduledir=$(libdir)/gnome-js
+moduledir=$(datadir)/gnome-js
 module_DATA= \
 	lang.js \
 	signals.js \
Index: gnome-js-common-0.1.1/modules/tweener/Makefile.am
===================================================================
--- gnome-js-common-0.1.1.orig/modules/tweener/Makefile.am	2009-08-26 22:15:39.497304301 +0200
+++ gnome-js-common-0.1.1/modules/tweener/Makefile.am	2009-08-26 22:15:43.601333928 +0200
@@ -1,4 +1,4 @@
-tweenerdir=$(libdir)/gnome-js/tweener
+tweenerdir=$(datadir)/gnome-js/tweener
 tweener_DATA= \
 	equations.js \
 	tweener.js \
