Author: Nicholas Bamber <nicholas@periapt.co.uk>
Subject: Unnecessary paths in .pm
Forwarded: no
Last-Update: 2011-04-09
--- a/NTLM.pm
+++ b/NTLM.pm
@@ -1,5 +1,3 @@
-#!/usr/local/bin/perl
-
 package Authen::NTLM;
 use strict;
 use Authen::NTLM::DES;
--- a/DES/DES.pm
+++ b/DES/DES.pm
@@ -1,5 +1,3 @@
-#!/usr/local/bin/perl
-#
 # This is an implementation of part of the DES specification.  According
 # to the code this is ported from, this code does NOT enable 2-way
 # encryption and is, hence, not a cypher and does not appear to come
--- a/MD4/MD4.pm
+++ b/MD4/MD4.pm
@@ -1,5 +1,3 @@
-#!/usr/local/bin/perl
-#
 # This is a partial implentation of the MD4 checksum code.
 #
 # NOTE
