Description: Fix spelling errors in insserv(8).
From: Kel Modderman <kel@otaku42.de>
---
--- a/insserv.8.in
+++ b/insserv.8.in
@@ -124,7 +124,7 @@ these tags is stopped.
 The optional X\-Interactive keyword implies that the script using this
 keyword should be started alone in a concurrent boot configuration
 because it interact with the user at the console.  Only the value
-`true' is recogniced.  All other are ignored.
+`true' is recognised.  All other are ignored.
 .PP
 The optional
 .B X\-Start\-Before
@@ -244,7 +244,7 @@ in the configuration file
 also knows the special facility
 .BR $all .
 This facility indicates that a service should be inserted
-at the end of all services at starting and at the very begining
+at the end of all services at starting and at the very beginning
 at stopping.  Clearly all services using this facility will be
 grouped into one starting or stopping order.
 .\"
