Index: modemmanager.fix-910736/src/77-mm-usb-device-blacklist.rules
===================================================================
--- modemmanager.fix-910736.orig/src/77-mm-usb-device-blacklist.rules	2012-01-02 13:52:27.976866000 +0700
+++ modemmanager.fix-910736/src/77-mm-usb-device-blacklist.rules	2012-01-02 13:55:27.610373581 +0700
@@ -77,5 +77,10 @@
 # QinHeng Electronics HL-340
 ATTRS{idVendor}=="1a86", ATTRS{idProduct}=="7523", ENV{ID_MM_DEVICE_IGNORE}="1"
 
+# Arduino devices, FT-232-based series and Atmega8U2 (Uno)
+ATTRS{idVendor}=="0403", ATTRS{idProduct}=="6001", ENV{ID_MM_DEVICE_IGNORE}="1"
+ATTRS{idVendor}=="03eb", ATTRS{idProduct}=="204b", ENV{ID_MM_DEVICE_IGNORE}="1"
+ATTRS{idVendor}=="2341", ENV{ID_MM_DEVICE_IGNORE}="1"
+
 LABEL="mm_usb_device_blacklist_end"
 
