Index: modemmanager-0.5/src/77-mm-usb-device-blacklist.rules
===================================================================
--- modemmanager-0.5.orig/src/77-mm-usb-device-blacklist.rules	2011-08-05 11:37:37.535618352 -0400
+++ modemmanager-0.5/src/77-mm-usb-device-blacklist.rules	2011-08-05 11:39:06.565883158 -0400
@@ -68,5 +68,14 @@
 # ATEN Intl UC-232A (Prolific)
 ATTRS{idVendor}=="0557", ATTRS{idProduct}=="2008", ENV{ID_MM_DEVICE_IGNORE}="1"
 
+# Prolific Technology, Inc. PL2303 Serial Port
+ATTRS{idVendor}=="067b", ATTRS{idProduct}=="2303", ENV{ID_MM_DEVICE_IGNORE}="1"
+
+# Cygnal Integrated Products, Inc. CP210x
+ATTRS{idVendor}=="10c4", ATTRS{idProduct}=="ea60", ENV{ID_MM_DEVICE_IGNORE}="1"
+
+# QinHeng Electronics HL-340
+ATTRS{idVendor}=="1a86", ATTRS{idProduct}=="7523", ENV{ID_MM_DEVICE_IGNORE}="1"
+
 LABEL="mm_usb_device_blacklist_end"
 
