--- a/cmake/admDetermineSystem.cmake
+++ b/cmake/admDetermineSystem.cmake
@@ -111,7 +111,6 @@ ELSE (X86_64_SUPPORTED)
 
 		IF (POWERPC_SUPPORTED)
 			SET(ADM_CPU_PPC 1)
-			SET(ADM_ALTIVEC_FLAGS "-mabi=altivec -maltivec")
 
 			IF (APPLE)
 				SET(ADM_ALTIVEC_FLAGS "${ADM_ALTIVEC_FLAGS} -faltivec")
@@ -119,10 +118,6 @@ ELSE (X86_64_SUPPORTED)
 
 			PERFORM_SYSTEM_TEST(cpu_altivec_check.cpp "AltiVec capable" ALTIVEC_SUPPORTED "${ADM_ALTIVEC_FLAGS}")
 
-			IF (ALTIVEC_SUPPORTED)
-				SET(ADM_CPU_ALTIVEC 1)
-			ENDIF (ALTIVEC_SUPPORTED)
-
 			PERFORM_SYSTEM_TEST(cpu_dcbzl_check.cpp "dcbzl capable" DCBZL_SUPPORTED)
 
 			IF (DCBZL_SUPPORTED)
@@ -156,4 +151,4 @@ IF (CMAKE_WORDS_BIGENDIAN)
 	SET(ADM_BIG_ENDIAN 1)
 ENDIF (CMAKE_WORDS_BIGENDIAN)
 
-MESSAGE("")
\ No newline at end of file
+MESSAGE("")
