--- a/plugins/ADM_videoEncoder/ADM_vidEnc_x264/x264Options.cpp
+++ b/plugins/ADM_videoEncoder/ADM_vidEnc_x264/x264Options.cpp
@@ -417,7 +417,7 @@ void x264Options::setCabac(bool cabac)
 	_param.b_cabac = cabac;
 }
 
-#if X264_BUILD > 101
+#if X264_BUILD > 101 && X264_BUILD < 115
 unsigned int x264Options::getOpenGopMode(void)
 {
 #if X264_BUILD > 114
@@ -449,6 +449,16 @@ void x264Options::setOpenGopMode(unsigne
 		_param.i_open_gop = openGopMode;
 #endif
 }
+#elif X264_BUILD > 114
+unsigned int x264Options::getOpenGopMode(void)
+{
+	return _param.b_open_gop;
+}
+void x264Options::setOpenGopMode(unsigned int openGopMode)
+{
+	if (openGopMode < 3)
+		_param.b_open_gop = openGopMode;
+}
 #endif
 
 unsigned int x264Options::getInterlaced(void)
