diff -ur libxspf_1.2.0/examples/read/read.cpp xspf/examples/read/read.cpp
--- libxspf_1.2.0/examples/read/read.cpp	2009-02-21 15:13:25.000000000 -0500
+++ xspf/examples/read/read.cpp	2012-12-29 07:58:06.750151917 -0500
@@ -56,6 +56,7 @@
 #  define PORT_GETCWD _getcwd
 # endif
 #else
+# include <unistd.h>
 # define PORT_PATH_MAX PATH_MAX
 # define DIR_SEPERATOR '/'
 # define uriFilenameToUriString uriUnixFilenameToUriStringA
