Author: Daniel Schaal <farbing@web.de>
Description: Set a multi-arch library directory.
Last-Update: 2012-03-30
Forwarded: no
Bug-Debian: http://bugs.debian.org/636042

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -20,10 +20,10 @@ endif (_HAVE_VISIBILITY AND NOT WIN32)
 #
 ### lib64/32 support kde like
 #
-set(LIB_SUFFIX "" CACHE STRING "Define suffix for lib directory (32/64)")
+# set(LIB_SUFFIX "" CACHE STRING "Define suffix for lib directory (32/64)")
 
 set(BIN_INSTALL_DIR ${CMAKE_INSTALL_PREFIX}/bin)
-set(LIB_INSTALL_DIR ${CMAKE_INSTALL_PREFIX}/lib${LIB_SUFFIX})
+set(LIB_INSTALL_DIR ${CMAKE_INSTALL_PREFIX}/lib/${CMAKE_LIBRARY_ARCHITECTURE})
 
 set(INSTALL_TARGETS_DEFAULT_ARGS  RUNTIME DESTINATION "${BIN_INSTALL_DIR}"
                                   LIBRARY DESTINATION "${LIB_INSTALL_DIR}"
