Description: Fix test suite for Ruby 1.8.7
Origin: https://github.com/thumblemonks/riot/commit/a782fb4474bfc240c651ed9fd1ba58a4b8699801
Author: Justin Knowlden <gus@gusg.us>
Applied-Upstream: yes
Reviewed-by: Cédric Boutillier <boutil@debian.org>
Last-Update: 2013-03-07

--- a/test/core/reports/dot_matrix_reporter_test.rb
+++ b/test/core/reports/dot_matrix_reporter_test.rb
@@ -44,7 +44,13 @@
     asserts_topic('puts the full context + assertion name').matches('whatever asserts bang')
     asserts_topic('puts the exception message').matches('BOOM')
     # <file path>:<one or more number><two newlines><anything till end of line><newline> is the last thing in the stack trace
-    asserts_topic('puts the filtered exception backtrace').matches(/#{__FILE__}:\d+:[^\n]*\n\n.*$\n\z/)
+    asserts_topic('puts the filtered exception backtrace').matches do
+      if RUBY_VERSION =~ /^1.8.\d+/
+        /#{__FILE__}:\d+\n\n.*$\n\z/
+      else
+        /#{__FILE__}:\d+:[^\n]*\n\n.*$\n\z/
+      end
+    end
   end
 end
 
