Description: add gdm3 path to X abstraction. (Upstream r2035)
Author: Intrigeri <intrigeri@debian.org>
Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=660079

Index: apparmor-debian/profiles/apparmor.d/abstractions/X
===================================================================
--- apparmor-debian.orig/profiles/apparmor.d/abstractions/X	2012-04-24 11:03:46.506994000 -0700
+++ apparmor-debian/profiles/apparmor.d/abstractions/X	2012-04-24 12:56:42.255783258 -0700
@@ -17,7 +17,7 @@
 
   # .Xauthority files required for X connections, per user
   @{HOME}/.Xauthority           r,
-  owner /{,var/}run/gdm/*/database r,
+  owner /{,var/}run/gdm{,3}/*/database r,
   owner /{,var/}run/lightdm/authority/[0-9]* r,
 
   # the unix socket to use to connect to the display
