Description: include IceWeasel in FireFox abstraction. (Upstream r2036)
Author: Intrigeri <intrigeri@debian.org>
Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=661176

Index: apparmor-debian/profiles/apparmor.d/abstractions/ubuntu-browsers
===================================================================
--- apparmor-debian.orig/profiles/apparmor.d/abstractions/ubuntu-browsers	2012-04-24 11:03:46.506994000 -0700
+++ apparmor-debian/profiles/apparmor.d/abstractions/ubuntu-browsers	2012-04-24 13:01:22.499517948 -0700
@@ -29,8 +29,8 @@
 
   # this should cover all firefox browsers and versions (including shiretoko
   # and abrowser)
-  /usr/bin/firefox Cxr -> sanitized_helper,
-  /usr/lib/firefox*/firefox*.sh Cx -> sanitized_helper,
+  /usr/bin/{firefox,iceweasel} Cxr -> sanitized_helper,
+  /usr/lib/{firefox*,iceweasel}/{firefox*.sh,iceweasel} Cx -> sanitized_helper,
 
   # some unpackaged, but popular browsers
   /usr/lib/icecat-*/icecat Cx -> sanitized_helper,
