Description: handle missing network mediation support.
Author: John Johansen <john.johansen@canonical.com>
Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=679597

Index: apparmor-debian/parser/parser_main.c
===================================================================
--- apparmor-debian.orig/parser/parser_main.c	2012-06-28 22:17:27.433492000 -0700
+++ apparmor-debian/parser/parser_main.c	2012-07-16 11:51:23.742020178 -0700
@@ -1182,7 +1182,10 @@
 		write_cache = 0;
 		skip_read_cache = 1;
 		return;
-	}
+	} else if (strstr(flags_string, "network"))
+		kernel_supports_network = 1;
+	else
+		kernel_supports_network = 0;
 
 	/*
          * Deal with cache directory versioning:
