Description: pass CPPFLAGS into build. (Upstream r2037)
Author: Kees Cook <kees@debian.org>

Index: apparmor-debian/libraries/libapparmor/src/Makefile.am
===================================================================
--- apparmor-debian.orig/libraries/libapparmor/src/Makefile.am	2012-06-29 00:04:24.308135919 -0700
+++ apparmor-debian/libraries/libapparmor/src/Makefile.am	2012-06-29 00:04:27.000000000 -0700
@@ -3,7 +3,8 @@
 BUILT_SOURCES = grammar.h scanner.h af_protos.h
 AM_LFLAGS = -v
 AM_YFLAGS = -d -p aalogparse_
-AM_CFLAGS = @CFLAGS@ -D_GNU_SOURCE -Wall
+AM_CFLAGS = -Wall
+AM_CPPFLAGS = -D_GNU_SOURCE
 scanner.h: scanner.l
 	$(LEX) -v $<
 
Index: apparmor-debian/libraries/libapparmor/swig/perl/Makefile.PL.in
===================================================================
--- apparmor-debian.orig/libraries/libapparmor/swig/perl/Makefile.PL.in	2012-06-29 00:04:24.296135745 -0700
+++ apparmor-debian/libraries/libapparmor/swig/perl/Makefile.PL.in	2012-06-29 00:13:38.452136217 -0700
@@ -10,8 +10,9 @@
 	'FIRST_MAKEFILE' => 'Makefile.perl',
 	'ABSTRACT' => q[Perl interface to AppArmor] ,
 	'VERSION' => q[@VERSION@],
-	'INC' => q[-I@top_srcdir@/src @CFLAGS@],
+	'INC' => q[@CPPFLAGS@ -I@top_srcdir@/src @CFLAGS@],
 	'LIBS' => q[-L@top_builddir@/src/.libs/ -lapparmor @LIBS@],
 	'OBJECT' => 'libapparmor_wrap.o', # $(OBJ_EXT)
+	'dynamic_lib' => { 'OTHERLDFLAGS' => q[@LDFLAGS@], },
 ) ;
 
Index: apparmor-debian/libraries/libapparmor/m4/ac_python_devel.m4
===================================================================
--- apparmor-debian.orig/libraries/libapparmor/m4/ac_python_devel.m4	2012-06-29 00:04:24.276135455 -0700
+++ apparmor-debian/libraries/libapparmor/m4/ac_python_devel.m4	2012-06-29 00:04:27.000000000 -0700
@@ -158,6 +158,8 @@
         AC_MSG_CHECKING([consistency of all components of python development environment])
         AC_LANG_PUSH([C])
         # save current global flags
+        ac_save_LIBS="$LIBS"
+        ac_save_CPPFLAGS="$CPPFLAGS"
         LIBS="$ac_save_LIBS $PYTHON_LDFLAGS"
         CPPFLAGS="$ac_save_CPPFLAGS $PYTHON_CPPFLAGS"
         AC_TRY_LINK([
Index: apparmor-debian/libraries/libapparmor/testsuite/Makefile.am
===================================================================
--- apparmor-debian.orig/libraries/libapparmor/testsuite/Makefile.am	2012-06-29 00:04:24.288135629 -0700
+++ apparmor-debian/libraries/libapparmor/testsuite/Makefile.am	2012-06-29 00:04:27.000000000 -0700
@@ -10,8 +10,7 @@
 noinst_PROGRAMS = test_multi.multi
 
 test_multi_multi_SOURCES	= test_multi.c
-test_multi_multi_CFLAGS		= $(CFLAGS) -Wall
-test_multi_multi_LDFLAGS	= $(LDFLAGS)
+test_multi_multi_CFLAGS		= -Wall
 test_multi_multi_LDADD		= -L../src/.libs -lapparmor
 
 clean-local:
Index: apparmor-debian/changehat/pam_apparmor/Makefile
===================================================================
--- apparmor-debian.orig/changehat/pam_apparmor/Makefile	2012-06-29 00:04:24.320136093 -0700
+++ apparmor-debian/changehat/pam_apparmor/Makefile	2012-06-29 00:04:27.000000000 -0700
@@ -27,7 +27,7 @@
 endif
 
 EXTRA_CFLAGS=$(CFLAGS) -fPIC -shared -Wall -I../../libraries/libapparmor/src/
-LINK_FLAGS=-Xlinker -x -L../../libraries/libapparmor/src/.libs
+LINK_FLAGS=-Xlinker -x -L../../libraries/libapparmor/src/.libs $(LDFLAGS)
 LIBS=-lpam -lapparmor
 OBJECTS=${NAME}.o get_options.o
 
Index: apparmor-debian/parser/Makefile
===================================================================
--- apparmor-debian.orig/parser/Makefile	2012-06-29 00:04:24.332136265 -0700
+++ apparmor-debian/parser/Makefile	2012-06-29 00:04:27.000000000 -0700
@@ -85,7 +85,7 @@
 AAREDIR= libapparmor_re
 AAREOBJECT = ${AAREDIR}/libapparmor_re.a
 AAREOBJECTS = $(AAREOBJECT) libstdc++.a
-AARE_LDFLAGS=-static-libgcc -L.
+AARE_LDFLAGS=-static-libgcc -L. $(LDFLAGS)
 
 LEX_C_FILES	= parser_lex.c
 YACC_C_FILES	= parser_yacc.c parser_yacc.h
