Index: lightdm/src/xserver-local.c
===================================================================
--- lightdm.orig/src/xserver-local.c	2012-07-24 17:17:58.382128000 +1000
+++ lightdm/src/xserver-local.c	2012-07-24 17:53:18.878629448 +1000
@@ -438,6 +438,9 @@
     if (server->priv->layout)
         g_string_append_printf (command, " -layout %s", server->priv->layout);
 
+    /* We always want apport to be able to catch crashes */
+    g_string_append (command, " -core");
+
     gethostname (hostname, 1024);
     number = g_strdup_printf ("%d", xserver_get_display_number (XSERVER (server)));
     if (!server->priv->xdmcp_key)
