From: Stephane Glondu <steph@glondu.net>
Date: Fri, 21 May 2010 17:16:12 +0200
Subject: Create man dir before installing into it

Signed-off-by: Stephane Glondu <steph@glondu.net>
---
 Makefile.in |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile.in b/Makefile.in
index 07301a8..d4e835e 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -187,7 +187,7 @@ install-byte-contrib: install-byte-dot install-byte-lablgtk
 
 
 install-bin:
-	mkdir -p $(BINDIR)
+	mkdir -p $(BINDIR) $(MANDIR)/man1
 	cp -f $(BUILD)$(TOOL) $(BINDIR)/mlpost
 	cp -f mlpost.1 $(MANDIR)/man1
 
-- 
