From: Stephane Glondu <steph@glondu.net>
Date: Sat, 22 May 2010 22:29:34 +0200
Subject: myocamlmacroparser: use .byte instead of .native

Signed-off-by: Stephane Glondu <steph@glondu.net>
---
 myocamlbuild.ml.in |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/myocamlbuild.ml.in b/myocamlbuild.ml.in
index e0f659c..f60cfaf 100644
--- a/myocamlbuild.ml.in
+++ b/myocamlbuild.ml.in
@@ -47,7 +47,7 @@ let syntaxes () =
      "bitstring", cmd, cmd in
   let s = string_flatten (gen_def_string ()) in
   let compile_cmd = "@CAMLP4O@ Camlp4MacroParser.cmo " ^ s in
-  let doc_cmd = "./myocamlmacroparser.native " ^ s in
+  let doc_cmd = "./myocamlmacroparser.byte " ^ s in
   let mymacroparser_entry = "mymacroparser",compile_cmd, doc_cmd in
   let macro_entry = "macro", compile_cmd, compile_cmd in
   [ bitstring_entry ; mymacroparser_entry ; macro_entry ]
@@ -101,7 +101,7 @@ let _ = dispatch begin function
            Cmd(Sh "rm -rf doc && cp -r mlpost.docdir doc && cp -r img doc"));
       flag ["ocaml"; "compile"; "use_ocamldoc"] & S[A"-I"; A "+ocamldoc"];
       dep ["ocaml";"doc"] ["customdoc/img.cmo"];
-      dep ["ocaml"; "syntax_mymacroparser"] ["myocamlmacroparser.native"];
+      dep ["ocaml"; "syntax_mymacroparser"] ["myocamlmacroparser.byte"];
       flag ["ocaml";"doc"] &
         S([A"-hide" ; A "Mlpost"; A"-g"; A "customdoc/img.cmo"]
             @ (if "@LABLGTK2@" = "yes"
-- 
