From: Mehdi Dogguy <mehdi@debian.org>
Date: Tue, 12 Jun 2012 15:00:26 +0200
Subject: Fully qualify name of input'ed files

---
 metapost.ml |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/metapost.ml b/metapost.ml
index 71ed108..4469d51 100644
--- a/metapost.ml
+++ b/metapost.ml
@@ -25,8 +25,8 @@ let print i fmt c =
   fprintf fmt "@[beginfig(%d)@\n  @[%a@] endfig;@]@." i MPprint.command c
 
 let print_prelude ?(eps=false) s fmt () =
-  fprintf fmt "input mp-tool ; %% some initializations and auxiliary macros
-input mp-spec ; %% macros that support special features
+  fprintf fmt "input mp-tool.mpii ; %% some initializations and auxiliary macros
+input mp-spec.mpii ; %% macros that support special features
 
 %%redefinition
 def doexternalfigure (expr filename) text transformation =
-- 
