Index: cairo/src/Makefile.am
===================================================================
--- cairo.orig/src/Makefile.am	2012-02-02 01:48:14.000000000 +0100
+++ cairo/src/Makefile.am	2012-03-27 10:45:59.335499749 +0200
@@ -44,7 +44,7 @@
 	$(enabled_cairo_private) \
 	$(enabled_cairo_sources) \
 	$(NULL)
-libcairo_la_LDFLAGS = $(AM_LDFLAGS) -version-info $(CAIRO_LIBTOOL_VERSION_INFO) -no-undefined $(export_symbols)
+libcairo_la_LDFLAGS = $(AM_LDFLAGS) -version-info $(CAIRO_LIBTOOL_VERSION_INFO) -no-undefined $(export_symbols) -export-symbols-regex \^cairo_.*
 libcairo_la_LIBADD = $(CAIRO_LIBS) \
 	$(cairo_cxx_lib)
 libcairo_la_DEPENDENCIES = $(cairo_def_dependency) $(cairo_cxx_lib)
Index: cairo/util/cairo-gobject/Makefile.am
===================================================================
--- cairo.orig/util/cairo-gobject/Makefile.am	2010-08-26 13:57:19.000000000 +0200
+++ cairo/util/cairo-gobject/Makefile.am	2012-03-27 10:45:59.335499749 +0200
@@ -10,6 +10,6 @@
 	$(NULL)
 
 libcairo_gobject_la_CFLAGS = $(CAIRO_CFLAGS) $(GOBJECT_CFLAGS)
-libcairo_gobject_la_LDFLAGS = -version-info $(CAIRO_LIBTOOL_VERSION_INFO) -no-undefined $(export_symbols)
+libcairo_gobject_la_LDFLAGS = -version-info $(CAIRO_LIBTOOL_VERSION_INFO) -no-undefined $(export_symbols) -export-symbols-regex \^cairo_.*
 libcairo_gobject_la_LIBADD = $(top_builddir)/src/libcairo.la $(CAIRO_LIBS) $(GOBJECT_LIBS)
 
Index: cairo/util/cairo-script/Makefile.am
===================================================================
--- cairo.orig/util/cairo-script/Makefile.am	2012-02-02 01:48:15.000000000 +0100
+++ cairo/util/cairo-script/Makefile.am	2012-03-27 10:45:59.335499749 +0200
@@ -14,7 +14,7 @@
 	$(libcairo_script_interpreter_headers)	\
 	$(NULL)
 libcairo_script_interpreter_la_CFLAGS = $(CAIRO_CFLAGS)
-libcairo_script_interpreter_la_LDFLAGS = -version-info $(CAIRO_LIBTOOL_VERSION_INFO) -no-undefined $(export_symbols)
+libcairo_script_interpreter_la_LDFLAGS = -version-info $(CAIRO_LIBTOOL_VERSION_INFO) -no-undefined $(export_symbols) -export-symbols-regex \^cairo_.*
 libcairo_script_interpreter_la_LIBADD = $(top_builddir)/src/libcairo.la $(CAIRO_LIBS) -lz
 
 csi_replay_SOURCES = csi-replay.c
