Index: cairo/test/cairo-test-trace.c
===================================================================
--- cairo.orig/test/cairo-test-trace.c	2012-03-30 09:38:27.937300339 +0200
+++ cairo/test/cairo-test-trace.c	2012-03-30 09:48:31.833279021 +0200
@@ -541,7 +541,11 @@
 
     base = mmap (NULL, DATA_SIZE,
 		 PROT_READ | PROT_WRITE,
+#ifdef MAP_NORESERVE
 		 MAP_SHARED | MAP_NORESERVE,
+#else
+		 MAP_SHARED,
+#endif
 		 fd, 0);
     close (fd);
 
Index: cairo/util/cairo-sphinx/sphinx.c
===================================================================
--- cairo.orig/util/cairo-sphinx/sphinx.c	2012-03-30 09:38:28.269300327 +0200
+++ cairo/util/cairo-sphinx/sphinx.c	2012-03-30 09:48:00.657280122 +0200
@@ -1325,7 +1325,11 @@
 
     base = mmap (NULL, DATA_SIZE,
 		 PROT_READ | PROT_WRITE,
+#ifdef MAP_NORESERVE
 		 MAP_SHARED | MAP_NORESERVE,
+#else
+		 MAP_SHARED,
+#endif
 		 fd, 0);
     close (fd);
 
