--- gnome-vfs-mime-magic.orig	2004-01-18 00:28:59.000000000 +0100
+++ gnome-vfs-mime-magic	2004-01-18 00:29:44.000000000 +0100
@@ -133,6 +133,7 @@
 0:64	string		\<svg					image/svg+xml
 0:64	string		\<chemistry				application/x-gchempaint
 0:64	string		\<crystal				application/x-gcrystal
+0:256	string		\<display				application/x-gdesklets-display
 # php rules need to go before html
 0:64	string		\<?php					application/x-php
 0:64	string		\<smil\>				application/x-smil
