Author: Martin Pitt <mpitt@debian.org>
Description: Update tutorial README for required build dependencies.
Forwarded: No, Debian specific

Index: postgresql-9.0-9.0~beta1/src/tutorial/README
===================================================================
--- postgresql-9.0-9.0~beta1.orig/src/tutorial/README	2008-03-21 14:23:29.000000000 +0100
+++ postgresql-9.0-9.0~beta1/src/tutorial/README	2010-05-01 15:01:42.747229787 +0200
@@ -6,8 +6,7 @@
 This directory contains SQL tutorial scripts.  To look at them, first do a
 	% make
 to compile all the scripts and C files for the user-defined functions
-and types.  (make needs to be GNU make --- it may be named something
-different on your system, often gmake)
+and types. This requires postgresql-server-dev-9.1 to be installed.
 
 Then, run psql with the -s (single-step) flag:
 	% psql -s
