Index: gconf-editor-3.0.0/data/gconf-editor.desktop.in.in
===================================================================
--- gconf-editor-3.0.0.orig/data/gconf-editor.desktop.in.in	2011-06-05 00:31:37.173914134 +0200
+++ gconf-editor-3.0.0/data/gconf-editor.desktop.in.in	2011-06-05 00:31:45.209953416 +0200
@@ -6,6 +6,7 @@ Terminal=false
 Type=Application
 Icon=gconf-editor
 StartupNotify=true
+NoDisplay=true
 Categories=GNOME;GTK;System;
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=gconf-editor
