#Upstream : https://bugs.launchpad.net/screenlets/+bug/245314
#Debian : #488952 #472897
diff -ur screenlets.orig/src/bin/screenletsd screenlets/src/bin/screenletsd
--- screenlets.orig/src/bin/screenletsd	2008-06-05 22:09:02.000000000 +0200
+++ screenlets/src/bin/screenletsd	2008-06-05 22:12:00.000000000 +0200
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/bin/sh
 # This application is released under the GNU General Public License
 # v3 (or, at your option, any later version). You can find the full
 # text of the license under [url]http://www.gnu.org/licenses/gpl.txt[/url].
@@ -17,10 +17,10 @@
 runDaemon()
 {
 if [ -z "`ps -f -C python | grep screenlets-daemon`" ]; then            #Check if Daemon is running
-        echo "Loading Daemon"           #Load Daemon
+        printf "Loading Daemon"           #Load Daemon
         $PREFIX/share/screenlets-manager/screenlets-daemon.py > /dev/null &
 else
-        echo "Daemon is already running"
+        printf "Daemon is already running"
 fi
 }
 
@@ -32,7 +32,7 @@
             if [ -d $l/$d ]; then
                 screenletpy="$l/$d/${d}Screenlet.py"
                 if [ -e $screenletpy ]; then
-                    echo -e $d
+                    printf "$d\n"
                 fi
             fi
         done
@@ -56,7 +56,7 @@
     fi
 done
 if [ -z $location ]; then
-    echo -e "There is no screenlet named $screenlet"
+    printf "There is no screenlet named $screenlet"
     exit 1
 fi
 }
@@ -64,11 +64,11 @@
 auScreenlet()
 {
 AU_DIR="$HOME/.config/autostart"                #Autostart folder
-echo -e "[Desktop Entry]" > $AU_DIR/${1}Screenlet.desktop              #Make a .desktop entry to autostart the screenlet
-echo -e "Name=${1}Screenlet" >> $AU_DIR/${1}Screenlet.desktop
-echo -e "Version=1.0" >> $AU_DIR/${1}Screenlet.desktop
-echo -e "Exec=python -u $location > /dev/null" >> $AU_DIR/${1}Screenlet.desktop
-echo -e "X-GNOME-Autostart-enabled=true" >> $AU_DIR/${1}Screenlet.desktop
+printf "[Desktop Entry]\n" > $AU_DIR/${1}Screenlet.desktop              #Make a .desktop entry to autostart the screenlet
+printf "Name=${1}Screenlet\n" >> $AU_DIR/${1}Screenlet.desktop
+printf "Version=1.0\n" >> $AU_DIR/${1}Screenlet.desktop
+printf "Exec=python -u $location > /dev/null\n" >> $AU_DIR/${1}Screenlet.desktop
+printf "X-GNOME-Autostart-enabled=true\n" >> $AU_DIR/${1}Screenlet.desktop
 }
 
 runScreenlet()
@@ -78,30 +78,30 @@
 
 help_me()                       #Help
 {
-echo -e "  Options:"
-echo -e "\t --gui \t\t\t Open up the Graphical Interface"
-echo -e "\t --cli \t\t\t Use the Command Line interface"
-echo -e "  Cli Options (only use one at a time):"
-echo -e "\t    run \t\t\t Run the Screenlet"
-echo -e "\t    au \t\t\t Add the Screenlet to your autostart list"
-echo -e "\t    list \t\t\t List all installed screenlets"
+printf "  Options:\n"
+printf "\t --gui \t\t\t Open up the Graphical Interface\n"
+printf "\t --cli \t\t\t Use the Command Line interface\n"
+printf "  Cli Options (only use one at a time):\n"
+printf "\t    run \t\t\t Run the Screenlet\n"
+printf "\t    au \t\t\t Add the Screenlet to your autostart list\n"
+printf "\t    list \t\t\t List all installed screenlets\n"
 }
 
-if [ x$1 == x"--gui" ]; then            #Decide what to do
+if [ x$1 = x"--gui" ]; then            #Decide what to do
         runDaemon
-    MANAGER_OPTIONS=`echo $@ | sed -e s/$1//`
+    MANAGER_OPTIONS=`printf $@ | sed -e s/$1//`
         startgui $MANAGER_OPTIONS
         exit 0
-elif [ x$1 == x"--cli" ]; then
-        if [ x$2 == x"run" ]; then
+elif [ x$1 = x"--cli" ]; then
+        if [ x$2 = x"run" ]; then
                 findScreenlet $3
                 runScreenlet
                 exit 0
-        elif [ x$2 == x"au" ]; then
+        elif [ x$2 = x"au" ]; then
                 findScreenlet $3
                 auScreenlet $3
                 exit 0
-    elif [ x$2 == x"list" ]; then
+    elif [ x$2 = x"list" ]; then
         listScreenlets
         exit 0
         else
Seulement dans screenlets/src/bin: screenletsd~
