#Upstream: https://bugs.launchpad.net/screenlets/+bug/245311
diff -ur screenlets.orig/src/share/screenlets/ACPIBattery/ACPIBatteryScreenlet.py screenlets/src/share/screenlets/ACPIBattery/ACPIBatteryScreenlet.py
--- screenlets.orig/src/share/screenlets/ACPIBattery/ACPIBatteryScreenlet.py	2008-06-05 22:09:02.000000000 +0200
+++ screenlets/src/share/screenlets/ACPIBattery/ACPIBatteryScreenlet.py	2008-06-06 17:24:10.000000000 +0200
@@ -103,7 +103,11 @@
 				self.__dict__['update_interval'] = 1
 		elif name == "file_auto":
 			if value:
-				dirs=listdir('/proc/acpi/battery/');
+				try:
+					dirs=listdir('/proc/acpi/battery/');
+				except OSError:
+					dirs=[];
+					pass
 				dirs.sort();
 				try:
 					self.__dict__['file_path']='/proc/acpi/battery/'+dirs[0]+'/'
Seulement dans screenlets/src/share/screenlets/ACPIBattery: ACPIBatteryScreenlet.py~

