#upstream: https://bugs.launchpad.net/screenlets/+bug/245312
diff -ur screenlets.orig/src/share/screenlets/WallpaperClock/WallpaperClockScreenlet.py screenlets/src/share/screenlets/WallpaperClock/WallpaperClockScreenlet.py
--- screenlets.orig/src/share/screenlets/WallpaperClock/WallpaperClockScreenlet.py	2008-06-05 22:09:02.000000000 +0200
+++ screenlets/src/share/screenlets/WallpaperClock/WallpaperClockScreenlet.py	2008-06-06 17:29:36.000000000 +0200
@@ -78,12 +78,12 @@
 	p_layout = None
 	home = getoutput("echo $HOME")
 	xfce_workarround = False
-	wall_sel = os.listdir(os.environ['HOME'] + '/.screenlets/WallpaperClock/' + 'wallpapers')
-	imagepath = os.environ['HOME'] + '/.screenlets/WallpaperClock/'
 	if not os.path.exists(os.environ['HOME'] + '/.screenlets/WallpaperClock'):
 		os.system('mkdir ' + os.environ['HOME'] + '/.screenlets/WallpaperClock')
 	if not os.path.exists(os.environ['HOME'] + '/.screenlets/WallpaperClock/wallpapers'):
 		os.system('mkdir ' + os.environ['HOME'] + '/.screenlets/WallpaperClock/wallpapers')
+	wall_sel = os.listdir(os.environ['HOME'] + '/.screenlets/WallpaperClock/' + 'wallpapers')
+	imagepath = os.environ['HOME'] + '/.screenlets/WallpaperClock/'
 	# --------------------------------------------------------------------------
 	# constructor and internals
 	# --------------------------------------------------------------------------
Seulement dans screenlets/src/share/screenlets/WallpaperClock: WallpaperClockScreenlet.py~
