#Upstream : https://bugs.launchpad.net/screenlets/+bug/256329
diff -Nur -x '*.orig' -x '*~' screenlets-0.1.2/src/lib/menu.py screenlets-0.1.2.new/src/lib/menu.py
--- screenlets-0.1.2/src/lib/menu.py	2008-06-04 14:31:25.000000000 +0200
+++ screenlets-0.1.2.new/src/lib/menu.py	2008-10-21 21:18:17.000000000 +0200
@@ -321,9 +321,8 @@
 		super(ImageMenuItem, self).__init__(stock)
 		
 		# get the label and image for later
-		children = self.get_children()
-		self.label = children[0]
-		self.image = children[1]
+		self.label = self.get_child()
+		self.image = self.get_image()
 		
 		# set the label to custom text
 		if label is not None:
