#Upstream : https://bugs.launchpad.net/ubuntu/+source/screenlets/+bug/269256

Index: trunk/src/lib/plugins/mpdclient2.py
===================================================================
--- trunk.orig/src/lib/plugins/mpdclient2.py	2009-04-29 23:26:50.000000000 +0200
+++ trunk/src/lib/plugins/mpdclient2.py	2009-04-29 23:46:35.000000000 +0200
@@ -325,9 +325,10 @@
         password = kw_password
     if kw_host:
         host = kw_host
-
-    conn = mpd_connection(host, port)
-    if password:
-        conn.password(password)
-    return conn
-
+    try:
+        conn = mpd_connection(host, port)
+        if password:
+            conn.password(password)
+        return conn
+    except:
+        return False
