#Upstream https://bugs.launchpad.net/ubuntu/+source/screenlets/+bug/264809

Index: trunk/src/share/screenlets/ClearWeather/ClearWeatherScreenlet.py
===================================================================
--- trunk.orig/src/share/screenlets/ClearWeather/ClearWeatherScreenlet.py	2009-04-28 01:03:15.000000000 +0200
+++ trunk/src/share/screenlets/ClearWeather/ClearWeatherScreenlet.py	2009-04-28 01:04:54.000000000 +0200
@@ -11,6 +11,7 @@
 
 import re
 from urllib import urlopen
+import socket # for socket.error
 import screenlets
 from screenlets.options import StringOption, BoolOption, ColorOption, FontOption
 import pygtk
@@ -164,8 +165,9 @@
 				dcstop = data.find('</day>',dcstart)   #####10-day forecast
 				day = data[dcstart:dcstop]
 				forecast.append(self.tokenizeForecast(day))
-		except:
-			pass
+		except (IOError, socket.error), e:
+			print "Error retrieving weather data", e
+			self.show_error(("Error retrieving weather data", e))
 
 		return forecast
 
@@ -186,8 +188,9 @@
 				dcstop = data.find('</hour>',dcstart)   ####hourly forecast
 				hour = data[dcstart:dcstop]
 				hforecast.append(self.tokenizeForecastHourly(hour))
-		except IOError, e:
+		except (IOError, socket.error), e:
 			print "Error retrieving weather data", e
+			self.show_error(("Error retrieving weather data", e))
 
 		return hforecast
 
@@ -513,7 +516,7 @@
 
 		
 			if y >= 75 and x <= 132 and x >= 110:
-				os.system('xdg-open weather.com')
+				os.system('xdg-open http://weather.com')
 
 	def on_draw_shape(self,ctx):
 		if self.theme:
@@ -553,13 +556,15 @@
 
 
 
-	def show_error(self):
+	def show_error(self, reason=None):
 
 		dialog = gtk.Dialog("Zip Code", self.window)
 		dialog.resize(300, 100)
 		dialog.add_buttons(gtk.STOCK_OK, gtk.RESPONSE_OK)
 
-		label = gtk.Label("Could not reach weather.com.  Check your internet connection and location and try again.")
+		reasonstr = "\nReason: %s" % reason if reason is not None else ""
+
+		label = gtk.Label("Could not reach weather.com.  Check your internet connection and location and try again."+reasonstr)
 		dialog.vbox.add(label)
 		check = gtk.CheckButton("Do not show this again")
 		dialog.vbox.add(check)
