#Upstream : https://bugs.launchpad.net/ubuntu/+source/screenlets/+bug/269256

Index: trunk/src/share/screenlets/NowPlaying/NowPlayingScreenlet.py
===================================================================
--- trunk.orig/src/share/screenlets/NowPlaying/NowPlayingScreenlet.py	2009-04-29 23:20:25.000000000 +0200
+++ trunk/src/share/screenlets/NowPlaying/NowPlayingScreenlet.py	2009-04-29 23:21:47.000000000 +0200
@@ -58,7 +58,8 @@
 PLAYER_LIST = {'Rhythmbox':'RhythmboxAPI',
                     'Listen':'ListenAPI',
                     'Banshee':'BansheeAPI',
-                    'Amarok':'AmarokAPI',
+#TODO Port to Amarok 2.X API
+#                    'Amarok':'AmarokAPI',
                     'Exaile':'ExaileAPI',
                     'Sonata':'SonataAPI',
                     'Kaffeine':'KaffeineAPI',
@@ -68,7 +69,7 @@
                     'LastFMProxy':'LastFMProxyAPI',
                     'Songbird':'SongbirdAPI',}
 
-PLAYER_LIST_LAUNCH = ['amarok','audacious','banshee','exaile','juk','kaffeine','listen','quodlibet','rhythmbox','sonata','songbird']
+PLAYER_LIST_LAUNCH = ['audacious','banshee','exaile','juk','kaffeine','listen','quodlibet','rhythmbox','sonata','songbird']
 # The Screenlet
 class NowPlayingScreenlet(screenlets.Screenlet):
 	"""Shows Song Info"""
