#Upstream : https://bugs.edge.launchpad.net/screenlets/+bug/369565

Index: trunk/src/lib/plugins/Rhythmbox.py
===================================================================
--- trunk.orig/src/lib/plugins/Rhythmbox.py	2009-04-29 23:58:39.000000000 +0200
+++ trunk/src/lib/plugins/Rhythmbox.py	2009-04-29 23:58:45.000000000 +0200
@@ -74,8 +74,12 @@
 
 
 	def is_playing(self):
-		if self.playerAPI.getPlaying() == 1: return True
-		else: return False
+		try:
+			test_playing = self.playerAPI.getPlaying()
+			if self.playerAPI.getPlaying() == 1: return True
+			else: return False
+		except DBusException:
+			return False
 
 	def play_pause(self):
 		if self.is_playing:
