Description: Makes Makefile compliant with Debian and implements hardening.
Author: Joao Eriberto Mota Filho <eriberto@eriberto.pro.br>
Last-Update: 2013-04-11
--- a/Makefile
+++ b/Makefile
@@ -1,6 +1,6 @@
 SHELL = /bin/sh
 
-prefix = /usr
+prefix = $(DESTDIR)/usr
 exec_prefix = ${prefix}
 bindir = ${exec_prefix}/bin
 mandir = ${prefix}/share/man/man1
@@ -9,6 +9,10 @@
 DEBUG = -g
 CFLAGS ?= -O2
 CFLAGS += $(DEBUG)
+CPPFLAGS:=$(shell dpkg-buildflags --get CPPFLAGS)
+CFLAGS:=$(shell dpkg-buildflags --get CFLAGS)
+LDFLAGS:=$(shell dpkg-buildflags --get LDFLAGS)
+
 SRC = src
 DOC = doc
 
@@ -19,10 +23,10 @@
 all: bittwist bittwiste
 
 bittwist:
-	$(CC) $(CFLAGS) $(SRC)/bittwist.c -o $(SRC)/bittwist -I/usr/local/include -L/usr/local/lib -lpcap
+	$(CC) $(CFLAGS) $(CPPFLAGS) $(LDFLAGS) $(SRC)/bittwist.c -o $(SRC)/bittwist -I/usr/local/include -L/usr/local/lib -lpcap
 
 bittwiste:
-	$(CC) $(CFLAGS) $(SRC)/bittwiste.c -o $(SRC)/bittwiste -I/usr/local/include -L/usr/local/lib -lpcap
+	$(CC) $(CFLAGS) $(CPPFLAGS) $(LDFLAGS) $(SRC)/bittwiste.c -o $(SRC)/bittwiste -I/usr/local/include -L/usr/local/lib -lpcap
 
 clean:
 	rm -f $(SRC)/bittwist $(SRC)/bittwiste
