--- a/scripts/xdg-open.in
+++ b/scripts/xdg-open.in
@@ -139,7 +139,8 @@ DEBUG 3 "$xdg_user_dir:$xdg_system_dirs"
                 command="`grep -E "^Exec(\[[^]=]*])?=" "$file" | cut -d= -f 2- | first_word`"
                 command_exec=`which $command 2>/dev/null`
                 arguments="`grep -E "^Exec(\[[^]=]*])?=" "$file" | cut -d= -f 2- | last_word`"
-                arguments_exec="`echo $arguments | sed -e 's*%[fFuU]*"'"$1"'"*g'`"
+                local sed_escaped_url="$(echo "$1" | sed -e 's/[&\\]/\\&/g')"
+                arguments_exec="`echo $arguments | sed -e 's*%[fFuU]*"'"$sed_escaped_url"'"*g'`"
                 if [ -x "$command_exec" ] ; then
                     if echo $arguments | grep -iq '%[fFuU]' ; then
                         eval $command_exec $arguments_exec
