--- a/scripts/xdg-open.in
+++ b/scripts/xdg-open.in
@@ -144,7 +144,7 @@ DEBUG 3 "$xdg_user_dir:$xdg_system_dirs"
                     if echo $arguments | grep -iq '%[fFuU]' ; then
                         eval $command_exec $arguments_exec
                     else
-                        eval $command_exec $arguments_exec "$1"
+                        eval $command_exec $arguments_exec '"$1"'
                     fi
 
                     if [ $? -eq 0 ]; then
