Description: Fix support for multi-word $BROWSER commands.
Author: Per Olofsson <pelle@debian.org>
Bug-Debian: http://bugs.debian.org/612339
--- a/scripts/xdg-open.in
+++ b/scripts/xdg-open.in
@@ -206,8 +206,10 @@ open_generic()
 
     open_generic_xdg_x_scheme_handler "$1"
 
+    OLDIFS="$IFS"
     IFS=":"
     for browser in $BROWSER; do
+        IFS="$OLDIFS"
         if [ x"$browser" != x"" ]; then
 
             browser_with_arg=`printf "$browser" "$1" 2>/dev/null`
