Description: Add $MAILER support to xdg-email.
Bug: https://bugs.freedesktop.org/show_bug.cgi?id=6615
Bug-Debian: http://bugs.debian.org/574131
Author: Per Olofsson <pelle@debian.org>

--- a/scripts/xdg-email.in
+++ b/scripts/xdg-email.in
@@ -183,6 +183,23 @@ open_xfce()
     fi
 }
 
+open_envvar()
+{
+    local OLDIFS="$IFS"
+    IFS=:
+    for i in $MAILER; do
+	IFS="$OLDIFS"
+
+	eval "$i" '"$1"'
+
+	if [ $? -eq 0 ]; then
+	    exit_success
+	fi
+    done
+
+    exit_failure_operation_failed
+}
+
 open_generic()
 {
     IFS=":"
@@ -376,6 +393,10 @@ if [ x"$DE" = x"" ]; then
     DE=generic
 fi
 
+if [ x"$MAILER" != x"" ]; then
+    DE=envvar
+fi
+
 # if BROWSER variable is not set, check some well known browsers instead
 if [ x"$BROWSER" = x"" ]; then
     BROWSER=links2:elinks:links:lynx:w3m
@@ -385,6 +406,10 @@ if [ x"$BROWSER" = x"" ]; then
 fi
 
 case "$DE" in
+    envvar)
+    open_envvar "${mailto}"
+    ;;
+
     kde)
     open_kde "${mailto}"
     ;;
