--- aotcompile.py-old	2007-11-26 11:18:40.000000000 +0100
+++ aotcompile.py	2007-11-26 11:19:19.000000000 +0100
@@ -31,23 +31,8 @@
 GCJFLAGS = ["-g", "-O2", "-fPIC", "-findirect-dispatch", "-fjni"]
 LDFLAGS = ["-Wl,-Bsymbolic"]
 
-MAX_CLASSES_PER_JAR = 1024
-MAX_BYTES_PER_JAR = 1048576
-
-# Calculate the maximum jar size/maximum classes per jar.
-try:
-    for line in file('/proc/meminfo'):
-        if line.startswith('MemTotal:'):
-            memtotal = int(line.split()[1])
-            break
-    if memtotal < 2200000:
-        MAX_CLASSES_PER_JAR = 512
-        MAX_BYTES_PER_JAR = 524288
-    if memtotal < 1100000:
-        MAX_CLASSES_PER_JAR = 256
-        MAX_BYTES_PER_JAR = 262144
-except:
-    pass
+MAX_CLASSES_PER_JAR = 256
+MAX_BYTES_PER_JAR = 262144
 
 MAKEFILE = "Makefile"
 
