Index: gnome-panel-3.4.2.1/gnome-panel/panel-menu-button.c
===================================================================
--- gnome-panel-3.4.2.1.orig/gnome-panel/panel-menu-button.c	2012-03-15 09:56:50.000000000 +0100
+++ gnome-panel-3.4.2.1/gnome-panel/panel-menu-button.c	2012-05-21 19:22:50.861191796 +0200
@@ -73,7 +73,7 @@ typedef struct {
 } MenuPathRootItem;
 
 static MenuPathRootItem root_items [] = {
-	{ APPLICATIONS_MENU, "applications", "applications.menu" },
+	{ APPLICATIONS_MENU, "applications", "gnome-applications.menu" },
 	{ GNOMECC_MENU,      "gnomecc",      "gnomecc.menu"      }
 };
 
Index: gnome-panel-3.4.2.1/gnome-panel/panel-addto.c
===================================================================
--- gnome-panel-3.4.2.1.orig/gnome-panel/panel-addto.c	2012-03-15 09:56:50.000000000 +0100
+++ gnome-panel-3.4.2.1/gnome-panel/panel-addto.c	2012-05-21 19:30:16.079367262 +0200
@@ -674,7 +674,7 @@ panel_addto_make_application_model (Pane
 
 	if (tree != NULL && (root = gmenu_tree_get_root_directory (tree))) {
 		panel_addto_make_application_list (&dialog->application_list,
-						   root, "applications.menu",
+						   root, "gnome-applications.menu",
 						   PANEL_ADDTO_MENU_SHOW_ALL);
 		panel_addto_populate_application_model (store, NULL, dialog->application_list);
 
