Debian #524117
GNOME #513347

Index: gnome-panel-3.4.1/applets/wncklet/window-list.c
===================================================================
--- gnome-panel-3.4.1.orig/applets/wncklet/window-list.c	2012-03-15 09:56:49.000000000 +0100
+++ gnome-panel-3.4.1/applets/wncklet/window-list.c	2012-04-19 17:47:50.400533701 +0200
@@ -8,6 +8,8 @@
  *
  */
 
+#define WNCK_I_KNOW_THIS_IS_NOT_UPSTREAM 1
+
 #ifdef HAVE_CONFIG_H
 #include <config.h>
 #endif
@@ -112,7 +114,7 @@
 		return;
   
 	tasklist->orientation = new_orient;
-
+        wnck_tasklist_set_orientation (tasklist->tasklist, new_orient);
 	tasklist_update (tasklist);
 }
 
@@ -387,6 +389,8 @@
 
 	tasklist->tasklist = wnck_tasklist_new ();
 
+        wnck_tasklist_set_orientation (tasklist->tasklist, tasklist->orientation);
+
         wnck_tasklist_set_icon_loader (WNCK_TASKLIST (tasklist->tasklist),
                                        icon_loader_func,
                                        tasklist,
