Index: gnome-panel-3.0.0.1/gnome-panel/main.c
===================================================================
--- gnome-panel-3.0.0.1.orig/gnome-panel/main.c	2011-04-16 22:20:07.643885619 +0200
+++ gnome-panel-3.0.0.1/gnome-panel/main.c	2011-04-16 22:27:00.681904861 +0200
@@ -36,9 +36,11 @@ GSList *panels = NULL;
 GSList *panel_list = NULL;
 
 static gboolean  replace = FALSE;
+static gint      screen = 0;
 
 static const GOptionEntry options[] = {
   { "replace", 0, 0, G_OPTION_ARG_NONE, &replace, N_("Replace a currently running panel"), NULL },
+  { "screen", 0, G_OPTION_FLAG_HIDDEN, G_OPTION_ARG_INT, &screen, "Ignored - for compatibility", "num"},
   { NULL }
 };
 
