Index: gnome-panel-3.4.2.1/gnome-panel/panel-menu-items.c
===================================================================
--- gnome-panel-3.4.2.1.orig/gnome-panel/panel-menu-items.c	2012-05-21 17:42:15.807995862 +0200
+++ gnome-panel-3.4.2.1/gnome-panel/panel-menu-items.c	2012-05-22 07:51:59.548107179 +0200
@@ -1377,11 +1377,6 @@
 		add_menu_separator (desktop_menu);
 #endif
 
-	item = panel_menu_item_desktop_new ("gnome-online-accounts-panel.desktop",
-					    NULL, FALSE);
-	if (item)
-		gtk_menu_shell_append (GTK_MENU_SHELL (desktop_menu), item);
-
 	/* Do not force the string like in gnome-shell, but just use the one
 	 * from the .desktop file */
 	item = panel_menu_item_desktop_new ("gnome-control-center.desktop",
