Author: Steffen Moeller <moeller@debian.org>
Description: Increase of buffers describing client state.

Index: boinc/client/client_state.cpp
===================================================================
--- boinc.orig/client/client_state.cpp	2012-05-03 13:53:07.275804022 +0200
+++ boinc/client/client_state.cpp	2012-05-03 13:53:25.599553075 +0200
@@ -293,7 +293,7 @@
 int CLIENT_STATE::init() {
     int retval;
     unsigned int i;
-    char buf[256];
+    char buf[2048];
     PROJECT* p;
 
     srand((unsigned int)time(0));
@@ -1059,7 +1059,7 @@
         }
 
         if (!strcmp(file_ref.open_name, GRAPHICS_APP_FILENAME)) {
-            char relpath[512], path[512];
+            char relpath[1024], path[2048];
             get_pathname(fip, relpath, sizeof(relpath));
             relative_to_absolute(relpath, path);
             strlcpy(avp->graphics_exec_path, path, sizeof(avp->graphics_exec_path));
