Author: Steffen Moeller <moeller@debian.org>
Description: Adding now truly missing const flags to icons.

Index: boinc/clientgui/res/linuxicon.xpm
===================================================================
--- boinc.orig/clientgui/res/linuxicon.xpm	2011-07-24 19:39:09.000000000 +0200
+++ boinc/clientgui/res/linuxicon.xpm	2011-11-13 19:53:15.000000000 +0100
@@ -1,5 +1,5 @@
 /* XPM */
-static char *Linux_xpm[] = {
+static const char *Linux_xpm[] = {
 /* columns rows colors chars-per-pixel */
 "14 14 17 1",
 ". c #8F8979",
Index: boinc/clientgui/res/macosicon.xpm
===================================================================
--- boinc.orig/clientgui/res/macosicon.xpm	2011-07-24 19:39:09.000000000 +0200
+++ boinc/clientgui/res/macosicon.xpm	2011-11-13 19:54:13.000000000 +0100
@@ -1,5 +1,5 @@
 /* XPM */
-static char *MacOS_xpm[] = {
+static const char *MacOS_xpm[] = {
 /* columns rows colors chars-per-pixel */
 "14 14 14 1",
 "= c #295A9C",
Index: boinc/clientgui/res/windowsicon.xpm
===================================================================
--- boinc.orig/clientgui/res/windowsicon.xpm	2011-07-24 19:39:09.000000000 +0200
+++ boinc/clientgui/res/windowsicon.xpm	2011-11-13 19:56:04.000000000 +0100
@@ -1,5 +1,5 @@
 /* XPM */
-static char *Win32_xpm[] = {
+static const char *Win32_xpm[] = {
 /* columns rows colors chars-per-pixel */
 "14 14 23 1",
 "4 c #6CC0EE",
