Author: Steffen Moeller <moeller@debian.org>
Description: Improving initalisation of app_init_data

Index: boinc/lib/app_ipc.cpp
===================================================================
--- boinc.orig/lib/app_ipc.cpp	2012-05-06 14:00:20.000000000 +0200
+++ boinc/lib/app_ipc.cpp	2012-05-06 14:04:04.956192691 +0200
@@ -137,7 +137,7 @@
 }
 
 int write_init_data_file(FILE* f, APP_INIT_DATA& ai) {
-    char buf[2048];
+    char buf[9192];
     fprintf(f,
         "<app_init_data>\n"
         "<major_version>%d</major_version>\n"
@@ -469,7 +469,7 @@
     const char *virtual_name, char *physical_name, int len
 ) {
     FILE *fp;
-    char buf[512], *p;
+    char buf[4096], *p;
 
     if (!virtual_name) return ERR_NULL;
     strlcpy(physical_name, virtual_name, len);
@@ -502,7 +502,7 @@
 // same, std::string version
 //
 int boinc_resolve_filename_s(const char *virtual_name, string& physical_name) {
-    char buf[512], *p;
+    char buf[4096], *p;
     if (!virtual_name) return ERR_NULL;
     physical_name = virtual_name;
 #ifndef _WIN32
@@ -520,7 +520,7 @@
 }
 
 void url_to_project_dir(char* url, char* dir) {
-    char buf[256];
+    char buf[4096];
     escape_project_url(url, buf);
     sprintf(dir, "%s/%s", PROJECT_DIR, buf);
 }
Index: boinc/lib/app_ipc.h
===================================================================
--- boinc.orig/lib/app_ipc.h	2012-05-03 17:05:19.000000000 +0200
+++ boinc/lib/app_ipc.h	2012-05-06 14:04:04.960192637 +0200
@@ -193,7 +193,7 @@
 
     // info for GPU apps
     //
-    char gpu_type[64];
+    char gpu_type[512];
     int gpu_device_num;
     int gpu_opencl_dev_index;
 
