Author: Steffen Moeller <steffen.moeller@uk-sh.de>
Description: Tested compatibility with clang.
--- a/lib/Makefile.am
+++ b/lib/Makefile.am
@@ -147,13 +147,13 @@
 libboinc_la_CFLAGS = $(AM_CFLAGS) $(PICFLAGS) $(PTHREAD_CFLAGS)
 libboinc_la_CXXFLAGS = $(AM_CXXFLAGS) $(PICFLAGS) $(PTHREAD_CFLAGS)
 libboinc_la_LDFLAGS = -L$(libdir) -rpath $(libdir) -static -version-number $(LIBBOINC_VERSION)
-libboinc_la_LIBADD =
+libboinc_la_LIBADD = -lm -lstdc++
 
 libboinc_crypt_la_SOURCES = crypt.cpp
 libboinc_crypt_la_CFLAGS = $(AM_CFLAGS) $(PICFLAGS) $(PTHREAD_CFLAGS) $(SSL_CFLAGS)
 libboinc_crypt_la_CXXFLAGS = $(AM_CXXFLAGS) $(PICFLAGS) $(PTHREAD_CFLAGS) $(SSL_CXXFLAGS)
 libboinc_crypt_la_LDFLAGS = -L$(libdir) -rpath $(libdir) -static -version-number $(LIBBOINC_VERSION)
-libboinc_crypt_la_LIBADD = 
+libboinc_crypt_la_LIBADD =  -lm -lstdc++
 
 
 if ENABLE_FCGI
--- a/sched/make_work.cpp
+++ b/sched/make_work.cpp
@@ -176,9 +176,6 @@
     char keypath[MAXPATHLEN];
     char buf[BLOB_SIZE];
     R_RSA_PRIVATE_KEY key;
-    int nwu_names = wu_names.size();
-    DB_WORKUNIT wus[nwu_names];
-    int i;
     static int index=0;
 
     retval = config.parse_file();
@@ -193,8 +190,11 @@
         exit(1);
     }
 
-    for (i=0; i<nwu_names; i++) {
-        DB_WORKUNIT& wu = wus[i];
+    const int nwu_names = wu_names.size();
+    DB_WORKUNIT *wus = new DB_WORKUNIT[nwu_names]();
+
+    for (int i=0; i<nwu_names; i++) {
+	DB_WORKUNIT& wu=wus[i];
         sprintf(buf, "where name='%s'", wu_names[i].c_str());
         retval = wu.lookup(buf);
         if (retval) {
@@ -265,6 +265,8 @@
 
         wait_for_results(new_wu_id);
     }
+
+    delete [] wus;
 }
 
 void usage(char *name) {
