Author: Steffen Moeller <moeller@debian.org>
Description: Extension of path lengths

Index: boinc/client/acct_mgr.h
===================================================================
--- boinc.orig/client/acct_mgr.h	2012-03-27 14:02:21.716769660 +0200
+++ boinc/client/acct_mgr.h	2012-03-30 18:31:27.906016323 +0200
@@ -49,7 +49,7 @@
         // use of cookies are required during initial signup
         // NOTE: This bool gets dropped after the client has
         //   successfully attached to an account manager
-    char cookie_failure_url[256];
+    char cookie_failure_url[1024];
         // if the cookies could not be detected, provide a
         // link to a website to go to so the user can find
         // what login name and password they have been assigned
@@ -120,7 +120,7 @@
     std::vector<AM_ACCOUNT> accounts;
     double repeat_sec;
     char* global_prefs_xml;
-    char host_venue[256];
+    char host_venue[1024];
     bool got_rss_feeds;
     std::vector<RSS_FEED>rss_feeds;
 
Index: boinc/client/app.h
===================================================================
--- boinc.orig/client/app.h	2012-03-27 14:02:21.728769496 +0200
+++ boinc/client/app.h	2012-03-30 18:30:24.810879369 +0200
@@ -110,9 +110,9 @@
     double bytes_sent;
         // reported by the app if it does network I/O
     double bytes_received;
-    char slot_dir[256];
+    char slot_dir[1024];
         // directory where process runs (relative)
-    char slot_path[512];
+    char slot_path[1024];
         // same, absolute
         // This is used only to run graphics apps
         // (that way don't have to worry about top-level dirs
@@ -151,8 +151,8 @@
         // but not descendants of the main process
         // (e.g. VMs created by vboxwrapper)
         // These are communicated via the app_status message channel
-    char web_graphics_url[256];
-    char remote_desktop_addr[256];
+    char web_graphics_url[1024];
+    char remote_desktop_addr[1024];
     ASYNC_COPY* async_copy;
 
     void set_task_state(int, const char*);
Index: boinc/client/client_state.h
===================================================================
--- boinc.orig/client/client_state.h	2012-03-27 14:02:21.764769005 +0200
+++ boinc/client/client_state.h	2012-03-30 18:27:02.037653176 +0200
@@ -125,16 +125,16 @@
     bool in_abort_sequence;
         // Determine when it is safe to leave the quit_client() handler
         // and to finish cleaning up.
-    char detach_project_url[256];
+    char detach_project_url[1024];
         // stores URL for -detach_project option
-    char reset_project_url[256];
+    char reset_project_url[1024];
         // stores URL for -reset_project option
-    char update_prefs_url[256];
+    char update_prefs_url[1024];
         // stores URL for -update_prefs option
-    char main_host_venue[256];
+    char main_host_venue[1024];
         // venue from project or AMS that gave us general prefs
-    char attach_project_url[256];
-    char attach_project_auth[256];
+    char attach_project_url[1024];
+    char attach_project_auth[1024];
     bool exit_before_upload;
         // exit when about to upload a file
 #ifndef _WIN32
Index: boinc/client/cs_notice.h
===================================================================
--- boinc.orig/client/cs_notice.h	2012-03-27 14:02:21.800768513 +0200
+++ boinc/client/cs_notice.h	2012-03-30 18:27:51.556975760 +0200
@@ -86,9 +86,9 @@
 extern NOTICES notices;
 
 struct RSS_FEED {
-    char url[256];
-    char url_base[256];
-    char project_name[256];
+    char url[1024];
+    char url_base[1024];
+    char project_name[1024];
     double poll_interval;
     double next_poll_time;
     bool found;
Index: boinc/client/file_xfer.h
===================================================================
--- boinc.orig/client/file_xfer.h	2012-03-27 14:02:21.840767966 +0200
+++ boinc/client/file_xfer.h	2012-03-30 18:25:44.078719674 +0200
@@ -33,8 +33,8 @@
 class FILE_XFER : public HTTP_OP {
 public:
     FILE_INFO* fip;
-    char pathname[256];
-    char header[4096];
+    char pathname[2048];
+    char header[8192];
     bool file_size_query;
     bool is_upload;
     double starting_size;
Index: boinc/client/http_curl.h
===================================================================
--- boinc.orig/client/http_curl.h	2012-03-27 14:02:21.868767583 +0200
+++ boinc/client/http_curl.h	2012-03-30 18:23:36.392465752 +0200
@@ -57,15 +57,15 @@
     PROXY_INFO pi;
     PROJECT* project;   // associated project, if any
 
-	char m_url[256];  
-	char m_curl_ca_bundle_location[256];
+	char m_url[1024];  
+	char m_curl_ca_bundle_location[1024];
         // string needed for ssl support
-	char m_curl_user_credentials[128];
+	char m_curl_user_credentials[1024];
         // string needed for proxy username/password
 
     int content_length;
     unsigned int trace_id;
-    char request_header[4096];
+    char request_header[8192];
 
     FILE* fileIn;
 	FILE* fileOut;
@@ -85,7 +85,7 @@
     //
     char* req1;
         // if not NULL, send this string first
-    char infile[256];
+    char infile[1024];
         // then (is nonempty) this file
     double file_offset;
         // starting at this offset
@@ -93,7 +93,7 @@
     // reply message stuff
     //
     // reply is always written to a file
-    char outfile[256];
+    char outfile[1024];
         // if specified, it's written to this file w/ optional offset 
         // otherwise it's written to a temp file
     // if type POST2, copy output to req1 buffer
@@ -102,7 +102,7 @@
 
 	long lSeek;
         // offset within the file or memory buffer we're reading,
-    char error_msg[256];
+    char error_msg[1024];
         // put Curl error message here
 	bool bTempOutfile;
         // CMC -- flag that outfile is really a tempfile we should delete
Index: boinc/client/scheduler_op.h
===================================================================
--- boinc.orig/client/scheduler_op.h	2012-03-27 14:02:21.924766818 +0200
+++ boinc/client/scheduler_op.h	2012-03-30 18:19:03.480197856 +0200
@@ -104,8 +104,8 @@
     char* project_prefs_xml;
         // not including <project_preferences> tags
         // may include <venue> elements
-    char master_url[256];
-    char host_venue[256];
+    char master_url[1024];
+    char host_venue[1024];
     unsigned int user_create_time;
     std::vector<APP> apps;
     std::vector<FILE_INFO> file_infos;
