--- a/api/gutil.cpp
+++ b/api/gutil.cpp
@@ -24,7 +24,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
-#include <math.h>
+#include <cmath>
 #include <setjmp.h>
 
 #ifdef _WIN32
--- a/api/gutil_text.cpp
+++ b/api/gutil_text.cpp
@@ -26,7 +26,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
-#include <math.h>
+#include <cmath>
 #include <setjmp.h>
 
 #ifdef _WIN32
--- a/client/client_state.cpp
+++ b/client/client_state.cpp
@@ -26,7 +26,7 @@
 #include <ctime>
 #include <cstdarg>
 #include <cstring>
-#include <math.h>
+#include <cmath>
 #if HAVE_SYS_SOCKET_H
 #include <sys/socket.h>
 #endif
--- a/client/sim_util.cpp
+++ b/client/sim_util.cpp
@@ -35,7 +35,7 @@
 #define finite _finite
 #endif
 
-#include <math.h>
+#include <cmath>
 
 #include "error_numbers.h"
 #include "filesys.h"
--- a/client/whetstone.cpp
+++ b/client/whetstone.cpp
@@ -29,7 +29,7 @@
 #include <stdlib.h>
 #include <stdio.h>
 #include <string.h>
-#include <math.h>
+#include <cmath>
 #include <time.h>
 #endif
 
--- a/clientscr/ss_app.cpp
+++ b/clientscr/ss_app.cpp
@@ -22,7 +22,7 @@
 #ifdef _WIN32
 #include "boinc_win.h"
 #else
-#include <math.h>
+#include <cmath>
 #endif
 #include <string>
 #include <vector>
--- a/db/boinc_db.cpp
+++ b/db/boinc_db.cpp
@@ -22,7 +22,7 @@
 #include <ctime>
 #include <unistd.h>
 #include <cmath>
-#include <math.h>
+#include <cmath>
 
 // For machines with finite() defined in ieeefp.h
 #if HAVE_IEEEFP_H
--- a/lib/boinc_win.h
+++ b/lib/boinc_win.h
@@ -197,7 +197,7 @@
 #include <assert.h>
 #include <ctype.h>
 #include <errno.h>
-#include <math.h>
+#include <cmath>
 #include <setjmp.h>
 #include <signal.h>
 #include <stdarg.h>
--- a/lib/md5.c
+++ b/lib/md5.c
@@ -88,7 +88,7 @@
  * For reference, here is the program that computed the T values.
  */
 #if 0
-#include <math.h>
+#include <cmath>
 main()
 {
     int i;
--- a/lib/parse.cpp
+++ b/lib/parse.cpp
@@ -31,7 +31,7 @@
 #include <cstring>
 #include <cstdlib>
 #include <string>
-#include <math.h>
+#include <cmath>
 #include <ctype.h>
 #include <errno.h>
 #if HAVE_IEEEFP_H
--- a/lib/str_util.cpp
+++ b/lib/str_util.cpp
@@ -27,7 +27,7 @@
 #ifndef _WIN32
 #include "config.h"
 #include <string>
-#include <math.h>
+#include <cmath>
 #include <string.h>
 #include <stdlib.h>
 #include <ctype.h>
--- a/lib/util.cpp
+++ b/lib/util.cpp
@@ -50,7 +50,7 @@
 #include <errno.h>
 #include <string>
 #include <cstring>
-#include <math.h>
+#include <cmath>
 #if HAVE_IEEEFP_H
 #include <ieeefp.h>
 extern "C" {
--- a/samples/atiopencl/atiopencl.hpp
+++ b/samples/atiopencl/atiopencl.hpp
@@ -28,7 +28,7 @@
 #endif
 
 #include <string.h>
-#include <math.h>
+#include <cmath>
 #include <time.h>
 #include <cstdlib>
 #include <iostream>
--- a/samples/example_app/uc2_graphics.cpp
+++ b/samples/example_app/uc2_graphics.cpp
@@ -33,7 +33,7 @@
 #ifdef _WIN32
 #include "boinc_win.h"
 #else
-#include <math.h>
+#include <cmath>
 #endif
 
 #include "parse.h"
--- a/samples/nvcuda/cuda_kernel.cu
+++ b/samples/nvcuda/cuda_kernel.cu
@@ -25,7 +25,7 @@
 // identity matrix and the product A * incerse(A). For debugging...
 //#define VERIFY 1
 #include <stdio.h>
-#include <math.h>
+#include <cmath>
 #include <time.h>
 #include "cuda_config.h"
 
--- a/samples/nvopencl/nvopencl.hpp
+++ b/samples/nvopencl/nvopencl.hpp
@@ -24,7 +24,7 @@
 #include <CL/cl.h>
 #include <oclUtils.h>
 #include <string.h>
-#include <math.h>
+#include <cmath>
 #include <time.h>
 #include <cstdlib>
 #include <iostream>
--- a/samples/vboxwrapper/vboxwrapper.cpp
+++ b/samples/vboxwrapper/vboxwrapper.cpp
@@ -56,7 +56,7 @@
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <stdio.h>
-#include <math.h>
+#include <cmath>
 #include <string>
 #include <unistd.h>
 #endif
--- a/samples/wrappture/fermi.cpp
+++ b/samples/wrappture/fermi.cpp
@@ -2,7 +2,7 @@
 
 #include <stdlib.h>
 #include <stdio.h>
-#include <math.h>
+#include <cmath>
 #include <unistd.h>
 
 int main(int argc, char * argv[]) {
--- a/samples/wrappture/wrappture_example.cpp
+++ b/samples/wrappture/wrappture_example.cpp
@@ -15,7 +15,7 @@
 
 #include <stdlib.h>
 #include <stdio.h>
-#include <math.h>
+#include <cmath>
 #ifndef _WIN32
 #include <unistd.h>
 #endif
--- a/vda/vda_lib.cpp
+++ b/vda/vda_lib.cpp
@@ -19,7 +19,7 @@
 
 #include <algorithm>
 #include <limits.h>
-#include <math.h>
+#include <cmath>
 #include <set>
 #include <stdio.h>
 #include <vector>
