--- a/Makefile.incl
+++ b/Makefile.incl
@@ -38,22 +38,22 @@
 # dependencies to make sure libs gets compiled before
 # programs linking to them:
 
-LIBSCHED = $(top_builddir)/sched/libsched.la
+LIBSCHED = -lsched
 $(LIBSCHED):
 	cd $(top_builddir)/sched; ${MAKE} libsched.la
-LIBSCHED_FCGI = $(top_builddir)/sched/libsched_fcgi.la
+LIBSCHED_FCGI = -lsched_fcgi
 $(LIBSCHED_FCGI):
 	cd $(top_builddir)/sched; ${MAKE} libsched_fcgi.la
-LIBBOINC = $(top_builddir)/lib/libboinc.la
+LIBBOINC = -lboinc
 $(LIBBOINC):
 	cd $(top_builddir)/lib; ${MAKE} libboinc.la
-LIBBOINC_CRYPT = $(top_builddir)/lib/libboinc_crypt.la
+LIBBOINC_CRYPT = -lboinc_crypt
 $(LIBBOINC_CRYPT):
 	cd $(top_builddir)/lib; ${MAKE} libboinc.la
-LIBBOINC_FCGI = $(top_builddir)/lib/libboinc_fcgi.la
+LIBBOINC_FCGI = -lboinc_fcgi
 $(LIBBOINC_FCGI):
 	cd $(top_builddir)/lib; ${MAKE} libboinc_fcgi.la
-LIBAPI = $(top_builddir)/api/libboinc_api.la
+LIBAPI = -lboinc_api
 $(LIBAPI):
 	cd $(top_builddir)/api; ${MAKE} libboinc_api.la
 
@@ -61,3 +61,4 @@
 SERVERLIBS_FCGI = $(LIBSCHED_FCGI) $(LIBBOINC_CRYPT) $(LIBBOINC_FCGI) -lfcgi $(MYSQL_LIBS) $(PTHREAD_LIBS) $(RSA_LIBS) $(SSL_LIBS)
 APPLIBS = $(LIBAPI) $(LIBBOINC)
 
+.PHONY: $(LIBBOINC) $(LIBBOINC_CRYPT) $(LIBBOINC_FCGI) $(LIBAPI) $(LIBSCHED) $(LIBSCHED_FCGI)
