Author: Steffen Moeller <moeller@debian.org>
Description: Helping verbosity of error messages.

--- a/lib/filesys.cpp
+++ b/lib/filesys.cpp
@@ -144,7 +144,10 @@
     dirp->handle = INVALID_HANDLE_VALUE;
 #else
     dirp = opendir(p);
-    if (!dirp) return NULL;
+    if (!dirp) {
+        fprintf(stderr,"dir_open: Could not open directory '%s'.\n",p);
+        return NULL;
+    }
 #endif
     return dirp;
 }
