Author: Frank Thomas
Description: We do not build the screensaver.

Index: boinc/Makefile.am
===================================================================
--- boinc.orig/Makefile.am	2011-10-08 14:11:56.000000000 +0200
+++ boinc/Makefile.am	2011-10-08 14:32:51.000000000 +0200
@@ -22,9 +22,9 @@
 
 if ENABLE_MANAGER
    CLIENTGUI_SUBDIRS = clientgui clientgui/res locale
-if BUILD_X11_SCREENSAVER
-   CLIENTGUI_SUBDIRS += clientscr
-endif
+#if BUILD_X11_SCREENSAVER
+#   CLIENTGUI_SUBDIRS += clientscr
+#endif
 endif
 
 # ORDER MATTERS below.  One must build dependencies FIRST, then things
