--- a/api/graphics2_util.cpp
+++ b/api/graphics2_util.cpp
@@ -37,7 +37,7 @@
 static key_t get_shmem_name(const char* prog_name) {
     char cwd[MAXPATHLEN], path[MAXPATHLEN];
     boinc_getcwd(cwd);
-    sprintf(path, "%s/init_data.xml", cwd);
+    snprintf(path, sizeof(path), "%s/init_data.xml", cwd);
     return ftok(path, 2);
 }
 #else
@@ -47,7 +47,7 @@
     APP_INIT_DATA aid;
     int retval = boinc_get_init_data(aid);
     if (retval) aid.slot = 0;
-    sprintf(shmem_name, "boinc_%s_%d", prog_name, aid.slot);
+    snprintf(shmem_name, MAXPATHLEN, "boinc_%s_%d", prog_name, aid.slot);
 }
 #endif
 
--- a/lib/filesys.cpp
+++ b/lib/filesys.cpp
@@ -433,7 +433,7 @@
     while (1) {
         retval = dir_scan(filename, dirp, sizeof(filename));
         if (retval) break;
-        sprintf(subdir, "%s/%s", dirpath, filename);
+        snprintf(subdir, sizeof(subdir), "%s/%s", dirpath, filename);
 
         if (is_dir(subdir)) {
             if (recurse) {
@@ -551,8 +551,8 @@
     }
     return 0;
 #elif defined(__EMX__)
-    char cmd[2*MAXPATHLEN];
-    sprintf(cmd, "copy \"%s\" \"%s\"", orig, newf);
+    char cmd[2*MAXPATHLEN+5];
+    snprintf(cmd, sizeof(cmd), "copy \"%s\" \"%s\"", orig, newf);
     return system(cmd);
 #else
     // POSIX requires that shells run from an application will use the 
@@ -683,7 +683,7 @@
         p = strchr(q, '/');
         if (!p) break;
         *p = 0;
-        sprintf(newpath, "%s/%s", oldpath, q);
+        snprintf(newpath, sizeof(newpath), "%s/%s", oldpath, q);
         retval = boinc_mkdir(newpath);
         if (retval) return retval;
         strcpy(oldpath, newpath);
--- a/lib/network.cpp
+++ b/lib/network.cpp
@@ -71,7 +71,7 @@
     case WSAENOTSOCK:
         return "not a socket";
     }
-    sprintf(buf, "error %d", e);
+    snprintf(buf, sizeof(buf), "error %d", e);
     return buf;
 #else
     switch (h_errno) {
@@ -85,11 +85,11 @@
         return "host not found or server failure";
 #ifdef NETDB_INTERNAL
     case NETDB_INTERNAL:
-        sprintf(buf,"network internal error %d",errno);
+        snprintf(buf, sizeof(buf), "network internal error %d",errno);
         return buf;
 #endif
     }
-    sprintf(buf, "error %d", h_errno);
+    snprintf(buf, sizeof(buf), "error %d", h_errno);
     return buf;
 #endif
 }
@@ -107,7 +107,7 @@
         case AF_INET6: {
             sockaddr_in6* sin = (sockaddr_in6*)&s;
             char buf[256];
-            inet_ntop(AF_INET6, (void*)(&sin->sin6_addr), buf, 256);
+            inet_ntop(AF_INET6, (void*)(&sin->sin6_addr), buf, sizeof(buf));
             return (strcmp(buf, "::1") == 0);
             break;
         }
