Index: boinc/client/coproc_detect.cpp
===================================================================
--- boinc.orig/client/coproc_detect.cpp	2012-05-03 14:01:32.344886723 +0200
+++ boinc/client/coproc_detect.cpp	2012-05-03 14:01:48.736662216 +0200
@@ -1024,21 +1024,33 @@
         return;
     }
 
+#ifndef __cuda_cuda_h__
     retval = (*__cuInit)(0);
+#else
+    retval = cuInit(0);
+#endif
     if (retval) {
         sprintf(buf, "NVIDIA drivers present but no GPUs found");
         warnings.push_back(buf);
         return;
     }
 
+#ifndef __cuda_cuda_h__
     retval = (*__cuDriverGetVersion)(&cuda_version);
+#else
+    retval = cuDriverGetVersion(&cuda_version);
+#endif
     if (retval) {
         sprintf(buf, "cuDriverGetVersion() returned %d", retval);
         warnings.push_back(buf);
         return;
     }
 
+#ifndef __cuda_cuda_h__
     retval = (*__cuDeviceGetCount)(&cuda_ndevs);
+#else
+    retval = cuDeviceGetCount(&cuda_ndevs);
+#endif
     if (retval) {
         sprintf(buf, "cuDeviceGetCount() returned %d", retval);
         warnings.push_back(buf);
@@ -1061,14 +1073,23 @@
  nvidia_gpus.push_back(cc);
 }
 #endif
+
+#ifndef __cuda_cuda_h__
         retval = (*__cuDeviceGet)(&device, j);
+#else
+        retval = cuDeviceGet(&device, j);
+#endif
         if (retval) {
             sprintf(buf, "cuDeviceGet(%d) returned %d", j, retval);
             warnings.push_back(buf);
             return;
         }
         cc.prop.deviceHandle = device;
-        (*__cuDeviceGetName)(cc.prop.name, 256, device);
+#ifndef __cuda_cuda_h__
+        retval = (*__cuDeviceGetName)(cc.prop.name, 256, device);
+#else
+        retval = cuDeviceGetName(cc.prop.name, 256, device);
+#endif
         if (retval) {
             sprintf(buf, "cuDeviceGetName(%d) returned %d", j, retval);
             warnings.push_back(buf);
@@ -1195,7 +1216,11 @@
     void* ctx;
     
     available_ram = prop.totalGlobalMem;
+#ifndef __cuda_cuda_h__
     retval = (*__cuDeviceGet)(&device, device_num);
+#else
+    retval = cuDeviceGet(&device, device_num);
+#endif
     if (retval) {
         if (log_flags.coproc_debug) {
             msg_printf(0, MSG_INFO,
@@ -1206,7 +1231,11 @@
     }
 
     CUcontext ctx; // no immediate better idea
+#ifndef __cuda_cuda_h__
     retval = (*__cuCtxCreate)(&ctx, 0, device);
+#else
+    retval = cuCtxCreate(&ctx, 0, device);
+#endif
     if (retval) {
         if (log_flags.coproc_debug) {
             msg_printf(0, MSG_INFO,
@@ -1217,17 +1246,29 @@
     }
 
     size_t memfree, memtotal;
+#ifndef __cuda_cuda_h__
     retval = (*__cuMemGetInfo)(&memfree, &memtotal);
+#else
+    retval = cuMemGetInfo(&memfree, &memtotal);
+#endif
     if (retval) {
         if (log_flags.coproc_debug) {
             msg_printf(0, MSG_INFO,
                 "[coproc] cuMemGetInfo(%d) returned %d", device_num, retval
             );
         }
+#ifndef __cuda_cuda_h__
         (*__cuCtxDestroy)(ctx);
+#else
+        cuCtxDestroy(ctx);
+#endif
         return;
     }
+#ifndef __cuda_cuda_h__
     (*__cuCtxDestroy)(ctx);
+#else
+        cuCtxDestroy(ctx);
+#endif
     available_ram = (double) memfree;
 }
 
