Author: Steffen Moeller
Description: Increased buffers.

Index: boinc/client/coproc_detect.cpp
===================================================================
--- boinc.orig/client/coproc_detect.cpp	2012-05-03 13:38:04.292171877 +0200
+++ boinc/client/coproc_detect.cpp	2012-05-03 13:53:32.499458579 +0200
@@ -77,7 +77,7 @@
     vector<int>& ignore_ati_dev
 ) {
     unsigned int i;
-    char buf[256], buf2[256];
+    char buf[1024], buf2[1024];
 
 #ifdef _WIN32
     try {
@@ -865,7 +865,7 @@
     vector<int>& ignore_devs
 ) {
     int cuda_ndevs, retval;
-    char buf[256];
+    char buf[1024];
 
 #ifdef _WIN32
     HMODULE cudalib = LoadLibrary("nvcuda.dll");
Index: boinc/client/scheduler_op.cpp
===================================================================
--- boinc.orig/client/scheduler_op.cpp	2012-05-03 13:38:04.376170725 +0200
+++ boinc/client/scheduler_op.cpp	2012-05-03 13:53:32.503458525 +0200
@@ -228,7 +228,7 @@
 //
 int SCHEDULER_OP::start_rpc(PROJECT* p) {
     int retval;
-    char request_file[1024], reply_file[1024], buf[256];
+    char request_file[1024], reply_file[1024], buf[2048];
     const char *trickle_up_msg;
 
     safe_strcpy(scheduler_url, p->get_scheduler_url(url_index, url_random));
@@ -241,7 +241,7 @@
         } else {
             trickle_up_msg = "";
         }
-        request_string(buf);
+        request_string(buf); // no size check of buf in request_string
         if (strlen(buf)) {
             if (p->nresults_returned) {
                 msg_printf(p, MSG_INFO,
@@ -323,7 +323,7 @@
 // parse a master file.
 //
 int SCHEDULER_OP::parse_master_file(PROJECT* p, vector<std::string> &urls) {
-    char buf[256], buf2[256];
+    char buf[1024], buf2[1024];
     char master_filename[256];
     std::string str;
     FILE* f;
@@ -336,7 +336,7 @@
         return ERR_FOPEN;
     }
     p->scheduler_urls.clear();
-    while (fgets(buf, 256, f)) {
+    while (fgets(buf, sizeof(buf), f)) {
 
         // allow for the possibility of > 1 tag per line here
         // (UMTS may collapse lines)
@@ -579,7 +579,7 @@
 // Others are copied straight to the PROJECT
 //
 int SCHEDULER_REPLY::parse(FILE* in, PROJECT* project) {
-    char buf[256], msg_buf[1024], pri_buf[256], attr_buf[256];
+    char buf[2048], msg_buf[4096], pri_buf[2048], attr_buf[2048];
     int retval;
     MIOFILE mf;
     XML_PARSER xp(&mf);
