--- a/lib/filesys.cpp
+++ b/lib/filesys.cpp
@@ -585,9 +585,9 @@
     fclose(src);
     fclose(dst);
     // Copy file's ownership, permissions to the extent we are allowed
-    lstat(orig, &sbuf);             // Get source file's info
-    chown(newf, sbuf.st_uid, sbuf.st_gid);
-    chmod(newf, sbuf.st_mode);
+    if (lstat(orig, &sbuf)) retval=ERR_FWRITE;             // Get source file's info
+    if (chown(newf, sbuf.st_uid, sbuf.st_gid)) retval=ERR_FWRITE;
+    if (chmod(newf, sbuf.st_mode)) retval=ERR_FWRITE;
     return retval;
 #endif
 }
--- a/lib/diagnostics.cpp
+++ b/lib/diagnostics.cpp
@@ -74,14 +74,14 @@
 
 static int         diagnostics_initialized = false;
 static int         flags;
-static char        stdout_log[256];
-static char        stdout_archive[256];
+static char        stdout_log[MAXPATHLEN];
+static char        stdout_archive[MAXPATHLEN];
 static FILE*       stdout_file;
-static char        stderr_log[256];
-static char        stderr_archive[256];
+static char        stderr_log[MAXPATHLEN];
+static char        stderr_archive[MAXPATHLEN];
 static FILE*       stderr_file;
-static char        boinc_dir[256];
-static char        boinc_install_dir[256];
+static char        boinc_dir[MAXPATHLEN];
+static char        boinc_install_dir[MAXPATHLEN];
 static int         boinc_proxy_enabled;
 static char        boinc_proxy[256];
 static char        symstore[256];
@@ -305,7 +305,7 @@
 
     // Store various pieces of inforation for future use.
     if (flags & BOINC_DIAG_BOINCAPPLICATION) {
-        char    buf[256];
+        char    buf[MAXPATHLEN];
         char    proxy_address[256];
         int     proxy_port;
         MIOFILE mf;
@@ -330,13 +330,13 @@
 			mf.init_file(p);
 			while(mf.fgets(buf, sizeof(buf))) {
 				if (match_tag(buf, "</app_init_data>")) break;
-				else if (parse_str(buf, "<boinc_dir>", boinc_dir, 256)) continue;
-				else if (parse_str(buf, "<symstore>", symstore, 256)) continue;
+				else if (parse_str(buf, "<boinc_dir>", boinc_dir, sizeof(boinc_dir))) continue;
+				else if (parse_str(buf, "<symstore>", symstore, sizeof(symstore))) continue;
 				else if (match_tag(buf, "<use_http_proxy/>")) {
 					boinc_proxy_enabled = true;
 					continue;
 				}
-				else if (parse_str(buf, "<http_server_name>", proxy_address, 256)) continue;
+				else if (parse_str(buf, "<http_server_name>", proxy_address, sizeof(proxy_address))) continue;
 				else if (parse_int(buf, "<http_server_port>", proxy_port)) continue;
 			}
 			fclose(p);
--- a/client/sandbox.cpp
+++ b/client/sandbox.cpp
@@ -207,7 +207,7 @@
         strcpy(filename, "");
         retval = dir_scan(filename, dirp, sizeof(filename));
         if (retval) break;
-        sprintf(path, "%s/%s", dirpath,  filename);
+        snprintf(path, sizeof(path), "%s/%s", dirpath,  filename);
         if (is_dir(path)) {
             retval = client_clean_out_dir(path, NULL);
             if (retval) final_retval = retval;
