Author: Steffen Moeller <moeller@debian.org>
Description: Improved memory handling.

--- a/lib/parse.cpp
+++ b/lib/parse.cpp
@@ -178,7 +178,12 @@
         int n = (int)strlen(line);
         if (nused + n >= bufsize) {
             bufsize *= 2;
-            buf = (char*)realloc(buf, bufsize);
+            char *b = buf;
+            buf = (char*)realloc(b, bufsize);
+            if (!buf) {
+                free(b); // b was buf and was allocated in this fun
+                return ERR_XML_PARSE;
+            }
         }
         strcpy(buf+nused, line);
         nused += n;
--- a/lib/procinfo_win.cpp
+++ b/lib/procinfo_win.cpp
@@ -25,6 +25,9 @@
     tNTQSI   pNTQSI = (tNTQSI)GetProcAddress(hNTDllLib, "NtQuerySystemInformation");
     ULONG    cbBuffer = 0;
 
+    if (!ppBuffer) return ERR_NULL;
+    if (!pcbBuffer) return ERR_NULL;
+
     while (1) {
         // Store the buffer size since it appears that somebody is monkeying around
         //   with the return values on some systems.
@@ -55,9 +58,9 @@
         } else if (!NT_SUCCESS(Status)) {
             HeapFree(hHeap, NULL, *ppBuffer);
             return ERR_GETRUSAGE;
-		} else {
-			return 0;
-		}
+        } else {
+            return 0;
+        }
     }
     return 0;	// never reached
 }
