Author: Steffen Moeller <moeller@debian.org>
Description: Increased buffers and eliminated memory sink.
--- a/sched/sched_driver.cpp
+++ b/sched/sched_driver.cpp
@@ -47,6 +47,8 @@
 
 #include "util.h"
 #include "svn_version.h"
+#include "str_replace.h"  // strlcpy
+#include "str_util.h"     // safe_strcpy
 
 using std::vector;
 
@@ -61,7 +63,7 @@
 double max_time = 1;
 
 void read_hosts() {
-    char buf[256], buf2[256];
+    char buf[1024], buf2[1024];
     host_descs.clear();
     FILE* f = fopen("host_descs.txt", "r");
     if (!f) {
@@ -72,15 +74,15 @@
         HOST_DESC hd;
         strcpy(buf2, buf);
         char* p1 = strtok(buf2, "\t\n");
-        strcpy(hd.os_name, p1);
+        safe_strcpy(hd.os_name, p1);
         char* p2 = strtok(0, "\t\n");
-        strcpy(hd.p_vendor, p2);
+        safe_strcpy(hd.p_vendor, p2);
         char* p3 = strtok(0, "\t\n");
         if (!p3) {
             fprintf(stderr, "bad line: %s\n", buf);
             exit(1);
         }
-        strcpy(hd.p_model, p3);
+        safe_strcpy(hd.p_model, p3);
         host_descs.push_back(hd);
     }
     fclose(f);
