Author: Steffen Moeller <moeller@debian.org>
Forwarded: yes (PM to David)

Index: boinc_upstream/Makefile.am
===================================================================
--- boinc_upstream.orig/Makefile.am
+++ boinc_upstream/Makefile.am
@@ -13,7 +13,7 @@
 endif
 
 if ENABLE_SERVER
-   SERVER_SUBDIRS = db test py sched apps tools vda
+   SERVER_SUBDIRS = db test py sched apps tools vda html
 ## once contained samples/example_app which breaks out-of-source-tree builds
 endif
 
Index: boinc_upstream/html/Makefile.am
===================================================================
--- /dev/null
+++ boinc_upstream/html/Makefile.am
@@ -0,0 +1,5 @@
+
+htmldir=$(DESTDIR)/$(prefix)/share/boinc-server-maker/html
+install-data-local:
+	mkdir -p $(htmldir)
+	cp -r bt  drupal  inc  languages  ops  project.sample  queue  user $(htmldir)
Index: boinc_upstream/sched/Makefile.am
===================================================================
--- boinc_upstream.orig/sched/Makefile.am
+++ boinc_upstream/sched/Makefile.am
@@ -87,11 +87,12 @@
 
 if ENABLE_SERVER
 
-cgidir = $(libexecdir)/cgi-bin
-scheddir = $(libexecdir)/sched
+schedcgidir = $(prefix)/lib/boinc-server-maker/sched
+schedsharedir = $(prefix)/lib/boinc-server-maker/sched
+schedbindir = $(prefix)/lib/boinc-server-maker/sched
 
-bin_PROGRAMS = \
-	adjust_user_priority \
+schedbin_PROGRAMS = \
+    adjust_user_priority \
     delete_file \
     get_file \
     make_work \
@@ -100,9 +101,9 @@
     show_shmem \
     wu_check
 
-sched_PROGRAMS = \
+schedshare_PROGRAMS = \
     census \
-	credit_test \
+    credit_test \
     db_dump \
     db_purge \
     feeder \
@@ -120,12 +121,18 @@
     trickle_echo \
     update_stats
 
-cgi_PROGRAMS= \
+schedcgi_PROGRAMS= \
     cgi \
     file_upload_handler
 
+schedshare_DATA = \
+    db_dump_spec.xml \
+    assimilator.py \
+    pymw_assimilator.py \
+    transitioner_catchup.php
+
 # scripts that 'make install' should put in bindir
-bin_SCRIPTS = start stop status
+schedshare_SCRIPTS = start stop status
 
 noinst_HEADERS = \
     assimilate_handler.h \
@@ -276,8 +283,7 @@
 
 if ENABLE_FCGI
 
-cgi_PROGRAMS += fcgi \
-                fcgi_file_upload_handler
+schedcgi_PROGRAMS += fcgi fcgi_file_upload_handler
 
 fcgi_SOURCES = $(cgi_sources)
 fcgi_CPPFLAGS = -D_USING_FCGI_ $(AM_CPPFLAGS)
Index: boinc_upstream/configure.ac
===================================================================
--- boinc_upstream.orig/configure.ac
+++ boinc_upstream/configure.ac
@@ -975,6 +975,7 @@
                  db/Makefile
                  doc/Makefile
                  doc/manpages/Makefile
+                 html/Makefile
                  lib/Makefile
                  locale/Makefile
                  Makefile
Index: boinc_upstream/db/Makefile.am
===================================================================
--- boinc_upstream.orig/db/Makefile.am
+++ boinc_upstream/db/Makefile.am
@@ -9,4 +9,5 @@
 endif
 
 ## other stuff to go into the distribution
-EXTRA_DIST = constraints.sql schema.sql init_db
+extradir = $(prefix)/share/boinc-server-maker/db
+dist_extra_DATA = constraints.sql schema.sql init_db
Index: boinc_upstream/tools/Makefile.am
===================================================================
--- boinc_upstream.orig/tools/Makefile.am
+++ boinc_upstream/tools/Makefile.am
@@ -1,11 +1,13 @@
 ## -*- mode: makefile; tab-width: 4 -*-
 include $(top_srcdir)/Makefile.incl
 
-scheddir = $(libexecdir)/sched
+scheddir = $(prefix)/lib/boinc-server-maker/sched
+toolsdir = $(prefix)/lib/boinc-server-maker/tools
+toolbindir = $(prefix)/lib/boinc-server-maker/tools
 
-bin_PROGRAMS = create_work sign_executable dir_hier_path dir_hier_move cancel_jobs
+toolbin_PROGRAMS = create_work sign_executable dir_hier_path dir_hier_move cancel_jobs
 
-dist_bin_SCRIPTS = \
+dist_toolbin_SCRIPTS = \
 	    boinc_submit \
 		dbcheck_files_exist \
 		demo_submit \
@@ -23,6 +25,15 @@
 		makelog.sh \
 		run_in_ops
 
+dist_tools_SCRIPTS = \
+    appmgr \
+    db_query \
+    manage_privileges \
+    run_in_ops \
+    stage_file \
+    watch_tcp \
+    project.xml
+
 AM_CXXFLAGS += $(MYSQL_CFLAGS)
 AM_LDFLAGS += -static -L../sched -L../lib/.libs
 
Index: boinc_upstream/vda/Makefile.am
===================================================================
--- boinc_upstream.orig/vda/Makefile.am
+++ boinc_upstream/vda/Makefile.am
@@ -1,6 +1,7 @@
 include $(top_srcdir)/Makefile.incl
 
-bin_PROGRAMS = vda vdad ssim
+vdadir=$(prefix)/lib/boinc-server-maker/vda
+vda_PROGRAMS = vda vdad ssim
 
 AM_CXXFLAGS += $(MYSQL_CFLAGS)
 AM_LDFLAGS += -static -L../lib/.libs -L../sched
Index: boinc_upstream/apps/Makefile.am
===================================================================
--- boinc_upstream.orig/apps/Makefile.am
+++ boinc_upstream/apps/Makefile.am
@@ -3,7 +3,7 @@
 
 include $(top_srcdir)/Makefile.incl
 
-exampledir = $(libexecdir)/examples
+exampledir = $(prefix)/lib/boinc-apps-examples
 example_PROGRAMS = upper_case concat 1sec
 
 upper_case_SOURCES = upper_case.cpp
Index: boinc_upstream/lib/Makefile.am
===================================================================
--- boinc_upstream.orig/lib/Makefile.am
+++ boinc_upstream/lib/Makefile.am
@@ -8,7 +8,8 @@
 # Stuff needed for server builds goes here.
 if ENABLE_SERVER
 
-bin_PROGRAMS += crypt_prog parse_test
+serverbindir = $(prefix)/lib/boinc-server-maker/lib
+serverbin_PROGRAMS = crypt_prog parse_test
 
 endif
 # end of "if ENABLE_SERVER"
Index: boinc_upstream/py/Makefile.am
===================================================================
--- boinc_upstream.orig/py/Makefile.am
+++ boinc_upstream/py/Makefile.am
@@ -19,6 +19,10 @@
 	Boinc/tools.py \
 	Boinc/util.py
 
+pydir=$(prefix)/usr/share/boinc-server-maker/py/Boinc
+py_SCRIPTS=$(EXTRA_DIST)
+
+
 $(srcdir)/Boinc/boinc_db.py: $(top_srcdir)/db/boinc_db.h $(top_srcdir)/lib/common_defs.h
 	cat $^ | $(top_srcdir)/py/db_def_to_py > $@
 
