Index: boinc/lib/network.h
===================================================================
--- boinc.orig/lib/network.h	2012-05-28 12:30:21.326797803 +0200
+++ boinc/lib/network.h	2012-05-28 15:07:10.631217544 +0200
@@ -54,10 +54,7 @@
 extern const char* socket_error_str();
 extern void reset_dns();
 
-#if defined(_WIN32) && defined(USE_WINSOCK)
-typedef int BOINC_SOCKLEN_T;
-#endif
-
+typedef int BOINC_SOCKLEN_T;
 #if defined(_WIN32) && defined(USE_WINSOCK)
 extern int WinsockInitialize();
 extern int WinsockCleanup();
