--- a/tools/dir_hier_path.cpp
+++ b/tools/dir_hier_path.cpp
@@ -42,7 +42,7 @@
     int retval;
 
     if ( (argc == 1) ||  !strcmp(argv[1], "-h")  || !strcmp(argv[1],"--help") || (argc != 2) ) {
-        fprintf(stderr, usage);
+        fprintf(stderr,"%s", usage);
         exit(1);
     }
 
--- a/tools/dir_hier_move.cpp
+++ b/tools/dir_hier_move.cpp
@@ -43,14 +43,14 @@
     int retval;
     
     if ( (argc == 1) || !strcmp(argv[1], "-h")  || !strcmp(argv[1],"--help") || (argc != 4) ) {
-        fprintf(stderr, usage);
+        fprintf(stderr,"%s", usage);
         exit(1);
     }
     src_dir = argv[1];
     dst_dir = argv[2];
     fanout = atoi(argv[3]);
     if (!fanout) {
-        fprintf(stderr, usage);
+        fprintf(stderr,"%s", usage);
         exit(1);
     }
 
--- a/clientgui/sg_ProjectPanel.cpp
+++ b/clientgui/sg_ProjectPanel.cpp
@@ -390,8 +390,7 @@
     CMainDocument* pDoc = wxGetApp().GetDocument();
     ProjectSelectionData* selData;
     PROJECT* project;
-    char* ctrl_url;
-    int i, j, oldProjectSelection, newProjectSelection;
+    int oldProjectSelection, newProjectSelection;
 
     if ( pDoc->IsConnected() ) {
         int projCnt = pDoc->GetSimpleProjectCount();
@@ -399,10 +398,11 @@
         oldProjectSelection = m_ProjectSelectionCtrl->GetSelection();
         
         // If a new project has been added, figure out which one
-        for(i=0; i<projCnt; i++) {
+        for(int i=0; i<projCnt; i++) {
+            char* ctrl_url;
             project = pDoc->state.projects[i];
             bool found = false;
-            for(j=0; j<ctrlCount; j++) {
+            for(int j=0; j<ctrlCount; j++) {
                 ctrl_url = ((ProjectSelectionData*)m_ProjectSelectionCtrl->GetClientData(j))->project_url;
                 if (!strcmp(project->master_url, ctrl_url)) {
                     found = true;
@@ -414,7 +414,7 @@
             if ( !found ) {
                 wxString projname(project->project_name.c_str(), wxConvUTF8);
 #if SORTPROJECTLIST
-                int alphaOrder;
+                int alphaOrder,j;
                 for(j = 0; j < ctrlCount; ++j) {
                     alphaOrder = (m_ProjectSelectionCtrl->GetString(j)).CmpNoCase(projname);
                     if (alphaOrder > 0) {
@@ -446,8 +446,8 @@
         if ( projCnt < ctrlCount ) {
             project = NULL;
             // Check items in descending order so deletion won't change indexes of items yet to be checked
-            for(j=ctrlCount-1; j>=0; --j) {
-                ctrl_url = ((ProjectSelectionData*)m_ProjectSelectionCtrl->GetClientData(j))->project_url;
+            for(int j=ctrlCount-1; j>=0; --j) {
+                char* ctrl_url = ((ProjectSelectionData*)m_ProjectSelectionCtrl->GetClientData(j))->project_url;
                 project = pDoc->state.lookup_project(ctrl_url);
                 if ( project == NULL ) {
                     selData = (ProjectSelectionData*)m_ProjectSelectionCtrl->GetClientData(j);
@@ -475,9 +475,9 @@
     
         // Check to see if we need to reload the project icon
         ctrlCount = m_ProjectSelectionCtrl->GetCount();
-        for(j=0; j<ctrlCount; j++) {
+        for(int j=0; j<ctrlCount; j++) {
             selData = (ProjectSelectionData*)m_ProjectSelectionCtrl->GetClientData(j);
-            ctrl_url = selData->project_url;
+            char* ctrl_url = selData->project_url;
             project = pDoc->state.lookup_project(ctrl_url);
             if ( project->project_files_downloaded_time > selData->project_files_downloaded_time ) {
                 wxBitmap* projectBM = GetProjectSpecificBitmap(ctrl_url);
