--- a/client/cs_files.cpp
+++ b/client/cs_files.cpp
@@ -177,7 +177,9 @@
     //
     if (download_gzipped && !boinc_file_exists(pathname)) {
         char gzpath[MAXPATHLEN];
-        sprintf(gzpath, "%s.gz", pathname);
+        snprintf(gzpath, sizeof(gzpath), "%s.gz", pathname);
+       //FIXME: a distinction is missing for the case that the .gz suffix goes beyond the MAXPATHLEN and is hence not found because
+       //       one should then not reperform the download as intended below
         if (boinc_file_exists(gzpath) ) {
 			if (allow_async && nbytes > ASYNC_FILE_THRESHOLD) {
 				ASYNC_VERIFY* avp = new ASYNC_VERIFY;
