From: YunQiang Su <wzssyqa@gmail.com>
Date: Mon, 28 May 2012 18:28:28 +0800
Subject: fix autoconf automake

---
 configure.in       |   13 ++++++-------
 4 files changed, 20 insertions(+), 17 deletions(-)

diff --git a/configure.in b/configure.in
index 756003d..36f0ebb 100644
--- a/configure.in
+++ b/configure.in
@@ -84,7 +84,7 @@ VERSION=${LIBXSLT_VERSION}
 
 AM_INIT_AUTOMAKE($PACKAGE, $VERSION)
 
-# AM_MAINTAINER_MODE
+AM_MAINTAINER_MODE
 
 # Support silent build rules, requires at least automake-1.11. Disable
 # by either passing --disable-silent-rules to configure or passing V=1
@@ -646,14 +647,14 @@ AC_SUBST(PYTHON_SITE_PACKAGES)
 
 XSLT_LIBDIR='-L${libdir}'
 XSLT_INCLUDEDIR='-I${includedir}'
-XSLT_LIBS="-lxslt $LIBXML_LIBS $M_LIBS"
+XSLT_LIBS="-lxslt $LIBXML_LIBS $EXTRA_LIBS"
 AC_SUBST(XSLT_LIBDIR)
 AC_SUBST(XSLT_INCLUDEDIR)
 AC_SUBST(XSLT_LIBS)
 
 EXSLT_LIBDIR='-L${libdir}'
 EXSLT_INCLUDEDIR='-I${includedir}'
-EXSLT_LIBS="-lexslt $XSLT_LIBS $LIBGCRYPT_LIBS"
+EXSLT_LIBS="-lexslt $XSLT_LIBS"
 AC_SUBST(EXSLT_LIBDIR)
 AC_SUBST(EXSLT_INCLUDEDIR)
 AC_SUBST(EXSLT_LIBS)
