*** vtkPLOT3DReader.cxx.old	Fri Nov 10 10:11:24 2006
--- vtkPLOT3DReader.cxx	Tue Nov 14 14:52:34 2006
***************
*** 570,576 ****
      int ni, nj, nk;
      this->ReadIntBlock(fp, 1, &ni);
      this->ReadIntBlock(fp, 1, &nj);
!     this->ReadIntBlock(fp, 1, &nk);
      vtkDebugMacro("Q, block " << i << " dimensions: "
                    << ni << " " << nj << " " << nk);
  
--- 570,583 ----
      int ni, nj, nk;
      this->ReadIntBlock(fp, 1, &ni);
      this->ReadIntBlock(fp, 1, &nj);
!     if (!this->TwoDimensionalGeometry)
!       {
!       this->ReadIntBlock(fp, 1, &nk);
!       }
!     else
!       {
!       nk = 1;
!       }
      vtkDebugMacro("Q, block " << i << " dimensions: "
                    << ni << " " << nj << " " << nk);
  
***************
*** 958,963 ****
--- 965,971 ----
        nthOutput->GetPointData()->AddArray(se);
        se->Delete();
  
+       this->SkipByteCount(qFp);
        
        if ( this->FunctionList->GetNumberOfTuples() > 0 )
          {
*** vtkMultiBlockPLOT3DReader.cxx.old	Fri Nov 10 10:11:32 2006
--- vtkMultiBlockPLOT3DReader.cxx	Tue Nov 14 14:51:31 2006
***************
*** 589,595 ****
      int ni, nj, nk;
      this->ReadIntBlock(fp, 1, &ni);
      this->ReadIntBlock(fp, 1, &nj);
!     this->ReadIntBlock(fp, 1, &nk);
      vtkDebugMacro("Q, block " << i << " dimensions: "
                    << ni << " " << nj << " " << nk);
  
--- 589,602 ----
      int ni, nj, nk;
      this->ReadIntBlock(fp, 1, &ni);
      this->ReadIntBlock(fp, 1, &nj);
!     if (!this->TwoDimensionalGeometry)
!       {
!       this->ReadIntBlock(fp, 1, &nk);
!       }
!     else
!       {
!       nk = 1;
!       }
      vtkDebugMacro("Q, block " << i << " dimensions: "
                    << ni << " " << nj << " " << nk);
  
***************
*** 1012,1017 ****
--- 1019,1025 ----
        nthOutput->GetPointData()->AddArray(se);
        se->Delete();
  
+       this->SkipByteCount(qFp);
        
        if ( this->FunctionList->GetNumberOfTuples() > 0 )
          {

