Index: irda-utils/findchip/nsc.c
===================================================================
--- irda-utils.orig/findchip/nsc.c	2012-03-05 19:36:53.135098382 +0100
+++ irda-utils/findchip/nsc.c	2012-03-05 19:38:18.543521916 +0100
@@ -33,7 +33,7 @@
 #include <string.h>
 #include <unistd.h>
 #include <fcntl.h>
-#include <asm/io.h>
+#include <sys/io.h>
 
 #include "findchip.h"
 
Index: irda-utils/findchip/smc.c
===================================================================
--- irda-utils.orig/findchip/smc.c	2012-03-05 19:37:05.859161508 +0100
+++ irda-utils/findchip/smc.c	2012-03-05 19:38:15.111504814 +0100
@@ -32,7 +32,7 @@
 #include <string.h>
 #include <unistd.h>
 #include <fcntl.h>
-#include <asm/io.h>
+#include <sys/io.h>
 
 #include "findchip.h"
 
Index: irda-utils/findchip/winbond.c
===================================================================
--- irda-utils.orig/findchip/winbond.c	2012-03-05 19:37:10.279183396 +0100
+++ irda-utils/findchip/winbond.c	2012-03-05 19:38:10.683482941 +0100
@@ -32,7 +32,7 @@
 #include <stdio.h>
 #include <unistd.h>
 #include <fcntl.h>
-#include <asm/io.h>
+#include <sys/io.h>
 
 #include "findchip.h"
 
