Index: irda-utils/irattach/README
===================================================================
--- irda-utils.orig/irattach/README	2012-03-05 19:39:48.807969509 +0100
+++ irda-utils/irattach/README	2012-03-05 19:40:03.052040133 +0100
@@ -24,12 +24,13 @@
 irattach must be run as root or installed setuid root, as it requires root
 privileges
 
-Usage: irattach <dev> [-d dongle] [-s]
+Usage: irattach <dev> [-d dongle] [-s] [-b]
 
 where
        <dev>        usually the name of a tty, but can also be a FIR device
        -d dongle    attaches a dongle driver (see below for more info)
        -s           starts discovery of remote IrDA devices
+       -b           do not fork to background
 
 Dongles:
 
Index: irda-utils/man/man-irattach-inc.sgml
===================================================================
--- irda-utils.orig/man/man-irattach-inc.sgml	2012-03-05 19:40:45.352249907 +0100
+++ irda-utils/man/man-irattach-inc.sgml	2012-03-05 19:40:56.716306248 +0100
@@ -23,6 +23,7 @@
  <ARG><REPLACEABLE>-s</REPLACEABLE></ARG>
  <ARG><REPLACEABLE>-d dongle</REPLACEABLE></ARG>
  <ARG><REPLACEABLE>-v</REPLACEABLE></ARG>
+ <ARG><REPLACEABLE>-b</REPLACEABLE></ARG>
  <ARG><REPLACEABLE>-h</REPLACEABLE></ARG>
 </CMDSYNOPSIS>
 </REFSYNOPSISDIV>
@@ -136,6 +137,10 @@
 </PARA>
 
 <PARA>
+<OPTION>-b</OPTION> : do not fork to background
+</PARA>
+
+<PARA>
 <OPTION>-h</OPTION> : shows help information.
 </PARA>
 
