Index: irda-utils/irdadump/main.c
===================================================================
--- irda-utils.orig/irdadump/main.c	2012-03-05 19:41:47.348557315 +0100
+++ irda-utils/irdadump/main.c	2012-03-05 19:41:59.232616276 +0100
@@ -134,6 +134,7 @@
 			fprintf(stderr,"\t-p <n>\tDisable parsing/decoding\n");
 			fprintf(stderr,"\t-i device\tIrDA port to listen on\n");
 			fprintf(stderr,"\t-w log\tWrites out raw capture file\n");
+			fprintf(stderr,"\t-v\tPrint version number and exit\n");
 			fprintf(stderr,"\t-r log\tRead from raw capture file\n");
  			exit(1);
 		default:
