Description: Fix FTBFS with ld --as-needed
Author: Matthias Klose <doko@ubuntu.com>
Bug-Debian: http://bugs.debian.org/609263
Index: irda-utils/irdadump/Makefile
===================================================================
--- irda-utils.orig/irdadump/Makefile	2012-03-05 19:12:40.907897192 +0100
+++ irda-utils/irdadump/Makefile	2012-03-05 19:55:47.808724940 +0100
@@ -40,7 +40,7 @@
 
 irdadump: $(IRDADUMP_OBJS) $(LIBIRDADUMP_TARGET)
 	$(prn_cc_o)
-	$(ECMD)$(CC) $(CFLAGS) `pkg-config --libs glib-2.0` -o  $(IRDADUMP_TARGET) $< $(LIBIRDADUMP_TARGET)
+	$(ECMD)$(CC) $(CFLAGS) -o  $(IRDADUMP_TARGET) $< $(LIBIRDADUMP_TARGET) `pkg-config --libs glib-2.0`
 
 
 .c.o:
