=== modified file 'src/signond/signond.conf'
--- old/src/signond/signond.conf	2012-07-25 14:57:50 +0000
+++ new/src/signond/signond.conf	2012-09-12 17:37:32 +0000
@@ -2,7 +2,7 @@
 [General]
 StoragePath=~/.signon/
 ;0 - fatal, 1 - critical (default), 2 - info/debug
-LoggingLevel=2
+;LoggingLevel=1
 
 [SecureStorage]
 ; CryptoManager selects the encryption for the credentials FS. Possible values:

