Description: move -lm to Libs.private
Forwarded: fftw@fftw.org
Author: Julian Taylor <jtaylor.debian@googlemail.com>

--- a/fftw.pc.in
+++ b/fftw.pc.in
@@ -6,5 +6,6 @@
 Name: FFTW
 Description: fast Fourier transform library
 Version: @VERSION@
-Libs: -L${libdir} -lfftw3@PREC_SUFFIX@ @LIBQUADMATH@ -lm
+Libs: -L${libdir} -lfftw3@PREC_SUFFIX@ @LIBQUADMATH@
+Libs.private: -lm
 Cflags: -I${includedir}
