From: Mike Hommey <glandium@debian.org>
Date: Sat, 8 Dec 2007 19:24:40 +0100
Subject: Set javascript.options.showInConsole

---
 modules/libpref/src/init/all.js |    1 +
 1 file changed, 1 insertion(+)

diff --git a/modules/libpref/src/init/all.js b/modules/libpref/src/init/all.js
index be5bfa5..287f61b 100644
--- a/modules/libpref/src/init/all.js
+++ b/modules/libpref/src/init/all.js
@@ -701,6 +701,7 @@ pref("javascript.options.strict",           false);
 pref("javascript.options.strict.debug",     true);
 #endif
 pref("javascript.options.relimit",          true);
+pref("javascript.options.showInConsole",    true);
 pref("javascript.options.methodjit.content", true);
 pref("javascript.options.methodjit.chrome",  true);
 pref("javascript.options.pccounts.content", false);
