Description: Simple spelling fix
Author: Alberto Gonzalez Iniesta <agi@inittab.org>
Index: openvpn/occ.c
===================================================================
--- openvpn.orig/occ.c	2012-02-23 17:35:52.609808455 +0100
+++ openvpn/occ.c	2012-02-23 17:37:34.086311649 +0100
@@ -373,7 +373,7 @@
 	      && c->c2.max_send_size_local > TUN_MTU_MIN
 	      && (c->c2.max_recv_size_remote < c->c2.max_send_size_local
 		  || c->c2.max_recv_size_local < c->c2.max_send_size_remote))
-	    msg (M_INFO, "NOTE: This connection is unable to accomodate a UDP packet size of %d. Consider using --fragment or --mssfix options as a workaround.",
+	    msg (M_INFO, "NOTE: This connection is unable to accommodate a UDP packet size of %d. Consider using --fragment or --mssfix options as a workaround.",
 		 c->c2.max_send_size_local);
 	}
       event_timeout_clear (&c->c2.occ_mtu_load_test_interval);
