Description: Improve kFreeBSD support
Author: Gonéri Le Bouder <goneri@rulezlan.org>
Bug-Debian: http://bugs.debian.org/626062
Index: openvpn-2.2.1/route.c
===================================================================
--- openvpn-2.2.1.orig/route.c	2011-12-13 12:24:54.641739154 +0100
+++ openvpn-2.2.1/route.c	2011-12-13 12:41:46.990083056 +0100
@@ -1192,7 +1192,7 @@
   argv_msg (D_ROUTE, &argv);
   status = openvpn_execve_check (&argv, es, 0, "ERROR: Solaris route add command failed");
 
-#elif defined(TARGET_FREEBSD)
+#elif defined(TARGET_FREEBSD)||defined(__FreeBSD_kernel__)
 
   argv_printf (&argv, "%s add",
 		ROUTE_PATH);
Index: openvpn-2.2.1/tun.c
===================================================================
--- openvpn-2.2.1.orig/tun.c	2011-12-13 12:41:30.078294479 +0100
+++ openvpn-2.2.1/tun.c	2011-12-13 12:41:46.993083019 +0100
@@ -1069,7 +1069,7 @@
 	  add_route_connected_v6_net(tt, es);
 	}
 
-#elif defined(TARGET_FREEBSD)||defined(TARGET_DRAGONFLY)
+#elif defined(TARGET_FREEBSD)||defined(TARGET_DRAGONFLY)||defined(__FreeBSD_kernel__)
 
       /* example: ifconfig tun2 10.2.0.2 10.2.0.1 mtu 1450 netmask 255.255.255.255 up */
       if (tun)
