--- xserver-xorg-video-tdfx.orig/src/tdfx_driver.c
+++ xserver-xorg-video-tdfx/src/tdfx_driver.c
@@ -2689,8 +2689,16 @@
   TDFXPtr pTDFX;
 
   TDFXTRACE("TDFXValidMode start\n");
-  if ((mode->HDisplay>2048) || (mode->VDisplay>1536)) 
+  if (mode->HDisplay>2048) {
+    xf86DrvMsg(scrnIndex, X_INFO, "rejecting mode with horizontal resolution %d"
+               " exceeding maximum of 2048\n", mode->HDisplay);
     return MODE_BAD;
+  }
+  if (mode->VDisplay>1536) {
+    xf86DrvMsg(scrnIndex, X_INFO, "rejecting mode with vertical resolution %d"
+               " exceeding maximum of 1536\n", mode->VDisplay);
+    return MODE_BAD;
+  }
   /* Banshee doesn't support interlace, but Voodoo 3 and higher do. */
   pScrn = xf86Screens[scrnIndex];
   pTDFX = TDFXPTR(pScrn);
@@ -2704,13 +2712,19 @@
         return MODE_OK;
         break;
       default:
+        xf86DrvMsg(scrnIndex, X_INFO, "rejecting mode with unsupported"
+	           " interlace flag\n");
         return MODE_BAD;
         break;
     }
   }
   /* In clock doubled mode widths must be divisible by 16 instead of 8 */
-  if ((mode->Clock>TDFX2XCUTOFF) && (mode->HDisplay%16))
+  if ((mode->Clock>TDFX2XCUTOFF) && (mode->HDisplay%16)) {
+    xf86DrvMsg(scrnIndex, X_INFO, "rejecting mode with horizontal resolution %d"
+               " not divisibile by 16 and clock %d greater than %d\n",
+               mode->HDisplay, mode->Clock, TDFX2XCUTOFF);
     return MODE_BAD;
+  }
   return MODE_OK;
 }
 
