From: David Paleino <dapal@debian.org>
Subject: fix creation of image symlink
Origin: vendor
Forwarded: no

---
 Makefile |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- leocad.orig/Makefile
+++ leocad/Makefile
@@ -133,7 +133,7 @@ install-data: desktop
 	@install -c -m 0644 tools/icon/icon.svg $(DESTDIR)$(PREFIX)/share/pixmaps/leocad.svg
 	@install -d $(DESTDIR)$(PREFIX)/share/icons/hicolor/scalable/mimetypes/
 	@rm -f $(DESTDIR)$(PREFIX)/share/icons/hicolor/scalable/mimetypes/application-vnd.leocad.svg
-	@ln -s $(DESTDIR)$(PREFIX)/share/pixmaps/leocad.svg $(DESTDIR)$(PREFIX)/share/icons/hicolor/scalable/mimetypes/application-vnd.leocad.svg
+	@ln -s $(PREFIX)/share/pixmaps/leocad.svg $(DESTDIR)$(PREFIX)/share/icons/hicolor/scalable/mimetypes/application-vnd.leocad.svg
 	@install -d $(DESTDIR)$(PREFIX)/share/mime/packages
 	@install -c -m 0644 linux/leocad-mime.xml $(DESTDIR)$(PREFIX)/share/mime/packages/
 
