From: Petter Reinholdtsen <pere@debian.org>
Subject: get rid of some signed/unsigned warnings from the compiler.
Origin: vendor
Forwarded: no

---
 common/lc_zipfile.cpp |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- leocad.orig/common/lc_zipfile.cpp
+++ leocad/common/lc_zipfile.cpp
@@ -782,7 +782,7 @@ bool lcZipFile::AddFile(const char* File
 	lcZipFileInfo& Info = mFiles.Add();
 
 	Info.version = 0;
-    if (CompressedFile.GetLength() >= 0xffffffff || File.GetLength() >= 0xffffffff)
+    if (CompressedFile.GetLength() >= -1 || File.GetLength() >= -1)
 		Info.version_needed = 45;
 	else
 		Info.version_needed = 20;
