From: David Paleino <dapal@debian.org>
Subject: fix deprecated code
Origin: vendor
Forwarded: no

---
 common/project.cpp |   10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

--- leocad.orig/common/project.cpp
+++ leocad/common/project.cpp
@@ -3696,11 +3696,11 @@ void Project::HandleCommand(LC_COMMANDS
 				switch (opts.imopts.format)
 				{
 				default:
-				case LC_IMAGE_BMP: ext = ".bmp"; break;
-				case LC_IMAGE_GIF: ext = ".gif"; break;
-				case LC_IMAGE_JPG: ext = ".jpg"; break;
-				case LC_IMAGE_PNG: ext = ".png"; break;
-				case LC_IMAGE_AVI: ext = ".avi"; break;
+				case LC_IMAGE_BMP: ext = (char *)".bmp"; break;
+				case LC_IMAGE_GIF: ext = (char *)".gif"; break;
+				case LC_IMAGE_JPG: ext = (char *)".jpg"; break;
+				case LC_IMAGE_PNG: ext = (char *)".png"; break;
+				case LC_IMAGE_AVI: ext = (char *)".avi"; break;
 				}
 
 				if (m_bAnimation)
