Description: Don't regenerate svnversion.h
Forwarded: not-needed
Author: Balint Reczey <balint@balintreczey.hu>

Index: ./Makefile.am
===================================================================
--- ./Makefile.am	(revision 37943)
+++ ./Makefile.am	(working copy)
@@ -474,7 +474,7 @@
 # FORCE is the portable version of .PHONY
 FORCE:
 
-svnversion.h: FORCE
+svnversion.h:
 	$(PERL) $(srcdir)/make-version.pl $(srcdir)
 
 #

