Index: gnupg2/configure.ac
===================================================================
--- gnupg2.orig/configure.ac	2011-08-30 02:24:56.000000000 -0400
+++ gnupg2/configure.ac	2011-08-30 03:48:02.730942608 -0400
@@ -34,7 +34,7 @@
 m4_define([my_full_version], [my_version[]m4_if(my_issvn,[yes],
           [m4_if(git_revision,[],[-svn[]svn_revision],[-git[]git_revision])])])
 
-AC_INIT([gnupg],[my_full_version],[http://bugs.gnupg.org])
+AC_INIT([gnupg2],[my_full_version],[http://bugs.gnupg.org])
 # Set development_version to yes if the minor number is odd or you
 # feel that the default check for a development version is not
 # sufficient.
@@ -53,7 +53,7 @@
 
 
 PACKAGE=$PACKAGE_NAME
-PACKAGE_GT=${PACKAGE_NAME}2
+PACKAGE_GT=${PACKAGE_NAME}
 VERSION=$PACKAGE_VERSION
 
 AC_CONFIG_AUX_DIR(scripts)
