diff --git a/tools/src/common/Makefile b/tools/src/common/Makefile
index 4f81836..372bfa5 100644
--- a/tools/src/common/Makefile
+++ b/tools/src/common/Makefile
@@ -1,6 +1,6 @@
 
-CONFIG_ARCH=apple
-TAUROOT=/Users/khuck/tau2
+CONFIG_ARCH=default
+TAUROOT=
 TOOLS_SRC=$(TAUROOT)/tools/src
 
 
diff --git a/tools/src/perfdmf/Makefile b/tools/src/perfdmf/Makefile
index 3bab7dd..e6ef723 100644
--- a/tools/src/perfdmf/Makefile
+++ b/tools/src/perfdmf/Makefile
@@ -1,6 +1,6 @@
 # settings set automatically by the configure script.
-CONFIG_ARCH=apple
-TAUROOT=/Users/khuck/tau2
+CONFIG_ARCH=default
+TAUROOT=
 
 
 # these are classpaths for the build.  The second classpath is for the 
diff --git a/tools/src/vis/Makefile b/tools/src/vis/Makefile
index 3bb4664..1aecf97 100644
--- a/tools/src/vis/Makefile
+++ b/tools/src/vis/Makefile
@@ -1,6 +1,6 @@
 
-CONFIG_ARCH=i386_linux
-TAUROOT=/home/amorris/tau2
+CONFIG_ARCH=default
+TAUROOT=
 TOOLS_SRC=$(TAUROOT)/tools/src
 
 
diff --git a/tools/src/windows/examples/dll/Makefile b/tools/src/windows/examples/dll/Makefile
old mode 100755
new mode 100644
diff --git a/tools/src/windows/examples/threads/Makefile b/tools/src/windows/examples/threads/Makefile
old mode 100755
new mode 100644
diff --git a/utils/FixMakefile.sed b/utils/FixMakefile.sed
index 6bcad35..b235ec8 100644
--- a/utils/FixMakefile.sed
+++ b/utils/FixMakefile.sed
@@ -1,5 +1,33 @@
 s/\(.*\)#ENDIF#\(.*\)/\2\1#ENDIF#/g
 s/^CONFIG_ARCH=\(.*\)/CONFIG_ARCH=default/g
+s/^TAU_ARCH=\(.*\)/TAU_ARCH=default/g
+s@^TAUROOT=\(.*\)@TAUROOT=@g
+s@^TULIPDIR=\(.*\)@TULIPDIR=@g
+s@^TAUGCCLIBOPTS=\(.*\)@TAUGCCLIBOPTS=@g
+s@^TAUGCCLIBDIR=\(.*\)@TAUGCCLIBDIR=@g
+s@^PCLDIR=\(.*\)@PCLDIR=@g
+s@^PAPIDIR=\(.*\)@PAPIDIR=@g
+s@^PAPISUBDIR=\(.*\)@PAPISUBDIR=@g
+s@^PDTDIR=\(.*\)@PDTDIR=@g
+s@^PDTCOMPDIR=\(.*\)@PDTCOMPDIR=@g
+s@^DYNINSTDIR=\(.*\)@DYNINSTDIR=@g
+s@^JDKDIR=\(.*\)@JDKDIR=@g
+s@^OPARIDIR=\(.*\)@OPARIDIR=@g
+s@^TAU_OPARI_TOOL=\(.*\)@TAU_OPARI_TOOL=@g
+s@^EPILOGDIR=\(.*\)@EPILOGDIR=@g
+s@^EXTRADIR=\(.*\)@EXTRADIR=@g
+s@^EXTRADIRCXX=\(.*\)@EXTRADIRCXX=@g
+s@^VTFDIR=\(.*\)@VTFDIR=@g
+s@^OTFDIR=\(.*\)@OTFDIR=@g
+s@^SLOG2DIR=\(.*\)@SLOG2DIR=@g
+s@^PYTHON_INCDIR=\(.*\)@PYTHON_INCDIR=@g
+s@^PYTHON_LIBDIR=\(.*\)@PYTHON_LIBDIR=@g
+s@^TAU_CONFIG=\(.*\)@TAU_CONFIG=@g
+s@^TAU_MPI_INC=\(.*\)@TAU_MPI_INC=@g
+s@^TAU_MPI_LIB=\(.*\)@TAU_MPI_LIB=@g
+s@^TAU_MPI_FLIB=\(.*\)@TAU_MPI_FLIB=@g
+s@^TAU_MPILIB_DIR=\(.*\)@TAU_MPILIB_DIR=@g
+s@^TAU_PREFIX_INSTALL_DIR=\(.*\)@TAU_PREFIX_INSTALL_DIR=@g
 s/#PVM_INSTALLED#\(.*\)/\1#PVM_INSTALLED#/g
 s/^PVM_ARCH=\(.*\)/PVM_ARCH=default/g
 s@^PVM_DIR=\(.*\)@PVM_DIR=default@g
@@ -7,3 +35,8 @@ s@^CONFIG_CC=\(.*\)@CONFIG_CC=gcc@g
 s@^CONFIG_CXX=\(.*\)@CONFIG_CXX=g++@g
 s@^PCXX_OPT=\(.*\)@PCXX_OPT=-g@g
 s@^USER_OPT=\(.*\)@USER_OPT=-g@g
+s@^PDT_CXX=\(.*\)@PDT_CXX=@g
+s@^TAU_SHMEM_INC=\(.*\)@TAU_SHMEM_INC=@g
+s@^TAU_SHMEM_LIB=\(.*\)@TAU_SHMEM_LIB=@g
+s@^CHARMDIR=\(.*\)@CHARMDIR=@g
+s@^FORCEIA32=\(.*\)@FORCEIA32=@g
diff --git a/utils/csUI/Makefile b/utils/csUI/Makefile
index 0061432..a2b9b78 100644
--- a/utils/csUI/Makefile
+++ b/utils/csUI/Makefile
@@ -19,12 +19,12 @@ #**      Documentation   : See http://ww
 #****************************************************************************
 
 ########### Automatically modified by the configure script ############
-CONFIG_ARCH=solaris2
+CONFIG_ARCH=default
 CONFIG_CC=gcc
 CONFIG_CXX=g++
 PCXX_OPT=-g
-USER_OPT=
-TAUROOT=/tmp/sameer/tau2
+USER_OPT=-g
+TAUROOT=
 XFORMSROOT=/local/apps/xforms/FORMS
 #######################################################################
  
