diff --git a/Makefile b/Makefile
index 16dca22..00aba25 100644
--- a/Makefile
+++ b/Makefile
@@ -87,7 +87,7 @@ exports : 
          else echo System previously configured as a `utils/ConfigQuery -arch` ; fi
 	@echo "*********** RECURSIVELY MAKING SUBDIRECTORIES ***********"
 	@for i in ${EXPORTS}; do (echo "*** COMPILING $$i DIRECTORY"; cd $$i;\
-             $(MAKE) "MAKE=$(MAKE)" "CC=$(CC)" "CXX=$(CXX)" "LINKER=$(LINKER)" ); done
+             $(MAKE) "MAKE=$(MAKE)" "CC=$(CC)" "CXX=$(CXX)" "LINKER=$(LINKER)" ) || exit $?; done
 	@echo "***************** DONE ************************"
 
 tests: 
