diff --git a/src/Profile/Makefile b/src/Profile/Makefile
index 734bb42..be365f0 100644
--- a/src/Profile/Makefile
+++ b/src/Profile/Makefile
@@ -284,133 +284,133 @@ Profile.class	: Profile.java
 	$(JDKDIR)/bin/javac Profile.java 
 
 Profiler.o 	: Profiler.cpp $(HDRS)
-	$(CXX) $(CXXFLAGS) -c $(@:.o=.cpp)
+	$(CXX) $(CXXFLAGS) -c $<
 
 TulipTimers.o 	: TulipTimers.cpp $(HDRS)
-	$(CXX) $(CXXFLAGS) -c $(@:.o=.cpp)
+	$(CXX) $(CXXFLAGS) -c $<
 
 UserEvent.o 	: UserEvent.cpp $(HDRS)
-	$(CXX) $(CXXFLAGS) -c $(@:.o=.cpp)
+	$(CXX) $(CXXFLAGS) -c $<
 
 FunctionInfo.o 	: FunctionInfo.cpp $(HDRS)
-	$(CXX) $(CXXFLAGS) -c $(@:.o=.cpp)
+	$(CXX) $(CXXFLAGS) -c $<
 
 RtsLayer.o 	: RtsLayer.cpp $(HDRS)
-	$(CXX) $(CXXFLAGS) -c $(@:.o=.cpp)
+	$(CXX) $(CXXFLAGS) -c $<
 
 RtsThread.o 	: RtsThread.cpp $(HDRS)
-	$(CXX) $(CXXFLAGS) -c $(@:.o=.cpp)
+	$(CXX) $(CXXFLAGS) -c $<
 
 ProfileParam.o   : ProfileParam.cpp $(HDRS)
-	$(CXX) $(CXXFLAGS) -c $(@:.o=.cpp)
+	$(CXX) $(CXXFLAGS) -c $<
 
 pcxx_events.o   : pcxx_events.cpp $(HDRS)
-	$(CXX) $(CXXFLAGS) -c $(@:.o=.cpp)
+	$(CXX) $(CXXFLAGS) -c $<
 
 Tracer.o   	: Tracer.cpp $(HDRS)
-	$(CXX) $(CXXFLAGS) -c $(@:.o=.cpp)
+	$(CXX) $(CXXFLAGS) -c $<
 
 PthreadLayer.o   : PthreadLayer.cpp $(HDRS)
-	$(CXX) $(CXXFLAGS) -c $(@:.o=.cpp)
+	$(CXX) $(CXXFLAGS) -c $<
 
 SprocLayer.o   : SprocLayer.cpp $(HDRS)
-	$(CXX) $(CXXFLAGS) -c $(@:.o=.cpp)
+	$(CXX) $(CXXFLAGS) -c $<
 
 JavaThreadLayer.o   : JavaThreadLayer.cpp $(HDRS)
-	$(CXX) $(CXXFLAGS) -c $(@:.o=.cpp)
+	$(CXX) $(CXXFLAGS) -c $<
 
 TulipThreadLayer.o   : TulipThreadLayer.cpp $(HDRS)
-	$(CXX) $(CXXFLAGS) -c $(@:.o=.cpp)
+	$(CXX) $(CXXFLAGS) -c $<
 
 TauCAPI.o   : TauCAPI.cpp $(HDRS)
-	$(CXX) $(CXXFLAGS) -c $(@:.o=.cpp)
+	$(CXX) $(CXXFLAGS) -c $<
 
 TauFAPI.o   : TauFAPI.cpp $(HDRS)
-	$(CXX) $(CXXFLAGS) -c $(@:.o=.cpp)
+	$(CXX) $(CXXFLAGS) -c $<
 
 TauJAPI.o   : TauJAPI.cpp $(HDRS) 
-	$(CXX) $(CXXFLAGS) -c $(@:.o=.cpp)
+	$(CXX) $(CXXFLAGS) -c $<
 
 TauMapping.o   : TauMapping.cpp $(HDRS)
-	$(CXX) $(CXXFLAGS) -c $(@:.o=.cpp)
+	$(CXX) $(CXXFLAGS) -c $<
 
 TauHooks.o   : TauHooks.cpp $(HDRS)
-	$(CXX) $(CXXFLAGS) -c $(@:.o=.cpp)
+	$(CXX) $(CXXFLAGS) -c $<
 
 TauJava.o :	TauJava.cpp
-	$(CXX) $(CXXFLAGS) -c TauJava.cpp
+	$(CXX) $(CXXFLAGS) -c $<
 
 PclLayer.o 	: PclLayer.cpp $(HDRS)
-	$(CXX) $(CXXFLAGS) -c $(@:.o=.cpp)
+	$(CXX) $(CXXFLAGS) -c $<
 
 PapiLayer.o 	: PapiLayer.cpp $(HDRS)
-	$(CXX) $(CXXFLAGS) -c $(@:.o=.cpp)
+	$(CXX) $(CXXFLAGS) -c $<
 
 MultipleCounters.o : MultipleCounters.cpp $(HDRS)
-	$(CXX) $(CXXFLAGS) -c $(@:.o=.cpp)
+	$(CXX) $(CXXFLAGS) -c $<
 
 TauDisable.o 	: TauDisable.cpp $(HDRS)
-	$(CXX) $(CXXFLAGS) -c $(@:.o=.cpp)
+	$(CXX) $(CXXFLAGS) -c $<
 
 OpenMPLayer.o 	: OpenMPLayer.cpp $(HDRS)
-	$(CXX) $(CXXFLAGS) -c $(@:.o=.cpp)
+	$(CXX) $(CXXFLAGS) -c $<
 
 TauOpari.o 	: TauOpari.cpp $(HDRS)
-	$(CXX) $(CXXFLAGS) -c $(@:.o=.cpp)
+	$(CXX) $(CXXFLAGS) -c $<
 
 TauKojakOpari.o : TauKojakOpari.cpp $(HDRS)
-	$(CXX) $(CXXFLAGS) -c $(@:.o=.cpp)
+	$(CXX) $(CXXFLAGS) -c $<
 
 TauLinuxTimers.o : TauLinuxTimers.c $(HDRS)
-	gcc -c $(@:.o=.c)
+	gcc -c $<
 
 TauCallPath.o 	: TauCallPath.cpp $(HDRS)
-	$(CXX) $(CXXFLAGS) -c $(@:.o=.cpp)
+	$(CXX) $(CXXFLAGS) -c $<
 
 PyGroups.o 	: PyGroups.cpp $(HDRS)
-	$(CXX) $(CXXFLAGS) -c $(@:.o=.cpp)
+	$(CXX) $(CXXFLAGS) -c $<
 
 PyExceptions.o 	: PyExceptions.cpp $(HDRS)
-	$(CXX) $(CXXFLAGS) -c $(@:.o=.cpp)
+	$(CXX) $(CXXFLAGS) -c $<
 
 PyDatabase.o 	: PyDatabase.cpp $(HDRS)
-	$(CXX) $(CXXFLAGS) -c $(@:.o=.cpp)
+	$(CXX) $(CXXFLAGS) -c $<
 
 PyBindings.o 	: PyBindings.cpp $(HDRS)
-	$(CXX) $(CXXFLAGS) -c $(@:.o=.cpp)
+	$(CXX) $(CXXFLAGS) -c $<
 
 PyTimer.o 	: PyTimer.cpp $(HDRS)
-	$(CXX) $(CXXFLAGS) -c $(@:.o=.cpp)
+	$(CXX) $(CXXFLAGS) -c $<
 
 TauMuse.o 	: TauMuse.cpp $(HDRS)
-	$(CXX) $(CXXFLAGS) -c $(@:.o=.cpp)
+	$(CXX) $(CXXFLAGS) -c $<
 
 TauMuseFilters.o : TauMuseFilters.cpp $(HDRS)
-	$(CXX) $(CXXFLAGS) -c $(@:.o=.cpp)
+	$(CXX) $(CXXFLAGS) -c $<
 
 TauMuseHandlers.o : TauMuseHandlers.cpp $(HDRS)
-	$(CXX) $(CXXFLAGS) -c $(@:.o=.cpp)
+	$(CXX) $(CXXFLAGS) -c $<
 
 TauMusePackages.o : TauMusePackages.cpp $(HDRS)
-	$(CXX) $(CXXFLAGS) -c $(@:.o=.cpp)
+	$(CXX) $(CXXFLAGS) -c $<
 
 TauCompensate.o : TauCompensate.cpp $(HDRS)
-	$(CXX) $(CXXFLAGS) -c $(@:.o=.cpp)
+	$(CXX) $(CXXFLAGS) -c $<
 
 TauHandler.o : TauHandler.cpp $(HDRS)
-	$(CXX) $(CXXFLAGS) -c $(@:.o=.cpp)
+	$(CXX) $(CXXFLAGS) -c $<
 
 TauMemory.o : TauMemory.cpp $(HDRS)
-	$(CXX) $(CXXFLAGS) -c $(@:.o=.cpp)
+	$(CXX) $(CXXFLAGS) -c $<
 
 ktau_proc_interface.o : ktau_proc_interface.c $(HDRS)
-	$(CC) $(CXXFLAGS) -c $(@:.o=.c)
+	$(CC) $(CXXFLAGS) -c $<
 
-$(TAUROOT)/src/Profile/TauPGIHelper.o 	: TauPGIHelper.cpp $(HDRS)
-	$(CXX) $(CXXFLAGS) -c $(@:.o=.cpp) -o $@
+TauPGIHelper.o 	: TauPGIHelper.cpp $(HDRS)
+	$(CXX) $(CXXFLAGS) -c $<
 
 
 clean : 
