diff --git a/src/Profile/Makefile b/src/Profile/Makefile
index be365f0..ef01e93 100644
--- a/src/Profile/Makefile
+++ b/src/Profile/Makefile
@@ -40,6 +40,9 @@ ############# For KAI KCC compiler, use 
 #KAI#CEXTRA			= --one_per	   	   #ENDIF#
 ##############################################
 
+# Use -fPIC, so that the shared lib is correct
+#GNU#CEXTRA			= -fPIC		   	   #ENDIF#
+
 ##############################################
 #PGICC#PRELINK_PHASE = $(CXX) --prelink_objects $(OBJS) #ENDIF#
 #PGINOPRELINK#PRELINK_PHASE = #ENDIF#
