diff --git a/examples/fork/Makefile b/examples/fork/Makefile
index 1ebbcfc..2ff28a2 100644
--- a/examples/fork/Makefile
+++ b/examples/fork/Makefile
@@ -13,7 +13,7 @@ ########################################
  
 TAUROOTDIR	= ../..
 
-include $(TAUROOTDIR)/include/Makefile
+include $(TAUROOTDIR)/default/lib/Makefile.tau
 
 CXX		= $(TAU_CXX)
 
diff --git a/examples/instrument/Makefile b/examples/instrument/Makefile
index c1df0b3..d0ce51e 100644
--- a/examples/instrument/Makefile
+++ b/examples/instrument/Makefile
@@ -13,7 +13,7 @@ ########################################
  
 TAUROOTDIR	= ../..
 
-include $(TAUROOTDIR)/include/Makefile
+include $(TAUROOTDIR)/default/lib/Makefile.tau
 
 CXX		= $(TAU_CXX)
 
diff --git a/examples/mapping/embedded/Makefile b/examples/mapping/embedded/Makefile
index 4928de3..847692d 100644
--- a/examples/mapping/embedded/Makefile
+++ b/examples/mapping/embedded/Makefile
@@ -13,7 +13,7 @@ ########################################
  
 TAUROOTDIR	= ../../..
 
-include $(TAUROOTDIR)/include/Makefile
+include $(TAUROOTDIR)/default/lib/Makefile.tau
 
 CXX		= $(TAU_CXX)
 
diff --git a/examples/mapping/external/Makefile b/examples/mapping/external/Makefile
index a84d29c..4472379 100644
--- a/examples/mapping/external/Makefile
+++ b/examples/mapping/external/Makefile
@@ -13,7 +13,7 @@ ########################################
  
 TAUROOTDIR	= ../../..
 
-include $(TAUROOTDIR)/include/Makefile
+include $(TAUROOTDIR)/default/lib/Makefile.tau
 
 CXX		= $(TAU_CXX)
 
diff --git a/examples/selectiveAccess/Makefile b/examples/selectiveAccess/Makefile
index b0b20f7..3f7c191 100644
--- a/examples/selectiveAccess/Makefile
+++ b/examples/selectiveAccess/Makefile
@@ -13,7 +13,7 @@ ########################################
  
 TAUROOTDIR	= ../..
 
-include $(TAUROOTDIR)/include/Makefile
+include $(TAUROOTDIR)/default/lib/Makefile.tau
 
 CXX		= $(TAU_CXX)
 
