diff -ur meanwhile_1.0.2/src/mw_debug.c ourmw/src/mw_debug.c
--- meanwhile_1.0.2/src/mw_debug.c	2005-12-02 22:46:10.000000000 -0500
+++ ourmw/src/mw_debug.c	2013-01-14 23:18:49.861449024 -0500
@@ -19,7 +19,7 @@
 */
 
 
-#include <glib/gstring.h>
+#include <glib.h>
 
 #include "mw_debug.h"
 
diff -ur meanwhile_1.0.2/src/mw_message.h ourmw/src/mw_message.h
--- meanwhile_1.0.2/src/mw_message.h	2005-12-14 19:30:28.000000000 -0500
+++ ourmw/src/mw_message.h	2013-01-14 23:14:37.474786909 -0500
@@ -22,7 +22,7 @@
 #define _MW_MESSAGE_H
 
 
-#include <glib/glist.h>
+#include <glib.h>
 #include "mw_common.h"
 
 
diff -ur meanwhile_1.0.2/src/mw_srvc_conf.h ourmw/src/mw_srvc_conf.h
--- meanwhile_1.0.2/src/mw_srvc_conf.h	2005-12-14 19:30:28.000000000 -0500
+++ ourmw/src/mw_srvc_conf.h	2013-01-14 23:16:50.984784501 -0500
@@ -22,7 +22,7 @@
 #define _MW_SRVC_CONF_H
 
 
-#include <glib/glist.h>
+#include <glib.h>
 #include "mw_common.h"
 
 
diff -ur meanwhile_1.0.2/src/mw_srvc_dir.h ourmw/src/mw_srvc_dir.h
--- meanwhile_1.0.2/src/mw_srvc_dir.h	2005-12-14 19:30:28.000000000 -0500
+++ ourmw/src/mw_srvc_dir.h	2013-01-14 23:16:21.048118375 -0500
@@ -22,7 +22,6 @@
 
 
 #include <glib.h>
-#include <glib/glist.h>
 
 
 #ifdef __cplusplus
diff -ur meanwhile_1.0.2/src/mw_srvc_place.h ourmw/src/mw_srvc_place.h
--- meanwhile_1.0.2/src/mw_srvc_place.h	2005-12-14 19:30:28.000000000 -0500
+++ ourmw/src/mw_srvc_place.h	2013-01-14 23:20:21.348114039 -0500
@@ -22,7 +22,7 @@
 #define _MW_SRVC_PLACE_H
 
 
-#include <glib/glist.h>
+#include <glib.h>
 #include "mw_common.h"
 
 
diff -ur meanwhile_1.0.2/src/mw_srvc_resolve.h ourmw/src/mw_srvc_resolve.h
--- meanwhile_1.0.2/src/mw_srvc_resolve.h	2005-12-14 19:30:28.000000000 -0500
+++ ourmw/src/mw_srvc_resolve.h	2013-01-14 23:19:49.651447946 -0500
@@ -23,7 +23,6 @@
 
 
 #include <glib.h>
-#include <glib/glist.h>
 
 
 #ifdef __cplusplus
diff -ur meanwhile_1.0.2/src/mw_st_list.h ourmw/src/mw_st_list.h
--- meanwhile_1.0.2/src/mw_st_list.h	2005-12-14 19:30:28.000000000 -0500
+++ ourmw/src/mw_st_list.h	2013-01-14 23:18:00.071449923 -0500
@@ -30,7 +30,6 @@
 
 
 #include <glib.h>
-#include <glib/glist.h>
 #include "mw_common.h"
 
 
diff -ur meanwhile_1.0.2/src/mw_util.h ourmw/src/mw_util.h
--- meanwhile_1.0.2/src/mw_util.h	2004-12-28 15:41:26.000000000 -0500
+++ ourmw/src/mw_util.h	2013-01-14 23:13:55.454787668 -0500
@@ -23,9 +23,6 @@
 
 
 #include <glib.h>
-#include <glib/ghash.h>
-#include <glib/glist.h>
-
 
 #define map_guint_new() \
   g_hash_table_new(g_direct_hash, g_direct_equal)
diff -ur meanwhile_1.0.2/src/srvc_aware.c ourmw/src/srvc_aware.c
--- meanwhile_1.0.2/src/srvc_aware.c	2005-12-08 17:02:11.000000000 -0500
+++ ourmw/src/srvc_aware.c	2013-01-14 23:15:27.071452681 -0500
@@ -19,8 +19,6 @@
 */
 
 #include <glib.h>
-#include <glib/ghash.h>
-#include <glib/glist.h>
 #include <string.h>
 
 #include "mw_channel.h"
diff -ur meanwhile_1.0.2/src/srvc_conf.c ourmw/src/srvc_conf.c
--- meanwhile_1.0.2/src/srvc_conf.c	2005-12-27 21:46:54.000000000 -0500
+++ ourmw/src/srvc_conf.c	2013-01-14 23:17:02.878117620 -0500
@@ -19,8 +19,6 @@
 */
 
 #include <glib.h>
-#include <glib/ghash.h>
-#include <glib/glist.h>
 
 #include <stdio.h>
 #include <stdlib.h>
diff -ur meanwhile_1.0.2/src/srvc_dir.c ourmw/src/srvc_dir.c
--- meanwhile_1.0.2/src/srvc_dir.c	2005-09-15 16:30:20.000000000 -0400
+++ ourmw/src/srvc_dir.c	2013-01-14 23:15:52.858118883 -0500
@@ -18,7 +18,7 @@
   Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
 */
 
-#include <glib/ghash.h>
+#include <glib.h>
 
 #include "mw_channel.h"
 #include "mw_common.h"
diff -ur meanwhile_1.0.2/src/srvc_ft.c ourmw/src/srvc_ft.c
--- meanwhile_1.0.2/src/srvc_ft.c	2005-09-15 16:30:37.000000000 -0400
+++ ourmw/src/srvc_ft.c	2013-01-14 23:17:30.014783797 -0500
@@ -19,7 +19,7 @@
 */
 
 
-#include <glib/glist.h>
+#include <glib.h>
 
 #include "mw_channel.h"
 #include "mw_common.h"
diff -ur meanwhile_1.0.2/src/srvc_im.c ourmw/src/srvc_im.c
--- meanwhile_1.0.2/src/srvc_im.c	2005-12-27 21:46:54.000000000 -0500
+++ ourmw/src/srvc_im.c	2013-01-14 23:20:49.758113529 -0500
@@ -19,7 +19,6 @@
 */
 
 #include <glib.h>
-#include <glib/glist.h>
 #include <string.h>
 
 #include "mw_channel.h"
diff -ur meanwhile_1.0.2/src/srvc_place.c ourmw/src/srvc_place.c
--- meanwhile_1.0.2/src/srvc_place.c	2005-12-02 21:00:29.000000000 -0500
+++ ourmw/src/srvc_place.c	2013-01-14 23:20:11.918114213 -0500
@@ -19,8 +19,6 @@
 */
 
 #include <glib.h>
-#include <glib/ghash.h>
-#include <glib/glist.h>
 
 #include <stdio.h>
 #include <stdlib.h>
diff -ur meanwhile_1.0.2/src/srvc_resolve.c ourmw/src/srvc_resolve.c
--- meanwhile_1.0.2/src/srvc_resolve.c	2005-10-26 20:10:06.000000000 -0400
+++ ourmw/src/srvc_resolve.c	2013-01-14 23:19:36.294781520 -0500
@@ -18,7 +18,7 @@
   Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
 */
 
-#include <glib/ghash.h>
+#include <glib.h>
 
 #include "mw_channel.h"
 #include "mw_common.h"
diff -ur meanwhile_1.0.2/src/srvc_store.c ourmw/src/srvc_store.c
--- meanwhile_1.0.2/src/srvc_store.c	2005-11-18 20:52:42.000000000 -0500
+++ ourmw/src/srvc_store.c	2013-01-14 23:19:13.678115262 -0500
@@ -18,7 +18,7 @@
   Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
 */
 
-#include <glib/glist.h>
+#include <glib.h>
 
 #include "mw_channel.h"
 #include "mw_debug.h"
diff -ur meanwhile_1.0.2/src/st_list.c ourmw/src/st_list.c
--- meanwhile_1.0.2/src/st_list.c	2005-12-08 17:01:27.000000000 -0500
+++ ourmw/src/st_list.c	2013-01-14 23:18:26.541449444 -0500
@@ -20,7 +20,7 @@
 
 #include <stdio.h>
 #include <string.h>
-#include <glib/gstring.h>
+#include <glib.h>
 
 #include "mw_debug.h"
 #include "mw_util.h"
