From: =?UTF-8?q?Guido=20G=C3=BCnther?= <agx@sigxcpu.org>
Date: Thu, 17 Nov 2011 21:04:30 +0100
Subject: Fix build with xulrunner-8.0

---
 plugin/npshell.c |    4 ++--
 plugin/npunix.c  |    2 +-
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/plugin/npshell.c b/plugin/npshell.c
index ca4e78b..d8f826c 100644
--- a/plugin/npshell.c
+++ b/plugin/npshell.c
@@ -91,10 +91,10 @@
  *
  ***********************************************************************/
 
-char *
+const char *
 NPP_GetMIMEDescription(void)
 {
-  return (char *) MIME_TYPES_HANDLED;
+  return (const char *) MIME_TYPES_HANDLED;
 }
 
 NPError
diff --git a/plugin/npunix.c b/plugin/npunix.c
index 83ac628..54d8a41 100644
--- a/plugin/npunix.c
+++ b/plugin/npunix.c
@@ -537,7 +537,7 @@ Private_HandleEvent(NPP instance, void* event)
  *  - Netscape uses the return value to identify when an object instance
  *    of this plugin should be created.
  */
-char *
+const char *
 NP_GetMIMEDescription(void)
 {
     return NPP_GetMIMEDescription();
