diff --git a/all.dem b/all.dem
index 863a6c8..050cebf 100644
--- a/all.dem
+++ b/all.dem
@@ -41,8 +41,8 @@ print "******************** file mgr.dem ********************"
 load "mgr.dem"
 reset
 
-print "******************** file fit.dem ********************"
-load "fit.dem"
+print "******************** file <zcat fit.dem.gz ********************"
+load "<zcat fit.dem.gz"
 reset
 
 print "******************** file param.dem ********************"
@@ -53,8 +53,8 @@ print "******************** file polar.dem ********************"
 load "polar.dem"
 reset
 
-print "******************** file poldat.dem ********************"
-load "poldat.dem"
+print "******************** file <zcat poldat.dem.gz ********************"
+load "<zcat poldat.dem.gz"
 reset
 
 print "******************** file multiplt.dem ********************"
@@ -62,8 +62,8 @@ load "multiplt.dem"
 load "layout.dem"
 reset
 
-print "******************** file surface1.dem ********************"
-load "surface1.dem"
+print "******************** file <zcat surface1.dem.gz ********************"
+load "<zcat surface1.dem.gz"
 reset
 
 print "******************** file discrete.dem ********************"
@@ -83,16 +83,16 @@ load "world.dem"
 load "world2.dem"
 reset
 
-print "******************** file prob.dem ********************"
-load "prob.dem"
+print "******************** file <zcat prob.dem.gz ********************"
+load "<zcat prob.dem.gz"
 reset
 
-print "******************** file prob2.dem ********************"
-load "prob2.dem"
+print "******************** file <zcat prob2.dem.gz ********************"
+load "<zcat prob2.dem.gz"
 reset
 
-print "******************** file random.dem ********************"
-load "random.dem"
+print "******************** file <zcat random.dem.gz ********************"
+load "<zcat random.dem.gz"
 reset
 
 print "******************** file smooth.dem ********************"
@@ -119,16 +119,16 @@ print "******************** file singulr.dem ********************"
 load "singulr.dem"
 reset
 
-print "******************** file airfoil.dem ********************"
-load "airfoil.dem"
+print "******************** file <zcat airfoil.dem.gz ********************"
+load "<zcat airfoil.dem.gz"
 reset
 
 print "******************** file surface2.dem ********************"
 load "surface2.dem"
 reset
 
-print "******************** file contours.dem ********************"
-load "contours.dem"
+print "******************** file <zcat contours.dem.gz ********************"
+load "<zcat contours.dem.gz"
 reset
 
 print "******************** file multimsh.dem ********************"
@@ -155,12 +155,12 @@ print "********************** file rgb_variable.dem *********************"
 load "rgb_variable.dem"
 reset
 
-print "********************** file pm3d.dem *********************"
-load "pm3d.dem"
+print "********************** file <zcat pm3d.dem.gz *********************"
+load "<zcat pm3d.dem.gz"
 reset
 
-print "********************** file pm3dcolors.dem *********************"
-load "pm3dcolors.dem"
+print "********************** file <zcat pm3dcolors.dem.gz *********************"
+load "<zcat pm3dcolors.dem.gz"
 reset
 
 print "********************** file heatmaps.dem *********************"
@@ -217,8 +217,8 @@ load "histograms2.dem"
 reset
 
 print "**********************    Image formats    *********************"
-load "image.dem"
-load "image2.dem"
+load "<zcat image.dem.gz"
+load "<zcat image2.dem.gz"
 load "rgbalpha.dem"
 reset
 
@@ -242,8 +242,8 @@ print "********************** file ellipses_style.dem *********************"
 load "ellipses_style.dem"
 reset
 
-print "********************** file key.dem *********************"
-load "key.dem"
+print "********************** file <zcat key.dem.gz *********************"
+load "<zcat key.dem.gz"
 reset
 
 print "********************** file borders.dem *********************"
diff --git a/animate.dem b/animate.dem
index 9a1f031..db669f0 100644
--- a/animate.dem
+++ b/animate.dem
@@ -1,7 +1,7 @@
 #
 # $Id: animate.dem,v 1.7 2006/01/07 23:21:02 sfeam Exp $
 #
-# Demo animation, tumbling around 'glass.dat'.
+# Demo animation, tumbling around '<zcat glass.dat.gz'.
 #
 # History:
 #   - 1. 1. 2006 Dan Sebald:  Defined variables for more generic rotate
@@ -20,7 +20,7 @@ zrot_delta = 10
 xview(xrot)=(50.+30.*sin((xrot%180)/180.*pi))
 zview(zrot)=(60.+45.*sin(zrot/180.*pi))
 set view xview(xrot),zview(zrot)
-splot "glass.dat"
+splot "<zcat glass.dat.gz"
 
 limit_iterations=40 # limits number of iterations if nonzero
 
diff --git a/animate2.dem b/animate2.dem
index d129b72..7613a27 100644
--- a/animate2.dem
+++ b/animate2.dem
@@ -46,7 +46,7 @@ set view xview(xrot), zview(zrot), 2, 1
 set size square
 
 splot cos(u)*cos(v),cos(u)*sin(v),sin(u) notitle with lines lt 5, \
-      'world.dat' notitle with lines lt 2 lw 3
+      '<zcat world.dat.gz' notitle with lines lt 2 lw 3
 
 iteration_count=0
 xrot =(xrot+xrot_delta)%360
diff --git a/circles.dem b/circles.dem
index b8ab599..9598b6c 100644
--- a/circles.dem
+++ b/circles.dem
@@ -38,7 +38,7 @@ set border
 
 set yrange [-70:*]
 
-plot 'world.dat' with filledcurves lc rgb "light-green" , \
+plot '<zcat world.dat.gz' with filledcurves lc rgb "light-green" , \
      'world.cor' using 1:2:(7.*rand(0)) with circles lt 3 \
                  fs transparent solid 0.5 noborder
 
@@ -74,7 +74,7 @@ keyr = 25.
 set obj 20 rect from graph 0,0 to graph 1,1 behind fc rgb "cyan" fs solid 0.2
 set angle degree
 
-plot 'world.dat' with filledcurves lc rgb "light-green" notit, \
+plot '<zcat world.dat.gz' with filledcurves lc rgb "light-green" notit, \
      'energy_circles.dat' using 2:1:(sqrt($9)*r):6:($6+$7):5 \
             with circles lc pal fs solid 1.0 border rgb "gray",\
      for [i=0:types-1] '' using (keyx):(keyy):(keyr-5):(-(i+1)*360./types):(-(i)*360./types):5 \
@@ -111,7 +111,7 @@ unset key
 unset border
 set yrange [] reverse
 
-plot 'lena.rgb' binary array=(128,128) dx=4 dy=4 format='%uchar' with rgbimage, \
+plot '<zcat lena.rgb.gz' binary array=(128,128) dx=4 dy=4 format='%uchar' with rgbimage, \
      'lena-keypoints.bin' binary format='%double' with circles lc rgb "yellow"
 
 pause -1 "Hit return to continue"
@@ -123,7 +123,7 @@ set title "Delaunay triangulation of Hemisphere points, some empty circles in re
 unset key
 
 plot 'empty-circles.dat' with circles lw 2 lc rgb "red", \
-     'delaunay-edges.dat' with lines lc rgb "forest-green", \
-     'hemisphr.dat' u (100*$1):(100*$2) with points pt 7 lc rgb "black"
+     '<zcat delaunay-edges.dat.gz' with lines lc rgb "forest-green", \
+     '<zcat hemisphr.dat.gz' u (100*$1):(100*$2) with points pt 7 lc rgb "black"
 
 pause -1 "Hit return to continue"
diff --git a/ellipse.dem b/ellipse.dem
index fa20820..16b4e5e 100644
--- a/ellipse.dem
+++ b/ellipse.dem
@@ -1,7 +1,7 @@
 #
 # Ethan A Merritt - Feb 2007
 # This demo requires support for 'set object ellipse'
-# Also it uses the file 'random.tmp' created by random.dem
+# Also it uses the file '<zcat random.tmp.gz' created by <zcat random.dem.gz
 #
 if ((GPVAL_VERSION == 4.3 || GPVAL_VERSION == 4.2) \
 &&  (!strstrt(GPVAL_COMPILE_OPTIONS,"+OBJECTS"))) \
@@ -27,7 +27,7 @@ set object 1 ellipse center 1.5,1.0 size 2.,4. angle 20. front fillstyle empty b
 set obj 2 ellipse center 1.5, 1  size 6, 12  angle 20 front fs empty bo 3
 
 A = pi/9.
-plot 'random.tmp' using (1.5 + $1*cos(A)-2.*$2*sin(A)):(1.0 + $1*sin(A)+2.*$2*cos(A)) with dots
+plot '<zcat random.tmp.gz' using (1.5 + $1*cos(A)-2.*$2*sin(A)):(1.0 + $1*sin(A)+2.*$2*cos(A)) with dots
 
 pause -1 "Hit return to continue"
 
diff --git a/fillcrvs.dem b/fillcrvs.dem
index 68ef4ee..1d129c4 100644
--- a/fillcrvs.dem
+++ b/fillcrvs.dem
@@ -63,7 +63,7 @@ pause -1 'Press Return to continue'
 
 reset
 
-set title "world.dat plotted with filledcurves"
+set title "<zcat world.dat.gz plotted with filledcurves"
 set format x ""
 set format y ""
 set grid layerdefault linewidth 0.5 
@@ -71,7 +71,7 @@ set object  1 rect from graph 0, 0 to graph 1, 1 behind fc  rgb "#afffff" fillst
 set xrange [ -180.000 : 180.000 ]
 set yrange [ -70.0000 : 80.0000 ]
 set lmargin  1
-plot 'world.dat' with filledcurve notitle fs solid 1.0 lc rgb 'dark-goldenrod'
+plot '<zcat world.dat.gz' with filledcurve notitle fs solid 1.0 lc rgb 'dark-goldenrod'
 pause -1 'Press Return to continue'
 
 reset
diff --git a/hidden.dem b/hidden.dem
index 07e895d..0938c96 100644
--- a/hidden.dem
+++ b/hidden.dem
@@ -48,7 +48,7 @@ set style data lines
 # autoranging loses the verticals
 set xrange [0:15]
 set yrange [0:15]
-splot "glass.dat" using 1
+splot "<zcat glass.dat.gz" using 1
 pause -1 "Hit return to continue (6)"
 
 set view 50
diff --git a/kdensity2d.dem b/kdensity2d.dem
index 697de2a..9403d5c 100644
--- a/kdensity2d.dem
+++ b/kdensity2d.dem
@@ -1,8 +1,8 @@
-# Run 'random.dem' to generate 'random.tmp'
+# Run '<zcat random.dem.gz' to generate '<zcat random.tmp.gz'
 
 reset
 set title "How to plot a kernel density estimate for this 2D dataset?"
-plot 'random.tmp' using 1:2
+plot '<zcat random.tmp.gz' using 1:2
 
 pause -1 "Press Return to continue - the plot may take some time to appear"
 
@@ -11,7 +11,7 @@ set palette defined (0 'white', 1 'red')
 set dgrid3d 50,50 gauss kdensity
 set title "set dgrid3d 50,50 gauss kdensity"
 
-splot 'random.tmp' using 1:2:(1) with pm3d
+splot '<zcat random.tmp.gz' using 1:2:(1) with pm3d
 
 pause -1 "Press Return to continue - the plot may take some time to appear"
 set dgrid3d 50,50 gauss kdensity 0.1
diff --git a/molecule.dem b/molecule.dem
index ef77958..2abaed6 100644
--- a/molecule.dem
+++ b/molecule.dem
@@ -6,8 +6,8 @@ unset xtics
 unset ytics
 unset ztics
 
-splot 'GM1_sugar.pdb' using 6:7:8 with points lt 3 pt 7 ps 3 notitle, \
-      'GM1_bonds.r3d' using 1:2:3:($5-$1):($6-$2):($7-$3) with vectors nohead lw 3 lt 1 \
+splot '<zcat GM1_sugar.pdb.gz' using 6:7:8 with points lt 3 pt 7 ps 3 notitle, \
+      '<zcat GM1_bonds.r3d.gz' using 1:2:3:($5-$1):($6-$2):($7-$3) with vectors nohead lw 3 lt 1 \
       title "GM1 pentasaccharide"
 
 pause -1 "Hit return to continue"
diff --git a/multimsh.dem b/multimsh.dem
index 03b7a88..3fe62eb 100644
--- a/multimsh.dem
+++ b/multimsh.dem
@@ -9,40 +9,40 @@ set yrange [-4:4]
 set zrange [-2:2]
 set style data line
 set title "Demo of multiple mesh per file capability - Digitized Blue Whale"
-splot "whale.dat"
+splot "<zcat whale.dat.gz"
 pause -1 "Press Return"
 set title "Demo of multiple mesh per file capability - Digitized Blue Whale"
 set xlabel "Mesh or Network 0"
-splot "whale.dat" index 0
+splot "<zcat whale.dat.gz" index 0
 pause -1 "Press Return"
 
 set autoscale z
 set xlabel "Mesh or Network 6"
 set autoscale z
-splot "whale.dat" index 6 using 3:2:1 
+splot "<zcat whale.dat.gz" index 6 using 3:2:1 
 pause -1 "Press Return"
 
 set zrange [-2:2]
 set xlabel "Mesh or Network 12"
-splot "whale.dat" index 12 with points
+splot "<zcat whale.dat.gz" index 12 with points
 pause -1 "Press Return"
 
 set xlabel "Mesh or Network 13"
-splot "whale.dat" i 13
+splot "<zcat whale.dat.gz" i 13
 pause -1 "Press Return - A Loop over Indices could be done with reread"
 
 #  #file "loop"
 #  maxmsh = 16
 #  ind = (ind + 1)%maxmsh
-#  splot "whale.dat" i ind
+#  splot "<zcat whale.dat.gz" i ind
 #  reread "loop"
 
 set xlabel "Mesh or Network 4"
-splot "whale.dat" i 4
+splot "<zcat whale.dat.gz" i 4
 pause -1 "Press Return"
 
 set xlabel "Mesh or Network 5"
-splot "whale.dat" i 5
+splot "<zcat whale.dat.gz" i 5
 pause -1 "Press Return"
 
 reset
diff --git a/orbits.dem b/orbits.dem
index 86f1ea9..3cbf52d 100644
--- a/orbits.dem
+++ b/orbits.dem
@@ -22,7 +22,7 @@ set datafile separator "\t"
 if (strstrt(GNUTERM,"wxt")) \
     set label 1 "Use Ctrl+mousewheel to zoom!" at graph 0.5,0.05 center 
 
-fn = 'orbital_elements.dat'
+fn = '<zcat orbital_elements.dat.gz'
 
 # functions to calculate the parameters of the ellipses from orbital elements
 # the actual 3D orbits are projected onto the ecliptic plane
diff --git a/pointsize.dem b/pointsize.dem
index 64a81e7..55a14e0 100644
--- a/pointsize.dem
+++ b/pointsize.dem
@@ -13,7 +13,7 @@ unset ztics
 # plot a 2D map with locations marked by variable sized points
 #
 set title "plot with variable size points"
-plot 'world.dat' with lines lt 3, \
+plot '<zcat world.dat.gz' with lines lt 3, \
      'world.cor' using 1:2:(5.*rand(0)) with points lt 1 pt 6 ps variable
 pause -1 "Hit return to continue"
 
@@ -21,7 +21,7 @@ pause -1 "Hit return to continue"
 set title "splot with variable size points\nit is possible to specify size and color separately"
 set view map
 unset hidden3d
-splot 'world.dat' using 1:2:(0) with lines lt 3, \
+splot '<zcat world.dat.gz' using 1:2:(0) with lines lt 3, \
       'world.cor' using 1:2:(0.5-rand(0)):(5.*rand(0)) with points pt 5 ps var lt palette
 pause -1 "Hit return to continue"
 
@@ -38,7 +38,7 @@ set isosamples 9
 set urange [-90:90]
 set vrange [0:360]
 splot cos(u)*cos(v),cos(u)*sin(v),sin(u) with lines lt 5,\
-      'world.dat' with lines lt 3, \
+      '<zcat world.dat.gz' with lines lt 3, \
       'world.cor' using 1:2:(1):(5.*rand(0)) with points lt 1 pt 6 ps variable
 pause -1 "Hit return to continue"
 
@@ -52,7 +52,7 @@ set arrow from 0,0,-1.2 to 0,0,1.2 lt 5 lw 2
 set arrow from -1.2, 0, 0 to 1.2, 0, 0 nohead lt 5 lw 1 
 set arrow from 0, -1.2, 0 to 0, 1.2, 0 nohead lt 5 lw 1
 splot cos(u)*cos(v),-cos(u)*sin(v),sin(u) with lines lt 5,\
-      'world.dat' u 1:2:(1.001) with lines lt 3, \
+      '<zcat world.dat.gz' u 1:2:(1.001) with lines lt 3, \
       'world.cor' using 1:2:(1):(5.*rand(0))  with points lt 1 pt 6 ps var
 pause -1 "Hit return to continue"
 
diff --git a/rectangle.dem b/rectangle.dem
index 60d8a03..689c4f2 100644
--- a/rectangle.dem
+++ b/rectangle.dem
@@ -44,7 +44,7 @@ pause -1 "Hit return to continue"
 reset
 
 #
-# Requires data file "using.dat" from this directory,
+# Requires data file "<zcat using.dat.gz" from this directory,
 # so change current working directory to this directory before running.
 #
 
@@ -60,9 +60,9 @@ set obj rect from 3, graph 0 to 4, graph 1
 set obj rect from 5, graph 0 to 6, graph 1
 set obj rect from 7, graph 0 to 8, graph 1
 
-plot 'using.dat' using 3:4 title "Logged in" with impulses,\
-     'using.dat' using 3:5 t "Load average" with points,\
-     'using.dat' using 3:6 t "%CPU used" with lines
+plot '<zcat using.dat.gz' using 3:4 title "Logged in" with impulses,\
+     '<zcat using.dat.gz' using 3:5 t "Load average" with points,\
+     '<zcat using.dat.gz' using 3:6 t "%CPU used" with lines
 
 pause -1 "Hit return to continue"
 
diff --git a/rgbalpha.dem b/rgbalpha.dem
index 9d355f7..8a0c8f5 100644
--- a/rgbalpha.dem
+++ b/rgbalpha.dem
@@ -18,7 +18,7 @@ set yrange [ 0. : 128. ]
 # Alpha =  linear gradient on x
 
 plot 100.*(.4+sin(x/5.)/(x/5.)) lw 5 title 'solid line', \
-     'lena.rgb' binary array=(128,128) format="%uchar" flipy using 1:2:3:(2.*column(0)) \
+     '<zcat lena.rgb.gz' binary array=(128,128) format="%uchar" flipy using 1:2:3:(2.*column(0)) \
                 with rgbalpha title "Lena with linear\nalpha gradient"
 
 pause -1 "Hit return to continue"
@@ -28,7 +28,7 @@ pause -1 "Hit return to continue"
 focus(x,y) = ((column(0)-x)**2 + (column(-1)-(127-y))**2) > 400 ? 0 : 255
 
 plot 100.*(.4+sin(x/5.)/(x/5.)) lw 5 title 'solid line', \
-     'lena.rgb' binary array=(128,128) format="%uchar" flipy using 1:2:3:(focus(70,50)) \
+     '<zcat lena.rgb.gz' binary array=(128,128) format="%uchar" flipy using 1:2:3:(focus(70,50)) \
                 with rgbalpha title "Lena with circular mask"
 
 pause -1 "Hit return to continue"
diff --git a/scatter.dem b/scatter.dem
index 6f1eb45..cb1424a 100644
--- a/scatter.dem
+++ b/scatter.dem
@@ -13,30 +13,30 @@ set parametric
 set style data points
 set xlabel "data style point - no dgrid"
 set key box
-splot "hemisphr.dat"
+splot "<zcat hemisphr.dat.gz"
 pause -1 "Hit return to continue (1)"
 
 set dgrid3d 10,10,1
 set xlabel " data style lines, dgrid3d 10,10,1"
 set style data lines
-splot "hemisphr.dat"
+splot "<zcat hemisphr.dat.gz"
 pause -1 "Hit return to continue (2)"
 
 set dgrid3d ,,4
 set xlabel " data style lines, dgrid3d ,,4 "
 set style data lines
-splot "hemisphr.dat"
+splot "<zcat hemisphr.dat.gz"
 pause -1 "Hit return to continue (3)"
 
 set dgrid3d ,,16
 set xlabel " data style lines, dgrid3d ,,16"
 set style data lines
-splot "hemisphr.dat"
+splot "<zcat hemisphr.dat.gz"
 pause -1 "Hit return to continue (4)"
 
 set contour
 set xlabel "data style lines, dgrid3d ,,16, contour"
-splot "hemisphr.dat"
+splot "<zcat hemisphr.dat.gz"
 pause -1 "Hit return to continue (5)"
 
 unset dgrid3d
diff --git a/singulr.dem b/singulr.dem
index 76d0d74..daa0750 100644
--- a/singulr.dem
+++ b/singulr.dem
@@ -163,14 +163,14 @@ unset hidden3d
 set ticslevel 0.
 set view 50,15	#HBB: ,1,1.7
 set title "Klein bottle, glassblowers' version (look-through)"
-splot "klein.dat"
+splot "<zcat klein.dat.gz"
 pause -1 "Hit return to continue (19)"
 
 
 set hidden3d
 set view 70,305
 set title "Klein bottle, glassblowers' version (solid)"
-splot "klein.dat"
+splot "<zcat klein.dat.gz"
 pause -1 "Hit return to continue (20)"
 
 reset
diff --git a/smooth.dem b/smooth.dem
index 86c47f7..417d4d3 100644
--- a/smooth.dem
+++ b/smooth.dem
@@ -6,7 +6,7 @@ set zeroaxis
 set title "Uniform Distribution"
 set key top right
 set boxwidth 0.05
-plot [-0.1:1.1][-0.4:1.5] "random-points" u 1:(0.25*rand(0)-.35) t '', \
+plot [-0.1:1.1][-0.4:1.5] "<zcat random-points.gz" u 1:(0.25*rand(0)-.35) t '', \
      "" u (bin($1,0.05)):(20/300.) s f t 'smooth frequency' w boxes, \
      "" u 1:(1/300.) s cumul t 'smooth cumulative'
 pause -1 "Hit enter to continue"
@@ -16,7 +16,7 @@ pause -1 "Hit enter to continue"
 set title "Normal Distribution"
 set key top left
 set boxwidth 0.05
-plot "random-points" u 2:(0.25*rand(0)-.35) t '', \
+plot "<zcat random-points.gz" u 2:(0.25*rand(0)-.35) t '', \
      "" u (bin($2,0.05)):(20/300.) s f t 'smooth frequency' w boxes, \
      "" u 2:(1/300.) s cumul t 'smooth cumulative'
 pause -1 "Hit enter to continue"
@@ -26,7 +26,7 @@ pause -1 "Hit enter to continue"
 set title "Lognormal Distribution"
 set key top right
 set boxwidth 0.1
-plot [0:] "random-points" u 3:(0.25*rand(0)-.35) t '', \
+plot [0:] "<zcat random-points.gz" u 3:(0.25*rand(0)-.35) t '', \
     "" u (bin($3,0.1)):(10/300.) s f t 'smooth frequency' w boxes, \
     "" u 3:(1/300.) s cumul t 'smooth cumulative'
 pause -1 "Hit enter to continue"
@@ -36,7 +36,7 @@ pause -1 "Hit enter to continue"
 set title "Mixed Distribution (Lognormal with shifted Gaussian)"
 set key top right
 set boxwidth 0.1
-plot "random-points" u 4:(0.25*rand(0)-.35) t '', \
+plot "<zcat random-points.gz" u 4:(0.25*rand(0)-.35) t '', \
      "" u (bin($4,0.1)):(10/300.) s f t 'smooth frequency' w boxes, \
      "" u 4:(1/300.) s cumul t 'smooth cumulative'
 pause -1 "Hit enter to continue"
diff --git a/starmap.dem b/starmap.dem
index 830ec6e..04ba344 100644
--- a/starmap.dem
+++ b/starmap.dem
@@ -14,7 +14,7 @@ set colorbox horiz user origin .1,.1 size .8,.04
 set title "Gliese star catalog - 7 parsec neighborhood centered on Earth"
 set cblabel "Luminosity" offset 0,1
 #
-splot 'nearmap.csv' using 5:6:7:11 with points pt 6 ps 2 pal notitle, \
+splot '<zcat nearmap.csv.gz' using 5:6:7:11 with points pt 6 ps 2 pal notitle, \
       ''            using 5:6:7:11 with points pt 7 ps 1 pal notitle, \
       '' using 5:6:7:2 with labels font "Helvetica,9" left \
 	   point pt 0 offset 1,0 notitle
diff --git a/stats.dem b/stats.dem
index 86392a5..558a3da 100644
--- a/stats.dem
+++ b/stats.dem
@@ -13,14 +13,14 @@ set offset 0,0,.5,.5
 set autoscale fix
 set key left Left
 
-stats 'orbital_elements.dat' index 1 using 2 prefix "A"
+stats '<zcat orbital_elements.dat.gz' index 1 using 2 prefix "A"
 
 set arrow 1 from A_index_min, graph 0.1 to A_index_min, A_min fill
 set arrow 2 from A_index_max, graph 0.9 to A_index_max, A_max fill
 set label 1 at A_index_min, graph 0.1 "min" center offset 0,-1
 set label 2 at A_index_max, graph 0.9 "max" center offset 0,1
 
-plot 'orbital_elements.dat' index 1 using 0:2 title "  Data" lw 2, \
+plot '<zcat orbital_elements.dat.gz' index 1 using 0:2 title "  Data" lw 2, \
      A_mean title "  Mean"
 
 pause -1 "Hit return to continue"
@@ -29,14 +29,14 @@ set title "Use of stats command to find min/max/mean before plotting\nTwo data c
 
 f(x) = log(1+x)
 
-stats 'orbital_elements.dat' index 1 using (f($0)):2 prefix "B"
+stats '<zcat orbital_elements.dat.gz' index 1 using (f($0)):2 prefix "B"
 
 set arrow 1 from B_pos_min_y, graph 0.1 to B_pos_min_y, B_min_y fill
 set arrow 2 from B_pos_max_y, graph 0.9 to B_pos_max_y, B_max_y fill
 set label 1 at B_pos_min_y, graph 0.1 "min" center offset 0,-1
 set label 2 at B_pos_max_y, graph 0.9 "max" center offset 0,1
 
-plot 'orbital_elements.dat' index 1 using (f($0)):2 title "  Data" lw 2, \
+plot '<zcat orbital_elements.dat.gz' index 1 using (f($0)):2 title "  Data" lw 2, \
      B_mean_y title "  Mean", \
      B_slope * x + B_intercept title "Linear fit"
 
diff --git a/steps.dem b/steps.dem
index 0f15dc9..160cf65 100644
--- a/steps.dem
+++ b/steps.dem
@@ -17,7 +17,7 @@ pause -1 "Hit return for demonstration of automatic histogram creation"
 set title "Histogram built from unsorted data by 'smooth frequency'"
 set ylabel 'counts per bin'
 set xlabel 'bins'
-plot 'hemisphr.dat' u (floor($1*20)):(1) smooth frequency with histeps
+plot '<zcat hemisphr.dat.gz' u (floor($1*20)):(1) smooth frequency with histeps
 unset xlabel
 unset ylabel
 
diff --git a/triangle.dat b/triangle.dat
index a496168..09fb096 100644
--- a/triangle.dat
+++ b/triangle.dat
@@ -1,4 +1,4 @@
-# pm3d demo - datafile with different nb of points in scans and options
+# <zcat pm3d.dem.gzo - datafile with different nb of points in scans and options
 # set pm3d flush [begin | center | end]
 
 0 0 1
diff --git a/using.dem b/using.dem
index 66b46dd..742a1cc 100644
--- a/using.dem
+++ b/using.dem
@@ -1,15 +1,15 @@
 #
 # $Id: using.dem,v 1.6 2003/10/28 05:35:54 sfeam Exp $
 #
-# Requires data file "using.dat" from this directory,
+# Requires data file "<zcat using.dat.gz" from this directory,
 # so change current working directory to this directory before running.
 #
 
 set title "Convex     November 1-7 1989    Circadian"
 set key left box
 set xrange[-1:24]
-plot 'using.dat' using 2:4 title "Logged in" with impulses,\
-     'using.dat' using 2:4 title "Logged in" with points
+plot '<zcat using.dat.gz' using 2:4 title "Logged in" with impulses,\
+     '<zcat using.dat.gz' using 2:4 title "Logged in" with points
 pause -1 "Hit return to continue"
 
 set xrange [1:8]
@@ -17,9 +17,9 @@ set xrange [1:8]
 set title "Convex     November 1-7 1989"
 set key below
 set label "(Weekend)" at 5,25 center
-plot 'using.dat' using 3:4 title "Logged in" with impulses,\
-     'using.dat' using 3:5 t "Load average" with points,\
-     'using.dat' using 3:6 t "%CPU used" with lines
+plot '<zcat using.dat.gz' using 3:4 title "Logged in" with impulses,\
+     '<zcat using.dat.gz' using 3:5 t "Load average" with points,\
+     '<zcat using.dat.gz' using 3:6 t "%CPU used" with lines
 unset label
 pause -1 "Hit return to continue"
 reset
diff --git a/vector.dem b/vector.dem
index 387cb24..3565373 100644
--- a/vector.dem
+++ b/vector.dem
@@ -62,19 +62,19 @@ splot vtot(x,y) w l
 print "Now create a file with equipotential lines"
 pause -1 "Hit return to continue"
 
-set table "equipo2.tmp"
+set table "<zcat equipo2.tmp.gz"
 replot
 unset table
 reset
 pause 0
 
-plot "equipo2.tmp" w l
+plot "<zcat equipo2.tmp.gz" w l
 print "Now create a x/y datafile for plotting with vectors "
 print "and display vectors parallel to the electrostatic field"
 pause -1 "Hit return to continue"
 set isosam 31,31
 
-set table "field2xy.tmp"
+set table "<zcat field2xy.tmp.gz"
 splot vtot(x,y) w l
 unset table
 pause 0
@@ -84,8 +84,8 @@ set xr [xmin:xmax]
 set yr [ymin:ymax]
 set isosam 31,31
 set key under Left reverse
-plot "field2xy.tmp" u 1:2:(coef*dx1($1,$2)):(coef*dy1($1,$2)) w vec, \
-     "equipo2.tmp" w l
+plot "<zcat field2xy.tmp.gz" u 1:2:(coef*dx1($1,$2)):(coef*dy1($1,$2)) w vec, \
+     "<zcat equipo2.tmp.gz" w l
 pause -1 "Hit return to continue"
 
 reset
diff --git a/world.dem b/world.dem
index f7e47c9..79b9e8a 100644
--- a/world.dem
+++ b/world.dem
@@ -10,7 +10,7 @@ unset xtics
 unset ytics
 #
 # plot world map and correspondent locations as a +
-plot 'world.dat' with lines lt 3 , 'world.cor' with points lt 1 pt 2
+plot '<zcat world.dat.gz' with lines lt 3 , 'world.cor' with points lt 1 pt 2
 set title ""
 set key on
 set border
@@ -37,7 +37,7 @@ set isosamples 9
 set urange [-90:90]
 set vrange [0:360]
 splot cos(u)*cos(v),cos(u)*sin(v),sin(u) with lines lc rgb "cyan" ,\
-'world.dat' with lines lc rgb "blue" , 'world.cor' with points lt 1 pt 2
+'<zcat world.dat.gz' with lines lc rgb "blue" , 'world.cor' with points lt 1 pt 2
 pause -1 "Hit return to continue"
 
 # HBB 20000715: new demo:
@@ -49,7 +49,7 @@ set arrow from 0,0,-1.2 to 0,0,1.2 lc rgb "cyan" lw 2
 set arrow from -1.2, 0, 0 to 1.2, 0, 0 nohead lc rgb "cyan" lw 1 
 set arrow from 0, -1.2, 0 to 0, 1.2, 0 nohead lc rgb "cyan" lw 1
 splot cos(u)*cos(v),-cos(u)*sin(v),sin(u) with lines lc rgb "cyan" ,\
-'world.dat' u 1:2:(1.001) with lines lc rgb "blue" , 'world.cor' with points lt 1 pt 2
+'<zcat world.dat.gz' u 1:2:(1.001) with lines lc rgb "blue" , 'world.cor' with points lt 1 pt 2
 pause -1 "Hit return to continue"
 
 unset arrow
@@ -66,6 +66,6 @@ set isosamples 13
 set urange [-180:180]
 set vrange [-90:90]
 splot cos(u),sin(u),v with lines lc rgb "cyan" ,\
-'world.dat' with lines lc rgb "blue" lw 2, 'world.cor' with points lt 1 pt 2
+'<zcat world.dat.gz' with lines lc rgb "blue" lw 2, 'world.cor' with points lt 1 pt 2
 pause -1 "Hit return to continue"
 reset
diff --git a/world2.dem b/world2.dem
index 532ed44..e01ecc7 100644
--- a/world2.dem
+++ b/world2.dem
@@ -22,7 +22,7 @@ set cbrange [0:8]
 set colorb vert user size 0.02, 0.75
 unset hidden
 splot cos(u)*cos(v),cos(u)*sin(v),sin(u) notitle with lines lt 5, \
-      'world.dat' notitle with lines lt 2, \
+      '<zcat world.dat.gz' notitle with lines lt 2, \
       'srl.dat' using 3:2:(1):1:4 with labels notitle point pt 6 lw .1 left offset 1,0 font "Helvetica,7" tc pal
 pause -1 "Same plot with hidden line removal"
 set title "Labels with hidden line removal"
