Index: llvm-3.1-3.1/lib/Support/CommandLine.cpp
===================================================================
--- llvm-3.1-3.1.orig/lib/Support/CommandLine.cpp	2012-03-12 11:51:06.000000000 +0100
+++ llvm-3.1-3.1/lib/Support/CommandLine.cpp	2012-09-18 11:13:05.154409249 +0200
@@ -31,6 +31,7 @@
 #include "llvm/ADT/StringMap.h"
 #include "llvm/ADT/Twine.h"
 #include "llvm/Config/config.h"
+#include "../../debian.h"
 #include <cerrno>
 #include <cstdlib>
 using namespace llvm;
@@ -1351,6 +1352,10 @@
     OS << LLVM_VERSION_INFO;
 #endif
     OS << "\n  ";
+#ifdef LLVM_DEBIAN_INFO
+    OS << LLVM_DEBIAN_INFO;
+#endif
+    OS << "\n  ";
 #ifndef __OPTIMIZE__
     OS << "DEBUG build";
 #else
