Index: llvm-3.2-3.2~rc1/lib/Target/PowerPC/MCTargetDesc/PPCMCTargetDesc.h
===================================================================
--- llvm-3.2-3.2~rc1.orig/lib/Target/PowerPC/MCTargetDesc/PPCMCTargetDesc.h	2012-11-13 21:14:34.000000000 +0100
+++ llvm-3.2-3.2~rc1/lib/Target/PowerPC/MCTargetDesc/PPCMCTargetDesc.h	2012-11-15 11:04:00.000000000 +0100
@@ -44,6 +44,10 @@
                                          uint8_t OSABI);
 } // End llvm namespace
 
+// Generated files will use "namespace PPC". To avoid symbol clash,
+// undefine PPC here. PPC may be predefined on some hosts.
+#undef PPC
+
 // Defines symbolic names for PowerPC registers.  This defines a mapping from
 // register name to register number.
 //
Index: llvm-3.2-3.2~rc1/lib/Target/PowerPC/MCTargetDesc/PPCPredicates.h
===================================================================
--- llvm-3.2-3.2~rc1.orig/lib/Target/PowerPC/MCTargetDesc/PPCPredicates.h	2012-11-13 21:14:34.000000000 +0100
+++ llvm-3.2-3.2~rc1/lib/Target/PowerPC/MCTargetDesc/PPCPredicates.h	2012-11-15 11:04:00.000000000 +0100
@@ -14,6 +14,10 @@
 #ifndef LLVM_TARGET_POWERPC_PPCPREDICATES_H
 #define LLVM_TARGET_POWERPC_PPCPREDICATES_H
 
+// Generated files will use "namespace PPC". To avoid symbol clash,
+// undefine PPC here. PPC may be predefined on some hosts.
+#undef PPC
+
 namespace llvm {
 namespace PPC {
   /// Predicate - These are "(BI << 5) | BO"  for various predicates.
Index: llvm-3.2-3.2~rc1/lib/Target/PowerPC/MCTargetDesc/PPCFixupKinds.h
===================================================================
--- llvm-3.2-3.2~rc1.orig/lib/Target/PowerPC/MCTargetDesc/PPCFixupKinds.h	2012-11-13 21:14:34.000000000 +0100
+++ llvm-3.2-3.2~rc1/lib/Target/PowerPC/MCTargetDesc/PPCFixupKinds.h	2012-11-15 11:04:00.000000000 +0100
@@ -12,6 +12,8 @@
 
 #include "llvm/MC/MCFixup.h"
 
+#undef PPC
+
 namespace llvm {
 namespace PPC {
 enum Fixups {
