Index: llvm-3.2-3.2~rc1/autoconf/configure.ac
===================================================================
--- llvm-3.2-3.2~rc1.orig/autoconf/configure.ac	2012-11-13 21:14:34.000000000 +0100
+++ llvm-3.2-3.2~rc1/autoconf/configure.ac	2012-11-15 14:36:20.000000000 +0100
@@ -31,7 +31,7 @@
 dnl===-----------------------------------------------------------------------===
 dnl Initialize autoconf and define the package name, version number and
 dnl address for reporting bugs.
-AC_INIT([LLVM],[3.2svn],[http://llvm.org/bugs/])
+AC_INIT([LLVM],[3.2],[http://llvm.org/bugs/])
 AC_DEFINE([LLVM_VERSION_MAJOR], [3], [Major version of the LLVM API])
 AC_DEFINE([LLVM_VERSION_MINOR], [2], [Minor version of the LLVM API])
 
Index: llvm-3.2-3.2~rc1/configure
===================================================================
--- llvm-3.2-3.2~rc1.orig/configure	2012-11-15 12:28:46.000000000 +0100
+++ llvm-3.2-3.2~rc1/configure	2012-11-15 14:36:15.000000000 +0100
@@ -561,7 +561,7 @@
 # Identity of this package.
 PACKAGE_NAME='LLVM'
 PACKAGE_TARNAME='llvm'
-PACKAGE_VERSION='3.2svn'
+PACKAGE_VERSION='3.2'
 PACKAGE_STRING='LLVM 3.2svn'
 PACKAGE_BUGREPORT='http://llvm.org/bugs/'
 
