--- libreadonly-xs-perl.orig/XS.pm
+++ libreadonly-xs-perl/XS.pm
@@ -33,7 +33,7 @@
     if ($MAGIC_COOKIE ne "Do NOT use or require Readonly::XS unless you're me.")
     {
         require Carp;
-        Carp::croak "Readonly::XS is not a standalone module. You should not use it directly.";
+        Carp::croak("Readonly::XS is not a standalone module. You should not use it directly.");
     }
 }
 
