Index: haskell-debian-3.69.1/debian.cabal
===================================================================
--- haskell-debian-3.69.1.orig/debian.cabal	2012-11-20 01:37:05.000000000 +0100
+++ haskell-debian-3.69.1/debian.cabal	2012-12-01 23:56:33.087981810 +0100
@@ -29,7 +29,6 @@
    directory,
    filepath,
    HaXml >= 1.20,
-   HUnit,
    mtl,
    network >= 2.4,
    old-locale,
@@ -83,11 +82,7 @@
         Debian.URI,
         Debian.Util.FakeChanges
  other-modules:
-        Debian.Version.Internal,
-        Test.Changes,
-        Test.Dependencies,
-        Test.SourcesList,
-        Test.Versions
+        Debian.Version.Internal
 
 Executable fakechanges
  Main-is: utils/FakeChanges.hs
Index: haskell-debian-3.69.1/Setup.hs
===================================================================
--- haskell-debian-3.69.1.orig/Setup.hs	2012-11-20 01:37:05.000000000 +0100
+++ haskell-debian-3.69.1/Setup.hs	2012-12-01 23:56:33.087981810 +0100
@@ -6,8 +6,7 @@
 import System.Exit
 
 main = defaultMainWithHooks simpleUserHooks {
-         postBuild = \ _ _ _ _ -> runTestScript
-       , runTests = \ _ _ _ _ -> runTestScript
+       runTests = \ _ _ _ _ -> runTestScript
        }
 
 runTestScript =
