--- emma-0.6.orig/emmalib/__init__.py
+++ emma-0.6/emmalib/__init__.py
@@ -28,11 +28,6 @@
 import bz2
 
 try:
-	import gtk
-	from gtk import keysyms
-	import gobject
-	import gtk.gdk
-	import gtk.glade
 	if __name__ != "__main__":
 		from emmalib import __file__ as emmalib_file
 		from emmalib.mysql_host import *
@@ -41,6 +36,11 @@
 		emmalib_file = __file__
 		from mysql_host import *
 		from mysql_query_tab import *
+	import gtk
+	from gtk import keysyms
+	import gobject
+	import gtk.gdk
+	import gtk.glade
 except:
 	print "no gtk. you will not be able to start emma."
