Author: Chris Mason <chris.mason@oracle.com>
Description: Allow /'s in labels.

diff -Naurp btrfs-tools.orig/mkfs.c btrfs-tools/mkfs.c
--- btrfs-tools.orig/mkfs.c	2011-11-19 02:05:30.715938612 +0100
+++ btrfs-tools/mkfs.c	2011-11-19 02:26:09.494076467 +0100
@@ -339,7 +339,6 @@ static u64 parse_profile(char *s)
 
 static char *parse_label(char *input)
 {
-	int i;
 	int len = strlen(input);
 
 	if (len >= BTRFS_LABEL_SIZE) {
@@ -347,12 +346,6 @@ static char *parse_label(char *input)
 			BTRFS_LABEL_SIZE - 1);
 		exit(1);
 	}
-	for (i = 0; i < len; i++) {
-		if (input[i] == '/' || input[i] == '\\') {
-			fprintf(stderr, "invalid label %s\n", input);
-			exit(1);
-		}
-	}
 	return strdup(input);
 }
 
