Author: Luk Claes <luk@debian.org>
Description: Fixes FTBFS with --no-add-needed (Closes: #554059).

diff -Naurp btrfs-tools.orig/Makefile btrfs-tools/Makefile
--- btrfs-tools.orig/Makefile	2011-08-28 08:46:24.000000000 +0200
+++ btrfs-tools/Makefile	2011-08-28 09:02:30.524165410 +0200
@@ -14,7 +14,7 @@ DEPFLAGS = -Wp,-MMD,$(@D)/.$(@F).d,-MT,$
 INSTALL = install
 prefix ?= /usr/local
 bindir = $(prefix)/bin
-LIBS=-luuid -lm
+LIBS=-luuid -lm -lcom_err
 RESTORE_LIBS=-lz
 
 progs = btrfsctl mkfs.btrfs btrfs-debug-tree btrfs-show btrfs-vol btrfsck \
