Author: Donald Gordon <don@dis.org.nz>
Description: Correcting getopt (Closes: #665434).

diff -Naurp btrfs-tools.orig//cmds-scrub.c btrfs-tools/cmds-scrub.c
--- btrfs-tools.orig//cmds-scrub.c	2012-04-16 09:15:14.863734952 +0000
+++ btrfs-tools/cmds-scrub.c	2012-04-16 09:16:17.512169123 +0000
@@ -1590,7 +1590,7 @@ static int cmd_scrub_status(int argc, ch
 	int i;
 	int print_raw = 0;
 	int do_stats_per_dev = 0;
-	char c;
+	int c;
 	char fsid[37];
 	int fdres = -1;
 	int err = 0;
