Author: Daniel Baumann <daniel.baumann@progress-technologies.net>
Description: Update manpage to match /sbin/btrfs (Closes: #638778, #642302).

diff -Naurp btrfs-tools.orig/man/btrfs.8.in btrfs-tools/man/btrfs.8.in
--- btrfs-tools.orig/man/btrfs.8.in	2012-12-10 10:48:36.271412234 +0100
+++ btrfs-tools/man/btrfs.8.in	2012-12-10 10:51:53.679466625 +0100
@@ -21,6 +21,8 @@ btrfs \- control a btrfs filesystem
 [-s \fIstart\fR] [-t \fIsize\fR] -[vf] <\fIfile\fR>|<\fIdir\fR> \
 [<\fIfile\fR>|<\fIdir\fR>...]
 .PP
+\fBbtrfs\fP \fBfilesystem df\fP\fI <path> \fP
+.PP
 \fBbtrfs\fP \fBfilesystem sync\fP\fI <path> \fP
 .PP
 \fBbtrfs\fP \fBfilesystem resize\fP\fI [devid:][+/\-]<size>[gkm]|[devid:]max <filesystem>\fP
@@ -31,9 +33,9 @@ btrfs \- control a btrfs filesystem
 .PP
 \fBbtrfs\fP \fBfilesystem balance\fP\fI <path> \fP
 .PP
-\fBbtrfs\fP \fBdevice scan\fP\fI [--all-devices|<device> [<device>...]]\fP
+\fBbtrfs\fP \fBfilesystem scan\fP\fI [--all-devices|<device> [<device>...]]\fP
 .PP
-\fBbtrfs\fP \fBdevice show\fP\fI [--all-devices|<uuid>|<label>]\fP
+\fBbtrfs\fP \fBfilesystem show\fP\fI [--all-devices|<uuid>|<label>]\fP
 .PP
 \fBbtrfs\fP \fBdevice add\fP\fI <device> [<device>...] <path> \fP
 .PP
@@ -160,6 +162,10 @@ Get the default subvolume of the filesys
 is similar to \fBsubvolume list\fR command.
 .TP
 
+\fBfilesystem df\fR\fI <path>\fR
+Resize a filesystem identified by \fI<path>\fR.
+.TP
+
 \fBfilesystem defragment\fP -c[zlib|lzo] [-l \fIlen\fR] [-s \fIstart\fR] \
 [-t \fIsize\fR] -[vf] <\fIfile\fR>|<\fIdir\fR> [<\fIfile\fR>|<\fIdir\fR>...]
 
