Index: scalasca-1.4.2/cube-3.0/src/GUI-qt/Makefile
===================================================================
--- scalasca-1.4.2.orig/cube-3.0/src/GUI-qt/Makefile	2012-07-11 10:44:03.000000000 +0200
+++ scalasca-1.4.2/cube-3.0/src/GUI-qt/Makefile	2013-01-20 22:48:50.372376334 +0100
@@ -56,4 +56,4 @@
 
 install-doc:
 	$(MKDIR) -m 755 $(DOCDIR)/manuals
-	$(INSTALL) -c -m 644 ../../doc/CubeGuide.pdf $(DOCDIR)/manuals
+
Index: scalasca-1.4.2/utils/build/RealMakefile
===================================================================
--- scalasca-1.4.2.orig/utils/build/RealMakefile	2012-07-11 10:43:59.000000000 +0200
+++ scalasca-1.4.2/utils/build/RealMakefile	2013-01-20 22:48:33.184376717 +0100
@@ -242,9 +242,6 @@
 	$(INSTALL) -c -m 644 LICENSE                $(DOCDIR)
 	$(INSTALL) -c -m 644 README.1st             $(DOCDIR)
 	$(INSTALL) -c -m 644 OPEN_ISSUES            $(DOCDIR)
-	$(INSTALL) -c -m 644 doc/UserGuide.pdf      $(DOCDIR)/manuals
-	$(INSTALL) -c -m 644 doc/QuickReference.pdf $(DOCDIR)/manuals
-	$(INSTALL) -c -m 644 doc/index.html         $(DOCDIR)/html
 
 install-config:
 	$(MKDIR) -m 755 ${DOCDIR}
