Index: scalasca-1.4.2/epik/epilog/tools/elg_paraver.c
===================================================================
--- scalasca-1.4.2.orig/epik/epilog/tools/elg_paraver.c	2012-07-11 10:43:58.000000000 +0200
+++ scalasca-1.4.2/epik/epilog/tools/elg_paraver.c	2012-10-22 04:38:02.684899471 +0200
@@ -358,8 +358,8 @@
 static void write_paraver_cfgfile() {
   int a, i, firstid;
 
-  fprintf(pv_cfg, cfg_default);
-  fprintf(pv_cfg, cfg_states1);
+  fprintf(pv_cfg, "%s", cfg_default);
+  fprintf(pv_cfg, "%s", cfg_states1);
   /* map each non-default activity to a state */
   for (a=0; a<numact; a++) {
     if ( acttab[a].id == ELG_NO_ID ) break;
@@ -372,8 +372,8 @@
     if ( strcmp(acttab[a].name, "OMP-SYNC") == 0 ) continue;
     fprintf(pv_cfg, "%d    %s\n", ST_LAST + acttab[a].id, acttab[a].name);
   }
-  fprintf(pv_cfg, cfg_states2);
-  fprintf(pv_cfg, cfg_gradient);
+  fprintf(pv_cfg, "%s", cfg_states2);
+  fprintf(pv_cfg, "%s", cfg_gradient);
 
   /* Basic event types: Application, Flushing, Tracing */
   fprintf(pv_cfg, cfg_basic_events, ET_APPLIC, ET_FLUSH, ET_TRACING);
