Index: scalasca-1.4.2/mf/Makefile.defs.linux-gomp
===================================================================
--- scalasca-1.4.2.orig/mf/Makefile.defs.linux-gomp	2012-07-11 10:44:00.000000000 +0200
+++ scalasca-1.4.2/mf/Makefile.defs.linux-gomp	2012-10-22 04:44:48.264890421 +0200
@@ -23,15 +23,15 @@
 
 OPTFLAGS = -Wall
 CC       = gcc
-CFLAGS   = $(PFLAG) $(OPTFLAGS) -Wno-unknown-pragmas
+CFLAGS   = $(PFLAG) $(OPTCFLAGS) -Wno-unknown-pragmas
 CXX      = g++
-CXXFLAGS = $(PFLAG) $(OPTFLAGS) -Wno-unknown-pragmas -O3
+CXXFLAGS = $(PFLAG) $(OPTCXXFLAGS) -Wno-unknown-pragmas -O3
 F77      = gfortran
 F90      = $(F77)
-FFLAGS   = $(PFLAG) $(OPTFLAGS)
+FFLAGS   = $(PFLAG) $(OPTFFLAGS)
 FPP      =
 FDOPT    = -D
-LDFLAGS  = $(PFLAG) $(OPTFLAGS)
+LDFLAGS  = $(PFLAG) $(OPTLDFLAGS)
 UTILLIB  =
 FOBASE   = pomp2_fwrapper_base.o
 
