Index: hdf5-1.8.3/src/H5public.h
===================================================================
--- hdf5-1.8.3.orig/src/H5public.h	2009-08-06 16:37:19.000000000 +0200
+++ hdf5-1.8.3/src/H5public.h	2009-08-06 16:37:51.000000000 +0200
@@ -55,9 +55,6 @@
 #endif
 #ifdef H5_HAVE_PARALLEL
 #   include <mpi.h>
-#ifndef MPI_FILE_NULL		/*MPIO may be defined in mpi.h already       */
-#   include <mpio.h>
-#endif
 #endif
 
 
