Index: hdf5-1.8.10-patch1/src/H5F.c
===================================================================
--- hdf5-1.8.10-patch1.orig/src/H5F.c	2013-01-22 20:12:27.000000000 +0100
+++ hdf5-1.8.10-patch1/src/H5F.c	2013-01-26 09:48:33.814192347 +0100
@@ -129,6 +129,10 @@
 }};
 
 
+#ifndef PATH_MAX
+#define PATH_MAX    4096
+#endif  /* !PATH_MAX */
+
 
 /*-------------------------------------------------------------------------
  * Function:	H5F_init
Index: hdf5-1.8.10-patch1/testpar/t_cache.c
===================================================================
--- hdf5-1.8.10-patch1.orig/testpar/t_cache.c	2013-01-22 20:12:26.000000000 +0100
+++ hdf5-1.8.10-patch1/testpar/t_cache.c	2013-01-26 09:48:33.814192347 +0100
@@ -35,6 +35,10 @@
 
 #include "H5Fpkg.h"
 
+#ifndef PATH_MAX
+#define PATH_MAX    512
+#endif  /* !PATH_MAX */
+
 
 int     	nerrors = 0;
 int		failures = 0;
