Index: hdf5.git/src/H5public.h
===================================================================
--- hdf5.git.orig/src/H5public.h	2012-09-30 14:59:47.443402440 +0200
+++ hdf5.git/src/H5public.h	2012-11-25 11:14:18.767333174 +0100
@@ -58,6 +58,8 @@
 #   include <stddef.h>
 #endif
 #ifdef H5_HAVE_PARALLEL
+#   define MPICH_SKIP_MPICXX 1
+#   define OMPI_SKIP_MPICXX 1
 #   include <mpi.h>
 #endif
 
