Index: hdf5-1.8.10-patch1/src/H5Tconv.c
===================================================================
--- hdf5-1.8.10-patch1.orig/src/H5Tconv.c	2013-01-22 20:12:27.000000000 +0100
+++ hdf5-1.8.10-patch1/src/H5Tconv.c	2013-01-26 09:48:37.898192256 +0100
@@ -8514,7 +8514,6 @@
  *
  *-------------------------------------------------------------------------
  */
-#if H5T_CONV_INTERNAL_FP_ULLONG
 herr_t
 H5T__conv_float_ullong (hid_t src_id, hid_t dst_id, H5T_cdata_t *cdata,
 		       size_t nelmts, size_t buf_stride,
@@ -8523,7 +8522,6 @@
 {
     H5T_CONV_Fx(FLOAT, ULLONG, float, unsigned long long, 0, ULLONG_MAX);
 }
-#endif /*H5T_CONV_INTERNAL_FP_ULLONG*/
 
 
 /*-------------------------------------------------------------------------
@@ -8568,7 +8566,6 @@
  *
  *-------------------------------------------------------------------------
  */
-#if H5T_CONV_INTERNAL_FP_ULLONG
 herr_t
 H5T__conv_double_ullong (hid_t src_id, hid_t dst_id, H5T_cdata_t *cdata,
 		       size_t nelmts, size_t buf_stride,
@@ -8577,7 +8574,6 @@
 {
     H5T_CONV_Fx(DOUBLE, ULLONG, double, unsigned long long, 0, ULLONG_MAX);
 }
-#endif /*H5T_CONV_INTERNAL_FP_ULLONG*/
 
 
 /*-------------------------------------------------------------------------
@@ -8622,7 +8618,6 @@
  *
  *-------------------------------------------------------------------------
  */
-#if H5T_CONV_INTERNAL_LDOUBLE_ULLONG
 herr_t
 H5T__conv_ldouble_ullong (hid_t src_id, hid_t dst_id, H5T_cdata_t *cdata,
 		       size_t nelmts, size_t buf_stride,
@@ -8631,7 +8626,6 @@
 {
     H5T_CONV_Fx(LDOUBLE, ULLONG, long double, unsigned long long, 0, ULLONG_MAX);
 }
-#endif /*H5T_CONV_INTERNAL_LDOUBLE_ULLONG*/
 
 
 /*-------------------------------------------------------------------------
