From 49b3ccb16657599ed08e5e7018e79d63e1bde2a9 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Hugo=20Beauz=C3=A9e-Luyssen?= <beauze.h@gmail.com>
Date: Mon, 27 Aug 2012 23:35:16 +0300
Subject: [PATCH] key parser: Fix resource leak

---
 src/file/keydbcfg-parser.y |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/file/keydbcfg-parser.y b/src/file/keydbcfg-parser.y
index dd9506c..11a902b 100644
--- a/src/file/keydbcfg-parser.y
+++ b/src/file/keydbcfg-parser.y
@@ -449,6 +449,8 @@ int keydbcfg_parse_config(config_file *cfgfile, const char *path)
   cfgfile->host_cert_list = head_clist;
   cfgfile->list = head_celist;
 
+  fclose(fp);
+
   if (retval)
     return 0;
 
-- 
1.7.10.4

