diff --git a/scripts/bluetooth.rules.in b/scripts/bluetooth.rules.in
index 75218f7..d6117bb 100644
--- a/scripts/bluetooth.rules.in
+++ b/scripts/bluetooth.rules.in
@@ -1,4 +1,4 @@
 # Run helper every time a Bluetooth device appears
 # On remove actions, bluetoothd should go away by itself
-ACTION=="add", SUBSYSTEM=="bluetooth", RUN+="@prefix@/sbin/bluetoothd --udev"
-ACTION=="change", SUBSYSTEM=="bluetooth", RUN+="@prefix@/sbin/bluetoothd --udev"
+ACTION=="add", SUBSYSTEM=="bluetooth", RUN+="/lib/udev/bluez-udev --udev"
+ACTION=="change", SUBSYSTEM=="bluetooth", RUN+="/lib/udev/bluez-udev --udev"
