Index: ftgl-2.1.3~rc5/src/FTLayout/FTLayoutGlue.cpp
===================================================================
--- ftgl-2.1.3~rc5.orig/src/FTLayout/FTLayoutGlue.cpp	2009-11-21 02:07:38.000000000 -0500
+++ ftgl-2.1.3~rc5/src/FTLayout/FTLayoutGlue.cpp	2009-11-21 02:07:45.000000000 -0500
@@ -76,13 +76,13 @@
 
 // virtual FTBBox FTLayout::BBox(const char* string)
 extern "C++" {
-C_FUN(static FTBBox, _ftgGetlLayoutBBox, (FTGLlayout *l, const char *s),
+C_FUN(static FTBBox, _ftglGetLayoutBBox, (FTGLlayout *l, const char *s),
       return static_ftbbox, BBox, (s));
 }
 
-void ftgGetlLayoutBBox(FTGLlayout *l, const char * s, float c[6])
+void ftglGetLayoutBBox(FTGLlayout *l, const char * s, float c[6])
 {
-    FTBBox ret = _ftgGetlLayoutBBox(l, s);
+    FTBBox ret = _ftglGetLayoutBBox(l, s);
     FTPoint lower = ret.Lower(), upper = ret.Upper();
     c[0] = lower.Xf(); c[1] = lower.Yf(); c[2] = lower.Zf();
     c[3] = upper.Xf(); c[4] = upper.Yf(); c[5] = upper.Zf();
