From: Marcus Better <marcus@better.se>
Subject: fix off-by-one error in input-kbd
Bug-Debian: http://bugs.debian.org/388677
Forwarded: no

Signed-off-by: Marcus Better <marcus@better.se>

---
 input-kbd.c |    1 -
 1 files changed, 0 insertions(+), 1 deletions(-)

diff --git a/input-kbd.c b/input-kbd.c
index aaf23b9..6825ea9 100644
--- a/input-kbd.c
+++ b/input-kbd.c
@@ -50,7 +50,6 @@ static struct kbd_map* kbd_map_read(int fd, unsigned int version)
 			entry.keycode  = KEY_RESERVED;
 			rc = ioctl(fd, EVIOCGKEYCODE, &entry);
 			if (rc < 0) {
-				map->size--;
 				break;
 			}
 		} else {
-- 
tg: (cccd7bf..) features/fix-kbdmap-off-by-one (depends on: master)
