From: Marcus Better <marcus@better.se>
Subject: fix incorrect man page title for input-events
Bug-Debian: http://bugs.debian.org/413703
Forwarded: no

---
 input-events.man |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/input-events.man b/input-events.man
index da381c7..0a23cad 100644
--- a/input-events.man
+++ b/input-events.man
@@ -1,7 +1,7 @@
 .TH INPUT-EVENTS 8 "July 2005" "" ""
 .SH NAME
 
-input-events \- list input devices
+input-events \- print input events
 
 .SH SYNOPSIS
 \fBinput\-events \fR [\fB\-t\fR\ \fIsec\fR] [\fB\-g\fR]\ \fIdevnr\fR
-- 
tg: (28d1d45..) features/fix-input-events-manpage-title (depends on: upstream)
