From: Steve Langasek <vorlon@debian.org>
Subject: output the numeric value for keycodes in addition to the symbolic name
Forwarded: no

---
 input.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/input.c b/input.c
index ba9565d..9fc6963 100644
--- a/input.c
+++ b/input.c
@@ -149,8 +149,8 @@ void print_event(struct input_event *event)
 		EV_NAME[event->type]);
 	switch (event->type) {
 	case EV_KEY:
-		printf(" %s %s",
-			ev_type_name(EV_KEY, event->code),
+		printf(" %s (0x%x) %s",
+			ev_type_name(EV_KEY, event->code), event->code,
 			event->value ? "pressed" : "released");
 		break;
 	case EV_REL:
-- 
tg: (ef249de..) features/show-numeric-key-values (depends on: features/stdout-not-stderr)
