From: Marcus Better <marcus@better.se>
Subject: use up-to-date input.h from linux-libc-dev
Bug-Debian: http://bugs.debian.org/447550
Forwarded: not-needed

Signed-off-by: Marcus Better <marcus@better.se>

---
 input.h |    2 +-
 lirc.sh |    2 +-
 name.sh |    2 +-
 3 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/input.h b/input.h
index 0abdc8a..0576490 100644
--- a/input.h
+++ b/input.h
@@ -1,7 +1,7 @@
 #include <inttypes.h>
 
 #include <sys/ioctl.h>
-#include "linux-input.h"
+#include <linux/input.h>
 
 #define ev_name(code)  ((code) < EV_MAX  && EV_NAME[code]  ? EV_NAME[code]  : "???")
 #define ev_type_name(type, code) ((code) < EV_TYPE_MAX[type] && EV_TYPE_NAME[type][code] ? EV_TYPE_NAME[type][code] : "???")
diff --git a/lirc.sh b/lirc.sh
index 74a6169..2aec1fd 100755
--- a/lirc.sh
+++ b/lirc.sh
@@ -1,5 +1,5 @@
 #!/bin/sh
-INPUT="linux-input.h"
+INPUT="/usr/include/linux/input.h"
 cat <<EOF
 begin remote
 	name linux-input-layer
diff --git a/name.sh b/name.sh
index 562e0b4..744e9b8 100755
--- a/name.sh
+++ b/name.sh
@@ -1,7 +1,7 @@
 #!/bin/sh
 
 TYPE="$1"
-INPUT="linux-input.h"
+INPUT="/usr/include/linux/input.h"
 
 awk "
 	/EV_VERSION/	{ next };
-- 
tg: (cccd7bf..) features/use-system-input.h (depends on: master)
