Index: policykit-1/data/org.freedesktop.PolicyKit1.service.in
===================================================================
--- policykit-1.orig/data/org.freedesktop.PolicyKit1.service.in	2012-02-01 01:54:58.291191682 +0100
+++ policykit-1/data/org.freedesktop.PolicyKit1.service.in	2012-02-11 23:45:15.946856853 +0100
@@ -2,3 +2,4 @@
 Name=org.freedesktop.PolicyKit1
 Exec=@libexecdir@/polkitd --no-debug
 User=root
+SystemdService=polkitd.service
