Index: freeglut-2.6.0/src/freeglut_joystick.c
===================================================================
--- freeglut-2.6.0.orig/src/freeglut_joystick.c	2010-03-20 23:29:10.000000000 +0000
+++ freeglut-2.6.0/src/freeglut_joystick.c	2010-03-20 23:29:15.000000000 +0000
@@ -1094,6 +1094,8 @@
     joy->num_axes = joy->num_buttons = 0;
     joy->name[ 0 ] = '\0';
 
+    i = 0;
+
 #if TARGET_HOST_MACINTOSH
     /* XXX FIXME: get joystick name in Mac */
 
