Description: Fix build failures with --no-add-needed.
Author: Matthias Klose <doko@ubuntu.com>

--- freeglut-2.6.0.orig/progs/demos/spaceball/Makefile.am
+++ freeglut-2.6.0/progs/demos/spaceball/Makefile.am
@@ -3,4 +3,5 @@
 noinst_PROGRAMS = spaceball
 spaceball_SOURCES = spaceball.c vmath.c
 spaceball_LDFLAGS = -export-dynamic ../../../src/lib@LIBRARY@.la
-spaceball_CFLAGS = -I$(top_srcdir)/include $(X_CFLAGS)
\ No newline at end of file
+spaceball_CFLAGS = -I$(top_srcdir)/include $(X_CFLAGS)
+spaceball_LDADD = -lm
--- freeglut-2.6.0.orig/progs/demos/Lorenz/Makefile.am
+++ freeglut-2.6.0/progs/demos/Lorenz/Makefile.am
@@ -4,3 +4,4 @@
 lorenz_SOURCES = lorenz.c
 lorenz_LDFLAGS = -export-dynamic ../../../src/lib@LIBRARY@.la
 lorenz_CFLAGS = -I$(top_srcdir)/include $(X_CFLAGS)
+lorenz_LDADD = -lm
