Author: Daniel Baumann <daniel.baumann@progress-technologies.net>
Description: Ship configuration file with default theme set (Closes: #594999).

diff -Naurp plymouth.orig/src/plymouthd.conf plymouth/src/plymouthd.conf
--- plymouth.orig/src/plymouthd.conf	2012-09-26 18:50:53.000000000 +0200
+++ plymouth/src/plymouthd.conf	2012-10-27 19:39:09.271114581 +0200
@@ -1,3 +1,3 @@
 # Administrator customizations go in this file
 #[Daemon]
-#Theme=fade-in
+#Theme=text
diff -Naurp plymouth.orig/src/plymouthd.defaults plymouth/src/plymouthd.defaults
--- plymouth.orig/src/plymouthd.defaults	2012-10-27 19:27:05.334050283 +0200
+++ plymouth/src/plymouthd.defaults	2012-10-27 19:39:14.547151963 +0200
@@ -1,4 +1,4 @@
 # Distribution defaults. Changes to this file will get overwritten during
 # upgrades.
 [Daemon]
-Theme=spinner
+Theme=text
