Description: Fix typos found by Lintian.
Author: Benjamin Drung <bdrung@debian.org>

--- a/src/clients/nycli/xmms2.1
+++ b/src/clients/nycli/xmms2.1
@@ -303,7 +303,7 @@
 \fB\-f\fR, \fB\-\-format\fR \fIformat\fR
 .RE
 .RS 8
-Set the format string used to display status informations, instead of the one
+Set the format string used to display status information, instead of the one
 from the configuration file (see \fBFORMAT STRING\fR below).
 .RE
 .PP
--- a/src/xmms/xmms2d.1
+++ b/src/xmms/xmms2d.1
@@ -77,7 +77,7 @@
 .BI \-i\  url ,\ --ipc-socket= url
 Listen to socket at
 .IR url .
-Multiple sockets can be specified seperated by
+Multiple sockets can be specified separated by
 .BR ; -characters.
 See section
 .BR IPC\ SOCKET .
--- a/src/clients/lib/python/xmmsapi.c
+++ b/src/clients/lib/python/xmmsapi.c
@@ -26471,7 +26471,7 @@
  */
 
 static PyObject *__pyx_pf_7xmmsapi_7XmmsApi_89stats(PyObject *__pyx_v_self, PyObject *__pyx_args, PyObject *__pyx_kwds); /*proto*/
-static char __pyx_doc_7xmmsapi_7XmmsApi_89stats[] = "\n\t\tstats(cb=None) -> XmmsResult\n\n\t\tGet statistics informations from the server\n\t\t@rtype: L{XmmsResult}\n\t\t@return: The result of the operation.\n\t\t";
+static char __pyx_doc_7xmmsapi_7XmmsApi_89stats[] = "\n\t\tstats(cb=None) -> XmmsResult\n\n\t\tGet statistics information from the server\n\t\t@rtype: L{XmmsResult}\n\t\t@return: The result of the operation.\n\t\t";
 static PyObject *__pyx_pf_7xmmsapi_7XmmsApi_89stats(PyObject *__pyx_v_self, PyObject *__pyx_args, PyObject *__pyx_kwds) {
   PyObject *__pyx_v_cb = 0;
   PyObject *__pyx_r = NULL;
--- a/src/clients/lib/python/xmmsapi.pyx
+++ b/src/clients/lib/python/xmmsapi.pyx
@@ -1661,7 +1661,7 @@
 		"""
 		stats(cb=None) -> XmmsResult
 
-		Get statistics informations from the server
+		Get statistics information from the server
 		@rtype: L{XmmsResult}
 		@return: The result of the operation.
 		"""
--- a/src/clients/lib/xmmsclient/stats.c
+++ b/src/clients/lib/xmmsclient/stats.c
@@ -27,7 +27,7 @@
 /**
  * @defgroup StatsFunctions StatsFunctions
  * @ingroup XMMSClient
- * @brief This module contains functions to access informations and statistics of the XMMS server.
+ * @brief This module contains functions to access information and statistics of the XMMS server.
  *
  * @{
  */
--- a/src/clients/lib/python/xmmsvalue.c
+++ b/src/clients/lib/python/xmmsvalue.c
@@ -879,7 +879,7 @@
 static char __pyx_k_33[] = "Index out of range";
 static char __pyx_k_40[] = "Bad collection";
 static char __pyx_k_44[] = "All Media";
-static char __pyx_k_45[] = "Unkown collection type";
+static char __pyx_k_45[] = "Unknown collection type";
 static char __pyx_k_47[] = "";
 static char __pyx_k_49[] = "Unable to parse the pattern";
 static char __pyx_k_51[] = "\nPython bindings for xmmsv manipulations.\n";
@@ -14193,7 +14193,7 @@
  * 	elif colltype == XMMS_COLLECTION_TYPE_PARTYSHUFFLE:
  * 		c.__class__ = PShuffle             # <<<<<<<<<<<<<<
  * 	else:
- * 		raise RuntimeError("Unkown collection type")
+ * 		raise RuntimeError("Unknown collection type")
  */
     __pyx_t_2 = __Pyx_GetName(__pyx_m, __pyx_n_s__PShuffle); if (unlikely(!__pyx_t_2)) {__pyx_filename = __pyx_f[0]; __pyx_lineno = 795; __pyx_clineno = __LINE__; goto __pyx_L1_error;}
     __Pyx_GOTREF(__pyx_t_2);
@@ -14206,7 +14206,7 @@
     /* "xmmsvalue.pyx":797
  * 		c.__class__ = PShuffle
  * 	else:
- * 		raise RuntimeError("Unkown collection type")             # <<<<<<<<<<<<<<
+ * 		raise RuntimeError("Unknown collection type")             # <<<<<<<<<<<<<<
  * 
  * 	c.set_collection(coll)
  */
@@ -14219,7 +14219,7 @@
   __pyx_L3:;
 
   /* "xmmsvalue.pyx":799
- * 		raise RuntimeError("Unkown collection type")
+ * 		raise RuntimeError("Unknown collection type")
  * 
  * 	c.set_collection(coll)             # <<<<<<<<<<<<<<
  * 
@@ -17027,7 +17027,7 @@
   /* "xmmsvalue.pyx":797
  * 		c.__class__ = PShuffle
  * 	else:
- * 		raise RuntimeError("Unkown collection type")             # <<<<<<<<<<<<<<
+ * 		raise RuntimeError("Unknown collection type")             # <<<<<<<<<<<<<<
  * 
  * 	c.set_collection(coll)
  */
--- a/src/clients/lib/python/xmmsvalue.pyx
+++ b/src/clients/lib/python/xmmsvalue.pyx
@@ -794,7 +794,7 @@
 	elif colltype == XMMS_COLLECTION_TYPE_PARTYSHUFFLE:
 		c.__class__ = PShuffle
 	else:
-		raise RuntimeError("Unkown collection type")
+		raise RuntimeError("Unknown collection type")
 
 	c.set_collection(coll)
 
