Description: Create a symlink from nyxmms2.1 to xmms2.1.
Author: Benjamin Drung <bdrung@debian.org>
Bug: http://bugs.xmms2.org/view.php?id=2475

--- a/src/clients/nycli/wscript
+++ b/src/clients/nycli/wscript
@@ -35,6 +35,8 @@
 
     bld.symlink_as('${PREFIX}/bin/nyxmms2', 'xmms2')
 
+    bld.symlink_as('${PREFIX}/share/man/man1/nyxmms2.1.gz', 'xmms2.1.gz')
+
 
 def configure(conf):
     if Options.platform == 'win32':
