Description: Do not provide facebook.png
Bug-Debian: http://bugs.debian.org/609168
Author: Iulian Udrea <iulian@ubuntu.com>

Index: pidgin-facebookchat-1.69/Makefile
===================================================================
--- pidgin-facebookchat-1.69.orig/Makefile	2011-02-03 13:01:21.575381001 +0000
+++ pidgin-facebookchat-1.69/Makefile	2011-02-03 13:03:30.987381002 +0000
@@ -59,9 +59,9 @@
 	cp libfacebook64.so /usr/lib64/purple-2/
 	cp libfacebookarm.so /usr/lib/pidgin/
 	cp libfacebookppc.so /usr/lib/purple-2/
-	cp facebook16.png /usr/share/pixmaps/pidgin/protocols/16/facebook.png
-	cp facebook22.png /usr/share/pixmaps/pidgin/protocols/22/facebook.png
-	cp facebook48.png /usr/share/pixmaps/pidgin/protocols/48/facebook.png
+	#cp facebook16.png /usr/share/pixmaps/pidgin/protocols/16/facebook.png
+	#cp facebook22.png /usr/share/pixmaps/pidgin/protocols/22/facebook.png
+	#cp facebook48.png /usr/share/pixmaps/pidgin/protocols/48/facebook.png
 
 installers:	pidgin-facebookchat.exe pidgin-facebookchat.deb pidgin-facebookchat.tar.bz2
 
@@ -110,9 +110,9 @@
 pidgin-facebookchat.deb:	libfacebook.so
 	echo "Dont forget to update version number"
 	cp libfacebook.so ${DEB_PACKAGE_DIR}/usr/lib/purple-2/
-	cp facebook16.png ${DEB_PACKAGE_DIR}/usr/share/pixmaps/pidgin/protocols/16/facebook.png
-	cp facebook22.png ${DEB_PACKAGE_DIR}/usr/share/pixmaps/pidgin/protocols/22/facebook.png
-	cp facebook48.png ${DEB_PACKAGE_DIR}/usr/share/pixmaps/pidgin/protocols/48/facebook.png
+	#cp facebook16.png ${DEB_PACKAGE_DIR}/usr/share/pixmaps/pidgin/protocols/16/facebook.png
+	#cp facebook22.png ${DEB_PACKAGE_DIR}/usr/share/pixmaps/pidgin/protocols/22/facebook.png
+	#cp facebook48.png ${DEB_PACKAGE_DIR}/usr/share/pixmaps/pidgin/protocols/48/facebook.png
 
 pidgin-facebookchat.tar.bz2:	pidgin-facebookchat.deb
 	tar --bzip2 --directory ${DEB_PACKAGE_DIR} -cf $@ usr/
