--- ./swig/ruby/ruby_wrap.c_orig	2009-02-17 18:30:40.000000000 +0000
+++ ./swig/ruby/ruby_wrap.c	2012-07-18 06:43:04.000000000 +0000
@@ -1309,7 +1309,7 @@
 /* Error manipulation */
 
 #define SWIG_ErrorType(code)                            SWIG_Ruby_ErrorType(code)               
-#define SWIG_Error(code, msg)            		rb_raise(SWIG_Ruby_ErrorType(code), msg)
+#define SWIG_Error(code, msg)            		rb_raise(SWIG_Ruby_ErrorType(code), "%s", msg)
 #define SWIG_fail                        		goto fail				 
 
 
