Description: libss7: FTBFS with clang 3.1
Bug: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=686546
 
Index: libss7-1.0.2/Makefile
===================================================================
--- libss7-1.0.2.orig/Makefile	2008-08-07 04:49:06.000000000 +1000
+++ libss7-1.0.2/Makefile	2012-10-06 17:05:47.000000000 +1000
@@ -1,4 +1,4 @@
-CC=gcc
+CC?=gcc
 GREP=grep
 AWK=awk
 
Index: libss7-1.0.2/isup.c
===================================================================
--- libss7-1.0.2.orig/isup.c	2008-08-10 02:54:23.000000000 +1000
+++ libss7-1.0.2/isup.c	2012-10-06 17:07:25.000000000 +1000
@@ -1394,7 +1394,7 @@
 
 	switch ((parm[1] >> 4) & 0xf) {
 		case 0:
-			orig_redir_reas = "Unknown/not available";
+			redir_reas = "Unknown/not available";
 			break;
 		case 1:
 			redir_reas = "User busy";
Index: libss7-1.0.2/mtp3.c
===================================================================
--- libss7-1.0.2.orig/mtp3.c	2008-08-06 08:27:45.000000000 +1000
+++ libss7-1.0.2/mtp3.c	2012-10-06 17:06:20.000000000 +1000
@@ -614,7 +614,7 @@
 	int i;
 
 	for (i = 0; i < ss7->numlinks; i++) {
-		if ((ss7->mtp2_linkstate[i] == MTP2_LINKSTATE_DOWN)) {
+		if (ss7->mtp2_linkstate[i] == MTP2_LINKSTATE_DOWN) {
 			mtp2_start(ss7->links[i], 1);
 			ss7->mtp2_linkstate[i] = MTP2_LINKSTATE_ALIGNING;
 		}
