--- a/startpar/makeboot.c	(revision 1961)
+++ b/startpar/makeboot.c	(working copy)
@@ -485,10 +485,10 @@
 		}
 	}
 	printf("\"\n");
-	printf("skipped_service=\"");
+	printf("skipped_service_not_installed=\"");
 	i = r = 0;
 	for (i = 0; i < tree_entries; i++) {
-		if (resvec[i] == 5 || resvec[i] == 6) {
+		if (resvec[i] == 5) {
 			if (r)
 				printf(" ");
 			printf("%s", nodevec[i]->name);
@@ -496,6 +496,17 @@
 		}
 	}
 	printf("\"\n");
+	printf("skipped_service_not_configured=\"");
+	i = r = 0;
+	for (i = 0; i < tree_entries; i++) {
+		if (resvec[i] == 6) {
+			if (r)
+				printf(" ");
+			printf("%s", nodevec[i]->name);
+			r++;
+		}
+	}
+	printf("\"\n");
 }
 
 #if DEBUG
