Description: Build with -fvisiblity=hidden.
Author: Felix Geyer <debfx-pkg@fobos.de>
Forwarded: yes, http://quickgit.kde.org/?p=libqzeitgeist.git&a=commit&h=37edd54b446aa754e3ef9ecc8c0c7c861abeb127

--- libqzeitgeist-0.7.0.orig/CMakeLists.txt
+++ libqzeitgeist-0.7.0/CMakeLists.txt
@@ -7,6 +7,13 @@ set(CMAKE_MODULE_PATH ${CMAKE_SOURCE_DIR
 
 find_package(Qt4 4.7.0 COMPONENTS QtCore QtDBus QtTest REQUIRED)
 
+include(CheckCXXCompilerFlag)
+
+check_cxx_compiler_flag(-fvisibility=hidden __HAVE_GCC_VISIBILITY)
+if (__HAVE_GCC_VISIBILITY AND NOT WIN32)
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fvisibility=hidden")
+endif (__HAVE_GCC_VISIBILITY AND NOT WIN32)
+
 
 set(CPACK_PACKAGE_VERSION_MAJOR "0")
 set(CPACK_PACKAGE_VERSION_MINOR "7")
--- libqzeitgeist-0.7.0.orig/include/QtZeitgeist/qtzeitgeist.h
+++ libqzeitgeist-0.7.0/include/QtZeitgeist/qtzeitgeist.h
@@ -22,6 +22,8 @@
 #ifndef QTZEITGEIST_H_
 #define QTZEITGEIST_H_
 
+#include <QtGlobal>
+
 namespace QtZeitgeist
 {
 
@@ -33,7 +35,7 @@ namespace QtZeitgeist
  * registering custom types.
  * Note: It's really important to call this function before using the library.
  */
-void init();
+Q_DECL_EXPORT void init();
 
 };
 #endif // QTZEITGEIST_H_
--- libqzeitgeist-0.7.0.orig/include/QtZeitgeist/DataModel/datasource.h
+++ libqzeitgeist-0.7.0/include/QtZeitgeist/DataModel/datasource.h
@@ -145,6 +145,11 @@ private:
             DataSource &datasource);
 };
 
+Q_DECL_EXPORT QDBusArgument & operator << (QDBusArgument &argument,
+               const DataSource &datasource);
+Q_DECL_EXPORT const QDBusArgument & operator >> (const QDBusArgument &argument,
+               DataSource &datasource);
+
 // Create additional types.
 typedef QList<DataSource> DataSourceList;
 
--- libqzeitgeist-0.7.0.orig/include/QtZeitgeist/DataModel/event.h
+++ libqzeitgeist-0.7.0/include/QtZeitgeist/DataModel/event.h
@@ -219,6 +219,11 @@ private:
             Event &event);
 };
 
+Q_DECL_EXPORT QDBusArgument & operator << (QDBusArgument &argument,
+               const Event &event);
+Q_DECL_EXPORT const QDBusArgument & operator >> (const QDBusArgument &argument,
+               Event &event);
+
 // Create additional types.
 typedef QList<Event> EventList;
 typedef QList<quint32> EventIdList;
--- libqzeitgeist-0.7.0.orig/include/QtZeitgeist/DataModel/timerange.h
+++ libqzeitgeist-0.7.0/include/QtZeitgeist/DataModel/timerange.h
@@ -141,6 +141,11 @@ private:
             TimeRange &event);
 };
 
+Q_DECL_EXPORT QDBusArgument & operator << (QDBusArgument &argument,
+               const TimeRange &event);
+Q_DECL_EXPORT const QDBusArgument & operator >> (const QDBusArgument &argument,
+               TimeRange &event);
+
 };
 
 };
