Index: xutils-dev-1.0.2/xorg-cf-files/linux.cf
===================================================================
--- xutils-dev-1.0.2.orig/xorg-cf-files/linux.cf	2006-04-16 16:53:09.000000000 +0800
+++ xutils-dev-1.0.2/xorg-cf-files/linux.cf	2006-04-16 16:54:41.000000000 +0800
@@ -86,6 +86,19 @@
 # define BuildPDFdocs NO
 #endif
 
+#ifndef ProjectRoot
+# define ProjectRoot   /usr
+#endif
+#ifndef ManDirectoryRoot
+# define ManDirectoryRoot      /usr/share/man
+#endif
+#ifndef AlternateUsrLibDir
+# define AlternateUsrLibDir NO
+#endif
+#ifndef AlternateIncRoot
+# define AlternateIncRoot NO
+#endif
+
 #ifndef LinuxBinUtilsMajorVersion
 # define LinuxBinUtilsMajorVersion	DefaultLinuxBinUtilsMajorVersion
 #endif
