Index: xutils-dev/xorg-cf-files/X11.tmpl
===================================================================
--- xutils-dev.orig/xorg-cf-files/X11.tmpl	2006-12-19 20:55:05.000000000 -0500
+++ xutils-dev/xorg-cf-files/X11.tmpl	2006-12-19 20:56:03.000000000 -0500
@@ -1512,10 +1512,10 @@
 #define DocPdfDir $(DOCDIR)/PDF
 #endif
 #ifndef FontDir
-#define FontDir $(LIBDIR)/fonts
+#define FontDir /usr/share/fonts/X11
 #endif
 #ifndef FontEncDir
-#define FontEncDir $(LIBDIR)/fonts/encodings
+#define FontEncDir /usr/share/fonts/X11/encodings
 #endif
 #ifndef AdmDir
 #define AdmDir /usr/adm
