Description: spelling mistake
Origin: vendor
Bug: https://rt.cpan.org/Ticket/Display.html?id=77720
Forwarded: https://rt.cpan.org/Ticket/Display.html?id=77720
Author: gregor herrmann <gregoa@debian.org>
Last-Update: 2012-06-09

--- a/lib/Sub/Exporter/Util.pm
+++ b/lib/Sub/Exporter/Util.pm
@@ -223,7 +223,7 @@
 
 This module provides a number of utility functions for performing common or
 useful operations when setting up a Sub::Exporter configuration.  All of the
-utilites may be exported, but none are by default.
+utilities may be exported, but none are by default.
 
 =head1 THE UTILITIES
 
