Description: config universal variables
 Eliminate non-universal variables.
Author: Obey Arthur Liu <arthur@milliways.fr>
Last-Update: 2012-05-23
--- a/binsrc/samples/demo/default-virtuoso.ini
+++ b/binsrc/samples/demo/default-virtuoso.ini
@@ -151,7 +151,7 @@ VDBDisconnectTimeout		= 1000
 KeepConnectionOnFixedThread	= 0
 
 [Replication]
-ServerName			= @REPLNAME@
+ServerName			= db
 ServerEnable			= 1
 QueueMax			= 50000
 
@@ -203,7 +203,7 @@ Segment2			= 100M, db-seg2-1.db
 
 
 [Zero Config]
-ServerName			= @ZNAME@
+ServerName			= virtuoso
 ;ServerDSN			= ZDSN
 ;SSLServerName			= 
 ;SSLServerDSN			= 
