--- a/binsrc/VirtuosoClient.Net/Makefile.mono
+++ b/binsrc/VirtuosoClient.Net/Makefile.mono
@@ -86,7 +86,7 @@ RESOURCES= \
 
 ifeq  (ADONET2, $(ADONETVERSION))
 
-CSC		= gmcs
+CSC		= mono-csc
 
 #CSFLAGS	= -debug+ /d:MONO /d:DEBUG /d:DEBUG_IO /d:ADONET2
 CSFLAGS		= /optimize+ /d:MONO /d:ADONET2
@@ -101,7 +101,7 @@ REFERENCES	= /r:System.Data /r:System.En
 
 else
 
-CSC		= mcs
+CSC		= mono-csc
 
 #CSFLAGS	= -debug+ /d:MONO /d:DEBUG /d:DEBUG_IO
 CSFLAGS		= /optimize+ /d:MONO
