diff -NurbBp a/configure.in b/configure.in
--- a/configure.in	2009-12-05 08:26:45.000000000 +0100
+++ b/configure.in	2009-12-20 14:17:56.000000000 +0100
@@ -524,7 +524,6 @@ AC_CONFIG_FILES([Makefile
                  include/vigra_ext/Makefile
                  src/Makefile
                  src/vigra_impex/Makefile
-                 src/win32helpers/Makefile
                  doc/Makefile])
 AC_OUTPUT
 
diff -NurbBp a/src/Makefile.am b/src/Makefile.am
--- a/src/Makefile.am	2009-12-02 08:27:29.000000000 +0100
+++ b/src/Makefile.am	2009-12-20 14:18:42.000000000 +0100
@@ -3,7 +3,7 @@ AM_CFLAGS = $(EXTRACFLAGS)
 AM_CXXFLAGS = $(EXTRACXXFLAGS)
 AM_LDFLAGS = $(EXTRALDFLAGS)
 
-SUBDIRS = vigra_impex win32helpers
+SUBDIRS = vigra_impex
 
 bin_PROGRAMS = enblend enfuse
 
