diff -NurbBp a/m4/ax_check_glut.m4 b/m4/ax_check_glut.m4
--- a/m4/ax_check_glut.m4	2009-09-06 15:14:31.000000000 +0200
+++ b/m4/ax_check_glut.m4	2009-12-20 14:23:34.000000000 +0100
@@ -72,12 +72,6 @@ m4_define([AX_CHECK_GLUT_PROGRAM],
 # endif]],
                            [[glutMainLoop()]])])
 
-#
-# If X is present, assume GLUT depends on it.
-#
-AS_IF([test X$no_x != Xyes],
-      [GLUT_LIBS="${X_PRE_LIBS} -lXmu -lXi ${X_EXTRA_LIBS} ${GLUT_LIBS}"])
-
 AC_CACHE_CHECK([for GLUT library], [ax_cv_check_glut_libglut],
 [ax_cv_check_glut_libglut="no"
 AC_LANG_PUSH(C)
