--- libxml-dom-perl.orig/t/dom_jp_attr.t
+++ libxml-dom-perl/t/dom_jp_attr.t
@@ -90,7 +90,9 @@
 my $doc = $parser->parse ($str);
 assert_ok (not $@);
 
+use bytes;
 my $out = $doc->toString;
+use utf8;
 $out =~ tr/\012/\n/;
 $out =~ s/(\&\#(\d+);)/sprintf("%s",charRef2U8($2))/eg;
 assert_ok ($out eq $str);
--- libxml-dom-perl.orig/t/dom_jp_cdata.t
+++ libxml-dom-perl/t/dom_jp_cdata.t
@@ -44,7 +44,9 @@
 my $doc = $parser->parse ($str);
 assert_ok (not $@);
 
+use bytes;
 my $out = $doc->toString;
+use utf8;
 $out =~ tr/\012/\n/;
 my $result = ($XML::Parser::VERSION >= 2.19) ? $str : $oldStr;
 assert_ok ($out eq $result);
--- libxml-dom-perl.orig/t/dom_jp_minus.t
+++ libxml-dom-perl/t/dom_jp_minus.t
@@ -46,6 +46,8 @@
 my $parser = new XML::DOM::Parser;
 my $doc = $parser->parse ($str);
 
+use bytes;
 my $out = $doc->toString;
+use utf8;
 $out =~ tr/\012/\n/;
 assert_ok ($out eq $str);
--- libxml-dom-perl.orig/t/dom_jp_modify.t
+++ libxml-dom-perl/t/dom_jp_modify.t
@@ -180,4 +180,5 @@
 </ケニー></サウスパーク>
 END
 
+use bytes;
 assert_ok ($str eq $end);
--- libxml-dom-perl.orig/t/dom_jp_print.t
+++ libxml-dom-perl/t/dom_jp_print.t
@@ -83,7 +83,9 @@
 
 my $parser = new XML::DOM::Parser (NoExpand => 1);
 my $doc = $parser->parse ($str);
+use bytes;
 my $out = $doc->toString;
+use utf8;
 $out =~ tr/\012/\n/;
 $out =~ s/(\&\#(\d+);)/sprintf("%s",charRef2U8($2))/eg;
 
