Description: Sanitise username when used in dbus object paths
Author: Christopher James Halse Rogers <raof@ubuntu.com>
Bug: https://bugs.freedesktop.org/show_bug.cgi?id=51031
Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=675852
Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/+source/colord/+bug/1021374

Index: colord/src/cd-device.c
===================================================================
--- colord.orig/src/cd-device.c	2012-09-13 14:06:46.476688685 +1000
+++ colord/src/cd-device.c	2012-09-13 14:08:33.172692354 +1000
@@ -244,19 +244,21 @@
 	gchar *path_owner;
 	struct passwd *pw;
 
-	/* make sure object path is sane */
-	path_tmp = cd_main_ensure_dbus_path (device->priv->id);
-
 	/* append the uid to the object path */
 	pw = getpwuid (device->priv->owner);
 	if (device->priv->owner == 0 ||
 	    g_strcmp0 (pw->pw_name, DAEMON_USER) == 0) {
-		path_owner = g_strdup (path_tmp);
+		path_tmp = g_strdup (device->priv->id);
 	} else {
-		path_owner = g_strdup_printf ("%s_%s",
-					      path_tmp,
-					      pw->pw_name);
+		path_tmp = g_strdup_printf ("%s_%s_%d",
+					    device->priv->id,
+					    pw->pw_name,
+					    device->priv->owner);
 	}
+
+	/* make sure object path is sane */
+	path_owner = cd_main_ensure_dbus_path (path_tmp);
+
 	device->priv->object_path = g_build_filename (COLORD_DBUS_PATH,
 						      "devices",
 						      path_owner,
Index: colord/src/cd-profile.c
===================================================================
--- colord.orig/src/cd-profile.c	2012-09-13 14:06:46.476688685 +1000
+++ colord/src/cd-profile.c	2012-09-13 14:08:33.172692354 +1000
@@ -171,19 +171,21 @@
 	gchar *path_owner;
 	struct passwd *pw;
 
-	/* make sure object path is sane */
-	path_tmp = cd_main_ensure_dbus_path (profile->priv->id);
 
 	/* append the uid to the object path */
 	pw = getpwuid (profile->priv->owner);
 	if (profile->priv->owner == 0 ||
 	    g_strcmp0 (pw->pw_name, DAEMON_USER) == 0) {
-		path_owner = g_strdup (path_tmp);
+		path_tmp = g_strdup (profile->priv->id);
 	} else {
-		path_owner = g_strdup_printf ("%s_%s",
-					      path_tmp,
-					      pw->pw_name);
+		path_tmp = g_strdup_printf ("%s_%s_%d",
+					    profile->priv->id,
+					    pw->pw_name,
+					    profile->priv->owner);
 	}
+	/* make sure object path is sane */
+	path_owner = cd_main_ensure_dbus_path (path_tmp);
+
 	profile->priv->object_path = g_build_filename (COLORD_DBUS_PATH,
 						       "profiles",
 						       path_owner,
