diff --git a/invesalius/constants.py b/invesalius/constants.py
index 3dfc5df..610c3bd 100644
--- a/invesalius/constants.py
+++ b/invesalius/constants.py
@@ -295,7 +295,7 @@ WINDOW_LEVEL = {_("Abdomen"):(350,50),
 REDUCE_IMAGEDATA_QUALITY = 0
 
 ICON_DIR = os.path.abspath(os.path.join('..', 'icons'))
-SAMPLE_DIR = os.path.abspath(os.path.join('..', 'samples'))
+SAMPLE_DIR = "/usr/share/doc/invesalius-examples/examples/"
 DOC_DIR = os.path.abspath(os.path.join('..', 'docs'))
 
 
