Index: timidity-2.13.2/debian/timidity-daemon.timidity.init
===================================================================
--- timidity-2.13.2.orig/debian/timidity-daemon.timidity.init	2010-05-21 11:39:31.000000000 +0200
+++ timidity-2.13.2/debian/timidity-daemon.timidity.init	2010-05-21 11:40:08.000000000 +0200
@@ -72,7 +72,7 @@
 	log_begin_msg "Starting $DESC..."
     mkdir -p /var/run/timidity
     chown timidity /var/run/timidity
-	if start-stop-daemon ${START} >/dev/null; then
+	if start-stop-daemon ${START} < /dev/null >/dev/null 2>&1; then
 		log_end_msg 0
 	else
 		log_end_msg 1
