Index: libunwind-1.0.1/src/ptrace/_UPT_get_dyn_info_list_addr.c
===================================================================
--- libunwind-1.0.1.orig/src/ptrace/_UPT_get_dyn_info_list_addr.c	2011-09-11 05:06:41.000000000 +0200
+++ libunwind-1.0.1/src/ptrace/_UPT_get_dyn_info_list_addr.c	2012-07-07 13:36:27.632998530 +0200
@@ -62,7 +62,7 @@
 
       Debug (16, "checking object %s\n", path);
 
-      di = _UPTi_find_unwind_table (ui, as, path, lo, off);
+      di = _UPTi_find_unwind_table (ui, as, path, lo, off, 0);
       if (di)
 	{
 	  res = _Uia64_find_dyn_list (as, di, arg);
Index: libunwind-1.0.1/src/ptrace/_UPT_internal.h
===================================================================
--- libunwind-1.0.1.orig/src/ptrace/_UPT_internal.h	2011-09-11 05:06:41.000000000 +0200
+++ libunwind-1.0.1/src/ptrace/_UPT_internal.h	2012-07-07 13:37:07.880999763 +0200
@@ -64,7 +64,7 @@
 
 extern int _UPT_reg_offset[UNW_REG_LAST + 1];
 
-extern int _UPTi_find_unwind_table (struct UPT_info *ui,
+extern unw_dyn_info_t *UPTi_find_unwind_table (struct UPT_info *ui,
 				    unw_addr_space_t as,
 				    char *path,
 				    unw_word_t segbase,
