Index: pkg-libunwind/tests/Makefile.am
===================================================================
--- pkg-libunwind.orig/tests/Makefile.am	2012-04-22 21:58:15.000000000 +0900
+++ pkg-libunwind/tests/Makefile.am	2012-04-22 21:59:54.000000000 +0900
@@ -128,21 +128,21 @@
 
 test_async_sig_LDADD = $(LIBUNWIND_local) -lpthread
 test_flush_cache_LDADD = $(LIBUNWIND_local)
-test_init_remote_LDADD = $(LIBUNWIND)
-test_mem_LDADD = $(LIBUNWIND)
+test_init_remote_LDADD = $(LIBUNWIND) $(LIBUNWIND_local)
+test_mem_LDADD = $(LIBUNWIND) $(LIBUNWIND_local)
 test_ptrace_LDADD = $(LIBUNWIND_ptrace) $(LIBUNWIND)
-test_proc_info_LDADD = $(LIBUNWIND)
-test_static_link_LDADD = $(LIBUNWIND)
-test_strerror_LDADD = $(LIBUNWIND)
-rs_race_LDADD = $(LIBUNWIND) -lpthread
+test_proc_info_LDADD = $(LIBUNWIND) $(LIBUNWIND_local)
+test_static_link_LDADD = $(LIBUNWIND) $(LIBUNWIND_local)
+test_strerror_LDADD = $(LIBUNWIND) $(LIBUNWIND_local)
+rs_race_LDADD = $(LIBUNWIND) $(LIBUNWIND_local) -lpthread
 test_varargs_LDADD = @BACKTRACELIB@
 
-Gtest_bt_LDADD = $(LIBUNWIND)
-Gtest_concurrent_LDADD = $(LIBUNWIND) -lpthread
+Gtest_bt_LDADD = $(LIBUNWIND) $(LIBUNWIND_local)
+Gtest_concurrent_LDADD = $(LIBUNWIND) $(LIBUNWIND_local) -lpthread
 Gtest_dyn1_LDADD = $(LIBUNWIND) $(LIBUNWIND_local)
-Gtest_exc_LDADD = $(LIBUNWIND)
-Gtest_init_LDADD = $(LIBUNWIND)
-Gtest_resume_sig_LDADD = $(LIBUNWIND)
+Gtest_exc_LDADD = $(LIBUNWIND) $(LIBUNWIND_local)
+Gtest_init_LDADD = $(LIBUNWIND) $(LIBUNWIND_local)
+Gtest_resume_sig_LDADD = $(LIBUNWIND) $(LIBUNWIND_local)
 Gperf_simple_LDADD = $(LIBUNWIND) $(LIBUNWIND_local)
 Gtest_trace_LDADD=$(LIBUNWIND) $(LIBUNWIND_local)
 Gperf_trace_LDADD = $(LIBUNWIND) $(LIBUNWIND_local)
