diff -Nur -x '*.orig' -x '*~' cpptest-1.1.0/Makefile.in cpptest-1.1.0.new/Makefile.in
--- cpptest-1.1.0/Makefile.in	2008-07-17 15:27:24.000000000 +0200
+++ cpptest-1.1.0.new/Makefile.in	2009-11-30 10:38:01.643135914 +0100
@@ -195,7 +195,7 @@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 x_doxygen = @x_doxygen@
-SUBDIRS = src test $(am__append_1)
+SUBDIRS = src $(am__append_1)
 
 ####### Distribution
 EXTRA_DIST = \
