Author: Elimar Riesebieter <riesebie@lxtec.de>
Description: Expand the as10k1 example Makefile with a distclean target.
Forwarded: no

Index: alsa-tools-1.0.25/as10k1/examples/Makefile
===================================================================
--- alsa-tools-1.0.25.orig/as10k1/examples/Makefile	2012-05-19 16:57:36.304424605 +0200
+++ alsa-tools-1.0.25/as10k1/examples/Makefile	2012-05-19 16:57:53.712423717 +0200
@@ -12,6 +12,8 @@
 clean:
 	rm -f *~ *.emu10k1
 
+distclean: clean
+
 install:
 
 distdir:
