Author: Elimar Riesebieter <riesebie@lxtec.de>
Description: Make usx2yloader work with udev.
Last-Update: 12-09-2007

Index: alsa-tools-1.0.25/usx2yloader/Makefile.am
===================================================================
--- alsa-tools-1.0.25.orig/usx2yloader/Makefile.am	2012-05-19 16:57:36.464424598 +0200
+++ alsa-tools-1.0.25/usx2yloader/Makefile.am	2012-05-19 16:57:52.992423754 +0200
@@ -11,8 +11,8 @@
 
 EXTRA_DIST = gitcompile depcomp tascam_fw.usermap tascam_fw.in tascam_fpga.in
 
-hotplug_files = tascam_fw.usermap tascam_fw tascam_fpga
-hotplugdir = /etc/hotplug/usb
+hotplug_files = tascam_fw tascam_fpga
+hotplugdir = /lib/udev
 hotplug_SCRIPTS = $(hotplug_files)
 
 
Index: alsa-tools-1.0.25/usx2yloader/Makefile.in
===================================================================
--- alsa-tools-1.0.25.orig/usx2yloader/Makefile.in	2012-05-19 16:57:36.464424598 +0200
+++ alsa-tools-1.0.25/usx2yloader/Makefile.in	2012-05-19 16:57:52.992423754 +0200
@@ -193,8 +193,8 @@
 AM_CFLAGS = -DDATAPATH=\"$(datadir)/alsa/firmware/$(MYNAME)\"
 usx2yloader_SOURCES = usx2yloader.c
 EXTRA_DIST = gitcompile depcomp tascam_fw.usermap tascam_fw.in tascam_fpga.in
-hotplug_files = tascam_fw.usermap tascam_fw tascam_fpga
-hotplugdir = /etc/hotplug/usb
+hotplug_files = tascam_fw tascam_fpga
+hotplugdir = /lib/udev
 hotplug_SCRIPTS = $(hotplug_files)
 all: all-am
 
