--- a/configure.ac
+++ b/configure.ac
@@ -414,7 +414,7 @@ AC_ARG_ENABLE([gforce],
   AS_HELP_STRING([--disable-gforce],
     [Do not build the G-Force plugin @<:@default=enabled@:>@]),
     [ENABLE_GFORCE=$enableval],
-    [ENABLE_GFORCE=yes])
+    [ENABLE_GFORCE=no])
 
 if test "$ENABLE_GFORCE" = "yes"; then
   build_actor_plugins="$build_actor_plugins G-Force"
@@ -558,28 +558,6 @@ AC_CONFIG_FILES([
 	plugins/actor/nastyfft/Makefile
 	plugins/actor/oinksie/Makefile
 	plugins/actor/pseudotoad_flower/Makefile
-	plugins/actor/G-Force/Makefile
-	plugins/actor/G-Force/docs/Makefile
-	plugins/actor/G-Force/Common/Makefile
-	plugins/actor/G-Force/Common/GeneralTools/Makefile
-	plugins/actor/G-Force/Common/GeneralTools/Headers/Makefile
-	plugins/actor/G-Force/Common/io/Makefile
-	plugins/actor/G-Force/Common/io/Headers/Makefile
-	plugins/actor/G-Force/Common/math/Makefile
-	plugins/actor/G-Force/Common/math/Headers/Makefile
-	plugins/actor/G-Force/Common/UI/Makefile
-	plugins/actor/G-Force/Common/UI/Headers/Makefile
-	plugins/actor/G-Force/GForceCommon/Makefile
-	plugins/actor/G-Force/GForceCommon/Headers/Makefile
-	plugins/actor/G-Force/unix/Makefile
-	plugins/actor/G-Force/unix/Headers/Makefile
-	plugins/actor/G-Force/unix/libmfl/Makefile
-	plugins/actor/G-Force/unix/libvisual/Makefile
-	plugins/actor/G-Force/GForceColorMaps/Makefile
-	plugins/actor/G-Force/GForceDeltaFields/Makefile
-	plugins/actor/G-Force/GForceParticles/Makefile
-	plugins/actor/G-Force/GForceWaveShapes/Makefile
-	plugins/actor/G-Force/NotWorkingWaveShapes/Makefile
 	plugins/input/Makefile
 	plugins/input/esd/Makefile
 	plugins/input/alsa/Makefile
--- a/plugins/actor/Makefile.am
+++ b/plugins/actor/Makefile.am
@@ -4,7 +4,6 @@ SUBDIRS = $(build_actor_plugins)
 DIST_SUBDIRS = \
   bumpscope \
   corona \
-  G-Force \
   gdkpixbuf \
   gstreamer \
   infinite \
