Index: debian/Makefile.am
===================================================================
--- debian.orig/Makefile.am
+++ debian/Makefile.am
@@ -1,5 +1,7 @@
 ## Process this file with automake to generate a Makefile.in
 
+ACLOCAL_AMFLAGS = -I m4
+
 SUBDIRS = plugins po
 
 EXTRA_DIST = libvisual-plugins.spec gettext.h po/Makevars.in
Index: debian/configure.ac
===================================================================
--- debian.orig/configure.ac
+++ debian/configure.ac
@@ -13,6 +13,7 @@
 
 AC_CONFIG_SRCDIR([config.h.in])
 AC_CONFIG_HEADER([config.h])
+AC_CONFIG_MACRO_DIR([m4])
 
 m4_define([libvisual_required_version], [lv_plugins_version])
 m4_define([esound_required_version], [0.2.28])
