Description: Avoids the useless udev rules reload
Author: Didier Raboud <didier@raboud.com>
Origin: vendor
Forwarded: yes (private mail)
Last-Update: 2011-06-20
--- a/Makefile
+++ b/Makefile
@@ -12,9 +12,9 @@
 
 clean:
 
-install: files-install db-install rules-reload
+install: files-install db-install
 
-install-packed: files-install db-install-packed rules-reload
+install-packed: files-install db-install-packed
 
 files-install:
 	install -d $(PREFIX)/share/usb_modeswitch
@@ -50,7 +50,7 @@
 		fi \
 	fi
 
-uninstall: files-uninstall rules-reload
+uninstall: files-uninstall
 
 files-uninstall:
 	$(RM) $(RULESDIR)/40-usb_modeswitch.rules
