From: Stephane Glondu <steph@glondu.net>
Date: Mon, 21 Jun 2010 13:54:48 -0700
Subject: Fix build on bytecode architectures

Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=586619
Signed-off-by: Stephane Glondu <steph@glondu.net>
---
 src/Makefile |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/Makefile b/src/Makefile
index 3f53e3d..e1f77e1 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -17,7 +17,7 @@ gfx_MODULES    = sdlgfx
 
 TARGETS  = sdl.cma $(patsubst %,sdl%.cma,$(PARTS))
 TARGETS += libsdlstub.$(A) $(patsubst %,libsdl%stub.$(A),$(PARTS))
-ifdef OCAMLOPT
+ifneq ($(OCAMLOPT),no)
 TARGETS += sdl.cmxa $(patsubst %,sdl%.cmxa,$(PARTS))
 endif
 
@@ -87,7 +87,7 @@ ifdef OCAMLFIND
 install-findlib :
 	$(OCAMLFIND) install sdl ../META \
           *.cma lib*.$(A) *.cmi *.mli \
-          $(if $(OCAMLOPT),*.cmxa sdl*.$(A) *.cmx) \
+          $(if $(filter-out no,$(OCAMLOPT)),*.cmxa sdl*.$(A) *.cmx) \
           $(if $(OCAMLMKLIB),dll*.so)
 ifeq ($(PLATFORM),Apple)
 	$(RANLIB) $$($(OCAMLFIND) printconf destdir)/sdl/*.$(A)
-- 
