From: Mike Hommey <mh@glandium.org>
Date: Fri, 23 Mar 2012 09:57:04 +0100
Subject: Don't hardcode page size on ia64 or sparc

---
 memory/mozjemalloc/jemalloc.c |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/memory/mozjemalloc/jemalloc.c b/memory/mozjemalloc/jemalloc.c
index 631e736..ba78597 100644
--- a/memory/mozjemalloc/jemalloc.c
+++ b/memory/mozjemalloc/jemalloc.c
@@ -1091,7 +1091,9 @@ static unsigned		ncpus;
  * controlling the malloc behavior are defined as compile-time constants
  * for best performance and cannot be altered at runtime.
  */
+#if !defined(__ia64__) && !defined(__sparc__)
 #define MALLOC_STATIC_SIZES 1
+#endif
 
 #ifdef MALLOC_STATIC_SIZES
 
