Index: pango-1.30.0/pango/modules.c
===================================================================
--- pango-1.30.0.orig/pango/modules.c	2012-03-31 21:42:48.366087473 +0200
+++ pango-1.30.0/pango/modules.c	2012-03-31 21:42:49.782087424 +0200
@@ -530,6 +530,11 @@
   FILE *module_file;
 
   char *file_str =  pango_config_key_get ("Pango/ModuleFiles");
+  char *module_files_d_str = g_build_filename (pango_get_lib_subdirectory (),
+					       MODULE_VERSION,
+					       "module-files.d",
+					       NULL);
+  char *list_str;
   char **files;
   int n;
 
@@ -540,7 +545,12 @@
 				 "pango.modules",
 				 NULL);
 
-  files = pango_split_file_list (file_str);
+  list_str = g_strjoin (G_SEARCHPATH_SEPARATOR_S,
+			file_str,
+			module_files_d_str,
+			NULL);
+
+  files = pango_split_file_list (list_str);
 
   n = 0;
   while (files[n])
@@ -586,6 +596,8 @@
     }
 
   g_strfreev (files);
+  g_free (list_str);
+  g_free (module_files_d_str);
   g_free (file_str);
 
   dlloaded_engines = g_slist_reverse (dlloaded_engines);
