diff -ur asio-1.5.3/include/asio/ssl/impl/context.ipp asio/include/asio/ssl/impl/context.ipp
--- asio-1.5.3/include/asio/ssl/impl/context.ipp	2011-03-21 19:04:00.000000000 -0400
+++ asio/include/asio/ssl/impl/context.ipp	2013-03-13 00:17:09.189359760 -0400
@@ -39,13 +39,13 @@
   switch (m)
   {
   case context::sslv2:
-    handle_ = ::SSL_CTX_new(::SSLv2_method());
+    //handle_ = ::SSL_CTX_new(::SSLv2_method());
     break;
   case context::sslv2_client:
-    handle_ = ::SSL_CTX_new(::SSLv2_client_method());
+    //handle_ = ::SSL_CTX_new(::SSLv2_client_method());
     break;
   case context::sslv2_server:
-    handle_ = ::SSL_CTX_new(::SSLv2_server_method());
+    //handle_ = ::SSL_CTX_new(::SSLv2_server_method());
     break;
   case context::sslv3:
     handle_ = ::SSL_CTX_new(::SSLv3_method());
