diff -ur asio-1.4.1/include/asio/ssl/detail/openssl_stream_service.hpp asio/include/asio/ssl/detail/openssl_stream_service.hpp
--- asio-1.4.1/include/asio/ssl/detail/openssl_stream_service.hpp	2009-03-02 21:22:54.000000000 -0500
+++ asio/include/asio/ssl/detail/openssl_stream_service.hpp	2013-03-13 00:55:41.093203194 -0400
@@ -90,7 +90,7 @@
       : base_handler<Stream>(io_service)
       , handler_(handler)
     {
-      set_func(boost::bind(
+      this->set_func(boost::bind(
         &io_handler<Stream, Handler>::handler_impl, 
         this, boost::arg<1>(), boost::arg<2>() ));
     }
@@ -114,7 +114,7 @@
       : base_handler<Stream>(io_service)
       , handler_(handler)
     {
-      set_func(boost::bind(
+      this->set_func(boost::bind(
         &handshake_handler<Stream, Handler>::handler_impl, 
         this, boost::arg<1>(), boost::arg<2>() ));
     }
@@ -139,7 +139,7 @@
       : base_handler<Stream>(io_service),
         handler_(handler)
     { 
-      set_func(boost::bind(
+      this->set_func(boost::bind(
         &shutdown_handler<Stream, Handler>::handler_impl, 
         this, boost::arg<1>(), boost::arg<2>() ));
     }
