diff -ru python2.5-2.5.2/Misc/python.man python2.5/Misc/python.man
--- python2.5-2.5.2/Misc/python.man	2010-04-26 09:17:43.000000000 -0700
+++ python2.5/Misc/python.man	2010-04-26 09:16:39.000000000 -0700
@@ -243,9 +243,9 @@
 from that file;
 when called with
 .B \-c
-.I command,
+.IR command ,
 it executes the Python statement(s) given as
-.I command.
+.IR command .
 Here
 .I command
 may contain multiple statements separated by newlines.
@@ -255,7 +255,7 @@
 .PP
 If available, the script name and additional arguments thereafter are
 passed to the script in the Python variable
-.I sys.argv ,
+.IR sys.argv ,
 which is a list of strings (you must first
 .I import sys
 to be able to access it).
@@ -269,14 +269,14 @@
 .I '-c'.
 Note that options interpreted by the Python interpreter itself
 are not placed in
-.I sys.argv.
+.IR sys.argv .
 .PP
 In interactive mode, the primary prompt is `>>>'; the second prompt
 (which appears when a command is not complete) is `...'.
 The prompts can be changed by assignment to
 .I sys.ps1
 or
-.I sys.ps2.
+.IR sys.ps2 .
 The interpreter quits when it reads an EOF at a prompt.
 When an unhandled exception occurs, a stack trace is printed and
 control returns to the primary prompt; in non-interactive mode, the
@@ -335,7 +335,7 @@
 inserted in the path in front of $PYTHONPATH.
 The search path can be manipulated from within a Python program as the
 variable
-.I sys.path .
+.IR sys.path .
 .IP PYTHONSTARTUP
 If this is the name of a readable file, the Python commands in that
 file are executed before the first prompt is displayed in interactive
