diff -ur old/haskell-crypto-4.2.3/Crypto.cabal haskell-crypto-4.2.3/Crypto.cabal
--- old/haskell-crypto-4.2.3/Crypto.cabal	2010-12-27 21:14:30.000000000 -0400
+++ haskell-crypto-4.2.3/Crypto.cabal	2011-04-25 17:48:48.737535576 -0400
@@ -46,7 +46,7 @@
                  Data.Digest.SHA512,
                  Data.LargeWord,
                  Data.HMAC
-        Build-Depends:  QuickCheck >= 2.4.0.1, HUnit
+        Build-Depends:  HUnit
         if flag(small_base)
             Build-Depends: base >= 3, base < 5, array, random, pretty
         else
@@ -58,49 +58,3 @@
                  Codec.Encryption.DESAux,
                  Codec.Encryption.AESAux,
                  Data.Digest.MD5Aux
-
-Executable      SymmetricTest
-  Main-Is:         SymmetricTest.hs
-  Ghc-options:     -fregs-graph
-  Other-modules:   Codec.Utils
-                   Codec.Encryption.Blowfish
-                   Codec.Encryption.Modes
-                   Codec.Encryption.Padding
-                   Codec.Encryption.DES
-                   Codec.Encryption.AES
-                   Data.LargeWord
-
-Executable      SHA1Test
-  Main-Is:         SHA1Test.hs
-  Ghc-options:     -fregs-graph
-  Other-modules:   Codec.Text.Raw
-                   Data.Digest.SHA1
-
-Executable      RSATest
-  Main-Is:         RSATest.hs
-  Ghc-options:     -fregs-graph
-  Other-modules:   Codec.Utils
-                   Data.Digest.SHA1
-                   Codec.Encryption.RSA.MGF
-                   Codec.Encryption.RSA.EMEOAEP
-                   Codec.Encryption.RSA
-
-Executable      QuickTest
-  Main-Is:         QuickTest.hs
-  Ghc-options:     -fregs-graph
-  Extensions: TypeSynonymInstances
-  Other-modules:   Codec.Utils
-                   Codec.Encryption.Blowfish
-                   Codec.Encryption.AES
-                   Codec.Encryption.Modes
-                   Codec.Encryption.Padding
-                   Data.LargeWord
-
-Executable      HMACTest
-  Main-Is:         HMACTest.hs
-  Other-modules:   Codec.Utils
-                   Data.HMAC
-
-Executable      WordListTest
-  Main-Is:         WordListTest.hs
-  Other-modules:   Data.LargeWord
