From: Reinhard Tartler <siretart@tauware.de>
Subject: Allow compilation with GCC 4.7
Forwarded-upstream: no

--- a/y4mdenoise/MotionSearcher.hh
+++ b/y4mdenoise/MotionSearcher.hh
@@ -2196,7 +2196,7 @@ MotionSearcher<PIXEL_NUM,DIM,PIXEL_TOL,P
 	{
 		// This region is too small to be bothered with.
 		// Just get rid of it.
-		DeleteRegion (a_pRegion);
+		this->DeleteRegion (a_pRegion);
 	}
 	else
 	{
