From: Daniel T Chen <crimsun@ubuntu.com>
Subject: Link explicitly against the built libmjpegutils. Fixes FTBFS.

--- a/lavtools/Makefile.am
+++ b/lavtools/Makefile.am
@@ -150,7 +150,7 @@ testrec_LDADD = @PTHREAD_LIBS@ $(LIBMJPE
 lav2yuv_SOURCES = lav2yuv.c lav_common.c
 lav2yuv_CPPFLAGS = $(AM_CPPFLAGS) $(LIBDV_CFLAGS)
 lav2yuv_LDADD = $(LIBMJPEGUTILS) \
-		liblavfile.la liblavjpeg.la
+		liblavfile.la liblavjpeg.la -ldv
 
 jpeg2yuv_SOURCES = jpeg2yuv.c
 jpeg2yuv_CPPFLAGS = $(AM_CPPFLAGS) $(JPEG_CFLAGS)
