From: Daniel T Chen <crimsun@ubuntu.com>
Subject: Link explicitly against the built libmjpegutils. Fixes FTBFS.

--- a/mplex/Makefile.am
+++ b/mplex/Makefile.am
@@ -84,5 +84,6 @@ mplex_SOURCES = main.cpp
 
 mplex_DEPENDENCIES = libmplex2.la
 
-mplex_LDADD = libmplex2.la @LIBGETOPT_LIB@ $(LIBM_LIBS)
+mplex_LDADD = libmplex2.la @LIBGETOPT_LIB@ $(LIBM_LIBS) \
+	$(top_builddir)/utils/libmjpegutils.la
 
