From: Daniel T Chen <crimsun@ubuntu.com>
Subject: Link explicitly against the built libmjpegutils. Fixes FTBFS.

--- a/y4mutils/Makefile.am
+++ b/y4mutils/Makefile.am
@@ -54,11 +54,13 @@ y4minterlace_LDADD = $(LIBMJPEGUTILS)
 if HAVE_LIBQUICKTIME
 y4mtoqt_SOURCES = y4mtoqt.c
 y4mtoqt_CPPFLAGS = $(AM_CPPFLAGS) -I$(top_srcdir)/lavtools $(LIBQUICKTIME_CFLAGS)
-y4mtoqt_LDADD = $(LIBMJPEGUTILS) $(top_builddir)/lavtools/liblavfile.la
+y4mtoqt_LDADD = $(LIBMJPEGUTILS) $(top_builddir)/lavtools/liblavfile.la \
+	-lquicktime
 
 qttoy4m_SOURCES = qttoy4m.c
 qttoy4m_CPPFLAGS = $(AM_CPPFLAGS) -I$(top_srcdir)/lavtools $(LIBQUICKTIME_CFLAGS)
-qttoy4m_LDADD = $(LIBMJPEGUTILS) $(top_builddir)/lavtools/liblavfile.la
+qttoy4m_LDADD = $(LIBMJPEGUTILS) $(top_builddir)/lavtools/liblavfile.la \
+	-lquicktime
 endif
 
 y4mtoyuv_SOURCES = y4mtoyuv.c
