From: =?UTF-8?q?Guido=20G=C3=BCnther?= <agx@sigxcpu.org>
Date: Sat, 21 Nov 2009 17:31:55 +0100
Subject: [PATCH] Make user_friendly_names compatible to multipath-tools 0.4.8 and earlier

---
 libmultipath/alias.c |   30 ++++++------------------------
 1 files changed, 6 insertions(+), 24 deletions(-)

diff --git a/libmultipath/alias.c b/libmultipath/alias.c
index 95506b4..02b384b 100644
--- a/libmultipath/alias.c
+++ b/libmultipath/alias.c
@@ -184,40 +184,22 @@ format_devname(char *name, int id, int len)
 {
 	int pos;
 
-	memset(name,0, len);
-	strcpy(name,"mpath");
-	for (pos = len - 1; pos >= 5; pos--) {
-		name[pos] = 'a' + id % 26;
-		if (id < 26)
-			break;
-		id /= 26;
-		id--;
-	}
-	memmove(name + 5, name + pos, len - pos);
-	name[5 + len - pos] = '\0';
-	return (5 + len - pos);
+	memset(name, 0, len);
+	return snprintf(name, len, "mpath%d", id);
 }
 
 static int
 scan_devname(char *alias)
 {
-	char *c;
-	int i, n = 0;
+	int n = 0;
 
 	if (strncmp(alias, "mpath", 5))
 		return -1;
 
-	c = alias + 5;
-	while (*c != '\0' && *c != ' ' && *c != '\t') {
-		i = *c - 'a';
-		n = ( n * 26 ) + i;
-		c++;
-		if (*c < 'a' || *c > 'z')
-			break;
-		n++;
-	}
+	if (sscanf(alias, "mpath%d", &n) >= 1)
+		return n;
 
-	return n;
+	return -1;
 }
 
 static int
-- 
