Description: Do not link against ncurses unnecessarily
Author: Sven Joachim <svenjoac@gmx.de>
Bug-Debian: http://bugs.debian.org/646148
Last-Update: <2011-11-05>

---

Index: mpath/multipathd/Makefile
===================================================================
--- mpath.orig/multipathd/Makefile	2012-02-09 18:17:13.000000000 +0530
+++ mpath/multipathd/Makefile	2012-02-09 18:20:18.994131886 +0530
@@ -6,7 +6,7 @@
 # basic flags setting
 #
 CFLAGS += -I$(multipathdir)
-LDFLAGS += -lpthread -ldevmapper -lreadline -lncurses -ldl \
+LDFLAGS += -lpthread -ldevmapper -lreadline -ldl \
 	   -L$(multipathdir) -lmultipath
 
 #
