Multipath Hardening Patch
Index: multipath-tools/kpartx/Makefile
===================================================================
--- multipath-tools.orig/kpartx/Makefile	2012-03-12 12:28:56.914148797 +0530
+++ multipath-tools/kpartx/Makefile	2012-03-12 12:39:06.505171593 +0530
@@ -12,7 +12,7 @@
 	CFLAGS += -DLIBDM_API_COOKIE
 endif
 
-LDFLAGS = -ldevmapper
+LDFLAGS += $(shell dpkg-buildflags --get LDFLAGS) -ldevmapper
 OBJS = bsd.o dos.o kpartx.o solaris.o unixware.o dasd.o sun.o \
        gpt.o mac.o crc32.o lopart.o xstrncpy.o devmapper.o
 EXEC = kpartx
Index: multipath-tools/Makefile.inc
===================================================================
--- multipath-tools.orig/Makefile.inc	2012-03-12 12:28:56.914148797 +0530
+++ multipath-tools/Makefile.inc	2012-03-12 12:38:23.568958684 +0530
@@ -36,8 +36,10 @@
 GZIP        = /bin/gzip -9 -c
 INSTALL_PROGRAM = install
 
-OPTFLAGS     = -pipe -g -Wall -Wunused -Wstrict-prototypes
-CFLAGS	     = $(OPTFLAGS) -fPIC -DLIB_STRING=\"${LIB}\"
+OPTFLAGS     = -pipe -Wall -Wunused -Wstrict-prototypes
+CFLAGS      = $(shell dpkg-buildflags --get CFLAGS) $(shell dpkg-buildflags --get CPPFLAGS) $(OPTFLAGS) -fPIC -DLIB_STRING=\"${LIB}\"
+LDFLAGS     = $(shell dpkg-buildflags --get LDFLAGS)
+
 SHARED_FLAGS = -shared
 
 %.o:	%.c
Index: multipath-tools/multipathd/Makefile
===================================================================
--- multipath-tools.orig/multipathd/Makefile	2012-03-12 12:29:27.954302713 +0530
+++ multipath-tools/multipathd/Makefile	2012-03-12 12:29:28.466305251 +0530
@@ -6,7 +6,7 @@
 # basic flags setting
 #
 CFLAGS += -I$(multipathdir)
-LDFLAGS += -lpthread -ldevmapper -lreadline -ldl \
+LDFLAGS += `dpkg-buildflags --get LDFLAGS` -lpthread -ldevmapper -lreadline -ldl \
 	   -L$(multipathdir) -lmultipath
 
 #
Index: multipath-tools/multipath/Makefile
===================================================================
--- multipath-tools.orig/multipath/Makefile	2012-03-12 12:28:56.914148797 +0530
+++ multipath-tools/multipath/Makefile	2012-03-12 12:40:48.061675186 +0530
@@ -7,7 +7,7 @@
 OBJS = main.o
 
 CFLAGS += -I$(multipathdir)
-LDFLAGS += -lpthread -ldevmapper -ldl -L$(multipathdir) -lmultipath
+LDFLAGS += $(shell dpkg-buildflags --get LDFLAGS) -lpthread -ldevmapper -ldl -lmultipath -L$(multipathdir)
 
 EXEC = multipath
 
