From: Marco Nenciarini <mnencia@debian.org>
Date: Fri, 11 Jan 2013 01:40:39 +0100
Subject: declare argparse dep

---
 setup.py |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index dca263e..553271d 100755
--- a/setup.py
+++ b/setup.py
@@ -12,7 +12,7 @@ setup(
     author_email='kislyuk@gmail.com',
     description='Bash tab completion for argparse',
     long_description=open('README.rst').read(),
-    install_requires=['distribute'],
+    install_requires=['distribute', 'argparse'],
     packages = find_packages(),
     scripts = glob.glob('scripts/*'),
     package_data={'argcomplete': ['bash_completion.d/python-argcomplete.sh']},
