=== modified file 'src/idoscalemenuitem.c'
--- a/src/idoscalemenuitem.c	2012-03-21 19:53:48 +0000
+++ b/src/idoscalemenuitem.c	2012-04-29 08:46:42 +0000
@@ -518,14 +518,14 @@
   event->x_root = x;
 
 #ifndef USE_GTK3
-  ubuntu_gtk_widget_set_has_grab (scale, TRUE);
+  GTK_OBJECT_FLAGS (scale) |= GTK_HAS_GRAB;
 #endif
 
   gtk_widget_event (priv->scale,
                     ((GdkEvent *)(void*)(event)));
 
 #ifndef USE_GTK3
-  ubuntu_gtk_widget_set_has_grab (scale, FALSE);
+  GTK_OBJECT_FLAGS (scale) &= ~(GTK_HAS_GRAB);
 #endif
 
   if (!priv->grabbed)

