From: Alexander Sack <asac@debian.org>
Date: Wed, 27 Jan 2010 19:05:18 +0100
Subject: fix installdir

---
 mail/app/Makefile.in |    6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/mail/app/Makefile.in b/mail/app/Makefile.in
index d5b4489..c386bb4 100644
--- a/mail/app/Makefile.in
+++ b/mail/app/Makefile.in
@@ -172,10 +172,10 @@ libs:: $(addprefix $(DIST)/branding/,$(BRANDED_ICON_FILES))
 	$(INSTALL) $(IFLAGS1) $^ $(DIST)/bin/chrome/icons/default
 
 install:: $(addprefix icons/$(MOZ_WIDGET_TOOLKIT)/,$(DESKTOP_ICON_FILES))
-	$(SYSINSTALL) $(IFLAGS1) $^ $(DESTDIR)$(mozappdir)/chrome/icons/default
+	$(SYSINSTALL) $(IFLAGS1) $^ $(DESTDIR)$(installdir)/chrome/icons/default
 
 install:: $(addprefix $(DIST)/branding,$(BRANDED_ICON_FILES))
-	$(SYSINSTALL) $(IFLAGS1) $^ $(DESTDIR)$(mozappdir)/chrome/icons/default
+	$(SYSINSTALL) $(IFLAGS1) $^ $(DESTDIR)$(installdir)/chrome/icons/default
 endif
 
 ifeq ($(MOZ_WIDGET_TOOLKIT),gtk2)
@@ -183,7 +183,7 @@ libs::
 	$(INSTALL) $(IFLAGS1) $(DIST)/branding/default* $(DIST)/bin/chrome/icons/default
 
 install::
-	$(SYSINSTALL) $(IFLAGS1) $(DIST)/branding/default* $(DESTDIR)$(mozappdir)/chrome/icons/default
+	$(SYSINSTALL) $(IFLAGS1) $(DIST)/branding/default* $(DESTDIR)$(installdir)/chrome/icons/default
 endif
 
 libs:: $(srcdir)/profile/prefs.js
