Index: libaacplus-2.0.1/src/Makefile.in
===================================================================
--- libaacplus-2.0.1.orig/src/Makefile.in	2010-11-05 09:27:35.000000000 +0100
+++ libaacplus-2.0.1/src/Makefile.in	2010-11-05 12:45:26.000000000 +0100
@@ -243,7 +243,6 @@
 top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-DISTCLEANFILES = 26410-800.zip $(top_srcdir)/26410-800.doc 26410-800-ANSI-C_source_code.zip $(library_sources_used) $(library_headers_used)
 EXTRA_DIST = libaacplus.sym
 INCLUDES = @FFTW3_INCFLAGS@ -I$(top_srcdir)/include
 LDADD = @FFTW3_LDFLAGS@
@@ -721,18 +720,6 @@
 	tags uninstall uninstall-am uninstall-libLTLIBRARIES
 
 
-sources_delete:
-	@echo echo "Deleting headers..."
-	for source in $(library_headers_used); \
-	do \
-	    rm -f $$source; \
-	done
-	@echo echo "Deleting sources..."
-	for source in $(library_sources_used); \
-	do \
-	    rm -f $$source; \
-	done
-
 sources_copy:
 	@echo echo "Copying headers..."
 	for source in $(library_headers_used); \
@@ -802,18 +789,6 @@
 	mv -f $(srcdir)/3GPP_enhanced_aacPlus_src_*/FloatFR_aacPlusenc $(srcdir)
 	rm -rf $(srcdir)/3GPP_enhanced_aacPlus_src_*
 
-$(srcdir)/26410-800-ANSI-C_source_code.zip: $(srcdir)/26410-800.zip
-	rm -f $(srcdir)/26410-800-ANSI-C_source_code.zip $(srcdir)/26410-800.doc
-	cd $(srcdir) && $(UNZIP) 26410-800.zip
-	mv $(srcdir)/26410-800.doc $(top_srcdir)/26410-800.doc
-	chmod og-w $(top_srcdir)/26410-800.doc
-
-$(srcdir)/26410-800.zip:
-	cd $(srcdir) && $(DOWNLOADER_PROG) http://www.3gpp.org/ftp/Specs/archive/26_series/26.410/26410-800.zip
-
-extraclean:
-	rm -f $(distdir)/($(library_sources_used)) $(distdir)/($(library_headers_used)) $(distdir)/26410-800.zip $(distdir)/../26410-800.doc $(distdir)/26410-800_ANSI-C_source_code.zip
-
 # WARNING: These commands use internal automake command and may fail in future version.
 extradist-gzip: distdir extraclean
 	tardir=$(distdir) && $(am__tar) | GZIP=$(GZIP_ENV) gzip -c >$(distdir).tar.gz
