Description: fix FTBSD on kfreeBSD
Author: Hideki Yamane <henrich@debian.org>
Bug-Debian: 591254
Index: libgpod/tools/generic-callout.c
===================================================================
--- libgpod.orig/tools/generic-callout.c	2011-08-12 19:08:31.149534658 +0800
+++ libgpod/tools/generic-callout.c	2011-08-12 19:08:37.789574397 +0800
@@ -584,6 +584,7 @@
 
 static char *mount_ipod (const char *dev_path, const char *fstype)
 {
+#ifdef __linux__
         char *filename;
         char *tmpname;
         int result;
@@ -612,6 +613,10 @@
         g_debug("device successfully mounted at %s", tmpname);
 
         return tmpname;
+#else
+#warning mount wrapper for your OS is missing
+        return NULL;
+#endif
 }
 
 static gboolean write_sysinfo_extended (const char *mountpoint,
