Index: qemu/configure
===================================================================
--- qemu.orig/configure	2012-05-24 20:19:08.825346337 -0700
+++ qemu/configure	2012-05-24 20:19:39.145496689 -0700
@@ -2305,6 +2305,14 @@
 #include <stddef.h>
 #include <fcntl.h>
 
+#if defined __stub_futimens || defined __stub___futimens
+choke me
+#endif
+
+#if defined __stub_utimensat || defined __stub___utimensat
+choke me
+#endif
+
 int main(void)
 {
     utimensat(AT_FDCWD, "foo", NULL, 0);
