---
 build.sh |    6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

--- luatex.orig/build.sh
+++ luatex/build.sh
@@ -190,9 +190,9 @@
 # (perhaps more files needed to be copied from TL?)
 
 (cd libs/zziplib; $MAKE all )
-(cd libs/zlib; $MAKE all )
-(cd libs/libpng; $MAKE all )
-(cd libs/poppler; $MAKE all )
+#(cd libs/zlib; $MAKE all )
+#(cd libs/libpng; $MAKE all )
+#(cd libs/poppler; $MAKE all )
 
 (cd texk/kpathsea; $MAKE )
 (cd texk/web2c; $MAKE $LUATEXEXE )
