Description: Fix FTBFS by linking with GLU
Author: Tormod Volden <debian.tormod@gmail.com>

Index: elmerfem-6.1.0.svn.5396.dfsg/ElmerGUI/Application/Application.pro
===================================================================
--- elmerfem-6.1.0.svn.5396.dfsg.orig/ElmerGUI/Application/Application.pro	2012-05-26 00:47:14.000000000 +0200
+++ elmerfem-6.1.0.svn.5396.dfsg/ElmerGUI/Application/Application.pro	2012-05-26 00:50:54.817404865 +0200
@@ -115,6 +115,8 @@
 #------------------------------------------------------------------------------
 win32: LIBS += -lpsapi
 
+LIBS += -lGLU
+
 #------------------------------------------------------------------------------
 # Input files:
 #------------------------------------------------------------------------------
Index: elmerfem-6.1.0.svn.5396.dfsg/ElmerGUI/Application/src/glwidget.h
===================================================================
--- elmerfem-6.1.0.svn.5396.dfsg.orig/ElmerGUI/Application/src/glwidget.h	2011-10-01 10:00:06.000000000 +0200
+++ elmerfem-6.1.0.svn.5396.dfsg/ElmerGUI/Application/src/glwidget.h	2012-05-26 00:48:46.137400923 +0200
@@ -56,6 +56,7 @@
 #include <QVector>
 #include "helpers.h"
 #include "meshutils.h"
+#include <GL/glu.h>
 
 #define DUMMY_NAME 0xffffffff
 
