Change link order for binutils-gold

Index: elmerfem/ElmerGUI/Application/Application.pro
===================================================================
--- elmerfem.orig/ElmerGUI/Application/Application.pro
+++ elmerfem/ElmerGUI/Application/Application.pro
@@ -107,7 +107,7 @@
 contains(DEFINES, EG_PYTHONQT) {
    INCLUDEPATH += $${PY_INCLUDEPATH} ../PythonQt/src
    LIBPATH += $${PY_LIBPATH} ../PythonQt/lib
-   LIBS += $${PY_LIBS} -lPythonQt
+   LIBS += -lPythonQt $${PY_LIBS}
 }
 
 #------------------------------------------------------------------------------
