Compatibility change for libav 0.7.1

Index: elmerfem/utils/ElmerClips/src/encoder.cpp
===================================================================
--- elmerfem.orig/utils/ElmerClips/src/encoder.cpp
+++ elmerfem/utils/ElmerClips/src/encoder.cpp
@@ -213,7 +213,7 @@
   }
 
   context->codec_id = codec_id;
-  context->codec_type = CODEC_TYPE_VIDEO;
+  context->codec_type = AVMEDIA_TYPE_VIDEO;
   context->qmin = 2;
   context->qmax = qMax(2, qMin(31, quality));
   context->width = widthYUV;
