Index: openslp-dfsg-1.2.1/configure.in
===================================================================
--- openslp-dfsg-1.2.1.orig/configure.in	2009-06-25 13:16:56.000000000 +0200
+++ openslp-dfsg-1.2.1/configure.in	2009-06-25 13:17:30.000000000 +0200
@@ -159,7 +159,7 @@
 dnl Checks for header files.
 dnl ***********************************************************************
 AC_HEADER_STDC
-AC_CHECK_HEADERS(string.h sys/types.h stdint.h netinet/in.h)
+AC_CHECK_HEADERS(string.h sys/types.h stdint.h netinet/in.h net/if_arp.h)
 
 dnl ***********************************************************************
 dnl Checks for data types.
Index: openslp-dfsg-1.2.1/common/slp_dhcp.c
===================================================================
--- openslp-dfsg-1.2.1.orig/common/slp_dhcp.c	2009-06-25 13:16:57.000000000 +0200
+++ openslp-dfsg-1.2.1/common/slp_dhcp.c	2009-06-25 13:17:30.000000000 +0200
@@ -65,7 +65,9 @@
 #include <sys/types.h>
 #include <sys/socket.h>
 #include <sys/ioctl.h>
+#ifdef HAVE_NET_IF_ARP_H
 #include <net/if_arp.h>
+#endif
 #include <sys/time.h>
 #endif
 
@@ -335,7 +337,7 @@
 		FreeLibrary(hmod);
 	}
 
-#elif defined(SIOCGARP)
+#elif defined(SIOCGARP) && defined(HAVE_NET_IF_ARP_H)
 
 	/* Query the ARP cache for our hardware address */
 	
