Subject: Ensure PyParser's parserline is initialized
Forwarded: no
Author: John Eikenberry <jae@zhar.net>
Bug-Debian: http://bugs.debian.org/608420

Index: b/runtime/autoload/pythoncomplete.vim
===================================================================
--- a/runtime/autoload/pythoncomplete.vim
+++ b/runtime/autoload/pythoncomplete.vim
@@ -377,6 +377,7 @@
     def __init__(self):
         self.top = Scope('global',0)
         self.scope = self.top
+        self.parserline = 0
 
     def _parsedotname(self,pre=None):
         #returns (dottedname, nexttoken)
