Description: Fix build failure with [-Werror=format-security]
Author: Michael Biebl <biebl@debian.org>
Bug: https://bugzilla.gnome.org/show_bug.cgi?id=661722
Index: libgnome-media-profiles-3.0.0/libgnome-media-profiles/audio-profile-edit.c
===================================================================
--- libgnome-media-profiles-3.0.0.orig/libgnome-media-profiles/audio-profile-edit.c	2010-10-31 02:01:20.000000000 +0200
+++ libgnome-media-profiles-3.0.0/libgnome-media-profiles/audio-profile-edit.c	2011-10-14 01:34:51.792485916 +0200
@@ -230,7 +230,7 @@
   /* get the dialog */
   builder = gmp_util_load_builder_file ("gnome-audio-profile-edit.ui", NULL, &error);
   if (error != NULL) {
-    g_warning (error->message);
+    g_warning ("%s", error->message);
     g_error_free (error);
     return NULL;
   }
Index: libgnome-media-profiles-3.0.0/libgnome-media-profiles/audio-profiles-edit.c
===================================================================
--- libgnome-media-profiles-3.0.0.orig/libgnome-media-profiles/audio-profiles-edit.c	2011-01-13 17:49:35.000000000 +0100
+++ libgnome-media-profiles-3.0.0/libgnome-media-profiles/audio-profiles-edit.c	2011-10-14 01:33:56.969175138 +0200
@@ -912,7 +912,7 @@
     builder = gmp_util_load_builder_file ("gnome-audio-profile-new.ui", transient_parent, &error);
 
     if (error != NULL) {
-      g_warning (error->message);
+      g_warning ("%s", error->message);
       g_error_free (error);
       return;
     }
Index: libgnome-media-profiles-3.0.0/libgnome-media-profiles/gmp-util.c
===================================================================
--- libgnome-media-profiles-3.0.0.orig/libgnome-media-profiles/gmp-util.c	2010-10-31 02:01:20.000000000 +0200
+++ libgnome-media-profiles-3.0.0/libgnome-media-profiles/gmp-util.c	2011-10-14 01:35:27.284039726 +0200
@@ -47,7 +47,7 @@
     goto end;
 
   if (error != NULL) {
-    g_warning (error->message);
+    g_warning ("%s", error->message);
     g_error_free (error);
     error = NULL;
   }
