From: Arto Jantunen <viiru@debian.org>
Date: Tue, 8 May 2012 14:31:56 +0300
Subject: [PATCH] Stop setting CFLAGS in the Makefile

---
 Makefile |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/Makefile b/Makefile
index 3e53d15..6a614f1 100644
--- a/Makefile
+++ b/Makefile
@@ -1,6 +1,6 @@
 
 #LINKOPT	=	/usr/local/lib/libdbmalloc.a
-CFLAGS	=	-g
+#CFLAGS	=	-g
 
 default:	all
 
-- 
