=== modified file 'bindings/mono/indicate-gtk/indicate-gtk-sharp-0.1.pc.in'
--- old/bindings/mono/indicate-gtk/indicate-gtk-sharp-0.1.pc.in	2010-09-09 20:00:04 +0000
+++ new/bindings/mono/indicate-gtk/indicate-gtk-sharp-0.1.pc.in	2010-09-09 20:06:35 +0000
@@ -1,7 +1,7 @@
 prefix=@prefix@
 exec_prefix=@exec_prefix@
 libdir=@libdir@
-assemblies_dir=${libdir}/indicate-gtk-sharp-0.1
+assemblies_dir=${libdir}/cli/indicate-gtk-sharp-0.1
 
 Name: indicate-gtk-sharp
 Description: indicate-gtk for .NET

=== modified file 'bindings/mono/indicate/indicate-sharp-0.1.pc.in'
--- old/bindings/mono/indicate/indicate-sharp-0.1.pc.in	2010-09-09 20:00:04 +0000
+++ new/bindings/mono/indicate/indicate-sharp-0.1.pc.in	2010-09-09 20:06:06 +0000
@@ -1,7 +1,7 @@
 prefix=@prefix@
 exec_prefix=@exec_prefix@
 libdir=@libdir@
-assemblies_dir=${libdir}/indicate-sharp-0.1
+assemblies_dir=${libdir}/cli/indicate-sharp-0.1
 
 Name: indicate-sharp
 Description: indicate for .NET

