Index: autogen/autoopts/autoopts.h
===================================================================
--- autogen.orig/autoopts/autoopts.h	2009-06-11 18:18:52.000000000 +0100
+++ autogen/autoopts/autoopts.h	2009-06-11 18:19:51.000000000 +0100
@@ -41,7 +41,7 @@
 #  ifdef PATH_MAX
 #    define AG_PATH_MAX         ((size_t)PATH_MAX)
 #  else
-#    define AG_PATH_MAX         ((size_t)4096)
+#    define AG_PATH_MAX         4096
 #  endif
 #else
 #  if defined(PATH_MAX) && (PATH_MAX > MAXPATHLEN)
Index: autogen/compat/compat.h
===================================================================
--- autogen.orig/compat/compat.h	2009-06-11 18:18:44.000000000 +0100
+++ autogen/compat/compat.h	2009-06-11 18:19:25.000000000 +0100
@@ -243,7 +243,7 @@
 #endif
 
 #if !defined (MAXPATHLEN)
-#  define MAXPATHLEN ((size_t)4096)
+#  define MAXPATHLEN 4096
 #endif /* MAXPATHLEN */
 
 #define AG_PATH_MAX  ((size_t)MAXPATHLEN)
