diff -ruN console-tools-0.2.3-old/screenfonttools/consolechars.c console-tools-0.2.3/screenfonttools/consolechars.c
--- console-tools-0.2.3-old/screenfonttools/consolechars.c	2003-11-16 16:30:02.000000000 +0000
+++ console-tools-0.2.3/screenfonttools/consolechars.c	2003-11-16 16:31:32.000000000 +0000
@@ -577,6 +577,7 @@
 fprintf (stderr, "WARNING: Not using a unimap may lead to erroneous display !\n");
    */
 
+#if 0
   /* Check that font-size is valid */
 #if !defined( PIO_FONTX ) || defined( __sparc__ )
   if (the_font->font.charcount != 256) 
@@ -585,6 +586,7 @@
       exit(1);
     }
 #endif
+#endif
 
   
   /*
