diff -ruN console-tools-0.2.3-old/doc/lct.sgml console-tools-0.2.3/doc/lct.sgml
--- console-tools-0.2.3-old/doc/lct.sgml	2004-01-10 21:24:24.000000000 +0000
+++ console-tools-0.2.3/doc/lct.sgml	2004-01-10 21:24:24.000000000 +0000
@@ -19,7 +19,7 @@
 
 <sect1>Other documents
 <p>
-The &lct; <url url="http://www.multimania.com/ydirson/en/lct/"
+The &lct; <url url="http://lct.sourceforge.net/"
 name="WWW site"> may contain additionnal informations, latest news,
 and such.
 
diff -ruN console-tools-0.2.3-old/doc/lct.txt console-tools-0.2.3/doc/lct.txt
--- console-tools-0.2.3-old/doc/lct.txt	1999-07-19 20:50:42.000000000 +0100
+++ console-tools-0.2.3/doc/lct.txt	2004-01-10 21:24:24.000000000 +0000
@@ -61,7 +61,7 @@
   11..11..  OOtthheerr ddooccuummeennttss
 
   The Linux Console Tools WWW site
-  <http://www.multimania.com/ydirson/en/lct/> may contain additionnal
+  <http://lct.sourceforge.net/> may contain additionnal
   informations, latest news, and such.
 
   Files in the doc/contrib/ directory are unsupported, and may be
diff -ruN console-tools-0.2.3-old/doc/man/consolechars.8.in console-tools-0.2.3/doc/man/consolechars.8.in
--- console-tools-0.2.3-old/doc/man/consolechars.8.in	1999-04-08 22:29:41.000000000 +0100
+++ console-tools-0.2.3/doc/man/consolechars.8.in	2004-01-10 21:25:30.000000000 +0000
@@ -285,4 +285,5 @@
 
 .SH "SEE ALSO"
 .BR psfaddtable (1),
-.BR psfstriptable (1).
+.BR psfstriptable (1),
+.BR setkeycodes (8).
diff -ruN console-tools-0.2.3-old/doc/man/dumpkeys.1.in console-tools-0.2.3/doc/man/dumpkeys.1.in
--- console-tools-0.2.3-old/doc/man/dumpkeys.1.in	1999-04-08 22:29:41.000000000 +0100
+++ console-tools-0.2.3/doc/man/dumpkeys.1.in	2004-01-10 21:25:43.000000000 +0000
@@ -239,4 +239,5 @@
 
 .SH "SEE ALSO"
 .BR loadkeys (1),
-.BR keymaps (5).
+.BR keymaps (5),
+.BR setkeycodes (8).
diff -ruN console-tools-0.2.3-old/doc/man/kbdrate.8 console-tools-0.2.3/doc/man/kbdrate.8
--- console-tools-0.2.3-old/doc/man/kbdrate.8	2004-01-10 21:24:24.000000000 +0000
+++ console-tools-0.2.3/doc/man/kbdrate.8	2004-01-10 21:24:24.000000000 +0000
@@ -63,6 +63,4 @@
 (The former usually works on an m68k machine, the latter for SPARC.)
 When these ioctls fail an ioport interface as on i386 is assumed.
 .SH FILES
-.I /etc/rc.local
-.br
 .I /dev/port
diff -ruN console-tools-0.2.3-old/doc/man/loadkeys.1.in console-tools-0.2.3/doc/man/loadkeys.1.in
--- console-tools-0.2.3-old/doc/man/loadkeys.1.in	1999-04-08 22:29:41.000000000 +0100
+++ console-tools-0.2.3/doc/man/loadkeys.1.in	2004-01-10 21:25:56.000000000 +0000
@@ -184,4 +184,5 @@
 .SH "SEE ALSO"
 .BR dumpkeys (1),
 .BR kbd_mode (1),
-.BR keymaps (5).
+.BR keymaps (5),
+.BR setkeycodes (8).
diff -ruN console-tools-0.2.3-old/doc/man/showkey.1 console-tools-0.2.3/doc/man/showkey.1
--- console-tools-0.2.3-old/doc/man/showkey.1	1999-04-08 22:29:41.000000000 +0100
+++ console-tools-0.2.3/doc/man/showkey.1	2004-01-10 21:26:14.000000000 +0000
@@ -102,4 +102,5 @@
 .SH "SEE ALSO"
 .BR loadkeys (1),
 .BR dumpkeys (1),
-.BR keymaps (5).
+.BR keymaps (5),
+.BR setkeycodes (8).
