diff -urN console-tools-0.2.3.orig/vttools/setvesablank.c console-tools-0.2.3/vttools/setvesablank.c
--- console-tools-0.2.3.orig/vttools/setvesablank.c	Thu Sep  5 13:08:46 2002
+++ console-tools-0.2.3/vttools/setvesablank.c	Thu Sep  5 13:18:40 2002
@@ -33,6 +33,8 @@
       return 2;
   else if (!strcmp(s, "off"))
      return 0;
+  else if (!strcmp(s, "OFF"))
+     return 0;
   else {
     fprintf(stderr, _("%s: argument '%s' should be 'ON', 'on' or 'off'\n"),
 	    progname, s);
