diff -ruN console-tools-0.2.3-old/kbdtools/setkeycodes.c console-tools-0.2.3/kbdtools/setkeycodes.c
--- console-tools-0.2.3-old/kbdtools/setkeycodes.c	2003-11-16 18:50:35.000000000 +0000
+++ console-tools-0.2.3/kbdtools/setkeycodes.c	2003-11-16 18:51:18.000000000 +0000
@@ -75,8 +75,7 @@
 	  a.scancode -= 0xe000;
 	  a.scancode += 128;
 	}
-      if (a.scancode > 255 || a.keycode > 127)
-	badusage(_("code outside bounds"));
+
       if (ioctl(fd,KDSETKEYCODE,&a)) 
 	{
 	  perror("KDSETKEYCODE");
