diff -ruN console-tools-0.2.3-old/screenfonttools/unicode_start console-tools-0.2.3/screenfonttools/unicode_start
--- console-tools-0.2.3-old/screenfonttools/unicode_start	1999-04-08 23:11:15.000000000 +0100
+++ console-tools-0.2.3/screenfonttools/unicode_start	2003-05-23 13:10:15.000000000 +0100
@@ -1,13 +1,20 @@
 #!/bin/sh
 # start unicode mode
 
-kbd_mode -u
 
-if vt-is-UTF8 --quiet ; then
-  echo "Already in UTF8 mode"
-  exit 0
+readlink /proc/self/fd/0 | grep -q -e /dev/vc -e /dev/tty -e /dev/console
+if [ $? -eq 1 ]; then
+    echo  "Won't set unicode mode: not a VT."
+    exit 0
+fi
+
+if /usr/bin/vt-is-UTF8 --quiet ; then
+    echo "unicode_start: Already in UTF8 mode"
+    exit 0
 fi
 
+kbd_mode -u
+
 case $# in
     2)
         consolechars --font=$1 --sfm=$2
diff -ruN console-tools-0.2.3-old/screenfonttools/unicode_stop console-tools-0.2.3/screenfonttools/unicode_stop
--- console-tools-0.2.3-old/screenfonttools/unicode_stop	1999-04-08 22:29:41.000000000 +0100
+++ console-tools-0.2.3/screenfonttools/unicode_stop	2003-05-23 13:09:51.000000000 +0100
@@ -1,11 +1,18 @@
 #!/bin/sh
 # stop unicode
 
-kbd_mode -a
 
-if ! vt-is-UTF8 --quiet ; then
-  echo "Already in byte-mode"
-  exit 0
+if ! /usr/bin/vt-is-UTF8 --quiet ; then
+    echo "unicode_stop: Already in byte-mode"
+    exit 0
+fi
+
+readlink /proc/self/fd/0 | grep -q  -e /dev/vc -e /dev/tty -e /dev/console
+if [ $? -eq 1 ]; then
+    echo "Not a virtual terminal : Not setting UTF-8 mode."
+    exit 0
 fi
 
+kbd_mode -a
+
 echo -n -e '\033%@'
