diff -ruN console-tools-0.2.3-old/fontfiletools/psfgettable.c console-tools-0.2.3/fontfiletools/psfgettable.c
--- console-tools-0.2.3-old/fontfiletools/psfgettable.c	1999-04-08 23:11:12.000000000 +0100
+++ console-tools-0.2.3/fontfiletools/psfgettable.c	2002-11-28 00:03:41.000000000 +0000
@@ -100,6 +100,7 @@
   int glyph;
   unicode unichar;
   int fontlen;
+  unsigned char buf[2];
 
   setuplocale();
   
@@ -158,8 +159,8 @@
       else
 	fprintf(out, "0x%03x\t", glyph);
 
-      while ( fread(&unichar, sizeof(unicode), 1, in),
-	      unichar != PSF_SEPARATOR )
+      while ( fread(buf, sizeof(buf), 1, in),
+	      (unichar = buf[0] + 256 * buf[1]) != PSF_SEPARATOR )
 	fprintf(out, " U+%04x", unichar);
 
       putc('\n', out);
