diff -ruN console-tools-0.2.3-old/vttools/openvt.c console-tools-0.2.3/vttools/openvt.c
--- console-tools-0.2.3-old/vttools/openvt.c	2004-04-13 11:26:07.000000000 +0000
+++ console-tools-0.2.3/vttools/openvt.c	2004-04-13 11:26:07.000000000 +0000
@@ -81,7 +81,7 @@
 
 void usage(char * progname)
 {
-  printf(_("Usage: %s [-c vtnumber] [-l] [-u] [-s] [-v] [-w] -- command_line\n"), progname);
+  printf(_("Usage: %s [-c vtnumber] [-l] [-u] [-s] [-v] [-w] command_line\n"), progname);
 }
 
 int main(int argc, char *argv[])
@@ -105,11 +105,11 @@
   setuplocale();
   
   /*
-   * I don't like using getopt for this, but otherwise this gets messy.
-   * POSIX/Gnu getopt forces the use of -- to separate child/program
-   * options. RTFM.
+   * The '+' at the beginning makes GNU getopt stop processing options
+   * as soon as a non-option argument is encountered.  That way the
+   * child options are left intact.
    */
-  while ((opt = getopt(argc, argv, "c:lsvfuw")) != -1)
+  while ((opt = getopt(argc, argv, "+c:lsvfuw")) != -1)
     {
       switch (opt)
 	{
diff -ruN console-tools-0.2.3-old/vttools/vt-is-UTF8.c console-tools-0.2.3/vttools/vt-is-UTF8.c
--- console-tools-0.2.3-old/vttools/vt-is-UTF8.c	2004-04-13 11:26:07.000000000 +0000
+++ console-tools-0.2.3/vttools/vt-is-UTF8.c	2004-04-13 11:26:37.000000000 +0000
@@ -36,6 +36,7 @@
   int result;			       /* option handling */
   int an_option;
   int quiet = 0;
+  char *term;
 
   setuplocale();
   
@@ -63,7 +64,13 @@
     }
   
   if (optind < argc)
-      badusage ("no non-option arguments are valid");
+      badusage (_("no non-option arguments are valid"));
+
+  term = getenv("TERM");
+  if (term == NULL)
+	  badusage(_("TERM must be set for vt-is-UTF8 to run"));
+  if (strstr(term,"linux")==NULL)
+	  badusage(_("vt-is-UTF8 only runs on linux virtual terminals"));
 
   x = is_in_UTF8_mode (get_console_fd(NULL));
   switch (x)
@@ -72,7 +79,7 @@
       if (quiet)
 	  exit (0);
       else
-	  fprintf (stderr, "UTF-8 unicode mode.\n");
+	  fprintf (stderr, _("UTF-8 unicode mode.\n"));
       break;
       
     case 0:
