diff -ruN console-tools-0.2.3-old/compat/Makefile.in console-tools-0.2.3/compat/Makefile.in
--- console-tools-0.2.3-old/compat/Makefile.in	2003-01-22 11:38:03.000000000 +0000
+++ console-tools-0.2.3/compat/Makefile.in	2003-01-22 11:38:03.000000000 +0000
@@ -97,7 +97,6 @@
 POFILES = @POFILES@
 POSUB = @POSUB@
 RANLIB = @RANLIB@
-RESIZECONS = @RESIZECONS@
 TRANSDIR = @TRANSDIR@
 UNICODEDATA = @UNICODEDATA@
 UNICODELISTS = @UNICODELISTS@
diff -ruN console-tools-0.2.3-old/configure console-tools-0.2.3/configure
--- console-tools-0.2.3-old/configure	2003-01-22 11:38:03.000000000 +0000
+++ console-tools-0.2.3/configure	2003-01-22 11:38:03.000000000 +0000
@@ -4243,13 +4243,6 @@
 
 
 
-# Enable `resizecons' only on i386 arch
-case $target_cpu in
-i?86)
-    RESIZECONS="resizecons"
-    ;;
-esac
-
 
 if test "$EXTRAS" != ""
 then
@@ -4527,7 +4520,6 @@
 s%@l@%$l%g
 s%@LOADKEYS@%$LOADKEYS%g
 s%@KEYCODES@%$KEYCODES%g
-s%@RESIZECONS@%$RESIZECONS%g
 s%@UNICODEDATA@%$UNICODEDATA%g
 s%@UNICODELISTS@%$UNICODELISTS%g
 s%@COMPAT@%$COMPAT%g
diff -ruN console-tools-0.2.3-old/configure.in console-tools-0.2.3/configure.in
--- console-tools-0.2.3-old/configure.in	1999-10-25 21:48:29.000000000 +0100
+++ console-tools-0.2.3/configure.in	2003-01-22 11:38:03.000000000 +0000
@@ -97,14 +97,6 @@
   KEYCODES="getkeycodes setkeycodes" ; EXTRAS="${EXTRAS} ${KEYCODES}" )
 AC_SUBST(KEYCODES)
 
-# Enable `resizecons' only on i386 arch
-case $target_cpu in
-i?86)
-    RESIZECONS="resizecons"
-    ;;
-esac
-AC_SUBST(RESIZECONS)
-
 if test "$EXTRAS" != ""
 then
     AC_MSG_WARN([The following programs may not build on all archs: $EXTRAS.
diff -ruN console-tools-0.2.3-old/doc/Makefile.in console-tools-0.2.3/doc/Makefile.in
--- console-tools-0.2.3-old/doc/Makefile.in	2003-01-22 11:38:03.000000000 +0000
+++ console-tools-0.2.3/doc/Makefile.in	2003-01-22 11:38:03.000000000 +0000
@@ -97,7 +97,6 @@
 POFILES = @POFILES@
 POSUB = @POSUB@
 RANLIB = @RANLIB@
-RESIZECONS = @RESIZECONS@
 TRANSDIR = @TRANSDIR@
 UNICODEDATA = @UNICODEDATA@
 UNICODELISTS = @UNICODELISTS@
diff -ruN console-tools-0.2.3-old/doc/man/Makefile.am console-tools-0.2.3/doc/man/Makefile.am
--- console-tools-0.2.3-old/doc/man/Makefile.am	2003-01-22 11:38:03.000000000 +0000
+++ console-tools-0.2.3/doc/man/Makefile.am	2003-01-22 11:38:03.000000000 +0000
@@ -7,7 +7,7 @@
 	psfgettable.1 psfstriptable.1 setleds.1 \
 	setmetamode.1 showkey.1 console.4 \
 	console_ioctls.4 keymaps.5 \
-	getkeycodes.8 setkeycodes.8 resizecons.8 \
+	getkeycodes.8 setkeycodes.8  \
 	unicode_start.1 unicode_stop.1 \
 	vt-is-UTF8.1 showcfont.1 charset.1 \
 	openvt.1 kbdrate.8 setvesablank.8 \
diff -ruN console-tools-0.2.3-old/doc/man/Makefile.in console-tools-0.2.3/doc/man/Makefile.in
--- console-tools-0.2.3-old/doc/man/Makefile.in	2003-01-22 11:38:03.000000000 +0000
+++ console-tools-0.2.3/doc/man/Makefile.in	2003-01-22 11:40:37.000000000 +0000
@@ -97,7 +97,6 @@
 POFILES = @POFILES@
 POSUB = @POSUB@
 RANLIB = @RANLIB@
-RESIZECONS = @RESIZECONS@
 TRANSDIR = @TRANSDIR@
 UNICODEDATA = @UNICODEDATA@
 UNICODELISTS = @UNICODELISTS@
@@ -111,7 +110,7 @@
 
 AUTOMAKE_OPTIONS = foreign
 
-KBD_UNSUBST_MAN = chvt.1 deallocvt.1 fgconsole.1 	kbd_mode.1 psfaddtable.1 	psfgettable.1 psfstriptable.1 setleds.1 	setmetamode.1 showkey.1 console.4 	console_ioctls.4 keymaps.5 	getkeycodes.8 setkeycodes.8 resizecons.8 	unicode_start.1 unicode_stop.1 	vt-is-UTF8.1 showcfont.1 charset.1 	openvt.1 kbdrate.8	setvesablank.8 codepage.1 mk_modmap.8 screendump.1 splitfont.1 font2psf.1 vcstime.8 writevt.8 fix_bs_and_del.1 
+KBD_UNSUBST_MAN = chvt.1 deallocvt.1 fgconsole.1 	kbd_mode.1 psfaddtable.1 	psfgettable.1 psfstriptable.1 setleds.1 	setmetamode.1 showkey.1 console.4 	console_ioctls.4 keymaps.5 	getkeycodes.8 setkeycodes.8 	unicode_start.1 unicode_stop.1 	vt-is-UTF8.1 showcfont.1 charset.1 	openvt.1 kbdrate.8	setvesablank.8 codepage.1 mk_modmap.8 screendump.1 splitfont.1 font2psf.1 vcstime.8 writevt.8 fix_bs_and_del.1 
 
 
 KBD_SUBST_MAN = consolechars.8 dumpkeys.1 loadkeys.1 
diff -ruN console-tools-0.2.3-old/doc/man/Makefile.in.rej console-tools-0.2.3/doc/man/Makefile.in.rej
--- console-tools-0.2.3-old/doc/man/Makefile.in.rej	2003-01-22 11:38:03.000000000 +0000
+++ console-tools-0.2.3/doc/man/Makefile.in.rej	1970-01-01 01:00:00.000000000 +0100
@@ -1,49 +0,0 @@
-***************
-*** 139,147 ****
-  all: all-redirect
-  .SUFFIXES:
-  $(srcdir)/Makefile.in: Makefile.am $(top_srcdir)/configure.in $(ACLOCAL_M4) 
-- 	cd $(top_srcdir) && $(AUTOMAKE) --foreign --include-deps doc/man/Makefile
-  
-- Makefile: $(srcdir)/Makefile.in  $(top_builddir)/config.status
-  	cd $(top_builddir) \
-  	  && CONFIG_FILES=$(subdir)/$@ CONFIG_HEADERS= $(SHELL) ./config.status
-  
---- 139,147 ----
-  all: all-redirect
-  .SUFFIXES:
-  $(srcdir)/Makefile.in: Makefile.am $(top_srcdir)/configure.in $(ACLOCAL_M4) 
-+ 	cd $(top_srcdir) && $(AUTOMAKE) --foreign doc/man/Makefile
-  
-+ Makefile: $(srcdir)/Makefile.in  $(top_builddir)/config.status $(BUILT_SOURCES)
-  	cd $(top_builddir) \
-  	  && CONFIG_FILES=$(subdir)/$@ CONFIG_HEADERS= $(SHELL) ./config.status
-  
-***************
-*** 294,303 ****
-  subdir = doc/man
-  
-  distdir: $(DISTFILES)
-  	@for file in $(DISTFILES); do \
-  	  d=$(srcdir); \
-  	  if test -d $$d/$$file; then \
-- 	    cp -pr $$/$$file $(distdir)/$$file; \
-  	  else \
-  	    test -f $(distdir)/$$file \
-  	    || ln $$d/$$file $(distdir)/$$file 2> /dev/null \
---- 294,308 ----
-  subdir = doc/man
-  
-  distdir: $(DISTFILES)
-+ 	here=`cd $(top_builddir) && pwd`; \
-+ 	top_distdir=`cd $(top_distdir) && pwd`; \
-+ 	distdir=`cd $(distdir) && pwd`; \
-+ 	cd $(top_srcdir) \
-+ 	  && $(AUTOMAKE) --include-deps --build-dir=$$here --srcdir-name=$(top_srcdir) --output-dir=$$top_distdir --foreign doc/man/Makefile
-  	@for file in $(DISTFILES); do \
-  	  d=$(srcdir); \
-  	  if test -d $$d/$$file; then \
-+ 	    cp -pr $$d/$$file $(distdir)/$$file; \
-  	  else \
-  	    test -f $(distdir)/$$file \
-  	    || ln $$d/$$file $(distdir)/$$file 2> /dev/null \
diff -ruN console-tools-0.2.3-old/doc/man/#openvt.1# console-tools-0.2.3/doc/man/#openvt.1#
--- console-tools-0.2.3-old/doc/man/#openvt.1#	2003-01-22 11:38:03.000000000 +0000
+++ console-tools-0.2.3/doc/man/#openvt.1#	1970-01-01 01:00:00.000000000 +0100
@@ -1,85 +0,0 @@
-
-
-
-.\" Copyright 1994-95 Jon Tombs (jon@gtex02.us.es, jon@robots.ox.ac.uk)
-.\" May be distributed under the GNU General Public License
-.TH OPENVT 1 "19 Jul 1996" "Console tools" "Linux User's Manual"
-.SH NAME
-openvt \- start a program on a new virtual terminal (VT).
-.SH SYNOPSIS
-.B openvt 
-[\-c vtnumber] [\-s] [\-u] [\-l] [\-v] [\-\-] command command_options
-.SH DESCRIPTION
-.B openvt
-will find the first available VT, and run on it the given 
-.BR command
-with the given 
-.BR "command options",
-standard input, output and error are directed to that terminal. The current
-search path ($PATH) is used to find the requested command. If no command is
-specified then the environment variable $SHELL is used.
-.SS OPTIONS
-.TP
-.I "\-c vtnumber" 
-Use the given VT number and not the first available. Note you
-must have write access to the supplied VT for this to work.
-.TP
-.I "\-s"
-Switch to the new VT when starting the command. The VT of the new command
-will be made the new current VT.
-.TP
-.I "\-u"
-Figure out the owner of the current VT, and run login as that user.
-Suitable to be called by init. Shouldn't be used with -c or -l.
-.TP
-.I "\-l"
-Make the command a login shell. A \- is prepended to the name of the command
-to be executed.
-.TP
-.I "\-v"
-Be a bit more verbose.
-.TP
-.I "\-w"
-wait for command to complete. If \-w and \-s are used together then
-.B openvt
-will switch back to the controlling terminal when the command completes.
-.TP
-.I "\-\-" 
-end of options to
-.BR openvt .
-.SH NOTE
-If
-.B openvt
-is compiled with a POSIX (Gnu) getopt() and you wish to set
-options to the command to be run, then you must supply
-the end of options \-\- flag before the command.
-.SH EXAMPLES
-.B openvt
-can be used to start a shell on the next free VT, by using the command:
-.TP
-.I "openvt bash"
-.TP
-
-To start the shell as a login shell, use:
-.TP
-.I "openvt -l bash"
-.TP
-
-To get a long listing you must supply the \-\- separator:
-.TP
-.I "openvt -- ls -l"
-.BR
-
-
-.SH "SEE ALSO"
-.BR chvt (1),
-.BR login (1).
-
-.SH AUTHOR
-Jon Tombs <jon@gtex02.us.es or jon@robots.ox.ac.uk>
-
-.SH HISTORY
-Earlier, openvt was called open.  It was written by Jon Tombs
-<jon@gtex02.us.es or jon@robots.ox.ac.uk>. The
-.br
-\-w idea from "sam".
diff -ruN console-tools-0.2.3-old/examples/Makefile.in console-tools-0.2.3/examples/Makefile.in
--- console-tools-0.2.3-old/examples/Makefile.in	2003-01-22 11:38:03.000000000 +0000
+++ console-tools-0.2.3/examples/Makefile.in	2003-01-22 11:38:03.000000000 +0000
@@ -97,7 +97,6 @@
 POFILES = @POFILES@
 POSUB = @POSUB@
 RANLIB = @RANLIB@
-RESIZECONS = @RESIZECONS@
 TRANSDIR = @TRANSDIR@
 UNICODELISTS = @UNICODELISTS@
 USE_INCLUDED_LIBINTL = @USE_INCLUDED_LIBINTL@
diff -ruN console-tools-0.2.3-old/Makefile.in console-tools-0.2.3/Makefile.in
--- console-tools-0.2.3-old/Makefile.in	2003-01-22 11:38:03.000000000 +0000
+++ console-tools-0.2.3/Makefile.in	2003-01-22 11:38:03.000000000 +0000
@@ -97,7 +97,6 @@
 POFILES = @POFILES@
 POSUB = @POSUB@
 RANLIB = @RANLIB@
-RESIZECONS = @RESIZECONS@
 TRANSDIR = @TRANSDIR@
 UNICODEDATA = @UNICODEDATA@
 UNICODELISTS = @UNICODELISTS@
diff -ruN console-tools-0.2.3-old/po/POTFILES.in console-tools-0.2.3/po/POTFILES.in
--- console-tools-0.2.3-old/po/POTFILES.in	1999-04-08 22:29:42.000000000 +0100
+++ console-tools-0.2.3/po/POTFILES.in	2003-01-22 11:38:03.000000000 +0000
@@ -25,7 +25,6 @@
 vttools/screendump.c
 vttools/setvesablank.c
 vttools/vt-is-UTF8.c
-vttools/resizecons.c
 vttools/deallocvt.c
 vttools/fgconsole.c
 vttools/chvt.c
diff -ruN console-tools-0.2.3-old/vttools/Makefile.am console-tools-0.2.3/vttools/Makefile.am
--- console-tools-0.2.3-old/vttools/Makefile.am	1999-04-08 22:29:40.000000000 +0100
+++ console-tools-0.2.3/vttools/Makefile.am	2003-01-22 11:38:03.000000000 +0000
@@ -4,12 +4,11 @@
 INCLUDES = -I$(top_srcdir)/include
 
 bin_PROGRAMS = chvt deallocvt writevt fgconsole screendump setvesablank \
-	vcstime vt-is-UTF8 openvt @RESIZECONS@
-EXTRA_PROGRAMS = resizecons
+	vcstime vt-is-UTF8 openvt 
+EXTRA_PROGRAMS = 
 
 LDADD = ../lib/libctlocal.a ../lib/libconsole.la ../lib/libctutils.la
 
 vcstime_LDADD = ../lib/libctlocal.a
 screendump_LDADD = ../lib/libctlocal.a ../lib/libctutils.la
 writevt_LDADD = ../lib/libctlocal.a ../lib/libctutils.la
-resizecons_LDADD =  ../lib/libcfont.la $(LDADD)
diff -ruN console-tools-0.2.3-old/vttools/Makefile.in console-tools-0.2.3/vttools/Makefile.in
--- console-tools-0.2.3-old/vttools/Makefile.in	2003-01-22 11:38:03.000000000 +0000
+++ console-tools-0.2.3/vttools/Makefile.in	2003-01-22 11:38:03.000000000 +0000
@@ -97,7 +97,6 @@
 POFILES = @POFILES@
 POSUB = @POSUB@
 RANLIB = @RANLIB@
-RESIZECONS = @RESIZECONS@
 TRANSDIR = @TRANSDIR@
 UNICODEDATA = @UNICODEDATA@
 UNICODELISTS = @UNICODELISTS@
@@ -112,16 +111,15 @@
 AUTOMAKE_OPTIONS = foreign
 INCLUDES = -I$(top_srcdir)/include
 
-bin_PROGRAMS = chvt deallocvt writevt fgconsole screendump setvesablank 	vcstime vt-is-UTF8 openvt @RESIZECONS@
+bin_PROGRAMS = chvt deallocvt writevt fgconsole screendump setvesablank 	vcstime vt-is-UTF8 openvt 
 
-EXTRA_PROGRAMS = resizecons
+EXTRA_PROGRAMS = 
 
 LDADD = ../lib/libctlocal.a ../lib/libconsole.la ../lib/libctutils.la
 
 vcstime_LDADD = ../lib/libctlocal.a
 screendump_LDADD = ../lib/libctlocal.a ../lib/libctutils.la
 writevt_LDADD = ../lib/libctlocal.a ../lib/libctutils.la
-resizecons_LDADD = ../lib/libcfont.la $(LDADD)
 mkinstalldirs = $(SHELL) $(top_srcdir)/mkinstalldirs
 CONFIG_HEADER = ../config.h
 CONFIG_CLEAN_FILES = 
@@ -132,11 +130,6 @@
 CPPFLAGS = @CPPFLAGS@
 LDFLAGS = @LDFLAGS@
 LIBS = @LIBS@
-resizecons_SOURCES = resizecons.c
-resizecons_OBJECTS =  resizecons.o
-resizecons_DEPENDENCIES =  ../lib/libcfont.la ../lib/libctlocal.a \
-../lib/libconsole.la ../lib/libctutils.la
-resizecons_LDFLAGS = 
 chvt_SOURCES = chvt.c
 chvt_OBJECTS =  chvt.o
 chvt_LDADD = $(LDADD)
@@ -198,10 +191,10 @@
 TAR = tar
 GZIP_ENV = --best
 DEP_FILES =  .deps/chvt.P .deps/deallocvt.P .deps/fgconsole.P \
-.deps/openvt.P .deps/resizecons.P .deps/screendump.P \
+.deps/openvt.P  .deps/screendump.P \
 .deps/setvesablank.P .deps/vcstime.P .deps/vt-is-UTF8.P .deps/writevt.P
-SOURCES = resizecons.c chvt.c deallocvt.c writevt.c fgconsole.c screendump.c setvesablank.c vcstime.c vt-is-UTF8.c openvt.c
-OBJECTS = resizecons.o chvt.o deallocvt.o writevt.o fgconsole.o screendump.o setvesablank.o vcstime.o vt-is-UTF8.o openvt.o
+SOURCES =  chvt.c deallocvt.c writevt.c fgconsole.c screendump.c setvesablank.c vcstime.c vt-is-UTF8.c openvt.c
+OBJECTS = chvt.o deallocvt.o writevt.o fgconsole.o screendump.o setvesablank.o vcstime.o vt-is-UTF8.o openvt.o
 
 all: all-redirect
 .SUFFIXES:
@@ -271,10 +264,6 @@
 
 maintainer-clean-libtool:
 
-resizecons: $(resizecons_OBJECTS) $(resizecons_DEPENDENCIES)
-	@rm -f resizecons
-	$(LINK) $(resizecons_LDFLAGS) $(resizecons_OBJECTS) $(resizecons_LDADD) $(LIBS)
-
 chvt: $(chvt_OBJECTS) $(chvt_DEPENDENCIES)
 	@rm -f chvt
 	$(LINK) $(chvt_LDFLAGS) $(chvt_OBJECTS) $(chvt_LDADD) $(LIBS)
diff -ruN console-tools-0.2.3-old/vttools/resizecons.c console-tools-0.2.3/vttools/resizecons.c
--- console-tools-0.2.3-old/vttools/resizecons.c	1999-04-08 23:11:15.000000000 +0100
+++ console-tools-0.2.3/vttools/resizecons.c	1970-01-01 01:00:00.000000000 +0100
@@ -1,567 +0,0 @@
-/*
- * resizecons.c - change console video mode
- *
- * Version 0.90
- *
- * How to use this:
- *
- * 1. Get svgalib, make restoretextmode, put it somewhere in your path.
- * 2. Put vga=ask in /etc/lilo/config, boot your machine a number of times,
- *    each time with a different vga mode, and run the command
- *        "restoretextmode -w COLSxROWS".
- *    For me this resulted in the files 80x25, 80x28, 80x50, 80x60, 100x40,
- *    132x25, 132x28, 132x44. Put these files in /usr/lib/kbd/videomodes
- *    (or in your current dir).
- * 3. Now "resizecons COLSxROWS" will change your video mode. (Assuming you have
- *    an appropriate kernel, and svgalib works for your video card.)
- *
- * Note: this is experimental, but it works for me. Comments are welcome.
- * You may have to be root to get the appropriate ioperm permissions.
- * It is not safe to make this program suid root.
- *
- * aeb@cwi.nl - 940924
- *
- * Harm Hanemaaijer added the -lines option, which reprograms the
- * number of scanlines. He writes:
- *
- * Added -lines option, which reprograms the number of scanlines and
- * the font height of the VGA hardware with register I/O, so that
- * switching is possible between textmodes with different numbers
- * of lines, in a VGA compatible way. It should work for 132 column
- * modes also, except that number of columns cannot be changed.
- *
- * Standard VGA textmode uses a 400 scanline screen which is refreshed
- * at 70 Hz. The following modes are supported that use this vertical
- * resolution (C is the number of columns, usually 80 or 132).
- *
- *	mode		font height
- *	C x 25		16
- *	C x 28		14
- *	C x 36		11	(non-standard height)
- *	C x 44		9	(8-line fonts are a good match)
- *	C x 50		8
- *
- * The following modes are supported with a 480 scanline resolution,
- * refresh at 60 Hz. Some not quite VGA compatible displays may not
- * support this (it uses the same vertical timing as standard VGA
- * 640x480x16 graphics mode).
- *
- *	mode		font height
- *	C x 30		16
- *	C x 34		14
- *	C x 40		12	(non-standard height)
- *	C x 60		8
- *
- * Two 12-line fonts are already in the consolefonts directory,
- * namely lat1-12.psf and lat2-12.psf.
- * For the 36 lines mode (11 line font), lat1-10.psf and lat2-10.psf
- * can be used.
- * 
- * hhanemaa@cs.ruu.nl - 941028
- * 
- * Notes:
- *
- * In the consolefonts directory there is 'default8x9' font file but
- * no 'default8x8'. Why is this? The standard VGA BIOS has an 8-line
- * font, and they are much more common in SVGA modes (e.g. 50 and 60
- * row modes). It is true that standard VGA textmode uses effectively
- * 9 pixel wide characters, but that has nothing to do with the font
- * data.
- */
-
-#include <stdlib.h>
-#include <unistd.h>
-#include <fcntl.h>
-#include <stdio.h>
-#include <errno.h>
-#include <signal.h>
-#include <string.h>
-#include <sys/ioctl.h>
-#if (__GNU_LIBRARY__ >= 6)
-# include <sys/perm.h>
-#else
-# include <linux/types.h>
-# include <linux/termios.h>
-#endif
-#include <linux/vt.h>
-
-#include <getopt.h>
-#include <lct/local.h>
-#include <lct/utils.h>
-#include <lct/console.h>
-#include <lct/font.h>				  /* findvideomode() */
-
-#define MODE_RESTORETEXTMODE	0
-#define MODE_VGALINES		1
-
-/* VGA textmode register tweaking. */
-static void vga_init_io();
-static void vga_400_scanlines();
-static void vga_480_scanlines();
-static void vga_set_fontheight( int );
-static int vga_get_fontheight();
-static void vga_set_cursor( int, int );
-static void vga_set_verticaldisplayend_lowbyte( int );
-
-static char *progname;
-
-static void usage(char *progname) 
-{
-  printf(_("Usage:  %s COLSxROWS\n"
-	   "        %s COLS ROWS\n"
-	   "        %s -lines ROWS, with ROWS one of 25, 28, 30, 34, 36, 40, 44, 50, 60\n"), 
-	  progname, progname, progname);
-}
-
-static void parse_cmdline (int argc, char **argv, 
-		    int *mode, int *rows, int *cols)
-{
-  int c;
-  char *p; 
-  const struct option long_opts[] =
-  {
-    { "cols", 1, NULL, 'c' },
-    { "lines", 1, NULL, 'l'},
-    { "rows", 1, NULL, 'r'},
-    { "help", no_argument, NULL, 'h' },
-    { "version", no_argument, NULL, 'v' },
-    { NULL, 0, NULL, 0 }
-  };
-
-  while ( (c = getopt_long (argc, argv, "-vhc:r:l:", long_opts, NULL)) != EOF)
-    {
-      switch (c)
-	{
-	case 'h':
-	  usage(progname);
-	  exit(0);
-	case 'v':
-	  version(progname);
-	  exit(0);
-	case 'c':
-	  *cols = atoi(optarg);
-	  if (*cols == 0)
-	    {
-	      fprintf(stderr, _("%s: Invalid number of columns: %s\n"),
-		      progname, optarg);
-	      exit(1);
-	    }
-	  break;
-	case 'r':
-	  *rows = atoi(optarg);
-	  if (*rows == 0)
-	    {
-	      fprintf(stderr,_("%s: Invalid number of rows: %s\n"),
-		      progname, optarg);
-	      exit(1);
-	    }
-	  break;
-	case 'l':			/* lines */
-	  *rows = atoi(optarg);
-	  *mode = MODE_VGALINES;
-	  break;
-	case '?':
-	  usage(progname);
-	  exit(1);
-	}
-  }
-
-   
-  if (argc == optind +1) /* COLSxROWS case */
-    {
-      if (*rows || *cols)
-	{
-	  fprintf(stderr,_("%s: Invalid arguments"),
-		  progname);
-	  exit(1);
-	}
-      if ((p = index (argv[optind],'x')) != 0)
-	{
-	  *cols = atoi (argv[optind]);
-	  *rows = atoi (p+1);
-	  return;
-	}
-      else
-	{
-	  fprintf(stderr,_("%s: Unrecognized argument"),
-		  progname);
-	  exit(1);
-	}
-      return;
-     
-    }
-   
-  if (argc == optind +2) /* COLS ROWS case */
-    {
-      if (*rows || *cols)
-	{
-	  fprintf(stderr,_("%s: Invalid arguments"),
-		  progname);
-	  exit(1);
-	}
-      *cols = atoi(argv[optind]);
-      *rows = atoi(argv[optind+1]);
-      return;
-    }
-       
-  if (argc == optind && *rows && *cols)
-    return;
-   
-  fprintf (stderr, _("%s: bad number of arguments\n"),
-	   progname);
-  exit(1);
-}
-  
-
-int main(int argc, char* argv[]) 
-{
-  int rr = 0, cc = 0, fd, i, mode = MODE_RESTORETEXTMODE;
-  struct vt_sizes vtsizes;
-  struct vt_stat vtstat;
-  struct winsize winsize;
-  char tty[12], cmd[80], infile[1024];
-  FILE *fin;
-  char *defaultfont;
-  char pathname[1024];
-
-  setuplocale();
-  
-  progname = strip_path(argv[0]);
-  parse_cmdline (argc, argv, &mode, &rr, &cc);
-  
-  if (mode == MODE_RESTORETEXTMODE) 
-    {
-      /* prepare for: restoretextmode -r 80x25 */
-      sprintf(infile, "%dx%d", cc, rr);
-      fin = findvideomode(infile, pathname, sizeof(pathname), stdin, NULL);
-      if (!fin) 
-	{
-	  fprintf(stderr, _("%s: cannot find videomode file %s\n"), progname, infile);
-	  exit(EX_NOINPUT);
-	}
-      fclose(fin);
-    }
-
-  if (-1 == (fd = get_console_fd(NULL))) exit (EX_OSERR);
-  
-  if(ioctl(fd, TIOCGWINSZ, &winsize)) 
-    {
-      perror("TIOCGWINSZ");
-      exit(EX_OSERR);
-    }
-  
-  if (mode == MODE_VGALINES) 
-    {
-      /* Get the number of columns. */
-      cc = winsize.ws_col;
-      if (rr != 25 && rr != 28 && rr !=30 && rr != 34 && rr != 36
-	  && rr != 40 && rr != 44 && rr != 50 && rr != 60) 
-	{
-	  fprintf(stderr, _("Invalid number of lines\n"));
-	  exit(EX_DATAERR);
-	}
-    }
-  
-  if(ioctl(fd, VT_GETSTATE, &vtstat)) 
-    {
-      perror("VT_GETSTATE");
-      exit(EX_OSERR);
-    }
-
-  vtsizes.v_rows = rr;
-  vtsizes.v_cols = cc;
-  vtsizes.v_scrollsize = 0;
-  
-  vga_init_io();		/* maybe only if (mode == MODE_VGALINES) */
-  
-  if(ioctl(fd, VT_RESIZE, &vtsizes)) 
-    {
-      perror("VT_RESIZE");
-      exit(EX_OSERR);
-    }
-
-    if (mode == MODE_VGALINES) 
-    {
-      /* Program the VGA registers. */
-      int scanlines_old;
-      int scanlines_new;
-      int fontheight;
-      
-      if (winsize.ws_row == 25 || winsize.ws_row == 28 ||
-	  winsize.ws_row == 36 || winsize.ws_row == 44 ||
-	  winsize.ws_row == 50)
-	  scanlines_old = 400;
-      else
-	  scanlines_old = 480;
-      
-      if (rr == 25 || rr == 28 || rr == 36 || rr == 44 || rr == 50)
-	  scanlines_new = 400;
-      else
-	  scanlines_new = 480;
-      
-      /* Switch to 400 or 480 scanline vertical timing if required. */
-      if (scanlines_old != 400 && scanlines_new == 400)
-	  vga_400_scanlines();
-      
-      if (scanlines_old != 480 && scanlines_new == 480)
-	  vga_480_scanlines();
-      
-      switch (rr) 
-	{
-	case 25 : fontheight = 16; break;
-	case 28 : fontheight = 14; break;
-        case 30 : fontheight = 16; break;
-        case 34 : fontheight = 14; break;
-        case 36 : fontheight = 12; break;
-        case 40 : fontheight = 12; break;
-        case 44 : fontheight = 9; break;
-        case 50 : fontheight = 8; break;
-        case 60 : fontheight = 8; break;
-        default : fontheight = 8; break;
-	}
-      
-      /* Set the VGA character height. */
-      vga_set_fontheight(fontheight);
-      
-      /* Set the line offsets within a character cell of the cursor. */
-      if (fontheight >= 10)
-	  vga_set_cursor(fontheight - 3, fontheight - 2);
-      else
-	  vga_set_cursor(fontheight - 2, fontheight - 1);
-      
-      /*
-       * If there are a few unused scanlines at the bottom of the
-       * screen, make sure they are not displayed, otherwise
-       * there is a annoying changing partial line at the bottom.
-       */
-      vga_set_verticaldisplayend_lowbyte((fontheight * rr - 1) & 0xff);
-      printf(_("Old mode: %dx%d  New mode: %dx%d\n"), winsize.ws_col,
-	     winsize.ws_row, cc, rr);
-      printf(_("Old #scanlines: %d  New #scanlines: %d  Character height: %d\n"),
-	     scanlines_old, scanlines_new, fontheight);
-    }
-
-  if (mode == MODE_RESTORETEXTMODE) 
-    {
-      /* do: restoretextmode -r 25x80 */
-      sprintf(cmd, "restoretextmode -r %s\n", pathname);
-      errno = 0;
-      if(system(cmd)) 
-	{
-	  if(errno)
-	      perror("restoretextmode");
-	  fprintf(stderr, _("%s: the command `%s' failed\n"), progname, cmd);
-	  exit(1);
-	}
-    }
-
-  /*
-   * for i in /dev/tty[0-9] /dev/tty[0-9][0-9]
-   * do
-   *     stty rows $rr cols $cc < $i
-   * done
-   */
-  winsize.ws_row = rr;
-  winsize.ws_col = cc;
-  for (i=0; i<16; i++)
-      if (vtstat.v_state & (1<<i)) 
-         {
-	   sprintf(tty, "/dev/tty%d", i);
-	   if ((fd = open(tty, 0)) > 0) 
-	     {
-	       if(ioctl(fd, TIOCSWINSZ, &winsize))
-		   perror("TIOCSWINSZ");
-	       close(fd);
-	     }
-	 }
-
-  /* do: setfont default8x16 */
-  /* (other people might wish other fonts - this should be settable) */
-  
-  /* We read the VGA font height register to be sure. */
-  /* There isn't much consistency in this. */
-  switch (vga_get_fontheight()) 
-    {
-    case 8 :
-    case 9 : defaultfont = "default8x9"; break;
-    case 10 : defaultfont = "lat1-10.psf"; break;
-    case 11 :
-    case 12 : defaultfont = "lat1-12.psf"; break;
-    case 13 :
-    case 14 : defaultfont = "iso01.f14"; break;
-    case 15 :
-    case 16 :
-    default : defaultfont = "default8x16"; break;
-    }
-
-  sprintf(cmd, "setfont %s", defaultfont);
-  errno = 0;
-  if(system(cmd)) 
-    {
-      if(errno)
-	  perror("setfont");
-      fprintf(stderr, _("%s: the command `%s' failed\n"), progname, cmd);
-      exit(1);
-    }
-
-  fprintf(stderr,
-	  _("%s: don't forget to change TERM (maybe to con%dx%d or linux-%dx%d)\n"),
-	  progname, cc, rr, cc, rr);
-  if (getenv("LINES") || getenv("COLUMNS"))
-      fprintf(stderr,
-	      _("Also the variables LINES and COLUMNS may need adjusting.\n"));
-  
-  return 0;
-}
-
-/*
- * The code below is used only with the option `-lines ROWS', and is
- * very hardware dependent, and requires root privileges.
- */
-
-/* Port I/O macros. Note that these are not compatible with the ones */
-/* defined in the kernel header files. */
-
-static inline void outb( int port, int value )
-{
-  __asm__ volatile ("outb %0,%1"
-		    : : "a" ((unsigned char)value), "d" ((unsigned short)port));
-}
-
-static inline int inb( int port )
-{
-  unsigned char value;
-  __asm__ volatile ("inb %1,%0"
-		    : "=a" (value)
-		    : "d" ((unsigned short)port));
-  return value;
-}
-
-
-/* VGA textmode register tweaking functions. */
-
-static int crtcport;
-
-static void vga_init_io() 
-{
-  if (iopl(3) < 0) 
-    {
-      fprintf(stderr, _("%s: cannot get I/O permissions.\n"), progname);
-      exit(1);
-    }
-  crtcport = 0x3d4;
-  if ((inb(0x3cc) & 0x01) == 0)
-      crtcport = 0x3b4;
-}
-
-static void vga_set_fontheight( int h ) 
-{
-  outb(crtcport, 0x09);
-  outb(crtcport + 1, (inb(crtcport + 1) & 0xe0) | (h - 1));
-}
-
-static int vga_get_fontheight() 
-{
-  outb(crtcport, 0x09);
-  return (inb(crtcport + 1) & 0x1f) + 1;
-}
-
-static void vga_set_cursor( int top, int bottom ) 
-{
-  outb(crtcport, 0x0a);
-  outb(crtcport + 1, (inb(crtcport + 1) & 0xc0) | top);
-  outb(crtcport, 0x0b);
-  outb(crtcport + 1, (inb(crtcport + 1) & 0xe0) | bottom);
-}
-
-static void vga_set_verticaldisplayend_lowbyte( int byte ) 
-{
-  /* CRTC register 0x12 */
-  /* vertical display end */
-  outb(crtcport, 0x12);
-  outb(crtcport + 1, byte);
-}
-
-static void vga_480_scanlines() 
-{
-  /* CRTC register 0x11 */
-  /* vertical sync end (also unlocks CR0-7) */
-  outb(crtcport, 0x11);
-  outb(crtcport + 1, 0x0c);
-  
-  /* CRTC register 0x06 */
-  /* vertical total */
-  outb(crtcport, 0x06);
-  outb(crtcport + 1, 0x0b);
-  
-  /* CRTC register 0x07 */
-  /* (vertical) overflow */
-  outb(crtcport, 0x07);
-  outb(crtcport + 1, 0x3e);
-
-  /* CRTC register 0x10 */
-  /* vertical sync start */
-  outb(crtcport, 0x10);
-  outb(crtcport + 1, 0xea);
-
-  /* CRTC register 0x12 */
-  /* vertical display end */
-  outb(crtcport, 0x12);
-  outb(crtcport + 1, 0xdf);
-
-  /* CRTC register 0x15 */
-  /* vertical blank start */
-  outb(crtcport, 0x15);
-  outb(crtcport + 1, 0xe7);
-
-  /* CRTC register 0x16 */
-  /* vertical blank end */
-  outb(crtcport, 0x16);
-  outb(crtcport + 1, 0x04);
-
-  /* Misc Output register */
-  /* Preserver clock select bits and set correct sync polarity */
-  outb(0x3c2, (inb(0x3cc) & 0x0d) | 0xe2);
-}
-
-static void vga_400_scanlines() 
-{
-  /* CRTC register 0x11 */
-  /* vertical sync end (also unlocks CR0-7) */
-  outb(crtcport, 0x11);
-  outb(crtcport + 1, 0x0e);
-  
-  /* CRTC register 0x06 */
-  /* vertical total */
-  outb(crtcport, 0x06);
-  outb(crtcport + 1, 0xbf);
-
-  /* CRTC register 0x07 */
-  /* (vertical) overflow */
-  outb(crtcport, 0x07);
-  outb(crtcport + 1, 0x1f);
-  
-  /* CRTC register 0x10 */
-  /* vertical sync start */
-  outb(crtcport, 0x10);
-  outb(crtcport + 1, 0x9c);
-
-  /* CRTC register 0x12 */
-  /* vertical display end */
-  outb(crtcport, 0x12);
-  outb(crtcport + 1, 0x8f);
-
-  /* CRTC register 0x15 */
-  /* vertical blank start */
-  outb(crtcport, 0x15);
-  outb(crtcport + 1, 0x96);
-  
-  /* CRTC register 0x16 */
-  /* vertical blank end */
-  outb(crtcport, 0x16);
-  outb(crtcport + 1, 0xb9);
-
-  /* Misc Output register */
-  /* Preserver clock select bits and set correct sync polarity */
-  outb(0x3c2, (inb(0x3cc) & 0x0d) | 0x62);
-}
