Ensure that -V works. 

diff -ruN console-tools-0.2.3-old/kbdtools/dumpkeys.c console-tools-0.2.3/kbdtools/dumpkeys.c
--- console-tools-0.2.3-old/kbdtools/dumpkeys.c	2003-02-18 21:23:20.000000000 +0000
+++ console-tools-0.2.3/kbdtools/dumpkeys.c	2003-02-18 21:31:40.000000000 +0000
@@ -539,7 +539,7 @@
 
 int main (int argc, char *argv[]) 
 {
-  const char *short_opts = "hilvsnf1S:c:";
+  const char *short_opts = "hilvVsnf1S:c:";
   const struct option long_opts[] = 
   {
     { "help",	no_argument,		NULL, 'h' },
