diff -ruN console-tools-0.2.3-old/lib/Makefile.in console-tools-0.2.3/lib/Makefile.in
--- console-tools-0.2.3-old/lib/Makefile.in	2003-05-22 14:56:29.000000000 +0100
+++ console-tools-0.2.3/lib/Makefile.in	2003-05-22 15:03:45.000000000 +0100
@@ -116,7 +116,7 @@
 
 INCLUDES = -I$(top_srcdir)/include $(FindfileLctDefs)
 
-lib_LTLIBRARIES = libcfont.la libconsole.la libctutils.la
+lib_LTLIBRARIES = libctutils.la libconsole.la  libcfont.la
 
 noinst_LIBRARIES = libctlocal.a
 
@@ -145,11 +145,11 @@
 AR = ar
 LTLIBRARIES =  $(lib_LTLIBRARIES)
 
-libcfont_la_LDFLAGS = 
+libcfont_la_LDFLAGS =  -L. -lctutils
 libcfont_la_LIBADD = 
 libcfont_la_OBJECTS =  sfm-rd-ascii.lo sfm-wr-ascii.lo sfm-wr-binary.lo \
 sfm-rd-binary.lo sfm-misc.lo psf-header.lo fontdata.lo fontstruct.lo
-libconsole_la_LDFLAGS = 
+libconsole_la_LDFLAGS =  -L. -lctutils
 libconsole_la_LIBADD = 
 libconsole_la_OBJECTS =  acm.lo misc-console-utils.lo ksyms.lo \
 kernelfont.lo testUTF8.lo
