diff -ruN console-tools-0.2.3-old/kbdtools/setleds.c console-tools-0.2.3/kbdtools/setleds.c
--- console-tools-0.2.3-old/kbdtools/setleds.c	2003-07-17 16:44:16.000000000 +0100
+++ console-tools-0.2.3/kbdtools/setleds.c	2003-07-17 16:45:43.000000000 +0100
@@ -156,7 +156,7 @@
     odefflags = ndefflags = ((oflags >> 4) & 7);
     oflags = nflags = (oflags & 7);
 
-    if (argc <= 1) {
+    if ((ndef | nval) == 0) {
 	if (optL) {
 	    nleds = 0xff;
 	    if (ioctl(0, KDSETLED, &nleds)) {
