diff -ruN console-tools-0.2.3-old/screenfonttools/unicode_start console-tools-0.2.3/screenfonttools/unicode_start
--- console-tools-0.2.3-old/screenfonttools/unicode_start	2004-04-18 13:36:42.000000000 +0100
+++ console-tools-0.2.3/screenfonttools/unicode_start	2004-04-18 19:26:14.000000000 +0100
@@ -1,16 +1,21 @@
 #!/bin/sh
 # start unicode mode
 
+if [ ! -e /proc/self/fd/0 ] ; then
+   echo "Won't set unicode mode: Can't determine console type;" >&2
+   echo " Please ensure that /proc is mounted." >&2
+   exit 1
+fi
 
-readlink /proc/self/fd/0 | grep -q -e /dev/vc -e /dev/tty -e /dev/console
+readlink /proc/self/fd/0 | grep -q -e /dev/vc -e '/dev/tty[^p]' -e /dev/console
 if [ $? -eq 1 ]; then
-    echo  "Won't set unicode mode: not a VT."
-    exit 0
+    echo  "Won't set unicode mode: not a VT." >&2
+    exit 1
 fi
 
 
 if /usr/bin/vt-is-UTF8 --quiet ; then
-    echo "unicode_start: Already in UTF8 mode"
+    echo "unicode_start: Already in UTF8 mode" >&2
     exit 0
 fi
 
@@ -32,5 +37,17 @@
 	;;
 esac
 
-# Since 1.2.1 - for older kernels, use: echo -n -e '\033%8'
-echo -n -e '\033%G'
+# set DEVICE_PREFIX depending on devfs/udev
+if [ -d /dev/vc ]; then
+    DEVICE_PREFIX="/dev/vc/"
+    else
+     DEVICE_PREFIX="/dev/tty"
+fi
+
+NUM=`fgconsole --next-available`
+NUM=`expr ${NUM} - 1 `
+for vc in `seq 0 ${NUM}` 
+do
+  # Since 1.2.1 - for older kernels, use: echo -n -e '\033%8'
+  /bin/echo -n -e '\033%G' > ${DEVICE_PREFIX}${vc}
+done
diff -ruN console-tools-0.2.3-old/screenfonttools/unicode_stop console-tools-0.2.3/screenfonttools/unicode_stop
--- console-tools-0.2.3-old/screenfonttools/unicode_stop	2004-04-18 13:36:42.000000000 +0100
+++ console-tools-0.2.3/screenfonttools/unicode_stop	2004-04-18 19:24:03.000000000 +0100
@@ -1,18 +1,35 @@
 #!/bin/sh
 # stop unicode
 
-
-if ! /usr/bin/vt-is-UTF8 --quiet ; then
-    echo "unicode_stop: Already in byte-mode"
-    exit 0
+if [ ! -e /proc/self/fd/0 ] ; then
+   echo "Not setting UTF-8 mode; cannot check console type." >&2
+   echo "  please ensure /proc is mounted." >&2
+   exit 1
 fi
 
-readlink /proc/self/fd/0 | grep -q  -e /dev/vc -e /dev/tty -e /dev/console
+readlink /proc/self/fd/0 | grep -q  -e /dev/vc -e '/dev/tty[^p]' -e /dev/console
 if [ $? -eq 1 ]; then
-    echo "Not a virtual terminal : Not setting UTF-8 mode."
+    echo "Not a virtual terminal : Not setting UTF-8 mode." >&2
+    exit 1
+fi
+
+if ! /usr/bin/vt-is-UTF8 --quiet ; then
+    echo "unicode_stop: Already in byte-mode" >&2
     exit 0
 fi
 
 kbd_mode -a
 
-echo -n -e '\033%@'
+# set DEVICE_PREFIX depending on devfs/udev
+if [ -d /dev/vc ]; then
+    DEVICE_PREFIX="/dev/vc/"
+    else
+    DEVICE_PREFIX="/dev/tty"
+fi
+
+NUM=`fgconsole --next-available`
+NUM=`expr ${NUM} - 1`
+for vc in `seq 0 ${NUM}`
+do 
+   /bin/echo -n -e '\033%@' > ${DEVICE_PREFIX}${vc}
+done
