diff -ruN console-tools-0.2.3-old/kbdtools/loadkeys.y console-tools-0.2.3/kbdtools/loadkeys.y
--- console-tools-0.2.3-old/kbdtools/loadkeys.y	2003-11-11 11:38:35.000000000 +0000
+++ console-tools-0.2.3/kbdtools/loadkeys.y	2003-11-11 11:41:20.000000000 +0000
@@ -509,7 +509,7 @@
 gotf:
   filename = xstrdup(pathname);
   if (!quiet)
-    fprintf(stderr, "Loading %s\n", pathname);
+    fprintf(stdout, "Loading %s\n", pathname);
   if (first_file)
     {
       yyin = f;
