diff -ruN console-tools-0.2.3-old/vttools/writevt.c console-tools-0.2.3/vttools/writevt.c
--- console-tools-0.2.3-old/vttools/writevt.c	2003-07-30 08:05:24.000000000 +0100
+++ console-tools-0.2.3/vttools/writevt.c	2003-07-30 08:06:02.000000000 +0100
@@ -56,6 +56,9 @@
     case 't':
       vterm = optarg;
       break;
+    case 'T':
+      text = optarg;
+      break;
     case '?':
       usage(progname);
       exit(1);
