diff -ruN console-tools-0.2.3-old/lib/findfile.c console-tools-0.2.3/lib/findfile.c
--- console-tools-0.2.3-old/lib/findfile.c	2003-08-12 23:06:14.000000000 +0100
+++ console-tools-0.2.3/lib/findfile.c	2003-08-12 23:14:47.000000000 +0100
@@ -315,6 +315,7 @@
 	  i = close (is_compressed ? file[1] : uncompressed[1]);
 	  assert (-1 != i);
 	  i = fclose (inf);
+	  inf = NULL; /* Signal we've closed this fd . HACK */
 	  assert (-1 != i);
 	}
     } /* if unseekable */
@@ -406,8 +407,10 @@
 		}
 	      else
 		{
-		  i = fclose (inf);
-		  assert (-1 != i);
+		  if (inf != NULL) {
+			  i = fclose (inf);
+			  assert (-1 != i);
+		}
 		}
 	      assert (-1 != i);
 	      i = close (identified[1]);
