diff -ruN console-tools-0.2.3-old/kbdtools/kbdrate.c console-tools-0.2.3/kbdtools/kbdrate.c
--- console-tools-0.2.3-old/kbdtools/kbdrate.c	2004-07-31 10:46:06 +0100
+++ console-tools-0.2.3/kbdtools/kbdrate.c	2004-07-31 12:25:32 +0100
@@ -76,13 +76,6 @@
 #include <sys/file.h>
 #include <sys/ioctl.h>
 
-#if 0
-#include "../defines.h"
-#ifdef HAVE_kd_h
-#include <linux/kd.h>
-#endif
-#endif
-
 #include <lct/local.h>
 #include <lct/utils.h>
 
@@ -91,6 +84,16 @@
 #include <asm/kbio.h>
 #endif
 
+#ifndef KDKBDREP
+/* usually defined in <linux/kd.h> */
+#define KDKBDREP        0x4B52  /* set keyboard delay/repeat rate;
+                                 * actually used values are returned */
+struct kbd_repeat {
+        int delay;        /* in msec; <= 0: don't change */
+        int period;       /* in msec; <= 0: don't change */
+};
+#endif
+
 #if 0
 #include "nls.h"
 #include "version.h"
@@ -107,13 +110,12 @@
 
 static int
 KDKBDREP_ioctl_ok(double rate, int delay, int silent) {
-#ifdef KDKBDREP
 	/* This ioctl is defined in <linux/kd.h> but is not
 	   implemented anywhere - must be in some m68k patches. */
 	struct kbd_repeat kbdrep_s;
 
 	/* don't change, just test */
-	kbdrep_s.rate = -1;
+	kbdrep_s.period = -1;
 	kbdrep_s.delay = -1;
 	if (ioctl( 0, KDKBDREP, &kbdrep_s )) {
 		if (errno == EINVAL)
@@ -124,11 +126,11 @@
 
 	/* do the change */
 	if (rate == 0)				/* switch repeat off */
-		kbdrep_s.rate = 0;
+		kbdrep_s.period = 0;
 	else
-		kbdrep_s.rate  = 1000.0 / rate;	/* convert cps to msec */
-	if (kbdrep_s.rate < 1)
-		kbdrep_s.rate = 1;
+		kbdrep_s.period  = 1000.0 / rate;	/* convert cps to msec */
+	if (kbdrep_s.period < 1)
+		kbdrep_s.period = 1;
 	kbdrep_s.delay = delay;
 	if (kbdrep_s.delay < 1)
 		kbdrep_s.delay = 1;
@@ -139,10 +141,10 @@
 	}
 
 	/* report */
-	if (kbdrep_s.rate == 0)
+	if (kbdrep_s.period == 0)
 		rate = 0;
 	else
-		rate = 1000.0 / (double) kbdrep_s.rate;
+		rate = 1000.0 / (double) kbdrep_s.period;
 
 	if (!silent)
 		printf( _("Typematic Rate set to %.1f cps (delay = %d ms)\n"),
@@ -150,9 +152,6 @@
 
 	return 1;			/* success! */
 
-#else /* no KDKBDREP */
-	return 0;
-#endif /* KDKBDREP */
 }
 
 static int
