diff -ruN console-tools-0.2.3-old/doc/man/Makefile.am console-tools-0.2.3/doc/man/Makefile.am
--- console-tools-0.2.3-old/doc/man/Makefile.am	2004-04-18 20:37:31.000000000 +0100
+++ console-tools-0.2.3/doc/man/Makefile.am	2004-04-18 20:37:32.000000000 +0100
@@ -12,7 +12,8 @@
 	vt-is-UTF8.1 showcfont.1 charset.1 \
 	openvt.1 kbdrate.8 setvesablank.8 \
 	codepage.1 mk_modmap.8 screendump.1 splitfont.1 \
-	font2psf.1 vcstime.8 writevt.8 fix_bs_and_del.1
+	font2psf.1 vcstime.8 writevt.8 fix_bs_and_del.1 \
+	setlogcons.8
 
 KBD_SUBST_MAN = consolechars.8 dumpkeys.1 loadkeys.1 
 
diff -ruN console-tools-0.2.3-old/doc/man/Makefile.in console-tools-0.2.3/doc/man/Makefile.in
--- console-tools-0.2.3-old/doc/man/Makefile.in	2004-04-18 20:37:32.000000000 +0100
+++ console-tools-0.2.3/doc/man/Makefile.in	2004-04-18 20:38:08.000000000 +0100
@@ -126,7 +126,7 @@
 
 AUTOMAKE_OPTIONS = foreign
 
-KBD_UNSUBST_MAN = chvt.1 deallocvt.1 fgconsole.1 	kbd_mode.1 psfaddtable.1 	psfgettable.1 psfstriptable.1 setleds.1 	setmetamode.1 showkey.1 console.4 	console_ioctls.4 keymaps.5 	getkeycodes.8 setkeycodes.8  	unicode_start.1 unicode_stop.1 	vt-is-UTF8.1 showcfont.1 charset.1 	openvt.1 kbdrate.8 setvesablank.8 	codepage.1 mk_modmap.8 screendump.1 splitfont.1 	font2psf.1 vcstime.8 writevt.8 fix_bs_and_del.1
+KBD_UNSUBST_MAN = chvt.1 deallocvt.1 fgconsole.1 	kbd_mode.1 psfaddtable.1 	psfgettable.1 psfstriptable.1 setleds.1 	setmetamode.1 showkey.1 console.4 	console_ioctls.4 keymaps.5 	getkeycodes.8 setkeycodes.8  	unicode_start.1 unicode_stop.1 	vt-is-UTF8.1 showcfont.1 charset.1 	openvt.1 kbdrate.8 setvesablank.8 	codepage.1 mk_modmap.8 screendump.1 splitfont.1 	font2psf.1 vcstime.8 writevt.8 fix_bs_and_del.1 setlogcons.8
 
 
 KBD_SUBST_MAN = consolechars.8 dumpkeys.1 loadkeys.1 
diff -ruN console-tools-0.2.3-old/doc/man/setlogcons.8 console-tools-0.2.3/doc/man/setlogcons.8
--- console-tools-0.2.3-old/doc/man/setlogcons.8	1970-01-01 01:00:00.000000000 +0100
+++ console-tools-0.2.3/doc/man/setlogcons.8	2004-04-18 20:37:32.000000000 +0100
@@ -0,0 +1,26 @@
+.TH SETLOGCONS 8 "18 Apr 2004" "Console tools" "Linux User's Manual"
+
+.SH NAME
+setlogcons \- Send kernel messages to console N 
+.SH SYNOPSIS
+.BI setlogcons
+.BI N
+
+.SH DESCRIPTION
+The command
+.B setlogcons
+.I N
+sets all kernel messges to the console specified as N.
+.P
+By default kernel messages are sent to the current console.
+.P
+To change the level of messages sent, use 
+.B dmesg
+
+.SH "AUTHORS"
+.BI setlogcons 
+was originally written by Andries Brouwer for the kbd package.
+.P
+This manual page was written by Alastair McKinstry.
+.SH "SEE ALSO"
+dmesg(1)
diff -ruN console-tools-0.2.3-old/vttools/Makefile.am console-tools-0.2.3/vttools/Makefile.am
--- console-tools-0.2.3-old/vttools/Makefile.am	2004-04-18 20:37:31.000000000 +0100
+++ console-tools-0.2.3/vttools/Makefile.am	2004-04-18 20:37:32.000000000 +0100
@@ -4,7 +4,7 @@
 INCLUDES = -I$(top_srcdir)/include
 
 bin_PROGRAMS = chvt deallocvt writevt fgconsole screendump setvesablank \
-	vcstime vt-is-UTF8 openvt 
+	vcstime vt-is-UTF8 openvt setlogcons
 EXTRA_PROGRAMS = 
 
 LDADD = ../lib/libctlocal.a ../lib/libconsole.la ../lib/libctutils.la
diff -ruN console-tools-0.2.3-old/vttools/Makefile.in console-tools-0.2.3/vttools/Makefile.in
--- console-tools-0.2.3-old/vttools/Makefile.in	2004-04-18 20:37:32.000000000 +0100
+++ console-tools-0.2.3/vttools/Makefile.in	2004-04-18 20:41:18.000000000 +0100
@@ -127,7 +127,7 @@
 AUTOMAKE_OPTIONS = foreign
 INCLUDES = -I$(top_srcdir)/include
 
-bin_PROGRAMS = chvt deallocvt writevt fgconsole screendump setvesablank 	vcstime vt-is-UTF8 openvt 
+bin_PROGRAMS = chvt deallocvt writevt fgconsole screendump setvesablank 	vcstime vt-is-UTF8 openvt setlogcons
 
 EXTRA_PROGRAMS = 
 
@@ -141,7 +141,7 @@
 CONFIG_CLEAN_FILES = 
 bin_PROGRAMS =  chvt$(EXEEXT) deallocvt$(EXEEXT) writevt$(EXEEXT) \
 fgconsole$(EXEEXT) screendump$(EXEEXT) setvesablank$(EXEEXT) \
-vcstime$(EXEEXT) vt-is-UTF8$(EXEEXT) openvt$(EXEEXT)
+vcstime$(EXEEXT) vt-is-UTF8$(EXEEXT) openvt$(EXEEXT) setlogcons$(EXEEXT)
 PROGRAMS =  $(bin_PROGRAMS)
 
 
@@ -155,6 +155,11 @@
 chvt_DEPENDENCIES =  ../lib/libctlocal.a ../lib/libconsole.la \
 ../lib/libctutils.la
 chvt_LDFLAGS = 
+setlogcons_SOURCES= setlogcons.c
+setlogcons_OBJECTS= setlogcons.$(OBJEXT)
+setlogcons_LDADD= $(LDADD)
+setlogcons_DEPENDENCIES = ../lib/libctlocal.a ../lib/libconsole.la \
+../lib/libctutils.la
 deallocvt_SOURCES = deallocvt.c
 deallocvt_OBJECTS =  deallocvt.$(OBJEXT)
 deallocvt_LDADD = $(LDADD)
@@ -211,9 +216,9 @@
 GZIP_ENV = --best
 DEP_FILES =  .deps/chvt.P .deps/deallocvt.P .deps/fgconsole.P \
 .deps/openvt.P .deps/screendump.P .deps/setvesablank.P .deps/vcstime.P \
-.deps/vt-is-UTF8.P .deps/writevt.P
-SOURCES = chvt.c deallocvt.c writevt.c fgconsole.c screendump.c setvesablank.c vcstime.c vt-is-UTF8.c openvt.c
-OBJECTS = chvt.$(OBJEXT) deallocvt.$(OBJEXT) writevt.$(OBJEXT) fgconsole.$(OBJEXT) screendump.$(OBJEXT) setvesablank.$(OBJEXT) vcstime.$(OBJEXT) vt-is-UTF8.$(OBJEXT) openvt.$(OBJEXT)
+.deps/vt-is-UTF8.P .deps/writevt.P .deps/setlogcons.P
+SOURCES = chvt.c deallocvt.c writevt.c fgconsole.c screendump.c setvesablank.c vcstime.c vt-is-UTF8.c openvt.c setlogcons.c
+OBJECTS = chvt.$(OBJEXT) deallocvt.$(OBJEXT) writevt.$(OBJEXT) fgconsole.$(OBJEXT) screendump.$(OBJEXT) setvesablank.$(OBJEXT) vcstime.$(OBJEXT) vt-is-UTF8.$(OBJEXT) openvt.$(OBJEXT) setlogcons.$(OBJEXT)
 
 all: all-redirect
 .SUFFIXES:
@@ -293,6 +298,10 @@
 	@rm -f chvt$(EXEEXT)
 	$(LINK) $(chvt_LDFLAGS) $(chvt_OBJECTS) $(chvt_LDADD) $(LIBS)
 
+setlogcons$(EXEEXT): $(setlogcons_OBJECTS) $(setlogcons_DEPENDENCIES)
+	@rm -f setlogcons$(EXEEXT)
+	$(LINK) $(setlogcons_LDFLAGS) $(setlogcons_OBJECTS) $(setlogcons_LDADD) $(LIBS)
+
 deallocvt$(EXEEXT): $(deallocvt_OBJECTS) $(deallocvt_DEPENDENCIES)
 	@rm -f deallocvt$(EXEEXT)
 	$(LINK) $(deallocvt_LDFLAGS) $(deallocvt_OBJECTS) $(deallocvt_LDADD) $(LIBS)
diff -ruN console-tools-0.2.3-old/vttools/setlogcons.c console-tools-0.2.3/vttools/setlogcons.c
--- console-tools-0.2.3-old/vttools/setlogcons.c	1970-01-01 01:00:00.000000000 +0100
+++ console-tools-0.2.3/vttools/setlogcons.c	2004-04-18 20:45:11.000000000 +0100
@@ -0,0 +1,36 @@
+/*
+ * setlogcons.c - aeb - 000523
+ *
+ * usage: setlogcons N
+ */
+
+/* Send kernel messages to the current console or to console N */
+#include <stdio.h>
+#include <fcntl.h>
+#include <stdlib.h>
+#include <sys/ioctl.h>
+#include <lct/local.h>
+
+int
+main(int argc, char **argv){
+	int fd, cons;
+	struct { char fn, subarg; } arg;
+
+	setlocale(LC_ALL, "");
+	bindtextdomain(PACKAGE, LOCALEDIR);
+	textdomain(PACKAGE);
+
+	if (argc == 2)
+		cons = atoi(argv[1]);
+	else
+		cons = 0;	/* current console */
+
+	fd = get_console_fd(NULL);
+	arg.fn = 11;		/* redirect kernel messages */
+	arg.subarg = cons;	/* to specified console */
+	if (ioctl(fd, TIOCLINUX, &arg)) {
+		perror("TIOCLINUX");
+		exit(1);
+	}
+	return 0;
+}
