diff -ruN console-tools-0.2.3-old/console-tools.pot console-tools-0.2.3/console-tools.pot
--- console-tools-0.2.3-old/console-tools.pot	2004-06-26 12:17:02.000000000 +0100
+++ console-tools-0.2.3/console-tools.pot	2004-06-26 12:17:44.000000000 +0100
@@ -124,7 +124,7 @@
 
 #: kbdtools/showkey.c:109
 #, c-format
-msgid "(Warnning: Currently running in a pseudoterminal.\n"
+msgid "(Warning: Currently running in a pseudoterminal.)\n"
 msgstr ""
 
 #: kbdtools/showkey.c:110
diff -ruN console-tools-0.2.3-old/kbdtools/showkey.c console-tools-0.2.3/kbdtools/showkey.c
--- console-tools-0.2.3-old/kbdtools/showkey.c	2004-06-26 12:17:01.000000000 +0100
+++ console-tools-0.2.3/kbdtools/showkey.c	2004-06-26 12:18:02.000000000 +0100
@@ -106,7 +106,7 @@
 		return; // Another error
 	}
 	if (major (buf.st_rdev) == 136) {
-		printf (_("(Warnning: Currently running in a pseudoterminal.\n"));
+		printf (_("(Warning: Currently running in a pseudoterminal.)\n"));
 		printf (_("The reported keycodes are probably wrong.\n\n"));
 	}
 }
