diff -ruN console-tools-0.2.3-old/include/lct/koi8.syms.h console-tools-0.2.3/include/lct/koi8.syms.h
--- console-tools-0.2.3-old/include/lct/koi8.syms.h	2004-07-03 15:09:26.000000000 +0300
+++ console-tools-0.2.3/include/lct/koi8.syms.h	2004-07-03 15:16:41.000000000 +0300
@@ -42,32 +42,32 @@
 	{ 0x2551, "" },
 	{ 0x2552, "" },
 	{ 0x0451, "cyrillic_small_letter_io" },
-	{ 0x2553, "ukrainian_cyrillic_small_letter_ie" },  /* koi8-u #164 */
+	{ 0x0454, "ukrainian_cyrillic_small_letter_ie" },  /* koi8-u #164 */
 	{ 0x2554, "" },
-	{ 0x2555, "ukrainian_cyrillic_small_letter_i" },   /* koi8-u #166 */
-	{ 0x2556, "ukrainian_cyrillic_small_letter_yi" },  /* koi8-u #167 */
+	{ 0x0456, "ukrainian_cyrillic_small_letter_i" },   /* koi8-u #166 */
+	{ 0x0457, "ukrainian_cyrillic_small_letter_yi" },  /* koi8-u #167 */
 	{ 0x2557, "" },                        /* 0250 */
 	{ 0x2558, "" },
 	{ 0x2559, "" },
 	{ 0x255a, "" },
 	{ 0x255b, "" },
-	{ 0x255c, "ukrainian_cyrillic_small_letter_ghe_with_upturn" }, /* koi8-u #173 */
+	{ 0x0491, "ukrainian_cyrillic_small_letter_ghe_with_upturn" }, /* koi8-u #173 */
 	{ 0x255d, "" },
 	{ 0x255e, "" },
 	{ 0x255f, "" },                        /* 0260 */
 	{ 0x2560, "" },
 	{ 0x2561, "" },
 	{ 0x0401, "cyrillic_capital_letter_io" },
-	{ 0x2562, "ukrainian_cyrillic_capital_letter_ie" },/* koi8-u #180 */
+	{ 0x0403, "ukrainian_cyrillic_capital_letter_ie" },/* koi8-u #180 */
 	{ 0x2563, "" },
-	{ 0x2564, "ukrainian_cyrillic_capital_letter_i" }, /* koi8-u #182 */
-	{ 0x2565, "ukrainian_cyrillic_capital_letter_yi" },/* koi8-u #183 */
+	{ 0x0406, "ukrainian_cyrillic_capital_letter_i" }, /* koi8-u #182 */
+	{ 0x0407, "ukrainian_cyrillic_capital_letter_yi" },/* koi8-u #183 */
 	{ 0x2566, "" },                        /* 0270 */
 	{ 0x2567, "" },
 	{ 0x2568, "" },
 	{ 0x2569, "" },
 	{ 0x256a, "" },
-	{ 0x256b, "ukrainian_cyrillic_capital_letter_ghe_with_upturn" },/* koi8-u #189 */
+	{ 0x0490, "ukrainian_cyrillic_capital_letter_ghe_with_upturn" },/* koi8-u #189 */
 	{ 0x256c, "" },
 	{ 0x00a9, "copyright" },
 	{ 0x044e, "cyrillic_small_letter_yu" },                        /* 0300 */
