diff -ruN console-tools-0.2.3-old/screenfonttools/unicode_start console-tools-0.2.3/screenfonttools/unicode_start
--- console-tools-0.2.3-old/screenfonttools/unicode_start	2004-07-29 16:27:56.000000000 +0100
+++ console-tools-0.2.3/screenfonttools/unicode_start	2004-07-29 16:34:54.000000000 +0100
@@ -19,9 +19,11 @@
     exit 0
 fi
 
+# We want kbd to be in unicode mode before we run loadkeys; 
+# otherwise confusion over keysyms reigns.
+kbd_mode -u
 dumpkeys | loadkeys --unicode > /dev/null
 
-kbd_mode -u
 
 case $# in
     2)
