diff -ruN console-tools-0.2.3-old/lib/Makefile.am console-tools-0.2.3/lib/Makefile.am
--- console-tools-0.2.3-old/lib/Makefile.am	1999-04-08 22:29:39.000000000 +0100
+++ console-tools-0.2.3/lib/Makefile.am	2004-07-30 14:35:18.000000000 +0100
@@ -8,7 +8,7 @@
 
 INCLUDES = -I$(top_srcdir)/include $(FindfileLctDefs)
 
-lib_LTLIBRARIES = libcfont.la libconsole.la libctutils.la
+lib_LTLIBRARIES = libctutils.la libcfont.la libconsole.la 
 
 noinst_LIBRARIES = libctlocal.a
 
@@ -28,5 +28,5 @@
 
 # Maybe one day libtool will support this:
 # Deps between uninstalled libs:
-#libcfont_la_LIBADD = ./libctutils.la
-#libconsole_la_LIBADD = ./libctutils.la
+libcfont_la_LIBADD = ./libctutils.la
+libconsole_la_LIBADD = ./libctutils.la
diff -ruN console-tools-0.2.3-old/lib/Makefile.in console-tools-0.2.3/lib/Makefile.in
--- console-tools-0.2.3-old/lib/Makefile.in	2004-07-30 14:34:43.000000000 +0100
+++ console-tools-0.2.3/lib/Makefile.in	2004-07-30 14:38:42.000000000 +0100
@@ -131,7 +131,7 @@
 
 INCLUDES = -I$(top_srcdir)/include $(FindfileLctDefs)
 
-lib_LTLIBRARIES = libcfont.la libconsole.la libctutils.la
+lib_LTLIBRARIES = libctutils.la libcfont.la libconsole.la
 
 noinst_LIBRARIES = libctlocal.a
 
@@ -160,11 +160,11 @@
 LTLIBRARIES =  $(lib_LTLIBRARIES)
 
 libcfont_la_LDFLAGS = 
-libcfont_la_LIBADD = 
+libcfont_la_LIBADD = ./libctutils.la
 libcfont_la_OBJECTS =  sfm-rd-ascii.lo sfm-wr-ascii.lo sfm-wr-binary.lo \
 sfm-rd-binary.lo sfm-misc.lo psf-header.lo fontdata.lo fontstruct.lo
 libconsole_la_LDFLAGS = 
-libconsole_la_LIBADD = 
+libconsole_la_LIBADD =  ./libctutils.la
 libconsole_la_OBJECTS =  acm.lo misc-console-utils.lo ksyms.lo \
 kernelfont.lo testUTF8.lo
 libctutils_la_LDFLAGS = 
