diff -ruN console-tools-0.2.3-old/lib/findfile.c console-tools-0.2.3/lib/findfile.c
--- console-tools-0.2.3-old/lib/findfile.c	2004-11-06 19:17:16 +0000
+++ console-tools-0.2.3/lib/findfile.c	2004-11-06 19:18:31 +0000
@@ -687,6 +687,7 @@
     {
       memset (&sa, 0, sizeof (sa));
       sa.sa_handler = chld_handler;
+      sa.sa_flags = SA_RESTART;
       sigaction (SIGCHLD, &sa, &old_sa);
     }
 
