diff -ruN console-tools-0.2.3-old/lib/ksyms.c console-tools-0.2.3/lib/ksyms.c
--- console-tools-0.2.3-old/lib/ksyms.c	2006-04-25 14:09:19.000000000 +0100
+++ console-tools-0.2.3/lib/ksyms.c	2006-04-25 14:28:25.000000000 +0100
@@ -1503,6 +1503,19 @@
 	"SCtrlR"
 };
 
+/* Keysyms whose KTYP is KT_BRL. */
+static const char *brl_syms[] = {
+	"Brl_blank",
+	"Brl_dot1",
+	"Brl_dot2",
+	"Brl_dot3",
+	"Brl_dot4",
+	"Brl_dot5",
+	"Brl_dot6",
+	"Brl_dot7",
+	"Brl_dot8"
+};
+
 #define E(x) { x, sizeof(x) / sizeof(x[0]) }
 
 syms_entry syms[] = {
@@ -1518,7 +1531,9 @@
 	E(ascii_syms),		/* KT_ASCII */
 	E(lock_syms),		/* KT_LOCK */
 	{ 0, 0 },		/* KT_LETTER */
-	E(sticky_syms)		/* KT_SLOCK */
+	E(sticky_syms),		/* KT_SLOCK */
+	{ 0, 0 },		/* */
+	E(brl_syms)		/* KT_BRL */
 };
 
 #undef E
