diff -ruN console-tools-0.2.3-old/doc/man/loadkeys.1.in console-tools-0.2.3/doc/man/loadkeys.1.in
--- console-tools-0.2.3-old/doc/man/loadkeys.1.in	2006-09-05 12:13:24.000000000 +0100
+++ console-tools-0.2.3/doc/man/loadkeys.1.in	2006-09-05 12:21:31.000000000 +0100
@@ -41,9 +41,9 @@
 ) option is given,
 .B loadkeys
 loads a default keymap, probably the file
-.I defkeymap.map
-either in
-.I /usr/share/keymaps
+.I defkeymap.map[.gz]
+typically in 
+.I /usr/share/keymaps/i386/qwerty/
 or in
 .IR /usr/src/linux/drivers/char .
 (Probably the former was user-defined, while the latter
