diff -ruN console-tools-0.2.3-old/screenfonttools/unicode_stop console-tools-0.2.3/screenfonttools/unicode_stop
--- console-tools-0.2.3-old/screenfonttools/unicode_stop	2010-03-02 21:00:17.000000000 +0000
+++ console-tools-0.2.3/screenfonttools/unicode_stop	2010-03-02 21:03:35.000000000 +0000
@@ -1,38 +1,30 @@
 #!/bin/sh
-# stop unicode
 
 if [ ! -e /proc/self/fd/0 ] ; then
-   echo "Not setting UTF-8 mode; cannot check console type." >&2
-   echo "  please ensure /proc is mounted." >&2
+   echo "Won't set ascii mode: Can't determine console type;" >&2
+   echo " Please ensure that /proc is mounted." >&2
    exit 1
 fi
 
-readlink /proc/self/fd/0 | grep -q  -e /dev/vc -e '/dev/tty[^p]' -e /dev/console
-if [ $? -eq 1 ]; then
-    echo "Not a virtual terminal : Not setting UTF-8 mode." >&2
-    exit 1
-fi
-
-if ! /usr/bin/vt-is-UTF8 --quiet ; then
-    echo "unicode_stop: Already in byte-mode" >&2
-    exit 0
-fi
+TTY="`/usr/bin/tty`"
+case "$TTY" in
+	/dev/console|/dev/vc*|/dev/tty[0-9]*)
+		;;
+	*)
+		echo "unicode_stop skipped on $TTY"
+		exit 0
+		;;
+esac
 
+# stop unicode
 kbd_mode -a
-
-# set DEVICE_PREFIX depending on devfs/udev
-if [ -d /dev/vc ]; then
-    DEVICE_PREFIX="/dev/vc/"
-    else
-    DEVICE_PREFIX="/dev/tty"
+if test -t ; then
+	printf '\033%%@'
 fi
+stty -iutf8
 
-NUM=`fgconsole --next-available`
-NUM=`expr ${NUM} - 1`
-# Cope with being called in S mode before getty starts the remaining
-# VTs:
-[ $NUM -eq 1 ] && NUM=6
-for vc in `seq 1 ${NUM}`
-do 
-   /bin/echo -n -e '\033%@' > ${DEVICE_PREFIX}${vc}
-done
+uid="`id -u 2>/dev/null`" ||:
+if [ "$uid" = '0' ]; then
+	[ ! -r "$HOME/.kbd/.keymap_sv" ] ||
+		loadkeys "$HOME/.kbd/.keymap_sv"
+fi
