diff -ur cinepaint/app/cms.c cinepaint-png15/app/cms.c
--- cinepaint/app/cms.c	2008-12-31 18:14:53.000000000 -0500
+++ cinepaint-png15/app/cms.c	2013-03-02 03:55:09.944689984 -0500
@@ -45,7 +45,8 @@
 #include <arpa/inet.h>  /* ntohl */
 #include <oyranos/oyranos_monitor.h>
 #else
-#include <oyranos/oyranos_alpha.h>
+#include <oyranos/oyProfiles_s.h>
+#include <oyranos/alpha/oyranos_alpha.h>
 #endif
 #endif
