From: Andreas Hildebrandt <andreas.hildebrandt@uni-mainz.de>
Date: Tue, 14 Aug 2012 17:23:22 +0200
Subject: Fix compilation of hashMap.h on gcc-4.7

---
 include/BALL/DATATYPE/hashMap.h |    9 ++-------
 1 file changed, 2 insertions(+), 7 deletions(-)

diff --git a/include/BALL/DATATYPE/hashMap.h b/include/BALL/DATATYPE/hashMap.h
index 329db2b..f185b8c 100644
--- a/include/BALL/DATATYPE/hashMap.h
+++ b/include/BALL/DATATYPE/hashMap.h
@@ -236,7 +236,7 @@ namespace BALL
 	template <class Key, class T>
 	const T& HashMap<Key, T>::operator [] (const Key& key) const
 	{
-		ConstIterator it = find(key);
+		ConstIterator it = this->find(key);
 		if (it == Base::end())
 		{
 			throw IllegalKey(__FILE__, __LINE__);
@@ -272,12 +272,7 @@ namespace BALL
 	T& HashMap<Key, T>::operator [] (const Key& key)
 		
 	{
-		Iterator it = find(key);
-		if (it == Base::end())
-		{
-			it = insert(ValueType(key, T())).first;
-		}
-		return it->second;
+		return BALL_MAP_NAME::operator[] (key);
 	}
 
 } // namespace BALL
