=== modified file 'configure.ac'
--- a/configure.ac	2012-04-11 16:54:01 +0000
+++ b/configure.ac	2012-06-12 06:28:06 +0000
@@ -121,7 +121,8 @@
                                   gio-unix-2.0  >= $GIO_UNIX_REQUIRED_VERSION,
                                   [have_tests=yes]
 )
-PKG_CHECK_MODULES(DBUSMENUTESTSVALGRIND, valgrind)
+PKG_CHECK_MODULES(DBUSMENUTESTSVALGRIND, valgrind, have_valgrind=yes, have_valgrind=no)
+AM_CONDITIONAL([HAVE_VALGRIND], [test "x$have_valgrind" = "xyes"])
 ])
 
 AC_SUBST(DBUSMENUTESTS_CFLAGS)

