From: Dominic Hargreaves <dom@earth.li>
Subject: Append $Config{ccflags} to CCFLAGS

Bug-Debian: http://bugs.debian.org/636649

Index: libgd-gd2-perl/Makefile.PL
===================================================================
--- libgd-gd2-perl.orig/Makefile.PL	2011-08-04 21:26:48.000000000 +0100
+++ libgd-gd2-perl/Makefile.PL	2011-08-04 21:27:01.000000000 +0100
@@ -245,7 +245,7 @@
     'PL_FILES'  => {'bdf_scripts/bdf2gdfont.PLS' => 'bdf_scripts/bdf2gdfont.pl',
 		    'GD/Image.pm.PLS' => 'GD/Image.pm',
 		   },
-    'CCFLAGS'    => '-Wformat=0',
+    'CCFLAGS'    => '-Wformat=0 ' . $Config{ccflags},
     'EXE_FILES' => ['bdf_scripts/bdf2gdfont.pl'],
     'AUTHOR'    => 'Lincoln Stein (lstein@cshl.org)',
     'ABSTRACT'  => 'Interface to Gd Graphics Library',
