From: Paul Wise <pabs3@bonedaddy.net>
Date: Mon, 4 Feb 2013 09:23:20 +0800
Subject: Fix python3 compatibility for iotop when installed
Forwarded: Guillaume Chazarain <guichaz@gmail.com>

---
 sbin/iotop | 9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/sbin/iotop b/sbin/iotop
index c4aa720..c5202d3 100755
--- a/sbin/iotop
+++ b/sbin/iotop
@@ -3,14 +3,15 @@
 # Copyright (c) 2007, 2008 Guillaume Chazarain <guichaz@gmail.com>, GPLv2
 # See iotop --help for some help
 
+from __future__ import print_function
 import sys
 
 try:
     from iotop.ui import main
-except ImportError, e:
-    print e
-    print 'To run an uninstalled copy of iotop,'
-    print 'launch iotop.py in the top directory'
+except ImportError as e:
+    print(e)
+    print('To run an uninstalled copy of iotop,')
+    print('launch iotop.py in the top directory')
 else:
     try:
         main()
-- 
1.8.3.rc1

