From: Paul Wise <pabs3@bonedaddy.net>
Date: Sun, 5 May 2013 13:56:20 +0800
Subject: Exit when a SIGTERM signal is received in interactive mode.
Forwarded: Guillaume Chazarain <guichaz@gmail.com>
Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/+source/iotop/+bug/1156754

---
 iotop/ui.py | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/iotop/ui.py b/iotop/ui.py
index bcc22a5..22be478 100644
--- a/iotop/ui.py
+++ b/iotop/ui.py
@@ -477,9 +477,14 @@ class IOTopUI(object):
                                 curses.A_BOLD)
             self.win.refresh()
 
+def signal_handler(signum, frame):
+    sys.exit(0)
+
 def run_iotop_window(win, options):
     if options.batch:
         signal.signal(signal.SIGPIPE, signal.SIG_DFL)
+    else:
+        signal.signal(signal.SIGTERM, signal_handler)
     taskstats_connection = TaskStatsNetlink(options)
     process_list = ProcessList(taskstats_connection, options)
     ui = IOTopUI(win, process_list, options)
-- 
1.8.3.rc1

