Author: Pino Toscano <pino@kde.org>
Description: Identify SuperH correctly.
Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=623281
Forwarded: no
Origin: vendor
Last-Update: 2011-04-27

---
 configure |    6 ++++++
 1 file changed, 6 insertions(+)

--- a/configure
+++ b/configure
@@ -3248,6 +3248,12 @@ if [ -z "${CFG_HOST_ARCH}" ]; then
             ;;
         esac
         ;;
+    *:*:sh4)
+        if [ "$OPT_VERBOSE" = "yes" ]; then
+            echo "    Renesas SuperH 4"
+        fi
+        CFG_HOST_ARCH=sh
+        ;;
     *:*:*)
         if [ "$OPT_VERBOSE" = "yes" ]; then
             echo "    Trying '$UNAME_MACHINE'..."
