Fix out-of-tree-build
--- a/audisp/plugins/prelude/Makefile.am
+++ b/audisp/plugins/prelude/Makefile.am
@@ -24,7 +24,7 @@ CONFIG_CLEAN_FILES = *.rej *.orig
 EXTRA_DIST = au-prelude.conf audisp-prelude.conf
 AUTOMAKE_OPTIONS = no-dependencies
 INCLUDES = -I${top_srcdir} -I${top_srcdir}/lib -I${top_srcdir}/auparse
-LIBS = -L${top_srcdir}/auparse/.libs -lauparse -lprelude 
+LIBS = -L${top_builddir}/auparse/.libs -lauparse -lprelude
 LDADD = -lpthread 
 prog_confdir = $(sysconfdir)/audisp
 prog_conf = audisp-prelude.conf
--- a/audisp/plugins/prelude/Makefile.in
+++ b/audisp/plugins/prelude/Makefile.in
@@ -140,7 +140,7 @@ LDFLAGS = @LDFLAGS@
 LIBOBJS = @LIBOBJS@
 LIBPRELUDE_CFLAGS = @LIBPRELUDE_CFLAGS@
 LIBPRELUDE_LDFLAGS = @LIBPRELUDE_LDFLAGS@
-LIBS = -L${top_srcdir}/auparse/.libs -lauparse -lprelude 
+LIBS = -L${top_builddir}/auparse/.libs -lauparse -lprelude
 LIBTOOL = @LIBTOOL@
 LIBTOOL_DEPS = @LIBTOOL_DEPS@
 LIBWRAP_LIBS = @LIBWRAP_LIBS@
