Replace condrestart with restart for Debian
--- a/system-config-audit/src/server.c
+++ b/system-config-audit/src/server.c
@@ -344,7 +344,7 @@ req_apply (void)
 
   assert (SOCKET_FILENO == STDIN_FILENO);
   /* system() uses "sh -c ...", so the "exec " avoids one fork (). */
-  res = system ("exec /sbin/service auditd condrestart "
+  res = system ("exec /sbin/service auditd restart "
 		STR(SOCKET_FILENO) "</dev/null");
   switch (res)
     {
