From: Bryan Donlan <bdonlan@gmail.com>
Date: Tue, 10 May 2011 16:17:39 -0400
Subject: [PATCH] Fix formatting of ofxdump usage message so help2man output is well-formed

---
 ofxdump/cmdline.ggo |    6 +++---
 1 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/ofxdump/cmdline.ggo b/ofxdump/cmdline.ggo
index 52fba2e..9ba7e07 100644
--- a/ofxdump/cmdline.ggo
+++ b/ofxdump/cmdline.ggo
@@ -3,9 +3,9 @@
 # Version of your program
 # don't use version if you're using automake
 purpose "ofxdump prints to stdout, in human readable form, everything the library 
- understands about a particular file or response, and sends errors to 
- stderr.  To know exactly what the library understands about of a particular
- ofx response file, just call ofxdump on that file."
+understands about a particular file or response, and sends errors to
+stderr.  To know exactly what the library understands about of a particular
+ofx response file, just call ofxdump on that file."
 
 # Options
 # option <long> <short> <desc> {argtype} {typestr="<type descr>"} {default="<default value>"} {required} {argoptional} {multiple}
-- 
