From 352b995a1a0418ea7c60961066b3f86c44fbbaa8 Mon Sep 17 00:00:00 2001
From: "Bernhard R. Link" <brlink@debian.org>
Date: Tue, 14 Jul 2009 15:40:00 +0200
Subject: rpshowall.sh needs bash, so use #!/bin/bash

Bug-Debian: 530176
---
 contrib/rpshowall.sh |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/contrib/rpshowall.sh b/contrib/rpshowall.sh
index cd69ce3..db35c16 100644
--- a/contrib/rpshowall.sh
+++ b/contrib/rpshowall.sh
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/bin/bash
 
 # rpshowall
 # ratpoison script to show all open windows
