Description: Avoid unnecessary dependency on libz
Forwarded: no
Author: Damyan Ivanov <dmn@debian.org>
Reviewed-by: Alessandro Ghedini <al3xbio@gmail.com>
Last-Update: 2012-02-24

--- a/inc/Module/Install/PRIVATE/Net/SSLeay.pm
+++ b/inc/Module/Install/PRIVATE/Net/SSLeay.pm
@@ -115,8 +115,8 @@
         $opts->{optimize} = '-O2 -g';
         push @{ $opts->{lib_links} },
              ($rsaref
-              ? qw( ssl crypto RSAglue rsaref z )
-              : qw( ssl crypto z )
+              ? qw( ssl crypto RSAglue rsaref )
+              : qw( ssl crypto )
              );
 
         if (($Config{cc} =~ /aCC/i) && $^O eq 'hpux') {
