diff -urNad lshw-02.06~/src/core/pcmcia.cc lshw-02.06/src/core/pcmcia.cc
--- lshw-02.06~/src/core/pcmcia.cc	2005-10-13 23:33:55.000000000 +0200
+++ lshw-02.06/src/core/pcmcia.cc	2005-10-18 10:32:20.000000000 +0200
@@ -1007,8 +1007,8 @@
     {
       if (stab[i][0] == 'S')
       {
-	unsigned int pos = stab[i].find(':');
-
+	string::size_type pos = stab[i].find(':');
+	      
 	socketname = "";
 	carddescription = "";
 
