--- a/libs/kworkspace/CMakeLists.txt
+++ b/libs/kworkspace/CMakeLists.txt
@@ -51,6 +51,7 @@ qt4_add_dbus_interface( kworkspace_LIB_S
 kde4_add_library(kworkspace SHARED ${kworkspace_LIB_SRCS})
 
 target_link_libraries(kworkspace  ${KDE4_KDEUI_LIBS} ${X11_LIBRARIES} ${KDE4_PLASMA_LIBS} )
+target_link_libraries(kworkspace  LINK_INTERFACE_LIBRARIES ${KDE4_KDEUI_LIBRARY} )
 
 set_target_properties(kworkspace PROPERTIES VERSION ${GENERIC_LIB_VERSION} SOVERSION ${GENERIC_LIB_SOVERSION} DEFINE_SYMBOL MAKE_KWORKSPACE_LIB)
 install(TARGETS kworkspace EXPORT kdeworkspaceLibraryTargets ${INSTALL_TARGETS_DEFAULT_ARGS} )
--- a/libs/taskmanager/CMakeLists.txt
+++ b/libs/taskmanager/CMakeLists.txt
@@ -50,6 +50,7 @@ if (NOT KDE_PLATFORM_FEATURE_BINARY_COMP
     target_link_libraries(taskmanager kephal)
  endif (NOT KDE_PLATFORM_FEATURE_BINARY_COMPATIBLE_FEATURE_REDUCTION)
 
+target_link_libraries(taskmanager LINK_INTERFACE_LIBRARIES ${QT_QTGUI_LIBRARY} )
 set_target_properties(taskmanager PROPERTIES VERSION ${GENERIC_LIB_VERSION} SOVERSION ${GENERIC_LIB_SOVERSION} )
 install(TARGETS taskmanager EXPORT kdeworkspaceLibraryTargets ${INSTALL_TARGETS_DEFAULT_ARGS} )
 
