From: José Manuel Santamaría Lema <panfaust@gmail.com>
From: Modestas Vainius <modax@debian.org>
Subject: initialize some primitive variables avoiding potential crashes with dbus 1.4.8
Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=623492
Last-Update: 2011-04-24
Forwarded: no
Origin: vendor

--- a/libs/kephal/service/xrandroutputs.cpp
+++ b/libs/kephal/service/xrandroutputs.cpp
@@ -94,7 +94,15 @@ namespace Kephal {
     }
 
     XRandROutput::XRandROutput(XRandROutputs * parent, RROutput rrId)
-            : BackendOutput(parent), m_productId(-1), m_serialNumber(0)
+            : BackendOutput(parent),
+            m_productId(-1),
+            m_serialNumber(0),
+            m_previousConnected(false),
+            m_previousActivated(false),
+            m_previousRotation(RotateNormal),
+            m_previousRate(0),
+            m_previousReflectX(false),
+            m_previousReflectY(false)
     {
         m_outputs = parent;
         m_rrId = rrId;
--- a/libs/kephal/kephal/simpleoutput.cpp
+++ b/libs/kephal/kephal/simpleoutput.cpp
@@ -24,7 +24,13 @@
 namespace Kephal {
 
     SimpleOutput::SimpleOutput(QObject * parent, QString id, QSize size, QPoint position, bool connected, bool activated)
-        : Output(parent)
+        : Output(parent),
+        m_productId(-1),
+        m_serialNumber(0),
+        m_rotation(RotateNormal),
+        m_reflectX(false),
+        m_reflectY(false),
+        m_rate(0)
     {
         m_id = id;
         m_size = size;
@@ -39,11 +45,23 @@ namespace Kephal {
         m_position(0, 0),
         m_connected(false),
         m_activated(false)
+        m_productId(-1),
+        m_serialNumber(0),
+        m_rotation(RotateNormal),
+        m_reflectX(false),
+        m_reflectY(false),
+        m_rate(0)
     {
     }
 
     SimpleOutput::SimpleOutput(QObject * parent, Output * output)
-        : Output(parent)
+        : Output(parent),
+        m_productId(-1),
+        m_serialNumber(0),
+        m_rotation(RotateNormal),
+        m_reflectX(false),
+        m_reflectY(false),
+        m_rate(0)
     {
         m_id = output->id();
         m_size = output->size();
