--- a/kdm/kfrontend/genkdmconf.c
+++ b/kdm/kfrontend/genkdmconf.c
@@ -3154,9 +3154,11 @@ int main(int argc, char **argv)
                 memcmp(oldkde + olen - sizeof(SHR_CONF) + 1,
                        SHR_CONF, sizeof(SHR_CONF)))
             {
+                /*
                 fprintf(stderr,
                         "Warning: --old-kde does not end with " SHR_CONF ". "
                         "Might wreak havoc.\n");
+                */
                 oldkdepfx = oldkde;
             } else
                 ASPrintf((char **)&oldkdepfx,
