--- a/kdm/config.def
+++ b/kdm/config.def
@@ -92,6 +92,12 @@
 <kdmrc>
  &kdm; master configuration file
  _
+ Please note: Settings in this file are sometimes ignored (overridden).
+ The default KDM startup script /etc/init.d/kdm looks in /etc/default/kdm.d
+ for theme-related settings which, if found, take precedence. The possibly 
+ overridden settings are: UseBackground, BackgroundCfg, UseTheme, Theme.
+ See /usr/share/doc/kdm/README.Debian for details
+ _
  Definition: the greeter is the login dialog, i.e., the part of &kdm;
  which the user sees.
  _
