--- a/kdm/kfrontend/kgreeter.cpp
+++ b/kdm/kfrontend/kgreeter.cpp
@@ -451,7 +451,6 @@ void
 KGreeter::putSession(const QString &type, const QString &name, bool hid, const char *exe)
 {
     int prio = exe ? (!strcmp(exe, "default") ? 0 :
-                      !strcmp(exe, "custom") ? 1 :
                       !strcmp(exe, "failsafe") ? 3 : 2) : 2;
     for (int i = 0; i < sessionTypes.size(); i++)
         if (sessionTypes[i].type == type) {
@@ -480,7 +479,6 @@ KGreeter::insertSessions()
                             dsk.readEntry("Exec").toLatin1());
             }
     putSession("default", i18nc("@item:inlistbox session type", "Default"), false, "default");
-    putSession("custom", i18nc("@item:inlistbox session type", "Custom"), false, "custom");
     putSession("failsafe", i18nc("@item:inlistbox session type", "Failsafe"), false, "failsafe");
     qSort(sessionTypes);
     for (int i = 0; i < sessionTypes.size() && !sessionTypes[i].hid; i++) {
