From b798928ac8b9c6acf804b5cd4101d8a2a860654a Mon Sep 17 00:00:00 2001
From: Marco Martin <notmart@gmail.com>
Date: Mon, 25 Jun 2012 14:37:08 +0200
Subject: [PATCH] make sure scripts are executed sorted

---
 libs/plasmagenericshell/scripting/scriptengine.cpp |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/libs/plasmagenericshell/scripting/scriptengine.cpp b/libs/plasmagenericshell/scripting/scriptengine.cpp
index 0fb2d38..777a56d 100644
--- a/libs/plasmagenericshell/scripting/scriptengine.cpp
+++ b/libs/plasmagenericshell/scripting/scriptengine.cpp
@@ -677,6 +677,8 @@ QStringList ScriptEngine::defaultLayoutScripts()
 {
     const QString appName = KGlobal::activeComponent().aboutData()->appName();
     QStringList scripts = KGlobal::dirs()->findAllResources("data", appName + "/init/*.js");
+    scripts.sort();
+
     QStringList scriptPaths;
 
     if (scripts.isEmpty()) {
-- 
1.7.10

