--- a/kcontrol/fonts/fonts.cpp
+++ b/kcontrol/fonts/fonts.cpp
@@ -523,13 +523,13 @@ KFonts::KFonts(QWidget *parent, const QV
   QFont f1("Monospace", 16;
   QFont f2("Sans Serif", 16); // toolbar
 #else
-  QFont f0("Sans Serif", 9); // general/menu/desktop
-  QFont f1("Monospace", 9);
-  QFont f2("Sans Serif", 8); // toolbar
+  QFont f0("DejaVu Sans", 9); // general/menu/desktop
+  QFont f1("DejaVu Sans Mono", 9);
+  QFont f2("DejaVu Sans", 8); // toolbar
 #endif
-  QFont f3("Sans Serif", 8); // window title
-  QFont f4("Sans Serif", 9); // taskbar
-  QFont f5("Sans Serif", 8); // smallestReadableFont
+  QFont f3("DejaVu Sans", 9, QFont::Bold); // window title
+  QFont f4("DejaVu Sans", 9); // taskbar
+  QFont f5("DejaVu Sans", 8); // smallestReadableFont
 
   defaultFontList << f0 << f1 << f5 << f2 << f0 << f3 << f4 << f0;
 
