--- metacity-2.28.0/src/core/window-props.c	2009-09-08 22:55:35.000000000 +0200
+++ metacity-2.28.0-tomaz/src/core/window-props.c	2010-02-16 11:38:50.000000000 +0100
@@ -337,8 +337,15 @@
       
   glibtop_get_proc_uid (&process_details, process);
 
-  *result = process_details.uid;
-  return TRUE;
+  if (process_details.flags & (1L << GLIBTOP_PROC_UID_UID) ) 
+    {
+      *result = process_details.uid;
+      return TRUE;
+    }
+  else
+    {
+      return FALSE;
+    }
 #else
   /* I don't know, maybe we could do something hairy like see whether
    * /proc/$PID exists and who owns it, in case they have procfs.
