Subject: Avoid regenerating dirac-stdint.h.
Author: Alessio Treglia <alessio@debian.org>
Forwarded: not-needed
---
 configure.ac |    1 -
 1 file changed, 1 deletion(-)

--- dirac.orig/configure.ac
+++ dirac/configure.ac
@@ -100,7 +100,6 @@ AC_LANG_POP(C++)
 dnl ** We probably need this line, but not now
 dnl AC_CHECK_HEADERS(stdint.h unistd.h fcntl.h)
 
-AX_CREATE_STDINT_H([libdirac_common/dirac-stdint.h])
 
 dnl -----------------------------------------------
 dnl Checks for libraries.
