Index: madlib/Mesh/MshTags.h
===================================================================
--- madlib.orig/Mesh/MshTags.h	2009-12-20 20:18:38.000000000 +0100
+++ madlib/Mesh/MshTags.h	2009-12-20 20:18:46.000000000 +0100
@@ -1,9 +1,9 @@
-// -*- C++ -*- 
+// -*- C++ -*-
 // -------------------------------------------------------------------
 // MAdLib - Copyright (C) 2008-2009 Universite catholique de Louvain
 //
-// See the Copyright.txt and License.txt files for license information. 
-// You should have received a copy of these files along with MAdLib. 
+// See the Copyright.txt and License.txt files for license information.
+// You should have received a copy of these files along with MAdLib.
 // If not, see <http://www.madlib.be/license/>
 //
 // Please report all bugs and problems to <contrib@madlib.be>
@@ -14,6 +14,7 @@
 #ifndef MSHTAGS__H
 #define MSHTAGS__H
 
+#include <cstdio>
 #include <string>
 #include <sstream>
 
@@ -91,7 +92,7 @@
     case MSH_TET_35 : return 35;
     case MSH_TET_34 : return 34;
     case MSH_TET_56 : return 56;
-    default: 
+    default:
       printf("Error (MshTags.h): Unknown type of element %d\n", type);
     }
     return 0;
@@ -102,38 +103,38 @@
   {
     switch (type) {
     case MSH_PNT    : return 0;
-    case MSH_LIN_2  : 
-    case MSH_LIN_3  : 
-    case MSH_LIN_4  : 
-    case MSH_LIN_5  : 
+    case MSH_LIN_2  :
+    case MSH_LIN_3  :
+    case MSH_LIN_4  :
+    case MSH_LIN_5  :
     case MSH_LIN_6  : return 1;
-    case MSH_TRI_3  : 
-    case MSH_TRI_6  : 
-    case MSH_TRI_9  : 
-    case MSH_TRI_10  : 
-    case MSH_TRI_12  : 
-    case MSH_TRI_15  : 
-    case MSH_TRI_15I : 
-    case MSH_TRI_21 : 
-    case MSH_QUA_4  : 
-    case MSH_QUA_8  : 
+    case MSH_TRI_3  :
+    case MSH_TRI_6  :
+    case MSH_TRI_9  :
+    case MSH_TRI_10  :
+    case MSH_TRI_12  :
+    case MSH_TRI_15  :
+    case MSH_TRI_15I :
+    case MSH_TRI_21 :
+    case MSH_QUA_4  :
+    case MSH_QUA_8  :
     case MSH_QUA_9  : return 2;
-    case MSH_TET_4  : 
-    case MSH_TET_10 : 
-    case MSH_HEX_8  : 
-    case MSH_HEX_20 : 
-    case MSH_HEX_27 : 
-    case MSH_PRI_6  : 
-    case MSH_PRI_15 : 
-    case MSH_PRI_18 : 
-    case MSH_PYR_5  : 
-    case MSH_PYR_13 : 
-    case MSH_PYR_14 : 
-    case MSH_TET_20 : 
-    case MSH_TET_35 : 
-    case MSH_TET_34 : 
+    case MSH_TET_4  :
+    case MSH_TET_10 :
+    case MSH_HEX_8  :
+    case MSH_HEX_20 :
+    case MSH_HEX_27 :
+    case MSH_PRI_6  :
+    case MSH_PRI_15 :
+    case MSH_PRI_18 :
+    case MSH_PYR_5  :
+    case MSH_PYR_13 :
+    case MSH_PYR_14 :
+    case MSH_TET_20 :
+    case MSH_TET_35 :
+    case MSH_TET_34 :
     case MSH_TET_56 : return 3;
-    default: 
+    default:
       printf("Error (MshTags.h): Unknown type of element %d\n", type);
     }
     return -1;
@@ -143,7 +144,7 @@
   std::string getElementName(int type) {
 
     std::string name;
-  
+
     switch (type) {
     case MSH_PNT     : name = "Point"; break;
     case MSH_LIN_2   : name = "Linear edge"; break;
@@ -180,7 +181,7 @@
     default:
       std::stringstream ss; break;
       ss <<  "Unknown type of element (tag " << type << ")"; break;
-      name = ss.str(); 
+      name = ss.str();
     }
     return name;
   }
