Description: Fix ftbfs with -Werror=format-security
Bug-Debian: http://bugs.debian.org/646261
Forwarded: https://bugs.launchpad.net/gimmix/+bug/880289
Author: Vincent Legout <vincent@legout.info>
Last-Update: 2011-10-23

Index: gimmix/src/gimmix-config.c
===================================================================
--- gimmix.orig/src/gimmix-config.c	2011-10-23 12:17:55.204200757 +0200
+++ gimmix/src/gimmix-config.c	2011-10-23 12:18:07.696200791 +0200
@@ -93,7 +93,6 @@
 	int cmpval = strncasecmp (cfg_get_key_value(conf,(char*)key), "true", 4);
 	if (cmpval == 0)
 	{
-		g_print ("");
 		ret = true;
 	}
 	
@@ -146,7 +145,7 @@
 		}
 		else
 		{
-			ret = g_strdup_printf (host);
+			ret = g_strdup_printf ("%s", host);
 		}
 	}
 
Index: gimmix/src/gimmix-playlist.c
===================================================================
--- gimmix.orig/src/gimmix-playlist.c	2011-10-23 12:17:55.212200757 +0200
+++ gimmix/src/gimmix-playlist.c	2011-10-23 12:18:07.696200791 +0200
@@ -532,7 +532,7 @@
 			}
 			else
 			{
-				title = g_markup_printf_escaped (g_path_get_basename(data->song->file));
+				title = g_markup_printf_escaped ("%s", g_path_get_basename(data->song->file));
 				gimmix_strip_file_ext (title);
 			}
 			if (data->song->artist)
Index: gimmix/src/gimmix-lyrics.c
===================================================================
--- gimmix.orig/src/gimmix-lyrics.c	2011-10-23 12:17:55.232200757 +0200
+++ gimmix/src/gimmix-lyrics.c	2011-10-23 12:18:07.696200791 +0200
@@ -168,7 +168,7 @@
 	nxml_get_string (nndata, &str);
 	if (str)
 	{
-		node->lyrics = g_strdup_printf (str);
+		node->lyrics = g_strdup_printf ("%s", str);
 		g_free (str);
 		ret = TRUE;
 	}
Index: gimmix/src/gimmix-covers.c
===================================================================
--- gimmix.orig/src/gimmix-covers.c	2011-10-23 12:17:55.220200757 +0200
+++ gimmix/src/gimmix-covers.c	2011-10-23 12:18:07.696200791 +0200
@@ -635,7 +635,7 @@
 	path = g_strdup_printf ("%s/%s-%s.albuminfo", temp, artist_e, album_e);
 	if ((fp=fopen(path, "w")))
 	{
-		fprintf (fp, info);
+		fprintf (fp, "%s", info);
 		fclose (fp);
 	}
 	g_free (temp);
Index: gimmix/src/gimmix.c
===================================================================
--- gimmix.orig/src/gimmix.c	2011-10-23 12:17:55.192200757 +0200
+++ gimmix/src/gimmix.c	2011-10-23 12:18:07.696200791 +0200
@@ -78,7 +78,7 @@
 	gchar	*error = NULL;
 
 	error = g_strdup_printf ("Error:(%d):%s\n", error_id, msg);
-	printf (error);
+	printf ("%s", error);
 	g_free (error);
 	if (last_error!=NULL)
 	{
@@ -266,7 +266,7 @@
 							g_ascii_strdown(APPNAME,strlen(APPNAME)),
 							VERSION,
 							"Copyright 2006, 2007, 2008, 2009 Priyank Gosalia");
-				fprintf (stdout, vstr);
+				fprintf (stdout, "%s", vstr);
 				g_free (vstr);
 				goto cleanup;
 				break;
