diff -ur tar_1.26/gnu/stdio.h tar/gnu/stdio.h
--- tar_1.26/gnu/stdio.h	2012-12-31 00:39:19.217510482 -0500
+++ tar/gnu/stdio.h	2012-12-31 00:41:02.537508618 -0500
@@ -472,12 +472,6 @@
                  "use gnulib module fflush for portable POSIX compliance");
 #endif
 
-/* It is very rare that the developer ever has full control of stdin,
-   so any use of gets warrants an unconditional warning.  Assume it is
-   always declared, since it is required by C89.  */
-#undef gets
-_GL_WARN_ON_USE (gets, "gets is a security hole - use fgets instead");
-
 #if 0
 # if 0
 #  if !(defined __cplusplus && defined GNULIB_NAMESPACE)
