diff -ur tar_1.26/gnu/stdio.in.h tar/gnu/stdio.in.h
--- tar_1.26/gnu/stdio.in.h	2011-03-12 04:14:33.000000000 -0500
+++ tar/gnu/stdio.in.h	2012-12-31 00:42:28.837507061 -0500
@@ -160,12 +160,6 @@
                  "use gnulib module fflush for portable POSIX compliance");
 #endif
 
-/* It is very rare that the developer ever has full control of stdin,
-   so any use of gets warrants an unconditional warning.  Assume it is
-   always declared, since it is required by C89.  */
-#undef gets
-_GL_WARN_ON_USE (gets, "gets is a security hole - use fgets instead");
-
 #if @GNULIB_FOPEN@
 # if @REPLACE_FOPEN@
 #  if !(defined __cplusplus && defined GNULIB_NAMESPACE)
