Index: lapack-3.4.2/install/make.inc.gfortran
===================================================================
--- lapack-3.4.2.orig/install/make.inc.gfortran	2012-04-13 20:13:07.000000000 +0200
+++ lapack-3.4.2/install/make.inc.gfortran	2012-09-26 12:58:45.025523212 +0200
@@ -42,7 +42,7 @@
 #
 # CC is the C compiler, normally invoked with options CFLAGS.
 #
-CC = gcc
+CC = /usr/bin/cc
 CFLAGS = -O3
 #
 # LAPACKE has also the interface to some routines from tmglib,
