Index: lapack-3.4.0/lapacke/make.inc
===================================================================
--- lapack-3.4.0.orig/lapacke/make.inc	2011-11-09 23:56:15.000000000 +0100
+++ lapack-3.4.0/lapacke/make.inc	2011-11-16 18:00:42.958453190 +0100
@@ -48,14 +48,14 @@
 # - gfortran, set: LINKER = gfortran
 #
 CC = gcc
-CFLAGS =
+CFLAGS = -fPIC -O2 -Wall
 LINKER = gfortran
 LDFLAGS =
 #
 # The name of the libraries to be created/linked to
 # Ensure that the libraries have the same data model (LP64/ILP64).
 #
-LAPACKE = lapacke.a
+LAPACKE = liblapacke.a
 LIBS =  /Users/julie/Documents/Boulot/lapack-dev/lapack/trunk/lapack_LINUX.a \
         /Users/julie/Documents/Boulot/lapack-dev/lapack/trunk/blas_LINUX.a  -lm
 #  The archiver and the flag(s) to use when building archive (library)
