Index: lapack-3.3.0/man/manl/zggev.l
===================================================================
--- lapack-3.3.0.orig/man/manl/zggev.l	2010-12-20 00:04:19.000000000 +0100
+++ lapack-3.3.0/man/manl/zggev.l	2010-12-20 00:04:38.000000000 +0100
@@ -77,6 +77,7 @@
 The leading dimension of B.  LDB >= max(1,N).
 .TP 8
 ALPHA   (output) COMPLEX*16 array, dimension (N)
+.TQ 8
 BETA    (output) COMPLEX*16 array, dimension (N)
 On exit, ALPHA(j)/BETA(j), j=1,...,N, will be the
 generalized eigenvalues.
