Description: Fix spelling errors found by lintian
Author: Alessandro Ghedini <al3xbio@gmail.com>
Origin: vendor
Forwarded: no
Last-Update: 2012-02-17

--- a/dbiproxy.PL
+++ b/dbiproxy.PL
@@ -108,7 +108,7 @@
 =item B<--localaddr=ip>
 
 By default a daemon is listening to any IP number that a machine
-has. This attribute allows to restrict the server to the given
+has. This attribute allows one to restrict the server to the given
 IP number.
 
 =item B<--localport=port>
--- a/lib/DBI/ProxyServer.pm
+++ b/lib/DBI/ProxyServer.pm
@@ -474,7 +474,7 @@
 =item I<localaddr> (B<--localaddr=ip>)
 
 By default a daemon is listening to any IP number that a machine
-has. This attribute allows to restrict the server to the given
+has. This attribute allows one to restrict the server to the given
 IP number.
 
 =item I<localport> (B<--localport=port>)
