Subject: A NTP clock adjustment of -0.00967s lead to FTBFS.
Reviewed-by: Nicholas Bamber <nicholas@periapt.co.uk>
Last-Update: 2010-10-07
Forwarded: not-needed
Author: gregor hermann <gregoa@debian.org>
--- a/t/40profile.t
+++ b/t/40profile.t
@@ -177,7 +177,7 @@
 this kind of clock behaviour. Please retry.
 EOT
     # don't treat small negative values as failure
-    $shortest = 0 if $shortest > -0.008;
+    $shortest = 0 if $shortest > -0.8;
 }
 
 
