Subject: syslogd isn't necessarily on the buildds
Reviewed-by: Nicholas Bamber <nicholas@periapt.co.uk>
Last-Update: 2010-10-07
Author: Nicholas Bamber <nicholas@periapt.co.uk>
--- a/t/80proxy.t
+++ b/t/80proxy.t
@@ -10,6 +10,10 @@
 require VMS::Filespec if $^O eq 'VMS';
 require Cwd;
 
+# DEBIAN
+print("1..1\nok 1 # skipped No syslogd in Debian build environments.\n");
+exit(0);
+
 my $haveFileSpec = eval { require File::Spec };
 my $failed_tests = 0;
 
