From 4b1819b285ed3e6b8ddae2acde20e349405a8c6e Mon Sep 17 00:00:00 2001
From: Robert Ancell <robert.ancell@canonical.com>
Date: Fri, 04 May 2012 01:45:37 +0000
Subject: swell-foop: Install into /usr/share/swell-foop not /usr/share/gnome-games

---
Index: gnome-games-3.4.2/swell-foop/data/Makefile.am
===================================================================
--- gnome-games-3.4.2.orig/swell-foop/data/Makefile.am	2012-05-23 07:32:31.627967707 +0200
+++ gnome-games-3.4.2/swell-foop/data/Makefile.am	2012-05-23 07:32:34.119967618 +0200
@@ -5,7 +5,7 @@
 @INTLTOOL_XML_NOMERGE_RULE@
 @GSETTINGS_RULES@
 
-swelldir=$(pkgdatadir)/swell-foop
+swelldir=$(datadir)/swell-foop
 swell_DATA = \
     preferences.ui
 
Index: gnome-games-3.4.2/swell-foop/data/themes/colors/Makefile.am
===================================================================
--- gnome-games-3.4.2.orig/swell-foop/data/themes/colors/Makefile.am	2012-05-23 07:32:31.627967707 +0200
+++ gnome-games-3.4.2/swell-foop/data/themes/colors/Makefile.am	2012-05-23 07:32:34.119967618 +0200
@@ -1,4 +1,4 @@
-themedir = $(pkgdatadir)/swell-foop/themes/colors
+themedir = $(datadir)/swell-foop/themes/colors
 theme_DATA = \
 	bkg.svg \
 	blue.svg \
Index: gnome-games-3.4.2/swell-foop/data/themes/shapesandcolors/Makefile.am
===================================================================
--- gnome-games-3.4.2.orig/swell-foop/data/themes/shapesandcolors/Makefile.am	2012-05-23 07:32:31.627967707 +0200
+++ gnome-games-3.4.2/swell-foop/data/themes/shapesandcolors/Makefile.am	2012-05-23 07:32:34.119967618 +0200
@@ -1,4 +1,4 @@
-themedir = $(pkgdatadir)/swell-foop/themes/shapesandcolors
+themedir = $(datadir)/swell-foop/themes/shapesandcolors
 theme_DATA = \
 	bkg.svg \
 	blue.svg \
Index: gnome-games-3.4.2/swell-foop/src/Makefile.am
===================================================================
--- gnome-games-3.4.2.orig/swell-foop/src/Makefile.am	2012-05-23 07:32:31.627967707 +0200
+++ gnome-games-3.4.2/swell-foop/src/Makefile.am	2012-05-23 07:32:34.119967618 +0200
@@ -17,8 +17,8 @@
 	-I$(top_srcdir)/libgames-support \
 	-DVERSION=\"$(VERSION)\" \
 	-DGETTEXT_PACKAGE=\"$(GETTEXT_PACKAGE)\" \
-	-DPKGDATADIR=\""$(pkgdatadir)/swell-foop"\" \
-	-DLOCALEDIR=\"$(pkgdatadir)/locale\" \
+	-DDATADIR=\""$(datadir)/swell-foop"\" \
+	-DLOCALEDIR=\"$(datadir)/locale\" \
 	$(GHTREAD_CFLAGS) \
 	$(GMODULE_CFLAGS) \
 	$(GTK_CFLAGS) \
Index: gnome-games-3.4.2/swell-foop/src/config.vapi
===================================================================
--- gnome-games-3.4.2.orig/swell-foop/src/config.vapi	2012-05-23 07:32:31.627967707 +0200
+++ gnome-games-3.4.2/swell-foop/src/config.vapi	2012-05-23 07:32:34.119967618 +0200
@@ -1,4 +1,4 @@
-public const string PKGDATADIR;
+public const string DATADIR;
 public const string LOCALEDIR;
 public const string GETTEXT_PACKAGE;
 public const string VERSION;
Index: gnome-games-3.4.2/swell-foop/src/game-view.vala
===================================================================
--- gnome-games-3.4.2.orig/swell-foop/src/game-view.vala	2012-05-23 07:32:31.627967707 +0200
+++ gnome-games-3.4.2/swell-foop/src/game-view.vala	2012-05-23 07:32:34.119967618 +0200
@@ -245,7 +245,7 @@
         try
         {
             for (int i = 0; i < 4; i++)
-                textures[i] = new Clutter.Texture.from_file (Path.build_filename (PKGDATADIR, "themes", name, colors[i] + ".svg"));
+                textures[i] = new Clutter.Texture.from_file (Path.build_filename (DATADIR, "themes", name, colors[i] + ".svg"));
         }
         catch (Clutter.TextureError e)
         {
Index: gnome-games-3.4.2/swell-foop/src/swell-foop.vala
===================================================================
--- gnome-games-3.4.2.orig/swell-foop/src/swell-foop.vala	2012-05-23 07:32:31.627967707 +0200
+++ gnome-games-3.4.2/swell-foop/src/swell-foop.vala	2012-05-23 07:32:34.119967618 +0200
@@ -229,7 +229,7 @@
         var preferences_builder = new Gtk.Builder ();
         try
         {
-            preferences_builder.add_from_file (Path.build_filename (PKGDATADIR, "preferences.ui", null));
+            preferences_builder.add_from_file (Path.build_filename (DATADIR, "preferences.ui", null));
         }
         catch (Error e)
         {
