From d59d8a2502ff78a2ad4040d96f3c4dbe7c995cb0 Mon Sep 17 00:00:00 2001
From: Dominique Leuenberger <dimstar@opensuse.org>
Date: Fri, 18 May 2012 20:15:39 +0000
Subject: glchess: fix phalanx/gnuchess output interpretation. Bug 675434

---
diff --git a/glchess/src/chess-engine-cecp.vala b/glchess/src/chess-engine-cecp.vala
index cb40419..4a7aff5 100644
--- a/glchess/src/chess-engine-cecp.vala
+++ b/glchess/src/chess-engine-cecp.vala
@@ -40,7 +40,7 @@ public class ChessEngineCECP : ChessEngine
 
             debug ("Read from engine: '%s'", line);
 
-            string[] move_prefixes = { "My move is: ", "my move is ", "move " };
+            string[] move_prefixes = { "My move is: ", "My move is : ", "my move is ", "move " };
             foreach (string prefix in move_prefixes)
             {
                 if (line.has_prefix (prefix))
@@ -48,7 +48,7 @@ public class ChessEngineCECP : ChessEngine
                     string move = line[prefix.length:line.length];
                     debug ("Engine moves %s", move);
                     moving = true;
-                    moved (move);
+                    moved (move.strip());
                 }
             }
 
diff --git a/glchess/src/chess-game.vala b/glchess/src/chess-game.vala
index 3c5ebce..ecd8b83 100644
--- a/glchess/src/chess-game.vala
+++ b/glchess/src/chess-game.vala
@@ -1027,7 +1027,7 @@ public class ChessState
                 r1 = (int) (move[i] - '1');
                 i++;
             }
-            if (move[i] == 'x')
+            if (move[i] == 'x' || move[i] == '-')
                 i++;
             if (move[i] >= 'a' && move[i] <= 'h')
             {

