From 1eb18e95bad9135b516fbf6f77f516fa1ab8b402 Mon Sep 17 00:00:00 2001
From: Robert Ancell <robert.ancell@canonical.com>
Date: Sat, 14 Jul 2012 05:13:12 +0000
Subject: glchess: Kill engine processes on new game/exit

Index: gnome-games-3.4.2/glchess/src/chess-engine.vala
===================================================================
--- gnome-games-3.4.2.orig/glchess/src/chess-engine.vala	2012-05-15 06:37:48.000000000 +0200
+++ gnome-games-3.4.2/glchess/src/chess-engine.vala	2012-08-05 23:13:21.586577206 +0200
@@ -35,9 +35,9 @@
         try
         {
             Process.spawn_async_with_pipes (null, argv, null,
-                                                 SpawnFlags.SEARCH_PATH,
-                                                 null,
-                                                 out pid, out stdin_fd, out stdout_fd, out stderr_fd);
+                                            SpawnFlags.SEARCH_PATH | SpawnFlags.DO_NOT_REAP_CHILD,
+                                            null,
+                                            out pid, out stdin_fd, out stdout_fd, out stderr_fd);
         }
         catch (SpawnError e)
         {
@@ -45,6 +45,8 @@
             return false;
         }
 
+        ChildWatch.add (pid, engine_stopped_cb);
+
         stdout_channel = new IOChannel.unix_new (stdout_fd);
         try
         {
@@ -60,7 +62,12 @@
 
         return true;
     }
-    
+
+    private void engine_stopped_cb (Pid pid, int status)
+    {
+        stopped ();
+    }
+
     public virtual void start_game ()
     {
     }
@@ -79,8 +86,8 @@
 
     public void stop ()
     {
-        // FIXME
-        stopped ();
+        if (pid != 0)
+            Posix.kill (pid, Posix.SIGTERM);        
     }
 
     private bool read_cb (IOChannel source, IOCondition condition)
Index: gnome-games-3.4.2/glchess/src/glchess.vala
===================================================================
--- gnome-games-3.4.2.orig/glchess/src/glchess.vala	2012-05-15 06:37:48.000000000 +0200
+++ gnome-games-3.4.2/glchess/src/glchess.vala	2012-08-05 23:13:21.586577206 +0200
@@ -124,6 +124,13 @@
         settings_changed_cb (settings, "show-3d");
     }
 
+    protected override void shutdown ()
+    {
+        base.shutdown ();
+        if (opponent_engine != null)
+            opponent_engine.stop ();
+    }
+
     public void quit_game ()
     {
         if (save_duration_timeout != 0)
@@ -298,7 +305,14 @@
             black_level = "normal";
 
         opponent = null;
-        opponent_engine = null;
+        if (opponent_engine != null)
+        {
+            opponent_engine.stop ();
+            opponent_engine.ready_changed.disconnect (engine_ready_cb);
+            opponent_engine.moved.disconnect (engine_move_cb);
+            opponent_engine.stopped.disconnect (engine_stopped_cb);
+            opponent_engine = null;
+        }
         if (white_engine != null)
         {
             opponent = game.white;
@@ -316,6 +330,7 @@
         {
             opponent_engine.ready_changed.connect (engine_ready_cb);
             opponent_engine.moved.connect (engine_move_cb);
+            opponent_engine.stopped.connect (engine_stopped_cb);
             opponent_engine.start ();
         }
 
@@ -436,6 +451,11 @@
         opponent.move (move);
     }
 
+    private void engine_stopped_cb (ChessEngine engine)
+    {
+        opponent.resign ();
+    }
+
     private void game_start_cb (ChessGame game)
     {
         if (opponent_engine != null)
