Index: scilab-5.4.0-alpha-20122105-824511b/modules/demo_tools/macros/add_demo.sci
===================================================================
--- scilab-5.4.0-alpha-20122105-824511b.orig/modules/demo_tools/macros/add_demo.sci	2012-05-21 14:59:42.000000000 +0200
+++ scilab-5.4.0-alpha-20122105-824511b/modules/demo_tools/macros/add_demo.sci	2012-05-21 14:59:50.000000000 +0200
@@ -35,7 +35,7 @@
   path = pathconvert(path, %f, %t);
 
   if ~isfile(path) then
-    error(msprintf(gettext("%s: The file %s does not exist.\n"), "add_demo", path));
+    error(msprintf(gettext("Demos are not available. Please install the package scilab-doc\n")));
   end
 
   if isdir(path) then
