Index: scilab-5.4.0-alpha-20122105-824511b/bin/scilab
===================================================================
--- scilab-5.4.0-alpha-20122105-824511b.orig/bin/scilab	2012-05-21 14:59:42.000000000 +0200
+++ scilab-5.4.0-alpha-20122105-824511b/bin/scilab	2012-05-21 14:59:51.000000000 +0200
@@ -739,7 +739,7 @@
 
 # . is used by the dynamic link. The build lib are copied into the PWD
 SCILIB=$SCI/bin:$SCI/lib/scilab/:$SCI/lib64/scilab/:.
-LD_LIBRARY_PATH=$SCILIB${LD_LIBRARY_PATH+:$LD_LIBRARY_PATH}
+#LD_LIBRARY_PATH=$SCILIB${LD_LIBRARY_PATH+:$LD_LIBRARY_PATH}
 
 case $OS in
     *darwin* | *Darwin*) # MacOS X
@@ -783,6 +783,7 @@
     *)
 esac
 
+LD_LIBRARY_PATH=/usr/lib/scilab/:/usr/lib64/scilab/:.${LD_LIBRARY_PATH:+:$LD_LIBRARY_PATH}
 
 if test $SCIVERBOSE -ne 0; then
     echo "LD_LIBRARY_PATH : $LD_LIBRARY_PATH"
@@ -1012,6 +1013,7 @@
         esac
     done
 
+LD_LIBRARY_PATH=/usr/lib/scilab/:/usr/lib64/scilab/:.${LD_LIBRARY_PATH:+:$LD_LIBRARY_PATH}
 
 # If the argument -a is not provided (extraarg) and we found unknown args
 # fail and provide help
