Index: scilab-5.3.0-beta-3/modules/scicos/macros/scicos_scicos/genmoc.sci
===================================================================
--- scilab-5.3.0-beta-3.orig/modules/scicos/macros/scicos_scicos/genmoc.sci	2010-09-12 15:24:04.000000000 +0200
+++ scilab-5.3.0-beta-3/modules/scicos/macros/scicos_scicos/genmoc.sci	2010-09-12 15:24:27.000000000 +0200
@@ -29,7 +29,7 @@
   //++ Check that modelica compiler is available
   //++ Otherwise, give some feedback and quit
   if ~with_modelica_compiler() then
-    error(sprintf(gettext("%s: Error: Modelica compiler (MODELICAC) is unavailable."), "genmoc"))
+    error(sprintf(gettext("%s: Error: Modelica compiler (MODELICAC) is unavailable.")+" This is probably due to the non support of ocamlopt on your architecture.", "genmoc"))
     return
   end
 
