Index: anjuta-3.0.0.0/src/about.c
===================================================================
--- anjuta-3.0.0.0.orig/src/about.c	2011-04-16 13:04:37.194307367 +0200
+++ anjuta-3.0.0.0/src/about.c	2011-04-16 13:05:00.834307378 +0200
@@ -28,7 +28,7 @@
 
 #include "about.h"
 
-#define LICENSE_FILE PACKAGE_DOC_DIR "/COPYING"
+#define LICENSE_FILE "/usr/share/common-licenses/GPL-2"
 
 #define ANJUTA_PIXMAP_LOGO			"anjuta_logo.png"
 #define ABOUT_AUTHORS				"AUTHORS"	
