Index: anjuta-3.4.1/configure.ac
===================================================================
--- anjuta-3.4.1.orig/configure.ac	2012-04-29 15:37:22.000000000 +0200
+++ anjuta-3.4.1/configure.ac	2012-04-30 16:22:45.174448737 +0200
@@ -298,7 +298,7 @@
 
 if test x$have_python = xyes; then
 	if test -x $PYTHON-config; then
-		PYTHON_CFLAGS=`$PYTHON-config --cflags`
+		PYTHON_CFLAGS=`unset CFLAGS; $PYTHON-config --cflags`
 		PYTHON_LIBS=`$PYTHON-config --libs`
 		AC_SUBST(PYTHON_CFLAGS)
 		AC_SUBST(PYTHON_LIBS)
Index: anjuta-3.4.1/configure
===================================================================
--- anjuta-3.4.1.orig/configure	2012-04-29 15:58:51.000000000 +0200
+++ anjuta-3.4.1/configure	2012-04-30 16:22:45.186448734 +0200
@@ -17947,7 +17947,7 @@
 
 if test x$have_python = xyes; then
 	if test -x $PYTHON-config; then
-		PYTHON_CFLAGS=`$PYTHON-config --cflags`
+		PYTHON_CFLAGS=`unset CFLAGS; $PYTHON-config --cflags`
 		PYTHON_LIBS=`$PYTHON-config --libs`
 
 
