Description: Transition to libnotify 0.7
Author: Laurent Bigonville <bigon@debian.org>
Date: Tue, 02 Aug 2011 21:31:13 +0200
Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=630262

--- a/applets/maintained/notification-daemon/daemon.c
+++ b/applets/maintained/notification-daemon/daemon.c
@@ -1487,7 +1487,7 @@
   if (fork() == 0)
   {
     notify_init("notify-send");
-    notify = notify_notification_new(summary, body, icon_str, NULL);
+    notify = notify_notification_new(summary, body, icon_str);
     notify_notification_set_category(notify, type);
     notify_notification_set_urgency(notify, urgency);
     notify_notification_set_timeout(notify, expire_timeout);
