Index: awn-extras-applets-0.4.0/applets/maintained/awnterm/keybinder.h
===================================================================
--- awn-extras-applets-0.4.0.orig/applets/maintained/awnterm/keybinder.h	2009-12-04 15:02:02.000000000 +0100
+++ awn-extras-applets-0.4.0/applets/maintained/awnterm/keybinder.h	2012-03-30 07:25:28.121048485 +0200
@@ -24,7 +24,7 @@
 #ifndef __AWN_KEY_BINDER_H__
 #define __AWN_KEY_BINDER_H__
 
-#include <glib/gtypes.h>
+#include <glib.h>
 
 G_BEGIN_DECLS
 
Index: awn-extras-applets-0.4.0/applets/maintained/notification-daemon/daemon.h
===================================================================
--- awn-extras-applets-0.4.0.orig/applets/maintained/notification-daemon/daemon.h	2009-11-18 23:57:02.000000000 +0100
+++ awn-extras-applets-0.4.0/applets/maintained/notification-daemon/daemon.h	2012-03-30 07:25:27.613048502 +0200
@@ -31,8 +31,6 @@
 
 #include <libawn/awn-applet.h>
 #include <libawn/awn-applet-simple.h>
-#include <glib/gmacros.h>
-#include <glib/gerror.h>
 
 #include <libawn/awn-dialog.h>
 #include <libawn/awn-cairo-utils.h>
Index: awn-extras-applets-0.4.0/applets/maintained/notification-daemon/engines.c
===================================================================
--- awn-extras-applets-0.4.0.orig/applets/maintained/notification-daemon/engines.c	2010-04-10 21:14:44.000000000 +0200
+++ awn-extras-applets-0.4.0/applets/maintained/notification-daemon/engines.c	2012-03-30 07:25:27.857048494 +0200
@@ -38,8 +38,7 @@
 
 
 #include <libawn/awn-applet.h>
-#include <glib/gmacros.h>
-#include <glib/gerror.h>
+#include <glib.h>
 
 #include <libawn/awn-dialog.h>
 #include <libawn/awn-applet-simple.h>
@@ -50,7 +49,6 @@
 
 #include <string.h>
 #include <gtk/gtk.h>
-#include <glib.h>
 #ifndef HAVE_GTK_URL_LABEL
 #include <libsexy/sexy-url-label.h>
 #endif
Index: awn-extras-applets-0.4.0/applets/unmaintained/awnsystemmonitor/awnsystemmonitor.h
===================================================================
--- awn-extras-applets-0.4.0.orig/applets/unmaintained/awnsystemmonitor/awnsystemmonitor.h	2009-05-29 16:06:00.000000000 +0200
+++ awn-extras-applets-0.4.0/applets/unmaintained/awnsystemmonitor/awnsystemmonitor.h	2012-03-30 07:25:28.653048466 +0200
@@ -25,7 +25,7 @@
 //#include <libawn/awn-title.h>
 #include <libawn/awn-tooltip.h>
 
-#include <glib/gtypes.h>
+#include <glib.h>
 #include <glibtop/cpu.h>
 #include <gconf/gconf-client.h>
 
Index: awn-extras-applets-0.4.0/applets/unmaintained/awnsystemmonitor/awntop_cairo_component.c
===================================================================
--- awn-extras-applets-0.4.0.orig/applets/unmaintained/awnsystemmonitor/awntop_cairo_component.c	2010-04-01 17:21:40.000000000 +0200
+++ awn-extras-applets-0.4.0/applets/unmaintained/awnsystemmonitor/awntop_cairo_component.c	2012-03-30 07:25:28.565048469 +0200
@@ -43,13 +43,11 @@
 
 
 #include <libawn/awn-applet.h>
-#include <glib/gmacros.h>
-#include <glib/gerror.h>
+#include <glib.h>
 #include <gconf/gconf-value.h>
 
 #include <libawn/awn-dialog.h>
 #include <libawn/awn-applet-simple.h>
-#include <glib.h>
 #include <gtk/gtk.h>
 
 
Index: awn-extras-applets-0.4.0/applets/unmaintained/awnsystemmonitor/cpu_component.c
===================================================================
--- awn-extras-applets-0.4.0.orig/applets/unmaintained/awnsystemmonitor/cpu_component.c	2009-07-07 22:18:56.000000000 +0200
+++ awn-extras-applets-0.4.0/applets/unmaintained/awnsystemmonitor/cpu_component.c	2012-03-30 07:25:29.373048440 +0200
@@ -18,13 +18,11 @@
  */
 
 #include <libawn/awn-applet.h>
-#include <glib/gmacros.h>
-#include <glib/gerror.h>
+#include <glib.h>
 #include <gconf/gconf-value.h>
 
 #include <libawn/awn-dialog.h>
 #include <libawn/awn-applet-simple.h>
-#include <glib.h>
 #include <gtk/gtk.h>
 #include <gdk/gdk.h>
 #include <string.h>
Index: awn-extras-applets-0.4.0/applets/unmaintained/awnsystemmonitor/cpumetergconf.c
===================================================================
--- awn-extras-applets-0.4.0.orig/applets/unmaintained/awnsystemmonitor/cpumetergconf.c	2009-05-29 16:06:00.000000000 +0200
+++ awn-extras-applets-0.4.0/applets/unmaintained/awnsystemmonitor/cpumetergconf.c	2012-03-30 07:25:28.961048455 +0200
@@ -20,13 +20,10 @@
  * Boston, MA 02111-1307, USA.
  */
 #include <string.h>
-#include <glib/gmacros.h>
-#include <glib/gerror.h>
+#include <glib.h>
 #include <gconf/gconf-value.h>
 
 #include <libawn/awn-applet.h>
-#include <glib/gmacros.h>
-#include <glib/gerror.h>
 #include <gconf/gconf-value.h>
 
 #include "cairo-utils.h"
Index: awn-extras-applets-0.4.0/applets/unmaintained/awnsystemmonitor/dashboard.h
===================================================================
--- awn-extras-applets-0.4.0.orig/applets/unmaintained/awnsystemmonitor/dashboard.h	2009-05-29 16:06:00.000000000 +0200
+++ awn-extras-applets-0.4.0/applets/unmaintained/awnsystemmonitor/dashboard.h	2012-03-30 07:25:29.053048452 +0200
@@ -23,13 +23,11 @@
 
 
 #include <libawn/awn-applet.h>
-#include <glib/gmacros.h>
-#include <glib/gerror.h>
+#include <glib.h>
 #include <gconf/gconf-value.h>
 
 #include <libawn/awn-dialog.h>
 #include <libawn/awn-applet-simple.h>
-#include <glib.h>
 #include <gtk/gtk.h>
 
 #include <glibtop/mem.h>
Index: awn-extras-applets-0.4.0/applets/unmaintained/awnsystemmonitor/date_time_component.c
===================================================================
--- awn-extras-applets-0.4.0.orig/applets/unmaintained/awnsystemmonitor/date_time_component.c	2009-07-07 22:18:56.000000000 +0200
+++ awn-extras-applets-0.4.0/applets/unmaintained/awnsystemmonitor/date_time_component.c	2012-03-30 07:25:29.257048444 +0200
@@ -19,13 +19,11 @@
 
 
 #include <libawn/awn-applet.h>
-#include <glib/gmacros.h>
-#include <glib/gerror.h>
+#include <glib.h>
 #include <gconf/gconf-value.h>
 
 #include <libawn/awn-dialog.h>
 #include <libawn/awn-applet-simple.h>
-#include <glib.h>
 #include <gtk/gtk.h>
 #include <gdk/gdk.h>
 #include <string.h>
Index: awn-extras-applets-0.4.0/applets/unmaintained/awnsystemmonitor/loadavg_component.c
===================================================================
--- awn-extras-applets-0.4.0.orig/applets/unmaintained/awnsystemmonitor/loadavg_component.c	2009-07-07 22:18:56.000000000 +0200
+++ awn-extras-applets-0.4.0/applets/unmaintained/awnsystemmonitor/loadavg_component.c	2012-03-30 07:25:28.757048462 +0200
@@ -19,13 +19,11 @@
 
 #include <glibtop/loadavg.h>
 #include <libawn/awn-applet.h>
-#include <glib/gmacros.h>
-#include <glib/gerror.h>
+#include <glib.h>
 #include <gconf/gconf-value.h>
 
 #include <libawn/awn-dialog.h>
 #include <libawn/awn-applet-simple.h>
-#include <glib.h>
 #include <gtk/gtk.h>
 #include <gdk/gdk.h>
 #include <string.h>
Index: awn-extras-applets-0.4.0/applets/unmaintained/awnsystemmonitor/sysmem_component.c
===================================================================
--- awn-extras-applets-0.4.0.orig/applets/unmaintained/awnsystemmonitor/sysmem_component.c	2009-07-07 22:18:56.000000000 +0200
+++ awn-extras-applets-0.4.0/applets/unmaintained/awnsystemmonitor/sysmem_component.c	2012-03-30 07:25:28.861048458 +0200
@@ -19,13 +19,11 @@
 
 #include <glibtop/mem.h>
 #include <libawn/awn-applet.h>
-#include <glib/gmacros.h>
-#include <glib/gerror.h>
+#include <glib.h>
 #include <gconf/gconf-value.h>
 
 #include <libawn/awn-dialog.h>
 #include <libawn/awn-applet-simple.h>
-#include <glib.h>
 #include <gtk/gtk.h>
 #include <gdk/gdk.h>
 #include <string.h>
Index: awn-extras-applets-0.4.0/applets/unmaintained/awnsystemmonitor/uptime_component.c
===================================================================
--- awn-extras-applets-0.4.0.orig/applets/unmaintained/awnsystemmonitor/uptime_component.c	2009-07-07 22:18:56.000000000 +0200
+++ awn-extras-applets-0.4.0/applets/unmaintained/awnsystemmonitor/uptime_component.c	2012-03-30 07:25:29.153048448 +0200
@@ -1,12 +1,10 @@
 
 #include <libawn/awn-applet.h>
-#include <glib/gmacros.h>
-#include <glib/gerror.h>
+#include <glib.h>
 #include <gconf/gconf-value.h>
 
 #include <libawn/awn-dialog.h>
 #include <libawn/awn-applet-simple.h>
-#include <glib.h>
 
 #include <glibtop/uptime.h>
 #include <glibtop/cpu.h>
Index: awn-extras-applets-0.4.0/applets/unmaintained/main-menu/applet.c
===================================================================
--- awn-extras-applets-0.4.0.orig/applets/unmaintained/main-menu/applet.c	2009-10-27 20:45:03.000000000 +0100
+++ awn-extras-applets-0.4.0/applets/unmaintained/main-menu/applet.c	2012-03-30 07:25:28.421048474 +0200
@@ -23,8 +23,7 @@
 
 #include <string.h>
 
-#include <glib/gmacros.h>
-#include <glib/gerror.h>
+#include <glib.h>
 #include <glib/gi18n.h>
 #include <gtk/gtk.h>
 #include <libdesktop-agnostic/fdo.h>
