Description: Add support for out-of-source documentation building.
Author: Jakub Wilk <jwilk@debian.org>
Forwarded: not-needed
Last-Update: 2010-04-16

--- a/docsrc/makeFunctionIndex.py
+++ b/docsrc/makeFunctionIndex.py
@@ -136,7 +136,7 @@
 # crosslinking of vigranumpy documentation.
 # Note that '::' are not allowed in reStructuedText link names, 
 # so we have to use '.' instead.
-replaces=open("../vigranumpy/docsrc/c_api_replaces.txt","w")
+replaces=open("../../vigranumpy/docsrc/c_api_replaces.txt","w")
 for i in range(len(functionList)):
     functionName = functionList[i][1]
     overloadDisambiguation = functionList[i][2]
