Index: b/vigranumpy/docsrc/CMakeLists.txt
===================================================================
--- a/vigranumpy/docsrc/CMakeLists.txt
+++ b/vigranumpy/docsrc/CMakeLists.txt
@@ -19,7 +19,7 @@
         -c ${CMAKE_CURRENT_BINARY_DIR}
         ${SPHINX_OPTS})
     file(MAKE_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/_static)
-    configure_file(_static/vigra-icon.ico ${CMAKE_CURRENT_BINARY_DIR}/_static/vigra-icon.ico COPYONLY)
+    configure_file(../../docsrc/vigra-icon.ico ${CMAKE_CURRENT_BINARY_DIR}/_static/vigra-icon.ico COPYONLY)
     
     DEPENDENCY_PATH(VIGRAIMPEX_PATH vigraimpex)
     STRING(REGEX REPLACE "/vigra$" "" VIGRANUMPY_TMP_PATH ${vigranumpy_tmp_dir})   
