Description: Correct output of ‘vigra-config --docdir’.
Author: Jakub Wilk <jwilk@debian.org>
Forwarded: not-needed
Last-Update: 2011-01-22
Index: b/config/vigra-config.in
===================================================================
--- a/config/vigra-config.in
+++ b/config/vigra-config.in
@@ -91,4 +91,4 @@
     print '@CMAKE_INSTALL_PREFIX@/include'
 
 if op.docdir:
-    print '@DOCDIR@'
+    print '/usr/share/doc/libvigraimpex-dev/html/'
