From: Modestas Vainius <modax@debian.org>
Subject: Exclude QtUiTools symbols from public libraries
 Currently, libplasma and libkjsembed are affected.
Forwarded: not-needed

--- a/plasma/CMakeLists.txt
+++ b/plasma/CMakeLists.txt
@@ -270,6 +270,9 @@ target_link_libraries(plasma ${QT_QTUITO
                              ${QT_QTSCRIPT_LIBRARY} ${QT_QTNETWORK_LIBRARY} ${QT_QTXML_LIBRARY} ${QT_QTSQL_LIBRARY} ${QT_QTDECLARATIVE_LIBRARY}
                              ${KDE4_KDEUI_LIBS} ${KDE4_KDNSSD_LIBS} ${KDE4_THREADWEAVER_LIBS} ${PLASMA_EXTRA_LIBS})
 
+# Do not export QtUiTools internal symbols
+set_target_properties(plasma PROPERTIES LINK_FLAGS_DEBIAN "-Wl,--exclude-libs -Wl,libQtUiTools.a")
+
 if(QCA2_FOUND)
    target_link_libraries(plasma ${QCA2_LIBRARIES})
 endif(QCA2_FOUND)
--- a/kjsembed/kjsembed/CMakeLists.txt
+++ b/kjsembed/kjsembed/CMakeLists.txt
@@ -71,7 +71,8 @@ target_link_libraries(${KJSEMBEDLIBNAME}
 set_target_properties(${KJSEMBEDLIBNAME} PROPERTIES VERSION ${GENERIC_LIB_VERSION} SOVERSION ${GENERIC_LIB_SOVERSION} )
 install(TARGETS ${KJSEMBEDLIBNAME} EXPORT kdelibsLibraryTargets ${INSTALL_TARGETS_DEFAULT_ARGS})
 
-
+# Do not export QtUiTools internal symbols
+set_target_properties(${KJSEMBEDLIBNAME} PROPERTIES LINK_FLAGS_DEBIAN "-Wl,--exclude-libs -Wl,libQtUiTools.a")
 
 
 
