--- a/kdewidgets/CMakeLists.txt
+++ b/kdewidgets/CMakeLists.txt
@@ -54,7 +54,7 @@ if(QT_QTDESIGNER_FOUND)
                          )
   endif(NOT WIN32)
 
-  install(TARGETS kdewidgets  DESTINATION ${PLUGIN_INSTALL_DIR}/plugins/designer )
+  install(TARGETS kdewidgets  DESTINATION /usr/lib/qt4/plugins/designer )
 
   # deprecated widgets
   if(NOT KDE_NO_DEPRECATED)
@@ -83,7 +83,7 @@ if(QT_QTDESIGNER_FOUND)
                            )
     endif(NOT WIN32)
 
-    install(TARGETS kdedeprecated  DESTINATION ${PLUGIN_INSTALL_DIR}/plugins/designer )
+    install(TARGETS kdedeprecated  DESTINATION /usr/lib/qt4/plugins/designer )
   endif(NOT KDE_NO_DEPRECATED)
 
 
@@ -117,7 +117,7 @@ if(QT_QTDESIGNER_FOUND)
                          )
   endif(NOT WIN32)
 
-  install(TARGETS kdewebkitwidgets  DESTINATION ${PLUGIN_INSTALL_DIR}/plugins/designer )
+  install(TARGETS kdewebkitwidgets  DESTINATION /usr/lib/qt4/plugins/designer )
 
 
   if (QT_QT3SUPPORT_FOUND)
@@ -152,7 +152,7 @@ if(QT_QTDESIGNER_FOUND)
                              )
      endif(NOT WIN32)
 
-     install(TARGETS kde3supportwidgets  DESTINATION ${PLUGIN_INSTALL_DIR}/plugins/designer )
+     install(TARGETS kde3supportwidgets  DESTINATION /usr/lib/qt4/plugins/designer )
   endif (QT_QT3SUPPORT_FOUND)
 endif (QT_QTDESIGNER_FOUND)
 
