From: Markus Koschany <apo@gambaru.de>
Date: Mon, 14 Jan 2013 01:49:09 +0100
Subject: change glui include path

---
 Demos/AllBulletDemos/Main.cpp |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Demos/AllBulletDemos/Main.cpp b/Demos/AllBulletDemos/Main.cpp
index c59e18d..1f30a45 100644
--- a/Demos/AllBulletDemos/Main.cpp
+++ b/Demos/AllBulletDemos/Main.cpp
@@ -18,7 +18,7 @@
 
 #include <stdio.h>
 
-#include "glui/GL/glui.h"
+#include "<GL/glui.h>"
 #include "LinearMath/btScalar.h"
 #include "LinearMath/btMinMax.h"
 #include "BulletDynamics/ConstraintSolver/btSequentialImpulseConstraintSolver.h"
