From: Markus Koschany <apo@gambaru.de>
Date: Sun, 13 Jan 2013 21:56:52 +0100
Subject: do not build with embedded glui library

---
 Extras/CMakeLists.txt |    6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/Extras/CMakeLists.txt b/Extras/CMakeLists.txt
index a92f647..24fc0f1 100644
--- a/Extras/CMakeLists.txt
+++ b/Extras/CMakeLists.txt
@@ -2,6 +2,6 @@ SUBDIRS( Serialize ConvexDecomposition HACD GIMPACTUtils )
 
 #Maya Dynamica plugin is moved to http://dynamica.googlecode.com
 
-IF (USE_GLUT AND GLUT_FOUND)
-	SUBDIRS (glui)
-ENDIF ()
+#IF (USE_GLUT AND GLUT_FOUND)
+#	SUBDIRS (glui)
+#ENDIF ()
