From: Markus Koschany <apo@gambaru.de>
Date: Tue, 15 Jan 2013 22:20:29 +0100
Subject: do not build with embedded tinyxml library

---
 .../BulletXmlWorldImporter/CMakeLists.txt          |   23 ++++++++++++++------
 .../btBulletXmlWorldImporter.cpp                   |    2 +-
 2 files changed, 17 insertions(+), 8 deletions(-)

diff --git a/Extras/Serialize/BulletXmlWorldImporter/CMakeLists.txt b/Extras/Serialize/BulletXmlWorldImporter/CMakeLists.txt
index 6f311af..5f8819a 100644
--- a/Extras/Serialize/BulletXmlWorldImporter/CMakeLists.txt
+++ b/Extras/Serialize/BulletXmlWorldImporter/CMakeLists.txt
@@ -10,19 +10,28 @@ ADD_LIBRARY(
 	btBulletXmlWorldImporter.h 
 	string_split.cpp
 	string_split.h
-	tinyxml.cpp
-	tinyxml.h
-	tinystr.cpp
-	tinystr.h
-	tinyxmlerror.cpp
-	tinyxmlparser.cpp
 )
 
+# Look for the header file.
+FIND_PATH(TINYXML_INCLUDE_DIR NAMES tinyxml.h)
+MARK_AS_ADVANCED(TINYXML_INCLUDE_DIR)
+
+# # Look for the library.
+FIND_LIBRARY(TINYXML_LIBRARY NAMES tinyxml)
+MARK_AS_ADVANCED(TINYXML_LIBRARY)
+
+# all listed variables are TRUE
+INCLUDE(${CMAKE_ROOT}/Modules/FindPackageHandleStandardArgs.cmake)
+FIND_PACKAGE_HANDLE_STANDARD_ARGS(TinyXML DEFAULT_MSG TINYXML_LIBRARY TINYXML_INCLUDE_DIR)
+
+SET(TINYXML_LIBRARIES ${TINYXML_LIBRARY})
+SET(TINYXML_INCLUDE_DIRS ${TINYXML_INCLUDE_DIR})
+
 SET_TARGET_PROPERTIES(BulletXmlWorldImporter  PROPERTIES VERSION ${BULLET_VERSION})
 SET_TARGET_PROPERTIES(BulletXmlWorldImporter PROPERTIES SOVERSION ${BULLET_VERSION})
 
 IF (BUILD_SHARED_LIBS)
-        TARGET_LINK_LIBRARIES(BulletXmlWorldImporter BulletWorldImporter BulletDynamics BulletCollision BulletFileLoader LinearMath)
+        TARGET_LINK_LIBRARIES(BulletXmlWorldImporter tinyxml BulletWorldImporter BulletDynamics BulletCollision BulletFileLoader LinearMath)
 ENDIF (BUILD_SHARED_LIBS)
 
 IF (INSTALL_EXTRA_LIBS)
diff --git a/Extras/Serialize/BulletXmlWorldImporter/btBulletXmlWorldImporter.cpp b/Extras/Serialize/BulletXmlWorldImporter/btBulletXmlWorldImporter.cpp
index 5419572..cabfee8 100644
--- a/Extras/Serialize/BulletXmlWorldImporter/btBulletXmlWorldImporter.cpp
+++ b/Extras/Serialize/BulletXmlWorldImporter/btBulletXmlWorldImporter.cpp
@@ -14,7 +14,7 @@ subject to the following restrictions:
 */
 
 #include "btBulletXmlWorldImporter.h"
-#include "tinyxml.h"
+#include <tinyxml.h>
 #include "btBulletDynamicsCommon.h"
 #include "string_split.h"
 
