--- a/src/tests/Makefile.am
+++ b/src/tests/Makefile.am
@@ -143,9 +143,6 @@
 
 include cluster.mk
 include sasl.mk
-if SSL
-include ssl.mk
-endif
 
 # Test programs that are installed and therefore built as part of make, not make check
 
@@ -318,7 +315,6 @@
   topictest								\
   run_header_test							\
   header_test.py							\
-  ssl_test								\
   config.null								\
   ais_check								\
   run_federation_tests							\
