--- a/bindings/qmf2/examples/cpp/Makefile.am
+++ b/bindings/qmf2/examples/cpp/Makefile.am
@@ -24,13 +24,13 @@
 noinst_PROGRAMS=agent event_driven_list_agents list_agents print_events
 
 agent_SOURCES=agent.cpp
-agent_LDADD=$(top_builddir)/src/libqmf2.la
+agent_LDADD=$(top_builddir)/src/libqmf2.la -lqpidtypes -lqpidmessaging
 
 list_agents_SOURCES=list_agents.cpp
-list_agents_LDADD=$(top_builddir)/src/libqmf2.la
+list_agents_LDADD=$(top_builddir)/src/libqmf2.la -lqpidmessaging
 
 event_driven_list_agents_SOURCES=event_driven_list_agents.cpp
-event_driven_list_agents_LDADD=$(top_builddir)/src/libqmf2.la
+event_driven_list_agents_LDADD=$(top_builddir)/src/libqmf2.la -lqpidtypes -lqpidmessaging
 
 print_events_SOURCES=print_events.cpp
-print_events_LDADD=$(top_builddir)/src/libqmf2.la
+print_events_LDADD=$(top_builddir)/src/libqmf2.la -lqpidtypes -lqpidmessaging
--- a/examples/messaging/Makefile.am
+++ b/examples/messaging/Makefile.am
@@ -33,10 +33,10 @@
 hello_xml_LDADD=$(CLIENT_LIB)
 
 drain_SOURCES=drain.cpp OptionParser.h OptionParser.cpp
-drain_LDADD=$(CLIENT_LIB)
+drain_LDADD=$(CLIENT_LIB) -lqpidtypes
 
 spout_SOURCES=spout.cpp OptionParser.h OptionParser.cpp
-spout_LDADD=$(CLIENT_LIB)
+spout_LDADD=$(CLIENT_LIB) -lqpidtypes
 
 client_SOURCES=client.cpp
 client_LDADD=$(CLIENT_LIB)
@@ -45,10 +45,10 @@
 server_LDADD=$(CLIENT_LIB)
 
 map_sender_SOURCES=map_sender.cpp
-map_sender_LDADD=$(CLIENT_LIB)
+map_sender_LDADD=$(CLIENT_LIB) -lqpidtypes
 
 map_receiver_SOURCES=map_receiver.cpp
-map_receiver_LDADD=$(CLIENT_LIB)
+map_receiver_LDADD=$(CLIENT_LIB) -lqpidtypes
 
 examples_DATA=                 \
 	hello_world.cpp	       \
--- a/examples/old_api/direct/Makefile.am
+++ b/examples/old_api/direct/Makefile.am
@@ -23,13 +23,13 @@
 
 noinst_PROGRAMS=direct_producer listener declare_queues
 direct_producer_SOURCES=direct_producer.cpp
-direct_producer_LDADD=$(CLIENT_LIB)
+direct_producer_LDADD=$(CLIENT_LIB) -lqpidcommon
 
 listener_SOURCES=listener.cpp
-listener_LDADD=$(CLIENT_LIB)
+listener_LDADD=$(CLIENT_LIB) -lqpidcommon
 
 declare_queues_SOURCES=declare_queues.cpp
-declare_queues_LDADD=$(CLIENT_LIB)
+declare_queues_LDADD=$(CLIENT_LIB) -lqpidcommon
 
 examples_DATA=               \
 	direct_producer.cpp  \
--- a/examples/old_api/failover/Makefile.am
+++ b/examples/old_api/failover/Makefile.am
@@ -24,13 +24,13 @@
 noinst_PROGRAMS=declare_queues resuming_receiver replaying_sender
 
 declare_queues_SOURCES=declare_queues.cpp
-declare_queues_LDADD=$(CLIENT_LIB)
+declare_queues_LDADD=$(CLIENT_LIB) -lqpidcommon
 
 resuming_receiver_SOURCES=resuming_receiver.cpp
-resuming_receiver_LDADD=$(CLIENT_LIB)
+resuming_receiver_LDADD=$(CLIENT_LIB) -lqpidcommon
 
 replaying_sender_SOURCES=replaying_sender.cpp
-replaying_sender_LDADD=$(CLIENT_LIB)
+replaying_sender_LDADD=$(CLIENT_LIB) -lqpidcommon
 
 examples_DATA=                \
 	declare_queues.cpp    \
--- a/examples/old_api/fanout/Makefile.am
+++ b/examples/old_api/fanout/Makefile.am
@@ -26,7 +26,7 @@
 fanout_producer_LDADD=$(CLIENT_LIB)
 
 listener_SOURCES=listener.cpp
-listener_LDADD=$(CLIENT_LIB)
+listener_LDADD=$(CLIENT_LIB) -lqpidcommon
 
 examples_DATA=               \
 	fanout_producer.cpp  \
--- a/examples/old_api/pub-sub/Makefile.am
+++ b/examples/old_api/pub-sub/Makefile.am
@@ -24,10 +24,10 @@
 noinst_PROGRAMS=topic_listener topic_publisher
 
 topic_listener_SOURCES=topic_listener.cpp
-topic_listener_LDADD=$(CLIENT_LIB)
+topic_listener_LDADD=$(CLIENT_LIB) -lqpidcommon
 
 topic_publisher_SOURCES=topic_publisher.cpp
-topic_publisher_LDADD=$(CLIENT_LIB)
+topic_publisher_LDADD=$(CLIENT_LIB) -lqpidcommon
 
 examples_DATA=               \
 	topic_listener.cpp   \
--- a/examples/old_api/request-response/Makefile.am
+++ b/examples/old_api/request-response/Makefile.am
@@ -24,10 +24,10 @@
 noinst_PROGRAMS=client server
 
 client_SOURCES=client.cpp
-client_LDADD=$(CLIENT_LIB)
+client_LDADD=$(CLIENT_LIB) -lqpidcommon
 
 server_SOURCES=server.cpp
-server_LDADD=$(CLIENT_LIB)
+server_LDADD=$(CLIENT_LIB) -lqpidcommon
 
 examples_DATA=               \
 	server.cpp           \
--- a/examples/old_api/tradedemo/Makefile.am
+++ b/examples/old_api/tradedemo/Makefile.am
@@ -24,13 +24,13 @@
 noinst_PROGRAMS=topic_listener topic_publisher declare_queues
 
 topic_listener_SOURCES=topic_listener.cpp
-topic_listener_LDADD=$(CLIENT_LIB)
+topic_listener_LDADD=$(CLIENT_LIB) -lqpidcommon
 
 topic_publisher_SOURCES=topic_publisher.cpp 
-topic_publisher_LDADD=$(CLIENT_LIB)
+topic_publisher_LDADD=$(CLIENT_LIB) -lqpidcommon
 
 declare_queues_SOURCES=declare_queues.cpp
-declare_queues_LDADD=$(CLIENT_LIB)
+declare_queues_LDADD=$(CLIENT_LIB) -lqpidcommon
 
 
 examples_DATA=               \
--- a/examples/old_api/xml-exchange/Makefile.am
+++ b/examples/old_api/xml-exchange/Makefile.am
@@ -24,13 +24,13 @@
 noinst_PROGRAMS=declare_queues xml_producer listener
 
 declare_queues_SOURCES=declare_queues.cpp
-declare_queues_LDADD=$(CLIENT_LIB)
+declare_queues_LDADD=$(CLIENT_LIB) -lqpidcommon
 
 xml_producer_SOURCES=xml_producer.cpp
-xml_producer_LDADD=$(CLIENT_LIB)
+xml_producer_LDADD=$(CLIENT_LIB) -lqpidcommon
 
 listener_SOURCES=listener.cpp
-listener_LDADD=$(CLIENT_LIB)
+listener_LDADD=$(CLIENT_LIB) -lqpidcommon
 
 EXTRA_DIST=                \
 	README.txt             \
--- a/examples/qmf-console/Makefile.am
+++ b/examples/qmf-console/Makefile.am
@@ -25,19 +25,19 @@
 noinst_PROGRAMS=console printevents ping queuestats cluster-qmon
 
 console_SOURCES=console.cpp
-console_LDADD=$(CONSOLE_LIB)
+console_LDADD=$(CONSOLE_LIB) -lqpidcommon -lqpidclient
 
 printevents_SOURCES=printevents.cpp
-printevents_LDADD=$(CONSOLE_LIB)
+printevents_LDADD=$(CONSOLE_LIB) -lqpidcommon -lqpidclient
 
 ping_SOURCES=ping.cpp
-ping_LDADD=$(CONSOLE_LIB)
+ping_LDADD=$(CONSOLE_LIB) -lqpidcommon -lqpidclient
 
 queuestats_SOURCES=queuestats.cpp
-queuestats_LDADD=$(CONSOLE_LIB)
+queuestats_LDADD=$(CONSOLE_LIB) -lqpidcommon -lqpidclient
 
 cluster_qmon_SOURCES=cluster-qmon.cpp
-cluster_qmon_LDADD=$(CONSOLE_LIB)
+cluster_qmon_LDADD=$(CONSOLE_LIB) -lqpidcommon -lqpidclient
 
 examples_DATA= \
 	console.cpp \
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -145,7 +145,8 @@
 
 qpidd_LDADD =					\
   libqpidbroker.la				\
-  libqpidcommon.la
+  libqpidcommon.la              \
+  -lboost_program_options
 
 posix_qpidd_src = posix/QpiddBroker.cpp
 
--- a/src/tests/Makefile.am
+++ b/src/tests/Makefile.am
@@ -68,7 +68,7 @@
 TESTS+=unit_test
 check_PROGRAMS+=unit_test
 unit_test_LDADD=-lboost_unit_test_framework \
-	$(lib_messaging) $(lib_broker) $(lib_console) $(lib_qmf2)
+	$(lib_messaging) $(lib_broker) $(lib_console) $(lib_qmf2) $(lib_client) -lqpidtypes $(lib_common)
 
 unit_test_SOURCES= unit_test.cpp unit_test.h \
 	MessagingSessionTests.cpp \
@@ -157,7 +157,7 @@
   receiver.cpp \
   TestOptions.h \
   ConnectionOptions.h
-receiver_LDADD = $(lib_client)
+receiver_LDADD = $(lib_client) -lboost_program_options -lqpidcommon
 
 qpidexectest_PROGRAMS += sender
 sender_SOURCES = \
@@ -165,7 +165,7 @@
   TestOptions.h \
   ConnectionOptions.h \
   Statistics.cpp
-sender_LDADD = $(lib_messaging)
+sender_LDADD = $(lib_messaging) -lboost_program_options -lqpidcommon -lqpidtypes -lqpidclient
 
 qpidexectest_PROGRAMS += qpid-receive
 qpid_receive_SOURCES = \
@@ -174,7 +174,7 @@
   ConnectionOptions.h \
   Statistics.h \
   Statistics.cpp
-qpid_receive_LDADD = $(lib_messaging)
+qpid_receive_LDADD = $(lib_messaging) -lboost_program_options -lqpidcommon -lqpidtypes
 
 qpidexectest_PROGRAMS += qpid-send
 qpid_send_SOURCES = \
@@ -183,42 +183,42 @@
   ConnectionOptions.h \
   Statistics.h \
   Statistics.cpp
-qpid_send_LDADD = $(lib_messaging)
+qpid_send_LDADD = $(lib_messaging) -lboost_program_options -lqpidcommon -lqpidtypes
 
 qpidexectest_PROGRAMS+=qpid-perftest
 qpid_perftest_SOURCES=qpid-perftest.cpp test_tools.h TestOptions.h ConnectionOptions.h
 qpid_perftest_INCLUDES=$(PUBLIC_INCLUDES)
-qpid_perftest_LDADD=$(lib_client) 
+qpid_perftest_LDADD=$(lib_client) -lboost_program_options -lqpidcommon -lqpidtypes
 
 qpidexectest_PROGRAMS+=qpid-txtest
 qpid_txtest_INCLUDES=$(PUBLIC_INCLUDES)
 qpid_txtest_SOURCES=qpid-txtest.cpp  TestOptions.h ConnectionOptions.h
-qpid_txtest_LDADD=$(lib_client) 
+qpid_txtest_LDADD=$(lib_client) -lboost_program_options -lqpidcommon -lqpidtypes
 
 qpidexectest_PROGRAMS+=qpid-latency-test
 qpid_latency_test_INCLUDES=$(PUBLIC_INCLUDES)
 qpid_latency_test_SOURCES=qpid-latency-test.cpp TestOptions.h ConnectionOptions.h
-qpid_latency_test_LDADD=$(lib_client) 
+qpid_latency_test_LDADD=$(lib_client) -lboost_program_options -lqpidcommon -lqpidtypes
 
 qpidexectest_PROGRAMS+=qpid-client-test
 qpid_client_test_INCLUDES=$(PUBLIC_INCLUDES)
 qpid_client_test_SOURCES=qpid-client-test.cpp TestOptions.h ConnectionOptions.h
-qpid_client_test_LDADD=$(lib_client) 
+qpid_client_test_LDADD=$(lib_client) -lboost_program_options -lqpidcommon -lqpidtypes
 
 qpidexectest_PROGRAMS+=qpid-topic-listener
 qpid_topic_listener_INCLUDES=$(PUBLIC_INCLUDES)
 qpid_topic_listener_SOURCES=qpid-topic-listener.cpp TestOptions.h ConnectionOptions.h
-qpid_topic_listener_LDADD=$(lib_client) 
+qpid_topic_listener_LDADD=$(lib_client) -lboost_program_options -lqpidcommon -lqpidtypes
 
 qpidexectest_PROGRAMS+=qpid-topic-publisher
 qpid_topic_publisher_INCLUDES=$(PUBLIC_INCLUDES)
 qpid_topic_publisher_SOURCES=qpid-topic-publisher.cpp TestOptions.h ConnectionOptions.h
-qpid_topic_publisher_LDADD=$(lib_client) 
+qpid_topic_publisher_LDADD=$(lib_client) -lboost_program_options -lqpidcommon -lqpidtypes
 
 qpidexectest_PROGRAMS+=qpid-ping
 qpid_ping_INCLUDES=$(PUBLIC_INCLUDES)
 qpid_ping_SOURCES=qpid-ping.cpp test_tools.h TestOptions.h ConnectionOptions.h
-qpid_ping_LDADD=$(lib_client)
+qpid_ping_LDADD=$(lib_client) -lboost_program_options -lqpidcommon -lqpidtypes
 
 #
 # Other test programs
@@ -227,74 +227,74 @@
 check_PROGRAMS+=echotest
 echotest_INCLUDES=$(PUBLIC_INCLUDES)
 echotest_SOURCES=echotest.cpp TestOptions.h ConnectionOptions.h
-echotest_LDADD=$(lib_client)
+echotest_LDADD=$(lib_client) $(lib_common) -lboost_program_options
 
 check_PROGRAMS+=publish
 publish_INCLUDES=$(PUBLIC_INCLUDES)
 publish_SOURCES=publish.cpp TestOptions.h ConnectionOptions.h
-publish_LDADD=$(lib_client) 
+publish_LDADD=$(lib_client) $(lib_common) -lboost_program_options
 
 check_PROGRAMS+=consume
 consume_INCLUDES=$(PUBLIC_INCLUDES)
 consume_SOURCES=consume.cpp  TestOptions.h ConnectionOptions.h
-consume_LDADD=$(lib_client) 
+consume_LDADD=$(lib_client) $(lib_common) -lboost_program_options
 
 check_PROGRAMS+=header_test
 header_test_INCLUDES=$(PUBLIC_INCLUDES)
 header_test_SOURCES=header_test.cpp TestOptions.h ConnectionOptions.h
-header_test_LDADD=$(lib_client) 
+header_test_LDADD=$(lib_client) $(lib_common) -lboost_program_options
 
 check_PROGRAMS+=failover_soak
 failover_soak_INCLUDES=$(PUBLIC_INCLUDES)
 failover_soak_SOURCES=failover_soak.cpp  ForkedBroker.h ForkedBroker.cpp
-failover_soak_LDADD=$(lib_client) $(lib_broker)
+failover_soak_LDADD=$(lib_client) $(lib_broker) $(lib_common) -lboost_program_options
 
 check_PROGRAMS+=declare_queues
 declare_queues_INCLUDES=$(PUBLIC_INCLUDES)
 declare_queues_SOURCES=declare_queues.cpp  
-declare_queues_LDADD=$(lib_client) 
+declare_queues_LDADD=$(lib_client) $(lib_common) -lboost_program_options
 
 check_PROGRAMS+=replaying_sender
 replaying_sender_INCLUDES=$(PUBLIC_INCLUDES)
 replaying_sender_SOURCES=replaying_sender.cpp  
-replaying_sender_LDADD=$(lib_client) 
+replaying_sender_LDADD=$(lib_client) $(lib_common) -lboost_program_options
 
 check_PROGRAMS+=resuming_receiver
 resuming_receiver_INCLUDES=$(PUBLIC_INCLUDES)
 resuming_receiver_SOURCES=resuming_receiver.cpp  
-resuming_receiver_LDADD=$(lib_client) 
+resuming_receiver_LDADD=$(lib_client) $(lib_common) -lboost_program_options
 
 check_PROGRAMS+=txshift
 txshift_INCLUDES=$(PUBLIC_INCLUDES)
 txshift_SOURCES=txshift.cpp  TestOptions.h ConnectionOptions.h
-txshift_LDADD=$(lib_client) 
+txshift_LDADD=$(lib_client) $(lib_common) -lboost_program_options
 
 check_PROGRAMS+=txjob
 txjob_INCLUDES=$(PUBLIC_INCLUDES)
 txjob_SOURCES=txjob.cpp  TestOptions.h ConnectionOptions.h
-txjob_LDADD=$(lib_client) 
+txjob_LDADD=$(lib_client)  $(lib_common) -lboost_program_options
 
 check_PROGRAMS+=PollerTest
 PollerTest_SOURCES=PollerTest.cpp
-PollerTest_LDADD=$(lib_common) $(lib_client) $(SOCKLIBS)
+PollerTest_LDADD=$(lib_common) $(lib_client) $(SOCKLIBS) $(lib_common) -lboost_program_options
 
 check_PROGRAMS+=DispatcherTest
 DispatcherTest_SOURCES=DispatcherTest.cpp
-DispatcherTest_LDADD=$(lib_common) $(lib_client) $(SOCKLIBS)
+DispatcherTest_LDADD=$(lib_common) $(lib_client) $(SOCKLIBS) $(lib_common) -lboost_program_options
 
 check_PROGRAMS+=datagen
 datagen_SOURCES=datagen.cpp
-datagen_LDADD=$(lib_common) $(lib_client)
+datagen_LDADD=$(lib_common) $(lib_client) $(lib_common) -lboost_program_options
 
 check_PROGRAMS+=qpid-stream
 qpid_stream_INCLUDES=$(PUBLIC_INCLUDES)
 qpid_stream_SOURCES=qpid-stream.cpp
-qpid_stream_LDADD=$(lib_messaging)
+qpid_stream_LDADD=$(lib_messaging) $(lib_common) -lboost_program_options -lqpidtypes
 
 check_PROGRAMS+=msg_group_test
 msg_group_test_INCLUDES=$(PUBLIC_INCLUDES)
 msg_group_test_SOURCES=msg_group_test.cpp
-msg_group_test_LDADD=$(lib_messaging)
+msg_group_test_LDADD=$(lib_messaging) $(lib_common) -lboost_program_options -lqpidtypes
 
 TESTS_ENVIRONMENT = \
     VALGRIND=$(VALGRIND) \
--- a/src/tests/cluster.mk
+++ b/src/tests/cluster.mk
@@ -97,7 +97,7 @@
 # build and so we need to make sure it is present in the tarball
 EXTRA_DIST += StoreStatus.cpp
 
-cluster_test_LDADD=$(lib_client) $(lib_broker) ../cluster.la -lboost_unit_test_framework
+cluster_test_LDADD=$(lib_client) $(lib_broker) ../cluster.la -lboost_unit_test_framework -lqpidcommon
 
 qpidexectest_SCRIPTS += run_cluster_tests brokertest.py cluster_tests.py cluster_test_logs.py run_long_cluster_tests long_cluster_tests.py testlib.py cluster_tests.fail
 qpidexectest_SCRIPTS += $(CLUSTER_TEST_SCRIPTS_LIST)
--- a/src/tests/testagent.mk
+++ b/src/tests/testagent.mk
@@ -46,6 +46,6 @@
 qpidexectest_PROGRAMS+=testagent
 testagent_CXXFLAGS=$(CXXFLAGS) -Itestagent_gen
 testagent_SOURCES=testagent.cpp $(TESTAGENT_GEN_SRC)
-testagent_LDADD=$(top_builddir)/src/libqmf.la
+testagent_LDADD=$(top_builddir)/src/libqmf.la -lqpidcommon -lqpidtypes -lqpidclient
 
 EXTRA_DIST+=testagent.xml
--- a/src/tests/sasl.mk
+++ b/src/tests/sasl.mk
@@ -23,7 +23,7 @@
 check_PROGRAMS+=cluster_authentication_soak
 cluster_authentication_soak_INCLUDES=$(PUBLIC_INCLUDES)
 cluster_authentication_soak_SOURCES=cluster_authentication_soak.cpp  ForkedBroker.h ForkedBroker.cpp
-cluster_authentication_soak_LDADD=$(lib_client) $(lib_broker)
+cluster_authentication_soak_LDADD=$(lib_client) $(lib_broker) $(lib_common)
 
 # Note: sasl_version is not a test -- it is a tool used by tests.
 check_PROGRAMS+=sasl_version
