--- a/bindings/qpid/Makefile.am
+++ b/bindings/qpid/Makefile.am
@@ -44,17 +44,18 @@
 
 perl/Makefile: perl/cqpid_perl.cpp
 	cd perl; \
-		$(PERL) Makefile.PL PREFIX=$(prefix) LIB=$(PERL_ARCHLIB) ; \
+		$(PERL) Makefile.PL INSTALLDIRS=vendor ; \
 	cd ..
 
 all-local: perl/Makefile
 	cd perl; \
-		$(MAKE) OPTIMIZE="$(CXXFLAGS)" ; \
+		$(MAKE) OPTIMIZE="$(CXXFLAGS) $(CPPFLAGS) -Wno-error=format-security" ; \
+		find -name '*.so' -exec chrpath -d "{}" \; ; \
 	cd ..
 
 install-exec-local:
 	cd perl ; \
-		$(MAKE) pure_install DESTDIR=$(prefix) ; \
+		$(MAKE) pure_install DESTDIR=$(DESTDIR) ; \
 	cd ..
 
 clean-local:
--- a/bindings/qpid/perl/Makefile.PL.in
+++ b/bindings/qpid/perl/Makefile.PL.in
@@ -13,6 +13,10 @@
     DEFINE            => '',
     INC               => '-I@top_srcdir@/include -I@top_builddir@/include -I@top_srcdir@/src -I@top_builddir@/src -I@PERL_INC@',
     C                 => ['cqpid_perl.cpp'],
-	# Un-comment this if you add C files to link with later:
-     OBJECT            => 'cqpid_perl.o',
+    # Un-comment this if you add C files to link with later:
+    OBJECT            => 'cqpid_perl.o',
+    # $Config{ccflags}/$Config{lddlflags} doesn't contain hardening flags
+    # (yet), add them manually.
+    CCFLAGS           => "$Config{ccflags} $ENV{CFLAGS} $ENV{CPPFLAGS} $ENV{NGX_PM_CFLAGS}",
+    LDDLFLAGS         => "$Config{lddlflags} $ENV{LDFLAGS}",
 );
