--- a/src/qpid/RefCountedBuffer.cpp
+++ b/src/qpid/RefCountedBuffer.cpp
@@ -32,12 +32,13 @@
 
 BufferRef RefCountedBuffer::create(size_t n) {
     void* store=::malloc (n + sizeof(RefCountedBuffer));
+
     if (NULL == store)
         throw std::bad_alloc();
     new(store) RefCountedBuffer;
     char* start = reinterpret_cast<char *>(store) + sizeof(RefCountedBuffer);
     return BufferRef(
-        boost::intrusive_ptr<RefCounted>(reinterpret_cast<RefCountedBuffer*>(store)),
+        boost::intrusive_ptr<RefCounted>(reinterpret_cast<RefCountedBuffer*>((void*)store)),
         start, start+n);
 }
 
