Index: perl-tk-804.029/PNG/libpng/contrib/pngsuite/README
===================================================================
--- perl-tk-804.029.orig/PNG/libpng/contrib/pngsuite/README	2012-06-14 20:02:17.671593125 +0100
+++ perl-tk-804.029/PNG/libpng/contrib/pngsuite/README	2012-06-14 20:03:00.067071323 +0100
@@ -3,8 +3,8 @@
 --------
 (c) Willem van Schaik, 1999
 
-Permission to use, copy, and distribute these images for any purpose and
-without fee is hereby granted.
+Permission to use, copy, modify, and distribute these images for any
+purpose and without fee is hereby granted.
 
 These 15 images are part of the much larger PngSuite test-set of 
 images, available for developers of PNG supporting software. The 
