Last-Update: 2012-08-18
Author: Dmitry Smirnov <onlyjob@member.fsf.org>
Forwarded: not-needed
Bug-Winswitch: https://winswitch.org/trac/ticket/212
Description: Disable local VBOX sessions by default.

--- a/skel/share/applications/winswitch.desktop
+++ b/skel/share/applications/winswitch.desktop
@@ -2,7 +2,7 @@
 Name=Window Switch
 Comment=Move running applications between networked computers
 TryExec=winswitch_applet
-Exec=winswitch_applet --open_server_config=%f
+Exec=winswitch_applet --no-virtualbox --open_server_config=%f
 Icon=winswitch_applet
 Terminal=false
 Type=Application
