diff -ru yelp-2.20.0.orig/src/yelp-man-parser.c yelp-2.20.0/src/yelp-man-parser.c
--- yelp-2.20.0.orig/src/yelp-man-parser.c	2007-09-17 21:49:09.000000000 +0100
+++ yelp-2.20.0/src/yelp-man-parser.c	2008-02-11 10:27:31.000000000 +0000
@@ -99,12 +99,30 @@
 {
     GError *errormsg = NULL;
     /*gchar *ptr = NULL;*/
+    gchar **recode_argv;
+    gint recode_stdout_fd;
 
     g_return_val_if_fail (parser != NULL, NULL);
     g_return_val_if_fail (file != NULL, NULL);
     g_return_val_if_fail (encoding != NULL, NULL);
 	
-    parser->channel = yelp_io_channel_new_file (file, NULL);
+    recode_argv = g_new (gchar *, 5);
+    recode_argv[0] = g_strdup ("man");
+    recode_argv[1] = g_strdup ("--recode");
+    recode_argv[2] = g_strdup ("UTF-8");
+    recode_argv[3] = g_strdup (file);
+    recode_argv[4] = NULL;
+    if (g_spawn_async_with_pipes (NULL, recode_argv, NULL,
+				  G_SPAWN_SEARCH_PATH |
+				  G_SPAWN_STDERR_TO_DEV_NULL,
+				  NULL, NULL, NULL,
+				  NULL, &recode_stdout_fd, NULL,
+				  NULL)) {
+	parser->channel = g_io_channel_unix_new (recode_stdout_fd);
+	encoding = "UTF-8";
+    } else
+	parser->channel = yelp_io_channel_new_file (file, NULL);
+    g_strfreev (recode_argv);
 
     if (!parser->channel)
 	return NULL;
