Debian #461625
GNOME #533718

Index: yelp-3.2.0/libyelp/yelp-info-parser.c
===================================================================
--- yelp-3.2.0.orig/libyelp/yelp-info-parser.c	2011-10-12 22:57:01.429024990 +0200
+++ yelp-3.2.0/libyelp/yelp-info-parser.c	2011-10-12 22:57:03.209002612 +0200
@@ -661,10 +661,12 @@
   gchar *title;
   gchar *reftitle;
 
-  if (!it) {
+  if (!it || !comp) {
     return NULL;
   }
 
+  g_return_val_if_fail (GTK_IS_TREE_MODEL (model), NULL);
+
   r = gtk_tree_iter_copy (it);
   tmp = gtk_tree_iter_copy (it);
 
