Change some default directories.

Index: a/tv/linux/plat/config.py
===================================================================
--- a/tv/linux/plat/config.py	2011-06-07 23:47:12.000000000 +0200
+++ b/tv/linux/plat/config.py	2011-06-07 23:47:28.000000000 +0200
@@ -164,10 +164,10 @@
 
     if descriptor == prefs.MOVIES_DIRECTORY:
         value = os.path.expanduser(os.path.join(options.user_home,
-                                                'Videos/Miro'))
+                                                '.miro/Movies'))
 
     elif descriptor == prefs.NON_VIDEO_DIRECTORY:
-        value = os.path.expanduser(os.path.join(options.user_home, 'Desktop'))
+        value = os.path.expanduser(os.path.join(options.user_home, '.miro/Nonvideo'))
 
     elif descriptor == prefs.GETTEXT_PATHNAME:
         value = resources.path("../../locale")
