http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=640986

--- a/tv/linux/miro.desktop.orig	2011-08-10 17:53:51.000000000 +0000
+++ b/tv/linux/miro.desktop	2011-09-09 08:15:00.000000000 +0000
@@ -1,7 +1,8 @@
 [Desktop Entry]
 Type=Application
-Name=Miro music and video player
+Name=Miro
 GenericName=Media player
+GenericName[de]=Medienwiedergabe
 Comment=Fully featured video and music player, podcast client, and bittorrent client
 Comment[de]=Multimedia Player
 Comment[es]=Reproductor multimedia
