Fix non-working gdb invokation.

--- a/linux/miro.orig	2011-07-03 18:55:22.000000000 +0200
+++ b/linux/miro	2011-07-03 18:56:39.000000000 +0200
@@ -38,9 +38,9 @@
     esac
 done
 
-if [ -e "./miro.real" ]
+if [ -e "/usr/bin/miro.real" ]
 then
-    MIROREAL=./miro.real
+    MIROREAL=/usr/bin/miro.real
 else
     MIROREAL=`which gdb`
 fi
@@ -60,7 +60,7 @@
     fi
 
     $GDB -ex 'set breakpoint pending on' \
-        -ex 'run' --args $PYTHON ./miro.real --sync "$@"
+        -ex 'run' --args $PYTHON /usr/bin/miro.real --sync "$@"
 else
     miro.real "$@"
 fi
