Index: camlimages/src/bitmap.ml
===================================================================
--- camlimages.orig/src/bitmap.ml	2011-10-17 00:02:17.941672767 +0200
+++ camlimages/src/bitmap.ml	2011-10-17 00:04:03.732755736 +0200
@@ -186,7 +186,7 @@
 
   let swap_out_eldest words =
     let sorted =
-      Sort.list (fun b1 b2 -> b1.last_used < b2.last_used) !swappable_blocks in
+      List.sort (fun b1 b2 -> compare b1.last_used b2.last_used) !swappable_blocks in
     let rec swapper sorted i =
      match sorted with
       | [] -> ()
