From: Ansgar Burchardt <ansgar@43-1.org>
Date: Thu, 26 Aug 2010 17:15:34 +0900
Origin: vendor
Subject: Fix spelling errors in documentation

--- a/lib/Date/Calendar/Year.pod
+++ b/lib/Date/Calendar/Year.pod
@@ -345,7 +345,7 @@
 can be queried with the "labels()" and "search()" methods,
 and appears when printing a calendar, for instance.
 
-The character ":" (mnemonic: divided into two halfs) specifies
+The character ":" (mnemonic: divided into two halves) specifies
 that the date in question is only a "half" holiday, i.e., you
 only get half a day off instead of a full day. Some companies
 have this sort of thing. C<:-)>
--- a/lib/Date/Calc/Object.pod
+++ b/lib/Date/Calc/Object.pod
@@ -483,7 +483,7 @@
 
 Invalid Dates
 
-Only "new()" allows to create objects containing possibly invalid
+Only "new()" allows one to create objects containing possibly invalid
 dates (needed for reading in and evaluating user input, for example).
 
 =item *
--- a/lib/Date/Calendar.pod
+++ b/lib/Date/Calendar.pod
@@ -398,7 +398,7 @@
 which your profile contains "half" holidays, or when you add
 or subtract fractions of a day.
 
-Sometimes it is not possible to accomodate the requested number
+Sometimes it is not possible to accommodate the requested number
 of work days, and a rest remains.
 
 This rest can currently only assume the value "0.0" (zero),
--- a/lib/Date/Calendar/Profiles.pod
+++ b/lib/Date/Calendar/Profiles.pod
@@ -62,7 +62,7 @@
 responsible for parsing the calendar schemes contained
 here in the Date::Calendar::Profiles module.
 
-This method offers a "mini-language" which allows to
+This method offers a "mini-language" which allows one to
 specify common date formulas, like for instance a simple
 fixed date (in various different formats, e.g. american
 or european), or things like "the second Sunday of May"
