From f0487376671ffd6ac3fc121657f1fbd0acea3cb0 Mon Sep 17 00:00:00 2001
From: Nikias Bassen
Date: Tue, 07 Jun 2011 15:59:22 +0000
Subject: lockdown: fix support for iOS 5

---
diff --git a/src/lockdown.c b/src/lockdown.c
index 935f24e..15b3d69 100644
--- a/src/lockdown.c
+++ b/src/lockdown.c
@@ -86,7 +86,9 @@ static int lockdown_check_result(plist_t dict, const char *query_match)
 
 	plist_t result_node = plist_dict_get_item(dict, "Result");
 	if (!result_node) {
-		return ret;
+		/* iOS 5: the 'Result' key is not present anymore.
+		   Just assume success here */
+		return RESULT_SUCCESS;
 	}
 
 	plist_type result_type = plist_get_node_type(result_node);
--
cgit v0.8.3.1-34-gbf3d
