Index: libimobiledevice/swig/Makefile.am
===================================================================
--- libimobiledevice.orig/swig/Makefile.am	2011-05-24 01:36:43.000000000 +0200
+++ libimobiledevice/swig/Makefile.am	2011-07-05 22:17:56.000000000 +0200
@@ -25,7 +25,7 @@
 nodist__imobiledevice_la_SOURCES = imobiledevice_wrap.cxx $(SWIG_SOURCES)
 _imobiledevice_la_CFLAGS = $(PYTHON_CPPFLAGS) -I$(top_srcdir)/src
 _imobiledevice_la_LDFLAGS = -module -avoid-version $(PYTHON_LDFLAGS)
-_imobiledevice_la_LIBADD = $(top_builddir)/src/libimobiledevice.la $(libplistmm_LIBS)
+_imobiledevice_la_LIBADD = $(top_builddir)/src/libimobiledevice.la $(libplistmm_LIBS) $(libplist_LIBS)
 
 imobiledevice_wrap.cxx : $(SWIG_SOURCES)
 	$(SWIG) $(SWIG_PYTHON_OPT) $(AM_CPPFLAGS) -I$(top_srcdir)/src -o $@ $<
Index: libimobiledevice/swig/Makefile.in
===================================================================
--- libimobiledevice.orig/swig/Makefile.in	2011-05-24 01:36:43.000000000 +0200
+++ libimobiledevice/swig/Makefile.in	2011-07-05 22:17:56.000000000 +0200
@@ -82,7 +82,7 @@
 am__DEPENDENCIES_1 =
 @HAVE_SWIG_TRUE@_imobiledevice_la_DEPENDENCIES =  \
 @HAVE_SWIG_TRUE@	$(top_builddir)/src/libimobiledevice.la \
-@HAVE_SWIG_TRUE@	$(am__DEPENDENCIES_1)
+@HAVE_SWIG_TRUE@	$(am__DEPENDENCIES_1) $(am__DEPENDENCIES_1)
 am__objects_1 =
 @HAVE_SWIG_TRUE@nodist__imobiledevice_la_OBJECTS =  \
 @HAVE_SWIG_TRUE@	imobiledevice_wrap.lo $(am__objects_1)
@@ -311,7 +311,7 @@
 @HAVE_SWIG_TRUE@nodist__imobiledevice_la_SOURCES = imobiledevice_wrap.cxx $(SWIG_SOURCES)
 @HAVE_SWIG_TRUE@_imobiledevice_la_CFLAGS = $(PYTHON_CPPFLAGS) -I$(top_srcdir)/src
 @HAVE_SWIG_TRUE@_imobiledevice_la_LDFLAGS = -module -avoid-version $(PYTHON_LDFLAGS)
-@HAVE_SWIG_TRUE@_imobiledevice_la_LIBADD = $(top_builddir)/src/libimobiledevice.la $(libplistmm_LIBS)
+@HAVE_SWIG_TRUE@_imobiledevice_la_LIBADD = $(top_builddir)/src/libimobiledevice.la $(libplistmm_LIBS) $(libplist_LIBS)
 all: $(BUILT_SOURCES)
 	$(MAKE) $(AM_MAKEFLAGS) all-am
 
