Index: libimobiledevice/dev/Makefile.am
===================================================================
--- libimobiledevice.orig/dev/Makefile.am	2011-07-05 22:38:19.000000000 +0200
+++ libimobiledevice/dev/Makefile.am	2011-07-05 22:40:55.000000000 +0200
@@ -1,7 +1,7 @@
 AM_CPPFLAGS = -I$(top_srcdir)/include
 
 AM_CFLAGS = $(GLOBAL_CFLAGS) $(libglib2_CFLAGS) $(libgnutls_CFLAGS) $(libtasn1_CFLAGS) $(libgthread2_CFLAGS) $(LFS_CFLAGS)
-AM_LDFLAGS = $(libglib2_LIBS) $(libgnutls_LIBS) $(libtasn1_LIBS) $(libgthread2_LIBS)
+AM_LDFLAGS = $(libglib2_LIBS) $(libgnutls_LIBS) $(libtasn1_LIBS) $(libgthread2_LIBS) $(libplist_LIBS)
 
 if ENABLE_DEVTOOLS
 noinst_PROGRAMS = ideviceclient lckd-client afccheck filerelaytest housearresttest
Index: libimobiledevice/tools/Makefile.am
===================================================================
--- libimobiledevice.orig/tools/Makefile.am	2011-07-05 22:41:04.000000000 +0200
+++ libimobiledevice/tools/Makefile.am	2011-07-05 22:41:59.000000000 +0200
@@ -1,7 +1,7 @@
 AM_CPPFLAGS = -I$(top_srcdir)/include
 
 AM_CFLAGS = $(GLOBAL_CFLAGS) $(libglib2_CFLAGS) $(libgnutls_CFLAGS) $(libtasn1_CFLAGS) $(libgthread2_CFLAGS) $(LFS_CFLAGS)
-AM_LDFLAGS = $(libglib2_LIBS) $(libgnutls_LIBS) $(libtasn1_LIBS) $(libgthread2_LIBS)
+AM_LDFLAGS = $(libglib2_LIBS) $(libgnutls_LIBS) $(libtasn1_LIBS) $(libgthread2_LIBS) $(libplist_LIBS)
 
 bin_PROGRAMS = idevice_id ideviceinfo idevicepair idevicesyslog idevicebackup idevicebackup2 ideviceimagemounter idevicescreenshot ideviceenterrecovery idevicedate
 
