Description: Remove check and link to gcrypt
Author: Julien Lavergne <julien.lavergne@gmail.com>
Bug: http://libiphone.lighthouseapp.com/projects/27916-libiphone/tickets/239-remove-check-for-gcry-revert-e449f1b3699f356b4e3238ca287e371656c4136c
Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=638417
Last-Update: 2011-10-25

Index: libimobiledevice/configure.ac
===================================================================
--- libimobiledevice.orig/configure.ac	2011-10-25 00:32:52.000000000 +0200
+++ libimobiledevice/configure.ac	2011-10-25 00:34:56.000000000 +0200
@@ -33,7 +33,6 @@
 PKG_CHECK_MODULES(libtasn1, libtasn1 >= 1.1)
 PKG_CHECK_MODULES(libplist, libplist >= 0.15)
 PKG_CHECK_MODULES(libplistmm, libplist++ >= 0.15)
-AC_CHECK_LIB(gcrypt, gcry_control, [AC_SUBST(libgcrypt_LIBS,[-lgcrypt])], [AC_MSG_ERROR([libgcrypt is required to build libimobiledevice])])
 
 # Checks for header files.
 AC_HEADER_STDC
Index: libimobiledevice/src/Makefile.am
===================================================================
--- libimobiledevice.orig/src/Makefile.am	2011-10-25 00:33:04.000000000 +0200
+++ libimobiledevice/src/Makefile.am	2011-10-25 00:34:46.000000000 +0200
@@ -1,7 +1,7 @@
 AM_CPPFLAGS = -I$(top_srcdir)/include
 
 AM_CFLAGS = $(GLOBAL_CFLAGS) $(libusbmuxd_CFLAGS) $(libglib2_CFLAGS) $(libgnutls_CFLAGS) $(libtasn1_CFLAGS) $(libgthread2_CFLAGS) $(libplist_CFLAGS) $(LFS_CFLAGS)
-AM_LDFLAGS = $(libglib2_LIBS) $(libgnutls_LIBS) $(libtasn1_LIBS) $(libgthread2_LIBS) $(libplist_LIBS) $(libusbmuxd_LIBS) $(libgcrypt_LIBS)
+AM_LDFLAGS = $(libglib2_LIBS) $(libgnutls_LIBS) $(libtasn1_LIBS) $(libgthread2_LIBS) $(libplist_LIBS) $(libusbmuxd_LIBS)
 
 lib_LTLIBRARIES = libimobiledevice.la
 libimobiledevice_la_LDFLAGS = $(AM_LDFLAGS) -version-info $(LIBIMOBILEDEVICE_SO_VERSION) -no-undefined
