Index: icedtea-web-1.3/Makefile.am
===================================================================
--- icedtea-web-1.3.orig/Makefile.am	2012-09-06 13:43:55.047975390 +0000
+++ icedtea-web-1.3/Makefile.am	2012-09-06 14:04:59.508015010 +0000
@@ -259,7 +259,7 @@
 	mkdir -p $(PLUGIN_DIR) && \
 	cd $(PLUGIN_DIR) && \
 	$(CXX) $(CXXFLAGS) \
-	   $(DEFS) $(VERSION_DEFS) \
+	   $(DEFS) \
 	  -DJDK_UPDATE_VERSION="\"$(JDK_UPDATE_VERSION)\"" \
 	  -DPLUGIN_NAME="\"IcedTea-Web Plugin\"" \
 	  -DPLUGIN_VERSION="\"$(PLUGIN_VERSION)\"" \
@@ -267,6 +267,7 @@
 	  -DMOZILLA_VERSION_COLLAPSED="$(MOZILLA_VERSION_COLLAPSED)" \
 	  -DICEDTEA_WEB_JRE="\"$(SYSTEM_JRE_DIR)\"" \
 	  -DPLUGIN_BOOTCLASSPATH=$(PLUGIN_BOOTCLASSPATH) \
+	  $(VERSION_DEFS) \
 	  $(GLIB_CFLAGS) \
 	  $(GTK_CFLAGS) \
 	  $(MOZILLA_CFLAGS) \
Index: icedtea-web-1.3/Makefile.in
===================================================================
--- icedtea-web-1.3.orig/Makefile.in	2012-08-27 19:36:58.431461082 +0000
+++ icedtea-web-1.3/Makefile.in	2012-09-06 14:04:59.508015010 +0000
@@ -780,7 +780,7 @@
 @ENABLE_PLUGIN_TRUE@	mkdir -p $(PLUGIN_DIR) && \
 @ENABLE_PLUGIN_TRUE@	cd $(PLUGIN_DIR) && \
 @ENABLE_PLUGIN_TRUE@	$(CXX) $(CXXFLAGS) \
-@ENABLE_PLUGIN_TRUE@	   $(DEFS) $(VERSION_DEFS) \
+@ENABLE_PLUGIN_TRUE@	   $(DEFS) \
 @ENABLE_PLUGIN_TRUE@	  -DJDK_UPDATE_VERSION="\"$(JDK_UPDATE_VERSION)\"" \
 @ENABLE_PLUGIN_TRUE@	  -DPLUGIN_NAME="\"IcedTea-Web Plugin\"" \
 @ENABLE_PLUGIN_TRUE@	  -DPLUGIN_VERSION="\"$(PLUGIN_VERSION)\"" \
@@ -788,6 +788,7 @@
 @ENABLE_PLUGIN_TRUE@	  -DMOZILLA_VERSION_COLLAPSED="$(MOZILLA_VERSION_COLLAPSED)" \
 @ENABLE_PLUGIN_TRUE@	  -DICEDTEA_WEB_JRE="\"$(SYSTEM_JRE_DIR)\"" \
 @ENABLE_PLUGIN_TRUE@	  -DPLUGIN_BOOTCLASSPATH=$(PLUGIN_BOOTCLASSPATH) \
+@ENABLE_PLUGIN_TRUE@	  $(VERSION_DEFS) \
 @ENABLE_PLUGIN_TRUE@	  $(GLIB_CFLAGS) \
 @ENABLE_PLUGIN_TRUE@	  $(GTK_CFLAGS) \
 @ENABLE_PLUGIN_TRUE@	  $(MOZILLA_CFLAGS) \
