--- icedtea-web-1.3.orig/Makefile.am
+++ icedtea-web-1.3/Makefile.am
@@ -275,7 +275,7 @@ $(PLUGIN_DIR)/%.o: $(PLUGIN_SRCDIR)/%.cc
 
 $(PLUGIN_DIR)/$(BUILT_PLUGIN_LIBRARY): $(addprefix $(PLUGIN_DIR)/,$(PLUGIN_OBJECTS))
 	cd $(PLUGIN_DIR) && \
-	$(CXX) $(CXXFLAGS) \
+	$(CXX) $(LDFLAGS) $(CXXFLAGS) \
 	  $(PLUGIN_OBJECTS) \
 	  $(GLIB_LIBS) \
 	  $(GTK_LIBS) \
--- icedtea-web-1.3.orig/Makefile.in
+++ icedtea-web-1.3/Makefile.in
@@ -796,7 +818,7 @@ uninstall-local:
 
 @ENABLE_PLUGIN_TRUE@$(PLUGIN_DIR)/$(BUILT_PLUGIN_LIBRARY): $(addprefix $(PLUGIN_DIR)/,$(PLUGIN_OBJECTS))
 @ENABLE_PLUGIN_TRUE@	cd $(PLUGIN_DIR) && \
-@ENABLE_PLUGIN_TRUE@	$(CXX) $(CXXFLAGS) \
+@ENABLE_PLUGIN_TRUE@	$(CXX) $(LDFLAGS) $(CXXFLAGS) \
 @ENABLE_PLUGIN_TRUE@	  $(PLUGIN_OBJECTS) \
 @ENABLE_PLUGIN_TRUE@	  $(GLIB_LIBS) \
 @ENABLE_PLUGIN_TRUE@	  $(GTK_LIBS) \
