Subject: install Debian specific Python modules
Author: Jan Dittberner <jandd@debian.org>
--- a/python/setup.py.in
+++ b/python/setup.py.in
@@ -42,7 +42,7 @@
     author_email="jan@dittberner.info",
     url="http://cracklib.sourceforge.net/",
     license="GPLv2+",
-    py_modules=['cracklib'],
+    py_modules=['cracklib', 'test_cracklib', 'crack'],
     ext_modules=extensions,
     zip_safe=False,
     classifiers=[
