Subject: add support for dictionary path to cracklib.test
Author: Jan Dittberner <jandd@debian.org>
--- a/python/cracklib.py
+++ b/python/cracklib.py
@@ -218,11 +218,11 @@
     return new
 
 
-def test(verbosity=1, repeat=1):
+def test(verbosity=1, repeat=1, dictpath=None):
     """Test cracklib methods."""
     import test_cracklib
     import sys
-    result = test_cracklib.run(verbosity=verbosity, repeat=repeat)
+    result = test_cracklib.run(verbosity=verbosity, repeat=repeat, use_dictpath=dictpath)
     if result.wasSuccessful():
         sys.exit(0)
     sys.exit(1)
