$Id: mimetypes.patch 379 2012-09-20 11:45:26Z tg $

--- a/includes/DefaultSettings.php
+++ b/includes/DefaultSettings.php
@@ -879,8 +879,8 @@ $wgAntivirusRequired = true;
 $wgVerifyMimeType = true;
 
 /** Sets the mime type definition file to use by MimeMagic.php. */
-$wgMimeTypeFile = "includes/mime.types";
-#$wgMimeTypeFile= "/etc/mime.types";
+#$wgMimeTypeFile= "includes/mime.types";
+$wgMimeTypeFile = "/etc/mime.types";
 #$wgMimeTypeFile= null; #use built-in defaults only.
 
 /** Sets the mime type info file to use by MimeMagic.php. */
