Index: xmonad-0.9.2/XMonad/Core.hs
===================================================================
--- xmonad-0.9.2.orig/XMonad/Core.hs	2011-04-05 21:39:02.000000000 +0530
+++ xmonad-0.9.2/XMonad/Core.hs	2011-04-05 21:39:13.000000000 +0530
@@ -1,7 +1,7 @@
 {-# LANGUAGE ExistentialQuantification, FlexibleInstances, GeneralizedNewtypeDeriving,
              MultiParamTypeClasses, TypeSynonymInstances, CPP, DeriveDataTypeable #-}
 -- required for deriving Typeable
-{- # OPTIONS_GHC -fglasgow-exts #-}
+{-# OPTIONS_GHC -fglasgow-exts #-}
 
 -----------------------------------------------------------------------------
 -- |
Index: xmonad-0.9.2/XMonad/Layout.hs
===================================================================
--- xmonad-0.9.2.orig/XMonad/Layout.hs	2011-04-05 21:39:03.000000000 +0530
+++ xmonad-0.9.2/XMonad/Layout.hs	2011-04-05 21:39:13.000000000 +0530
@@ -1,4 +1,4 @@
-{- # OPTIONS_GHC -fglasgow-exts    #-} -- For deriving Data/Typeable
+{-# OPTIONS_GHC -fglasgow-exts    #-} -- For deriving Data/Typeable
 {-# LANGUAGE FlexibleInstances, MultiParamTypeClasses, PatternGuards, TypeSynonymInstances, DeriveDataTypeable #-}
 
 -- --------------------------------------------------------------------------
Index: xmonad-0.9.2/XMonad/Operations.hs
===================================================================
--- xmonad-0.9.2.orig/XMonad/Operations.hs	2011-04-05 21:39:02.000000000 +0530
+++ xmonad-0.9.2/XMonad/Operations.hs	2011-04-05 21:41:11.000000000 +0530
@@ -1,7 +1,6 @@
-{-# OPTIONS_GHC -fno-warn-orphans #-}
+{-# OPTIONS_GHC -fglasgow-exts -fno-warn-orphans #-}
 {-# LANGUAGE FlexibleInstances, MultiParamTypeClasses, PatternGuards, TypeSynonymInstances, DeriveDataTypeable #-}
 
-{- # OPTIONS_GHC -fglasgow-exts    #-} -- For deriving Data/Typeable
 -- --------------------------------------------------------------------------
 -- |
 -- Module      :  XMonad.Operations
