Index: xmonad-0.9.1/XMonad/Config.hs
===================================================================
--- xmonad-0.9.1.orig/XMonad/Config.hs	2010-02-11 20:53:31.000000000 +0100
+++ xmonad-0.9.1/XMonad/Config.hs	2010-02-11 20:55:39.000000000 +0100
@@ -167,7 +167,7 @@
 -- | The preferred terminal program, which is used in a binding below and by
 -- certain contrib modules.
 terminal :: String
-terminal = "xterm"
+terminal = "x-terminal-emulator"
 
 -- | Whether focus follows the mouse pointer.
 focusFollowsMouse :: Bool
