Add support for $(DEBIAN_CFLAGS).

Index: man-1.6g/man2html/Makefile.in
===================================================================
--- man-1.6g.orig/man2html/Makefile.in	2010-12-31 21:28:46.000000000 +0100
+++ man-1.6g/man2html/Makefile.in	2011-01-09 11:18:08.000000000 +0100
@@ -1,11 +1,12 @@
-CC = @CC@
-CFLAGS += -Wall -Wstrict-prototypes -Wmissing-prototypes
+CFLAGS += -Wall -Wstrict-prototypes -Wmissing-prototypes $(DEBIAN_CFLAGS)
 OBJECTS = man2html.o cgibase.o abbrev.o strdefs.o
 EXEEXT = @EXEEXT@
 bindir = $(DESTDIR)$(PREFIX)/usr/bin
 mandir = $(DESTDIR)$(PREFIX)@mandir@
 vardir = $(DESTDIR)$(PREFIX)/var
 httpdir = $(DESTDIR)$(PREFIX)/home/httpd
+cgidir = $(DESTDIR)$(PREFIX)/usr/lib/cgi-bin/man
+sharedir = $(DESTDIR)$(PREFIX)/usr/share/man2html
 cgiowner = nobody
 cgigroup = nobody
 
@@ -29,10 +30,10 @@
 #   http://localhost/cgi-bin/man/man2html
 # to work.
 install-man-scripts:
-	mkdir -p $(httpdir)/cgi-bin/man
-	mkdir -p $(httpdir)/cgi-aux/man
-	install -m 755 scripts/cgi-bin/man/* $(httpdir)/cgi-bin/man
-	install -m 644 scripts/cgi-aux/man/* $(httpdir)/cgi-aux/man
+	mkdir -p $(cgidir)
+	mkdir -p $(sharedir)
+	install -m 755 scripts/cgi-bin/man/* $(cgidir)
+	install -m 644 scripts/cgi-aux/man/* $(sharedir)
 	install -d -o $(cgiowner) -g $(cgigroup) -m 775 $(vardir)/man2html
 # (aux was renamed to cgi-aux since aux causes problems under DOS)
 
