Remove bashism from hman

diff -ruN man-1.6c-old/man2html/hman.sh man-1.6c/man2html/hman.sh
--- man-1.6c-old/man2html/hman.sh	2003-03-11 16:53:29.000000000 +0100
+++ man-1.6c/man2html/hman.sh	2004-07-27 22:17:05.000000000 +0200
@@ -42,7 +42,7 @@
 fi
 
 # Interface to a live (already running) netscape browser.
-function nsfunc () {
+nsfunc () {
 	if ( /bin/ps xc | grep -q 'netscape$' ) ; then
 		if [ -x  netscape-remote ] ; then
 			exec netscape-remote  -remote "openURL($1,new_window)"
