Change paths to conform FHS and Debian Policy.

diff -ruN -x config.guess -x config.sub man-1.6e-old/man2html/cgibase.c man-1.6e/man2html/cgibase.c
--- man-1.6e-old/man2html/cgibase.c	2007-01-31 01:29:06.000000000 +0100
+++ man-1.6e/man2html/cgibase.c	2007-01-31 01:29:06.000000000 +0100
@@ -18,7 +18,7 @@
 /*
  * Either the user is non-local (or local, but using httpd),
  * in which case we use http:/cgi-bin, or the user is local
- * and uses lynx, and we use lynxcgi:/home/httpd/cgi-bin.
+ * and uses lynx, and we use lynxcgi:/usr/lib/cgi-bin.
  */
 
 static char *man2htmlpath = "/cgi-bin/man/man2html"; 	/* default */
diff -ruN -x config.guess -x config.sub man-1.6e-old/man2html/hman.sh man-1.6e/man2html/hman.sh
--- man-1.6e-old/man2html/hman.sh	2007-01-31 01:29:06.000000000 +0100
+++ man-1.6e/man2html/hman.sh	2007-01-31 01:29:06.000000000 +0100
@@ -57,7 +57,7 @@
 case "$HMAN_BROWSER" in
      lynxcgi)
 	HMAN_BROWSER=lynx
-	CG="lynxcgi:/home/httpd/cgi-bin/man"
+	CG="lynxcgi:/usr/lib/cgi-bin/man"
 	;;
      netscape)
         HMAN_BROWSER=nsfunc
diff -ruN -x config.guess -x config.sub man-1.6e-old/man2html/man2html.1 man-1.6e/man2html/man2html.1
--- man-1.6e-old/man2html/man2html.1	2007-01-31 01:29:06.000000000 +0100
+++ man-1.6e/man2html/man2html.1	2007-01-31 01:29:06.000000000 +0100
@@ -88,7 +88,7 @@
 .TP
 .B \-\^l
 Set method:cgipath to
-.RI lynxcgi: /home/httpd .
+.RI lynxcgi: /usr/lib .
 .TP
 .BI \-\^L " dir"
 Set method:cgipath to
@@ -115,7 +115,7 @@
 When some http daemon is running, lynx, or any other browser,
 can be used to browse the man pages, using the http method.
 The option \-l (for `lynxcgi') selects the former behaviour.
-With it, the default cgipath is \fI/home/httpd\fP.
+With it, the default cgipath is \fI/usr/lib\fP.
 
 In general, a cgi script can be called by
 .IP
diff -ruN -x config.guess -x config.sub man-1.6e-old/man2html/man2html.c man-1.6e/man2html/man2html.c
--- man-1.6e-old/man2html/man2html.c	2007-01-31 01:29:06.000000000 +0100
+++ man-1.6e/man2html/man2html.c	2007-01-31 01:29:06.000000000 +0100
@@ -3109,7 +3109,7 @@
 	 case 'H':
 	      set_cgibase(optarg); break;
 	 case 'l':
-	      set_lynxcgibase("/home/httpd"); break;
+	      set_lynxcgibase("/usr/lib"); break;
 	 case 'L':
 	      set_lynxcgibase(optarg); break;
 	 case 'M':
diff -ruN -x config.guess -x config.sub man-1.6e-old/man2html/README man-1.6e/man2html/README
--- man-1.6e-old/man2html/README	2007-01-31 01:29:06.000000000 +0100
+++ man-1.6e/man2html/README	2007-01-31 01:29:06.000000000 +0100
@@ -10,9 +10,9 @@
 The contents of FILENAME (or STDIN, in case FILENAME is "-" or absent)
 are converted from man-style nroff to html, and printed on STDOUT.
 
-With "-l" URLs of the form "lynxcgi:/home/httpd/cgi-bin/..." are generated.
+With "-l" URLs of the form "lynxcgi:/usr/lib/cgi-bin/..." are generated.
 With "-H host" we make URLs of the form "http://host/cgi-bin/...".
-The default is "http://localhost/cgi-bin/...".
+The default is "/cgi-bin/...".
 
 2. A collection of scripts
 
@@ -29,9 +29,9 @@
 2A. man stuff
 
 This first part (that can be done separately with "make install-man-scripts")
-puts various scripts under /home/httpd/cgi-bin and /home/httpd/cgi-aux
+puts various scripts under /usr/lib/cgi-bin and /usr/share/man2html
 in a subdirectory man.
-It will create a directory /var/man2html to hold the indices.
+It will create a directory /var/cache/man2html to hold the indices.
 (This directory should be writable by the cgi scripts;
 probably that means that the owner should be nobody.
 Choose a group and add all non-httpd users that should be
diff -ruN -x config.guess -x config.sub man-1.6e-old/man2html/scripts/cgi-aux/man/man.aux man-1.6e/man2html/scripts/cgi-aux/man/man.aux
--- man-1.6e-old/man2html/scripts/cgi-aux/man/man.aux	2006-06-19 07:19:27.000000000 +0200
+++ man-1.6e/man2html/scripts/cgi-aux/man/man.aux	2007-01-31 01:29:06.000000000 +0100
@@ -4,7 +4,7 @@
 <HEAD>
 <TITLE>Manual Pages - Main Contents</TITLE>
 <!-- Note: this is not html, but requires preprocessing -->
-<!-- It is used by /home/httpd/cgi-bin/man/man2html -->
+<!-- It is used by /usr/lib/cgi-bin/man/man2html -->
 </HEAD>
 <BODY>
 <H1>Manual Pages - Main Contents</H1>
diff -ruN -x config.guess -x config.sub man-1.6e-old/man2html/scripts/cgi-bin/man/man2html man-1.6e/man2html/scripts/cgi-bin/man/man2html
--- man-1.6e-old/man2html/scripts/cgi-bin/man/man2html	2005-08-21 01:26:06.000000000 +0200
+++ man-1.6e/man2html/scripts/cgi-bin/man/man2html	2007-01-31 01:29:06.000000000 +0100
@@ -1,10 +1,10 @@
 #!/bin/sh
 # man2html cgi script - uses /usr/bin/man2html to format man pages
-# auxiliary text files in /home/httpd/cgi-aux/man
+# auxiliary text files in /usr/share/man2html
 # aeb@cwi.nl - 980109
 
 MAN2HTML="/usr/bin/man2html"
-MANX="/home/httpd/cgi-aux/man/man.aux"
+MANX="/usr/share/man2html/man.aux"
 
 # Do we need lynxcgi URLs? For the moment our criterion is
 # 1) HTTP_USER_AGENT=Lynx*  and 2) HTTP_HOST is unset.
@@ -25,7 +25,7 @@
 case "$HH" in
     nohh)
 	LL="-l"
-	CG="lynxcgi:/home/httpd/cgi-bin/man"
+	CG="lynxcgi:/usr/lib/cgi-bin/man"
 	;;
     *)
 	LL="-H$SERVER"
diff -ruN -x config.guess -x config.sub man-1.6e-old/man2html/scripts/cgi-bin/man/mansearch man-1.6e/man2html/scripts/cgi-bin/man/mansearch
--- man-1.6e-old/man2html/scripts/cgi-bin/man/mansearch	2005-08-21 01:26:06.000000000 +0200
+++ man-1.6e/man2html/scripts/cgi-bin/man/mansearch	2007-01-31 01:29:06.000000000 +0100
@@ -23,7 +23,7 @@
 SERVER="${SERVER_NAME-localhost}"
 case "$HH" in
     nohh)
-	CG="lynxcgi:/home/httpd/cgi-bin/man"
+	CG="lynxcgi:/usr/lib/cgi-bin/man"
 	;;
     *)
 	CG="http://$SERVER/cgi-bin/man"
@@ -56,7 +56,7 @@
 
 BEGIN {
 
-  searchdocument = "/home/httpd/cgi-aux/man/mansearch.aux";
+  searchdocument = "/usr/share/man2html/mansearch.aux";
   quote = ENVIRON["QUOTE"];
   cgipath = ENVIRON["CG"];
   sedcmd = ENVIRON["SED"];
diff -ruN -x config.guess -x config.sub man-1.6e-old/man2html/scripts/cgi-bin/man/mansearchhelp man-1.6e/man2html/scripts/cgi-bin/man/mansearchhelp
--- man-1.6e-old/man2html/scripts/cgi-bin/man/mansearchhelp	2005-08-21 01:26:06.000000000 +0200
+++ man-1.6e/man2html/scripts/cgi-bin/man/mansearchhelp	2007-01-31 01:29:06.000000000 +0100
@@ -1,6 +1,6 @@
 #!/bin/sh
 MAN2HTML=/usr/bin/man2html
-MANSH=/home/httpd/cgi-aux/man/mansearchhelp.aux
+MANSH=/usr/share/man2html/mansearchhelp.aux
 
 # Do we need lynxcgi URLs? For the moment our criterion is
 # 1) HTTP_USER_AGENT=Lynx*  and 2) HTTP_HOST is unset.
@@ -17,7 +17,7 @@
 SERVER="${SERVER_NAME-localhost}"
 case "$HH" in
     nohh)
-	CG="lynxcgi:/home/httpd/cgi-bin/man"
+	CG="lynxcgi:/usr/lib/cgi-bin/man"
 	;;
     *)
 	CG="http://$SERVER/cgi-bin/man"
