file:/path --> file:///path 

diff -ruN -x config.guess -x config.sub man-1.6e-old/man2html/cgibase.c man-1.6e/man2html/cgibase.c
--- man-1.6e-old/man2html/cgibase.c	2007-01-31 01:29:53.000000000 +0100
+++ man-1.6e/man2html/cgibase.c	2007-01-31 01:29:54.000000000 +0100
@@ -97,7 +97,7 @@
 
 void
 include_file_html(char *g) {
-     printf("<A HREF=\"file:/usr/include/%s\">%s</A>&gt;", g,g);
+     printf("<A HREF=\"file:///usr/include/%s\">%s</A>&gt;", g,g);
 }
 
 void
diff -ruN -x config.guess -x config.sub man-1.6e-old/man2html/man2html.c man-1.6e/man2html/man2html.c
--- man-1.6e-old/man2html/man2html.c	2007-01-31 01:29:53.000000000 +0100
+++ man-1.6e/man2html/man2html.c	2007-01-31 01:29:54.000000000 +0100
@@ -118,11 +118,11 @@
     ** www.host.name           -> http://www.host.name
     ** ftp.host.name           -> ftp://ftp.host.name
     ** name@host               -> mailto:name@host
-    ** <name.h>                -> file:/usr/include/name.h   (guess)
+    ** <name.h>                -> file:///usr/include/name.h   (guess)
     **
     ** Other possible links to add in the future:
     **
-    ** /dir/dir/file  -> file:/dir/dir/file
+    ** /dir/dir/file  -> file:///dir/dir/file
     */
     int i,j,nr;
     char *f, *g, *h;
@@ -3271,7 +3271,7 @@
 		   "The requested file %s is not a valid (unformatted) "
 		   "man page.\nIf the file is a formatted man page, "
 		   "you could try to load the\n"
-		   "<A HREF=\"file://localhost%s\">plain file</A>.\n",
+		   "<A HREF=\"file://%s\">plain file</A>.\n",
 		   filename, filename);
 	else
 	     error_page(403, "Invalid Man Page",
