Use sensible-browser if lynx can't be found.

diff -ruN man-1.6c-old/man2html/hman.1 man-1.6c/man2html/hman.1
--- man-1.6c-old/man2html/hman.1	2003-03-11 16:53:30.000000000 +0100
+++ man-1.6c/man2html/hman.1	2004-07-27 22:21:22.000000000 +0200
@@ -39,7 +39,12 @@
 .B MANHTMLPAGER
 environment variable.
 The default is the non-httpd version of
-.BR lynx .
+.BR lynx ,
+or
+.BR sensible-browser
+if 
+.B lynx
+cannot be found.
 .TP
 .B \-\^H " host"
 Specify from what host to get the man pages.
@@ -62,6 +67,7 @@
 .BR man2html (1),
 .BR arena (1),
 .BR lynx (1),
+.BR sensible-browser (1),
 .BR netscape (1),
 .BR xmosaic (1),
 .BR glimpse (1)
diff -ruN man-1.6c-old/man2html/hman.sh man-1.6c/man2html/hman.sh
--- man-1.6c-old/man2html/hman.sh	2004-07-27 22:21:22.000000000 +0200
+++ man-1.6c/man2html/hman.sh	2004-07-27 22:21:22.000000000 +0200
@@ -24,7 +24,12 @@
 
 # The user has to set MANHTMLPAGER (or he will get httpd-free lynx).
 # Pick your favorite browser: lynx, xmosaic, netscape, arena, amaya, grail, ...
-HMAN_BROWSER=${MANHTMLPAGER-lynxcgi}
+if [ x"$MANHTMLPAGER" = x ]  && ! which lynx > /dev/null ; then
+	HMAN_BROWSER=sensible-browser
+else
+	HMAN_BROWSER=${MANHTMLPAGER-lynxcgi}
+fi
+
 #
 # If the man pages are on a remote host, specify it in MANHTMLHOST.
 HOST=${MANHTMLHOST-localhost}
