Add support for \N'n' groff sequences (closes: #348553).

diff -ruN man-1.6c-old/man2html/man2html.c man-1.6c/man2html/man2html.c
--- man-1.6c-old/man2html/man2html.c	2006-02-09 19:19:09.000000000 +0100
+++ man-1.6c/man2html/man2html.c	2006-02-09 19:17:53.000000000 +0100
@@ -511,7 +511,7 @@
 static char *
 scan_escape(char *c) {
     char *h=NULL;
-    char b[5];
+    char b[10];
     INTDEF *intd;
     int exoutputp,exskipescape;
     int i,j;
@@ -661,6 +661,20 @@
 	output_possible=exoutputp;
 	skip_escape=exskipescape;
 	break;
+    case 'N':	
+	/* convert \N'ddd' into &#ddd; */
+	c++;
+	i=*c;
+	j=0;
+	b[j++] = '&';
+	b[j++] = '#';
+	while (*(++c) != i)
+	    if (isdigit(*c) && j < sizeof(b) - 2)
+		    b[j++] = *c;
+	b[j++] = ';';
+	b[j] = '\0';
+	h = b;
+	break;
     case 'c': no_newline_output=1; break;
     case '{': newline_for_fun++; h="";break;
     case '}': if (newline_for_fun) newline_for_fun--; h="";break;
