Index: gtk+-3.4.1/gtk/gtksettings.c
===================================================================
--- gtk+-3.4.1.orig/gtk/gtksettings.c	2012-04-04 03:06:04.000000000 +0200
+++ gtk+-3.4.1/gtk/gtksettings.c	2012-04-15 20:55:20.822307902 +0200
@@ -429,7 +429,7 @@
                                              g_param_spec_string ("gtk-fallback-icon-theme",
                                                                   P_("Fallback Icon Theme Name"),
                                                                   P_("Name of a icon theme to fall back to"),
-                                                                  NULL,
+                                                                  "gnome",
                                                                   GTK_PARAM_READWRITE),
                                              NULL);
   g_assert (result == PROP_FALLBACK_ICON_THEME);
