Author: Daniel Baumann <daniel.baumann@progress-technologies.net>
Description: Exclude debian directory from subdirs.

diff -Naurp traceroute.orig/Make.rules traceroute/Make.rules
--- traceroute.orig/Make.rules	2008-04-18 13:22:05.000000000 +0200
+++ traceroute/Make.rules	2008-05-03 09:18:15.000000000 +0200
@@ -72,7 +72,7 @@ DESTDIR =
 
 
 #  layout stuff
-SKIPDIRS = tmp%
+SKIPDIRS = tmp% debian
 INCLUDEDIRS = include%
 LIBDIRS = lib%
 MODDIRS = mod%
