Author: Daniel Baumann <daniel.baumann@progress-technologies.net>
Description: Adjust path.

diff -Naurp traceroute.orig/wrappers/lft traceroute/wrappers/lft
--- traceroute.orig/wrappers/lft	2008-04-21 18:54:41.000000000 +0200
+++ traceroute/wrappers/lft	2008-05-03 09:18:47.000000000 +0200
@@ -113,5 +113,5 @@ opts="$opts -w $sec.${timeout#$sec}"
 
 opts="$method $opts"
 
-exec traceroute $opts $host $length
+exec /usr/bin/traceroute.db $opts $host $length
 
diff -Naurp traceroute.orig/wrappers/tcptraceroute traceroute/wrappers/tcptraceroute
--- traceroute.orig/wrappers/tcptraceroute	2008-04-21 18:54:46.000000000 +0200
+++ traceroute/wrappers/tcptraceroute	2008-05-03 09:18:47.000000000 +0200
@@ -68,5 +68,5 @@ shift
     shift
 }
 
-exec traceroute $opts $host $length
+exec /usr/bin/traceroute.db $opts $host $length
 
diff -Naurp traceroute.orig/wrappers/traceproto traceroute/wrappers/traceproto
--- traceroute.orig/wrappers/traceproto	2008-04-21 18:55:06.000000000 +0200
+++ traceroute/wrappers/traceproto	2008-05-03 09:18:47.000000000 +0200
@@ -122,6 +122,6 @@ opts="$opts -z $sendwait"
 while [ "$cont" -gt 0 ]
 do
     cont=$(($cont - 1))
-    traceroute $opts $host $length
+    /usr/bin/traceroute.db $opts $host $length
 done
 
diff -Naurp traceroute.orig/wrappers/traceroute-nanog traceroute/wrappers/traceroute-nanog
--- traceroute.orig/wrappers/traceroute-nanog	2008-04-21 18:55:17.000000000 +0200
+++ traceroute/wrappers/traceroute-nanog	2008-05-03 09:18:47.000000000 +0200
@@ -71,5 +71,5 @@ done
 
 [ -z "$spray" ] && opts="$opts -N 1"
 
-exec traceroute $opts $1 $2
+exec /usr/bin/traceroute.db $opts $1 $2
 
