Author: Daniel Baumann <daniel.baumann@progress-technologies.net>
Description: Avoid overwriting build environment.

diff -Naurp traceroute.orig/Make.rules traceroute/Make.rules
--- traceroute.orig/Make.rules	2012-04-26 09:43:18.442365190 +0200
+++ traceroute/Make.rules	2012-05-18 12:39:37.970036900 +0200
@@ -46,9 +46,9 @@ $(call set, INSTALL, cp)
 $(call set, INDENT, true)
 
 gcc = $(findstring gcc,$(CC))
-$(call set, CFLAGS, $(if $(gcc), -O2 -Wall, -O))
-$(call set, CPPFLAGS, )
-$(call set, LDFLAGS, -s)
+#$(call set, CFLAGS, $(if $(gcc), -O2 -Wall, -O))
+#$(call set, CPPFLAGS, )
+#$(call set, LDFLAGS, -s)
 $(call set, LIBS, )
 
 
