--- a/nls/compose.dir.pre
+++ b/nls/compose.dir.pre
@@ -487,7 +487,9 @@ en_US.UTF-8/Compose:		sv_SE.UTF-8
 en_US.UTF-8/Compose:		ta_IN.UTF-8
 en_US.UTF-8/Compose:		te_IN.UTF-8
 en_US.UTF-8/Compose:		tg_TJ.UTF-8
-en_US.UTF-8/Compose:		th_TH.UTF-8
+XCOMM Following entry commented to allow Thai XIM as default for X apps
+XCOMM See Debian bug#520509
+XCOMM en_US.UTF-8/Compose:		th_TH.UTF-8
 en_US.UTF-8/Compose:       ti_ER.UTF-8
 en_US.UTF-8/Compose:       ti_ET.UTF-8
 en_US.UTF-8/Compose:       tig_ER.UTF-8
