Index: tango-icon-theme-0.8.90/index.theme.in
===================================================================
--- tango-icon-theme-0.8.90.orig/index.theme.in	2006-06-07 15:19:56.000000000 +0200
+++ tango-icon-theme-0.8.90/index.theme.in	2009-03-08 17:41:43.000000000 +0100
@@ -457,10 +457,10 @@
 MaxSize=256
 
 [scalable/emblems]
-Size=48
+Size=96
 Context=Emblems
 Type=Scalable
-MinSize=32
+MinSize=1
 MaxSize=256
 
 [scalable/emotes]
