From 3822e9ecd0783a5743dd48cda86ae17ba702c468 Mon Sep 17 00:00:00 2001
From: Abramo Bagnara <abramo.bagnara@bugseng.com>
Date: Wed, 30 Jan 2013 09:20:06 +0100
Subject: [PATCH] Added missing line.

---
 src/Float_defs.hh |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

--- a/src/Float.defs.hh
+++ b/src/Float.defs.hh
@@ -261,6 +261,7 @@
   static const int EXPONENT_MIN = -EXPONENT_MAX + 1;
   static const int EXPONENT_MIN_DENORM = EXPONENT_MIN
 					- static_cast<int>(MANTISSA_BITS);
+  static const Floating_Point_Format floating_point_format = IEEE754_QUAD;
   int inf_sign() const;
   bool is_nan() const;
   int zero_sign() const;
