Index: gnome-tweak-tool-3.2.1/data/gnome-tweak-tool.desktop.in
===================================================================
--- gnome-tweak-tool-3.2.1.orig/data/gnome-tweak-tool.desktop.in	2011-08-06 01:24:18.000000000 +0200
+++ gnome-tweak-tool-3.2.1/data/gnome-tweak-tool.desktop.in	2011-11-07 17:19:14.847146550 +0100
@@ -6,5 +6,5 @@
 Terminal=false
 Type=Application
 StartupNotify=true
-Categories=GNOME;GTK;Utility;
+Categories=GNOME;GTK;Settings;
 OnlyShowIn=GNOME;Unity;
