From 7dc53cc3b792775369bf0b2f053a3f4ed5d87e3d Mon Sep 17 00:00:00 2001
From: Vasiliy Kulikov <segoon@openwall.com>
Date: Tue, 4 Oct 2011 18:12:26 +0400
Subject: [PATCH] Really exit on privsep init failure.

---
 radvd.c |    4 +++-
 1 files changed, 3 insertions(+), 1 deletions(-)

diff --git a/radvd.c b/radvd.c
index 3962d24..109b83a 100644
--- a/radvd.c
+++ b/radvd.c
@@ -270,8 +270,10 @@ main(int argc, char *argv[])
 	if (username) {
 		if (!singleprocess) {
 		 	dlog(LOG_DEBUG, 3, "Initializing privsep");
-		 	if (privsep_init() < 0)
+			if (privsep_init() < 0) {
 				perror("Failed to initialize privsep.");
+				exit(1);
+			}
 		}
 
 		if (drop_root_privileges(username) < 0) {
-- 
1.7.6.3

