From 7de1b9abf87b747ee2611c0d2a94dfeee79878b4 Mon Sep 17 00:00:00 2001
From: Vasiliy Kulikov <segoon@openwall.com>
Date: Tue, 4 Oct 2011 00:35:57 -0700
Subject: [PATCH] process_ra() has numerous missed len checks.  It leads to
 buffer overreads.

---
 process.c |   11 +++++++++++
 1 files changed, 11 insertions(+), 0 deletions(-)

diff --git a/process.c b/process.c
index 3307daa..f30d147 100644
--- a/process.c
+++ b/process.c
@@ -310,6 +310,8 @@ process_ra(struct Interface *iface, unsigned char *msg, int len,
 		{
 		case ND_OPT_MTU:
 			mtu = (struct nd_opt_mtu *)opt_str;
+			if (len < sizeof(*mtu))
+				return;
 
 			if (iface->AdvLinkMTU && (ntohl(mtu->nd_opt_mtu_mtu) != iface->AdvLinkMTU))
 			{
@@ -319,6 +321,8 @@ process_ra(struct Interface *iface, unsigned char *msg, int len,
 			break;
 		case ND_OPT_PREFIX_INFORMATION:
 			pinfo = (struct nd_opt_prefix_info *) opt_str;
+			if (len < sizeof(*pinfo))
+				return;
 			preferred = ntohl(pinfo->nd_opt_pi_preferred_time);
 			valid = ntohl(pinfo->nd_opt_pi_valid_time);
 
@@ -373,6 +377,8 @@ process_ra(struct Interface *iface, unsigned char *msg, int len,
 			break;
 		case ND_OPT_RDNSS_INFORMATION:
 			rdnssinfo = (struct nd_opt_rdnss_info_local *) opt_str;
+			if (len < sizeof(*rdnssinfo))
+				return;
 			count = rdnssinfo->nd_opt_rdnssi_len;
 
 			/* Check the RNDSS addresses received */
-- 
1.7.6.3

