From 2591d0189257caeaae2057dfed0a260310497a61 Mon Sep 17 00:00:00 2001
From: Reuben Hawkins <reubenhwk@gmail.com>
Date: Tue, 4 Oct 2011 13:18:47 -0700
Subject: [PATCH] removing mdelay in unicast only case

---
 process.c |    1 -
 1 files changed, 0 insertions(+), 1 deletions(-)

--- a/process.c
+++ b/process.c
@@ -193,7 +193,8 @@
 	delay = MAX_RA_DELAY_TIME * rand() / (RAND_MAX +1.0);
 
 	if (iface->UnicastOnly) {
-		send_ra_forall(iface, &addr->sin6_addr);
+	        mdelay(delay);
+	        send_ra_forall(iface, &addr->sin6_addr);
 	}
 	else if ( timevaldiff(&tv, &iface->last_multicast) / 1000.0 < iface->MinDelayBetweenRAs ) {
 		/* last RA was sent only a few moments ago, don't send another immediately. */
