From 7a1471b62da88373e8f4209d503307c5d841b81f Mon Sep 17 00:00:00 2001
From: Vasiliy Kulikov <segoon@openwall.com>
Date: Sat, 8 Oct 2011 09:26:58 -0700
Subject: [PATCH] checking iface name more carefully

modified:   device-linux.c
---
 device-linux.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

--- a/device-linux.c
+++ b/device-linux.c
@@ -247,7 +247,7 @@
 		return -1;
 
 	/* No path traversal */
-	if (strstr(name, "..") || strchr(name, '/'))
+	if (!iface[0] || !strcmp(iface, ".") || !strcmp(iface, "..") || strchr(iface, '/'))
 		return -1;
 
 	if (access(spath, F_OK) != 0)
