--- a/configure.ac
+++ b/configure.ac
@@ -26,6 +26,12 @@
 		AC_DEFINE([_GNU_SOURCE], [], [whether compiling on Linux, glibc>=2.8 doesn't expose in6_pktinfo otherwise..])
                 arch=linux
         ;;
+        *kfreebsd*gnu*)
+                AC_MSG_RESULT(kfreebsd-gnu)
+                AC_DEFINE([_GNU_SOURCE], [], [glibc>=2.8 doesn't expose in6_pktinfo otherwise..])
+                arch=bsd44
+        ;;
+
         *bsd*)
 		AC_MSG_RESULT(bsd44)
                 arch=bsd44
