Description: Adds missing headers to build with the GCC 4.3.
Origin: vendor : Debian
Forwarded: https://sourceforge.net/support/tracker.php?aid=2013397
--- srf-0.1.orig/src/SRF_util.hh
+++ srf-0.1/src/SRF_util.hh
@@ -32,6 +32,7 @@
 #include <iostream>
 #include <fstream>
 #include <string>
+#include <cstring>
 #include <vector>
 #include <io_lib/ztr.h>
 #include <io_lib/Read.h>
--- srf-0.1.orig/src/SRF_Ztr.cc
+++ srf-0.1/src/SRF_Ztr.cc
@@ -13,6 +13,8 @@
 //   limitations under the License.
 //
 //
+#include <cstdlib>
+#include <cstring>
 #include <ios>
 #include <iostream>
 #include <fstream>
--- srf-0.1.orig/test/testWrite.cc
+++ srf-0.1/test/testWrite.cc
@@ -1,4 +1,5 @@
 //
+#include <cstdlib>
 #include <ios>
 #include <iostream>
 #include <fstream>
