Index: qmf-1.0.7~2011w23.2/src/tools/messageserver/messageserver.cpp
===================================================================
--- qmf-1.0.7~2011w23.2.orig/src/tools/messageserver/messageserver.cpp	2011-06-15 08:53:17.000000000 +0000
+++ qmf-1.0.7~2011w23.2/src/tools/messageserver/messageserver.cpp	2012-05-23 23:13:20.539868874 +0000
@@ -57,6 +57,7 @@
 #include <sys/socket.h>
 #endif
 #include <signal.h>
+#include <unistd.h>
 }
 
 #if defined(Q_OS_UNIX)
