From: Mehdi Dogguy <mehdi@debian.org>
Date: Fri, 8 Oct 2010 22:03:07 +0200
Subject: Fix META file for bitstring.syntax

Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=599552
Signed-off-by: Mehdi Dogguy <mehdi@debian.org>
---
 META.in |    5 +++--
 1 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/META.in b/META.in
index 6e99669..699ba44 100644
--- a/META.in
+++ b/META.in
@@ -16,7 +16,8 @@ package "persistent" (
 package "syntax" (
   version = "@PACKAGE_VERSION@"
   description = "Syntax extension: bitstring operators"
-  archive(syntax,preprocessor) = "unix.cma bitstring.cma bitstring_persistent.cma pa_bitstring.cmo"
-  archive(syntax,toploop) = "unix.cma bitstring.cma bitstring_persistent.cma pa_bitstring.cmo"
+  requires = "bitstring.persistent"
+  archive(syntax,preprocessor) = "pa_bitstring.cmo"
+  archive(syntax,toploop) = "pa_bitstring.cmo"
   preprocessor = "camlp4of"
 )
-- 
