From: Nicolas Dandrimont <Nicolas.Dandrimont@crans.org>
Date: Mon, 18 Apr 2011 16:17:36 +0200
Subject: Fix preprocessor options for the syntax META file

Origin: upstream, http://code.google.com/p/bitstring/source/detail?r=185
Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=623230
Signed-off-by: Nicolas Dandrimont <Nicolas.Dandrimont@crans.org>
Signed-off-by: Stephane Glondu <steph@glondu.net>
---
 META.in |    3 +--
 1 files changed, 1 insertions(+), 2 deletions(-)

diff --git a/META.in b/META.in
index 699ba44..049bbb4 100644
--- a/META.in
+++ b/META.in
@@ -16,8 +16,7 @@ package "persistent" (
 package "syntax" (
   version = "@PACKAGE_VERSION@"
   description = "Syntax extension: bitstring operators"
-  requires = "bitstring.persistent"
+  requires = "bitstring.persistent, camlp4"
   archive(syntax,preprocessor) = "pa_bitstring.cmo"
   archive(syntax,toploop) = "pa_bitstring.cmo"
-  preprocessor = "camlp4of"
 )
-- 
