--- a/spectro/usbio.c
+++ b/spectro/usbio.c
@@ -203,10 +203,6 @@ struct _icoms *p
 #ifdef ENABLE_USB
 	struct usb_bus *bus;
 
-	/* Check that we've got an up to date version of libusb */
-	if (usb_argyll_patched() < 2)
-		error("usblib isn't up to date to work with this version of Argyll");
-
 	if (p->debug > 8)
 		usb_set_debug(p->debug);
 
@@ -738,7 +734,7 @@ char **pnames		/* List of process names
 			if (libusb_get_device_descriptor(p->ppath->dev, &descriptor) != LIBUSB_SUCCESS)
 				error("Get device descriptor on USB port '%s' failed with %d (%s)",p->ppath->path,rv,libusb_strerror(rv));
 #else
-			descriptor = dev->descriptor;	/* Copy */
+			descriptor = p->ppath->dev->descriptor;	/* Copy */
 #endif
 
 			p->vid = p->ppath->vid;
