Index: gnome-bluetooth-3.4.1/applet/Makefile.am
===================================================================
--- gnome-bluetooth-3.4.1.orig/applet/Makefile.am	2011-12-16 17:19:18.000000000 +0100
+++ gnome-bluetooth-3.4.1/applet/Makefile.am	2012-06-13 14:41:04.172029773 +0200
@@ -23,7 +23,7 @@
 ui_DATA = popup-menu.ui authorisation-dialogue.ui confirm-dialogue.ui passkey-dialogue.ui
 uidir = $(pkgdatadir)
 
-autostartdir = $(sysconfdir)/xdg/autostart
+autostartdir = $(datadir)/gnome/autostart
 
 autostart_in_in_files = bluetooth-applet.desktop.in.in
 autostart_in_files = bluetooth-applet.desktop.in
Index: gnome-bluetooth-3.4.1/applet/Makefile.in
===================================================================
--- gnome-bluetooth-3.4.1.orig/applet/Makefile.in	2012-06-12 19:12:23.000000000 +0200
+++ gnome-bluetooth-3.4.1/applet/Makefile.in	2012-06-13 14:41:04.172029773 +0200
@@ -380,7 +380,7 @@
 man_MANS = bluetooth-applet.1
 ui_DATA = popup-menu.ui authorisation-dialogue.ui confirm-dialogue.ui passkey-dialogue.ui
 uidir = $(pkgdatadir)
-autostartdir = $(sysconfdir)/xdg/autostart
+autostartdir = $(datadir)/gnome/autostart
 autostart_in_in_files = bluetooth-applet.desktop.in.in
 autostart_in_files = bluetooth-applet.desktop.in
 autostart_DATA = $(autostart_in_files:.desktop.in=.desktop)
