From 09e35c8990457c834cfd079e11ffdc31058c09d2 Mon Sep 17 00:00:00 2001
From: Yann Dirson <ydirson@free.fr>
Date: Mon, 22 Aug 2011 23:49:05 +0200
Subject: [PATCH] Comment out INSTALL_RPATH flag on executables.

---
 software/tulip/src/CMakeLists.txt |    3 ---
 1 file changed, 3 deletions(-)

diff --git a/software/tulip/src/CMakeLists.txt b/software/tulip/src/CMakeLists.txt
index f24ff9c..f9f08f6 100755
--- a/software/tulip/src/CMakeLists.txt
+++ b/software/tulip/src/CMakeLists.txt
@@ -40,7 +40,6 @@ SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DQT_MINOR_REL=${QT_VERSION_MINOR}")
 
 IF(WIN32)
   ADD_EXECUTABLE( tulip TulipLauncher.cpp)
-  SET_TARGET_PROPERTIES(tulip PROPERTIES INSTALL_RPATH ${TulipLibInstallDir})
   IF(NOT MSVC)
     TARGET_LINK_LIBRARIES( tulip -mwindows )
   ENDIF()
@@ -48,11 +47,9 @@ IF(WIN32)
 ENDIF(WIN32)
 
 ADD_EXECUTABLE( tulip_need_restart TulipNeedRestart.cpp)
-SET_TARGET_PROPERTIES(tulip_need_restart PROPERTIES INSTALL_RPATH ${TulipLibInstallDir})
 TARGET_LINK_LIBRARIES( tulip_need_restart ${QT_LIBRARIES} ${LibTulipName} ${LibTulipOglName} ${LibTulipQtName})
 
 ADD_EXECUTABLE( tulip_app ${tulip_UI_SRCS} ${tulip_MOC_SRCS} ${tulip_SRCS} ${tulip_RCC_SRCS})
-SET_TARGET_PROPERTIES(tulip_app PROPERTIES INSTALL_RPATH ${TulipLibInstallDir})
 IF(WIN32 AND NOT ${CMAKE_GENERATOR} MATCHES "Visual Studio" AND NOT ${CMAKE_GENERATOR} MATCHES "NMake Makefiles")
   TARGET_LINK_LIBRARIES( tulip_app -mwindows )
 ENDIF()
-- 
1.7.10

