From 8bc532682f3cbd812c31f53af1f69829f5018fda Mon Sep 17 00:00:00 2001
From: Yann Dirson <ydirson@free.fr>
Date: Tue, 23 Aug 2011 23:49:47 +0200
Subject: [PATCH] Use Debian-packaged jars.

---
 CMakeLists.txt |    9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7650549..edfbb89 100755
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -106,9 +106,12 @@ SET(UBUNTU_PPA_BUILD OFF CACHE INTERNAL "")
 ## -----------------------------------------------------------------------------------------------
 ## Jar files for doc
 ## -----------------------------------------------------------------------------------------------
-SET(SAXON_JAR "${PROJECT_SOURCE_DIR}/thirdparty/saxon/saxon.jar")
-SET(SAXON653_JAR "${PROJECT_SOURCE_DIR}/thirdparty/saxon/saxon653.jar")
-SET(RESOLVER_JAR "${PROJECT_SOURCE_DIR}/thirdparty/xml-commons/resolver.jar")
+#SET(SAXON_JAR "${PROJECT_SOURCE_DIR}/thirdparty/saxon/saxon.jar")
+#SET(SAXON653_JAR "${PROJECT_SOURCE_DIR}/thirdparty/saxon/saxon653.jar")
+#SET(RESOLVER_JAR "${PROJECT_SOURCE_DIR}/thirdparty/xml-commons/resolver.jar")
+SET(SAXON_JAR "/usr/share/java/saxon.jar")
+SET(SAXON653_JAR "/usr/share/java/saxon.jar")
+SET(RESOLVER_JAR "/usr/share/java/xml-resolver.jar")
 
 ## -----------------------------------------------------------------------------------------------
 ## Find need packages
-- 
1.7.10

