From 493fa2f7bedb367b4fa4e0e2b83a3ce1ddb9c0a9 Mon Sep 17 00:00:00 2001
From: Yann Dirson <ydirson@free.fr>
Date: Thu, 10 May 2012 23:59:09 +0200
Subject: [PATCH] Use bash for script with bashism

---
 tulip-config.in |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tulip-config.in b/tulip-config.in
index 99bf613..7062443 100755
--- a/tulip-config.in
+++ b/tulip-config.in
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/bin/bash
 
 thisdir=$(dirname "$0")
 if test "$thisdir" = "." ; then
-- 
1.7.10

