From: "Barak A. Pearlmutter" <barak+git@cs.nuim.ie>
Date: Thu, 25 Oct 2012 23:22:21 +0100
Subject: spelling: expresion

---
 plug-ins/sslstrip/sslstrip.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/plug-ins/sslstrip/sslstrip.c b/plug-ins/sslstrip/sslstrip.c
index 5394cec..59ec4da 100644
--- a/plug-ins/sslstrip/sslstrip.c
+++ b/plug-ins/sslstrip/sslstrip.c
@@ -1012,7 +1012,7 @@ static void http_remove_https(struct http_connection *connection)
 
 		if (regcomp(find_https_re, URL_PATTERN, REG_EXTENDED | REG_NEWLINE | REG_ICASE))
 		{
-			ERROR_MSG("SSLStrip: Error compiling regular expresion\n");
+			ERROR_MSG("SSLStrip: Error compiling regular expression\n");
 			return;
 		}
 	}
