Index: b/doc/build
===================================================================
--- a/doc/build	2010-01-31 18:29:47.000000000 +0000
+++ b/doc/build	2010-01-31 18:32:01.000000000 +0000
@@ -1,4 +1,4 @@
-#!/usr/local/bin/perl5
+#!/usr/bin/perl
 #
 # >>Title::     Build Documentation
 #
Index: b/perllib/sdf/home/sdf.ini
===================================================================
--- a/perllib/sdf/home/sdf.ini	2010-01-31 18:29:47.000000000 +0000
+++ b/perllib/sdf/home/sdf.ini	2010-01-31 18:32:01.000000000 +0000
@@ -63,9 +63,9 @@
 mif     =mif_
 pdf     =pdf_mif
 pod     =pod_
-ps      =ps_fm
+ps      =ps_sgml
 raw     =sdf_raw
-rtf     =rtf_fm
+rtf     =rtf_sgml
 sgml    =sgml_
 topics  =html_topics
 txt     =txt_
@@ -90,7 +90,8 @@
 #
 [Conversions]
 Context To      From    Action
-html    gif     eps     ps2gif -crop $source
+html    png     eps     pstoimg $source
+#html    gif     eps     ps2gif -crop $source
 html    gif     ps      ps2gif -crop $source
 html    gif     prn     prn2ps -ops $source; ps2gif -crop $base.ps; rm $base.ps
 ps      eps     ps      ps2epsi $source $dest
@@ -185,7 +186,7 @@
 [PostProcessing]
 update = &SdfBatch("-u"); &SdfBookClean("doc.backup")
 ps     = &SdfDelete("$long.$out_ext.ps"); &SdfBatch("-pfile -PA4")
-pdf    = &SdfSystem("distill -quiet $long.ps")
+pdf    = &SdfSystem("ps2pdf -quiet $long.ps")
 clean  = &SdfDelete("$long.$out_ext"); &SdfBookClean("doc")
 doc    = &SdfBatch("-sdoc -Sd")
 fvo    = &SdfBatch("-sfvo -Sl")
