Index: b/perllib/sdf/topod.pl
===================================================================
--- a/perllib/sdf/topod.pl	2010-01-31 18:29:47.000000000 +0000
+++ b/perllib/sdf/topod.pl	2010-01-31 18:32:01.000000000 +0000
@@ -450,7 +450,21 @@
 
     # For headings, map the tag to a command
     if ($tag =~ /^head/) {
-        return "=$tag $text\n";
+        if ($SDF_USER'var{'MAN_CONVENTIONS'}) {
+            if ($tag =~ /^head1/ && !$seen_name) {
+                $seen_name = 1;
+                return "=$tag NAME\n\n$text\n";
+            }
+            elsif ($tag =~ /^head2/) {
+                return "=head1 \U$text\E\n";
+            }
+            else {
+                return "=$tag $text\n";
+            }
+        }
+        else {
+            return "=$tag $text\n";
+        }
     }
 
     # Otherwise, format as a plain paragraph
