Index: b/perllib/sdf/table.pl
===================================================================
--- a/perllib/sdf/table.pl	2010-01-31 18:29:47.000000000 +0000
+++ b/perllib/sdf/table.pl	2010-01-31 18:32:01.000000000 +0000
@@ -915,7 +915,7 @@
     local($packfmt);
 
     $packfmt = '';
-    while ($format =~ s/\w+\s+//e) {
+    while ($format =~ s/\w+\s+//) {
         $packfmt .= 'A' . length($&);
     }
     $packfmt .= 'A*';
