Description: Skip some Debian packaging files in MANIFEST check
Author: Colin Watson <cjwatson@debian.org>
Forwarded: not-needed
Last-Update: 2011-05-30

Index: b/MANIFEST.SKIP
===================================================================
--- a/MANIFEST.SKIP
+++ b/MANIFEST.SKIP
@@ -7,3 +7,9 @@
 pm_to_blib$
 ^sdf.prj$
 ^\.sdf.prcs_aux$
+
+# Added for Debian packaging
+^\.bzr/
+^\.bzrignore
+^\.pc/
+^debian/
