Description: Use POSIX::ctime instead of Perl 4's ctime.pl
Author: Colin Watson <cjwatson@debian.org>
Forwarded: no
Last-Update: 2011-11-09

Index: b/bin/mif2rtf
===================================================================
--- a/bin/mif2rtf
+++ b/bin/mif2rtf
@@ -223,7 +223,7 @@
 #
 #------------------------------------------------------------------------------
 
-require 'ctime.pl';	# for date
+use POSIX qw(ctime);
 
 $winhelp=1; 	# this controls the handling of certain constructs that
 		# are *naughty* in the context of windows help files
Index: b/perllib/sdf/tests.pl
===================================================================
--- a/perllib/sdf/tests.pl
+++ b/perllib/sdf/tests.pl
@@ -37,7 +37,7 @@
 # * checking that additional files were not generated.
 #
 
-!require "ctime.pl";
+use POSIX qw(ctime);
 
 ######### Constants #########
 
