From: Ansgar Burchardt
Date: Sun, 11 Apr 2010 17:53:55 +0900
Bug-Debian: http://bugs.debian.org/577288
Subject: Do not link libz
Forwarded: https://rt.cpan.org/Ticket/Display.html?id=56479
--- libfile-libmagic-perl.orig/Makefile.PL
+++ libfile-libmagic-perl/Makefile.PL
@@ -15,7 +15,7 @@
     'PREREQ_PM'		=> { },
     'ABSTRACT_FROM'     => 'LibMagic.pm', # retrieve abstract from module
     'AUTHOR'            => 'Andreas Fitzner <andreas.fitzner@fv-berlin.de>',
-    'LIBS'		=> ['-lmagic -lz'], # e.g., '-lm'
+    'LIBS'		=> ['-lmagic'], # e.g., '-lm'
     'DEFINE'		=> '', # e.g., '-DHAVE_SOMETHING'
     'INC'		=> '-I.', # e.g., '-I. -I/usr/include/other'
 	# Un-comment this if you add C files to link with later:
