Description: adjust for magic 5.10
Origin: vendor
Bug: https://rt.cpan.org/Ticket/Display.html?id=75457
Bug-Debian: http://bugs.debian.org/661800
Forwarded: https://rt.cpan.org/Ticket/Display.html?id=75457
Author: gregor herrmann <gregoa@debian.org>
Last-Update: 2012-03-01

--- a/t/2-easy.t
+++ b/t/2-easy.t
@@ -7,7 +7,7 @@
 is( MagicBuffer("Hello World\n"),   'ASCII text'           );
 
 is( MagicFile('t/samples/foo.txt'), 'ASCII text'           );
-is( MagicFile('t/samples/foo.c'  ), 'ASCII C program text' );
+is( MagicFile('t/samples/foo.c'  ), 'C source, ASCII text' );
 
 # check the error handling
 eval { MagicBuffer(undef) };
--- a/t/3-complete.t
+++ b/t/3-complete.t
@@ -32,7 +32,7 @@
     is( magic_buffer( $handle, "Footastic\n" ), 'A foo file' );
 
     is( magic_file( $handle, 't/samples/foo.txt' ), 'ASCII text'           );
-    is( magic_file( $handle, 't/samples/foo.c'   ), 'ASCII C program text' );
+    is( magic_file( $handle, 't/samples/foo.c'   ), 'ASCII text' );
     is( magic_file( $handle, 't/samples/foo.foo' ), 'A foo file' );
 
     magic_close($handle);
@@ -45,7 +45,7 @@
     is( magic_buffer( $handle, "Hello World\n" ), 'ASCII text' );
 
     is( magic_file( $handle, 't/samples/foo.txt' ), 'ASCII text'           );
-    is( magic_file( $handle, 't/samples/foo.c'   ), 'ASCII C program text' );
+    is( magic_file( $handle, 't/samples/foo.c'   ), 'C source, ASCII text' );
     is( magic_file( $handle, 't/samples/foo.foo' ), 'ASCII text' );
 
     magic_close($handle);
@@ -58,7 +58,7 @@
     is( magic_buffer( $handle, "Hello World\n" ), 'ASCII text' );
 
     is( magic_file( $handle, 't/samples/foo.txt' ), 'ASCII text'           );
-    is( magic_file( $handle, 't/samples/foo.c'   ), 'ASCII C program text' );
+    is( magic_file( $handle, 't/samples/foo.c'   ), 'C source, ASCII text' );
     is( magic_file( $handle, 't/samples/foo.foo' ), 'ASCII text' );
 
     magic_close($handle);
--- a/t/4-all.t
+++ b/t/4-all.t
@@ -8,7 +8,7 @@
 is( MagicBuffer("Hello World\n"),   'ASCII text'           );
 
 is( MagicFile('t/samples/foo.txt'), 'ASCII text'           );
-is( MagicFile('t/samples/foo.c'  ), 'ASCII C program text' );
+is( MagicFile('t/samples/foo.c'  ), 'C source, ASCII text' );
 
 # subs from :complete
 my $handle = magic_open(MAGIC_NONE);
@@ -16,7 +16,7 @@
 is( magic_buffer( $handle, "Hello World\n" ), 'ASCII text' );
 
 is( magic_file( $handle, 't/samples/foo.txt' ), 'ASCII text'           );
-is( magic_file( $handle, 't/samples/foo.c'   ), 'ASCII C program text' );
+is( magic_file( $handle, 't/samples/foo.c'   ), 'C source, ASCII text' );
 
 magic_close($handle);
 
--- a/t/6-oo.t
+++ b/t/6-oo.t
@@ -6,12 +6,12 @@
 
 my %standard = (
     'foo.foo' => [ 'ASCII text',           'text/plain; charset=us-ascii' ],
-    'foo.c'   => [ 'ASCII C program text', 'text/x-c; charset=us-ascii'   ],
+    'foo.c'   => [ 'C source, ASCII text', 'text/x-c; charset=us-ascii'   ],
 );
 
 my %custom = (
     'foo.foo' => [ 'A foo file',           'text/plain; charset=us-ascii' ],
-    'foo.c'   => [ 'ASCII C program text', 'text/x-c; charset=us-ascii'   ],
+    'foo.c'   => [ 'ASCII text',           'text/plain; charset=us-ascii'   ],
 );
 
 plan tests => 4 + 4*(keys %standard) + 4*(keys %custom);
