Description: Trigger build failure when libmagic is not found
Origin: vendor
Forwarded: not-needed
Author: Alessandro Ghedini <ghedo@debian.org>
Reviewed-by: Alessandro Ghedini <ghedo@debian.org>
Last-Update: 2012-08-07

--- a/Makefile.PL
+++ b/Makefile.PL
@@ -7,7 +7,7 @@
 use lib qw(inc);
 use Devel::CheckLib;
 
-check_lib_or_exit( lib => 'magic', header => 'magic.h' );
+assert_lib( lib => 'magic', header => 'magic.h' );
 
 WriteMakefile(
     'NAME'		=> 'File::LibMagic',
