Description: fix pod
Origin: vendor
Forwarded: no
Author: gregor herrmann <gregoa@debian.org>
Last-Update: 2011-11-26

--- a/lib/Data/Random.pm
+++ b/lib/Data/Random.pm
@@ -682,7 +682,7 @@
 
 =head2 rand_date()
 
-This returns a random date in the form "YYYY-MM-DD".  2-digit years are not currently supported.  Efforts are made to make sure you're returned a truly valid date--ie, you'll never be returned the date February 31st.  See the options below to find out how to control the date range.  Here are a few examples:
+This returns a random date in the form "YYYY-MM-DD".  2-digit years are not currently supported.  Efforts are made to make sure you're returned a truly valid date - ie, you'll never be returned the date February 31st.  See the options below to find out how to control the date range.  Here are a few examples:
 
     # returns a date somewhere in between the current date, and one year from the current date
     $date = rand_date();
