Default to Debian alternative x-terminal-emulator
--- a/bindings.lisp
+++ b/bindings.lisp
@@ -75,8 +75,8 @@
   *escape-key* '*root-map*)
 
 (fill-keymap *root-map*
-  (kbd "c")   "exec xterm"
-  (kbd "C-c") "exec xterm"
+  (kbd "c")   "exec x-terminal-emulator"
+  (kbd "C-c") "exec x-terminal-emulator"
   (kbd "e")   "emacs"
   (kbd "C-e") "emacs"
   (kbd "b")   "banish"
--- a/sample-stumpwmrc.lisp
+++ b/sample-stumpwmrc.lisp
@@ -19,7 +19,7 @@
 ;; Browse somewhere
 (define-key *root-map* (kbd "b") "colon1 exec firefox http://www.")
 ;; Ssh somewhere
-(define-key *root-map* (kbd "C-s") "colon1 exec xterm -e ssh ")
+(define-key *root-map* (kbd "C-s") "colon1 exec x-terminal-emulator -e ssh ")
 ;; Lock screen
 (define-key *root-map* (kbd "C-l") "exec xlock")
 
--- a/contrib/wmii-like-stumpwmrc.lisp
+++ b/contrib/wmii-like-stumpwmrc.lisp
@@ -19,7 +19,7 @@
 
 ;;; Wmii-like keybindings
 
-(defvar *terminal* "xterm"
+(defvar *terminal* "x-terminal-emulator"
   "The command used to start a terminal. It should understand the -e
   parameter.")
 
--- a/stumpwm.texi.in
+++ b/stumpwm.texi.in
@@ -229,7 +229,7 @@
 Once you have StumpWM up and running, the first thing you might want
 to do is start @command{emacs}. Type @kbd{C-t e}, or in other words
 @kbd{Control + t} followed by @kbd{e}. Now perhaps you want an
-@command{xterm}. Type @kbd{C-t c}. Now you have some programs running.
+@command{x-terminal-emulator}. Type @kbd{C-t c}. Now you have some programs running.
 
 To see a list of windows StumpWM is managing, type @kbd{C-t w}. The
 highlighted window is the one you're looking at right now. It's the
@@ -244,16 +244,16 @@
 executed from the input bar or bound to a key.
 
 At this point you probably want to switch back from your new
-@command{xterm} to @command{emacs}. Type @kbd{C-t C-t}. This runs the
-@command{other} command. Type it again and you're back to xterm.
+@command{x-terminal-emulator} to @command{emacs}. Type @kbd{C-t C-t}. This runs the
+@command{other} command. Type it again and you're back to x-terminal-emulator.
 
-Perhaps you'd like to see @command{emacs} and @command{xterm}
+Perhaps you'd like to see @command{emacs} and @command{x-terminal-emulator}
 side-by-side. Type @kbd{C-t s}. You have now split the screen into 2
 @command{frames}. For more information see @ref{Frames}. To switch to the
-empty frame type @kbd{C-t TAB}. Now let's pull the xterm window into
+empty frame type @kbd{C-t TAB}. Now let's pull the x-terminal-emulator window into
 this empty frame. Type @kbd{C-t w} for a window listing. Find the
-@command{xterm} window. See the number beside it? Type @kbd{C-t} followed
-by @command{xterm}'s window number.
+@command{x-terminal-emulator} window. See the number beside it? Type @kbd{C-t} followed
+by @command{x-terminal-emulator}'s window number.
 
 Another common activity is browsing the interwebs. Type @kbd{C-t !}.
 The input bar pops up again. You can now run a shell command.  Let's
@@ -369,7 +369,7 @@
 
 @item C-t c
 @itemx C-t C-c
-Run an X terminal; by default @command{xterm}
+Run an X terminal; by default @command{x-terminal-emulator}
 
 @item C-t e
 @itemx C-t C-e
