Default to Debian alternative x-www-browser
--- a/sample-stumpwmrc.lisp
+++ b/sample-stumpwmrc.lisp
@@ -17,7 +17,7 @@
 ;; Read some doc
 (define-key *root-map* (kbd "d") "exec gv")
 ;; Browse somewhere
-(define-key *root-map* (kbd "b") "colon1 exec firefox http://www.")
+(define-key *root-map* (kbd "b") "colon1 exec x-www-browser http://www.")
 ;; Ssh somewhere
 (define-key *root-map* (kbd "C-s") "colon1 exec x-terminal-emulator -e ssh ")
 ;; Lock screen
@@ -29,8 +29,8 @@
     (substitute #\+ #\Space search)
     (run-shell-command (concatenate 'string ,prefix search))))
 
-(make-web-jump "google" "firefox http://www.google.fr/search?q=")
-(make-web-jump "imdb" "firefox http://www.imdb.com/find?q=")
+(make-web-jump "google" "x-www-browser http://www.google.fr/search?q=")
+(make-web-jump "imdb" "x-www-browser http://www.imdb.com/find?q=")
 
 ;; C-t M-s is a terrble binding, but you get the idea.
 (define-key *root-map* (kbd "M-s") "google")
