From: Martin Erik Werner <martinerikwerner@gmail.com>
Date: Wed, 11 Jan 2012 03:21:42 +0100
Subject: Fix syntax error in setnonhinting script
Forwarded: http://code.google.com/p/googlefontdirectory/issues/detail?id=110


--- a/tools/nonhinting/setnonhinting-fontforge.py
+++ b/tools/nonhinting/setnonhinting-fontforge.py
@@ -52,8 +52,6 @@
     font_in = argv[0]
     font = fontforge.open(font_in)
 
-    font.
-
 #   If there is a second font file specified on the command line, output to that
     if len(argv) == 2:
         font_out = argv[1]
