diff --git a/smalltalk-mode-init.el.in b/smalltalk-mode-init.el.in
index 4c7c041..56a9aa1 100644
--- a/smalltalk-mode-init.el.in
+++ b/smalltalk-mode-init.el.in
@@ -15,6 +15,11 @@
 
 (push "\\.star\\'" inhibit-first-line-modes-regexps)
 
-(autoload 'smalltalk-mode "@lispdir@/smalltalk-mode.elc" "" t)
-@WITH_EMACS_COMINT_TRUE@(autoload 'gst "@lispdir@/gst-mode.elc" "" t)
+(autoload 'smalltalk-mode "smalltalk-mode" "" t)
+(autoload 'gst-mode "gst-mode" "" t)
 
+(debian-pkg-add-load-path-item 
+	(concat "/usr/share/"
+		(symbol-name debian-emacs-flavor)
+		"/site-lisp/gnu-smalltalk-el")
+	)
