Description: Shared lib did not had soname set
Author: Mathieu Malaterre <malat@debian.org>
Forwarded: https://groups.google.com/group/openvdb-forum/msg/23279e7b14670783

Index: openvdb/Makefile
===================================================================
--- openvdb.orig/Makefile	2013-03-16 01:11:08.000000000 +0100
+++ openvdb/Makefile	2013-03-16 16:07:03.590380199 +0100
@@ -364,11 +364,13 @@
 LIB_PATCH_VERSION=$(shell grep 'OPENVDB_LIBRARY_PATCH_VERSION = ' \
     version.h | sed 's/[^0-9]*//g')
 
-LIB_VERSION=$(LIB_MINOR_VERSION).$(LIB_PATCH_VERSION)
+LIB_VERSION=$(LIB_MAJOR_VERSION).$(LIB_MINOR_VERSION).$(LIB_PATCH_VERSION)
+LIB_SOVERSION=$(LIB_MAJOR_VERSION).$(LIB_MINOR_VERSION)
 
 LIBOPENVDB_NAME=libopenvdb
 LIBOPENVDB_STATIC := $(LIBOPENVDB_NAME).a
 LIBOPENVDB_SHARED := $(LIBOPENVDB_NAME).so.$(LIB_VERSION)
+LIBOPENVDB_SONAME := $(LIBOPENVDB_NAME).so.$(LIB_SOVERSION)
 
 ifeq (no,$(strip $(shared)))
     LIBOPENVDB := $(LIBOPENVDB_STATIC)
@@ -420,12 +422,15 @@
 # Build shared library
 lib: $(LIBOPENVDB_NAME).so
 
-$(LIBOPENVDB_NAME).so: $(LIBOPENVDB_SHARED)
+$(LIBOPENVDB_NAME).so: $(LIBOPENVDB_SONAME)
+	ln -f -s $< $@
+
+$(LIBOPENVDB_SONAME): $(LIBOPENVDB_SHARED)
 	ln -f -s $< $@
 
 $(LIBOPENVDB_SHARED): $(OBJ_NAMES)
 	@echo "Building $@ because of $(call list_deps)"
-	$(CXX) $(CXXFLAGS) -shared -o $@ $^ $(LIBS_RPATH)
+	$(CXX) $(CXXFLAGS) -shared -o $@ $^ $(LIBS_RPATH) -Wl,-soname,$(LIBOPENVDB_SONAME)
 
 else
 
@@ -522,6 +527,7 @@
 	cp -f $(LIBOPENVDB) $(INSTALL_DIR)/lib
 	pushd $(INSTALL_DIR)/lib > /dev/null; \
 	    if [ -f $(LIBOPENVDB_SHARED) ]; then \
+	        ln -f -s $(LIBOPENVDB_SHARED) $(LIBOPENVDB_SONAME); \
 	        ln -f -s $(LIBOPENVDB_SHARED) $(LIBOPENVDB_NAME).so; fi; \
 	    popd > /dev/null
 	@echo "Copied libopenvdb to $(INSTALL_DIR)/lib/"
