Description: rework build system to allow multi-arch install
Author: Mathieu Malaterre <malat@debian.org>
Forwarded: not-needed

Index: openvdb/Makefile
===================================================================
--- openvdb.orig/Makefile	2013-03-16 16:07:03.590380199 +0100
+++ openvdb/Makefile	2013-03-16 16:07:08.566380132 +0100
@@ -68,6 +68,8 @@
 # The directory into which to install libraries, executables and header files
 INSTALL_DIR := /tmp/OpenVDB
 
+INSTALL_LIB_DIR := $(INSTALL_LIB_DIR)/lib
+
 # The parent directory of the boost/ header directory
 BOOST_INCL_DIR := $(HT)/include
 
@@ -522,15 +524,15 @@
 	for f in $(INCLUDE_NAMES); \
 	    do cp -f $$f $(INSTALL_DIR)/include/openvdb/$$f; done
 	@#
-	mkdir -p $(INSTALL_DIR)/lib
-	@echo "Created $(INSTALL_DIR)/lib/"
-	cp -f $(LIBOPENVDB) $(INSTALL_DIR)/lib
-	pushd $(INSTALL_DIR)/lib > /dev/null; \
+	mkdir -p $(INSTALL_LIB_DIR)
+	@echo "Created $(INSTALL_LIB_DIR)"
+	cp -f $(LIBOPENVDB) $(INSTALL_LIB_DIR)
+	pushd $(INSTALL_LIB_DIR) > /dev/null; \
 	    if [ -f $(LIBOPENVDB_SHARED) ]; then \
 	        ln -f -s $(LIBOPENVDB_SHARED) $(LIBOPENVDB_SONAME); \
 	        ln -f -s $(LIBOPENVDB_SHARED) $(LIBOPENVDB_NAME).so; fi; \
 	    popd > /dev/null
-	@echo "Copied libopenvdb to $(INSTALL_DIR)/lib/"
+	@echo "Copied libopenvdb to $(INSTALL_LIB_DIR)"
 	@#
 	mkdir -p $(INSTALL_DIR)/bin
 	@echo "Created $(INSTALL_DIR)/bin/"
