Description: Remove hard-coded rpath from original build system
Author: Mathieu Malaterre <malat@debian.org>
Forwarded: not-needed

Index: openvdb/Makefile
===================================================================
--- openvdb.orig/Makefile	2013-03-16 16:07:08.566380132 +0100
+++ openvdb/Makefile	2013-03-16 16:07:11.062380097 +0100
@@ -491,7 +491,7 @@
 vdb_view: $(LIBOPENVDB) $(VIEWER_OBJ_NAMES)
 	@echo "Building $@ because of $(call list_deps)"
 	$(CXX) $(CXXFLAGS) -o $@ $(VIEWER_OBJ_NAMES) -I . \
-	    -Wl,-rpath,$(GLFW_LIB_DIR) -L$(GLFW_LIB_DIR) $(GLFW_LIB) -lGL -lGLU \
+	    -L$(GLFW_LIB_DIR) $(GLFW_LIB) -lGL -lGLU \
 	    $(LIBS_RPATH) $(LIBOPENVDB_RPATH) -L$(PWD) $(LIBOPENVDB)
 endif
 endif
