Description: Run test suite in verbose mode
Author: Mathieu Malaterre <malat@debian.org>
Forwarded: not-needed

Index: openvdb/Makefile
===================================================================
--- openvdb.orig/Makefile	2013-03-16 16:07:11.062380097 +0100
+++ openvdb/Makefile	2013-03-16 16:07:13.078380070 +0100
@@ -508,7 +508,7 @@
 	    $(LIBOPENVDB_RPATH) -L$(PWD) $(LIBOPENVDB)
 
 test: vdb_test
-	export LD_LIBRARY_PATH=${LD_LIBRARY_PATH}:$(PWD); ./vdb_test
+	export LD_LIBRARY_PATH=${LD_LIBRARY_PATH}:$(PWD); ./vdb_test -v
 else
 vdb_test:
 	@echo "$@"': $$(CPPUNIT_INCL_DIR) is undefined'
