Description: Fix a spelling error calender -> calender
Origin: vendor
Forwarded: no
Author: Alessandro Ghedini <al3xbio@gmail.com>
Last-Update: 2011-11-21

--- a/libev/ev.pod
+++ b/libev/ev.pod
@@ -2192,7 +2192,7 @@
 
 Unlike C<ev_timer>, periodic watchers are not based on real time (or
 relative time, the physical time that passes) but on wall clock time
-(absolute time, the thing you can read on your calender or clock). The
+(absolute time, the thing you can read on your calendar or clock). The
 difference is that wall clock time can run faster or slower than real
 time, and time jumps are not uncommon (e.g. when you adjust your
 wrist-watch).
