Index: haskell-simpleirc-0.2.1/Network/SimpleIRC/Core.hs
===================================================================
--- haskell-simpleirc-0.2.1.orig/Network/SimpleIRC/Core.hs	2011-11-30 17:04:43.000000000 +0000
+++ haskell-simpleirc-0.2.1/Network/SimpleIRC/Core.hs	2012-11-09 05:27:05.666355483 +0000
@@ -47,6 +47,7 @@
 import Control.Concurrent
 import Control.Concurrent.Chan
 import Control.Concurrent.MVar
+import Control.Exception.Base (try)
 import Network.SimpleIRC.Messages
 import Data.Unique
 import System.IO.Error
