error message when using --delete-delay is not correct.

diff -Naur rsync-3.0.7/flist.c rsync-3.0.7-patch/flist.c
--- rsync-3.0.7/flist.c	2009-12-21 23:40:41.000000000 +0100
+++ rsync-3.0.7-patch/flist.c	2010-02-02 21:49:05.167469731 +0100
@@ -1791,7 +1791,12 @@
 	/* This (sadly) can only happen when pushing data because
 	 * the sender does not know about what kind of delete
 	 * is in effect on the receiving side when pulling. */
-	rprintf(FERROR_XFER, "FATAL I/O ERROR: dying to avoid a --delete-during issue with a pre-3.0.7 receiver.\n");
+	if(delete_during == 2) {
+		rprintf(FERROR_XFER, "FATAL I/O ERROR: dying to avoid a --delete-delay issue with a pre-3.0.7 receiver.\n");
+	}
+	else {
+		rprintf(FERROR_XFER, "FATAL I/O ERROR: dying to avoid a --delete-during issue with a pre-3.0.7 receiver.\n");
+	}
 	exit_cleanup(RERR_UNSUPPORTED);
 }
 
