--- rsync-2.6.9.orig/support/rsyncstats.orig	2006-11-20 15:50:42.711007850 +0100
+++ rsync-2.6.9/support/rsyncstats	2006-11-20 15:51:05.531353306 +0100
@@ -12,7 +12,7 @@
 use Getopt::Long;
 
 # You may wish to edit the next line to customize for your default log file.
-$usage_file = "/var/log/rsyncd.log";
+$usage_file = "/var/log/rsyncd";
 
 # Edit the following lines for default report settings.
 # Entries defined here will be over-ridden by the command line.
