--- a/cleanup.c	2012-04-15 14:14:02.913993934 +0200
+++ b/cleanup.c	2012-04-15 14:13:58.834033097 +0200
@@ -157,8 +157,9 @@
 				flush_write_file(cleanup_fd_w);
 				close(cleanup_fd_w);
 			}
+			cleanup_file->modtime = 0; /* solves problem with --partial --update */
 			finish_transfer(cleanup_new_fname, fname, NULL, NULL,
-					cleanup_file, 0, !partial_dir);
+					cleanup_file, 1, !partial_dir);
 		}
 
 		/* FALLTHROUGH */
