From: Mark Baker <mark@mnb.org.uk>
Description: Change soname to what debian use
Index: pcre3-8.31/configure
===================================================================
--- pcre3-8.31.orig/configure	2012-07-06 10:02:02.000000000 +0100
+++ pcre3-8.31/configure	2012-09-13 19:47:13.000000000 +0100
@@ -17190,13 +17190,13 @@
 # (Note: The libpcre*_version bits are m4 variables, assigned above)
 
 EXTRA_LIBPCRE_LDFLAGS="$EXTRA_LIBPCRE_LDFLAGS \
-                       $NO_UNDEFINED -version-info 1:1:0"
+                       $NO_UNDEFINED -version-info 16:1:13"
 
 EXTRA_LIBPCRE16_LDFLAGS="$EXTRA_LIBPCRE16_LDFLAGS \
-                       $NO_UNDEFINED -version-info 0:1:0"
+                       $NO_UNDEFINED -version-info 16:1:13"
 
 EXTRA_LIBPCREPOSIX_LDFLAGS="$EXTRA_LIBPCREPOSIX_LDFLAGS \
-                            $NO_UNDEFINED -version-info 0:1:0"
+                            $NO_UNDEFINED -version-info 16:1:13"
 
 EXTRA_LIBPCRECPP_LDFLAGS="$EXTRA_LIBPCRECPP_LDFLAGS \
                           $NO_UNDEFINED -version-info 0:0:0 \
