---
 localedata/locales/cy_GB |    2 ++
 1 file changed, 2 insertions(+)

--- a/localedata/locales/cy_GB
+++ b/localedata/locales/cy_GB
@@ -253,6 +253,8 @@
 date_fmt    "<U0025><U0061><U0020><U0025><U0062><U0020><U0025><U0065>/
 <U0020><U0025><U0048><U003A><U0025><U004D><U003A><U0025><U0053><U0020>/
 <U0025><U005A><U0020><U0025><U0059>"
+week    7;19971130;4
+first_weekday 2
 END LC_TIME
 
 LC_MESSAGES
