---
 localedata/Makefile |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/localedata/Makefile
+++ b/localedata/Makefile
@@ -238,7 +238,7 @@
 	echo -n `echo $$locale | sed 's/\([^\@]*\)\(\@.*\)*/\2/'`; \
 	echo -n '...'; \
 	input=`echo $$locale | sed 's/\([^.]*\)[^@]*\(.*\)/\1\2/'`; \
-	$(LOCALEDEF) --alias-file=../intl/locale.alias \
+	$(LOCALEDEF) --alias-file=../intl/locale.alias --no-archive \
 		     -i locales/$$input -c -f charmaps/$$charset \
 		     $(addprefix --prefix=,$(install_root)) $$locale; \
 	echo ' done'; \
