---
 localedata/SUPPORTED |   14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

--- a/localedata/SUPPORTED
+++ b/localedata/SUPPORTED
@@ -233,8 +233,8 @@
 hne_IN/UTF-8 \
 hr_HR.UTF-8/UTF-8 \
 hr_HR/ISO-8859-2 \
-hsb_DE/ISO-8859-2 \
 hsb_DE.UTF-8/UTF-8 \
+hsb_DE/ISO-8859-2 \
 ht_HT/UTF-8 \
 hu_HU.UTF-8/UTF-8 \
 hu_HU/ISO-8859-2 \
@@ -254,8 +254,8 @@
 iu_CA/UTF-8 \
 iw_IL.UTF-8/UTF-8 \
 iw_IL/ISO-8859-8 \
-ja_JP.EUC-JP/EUC-JP \
 ja_JP.UTF-8/UTF-8 \
+ja_JP.EUC-JP/EUC-JP \
 ka_GE.UTF-8/UTF-8 \
 ka_GE/GEORGIAN-PS \
 kk_KZ.UTF-8/UTF-8 \
@@ -264,8 +264,8 @@
 kl_GL/ISO-8859-1 \
 km_KH/UTF-8 \
 kn_IN/UTF-8 \
-ko_KR.EUC-KR/EUC-KR \
 ko_KR.UTF-8/UTF-8 \
+ko_KR.EUC-KR/EUC-KR \
 kok_IN/UTF-8 \
 ks_IN/UTF-8 \
 ks_IN@devanagari/UTF-8 \
@@ -339,8 +339,8 @@
 pt_PT@euro/ISO-8859-15 \
 ro_RO.UTF-8/UTF-8 \
 ro_RO/ISO-8859-2 \
-ru_RU.KOI8-R/KOI8-R \
 ru_RU.UTF-8/UTF-8 \
+ru_RU.KOI8-R/KOI8-R \
 ru_RU/ISO-8859-5 \
 ru_UA.UTF-8/UTF-8 \
 ru_UA/KOI8-U \
@@ -411,9 +411,9 @@
 uz_UZ@cyrillic/UTF-8 \
 ve_ZA/UTF-8 \
 vi_VN/UTF-8 \
+wa_BE.UTF-8/UTF-8 \
 wa_BE/ISO-8859-1 \
 wa_BE@euro/ISO-8859-15 \
-wa_BE.UTF-8/UTF-8 \
 wae_CH/UTF-8 \
 wal_ET/UTF-8 \
 wo_SN/UTF-8 \
@@ -423,17 +423,17 @@
 yi_US/CP1255 \
 yo_NG/UTF-8 \
 yue_HK/UTF-8 \
+zh_CN.UTF-8/UTF-8 \
 zh_CN.GB18030/GB18030 \
 zh_CN.GBK/GBK \
-zh_CN.UTF-8/UTF-8 \
 zh_CN/GB2312 \
 zh_HK.UTF-8/UTF-8 \
 zh_HK/BIG5-HKSCS \
 zh_SG.UTF-8/UTF-8 \
 zh_SG.GBK/GBK \
 zh_SG/GB2312 \
-zh_TW.EUC-TW/EUC-TW \
 zh_TW.UTF-8/UTF-8 \
+zh_TW.EUC-TW/EUC-TW \
 zh_TW/BIG5 \
 zu_ZA.UTF-8/UTF-8 \
 zu_ZA/ISO-8859-1 \
