---
 localedata/locales/sv_SE |   19 ++++++++++++++-----
 1 file changed, 14 insertions(+), 5 deletions(-)

--- a/localedata/locales/sv_SE
+++ b/localedata/locales/sv_SE
@@ -169,14 +169,23 @@
           "<U006F><U006B><U0074><U006F><U0062><U0065><U0072>";/
           "<U006E><U006F><U0076><U0065><U006D><U0062><U0065><U0072>";/
           "<U0064><U0065><U0063><U0065><U006D><U0062><U0065><U0072>"
-d_t_fmt   "<U0025><U0061><U0020><U0025><U0065><U0020><U0025><U0062><U0020><U0025><U0059><U0020><U0025><U0048><U002E><U0025><U004D><U002E><U0025><U0053>"
+%
+% Appropriate date and time representation (%c)
+%       "%a %e %b %Y %H:%M:%S"
+d_t_fmt   "<U0025><U0061><U0020><U0025><U0065><U0020><U0025><U0062><U0020><U0025><U0059><U0020><U0025><U0048><U003A><U0025><U004D><U003A><U0025><U0053>"
+%
+% Appropriate date representation (%x)   "%Y-%m-%d"
 d_fmt     "<U0025><U0059><U002D><U0025><U006D><U002D><U0025><U0064>"
-t_fmt     "<U0025><U0048><U002E><U0025><U004D><U002E><U0025><U0053>"
+%
+% Appropriate time representation (%X)   "%H:%M:%S"
+t_fmt     "<U0025><U0048><U003A><U0025><U004D><U003A><U0025><U0053>"
 am_pm     "";""
 t_fmt_ampm  ""
-date_fmt       "<U0025><U0061><U0020><U0025><U0062><U0020><U0025><U0065>/
-<U0020><U0025><U0048><U003A><U0025><U004D><U003A><U0025><U0053><U0020>/
-<U0025><U005A><U0020><U0025><U0059>"
+%
+% Appropriate date representation (date(1))   "%a %e %b %Y %H:%M:%S %Z"
+date_fmt       "<U0025><U0061><U0020><U0025><U0065><U0020><U0025><U0062>/
+<U0020><U0025><U0059><U0020><U0025><U0048><U003A><U0025><U004D><U003A>/
+<U0025><U0053><U0020><U0025><U005A>"
 first_weekday 2
 first_workday 2
 END LC_TIME
