Author: <hesso@pool.math.tu-berlin.de>
Description: Get rid of several minor gcc warnings.
Index: screen/attacher.c
===================================================================
--- screen.orig/attacher.c	2011-10-08 19:58:12.000000000 +0200
+++ screen/attacher.c	2011-10-08 19:58:41.000000000 +0200
@@ -848,7 +848,7 @@
   pam_handle_t *pamh = 0;
   int pam_error;
 #endif
-  char *pass, mypass[16 + 1], salt[3];
+  char *pass = 0, mypass[16 + 1], salt[3];
   int using_pam = 1;
 
 #ifdef USE_PAM
Index: screen/layer.c
===================================================================
--- screen.orig/layer.c	2011-10-08 19:54:20.000000000 +0200
+++ screen/layer.c	2011-10-08 19:58:41.000000000 +0200
@@ -492,7 +492,7 @@
 void
 LClearLine(l, y, xs, xe, bce, ol)
 struct layer *l;
-int xs, xe, bce;
+int y, xs, xe, bce;
 struct mline *ol;
 {
   struct canvas *cv;
