Author: <hesso@pool.math.tu-berlin.de>
Description: Remove an unconditional nethack message. (# 512299)
Index: screen/attacher.c
===================================================================
--- screen.orig/attacher.c	2011-10-08 19:58:41.000000000 +0200
+++ screen/attacher.c	2011-10-08 19:59:12.000000000 +0200
@@ -621,7 +621,7 @@
         {
 	  fcntl(0, F_SETFL, 0);
 	  SetTTY(0, &attach_Mode);
-	  printf("\nSuddenly the Dungeon collapses!! - You die...\n");
+	  printf("\nError: Cannot find master process to attach to!\n");
 	  eexit(1);
         }
 #ifdef BSDJOBS
