Fixes directory creation race condition (#653434)

Index: screen/screen.c
===================================================================
--- screen.orig/screen.c	2012-06-21 01:43:09.000000000 +0200
+++ screen/screen.c	2012-06-21 01:44:31.000000000 +0200
@@ -1127,7 +1127,7 @@
 	  sprintf(SockPath, "%s/S-%s", SockDir, LoginName);
 	  if (access(SockPath, F_OK))
 	    {
-	      if (mkdir(SockPath, 0700) == -1)
+	      if (mkdir(SockPath, 0700) == -1 && errno != EEXIST)
 		Panic(errno, "Cannot make directory '%s'", SockPath);
 	      (void) chown(SockPath, real_uid, real_gid);
 	    }
