Index: icon-naming-utils-0.8.90/Makefile.am
===================================================================
--- icon-naming-utils-0.8.90.orig/Makefile.am	2007-08-18 14:24:39.000000000 +0200
+++ icon-naming-utils-0.8.90/Makefile.am	2009-03-08 14:41:23.000000000 +0100
@@ -11,7 +11,7 @@
 pkgconfigdir = $(datadir)/pkgconfig
 pkgconfig_DATA = $(pkgconfig_in_files:.pc.in=.pc)
 
-dtddir = $(datadir)/dtds
+dtddir = $(pkgdatadir)
 dtd_DATA = legacy-icon-mapping.dtd
 
 pkgdata_DATA = legacy-icon-mapping.xml
Index: icon-naming-utils-0.8.90/Makefile.in
===================================================================
--- icon-naming-utils-0.8.90.orig/Makefile.in	2009-02-21 04:40:00.000000000 +0100
+++ icon-naming-utils-0.8.90/Makefile.in	2009-03-08 14:41:23.000000000 +0100
@@ -151,7 +151,7 @@
 
 pkgconfigdir = $(datadir)/pkgconfig
 pkgconfig_DATA = $(pkgconfig_in_files:.pc.in=.pc)
-dtddir = $(datadir)/dtds
+dtddir = $(pkgdatadir)
 dtd_DATA = legacy-icon-mapping.dtd
 pkgdata_DATA = legacy-icon-mapping.xml
 EXTRA_DIST = \
