Index: mingetty-1.08/mingetty.c
===================================================================
--- mingetty-1.08.orig/mingetty.c	2012-05-01 12:46:14.828486093 +0100
+++ mingetty-1.08/mingetty.c	2012-05-17 15:34:27.326610315 +0100
@@ -47,6 +47,9 @@
 /* some information about this host */
 static struct utsname uts;
 /* the hostname */
+#ifndef MAXHOSTNAMELEN
+#define MAXHOSTNAMELEN 64
+#endif
 static char hn[MAXHOSTNAMELEN + 1];
 /* process and session ID of this program */
 static pid_t pid, sid;
@@ -235,7 +238,9 @@
 		printf ("%s", uts.machine);
 		break;
 	case 'o':
+#ifdef __gnu_linux__
 		printf ("%s", uts.domainname);
+#endif
 		break;
 	case 'd':
 	case 't':
