Description: In Debian, lua library is called lua5.1.
Author: Michal Čihař <nijel@debian.org>
Forwarded: http://rpm.org/ticket/88
--- a/configure.ac
+++ b/configure.ac
@@ -703,7 +703,7 @@
 
 AS_IF([test "$with_lua" != no],[
   PKG_CHECK_MODULES([LUA],
-    [lua >= 5.1],
+    [lua5.1 >= 5.1],
     [AC_DEFINE(WITH_LUA, 1, [Build with lua support?])],
     [AC_MSG_ERROR([lua not present (--without-lua to disable)])])
   AC_SUBST(LUA_CFLAGS)
