Author: Thomas Viehmann <tv@beamnet.de>
Description: Fix for python memory handling
Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=445392
Last-Updated: 2009-04-20
Forwarded: no

diff -ruN newt-0.52.11-old/snackmodule.c newt-0.52.11/snackmodule.c
--- newt-0.52.11-old/snackmodule.c	2008-07-30 13:42:55.000000000 +0100
+++ newt-0.52.11/snackmodule.c	2009-04-20 21:31:38.000000000 +0100
@@ -273,7 +273,7 @@
 static snackWidget * snackWidgetNew (void) {
     snackWidget * widget;
      
-    widget = PyObject_NEW(snackWidget, &snackWidgetType);
+    widget = PyObject_New(snackWidget, &snackWidgetType);
 
     widget->scs.cb = NULL;
     widget->scs.data = NULL;
@@ -790,7 +790,7 @@
     if (help == Py_None)
 	help = NULL;
 
-    form = PyObject_NEW(snackForm, &snackFormType);
+    form = PyObject_New(snackForm, &snackFormType);
     form->fo = newtForm(NULL, help, 0);
 
     return form;
@@ -802,7 +802,7 @@
 
     if (!PyArg_ParseTuple(args, "ii", &cols, &rows)) return NULL;
 
-    grid = PyObject_NEW(snackGrid, &snackGridType);
+    grid = PyObject_New(snackGrid, &snackGridType);
     grid->grid = newtCreateGrid(cols, rows);
 
     return grid;
