Author: Alastair McKinstry <mckinstry@debian.org>
Description: newtInit() should only be called once
Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=557960
Forwarded: no
Last-Updated: 2011-12-04

Index: newt-0.52.14/snackmodule.c
===================================================================
--- newt-0.52.14.orig/snackmodule.c	2011-12-04 01:31:56.000000000 +0000
+++ newt-0.52.14/snackmodule.c	2011-12-04 01:33:58.000000000 +0000
@@ -284,10 +284,14 @@
 }
 
 static PyObject * initScreen(PyObject * s, PyObject * args) {
+    static int init_newt = 1;
     suspend.cb = NULL;
     suspend.data = NULL;
     
-    newtInit();
+    if (init_newt) { 
+	newtInit();
+	init_newt = 0;
+    }
     newtCls();
 
     Py_INCREF(Py_None);
