Author: Michael Vogt at Ubuntu
Description: Build for python-dbg package
Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=531725
Forwarded: not-needed
Last-Updated: 2012-01-09

Index: newt-0.52.14/Makefile.in
===================================================================
--- newt-0.52.14.orig/Makefile.in	2011-12-03 10:32:16.000000000 +0000
+++ newt-0.52.14/Makefile.in	2011-12-03 10:33:48.000000000 +0000
@@ -85,6 +85,8 @@
 			PLFLAGS=`$$ver-config --libs`; \
 	        $(CC) $(CPPFLAGS) $$PIFLAGS $$PCFLAGS -c -o $$ver/snackmodule.o snackmodule.c ;\
 	        $(CC) --shared $$PLDFLAGS $$PLFLAGS $(LDFLAGS) -o $$ver/_snackmodule.so $$ver/snackmodule.o -L.  -lnewt $(LIBS);\
+                $(CC) $(subst -O2,-O0,$(CFLAGS)) -I/usr/include/$${ver}_d -fPIC -c -o $$ver/snackmodule_d.o snackmodule.c ;\
+                $(CC) --shared $(SHCFLAGS) -o $$ver/_snackmodule_d.so $$ver/snackmodule_d.o -L .  -lnewt  ;\
 	done || :
 	touch $@
 
@@ -144,6 +146,7 @@
 	[ -n "$(PYTHONVERS)" ] && for ver in $(PYTHONVERS) ; do \
 	   [ -d $(instroot)/$(libdir)/$$ver/site-packages ] || install -m 755 -d $(instroot)/$(libdir)/$$ver/site-packages ;\
 	   install -m 755 $$ver/_snackmodule.so $(instroot)/$(libdir)/$$ver/site-packages ;\
+	   install -m 755 $$ver/_snackmodule_d.so $(instroot)/$(libdir)/$$ver/site-packages ;\
 	   install -m 644 snack.py $(instroot)/$(libdir)/$$ver/site-packages ;\
 	done || :
 
