Index: libgphoto2-2.4.10.1/camlibs/directory/directory.c
===================================================================
--- libgphoto2-2.4.10.1.orig/camlibs/directory/directory.c	2010-12-05 15:36:33.000000000 +0200
+++ libgphoto2-2.4.10.1/camlibs/directory/directory.c	2010-12-05 15:36:33.000000000 +0200
@@ -86,6 +86,8 @@
 	{"avi",  GP_MIME_AVI},
 	{"mov",  GP_MIME_QUICKTIME},
 	{"mpg",  GP_MIME_MPEG},
+	{"mts",  GP_MIME_AVCHD},
+	{"m2ts", GP_MIME_AVCHD},
 	{"pbm", "image/x-portable-bitmap"},
 	{"crw",  GP_MIME_CRW},
 	{"cr2",  GP_MIME_RAW},
Index: libgphoto2-2.4.10.1/gphoto2/gphoto2-file.h
===================================================================
--- libgphoto2-2.4.10.1.orig/gphoto2/gphoto2-file.h	2010-12-05 15:36:33.000000000 +0200
+++ libgphoto2-2.4.10.1/gphoto2/gphoto2-file.h	2010-12-05 15:36:33.000000000 +0200
@@ -51,6 +51,7 @@
 #define GP_MIME_WMA       "audio/x-wma"
 #define GP_MIME_ASF       "audio/x-asf"
 #define GP_MIME_MPEG      "video/mpeg"
+#define GP_MIME_AVCHD     "video/avchd"
 
 /**
  * \brief The type of view on the specified file.
Index: libgphoto2-2.4.10.1/libgphoto2/gphoto2-file.c
===================================================================
--- libgphoto2-2.4.10.1.orig/libgphoto2/gphoto2-file.c	2010-12-05 15:36:33.000000000 +0200
+++ libgphoto2-2.4.10.1/libgphoto2/gphoto2-file.c	2010-12-05 15:36:33.000000000 +0200
@@ -538,6 +538,8 @@
             "asf",  GP_MIME_ASF,
             "ogg",  GP_MIME_OGG,
             "mpg",  GP_MIME_MPEG,
+            "mts",  GP_MIME_AVCHD,
+            "m2ts", GP_MIME_AVCHD,
             NULL};
 
 	CHECK_NULL (file && filename);
