Description: Patch fixing build 
Author: Chris Cannam <chris.cannam@breakfastquay.com>
Forwarded: no

Index: dataquay/tests/main.cpp
===================================================================
--- dataquay.orig/tests/main.cpp	2012-11-15 09:21:06.534763985 +0100
+++ dataquay/tests/main.cpp	2012-11-15 13:05:42.641588378 +0100
@@ -42,7 +42,7 @@
 {
     int good = 0, bad = 0;
 
-    QApplication app(argc, argv);
+    QCoreApplication app(argc, argv);
 
     Dataquay::TestBasicStore tbs;
     if (QTest::qExec(&tbs, argc, argv) == 0) ++good;
