# DP: adjust locations of directories to Debian policy

Index: python2.6-2.6.5+20100521/Demo/scripts/ftpstats.py
===================================================================
--- python2.6-2.6.5+20100521.orig/Demo/scripts/ftpstats.py	2010-05-21 15:32:26.476709483 +0200
+++ python2.6-2.6.5+20100521/Demo/scripts/ftpstats.py	2010-05-21 15:32:35.875712007 +0200
@@ -6,7 +6,8 @@
 # ftpstats [-m maxitems] [-s search] [file]
 # -m maxitems: restrict number of items in "top-N" lists, default 25.
 # -s string:   restrict statistics to lines containing this string.
-# Default file is /usr/adm/ftpd;  a "-" means read standard input.
+# Default file for Debian GNU/Linux is /var/log/xferlog;
+# a "-" means read stdandard input.
 
 # The script must be run on the host where the ftp daemon runs.
 # (At CWI this is currently buizerd.)
@@ -34,7 +35,7 @@
             maxitems = string.atoi(a)
         if o == '-s':
             search = a
-    file = '/usr/adm/ftpd'
+    file = '/var/log/xferlog'
     if args: file = args[0]
     if file == '-':
         f = sys.stdin
Index: python2.6-2.6.5+20100521/Demo/tkinter/guido/ManPage.py
===================================================================
--- python2.6-2.6.5+20100521.orig/Demo/tkinter/guido/ManPage.py	2010-05-21 15:32:26.436711731 +0200
+++ python2.6-2.6.5+20100521/Demo/tkinter/guido/ManPage.py	2010-05-21 15:32:35.871745929 +0200
@@ -189,8 +189,9 @@
 def test():
     import os
     import sys
-    # XXX This directory may be different on your system
-    MANDIR = '/usr/local/man/mann'
+    # XXX This directory may be different on your system,
+    #     it is here set for Debian GNU/Linux.
+    MANDIR = '/usr/share/man'
     DEFAULTPAGE = 'Tcl'
     formatted = 0
     if sys.argv[1:] and sys.argv[1] == '-f':
Index: python2.6-2.6.5+20100521/Demo/tkinter/guido/tkman.py
===================================================================
--- python2.6-2.6.5+20100521.orig/Demo/tkinter/guido/tkman.py	2010-05-21 15:32:26.456719791 +0200
+++ python2.6-2.6.5+20100521/Demo/tkinter/guido/tkman.py	2010-05-21 15:32:35.875712007 +0200
@@ -9,8 +9,8 @@
 from Tkinter import *
 from ManPage import ManPage
 
-MANNDIRLIST = ['/depot/sundry/man/mann','/usr/local/man/mann']
-MAN3DIRLIST = ['/depot/sundry/man/man3','/usr/local/man/man3']
+MANNDIRLIST = ['/depot/sundry/man/mann','/usr/share/man/mann']
+MAN3DIRLIST = ['/depot/sundry/man/man3','/usr/share/man/man3']
 
 foundmanndir = 0
 for dir in MANNDIRLIST:
Index: python2.6-2.6.5+20100521/Lib/pydoc.py
===================================================================
--- python2.6-2.6.5+20100521.orig/Lib/pydoc.py	2010-05-21 15:32:26.372710645 +0200
+++ python2.6-2.6.5+20100521/Lib/pydoc.py	2010-05-21 15:32:35.867712036 +0200
@@ -27,6 +27,10 @@
 
 Module docs for core modules are assumed to be in
 
+    /usr/share/doc/pythonX.Y/html/library
+
+if the pythonX.Y-doc package is installed or in
+
     http://docs.python.org/library/
 
 This can be overridden by setting the PYTHONDOCS environment variable
@@ -349,6 +353,9 @@
 
         docloc = os.environ.get("PYTHONDOCS",
                                 "http://docs.python.org/library")
+        docdir = '/usr/share/doc/python%s/html/library' % sys.version[:3]
+        if not os.environ.has_key("PYTHONDOCS") and os.path.isdir(docdir):
+            docloc = docdir
         basedir = os.path.join(sys.exec_prefix, "lib",
                                "python"+sys.version[0:3])
         if (isinstance(object, type(os)) and
Index: python2.6-2.6.5+20100521/Misc/python.man
===================================================================
--- python2.6-2.6.5+20100521.orig/Misc/python.man	2010-05-21 15:32:26.495709663 +0200
+++ python2.6-2.6.5+20100521/Misc/python.man	2010-05-21 15:32:35.879712096 +0200
@@ -323,7 +323,7 @@
 These are subject to difference depending on local installation
 conventions; ${prefix} and ${exec_prefix} are installation-dependent
 and should be interpreted as for GNU software; they may be the same.
-The default for both is \fI/usr/local\fP.
+On Debian GNU/{Hurd,Linux} the default for both is \fI/usr\fP.
 .IP \fI${exec_prefix}/bin/python\fP
 Recommended location of the interpreter.
 .PP
Index: python2.6-2.6.5+20100521/Tools/faqwiz/faqconf.py
===================================================================
--- python2.6-2.6.5+20100521.orig/Tools/faqwiz/faqconf.py	2010-05-21 15:32:26.392715284 +0200
+++ python2.6-2.6.5+20100521/Tools/faqwiz/faqconf.py	2010-05-21 15:32:35.867712036 +0200
@@ -21,7 +21,7 @@
 OWNEREMAIL = "nobody@anywhere.org"      # Email for feedback
 HOMEURL = "http://www.python.org"       # Related home page
 HOMENAME = "Python home"                # Name of related home page
-RCSBINDIR = "/usr/local/bin/"           # Directory containing RCS commands
+RCSBINDIR = "/usr/bin/"			# Directory containing RCS commands
                                         # (must end in a slash)
 
 # Parameters you can normally leave alone
Index: python2.6-2.6.5+20100521/Tools/webchecker/webchecker.py
===================================================================
--- python2.6-2.6.5+20100521.orig/Tools/webchecker/webchecker.py	2010-05-21 15:32:26.416707302 +0200
+++ python2.6-2.6.5+20100521/Tools/webchecker/webchecker.py	2010-05-21 15:32:35.867712036 +0200
@@ -19,7 +19,8 @@
 a directory listing is returned.  Now, you can point webchecker to the
 document tree in the local file system of your HTTP daemon, and have
 most of it checked.  In fact the default works this way if your local
-web tree is located at /usr/local/etc/httpd/htdpcs (the default for
+web tree is located at /var/www, which is the default for Debian
+GNU/Linux. Other systems use /usr/local/etc/httpd/htdocs (the default for
 the NCSA HTTP daemon and probably others).
 
 Report printed:
