# DP: Don't try to download documentation tools

Index: python2.6-2.6.5+20100521/Doc/Makefile
===================================================================
--- python2.6-2.6.5+20100521.orig/Doc/Makefile	2010-05-21 15:52:06.072720141 +0200
+++ python2.6-2.6.5+20100521/Doc/Makefile	2010-05-21 15:52:06.076709754 +0200
@@ -56,7 +56,7 @@
 
 update: clean checkout
 
-build: checkout
+build:
 	mkdir -p build/$(BUILDER) build/doctrees
 	$(PYTHON) tools/sphinx-build.py $(ALLSPHINXOPTS)
 	@echo
