# DP: Enable the build of the fpectl module.

Index: python2.6-2.6.5+20100521/setup.py
===================================================================
--- python2.6-2.6.5+20100521.orig/setup.py	2010-05-21 16:21:35.967707776 +0200
+++ python2.6-2.6.5+20100521/setup.py	2010-05-21 16:21:35.975709770 +0200
@@ -1120,6 +1120,9 @@
         else:
             missing.append('_curses_panel')
 
+        #fpectl fpectlmodule.c ...
+        exts.append( Extension('fpectl', ['fpectlmodule.c']) )
+
         # Andrew Kuchling's zlib module.  Note that some versions of zlib
         # 1.1.3 have security problems.  See CERT Advisory CA-2002-07:
         # http://www.cert.org/advisories/CA-2002-07.html
