# DP: Build the ossaudio extension on GNU/kFreeBSD.

Index: setup.py
===================================================================
--- a/setup.py	(Revision 80320)
+++ b/setup.py	(Arbeitskopie)
@@ -1361,8 +1361,9 @@
         else:
             missing.append('linuxaudiodev')
 
-        if platform in ('linux2', 'freebsd4', 'freebsd5', 'freebsd6',
-                        'freebsd7', 'freebsd8'):
+        if (platform in ('linux2', 'freebsd4', 'freebsd5', 'freebsd6',
+                        'freebsd7', 'freebsd8')
+            or platform.startswith("gnukfreebsd")):
             exts.append( Extension('ossaudiodev', ['ossaudiodev.c']) )
         else:
             missing.append('ossaudiodev')
