# DP: Proper handling of packages in linecache.py

Index: python2.6-2.6.5+20100521/Lib/linecache.py
===================================================================
--- python2.6-2.6.5+20100521.orig/Lib/linecache.py	2010-05-21 15:51:47.151717677 +0200
+++ python2.6-2.6.5+20100521/Lib/linecache.py	2010-05-21 15:51:47.155720211 +0200
@@ -108,6 +108,11 @@
         if os.path.isabs(filename):
             return []
 
+        # Take care to handle packages
+        if basename == '__init__.py':
+            # filename referes to a package
+            basename = filename
+
         for dirname in sys.path:
             # When using imputil, sys.path may contain things other than
             # strings; ignore them when it happens.
