# DP: Fix bashism in makesetup shell script

Index: python2.6-2.6.5+20100521/Modules/makesetup
===================================================================
--- python2.6-2.6.5+20100521.orig/Modules/makesetup	2010-05-21 16:02:19.563709916 +0200
+++ python2.6-2.6.5+20100521/Modules/makesetup	2010-05-21 16:02:19.567710355 +0200
@@ -281,7 +281,7 @@
 	-)	;;
 	*)	sedf="@sed.in.$$"
 		trap 'rm -f $sedf' 0 1 2 3
-		echo "1i\\" >$sedf
+		printf "1i\\" >$sedf
 		str="# Generated automatically from $makepre by makesetup."
 		echo "$str" >>$sedf
 		echo "s%_MODOBJS_%$OBJS%" >>$sedf
