# DP: disable large file support for GNU/Hurd

Index: python2.6-2.6.5+20100521/configure.in
===================================================================
--- python2.6-2.6.5+20100521.orig/configure.in	2010-05-21 16:28:55.908709516 +0200
+++ python2.6-2.6.5+20100521/configure.in	2010-05-21 16:29:04.796713755 +0200
@@ -1408,6 +1408,9 @@
   use_lfs=no
 fi
 
+# Don't use largefile support anyway.
+use_lfs=no
+
 if test "$use_lfs" = "yes"; then
 # Two defines needed to enable largefile support on various platforms
 # These may affect some typedefs
