Index: python2.6-2.6.6/Modules/_ssl.c
===================================================================
--- python2.6-2.6.6.orig/Modules/_ssl.c	2011-04-20 09:23:09.553632124 +0000
+++ python2.6-2.6.6/Modules/_ssl.c	2011-04-20 09:29:02.995384745 +0000
@@ -303,7 +303,14 @@
     else if (proto_version == PY_SSL_VERSION_SSL3)
         self->ctx = SSL_CTX_new(SSLv3_method()); /* Set up context */
     else if (proto_version == PY_SSL_VERSION_SSL2)
+#ifdef OPENSSL_NO_SSL2
+    {
+        errstr = ERRSTR("SSLv2 support not provided anymore on Debian");
+        goto fail;
+    }
+#else
         self->ctx = SSL_CTX_new(SSLv2_method()); /* Set up context */
+#endif
     else if (proto_version == PY_SSL_VERSION_SSL23)
         self->ctx = SSL_CTX_new(SSLv23_method()); /* Set up context */
     PySSL_END_ALLOW_THREADS
