Index: lens-3.7.1.2/src/Data/Complex/Lens.hs
===================================================================
--- lens-3.7.1.2.orig/src/Data/Complex/Lens.hs	2013-01-01 22:43:28.000000000 +0000
+++ lens-3.7.1.2/src/Data/Complex/Lens.hs	2013-01-01 22:45:08.230678712 +0000
@@ -32,7 +32,7 @@
 
 -- $setup
 -- >>> import Debug.SimpleReflect
--- >>> let { a ≈ b = abs (a - b) < 1e-6; infix 4 ≈ }
+-- >>> let { a ~~ b = abs (a - b) < 1e-6; infix 4 ~~ }
 
 -- | Access the 'realPart' of a 'Complex' number
 --
@@ -102,7 +102,7 @@
 
 -- | Access the 'phase' of a 'Complex' number
 --
--- >>> (mkPolar 10 (2-pi) & _phase +~ pi & view _phase) ≈ 2
+-- >>> (mkPolar 10 (2-pi) & _phase +~ pi & view _phase) ~~ 2
 -- True
 --
 -- This isn't /quite/ a legal lens. Notably the
@@ -123,7 +123,7 @@
 -- >>> (2.0 :+ 3.0) & _conjugate . _imagPart -~ 1
 -- 2.0 :+ 4.0
 --
--- >>> (mkPolar 10.0 2.0 ^. _conjugate . _phase) ≈ (-2.0)
+-- >>> (mkPolar 10.0 2.0 ^. _conjugate . _phase) ~~ (-2.0)
 -- True
 _conjugate :: RealFloat a => Simple Iso (Complex a) (Complex a)
 _conjugate = iso conjugate conjugate
