Author: Enrico Tassi
Description: avoid name mangling while building with C++
Index: lua5.1-5.1.4/src/luaconf.h
===================================================================
--- lua5.1-5.1.4.orig/src/luaconf.h	2011-12-24 15:59:53.000000000 +0100
+++ lua5.1-5.1.4/src/luaconf.h	2011-12-24 16:01:42.000000000 +0100
@@ -166,7 +166,11 @@
 
 #else
 
+#ifdef __cplusplus
+#define LUA_API		extern "C"
+#else
 #define LUA_API		extern
+#endif
 
 #endif
 
