Author: John V. Belmonte <jbelmonte@debian.org>
Description: Set Lua's default PATH and CPATH.
Index: lua5.1-5.1.5/src/luaconf.h
===================================================================
--- lua5.1-5.1.5.orig/src/luaconf.h	2008-02-11 17:25:08.000000000 +0100
+++ lua5.1-5.1.5/src/luaconf.h	2012-07-16 12:23:12.000000000 +0200
@@ -94,14 +94,21 @@
 	".\\?.dll;"  LUA_CDIR"?.dll;" LUA_CDIR"loadall.dll"
 
 #else
+/* This defines DEB_HOST_MULTIARCH */
+#include "lua5.1-deb-multiarch.h"
 #define LUA_ROOT	"/usr/local/"
+#define LUA_ROOT2	"/usr/"
 #define LUA_LDIR	LUA_ROOT "share/lua/5.1/"
+#define LUA_LDIR2	LUA_ROOT2 "share/lua/5.1/"
 #define LUA_CDIR	LUA_ROOT "lib/lua/5.1/"
+#define LUA_CDIR2	LUA_ROOT2 "lib/" DEB_HOST_MULTIARCH "/lua/5.1/"
+#define LUA_CDIR3	LUA_ROOT2 "lib/lua/5.1/"
 #define LUA_PATH_DEFAULT  \
 		"./?.lua;"  LUA_LDIR"?.lua;"  LUA_LDIR"?/init.lua;" \
-		            LUA_CDIR"?.lua;"  LUA_CDIR"?/init.lua"
+		            LUA_CDIR"?.lua;"  LUA_CDIR"?/init.lua;" \
+		            LUA_LDIR2"?.lua;"  LUA_LDIR2"?/init.lua"
 #define LUA_CPATH_DEFAULT \
-	"./?.so;"  LUA_CDIR"?.so;" LUA_CDIR"loadall.so"
+	"./?.so;" LUA_CDIR"?.so;" LUA_CDIR2"?.so;" LUA_CDIR3"?.so;" LUA_CDIR"loadall.so"
 #endif
 
 
