diff -NurBbp gnutls-2.8.6.orig/configure gnutls-2.8.6/configure
--- gnutls-2.8.6.orig/configure	2010-03-15 11:29:16.000000000 +0100
+++ gnutls-2.8.6/configure	2010-03-20 16:01:07.000000000 +0100
@@ -7026,7 +7026,7 @@ fi
   rpathdirs=
   ltrpathdirs=
   names_already_handled=
-  names_next_round='gcrypt gpg-error'
+  names_next_round='gcrypt'
   while test -n "$names_next_round"; do
     names_this_round="$names_next_round"
     names_next_round=
diff -NurBbp gnutls-2.8.6.orig/lib/configure gnutls-2.8.6/lib/configure
--- gnutls-2.8.6.orig/lib/configure	2010-03-15 11:28:38.000000000 +0100
+++ gnutls-2.8.6/lib/configure	2010-03-20 16:00:59.000000000 +0100
@@ -12102,7 +12102,7 @@ fi
   rpathdirs=
   ltrpathdirs=
   names_already_handled=
-  names_next_round='gcrypt gpg-error'
+  names_next_round='gcrypt'
   while test -n "$names_next_round"; do
     names_this_round="$names_next_round"
     names_next_round=
diff -NurBbp gnutls-2.8.6.orig/libextra/configure gnutls-2.8.6/libextra/configure
--- gnutls-2.8.6.orig/libextra/configure	2010-03-15 11:28:58.000000000 +0100
+++ gnutls-2.8.6/libextra/configure	2010-03-20 16:00:53.000000000 +0100
@@ -11509,7 +11509,7 @@ fi
   rpathdirs=
   ltrpathdirs=
   names_already_handled=
-  names_next_round='gcrypt gpg-error'
+  names_next_round='gcrypt'
   while test -n "$names_next_round"; do
     names_this_round="$names_next_round"
     names_next_round=
