Description: Link against debian libopts.
Author: Andreas Metzler <ametzler@debian.org>

--- gnutls28-3.0.15.orig/configure
+++ gnutls28-3.0.15/configure
@@ -14261,6 +14261,7 @@ _ACEOF
   NEED_LIBOPTS_DIR=''
 
   # Check whether --enable-local-libopts was given.
+enable_local_libopts='no'
 if test "${enable_local_libopts+set}" = set; then :
   enableval=$enable_local_libopts;
     if test x$enableval = xyes ; then
