From 43269acc9bef22a9735047f35f1eb4577b492c18 Mon Sep 17 00:00:00 2001
From: Nikos Mavrogiannopoulos <nmav@gnutls.org>
Date: Fri, 12 Oct 2012 19:57:05 +0200
Subject: [PATCH] corrected typo

diff --git a/src/danetool-args.def b/src/danetool-args.def
index d13e52a..90017ea 100644
--- a/src/danetool-args.def
+++ b/src/danetool-args.def
@@ -126,12 +126,12 @@ doc-section = {
 To create a DANE TLSA resource record for a CA signed certificate use the following commands.
 
 @example
-$ certtool --tlsa-rr --host www.example.com --load-certificate cert.pem
+$ danetool --tlsa-rr --host www.example.com --load-certificate cert.pem
 @end example
 
 For a self signed certificate use:
 @example
-$ certtool --tlsa-rr --host www.example.com --load-certificate cert.pem \
+$ danetool --tlsa-rr --host www.example.com --load-certificate cert.pem \
   --local
 @end example
 
@@ -141,7 +141,7 @@ certificate using DANE.
 
 In order to create a record for the signer of your certificate use:
 @example
-$ certtool --tlsa-rr --host www.example.com --load-certificate cert.pem \
+$ danetool --tlsa-rr --host www.example.com --load-certificate cert.pem \
   --ca
 @end example
 
