Description: fix for newer doxygen
Origin: upstream, from ticket
Bug: http://bugs.icu-project.org/trac/ticket/9277
Bug-Debian: http://bugs.debian.org/674382

Index: icu-4.8.1.1/source/Makefile.in
===================================================================
--- icu-4.8.1.1.orig/source/Makefile.in	2011-10-18 19:15:50.000000000 -0400
+++ icu-4.8.1.1/source/Makefile.in	2012-05-24 13:52:31.506776284 -0400
@@ -16,9 +16,8 @@
 
 docdir = $(datadir)/doc
 docsubdir = $(PACKAGE)$(ICULIBDASHSUFFIX)/html
-docsubsrchdir = $(docsubdir)/search
 docfilesdir = doc/html
-docfiles = $(docfilesdir)/*.png $(docfilesdir)/*.html $(docfilesdir)/*.css $(docfilesdir)/*.tag $(docfilesdir)/installdox
+docfiles = $(docfilesdir)/*.png $(docfilesdir)/*.html $(docfilesdir)/*.css $(docfilesdir)/*.tag
 docsrchdir = $(docfilesdir)/search
 docsrchfiles = $(docsrchdir)/*
 
@@ -163,9 +162,8 @@
 else
 install-doc: doc
 	$(RM) -r $(DESTDIR)$(docdir)/$(docsubdir)
-	$(MKINSTALLDIRS) $(DESTDIR)$(docdir)/$(docsubsrchdir)
+	$(MKINSTALLDIRS) $(DESTDIR)$(docdir)/$(docsubdir)
 	$(INSTALL_DATA) $(docfiles) $(DESTDIR)$(docdir)/$(docsubdir)
-	$(INSTALL_DATA) $(docsrchfiles) $(DESTDIR)$(docdir)/$(docsubsrchdir)
 endif
 
 $(DESTDIR)$(pkglibdir)/%: $(top_srcdir)/../data/%
