Index: icu-4.8.1.1/source/config/mh-linux
===================================================================
--- icu-4.8.1.1.orig/source/config/mh-linux	2011-10-18 19:13:00.000000000 -0400
+++ icu-4.8.1.1/source/config/mh-linux	2012-01-04 09:50:49.856503236 -0500
@@ -21,7 +21,9 @@
 LD_RPATH_PRE = -Wl,-rpath,
 
 ## These are the library specific LDFLAGS
-LDFLAGSICUDT=-nodefaultlibs -nostdlib
+#LDFLAGSICUDT=-nodefaultlibs -nostdlib
+# Debian change: linking icudata as data only causes too many problems.
+LDFLAGSICUDT=
 
 ## Compiler switch to embed a library name
 # The initial tab in the next line is to prevent icu-config from reading it.
