Description: fix configure for 64-bit kfreebsd
Author: Petr Salinger <Petr.Salinger@seznam.cz>
Bug: http://bugs.icu-project.org/trac/ticket/8798
Bug-Debian: http://bugs.debian.org/630517

Index: icu-4.8.1.1/source/configure
===================================================================
--- icu-4.8.1.1.orig/source/configure	2011-11-09 09:56:31.163396717 -0500
+++ icu-4.8.1.1/source/configure	2011-11-09 09:56:44.947396524 -0500
@@ -6239,7 +6239,7 @@
 # Check to see if genccode can generate simple assembly.
 GENCCODE_ASSEMBLY=
 case "${host}" in
-*-linux*|i*86-*-*bsd*|i*86-pc-gnu)
+*-linux*|*-kfreebsd*-gnu*|i*86-*-*bsd*|i*86-pc-gnu)
     if test "$GCC" = yes; then
         # We're using gcc, and the simple -a gcc command line works for genccode
         GENCCODE_ASSEMBLY="-a gcc"
Index: icu-4.8.1.1/source/configure.in
===================================================================
--- icu-4.8.1.1.orig/source/configure.in	2011-11-09 09:56:31.171396718 -0500
+++ icu-4.8.1.1/source/configure.in	2011-11-09 09:56:44.947396524 -0500
@@ -604,7 +604,7 @@
 # Check to see if genccode can generate simple assembly.
 GENCCODE_ASSEMBLY=
 case "${host}" in
-*-linux*|i*86-*-*bsd*|i*86-pc-gnu)
+*-linux*|*-kfreebsd*-gnu*|i*86-*-*bsd*|i*86-pc-gnu)
     if test "$GCC" = yes; then
         # We're using gcc, and the simple -a gcc command line works for genccode
         GENCCODE_ASSEMBLY="-a gcc"
