Description: work around PATH_MAX for gnu-hurd
Author: Pino Toscano <pino@debian.org>
Bug-Debian: http://bugs.debian.org/643661

Index: icu-4.8.1.1/source/common/putil.c
===================================================================
--- icu-4.8.1.1.orig/source/common/putil.c	2011-11-09 09:56:31.127396722 -0500
+++ icu-4.8.1.1/source/common/putil.c	2011-11-09 09:56:45.187396519 -0500
@@ -149,6 +149,11 @@
 #include "icucfg.h"
 #endif
 
+/* Define PATH_MAX if not provided */
+#ifndef PATH_MAX
+#define PATH_MAX 4096
+#endif
+
 /* Define the extension for data files, again... */
 #define DATA_TYPE "dat"
 
