diff --git a/module/splat/splat-vnode.c b/module/splat/splat-vnode.c
index 0cd28a6..f5f010c 100644
--- a/module/splat/splat-vnode.c
+++ b/module/splat/splat-vnode.c
@@ -414,7 +414,7 @@ fd_uninstall(int fd)
                 goto out_unlock;
 
         rcu_assign_pointer(fdt->fd[fd], NULL);
-        FD_CLR(fd, fdt->close_on_exec);
+	__clear_close_on_exec(fd, fdt);
 #else
         spin_lock(&files->file_lock);
         if (fd >= files->max_fds)
