From: Damien Leone <damien.leone@fensalir.fr>
Subject: Check return values of system() calls.
Index: ncmpcpp-0.5.10/src
===================================================================
--- ncmpcpp-0.5.10.orig/src/lyrics.cpp	2012-04-01 09:49:09.000000000 -0700
+++ ncmpcpp-0.5.10/src/lyrics.cpp	2012-05-12 17:43:11.961233289 -0700
@@ -377,15 +377,20 @@
 	
 	if (Config.use_console_editor)
 	{
-		system(("/bin/sh -c \"" + Config.external_editor + " \\\"" + itsFilename + "\\\"\"").c_str());
+		if (system(("/bin/sh -c \"" + Config.external_editor + " \\\"" + itsFilename + "\\\"\"").c_str()) < 0)
+		{
+			ShowMessage("Couldn't edit lyrics");
+			return;
+		}
+
 		// below is needed as screen gets cleared, but apparently
 		// ncurses doesn't know about it, so we need to reload main screen
 		endwin();
 		initscr();
 		curs_set(0);
 	}
-	else
-		system(("nohup " + Config.external_editor + " \"" + itsFilename + "\" > /dev/null 2>&1 &").c_str());
+	else if (system(("nohup " + Config.external_editor + " \"" + itsFilename + "\" > /dev/null 2>&1 &").c_str()) < 0)
+		ShowMessage("Couldn't edit lyrics");
 }
 
 #ifdef HAVE_CURL_CURL_H
--- ncmpcpp-0.5.10.orig/src/status.cpp	2012-04-01 09:49:09.000000000 -0700
+++ ncmpcpp-0.5.10/src/status.cpp	2012-05-12 17:55:01.753208202 -0700
@@ -401,7 +401,8 @@
 		if (myPlaylist->isPlaying())
 		{
 			if (!Config.execute_on_song_change.empty())
-				system(Config.execute_on_song_change.c_str());
+				if (system(Config.execute_on_song_change.c_str()) < 0)
+					ShowMessage("Couldn't execute \"%s\"", Config.execute_on_song_change.c_str());
 			
 #			ifdef HAVE_CURL_CURL_H
 			if (Config.fetch_lyrics_in_background)
