--- a/release/makedpkg.in	(.../tags/1.2.1)	(revision 861)
+++ b/release/makedpkg.in	(.../branches/1.2.x)	(revision 861)
@@ -67,6 +67,10 @@
 		mkdir -p $TMPDIR/usr/include
 		mv $TMPDIR/opt/$DIRNAME/include/turbojpeg.h $TMPDIR/usr/include
 		ln -fs /usr/include/turbojpeg.h $TMPDIR/opt/$DIRNAME/include/
+	else
+		mkdir -p $TMPDIR/usr/lib/i386-linux-gnu
+		ln -fs /usr/lib32/libturbojpeg.so $TMPDIR/usr/lib/i386-linux-gnu/libturbojpeg.so
+		ln -fs /usr/lib32/libturbojpeg.a $TMPDIR/usr/lib/i386-linux-gnu/libturbojpeg.a
 	fi
 
 	sudo chown -Rh root:root $TMPDIR/*
--- a/turbojpeg.c	(.../tags/1.2.1)	(revision 861)
+++ b/turbojpeg.c	(.../branches/1.2.x)	(revision 861)
@@ -891,6 +891,7 @@
 	}
 
 	if(flags&TJFLAG_FASTUPSAMPLE) dinfo->do_fancy_upsampling=FALSE;
+	if(flags&TJFLAG_FASTDCT) dinfo->dct_method=JDCT_FASTEST;
 	dinfo->raw_data_out=TRUE;
 
 	jpeg_start_decompress(dinfo);
--- a/simd/CMakeLists.txt	(.../tags/1.2.1)	(revision 861)
+++ b/simd/CMakeLists.txt	(.../branches/1.2.x)	(revision 861)
@@ -43,16 +43,19 @@
   set(DEPFILE "")
   set(SIMD_SRC ${CMAKE_CURRENT_SOURCE_DIR}/${file}.asm)
   if(${file} MATCHES col)
-    set(DEPFILE ${SIMD_SRC})
+    set(DEPFILE ${file})
     string(REGEX REPLACE "col" "clr" DEPFILE ${DEPFILE})
+    set(DEPFILE ${CMAKE_CURRENT_SOURCE_DIR}/${DEPFILE}.asm)
   endif()
   if(${file} MATCHES mer)
-    set(DEPFILE ${SIMD_SRC})
+    set(DEPFILE ${file})
     string(REGEX REPLACE "mer" "mrg" DEPFILE ${DEPFILE})
+    set(DEPFILE ${CMAKE_CURRENT_SOURCE_DIR}/${DEPFILE}.asm)
   endif()
   if(${file} MATCHES gra)
-    set(DEPFILE ${SIMD_SRC})
+    set(DEPFILE ${file})
     string(REGEX REPLACE "gra" "gry" DEPFILE ${DEPFILE})
+    set(DEPFILE ${CMAKE_CURRENT_SOURCE_DIR}/${DEPFILE}.asm)
   endif()
   set(SIMD_OBJ ${OBJDIR}/${file}.obj)
   add_custom_command(OUTPUT ${SIMD_OBJ}
--- a/ChangeLog.txt	(.../tags/1.2.1)	(revision 861)
+++ b/ChangeLog.txt	(.../branches/1.2.x)	(revision 861)
@@ -1,3 +1,14 @@
+1.2.2 (not yet released)
+========================
+
+[1] The tjDecompressToYUV() function now supports the TJFLAG_FASTDCT flag.
+
+[2] The 32-bit supplementary package for amd64 Debian systems now provides
+symlinks in /usr/lib/i386-linux-gnu for the TurboJPEG libraries in /usr/lib32.
+This allows those libraries to be used on MultiArch-compatible systems (such as
+Ubuntu 11 and later) without setting the linker path.
+
+
 1.2.1
 =====
 
