Index: iputils/Makefile
===================================================================
--- iputils.orig/Makefile	2013-05-14 08:57:05.000000000 -0700
+++ iputils/Makefile	2013-05-14 09:11:54.686460217 -0700
@@ -106,7 +106,7 @@
 endif
 
 # -------------------------------------
-IPV4_TARGETS=tracepath ping clockdiff rdisc arping tftpd rarpd
+IPV4_TARGETS=tracepath ping clockdiff arping
 IPV6_TARGETS=tracepath6 traceroute6 ping6
 TARGETS=$(IPV4_TARGETS) $(IPV6_TARGETS)
 
