Subject: Fix syntax error in NetRead's ftp handling
Forwarded: no
Author: James McCoy <jamessan@debian.org>
Bug-Debian: http://bugs.debian.org/678513

--- a/runtime/autoload/netrw.vim
+++ b/runtime/autoload/netrw.vim
@@ -956,7 +956,7 @@
 "     call Decho("filter input: ".getline('.'))
     endif
 
-    if exists("g:netrw_uid" && g:netrw_uid != ""
+    if exists("g:netrw_uid") && g:netrw_uid != ""
      if exists("g:netrw_ftp") && g:netrw_ftp == 1
       keepj put =g:netrw_uid
 "       call Decho("filter input: ".getline('.'))
