From: Stefano Zacchiroli <zack@debian.org>
Date: Tue, 19 May 2009 17:24:14 +0200
Subject: Don't use rpath

---
 tools/ocamlmklib.mlp |    5 +++++
 1 file changed, 5 insertions(+)

diff --git a/tools/ocamlmklib.mlp b/tools/ocamlmklib.mlp
index b6c236e..b491fdd 100644
--- a/tools/ocamlmklib.mlp
+++ b/tools/ocamlmklib.mlp
@@ -38,6 +38,11 @@ and output_c = ref ""       (* Output name for C part of library *)
 and rpath = ref []          (* rpath options *)
 and verbose = ref false
 
+(* Debian specific: inhibit rpath *)
+let byteccrpath = ""
+and nativeccrpath = ""
+and mksharedlibrpath = ""
+
 let starts_with s pref =
   String.length s >= String.length pref &&
   String.sub s 0 (String.length pref) = pref
-- 
