From: Julien Cristau <julien.cristau@ens-lyon.org>
Date: Tue, 19 May 2009 17:24:52 +0200
Subject: Put manpages in section 3o instead of 3

---
 ocamldoc/Makefile |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ocamldoc/Makefile b/ocamldoc/Makefile
index 74c82d3..1ef43b3 100644
--- a/ocamldoc/Makefile
+++ b/ocamldoc/Makefile
@@ -344,7 +344,7 @@ test_texi: dummy
 stdlib_man/Pervasives.3o: $(STDLIB_MLIS)
 	$(MKDIR) stdlib_man
 	$(OCAMLDOC_RUN) -man -d stdlib_man $(INCLUDES) \
-	-t "OCaml library" -man-mini \
+	-t "OCaml library" -man-mini -man-section 3o \
 	$(STDLIB_MLIS)
 
 autotest_stdlib: dummy
-- 
