From: "Matteo F. Vescovi" <mfv.debian@gmail.com>
Date: Wed, 8 Aug 2012 15:56:40 +0200
Subject: Fix_FTBFS_on_kFreeBSD

---
 config/Makefile.kfreebsd |   13 ++++++++++---
 1 file changed, 10 insertions(+), 3 deletions(-)

diff --git a/config/Makefile.kfreebsd b/config/Makefile.kfreebsd
index b3747e1..d7d1612 100644
--- a/config/Makefile.kfreebsd
+++ b/config/Makefile.kfreebsd
@@ -2,7 +2,14 @@ NAME = $(GLEW_NAME)
 CC = cc
 LD = cc
 PICFLAG = -fPIC
-LDFLAGS.EXTRA = -L/usr/X11R6/lib
+M_ARCH ?= $(shell uname -m)
+ifeq (x86_64,${M_ARCH})
+LDFLAGS.EXTRA = -L/usr/X11R6/lib64 -L/usr/lib64
+LIBDIR = $(GLEW_DEST)/lib64
+else
+LDFLAGS.EXTRA = -L/usr/X11R6/lib -L/usr/lib
+LIBDIR = $(GLEW_DEST)/lib
+endif
 LDFLAGS.GL = -lXmu -lXi -lGL -lXext -lX11
 LDFLAGS.STATIC = -Wl,-Bstatic
 LDFLAGS.DYNAMIC = -Wl,-Bdynamic
@@ -14,9 +21,9 @@ LIB.SONAME    = lib$(NAME).so.$(SO_MAJOR)
 LIB.DEVLNK    = lib$(NAME).so
 LIB.SHARED    = lib$(NAME).so.$(SO_VERSION)
 LIB.STATIC    = lib$(NAME).a
-LDFLAGS.SO    = -shared -Wl,-soname $(LIB.SONAME)
+LDFLAGS.SO    = -shared -Wl,-soname=$(LIB.SONAME)
 LIB.SONAME.MX = lib$(NAME)mx.so.$(SO_MAJOR)
 LIB.DEVLNK.MX = lib$(NAME)mx.so
 LIB.SHARED.MX = lib$(NAME)mx.so.$(SO_VERSION)
 LIB.STATIC.MX = lib$(NAME)mx.a
-LDFLAGS.SO.MX = -shared -Wl,-soname $(LIB.SONAME.MX)
+LDFLAGS.SO.MX = -shared -Wl,-soname=$(LIB.SONAME.MX)
