Description: Fix perl path in the shebang line
Author: Rafael Laboissiere <rafael@debian.org>
--- a/bin/ps2eps
+++ b/bin/ps2eps
@@ -1,5 +1,4 @@
-eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}' && eval 'exec perl -S $0 $argv:q' # -*-perl-*-
-        if 0;
+#! /usr/bin/perl
 # The expression in the previous line replaces the unix specific line 
 # {#!/usr/bin/perl}.   
 # ps2eps - convert PostScript to EPS (Encapsulated PostScript) files
