---
 autogen.sh      |   12 +++---------
 test/autogen.sh |   11 ++---------
 2 files changed, 5 insertions(+), 18 deletions(-)

--- a/autogen.sh
+++ b/autogen.sh
@@ -4,15 +4,9 @@
 echo "This may take a while ..."
 
 # Regenerate configuration files
-cp acinclude.m4 aclocal.m4
-found=false
-for autoconf in autoconf autoconf259 autoconf-2.59
-do if which $autoconf >/dev/null 2>&1; then $autoconf && found=true; break; fi
-done
-if test x$found = xfalse; then
-    echo "Couldn't find autoconf, aborting"
-    exit 1
-fi
+aclocal-1.9
+autoconf
+autoheader
 (cd test; sh autogen.sh)
 
 # Run configure for this platform
--- a/test/autogen.sh
+++ b/test/autogen.sh
@@ -1,12 +1,5 @@
 #!/bin/sh
 #
 # Regenerate configuration files
-cp acinclude.m4 aclocal.m4
-found=false
-for autoconf in autoconf autoconf259 autoconf-2.59
-do if which $autoconf >/dev/null 2>&1; then $autoconf && found=true; break; fi
-done
-if test x$found = xfalse; then
-    echo "Couldn't find autoconf, aborting"
-    exit 1
-fi
+aclocal-1.9
+autoconf
