---
 build-scripts/strip_fPIC.sh |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/build-scripts/strip_fPIC.sh
+++ b/build-scripts/strip_fPIC.sh
@@ -5,8 +5,8 @@
 command=""
 while [ $# -gt 0 ]; do
     case "$1" in
-        -?PIC)
-            # Ignore -fPIC and -DPIC options
+        -fPIC)
+            # Ignore -fPIC option
             ;;
         -fno-common)
             # Ignore -fPIC and -DPIC options
