Index: SDL-1.2.14/configure.in
===================================================================
--- SDL-1.2.14.orig/configure.in	2009-11-27 15:01:50.000000000 -0500
+++ SDL-1.2.14/configure.in	2009-11-27 15:01:50.000000000 -0500
@@ -882,19 +882,19 @@
 }
 
 
-dnl Do the iPod thing
-CheckIPod()
-{
-    AC_ARG_ENABLE(ipod,
-AC_HELP_STRING([--enable-ipod], [configure SDL to work with iPodLinux [[default=no]]]),
-        , enable_ipod=no)
-
-    if test x$enable_ipod = xyes; then
-        EXTRA_CFLAGS="$EXTRA_CFLAGS -DIPOD"
-        AC_DEFINE(SDL_VIDEO_DRIVER_IPOD)
-        SOURCES="$SOURCES $srcdir/src/video/ipod/*.c"
-    fi
-}
+#dnl Do the iPod thing
+#CheckIPod()
+#{
+#    AC_ARG_ENABLE(ipod,
+#AC_HELP_STRING([--enable-ipod], [configure SDL to work with iPodLinux [[default=no]]]),
+#        , enable_ipod=no)
+#
+#    if test x$enable_ipod = xyes; then
+#        EXTRA_CFLAGS="$EXTRA_CFLAGS -DIPOD"
+#        AC_DEFINE(SDL_VIDEO_DRIVER_IPOD)
+#        SOURCES="$SOURCES $srcdir/src/video/ipod/*.c"
+#    fi
+#}
 
 dnl Find the nanox include and library directories
 CheckNANOX()
