---
 include/SDL_events.h |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/include/SDL_events.h
+++ b/include/SDL_events.h
@@ -44,7 +44,7 @@
 #define SDL_PRESSED	1
 
 /* Event enumerations */
-typedef enum {
+typedef enum SDL_Events {
        SDL_NOEVENT = 0,			/* Unused (do not remove) */
        SDL_ACTIVEEVENT,			/* Application loses/gains visibility */
        SDL_KEYDOWN,			/* Keys pressed */
@@ -79,7 +79,7 @@
 
 /* Predefined event masks */
 #define SDL_EVENTMASK(X)	(1<<(X))
-typedef enum {
+typedef enum SDL_EventMasks {
 	SDL_ACTIVEEVENTMASK	= SDL_EVENTMASK(SDL_ACTIVEEVENT),
 	SDL_KEYDOWNMASK		= SDL_EVENTMASK(SDL_KEYDOWN),
 	SDL_KEYUPMASK		= SDL_EVENTMASK(SDL_KEYUP),
