---
 src/audio/pulse/SDL_pulseaudio.c |    7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

--- a/src/audio/pulse/SDL_pulseaudio.c
+++ b/src/audio/pulse/SDL_pulseaudio.c
@@ -340,11 +340,10 @@
 	SDL_memset(mixbuf, spec->silence, spec->size);
 	
 	/* Reduced prebuffering compared to the defaults. */
-	paattr.tlength = mixlen;
+	paattr.tlength = mixlen*2;
 	paattr.minreq = mixlen;
-	paattr.fragsize = mixlen;
-	paattr.prebuf = mixlen;
-	paattr.maxlength = mixlen * 4;
+	paattr.prebuf = mixlen * 2;
+	paattr.maxlength = mixlen * 2;
 	
 	/* The SDL ALSA output hints us that we use Windows' channel mapping */
 	/* http://bugzilla.libsdl.org/show_bug.cgi?id=110 */
