Index: libsdl1.2-1.2.13/src/video/x11/SDL_x11gl.c
===================================================================
--- libsdl1.2-1.2.13.orig/src/video/x11/SDL_x11gl.c	2009-10-01 16:53:43.000000000 -0400
+++ libsdl1.2-1.2.13/src/video/x11/SDL_x11gl.c	2009-10-01 16:55:39.000000000 -0400
@@ -70,6 +70,11 @@
 	/* It takes a bit of care to be fool-proof about parsing the
 	 * OpenGL extensions string. Don't be fooled by sub-strings, etc.
 	 */
+
+	/* http://bugs.debian.org/537487 */
+	if (extensions == NULL) {
+	      return 0;
+	}
 	
 	start = extensions;
 	
