Index: libsdl2/sdl2-config.in
===================================================================
--- libsdl2.orig/sdl2-config.in	2012-05-18 00:51:13.076995010 +0200
+++ libsdl2/sdl2-config.in	2012-05-18 00:51:13.089000887 +0200
@@ -42,7 +42,7 @@
       echo @SDL_VERSION@
       ;;
     --cflags)
-      echo -I@includedir@/SDL2 @SDL_CFLAGS@
+      echo -I@includedir@/SDL2 @SDL_CFLAGS@ @EXTRA_CFLAGS@
       ;;
 @ENABLE_SHARED_TRUE@    --libs)
 @ENABLE_SHARED_TRUE@      echo -L@libdir@ @SDL_RLD_FLAGS@ -lSDL2
Index: libsdl2/sdl2.pc.in
===================================================================
--- libsdl2.orig/sdl2.pc.in	2012-05-18 00:51:13.076995010 +0200
+++ libsdl2/sdl2.pc.in	2012-05-18 00:51:13.089000887 +0200
@@ -12,4 +12,4 @@
 Conflicts:
 Libs: -L${libdir} @SDL_RLD_FLAGS@ -lSDL2
 Libs.private: @SDL_STATIC_LIBS@
-Cflags: -I${includedir}/SDL2 @SDL_CFLAGS@
+Cflags: -I${includedir}/SDL2 @SDL_CFLAGS@ @EXTRA_CFLAGS@
