Index: libsdl2/src/joystick/linux/SDL_sysjoystick.c
===================================================================
--- a/src/joystick/linux/SDL_sysjoystick.c	2012-05-18 00:51:07.758390396 +0200
+++ b/src/joystick/linux/SDL_sysjoystick.c	2012-05-18 00:51:13.084998927 +0200
@@ -703,7 +703,7 @@
                 ++joystick->nbuttons;
             }
         }
-        for (i = 0; i < ABS_MISC; ++i) {
+        for (i = 0; i < ABS_MAX; ++i) {
             /* Skip hats */
             if (i == ABS_HAT0X) {
                 i = ABS_HAT3Y;
