Description: build the library as a shared library
Author: Scott Howard <showard@debian.org>

Index: mygui/Platforms/Ogre/OgrePlatform/CMakeLists.txt
===================================================================
--- mygui.orig/Platforms/Ogre/OgrePlatform/CMakeLists.txt	2013-01-29 19:59:02.325377418 -0500
+++ mygui/Platforms/Ogre/OgrePlatform/CMakeLists.txt	2013-01-29 19:59:29.921378074 -0500
@@ -8,11 +8,12 @@
 
 include(${PROJECTNAME}.list)
 
-add_library(${PROJECTNAME} ${HEADER_FILES} ${SOURCE_FILES})
+add_library(${PROJECTNAME} SHARED ${HEADER_FILES} ${SOURCE_FILES})
+set_target_properties(${PROJECTNAME} PROPERTIES VERSION 0debian0.0.0 SOVERSION 0debian0)
 
 add_dependencies(${PROJECTNAME} MyGUIEngine)
 
-target_link_libraries(${PROJECTNAME} ${OGRE_LIBRARIES})
+target_link_libraries(${PROJECTNAME} ${OGRE_LIBRARIES} MyGUIEngine)
 link_directories(${OGRE_LIB_DIR})
 
 install(FILES ${HEADER_FILES}
Index: mygui/Plugins/CMakeLists.txt
===================================================================
--- mygui.orig/Plugins/CMakeLists.txt	2013-01-29 19:59:02.325377418 -0500
+++ mygui/Plugins/CMakeLists.txt	2013-01-29 19:59:29.921378074 -0500
@@ -1,7 +1,7 @@
 option(MYGUI_BUILD_HIKARI_PLUGIN "Build Plugin HikariWidget" FALSE)
 option(MYGUI_BUILD_BERKELIUM_PLUGIN "Build Plugin BerkeliumWidget" FALSE)
 
-add_subdirectory(Plugin_StrangeButton)
+#add_subdirectory(Plugin_StrangeButton)
 
 if (MYGUI_BUILD_HIKARI_PLUGIN)
 	if (WIN32)
Index: mygui/Platforms/OpenGL/OpenGLPlatform/CMakeLists.txt
===================================================================
--- mygui.orig/Platforms/OpenGL/OpenGLPlatform/CMakeLists.txt	2013-01-29 19:59:02.325377418 -0500
+++ mygui/Platforms/OpenGL/OpenGLPlatform/CMakeLists.txt	2013-01-29 19:59:29.921378074 -0500
@@ -9,14 +9,15 @@
 
 include(${PROJECTNAME}.list)
 
-add_definitions(-DGLEW_STATIC)
+#add_definitions(-DGLEW_STATIC)
 add_definitions(-DGL_GLEXT_PROTOTYPES)
 
-add_library(${PROJECTNAME} ${HEADER_FILES} ${SOURCE_FILES})
+add_library(${PROJECTNAME} SHARED ${HEADER_FILES} ${SOURCE_FILES})
+set_target_properties(${PROJECTNAME} PROPERTIES VERSION 0debian0.0.0 SOVERSION 0debian0)
 
 add_dependencies(${PROJECTNAME} MyGUIEngine)
 
-target_link_libraries(${PROJECTNAME} ${OPENGL_LIBRARIES} ${PNG_LIBRARIES} ${ZLIB_LIBRARIES})
+target_link_libraries(${PROJECTNAME} ${OPENGL_LIBRARIES} ${PNG_LIBRARIES} ${ZLIB_LIBRARIES} MyGUIEngine GLEW)
 link_directories(${OPENGL_LIB_DIR} ${PNG_LIBRARY})
 
 # installation rules
Index: mygui/Platforms/OpenGL/OpenGLPlatform/MyGUI.OpenGLPlatform.list
===================================================================
--- mygui.orig/Platforms/OpenGL/OpenGLPlatform/MyGUI.OpenGLPlatform.list	2013-01-29 19:59:02.325377418 -0500
+++ mygui/Platforms/OpenGL/OpenGLPlatform/MyGUI.OpenGLPlatform.list	2013-01-29 19:59:29.925378074 -0500
@@ -8,7 +8,6 @@
   include/MyGUI_OpenGLVertexBuffer.h
 )
 set (SOURCE_FILES
-  include/GL/glew.c
   src/MyGUI_OpenGLDataManager.cpp
   src/MyGUI_OpenGLPlatform.cpp
   src/MyGUI_OpenGLRenderManager.cpp
@@ -17,7 +16,6 @@
   src/MyGUI_OpenGLVertexBuffer.cpp
 )
 SOURCE_GROUP("Source Files" FILES
-  include/GL/glew.c
   src/MyGUI_OpenGLDataManager.cpp
   src/MyGUI_OpenGLPlatform.cpp
   src/MyGUI_OpenGLRenderManager.cpp
Index: mygui/MyGUIEngine/CMakeLists.txt
===================================================================
--- mygui.orig/MyGUIEngine/CMakeLists.txt	2013-01-29 19:59:26.089377984 -0500
+++ mygui/MyGUIEngine/CMakeLists.txt	2013-01-29 19:59:52.945378621 -0500
@@ -45,6 +45,7 @@
 # setup MyGUIEngine target
 add_library(${PROJECTNAME} ${MYGUI_LIB_TYPE} ${HEADER_FILES} ${SOURCE_FILES})
 set_target_properties(${PROJECTNAME} PROPERTIES VERSION ${MYGUI_VERSION} SOVERSION ${MYGUI_VERSION_MAJOR})
+target_link_libraries(${PROJECTNAME} dl)
 if (MYGUI_USE_FREETYPE)
 	target_link_libraries(${PROJECTNAME}
 		${FREETYPE_LIBRARIES}
