---
 ports/sysdeps/arm/shlib-versions |    1 +
 1 file changed, 1 insertion(+)

--- a/ports/sysdeps/arm/shlib-versions
+++ b/ports/sysdeps/arm/shlib-versions
@@ -1,5 +1,6 @@
 arm.*-.*-linux-gnueabi.*	DEFAULT			GLIBC_2.4
 
+arm.*-.*-linux-gnueabihf	ld=ld-linux-armhf.so.3
 %ifdef HAVE_ARM_PCS_VFP
 # The EABI-derived hard-float ABI uses a new dynamic linker.
 arm.*-.*-linux-gnueabi.*       ld=ld-linux-armhf.so.3
