---
 harness/cases/12.t |    2 +-
 harness/cases/14.t |    2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

--- a/harness/cases/12.t
+++ b/harness/cases/12.t
@@ -35,7 +35,7 @@ int test_main(void)
 	if (pid == 0)
 		test_child();
 
-	res = waitpid(pid, &status, 0);
+	res = waitpid(pid, &status, 0);		assert(res != -1);
 
 	if (WIFEXITED(status)) {
 		int failed = (WEXITSTATUS(status) != 0);
--- a/harness/cases/14.t
+++ b/harness/cases/14.t
@@ -75,7 +75,7 @@ int test_main(void)
 	if (pid == 0)
 		test_child();
 
-	res = waitpid(pid, &status, 0);
+	res = waitpid(pid, &status, 0);		assert(res != -1);
 
 	if (WIFEXITED(status)) {
 		int failed = (WEXITSTATUS(status) != 0);
