---
 harness/Makefile |    7 +++++--
 src/Makefile     |   14 ++++++++------
 2 files changed, 13 insertions(+), 8 deletions(-)

--- a/harness/Makefile
+++ b/harness/Makefile
@@ -6,13 +6,16 @@ PROGS:=$(PARTPROGS) $(EXTRAPROGS)
 HARNESS_SRCS:=main.c
 # io_queue.c
 
-CFLAGS=-Wall -I../src -g -O
+BASE_CPPFLAGS := -I../src $(CPPFLAGS)
+CFLAGS = -Wall -g -O
 #-lpthread -lrt
+BASE_CFLAGS = $(CFLAGS)
+BASE_LDFLAGS = ../src/libaio.a $(LDFLAGS)
 
 all: $(PROGS)
 
 $(PROGS): %.p: %.t $(HARNESS_SRCS)
-	$(CC) $(CFLAGS) -DTEST_NAME=\"$<\" -o $@ main.c ../src/libaio.a
+	$(CC) $(BASE_CPPFLAGS) $(BASE_CFLAGS) -DTEST_NAME=\"$<\" -o $@ main.c $(BASE_LDFLAGS)
 
 clean:
 	rm -f $(PROGS) *.o runtests.out rofile wofile rwfile
--- a/src/Makefile
+++ b/src/Makefile
@@ -5,10 +5,12 @@ libdir=$(prefix)/lib
 libdevdir=$(prefix)/lib
 
 ARCH := $(shell uname -m | sed -e s/i.86/i386/)
-CFLAGS := -nostdlib -nostartfiles -Wall -I. -g -fomit-frame-pointer -O2 -fPIC
-SO_CFLAGS=-shared $(CFLAGS)
-L_CFLAGS=$(CFLAGS)
-LINK_FLAGS := -lgcc
+BASE_CFLAGS := -nostdlib -nostartfiles -fomit-frame-pointer -fPIC $(CFLAGS)
+BASE_CPPFLAGS := -I. $(CPPFLAGS)
+CFLAGS := -Wall -g -O2
+SO_CFLAGS = -shared $(BASE_CFLAGS)
+L_CFLAGS = $(BASE_CFLAGS)
+LINK_FLAGS := -lgcc $(LDFLAGS)
 
 soname=libaio.so.1
 minor=0
@@ -38,10 +40,10 @@ libaio_sobjs := $(patsubst %.c,%.os,$(li
 $(libaio_objs) $(libaio_sobjs): libaio.h vsys_def.h
 
 %.os: %.c
-	$(CC) $(SO_CFLAGS) -c -o $@ $<
+	$(CC) $(BASE_CPPFLAGS) $(SO_CFLAGS) -c -o $@ $<
 
 %.ol: %.c
-	$(CC) $(L_CFLAGS) -c -o $@ $<
+	$(CC) $(BASE_CPPFLAGS) $(L_CFLAGS) -c -o $@ $<
 
 
 libaio.a: $(libaio_objs)
