Description: Enable indicator support by default on Ubuntu.
Author: Andrew Starr-Bochicchio <a.starr.b@gmail.com>
Forwarded: not-needed
Bug-Ubuntu: https://bugs.launchpad.net/bugs/843805

--- a/deluge/ui/gtkui/gtkui.py
+++ b/deluge/ui/gtkui/gtkui.py
@@ -115,7 +115,7 @@
     "enable_system_tray": True,
     "close_to_tray": False,
     "start_in_tray": False,
-    "enable_appindicator": False,
+    "enable_appindicator": True,
     "lock_tray": False,
     "tray_password": "",
     "check_new_releases": False,
