--- ./utils/chemdiff/chemdiff.sh.orig	2012-03-22 23:23:05.231581376 +0100
+++ ./utils/chemdiff/chemdiff.sh	2012-03-22 23:23:11.907576706 +0100
@@ -1,11 +1,5 @@
-#!/bin/sh
+#!/bin/bash
 
 JAVA_OPTS="-Xss10m -Dawt.useSystemAAFontSettings=on"
 
-#Following symlinks
-path=$0
-while [ -h "$path" ]; do
-    path=$(ls -ld $path | sed -r 's/^.*->.//')
-done
-
-java $JAVA_OPTS -jar $(dirname $path)/chemdiff.jar "$@"
+java $JAVA_OPTS -jar /usr/share/java/chemdiff.jar "$@"
--- ./legio/legio.sh.orig	2012-03-22 23:27:03.179713313 +0100
+++ ./legio/legio.sh	2012-03-22 23:27:36.023581692 +0100
@@ -1,11 +1,5 @@
-#!/bin/sh
+#!/bin/bash
 
 JAVA_OPTS="-Xss10m -Dawt.useSystemAAFontSettings=on"
 
-#Following symlinks
-path=$0
-while [ -h "$path" ]; do
-    path=$(ls -ld $path | sed -r 's/^.*->.//')
-done
-
-java $JAVA_OPTS -jar $(dirname $path)/legio.jar "$@"
+java $JAVA_OPTS -jar /usr/share/java/legio.jar "$@"
