--- ./reaction/src/reaction_cml_loader.cpp.orig	2012-03-22 02:31:57.703147210 +0100
+++ ./reaction/src/reaction_cml_loader.cpp	2012-03-22 02:32:10.811157325 +0100
@@ -16,6 +16,7 @@
 #include "reaction/reaction.h"
 #include "base_cpp/tlscont.h"
 #include "base_cpp/scanner.h"
+#define TIXML_USE_STL
 #include "tinyxml.h"
 #include "molecule/molecule_cml_loader.h"
 
--- ./molecule/src/molecule_cml_loader.cpp.orig	2012-03-22 02:32:28.055149859 +0100
+++ ./molecule/src/molecule_cml_loader.cpp	2012-03-22 02:32:39.411147707 +0100
@@ -15,6 +15,7 @@
 #include "molecule/molecule_cml_loader.h"
 #include "base_cpp/scanner.h"
 #include "molecule/molecule.h"
+#define TIXML_USE_STL
 #include "tinyxml.h"
 #include "molecule/elements.h"
 #include "molecule/molecule_scaffold_detection.h"
