From 0d4d467dd55ac23151a32e52195d47b6ab7706ca Mon Sep 17 00:00:00 2001
From: Alejandro Piñeiro <apinheiro@igalia.com>
Date: Mon, 18 Jun 2012 09:30:25 +0000
Subject: a11y: Remove key event listener hash table if no longer required

https://bugzilla.gnome.org/show_bug.cgi?id=675183
---
diff --git a/clutter/cally/cally-util.c b/clutter/cally/cally-util.c
index e4c90a7..1fcbf83 100644
--- a/clutter/cally/cally-util.c
+++ b/clutter/cally/cally-util.c
@@ -261,6 +261,8 @@ cally_util_remove_key_event_listener (guint remove_listener)
 
   if (g_hash_table_size (key_listener_list) == 0)
     {
+      g_hash_table_destroy (key_listener_list);
+      key_listener_list = NULL;
       cally_util_simulate_snooper_remove ();
     }
 }
--
cgit v0.9.0.2
