Fix nice granularity

---
 resource.h |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)
--- hurd/hurd/resource.h.orig	2012-05-12 17:23:56.000000000 +0000
+++ hurd/hurd/resource.h	2012-05-12 17:24:01.000000000 +0000
@@ -43,8 +43,8 @@
 
 /* Convert between Mach priority values and the priority
    values used by getpriority, setpriority, and nice.  */
-#define MACH_PRIORITY_TO_NICE(prio) (2 * ((prio) - 12))
-#define NICE_TO_MACH_PRIORITY(nice) (12 + ((nice) / 2))
+#define MACH_PRIORITY_TO_NICE(prio) ((prio) - 20)
+#define NICE_TO_MACH_PRIORITY(nice) (20 + (nice))
 
 
 
