From: Samuel Thibault <samuel.thibault@ens-lyon.org>
Subject: [PATCH] The critical section lock _can_ be held in that place.

---
 hurd/thread-cancel.c |    1 -
 1 file changed, 1 deletion(-)

--- a/hurd/thread-cancel.c
+++ b/hurd/thread-cancel.c
@@ -52,7 +52,6 @@
       return 0;
     }
 
-  assert (! __spin_lock_locked (&ss->critical_section_lock));
   __spin_lock (&ss->critical_section_lock);
   __spin_lock (&ss->lock);
   err = __thread_suspend (thread);
