diff --git a/mach/Makefile b/mach/Makefile
index 00de7a5..eaf0787 100644
--- a/mach/Makefile
+++ b/mach/Makefile
@@ -112,7 +112,7 @@ ifndef mach-shortcuts
 # $(mach-shortcuts) will be set, and that will change how
 # mach_interface.defs is processed: it will get the -D flags below.
 user-interfaces := $(filter-out $(mach-interface-list:%=mach/%) \
-				mach/mach_port mach/mach_host mach/mach4 \
+				mach/mach_port mach/mach_host mach/mach4 mach/gnumach \
 				device/device_request,\
 				$(user-interfaces))
 endif
diff --git a/sysdeps/mach/configure b/sysdeps/mach/configure
index b41449e..8b1b80b 100644
--- a/sysdeps/mach/configure
+++ b/sysdeps/mach/configure
@@ -411,7 +411,7 @@ if test $libc_cv_mach_task_creation_time = no; then
 fi
 
 mach_interface_list=
-for ifc in mach mach4 \
+for ifc in mach mach4 gnumach \
 	   clock clock_priv host_priv host_security ledger lock_set \
 	   processor processor_set task thread_act vm_map \
 	   memory_object memory_object_default default_pager \
diff --git a/sysdeps/mach/configure.in b/sysdeps/mach/configure.in
index 2f82c3a..101a884 100644
--- a/sysdeps/mach/configure.in
+++ b/sysdeps/mach/configure.in
@@ -53,7 +53,7 @@ dnl but we don't do a check for that here because in a bare
 dnl environment the compile against those headers will fail.
 dnl
 mach_interface_list=
-for ifc in mach mach4 \
+for ifc in mach mach4 gnumach \
 	   clock clock_priv host_priv host_security ledger lock_set \
 	   processor processor_set task thread_act vm_map \
 	   memory_object memory_object_default default_pager \
