--- a/src/libgcc/config/arm/unwind-arm.h~	2011-09-13 14:48:33.000000000 +0200
+++ b/src/libgcc/config/arm/unwind-arm.h	2012-12-09 15:49:26.541327118 +0100
@@ -39,7 +39,8 @@
 #endif
   /* Decode an R_ARM_TARGET2 relocation.  */
   static inline _Unwind_Word
-  _Unwind_decode_typeinfo_ptr (_Unwind_Word base, _Unwind_Word ptr)
+  _Unwind_decode_typeinfo_ptr (_Unwind_Word base __attribute__ ((unused)),
+                               _Unwind_Word ptr)
     {
       _Unwind_Word tmp;
 
@@ -65,7 +66,9 @@
     }
 
   static inline _Unwind_Reason_Code
-  __gnu_unwind_24bit (_Unwind_Context * context, _uw data, int compact)
+  __gnu_unwind_24bit (_Unwind_Context * context __attribute__ ((unused)),
+                      _uw data __attribute__ ((unused)),
+                      int compact __attribute__ ((unused)))
     {
       return _URC_FAILURE;
     }
