--- a/src/libmudflap/Makefile.in	2012-12-08 08:32:41.301881153 +0100
+++ b/src/libmudflap/Makefile.in	2012-12-08 08:58:29.281874520 +0100
@@ -269,7 +269,7 @@
 @LIBMUDFLAPTH_FALSE@libmudflapth = 
 @LIBMUDFLAPTH_TRUE@libmudflapth = libmudflapth.la
 toolexeclib_LTLIBRARIES = libmudflap.la $(libmudflapth)
-libsubincludedir = $(libdir)/gcc/$(target_noncanonical)/$(gcc_version)/include
+libsubincludedir = $(libdir)/gcc-cross/$(target_noncanonical)/$(gcc_version)/include
 nobase_libsubinclude_HEADERS = mf-runtime.h
 libmudflap_la_SOURCES = \
 	mf-runtime.c \
--- a/src/libmudflap/Makefile.am	2012-12-08 08:32:41.301881153 +0100
+++ b/src/libmudflap/Makefile.am	2012-12-08 08:58:04.633876182 +0100
@@ -23,7 +23,7 @@
 
 toolexeclib_LTLIBRARIES = libmudflap.la $(libmudflapth)
 target_noncanonical = @target_noncanonical@
-libsubincludedir = $(libdir)/gcc/$(target_noncanonical)/$(gcc_version)/include
+libsubincludedir = $(libdir)/gcc-cross/$(target_noncanonical)/$(gcc_version)/include
 nobase_libsubinclude_HEADERS = mf-runtime.h
 
 
--- a/src/fixincludes/Makefile.in	2011-01-03 21:52:22.000000000 +0100
+++ b/src/fixincludes/Makefile.in	2012-12-08 08:53:27.029874709 +0100
@@ -52,9 +52,9 @@
 gcc_version := $(shell cat $(srcdir)/../gcc/BASE-VER)
 
 # Directory in which the compiler finds libraries etc.
-libsubdir = $(libdir)/gcc/$(target_noncanonical)/$(gcc_version)
+libsubdir = $(libdir)/gcc-cross/$(target_noncanonical)/$(gcc_version)
 # Directory in which the compiler finds executables
-libexecsubdir = $(libexecdir)/gcc/$(target_noncanonical)/$(gcc_version)
+libexecsubdir = $(libexecdir)/gcc-cross/$(target_noncanonical)/$(gcc_version)
 # Where our executable files go
 itoolsdir = $(libexecsubdir)/install-tools
 # Where our data files go
--- a/src/libgfortran/Makefile.in	2012-09-20 09:23:55.000000000 +0200
+++ b/src/libgfortran/Makefile.in	2012-12-08 08:50:26.369874316 +0100
@@ -499,12 +499,12 @@
 
 libgfortran_la_DEPENDENCIES = $(version_dep) libgfortran.spec $(LIBQUADLIB_DEP)
 myexeclib_LTLIBRARIES = libgfortranbegin.la
-myexeclibdir = $(libdir)/gcc/$(target_alias)/$(gcc_version)$(MULTISUBDIR)
+myexeclibdir = $(libdir)/gcc-cross/$(target_alias)/$(gcc_version)$(MULTISUBDIR)
 libgfortranbegin_la_SOURCES = fmain.c
 libgfortranbegin_la_LDFLAGS = -static
 libgfortranbegin_la_LINK = $(LINK) $(libgfortranbegin_la_LDFLAGS)
 cafexeclib_LTLIBRARIES = libcaf_single.la
-cafexeclibdir = $(libdir)/gcc/$(target_alias)/$(gcc_version)$(MULTISUBDIR)
+cafexeclibdir = $(libdir)/gcc-cross/$(target_alias)/$(gcc_version)$(MULTISUBDIR)
 libcaf_single_la_SOURCES = caf/single.c
 libcaf_single_la_LDFLAGS = -static
 libcaf_single_la_DEPENDENCIES = caf/libcaf.h
--- a/src/libgfortran/Makefile.am	2012-01-09 17:02:36.000000000 +0100
+++ b/src/libgfortran/Makefile.am	2012-12-08 08:49:41.957876998 +0100
@@ -42,13 +42,13 @@
 libgfortran_la_DEPENDENCIES = $(version_dep) libgfortran.spec $(LIBQUADLIB_DEP)
 
 myexeclib_LTLIBRARIES = libgfortranbegin.la
-myexeclibdir = $(libdir)/gcc/$(target_alias)/$(gcc_version)$(MULTISUBDIR)
+myexeclibdir = $(libdir)/gcc-cross/$(target_alias)/$(gcc_version)$(MULTISUBDIR)
 libgfortranbegin_la_SOURCES = fmain.c
 libgfortranbegin_la_LDFLAGS = -static
 libgfortranbegin_la_LINK = $(LINK) $(libgfortranbegin_la_LDFLAGS)
 
 cafexeclib_LTLIBRARIES = libcaf_single.la
-cafexeclibdir = $(libdir)/gcc/$(target_alias)/$(gcc_version)$(MULTISUBDIR)
+cafexeclibdir = $(libdir)/gcc-cross/$(target_alias)/$(gcc_version)$(MULTISUBDIR)
 libcaf_single_la_SOURCES = caf/single.c
 libcaf_single_la_LDFLAGS = -static
 libcaf_single_la_DEPENDENCIES = caf/libcaf.h
--- a/src/lto-plugin/Makefile.in	2011-08-10 10:48:37.000000000 +0200
+++ b/src/lto-plugin/Makefile.in	2012-12-08 09:00:17.861873944 +0100
@@ -227,7 +227,7 @@
 ACLOCAL_AMFLAGS = -I .. -I ../config
 AUTOMAKE_OPTIONS = no-dependencies
 gcc_version := $(shell cat $(top_srcdir)/../gcc/BASE-VER)
-libexecsubdir := $(libexecdir)/gcc/$(target_noncanonical)/$(gcc_version)
+libexecsubdir := $(libexecdir)/gcc-cross/$(target_noncanonical)/$(gcc_version)
 AM_CPPFLAGS = -I$(top_srcdir)/../include $(DEFS)
 AM_CFLAGS = @ac_lto_plugin_warn_cflags@
 AM_LIBTOOLFLAGS = --tag=disable-static
--- a/src/lto-plugin/Makefile.am	2011-08-10 10:48:37.000000000 +0200
+++ b/src/lto-plugin/Makefile.am	2012-12-08 08:59:54.621875067 +0100
@@ -5,7 +5,7 @@
 
 gcc_version := $(shell cat $(top_srcdir)/../gcc/BASE-VER)
 target_noncanonical := @target_noncanonical@
-libexecsubdir := $(libexecdir)/gcc/$(target_noncanonical)/$(gcc_version)
+libexecsubdir := $(libexecdir)/gcc-cross/$(target_noncanonical)/$(gcc_version)
 
 AM_CPPFLAGS = -I$(top_srcdir)/../include $(DEFS)
 AM_CFLAGS = @ac_lto_plugin_warn_cflags@
--- a/src/libitm/Makefile.in	2012-12-08 08:32:40.093881158 +0100
+++ b/src/libitm/Makefile.in	2012-12-08 08:54:51.929875619 +0100
@@ -306,8 +306,8 @@
 gcc_version := $(shell cat $(top_srcdir)/../gcc/BASE-VER)
 abi_version = -fabi-version=4
 search_path = $(addprefix $(top_srcdir)/config/, $(config_path)) $(top_srcdir)
-fincludedir = $(libdir)/gcc/$(target_alias)/$(gcc_version)/finclude
-libsubincludedir = $(libdir)/gcc/$(target_alias)/$(gcc_version)/include
+fincludedir = $(libdir)/gcc-cross/$(target_alias)/$(gcc_version)/finclude
+libsubincludedir = $(libdir)/gcc-cross/$(target_alias)/$(gcc_version)/include
 AM_CPPFLAGS = $(addprefix -I, $(search_path))
 AM_CFLAGS = $(XCFLAGS)
 AM_CXXFLAGS = $(XCFLAGS) -std=gnu++0x -funwind-tables -fno-exceptions \
--- a/src/libitm/Makefile.am	2012-02-14 14:14:27.000000000 +0100
+++ b/src/libitm/Makefile.am	2012-12-08 08:53:58.341873782 +0100
@@ -11,8 +11,8 @@
 config_path = @config_path@
 search_path = $(addprefix $(top_srcdir)/config/, $(config_path)) $(top_srcdir)
 
-fincludedir = $(libdir)/gcc/$(target_alias)/$(gcc_version)/finclude
-libsubincludedir = $(libdir)/gcc/$(target_alias)/$(gcc_version)/include
+fincludedir = $(libdir)/gcc-cross/$(target_alias)/$(gcc_version)/finclude
+libsubincludedir = $(libdir)/gcc-cross/$(target_alias)/$(gcc_version)/include
 
 vpath % $(strip $(search_path))
 
--- a/src/gcc/gcc.c	2012-08-06 16:34:27.000000000 +0200
+++ b/src/gcc/gcc.c	2012-12-08 08:42:06.353877392 +0100
@@ -3621,7 +3621,7 @@
 	 GCC_EXEC_PREFIX is typically a directory name with a trailing
 	 / (which is ignored by make_relative_prefix), so append a
 	 program name.  */
-      char *tmp_prefix = concat (gcc_exec_prefix, "gcc", NULL);
+      char *tmp_prefix = concat (gcc_exec_prefix, "gcc-cross", NULL);
       gcc_libexec_prefix = get_relative_prefix (tmp_prefix,
 						standard_exec_prefix,
 						standard_libexec_prefix);
@@ -3647,15 +3647,15 @@
     {
       int len = strlen (gcc_exec_prefix);
 
-      if (len > (int) sizeof ("/lib/gcc/") - 1
+      if (len > (int) sizeof ("/lib/gcc-cross/") - 1
 	  && (IS_DIR_SEPARATOR (gcc_exec_prefix[len-1])))
 	{
-	  temp = gcc_exec_prefix + len - sizeof ("/lib/gcc/") + 1;
+	  temp = gcc_exec_prefix + len - sizeof ("/lib/gcc-cross/") + 1;
 	  if (IS_DIR_SEPARATOR (*temp)
 	      && filename_ncmp (temp + 1, "lib", 3) == 0
 	      && IS_DIR_SEPARATOR (temp[4])
-	      && filename_ncmp (temp + 5, "gcc", 3) == 0)
-	    len -= sizeof ("/lib/gcc/") - 1;
+	      && filename_ncmp (temp + 5, "gcc-cross", 3) == 0)
+	    len -= sizeof ("/lib/gcc-cross/") - 1;
 	}
 
       set_std_prefix (gcc_exec_prefix, len);
--- a/src/gcc/Makefile.in	2012-12-08 08:32:41.337881153 +0100
+++ b/src/gcc/Makefile.in	2012-12-08 08:36:18.493883559 +0100
@@ -566,9 +566,9 @@
 # --------
 
 # Directory in which the compiler finds libraries etc.
-libsubdir = $(libdir)/gcc/$(target_noncanonical)/$(version)
+libsubdir = $(libdir)/gcc-cross/$(target_noncanonical)/$(version)
 # Directory in which the compiler finds executables
-libexecsubdir = $(libexecdir)/gcc/$(target_noncanonical)/$(version)
+libexecsubdir = $(libexecdir)/gcc-cross/$(target_noncanonical)/$(version)
 # Directory in which all plugin resources are installed
 plugin_resourcesdir = $(libsubdir)/plugin
  # Directory in which plugin headers are installed
@@ -2079,8 +2079,8 @@
 
 DRIVER_DEFINES = \
   -DSTANDARD_STARTFILE_PREFIX=\"$(unlibsubdir)/\" \
-  -DSTANDARD_EXEC_PREFIX=\"$(libdir)/gcc/\" \
-  -DSTANDARD_LIBEXEC_PREFIX=\"$(libexecdir)/gcc/\" \
+  -DSTANDARD_EXEC_PREFIX=\"$(libdir)/gcc-cross/\" \
+  -DSTANDARD_LIBEXEC_PREFIX=\"$(libexecdir)/gcc-cross/\" \
   -DDEFAULT_TARGET_VERSION=\"$(version)\" \
   -DDEFAULT_TARGET_MACHINE=\"$(target_noncanonical)\" \
   -DSTANDARD_BINDIR_PREFIX=\"$(bindir)/\" \
@@ -3980,7 +3980,7 @@
   -DTOOL_INCLUDE_DIR=\"$(gcc_tooldir)/include\" \
   -DNATIVE_SYSTEM_HEADER_DIR=\"$(NATIVE_SYSTEM_HEADER_DIR)\" \
   -DPREFIX=\"$(prefix)/\" \
-  -DSTANDARD_EXEC_PREFIX=\"$(libdir)/gcc/\" \
+  -DSTANDARD_EXEC_PREFIX=\"$(libdir)/gcc-cross/\" \
   @TARGET_SYSTEM_ROOT_DEFINE@
 
 CFLAGS-cppbuiltin.o += $(PREPROCESSOR_DEFINES) -DBASEVER=$(FULLVER_s)
--- a/src/libssp/Makefile.in	2011-02-13 12:45:53.000000000 +0100
+++ b/src/libssp/Makefile.in	2012-12-08 08:59:07.469875025 +0100
@@ -259,7 +259,7 @@
 @LIBSSP_USE_SYMVER_SUN_TRUE@@LIBSSP_USE_SYMVER_TRUE@version_dep = ssp.map-sun
 AM_CFLAGS = -Wall
 toolexeclib_LTLIBRARIES = libssp.la libssp_nonshared.la
-libsubincludedir = $(libdir)/gcc/$(target_noncanonical)/$(gcc_version)/include
+libsubincludedir = $(libdir)/gcc-cross/$(target_noncanonical)/$(gcc_version)/include
 nobase_libsubinclude_HEADERS = ssp/ssp.h ssp/string.h ssp/stdio.h ssp/unistd.h
 libssp_la_SOURCES = \
 	ssp.c gets-chk.c memcpy-chk.c memmove-chk.c mempcpy-chk.c \
--- a/src/libssp/Makefile.am	2010-12-06 01:50:04.000000000 +0100
+++ b/src/libssp/Makefile.am	2012-12-08 08:58:51.241873553 +0100
@@ -39,7 +39,7 @@
 toolexeclib_LTLIBRARIES = libssp.la libssp_nonshared.la
 
 target_noncanonical = @target_noncanonical@
-libsubincludedir = $(libdir)/gcc/$(target_noncanonical)/$(gcc_version)/include
+libsubincludedir = $(libdir)/gcc-cross/$(target_noncanonical)/$(gcc_version)/include
 nobase_libsubinclude_HEADERS = ssp/ssp.h ssp/string.h ssp/stdio.h ssp/unistd.h
 
 libssp_la_SOURCES = \
--- a/src/libquadmath/Makefile.in	2011-09-21 16:36:03.000000000 +0200
+++ b/src/libquadmath/Makefile.in	2012-12-08 08:49:10.557875680 +0100
@@ -319,7 +319,7 @@
 
 @BUILD_LIBQUADMATH_TRUE@libquadmath_la_DEPENDENCIES = $(version_dep) $(libquadmath_la_LIBADD)
 @BUILD_LIBQUADMATH_TRUE@nodist_libsubinclude_HEADERS = quadmath.h quadmath_weak.h
-@BUILD_LIBQUADMATH_TRUE@libsubincludedir = $(libdir)/gcc/$(target_alias)/$(gcc_version)/include
+@BUILD_LIBQUADMATH_TRUE@libsubincludedir = $(libdir)/gcc-cross/$(target_alias)/$(gcc_version)/include
 @BUILD_LIBQUADMATH_TRUE@libquadmath_la_SOURCES = \
 @BUILD_LIBQUADMATH_TRUE@  math/acoshq.c math/fmodq.c math/acosq.c math/frexpq.c \
 @BUILD_LIBQUADMATH_TRUE@  math/rem_pio2q.c math/asinhq.c math/hypotq.c math/remainderq.c \
--- a/src/libquadmath/Makefile.am	2011-09-21 16:36:03.000000000 +0200
+++ b/src/libquadmath/Makefile.am	2012-12-08 08:48:25.553878276 +0100
@@ -40,7 +40,7 @@
 libquadmath_la_DEPENDENCIES = $(version_dep) $(libquadmath_la_LIBADD)
 
 nodist_libsubinclude_HEADERS = quadmath.h quadmath_weak.h
-libsubincludedir = $(libdir)/gcc/$(target_alias)/$(gcc_version)/include
+libsubincludedir = $(libdir)/gcc-cross/$(target_alias)/$(gcc_version)/include
 
 libquadmath_la_SOURCES = \
   math/acoshq.c math/fmodq.c math/acosq.c math/frexpq.c \
--- a/src/libobjc/Makefile.in	2011-11-02 16:28:43.000000000 +0100
+++ b/src/libobjc/Makefile.in	2012-12-08 08:50:47.241873110 +0100
@@ -51,7 +51,7 @@
 -include ../boehm-gc/threads.mk
 
 libdir = $(exec_prefix)/lib
-libsubdir = $(libdir)/gcc/$(target_noncanonical)/$(gcc_version)
+libsubdir = $(libdir)/gcc-cross/$(target_noncanonical)/$(gcc_version)
 
 # Multilib support variables.
 MULTISRCTOP =
--- a/src/libada/Makefile.in	2012-08-06 16:34:27.000000000 +0200
+++ b/src/libada/Makefile.in	2012-12-08 08:53:01.321876031 +0100
@@ -62,7 +62,7 @@
 
 target_noncanonical:=@target_noncanonical@
 version := $(shell cat $(srcdir)/../gcc/BASE-VER)
-libsubdir := $(libdir)/gcc/$(target_noncanonical)/$(version)$(MULTISUBDIR)
+libsubdir := $(libdir)/gcc-cross/$(target_noncanonical)/$(version)$(MULTISUBDIR)
 ADA_RTS_DIR=$(GCC_DIR)/ada/rts$(subst /,_,$(MULTISUBDIR))
 ADA_RTS_SUBDIR=./rts$(subst /,_,$(MULTISUBDIR))
 
--- a/src/libgomp/Makefile.in	2012-09-20 09:23:55.000000000 +0200
+++ b/src/libgomp/Makefile.in	2012-12-08 08:45:32.157878288 +0100
@@ -291,8 +291,8 @@
 SUBDIRS = testsuite
 gcc_version := $(shell cat $(top_srcdir)/../gcc/BASE-VER)
 search_path = $(addprefix $(top_srcdir)/config/, $(config_path)) $(top_srcdir)
-fincludedir = $(libdir)/gcc/$(target_alias)/$(gcc_version)/finclude
-libsubincludedir = $(libdir)/gcc/$(target_alias)/$(gcc_version)/include
+fincludedir = $(libdir)/gcc-cross/$(target_alias)/$(gcc_version)/finclude
+libsubincludedir = $(libdir)/gcc-cross/$(target_alias)/$(gcc_version)/include
 AM_CPPFLAGS = $(addprefix -I, $(search_path))
 AM_CFLAGS = $(XCFLAGS)
 AM_LDFLAGS = $(XLDFLAGS) $(SECTION_LDFLAGS) $(OPT_LDFLAGS)
--- a/src/libgomp/Makefile.am	2012-02-27 14:51:50.000000000 +0100
+++ b/src/libgomp/Makefile.am	2012-12-08 08:44:48.913867574 +0100
@@ -9,8 +9,8 @@
 config_path = @config_path@
 search_path = $(addprefix $(top_srcdir)/config/, $(config_path)) $(top_srcdir)
 
-fincludedir = $(libdir)/gcc/$(target_alias)/$(gcc_version)/finclude
-libsubincludedir = $(libdir)/gcc/$(target_alias)/$(gcc_version)/include
+fincludedir = $(libdir)/gcc-cross/$(target_alias)/$(gcc_version)/finclude
+libsubincludedir = $(libdir)/gcc-cross/$(target_alias)/$(gcc_version)/include
 
 vpath % $(strip $(search_path))
 
--- a/src/libgcc/Makefile.in	2012-12-08 08:32:41.249881153 +0100
+++ b/src/libgcc/Makefile.in	2012-12-08 08:43:50.201879083 +0100
@@ -178,7 +178,7 @@
 STRIP_FOR_TARGET = $(STRIP)
 
 # Directory in which the compiler finds libraries etc.
-libsubdir = $(libdir)/gcc/$(host_noncanonical)/$(version)
+libsubdir = $(libdir)/gcc-cross/$(host_noncanonical)/$(version)
 # Used to install the shared libgcc.
 slibdir = @slibdir@
 # Maybe used for DLLs on Windows targets.
--- a/src/libjava/Makefile.in	2012-12-08 08:32:41.249881153 +0100
+++ b/src/libjava/Makefile.in	2012-12-08 08:51:43.365881984 +0100
@@ -785,8 +785,8 @@
 
 
 # This is required by TL_AC_GXX_INCLUDE_DIR.
-libsubdir = $(libdir)/gcc/$(target_noncanonical)/$(gcc_version)
-libexecsubdir = $(libexecdir)/gcc/$(target_noncanonical)/$(gcc_version)
+libsubdir = $(libdir)/gcc-cross/$(target_noncanonical)/$(gcc_version)
+libexecsubdir = $(libexecdir)/gcc-cross/$(target_noncanonical)/$(gcc_version)
 toolexeclib_LTLIBRARIES = libgcj.la libgij.la libgcj-tools.la \
 	$(am__append_2) $(am__append_3) $(am__append_4)
 toolexecmainlib_DATA = libgcj.spec
--- a/src/libjava/Makefile.am	2012-12-08 08:32:41.241881153 +0100
+++ b/src/libjava/Makefile.am	2012-12-08 08:51:13.481876463 +0100
@@ -34,9 +34,9 @@
 target_noncanonical = @target_noncanonical@
 
 # This is required by TL_AC_GXX_INCLUDE_DIR.
-libsubdir = $(libdir)/gcc/$(target_noncanonical)/$(gcc_version)
+libsubdir = $(libdir)/gcc-cross/$(target_noncanonical)/$(gcc_version)
 
-libexecsubdir = $(libexecdir)/gcc/$(target_noncanonical)/$(gcc_version)
+libexecsubdir = $(libexecdir)/gcc-cross/$(target_noncanonical)/$(gcc_version)
 
 ##
 ## What gets installed, and where.
--- a/src/libffi/include/Makefile.am	2006-09-12 18:51:43.000000000 +0200
+++ b/src/libffi/include/Makefile.am	2012-12-08 09:42:12.313863513 +0100
@@ -7,6 +7,6 @@
 
 # Where generated headers like ffitarget.h get installed.
 gcc_version   := $(shell cat $(top_srcdir)/../gcc/BASE-VER)
-toollibffidir := $(libdir)/gcc/$(target_alias)/$(gcc_version)/include
+toollibffidir := $(libdir)/gcc-cross/$(target_alias)/$(gcc_version)/include
 
 toollibffi_HEADERS = ffi.h ffitarget.h
--- a/src/libffi/include/Makefile.in	2012-12-08 09:12:36.913870891 +0100
+++ b/src/libffi/include/Makefile.in	2012-12-08 09:42:24.901862621 +0100
@@ -213,7 +213,7 @@
 
 # Where generated headers like ffitarget.h get installed.
 gcc_version := $(shell cat $(top_srcdir)/../gcc/BASE-VER)
-toollibffidir := $(libdir)/gcc/$(target_alias)/$(gcc_version)/include
+toollibffidir := $(libdir)/gcc-cross/$(target_alias)/$(gcc_version)/include
 toollibffi_HEADERS = ffi.h ffitarget.h
 all: all-am
 
