# DP: libffi:  Define FFI_MMAP_EXEC_WRIT on kfreebsd-*.

Index: b/src/libffi/configure.ac
===================================================================
--- a/src/libffi/configure.ac
+++ b/src/libffi/configure.ac
@@ -317,7 +317,7 @@
 fi
 
 case "$target" in
-     *-apple-darwin10* | *-*-freebsd* | *-*-openbsd* | *-pc-solaris*)
+     *-apple-darwin10* | *-*-freebsd* | *-*-kfreebsd* | *-*-openbsd* | *-pc-solaris*)
        AC_DEFINE(FFI_MMAP_EXEC_WRIT, 1,
                  [Cannot use malloc on this target, so, we revert to
                    alternative means])
