# DP: PR libffi/47248, force a read only eh frame section.

Index: b/src/libffi/configure.ac
===================================================================
--- a/src/libffi/configure.ac
+++ b/src/libffi/configure.ac
@@ -351,6 +351,8 @@
 		libffi_cv_ro_eh_frame=yes
 	    fi
 	fi
+	# FIXME: see PR libffi/47248
+	libffi_cv_ro_eh_frame=yes
 	rm -f conftest.*
     ])
 if test "x$libffi_cv_ro_eh_frame" = xyes; then
