# DP: Install libjava libraries to multiarch location

Index: b/src/libjava/configure.ac
===================================================================
--- a/src/libjava/configure.ac
+++ b/src/libjava/configure.ac
@@ -1603,6 +1603,10 @@
       .) toolexeclibdir=$toolexecmainlibdir ;; # Avoid trailing /.
       *) toolexeclibdir=$toolexecmainlibdir/$multi_os_directory ;;
     esac
+    multiarch=`$CC -print-multiarch`
+    if test -n "$multiarch"; then
+        toolexeclibdir=$toolexecmainlibdir/$multiarch
+    fi
     ;;
 esac
 AC_SUBST(toolexecdir)
@@ -1628,6 +1632,10 @@
    dbexecdir='$(libdir)/'$multi_os_directory/$gcjsubdir
    ;;
 esac
+multiarch=`$CC -print-multiarch`
+if test -n "$multiarch"; then
+    dbexecdir='$(libdir)/'$multiarch/$gcjsubdir
+fi
 AC_SUBST(dbexecdir)
 AC_SUBST(gcjsubdir)
 
Index: b/src/libjava/Makefile.am
===================================================================
--- a/src/libjava/Makefile.am
+++ b/src/libjava/Makefile.am
@@ -371,7 +371,7 @@
 	-DGCJ_VERSIONED_LIBDIR="\"$(dbexecdir)\"" \
 	-DPATH_SEPARATOR="\"$(CLASSPATH_SEPARATOR)\"" \
 	-DECJ_JAR_FILE="\"$(ECJ_JAR)\"" \
-	-DLIBGCJ_DEFAULT_DATABASE="\"$(dbexecdir)/$(db_name)\"" \
+	-DLIBGCJ_DEFAULT_DATABASE="\"/var/lib/$(MULTIARCH_DIR)/gcj-4.7/$(db_name)\"" \
 	-DLIBGCJ_DEFAULT_DATABASE_PATH_TAIL="\"$(db_pathtail)\""
 
 AM_GCJFLAGS = \
Index: b/src/libjava/Makefile.in
===================================================================
--- a/src/libjava/Makefile.in
+++ b/src/libjava/Makefile.in
@@ -1019,7 +1019,7 @@
 	-DGCJ_VERSIONED_LIBDIR="\"$(dbexecdir)\"" \
 	-DPATH_SEPARATOR="\"$(CLASSPATH_SEPARATOR)\"" \
 	-DECJ_JAR_FILE="\"$(ECJ_JAR)\"" \
-	-DLIBGCJ_DEFAULT_DATABASE="\"$(dbexecdir)/$(db_name)\"" \
+	-DLIBGCJ_DEFAULT_DATABASE="\"/var/lib/$(MULTIARCH_DIR)/gcj-4.7/$(db_name)\"" \
 	-DLIBGCJ_DEFAULT_DATABASE_PATH_TAIL="\"$(db_pathtail)\""
 
 AM_GCJFLAGS = \
Index: b/src/libjava/classpath/m4/acinclude.m4
===================================================================
--- a/src/libjava/classpath/m4/acinclude.m4
+++ b/src/libjava/classpath/m4/acinclude.m4
@@ -252,6 +252,10 @@
     .) toolexeclibdir=${libdir} ;; # Avoid trailing /.
     *) toolexeclibdir=${libdir}/${multi_os_directory} ;;
   esac
+  multiarch=`$CC -print-multiarch`
+  if test -n "$multiarch"; then
+    toolexeclibdir=${libdir}/${multiarch}
+  fi
   AC_SUBST(toolexeclibdir)
 ])
 
Index: b/src/libjava/classpath/configure.ac
===================================================================
--- a/src/libjava/classpath/configure.ac
+++ b/src/libjava/classpath/configure.ac
@@ -16,6 +16,8 @@
 
 AC_CANONICAL_TARGET
 
+dnl dummy change to run autoconf
+
 dnl GCJ LOCAL
 AC_ARG_ENABLE(java-maintainer-mode,
 	AS_HELP_STRING([--enable-java-maintainer-mode],
