# DP: Link libmudflapth.so with -lpthread

--- a/src/libmudflap/Makefile.am
+++ b/src/libmudflap/Makefile.am
@@ -44,7 +44,7 @@
 	mf-hooks2.c \
 	mf-hooks3.c
 libmudflapth_la_CFLAGS = -DLIBMUDFLAPTH
-libmudflapth_la_LIBADD =
+libmudflapth_la_LIBADD = -lpthread
 libmudflapth_la_DEPENDENCIES = $(libmudflapth_la_LIBADD)
 libmudflapth_la_LDFLAGS = -version-info `grep -v '^\#' $(srcdir)/libtool-version`
 
--- a/src/libmudflap/Makefile.in
+++ b/src/libmudflap/Makefile.in
@@ -288,7 +288,7 @@
 	mf-hooks3.c
 
 libmudflapth_la_CFLAGS = -DLIBMUDFLAPTH
-libmudflapth_la_LIBADD = 
+libmudflapth_la_LIBADD = -lpthread
 libmudflapth_la_DEPENDENCIES = $(libmudflapth_la_LIBADD)
 libmudflapth_la_LDFLAGS = -version-info `grep -v '^\#' $(srcdir)/libtool-version`
 
