diff --git a/src/kdecompat/kdecompat.c b/src/kdecompat/kdecompat.c
index 49a446d..33b75d8 100644
--- a/src/kdecompat/kdecompat.c
+++ b/src/kdecompat/kdecompat.c
@@ -246,8 +246,8 @@ kdecompatUpdatePreviews (CompWindow *w)
     {
 	if (format == 32 && actual == kd->kdePreviewAtom)
 	{
-	    int *data    = (int *) propData;
-	    int nPreview = *data++;
+	    long *data    = (long *) propData;
+	    int  nPreview = *data++;
 
 	    if (n == (6 * nPreview + 1))
 	    {
