Description: Show GNOME Documents in menus everywhere
Author: Jeremy Bicha <jbicha@ubuntu.com>
Index: gnome-documents-0.4.1/data/gnome-documents.desktop
===================================================================
--- gnome-documents-0.4.1.orig/data/gnome-documents.desktop	2012-04-16 17:10:14.000000000 +0200
+++ gnome-documents-0.4.1/data/gnome-documents.desktop	2012-04-17 22:02:33.094989102 +0200
@@ -120,7 +120,6 @@
 Terminal=false
 Type=Application
 StartupNotify=true
-OnlyShowIn=GNOME;
 Categories=GNOME;GTK;Utility;Core;
 Keywords=Docs;PDF;Document;
 Keywords[ar]=مستندات;المستندات;بي دي اف;
Index: gnome-documents-0.4.1/data/gnome-documents.desktop.in
===================================================================
--- gnome-documents-0.4.1.orig/data/gnome-documents.desktop.in	2012-04-16 17:10:10.000000000 +0200
+++ gnome-documents-0.4.1/data/gnome-documents.desktop.in	2012-04-17 22:02:33.094989102 +0200
@@ -6,6 +6,5 @@
 Terminal=false
 Type=Application
 StartupNotify=true
-OnlyShowIn=GNOME;
 Categories=GNOME;GTK;Utility;Core;
 _Keywords=Docs;PDF;Document;
Index: gnome-documents-0.4.1/data/gnome-documents.desktop.in.in
===================================================================
--- gnome-documents-0.4.1.orig/data/gnome-documents.desktop.in.in	2012-03-01 03:29:42.000000000 +0100
+++ gnome-documents-0.4.1/data/gnome-documents.desktop.in.in	2012-04-17 22:02:33.094989102 +0200
@@ -6,6 +6,5 @@
 Terminal=false
 Type=Application
 StartupNotify=true
-OnlyShowIn=GNOME;
 Categories=GNOME;GTK;Utility;Core;
 _Keywords=Docs;PDF;Document;
