Author: Aurelien Jarno <aurel32@debian.org>
Description: enable maintainer mode
--- a/configure.ac
+++ b/configure.ac
@@ -34,6 +34,7 @@ AC_CONFIG_SRCDIR([ggadget/common.h])
 AC_CONFIG_AUX_DIR([autotools])
 AC_CONFIG_MACRO_DIR([autotools])
 
+AM_MAINTAINER_MODE
 
 # Save the user defined CFLAGS and CXXFLAGS, then restore them later.
 # So that the value set by AC_PROG_CC and AC_PROG_CXX won't be used,
