Author: Aurelien Jarno <aurel32@debian.org>
Description: recognize GNU/kFreeBSD as "linux"
--- a/configure.ac
+++ b/configure.ac
@@ -495,7 +495,7 @@ case $host_os in
     GGL_PLATFORM_SHORT="\"mac\""
     GGL_PLATFORM="\"mac\""
     ;;
-  *linux* )
+  *linux* | *kfreebsd*-gnu* )
     ggl_host_type=linux
     PREDEFINED_MACROS="$PREDEFINED_MACROS -DGGL_HOST_LINUX=1"
     GGL_PLATFORM_SHORT="\"linux\""
