Author: Mike Hommey <glandium@debian.org>
Description: Tighten the GREVersionRange.
--- a/extensions/gtkmoz_browser_element/browser_child.cc
+++ b/extensions/gtkmoz_browser_element/browser_child.cc
@@ -1268,8 +1268,8 @@
   nsresult rv;
 
   static const GREVersionRange kGREVersion = {
-    "1.9a", PR_TRUE,
-    "1.9.4", PR_FALSE,
+    "1.9.1", PR_TRUE,
+    "1.9.2", PR_FALSE,
   };
 
   char xpcom_location[4096];
--- a/extensions/smjs_script_runtime/libmozjs_glue.cc
+++ b/extensions/smjs_script_runtime/libmozjs_glue.cc
@@ -198,8 +198,8 @@
   char xpcom_file[4096];
 
   static const GREVersionRange kGREVersion = {
-    "1.9a", PR_TRUE,
-    "1.9.4", PR_FALSE,
+    "1.9.1", PR_TRUE,
+    "1.9.2", PR_FALSE,
   };
 
   rv = GRE_GetGREPathWithProperties(&kGREVersion, 1, nsnull, 0,
