Author: Pino Toscano <pino@debian.org>
Description: recognize GNU/Hurd as "linux"
--- a/configure.ac
+++ b/configure.ac
@@ -495,7 +495,7 @@
     GGL_PLATFORM_SHORT="\"mac\""
     GGL_PLATFORM="\"mac\""
     ;;
-  *linux* | *kfreebsd*-gnu* )
+  *linux* | *kfreebsd*-gnu* | gnu* )
     ggl_host_type=linux
     PREDEFINED_MACROS="$PREDEFINED_MACROS -DGGL_HOST_LINUX=1"
     GGL_PLATFORM_SHORT="\"linux\""
