Index: sqlite3-3.7.4/Makefile.in
===================================================================
--- sqlite3-3.7.4.orig/Makefile.in
+++ sqlite3-3.7.4/Makefile.in
@@ -534,10 +534,20 @@
 
 # Rules to build the LEMON compiler generator
 #
-lemon$(BEXE):	$(TOP)/tool/lemon.c $(TOP)/src/lempar.c
+lemon-for-build$(BEXE):	$(TOP)/tool/lemon.c $(TOP)/src/lempar.c
 	$(BCC) -o $@ $(TOP)/tool/lemon.c
 	cp $(TOP)/src/lempar.c .
 
+lemon$(TEXE):	$(TOP)/tool/lemon.c $(TOP)/src/lempar.c
+	$(LTLINK) -o $@ $(TOP)/tool/lemon.c
+	cp $(TOP)/src/lempar.c .
+
+ifeq ($(CROSS_BUILDING),yes)
+LEMON_FOR_BUILD = lemon-for-build$(BEXE)
+else
+LEMON_FOR_BUILD = lemon$(TEXE)
+endif
+
 # Rules to build individual *.o files from generated *.c files. This
 # applies to:
 #
@@ -784,10 +794,10 @@
 #
 parse.h:	parse.c
 
-parse.c:	$(TOP)/src/parse.y lemon$(BEXE) $(TOP)/addopcodes.awk
+parse.c:	$(TOP)/src/parse.y $(LEMON_FOR_BUILD) $(TOP)/addopcodes.awk
 	cp $(TOP)/src/parse.y .
 	rm -f parse.h
-	./lemon$(BEXE) $(OPT_FEATURE_FLAGS) $(OPTS) parse.y
+	./$(LEMON_FOR_BUILD) $(OPT_FEATURE_FLAGS) $(OPTS) parse.y
 	mv parse.h parse.h.temp
 	$(NAWK) -f $(TOP)/addopcodes.awk parse.h.temp >parse.h
 
@@ -930,7 +940,10 @@
 	rm -f *.lo *.la *.o sqlite3$(TEXE) libsqlite3.la
 	rm -f sqlite3.h opcodes.*
 	rm -rf .libs .deps
-	rm -f lemon$(BEXE) lempar.c parse.* sqlite*.tar.gz
+	rm -f $(LEMON_FOR_BUILD) lempar.c parse.* sqlite*.tar.gz
+ifeq ($(CROSS_BUILDING), yes)
+	rm -f lemon$(TEXE)
+endif
 	rm -f mkkeywordhash$(BEXE) keywordhash.h
 	rm -f $(PUBLISH)
 	rm -f *.da *.bb *.bbg gmon.out
