Last-Update: 2012-09-13
Author: Dmitry Smirnov <onlyjob@member.fsf.org>
Bug-Debian: http://bugs.debian.org/660933
Forwarded: 2012-02-23
Description: replace large menu icon with smaller one

--- a/src/shell/shell.pro
+++ b/src/shell/shell.pro
@@ -47,7 +47,7 @@
     desktop.path = /applications/
 
     icons.target = ../../launcher/icons/tupi.png
-    icons.commands = cp ../../launcher/icons/tupi.png $(INSTALL_ROOT)/pixmaps
+    icons.commands = cp ../../launcher/icons/tupi_128x128.png $(INSTALL_ROOT)/pixmaps/tupi.png
     icons.path = /pixmaps/
 
     tupiman.target = ../components/help/help/man/tupi.1.gz
