Last-Update: 2012-09-13
Author: Dmitry Smirnov <onlyjob@member.fsf.org>
Bug-Debian: http://bugs.debian.org/676121
Forwarded: no
Description: update ruby scripts for compatibility with ruby1.9

--- a/configure.rb
+++ b/configure.rb
@@ -3,9 +3,9 @@
 # TODO: This script must detect if every command line given is valid 
 #       Currently, it just try to check if some of them are included or not
 
-require './qonf/configure'
-require './qonf/info'
-require './qonf/defaults'
+require_relative 'qonf/configure'
+require_relative 'qonf/info'
+require_relative 'qonf/defaults'
 
 begin
     conf = RQonf::Configure.new(ARGV)
--- a/qonf/configure.rb
+++ b/qonf/configure.rb
@@ -1,10 +1,10 @@
 # encoding: utf-8
 
-require './qonf/test'
-require './qonf/config'
-require './qonf/info'
-require './qonf/qonfexception'
-require './qonf/makefile'
+require_relative 'test'
+require_relative 'config'
+require_relative 'info'
+require_relative 'qonfexception'
+require_relative 'makefile'
 
 module RQonf
 
--- a/qonf/qmake.rb
+++ b/qonf/qmake.rb
@@ -1,5 +1,5 @@
-require 'qonf/info'
-require 'qonf/extensions'
+require_relative 'info'
+require_relative 'extensions'
 
 module RQonf
 
--- a/qonf/test.rb
+++ b/qonf/test.rb
@@ -1,10 +1,10 @@
 
-require 'qonf/qmake'
-require 'qonf/detectos'
+require_relative 'qmake'
+require_relative 'detectos'
 require 'rexml/parsers/sax2parser'
 require 'rexml/sax2listener'
 
-require 'qonf/info'
+require_relative 'info'
 
 module RQonf
 
