Description: Made error message tell about missing vboxnetflt module.
Author: Michael Meskes <meskes@debian.org>

diff -Nur virtualbox-ose-4.0.0-dfsg.orig/src/VBox/Devices/Network/DrvIntNet.cpp virtualbox-ose-4.0.0-dfsg/src/VBox/Devices/Network/DrvIntNet.cpp
--- virtualbox-ose-4.0.0-dfsg.orig/src/VBox/Devices/Network/DrvIntNet.cpp	2010-11-28 18:24:07.000000000 +0100
+++ virtualbox-ose-4.0.0-dfsg/src/VBox/Devices/Network/DrvIntNet.cpp	2011-01-01 23:39:33.408286663 +0100
@@ -1569,7 +1569,7 @@
             return VERR_PDM_NO_ATTACHED_DRIVER;
         }
         return PDMDrvHlpVMSetError(pDrvIns, rc, RT_SRC_POS,
-                                   N_("Failed to open/create the internal network '%s'"), pThis->szNetwork);
+                                   N_("Failed to open/create the internal network '%s' (you might need to modprobe vboxnetflt to make it accessible)"), pThis->szNetwork);
     }
 
     AssertRelease(OpenReq.hIf != INTNET_HANDLE_INVALID);
