Description: Adjusts failure message with Debian specific solution.
Author: Daniel Baumann <daniel@debian.org>

diff -urNad virtualbox-ose-4.0.2-dfsg/src/VBox/VMM/VMMR3/VM.cpp virtualbox-ose-3.0.8-dfsg/src/VBox/VMM/VMMR3/VM.cpp
--- virtualbox-ose-4.0.2-dfsg/src/VBox/VMM/VMMR3/VM.cpp	2011-01-20 11:58:26.000000000 +0100
+++ virtualbox-ose-4.0.2-dfsg/src/VBox/VMM/VMMR3/VM.cpp	2011-01-20 11:58:27.000000000 +0100
@@ -410,9 +410,11 @@
                     break;
                 case VERR_VERSION_MISMATCH:
                 case VERR_VM_DRIVER_VERSION_MISMATCH:
-                    pszError = N_("The VirtualBox support driver which is running is from a different "
-                                  "version of VirtualBox.  You can correct this by stopping all "
-                                  "running instances of VirtualBox and reinstalling the software.");
+                    pszError = N_("The version of the VirtualBox kernel modules and the version of "
+				  "VirtualBox application are not matching. You can correct this by "
+				  "either installing the correct virtualbox-ose-modules package "
+				  "(if available) through apt-get or by building it manually with: "
+				  "module-assistant auto-install virtualbox-ose");
                     break;
                 default:
                     pszError = N_("Unknown error initializing kernel driver");
