Description: Disable notifications about outdated guest additions.
Author: Felix Geyer <debfx-pkg@fobos.de>

diff --git a/src/VBox/Additions/x11/VBoxClient/hostversion.cpp b/src/VBox/Additions/x11/VBoxClient/hostversion.cpp
--- a/src/VBox/Additions/x11/VBoxClient/hostversion.cpp
+++ b/src/VBox/Additions/x11/VBoxClient/hostversion.cpp
@@ -40,6 +40,8 @@ public:
 
     virtual int showNotify(const char *pcHeader, const char *pcBody)
     {
+        return VINF_SUCCESS;
+
         int rc;
 # ifdef VBOX_WITH_DBUS
         DBusConnection *conn;
