diff --git a/gzip.c b/gzip.c
index 38e1373..0d9f499 100644
--- a/gzip.c
+++ b/gzip.c
@@ -223,7 +223,9 @@ static int handled_sig[] =
 #ifdef SIGHUP
     , SIGHUP
 #endif
+#ifdef SIGPIPE
     , SIGPIPE
+#endif
 #ifdef SIGTERM
     , SIGTERM
 #endif
@@ -556,7 +556,11 @@ int main (int argc, char **argv)
     ALLOC(ush, tab_prefix1, 1L<<(BITS-1));
 #endif
 
+#ifdef SIGPIPE
     exiting_signal = quiet ? SIGPIPE : 0;
+#else
+    exiting_signal = 0;
+#endif
     install_signal_handlers ();
 
     /* And get to work */
