diff -NurbB gnutls-2.7.10.orig/lib/po/Makevars gnutls-2.7.10/lib/po/Makevars
--- gnutls-2.7.10.orig/lib/po/Makevars	2009-05-11 18:15:43.000000000 +0200
+++ gnutls-2.7.10/lib/po/Makevars	2009-05-14 19:29:24.000000000 +0200
@@ -1,7 +1,7 @@
 # Makefile variables for PO directory in any package using GNU gettext.
 
 # Usually the message domain is the same as the package name.
-DOMAIN = $(PACKAGE)
+DOMAIN = $(PACKAGE)26
 
 # These two variables depend on the location of this directory.
 subdir = po
