Description: use settings from desktop-base by default
Author: Yves-Alexis Perez <corsac@debian.org>
Forwarded: not-needed

Index: libxfce4util-4.8.2/libxfce4util/xfce-resource.c
===================================================================
--- libxfce4util-4.8.2.orig/libxfce4util/xfce-resource.c	2011-09-12 20:45:56.000000000 +0200
+++ libxfce4util-4.8.2/libxfce4util/xfce-resource.c	2011-09-12 22:36:08.024034835 +0200
@@ -221,6 +221,12 @@
   _res_split_and_append (dirs, XFCE_RESOURCE_CONFIG);
   _res_split_and_append (SYSCONFDIR "/xdg", XFCE_RESOURCE_CONFIG);
 
+  /* add desktop-base config files to the list. quick and dirty
+   * patch for Etch
+   */
+  _res_split_and_append ("/usr/share/desktop-base/profiles/xdg-config", 
+		  XFCE_RESOURCE_CONFIG);
+
   /*
    * Icons dirs
    */
