Subject: forward porting to clutter 1.10
Author: Sjoerd Simons <sjoerd@debian.org>
Bug-Debian: http://bugs.debian.org/666749
Index: clutter-gesture-0.0.2.1/tests/test-events.c
===================================================================
--- clutter-gesture-0.0.2.1.orig/tests/test-events.c	2012-04-02 23:20:03.299751855 +0800
+++ clutter-gesture-0.0.2.1/tests/test-events.c	2012-04-02 23:20:07.571671343 +0800
@@ -247,7 +247,7 @@
     case CLUTTER_DELETE:
 //      g_print ("[%s] DELETE", source);
       break;
-    case CLUTTER_NOTHING:
+    default:
       return FALSE;
     }
 
Index: clutter-gesture-0.0.2.1/tests/test_engine.c
===================================================================
--- clutter-gesture-0.0.2.1.orig/tests/test_engine.c	2012-04-02 23:20:03.299751855 +0800
+++ clutter-gesture-0.0.2.1/tests/test_engine.c	2012-04-02 23:20:07.571671343 +0800
@@ -74,7 +74,9 @@
   else
     IsMotion = TRUE;
 
-  clutter_set_motion_events_enabled (IsMotion);
+  clutter_stage_set_motion_events_enabled (
+    CLUTTER_STAGE (clutter_actor_get_stage (actor)),
+    IsMotion);
 
   g_print ("*** Per actor motion events %s ***\n",
            IsMotion ? "enabled" : "disabled");
@@ -208,7 +210,7 @@
     case CLUTTER_DELETE:
       g_print ("[%s] DELETE", source);
       break;
-    case CLUTTER_NOTHING:
+    default:
       return FALSE;
     }
 
