Subject: Pod files lacking description in NAME section
Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=472860
Author: Nicholas Bamber <nicholas@periapt.co.uk>
Forwarded: no
Last-Update: 2011-04-29
--- a/Parser/Style/Objects.pm
+++ b/Parser/Style/Objects.pm
@@ -52,7 +52,7 @@
 
 =head1 NAME
 
-XML::Parser::Style::Objects
+XML::Parser::Style::Objects - Objects styler parser
 
 =head1 SYNOPSIS
 
@@ -75,4 +75,4 @@
 
 L<XML::Parser::Style::Tree>
 
-=cut
\ No newline at end of file
+=cut
--- a/Parser/Style/Subs.pm
+++ b/Parser/Style/Subs.pm
@@ -23,7 +23,7 @@
 
 =head1 NAME
 
-XML::Parser::Style::Subs
+XML::Parser::Style::Subs - glue for handling element callbacks
 
 =head1 SYNOPSIS
 
@@ -55,4 +55,4 @@
 
 Nothing special is returned by parse.
 
-=cut
\ No newline at end of file
+=cut
--- a/Parser/Style/Tree.pm
+++ b/Parser/Style/Tree.pm
@@ -49,7 +49,7 @@
 
 =head1 NAME
 
-XML::Parser::Style::Tree
+XML::Parser::Style::Tree - Tree style parser
 
 =head1 SYNOPSIS
 
