Description: spelling fix
Origin: vendor
Author: gregor herrmann <gregoa@debian.org>
Bug: https://rt.cpan.org/Ticket/Display.html?id=80671
Forwarded: https://rt.cpan.org/Ticket/Display.html?id=80671
Last-Update: 2012-11-06

--- a/lib/Font/TTF/Cmap.pm
+++ b/lib/Font/TTF/Cmap.pm
@@ -15,7 +15,7 @@
 
 =head1 INSTANCE VARIABLES
 
-The instance variables listed here are not preceeded by a space due to their
+The instance variables listed here are not preceded by a space due to their
 emulating structural information in the font.
 
 =over 4
@@ -31,7 +31,7 @@
 =back
 
 Each subtables also has its own instance variables which are, again, not
-preceeded by a space.
+preceded by a space.
 
 =over 4
 
--- a/lib/Font/TTF/OldCmap.pm
+++ b/lib/Font/TTF/OldCmap.pm
@@ -22,7 +22,7 @@
 
 =head1 INSTANCE VARIABLES
 
-The instance variables listed here are not preceeded by a space due to their
+The instance variables listed here are not preceded by a space due to their
 emulating structural information in the font.
 
 =over 4
@@ -38,7 +38,7 @@
 =back
 
 Each subtables also has its own instance variables which are, again, not
-preceeded by a space.
+preceded by a space.
 
 =over 4
 
--- a/lib/Font/TTF/Ttopen.pm
+++ b/lib/Font/TTF/Ttopen.pm
@@ -30,7 +30,7 @@
 
 =item OFFSET
 
-This variable is preceeded by a space and gives the offset from the start of the
+This variable is preceded by a space and gives the offset from the start of the
 table (not the table section) to the script table for this script
 
 =item REFTAG
@@ -59,7 +59,7 @@
 
 =item OFFSET
 
-This variable is preceeded by a a space and gives the offset from the start of
+This variable is preceded by a a space and gives the offset from the start of
 the whole table to the language table for this language
 
 =item REFTAG
@@ -105,7 +105,7 @@
 
 =item OFFSET
 
-This attribute is preceeded by a space and gives the offset relative to the start of the whole
+This attribute is preceded by a space and gives the offset relative to the start of the whole
 table of this particular feature.
 
 =item PARMS
@@ -132,7 +132,7 @@
 
 =item OFFSET
 
-This name is preceeded by a space and contains the offset from the start of the table to this
+This name is preceded by a space and contains the offset from the start of the table to this
 particular lookup
 
 =item TYPE
@@ -158,7 +158,7 @@
 
 =item OFFSET
 
-This is preceeded by a space and gives the offset relative to the start of the table for this
+This is preceded by a space and gives the offset relative to the start of the table for this
 subtable
 
 =item FORMAT
@@ -197,7 +197,7 @@
 
 =item PRE
 
-This array holds the sequence of elements preceeding the first match element
+This array holds the sequence of elements preceding the first match element
 and has the same form as the MATCH array.
 
 =item POST
