Description: fix POD issues
Origin: vendor
Author: gregor herrmann <gregoa@debian.org>
Reviewed-by: gregor herrmann <gregoa@debian.org>
Bug: https://rt.cpan.org/Ticket/Display.html?id=80671
Forwarded: https://rt.cpan.org/Ticket/Display.html?id=80671
Last-Update: 2012-11-06

--- a/lib/Font/TTF/AATutils.pm
+++ b/lib/Font/TTF/AATutils.pm
@@ -104,6 +104,10 @@
     }
 }
 
+=head1 NAME
+
+Font::TTF::AATutils - Module without a description
+
 =head2 ($classes, $states) = AAT_read_subtable($fh, $baseOffset, $subtableStart, $limits)
 
 =cut
--- a/lib/Font/TTF/Bsln.pm
+++ b/lib/Font/TTF/Bsln.pm
@@ -8,6 +8,8 @@
 
 =head1 INSTANCE VARIABLES
 
+=over
+
 =item version
 
 =item xformat
@@ -24,6 +26,8 @@
 
 =item lookup
 
+=back
+
 =head1 METHODS
 
 =cut
--- a/lib/Font/TTF/Coverage.pm
+++ b/lib/Font/TTF/Coverage.pm
@@ -12,6 +12,8 @@
 for each glyph is the corresponding index in the coverage table. The resulting
 data structure is that a Coverage table has the following fields:
 
+=over
+
 =item cover
 
 A boolean to indicate whether this table is a coverage table (TRUE) or a
@@ -31,6 +33,8 @@
 A count of the elements in a coverage table for use with add. Each subsequent
 addition is added with the current count and increments the count.
 
+=back
+
 =head1 METHODS
 
 =cut
--- a/lib/Font/TTF/Delta.pm
+++ b/lib/Font/TTF/Delta.pm
@@ -9,6 +9,8 @@
 Each device table corresponds to a set of deltas for a particular point over
 a range of ppem values.
 
+=over
+
 =item first
 
 The first ppem value in the range
@@ -27,6 +29,8 @@
 This is the fmt used (log2 of number bits per value) when the device table was
 read. It is recalculated on output.
 
+=back
+
 =head1 METHODS
 
 =cut
--- a/lib/Font/TTF/EBLC.pm
+++ b/lib/Font/TTF/EBLC.pm
@@ -21,6 +21,7 @@
 An array of tables of following information
 
 =over 8
+
 =item indexSubTableArrayOffset (R)
 =item indexTablesSize (R)
 =item numberOfIndexSubTables (R)
@@ -33,6 +34,7 @@
 =item ppemY
 =item bitDepth
 =item flags
+
 =back
 
 =item indexSubTableArray (R)
--- a/lib/Font/TTF/Fdsc.pm
+++ b/lib/Font/TTF/Fdsc.pm
@@ -8,12 +8,16 @@
 
 =head1 INSTANCE VARIABLES
 
+=over
+
 =item version
 
 =item descriptors
 
 Hash keyed by descriptor tags
 
+=back
+
 =head1 METHODS
 
 =cut
--- a/lib/Font/TTF/Kern.pm
+++ b/lib/Font/TTF/Kern.pm
@@ -35,10 +35,10 @@
 
 Array of subtables in the kerning table
 
-=over 4
-
 Each subtable has a number of instance variables.
 
+=over 4
+
 =item kern
 
 A two level hash array containing kerning values. The indexing is left
--- a/lib/Font/TTF/OldMort.pm
+++ b/lib/Font/TTF/OldMort.pm
@@ -8,6 +8,8 @@
 
 =head1 INSTANCE VARIABLES
 
+=over
+
 =item version
 
 table version number (Fixed: currently 1.0)
@@ -109,6 +111,8 @@
 
 =back
 
+=back
+
 =back
 
 =head1 METHODS
--- a/lib/Font/TTF/Prop.pm
+++ b/lib/Font/TTF/Prop.pm
@@ -8,6 +8,8 @@
 
 =head1 INSTANCE VARIABLES
 
+=over
+
 =item version
 
 =item default
@@ -18,6 +20,8 @@
 
 =item lookupFormat
 
+=back
+
 =head1 METHODS
 
 =cut
--- a/lib/Font/TTF/XMLparse.pm
+++ b/lib/Font/TTF/XMLparse.pm
@@ -55,9 +55,9 @@
 
 Anything else is taken as the element content to be stored in the parent.
 
-=back 4
+=back
 
-=back 4
+=back
 
 In addition, the context hash passed to these messages contains the following
 keys:
@@ -98,7 +98,7 @@
 Character processing is to insert all the characters into the text element of the
 context for available use later.
 
-=back 4
+=back
 
 =head1 METHODS
 
