Description: add NAME section to manpage
Author: Xavier Guimard <x.guimard@free.fr>
Bug: https://rt.cpan.org/Ticket/Display.html?id=80671
Forwarded: https://rt.cpan.org/Ticket/Display.html?id=80671
Last-Update: 2012-11-06

--- a/lib/Font/TTF/Gloc.pm
+++ b/lib/Font/TTF/Gloc.pm
@@ -1,6 +1,6 @@
 package Font::TTF::Gloc;
 
-=head1 TITLE
+=head1 NAME
 
 Font::TTF::Gloc - Offsets into Glat table for the start of the attributes for each glyph
 
