Index: gtk+2.0-2.22.0/gtk/gtkrc.c
===================================================================
--- gtk+2.0-2.22.0.orig/gtk/gtkrc.c	2010-09-23 05:00:37.000000000 +0200
+++ gtk+2.0-2.22.0/gtk/gtkrc.c	2010-09-23 20:29:35.145600147 +0200
@@ -450,7 +450,7 @@
       if (im_module_file)
 	result = g_strdup (im_module_file);
       else
-	result = g_build_filename (GTK_SYSCONFDIR, "gtk-2.0", "gtk.immodules", NULL);
+	result = g_build_filename (GTK_LIBDIR, "gtk-2.0", GTK_BINARY_VERSION, "gtk.immodules", NULL);
     }
 
   return result;
