Index: zvbi-0.2.33/daemon/Makefile.in
===================================================================
--- zvbi-0.2.33.orig/daemon/Makefile.in	2010-11-09 09:23:34.000000000 +0100
+++ zvbi-0.2.33/daemon/Makefile.in	2010-11-09 09:23:38.000000000 +0100
@@ -250,7 +250,7 @@
 
 AM_CPPFLAGS = \
 	-I$(top_srcdir) \
-	-DLIBZVBI_CHAINS_PATH=\"libzvbi-chains.so\" \
+	-DLIBZVBI_CHAINS_PATH=\"libzvbi-chains.so.0.0.0\" \
 	-D_REENTRANT \
 	-D_GNU_SOURCE
 
