Index: zvbi-0.2.33/src/exp-gfx.c
===================================================================
--- zvbi-0.2.33.orig/src/exp-gfx.c	2008-02-24 15:17:47.000000000 +0100
+++ zvbi-0.2.33/src/exp-gfx.c	2012-03-12 08:02:12.678422597 +0100
@@ -1672,11 +1672,19 @@
 	char title[80];
 	unsigned int i;
 
+#if PNG_LIBPNG_VER_MAJOR == 1 && PNG_LIBPNG_VER_MINOR >= 4
+	if (setjmp(png_jmpbuf(png_ptr)))
+#else
 	if (setjmp (png_ptr->jmpbuf))
+#endif
 		return FALSE;
 
 	png_set_write_fn (png_ptr,
+#if PNG_LIBPNG_VER_MAJOR == 1 && PNG_LIBPNG_VER_MINOR >= 4
+			  (void *) gfx,
+#else
 			  (voidp) gfx,
+#endif
 			  write_data,
 			  flush_data);
 
