Index: zvbi-0.2.33/src/proxy-msg.c
===================================================================
--- zvbi-0.2.33.orig/src/proxy-msg.c	2012-05-05 11:06:33.000000000 +0200
+++ zvbi-0.2.33/src/proxy-msg.c	2012-06-14 08:13:14.000000000 +0200
@@ -1039,7 +1039,6 @@
 static char * vbi_proxy_msg_resolve_symlinks( const char * p_dev_name )
 {
    struct stat stbuf;
-   char   link_name[MAXPATHLEN + 1];
    char * p_path;
    char * p_tmp;
    char * p_tmp2;
@@ -1054,6 +1053,7 @@
       res = lstat(p_path, &stbuf);
       if ((res == 0) && S_ISLNK(stbuf.st_mode))
       {
+         char link_name[stbuf.st_size + 1];
          name_len = readlink(p_path, link_name, sizeof(link_name));
          if ((name_len > 0) && (name_len < (int) sizeof(link_name)))
          {
