diff -Nurp k3b-1.0rc2/admin/cvs.sh k3b-1.0rc2-copy/admin/cvs.sh
--- k3b-1.0rc2/admin/cvs.sh	2006-12-13 14:40:56.000000000 +0000
+++ k3b-1.0rc2-copy/admin/cvs.sh	2006-12-18 14:36:23.000000000 +0000
@@ -554,10 +554,11 @@ for subdir in $dirs; do
    fi
    perl -e '$mes=0; while (<STDIN>) { next if (/^(if\s|else\s|endif)/); if (/^messages:/) { $mes=1; print $_; next; } if ($mes) { if (/$\\(XGETTEXT\)/ && / -o/) { s/ -o \$\(podir\)/ _translatorinfo.cpp -o \$\(podir\)/ } print $_; } else { print $_; } }' < Makefile.am | egrep -v '^include ' > _transMakefile
 
-   kdepotpath=${includedir:-`kde-config --expandvars --install include`}/kde.pot
-   if ! test -f $kdepotpath; then
-	kdepotpath=`kde-config --expandvars --prefix`/include/kde.pot
-   fi
+   #kdepotpath=${includedir:-`kde-config --expandvars --install include`}/kde.pot
+   #if ! test -f $kdepotpath; then
+	#kdepotpath=`kde-config --expandvars --prefix`/include/kde.pot
+   #fi
+   kdepotpath=/usr/include/kde/kde.pot
 
    $MAKE -s -f _transMakefile podir=$podir EXTRACTRC="$EXTRACTRC" PREPARETIPS="$PREPARETIPS" srcdir=. \
 	XGETTEXT="${XGETTEXT:-xgettext} --foreign-user -C -ci18n -ki18n -ktr2i18n -kI18N_NOOP -kI18N_NOOP2 -kaliasLocale -x $kdepotpath" messages
