---
 src/k3bsystemproblemdialog.cpp |    4 ++++
 1 file changed, 4 insertions(+)

--- a/src/k3bsystemproblemdialog.cpp
+++ b/src/k3bsystemproblemdialog.cpp
@@ -221,6 +221,7 @@ void K3b::SystemProblemDialog::checkSyst
                                                    i18n("Install a more recent version of the cdrtools.") ) );
             }
 
+/*
 #ifdef Q_OS_LINUX
 
             //
@@ -257,6 +258,7 @@ void K3b::SystemProblemDialog::checkSyst
             }
 #endif // CDRECORD_SUID_ROOT_CHECK
 #endif
+*/
         }
 
         if( !k3bcore->externalBinManager()->foundBin( "cdrdao" ) ) {
@@ -265,6 +267,7 @@ void K3b::SystemProblemDialog::checkSyst
                                                i18n("K3b uses cdrdao to actually write CDs."),
                                                i18n("Install the cdrdao package.") ) );
         }
+/*
         else {
 #ifdef Q_OS_LINUX
 #ifdef CDRECORD_SUID_ROOT_CHECK
@@ -280,6 +283,7 @@ void K3b::SystemProblemDialog::checkSyst
 #endif // CDRECORD_SUID_ROOT_CHECK
 #endif
         }
+*/
     }
 
 
