diff -Nurp k3b-1.0rc3-clean/src/k3bwelcomewidget.cpp k3b-1.0rc3/src/k3bwelcomewidget.cpp
--- k3b-1.0rc3-clean/src/k3bwelcomewidget.cpp	2006-12-29 20:41:04.000000000 +0000
+++ k3b-1.0rc3/src/k3bwelcomewidget.cpp	2007-01-08 11:50:52.000000000 +0000
@@ -313,6 +313,8 @@ void K3bWelcomeWidget::loadConfig( KConf
     sl.append( "file_new_data" );
     sl.append( "file_new_dvd" );
     sl.append( "tools_copy_cd" );
+    sl.append( "tools_write_cd_image" );
+    sl.append( "tools_write_dvd_iso" );
   }
 
   QPtrList<KAction> actions;
