---
 src/k3bsystemproblemdialog.cpp |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/src/k3bsystemproblemdialog.cpp
+++ b/src/k3bsystemproblemdialog.cpp
@@ -593,7 +593,7 @@ void K3b::SystemProblemDialog::checkSyst
     if( problems.isEmpty() ) {
         kDebug() << "          - none - ";
         if( level == AlwaysNotify ) {
-            K3b::PassivePopup::showPopup( i18n("No problems found in system configuration."), i18n("System configured properly") );
+//            K3b::PassivePopup::showPopup( i18n("No problems found in system configuration."), i18n("System configured properly") );
         }
     }
     else {
