---
 src/k3bsystemproblemdialog.cpp |    3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

--- a/src/k3bsystemproblemdialog.cpp
+++ b/src/k3bsystemproblemdialog.cpp
@@ -526,8 +526,7 @@ void K3b::SystemProblemDialog::checkSyst
                                                 "include MP3 support for legal reasons."),
                                            i18n("To enable MP3 support, please install the MAD MP3 decoding library as well as the "
                                                 "K3b MAD MP3 decoder plugin (the latter may already be installed but not functional "
-                                                "due to the missing libmad). Some distributions allow installation of MP3 support "
-                                                "via an online update tool.") ) );
+                                                "due to the missing libmad). To get MP3 support, please install the package libk3b6-extracodecs.") ) );
     }
 
 #ifdef HAVE_ICONV_H
