From f86c3980aac45ac81ab1737d3366c3cd5449fa3c Mon Sep 17 00:00:00 2001
From 60dcd4363cadd48efa847f2de2f46024eae6a67c
From: Pino Toscano <pino@kde.org>
Date: Mon, 20 Feb 2012 12:12:42 +0100
Date: Sat, 25 Feb 2012 16:10:15 +0100
Subject: [PATCH] k3bsetup: put the shebang in the first line
Subject: [PATCH] k3bsetup: fix the shebang line to be just /bin/sh

---
 k3bsetup/k3bsetup |    4 +---
 1 files changed, 1 insertions(+), 3 deletions(-)

diff --git a/k3bsetup/k3bsetup b/k3bsetup/k3bsetup
index fbbbeeb..6e32d01 100644
--- a/k3bsetup/k3bsetup
+++ b/k3bsetup/k3bsetup
@@ -1,3 +1,4 @@
+#!/bin/sh
 #
 # $Id$
 # Copyright (C) 2003 Sebastian Trueg <trueg@k3b.org>
@@ -12,7 +13,4 @@
 # See the file "COPYING" for the exact licensing terms.
 #
 
-
-#!/usr/bin/sh
-
 kcmshell4 k3bsetup
-- 
1.7.9

