Description: Find postgresql headers on Debian
Author: Pere Nubiola i Radigales <pere@nubiola.cat>
Author: Gregor Jasny <gjasny@googlemail.com>
Bug: http://bugs.debian.org/665721
Acked-By: Modestas Vainius <modax@debian.org>
Origin: other, http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=665721#17
Last-Update: 2012-08-18

--- a/Modules/FindPostgreSQL.cmake
+++ b/Modules/FindPostgreSQL.cmake
@@ -116,6 +116,7 @@
    # Look in other places.
    ${PostgreSQL_ROOT_DIRECTORIES}
   PATH_SUFFIXES
+    postgresql
     pgsql/server
     postgresql/server
     include/server
