Subject: Add -lpthread when using Boost::Thread.
Author: Kai Wasserbäch <debian@carbon-project.org>
Origin: vendor
Last-Update: 2011-01-16
Bug: http://cmake.org/Bug/view.php?id=10692
Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=563479
Forwarded: http://cmake.org/Bug/file_download.php?file_id=3629&type=bug
---
 Modules/FindBoost.cmake |    3 +++
 1 file changed, 3 insertions(+)

--- a/Modules/FindBoost.cmake
+++ b/Modules/FindBoost.cmake
@@ -1184,6 +1184,9 @@ set(Boost_ERROR_REASON)
             message (STATUS "  ${COMPONENT}")
           endif(NOT Boost_FIND_QUIETLY)
           set(Boost_LIBRARIES ${Boost_LIBRARIES} ${Boost_${UPPERCOMPONENT}_LIBRARY})
+          if(${UPPERCOMPONENT} STREQUAL "THREAD" AND UNIX)
+            set(Boost_LIBRARIES ${Boost_LIBRARIES} pthread)
+          endif(${UPPERCOMPONENT} STREQUAL "THREAD" AND UNIX)
         endif( Boost_${UPPERCOMPONENT}_FOUND )
       endforeach(COMPONENT)
   else()
