Index: sip4-4.13.3/siputils.py
===================================================================
--- sip4-4.13.3.orig/siputils.py	2012-06-21 11:18:06.533099607 -0400
+++ sip4-4.13.3/siputils.py	2012-06-21 11:21:56.865092801 -0400
@@ -1617,9 +1617,12 @@
         mfile.write("\n$(OFILES): $(HFILES)\n")
 
         for mf in self._build["moc_headers"].split():
-            root, discard = os.path.splitext(mf)
+            root, _ = os.path.splitext(mf)
             cpp = "moc_" + root + ".cpp"
 
+            if self._src_dir != self.dir:
+                mf = os.path.join(self._src_dir, mf)
+
             mfile.write("\n%s: %s\n" % (cpp, mf))
             mfile.write("\t$(MOC) -o %s %s\n" % (cpp, mf))
 
