Author: Daniel Baumann <daniel@debian.org>
Description: Fixes outdated smake rules (Closes: #250031, #471224).

diff -Naurp cdrdao.orig/scsilib/RULES/MKLINKS cdrdao/scsilib/RULES/MKLINKS
--- cdrdao.orig/scsilib/RULES/MKLINKS	2009-06-14 08:36:45.000000000 +0000
+++ cdrdao/scsilib/RULES/MKLINKS	2009-07-16 22:51:41.000000000 +0000
@@ -111,6 +111,8 @@ ln -s	i586-linux-cc.rul	mipsel-linux-cc.
 ln -s	i586-linux-gcc.rul	mipsel-linux-gcc.rul
 ln -s	i586-linux-cc.rul	s390-linux-cc.rul
 ln -s	i586-linux-gcc.rul	s390-linux-gcc.rul
+ln -s	i586-linux-cc.rul	s390x-linux-cc.rul
+ln -s	i586-linux-gcc.rul	s390x-linux-gcc.rul
 ln -s	i586-linux-cc.rul	parisc-linux-cc.rul
 ln -s	i586-linux-gcc.rul	parisc-linux-gcc.rul
 ln -s	i586-linux-cc.rul	parisc64-linux-cc.rul
diff -Naurp cdrdao.orig/scsilib/RULES/MKLINKS.orig cdrdao/scsilib/RULES/MKLINKS.orig
--- cdrdao.orig/scsilib/RULES/MKLINKS.orig	1970-01-01 00:00:00.000000000 +0000
+++ cdrdao/scsilib/RULES/MKLINKS.orig	2009-06-14 08:36:45.000000000 +0000
@@ -0,0 +1,174 @@
+#!/bin/sh
+###########################################################################
+#ident "@(#)MKLINKS	1.53 04/09/08 "
+###########################################################################
+# Written 1996 by J. Schilling
+###########################################################################
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2, or (at your option)
+# any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; see the file COPYING.  If not, write to the Free Software
+# Foundation, 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
+###########################################################################
+ln -s	mk-.id			mk-make.id
+ln -s	r-make.c4x		r-gmake.c4x
+ln -s	r-make.c4x-dep		r-gmake.c4x-dep
+ln -s	r-make.obj		r-gmake.obj
+ln -s	9000-725-hp-ux-cc.rul	9000-710-hp-ux-cc.rul
+ln -s	9000-725-hp-ux-cc.rul	9000-712-hp-ux-cc.rul
+ln -s	9000-725-hp-ux-cc.rul	9000-715-hp-ux-cc.rul
+ln -s	9000-725-hp-ux-cc.rul	9000-735-hp-ux-cc.rul
+ln -s	9000-725-hp-ux-cc.rul	9000-743-hp-ux-cc.rul
+ln -s	9000-725-hp-ux-cc.rul	9000-755-hp-ux-cc.rul
+ln -s	9000-725-hp-ux-cc.rul	9000-777-hp-ux-cc.rul
+ln -s	9000-725-hp-ux-cc.rul	9000-778-hp-ux-cc.rul
+ln -s	9000-725-hp-ux-cc.rul	9000-780-hp-ux-cc.rul
+ln -s	9000-725-hp-ux-cc.rul	9000-782-hp-ux-cc.rul
+ln -s	9000-725-hp-ux-cc.rul	9000-785-hp-ux-cc.rul
+ln -s	9000-725-hp-ux-cc.rul	9000-800-hp-ux-cc.rul
+ln -s	9000-725-hp-ux-cc.rul	9000-820-hp-ux-cc.rul
+ln -s	9000-725-hp-ux-cc.rul	9000-831-hp-ux-cc.rul
+ln -s	9000-725-hp-ux-cc.rul	9000-899-hp-ux-cc.rul
+ln -s	9000-725-hp-ux-gcc.rul	9000-710-hp-ux-gcc.rul
+ln -s	9000-725-hp-ux-gcc.rul	9000-712-hp-ux-gcc.rul
+ln -s	9000-725-hp-ux-gcc.rul	9000-715-hp-ux-gcc.rul
+ln -s	9000-725-hp-ux-gcc.rul	9000-735-hp-ux-gcc.rul
+ln -s	9000-725-hp-ux-gcc.rul	9000-743-hp-ux-gcc.rul
+ln -s	9000-725-hp-ux-gcc.rul	9000-755-hp-ux-gcc.rul
+ln -s	9000-725-hp-ux-gcc.rul	9000-777-hp-ux-gcc.rul
+ln -s	9000-725-hp-ux-gcc.rul	9000-778-hp-ux-gcc.rul
+ln -s	9000-725-hp-ux-gcc.rul	9000-780-hp-ux-gcc.rul
+ln -s	9000-725-hp-ux-gcc.rul	9000-782-hp-ux-gcc.rul
+ln -s	9000-725-hp-ux-gcc.rul	9000-785-hp-ux-gcc.rul
+ln -s	9000-725-hp-ux-gcc.rul	9000-800-hp-ux-gcc.rul
+ln -s	9000-725-hp-ux-gcc.rul	9000-820-hp-ux-gcc.rul
+ln -s	9000-725-hp-ux-gcc.rul	9000-831-hp-ux-gcc.rul
+ln -s	9000-725-hp-ux-gcc.rul	9000-899-hp-ux-gcc.rul
+ln -s	hppa-nextstep-cc.rul	i386-nextstep-cc.rul
+ln -s	hppa-nextstep-cc.rul	m68k-nextstep-cc.rul
+ln -s	hppa-nextstep-cc.rul	sparc-nextstep-cc.rul
+ln -s	hppa-nextstep-cc.rul	i386-nextstep-gcc.rul
+ln -s	hppa-nextstep-cc.rul	m68k-nextstep-gcc.rul
+ln -s	hppa-nextstep-cc.rul	sparc-nextstep-gcc.rul
+ln -s	os-irix.id		os-irix64.id
+ln -s	ip22-irix-cc.rul	ip17-irix-cc.rul
+ln -s	ip22-irix-cc.rul	ip20-irix-cc.rul
+ln -s	ip22-irix-cc.rul	ip27-irix-cc.rul
+ln -s	ip22-irix-cc.rul	ip28-irix-cc.rul
+ln -s	ip22-irix-cc.rul	ip30-irix-cc.rul
+ln -s	ip22-irix-cc.rul	ip32-irix-cc.rul
+ln -s	ip22-irix-gcc.rul	ip17-irix-gcc.rul
+ln -s	ip22-irix-gcc.rul	ip20-irix-gcc.rul
+ln -s	ip22-irix-gcc.rul	ip27-irix-gcc.rul
+ln -s	ip22-irix-gcc.rul	ip28-irix-gcc.rul
+ln -s	ip22-irix-gcc.rul	ip30-irix-gcc.rul
+ln -s	ip22-irix-gcc.rul	ip32-irix-gcc.rul
+ln -s	i386-freebsd-cc.rul	sparc64-freebsd-cc.rul
+ln -s	i386-freebsd-gcc.rul	sparc64-freebsd-gcc.rul
+ln -s	i386-netbsd-cc.rul	alpha-netbsd-cc.rul
+ln -s	i386-netbsd-gcc.rul	alpha-netbsd-gcc.rul
+ln -s	i386-netbsd-cc.rul	sparc-netbsd-cc.rul
+ln -s	i386-netbsd-gcc.rul	sparc-netbsd-gcc.rul
+ln -s	i386-netbsd-cc.rul	mac68k-netbsd-cc.rul
+ln -s	i386-netbsd-gcc.rul	mac68k-netbsd-gcc.rul
+ln -s	i386-netbsd-cc.rul	amiga-netbsd-cc.rul
+ln -s	i386-netbsd-gcc.rul	amiga-netbsd-gcc.rul
+ln -s	i386-netbsd-cc.rul	macppc-netbsd-cc.rul
+ln -s	i386-netbsd-gcc.rul	macppc-netbsd-gcc.rul
+ln -s	i386-bsd-os-cc.rul	sparc-bsd-os-cc.rul
+ln -s	i386-bsd-os-gcc.rul	sparc-bsd-os-gcc.rul
+ln -s	i386-bsd-os3-cc.rul	sparc-bsd-os3-cc.rul
+ln -s	i386-bsd-os3-gcc.rul	sparc-bsd-os3-gcc.rul
+ln -s	i586-linux-cc.rul	i386-linux-cc.rul
+ln -s	i586-linux-cc.rul	i486-linux-cc.rul
+ln -s	i586-linux-cc.rul	i686-linux-cc.rul
+ln -s	i586-linux-gcc.rul	i386-linux-gcc.rul
+ln -s	i586-linux-gcc.rul	i486-linux-gcc.rul
+ln -s	i586-linux-gcc.rul	i686-linux-gcc.rul
+ln -s	i586-linux-cc.rul	sparc-linux-cc.rul
+ln -s	i586-linux-gcc.rul	sparc-linux-gcc.rul
+ln -s	i586-linux-cc.rul	sparc64-linux-cc.rul
+ln -s	i586-linux-gcc.rul	sparc64-linux-gcc.rul
+ln -s	i586-linux-cc.rul	alpha-linux-cc.rul
+ln -s	i586-linux-gcc.rul	alpha-linux-gcc.rul
+ln -s	i586-linux-cc.rul	ppc-linux-cc.rul
+ln -s	i586-linux-gcc.rul	ppc-linux-gcc.rul
+ln -s	i586-linux-cc.rul	m68k-linux-cc.rul
+ln -s	i586-linux-gcc.rul	m68k-linux-gcc.rul
+ln -s	i586-linux-cc.rul	armv4l-linux-cc.rul
+ln -s	i586-linux-gcc.rul	armv4l-linux-gcc.rul
+ln -s	i586-linux-cc.rul	mips-linux-cc.rul
+ln -s	i586-linux-gcc.rul	mips-linux-gcc.rul
+ln -s	i586-linux-cc.rul	mipsel-linux-cc.rul
+ln -s	i586-linux-gcc.rul	mipsel-linux-gcc.rul
+ln -s	i586-linux-cc.rul	s390-linux-cc.rul
+ln -s	i586-linux-gcc.rul	s390-linux-gcc.rul
+ln -s	i586-linux-cc.rul	parisc-linux-cc.rul
+ln -s	i586-linux-gcc.rul	parisc-linux-gcc.rul
+ln -s	i586-linux-cc.rul	parisc64-linux-cc.rul
+ln -s	i586-linux-gcc.rul	parisc64-linux-gcc.rul
+ln -s	i586-linux-cc.rul	ia64-linux-cc.rul
+ln -s	i586-linux-gcc.rul	ia64-linux-gcc.rul
+ln -s	i586-linux-cc.rul	x86_64-linux-cc.rul
+ln -s	i586-linux-gcc.rul	x86_64-linux-gcc.rul
+ln -s	os-cygwin32_nt.id	os-cygwin_95-4.0.id
+ln -s	os-cygwin32_nt.id	os-cygwin_98-4.0.id
+ln -s	os-cygwin32_nt.id	os-cygwin_98-4.10.id
+ln -s	os-cygwin32_nt.id	os-cygwin_me-4.90.id 
+ln -s	os-cygwin32_nt.id	os-cygwin_nt-4.0.id
+ln -s	os-cygwin32_nt.id	os-cygwin_nt-5.0.id
+ln -s	os-cygwin32_nt.id	os-cygwin_nt-5.1.id
+ln -s	os-unixware.id		os-openunix.id
+ln -s	i486-cygwin32_nt-cc.rul	i386-cygwin32_nt-cc.rul
+ln -s	i486-cygwin32_nt-cc.rul	i586-cygwin32_nt-cc.rul
+ln -s	i486-cygwin32_nt-cc.rul	i686-cygwin32_nt-cc.rul
+ln -s	i486-cygwin32_nt-cc.rul	i786-cygwin32_nt-cc.rul
+ln -s	i486-cygwin32_nt-gcc.rul	i386-cygwin32_nt-gcc.rul
+ln -s	i486-cygwin32_nt-gcc.rul	i586-cygwin32_nt-gcc.rul
+ln -s	i486-cygwin32_nt-gcc.rul	i686-cygwin32_nt-gcc.rul
+ln -s	i486-cygwin32_nt-gcc.rul	i786-cygwin32_nt-gcc.rul
+ln -s	power-macintosh-rhapsody-cc.rul	power-macintosh-mac-os10-cc.rul
+ln -s	power-macintosh-rhapsody-gcc.rul	power-macintosh-mac-os10-gcc.rul
+ln -s	sun4-sunos4-cc.rul	sun4c-sunos4-cc.rul
+ln -s	sun4-sunos4-gcc.rul	sun4c-sunos4-gcc.rul
+ln -s	sun4-sunos5-cc.rul	sun4c-sunos5-cc.rul
+ln -s	sun4-sunos5-gcc.rul	sun4c-sunos5-gcc.rul
+ln -s	sun4-sunos4-cc.rul	sun4m-sunos4-cc.rul
+ln -s	sun4-sunos4-gcc.rul	sun4m-sunos4-gcc.rul
+ln -s	sun4-sunos5-cc.rul	sun4m-sunos5-cc.rul
+ln -s	sun4-sunos5-gcc.rul	sun4m-sunos5-gcc.rul
+ln -s	sun4-sunos5-cc.rul	sun4u-sunos5-cc.rul
+ln -s	sun4-sunos5-gcc.rul	sun4u-sunos5-gcc.rul
+ln -s	sun4-sunos5-cc.rul	sun4d-sunos5-cc.rul
+ln -s	sun4-sunos5-gcc.rul	sun4d-sunos5-gcc.rul
+ln -s	sun4-sunos5-cc.rul	sun4e-sunos5-cc.rul
+ln -s	sun4-sunos5-gcc.rul	sun4e-sunos5-gcc.rul
+ln -s	sun4-sunos5-cc.rul	sun4L-sunos5-cc.rul
+ln -s	sun4-sunos5-gcc.rul	sun4L-sunos5-gcc.rul
+ln -s	i386-unixware-cc.rul	i486-unixware-cc.rul
+ln -s	i386-unixware-gcc.rul	i486-unixware-gcc.rul
+ln -s	i386-unixware-cc.rul	i586-unixware-cc.rul
+ln -s	i386-unixware-gcc.rul	i586-unixware-gcc.rul
+ln -s	i386-unixware-cc.rul	pentium-iii-unixware-cc.rul
+ln -s	i386-unixware-gcc.rul	pentium-iii-unixware-gcc.rul
+ln -s	i386-unixware-cc.rul	pentium-pro-unixware-cc.rul
+ln -s	i386-unixware-gcc.rul	pentium-pro-unixware-gcc.rul
+ln -s	i386-ms-dos-gcc.rul	i486-ms-dos-gcc.rul
+ln -s	i386-ms-dos-gcc.rul	i586-ms-dos-gcc.rul
+ln -s	i386-ms-dos-gcc.rul	i686-ms-dos-gcc.rul
+ln -s	i386-ms-dos-gcc.rul	i786-ms-dos-gcc.rul
+#
+ln -s	mk-smake.id		mk-build.id
+ln -s	r-smake.c4x		r-build.c4x
+ln -s	r-smake.c4x-dep		r-build.c4x-dep
+ln -s	r-smake.dep		r-build.dep
+ln -s	r-smake.obj		r-build.obj
+ln -s	r-smake.tag		r-build.tag
diff -Naurp cdrdao.orig/scsilib/RULES/s390x-linux-cc.rul cdrdao/scsilib/RULES/s390x-linux-cc.rul
--- cdrdao.orig/scsilib/RULES/s390x-linux-cc.rul	1970-01-01 00:00:00.000000000 +0000
+++ cdrdao/scsilib/RULES/s390x-linux-cc.rul	2009-07-16 22:51:41.000000000 +0000
@@ -0,0 +1,70 @@
+#ident "@(#)i586-linux-cc.rul	1.5 02/10/15 "
+###########################################################################
+# Written 1996 by J. Schilling
+###########################################################################
+#
+# Platform dependent MACROS for Linux
+#
+###########################################################################
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2, or (at your option)
+# any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; see the file COPYING.  If not, write to
+# the Free Software Foundation, 675 Mass Ave, Cambridge, MA 02139, USA.
+###########################################################################
+include $(SRCROOT)/$(RULESDIR)/rules.prg
+###########################################################################
+
+CPPFLAGS=	$(CPPOPTS) $(CPPOPTX)
+CFLAGS=		$(COPTS) $(CWARNOPTS) $(COPTOPT) $(GCCOPTOPT) $(COPTX)
+
+CPPOPTS=	-I. -I$(ARCHDIR) -I$(OINCSDIR) $(INCDIRS:%=-I%) $(OSDEFS)
+COPTS=
+CWOPTS=		-Wall -Wtraditional \
+		-Wshadow -Wmissing-prototypes -Wstrict-prototypes
+COPTOPT=	-O
+KDEFINES=	-DKERNEL -D_KERNEL
+COPTDYN=	-fpic
+COPTGPROF=	-pg
+
+LIB_PREFIX=	lib
+LIB_SUFFIX=	.a
+SHL_SUFFIX=	.so.1.0
+
+#LIB_SOCKET=	-lsocket -lnsl -ldl
+LIB_SOCKET=	
+LIB_MATH=	-lm
+#LIB_KVM=	-lkvm
+LIB_KVM=	
+
+LIBS_PATH=	-L$(OLIBSDIR)
+
+LDFLAGS=	$(LDOPTS) $(LDOPTX)
+LDLIBS=		$(LIBS) $(LIBX)
+
+#LDOPTS=		$(LIBS_PATH) $(LDPATH) $(RUNPATH)
+LDOPTS=		$(LIBS_PATH) $(LDPATH)
+LDOPTDYN=	-shared -Wl,-soname,$(TARGET)
+LNDYNLIB=	@$(SYMLINK) $(TARGET) $(PTARGET_BASE).so
+
+FLOAT_OPTIONS=	
+
+CC=		@echo "	==> COMPILING \"$@\""; gcc
+LDCC=		@echo "	==> LINKING \"$@\""; gcc
+DYNLD=		@echo "	==> LINKING dynamic library \"$@\""; gcc
+RANLIB=		@echo "	==> RANDOMIZING ARCHIVE \"$@\""; true
+ARFLAGS=	cr
+LORDER=		echo
+TSORT=		cat
+
+RMDEP=		:
+MKDEP=		@echo "	==> MAKING DEPENDENCIES \"$@\""; $(RMDEP); gcc -M
+MKDEP_OUT=
diff -Naurp cdrdao.orig/scsilib/RULES/s390x-linux-gcc.rul cdrdao/scsilib/RULES/s390x-linux-gcc.rul
--- cdrdao.orig/scsilib/RULES/s390x-linux-gcc.rul	1970-01-01 00:00:00.000000000 +0000
+++ cdrdao/scsilib/RULES/s390x-linux-gcc.rul	2009-07-16 22:51:41.000000000 +0000
@@ -0,0 +1,70 @@
+#ident "@(#)i586-linux-gcc.rul	1.5 02/10/15 "
+###########################################################################
+# Written 1996 by J. Schilling
+###########################################################################
+#
+# Platform dependent MACROS for Linux
+#
+###########################################################################
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2, or (at your option)
+# any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; see the file COPYING.  If not, write to
+# the Free Software Foundation, 675 Mass Ave, Cambridge, MA 02139, USA.
+###########################################################################
+include $(SRCROOT)/$(RULESDIR)/rules.prg
+###########################################################################
+
+CPPFLAGS=	$(CPPOPTS) $(CPPOPTX)
+CFLAGS=		$(COPTS) $(CWARNOPTS) $(COPTOPT) $(GCCOPTOPT) $(COPTX)
+
+CPPOPTS=	-I. -I$(ARCHDIR) -I$(OINCSDIR) $(INCDIRS:%=-I%) $(OSDEFS)
+COPTS=
+CWOPTS=		-Wall -Wtraditional \
+		-Wshadow -Wmissing-prototypes -Wstrict-prototypes
+COPTOPT=	-O
+KDEFINES=	-DKERNEL -D_KERNEL
+COPTDYN=	-fpic
+COPTGPROF=	-pg
+
+LIB_PREFIX=	lib
+LIB_SUFFIX=	.a
+SHL_SUFFIX=	.so.1.0
+
+#LIB_SOCKET=	-lsocket -lnsl -ldl
+LIB_SOCKET=	
+LIB_MATH=	-lm
+#LIB_KVM=	-lkvm
+LIB_KVM=	
+
+LIBS_PATH=	-L$(OLIBSDIR)
+
+LDFLAGS=	$(LDOPTS) $(LDOPTX)
+LDLIBS=		$(LIBS) $(LIBX)
+
+#LDOPTS=		$(LIBS_PATH) $(LDPATH) $(RUNPATH)
+LDOPTS=		$(LIBS_PATH) $(LDPATH)
+LDOPTDYN=	-shared -Wl,-soname,$(TARGET)
+LNDYNLIB=	@$(SYMLINK) $(TARGET) $(PTARGET_BASE).so
+
+FLOAT_OPTIONS=	
+
+CC=		@echo "	==> COMPILING \"$@\""; gcc
+LDCC=		@echo "	==> LINKING \"$@\""; gcc
+DYNLD=		@echo "	==> LINKING dynamic library \"$@\""; gcc
+RANLIB=		@echo "	==> RANDOMIZING ARCHIVE \"$@\""; true
+ARFLAGS=	cr
+LORDER=		echo
+TSORT=		cat
+
+RMDEP=		:
+MKDEP=		@echo "	==> MAKING DEPENDENCIES \"$@\""; $(RMDEP); gcc -M
+MKDEP_OUT=
