Index: cdrdao-1.2.2/scsilib/include/schily.h
===================================================================
--- cdrdao-1.2.2.orig/scsilib/include/schily.h	2004-03-05 01:30:40.000000000 +0100
+++ cdrdao-1.2.2/scsilib/include/schily.h	2009-10-03 14:34:42.000000000 +0200
@@ -108,7 +108,7 @@
 		/* 6th arg not const, fexecv forces av[ac] = NULL */
 extern	int	fexecv __PR((const char *, FILE *, FILE *, FILE *, int,
 							char **));
-extern	int	fexecve __PR((const char *, FILE *, FILE *, FILE *,
+extern	int	fexecve_schily __PR((const char *, FILE *, FILE *, FILE *,
 					char * const *, char * const *));
 extern	int	fspawnv __PR((FILE *, FILE *, FILE *, int, char * const *));
 extern	int	fspawnl __PR((FILE *, FILE *, FILE *,
@@ -187,7 +187,7 @@
 extern	char	*findbytes __PR((const void *, int, char));
 extern	int	findline __PR((const char *, char, const char *,
 							int, char **, int));
-extern	int	getline __PR((char *, int));
+extern	int	getline_schily __PR((char *, int));
 extern	int	getstr __PR((char *, int));
 extern	int	breakline __PR((char *, char, char **, int));
 extern	int	getallargs __PR((int *, char * const**, const char *, ...));
Index: cdrdao-1.2.2/scsilib/libscg/scsitransp.c
===================================================================
--- cdrdao-1.2.2.orig/scsilib/libscg/scsitransp.c	2009-10-03 14:33:09.000000000 +0200
+++ cdrdao-1.2.2/scsilib/libscg/scsitransp.c	2009-10-03 14:33:09.000000000 +0200
@@ -324,7 +324,7 @@
 
 	js_printf("%s", msg);
 	flush();
-	if (getline(okbuf, sizeof (okbuf)) == EOF)
+	if (getline_schily(okbuf, sizeof (okbuf)) == EOF)
 		exit(EX_BAD);
 	if (streql(okbuf, "y") || streql(okbuf, "yes") ||
 	    streql(okbuf, "Y") || streql(okbuf, "YES"))
Index: cdrdao-1.2.2/scsilib/libschily/stdio/fgetline.c
===================================================================
--- cdrdao-1.2.2.orig/scsilib/libschily/stdio/fgetline.c	2004-08-08 13:02:05.000000000 +0200
+++ cdrdao-1.2.2/scsilib/libschily/stdio/fgetline.c	2009-10-03 14:33:09.000000000 +0200
@@ -64,7 +64,7 @@
 }
 
 EXPORT int
-getline(buf, len)
+getline_schily(buf, len)
 	char	*buf;
 	int	len;
 {
Index: cdrdao-1.2.2/scsilib/libschily/fexec.c
===================================================================
--- cdrdao-1.2.2.orig/scsilib/libschily/fexec.c	2009-10-03 14:35:26.000000000 +0200
+++ cdrdao-1.2.2/scsilib/libschily/fexec.c	2009-10-03 14:35:08.000000000 +0200
@@ -159,7 +159,7 @@
 	} while (p != NULL);
 	va_end(args);
 
-	ret = fexecve(name, in, out, err, av, env);
+	ret = fexecve_schily(name, in, out, err, av, env);
 	if (av != xav)
 		free(av);
 	return (ret);
@@ -173,11 +173,11 @@
 	char *av[];
 {
 	av[ac] = NULL;			/*  force list to be null terminated */
-	return (fexecve(name, in, out, err, av, environ));
+	return (fexecve_schily(name, in, out, err, av, environ));
 }
 
 EXPORT int
-fexecve(name, in, out, err, av, env)
+fexecve_schily(name, in, out, err, av, env)
 	const char *name;
 	FILE *in, *out, *err;
 	char * const av[], * const env[];
