--- a/configure
+++ b/configure
@@ -10072,7 +10072,7 @@
   scsilib_objs="ScsiIf-lib.o"
 
   case "$host" in
-    *-freebsd*)
+    *-freebsd* | *-kfreebsd*-gnu)
 			       scsilib_libs="$scsilib_libs -lcam"
 			       ;;
   esac
@@ -10090,7 +10090,7 @@
 	       scsilib_objs="ScsiIf-linux.o sg_err.o"
 	       ;;
 
-    *-freebsd*)
+    *-freebsd* | *-kfreebsd*-gnu)
 	       scsilib_libs="-lcam"
 	       scsilib_objs="ScsiIf-freebsd-cam.o"
 	       ;;
@@ -10148,7 +10148,7 @@
 esac
 
 case "$host" in
-  *-linux* | *-freebsd* | *-solaris*)
+  *-linux* | *-freebsd* | *-kfreebsd*-gnu | *-solaris*)
 	     sound_if_obj=SoundIF-ao.o
 	     ;;
 
--- a/configure.ac
+++ b/configure.ac
@@ -239,7 +239,7 @@
   scsilib_objs="ScsiIf-lib.o"
   
   case "$host" in
-    *-freebsd*)
+    *-freebsd* | *-kfreebsd*-gnu)
 			       scsilib_libs="$scsilib_libs -lcam"
 			       ;;
   esac
@@ -257,7 +257,7 @@
 	       scsilib_objs="ScsiIf-linux.o sg_err.o"
 	       ;;
     
-    *-freebsd*)
+    *-freebsd* | *-kfreebsd*-gnu)
 	       scsilib_libs="-lcam"
 	       scsilib_objs="ScsiIf-freebsd-cam.o"
 	       ;;
@@ -313,7 +313,7 @@
 
 dnl Sound device selection for xcdrdao
 case "$host" in
-  *-linux* | *-freebsd* | *-solaris*)
+  *-linux* | *-freebsd* | *-kfreebsd*-gnu | *-solaris*)
 	     sound_if_obj=SoundIF-ao.o
 	     ;;
 
--- a/dao/ScsiIf-freebsd-cam.cc
+++ b/dao/ScsiIf-freebsd-cam.cc
@@ -32,6 +32,7 @@
 
 #include "ScsiIf.h"
 #include "log.h"
+#include "util.h"
 
 #define DEF_RETRY_COUNT 1
 
