Description: run tests in verbose mode
Author: Lucas Nussbaum <lucas@debian.org>

Index: ruby1.9.1-1.9.3~preview1+svn33029/common.mk
===================================================================
--- ruby1.9.1-1.9.3~preview1+svn33029.orig/common.mk	2011-08-25 18:10:42.000000000 +0200
+++ ruby1.9.1-1.9.3~preview1+svn33029/common.mk	2011-08-25 18:10:06.000000000 +0200
@@ -488,15 +488,15 @@
 
 test-all: $(TEST_RUNNABLE)-test-all
 yes-test-all: PHONY
-	$(RUNRUBY) "$(srcdir)/test/runner.rb" --ruby="$(RUNRUBY)" $(TESTS)
+	$(RUNRUBY) "$(srcdir)/test/runner.rb" -v --ruby="$(RUNRUBY)" $(TESTS)
 TESTS_BUILD = mkmf
 no-test-all: PHONY
-	$(MINIRUBY) -I"$(srcdir)/lib" "$(srcdir)/test/runner.rb" $(TESTS_BUILD)
+	$(MINIRUBY) -I"$(srcdir)/lib" "$(srcdir)/test/runner.rb" -v $(TESTS_BUILD)
 
 test-ruby: $(TEST_RUNNABLE)-test-ruby
 no-test-ruby: PHONY
 yes-test-ruby: PHONY
-	$(RUNRUBY) "$(srcdir)/test/runner.rb" ruby
+	$(RUNRUBY) "$(srcdir)/test/runner.rb" -v ruby
 
 extconf: $(PREP)
 	$(Q) $(MAKEDIRS) "$(EXTCONFDIR)"
