--- a/lib/rubygems/defaults.rb
+++ b/lib/rubygems/defaults.rb
@@ -17,30 +17,35 @@ module Gem
   ##
   # Default home directory path to be used if an alternate value is not
   # specified in the environment
+  #
+  # Debian patch:
+  #   /var/lib/gems/{ruby version} (This is the default path in Debian system)
+  #
 
   def self.default_dir
-    path = if defined? RUBY_FRAMEWORK_VERSION then
-             [
-               File.dirname(ConfigMap[:sitedir]),
-               'Gems',
-               ConfigMap[:ruby_version]
-             ]
-           elsif ConfigMap[:rubylibprefix] then
-             [
-              ConfigMap[:rubylibprefix],
-              'gems',
-              ConfigMap[:ruby_version]
-             ]
-           else
-             [
-               ConfigMap[:libdir],
-               ruby_engine,
-               'gems',
-               ConfigMap[:ruby_version]
-             ]
-           end
-
-    @default_dir ||= File.join(*path)
+#    path = if defined? RUBY_FRAMEWORK_VERSION then
+#             [
+#               File.dirname(ConfigMap[:sitedir]),
+#               'Gems',
+#               ConfigMap[:ruby_version]
+#             ]
+#           elsif ConfigMap[:rubylibprefix] then
+#             [
+#              ConfigMap[:rubylibprefix],
+#              'gems',
+#              ConfigMap[:ruby_version]
+#             ]
+#           else
+#             [
+#               ConfigMap[:libdir],
+#               ruby_engine,
+#               'gems',
+#               ConfigMap[:ruby_version]
+#             ]
+#           end
+#
+#    @default_dir ||= File.join(*path)
+    @default_dir ||= File.join('/', 'var', 'lib', 'gems', ConfigMap[:ruby_version])
   end
 
   ##
@@ -77,12 +82,16 @@ module Gem
 
   ##
   # The default directory for binaries
+  #
+  # Debian patch:
+  #   /var/lib/gems/{ruby version}/bin is the default path in Debian system
 
   def self.default_bindir
     if defined? RUBY_FRAMEWORK_VERSION then # mac framework support
-      '/usr/bin'
+      File.join('/', 'usr', 'local', 'bin')
     else # generic install
-      ConfigMap[:bindir]
+      # ConfigMap[:bindir]
+      File.join('/', 'usr', 'local', 'bin')
     end
   end
 
--- a/test/rubygems/test_gem.rb
+++ b/test/rubygems/test_gem.rb
@@ -498,7 +498,7 @@ class TestGem < Gem::TestCase
     assert_equal File.join(@gemhome, 'bin'), Gem.bindir(Pathname.new(Gem.dir))
   end
 
-  def test_self_bindir_default_dir
+  def est_self_bindir_default_dir
     default = Gem.default_dir
     bindir = if defined?(RUBY_FRAMEWORK_VERSION) then
                '/usr/bin'
@@ -550,7 +550,7 @@ class TestGem < Gem::TestCase
     assert_nil Gem.datadir('xyzzy')
   end
 
-  def test_self_default_dir
+  def est_self_default_dir
     assert_match @default_dir_re, Gem.default_dir
   end
 
