--- a/Makefile.am
+++ b/Makefile.am
@@ -1,6 +1,6 @@
 AUTOMAKE_OPTIONS = foreign
 
-SUBDIRS = src data doc
+SUBDIRS = src doc
 
 MAINTAINERCLEANFILES = aclocal.m4 config.guess config.h.in \
 		       config.sub configure depcomp install-sh \
--- a/configure.ac
+++ b/configure.ac
@@ -518,9 +518,6 @@
 src/modules/Makefile
 src/modules/filters/Makefile
 src/modules/loaders/Makefile
-data/Makefile
-data/fonts/Makefile
-data/images/Makefile
 doc/Makefile
 imlib2-config
 README
