Hide loved radio menu item.  It's no longer supported.
--- a/src/SideBarView.cpp
+++ b/src/SideBarView.cpp
@@ -539,7 +539,7 @@
         {
             map[PlayPersonalRadio] = menu.addAction( MooseUtils::icon( "personal_radio" ), tr( "Play %1's Radio Station" ).arg( text ) );
             map[PlayNeighbourRadio] = menu.addAction( MooseUtils::icon( "neighbour_radio" ), tr( "Play %1's Neighbourhood" ).arg( text ) );
-            map[PlayLovedRadio] = menu.addAction( MooseUtils::icon( "loved_radio" ), tr( "Play %1's Loved Tracks" ).arg( text ) );
+            //map[PlayLovedRadio] = menu.addAction( MooseUtils::icon( "loved_radio" ), tr( "Play %1's Loved Tracks" ).arg( text ) );
             menu.addSeparator();
             map[GoToUserPage] = menu.addAction( tr( "Go To %1's Profile" ).arg( text ) );
 
