Description: fix module name in the help script
Author: Yves-Alexis Perez <corsac@debian.org>
Forwarded: not-needed

diff --git a/scripts/xfhelp4.in b/scripts/xfhelp4.in
index 57fe9a0..ebb124e 100644
--- a/scripts/xfhelp4.in
+++ b/scripts/xfhelp4.in
@@ -25,7 +25,7 @@ HELP_DIR="@_datadir_@/doc"
 if [ ! x"$1" = x"" ]; then
     MODULE="${1%.html}"
 else
-    MODULE="xfce-utils"
+    MODULE="xfce4-utils"
 fi
 
 # Check for translated documentation
@@ -49,7 +49,7 @@ elif [ -r "$HELP_DIR/$MODULE/html/C/index.html" ]
 then
   URL="$HELP_DIR/$MODULE/html/C/index.html"
 else
-  URL="$HELP_DIR/xfce-utils/html/C/index.html"
+  URL="$HELP_DIR/xfce4-utils/html/C/index.html"
 fi
 
 xfbrowser4 $URL
