--- a/scripts/xfterm4	2007-11-25 17:45:31.000000000 +0000
+++ b/scripts/xfterm4	2007-11-25 17:45:41.000000000 +0000
@@ -1,24 +1,15 @@
 #!/bin/sh
 
 if [ "$1" = "-e" ]; then
-	if [ -n "$2" ]; then
-		shift
-		ESTRING="$@"
-	else
-		shift
-	fi
+	shift
 fi	
 
 if [ -d "$*" ]; then
 	cd "$*"
-elif [ -x "$*" ]; then
-	cd `dirname "$*"`
-	ESTRING="`which pauseme`"
-	MSTRING="$*"
-elif [ -f "$*" ]; then
+elif [ -f "$*" -a ! -x "$*" ]; then
 	ESTRING="`which less`"
 	MSTRING="$*"
-elif [ "`echo $* | grep "http:/"`" -o "`echo $* | grep "ftp:"`" ]; then
+elif [ "`echo $* | grep "http:/"`" -o "`echo $* | grep "ftp:"`" -a ! -x "$*" ]; then
 	# This requires lynx or links. If you don't have it, don't drop URLs.
 	if which lynx >/dev/null 2>&1; then
 		ESTRING="`which lynx`"
@@ -26,6 +17,13 @@
 		ESTRING="`which links`"
 	fi
 	MSTRING="$*"
+else
+	if which pauseme >/dev/null 2>&1; then
+		ESTRING="`which pauseme`"
+		MSTRING="$*"
+	else
+		ESTRING="$@"
+	fi
 fi
 
 if [ x"$ESTRING" = x"" ]; then
