Index: aften-0.0.8~svn20100103/libaften/ppc/cpu_caps.c
===================================================================
--- aften-0.0.8~svn20100103.orig/libaften/ppc/cpu_caps.c	2012-04-03 22:29:59.682264682 +0200
+++ aften-0.0.8~svn20100103/libaften/ppc/cpu_caps.c	2012-04-03 22:54:35.276149211 +0200
@@ -58,7 +58,7 @@
 void cpu_caps_detect(void)
 {
     if (g_is_altivec_present == -1L) {
-        sig_t oldhandler;
+        __sighandler_t oldhandler;
         sigset_t signame;
         struct sigaction sa_new, sa_old;
 
