Index: aften-0.0.8~svn20100103/CMakeLists.txt
===================================================================
--- aften-0.0.8~svn20100103.orig/CMakeLists.txt	2010-01-03 10:03:28.000000000 +0100
+++ aften-0.0.8~svn20100103/CMakeLists.txt	2012-04-03 22:27:35.682282644 +0200
@@ -342,6 +342,6 @@
 
 INSTALL(TARGETS ${INSTALL_TARGETS} aften_exe wavinfo wavrms wavfilter
         RUNTIME DESTINATION bin
-        LIBRARY DESTINATION lib${LIB_SUFFIX}
-        ARCHIVE DESTINATION lib${LIB_SUFFIX})
+        LIBRARY DESTINATION lib${LIB_SUFFIX}/${CMAKE_INSTALL_LIBDIR}
+        ARCHIVE DESTINATION lib${LIB_SUFFIX}/${CMAKE_INSTALL_LIBDIR})
 INSTALL(FILES ${INSTALL_HEADERS} libaften/aften.h libaften/aften-types.h DESTINATION include/aften)
