diff --git a/data/org.gnome.settings-daemon.plugins.media-keys.gschema.xml.in.in b/data/org.gnome.settings-daemon.plugins.media-keys.gschema.xml.in.in
index 6a63236..db9b2c4 100644
--- a/data/org.gnome.settings-daemon.plugins.media-keys.gschema.xml.in.in
+++ b/data/org.gnome.settings-daemon.plugins.media-keys.gschema.xml.in.in
@@ -85,6 +85,11 @@
       <_summary>Stop playback</_summary>
       <_description>Binding to stop playback.</_description>
     </key>
+    <key name="video-out" type="s">
+      <default>'&lt;Super&gt;p'</default>
+      <_summary>Video output</_summary>
+      <_description>Binding to switch the video output device.</_description>
+    </key>
     <key name="volume-down" type="s">
       <default>'XF86AudioLowerVolume'</default>
       <_summary>Volume down</_summary>
diff --git a/plugins/media-keys/shortcuts-list.h b/plugins/media-keys/shortcuts-list.h
index 437ed2c..8a66902 100644
--- a/plugins/media-keys/shortcuts-list.h
+++ b/plugins/media-keys/shortcuts-list.h
@@ -123,7 +123,7 @@ static struct {
         { FORWARD_KEY, NULL, "XF86AudioForward" },
         { REPEAT_KEY, NULL, "XF86AudioRepeat" },
         { RANDOM_KEY, NULL, "XF86AudioRandomPlay"},
-        { VIDEO_OUT_KEY, NULL, "<Super>p" },
+        { VIDEO_OUT_KEY, "video-out", NULL, NULL },
         /* Key code of the XF86Display key (Fn-F7 on Thinkpads, Fn-F4 on HP machines, etc.) */
         { VIDEO_OUT_KEY, NULL, "XF86Display" },
         /* Key code of the XF86RotateWindows key (present on some tablets) */
