Index: gnome-settings-daemon-3.4.2+git20120925.a4c817/plugins/mouse/gsd-mouse-manager.c
===================================================================
--- gnome-settings-daemon-3.4.2+git20120925.a4c817.orig/plugins/mouse/gsd-mouse-manager.c	2012-09-29 12:09:15.000000000 +0200
+++ gnome-settings-daemon-3.4.2+git20120925.a4c817/plugins/mouse/gsd-mouse-manager.c	2012-09-29 12:11:23.119038602 +0200
@@ -577,7 +577,8 @@ set_disable_w_typing (GsdMouseManager *m
                 args[2] = "1.0";
                 args[3] = "-K";
                 args[4] = "-R";
-                args[5] = NULL;
+                args[5] = "-t";                
+                args[6] = NULL;
 
                 if (!have_program_in_path (args[0]))
                         return 0;
