Index: gnome-settings-daemon-3.4.2/plugins/power/gsd-power-manager.c
===================================================================
--- gnome-settings-daemon-3.4.2.orig/plugins/power/gsd-power-manager.c	2012-05-19 08:39:11.423704404 +0200
+++ gnome-settings-daemon-3.4.2/plugins/power/gsd-power-manager.c	2012-05-19 08:39:15.719704363 +0200
@@ -4123,6 +4123,12 @@
 {
         GsdPowerManager *manager = GSD_POWER_MANAGER (user_data);
 
+        /* Check session pointer as a proxy for whether the manager is in the
+           start or stop state */
+        if (manager->priv->session == NULL) {
+                return;
+        }
+
         g_debug ("Calling method '%s.%s' for Power",
                  interface_name, method_name);
 
@@ -4157,6 +4163,12 @@
         GsdPowerManager *manager = GSD_POWER_MANAGER (user_data);
         GVariant *retval = NULL;
 
+        /* Check session pointer as a proxy for whether the manager is in the
+           start or stop state */
+        if (manager->priv->session == NULL) {
+                return NULL;
+        }
+
         if (g_strcmp0 (property_name, "Icon") == 0) {
                 retval = engine_get_icon_property_variant (manager);
         } else if (g_strcmp0 (property_name, "Tooltip") == 0) {
