Index: gnome-settings-daemon-3.4.2/plugins/power/gsd-power-manager.c
===================================================================
--- gnome-settings-daemon-3.4.2.orig/plugins/power/gsd-power-manager.c	2012-05-15 12:14:49.000000000 +0200
+++ gnome-settings-daemon-3.4.2/plugins/power/gsd-power-manager.c	2012-05-19 08:39:11.423704404 +0200
@@ -1160,7 +1160,7 @@
 
         /* add to database */
         array = up_client_get_devices (manager->priv->up_client);
-        for (i=0;i<array->len;i++) {
+        for (i = 0; array != NULL && i < array->len; i++) {
                 device = g_ptr_array_index (array, i);
                 engine_device_add (manager, device);
                 engine_check_recall (manager, device);
