diff --git a/module/zfs/zfs_vfsops.c b/module/zfs/zfs_vfsops.c
index 998cfde..464a88b 100644
--- a/module/zfs/zfs_vfsops.c
+++ b/module/zfs/zfs_vfsops.c
@@ -66,6 +66,7 @@
 #include <sys/sa.h>
 #include <sys/zpl.h>
 #include "zfs_comutil.h"
+#include <linux/version.h>
 
 
 /*ARGSUSED*/
@@ -1200,7 +1201,11 @@ zfs_domount(struct super_block *sb, void *data, int silent)
 	}
 
 	/* Allocate a root dentry for the filesystem */
+#if LINUX_VERSION_CODE < KERNEL_VERSION(3,4,0)
 	sb->s_root = d_alloc_root(root_inode);
+#else
+	sb->s_root = d_make_root(root_inode);
+#endif
 	if (sb->s_root == NULL) {
 		(void) zfs_umount(sb);
 		error = ENOMEM;
