Index: gnome-menus-2.26.1/layout/applications.menu
===================================================================
--- gnome-menus-2.26.1.orig/layout/applications.menu	2009-04-14 00:01:00.000000000 +0200
+++ gnome-menus-2.26.1/layout/applications.menu	2009-06-17 22:54:29.254809753 +0200
@@ -7,7 +7,6 @@
   <Directory>X-GNOME-Menu-Applications.directory</Directory>
 
   <!-- Scan legacy dirs first, as later items take priority -->
-  <KDELegacyDirs/>
   <LegacyDir>/etc/X11/applnk</LegacyDir>
   <LegacyDir>/usr/share/gnome/apps</LegacyDir>
 
