Index: gnome-menus-3.4.2/layout/applications.menu
===================================================================
--- gnome-menus-3.4.2.orig/layout/applications.menu	2012-05-18 22:07:48.359105100 +0200
+++ gnome-menus-3.4.2/layout/applications.menu	2012-05-18 22:07:49.191105070 +0200
@@ -156,4 +156,11 @@
      <Merge type="files" />
    </Layout>
 
+  <!-- The Debian menu -->
+  <Menu>
+    <Name>Debian</Name>
+    <MergeFile>debian-menu.menu</MergeFile>
+    <Directory>Debian.directory</Directory>
+  </Menu>
+
 </Menu> <!-- End Applications -->
