Index: gnome-menus-3.0.1/layout/applications.menu
===================================================================
--- gnome-menus-3.0.1.orig/layout/applications.menu	2011-06-04 21:59:24.161265110 +0200
+++ gnome-menus-3.0.1/layout/applications.menu	2011-06-04 22:28:28.905794724 +0200
@@ -134,6 +134,34 @@
         <Not><Category>Settings</Category></Not>
       </And>
     </Include>
+    <Menu>
+      <Name>Preferences</Name>
+      <Directory>Settings.directory</Directory>
+      <Include>
+        <And>
+          <Category>Settings</Category>
+          <Not>
+            <Or>
+              <Category>System</Category>
+              <Category>X-GNOME-Settings-Panel</Category>
+            </Or>
+          </Not>
+        </And>
+      </Include>
+    </Menu>
+    <Menu>
+      <Name>Administration</Name>
+      <Directory>Settings-System.directory</Directory>
+      <Include>
+        <And>
+          <Category>Settings</Category>
+          <Category>System</Category>
+          <Not>
+            <Category>X-GNOME-Settings-Panel</Category>
+          </Not>
+        </And>
+      </Include>
+    </Menu>
   </Menu>   <!-- End System Tools -->
 
   <!-- Other -->
