Index: gnome-menus-2.18.3/simple-editor/gmenu-simple-editor.in
===================================================================
--- gnome-menus-2.18.3.orig/simple-editor/gmenu-simple-editor.in	2007-07-02 19:15:26.000000000 +0200
+++ gnome-menus-2.18.3/simple-editor/gmenu-simple-editor.in	2007-07-17 18:45:40.274547699 +0200
@@ -20,9 +20,6 @@
 
 import sys
 
-if not "@pyexecdir@" in sys.path:
-    sys.path.insert (0, "@pyexecdir@")
-
 import GMenuSimpleEditor.main
 
 GMenuSimpleEditor.main.main (sys.argv[1:])
