Index: gnome-menus-2.24.2/layout/applications.menu
===================================================================
--- gnome-menus-2.24.2.orig/layout/applications.menu	2008-12-26 17:08:38.587592137 +0100
+++ gnome-menus-2.24.2/layout/applications.menu	2008-12-26 17:10:25.727590587 +0100
@@ -65,10 +65,23 @@
     <Include>
       <And>
         <Category>Education</Category>
+        <Not><Category>Science</Category></Not>
       </And>
     </Include>
   </Menu> <!-- End Education -->
 
+  <!-- Science -->
+  <Menu>
+    <Name>Science</Name>
+    <Directory>GnomeScience.directory</Directory>
+    <Include>
+      <And>
+        <Category>Education</Category>
+        <Category>Science</Category>
+      </And>
+    </Include>
+  </Menu> <!-- End Science -->
+
   <!-- Games -->
   <Menu>
     <Name>Games</Name>
Index: gnome-menus-2.24.2/desktop-directories/Education.directory.in
===================================================================
--- gnome-menus-2.24.2.orig/desktop-directories/Education.directory.in	2008-11-25 14:01:14.000000000 +0100
+++ gnome-menus-2.24.2/desktop-directories/Education.directory.in	2008-12-26 17:10:25.731589699 +0100
@@ -1,4 +1,4 @@
 [Desktop Entry]
 _Name=Education
-Icon=applications-science
+Icon=applications-accessories
 Type=Directory
Index: gnome-menus-2.24.2/desktop-directories/Utility.directory.in
===================================================================
--- gnome-menus-2.24.2.orig/desktop-directories/Utility.directory.in	2008-12-26 17:10:52.531587812 +0100
+++ gnome-menus-2.24.2/desktop-directories/Utility.directory.in	2008-12-26 17:11:01.619588523 +0100
@@ -1,5 +1,5 @@
 [Desktop Entry]
 _Name=Accessories
 _Comment=Desktop accessories
-Icon=applications-accessories
+Icon=applications-utilities
 Type=Directory
