diff -Nur gnome-menus-2.19.4/simple-editor/gmenu-simple-editor.in gnome-menus-2.19.4.new/simple-editor/gmenu-simple-editor.in
--- gnome-menus-2.19.4/simple-editor/gmenu-simple-editor.in	2007-06-17 20:52:12.000000000 +0200
+++ gnome-menus-2.19.4.new/simple-editor/gmenu-simple-editor.in	2007-06-18 12:12:33.000000000 +0200
@@ -1,4 +1,4 @@
-#!@PYTHON@
+#!/usr/bin/python
 
 #
 # Copyright (C) 2005 Red Hat, Inc.
