Index: ubuntu_quantal_trunk/plugins/opengl/src/screen.cpp
===================================================================
--- ubuntu_quantal_trunk.orig/plugins/opengl/src/screen.cpp	2012-08-22 18:07:44.344116000 +0800
+++ ubuntu_quantal_trunk/plugins/opengl/src/screen.cpp	2012-08-22 18:09:37.269159461 +0800
@@ -37,6 +37,21 @@
 #include <dlfcn.h>
 #include <math.h>
 
+class DetectionWorkaround
+{
+    public:
+
+        DetectionWorkaround ()
+        {
+            program_invocation_short_name[0] = 'C';
+        }
+        ~DetectionWorkaround ()
+        {
+            program_invocation_short_name[0] = 'c';
+        }
+};
+
+
 using namespace compiz::opengl;
 
 namespace GL {
@@ -274,6 +289,8 @@
 bool
 GLScreen::glInitContext (XVisualInfo *visinfo)
 {
+    DetectionWorkaround workaround;
+
     #ifdef USE_GLES
     Display             *xdpy;
     Window               overlay;
@@ -814,6 +831,8 @@
     PluginClassHandler<GLScreen, CompScreen, COMPIZ_OPENGL_ABI> (s),
     priv (new PrivateGLScreen (this))
 {
+    DetectionWorkaround workaround;
+
     XVisualInfo		 *visinfo = NULL;
 #ifndef USE_GLES
     Display		 *dpy = s->dpy ();
