Description: Fix for Linux's makefile for 1.0.10
Author: Kartik Mistry <kartik@debian.org>
Forwarded: https://sourceforge.net/tracker/?func=detail&aid=3015139&group_id=5635&atid=105635
Last-Update: 2010-06-12

--- libmng-1.0.10.orig/makefiles/makefile.linux
+++ libmng-1.0.10/makefiles/makefile.linux
@@ -14,25 +14,25 @@ CC=gcc
 OPTIONS = -DMNG_BUILD_SO -DMNG_FULL_CMS
 
 # where "make install" puts libmng.a,libmng.so*,libmng.h,libmng_conf.h,libmng_types.h
-prefix=/usr/local
+prefix=/usr
 
 # Where the zlib library and include files are located
 #ZLIBLIB=../zlib
 #ZLIBINC=../zlib
-ZLIBLIB=/usr/local/lib
-ZLIBINC=/usr/local/include
+ZLIBLIB=/usr/lib
+ZLIBINC=/usr/include
 
 # Where the jpeg library and include files are located
 #JPEGLIB=../jpgsrc
 #JPEGINC=../jpgsrc
-JPEGLIB=/usr/local/lib
-JPEGINC=/usr/local/include
+JPEGLIB=/usr/lib
+JPEGINC=/usr/include
 
 # Where the lcms library and include files are located
 #LCMSLIB=../lcms/lib
 #LCMSINC=../lcms/source
-LCMSLIB=/usr/local/lib
-LCMSINC=/usr/local/include
+LCMSLIB=/usr/lib
+LCMSINC=/usr/include
 
 ALIGN=
 # for i386:
@@ -44,7 +44,7 @@ WARNMORE=-Wwrite-strings -Wpointer-arith
 
 # for pgcc version 2.95.1, -O3 is buggy; don't use it.
 
-CFLAGS=-I$(ZLIBINC) -I$(JPEGINC) -I$(LCMSINC) -Wall -O3 -funroll-loops \
+CFLAGS=-I$(ZLIBINC) -I$(JPEGINC) -I$(LCMSINC) -Wall -O3 -fPIC -funroll-loops \
 	$(OPTIONS) $(ALIGN) # $(WARNMORE) -g
 LDFLAGS=-L. -Wl,-rpath,. \
 	-L$(ZLIBLIB) -Wl,-rpath,$(ZLIBLIB) \
@@ -57,7 +57,7 @@ RANLIB=ranlib
 
 # current version numbers
 MNGMAJ = 1
-MNGMIN = 1.0.9
+MNGMIN = 1.0.10
 MNGVER = $(MNGMAJ).$(MNGMIN)
 
 INCPATH=$(prefix)/include
@@ -84,7 +84,7 @@ OBJS = \
 	libmng_write.o \
 	libmng_zlib.o
 
-OBJSDLL = $(OBJS:.0=.pic.o)
+OBJSDLL = $(OBJS:.o=.pic.o)
 
 .SUFFIXES:      .c .o .pic.o
 
