Index: hardware-monitor-1.4.3/po/POTFILES.in
===================================================================
--- hardware-monitor-1.4.3.orig/po/POTFILES.in
+++ hardware-monitor-1.4.3/po/POTFILES.in
@@ -1,5 +1,6 @@
 # List of source files containing translatable strings.
 # Please keep this file sorted alphabetically.
+HardwareMonitor.server.in
 HardwareMonitor.server.in.in
 src/applet.cpp
 src/bar-view.cpp
