Description: fix spelling errors discovered by lintian
Author: Florian Schlichting <fschlich@zedat.fu-berlin.de>
Forwarded: https://rt.cpan.org/Public/Bug/Display.html?id=75659

--- a/Vector.pod
+++ b/Vector.pod
@@ -1295,7 +1295,7 @@
 
 C<$vector-E<gt>from_Hex($string);>
 
-Allows to read in the contents of a bit vector from a hexadecimal
+Allows one to read in the contents of a bit vector from a hexadecimal
 string, such as returned by the method "C<to_Hex()>" (see above).
 
 Remember that the least significant bits are always to the right of a
@@ -3065,7 +3065,7 @@
 
 This method calculates the reflexive transitive closure of the
 given boolean matrix (stored as a bit vector) using Kleene's
-algoritm.
+algorithm.
 
 (See L<Math::Kleene(3)> for a brief introduction into the
 theory behind Kleene's algorithm.)
--- a/lib/Bit/Vector/String.pod
+++ b/lib/Bit/Vector/String.pod
@@ -72,7 +72,7 @@
 
 C<$vector-E<gt>from_Oct($string);>
 
-Allows to read in the contents of a bit vector from an octal string,
+Allows one to read in the contents of a bit vector from an octal string,
 such as returned by the method "C<to_Oct()>" (see above).
 
 Note that this method is not particularly efficient, since it is
@@ -219,7 +219,7 @@
 
 C<$type = $vector-E<gt>String_Import($string);>
 
-Allows to read in the contents of a bit vector from a string
+Allows one to read in the contents of a bit vector from a string
 which has previously been produced by "C<String_Export()>",
 "C<to_Bin()>", "C<to_Oct()>", "C<to_Dec()>", "C<to_Hex()>",
 "C<to_Enum()>", "C<Block_Read()>" or manually or by another
