Index: rpcbind-0.2.0/src/security.c
===================================================================
--- rpcbind-0.2.0.orig/src/security.c	2011-03-12 17:52:51.000000000 +0100
+++ rpcbind-0.2.0/src/security.c	2011-03-12 17:53:48.000000000 +0100
@@ -53,7 +53,7 @@
 #ifndef PORTMAP_LOG_SEVERITY
 # define PORTMAP_LOG_SEVERITY LOG_INFO
 #endif
-#ifdef __linux__
+#if defined(INET6) && !defined(IPV6PORT_RESERVED)
  #define IPV6PORT_RESERVED IPPORT_RESERVED
 #endif
 
