Subject: Two words incorrectly spelt in the documentation
Forwarded: no
Author: Nicholas Bamber <nicholas@periapt.co.uk>
Last-Update: 2010-08-10
--- a/lib/SGML/Parser/OpenSP.pm
+++ b/lib/SGML/Parser/OpenSP.pm
@@ -491,7 +491,7 @@
 
 =item refc 
 
-Warn about ommitted refc delimiters. 
+Warn about omitted refc delimiters. 
 
 =item temp-ms 
 
@@ -527,7 +527,7 @@
 
 =item missing-att-name 
 
-Warn about ommitted attribute names in start tags. 
+Warn about omitted attribute names in start tags. 
 
 =item comment-decl-s 
 
@@ -613,7 +613,7 @@
 
 =head1 PROCESSING FILES
 
-In order to start processing of a document and recieve events, the
+In order to start processing of a document and receive events, the
 C<parse> method must be called. It takes one argument specifying
 the path to a file (not a file handle). You must set an event handler
 using the C<handler> method prior to using this method. The return
