From: Julian Taylor <jtaylor.debian@googlemail.com>
Date: Mon, 5 Sep 2011 02:50:23 +0200
Subject: link libfspot with needed libm

Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=640436
Forwarded: https://bugzilla.gnome.org/show_bug.cgi?id=658210
---
 lib/libfspot/Makefile.in |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/lib/libfspot/Makefile.in b/lib/libfspot/Makefile.in
index ecb0bab..08fd448 100644
--- a/lib/libfspot/Makefile.in
+++ b/lib/libfspot/Makefile.in
@@ -366,6 +366,7 @@ libfspot_la_SOURCES = \
 
 libfspot_la_LIBADD = \
 	-lX11						\
+	-lm						\
 	$(F_LIBS)					\
 	$(LCMS_LIBS)					\
 	$(GIO_LIBS)
-- 
