Subject: Add -O2 to CFLAGS by default, -O0 if noopt is set.
Author: Alessio Treglia <alessio@debian.org>
Forwarded: not-needed
---
 configure |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- gpac.orig/configure
+++ gpac/configure
@@ -328,7 +328,7 @@ for opt do
             ;;
         --cpu=*) cpu=`echo $opt | cut -d '=' -f 2`
             ;;
-        --enable-debug) debuginfo="yes"; no_gcc_opt="yes"
+        --enable-debug) debuginfo="yes";
             ;;
         --disable-opt) no_gcc_opt="yes"
             ;;
@@ -612,7 +612,7 @@ fi
 
 #GCC opt
 if test "$no_gcc_opt" = "no"; then
-    CFLAGS="-O3 $CFLAGS"
+    CFLAGS="-O2 $CFLAGS"
 else
     CFLAGS="-O0 $CFLAGS"
 fi
