--- a/src/accerciser.in
+++ b/src/accerciser.in
@@ -20,10 +20,6 @@
 # We can't rely on prefix if we're installed by relocated RPM. Instead, we 
 # use __file__ and for now hope that lib is relative to bin.
 sys.prefix = '@prefix@'
-libs = os.path.join(sys.prefix, 'lib',
-                    'python@PYTHON_VERSION@', 'site-packages')
-# point to the proper site-packages path
-sys.path.insert(1, libs)

 # TODO: Remove completely gnome dependency
 # make this program accessible
