Description: fix a spelling mistake
Origin: vendor
Bug: https://rt.cpan.org/Ticket/Display.html?id=73143
Forwarded: https://rt.cpan.org/Ticket/Display.html?id=73143
Author: gregor herrmann <gregoa@debian.org>
Last-Update: 2011-12-09

--- a/lib/IO/Socket/INET6.pm
+++ b/lib/IO/Socket/INET6.pm
@@ -460,7 +460,7 @@
     ReusePort	Set SO_REUSEPORT before binding
     Broadcast	Set SO_BROADCAST before binding
     Timeout	Timeout	value for various operations
-    MultiHomed  Try all adresses for multi-homed hosts
+    MultiHomed  Try all addresses for multi-homed hosts
     Blocking    Determine if connection will be blocking mode
 
 If C<Listen> is defined then a listen socket is created, else if the
