Index: xterm/xterm.man
===================================================================
--- xterm.orig/xterm.man
+++ xterm/xterm.man
@@ -85,7 +85,7 @@
 .el       .ds '' ''
 .TH XTERM 1 "__app_date__" "__app_version__" "X Window System"
 .SH NAME
-\*n \- terminal emulator for X
+xterm \- terminal emulator for X
 .SH SYNOPSIS
 .B \*n
 [\-\fItoolkitoption\fP ...] [\-\fIoption\fP ...] [\fIshell\fP]
