Index: xterm/xterm.man
===================================================================
--- xterm.orig/xterm.man
+++ xterm/xterm.man
@@ -1707,7 +1707,7 @@ The default is \*(``true\*(''.
 .TP
 .B "allowFontOps (\fPclass\fB AllowFontOps)"
 Specifies whether control sequences that set/query the font should be allowed.
-The default is \*(``true\*(''.
+The default is \*(``false\*(''.
 .TP 8
 .B "allowScrollLock (\fPclass\fB AllowScrollLock)"
 Specifies whether control sequences that set/query
