--- mdds-0.6.1.orig/Makefile.in
+++ mdds-0.6.1/Makefile.in
@@ -207,7 +207,6 @@ install: $(HEADERS)
 	install -m 644 -t $(DESTDIR)@includedir@/mdds $(INCDIR)/mdds/*.inl
 	install -m 644 -t $(DESTDIR)@includedir@/mdds/compat $(INCDIR)/mdds/compat/*.hpp
 	install -m 644 -t $(DESTDIR)@includedir@/mdds/hash_container $(INCDIR)/mdds/hash_container/*.hpp
-	install -m 644 -t $(DESTDIR)@docdir@ AUTHORS COPYING NEWS README VERSION
 
 check: $(ALL_TESTS)
 
