Index: hicolor-icon-theme-0.12/index.theme
===================================================================
--- hicolor-icon-theme-0.12.orig/index.theme	2010-01-13 11:18:27.000000000 +0100
+++ hicolor-icon-theme-0.12/index.theme	2010-09-08 19:56:12.374834083 +0200
@@ -138,7 +138,7 @@ Type=Threshold
 [22x22/apps]
 Size=22
 Context=Applications
-Type=Threshold
+Type=Fixed
 
 [22x22/categories]
 Size=22
