Index: gvfs-exp/daemon/Makefile.am
===================================================================
--- gvfs-exp.orig/daemon/Makefile.am	2012-03-19 17:18:57.000000000 +0100
+++ gvfs-exp/daemon/Makefile.am	2012-03-21 09:01:10.620172442 +0100
@@ -18,12 +18,14 @@
 	-DMOUNTABLE_DIR=\"$(mountdir)/\" 	\
 	-DGVFS_LOCALEDIR=\""$(localedir)"\"
 
-noinst_LTLIBRARIES=libdaemon.la
+privlibdir = $(libdir)/gvfs
+
+privlib_LTLIBRARIES=libgvfsdaemon.la
 
 libraries =						\
-	libdaemon.la					\
+	libgvfsdaemon.la				\
 	$(top_builddir)/common/libgvfscommon.la		\
-	$(GLIB_LIBS) $(DBUS_LIBS) $(KEYRING_LIBS)
+	$(GLIB_LIBS) $(DBUS_LIBS)
 
 gvfs_gschemas =
 gvfs_gschemas_convert =
@@ -132,7 +134,7 @@
 	gvfsd-test			\
 	$(NULL)
 
-libdaemon_la_SOURCES = \
+libgvfsdaemon_la_SOURCES = \
 	gvfsdaemon.c gvfsdaemon.h \
 	gvfsbackend.c gvfsbackend.h \
 	gvfschannel.c gvfschannel.h \
@@ -180,6 +182,12 @@
 	gvfskeyring.h gvfskeyring.c \
         $(NULL)
 
+libgvfsdaemon_la_LDFLAGS = -avoid-version
+
+libgvfsdaemon_la_LIBADD = \
+	$(top_builddir)/common/libgvfscommon.la		\
+	$(GLIB_LIBS) $(DBUS_LIBS) $(KEYRING_LIBS)
+
 gvfsd_SOURCES = \
 	mount.c mount.h \
 	main.c
Index: gvfs-exp/common/Makefile.am
===================================================================
--- gvfs-exp.orig/common/Makefile.am	2012-03-19 17:18:57.000000000 +0100
+++ gvfs-exp/common/Makefile.am	2012-03-21 09:01:10.620172442 +0100
@@ -1,6 +1,8 @@
 NULL =
 
-lib_LTLIBRARIES=libgvfscommon.la
+privlibdir=$(libdir)/gvfs
+
+privlib_LTLIBRARIES=libgvfscommon.la
 
 INCLUDES = -I$(top_srcdir) -I$(top_srcdir)/gvfs \
 	-I$(top_srcdir)/daemon \
@@ -22,7 +24,7 @@
 	$(NULL)
 
 # needed by cygwin (see bug #564003)
-libgvfscommon_la_LDFLAGS = -no-undefined
+libgvfscommon_la_LDFLAGS = -no-undefined  -avoid-version
 
 libgvfscommon_la_LIBADD =	\
 	$(DBUS_LIBS) 		\
@@ -30,7 +32,7 @@
 	$(BLURAY_LIBS)
 
 if HAVE_AVAHI
-lib_LTLIBRARIES += libgvfscommon-dnssd.la
+privlib_LTLIBRARIES += libgvfscommon-dnssd.la
 
 libgvfscommon_dnssd_la_SOURCES = 	\
 	gvfsdnssdutils.c gvfsdnssdutils.h \
@@ -40,6 +42,8 @@
 libgvfscommon_dnssd_la_CFLAGS =	\
 	$(AVAHI_CFLAGS)
 
+libgvfscommon_dnssd_la_LDFLAGS = -avoid-version
+
 libgvfscommon_dnssd_la_LIBADD =	\
 	libgvfscommon.la \
 	$(AVAHI_LIBS) \
