---
 configure.ac    |    1 +
 src/Makefile.am |    1 +
 2 files changed, 2 insertions(+)

--- simple-scan.orig/configure.ac
+++ simple-scan/configure.ac
@@ -39,6 +39,7 @@ AM_CONDITIONAL(HAVE_COLORD, test $have_c
 
 AC_CHECK_HEADERS([sane/sane.h],[],[AC_MSG_ERROR([SANE not found])])
 AC_CHECK_HEADERS([sane/saneopts.h],[],[AC_MSG_ERROR([SANE not found])])
+AC_CHECK_HEADERS([zlib.h],[],[AC_MSG_ERROR([ZLib not found])])
 AC_CHECK_HEADERS([jpeglib.h],[],[AC_MSG_ERROR([libjpeg not found])])
 
 dnl ###########################################################################
--- simple-scan.orig/src/Makefile.am
+++ simple-scan/src/Makefile.am
@@ -40,6 +40,7 @@ simple_scan_LDADD = \
 	$(COLORD_LIBS) \
 	-lsane \
 	-ljpeg \
+	-lz \
 	-lm
 
 CLEANFILES = \
