--- a/tools/scripts/zonesigner
+++ b/tools/scripts/zonesigner
@@ -1983,6 +1983,7 @@ sub ssetkeytype
 
 	my $keys;				# Keys we're modifying.
 	my @keylist;				# Keys inna list.
+	my @args;
 
 	#
 	# Change the type of the given signing set to the specified type.
@@ -2018,6 +2019,11 @@ sub ssetkeytype
 				vmed_print("setting revoke bit($cwd): $cmd\n");
 
 				System($cmd);
+				if ($? != 0)
+				{
+				    print STDERR "'$cmd' returned $?\n" if($verbose);
+				    exit($?);
+				}
 
 				#
 				# Record the time when key is revoked.
@@ -2049,7 +2055,13 @@ sub ssetkeytype
 	if(! -e $archdir)
 	{
 		vmed_print("creating key archive directory $archdir\n\n");
-		System("$MKDIR -p -m 0700 $archdir");
+		@args = ($MKDIR, "-p", "-m 0700", $archdir);
+		System(@args);
+		if ($? != 0)
+		{
+		    print STDERR "'@args' returned $?\n" if($verbose);
+		    exit($?);
+		}
 	}
 	else
 	{
@@ -2077,7 +2089,13 @@ sub ssetkeytype
 			$newname = "$archdir/$kronos.$archfn";
 			vhigh_print("moving $key to $newname\n\n");
 
-			System("$MV $archfn $newname");
+			@args = ($MV, $archfn, $newname);
+			System(@args);
+			if ($? != 0)
+			{
+			    print STDERR "'@args' returned $?\n" if($verbose);
+			    exit($?);
+			}
 
 			if($archfn =~ /\.key$/)
 			{
@@ -2162,6 +2180,7 @@ sub revkeys
 sub keydirs
 {
 	my $cwd = getcwd();				# Current directory.
+	my @args;
 
 	vmed_print("checking key directories\n");
 
@@ -2184,12 +2203,24 @@ sub keydirs
 
 		foreach my $ksk (@kskcurlist)
 		{
-			System("$MV $ksk.* $kskdir");
+			@args = ($MV, "$ksk.*", $kskdir);
+			System(@args);
+			if ($? != 0)
+			{
+			    print STDERR "'@args' returned $?\n" if($verbose);
+			    exit($?)
+			}
 		}
 
 		foreach my $ksk (@kskpublist)
 		{
-			System("$MV $ksk.* $kskdir");
+			@args = ($MV, "$ksk.*", $kskdir);
+			System(@args);
+			if ($? != 0)
+			{
+			    print STDERR "'@args' returned $?\n" if($verbose);
+			    exit($?)
+			}
 		}
 
 	}
@@ -2213,17 +2244,35 @@ sub keydirs
 
 		foreach my $zsk (@zskcurlist)
 		{
-			System("$MV $zsk.* $zskdir");
+			@args = ($MV, "$zsk.*", $zskdir);
+			System(@args);
+			if ($? != 0)
+			{
+			    print STDERR "'@args' returned $?\n" if($verbose);
+			    exit($?)
+			}
 		}
 
 		foreach my $zsk (@zskpublist)
 		{
-			System("$MV $zsk.* $zskdir");
+			@args = ($MV, "$zsk.*", $zskdir);
+			System(@args);
+			if ($? != 0)
+			{
+			    print STDERR "'@args' returned $?\n" if($verbose);
+			    exit($?)
+			}
 		}
 
 		foreach my $zsk (@zsknewlist)
 		{
-			System("$MV $zsk.* $zskdir");
+			@args = ($MV, "$zsk.*", $zskdir);
+			System(@args);
+			if ($? != 0)
+			{
+			    print STDERR "'@args' returned $?\n" if($verbose);
+			    exit($?)
+			}
 		}
 	}
 
@@ -2240,6 +2289,7 @@ sub zoneincludes
 	my $file;				# Zone's contents.
 	my $flen;				# Zone file's length.
 	my $newserial;				# Zone's new serial number.
+	my @args;
 
 	vhigh_print("\n");
 	vprint("adding key includes to zone file\n");
@@ -2345,7 +2395,16 @@ sub zoneincludes
 		#
 		# Copy the zone data to a new file.
 		#
-		System("$CP $zonefile $zoneftmp") if($zonefile ne $zoneftmp);
+		if ($zonefile ne $zoneftmp)
+		{
+		    @args = ($CP, $zonefile, $zoneftmp);
+		    System(@args);
+		    if ($? != 0)
+		    {
+			print STDERR "'@args' returned $?\n" if($verbose);
+			exit($?);
+		    }
+		}
 		open(ZF,"+< $zonefile");
 		@zonestat = stat($zonefile);
 	}
@@ -2356,7 +2415,16 @@ sub zoneincludes
 		#
 		# Copy the zone data to a new file.
 		#
-		System("$CP $zonefile $zoneftmp") if($zonefile ne $zoneftmp);
+		@args = ($CP, $zonefile, $zoneftmp);
+		if($zonefile ne $zoneftmp)
+		{
+		    System(@args);
+		    if ($? != 0)
+		    {
+			print STDERR "'@args' returned $?\n" if($verbose);
+			exit($?);
+		    }
+		}
 
 		#
 		# Get the include-keys section.
@@ -2391,6 +2459,7 @@ sub zonesign
 	my $status;			# Execution return code.
 	my $zscmd;			# Zone-signing command line.
 	my $zcq = '-q';			# Quiet option for zone checker.
+	my @args;
 
 	#
 	# Get the most recent revoked keys if we're rolling KSKs or if
@@ -2459,8 +2528,8 @@ sub zonesign
 	#
 	vprint("checking zone\n");
 	$zcq = '' if($verbose > $VERBOSE_LOW);
-	$status = System("$zonecheck $zcopts $zcq $zone $szone");
-	if($status != 0)
+	System("$zonecheck $zcopts $zcq $zone $szone");
+	if($? != 0)
 	{
 		print STDERR "problems with zone signing\n";
 		System("$zonecheck $zcopts $zone $szone");
@@ -2590,10 +2659,22 @@ sub zonesign
 		{
 			my $now = time();
 			my $newzftmp = "$zoneftmp.$now";
-			System("$CP $zoneftmp $newzftmp");
+			@args = ($CP, $zoneftmp, $newzftmp);
+			System(@args);
+			if ($? != 0)
+			{
+			    print STDERR "'@args' returned $?\n" if($verbose);
+			    exit($?);
+			}
 		}
 
-		System("$RM $zoneftmp");
+		@args = ($RM, $zoneftmp);
+		System(@args);
+		if ($? != 0)
+		{
+		    print STDERR "'@args' returned $?\n" if($verbose);
+		    exit($?);
+		}
 	}
 
 	return(0);
