--- a/configure.in
+++ b/configure.in
@@ -24,15 +24,15 @@ AC_PATH_PROG(RM, rm)
 AC_ARG_ENABLE(bind-checks,
 [  --disable-bind-checks   Disable checks for bind dnssec utilities])
 if test "x$enable_bind_checks" != "xno"; then
-   AC_PATH_PROG(BIND_DNSSEC_KEYGEN, dnssec-keygen)
+   AC_PATH_PROG(BIND_DNSSEC_KEYGEN, dnssec-keygen, path = '/usr/sbin')
    if test -z "$BIND_DNSSEC_KEYGEN"; then
      AC_ERROR([Could not locate dnssec-keygen. Please install BIND utilities.])
    fi
-   AC_PATH_PROG(BIND_DNSSEC_SIGNZONE, dnssec-signzone)
+   AC_PATH_PROG(BIND_DNSSEC_SIGNZONE, dnssec-signzone, path = '/usr/sbin')
    if test -z "$BIND_DNSSEC_SIGNZONE"; then
      AC_ERROR([Could not locate dnssec-signzone. Please install BIND utilities.])
    fi
-   AC_PATH_PROG(BIND_DNSSEC_CHECKZONE, named-checkzone)
+   AC_PATH_PROG(BIND_DNSSEC_CHECKZONE, named-checkzone, path = '/usr/sbin')
    if test -z "$BIND_DNSSEC_CHECKZONE"; then
      AC_ERROR([Could not locate named-checkzone. Please install BIND utilities.])
    fi
