--- a/tools/modules/defaults.pm
+++ b/tools/modules/defaults.pm
@@ -33,12 +33,12 @@ my $installdir = getprefixdir() . "/sbin
 my %defaults =
 (
 	'admin-email'	   => "root",		# Admin's email address.
-	'archivedir'	   => getprefixdir() . "/var/key-archive",
-	'algorithm'	   => "rsasha1",	# Encryption algorithm.
+	'archivedir'	   => "/var/lib/dnssec-tools/archive",
+	'algorithm'	   => "rsasha256",	# Encryption algorithm.
 	'autosign'	   => 1,		# Auto-sign zone files flag.
 	'enddate'	   => "+2764800",	# Zone life, in seconds.
 	'entropy_msg'	   => 1,		# Display entropy message flag.
-	'keygen'	   => getprefixdir() . "/sbin/dnssec-keygen",
+	'keygen'	   => "/usr/sbin/dnssec-keygen",
 	'keygen-opts'	   => "",		# Options for key generator.
 	'kskcount'	   => 1,		# Number of KSK keys.
 	'ksklength'	   => 2048,		# Length of KSK key.
@@ -59,9 +59,9 @@ my %defaults =
 	'prog-zsk3'	   => 'default',	# Program for ZSK phase 3.
 	'prog-zsk4'	   => 'default',	# Program for ZSK phase 4.
 	'random'	   => "/dev/urandom",	# Random no. generator device.
-	'rndc'		   => getprefixdir() . "/sbin/rndc",
+	'rndc'		   => "/usr/sbin/rndc",
 	'roll_loadzone'	   => 1,		# Zone-reloading flag.
-	'roll_logfile'	   => makelocalstatedir() . "/log.rollerd",
+	'roll_logfile'	   => "/var/log/dnssec-tools/rollerd.log",
 	'roll_loglevel'	   => "phase",		# Rollerd's logging level.
 	'roll_phasemsg'	   => "long",		# Rollerd's phase logmsg length.
 	'roll_sleeptime'   => 3600,		# Rollerd's sleep time.
@@ -70,18 +70,18 @@ my %defaults =
 	'mailer-server'    => "localhost",	# Mail server.
 	'mailer-type'      => "smtp",		# Mail type.
 	'tacontact' 	   => "",
-	'tatmpdir' 	   => "/tmp/dnssec-tools/trustman",
-	'tadnsvalconffile' => getconfdir() . "/dnsval.conf",
-	'tanamedconffile'  => getconfdir() . "/named/named.conf",
+	'tatmpdir' 	   => "/var/run/dnssec-tools/trustman",
+	'tadnsvalconffile' => "/etc/dnssec-tools/dnsval.conf",
+	'tanamedconffile'  => "/etc/bind/named.conf",
 	'tasleeptime'      => 3600,
 	'tasmtpserver'     => "localhost",	# Trustman's SMTP server.
 	'taresolvconf'	   => "/etc/resolv.conf",	# resolv.conf file.
 	'usegui'	   => 0,		# Use GUI for option entry flag.
 #	'viewimage'	   => "/usr/X11R6/bin/viewimage",
 	'zone_errors'	   => 5,
-	'zonecheck'	   => getprefixdir() . "/sbin/named-checkzone",
+	'zonecheck'	   => "/usr/sbin/named-checkzone",
 	'zonecheck-opts'   => "-i local",	# Options for zone checker.
-	'zonesign'	   => getprefixdir() . "/sbin/dnssec-signzone",
+	'zonesign'	   => "/usr/sbin/dnssec-signzone",
 	'zonesign-opts'	   => "",		# Options for zone signer.
 	'zskcount'	   => 1,		# Number of Current ZSK keys.
 	'zsklength'	   => 1024,		# Length of ZSK key.
--- a/tools/modules/rollmgr.pm
+++ b/tools/modules/rollmgr.pm
@@ -468,7 +468,7 @@ my %port_archs =
 # 
 
 my $UNIX_ROLLMGR_DIR	  = makelocalstatedir("run");
-our $UNIX_ROLLMGR_PIDFILE = ($UNIX_ROLLMGR_DIR . "/rollmgr.pid");
+our $UNIX_ROLLMGR_PIDFILE = ($UNIX_ROLLMGR_DIR . "/rollerd.pid");
 
 my $PS = "/bin/ps";
 
@@ -1591,7 +1591,7 @@ sub rollmgr_channel
 		#
 		# Build the socket name and construct the socket data.
 		#
-		$unixsock = makelocalstatedir("/dnssec-tools") . $UNIXSOCK;
+		$unixsock = makelocalstatedir("run/dnssec-tools") . $UNIXSOCK;
 # print STDERR "rollmgr_channel:  unixsock - <$unixsock>\n";
 
 		#
--- a/tools/modules/rollrec.pm
+++ b/tools/modules/rollrec.pm
@@ -188,7 +188,7 @@ sub rollrec_lock
 	#
 	# Get the DNSSEC-Tools config directory.
 	#
-	$lockdir = makelocalstatedir("/dnssec-tools") || $DEFAULT_DNSSECTOOLS_DIR;
+	$lockdir = makelocalstatedir("run/dnssec-tools") || $DEFAULT_DNSSECTOOLS_DIR;
 
 	#
 	# Build our lock file.
