--- a/tools/etc/dnssec-tools/dnssec-tools.conf
+++ b/tools/etc/dnssec-tools/dnssec-tools.conf
@@ -19,10 +19,10 @@ rollchk		/usr/bin/rollchk
 rollctl		/usr/bin/rollctl
 zonesigner	/usr/bin/zonesigner
 
-keygen		/usr/local/sbin/dnssec-keygen
-rndc		/usr/local/sbin/rndc
-zonecheck	/usr/local/sbin/named-checkzone
-zonesign	/usr/local/sbin/dnssec-signzone
+keygen		/usr/sbin/dnssec-keygen
+rndc		/usr/sbin/rndc
+zonecheck	/usr/sbin/named-checkzone
+zonesign	/usr/sbin/dnssec-signzone
 
 zonecheck-opts	-i local
 
@@ -64,7 +64,7 @@ lifespan-min	3600
 # Settings that will be noticed by zonesigner.
 #
 # default_keyrec        output.krf
-archivedir      /usr/local/etc/dnssec-tools/KEY-SAFE
+archivedir      /var/lib/dnssec-tools/archive
 entropy_msg	1
 savekeys        1
 kskcount	1
@@ -75,7 +75,7 @@ zskcount	1
 #
 autosign	1
 roll_loadzone	1
-roll_logfile	/usr/local/etc/dnssec-tools/log-rollerd
+roll_logfile	/var/log/dnssec-tools/rollerd.log
 roll_loglevel	info
 roll_phasemsg	long
 roll_sleeptime	60
--- a/tools/modules/defaults.pm
+++ b/tools/modules/defaults.pm
@@ -28,7 +28,7 @@ our @EXPORT = qw(
 our $VERSION = "1.13";
 our $MODULE_VERSION = "1.13.0";
 
-my $installdir = getprefixdir() . "/bin"; # DNSSEC-Tools installation directory.
+my $installdir = getprefixdir() . "/sbin"; # DNSSEC-Tools installation directory.
 
 my %defaults =
 (
