--- a/tools/donuts/donuts
+++ b/tools/donuts/donuts
@@ -42,7 +42,7 @@ my %opts = (l => 5,
 	    c => $ENV{'HOME'} . "/.donuts.conf",
 	    T => 'port 53 || ip[6:2] & 0x1fff != 0',
 	    o => '%d.%t.pcap',
-	    r => "/usr/local/share/dnssec-tools/donuts/rules/*.txt," .
+	    r => "/usr/share/dnssec-tools/donuts/rules/*.txt," .
 	    $ENV{'HOME'} . "/.dnssec-tools/donuts/rules/*.txt");
 
 my $TCPDUMP = "tcpdump";
--- a/tools/donuts/Makefile.PL
+++ b/tools/donuts/Makefile.PL
@@ -20,7 +20,7 @@ WriteMakefile(%opts);
 sub MY::postamble {
     return '
 
-RULESDIR=$(DESTDIR)/$(PREFIX)/share/dnssec-tools/donuts/rules
+RULESDIR=$(DESTDIR)/usr/share/dnssec-tools/donuts/rules
 
 INSTALL=$(PERL) "-MExtUtils::Command" -e cp
 
