Description: Fix build with gcc 4.7, add missing includes.
Origin: vendor
Forwarded: not-needed
Author: Sebastian Ramacher <s.ramacher@gmx.at>, Steven Chamberlain <steven@pyro.eu.org>

--- a/src/engines/CurlDownload.cpp
+++ b/src/engines/CurlDownload.cpp
@@ -45,6 +45,7 @@
 #include <QtDebug>
 #include <iostream>
 #include <errno.h>
+#include <unistd.h>
 
 CurlDownload::CurlDownload()
 	: m_curl(0), m_nTotal(0), m_nStart(0), m_bAutoName(false), m_nUrl(0), m_file(0)
--- a/src/poller/EpollPoller.cpp
+++ b/src/poller/EpollPoller.cpp
@@ -30,6 +30,7 @@
 #include <sys/epoll.h>
 #include <errno.h>
 #include <alloca.h>
+#include <unistd.h>
 
 EpollPoller::EpollPoller(QObject* parent)
 	: Poller(parent)
--- a/src/engines/RapidshareFreeDownload.cpp
+++ b/src/engines/RapidshareFreeDownload.cpp
@@ -36,6 +36,7 @@
 #include <QApplication>
 #include <QMessageBox>
 #include <QMutex>
+#include <unistd.h>
 
 static QMutex m_mInstanceActive;
 
--- a/src/poller/KqueuePoller.cpp
+++ b/src/poller/KqueuePoller.cpp
@@ -31,6 +31,7 @@
 #include <sys/event.h>
 #include <sys/time.h>
 #include <alloca.h>
+#include <unistd.h>
 
 KqueuePoller::KqueuePoller(QObject* parent)
 	: Poller(parent)
