# Fix a few spelling errors in the source code.

diff -Naur fotowall-0.9.orig/3rdparty/posterazor/de.ts fotowall-0.9/3rdparty/posterazor/de.ts
--- fotowall-0.9.orig/3rdparty/posterazor/de.ts	2009-12-01 14:09:37.000000000 +0100
+++ fotowall-0.9/3rdparty/posterazor/de.ts	2012-02-24 20:07:21.535531098 +0100
@@ -47,8 +47,8 @@
     <message>
         <location line="+9"/>
         <source>Load an image by clicking the button with the open icon and selecting an image file, or by drag &amp; dropping an image file on the PosteRazor. The drag &amp; drop also works during the other steps.
-After loading the image, the most important informations are listed in the &lt;b&gt;%1&lt;/b&gt; fields.</source>
-        <comment>Wizard step 1. Place holders: %1 = &apos;Image informations&apos; (will be automatically inserted)</comment>
+After loading the image, the most important information are listed in the &lt;b&gt;%1&lt;/b&gt; fields.</source>
+        <comment>Wizard step 1. Place holders: %1 = &apos;Image information&apos; (will be automatically inserted)</comment>
         <translation type="unfinished">Laden Sie ein Ausgangsbild, indem sie auf das &apos;öffnen&apos;-Symbol klicken und eine Datei wählen, oder indem Sie eine Datei auf den PosteRazor fallen lassen (Drag &amp; Drop). Die Drag &amp; Drop-Funktion des Ausgangsbildes ist in allen Assistenten-Schritten vorhanden.
 Nach dem Laden eines Bildes werden in dem Feld &lt;b&gt;%1&lt;/b&gt; die wichtigsten Angaben aufgelistet.</translation>
     </message>
@@ -128,7 +128,7 @@
     <message>
         <location filename="wizard.cpp" line="+90"/>
         <location filename="wizardcontroller.cpp" line="+93"/>
-        <source>Image Informations</source>
+        <source>Image Information</source>
         <translation type="unfinished">Bildinformationen</translation>
     </message>
     <message>
@@ -397,7 +397,7 @@
     </message>
     <message>
         <location/>
-        <source>Image Informations</source>
+        <source>Image Information</source>
         <translation type="unfinished">Bildinformationen</translation>
     </message>
     <message>
diff -Naur fotowall-0.9.orig/3rdparty/posterazor/en.ts fotowall-0.9/3rdparty/posterazor/en.ts
--- fotowall-0.9.orig/3rdparty/posterazor/en.ts	2009-12-01 14:09:37.000000000 +0100
+++ fotowall-0.9/3rdparty/posterazor/en.ts	2012-02-24 20:07:21.539531311 +0100
@@ -55,8 +55,8 @@
     <message>
         <location line="+9"/>
         <source>Load an image by clicking the button with the open icon and selecting an image file, or by drag &amp; dropping an image file on the PosteRazor. The drag &amp; drop also works during the other steps.
-After loading the image, the most important informations are listed in the &lt;b&gt;%1&lt;/b&gt; fields.</source>
-        <comment>Wizard step 1. Place holders: %1 = &apos;Image informations&apos; (will be automatically inserted)</comment>
+After loading the image, the most important information are listed in the &lt;b&gt;%1&lt;/b&gt; fields.</source>
+        <comment>Wizard step 1. Place holders: %1 = &apos;Image information&apos; (will be automatically inserted)</comment>
         <translation type="unfinished"></translation>
     </message>
     <message>
@@ -163,7 +163,7 @@
     <message>
         <location line="-6"/>
         <location filename="wizardcontroller.cpp" line="+86"/>
-        <source>Image Informations</source>
+        <source>Image Information</source>
         <translation type="unfinished"></translation>
     </message>
     <message>
@@ -387,7 +387,7 @@
     </message>
     <message>
         <location/>
-        <source>Image Informations</source>
+        <source>Image Information</source>
         <translation type="unfinished"></translation>
     </message>
     <message>
diff -Naur fotowall-0.9.orig/3rdparty/posterazor/es.ts fotowall-0.9/3rdparty/posterazor/es.ts
--- fotowall-0.9.orig/3rdparty/posterazor/es.ts	2009-12-01 14:09:37.000000000 +0100
+++ fotowall-0.9/3rdparty/posterazor/es.ts	2012-02-24 20:07:21.543531443 +0100
@@ -47,8 +47,8 @@
     <message>
         <location line="+9"/>
         <source>Load an image by clicking the button with the open icon and selecting an image file, or by drag &amp; dropping an image file on the PosteRazor. The drag &amp; drop also works during the other steps.
-After loading the image, the most important informations are listed in the &lt;b&gt;%1&lt;/b&gt; fields.</source>
-        <comment>Wizard step 1. Place holders: %1 = &apos;Image informations&apos; (will be automatically inserted)</comment>
+After loading the image, the most important information are listed in the &lt;b&gt;%1&lt;/b&gt; fields.</source>
+        <comment>Wizard step 1. Place holders: %1 = &apos;Image information&apos; (will be automatically inserted)</comment>
         <translation type="unfinished">Cargue una imagen haciendo clic sobre el botón del icono de Abrir y seleccione un archivo de imagen, o arrastre y suelte un archivo de imagen dentro de PosteRazor. La función arrastrar-y-soltar también opera en los demás pasos.
 Después de cargar la imagen se presentan las informaciones más relevantes en los campos de &lt;b&gt;%1&lt;/b&gt;.</translation>
     </message>
@@ -128,7 +128,7 @@
     <message>
         <location filename="wizard.cpp" line="+90"/>
         <location filename="wizardcontroller.cpp" line="+93"/>
-        <source>Image Informations</source>
+        <source>Image Information</source>
         <translation type="unfinished">Informaciones de imagen</translation>
     </message>
     <message>
@@ -397,7 +397,7 @@
     </message>
     <message>
         <location/>
-        <source>Image Informations</source>
+        <source>Image Information</source>
         <translation type="unfinished">Informaciones de imagen</translation>
     </message>
     <message>
diff -Naur fotowall-0.9.orig/3rdparty/posterazor/fi.ts fotowall-0.9/3rdparty/posterazor/fi.ts
--- fotowall-0.9.orig/3rdparty/posterazor/fi.ts	2009-12-01 14:09:37.000000000 +0100
+++ fotowall-0.9/3rdparty/posterazor/fi.ts	2012-02-24 20:07:21.543531443 +0100
@@ -47,8 +47,8 @@
     <message>
         <location line="+9"/>
         <source>Load an image by clicking the button with the open icon and selecting an image file, or by drag &amp; dropping an image file on the PosteRazor. The drag &amp; drop also works during the other steps.
-After loading the image, the most important informations are listed in the &lt;b&gt;%1&lt;/b&gt; fields.</source>
-        <comment>Wizard step 1. Place holders: %1 = &apos;Image informations&apos; (will be automatically inserted)</comment>
+After loading the image, the most important information are listed in the &lt;b&gt;%1&lt;/b&gt; fields.</source>
+        <comment>Wizard step 1. Place holders: %1 = &apos;Image information&apos; (will be automatically inserted)</comment>
         <translation type="unfinished"></translation>
     </message>
     <message>
@@ -119,7 +119,7 @@
     <message>
         <location filename="wizard.cpp" line="+90"/>
         <location filename="wizardcontroller.cpp" line="+93"/>
-        <source>Image Informations</source>
+        <source>Image Information</source>
         <translation type="unfinished">Kuvan tiedot</translation>
     </message>
     <message>
@@ -389,7 +389,7 @@
     </message>
     <message>
         <location/>
-        <source>Image Informations</source>
+        <source>Image Information</source>
         <translation type="unfinished">Kuvan tiedot</translation>
     </message>
     <message>
diff -Naur fotowall-0.9.orig/3rdparty/posterazor/fr.ts fotowall-0.9/3rdparty/posterazor/fr.ts
--- fotowall-0.9.orig/3rdparty/posterazor/fr.ts	2009-12-01 14:09:37.000000000 +0100
+++ fotowall-0.9/3rdparty/posterazor/fr.ts	2012-02-24 20:07:21.543531443 +0100
@@ -47,10 +47,10 @@
     <message>
         <location line="+9"/>
         <source>Load an image by clicking the button with the open icon and selecting an image file, or by drag &amp; dropping an image file on the PosteRazor. The drag &amp; drop also works during the other steps.
-After loading the image, the most important informations are listed in the &lt;b&gt;%1&lt;/b&gt; fields.</source>
-        <comment>Wizard step 1. Place holders: %1 = &apos;Image informations&apos; (will be automatically inserted)</comment>
+After loading the image, the most important information are listed in the &lt;b&gt;%1&lt;/b&gt; fields.</source>
+        <comment>Wizard step 1. Place holders: %1 = &apos;Image information&apos; (will be automatically inserted)</comment>
         <translation type="unfinished">Chargez une image en cliquant sur l&apos;icône en forme de dossier. Choisissez le fichier en naviguant dans vos dossiers ou glissez-déposez l&apos;image dans la fenêtre. Le glissé-déposé marche aussi dans les autres étapes de l&apos;assistant.
-Après avoir chargé l&apos;image, les informations pratiques (taille, couleur, résolution) sont affichées dans la rubrique &lt;b&gt;%1&lt;/b&gt;.</translation>
+Après avoir chargé l&apos;image, les information pratiques (taille, couleur, résolution) sont affichées dans la rubrique &lt;b&gt;%1&lt;/b&gt;.</translation>
     </message>
     <message>
         <location line="+7"/>
@@ -128,8 +128,8 @@
     <message>
         <location filename="wizard.cpp" line="+90"/>
         <location filename="wizardcontroller.cpp" line="+93"/>
-        <source>Image Informations</source>
-        <translation type="unfinished">Informations</translation>
+        <source>Image Information</source>
+        <translation type="unfinished">Information</translation>
     </message>
     <message>
         <location line="+1"/>
@@ -398,8 +398,8 @@
     </message>
     <message>
         <location/>
-        <source>Image Informations</source>
-        <translation type="unfinished">Informations</translation>
+        <source>Image Information</source>
+        <translation type="unfinished">Information</translation>
     </message>
     <message>
         <location/>
diff -Naur fotowall-0.9.orig/3rdparty/posterazor/it.ts fotowall-0.9/3rdparty/posterazor/it.ts
--- fotowall-0.9.orig/3rdparty/posterazor/it.ts	2009-12-01 14:09:37.000000000 +0100
+++ fotowall-0.9/3rdparty/posterazor/it.ts	2012-02-24 20:07:21.547531715 +0100
@@ -47,8 +47,8 @@
     <message>
         <location line="+9"/>
         <source>Load an image by clicking the button with the open icon and selecting an image file, or by drag &amp; dropping an image file on the PosteRazor. The drag &amp; drop also works during the other steps.
-After loading the image, the most important informations are listed in the &lt;b&gt;%1&lt;/b&gt; fields.</source>
-        <comment>Wizard step 1. Place holders: %1 = &apos;Image informations&apos; (will be automatically inserted)</comment>
+After loading the image, the most important information are listed in the &lt;b&gt;%1&lt;/b&gt; fields.</source>
+        <comment>Wizard step 1. Place holders: %1 = &apos;Image information&apos; (will be automatically inserted)</comment>
         <translation>Carica l&apos;immagine premendo il bottone con l&apos;icona a forma di cartella e seleziona l&apos;immagine di origine. L&apos;immagine può essere anche trascinata all&apos;interno della finestra tramite drag &amp; drop. La procedura di drag &amp; drop funziona anche nei successivi passaggi.
 Dopo il caricamento dell&apos;immagine, le informazioni di dettaglio sono elencate nella casella &lt;b&gt;%1&lt;/b&gt;.</translation>
     </message>
@@ -128,7 +128,7 @@
     <message>
         <location filename="wizard.cpp" line="+90"/>
         <location filename="wizardcontroller.cpp" line="+93"/>
-        <source>Image Informations</source>
+        <source>Image Information</source>
         <translation>Informazioni sull&apos;immagine</translation>
     </message>
     <message>
@@ -397,7 +397,7 @@
     </message>
     <message>
         <location/>
-        <source>Image Informations</source>
+        <source>Image Information</source>
         <translation>Informazioni immagine</translation>
     </message>
     <message>
diff -Naur fotowall-0.9.orig/3rdparty/posterazor/nl.ts fotowall-0.9/3rdparty/posterazor/nl.ts
--- fotowall-0.9.orig/3rdparty/posterazor/nl.ts	2009-12-01 14:09:37.000000000 +0100
+++ fotowall-0.9/3rdparty/posterazor/nl.ts	2012-02-24 20:07:21.547531715 +0100
@@ -47,8 +47,8 @@
     <message>
         <location line="+9"/>
         <source>Load an image by clicking the button with the open icon and selecting an image file, or by drag &amp; dropping an image file on the PosteRazor. The drag &amp; drop also works during the other steps.
-After loading the image, the most important informations are listed in the &lt;b&gt;%1&lt;/b&gt; fields.</source>
-        <comment>Wizard step 1. Place holders: %1 = &apos;Image informations&apos; (will be automatically inserted)</comment>
+After loading the image, the most important information are listed in the &lt;b&gt;%1&lt;/b&gt; fields.</source>
+        <comment>Wizard step 1. Place holders: %1 = &apos;Image information&apos; (will be automatically inserted)</comment>
         <translation type="unfinished">Kies een afbeelding door op het icoon van de open map te klikken en te zoeken naar de plaats van de afbeelding, of maak gebruik van &quot;Drag and Drop&quot;. Sleep dan het afbeeldingbestand naar de PosteRazor. Dit &quot;Drag and Drop&quot; principe werkt ook bij de andere functies binnen PosteRazor.
 Veel bestands- en grafische formaten kunnen worden gebruikt. Na het kiezen van het invoerbestand worden de belangrijkste gegevens getoond in het &lt;b&gt;%1&lt;/b&gt; venster.</translation>
     </message>
@@ -128,7 +128,7 @@
     <message>
         <location filename="wizard.cpp" line="+90"/>
         <location filename="wizardcontroller.cpp" line="+93"/>
-        <source>Image Informations</source>
+        <source>Image Information</source>
         <translation type="unfinished">Afbeeldinginformatie</translation>
     </message>
     <message>
@@ -398,7 +398,7 @@
     </message>
     <message>
         <location/>
-        <source>Image Informations</source>
+        <source>Image Information</source>
         <translation type="unfinished">Afbeeldinginformatie</translation>
     </message>
     <message>
diff -Naur fotowall-0.9.orig/3rdparty/posterazor/pl.ts fotowall-0.9/3rdparty/posterazor/pl.ts
--- fotowall-0.9.orig/3rdparty/posterazor/pl.ts	2009-12-01 14:09:37.000000000 +0100
+++ fotowall-0.9/3rdparty/posterazor/pl.ts	2012-02-24 20:07:21.547531715 +0100
@@ -47,8 +47,8 @@
     <message>
         <location line="+9"/>
         <source>Load an image by clicking the button with the open icon and selecting an image file, or by drag &amp; dropping an image file on the PosteRazor. The drag &amp; drop also works during the other steps.
-After loading the image, the most important informations are listed in the &lt;b&gt;%1&lt;/b&gt; fields.</source>
-        <comment>Wizard step 1. Place holders: %1 = &apos;Image informations&apos; (will be automatically inserted)</comment>
+After loading the image, the most important information are listed in the &lt;b&gt;%1&lt;/b&gt; fields.</source>
+        <comment>Wizard step 1. Place holders: %1 = &apos;Image information&apos; (will be automatically inserted)</comment>
         <translation type="unfinished">Załaduj obraz klikając na przycisk i wybierając plik graficzny lub chwytając i przeciagając obraz na PosteRazor. Chwyć i przeciągij działa także w pozostałych krokach.
 Po załadowaniu obrazu, najważniejsze informacje są wypisane w polu &lt;b&gt;%1&lt;/b&gt;.</translation>
     </message>
@@ -126,7 +126,7 @@
     <message>
         <location filename="wizard.cpp" line="+90"/>
         <location filename="wizardcontroller.cpp" line="+93"/>
-        <source>Image Informations</source>
+        <source>Image Information</source>
         <translation type="unfinished">Informacje o obrazie</translation>
     </message>
     <message>
@@ -396,7 +396,7 @@
     </message>
     <message>
         <location/>
-        <source>Image Informations</source>
+        <source>Image Information</source>
         <translation type="unfinished">Informacje o obrazie</translation>
     </message>
     <message>
diff -Naur fotowall-0.9.orig/3rdparty/posterazor/pt_BR.ts fotowall-0.9/3rdparty/posterazor/pt_BR.ts
--- fotowall-0.9.orig/3rdparty/posterazor/pt_BR.ts	2009-12-01 14:09:37.000000000 +0100
+++ fotowall-0.9/3rdparty/posterazor/pt_BR.ts	2012-02-24 20:07:21.547531715 +0100
@@ -47,8 +47,8 @@
     <message>
         <location line="+9"/>
         <source>Load an image by clicking the button with the open icon and selecting an image file, or by drag &amp; dropping an image file on the PosteRazor. The drag &amp; drop also works during the other steps.
-After loading the image, the most important informations are listed in the &lt;b&gt;%1&lt;/b&gt; fields.</source>
-        <comment>Wizard step 1. Place holders: %1 = &apos;Image informations&apos; (will be automatically inserted)</comment>
+After loading the image, the most important information are listed in the &lt;b&gt;%1&lt;/b&gt; fields.</source>
+        <comment>Wizard step 1. Place holders: %1 = &apos;Image information&apos; (will be automatically inserted)</comment>
         <translation type="unfinished">Carregue a imagem clicando no botão com o ícone de abrir e selecione o arquivo de imagem, ou clique &amp; arraste uma imagem para o PosteRazor. O clique &amp; arraste também funciona nos outros passos.
 Após carregar uma imagem, a informações mais importantes são listadas nos campos &lt;b&gt;%1&lt;/b&gt;.</translation>
     </message>
@@ -128,7 +128,7 @@
     <message>
         <location filename="wizard.cpp" line="+90"/>
         <location filename="wizardcontroller.cpp" line="+93"/>
-        <source>Image Informations</source>
+        <source>Image Information</source>
         <translation type="unfinished">Informações da imagem</translation>
     </message>
     <message>
@@ -398,7 +398,7 @@
     </message>
     <message>
         <location/>
-        <source>Image Informations</source>
+        <source>Image Information</source>
         <translation type="unfinished">Informações da imagem</translation>
     </message>
     <message>
diff -Naur fotowall-0.9.orig/3rdparty/posterazor/pt_PT.ts fotowall-0.9/3rdparty/posterazor/pt_PT.ts
--- fotowall-0.9.orig/3rdparty/posterazor/pt_PT.ts	2009-12-01 14:09:37.000000000 +0100
+++ fotowall-0.9/3rdparty/posterazor/pt_PT.ts	2012-02-24 20:07:21.551531994 +0100
@@ -47,8 +47,8 @@
     <message>
         <location line="+9"/>
         <source>Load an image by clicking the button with the open icon and selecting an image file, or by drag &amp; dropping an image file on the PosteRazor. The drag &amp; drop also works during the other steps.
-After loading the image, the most important informations are listed in the &lt;b&gt;%1&lt;/b&gt; fields.</source>
-        <comment>Wizard step 1. Place holders: %1 = &apos;Image informations&apos; (will be automatically inserted)</comment>
+After loading the image, the most important information are listed in the &lt;b&gt;%1&lt;/b&gt; fields.</source>
+        <comment>Wizard step 1. Place holders: %1 = &apos;Image information&apos; (will be automatically inserted)</comment>
         <translation>Carregue a imagem clicando no botão com o ícone de abrir e seleccionando um ficheiro de imagem, ou arrastando um ficheiro de imagem para o PosteRazor. O arrastamento também funciona durante os outros passos.
 Após carregar a imagem, as informações mais importantes são listadas nos campos &lt;b&gt;%1&lt;/b&gt;.</translation>
     </message>
@@ -128,7 +128,7 @@
     <message>
         <location filename="wizard.cpp" line="+90"/>
         <location filename="wizardcontroller.cpp" line="+93"/>
-        <source>Image Informations</source>
+        <source>Image Information</source>
         <translation>Informações da Imagem</translation>
     </message>
     <message>
@@ -397,7 +397,7 @@
     </message>
     <message>
         <location/>
-        <source>Image Informations</source>
+        <source>Image Information</source>
         <translation type="unfinished">Informações da Imagem</translation>
     </message>
     <message>
diff -Naur fotowall-0.9.orig/3rdparty/posterazor/wizardcontroller.cpp fotowall-0.9/3rdparty/posterazor/wizardcontroller.cpp
--- fotowall-0.9.orig/3rdparty/posterazor/wizardcontroller.cpp	2009-08-31 00:46:21.000000000 +0200
+++ fotowall-0.9/3rdparty/posterazor/wizardcontroller.cpp	2012-02-24 20:07:13.887536536 +0100
@@ -195,9 +195,9 @@
     case WizardStepInputImage:
         result = QCoreApplication::translate("Help",
             "Load an image by clicking the button with the open icon and selecting an image file, or by drag & dropping an image file on the PosteRazor. The drag & drop also works during the other steps.\n"
-            "After loading the image, the most important informations are listed in the <b>%1</b> fields.",
-            "Wizard step 1. Place holders: %1 = 'Image informations' (will be automatically inserted)")
-            .arg(Types::cleanString(QCoreApplication::translate("Main window", "Image Informations")));
+            "After loading the image, the most important information are listed in the <b>%1</b> fields.",
+            "Wizard step 1. Place holders: %1 = 'Image information' (will be automatically inserted)")
+            .arg(Types::cleanString(QCoreApplication::translate("Main window", "Image Information")));
         break;
     case WizardStepPaperSize:
         result = QCoreApplication::translate("Help",
diff -Naur fotowall-0.9.orig/3rdparty/posterazor/wizard.cpp fotowall-0.9/3rdparty/posterazor/wizard.cpp
--- fotowall-0.9.orig/3rdparty/posterazor/wizard.cpp	2009-08-31 00:46:21.000000000 +0200
+++ fotowall-0.9/3rdparty/posterazor/wizard.cpp	2012-02-24 20:07:18.699534676 +0100
@@ -87,7 +87,7 @@
 
 void Wizard::retranslateUi()
 {
-    m_imageInfoGroup->setTitle(                     QCoreApplication::translate("Main window", "Image Informations"));
+    m_imageInfoGroup->setTitle(                     QCoreApplication::translate("Main window", "Image Information"));
     m_imageInformationColorTypeLabel->setText(      QCoreApplication::translate("Main window", "Color type:"));
     m_imageInformationResolutionLabel->setText(     QCoreApplication::translate("Main window", "Resolution:"));
     m_imageInformationSizeInPixelsLabel->setText(   QCoreApplication::translate("Main window", "Size (in pixels):"));
diff -Naur fotowall-0.9.orig/3rdparty/posterazor/wizard.ui fotowall-0.9/3rdparty/posterazor/wizard.ui
--- fotowall-0.9.orig/3rdparty/posterazor/wizard.ui	2009-08-31 00:46:21.000000000 +0200
+++ fotowall-0.9/3rdparty/posterazor/wizard.ui	2012-02-24 20:07:31.647532913 +0100
@@ -215,7 +215,7 @@
           <item>
            <widget class="QGroupBox" name="m_imageInfoGroup">
             <property name="title">
-             <string>Image Informations</string>
+             <string>Image Information</string>
             </property>
             <layout class="QGridLayout" name="_3">
              <property name="margin">
diff -Naur fotowall-0.9.orig/3rdparty/posterazor/zh_CN.ts fotowall-0.9/3rdparty/posterazor/zh_CN.ts
--- fotowall-0.9.orig/3rdparty/posterazor/zh_CN.ts	2009-12-01 14:09:37.000000000 +0100
+++ fotowall-0.9/3rdparty/posterazor/zh_CN.ts	2012-02-24 20:07:21.551531994 +0100
@@ -47,8 +47,8 @@
     <message>
         <location line="+9"/>
         <source>Load an image by clicking the button with the open icon and selecting an image file, or by drag &amp; dropping an image file on the PosteRazor. The drag &amp; drop also works during the other steps.
-After loading the image, the most important informations are listed in the &lt;b&gt;%1&lt;/b&gt; fields.</source>
-        <comment>Wizard step 1. Place holders: %1 = &apos;Image informations&apos; (will be automatically inserted)</comment>
+After loading the image, the most important information are listed in the &lt;b&gt;%1&lt;/b&gt; fields.</source>
+        <comment>Wizard step 1. Place holders: %1 = &apos;Image information&apos; (will be automatically inserted)</comment>
         <translation type="unfinished"></translation>
     </message>
     <message>
@@ -119,7 +119,7 @@
     <message>
         <location filename="wizard.cpp" line="+90"/>
         <location filename="wizardcontroller.cpp" line="+93"/>
-        <source>Image Informations</source>
+        <source>Image Information</source>
         <translation type="unfinished">绝对大小</translation>
     </message>
     <message>
@@ -388,7 +388,7 @@
     </message>
     <message>
         <location/>
-        <source>Image Informations</source>
+        <source>Image Information</source>
         <translation type="unfinished">绝对大小</translation>
     </message>
     <message>
