From: "Matteo F. Vescovi" <mfv.debian@gmail.com>
Date: Sat, 26 May 2012 16:15:53 +0200
Subject: Disable_FAST_MATH

---
 include/utilities/mathOptimizations.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/utilities/mathOptimizations.h b/include/utilities/mathOptimizations.h
index b228851..9dbffc5 100644
--- a/include/utilities/mathOptimizations.h
+++ b/include/utilities/mathOptimizations.h
@@ -53,7 +53,7 @@
 //# define M_SQRT2	1.41421356237309504880	/* sqrt(2) */
 //# define M_SQRT1_2	0.70710678118654752440	/* 1/sqrt(2) */
 
-#define FAST_MATH
+//#define FAST_MATH
 #define FAST_TRIG
 
 __BEGIN_YAFRAY
