From: Dmitrijs Ledkovs <xnox@ubuntu.com>
Date: Thu, 14 Mar 2013 10:41:45 +0100
Subject: Fix_FTBFS_against_python3.3

---
 src/bindings/CMakeLists.txt | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/bindings/CMakeLists.txt b/src/bindings/CMakeLists.txt
index a53d941..51607db 100644
--- a/src/bindings/CMakeLists.txt
+++ b/src/bindings/CMakeLists.txt
@@ -40,7 +40,7 @@ IF (WITH_YAF_PY_BINDINGS)
 	endif(YAF_PY_VERSION)
 	
 	MESSAGE("Searching for Python ${REQUIRED_PYTHON_VERSION}")
-	find_package(YafPythonLibs REQUIRED)
+	pkg_check_modules(PYTHON python-${REQUIRED_PYTHON_VERSION})
 	
 	if (NOT PYTHON_LIBRARIES)
 		message("Python wasn't found, please install it to build python bindings")
@@ -84,8 +84,8 @@ if (WITH_YAF_PY_BINDINGS)
 
 	endif(NOT WIN32)
 	
-	message("Python header path: ${PYTHON_INCLUDE_DIR}")
-	include_directories(${YAF_INCLUDE_DIRS} ${PYTHON_INCLUDE_DIR})
+	message("Python header path: ${PYTHON_INCLUDE_DIRS}")
+	include_directories(${YAF_INCLUDE_DIRS} ${PYTHON_INCLUDE_DIRS})
 	include(${SWIG_USE_FILE})
 
 	if(YAF_PY_VERSION GREATER 3)
