Description: fix clean target
Author: Dmitrijs Ledkovs <dmitrij.ledkov@ubuntu.com>

Index: b/Makefile
===================================================================
--- a/Makefile
+++ b/Makefile
@@ -14,5 +14,9 @@
 
 erase: erase.o
 
+install:
+	install -D -m 0755 flashbench $(DESTDIR)/usr/bin/flashbench
+	install -D -m 0755 erase $(DESTDIR)/usr/bin/flashbench-erase
+
 clean:
-	rm flashbench flashbench.o dev.o vm.o
+	rm -f flashbench flashbench.o dev.o vm.o
