--- antlr-2.7.7.orig/lib/csharp/antlr.astframe/AssemblyInfo.cs
+++ antlr-2.7.7/lib/csharp/antlr.astframe/AssemblyInfo.cs
@@ -31,12 +31,10 @@
 
 [assembly: CLSCompliantAttribute(true)]
 
-#if STRONGNAME
 #pragma warning disable 1699
 [assembly: AssemblyDelaySign(false)]
-[assembly: AssemblyKeyFile("org.antlr.snk")]
+[assembly: AssemblyKeyFile("/usr/share/cli-common/keys/mono.snk")]
 #pragma warning restore 1699
-#endif
 
 
 #if APTC
--- antlr-2.7.7.orig/lib/csharp/antlr.runtime/AssemblyInfo.cs
+++ antlr-2.7.7/lib/csharp/antlr.runtime/AssemblyInfo.cs
@@ -32,12 +32,10 @@
 
 [assembly: CLSCompliantAttribute(true)]
 
-#if STRONGNAME
 #pragma warning disable 1699
 [assembly: AssemblyDelaySign(false)]
-[assembly: AssemblyKeyFile("org.antlr.snk")]
+[assembly: AssemblyKeyFile("/usr/share/cli-common/keys/mono.snk")]
 #pragma warning restore 1699
-#endif
 
 #if APTC
 [assembly: System.Security.AllowPartiallyTrustedCallers]
