diff --git a/xfixes/xfixesint.h b/xfixes/xfixesint.h
index 334c71f..309954d 100644
--- a/xfixes/xfixesint.h
+++ b/xfixes/xfixesint.h
@@ -291,7 +291,7 @@  int
  SProcXFixesDestroyPointerBarrier(ClientPtr client);
 
 /* Xinerama */
-extern int (*PanoramiXSaveXFixesVector[XFixesNumberRequests]) (ClientPtr);
+_X_INTERNAL extern int (*PanoramiXSaveXFixesVector[XFixesNumberRequests]) (ClientPtr);
 void PanoramiXFixesInit(void);
 void PanoramiXFixesReset(void);
