--- a/kcontrol/menus/CMakeLists.txt
+++ b/kcontrol/menus/CMakeLists.txt
@@ -5,7 +5,7 @@
 if (WIN32)
 install( FILES kde-information.menu DESTINATION ${SHARE_INSTALL_DIR}/xdg/menus/)
 else (WIN32)
-install( FILES kde-information.menu DESTINATION ${SYSCONF_INSTALL_DIR}/xdg/menus/)
+install( FILES kde-information.menu DESTINATION ${SYSCONF_INSTALL_DIR}/xdg/menus/ RENAME kde4-information.menu )
 endif (WIN32)
 
-install( FILES kde-information.directory  DESTINATION  ${XDG_DIRECTORY_INSTALL_DIR} )
+install( FILES kde-information.directory  DESTINATION ${XDG_DIRECTORY_INSTALL_DIR} RENAME kde4-information.directory )
--- a/kcontrol/menus/kde-information.menu
+++ b/kcontrol/menus/kde-information.menu
@@ -3,7 +3,7 @@
 
 <Menu>
 	<Name>Information</Name>
-	<Directory>kde-information.directory</Directory>
+	<Directory>kde4-information.directory</Directory>
 	<Include>
 		<Category>X-KDE-information</Category>
 	</Include>
