--- a/khelpcenter/searchhandlers/khc_docbookdig.pl.cmake
+++ b/khelpcenter/searchhandlers/khc_docbookdig.pl.cmake
@@ -26,7 +26,7 @@
 
 use Getopt::Long;
 
-my $htdigdata = "/srv/www/htdig/common/";
+my $htdigdata = "/etc/htdig";
 my $htdigbin = "/usr/bin";
 my $kdeprefix = "@CMAKE_INSTALL_PREFIX@";
 chomp $kdeprefix;
--- a/khelpcenter/searchhandlers/khc_htdig.pl.cmake
+++ b/khelpcenter/searchhandlers/khc_htdig.pl.cmake
@@ -26,7 +26,7 @@
 
 use Getopt::Long;
 
-my $htdigdata = "/srv/www/htdig/common/";
+my $htdigdata = "/etc/htdig";
 my $htdigbin = "/usr/bin";
 my $kdeprefix = "@CMAKE_INSTALL_PREFIX@";
 chomp $kdeprefix;
--- a/khelpcenter/searchhandlers/khc_htsearch.pl
+++ b/khelpcenter/searchhandlers/khc_htsearch.pl
@@ -8,7 +8,7 @@
 use open IO => ':utf8';
 use open ':std';
 
-my $htsearchpath="/srv/www/cgi-bin/htsearch";
+my $htsearchpath="/usr/lib/cgi-bin/htsearch";
 my $config;
 my $format;
 my $method;
