--- a/sysdeps/posix/getaddrinfo.c
+++ b/sysdeps/posix/getaddrinfo.c
@@ -2267,8 +2267,20 @@
 		}
 
 	      socklen_t sl = sizeof (results[i].source_addr);
+#ifdef __FreeBSD_kernel__
+	      struct sockaddr_in6 sa_in6;
+	      const struct sockaddr *sa = &sa_in6; 
+	      memcpy(&sa_in6, q->ai_addr, q->ai_family == AF_INET6 ?
+		     sizeof(struct sockaddr_in6) : sizeof(struct sockaddr_in));
+	      /* The FreeBSD kernel doesn't allow connections on port 0. Use
+	         port 1 instead, as on the FreeBSD libc. */
+	      if (sa_in6.sin6_port == htons(0))
+		sa_in6.sin6_port = htons(1);
+#else
+	      const struct sockaddr *sa = q->ai_addr;
+#endif
 	      if (fd != -1
-		  && __connect (fd, q->ai_addr, q->ai_addrlen) == 0
+		  && __connect (fd, sa, q->ai_addrlen) == 0
 		  && __getsockname (fd,
 				    (struct sockaddr *) &results[i].source_addr,
 				    &sl) == 0)
