---
 malloc/memusage.c |    3 +++
 1 file changed, 3 insertions(+)

--- a/malloc/memusage.c
+++ b/malloc/memusage.c
@@ -674,6 +674,8 @@
   return result;
 }
 
+#include <_G_config.h>
+#ifdef _G_HAVE_MREMAP
 
 /* `mmap' replacement.  We do not have to keep track of the sizesince
    `munmap' will get it as a parameter.  */
@@ -739,6 +741,7 @@
   /* Return the pointer to the user buffer.  */
   return result;
 }
+#endif
 
 
 /* `munmap' replacement.  */
