---
 linuxthreads/old_pthread_atfork.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/linuxthreads/old_pthread_atfork.c
+++ b/linuxthreads/old_pthread_atfork.c
@@ -19,7 +19,7 @@
 
 #include <shlib-compat.h>
 
-#if SHLIB_COMPAT (libpthread, GLIBC_2_0, GLIBC_2_3)
+#if SHLIB_COMPAT (libpthread, GLIBC_2_0, GLIBC_2_3_2)
 # define __pthread_atfork __dyn_pthread_atfork
 # include "pthread_atfork.c"
 # undef __pthread_atfork
