Index: glib_2.33.14-SprezzOS1/configure.ac
===================================================================
--- glib_2.33.14-SprezzOS1.orig/configure.ac	2012-09-17 17:50:05.000000000 -0400
+++ glib_2.33.14-SprezzOS1/configure.ac	2012-10-03 21:45:57.000000000 -0400
@@ -534,6 +534,8 @@
   esac
 fi
 
+CPPFLAGS="$CPPFLAGS -DG_DISABLE_SINGLE_INCLUDES"
+
 AS_IF([test "$glib_native_win32" = "yes"], [
   if test x$enable_static = xyes -a x$enable_shared = xyes; then
     AC_MSG_ERROR([Can not build both shared and static at the same time on Windows.])
Index: glib_2.33.14-SprezzOS1/glib/deprecated/gallocator.h
===================================================================
--- glib_2.33.14-SprezzOS1.orig/glib/deprecated/gallocator.h	2012-07-03 15:43:59.000000000 -0400
+++ glib_2.33.14-SprezzOS1/glib/deprecated/gallocator.h	2012-10-03 21:45:57.000000000 -0400
@@ -15,7 +15,7 @@
  * Boston, MA 02111-1307, USA.
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib_2.33.14-SprezzOS1/glib/deprecated/gcache.h
===================================================================
--- glib_2.33.14-SprezzOS1.orig/glib/deprecated/gcache.h	2012-07-03 15:43:59.000000000 -0400
+++ glib_2.33.14-SprezzOS1/glib/deprecated/gcache.h	2012-10-03 21:45:57.000000000 -0400
@@ -24,7 +24,7 @@
  * GLib at ftp://ftp.gtk.org/pub/gtk/.
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib_2.33.14-SprezzOS1/glib/deprecated/gcompletion.h
===================================================================
--- glib_2.33.14-SprezzOS1.orig/glib/deprecated/gcompletion.h	2012-07-17 09:41:25.000000000 -0400
+++ glib_2.33.14-SprezzOS1/glib/deprecated/gcompletion.h	2012-10-03 21:45:57.000000000 -0400
@@ -24,7 +24,7 @@
  * GLib at ftp://ftp.gtk.org/pub/gtk/.
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib_2.33.14-SprezzOS1/glib/deprecated/grel.h
===================================================================
--- glib_2.33.14-SprezzOS1.orig/glib/deprecated/grel.h	2012-07-17 09:41:25.000000000 -0400
+++ glib_2.33.14-SprezzOS1/glib/deprecated/grel.h	2012-10-03 21:45:57.000000000 -0400
@@ -24,7 +24,7 @@
  * GLib at ftp://ftp.gtk.org/pub/gtk/.
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib_2.33.14-SprezzOS1/glib/deprecated/gthread.h
===================================================================
--- glib_2.33.14-SprezzOS1.orig/glib/deprecated/gthread.h	2012-07-17 09:41:25.000000000 -0400
+++ glib_2.33.14-SprezzOS1/glib/deprecated/gthread.h	2012-10-03 21:45:57.000000000 -0400
@@ -24,7 +24,7 @@
  * GLib at ftp://ftp.gtk.org/pub/gtk/.
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib_2.33.14-SprezzOS1/glib/galloca.h
===================================================================
--- glib_2.33.14-SprezzOS1.orig/glib/galloca.h	2012-07-03 15:43:59.000000000 -0400
+++ glib_2.33.14-SprezzOS1/glib/galloca.h	2012-10-03 21:45:57.000000000 -0400
@@ -24,7 +24,7 @@
  * GLib at ftp://ftp.gtk.org/pub/gtk/.
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib_2.33.14-SprezzOS1/glib/garray.h
===================================================================
--- glib_2.33.14-SprezzOS1.orig/glib/garray.h	2012-08-18 14:27:14.000000000 -0400
+++ glib_2.33.14-SprezzOS1/glib/garray.h	2012-10-03 21:45:57.000000000 -0400
@@ -24,7 +24,7 @@
  * GLib at ftp://ftp.gtk.org/pub/gtk/.
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib_2.33.14-SprezzOS1/glib/gasyncqueue.h
===================================================================
--- glib_2.33.14-SprezzOS1.orig/glib/gasyncqueue.h	2012-07-17 09:41:25.000000000 -0400
+++ glib_2.33.14-SprezzOS1/glib/gasyncqueue.h	2012-10-03 21:45:57.000000000 -0400
@@ -24,7 +24,7 @@
  * GLib at ftp://ftp.gtk.org/pub/gtk/.
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib_2.33.14-SprezzOS1/glib/gatomic.h
===================================================================
--- glib_2.33.14-SprezzOS1.orig/glib/gatomic.h	2012-07-29 09:17:09.000000000 -0400
+++ glib_2.33.14-SprezzOS1/glib/gatomic.h	2012-10-03 21:45:57.000000000 -0400
@@ -19,7 +19,7 @@
  * Author: Ryan Lortie <desrt@desrt.ca>
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib_2.33.14-SprezzOS1/glib/gbacktrace.h
===================================================================
--- glib_2.33.14-SprezzOS1.orig/glib/gbacktrace.h	2012-07-03 15:43:59.000000000 -0400
+++ glib_2.33.14-SprezzOS1/glib/gbacktrace.h	2012-10-03 21:45:57.000000000 -0400
@@ -24,7 +24,7 @@
  * GLib at ftp://ftp.gtk.org/pub/gtk/.
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib_2.33.14-SprezzOS1/glib/gbase64.h
===================================================================
--- glib_2.33.14-SprezzOS1.orig/glib/gbase64.h	2012-07-03 15:43:59.000000000 -0400
+++ glib_2.33.14-SprezzOS1/glib/gbase64.h	2012-10-03 21:45:57.000000000 -0400
@@ -18,7 +18,7 @@
  * Boston, MA 02111-1307, USA.
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib_2.33.14-SprezzOS1/glib/gbitlock.h
===================================================================
--- glib_2.33.14-SprezzOS1.orig/glib/gbitlock.h	2012-07-03 15:43:59.000000000 -0400
+++ glib_2.33.14-SprezzOS1/glib/gbitlock.h	2012-10-03 21:45:57.000000000 -0400
@@ -25,7 +25,7 @@
 
 #include <glib/gtypes.h>
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib_2.33.14-SprezzOS1/glib/gbookmarkfile.h
===================================================================
--- glib_2.33.14-SprezzOS1.orig/glib/gbookmarkfile.h	2012-07-03 15:43:59.000000000 -0400
+++ glib_2.33.14-SprezzOS1/glib/gbookmarkfile.h	2012-10-03 21:45:57.000000000 -0400
@@ -17,7 +17,7 @@
  * Free Software Foundation, Inc., 59 Temple Place - Suite 330,
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib_2.33.14-SprezzOS1/glib/gchecksum.h
===================================================================
--- glib_2.33.14-SprezzOS1.orig/glib/gchecksum.h	2012-08-18 14:27:14.000000000 -0400
+++ glib_2.33.14-SprezzOS1/glib/gchecksum.h	2012-10-03 21:45:57.000000000 -0400
@@ -18,7 +18,7 @@
  * Boston, MA 02111-1307, USA.
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib_2.33.14-SprezzOS1/glib/gconvert.h
===================================================================
--- glib_2.33.14-SprezzOS1.orig/glib/gconvert.h	2012-07-03 15:43:59.000000000 -0400
+++ glib_2.33.14-SprezzOS1/glib/gconvert.h	2012-10-03 21:45:57.000000000 -0400
@@ -24,7 +24,7 @@
  * GLib at ftp://ftp.gtk.org/pub/gtk/.
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib_2.33.14-SprezzOS1/glib/gdataset.h
===================================================================
--- glib_2.33.14-SprezzOS1.orig/glib/gdataset.h	2012-09-04 20:49:08.000000000 -0400
+++ glib_2.33.14-SprezzOS1/glib/gdataset.h	2012-10-03 21:45:57.000000000 -0400
@@ -24,7 +24,7 @@
  * GLib at ftp://ftp.gtk.org/pub/gtk/.
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib_2.33.14-SprezzOS1/glib/gdate.h
===================================================================
--- glib_2.33.14-SprezzOS1.orig/glib/gdate.h	2012-07-03 15:43:59.000000000 -0400
+++ glib_2.33.14-SprezzOS1/glib/gdate.h	2012-10-03 21:45:57.000000000 -0400
@@ -24,7 +24,7 @@
  * GLib at ftp://ftp.gtk.org/pub/gtk/.
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib_2.33.14-SprezzOS1/glib/gdatetime.h
===================================================================
--- glib_2.33.14-SprezzOS1.orig/glib/gdatetime.h	2012-07-03 15:43:59.000000000 -0400
+++ glib_2.33.14-SprezzOS1/glib/gdatetime.h	2012-10-03 21:45:57.000000000 -0400
@@ -23,7 +23,7 @@
  *          Ryan Lortie <desrt@desrt.ca>
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib_2.33.14-SprezzOS1/glib/gdir.h
===================================================================
--- glib_2.33.14-SprezzOS1.orig/glib/gdir.h	2012-07-03 15:43:59.000000000 -0400
+++ glib_2.33.14-SprezzOS1/glib/gdir.h	2012-10-03 21:45:57.000000000 -0400
@@ -21,7 +21,7 @@
  * Boston, MA 02111-1307, USA.
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib_2.33.14-SprezzOS1/glib/gerror.h
===================================================================
--- glib_2.33.14-SprezzOS1.orig/glib/gerror.h	2012-07-03 15:43:59.000000000 -0400
+++ glib_2.33.14-SprezzOS1/glib/gerror.h	2012-10-03 21:45:57.000000000 -0400
@@ -18,7 +18,7 @@
  *   Boston, MA 02111-1307, USA.
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib_2.33.14-SprezzOS1/glib/gfileutils.h
===================================================================
--- glib_2.33.14-SprezzOS1.orig/glib/gfileutils.h	2012-07-29 09:17:09.000000000 -0400
+++ glib_2.33.14-SprezzOS1/glib/gfileutils.h	2012-10-03 21:45:57.000000000 -0400
@@ -18,7 +18,7 @@
  *   Boston, MA 02111-1307, USA.
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib_2.33.14-SprezzOS1/glib/ghash.h
===================================================================
--- glib_2.33.14-SprezzOS1.orig/glib/ghash.h	2012-07-29 09:17:09.000000000 -0400
+++ glib_2.33.14-SprezzOS1/glib/ghash.h	2012-10-03 21:45:57.000000000 -0400
@@ -24,7 +24,7 @@
  * GLib at ftp://ftp.gtk.org/pub/gtk/.
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib_2.33.14-SprezzOS1/glib/ghmac.h
===================================================================
--- glib_2.33.14-SprezzOS1.orig/glib/ghmac.h	2012-07-29 09:17:09.000000000 -0400
+++ glib_2.33.14-SprezzOS1/glib/ghmac.h	2012-10-03 21:45:57.000000000 -0400
@@ -18,7 +18,7 @@
  * Boston, MA 02111-1307, USA.
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib_2.33.14-SprezzOS1/glib/ghook.h
===================================================================
--- glib_2.33.14-SprezzOS1.orig/glib/ghook.h	2012-07-03 15:43:59.000000000 -0400
+++ glib_2.33.14-SprezzOS1/glib/ghook.h	2012-10-03 21:45:57.000000000 -0400
@@ -24,7 +24,7 @@
  * GLib at ftp://ftp.gtk.org/pub/gtk/.
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib_2.33.14-SprezzOS1/glib/giochannel.h
===================================================================
--- glib_2.33.14-SprezzOS1.orig/glib/giochannel.h	2012-07-03 15:43:59.000000000 -0400
+++ glib_2.33.14-SprezzOS1/glib/giochannel.h	2012-10-03 21:45:57.000000000 -0400
@@ -24,7 +24,7 @@
  * GLib at ftp://ftp.gtk.org/pub/gtk/.
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib_2.33.14-SprezzOS1/glib/gkeyfile.h
===================================================================
--- glib_2.33.14-SprezzOS1.orig/glib/gkeyfile.h	2012-10-03 21:45:57.000000000 -0400
+++ glib_2.33.14-SprezzOS1/glib/gkeyfile.h	2012-10-03 21:45:57.000000000 -0400
@@ -20,7 +20,7 @@
  *   Boston, MA 02111-1307, USA.
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib_2.33.14-SprezzOS1/glib/glist.h
===================================================================
--- glib_2.33.14-SprezzOS1.orig/glib/glist.h	2012-07-29 09:17:09.000000000 -0400
+++ glib_2.33.14-SprezzOS1/glib/glist.h	2012-10-03 21:45:57.000000000 -0400
@@ -24,7 +24,7 @@
  * GLib at ftp://ftp.gtk.org/pub/gtk/.
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib_2.33.14-SprezzOS1/glib/gmacros.h
===================================================================
--- glib_2.33.14-SprezzOS1.orig/glib/gmacros.h	2012-07-17 09:41:25.000000000 -0400
+++ glib_2.33.14-SprezzOS1/glib/gmacros.h	2012-10-03 21:45:57.000000000 -0400
@@ -28,7 +28,7 @@
  * not refer to variables from glibconfig.h
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib_2.33.14-SprezzOS1/glib/gmain.h
===================================================================
--- glib_2.33.14-SprezzOS1.orig/glib/gmain.h	2012-07-29 09:17:09.000000000 -0400
+++ glib_2.33.14-SprezzOS1/glib/gmain.h	2012-10-03 21:45:57.000000000 -0400
@@ -17,7 +17,7 @@
  * Boston, MA 02111-1307, USA.
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib_2.33.14-SprezzOS1/glib/gmappedfile.h
===================================================================
--- glib_2.33.14-SprezzOS1.orig/glib/gmappedfile.h	2012-08-20 16:32:54.000000000 -0400
+++ glib_2.33.14-SprezzOS1/glib/gmappedfile.h	2012-10-03 21:45:57.000000000 -0400
@@ -19,7 +19,7 @@
  * Boston, MA 02111-1307, USA.
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib_2.33.14-SprezzOS1/glib/gmarkup.h
===================================================================
--- glib_2.33.14-SprezzOS1.orig/glib/gmarkup.h	2012-08-20 18:34:13.000000000 -0400
+++ glib_2.33.14-SprezzOS1/glib/gmarkup.h	2012-10-03 21:45:57.000000000 -0400
@@ -18,7 +18,7 @@
  *   Boston, MA 02111-1307, USA.
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib_2.33.14-SprezzOS1/glib/gmem.h
===================================================================
--- glib_2.33.14-SprezzOS1.orig/glib/gmem.h	2012-08-27 16:10:11.000000000 -0400
+++ glib_2.33.14-SprezzOS1/glib/gmem.h	2012-10-03 21:45:57.000000000 -0400
@@ -24,7 +24,7 @@
  * GLib at ftp://ftp.gtk.org/pub/gtk/.
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib_2.33.14-SprezzOS1/glib/gmessages.h
===================================================================
--- glib_2.33.14-SprezzOS1.orig/glib/gmessages.h	2012-07-03 15:43:59.000000000 -0400
+++ glib_2.33.14-SprezzOS1/glib/gmessages.h	2012-10-03 21:45:57.000000000 -0400
@@ -24,7 +24,7 @@
  * GLib at ftp://ftp.gtk.org/pub/gtk/.
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib_2.33.14-SprezzOS1/glib/gnode.h
===================================================================
--- glib_2.33.14-SprezzOS1.orig/glib/gnode.h	2012-07-17 09:41:25.000000000 -0400
+++ glib_2.33.14-SprezzOS1/glib/gnode.h	2012-10-03 21:45:57.000000000 -0400
@@ -24,7 +24,7 @@
  * GLib at ftp://ftp.gtk.org/pub/gtk/.
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib_2.33.14-SprezzOS1/glib/goption.h
===================================================================
--- glib_2.33.14-SprezzOS1.orig/glib/goption.h	2012-07-03 15:43:59.000000000 -0400
+++ glib_2.33.14-SprezzOS1/glib/goption.h	2012-10-03 21:45:57.000000000 -0400
@@ -18,7 +18,7 @@
  * Boston, MA 02111-1307, USA.
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib_2.33.14-SprezzOS1/glib/gpattern.h
===================================================================
--- glib_2.33.14-SprezzOS1.orig/glib/gpattern.h	2012-07-03 15:43:59.000000000 -0400
+++ glib_2.33.14-SprezzOS1/glib/gpattern.h	2012-10-03 21:45:57.000000000 -0400
@@ -17,7 +17,7 @@
  * Boston, MA 02111-1307, USA.
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib_2.33.14-SprezzOS1/glib/gprimes.h
===================================================================
--- glib_2.33.14-SprezzOS1.orig/glib/gprimes.h	2012-07-03 15:43:59.000000000 -0400
+++ glib_2.33.14-SprezzOS1/glib/gprimes.h	2012-10-03 21:45:57.000000000 -0400
@@ -24,7 +24,7 @@
  * GLib at ftp://ftp.gtk.org/pub/gtk/.
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib_2.33.14-SprezzOS1/glib/gqsort.h
===================================================================
--- glib_2.33.14-SprezzOS1.orig/glib/gqsort.h	2012-07-03 15:43:59.000000000 -0400
+++ glib_2.33.14-SprezzOS1/glib/gqsort.h	2012-10-03 21:45:57.000000000 -0400
@@ -24,7 +24,7 @@
  * GLib at ftp://ftp.gtk.org/pub/gtk/.
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib_2.33.14-SprezzOS1/glib/gquark.h
===================================================================
--- glib_2.33.14-SprezzOS1.orig/glib/gquark.h	2012-09-08 14:08:01.000000000 -0400
+++ glib_2.33.14-SprezzOS1/glib/gquark.h	2012-10-03 21:45:57.000000000 -0400
@@ -24,7 +24,7 @@
  * GLib at ftp://ftp.gtk.org/pub/gtk/.
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib_2.33.14-SprezzOS1/glib/gqueue.h
===================================================================
--- glib_2.33.14-SprezzOS1.orig/glib/gqueue.h	2012-07-03 15:43:59.000000000 -0400
+++ glib_2.33.14-SprezzOS1/glib/gqueue.h	2012-10-03 21:45:57.000000000 -0400
@@ -24,7 +24,7 @@
  * GLib at ftp://ftp.gtk.org/pub/gtk/.
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib_2.33.14-SprezzOS1/glib/grand.h
===================================================================
--- glib_2.33.14-SprezzOS1.orig/glib/grand.h	2012-07-03 15:43:59.000000000 -0400
+++ glib_2.33.14-SprezzOS1/glib/grand.h	2012-10-03 21:45:57.000000000 -0400
@@ -24,7 +24,7 @@
  * GLib at ftp://ftp.gtk.org/pub/gtk/.
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib_2.33.14-SprezzOS1/glib/gregex.h
===================================================================
--- glib_2.33.14-SprezzOS1.orig/glib/gregex.h	2012-07-29 09:17:09.000000000 -0400
+++ glib_2.33.14-SprezzOS1/glib/gregex.h	2012-10-03 21:45:57.000000000 -0400
@@ -19,7 +19,7 @@
  * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib_2.33.14-SprezzOS1/glib/gscanner.h
===================================================================
--- glib_2.33.14-SprezzOS1.orig/glib/gscanner.h	2012-07-03 15:43:59.000000000 -0400
+++ glib_2.33.14-SprezzOS1/glib/gscanner.h	2012-10-03 21:45:57.000000000 -0400
@@ -24,7 +24,7 @@
  * GLib at ftp://ftp.gtk.org/pub/gtk/.
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib_2.33.14-SprezzOS1/glib/gsequence.h
===================================================================
--- glib_2.33.14-SprezzOS1.orig/glib/gsequence.h	2012-07-03 15:44:00.000000000 -0400
+++ glib_2.33.14-SprezzOS1/glib/gsequence.h	2012-10-03 21:45:57.000000000 -0400
@@ -18,7 +18,7 @@
  * Boston, MA 02111-1307, USA.
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib_2.33.14-SprezzOS1/glib/gshell.h
===================================================================
--- glib_2.33.14-SprezzOS1.orig/glib/gshell.h	2012-07-03 15:44:00.000000000 -0400
+++ glib_2.33.14-SprezzOS1/glib/gshell.h	2012-10-03 21:45:57.000000000 -0400
@@ -18,7 +18,7 @@
  * Boston, MA 02111-1307, USA.
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib_2.33.14-SprezzOS1/glib/gslice.h
===================================================================
--- glib_2.33.14-SprezzOS1.orig/glib/gslice.h	2012-08-18 14:27:14.000000000 -0400
+++ glib_2.33.14-SprezzOS1/glib/gslice.h	2012-10-03 21:45:57.000000000 -0400
@@ -17,7 +17,7 @@
  * Boston, MA 02111-1307, USA.
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib_2.33.14-SprezzOS1/glib/gslist.h
===================================================================
--- glib_2.33.14-SprezzOS1.orig/glib/gslist.h	2012-07-29 09:17:09.000000000 -0400
+++ glib_2.33.14-SprezzOS1/glib/gslist.h	2012-10-03 21:45:57.000000000 -0400
@@ -24,7 +24,7 @@
  * GLib at ftp://ftp.gtk.org/pub/gtk/.
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib_2.33.14-SprezzOS1/glib/gspawn.h
===================================================================
--- glib_2.33.14-SprezzOS1.orig/glib/gspawn.h	2012-08-20 16:32:54.000000000 -0400
+++ glib_2.33.14-SprezzOS1/glib/gspawn.h	2012-10-03 21:45:57.000000000 -0400
@@ -18,7 +18,7 @@
  * Boston, MA 02111-1307, USA.
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib_2.33.14-SprezzOS1/glib/gstrfuncs.h
===================================================================
--- glib_2.33.14-SprezzOS1.orig/glib/gstrfuncs.h	2012-07-03 15:44:00.000000000 -0400
+++ glib_2.33.14-SprezzOS1/glib/gstrfuncs.h	2012-10-03 21:45:57.000000000 -0400
@@ -24,7 +24,7 @@
  * GLib at ftp://ftp.gtk.org/pub/gtk/.
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib_2.33.14-SprezzOS1/glib/gstringchunk.h
===================================================================
--- glib_2.33.14-SprezzOS1.orig/glib/gstringchunk.h	2012-07-03 15:44:00.000000000 -0400
+++ glib_2.33.14-SprezzOS1/glib/gstringchunk.h	2012-10-03 21:45:57.000000000 -0400
@@ -24,7 +24,7 @@
  * GLib at ftp://ftp.gtk.org/pub/gtk/.
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib_2.33.14-SprezzOS1/glib/gstring.h
===================================================================
--- glib_2.33.14-SprezzOS1.orig/glib/gstring.h	2012-07-29 09:17:09.000000000 -0400
+++ glib_2.33.14-SprezzOS1/glib/gstring.h	2012-10-03 21:45:57.000000000 -0400
@@ -24,7 +24,7 @@
  * GLib at ftp://ftp.gtk.org/pub/gtk/.
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib_2.33.14-SprezzOS1/glib/gtestutils.h
===================================================================
--- glib_2.33.14-SprezzOS1.orig/glib/gtestutils.h	2012-08-27 16:12:27.000000000 -0400
+++ glib_2.33.14-SprezzOS1/glib/gtestutils.h	2012-10-03 21:45:57.000000000 -0400
@@ -18,7 +18,7 @@
  * Boston, MA 02111-1307, USA.
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib_2.33.14-SprezzOS1/glib/gthread.h
===================================================================
--- glib_2.33.14-SprezzOS1.orig/glib/gthread.h	2012-07-03 15:44:00.000000000 -0400
+++ glib_2.33.14-SprezzOS1/glib/gthread.h	2012-10-03 21:45:57.000000000 -0400
@@ -24,7 +24,7 @@
  * GLib at ftp://ftp.gtk.org/pub/gtk/.
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib_2.33.14-SprezzOS1/glib/gthreadpool.h
===================================================================
--- glib_2.33.14-SprezzOS1.orig/glib/gthreadpool.h	2012-07-03 15:44:00.000000000 -0400
+++ glib_2.33.14-SprezzOS1/glib/gthreadpool.h	2012-10-03 21:45:57.000000000 -0400
@@ -24,7 +24,7 @@
  * GLib at ftp://ftp.gtk.org/pub/gtk/.
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib_2.33.14-SprezzOS1/glib/gtimer.h
===================================================================
--- glib_2.33.14-SprezzOS1.orig/glib/gtimer.h	2012-07-03 15:44:00.000000000 -0400
+++ glib_2.33.14-SprezzOS1/glib/gtimer.h	2012-10-03 21:45:57.000000000 -0400
@@ -24,7 +24,7 @@
  * GLib at ftp://ftp.gtk.org/pub/gtk/.
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib_2.33.14-SprezzOS1/glib/gtimezone.h
===================================================================
--- glib_2.33.14-SprezzOS1.orig/glib/gtimezone.h	2012-07-03 15:44:00.000000000 -0400
+++ glib_2.33.14-SprezzOS1/glib/gtimezone.h	2012-10-03 21:45:57.000000000 -0400
@@ -19,7 +19,7 @@
  * Author: Ryan Lortie <desrt@desrt.ca>
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib_2.33.14-SprezzOS1/glib/gtree.h
===================================================================
--- glib_2.33.14-SprezzOS1.orig/glib/gtree.h	2012-07-03 15:44:00.000000000 -0400
+++ glib_2.33.14-SprezzOS1/glib/gtree.h	2012-10-03 21:45:57.000000000 -0400
@@ -24,7 +24,7 @@
  * GLib at ftp://ftp.gtk.org/pub/gtk/.
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib_2.33.14-SprezzOS1/glib/gtypes.h
===================================================================
--- glib_2.33.14-SprezzOS1.orig/glib/gtypes.h	2012-07-17 09:41:25.000000000 -0400
+++ glib_2.33.14-SprezzOS1/glib/gtypes.h	2012-10-03 21:45:57.000000000 -0400
@@ -24,7 +24,7 @@
  * GLib at ftp://ftp.gtk.org/pub/gtk/.
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib_2.33.14-SprezzOS1/glib/gunicode.h
===================================================================
--- glib_2.33.14-SprezzOS1.orig/glib/gunicode.h	2012-07-29 09:17:09.000000000 -0400
+++ glib_2.33.14-SprezzOS1/glib/gunicode.h	2012-10-03 21:45:57.000000000 -0400
@@ -19,7 +19,7 @@
  *   Boston, MA 02111-1307, USA.
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib_2.33.14-SprezzOS1/glib/gurifuncs.h
===================================================================
--- glib_2.33.14-SprezzOS1.orig/glib/gurifuncs.h	2012-07-17 09:41:25.000000000 -0400
+++ glib_2.33.14-SprezzOS1/glib/gurifuncs.h	2012-10-03 21:45:57.000000000 -0400
@@ -20,7 +20,7 @@
  * Author: Alexander Larsson <alexl@redhat.com>
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib_2.33.14-SprezzOS1/glib/gutils.h
===================================================================
--- glib_2.33.14-SprezzOS1.orig/glib/gutils.h	2012-07-29 09:17:09.000000000 -0400
+++ glib_2.33.14-SprezzOS1/glib/gutils.h	2012-10-03 21:45:57.000000000 -0400
@@ -24,7 +24,7 @@
  * GLib at ftp://ftp.gtk.org/pub/gtk/.
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib_2.33.14-SprezzOS1/glib/gvariant.h
===================================================================
--- glib_2.33.14-SprezzOS1.orig/glib/gvariant.h	2012-08-20 17:27:46.000000000 -0400
+++ glib_2.33.14-SprezzOS1/glib/gvariant.h	2012-10-03 21:45:57.000000000 -0400
@@ -20,7 +20,7 @@
  * Author: Ryan Lortie <desrt@desrt.ca>
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib_2.33.14-SprezzOS1/glib/gvarianttype.h
===================================================================
--- glib_2.33.14-SprezzOS1.orig/glib/gvarianttype.h	2012-07-03 15:44:00.000000000 -0400
+++ glib_2.33.14-SprezzOS1/glib/gvarianttype.h	2012-10-03 21:45:57.000000000 -0400
@@ -20,7 +20,7 @@
  * Author: Ryan Lortie <desrt@desrt.ca>
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib_2.33.14-SprezzOS1/glib/gwin32.h
===================================================================
--- glib_2.33.14-SprezzOS1.orig/glib/gwin32.h	2012-07-03 15:44:00.000000000 -0400
+++ glib_2.33.14-SprezzOS1/glib/gwin32.h	2012-10-03 21:45:57.000000000 -0400
@@ -24,7 +24,7 @@
  * GLib at ftp://ftp.gtk.org/pub/gtk/.
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib_2.33.14-SprezzOS1/glib/gversionmacros.h
===================================================================
--- glib_2.33.14-SprezzOS1.orig/glib/gversionmacros.h	2012-07-29 09:17:09.000000000 -0400
+++ glib_2.33.14-SprezzOS1/glib/gversionmacros.h	2012-10-03 21:45:57.000000000 -0400
@@ -24,7 +24,7 @@
  * GLib at ftp://ftp.gtk.org/pub/gtk/.
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
