Index: glib-2.32.2/configure.ac
===================================================================
--- glib-2.32.2.orig/configure.ac	2012-05-01 02:06:39.000000000 +0200
+++ glib-2.32.2/configure.ac	2012-05-01 17:02:03.067885571 +0200
@@ -531,6 +531,8 @@
   esac
 fi
 
+CPPFLAGS="$CPPFLAGS -DG_DISABLE_SINGLE_INCLUDES"
+
 if test "$glib_native_win32" = "yes"; then
   if test x$enable_static = xyes -a x$enable_shared = xyes; then
     AC_MSG_ERROR([Can not build both shared and static at the same time on Windows.])
Index: glib-2.32.2/glib/deprecated/gallocator.h
===================================================================
--- glib-2.32.2.orig/glib/deprecated/gallocator.h	2012-04-13 01:39:20.000000000 +0200
+++ glib-2.32.2/glib/deprecated/gallocator.h	2012-05-01 17:02:03.071885572 +0200
@@ -15,7 +15,7 @@
  * Boston, MA 02111-1307, USA.
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib-2.32.2/glib/deprecated/gcache.h
===================================================================
--- glib-2.32.2.orig/glib/deprecated/gcache.h	2012-04-13 01:39:20.000000000 +0200
+++ glib-2.32.2/glib/deprecated/gcache.h	2012-05-01 17:02:03.075885574 +0200
@@ -24,7 +24,7 @@
  * GLib at ftp://ftp.gtk.org/pub/gtk/.
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib-2.32.2/glib/deprecated/gcompletion.h
===================================================================
--- glib-2.32.2.orig/glib/deprecated/gcompletion.h	2012-04-13 01:39:20.000000000 +0200
+++ glib-2.32.2/glib/deprecated/gcompletion.h	2012-05-01 17:02:03.075885574 +0200
@@ -24,7 +24,7 @@
  * GLib at ftp://ftp.gtk.org/pub/gtk/.
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib-2.32.2/glib/deprecated/grel.h
===================================================================
--- glib-2.32.2.orig/glib/deprecated/grel.h	2012-04-13 01:39:20.000000000 +0200
+++ glib-2.32.2/glib/deprecated/grel.h	2012-05-01 17:02:03.075885574 +0200
@@ -24,7 +24,7 @@
  * GLib at ftp://ftp.gtk.org/pub/gtk/.
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib-2.32.2/glib/deprecated/gthread.h
===================================================================
--- glib-2.32.2.orig/glib/deprecated/gthread.h	2012-04-13 01:39:20.000000000 +0200
+++ glib-2.32.2/glib/deprecated/gthread.h	2012-05-01 17:02:03.075885574 +0200
@@ -24,7 +24,7 @@
  * GLib at ftp://ftp.gtk.org/pub/gtk/.
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib-2.32.2/glib/galloca.h
===================================================================
--- glib-2.32.2.orig/glib/galloca.h	2012-04-13 01:39:20.000000000 +0200
+++ glib-2.32.2/glib/galloca.h	2012-05-01 17:02:03.075885574 +0200
@@ -24,7 +24,7 @@
  * GLib at ftp://ftp.gtk.org/pub/gtk/.
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib-2.32.2/glib/garray.h
===================================================================
--- glib-2.32.2.orig/glib/garray.h	2012-04-13 01:39:20.000000000 +0200
+++ glib-2.32.2/glib/garray.h	2012-05-01 17:02:03.075885574 +0200
@@ -24,7 +24,7 @@
  * GLib at ftp://ftp.gtk.org/pub/gtk/.
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib-2.32.2/glib/gasyncqueue.h
===================================================================
--- glib-2.32.2.orig/glib/gasyncqueue.h	2012-04-13 01:39:20.000000000 +0200
+++ glib-2.32.2/glib/gasyncqueue.h	2012-05-01 17:02:03.083885575 +0200
@@ -24,7 +24,7 @@
  * GLib at ftp://ftp.gtk.org/pub/gtk/.
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib-2.32.2/glib/gatomic.h
===================================================================
--- glib-2.32.2.orig/glib/gatomic.h	2012-04-13 01:39:20.000000000 +0200
+++ glib-2.32.2/glib/gatomic.h	2012-05-01 17:02:03.083885575 +0200
@@ -19,7 +19,7 @@
  * Author: Ryan Lortie <desrt@desrt.ca>
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib-2.32.2/glib/gbacktrace.h
===================================================================
--- glib-2.32.2.orig/glib/gbacktrace.h	2012-04-13 01:39:20.000000000 +0200
+++ glib-2.32.2/glib/gbacktrace.h	2012-05-01 17:02:03.083885575 +0200
@@ -24,7 +24,7 @@
  * GLib at ftp://ftp.gtk.org/pub/gtk/.
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib-2.32.2/glib/gbase64.h
===================================================================
--- glib-2.32.2.orig/glib/gbase64.h	2012-04-13 01:39:20.000000000 +0200
+++ glib-2.32.2/glib/gbase64.h	2012-05-01 17:02:03.083885575 +0200
@@ -18,7 +18,7 @@
  * Boston, MA 02111-1307, USA.
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib-2.32.2/glib/gbitlock.h
===================================================================
--- glib-2.32.2.orig/glib/gbitlock.h	2012-04-13 01:39:20.000000000 +0200
+++ glib-2.32.2/glib/gbitlock.h	2012-05-01 17:02:03.083885575 +0200
@@ -25,7 +25,7 @@
 
 #include <glib/gtypes.h>
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib-2.32.2/glib/gbookmarkfile.h
===================================================================
--- glib-2.32.2.orig/glib/gbookmarkfile.h	2012-04-13 01:39:20.000000000 +0200
+++ glib-2.32.2/glib/gbookmarkfile.h	2012-05-01 17:02:03.083885575 +0200
@@ -17,7 +17,7 @@
  * Free Software Foundation, Inc., 59 Temple Place - Suite 330,
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib-2.32.2/glib/gchecksum.h
===================================================================
--- glib-2.32.2.orig/glib/gchecksum.h	2012-04-13 01:39:20.000000000 +0200
+++ glib-2.32.2/glib/gchecksum.h	2012-05-01 17:02:03.087885575 +0200
@@ -18,7 +18,7 @@
  * Boston, MA 02111-1307, USA.
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib-2.32.2/glib/gconvert.h
===================================================================
--- glib-2.32.2.orig/glib/gconvert.h	2012-04-13 01:39:20.000000000 +0200
+++ glib-2.32.2/glib/gconvert.h	2012-05-01 17:02:03.087885575 +0200
@@ -24,7 +24,7 @@
  * GLib at ftp://ftp.gtk.org/pub/gtk/.
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib-2.32.2/glib/gdataset.h
===================================================================
--- glib-2.32.2.orig/glib/gdataset.h	2012-04-13 01:39:20.000000000 +0200
+++ glib-2.32.2/glib/gdataset.h	2012-05-01 17:02:03.087885575 +0200
@@ -24,7 +24,7 @@
  * GLib at ftp://ftp.gtk.org/pub/gtk/.
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib-2.32.2/glib/gdate.h
===================================================================
--- glib-2.32.2.orig/glib/gdate.h	2012-04-13 01:39:20.000000000 +0200
+++ glib-2.32.2/glib/gdate.h	2012-05-01 17:02:03.087885575 +0200
@@ -24,7 +24,7 @@
  * GLib at ftp://ftp.gtk.org/pub/gtk/.
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib-2.32.2/glib/gdatetime.h
===================================================================
--- glib-2.32.2.orig/glib/gdatetime.h	2012-04-13 01:39:20.000000000 +0200
+++ glib-2.32.2/glib/gdatetime.h	2012-05-01 17:02:03.087885575 +0200
@@ -23,7 +23,7 @@
  *          Ryan Lortie <desrt@desrt.ca>
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib-2.32.2/glib/gdir.h
===================================================================
--- glib-2.32.2.orig/glib/gdir.h	2012-04-13 01:39:20.000000000 +0200
+++ glib-2.32.2/glib/gdir.h	2012-05-01 17:02:03.087885575 +0200
@@ -21,7 +21,7 @@
  * Boston, MA 02111-1307, USA.
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib-2.32.2/glib/gerror.h
===================================================================
--- glib-2.32.2.orig/glib/gerror.h	2012-04-13 01:39:20.000000000 +0200
+++ glib-2.32.2/glib/gerror.h	2012-05-01 17:02:03.087885575 +0200
@@ -18,7 +18,7 @@
  *   Boston, MA 02111-1307, USA.
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib-2.32.2/glib/gfileutils.h
===================================================================
--- glib-2.32.2.orig/glib/gfileutils.h	2012-04-13 01:39:20.000000000 +0200
+++ glib-2.32.2/glib/gfileutils.h	2012-05-01 17:02:03.087885575 +0200
@@ -18,7 +18,7 @@
  *   Boston, MA 02111-1307, USA.
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib-2.32.2/glib/ghash.h
===================================================================
--- glib-2.32.2.orig/glib/ghash.h	2012-04-13 01:39:20.000000000 +0200
+++ glib-2.32.2/glib/ghash.h	2012-05-01 17:02:03.091885573 +0200
@@ -24,7 +24,7 @@
  * GLib at ftp://ftp.gtk.org/pub/gtk/.
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib-2.32.2/glib/ghmac.h
===================================================================
--- glib-2.32.2.orig/glib/ghmac.h	2012-04-13 01:39:20.000000000 +0200
+++ glib-2.32.2/glib/ghmac.h	2012-05-01 17:02:03.091885573 +0200
@@ -18,7 +18,7 @@
  * Boston, MA 02111-1307, USA.
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib-2.32.2/glib/ghook.h
===================================================================
--- glib-2.32.2.orig/glib/ghook.h	2012-04-13 01:39:20.000000000 +0200
+++ glib-2.32.2/glib/ghook.h	2012-05-01 17:02:03.091885573 +0200
@@ -24,7 +24,7 @@
  * GLib at ftp://ftp.gtk.org/pub/gtk/.
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib-2.32.2/glib/giochannel.h
===================================================================
--- glib-2.32.2.orig/glib/giochannel.h	2012-04-13 01:39:20.000000000 +0200
+++ glib-2.32.2/glib/giochannel.h	2012-05-01 17:02:03.091885573 +0200
@@ -24,7 +24,7 @@
  * GLib at ftp://ftp.gtk.org/pub/gtk/.
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib-2.32.2/glib/gkeyfile.h
===================================================================
--- glib-2.32.2.orig/glib/gkeyfile.h	2012-05-01 17:02:00.827885655 +0200
+++ glib-2.32.2/glib/gkeyfile.h	2012-05-01 17:02:03.091885573 +0200
@@ -20,7 +20,7 @@
  *   Boston, MA 02111-1307, USA.
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib-2.32.2/glib/glist.h
===================================================================
--- glib-2.32.2.orig/glib/glist.h	2012-04-13 01:39:20.000000000 +0200
+++ glib-2.32.2/glib/glist.h	2012-05-01 17:02:03.091885573 +0200
@@ -24,7 +24,7 @@
  * GLib at ftp://ftp.gtk.org/pub/gtk/.
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib-2.32.2/glib/gmacros.h
===================================================================
--- glib-2.32.2.orig/glib/gmacros.h	2012-04-13 01:39:20.000000000 +0200
+++ glib-2.32.2/glib/gmacros.h	2012-05-01 17:02:03.091885573 +0200
@@ -28,7 +28,7 @@
  * not refer to variables from glibconfig.h
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib-2.32.2/glib/gmain.h
===================================================================
--- glib-2.32.2.orig/glib/gmain.h	2012-04-13 01:39:20.000000000 +0200
+++ glib-2.32.2/glib/gmain.h	2012-05-01 17:02:03.091885573 +0200
@@ -17,7 +17,7 @@
  * Boston, MA 02111-1307, USA.
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib-2.32.2/glib/gmappedfile.h
===================================================================
--- glib-2.32.2.orig/glib/gmappedfile.h	2012-04-13 01:39:20.000000000 +0200
+++ glib-2.32.2/glib/gmappedfile.h	2012-05-01 17:02:03.091885573 +0200
@@ -19,7 +19,7 @@
  * Boston, MA 02111-1307, USA.
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib-2.32.2/glib/gmarkup.h
===================================================================
--- glib-2.32.2.orig/glib/gmarkup.h	2012-04-13 01:39:20.000000000 +0200
+++ glib-2.32.2/glib/gmarkup.h	2012-05-01 17:02:03.091885573 +0200
@@ -18,7 +18,7 @@
  *   Boston, MA 02111-1307, USA.
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib-2.32.2/glib/gmem.h
===================================================================
--- glib-2.32.2.orig/glib/gmem.h	2012-05-01 02:06:39.000000000 +0200
+++ glib-2.32.2/glib/gmem.h	2012-05-01 17:02:03.091885573 +0200
@@ -24,7 +24,7 @@
  * GLib at ftp://ftp.gtk.org/pub/gtk/.
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib-2.32.2/glib/gmessages.h
===================================================================
--- glib-2.32.2.orig/glib/gmessages.h	2012-04-13 01:39:20.000000000 +0200
+++ glib-2.32.2/glib/gmessages.h	2012-05-01 17:02:03.095885571 +0200
@@ -24,7 +24,7 @@
  * GLib at ftp://ftp.gtk.org/pub/gtk/.
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib-2.32.2/glib/gnode.h
===================================================================
--- glib-2.32.2.orig/glib/gnode.h	2012-04-13 01:39:20.000000000 +0200
+++ glib-2.32.2/glib/gnode.h	2012-05-01 17:02:03.095885571 +0200
@@ -24,7 +24,7 @@
  * GLib at ftp://ftp.gtk.org/pub/gtk/.
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib-2.32.2/glib/goption.h
===================================================================
--- glib-2.32.2.orig/glib/goption.h	2012-04-13 01:39:20.000000000 +0200
+++ glib-2.32.2/glib/goption.h	2012-05-01 17:02:03.095885571 +0200
@@ -18,7 +18,7 @@
  * Boston, MA 02111-1307, USA.
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib-2.32.2/glib/gpattern.h
===================================================================
--- glib-2.32.2.orig/glib/gpattern.h	2012-04-13 01:39:20.000000000 +0200
+++ glib-2.32.2/glib/gpattern.h	2012-05-01 17:02:03.095885571 +0200
@@ -17,7 +17,7 @@
  * Boston, MA 02111-1307, USA.
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib-2.32.2/glib/gprimes.h
===================================================================
--- glib-2.32.2.orig/glib/gprimes.h	2012-04-13 01:39:20.000000000 +0200
+++ glib-2.32.2/glib/gprimes.h	2012-05-01 17:02:03.095885571 +0200
@@ -24,7 +24,7 @@
  * GLib at ftp://ftp.gtk.org/pub/gtk/.
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib-2.32.2/glib/gqsort.h
===================================================================
--- glib-2.32.2.orig/glib/gqsort.h	2012-04-13 01:39:20.000000000 +0200
+++ glib-2.32.2/glib/gqsort.h	2012-05-01 17:02:03.095885571 +0200
@@ -24,7 +24,7 @@
  * GLib at ftp://ftp.gtk.org/pub/gtk/.
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib-2.32.2/glib/gquark.h
===================================================================
--- glib-2.32.2.orig/glib/gquark.h	2012-04-13 01:39:20.000000000 +0200
+++ glib-2.32.2/glib/gquark.h	2012-05-01 17:02:03.095885571 +0200
@@ -24,7 +24,7 @@
  * GLib at ftp://ftp.gtk.org/pub/gtk/.
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib-2.32.2/glib/gqueue.h
===================================================================
--- glib-2.32.2.orig/glib/gqueue.h	2012-04-13 01:39:20.000000000 +0200
+++ glib-2.32.2/glib/gqueue.h	2012-05-01 17:02:03.095885571 +0200
@@ -24,7 +24,7 @@
  * GLib at ftp://ftp.gtk.org/pub/gtk/.
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib-2.32.2/glib/grand.h
===================================================================
--- glib-2.32.2.orig/glib/grand.h	2012-04-13 01:39:20.000000000 +0200
+++ glib-2.32.2/glib/grand.h	2012-05-01 17:02:03.095885571 +0200
@@ -24,7 +24,7 @@
  * GLib at ftp://ftp.gtk.org/pub/gtk/.
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib-2.32.2/glib/gregex.h
===================================================================
--- glib-2.32.2.orig/glib/gregex.h	2012-04-13 01:39:20.000000000 +0200
+++ glib-2.32.2/glib/gregex.h	2012-05-01 17:02:03.095885571 +0200
@@ -19,7 +19,7 @@
  * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib-2.32.2/glib/gscanner.h
===================================================================
--- glib-2.32.2.orig/glib/gscanner.h	2012-04-13 01:39:20.000000000 +0200
+++ glib-2.32.2/glib/gscanner.h	2012-05-01 17:02:03.095885571 +0200
@@ -24,7 +24,7 @@
  * GLib at ftp://ftp.gtk.org/pub/gtk/.
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib-2.32.2/glib/gsequence.h
===================================================================
--- glib-2.32.2.orig/glib/gsequence.h	2012-04-13 01:39:20.000000000 +0200
+++ glib-2.32.2/glib/gsequence.h	2012-05-01 17:02:03.095885571 +0200
@@ -18,7 +18,7 @@
  * Boston, MA 02111-1307, USA.
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib-2.32.2/glib/gshell.h
===================================================================
--- glib-2.32.2.orig/glib/gshell.h	2012-04-13 01:39:20.000000000 +0200
+++ glib-2.32.2/glib/gshell.h	2012-05-01 17:02:03.099885570 +0200
@@ -18,7 +18,7 @@
  * Boston, MA 02111-1307, USA.
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib-2.32.2/glib/gslice.h
===================================================================
--- glib-2.32.2.orig/glib/gslice.h	2012-04-13 01:39:20.000000000 +0200
+++ glib-2.32.2/glib/gslice.h	2012-05-01 17:02:03.099885570 +0200
@@ -17,7 +17,7 @@
  * Boston, MA 02111-1307, USA.
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib-2.32.2/glib/gslist.h
===================================================================
--- glib-2.32.2.orig/glib/gslist.h	2012-04-13 01:39:20.000000000 +0200
+++ glib-2.32.2/glib/gslist.h	2012-05-01 17:02:03.099885570 +0200
@@ -24,7 +24,7 @@
  * GLib at ftp://ftp.gtk.org/pub/gtk/.
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib-2.32.2/glib/gspawn.h
===================================================================
--- glib-2.32.2.orig/glib/gspawn.h	2012-04-13 01:39:20.000000000 +0200
+++ glib-2.32.2/glib/gspawn.h	2012-05-01 17:02:03.099885570 +0200
@@ -18,7 +18,7 @@
  * Boston, MA 02111-1307, USA.
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib-2.32.2/glib/gstrfuncs.h
===================================================================
--- glib-2.32.2.orig/glib/gstrfuncs.h	2012-04-13 01:39:20.000000000 +0200
+++ glib-2.32.2/glib/gstrfuncs.h	2012-05-01 17:02:03.099885570 +0200
@@ -24,7 +24,7 @@
  * GLib at ftp://ftp.gtk.org/pub/gtk/.
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib-2.32.2/glib/gstringchunk.h
===================================================================
--- glib-2.32.2.orig/glib/gstringchunk.h	2012-04-13 01:39:20.000000000 +0200
+++ glib-2.32.2/glib/gstringchunk.h	2012-05-01 17:02:03.099885570 +0200
@@ -24,7 +24,7 @@
  * GLib at ftp://ftp.gtk.org/pub/gtk/.
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib-2.32.2/glib/gstring.h
===================================================================
--- glib-2.32.2.orig/glib/gstring.h	2012-04-13 01:39:20.000000000 +0200
+++ glib-2.32.2/glib/gstring.h	2012-05-01 17:02:03.099885570 +0200
@@ -24,7 +24,7 @@
  * GLib at ftp://ftp.gtk.org/pub/gtk/.
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib-2.32.2/glib/gtestutils.h
===================================================================
--- glib-2.32.2.orig/glib/gtestutils.h	2012-04-13 01:39:20.000000000 +0200
+++ glib-2.32.2/glib/gtestutils.h	2012-05-01 17:02:03.099885570 +0200
@@ -18,7 +18,7 @@
  * Boston, MA 02111-1307, USA.
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib-2.32.2/glib/gthread.h
===================================================================
--- glib-2.32.2.orig/glib/gthread.h	2012-04-13 01:39:20.000000000 +0200
+++ glib-2.32.2/glib/gthread.h	2012-05-01 17:02:03.099885570 +0200
@@ -24,7 +24,7 @@
  * GLib at ftp://ftp.gtk.org/pub/gtk/.
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib-2.32.2/glib/gthreadpool.h
===================================================================
--- glib-2.32.2.orig/glib/gthreadpool.h	2012-04-13 01:39:20.000000000 +0200
+++ glib-2.32.2/glib/gthreadpool.h	2012-05-01 17:02:03.099885570 +0200
@@ -24,7 +24,7 @@
  * GLib at ftp://ftp.gtk.org/pub/gtk/.
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib-2.32.2/glib/gtimer.h
===================================================================
--- glib-2.32.2.orig/glib/gtimer.h	2012-04-13 01:39:20.000000000 +0200
+++ glib-2.32.2/glib/gtimer.h	2012-05-01 17:02:03.099885570 +0200
@@ -24,7 +24,7 @@
  * GLib at ftp://ftp.gtk.org/pub/gtk/.
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib-2.32.2/glib/gtimezone.h
===================================================================
--- glib-2.32.2.orig/glib/gtimezone.h	2012-04-13 01:39:20.000000000 +0200
+++ glib-2.32.2/glib/gtimezone.h	2012-05-01 17:02:03.103885572 +0200
@@ -19,7 +19,7 @@
  * Author: Ryan Lortie <desrt@desrt.ca>
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib-2.32.2/glib/gtree.h
===================================================================
--- glib-2.32.2.orig/glib/gtree.h	2012-04-13 01:39:20.000000000 +0200
+++ glib-2.32.2/glib/gtree.h	2012-05-01 17:02:03.103885572 +0200
@@ -24,7 +24,7 @@
  * GLib at ftp://ftp.gtk.org/pub/gtk/.
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib-2.32.2/glib/gtypes.h
===================================================================
--- glib-2.32.2.orig/glib/gtypes.h	2012-04-13 01:39:20.000000000 +0200
+++ glib-2.32.2/glib/gtypes.h	2012-05-01 17:02:03.103885572 +0200
@@ -24,7 +24,7 @@
  * GLib at ftp://ftp.gtk.org/pub/gtk/.
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib-2.32.2/glib/gunicode.h
===================================================================
--- glib-2.32.2.orig/glib/gunicode.h	2012-04-13 01:39:20.000000000 +0200
+++ glib-2.32.2/glib/gunicode.h	2012-05-01 17:02:03.103885572 +0200
@@ -19,7 +19,7 @@
  *   Boston, MA 02111-1307, USA.
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib-2.32.2/glib/gurifuncs.h
===================================================================
--- glib-2.32.2.orig/glib/gurifuncs.h	2012-04-13 01:39:20.000000000 +0200
+++ glib-2.32.2/glib/gurifuncs.h	2012-05-01 17:02:03.103885572 +0200
@@ -20,7 +20,7 @@
  * Author: Alexander Larsson <alexl@redhat.com>
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib-2.32.2/glib/gutils.h
===================================================================
--- glib-2.32.2.orig/glib/gutils.h	2012-04-30 18:24:02.000000000 +0200
+++ glib-2.32.2/glib/gutils.h	2012-05-01 17:02:03.103885572 +0200
@@ -24,7 +24,7 @@
  * GLib at ftp://ftp.gtk.org/pub/gtk/.
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib-2.32.2/glib/gvariant.h
===================================================================
--- glib-2.32.2.orig/glib/gvariant.h	2012-04-13 01:39:20.000000000 +0200
+++ glib-2.32.2/glib/gvariant.h	2012-05-01 17:02:03.103885572 +0200
@@ -20,7 +20,7 @@
  * Author: Ryan Lortie <desrt@desrt.ca>
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib-2.32.2/glib/gvarianttype.h
===================================================================
--- glib-2.32.2.orig/glib/gvarianttype.h	2012-04-13 01:39:20.000000000 +0200
+++ glib-2.32.2/glib/gvarianttype.h	2012-05-01 17:02:03.103885572 +0200
@@ -20,7 +20,7 @@
  * Author: Ryan Lortie <desrt@desrt.ca>
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib-2.32.2/glib/gwin32.h
===================================================================
--- glib-2.32.2.orig/glib/gwin32.h	2012-04-13 01:39:20.000000000 +0200
+++ glib-2.32.2/glib/gwin32.h	2012-05-01 17:02:03.103885572 +0200
@@ -24,7 +24,7 @@
  * GLib at ftp://ftp.gtk.org/pub/gtk/.
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
Index: glib-2.32.2/glib/gversionmacros.h
===================================================================
--- glib-2.32.2.orig/glib/gversionmacros.h	2012-05-01 02:06:39.000000000 +0200
+++ glib-2.32.2/glib/gversionmacros.h	2012-05-01 17:02:03.103885572 +0200
@@ -24,7 +24,7 @@
  * GLib at ftp://ftp.gtk.org/pub/gtk/.
  */
 
-#if !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
+#if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
 #error "Only <glib.h> can be included directly."
 #endif
 
