Index: glib-2.32.2/gio/giomodule.c
===================================================================
--- glib-2.32.2.orig/gio/giomodule.c	2012-04-30 18:24:02.000000000 +0200
+++ glib-2.32.2/gio/giomodule.c	2012-05-01 17:02:18.503885029 +0200
@@ -905,6 +905,8 @@
 
       /* Then load the compiled in path */
       g_io_modules_scan_all_in_directory_with_scope (GIO_MODULE_DIR, scope);
+      /* Hard-coded fallback directory for pre-multiarch compatibility */
+      g_io_modules_scan_all_in_directory ("/usr/lib/gio/modules");
 
       g_io_module_scope_free (scope);
 
