From bc5f333a125db5bae616562f2e84888d684987b3 Mon Sep 17 00:00:00 2001
From: Javier Jardón <jjardon@gnome.org>
Date: Wed, 06 Oct 2010 22:13:54 +0000
Subject: GtkObject is gone from GTK+3

gtk_adjustment_new() returns now a Gtkdjustment*
---
diff --git a/src/evolution-webcal-notify.c b/src/evolution-webcal-notify.c
index fa23481..450982a 100644
--- a/src/evolution-webcal-notify.c
+++ b/src/evolution-webcal-notify.c
@@ -217,7 +217,11 @@ void e_webcal_query_user (const gchar * name, const gchar * desc,
                           const gchar * caluri,
                           gboolean has_events, gboolean has_tasks) {
   EIcalShareDialog * dialog;
+#if GTK_CHECK_VERSION (2,91,0)
+  GtkAdjustment * spinadj;
+#else
   GtkObject * spinadj;
+#endif
   GtkWidget * tspinb, * combo;
   GtkWidget * vbox, * hbox;
   GtkWidget * lbox, * wbox, * xbox;
--
cgit v0.9.0.2
