Description: Add whatis entry to POD/manpage
Origin: vendor
Forwarded: no
Author: gregor herrmann <gregoa@debian.org>
Reviewed-by: Jonathan Yu <frequency@cpan.org>
--- a/lib/Class/MethodMaker/Constants.pm
+++ b/lib/Class/MethodMaker/Constants.pm
@@ -4,7 +4,7 @@
 
 =head1 NAME
 
-Class::MethodMaker::Constants
+Class::MethodMaker::Constants - The Constants Class for Class::MethodMaker
 
 =head1 SYNOPSIS
 
