Description: fix a spelling mistake
Origin: vendor
Forwarded: no
Author: gregor herrmann <gregoa@debian.org>
Last-Update: 2011-06-05

--- a/lib/Class/MethodMaker.pm
+++ b/lib/Class/MethodMaker.pm
@@ -460,7 +460,7 @@
 
 Setting the target class in this way does B<not> persist over multiple calls
 to C<use>/C<import>.  A subsequent call to either will use the
-default-determined class as target (unless again overriden by
+default-determined class as target (unless again overridden by
 C<-target_class>).
 
 =back
