From: Matthias Klose <doko@ubuntu.com>
Date: Sun, 05 Dec 2010 15:04:54 +0100
Subject: Fix FTBFS with ld --no-add-needed
Bug-Debian: http://bugs.debian.org/606006

--- ming.orig/util/Makefile.am
+++ ming/util/Makefile.am
@@ -114,16 +114,16 @@
 makefdb_CFLAGS = -DMAKE_FDB
 
 gif2mask_SOURCES = gif2mask.c
-gif2mask_LDADD = $(top_builddir)/src/libming.la
+gif2mask_LDADD = $(top_builddir)/src/libming.la $(GIFLIB) $(MATHLIB) $(ZLIB)
 
 png2dbl_SOURCES = png2dbl.c
-png2dbl_LDADD = $(top_builddir)/src/libming.la
+png2dbl_LDADD = $(top_builddir)/src/libming.la $(PNGLIB) $(MATHLIB) $(ZLIB)
 
 dbl2png_SOURCES = dbl2png.c
-dbl2png_LDADD = $(top_builddir)/src/libming.la
+dbl2png_LDADD = $(top_builddir)/src/libming.la $(PNGLIB) $(ZLIB)
 
 gif2dbl_SOURCES = gif2dbl.c
-gif2dbl_LDADD = $(top_builddir)/src/libming.la
+gif2dbl_LDADD = $(top_builddir)/src/libming.la $(GIFLIB) $(MATHLIB) $(ZLIB)
 
 makeswf_SOURCES = makeswf.c makeswf_utils.c
 makeswf_LDADD = $(top_builddir)/src/libming.la
