From: Nobuhiro Iwamatsu <iwamatsu@nigauri.org>
Date: Sat, 30 Jul 2011 16:45:17 +0900
Subject: FTBFS with libpng 1.5
Bug-Debian: http://bugs.debian.org/636005

--- a/util/png2dbl.c
+++ b/util/png2dbl.c
@@ -91,7 +91,11 @@ struct pngdata readPNG(FILE *fp)
     error("Couldn't create end_info\n");
   }
 
-  if(setjmp(png_jmpbuf(png_ptr)))
+#if PNG_LIBPNG_VER_MAJOR >= 1 && PNG_LIBPNG_VER_MINOR >= 4
+  if (setjmp(png_jmpbuf((png_ptr))))
+#else
+  if(setjmp(png_ptr->jmpbuf))
+#endif
   {
     png_destroy_read_struct(&png_ptr, &info_ptr, &end_info);
     fclose(fp);
--- a/src/blocks/pngdbl.c
+++ b/src/blocks/pngdbl.c
@@ -61,7 +61,12 @@ static png_structp openPngFromFile(FILE
 }
 
 static int pngReadFunc(png_structp png, unsigned char *buf, int len)
-{	SWFInput input = (SWFInput) png_get_io_ptr(png);
+{
+#if PNG_LIBPNG_VER_MAJOR >= 1 && PNG_LIBPNG_VER_MINOR >= 4
+	SWFInput input = (SWFInput) png_get_io_ptr(png);
+#else
+	SWFInput input = (SWFInput) png->io_ptr;
+#endif
 	return SWFInput_read(input, buf, len);
 }
 
@@ -127,7 +132,11 @@ static int readPNG(png_structp png_ptr,
 		return 0;
 	}
 
+#if PNG_LIBPNG_VER_MAJOR >= 1 && PNG_LIBPNG_VER_MINOR >= 4
+	if (setjmp(png_jmpbuf(png_ptr)))
+#else
 	if(setjmp(png_jmpbuf(png_ptr)))
+#endif
 	{
 		png_destroy_read_struct(&png_ptr, &info_ptr, &end_info);
     	return 0;
