# Explain mount options.

diff -Naur gphotofs-0.4.0.orig/README gphotofs-0.4.0/README
--- gphotofs-0.4.0.orig/README	2009-04-25 01:14:25.000000000 +0200
+++ gphotofs-0.4.0/README	2009-04-25 01:15:25.000000000 +0200
@@ -49,6 +49,21 @@
 
    fusermount -u <mountpoint>
 
+
+Alternatively you can put an entry into /etc/fstab (please adapt the
+mount options to your needs):
+
+   gphotofs /media/camera fuse users,rw,noauto,defaults 0   0
+
+Make sure in that case /media/camera exists:
+
+   [ -d /media/camera ] || mkdir /media/camera
+
+Now you should be able to mount your camera by simpy issuing:
+
+   mount /media/camera
+
+
 Usage
 -----
 
