Patch by Ryan Niebur from #521924.

index 00c73d5..550254f 100644
--- a/include/wx/defs.h
+++ b/include/wx/defs.h
@@ -494,19 +494,19 @@ typedef int wxWindowID;
 /*  ---------------------------------------------------------------------------- */
 
 /*  Printf-like attribute definitions to obtain warnings with GNU C/C++ */
-#ifndef ATTRIBUTE_PRINTF
+#ifndef WX_ATTRIBUTE_PRINTF
 #   if defined(__GNUC__) && !wxUSE_UNICODE
-#       define ATTRIBUTE_PRINTF(m, n) __attribute__ ((__format__ (__printf__, m, n)))
+#       define WX_ATTRIBUTE_PRINTF(m, n) __attribute__ ((__format__ (__printf__, m, n)))
 #   else
-#       define ATTRIBUTE_PRINTF(m, n)
+#       define WX_ATTRIBUTE_PRINTF(m, n)
 #   endif
 
-#   define ATTRIBUTE_PRINTF_1 ATTRIBUTE_PRINTF(1, 2)
-#   define ATTRIBUTE_PRINTF_2 ATTRIBUTE_PRINTF(2, 3)
-#   define ATTRIBUTE_PRINTF_3 ATTRIBUTE_PRINTF(3, 4)
-#   define ATTRIBUTE_PRINTF_4 ATTRIBUTE_PRINTF(4, 5)
-#   define ATTRIBUTE_PRINTF_5 ATTRIBUTE_PRINTF(5, 6)
-#endif /* !defined(ATTRIBUTE_PRINTF) */
+#   define WX_ATTRIBUTE_PRINTF_1 WX_ATTRIBUTE_PRINTF(1, 2)
+#   define WX_ATTRIBUTE_PRINTF_2 WX_ATTRIBUTE_PRINTF(2, 3)
+#   define WX_ATTRIBUTE_PRINTF_3 WX_ATTRIBUTE_PRINTF(3, 4)
+#   define WX_ATTRIBUTE_PRINTF_4 WX_ATTRIBUTE_PRINTF(4, 5)
+#   define WX_ATTRIBUTE_PRINTF_5 WX_ATTRIBUTE_PRINTF(5, 6)
+#endif /* !defined(WX_ATTRIBUTE_PRINTF) */
 
 /*  Macro to issue warning when using deprecated functions with gcc3 or MSVC7: */
 #if wxCHECK_GCC_VERSION(3, 1)
diff --git a/include/wx/log.h b/include/wx/log.h
index 288e8e5..0fdb128 100644
--- a/include/wx/log.h
+++ b/include/wx/log.h
@@ -492,14 +492,14 @@ WXDLLIMPEXP_BASE const wxChar* wxSysErrorMsg(unsigned long nErrCode = 0);
 extern void WXDLLIMPEXP_BASE wxVLog##level(const wxChar *szFormat,          \
                                            va_list argptr);                 \
 extern void WXDLLIMPEXP_BASE wxLog##level(const wxChar *szFormat,           \
-                                          ...) ATTRIBUTE_PRINTF_1
+                                          ...) WX_ATTRIBUTE_PRINTF_1
 #define DECLARE_LOG_FUNCTION2_EXP(level, argclass, arg, expdecl)            \
 extern void expdecl wxVLog##level(argclass arg,                             \
                                   const wxChar *szFormat,                   \
                                   va_list argptr);                          \
 extern void expdecl wxLog##level(argclass arg,                              \
                                  const wxChar *szFormat,                    \
-                                 ...) ATTRIBUTE_PRINTF_2
+                                 ...) WX_ATTRIBUTE_PRINTF_2
 #else // !wxUSE_LOG
 
 // log functions do nothing at all
diff --git a/include/wx/memory.h b/include/wx/memory.h
index adb8ed2..4c66cf9 100644
--- a/include/wx/memory.h
+++ b/include/wx/memory.h
@@ -344,8 +344,8 @@ static wxDebugContextDumpDelayCounter wxDebugContextDumpDelayCounter_File;
     static wxDebugContextDumpDelayCounter wxDebugContextDumpDelayCounter_Extra;
 
 // Output a debug message, in a system dependent fashion.
-void WXDLLIMPEXP_BASE wxTrace(const wxChar *fmt ...) ATTRIBUTE_PRINTF_1;
-void WXDLLIMPEXP_BASE wxTraceLevel(int level, const wxChar *fmt ...) ATTRIBUTE_PRINTF_2;
+void WXDLLIMPEXP_BASE wxTrace(const wxChar *fmt ...) WX_ATTRIBUTE_PRINTF_1;
+void WXDLLIMPEXP_BASE wxTraceLevel(int level, const wxChar *fmt ...) WX_ATTRIBUTE_PRINTF_2;
 
 #define WXTRACE wxTrace
 #define WXTRACELEVEL wxTraceLevel
diff --git a/include/wx/msgout.h b/include/wx/msgout.h
index 6130849..942448f 100755
--- a/include/wx/msgout.h
+++ b/include/wx/msgout.h
@@ -30,7 +30,7 @@ public:
     virtual ~wxMessageOutput() { }
 
     // show a message to the user
-    virtual void Printf(const wxChar* format, ...)  ATTRIBUTE_PRINTF_2 = 0;
+    virtual void Printf(const wxChar* format, ...)  WX_ATTRIBUTE_PRINTF_2 = 0;
 
     // gets the current wxMessageOutput object (may be NULL during
     // initialization or shutdown)
@@ -54,7 +54,7 @@ class WXDLLIMPEXP_BASE wxMessageOutputBest : public wxMessageOutput
 public:
     wxMessageOutputBest() { }
 
-    virtual void Printf(const wxChar* format, ...) ATTRIBUTE_PRINTF_2;
+    virtual void Printf(const wxChar* format, ...) WX_ATTRIBUTE_PRINTF_2;
 };
 
 // ----------------------------------------------------------------------------
@@ -66,7 +66,7 @@ class WXDLLIMPEXP_BASE wxMessageOutputStderr : public wxMessageOutput
 public:
     wxMessageOutputStderr() { }
 
-    virtual void Printf(const wxChar* format, ...) ATTRIBUTE_PRINTF_2;
+    virtual void Printf(const wxChar* format, ...) WX_ATTRIBUTE_PRINTF_2;
 };
 
 // ----------------------------------------------------------------------------
@@ -80,7 +80,7 @@ class WXDLLIMPEXP_CORE wxMessageOutputMessageBox : public wxMessageOutput
 public:
     wxMessageOutputMessageBox() { }
 
-    virtual void Printf(const wxChar* format, ...) ATTRIBUTE_PRINTF_2;
+    virtual void Printf(const wxChar* format, ...) WX_ATTRIBUTE_PRINTF_2;
 };
 
 #endif // wxUSE_GUI
@@ -94,7 +94,7 @@ class WXDLLIMPEXP_BASE wxMessageOutputDebug : public wxMessageOutput
 public:
     wxMessageOutputDebug() { }
 
-    virtual void Printf(const wxChar* format, ...) ATTRIBUTE_PRINTF_2;
+    virtual void Printf(const wxChar* format, ...) WX_ATTRIBUTE_PRINTF_2;
 };
 
 // ----------------------------------------------------------------------------
@@ -106,7 +106,7 @@ class WXDLLIMPEXP_BASE wxMessageOutputLog : public wxMessageOutput
 public:
     wxMessageOutputLog() { }
 
-    virtual void Printf(const wxChar* format, ...) ATTRIBUTE_PRINTF_2;
+    virtual void Printf(const wxChar* format, ...) WX_ATTRIBUTE_PRINTF_2;
 };
 
 #endif
diff --git a/include/wx/string.h b/include/wx/string.h
index 7a57f24..c06eb82 100644
--- a/include/wx/string.h
+++ b/include/wx/string.h
@@ -1200,12 +1200,12 @@ public:
   // formatted input/output
     // as sprintf(), returns the number of characters written or < 0 on error
     // (take 'this' into account in attribute parameter count)
-  int Printf(const wxChar *pszFormat, ...) ATTRIBUTE_PRINTF_2;
+  int Printf(const wxChar *pszFormat, ...) WX_ATTRIBUTE_PRINTF_2;
     // as vprintf(), returns the number of characters written or < 0 on error
   int PrintfV(const wxChar* pszFormat, va_list argptr);
 
     // returns the string containing the result of Printf() to it
-  static wxString Format(const wxChar *pszFormat, ...) ATTRIBUTE_PRINTF_1;
+  static wxString Format(const wxChar *pszFormat, ...) WX_ATTRIBUTE_PRINTF_1;
     // the same as above, but takes a va_list
   static wxString FormatV(const wxChar *pszFormat, va_list argptr);
 
@@ -1237,7 +1237,7 @@ public:
 
   // use Printf()
   // (take 'this' into account in attribute parameter count)
-  int sprintf(const wxChar *pszFormat, ...) ATTRIBUTE_PRINTF_2;
+  int sprintf(const wxChar *pszFormat, ...) WX_ATTRIBUTE_PRINTF_2;
 
     // use Cmp()
   inline int CompareTo(const wxChar* psz, caseCompare cmp = exact) const
diff --git a/include/wx/wxchar.h b/include/wx/wxchar.h
index bbdcab1..b33db89 100644
--- a/include/wx/wxchar.h
+++ b/include/wx/wxchar.h
@@ -1017,7 +1017,7 @@ WXDLLIMPEXP_BASE bool wxOKlibc(); /* for internal use */
 #ifndef wxSnprintf_
     /* no snprintf(), cook our own */
     WXDLLIMPEXP_BASE int
-    wxSnprintf_(wxChar *buf, size_t len, const wxChar *format, ...) ATTRIBUTE_PRINTF_3;
+    wxSnprintf_(wxChar *buf, size_t len, const wxChar *format, ...) WX_ATTRIBUTE_PRINTF_3;
 #endif
 #ifndef wxVsnprintf_
     /* no (suitable) vsnprintf(), cook our own */
@@ -1057,13 +1057,13 @@ WXDLLIMPEXP_BASE bool wxOKlibc(); /* for internal use */
         either because we don't have them at all or because they don't have the
         semantics we need
      */
-    int wxScanf( const wxChar *format, ... ) ATTRIBUTE_PRINTF_1;
-    int wxSscanf( const wxChar *str, const wxChar *format, ... ) ATTRIBUTE_PRINTF_2;
-    int wxFscanf( FILE *stream, const wxChar *format, ... ) ATTRIBUTE_PRINTF_2;
+    int wxScanf( const wxChar *format, ... ) WX_ATTRIBUTE_PRINTF_1;
+    int wxSscanf( const wxChar *str, const wxChar *format, ... ) WX_ATTRIBUTE_PRINTF_2;
+    int wxFscanf( FILE *stream, const wxChar *format, ... ) WX_ATTRIBUTE_PRINTF_2;
     int wxVsscanf( const wxChar *str, const wxChar *format, va_list ap );
-    int wxPrintf( const wxChar *format, ... ) ATTRIBUTE_PRINTF_1;
-    int wxSprintf( wxChar *str, const wxChar *format, ... ) ATTRIBUTE_PRINTF_2;
-    int wxFprintf( FILE *stream, const wxChar *format, ... ) ATTRIBUTE_PRINTF_2;
+    int wxPrintf( const wxChar *format, ... ) WX_ATTRIBUTE_PRINTF_1;
+    int wxSprintf( wxChar *str, const wxChar *format, ... ) WX_ATTRIBUTE_PRINTF_2;
+    int wxFprintf( FILE *stream, const wxChar *format, ... ) WX_ATTRIBUTE_PRINTF_2;
     int wxVfprintf( FILE *stream, const wxChar *format, va_list ap );
     int wxVprintf( const wxChar *format, va_list ap );
     int wxVsprintf( wxChar *str, const wxChar *format, va_list ap );
@@ -1076,7 +1076,7 @@ WXDLLIMPEXP_BASE bool wxOKlibc(); /* for internal use */
    anything as our own wxVsnprintf_() already behaves as needed.
 */
 #if defined(wxNEED_PRINTF_CONVERSION) && defined(wxVsnprintf_)
-    int wxSnprintf( wxChar *str, size_t size, const wxChar *format, ... ) ATTRIBUTE_PRINTF_3;
+    int wxSnprintf( wxChar *str, size_t size, const wxChar *format, ... ) WX_ATTRIBUTE_PRINTF_3;
     int wxVsnprintf( wxChar *str, size_t size, const wxChar *format, va_list ap );
 #else
     #define wxSnprintf wxSnprintf_
