Author: Daniel Baumann <daniel@debian.org>
Description: Adding info header (Closes: #528867).

diff -Naurp barcode.orig/doc/barcode.info barcode/doc/barcode.info
--- barcode.orig/doc/barcode.info	2009-07-16 23:49:31.000000000 +0200
+++ barcode/doc/barcode.info	2009-07-17 00:08:22.000000000 +0200
@@ -1,3 +1,8 @@
+START-INFO-DIR-ENTRY
+* Barcode: (barcode).        Utility and library for barcode generation
+END-INFO-DIR-ENTRY
+INFO-DIR-SECTION Graphics
+
 This is barcode.info, produced by makeinfo version 4.0 from
 barcode.texinfo.
 
