From: Ansgar Burchardt <ansgar@43-1.org>
Date: Fri, 29 Jan 2010 01:41:34 +0900
Subject: Correct POD syntax errors
--- libxml-perl.orig/lib/XML/Parser/PerlSAX.pm
+++ libxml-perl/lib/XML/Parser/PerlSAX.pm
@@ -525,6 +525,8 @@
 entity.  The column, line, and byte positions I<are> of the current
 entity being parsed.
 
+=back
+
 =head1 OPTIONS
 
 The following options are supported by C<XML::Parser::PerlSAX>:
