From: Ansgar Burchardt <ansgar@43-1.org>
Date: Fri, 29 Jan 2010 01:40:17 +0900
Subject: correct spelling errors
--- libxml-perl.orig/lib/XML/Handler/XMLWriter.pm
+++ libxml-perl/lib/XML/Handler/XMLWriter.pm
@@ -256,7 +256,7 @@
 A true or false value; if this parameter is present and its value is
 true, then the module will insert an extra newline before the closing
 delimiter of start, end, and empty tags to guarantee that the document
-does not end up as a single, long line.  If the paramter is not
+does not end up as a single, long line.  If the parameter is not
 present, the module will not insert the newlines.
 
 =item IsSGML
--- libxml-perl.orig/lib/XML/PatAct/MatchName.pm
+++ libxml-perl/lib/XML/PatAct/MatchName.pm
@@ -76,7 +76,7 @@
 XML::PatAct::MatchName is a pattern module for use with PatAct drivers
 for applying pattern-action lists to XML parses or trees.
 XML::PatAct::MatchName is a simple pattern module that uses just
-element names to match on.  If multiple names are supplied seperated
+element names to match on.  If multiple names are supplied separated
 by `C</>' characters, then all of the parent element names must match
 as well.
 
