Author: Fabian Greffrath <fabian+debian@greffrath.com>
Subject: Fix format string security errors.

Forwarded: no
--- a/libhb/deccc608sub.c
+++ b/libhb/deccc608sub.c
@@ -1607,7 +1607,7 @@
     if (debug_608)
     {
         hb_log ("\n- - - SRT caption - - -\n");
-        hb_log (timeline);
+        hb_log ("%s", timeline);
     }
     //fwrite (enc_buffer,enc_buffer_used,1,wb->fh);		
     XMLRPC_APPEND(wb->enc_buffer,wb->enc_buffer_used);
--- a/libhb/decmpeg2.c
+++ b/libhb/decmpeg2.c
@@ -644,7 +644,7 @@
                 hb_audio_t *audio = hb_list_item( m->title->list_audio, 0 );
                 if( audio )
                 {
-                    snprintf( subtitle->iso639_2, sizeof( subtitle->iso639_2 ), 
+                    snprintf( "%s", subtitle->iso639_2, sizeof( subtitle->iso639_2 ),
                               audio->config.lang.iso639_2);
                 } else {
                     snprintf( subtitle->iso639_2, sizeof( subtitle->iso639_2 ), "und");
