--- tuxpaint-0.9.21/src/tuxpaint.c.png15	2012-03-22 09:32:19.347356715 -0400
+++ tuxpaint-0.9.21/src/tuxpaint.c	2012-03-22 09:45:06.547102782 -0400
@@ -12898,12 +12898,16 @@ static int do_png_save(FILE * fi, const
       {
 	png_init_io(png_ptr, fi);
 
+#if (PNG_LIBPNG_VER_MAJOR >= 1 && PNG_LIBPNG_VER_MINOR >= 4)
+	png_set_IHDR(png_ptr, info_ptr, surf->w, surf->h, 8, PNG_COLOR_TYPE_RGB, 1, PNG_COMPRESSION_TYPE_DEFAULT, PNG_FILTER_TYPE_DEFAULT);
+#else
 	info_ptr->width = surf->w;
 	info_ptr->height = surf->h;
 	info_ptr->bit_depth = 8;
 	info_ptr->color_type = PNG_COLOR_TYPE_RGB;
 	info_ptr->interlace_type = 1;
 	info_ptr->valid = 0;	/* will be updated by various png_set_FOO() functions */
+#endif
 
 	png_set_sRGB_gAMA_and_cHRM(png_ptr, info_ptr,
 				   PNG_sRGB_INTENT_PERCEPTUAL);

