From: Otavio Salvador <otavio@debian.org>
Date: Fri, 30 Nov 2012 08:09:12 +0100
Subject: 01_no_epydoc

---
 Makefile |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index af7920f..229e86d 100644
--- a/Makefile
+++ b/Makefile
@@ -17,7 +17,9 @@ cups.so: force
 
 doc:	cups.so
 	rm -rf html
-	epydoc -o html --html $<
+	mkdir -p html
+	pydoc -w $(patsubst %.so,%,$<)
+	mv $(patsubst %.so,%.html,$<) html
 
 doczip:	doc
 	cd html && zip ../cups-html.zip *
