From 0d2dedb78836ff04804c75bd258306672eecb24f Mon Sep 17 00:00:00 2001
From: nick black <nick.black@sprezzatech.com>
Date: Thu, 28 Mar 2013 18:31:48 -0400
Subject: [PATCH] replace deprecated av_find_stream_info()

---
 addons/video/ffmpeg.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/addons/video/ffmpeg.c b/addons/video/ffmpeg.c
index 937c62a..b16e628 100644
--- a/addons/video/ffmpeg.c
+++ b/addons/video/ffmpeg.c
@@ -1112,7 +1112,7 @@ static bool open_video(ALLEGRO_VIDEO *video)
       return false;
    }
 
-   if (av_find_stream_info(is->format_context) < 0) {
+   if (avformat_find_stream_info(is->format_context, NULL) < 0) {
       av_free(is);
       return false;
    }
-- 
1.8.2

