Description: Pass -fvisibility-inlines-hidden to the compiler
Origin: vendor
Author: George Kiagiadakis <george.kiagiadakis@collabora.com>

--- a/cmake/modules/TelepathyDefaults.cmake
+++ b/cmake/modules/TelepathyDefaults.cmake
@@ -47,6 +47,11 @@
         set(VISIBILITY_HIDDEN_FLAGS)
     endif (CXX_FVISIBILITY_HIDDEN)
 
+    CHECK_CXX_ACCEPTS_FLAG("-fvisibility-inlines-hidden" CXX_FVISIBILITY_INLINES_HIDDEN)
+    if (CXX_FVISIBILITY_INLINES_HIDDEN)
+        set(VISIBILITY_HIDDEN_FLAGS "${VISIBILITY_HIDDEN_FLAGS} -fvisibility-inlines-hidden")
+    endif (CXX_FVISIBILITY_INLINES_HIDDEN)
+
     CHECK_CXX_ACCEPTS_FLAG("-Wdeprecated-declarations" CXX_DEPRECATED_DECLARATIONS)
     if (CXX_DEPRECATED_DECLARATIONS)
         set(DEPRECATED_DECLARATIONS_FLAGS "-Wdeprecated-declarations -DTP_QT_DEPRECATED_WARNINGS")
