Package:     fox1.6
Subject:     prepare for external buildtree
Author:      Torsten Landschoff <torsten at debian.org>
Forwarded:   no
Last-Update: 2012-05-13
---

diff -urN s00/adie/Makefile.am s01/adie/Makefile.am
--- s00/adie/Makefile.am	2012-04-27 18:38:51.000000000 +0200
+++ s01/adie/Makefile.am	2012-05-13 10:27:14.311528911 +0200
@@ -77,7 +77,7 @@
 $(ICONS)
 
 
-Adie.cpp: icons.h icons.cpp
+Adie.lo: icons.h icons.cpp
 
 
 icons.h: $(ICONS)
diff -urN s00/calculator/Makefile.am s01/calculator/Makefile.am
--- s00/calculator/Makefile.am	2012-04-27 18:38:51.000000000 +0200
+++ s01/calculator/Makefile.am	2012-05-13 10:27:14.369739589 +0200
@@ -35,7 +35,7 @@
 $(ICONS)
 
 
-Calculator.cpp: icons.h icons.cpp
+Calculator.lo: icons.h icons.cpp
 
 
 icons.h: $(ICONS)
diff -urN s00/pathfinder/Makefile.am s01/pathfinder/Makefile.am
--- s00/pathfinder/Makefile.am	2012-04-27 18:38:51.000000000 +0200
+++ s01/pathfinder/Makefile.am	2012-05-13 10:27:14.414409237 +0200
@@ -63,7 +63,7 @@
 LICENSE \
 $(ICONS)
 
-PathFinder.cpp: icons.h icons.cpp
+PathFinder.lo: icons.h icons.cpp
 
 icons.h: $(ICONS)
 	$(RESWRAP) -i -o icons.h $^
diff -urN s00/shutterbug/Makefile.am s01/shutterbug/Makefile.am
--- s00/shutterbug/Makefile.am	2012-04-27 18:38:51.000000000 +0200
+++ s01/shutterbug/Makefile.am	2012-05-13 10:27:14.458935500 +0200
@@ -36,7 +36,7 @@
 $(ICONS)
 
 
-ShutterBug.cpp: icons.h icons.cpp
+ShutterBug.lo: icons.h icons.cpp
 
 
 icons.h: $(ICONS)
diff -urN s00/src/Makefile.am s01/src/Makefile.am
--- s00/src/Makefile.am	2012-04-27 18:38:51.000000000 +0200
+++ s01/src/Makefile.am	2012-05-13 10:27:14.275411164 +0200
@@ -447,14 +447,14 @@
 $(CURSORS) \
 $(ICONS)
 
-FXApp.cpp: icons.h
+FXApp.lo: icons.h
 
 icons.h: $(ICONS)
-	$(RESWRAP) -i -n FX -o icons.h $(ICONS)
+	$(RESWRAP) -i -n FX -o icons.h $^
 
 
 icons.cpp: $(ICONS)
-	$(RESWRAP) -e -n FX -o icons.cpp $(ICONS)
+	$(RESWRAP) -e -n FX -o icons.cpp $^
 
 
 EXTRA_DIST = Makefile.bc \
