Package:     fox1.6
Subject:     fix automake cxxflags
Author:      Joachim Wiedorn <ad_debian at joonet.de>
Forwarded:   no
Last-Update: 2012-05-13
---

diff -urN s01/adie/Makefile.am s02/adie/Makefile.am
--- s01/adie/Makefile.am	2012-05-13 10:27:14.311528911 +0200
+++ s02/adie/Makefile.am	2012-05-13 10:27:30.611410121 +0200
@@ -1,6 +1,6 @@
 ## Process this file with automake to produce Makefile.in
 
-CXXFLAGS = @CXXFLAGS@ @X_CFLAGS@
+AM_CXXFLAGS = @CXXFLAGS@ @X_CFLAGS@
 INCLUDES = -I$(top_srcdir)/include -I$(top_builddir)/include
 RESWRAP  = $(top_builddir)/utils/reswrap
 SUFFIXES = .gif .bmp
diff -urN s01/calculator/Makefile.am s02/calculator/Makefile.am
--- s01/calculator/Makefile.am	2012-05-13 10:27:14.369739589 +0200
+++ s02/calculator/Makefile.am	2012-05-13 10:27:30.656008249 +0200
@@ -1,6 +1,6 @@
 ## Process this file with automake to produce Makefile.in
 
-CXXFLAGS = @CXXFLAGS@ @X_CFLAGS@
+AM_CXXFLAGS = @CXXFLAGS@ @X_CFLAGS@
 INCLUDES = -I$(top_srcdir)/include -I$(top_builddir)/include
 RESWRAP  = $(top_builddir)/utils/reswrap
 SUFFIXES = .gif .bmp
diff -urN s01/chart/Makefile.am s02/chart/Makefile.am
--- s01/chart/Makefile.am	2012-04-27 18:38:51.000000000 +0200
+++ s02/chart/Makefile.am	2012-05-13 10:27:30.670411894 +0200
@@ -1,6 +1,6 @@
 ## Process this file with automake to produce Makefile.in
 
-CXXFLAGS = @CXXFLAGS@ @X_CFLAGS@
+AM_CXXFLAGS = @CXXFLAGS@ @X_CFLAGS@
 INCLUDES = -I$(top_srcdir)/include -I$(top_builddir)/include
 RESWRAP  = $(top_builddir)/utils/reswrap
 SUFFIXES = .gif .bmp
diff -urN s01/pathfinder/Makefile.am s02/pathfinder/Makefile.am
--- s01/pathfinder/Makefile.am	2012-05-13 10:27:14.414409237 +0200
+++ s02/pathfinder/Makefile.am	2012-05-13 10:27:30.722908339 +0200
@@ -1,6 +1,6 @@
 ## Process this file with automake to produce Makefile.in
 
-CXXFLAGS = @CXXFLAGS@ @X_CFLAGS@
+AM_CXXFLAGS = @CXXFLAGS@ @X_CFLAGS@
 INCLUDES = -I$(top_srcdir)/include -I$(top_builddir)/include
 RESWRAP  = $(top_builddir)/utils/reswrap
 SUFFIXES = .gif .bmp
diff -urN s01/shutterbug/Makefile.am s02/shutterbug/Makefile.am
--- s01/shutterbug/Makefile.am	2012-05-13 10:27:14.458935500 +0200
+++ s02/shutterbug/Makefile.am	2012-05-13 10:27:30.756325104 +0200
@@ -1,6 +1,6 @@
 ## Process this file with automake to produce Makefile.in
 
-CXXFLAGS = @CXXFLAGS@ @X_CFLAGS@
+AM_CXXFLAGS = @CXXFLAGS@ @X_CFLAGS@
 INCLUDES = -I$(top_srcdir)/include -I$(top_builddir)/include
 RESWRAP  = $(top_builddir)/utils/reswrap
 SUFFIXES = .gif .bmp
diff -urN s01/src/Makefile.am s02/src/Makefile.am
--- s01/src/Makefile.am	2012-05-13 10:27:14.275411164 +0200
+++ s02/src/Makefile.am	2012-05-13 10:27:30.767473570 +0200
@@ -1,6 +1,6 @@
 ## Process this file with automake to produce Makefile.in
 
-CXXFLAGS = @CXXFLAGS@ @X_CFLAGS@
+AM_CXXFLAGS = @CXXFLAGS@ @X_CFLAGS@
 INCLUDES = -I$(top_srcdir)/include -I$(top_builddir)/include
 RESWRAP  = $(top_builddir)/utils/reswrap
 SUFFIXES = .gif .bmp
diff -urN s01/tests/Makefile.am s02/tests/Makefile.am
--- s01/tests/Makefile.am	2012-04-27 18:38:51.000000000 +0200
+++ s02/tests/Makefile.am	2012-05-13 10:27:30.791941110 +0200
@@ -1,6 +1,6 @@
 ## Process this file with automake to produce Makefile.in
 
-CXXFLAGS = @CXXFLAGS@ @X_CFLAGS@
+AM_CXXFLAGS = @CXXFLAGS@ @X_CFLAGS@
 INCLUDES = -I$(top_srcdir)/include -I$(top_builddir)/include
 RESWRAP  = $(top_builddir)/utils/reswrap
 SUFFIXES = .gif .bmp
diff -urN s01/utils/Makefile.am s02/utils/Makefile.am
--- s01/utils/Makefile.am	2012-04-27 18:38:51.000000000 +0200
+++ s02/utils/Makefile.am	2012-05-13 10:27:30.810452001 +0200
@@ -1,6 +1,6 @@
 ## Process this file with automake to produce Makefile.in
 
-CXXFLAGS = @CXXFLAGS@ @X_CFLAGS@
+AM_CXXFLAGS = @CXXFLAGS@ @X_CFLAGS@
 
 LIBS=
 
