Package:     fox1.6
Subject:     link to freetype
Author:      Torsten Landschoff <torsten at debian.org>
Forwarded:   no
Last-Update: 2012-05-13
---

diff -urN s03/configure.ac s04/configure.ac
--- s03/configure.ac	2012-04-27 18:39:00.000000000 +0200
+++ s04/configure.ac	2012-05-13 10:28:01.687411560 +0200
@@ -280,6 +280,19 @@
 # Substitute variables in output files
 AC_SUBST(GL_LIBS)
 
+
+AC_MSG_CHECKING([if we should link against libfreetype2])
+link_to_freetype="no"
+if test "x$with_opengl" != "xno" -a  "x$with_xft" == "xyes"; then
+  FREETYPE2LIBS=`pkg-config --libs freetype2`
+  if test "x$FREETYPE2LIBS" != "x"; then
+    X_BASE_LIBS="$X_BASE_LIBS $FREETYPE2LIBS"
+    link_to_freetype="yes ($FREETYPE2LIBS)"
+  fi
+fi
+AC_MSG_RESULT([$link_to_freetype])
+
+
 # Done.
 AC_OUTPUT(Makefile
 utils/Makefile
