Package:     fox1.6
Subject:     link to fontconfig
Author:      Torsten Landschoff <torsten at debian.org>
Forwarded:   no
Last-Update: 2012-05-13
---

diff -urN s04/configure.ac s05/configure.ac
--- s04/configure.ac	2012-05-13 10:28:01.687411560 +0200
+++ s05/configure.ac	2012-05-13 10:28:46.683411663 +0200
@@ -293,6 +293,18 @@
 AC_MSG_RESULT([$link_to_freetype])
 
 
+AC_MSG_CHECKING([if we should link libfontconfig])
+link_to_fontconfig="no"
+if test "x$with_xft" == "xyes"; then
+  FONTCONFIGLIBS=`pkg-config --libs fontconfig`
+  if test "x$FONTCONFIGLIBS" != "x"; then
+    X_BASE_LIBS="$X_BASE_LIBS $FONTCONFIGLIBS"
+    link_to_fontconfig="yes ($FONTCONFIGLIBS)"
+  fi
+fi
+AC_MSG_RESULT([$link_to_fontconfig])
+
+
 # Done.
 AC_OUTPUT(Makefile
 utils/Makefile
