diff -ru orig/libproxy-0.3.1/src/lib/pac.c libproxy-0.3.1/src/lib/pac.c
--- orig/libproxy-0.3.1/src/lib/pac.c	2009-09-29 21:52:50.000000000 +0200
+++ libproxy-0.3.1/src/lib/pac.c	2010-10-14 16:13:58.273700566 +0200
@@ -19,6 +19,7 @@
 
 #include <unistd.h>
 #include <string.h>
+#include <strings.h>
 #include <time.h>
 #include <sys/types.h>
 #include <sys/stat.h>
@@ -145,11 +146,11 @@
 		/* Check for correct mime type and content length */
 		while (strcmp(line, "\r")) {
 			/* Check for content type */
-			if (strstr(line, "Content-Type: ") == line && strstr(line, PAC_MIME_TYPE))
+			if (strncasecmp(line, "Content-Type:", strlen("Content-Type:")) == 0 && strstr(line, PAC_MIME_TYPE))
 				correct_mime_type = true;
 
 			/* Check for content length */
-			else if (strstr(line, "Content-Length: ") == line)
+			else if (strncasecmp(line, "Content-Length:", strlen("Content-Length:")) == 0)
 				content_length = atoi(line + strlen("Content-Length: "));
 
 			/* Get new line */
