Description: Fix build failure with [-Werror=format-security]
Author: Michael Biebl <biebl@debian.org>
Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=643433
Index: libproxy-0.3.1/src/bin/proxy.c
===================================================================
--- libproxy-0.3.1.orig/src/bin/proxy.c	2009-09-29 21:52:50.000000000 +0200
+++ libproxy-0.3.1/src/bin/proxy.c	2011-10-25 01:38:38.564562709 +0200
@@ -89,7 +89,7 @@
 {
 	for (int j = 0; proxies[j] ; j++)
 	{
-		printf(proxies[j]);
+		printf("%s", proxies[j]);
 		if (proxies[j+1])
 			printf(" ");
 		else
