From: Juan Cespedes <cespedes@debian.org>
Subject: Fix 'p' (previous image) key.

diff -urNad 04.xloadimage.tmp/xloadimage.c 04.xloadimage/xloadimage.c
--- 04.xloadimage.tmp/xloadimage.c	2003-04-02 19:28:25.000000000 +0100
+++ 04.xloadimage/xloadimage.c	2003-04-02 19:27:44.000000000 +0100
@@ -493,6 +493,8 @@
       if (!tmpset)
 	goto redisplay_in_window; /* ick */
       optset= tmpset;
+      freeImage(dispimage);
+      dispimage= NULL;
       goto get_another_image; /* ick */
     case '<':
       if ((opt = getOption(optset,ZOOM)) == NULL) {
