From: James Troup <james@nocrew.org>
Subject: Define SYSPATHFILE during build.

diff -urNad 07.xloadimage.tmp/Makefile.in 07.xloadimage/Makefile.in
--- 07.xloadimage.tmp/Makefile.in	2003-04-02 23:40:49.000000000 +0100
+++ 07.xloadimage/Makefile.in	2003-04-02 23:40:15.000000000 +0100
@@ -3,7 +3,7 @@
 #
 
 CC = @CC@
-DEFS = @DEFS@
+DEFS = @DEFS@ -DSYSPATHFILE=\"/etc/X11/Xloadimage\"
 CFLAGS = @CFLAGS@
 XLIB = @X_LIBS@ -lX11 @X_EXTRA_LIBS@
 LDFLAGS = @LDFLAGS@
