From: Austin Donnelly <and1000@debian.org>
Subject: Include ~images in path for system-wide configuration file.

diff -urNad 09.xloadimage.tmp/xloadimagerc 09.xloadimage/xloadimagerc
--- 09.xloadimage.tmp/xloadimagerc	1993-10-21 22:30:32.000000000 +0100
+++ 09.xloadimage/xloadimagerc	2003-04-02 23:44:16.000000000 +0100
@@ -1,6 +1,9 @@
 # Sample .xloadimagerc file
 
-path = /usr/local/images
+# Directories to search for images
+path = ~/images /usr/local/images
+
+# Default extensions to try tacking onto the end of a filename, in order
 extension = .niff             # NIFF image
             .jpg .jpeg        # JPEG image
             .gif              # CompuServe GIF image
