From: Jens Peter Secher <jpsecher@diku.dk>
Subject: Make -fork imply -quiet as documented.

diff -urNad 11.xloadimage.tmp/options.c 11.xloadimage/options.c
--- 11.xloadimage.tmp/options.c	2003-04-05 14:13:53.000000000 +0100
+++ 11.xloadimage/options.c	2003-04-05 14:13:30.000000000 +0100
@@ -551,6 +551,7 @@
 	      optionName(FORK));
       continue;
 #else
+      killOption(global_options, VERBOSE);
       global_opt= 1;
       break;
 #endif
