From: Austin Donnelly <and1000@debian.org>
Subject: Check libpng version and call right function

--- a/png.c	2011-06-08 10:28:55.000000000 +0900
+++ b/png.c	2011-06-08 10:32:20.000000000 +0900
@@ -75,7 +75,11 @@
 {
   debug(" #error ");
   output_warn( png_ptr, str);
+#if PNG_LIBPNG_VER_MAJOR >= 1 && PNG_LIBPNG_VER_MINOR >= 4
+  longjmp(png_jmpbuf((png_ptr)),1);
+#else
   longjmp(png_ptr->jmpbuf, 1);	/* return control to outer routine */
+#endif
 }
 
 
@@ -164,7 +168,11 @@
     png_destroy_read_struct(png_pp, info_pp, end_pp);
     return 0;
   }
+#if PNG_LIBPNG_VER_MAJOR >= 1 && PNG_LIBPNG_VER_MINOR >= 4
+  if (setjmp(png_jmpbuf(*png_pp))) {
+#else
   if (setjmp((*png_pp)->jmpbuf)) {
+#endif
     /* On error */
     png_destroy_read_struct(png_pp, info_pp, end_pp);
     return 0;
@@ -220,7 +228,12 @@
     zclose(zinput_file);
     return 0;
   }
+
+#if PNG_LIBPNG_VER_MAJOR >= 1 && PNG_LIBPNG_VER_MINOR >= 4
+  if (setjmp(png_jmpbuf(png_ptr))) {
+#else
   if (setjmp(png_ptr->jmpbuf)) {
+#endif
     /* On error */
     freeImage(image);
     png_destroy_read_struct(&png_ptr, &info_ptr, &end_info);

