From: Austin Donnelly <and1000@debian.org>
Subject: Check libpng version and call right function

--- a/png.c	2011-06-08 21:56:57.000000000 +0900
+++ b/png.c	2011-06-08 21:58:16.000000000 +0900
@@ -321,7 +321,11 @@
   case PNG_COLOR_TYPE_GRAY_ALPHA:
   case PNG_COLOR_TYPE_GRAY:
     if (bit_depth < 8)
+#if PNG_LIBPNG_VER_MAJOR >= 1 && PNG_LIBPNG_VER_MINOR >= 4
+      png_set_expand_gray_1_2_4_to_8(png_ptr);  /* 1 pixlel 1 byte */
+#else
       png_set_gray_1_2_4_to_8(png_ptr);  /* 1 pixlel 1 byte */
+#endif
     image = newRGBImage(width, height, 8);
     image->rgb.used = 256;
     for (i = 0; i < 256; i++) {
