From: Colin Watson <cjwatson@ubuntu.com>
Subject: Fix jpeg.c build if C_ARITH_CODING_SUPPORTED is defined.
Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=662619
Bug-Ubuntu: https://bugs.launchpad.net/bugs/935491

Index: b/jpeg.c
===================================================================
--- a/jpeg.c
+++ b/jpeg.c
@@ -321,8 +321,8 @@
       cinfo->arith_code = TRUE;
 #else
       fprintf(stderr, "jpegDump: sorry, arithmetic coding not supported\n");
-    }
 #endif
+    }
     else if (!strncmp("grayscale", name, strlen(name)) ||
 	     !strncmp("greyscale", name, strlen(name)) ||
 	     !strncmp("monochrome", name, strlen(name))) {
