From: Modestas Vainius <modax@debian.org>
Subject: generate a tagfile while building soprano API documentation
Forwarded: not-needed
Origin: vendor
Last-Update: 2012-05-24

--- a/Doxyfile.cmake
+++ b/Doxyfile.cmake
@@ -239,7 +239,7 @@ SKIP_FUNCTION_MACROS   = YES
 # Configuration::additions related to external references   
 #---------------------------------------------------------------------------
 TAGFILES               = ${soprano_SOURCE_DIR}/qt4.tag=${QTDOCS}
-GENERATE_TAGFILE       = 
+GENERATE_TAGFILE       = docs/soprano.tag
 ALLEXTERNALS           = NO
 EXTERNAL_GROUPS        = YES
 PERL_PATH              = /usr/bin/perl
