Description: fix wrong check. X11 isn't required.
Author: Fathi Boudra <fabo@debian.org>
Forwarded: not-needed

--- a/cmake/modules/FindQt4.cmake
+++ b/cmake/modules/FindQt4.cmake
@@ -1211,7 +1211,7 @@ QT4_ADD_DBUS_INTERFACES(<srcList> <files
   IF(UNIX)
     # on OS X X11 may not be required
     IF (Q_WS_X11)
-      FIND_PACKAGE(X11 REQUIRED)
+      FIND_PACKAGE(X11)
     ENDIF (Q_WS_X11)
     FIND_PACKAGE(Threads)
     SET(QT_QTCORE_LIBRARY ${QT_QTCORE_LIBRARY} ${CMAKE_THREAD_LIBS_INIT})
