---
 prog/sensord/sense.c |    9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

--- a/prog/sensord/sense.c
+++ b/prog/sensord/sense.c
@@ -239,9 +239,14 @@
 {
 	int ret = 0;
 
-	sensorLog(LOG_DEBUG, "sensor read started");
+	/*
+	 * Debian mod: kill the debug messages, since they spam syslog every
+	 * minute. -- JEH
+	 */
+
+	/* sensorLog(LOG_DEBUG, "sensor read started"); */
 	ret = doChips(DO_READ);
-	sensorLog(LOG_DEBUG, "sensor read finished");
+	/* sensorLog(LOG_DEBUG, "sensor read finished"); */
 
 	return ret;
 }
