---
 etc/sensors.conf.default |    4 ++++
 1 file changed, 4 insertions(+)

--- a/etc/sensors.conf.default
+++ b/etc/sensors.conf.default
@@ -11,6 +11,10 @@
 #
 # Such custom configuration files for specific mainboards can be found at
 # http://www.lm-sensors.org/wiki/Configurations
+#
+# It is recommended not to modify this file, but to drop your local
+# changes in /etc/sensors.d/. File with names that start with a dot
+# are ignored.
 
 chip "lm78-*" "lm79-*" "lm80-*" "lm96080-*"
 
