---
 prog/detect/sensors-detect |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/prog/detect/sensors-detect
+++ b/prog/detect/sensors-detect
@@ -603,7 +603,7 @@
 		i2c_detect => sub { adt7467_detect(@_, 2); },
 	}, {
 		name => "Analog Devices ADT7466",
-		driver => "to-be-written",
+		driver => "lm85",
 		i2c_addrs => [0x4c],
 		i2c_detect => sub { adt7467_detect(@_, 3); },
 	}, {
