---
 prog/detect/sensors-detect |    5 +++++
 1 file changed, 5 insertions(+)

--- a/prog/detect/sensors-detect
+++ b/prog/detect/sensors-detect
@@ -1861,6 +1861,11 @@
 		driver => "not-a-sensor",
 		devid => 0x83,
 	}, {
+		name => "SMSC SCH5524-NS Super IO",
+		# No datasheet
+		driver => "not-a-sensor",
+		devid => 0x8d,
+	}, {
 		name => "SMSC SCH5627 Super IO",
 		driver => "sch5627",
 		devid => 0xc6,
