---
 prog/pwm/pwmconfig |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/prog/pwm/pwmconfig
+++ b/prog/pwm/pwmconfig
@@ -471,8 +471,8 @@
 					exit 1
 				fi
 			else
-				fanactive="$j+${fanactive}" #not supported yet by fancontrol
-				fanactive_min="$S+${fanactive_min}"
+				fanactive="$(($j+${fanactive}))" #not supported yet by fancontrol
+				fanactive_min="$(($S+${fanactive_min}))"
 			fi
 			S=`cat $j`
 			if [ $S -lt $threshold ]
