From: Pino Toscano <pino@debian.org>
From: Modestas Vainius <modax@debian.org>
Subject: Qt doesn't expose the X11 headers to the world, so no need to make it required.
Forwarded: not-needed
Origin: vendor
Last-Update: 2012-02-19

---
 cmake/Qt4ConfigDependentSettings.cmake |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/cmake/Qt4ConfigDependentSettings.cmake
+++ b/cmake/Qt4ConfigDependentSettings.cmake
@@ -27,7 +27,7 @@
 IF(UNIX)
   # on OS X X11 may not be required
   IF (Q_WS_X11)
-    FIND_PACKAGE(X11 REQUIRED)
+    FIND_PACKAGE(X11)
   ENDIF (Q_WS_X11)
   FIND_PACKAGE(Threads)
   SET(QT_QTCORE_LIBRARY ${QT_QTCORE_LIBRARY} ${CMAKE_THREAD_LIBS_INIT})
