From: Sune Vuorela <debian@pusling.com>
Subject: Install headers where they were earlier
Forwarded: no

---
 includes/CMakeLists.txt |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/includes/CMakeLists.txt
+++ b/includes/CMakeLists.txt
@@ -91,7 +91,7 @@ install( FILES
   old/Phonon/VolumeFaderEffect
   old/Phonon/VolumeFaderInterface
   old/Phonon/VolumeSlider
-DESTINATION ${INCLUDE_INSTALL_DIR}/KDE/Phonon COMPONENT Devel)
+DESTINATION ${INCLUDE_INSTALL_DIR}/phonon/Phonon COMPONENT Devel)
 
 if (PHONON_BUILD_EXPERIMENTAL)
     install(FILES
