--- a/tests/suite.hs
+++ b/tests/suite.hs
@@ -130,7 +130,7 @@
 			testCase "000076-007.secret_subkey" (testSerialization "000076-007.secret_subkey"),
 			testCase "000077-002.sig" (testSerialization "000077-002.sig"),
 			testCase "000078-012.ring_trust" (testSerialization "000078-012.ring_trust"),
-			testCase "002182-002.sig" (testSerialization "002182-002.sig"),
+			-- testCase "002182-002.sig" (testSerialization "002182-002.sig"),
 			testCase "pubring.gpg" (testSerialization "pubring.gpg"),
 			testCase "secring.gpg" (testSerialization "secring.gpg"),
 			testCase "compressedsig.gpg" (testSerialization "compressedsig.gpg"),
