From 40ef398290064e0c52be9e0d2127901b0f91f20d Mon Sep 17 00:00:00 2001
From: Ritesh Raj Sarraf <rrs@researchut.com>
Date: Tue, 2 Feb 2010 11:43:29 +0530
Subject: [PATCH 32/43] make iscsistart a dynamic binary

---
 usr/Makefile |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

Index: open-iscsi/usr/Makefile
===================================================================
--- open-iscsi.orig/usr/Makefile	2012-04-07 19:36:03.299049291 +0530
+++ open-iscsi/usr/Makefile	2012-04-07 19:39:32.276085546 +0530
@@ -61,7 +61,7 @@
 
 iscsistart: $(ISCSI_LIB_SRCS) $(INITIATOR_SRCS) $(FW_BOOT_SRCS) \
 		iscsistart.o statics.o
-	$(CC) $(CFLAGS) -static $^ -o $@
+	$(CC) $(CFLAGS) $^ -o $@
 clean:
 	rm -f *.o $(PROGRAMS) .depend $(LIBSYS)
 
