diff -ur postgresql-9.1-9.1~beta1/src/backend/main/main.c postgresql-9.1-9.1~beta1.new/src/backend/main/main.c
--- postgresql-9.1-9.1~beta1/src/backend/main/main.c	2010-01-02 17:57:45.000000000 +0100
+++ postgresql-9.1-9.1~beta1.new/src/backend/main/main.c	2010-05-01 14:41:42.997229019 +0200
@@ -328,7 +328,7 @@
 check_root(const char *progname)
 {
 #ifndef WIN32
-	if (geteuid() == 0)
+	if (0 && geteuid() == 0)
 	{
 		write_stderr("\"root\" execution of the PostgreSQL server is not permitted.\n"
 					 "The server must be started under an unprivileged user ID to prevent\n"
diff -ur postgresql-9.1-9.1~beta1/src/bin/initdb/initdb.c postgresql-9.1-9.1~beta1.new/src/bin/initdb/initdb.c
--- postgresql-9.1-9.1~beta1/src/bin/initdb/initdb.c	2010-02-26 03:01:15.000000000 +0100
+++ postgresql-9.1-9.1~beta1.new/src/bin/initdb/initdb.c	2010-05-01 14:41:42.997229019 +0200
@@ -648,7 +648,7 @@
 
 	struct passwd *pw;
 
-	if (geteuid() == 0)			/* 0 is root's uid */
+	if (0 && geteuid() == 0)		/* 0 is root's uid */
 	{
 		fprintf(stderr,
 				_("%s: cannot be run as root\n"
diff -ur postgresql-9.1-9.1~beta1/src/bin/pg_ctl/pg_ctl.c postgresql-9.1-9.1~beta1.new/src/bin/pg_ctl/pg_ctl.c
--- postgresql-9.1-9.1~beta1/src/bin/pg_ctl/pg_ctl.c	2010-04-07 05:48:51.000000000 +0200
+++ postgresql-9.1-9.1~beta1.new/src/bin/pg_ctl/pg_ctl.c	2010-05-01 14:41:42.997229019 +0200
@@ -1750,7 +1750,7 @@
 	 * Disallow running as root, to forestall any possible security holes.
 	 */
 #ifndef WIN32
-	if (geteuid() == 0)
+	if (0 && geteuid() == 0)
 	{
 		write_stderr(_("%s: cannot be run as root\n"
 					   "Please log in (using, e.g., \"su\") as the "
