--- a/modules/ntrack-libnl.c	2012-01-14 12:27:44 +0000
+++ b/modules/ntrack-libnl.c	2013-05-23 04:24:57 +0000
@@ -541,7 +541,7 @@
 		op_default_route_disappeared ((struct _ntrack_monitor_arch*) self, nl_info);
 	} else if (nl_info->topmost_route && topmost_route) {
 		int diff_bits;
-		if ((diff_bits = (route_obj_ops.oo_id_attrs | ROUTE_ATTR_OIF
+		if ((diff_bits = (/*route_obj_ops.oo_id_attrs |*/ ROUTE_ATTR_OIF
 		                  | ROUTE_ATTR_GATEWAY) &
 		                  nl_object_diff (OBJ_CAST (nl_info->topmost_route),
 		                                  OBJ_CAST (topmost_route)))) {

