Description: autotools_dev results
Author: Mathieu Malaterre <malat@debian.org>

--- docbook-utils-0.6.14.orig/Makefile.in
+++ docbook-utils-0.6.14/Makefile.in
@@ -128,7 +128,7 @@ sharedstatedir = @sharedstatedir@
 sysconfdir = @sysconfdir@
 target_alias = @target_alias@
 SUBDIRS = backends bin doc frontends helpers
-sgmldudir = $(prefix)/share/sgml/docbook/utils-@VERSION@
+sgmldudir = $(prefix)/share/docbook-utils
 sgmldu_DATA = docbook-utils.dsl
 EXTRA_DIST = AUTHORS README COPYING INSTALL NEWS ChangeLog TODO \
 	$(sgmldu_DATA) docbook-utils.spec
--- docbook-utils-0.6.14.orig/Makefile.am
+++ docbook-utils-0.6.14/Makefile.am
@@ -1,6 +1,6 @@
 SUBDIRS = backends bin doc frontends helpers
 
-sgmldudir = $(prefix)/share/sgml/docbook/utils-@VERSION@
+sgmldudir = $(prefix)/share/docbook-utils
 sgmldu_DATA = docbook-utils.dsl
 
 EXTRA_DIST = AUTHORS README COPYING INSTALL NEWS ChangeLog TODO \
--- docbook-utils-0.6.14.orig/backends/Makefile.in
+++ docbook-utils-0.6.14/backends/Makefile.in
@@ -104,7 +104,7 @@ sbindir = @sbindir@
 sharedstatedir = @sharedstatedir@
 sysconfdir = @sysconfdir@
 target_alias = @target_alias@
-sgmldudir = $(prefix)/share/sgml/docbook/utils-@VERSION@
+sgmldudir = $(prefix)/share/docbook-utils
 sgmldubedir = $(sgmldudir)/backends
 sgmldube_DATA = dvi html man pdf ps rtf tex texi txt
 EXTRA_DIST = dvi html man.in pdf ps rtf tex texi.in txt
--- docbook-utils-0.6.14.orig/backends/Makefile.am
+++ docbook-utils-0.6.14/backends/Makefile.am
@@ -1,4 +1,4 @@
-sgmldudir = $(prefix)/share/sgml/docbook/utils-@VERSION@
+sgmldudir = $(prefix)/share/docbook-utils
 sgmldubedir = $(sgmldudir)/backends
 sgmldube_DATA = dvi html man pdf ps rtf tex texi txt
 
--- docbook-utils-0.6.14.orig/backends/man.in
+++ docbook-utils-0.6.14/backends/man.in
@@ -3,7 +3,7 @@
 # This program is under GPL license. See LICENSE file for details.
 
 # Convert to *roff
-HELPER=$SGML_BASE_DIR/docbook/utils-@VERSION@/helpers/docbook2man-spec.pl
+HELPER=docbook2man-spec.pl
 TMPDIR=`mktemp -d /tmp/man.XXXXXX` || \
  { echo >&2 "man backend: could not create secure temporary directory"; exit 1;}
 trap 'rm -rf "${TMPDIR}"' EXIT
--- docbook-utils-0.6.14.orig/backends/texi.in
+++ docbook-utils-0.6.14/backends/texi.in
@@ -3,7 +3,7 @@
 # This program is under GPL license. See LICENSE file for details.
 
 # Convert to texinfo
-nsgmls $SGML_FILE | sgmlspl $SGML_BASE_DIR/docbook/utils-@VERSION@/helpers/docbook2texi-spec.pl >$SGML_FILE_NAME.texi
+nsgmls $SGML_FILE | sgmlspl docbook2texi-spec.pl >$SGML_FILE_NAME.texi
 if [ $? -ne 0 ]
 then exit 1
 fi
--- docbook-utils-0.6.14.orig/bin/jw.in
+++ docbook-utils-0.6.14/bin/jw.in
@@ -87,10 +87,10 @@ then
 fi
 
 # Set frontend to use
-SGML_FRONTEND="$SGML_BASE_DIR/docbook/utils-@VERSION@/frontends/docbook"
+SGML_FRONTEND="@prefix@/share/docbook-utils/frontends/docbook"
 
 # Set backend to use
-SGML_BACKEND="$SGML_BASE_DIR/docbook/utils-@VERSION@/backends/html"
+SGML_BACKEND="@prefix@/share/docbook-utils/backends/html"
 
 # Set main stylesheet
 SGML_STYLESHEET="default"
@@ -120,14 +120,14 @@ while [ $# -gt 0 ]
 do case $1 in
   -f|--frontend) case "$2" in
 		  /*) SGML_FRONTEND="$2" ;;
-		   *) d=$SGML_BASE_DIR/docbook/utils-@VERSION@/frontends
+		   *) d=@prefix@/share/docbook-utils/frontends
 		      SGML_FRONTEND="$d/$2" ;;
 		esac
 		shift 2
 		;;
   -b|--backend)	case "$2" in
 		  /*) SGML_BACKEND="$2" ;;
-		   *) d=$SGML_BASE_DIR/docbook/utils-@VERSION@/backends
+		   *) d=@prefix@/share/docbook-utils/backends
 		      SGML_BACKEND="$d/$2" ;;
 		esac
 		shift 2
@@ -186,8 +186,8 @@ do case $1 in
 		shift 2
 		;;
   -w|--warning)	case $2 in
-		  list) echo -e "List of allowed warning types (multiple -w options are allowed):\n"
-			echo -e $SGML_WARNINGS_LIST
+		  list) printf "List of allowed warning types (multiple -w options are allowed):\n\n"
+			printf "$SGML_WARNINGS_LIST\n"
 			exit 0
 			;;
 		  xml|mixed|sgmldecl|should|default|duplicate|undefined|unclosed|empty|net|min-tag|unused-map|unused-param|notation-sysid|all|no-xml|no-mixed|no-sgmldecl|no-should|no-default|no-duplicate|no-undefined|no-unclosed|no-empty|no-net|no-min-tag|no-unused-map|no-unused-param|no-notation-sysid) \
@@ -195,14 +195,14 @@ do case $1 in
 			shift 2
 			;;
 		  *)	echo "Unknown warning type \"$2\". Allowed warning types are:" >&2
-		        echo -e $SGML_WARNINGS_LIST >&2
+		        printf "$SGML_WARNINGS_LIST\n" >&2
 			exit 1
 			;;
 		esac
 		;;
   -e|--error)	case $2 in
-		  list) echo -e "List of allowed error types (multiple -e options are allowed):\n"
-			echo -e $SGML_ERRORS_LIST
+		  list) printf "List of allowed error types (multiple -e options are allowed):\n\n"
+			printf "$SGML_ERRORS_LIST\n"
 			exit 0
 			;;
 		  no-idref|no-significant|no-valid) \
@@ -210,19 +210,19 @@ do case $1 in
 			shift 2
 			;;
 		  *)	echo "Unknown error type \"$2\". Allowed error types are:" >&2
-		        echo -e $SGML_ERRORS_LIST >&2
+		        printf "$SGML_ERRORS_LIST\n" >&2
 			exit 1
 			;;
 		esac
 		;;
-  -h|--help)	echo -e "`basename $0` - Jade Wrapper\n"
-		echo -e "$SGML_HELP_MESSAGE"
+  -h|--help)	printf "`basename $0` - Jade Wrapper\n\n"
+		printf "$SGML_HELP_MESSAGE\n"
 		exit 0
 		;;
-  -v|--version)	echo -e "$SGML_VERSION_MESSAGE"
+  -v|--version)	printf "$SGML_VERSION_MESSAGE\n"
 		exit 0
 		;;
-  -*)		echo -e "$SGML_HELP_MESSAGE" >&2
+  -*)		printf "$SGML_HELP_MESSAGE\n" >&2
 		exit 1
 		;;
   *)		if [ -z "$SGML_FILE" ]
@@ -233,7 +233,7 @@ do case $1 in
 		  esac
 		  shift 1
 		else
-		  echo -e "$SGML_HELP_MESSAGE" >&2
+		  printf "$SGML_HELP_MESSAGE\n" >&2
 		  exit 1
 		fi
 		;;
@@ -243,7 +243,7 @@ done
 # Check that we have a frontend
 if [ -z "$SGML_FRONTEND" ]
 then
-  echo -e $SGML_HELP_MESSAGE >&2
+  printf "$SGML_HELP_MESSAGE\n" >&2
   exit 1
 fi
 if [ ! -s $SGML_FRONTEND ]
@@ -255,7 +255,7 @@ fi
 # Check that we have a backend
 if [ -z "$SGML_BACKEND" ]
 then
-  echo -e $SGML_HELP_MESSAGE >&2
+  printf "$SGML_HELP_MESSAGE\n" >&2
   exit 1
 fi
 if [ ! -s $SGML_BACKEND ]
@@ -267,7 +267,7 @@ fi
 # Check that we have a file to process
 if [ -z "$SGML_FILE" ]
 then
-  echo -e $SGML_HELP_MESSAGE >&2
+  printf "$SGML_HELP_MESSAGE\n" >&2
   exit 1
 fi
 if [ ! -s $SGML_FILE ]
@@ -290,7 +290,7 @@ fi
 
 # Use SGML declaration for XML if needed
 if [ "$SGML_DECL" = "" -a "$SGML_XML" = "xml" ]
-then SGML_DECL=${SGML_BASE_DIR}/xml.dcl
+then SGML_DECL=${SGML_BASE_DIR}/declaration/xml.dcl
 fi
 
 # Try to find the SGML normalizer
@@ -402,7 +402,7 @@ fi
 # Create output directory if not available
 if [ -z "$SGML_OUTPUT_DIRECTORY" ]
 then
-  echo -e $SGML_HELP_MESSAGE >&2
+  printf "$SGML_HELP_MESSAGE\n" >&2
   exit 1
 fi
 if [ ! -d $SGML_OUTPUT_DIRECTORY ]
--- docbook-utils-0.6.14.orig/doc/refentry/jw.sgml
+++ docbook-utils-0.6.14/doc/refentry/jw.sgml
@@ -230,7 +230,7 @@
     <para>Here we have chosen to generate HTML output. In fact we can
     use any of the backends stored in the <filename>backends/</filename>
     subdirectory of the DocBook-utils distribution directory (usually
-    <filename>/usr/share/sgml/docbook/utils-&version;</filename>).
+    <filename>/usr/share/docbook-utils</filename>).
     Similarly, you can use any frontend defined in the
     <filename>frontends/</filename> subdirectory to convert from another
     input format.</para>
@@ -592,21 +592,21 @@
       </varlistentry>
 
       <varlistentry>
-	<term><filename>/usr/share/sgml/docbook/utils-&version;/backends</filename></term>
+	<term><filename>/usr/share/docbook-utils/backends</filename></term>
 	<listitem>
           <para>The various backends</para>
 	</listitem>
       </varlistentry>
 
       <varlistentry>
-	<term><filename>/usr/share/sgml/docbook/utils-&version;/frontends</filename></term>
+	<term><filename>/usr/share/docbook-utils/frontends</filename></term>
 	<listitem>
           <para>The various frontends</para>
 	</listitem>
       </varlistentry>
 
       <varlistentry>
-	<term><filename>/usr/share/sgml/docbook/utils-&version;/helpers</filename></term>
+	<term><filename>/usr/share/perl5/sgmlspl-specs</filename></term>
 	<listitem>
           <para>The various helper scripts like docbook2man or docbook2texi</para>
 	</listitem>
--- docbook-utils-0.6.14.orig/doc/man/jw.1
+++ docbook-utils-0.6.14/doc/man/jw.1
@@ -128,7 +128,7 @@ be in any other directory than the curre
 Here we have chosen to generate HTML output. In fact we can
 use any of the backends stored in the \fIbackends/\fR
 subdirectory of the DocBook-utils distribution directory (usually
-\fI/usr/share/sgml/docbook/utils-0.6.14\fR).
+\fI/usr/share/docbook-utils\fR).
 Similarly, you can use any frontend defined in the
 \fIfrontends/\fR subdirectory to convert from another
 input format.
@@ -355,13 +355,13 @@ Centralized SGML open catalog. This file
 vary if another frontend than \fIdocbook\fR
 is used.
 .TP
-\fB\fI/usr/share/sgml/docbook/utils-0.6.14/backends\fB\fR
+\fB\fI/usr/share/docbook-utils/backends\fB\fR
 The various backends
 .TP
-\fB\fI/usr/share/sgml/docbook/utils-0.6.14/frontends\fB\fR
+\fB\fI/usr/share/docbook-utils/frontends\fB\fR
 The various frontends
 .TP
-\fB\fI/usr/share/sgml/docbook/utils-0.6.14/helpers\fB\fR
+\fB\fI/usr/share/perl5/sgmlspl-specs\fB\fR
 The various helper scripts like docbook2man or docbook2texi
 .SH "AUTHORS"
 .PP
--- docbook-utils-0.6.14.orig/doc/HTML/Makefile.in
+++ docbook-utils-0.6.14/doc/HTML/Makefile.in
@@ -103,7 +103,7 @@ sbindir = @sbindir@
 sharedstatedir = @sharedstatedir@
 sysconfdir = @sysconfdir@
 target_alias = @target_alias@
-htmldir = $(docdir)/html/docbook-utils-@VERSION@
+htmldir = $(docdir)/html/
 html_DATA = api.html \
 	backend-spec.html \
 	docbook2man.html \
--- docbook-utils-0.6.14.orig/frontends/Makefile.in
+++ docbook-utils-0.6.14/frontends/Makefile.in
@@ -104,7 +104,7 @@ sbindir = @sbindir@
 sharedstatedir = @sharedstatedir@
 sysconfdir = @sysconfdir@
 target_alias = @target_alias@
-sgmldudir = $(prefix)/share/sgml/docbook/utils-@VERSION@
+sgmldudir = $(prefix)/share/docbook-utils
 sgmldufedir = $(sgmldudir)/frontends
 sgmldufe_DATA = docbook
 all: all-am
--- docbook-utils-0.6.14.orig/frontends/Makefile.am
+++ docbook-utils-0.6.14/frontends/Makefile.am
@@ -1,3 +1,3 @@
-sgmldudir = $(prefix)/share/sgml/docbook/utils-@VERSION@
+sgmldudir = $(prefix)/share/docbook-utils
 sgmldufedir = $(sgmldudir)/frontends
 sgmldufe_DATA = docbook
--- docbook-utils-0.6.14.orig/frontends/docbook.in
+++ docbook-utils-0.6.14/frontends/docbook.in
@@ -27,10 +27,7 @@ case "$1" in
 	done
 	if [ -z "$SGML_CATALOG" ]
 	then
-	  if [ "${SGML_XML}" != "sgml" ]
-	  then SGML_CATALOG=${SGML_CATALOGS_DIR}/${SGML_XML}-docbook.cat
-	  else SGML_CATALOG=${SGML_CATALOGS_DIR}/catalog
-	  fi
+	  SGML_CATALOG=${SGML_CATALOGS_DIR}/catalog
 	fi
 	echo "$SGML_CATALOG"
 	;;
@@ -38,7 +35,7 @@ case "$1" in
   style-sheet)
 	case $SGML_STYLESHEET in
 	  default)
-	    echo "${SGML_BASE_DIR}/docbook/utils-@VERSION@/docbook-utils.dsl#${SGML_TYPE}"
+	    echo "/usr/share/docbook-utils/docbook-utils.dsl#${SGML_TYPE}"
 	    ;;
 	  none)
 	    find $SGML_BASE_DIR -name docbook.dsl | grep "$SGML_TYPE/docbook.dsl" | awk '{print $1}'
--- docbook-utils-0.6.14.orig/helpers/docbook2man-spec.pl
+++ docbook-utils-0.6.14/helpers/docbook2man-spec.pl
@@ -1199,8 +1199,8 @@ sgml('cdata', sub
 	$_[0] =~ s/\\/\\\\/g;
 
 	# Escape dots and single quotes in column 1
-	$_[0] =~ s/^\./\\\&\./;
-	$_[0] =~ s/^\'/\\\&\'/;
+	$_[0] =~ s/^[ \t]*\./\\\&\./;
+	$_[0] =~ s/^[ \t]*\'/\\\&\'/;
 
 	# In non-'pre'-type elements:
 	if(!$nocollapse_whitespace) {
--- docbook-utils-0.6.14.orig/helpers/Makefile.in
+++ docbook-utils-0.6.14/helpers/Makefile.in
@@ -103,8 +103,8 @@ sbindir = @sbindir@
 sharedstatedir = @sharedstatedir@
 sysconfdir = @sysconfdir@
 target_alias = @target_alias@
-sgmldudir = $(prefix)/share/sgml/docbook/utils-@VERSION@
-sgmlduhedir = $(sgmldudir)/helpers
+sgmldudir = $(prefix)/share
+sgmlduhedir = $(sgmldudir)/perl5/sgmlspl-specs
 sgmlduhe_DATA = docbook2man-spec.pl docbook2texi-spec.pl
 EXTRA_DIST = $(sgmlduhe_DATA)
 all: all-am
--- docbook-utils-0.6.14.orig/helpers/Makefile.am
+++ docbook-utils-0.6.14/helpers/Makefile.am
@@ -1,5 +1,5 @@
-sgmldudir = $(prefix)/share/sgml/docbook/utils-@VERSION@
-sgmlduhedir = $(sgmldudir)/helpers
+sgmldudir = $(prefix)/share
+sgmlduhedir = $(sgmldudir)/perl5/sgmlspl-specs
 sgmlduhe_DATA = docbook2man-spec.pl docbook2texi-spec.pl
 
 EXTRA_DIST = $(sgmlduhe_DATA)
--- docbook-utils-0.6.14.orig/helpers/docbook2texi-spec.pl
+++ docbook-utils-0.6.14/helpers/docbook2texi-spec.pl
@@ -774,7 +774,7 @@ texi_sgml('<BEGINPAGE>', "\n\@page\n");
 ## Print the URL
 sgml('<ULINK>', sub {
 	if($skip_inline++) { return; }	# hopefully doesn't happen
-	output '@uref{', output $_[0]->attribute('URL'), ', '
+	output '@uref{', $_[0]->attribute('URL')->value, ', '
 });
 sgml('</ULINK>', sub {
 	output '}' unless --$skip_inline;
@@ -807,6 +807,7 @@ sgml('<XREF>', sub {
 sgml('|[lt    ]|', "<");
 sgml('|[gt    ]|', ">");
 sgml('|[amp   ]|', "&");
+sgml('|[ndash ]|', "-");
 
 #
 # Default handlers (uncomment these if needed).  Right now, these are set
