--- blender-2.65a/source/blender/blenkernel/intern/writeffmpeg.c	2012-11-21 11:57:35.000000000 +0000
+++ blend/source/blender/blenkernel/intern/writeffmpeg.c	2012-12-30 21:52:00.781349450 +0000
@@ -373,7 +373,7 @@
 {
 	char name[128];
 	char *param;
-	const AVOption *rv = NULL;
+	int res = -1;
 
 	PRINT("FFMPEG expert option: %s: ", prop->name);
 
@@ -388,30 +388,30 @@
 	switch (prop->type) {
 		case IDP_STRING:
 			PRINT("%s.\n", IDP_String(prop));
-			av_set_string3(c, prop->name, IDP_String(prop), 1, &rv);
+			res = av_opt_set(c, prop->name, IDP_String(prop), 0);
 			break;
 		case IDP_FLOAT:
 			PRINT("%g.\n", IDP_Float(prop));
-			rv = av_set_double(c, prop->name, IDP_Float(prop));
+			res = av_opt_set_double(c, prop->name, IDP_Float(prop), 0);
 			break;
 		case IDP_INT:
 			PRINT("%d.\n", IDP_Int(prop));
 
 			if (param) {
 				if (IDP_Int(prop)) {
-					av_set_string3(c, name, param, 1, &rv);
+					av_opt_set(c, name, param, 0);
 				}
 				else {
 					return;
 				}
 			}
 			else {
-				rv = av_set_int(c, prop->name, IDP_Int(prop));
+				res = av_opt_set_int(c, prop->name, IDP_Int(prop), 0);
 			}
 			break;
 	}
 
-	if (!rv) {
+	if (!res) {
 		PRINT("ffmpeg-option not supported: %s! Skipping.\n", prop->name);
 	}
 }
@@ -464,7 +464,7 @@
 
 	error[0] = '\0';
 
-	st = av_new_stream(of, 0);
+	st = avformat_new_stream(of, 0);
 	if (!st) return NULL;
 
 	/* Set up the codec context */
@@ -539,7 +539,7 @@
 			c->pix_fmt = PIX_FMT_RGB32;
 		}
 		else {
-			c->pix_fmt = PIX_FMT_BGR0;
+			c->pix_fmt = PIX_FMT_BGRA;
 		}
 #else
 		c->pix_fmt = PIX_FMT_RGB32;
@@ -575,7 +575,7 @@
 
 	set_ffmpeg_properties(rd, c, "video");
 	
-	if (avcodec_open(c, codec) < 0) {
+	if (avcodec_open2(c, codec, NULL) < 0) {
 		BLI_strncpy(error, IMB_ffmpeg_last_error(), error_size);
 		return NULL;
 	}
@@ -612,7 +612,7 @@
 
 	error[0] = '\0';
 
-	st = av_new_stream(of, 1);
+	st = avformat_new_stream(of, NULL);
 	if (!st) return NULL;
 
 	c = st->codec;
@@ -635,7 +635,7 @@
 
 	set_ffmpeg_properties(rd, c, "audio");
 
-	if (avcodec_open(c, codec) < 0) {
+	if (avcodec_open2(c, codec, NULL) < 0) {
 		//XXX error("Couldn't initialize audio codec");
 		BLI_strncpy(error, IMB_ffmpeg_last_error(), error_size);
 		return NULL;
@@ -1144,7 +1144,7 @@
 	
 	val.i = 0;
 
-	avcodec_get_context_defaults(&c);
+	avcodec_get_context_defaults3(&c, NULL);
 
 	o = c.av_class->option + opt_index;
 	parent = c.av_class->option + parent_index;
@@ -1175,23 +1175,23 @@
 	}
 
 	switch (o->type) {
-		case FF_OPT_TYPE_INT:
-		case FF_OPT_TYPE_INT64:
+		case AV_OPT_TYPE_INT:
+		case AV_OPT_TYPE_INT64:
 			val.i = FFMPEG_DEF_OPT_VAL_INT(o);
 			idp_type = IDP_INT;
 			break;
-		case FF_OPT_TYPE_DOUBLE:
-		case FF_OPT_TYPE_FLOAT:
+		case AV_OPT_TYPE_DOUBLE:
+		case AV_OPT_TYPE_FLOAT:
 			val.f = FFMPEG_DEF_OPT_VAL_DOUBLE(o);
 			idp_type = IDP_FLOAT;
 			break;
-		case FF_OPT_TYPE_STRING:
+		case AV_OPT_TYPE_STRING:
 			val.string.str = (char *)"                                                                               ";
 			val.string.len = 80;
 /*		val.str = (char *)"                                                                               ";*/
 			idp_type = IDP_STRING;
 			break;
-		case FF_OPT_TYPE_CONST:
+		case AV_OPT_TYPE_CONST:
 			val.i = 1;
 			idp_type = IDP_INT;
 			break;
@@ -1231,7 +1231,7 @@
 	char *param;
 	IDProperty *prop = NULL;
 	
-	avcodec_get_context_defaults(&c);
+	avcodec_get_context_defaults3(&c, NULL);
 
 	strncpy(name_, str, sizeof(name_));
 
@@ -1252,10 +1252,10 @@
 	if (!o) {
 		return 0;
 	}
-	if (param && o->type == FF_OPT_TYPE_CONST) {
+	if (param && o->type == AV_OPT_TYPE_CONST) {
 		return 0;
 	}
-	if (param && o->type != FF_OPT_TYPE_CONST && o->unit) {
+	if (param && o->type != AV_OPT_TYPE_CONST && o->unit) {
 		p = my_av_find_opt(&c, param, o->unit, 0, 0);
 		if (p) {
 			prop = BKE_ffmpeg_property_add(rd, (char *) type, p - c.av_class->option, o - c.av_class->option);
--- blend/source/blender/imbuf/intern/anim_movie.c	2012-12-06 21:08:24.000000000 +0000
+++ blender_2.65a/source/blender/imbuf/intern/anim_movie.c	2012-12-30 22:01:31.344672494 +0000
@@ -486,7 +486,7 @@
 		return -1;
 	}
 
-	if (av_find_stream_info(pFormatCtx) < 0) {
+	if (avformat_find_stream_info(pFormatCtx, NULL) < 0) {
 		av_close_input_file(pFormatCtx);
 		return -1;
 	}
@@ -523,7 +523,7 @@
 
 	pCodecCtx->workaround_bugs = 1;
 
-	if (avcodec_open(pCodecCtx, pCodec) < 0) {
+	if (avcodec_open2(pCodecCtx, pCodec, NULL) < 0) {
 		av_close_input_file(pFormatCtx);
 		return -1;
 	}
--- blender_2.65/source/blender/imbuf/intern/indexer.c	2012-12-10 19:25:19.000000000 +0000
+++ b/source/blender/imbuf/intern/indexer.c	2012-12-31 19:18:24.799957539 +0000
@@ -496,7 +496,7 @@
 
 	fprintf(stderr, "Starting work on proxy: %s\n", rv->of->filename);
 
-	rv->st = av_new_stream(rv->of, 0);
+	rv->st = avformat_new_stream(rv->of, 0);
 	rv->c = rv->st->codec;
 	rv->c->codec_type = AVMEDIA_TYPE_VIDEO;
 	rv->c->codec_id = CODEC_ID_MJPEG;
@@ -531,8 +531,8 @@
 	/* there's no  way to set JPEG quality in the same way as in AVI JPEG and image sequence,
 	 * but this seems to be giving expected quality result */
 	ffmpeg_quality = (int)(1.0f + 30.0f * (1.0f - (float)quality / 100.0f) + 0.5f);
-	av_set_int(rv->c, "qmin", ffmpeg_quality);
-	av_set_int(rv->c, "qmax", ffmpeg_quality);
+	av_opt_set_int(rv->c, "qmin", ffmpeg_quality, 0);
+	av_opt_set_int(rv->c, "qmax", ffmpeg_quality, 0);
 
 	if (rv->of->flags & AVFMT_GLOBALHEADER) {
 		rv->c->flags |= CODEC_FLAG_GLOBAL_HEADER;
@@ -545,7 +545,7 @@
 		return 0;
 	}
 
-	avcodec_open(rv->c, rv->codec);
+	avcodec_open2(rv->c, rv->codec, NULL);
 
 	rv->video_buffersize = 2000000;
 	rv->video_buffer = (uint8_t *)MEM_mallocN(
@@ -758,7 +758,7 @@
 		return NULL;
 	}
 
-	if (av_find_stream_info(context->iFormatCtx) < 0) {
+	if (avformat_find_stream_info(context->iFormatCtx, NULL) < 0) {
 		av_close_input_file(context->iFormatCtx);
 		MEM_freeN(context);
 		return NULL;
@@ -797,7 +797,7 @@
 
 	context->iCodecCtx->workaround_bugs = 1;
 
-	if (avcodec_open(context->iCodecCtx, context->iCodec) < 0) {
+	if (avcodec_open2(context->iCodecCtx, context->iCodec, NULL) < 0) {
 		av_close_input_file(context->iFormatCtx);
 		MEM_freeN(context);
 		return NULL;
--- blender_2.65/source/blender/imbuf/intern/util.c	2012-12-10 19:25:19.000000000 +0000
+++ b/source/blender/imbuf/intern/util.c	2012-12-31 19:36:45.946604348 +0000
@@ -306,7 +306,7 @@
 		return 0;
 	}
 
-	if (av_find_stream_info(pFormatCtx) < 0) {
+	if (avformat_find_stream_info(pFormatCtx, NULL) < 0) {
 		if (UTIL_DEBUG) fprintf(stderr, "isffmpeg: av_find_stream_info failed\n");
 		av_close_input_file(pFormatCtx);
 		return 0;
@@ -340,7 +340,7 @@
 		return 0;
 	}
 
-	if (avcodec_open(pCodecCtx, pCodec) < 0) {
+	if (avcodec_open2(pCodecCtx, pCodec, NULL) < 0) {
 		av_close_input_file(pFormatCtx);
 		return 0;
 	}
