Author: Pino Toscano <pino@debian.org>
Subject: cmake options for GNU/Hurd
Bug: http://bugs.mysql.com/bug.php?id=64685
Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=651002
Last-Update: 2012-03-18
Reviewed-by: Nicholas Bamber <nicholas@periapt.co.uk>
Index: mysql-5.5.29/cmake/os/GNU.cmake
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ mysql-5.5.29/cmake/os/GNU.cmake	2013-01-02 15:38:08.271735374 -0800
@@ -0,0 +1,20 @@
+# This file includes GNU/Hurd specific options and quirks, related to system checks
+
+INCLUDE(CheckSymbolExists)
+
+SET(_GNU_SOURCE 1)
+
+# Fix CMake (< 2.8) flags. -rdynamic exports too many symbols.
+FOREACH(LANG C CXX)
+  STRING(REPLACE "-rdynamic" ""
+  CMAKE_SHARED_LIBRARY_LINK_${LANG}_FLAGS
+  "${CMAKE_SHARED_LIBRARY_LINK_${LANG}_FLAGS}"
+  )
+ENDFOREACH()
+
+# Ensure we have clean build for shared libraries
+# without unresolved symbols
+SET(LINK_FLAG_NO_UNDEFINED "-Wl,--no-undefined")
+
+# 64 bit file offset support flag
+SET(_FILE_OFFSET_BITS 64)
Index: mysql-5.5.29/mysql-test/lib/My/Platform.pm
===================================================================
--- mysql-5.5.29.orig/mysql-test/lib/My/Platform.pm	2013-01-02 15:36:15.398353757 -0800
+++ mysql-5.5.29/mysql-test/lib/My/Platform.pm	2013-01-02 15:38:08.271735374 -0800
@@ -110,6 +110,9 @@
   # This may not be true, but we can't test for it on AIX due to Perl bug
   # See Bug #45771
   return 0 if ($^O eq 'aix');
+  # Similarly the path length is hidden.
+  # See Debian bug #651002
+  return 0 if ($^O eq 'gnu');
 
   require IO::Socket::UNIX;
 
