Author: Clint Byrum <clint@ubuntu.com>
Subject: Convenient place to disable tests that are currently failing
Forwarded: not-needed
Last-Update: 2012-03-07
Reviewed-by: Nicholas Bamber <nicholas@periapt.co.uk>
Index: mysql-5.5.29/mysql-test/t/disabled.def
===================================================================
--- mysql-5.5.29.orig/mysql-test/t/disabled.def	2013-01-02 15:36:15.294359855 -0800
+++ mysql-5.5.29/mysql-test/t/disabled.def	2013-01-02 15:38:11.455548678 -0800
@@ -15,3 +15,8 @@
 archive-big              : Bug#11817185 2011-03-10 Anitha Disabled since this leads to timeout on Solaris Sparc
 log_tables-big           : Bug#11756699 2010-11-15 mattiasj report already exists
 mysql_embedded           : Bug#12561297 2011-05-14 Anitha Dependent on PB2 changes - eventum#41836
+partition_rename_longfilename : Fails when building with sbuild and schroots
+file_contents            : Fails without bzr revision id
+mysqlslap		 : n/a	2012-04-29 Failed once on kfreebsd
+mysqlhotcopy_isam	 : n/a  2012-04-29 Olaf van der Spek reported this failure
+mysqlhotcopy_archive	 : n/a  2012-04-29 Olaf van der Spek reported this failure
