Author: Nicholas Bamber <nicholas@periapt.co.uk>
Subject: FTBS when in build directory containing '+'
Bug-Debian: http://bugs.debian.org/674210
Bug: http://bugs.mysql.com/bug.php?id=65408
Last-Update: 2012-05-24
Index: mysql-5.5.29/mysql-test/include/mysqld--help.inc
===================================================================
--- mysql-5.5.29.orig/mysql-test/include/mysqld--help.inc	2013-01-02 15:36:15.078372518 -0800
+++ mysql-5.5.29/mysql-test/include/mysqld--help.inc	2013-01-02 15:38:16.403258546 -0800
@@ -47,7 +47,11 @@
     s/\b4294967295\b/18446744073709551615/;
     s/\b2146435072\b/9223372036853727232/;
     s/\b196608\b/262144/;
-    foreach $var (@env) { s/$ENV{$var}/$var/ }
+    foreach $var (@env) {
+	my $re = $ENV{$var};
+	$re =~ s/\+/\\\+/g;
+	s/$re/$var/
+    }
     next if /use --skip-(use-)?symbolic-links to disable/; # for valgrind, again
     next if $skip;
     print;
Index: mysql-5.5.29/mysql-test/suite/sys_vars/t/character_sets_dir_basic.test
===================================================================
--- mysql-5.5.29.orig/mysql-test/suite/sys_vars/t/character_sets_dir_basic.test	2013-01-02 15:36:15.078372518 -0800
+++ mysql-5.5.29/mysql-test/suite/sys_vars/t/character_sets_dir_basic.test	2013-01-02 15:38:16.403258546 -0800
@@ -7,6 +7,7 @@
 # TODO: fix with a proper comparison in mysqltest
 let $rcd= `SELECT REPLACE('$MYSQL_CHARSETSDIR', '\\\\\', '.')`;
 let $rcd= `SELECT REPLACE('$rcd', '/', '.')`;
+let $rcd= `SELECT REPLACE('$rcd', '+', '.')`;
 let $regex_charsetdir= `SELECT '/$rcd[\\\\\/\\\\\]/MYSQL_CHARSETSDIR/'`;
 
 --replace_regex $regex_charsetdir
