Author: Nicholas Bamber <nicholas@periapt.co.uk>
Bug: http://bugs.mysql.com/bug.php?id=64386
Debian-Bug: http://bugs.debian.org/660686
Subject: versioned symbols got dropped from the build
Last-Update: 2012-06-01
Index: mysql-5.5.29/libmysql/CMakeLists.txt
===================================================================
--- mysql-5.5.29.orig/libmysql/CMakeLists.txt	2013-01-02 15:36:15.014376269 -0800
+++ mysql-5.5.29/libmysql/CMakeLists.txt	2013-01-02 15:38:17.931168951 -0800
@@ -136,6 +136,8 @@
 
 )
 
+CONFIGURE_FILE(libmysql.ver.in ${CMAKE_BINARY_DIR}/libmysql/libmysql.ver)
+
 SET(CLIENT_SOURCES
   get_password.c 
   libmysql.c
@@ -209,7 +211,7 @@
         SET(libmysql_link_flags)
       ENDIF()
       SET_TARGET_PROPERTIES(libmysql PROPERTIES LINK_FLAGS 
-        "${libmysql_link_flags} ${LINK_FLAG_NO_UNDEFINED}")
+        "${libmysql_link_flags} ${LINK_FLAG_NO_UNDEFINED} -Wl,--version-script=libmysql.ver")
     ENDIF() 
     # clean direct output needs to be set several targets have the same name
     #(mysqlclient in this case)
