fix syntax errors in setup.py
Index: seascope/setup.py
===================================================================
--- seascope.orig/setup.py	2013-03-08 12:26:36.885917162 +0530
+++ seascope/setup.py	2013-03-08 12:27:29.449916683 +0530
@@ -5,17 +5,18 @@
 #
 # License: BSD 
 
+import textwrap
 from distutils.core import setup
 
 setup(name='Seascope',
 	version='0.7',
-	description='A multi-platform multi-language source code browsing tool'
-	long_description='A pyQt GUI front-end for idutils, cscope and gtags. Written in python using pyQt, QScintilla libraries.',
+	description='A multi-platform multi-language source code browsing tool',
+	long_description= textwrap.dedent("""A pyQt GUI front-end for idutils, cscope and gtags. Written in python using pyQt, QScintilla libraries."""),
 	url='http://seascope.googlecode.com',
 	packages=['Seascope',
 		'Seascope.backend',
 		'Seascope.backend.plugins',
-		'Seascope.backend.plugins.idutils'
+		'Seascope.backend.plugins.idutils',
 		'Seascope.backend.plugins.cscope',
 		'Seascope.backend.plugins.gtags',
 		'Seascope.view',
