---
 Makefile |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/tools/Makefile
+++ b/tools/Makefile
@@ -20,7 +20,7 @@
                  llc llvm-ranlib llvm-ar llvm-nm \
                  llvm-ld llvm-prof llvm-link \
                  lli llvm-extract llvm-mc \
-                 bugpoint llvm-bcanalyzer llvm-stub \
+                 bugpoint llvm-bcanalyzer \
                  llvmc llvm-diff
 
 # Let users override the set of tools to build from the command line.
