--- a/term/gd.trm
+++ b/term/gd.trm
@@ -981,7 +981,7 @@
 	if (external_default)
 		png_state.ttffont = gp_strdup(external_default);
 	else	/* Might as well try some plausible font; it's no worse than failing immediately */
-		png_state.ttffont = gp_strdup("arial");
+		png_state.ttffont = gp_strdup("/usr/share/fonts/truetype/liberation/LiberationSans-Regular.ttf");
 
 	free(png_state.default_ttffont);
 	png_state.default_ttffont = gp_strdup(png_state.ttffont);
