Description: Update libtomboy dllmap to use absolute path
Index: tomboy.git/Tomboy/Tomboy.exe.config.in
===================================================================
--- tomboy.git.orig/Tomboy/Tomboy.exe.config.in	2011-04-12 19:37:20.634328063 +0100
+++ tomboy.git/Tomboy/Tomboy.exe.config.in	2011-04-12 19:37:48.253245653 +0100
@@ -1,7 +1,7 @@
 <configuration>
-  <dllmap dll="libglib-2.0-0.dll" target="libglib-2.0.so.0"/>
+  <dllmap dll="glib-2.0.dll" target="libglib-2.0.so.0"/>
   <dllmap dll="libgtk-win32-2.0-0.dll" target="libgtk-x11-2.0.so.0"/>	
   <dllmap dll="libgtkspell" target="libgtkspell.so.0"/>
-  <dllmap dll="libtomboy" target="libtomboy.so"/>
+  <dllmap dll="libtomboy" target="/usr/lib/tomboy/libtomboy.so"/>
   <dllmap dll="panel-applet-2" target="libpanel-applet-2.so.0"/>
 </configuration>
