# DP: document readline header location on Debian systems

--- a/bash/examples/loadables/README
+++ b/bash/examples/loadables/README
@@ -32,6 +32,9 @@
 guide'.  The file template.c provides a template to use for creating
 new loadable builtins.
 
+On Debian GNU/Linux systems, the bash headers are in /usr/include/bash.
+The appropriate options are already set in the example Makefile.
+
 basename.c	Return non-directory portion of pathname.
 cat.c		cat(1) replacement with no options - the way cat was intended.
 cut.c		cut(1) replacement.
