Index: guake-0.4.2/src/guake.py
===================================================================
--- guake-0.4.2.orig/src/guake.py	2012-05-21 17:19:54.000000000 +0200
+++ guake-0.4.2/src/guake.py	2012-05-21 17:21:49.000000000 +0200
@@ -1125,6 +1125,7 @@
         box.terminal.grab_focus()
         box.terminal.connect('button-press-event', self.show_context_menu)
         box.terminal.connect('child-exited', self.on_terminal_exited, box)
+	box.terminal.set_alternate_screen_scroll(True)
         box.show()
 
         self.term_list.append(box.terminal)
