Index: guake-0.4.3/src/guake
===================================================================
--- guake-0.4.3.orig/src/guake	2012-06-06 21:31:37.000000000 +0200
+++ guake-0.4.3/src/guake	2012-06-07 10:17:43.000000000 +0200
@@ -787,9 +787,9 @@
             self.add_tab()
 
         window_rect = self.get_final_window_rect()
-        self.window.move(window_rect.x, window_rect.y)
         self.window.resize(window_rect.width, window_rect.height)
         self.window.show_all()
+        self.window.move(window_rect.x, window_rect.y)
 
         try:
             # does it work in other gtk backends
