Index: guake-0.4.3/src/guake
===================================================================
--- guake-0.4.3.orig/src/guake	2012-06-07 10:17:43.000000000 +0200
+++ guake-0.4.3/src/guake	2012-06-07 10:22:17.000000000 +0200
@@ -968,7 +968,10 @@
             if not val:
                 self.toolbar.hide()
         else:
+            window_rect = self.get_final_window_rect()
             self.window.unfullscreen()
+            self.window.resize(window_rect.width, window_rect.height)
+            self.window.move(window_rect.x, window_rect.y)
             self.fullscreen = False
 
             # making sure that tabbar and resizer will come back to
