Description: libccrtp: ftbfs gcc46
Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=626065

Index: libccrtp-2.0.1/src/ccrtp/sources.h
===================================================================
--- libccrtp-2.0.1.orig/src/ccrtp/sources.h	2011-05-29 07:47:58.000000000 +1000
+++ libccrtp-2.0.1/src/ccrtp/sources.h	2011-10-29 14:02:14.000000000 +1100
@@ -46,6 +46,7 @@
 
 #include <string>
 #include <ccrtp/rtcppkt.h>
+#include <cstddef>
 
 NAMESPACE_COMMONCPP
 
