Author: Reinhard Tartler <siretart@tauware.de>
Description: Disable configuration output warnings
Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=619530
Bug-Ubuntu: https://launchpad.net/bugs/765357
Forwarded: not-needed

--- a/cmdutils.c
+++ b/cmdutils.c
@@ -458,7 +458,9 @@ void print_error(const char *filename, i
     av_log(NULL, AV_LOG_ERROR, "%s: %s\n", filename, errbuf_ptr);
 }
 
-static int warned_cfg = 0;
+// Debian/Ubuntu: see http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=619530
+//                    https://launchpad.net/bugs/765357
+static int warned_cfg = 1;
 
 #define INDENT        1
 #define SHOW_VERSION  2
