Description: Fix headers includes
Author: Fathi Boudra <fabo@debian.org>
Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=549757

--- a/src/libicalcap/icalcap.h
+++ b/src/libicalcap/icalcap.h
@@ -1,7 +1,7 @@
 #ifndef __ICALCAP_H__
 #define __ICALCAP_H__
 
-#include <ical.h>
+#include <libical/ical.h>
 
 /*
  * Opaque objects
--- a/src/libicalcap/icalcap_message_impl.h
+++ b/src/libicalcap/icalcap_message_impl.h
@@ -3,7 +3,7 @@
 
 #ifdef WITH_RR
 
-#include <ical.h>
+#include <libical/ical.h>
 #include <librr/rr.h>
 
 #define	ICALCAP_MESSAGE_CMD	1
--- a/src/libicalss/icalcalendar.h
+++ b/src/libicalss/icalcalendar.h
@@ -30,7 +30,7 @@
 #define ICALCALENDAR_H
 
 #include <libical/ical.h>
-#include <icalset.h>
+#include "icalset.h"
 
 /* icalcalendar
  * Routines for storing calendar data in a file system. The calendar 
--- a/src/libicalss/icalclassify.h
+++ b/src/libicalss/icalclassify.h
@@ -28,8 +28,8 @@
 #define ICALCLASSIFY_H
 
 #include <libical/ical.h>
-#include <icalset.h>
-#include <icalcomponent.h>
+#include "icalset.h"
+#include "icalcomponent.h"
 
 icalproperty_xlicclass icalclassify(icalcomponent* c,icalcomponent* match, 
 			      const char* user);
--- a/src/libicalss/icalcluster.h
+++ b/src/libicalss/icalcluster.h
@@ -30,7 +30,7 @@
 #define ICALCLUSTER_H
 
 #include <libical/ical.h>
-#include <icalset.h>
+#include "icalset.h"
 
 typedef struct icalcluster_impl icalcluster;
 
--- a/src/libicalss/icaldirset.h
+++ b/src/libicalss/icaldirset.h
@@ -30,9 +30,9 @@
 #define ICALDIRSET_H
 
 #include <libical/ical.h>
-#include <icalset.h>
-#include <icalcluster.h>
-#include <icalgauge.h>
+#include "icalset.h"
+#include "icalcluster.h"
+#include "icalgauge.h"
 
 /* icaldirset Routines for storing, fetching, and searching for ical
  * objects in a database */
--- a/src/libicalss/icalfileset.h
+++ b/src/libicalss/icalfileset.h
@@ -30,9 +30,9 @@
 #define ICALFILESET_H
 
 #include <libical/ical.h>
-#include <icalset.h>
-#include <icalcluster.h>
-#include <icalgauge.h>
+#include "icalset.h"
+#include "icalcluster.h"
+#include "icalgauge.h"
 #include <sys/types.h> /* For open() flags and mode */
 #include <sys/stat.h> /* For open() flags and mode */
 #include <fcntl.h> /* For open() flags and mode */
--- a/src/libicalss/icalgauge.h
+++ b/src/libicalss/icalgauge.h
@@ -29,7 +29,7 @@
 #ifndef ICALGAUGE_H
 #define ICALGAUGE_H
 
-#include <icalcomponent.h>
+#include "icalcomponent.h"
 
 /** @file icalgauge.h
  *  @brief Routines implementing a filter for ical components
--- a/src/libicalss/icalset.h
+++ b/src/libicalss/icalset.h
@@ -41,7 +41,7 @@
 
 #include <limits.h> /* For PATH_MAX */
 #include <libical/ical.h>
-#include <icalgauge.h>
+#include "icalgauge.h"
 
 #ifdef PATH_MAX
 #define ICAL_PATH_MAX PATH_MAX
--- a/src/libicalss/icalspanlist.h
+++ b/src/libicalss/icalspanlist.h
@@ -26,7 +26,7 @@
 #define ICALSPANLIST_H
 
 #include <libical/ical.h>
-#include <icalset.h>
+#include "icalset.h"
 
 /** @file icalspanlist.h
  *  @brief Code that supports collections of free/busy spans of time
