From: Gergely Nagy <algernon@madhouse-project.org>
Date: Thu, 9 Feb 2012 10:14:42 +0100
Subject: libmikmod/Makefile.in: Respect $(CPPFLAGS) and $(LDFLAGS)

---
 libmikmod/Makefile.in |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/libmikmod/Makefile.in b/libmikmod/Makefile.in
index bb12574..ee6e1b6 100644
--- a/libmikmod/Makefile.in
+++ b/libmikmod/Makefile.in
@@ -32,7 +32,7 @@ MKINSTALLDIRS=install -d
 
 DEFS=@DEFS@
 CFLAGS=@CFLAGS@ -Dunix
-COMPILE=$(LIBTOOL) --mode=compile $(CC) $(DEFS) $(CFLAGS) -I$(top_srcdir)/include -I$(top_builddir) -I$(top_builddir)/include -DMIKMOD_H=$(top_srcdir)/include/mikmod.h
+COMPILE=$(LIBTOOL) --mode=compile $(CC) $(DEFS) $(CPPFLAGS) $(CFLAGS) -I$(top_srcdir)/include -I$(top_builddir) -I$(top_builddir)/include -DMIKMOD_H=$(top_srcdir)/include/mikmod.h
 
 LIB = libmikmod.la
 OBJ = $(LIBOBJS) $(DLOBJS) \
@@ -74,7 +74,7 @@ distclean:	clean
 	rm -f Makefile
 
 $(LIB): $(OBJ)
-	$(LIBTOOL) --mode=link $(CC) -version-info 2:4:0 -o $@ $(OBJ) $(LIBRARY_LIB) -rpath $(DESTDIR)$(libdir)
+	$(LIBTOOL) --mode=link $(CC) -version-info 2:4:0 $(LDFLAGS) -o $@ $(OBJ) $(LIBRARY_LIB) -rpath $(DESTDIR)$(libdir)
 
 dl_hpux.lo:	$(top_srcdir)/dlapi/dl_hpux.c \
 			$(top_srcdir)/dlapi/dlfcn.h
