From: Pantelis Koukousoulas <pktoss@gmail.com>
Date: Sun, 9 Sep 2012 11:54:09 +0300
Subject: fix building of ESD

---
 configure.in |   86 ++++------------------------------------------------------
 1 file changed, 6 insertions(+), 80 deletions(-)

diff --git a/configure.in b/configure.in
index 2a6a86c..26c74e7 100644
--- a/configure.in
+++ b/configure.in
@@ -62,15 +62,6 @@ else
   libmikmod_driver_alsa=no
 fi])
 
-AC_ARG_ENABLE(esd,
-[  --enable-esd            include Enlightened sound daemon driver [guessed]],
-[if test "$enableval" = "yes"
-then
-  libmikmod_driver_esd=yes
-else
-  libmikmod_driver_esd=no
-fi])
-
 AC_ARG_ENABLE(oss,
 [  --enable-oss            include OSS driver [guessed]],
 [if test "$enableval" = "yes"
@@ -266,27 +257,6 @@ else
 	libmikmod_driver_alsa=no
 fi
 
-# esd
-# ---
-if test $libmikmod_driver_esd = yes
-then
-	libmikmod_driver_esd=no
-	AM_PATH_ESD(0.2.6,libmikmod_driver_esd=yes)
-# We also need to know if esd is compiled with alsa support
-	if test $libmikmod_driver_esd = yes
-	then
-		AC_CACHE_CHECK([if esd requires alsa],libmikmod_cv_esd_needs_alsa,
-		if (esd-config --libs 2>&1 |grep asound >/dev/null) 2>/dev/null
-		then
-			libmikmod_cv_esd_needs_alsa=yes
-		else
-			libmikmod_cv_esd_needs_alsa=no
-		fi)
-	else
-		libmikmod_cv_esd_needs_alsa=no
-	fi
-fi
-
 # sgi audio - IRIX only
 # ---------------------
 case `uname` in
@@ -456,23 +426,6 @@ then
 	fi
 fi
 
-if test $libmikmod_driver_esd = yes
-then
-	libmikmod_driverlist="$libmikmod_driverlist esd"
-	AC_DEFINE(DRV_ESD)
-	CFLAGS="$CFLAGS $ESD_CFLAGS"
-
-	# Under Solaris and perhaps other systems, we have to forward esd
-	# dependencies to libmikmod's dependency list.
-	# However, no need to keep libraries we won't require (libossaudio on BSD
-	# systems, libaudiofile, libasound on Linux)
-	if test $libmikmod_dynload != no
-	then
-		ESD_LIBS="`echo $ESD_LIBS|sed -e 's/-lesd//'|sed -e 's/-lasound//'|sed -e 's/-laudiofile//'|sed -e 's/-lossaudio//'`"
-	fi
-	LIBRARY_LIB="$ESD_LIBS $LIBRARY_LIB"
-fi
-
 if test $libmikmod_driver_hp = yes
 then
 	libmikmod_driverlist="hp $libmikmod_driverlist"
@@ -508,6 +461,12 @@ then
 	fi
 fi
 
+if test $libmikmod_driver_esd = yes
+then
+	libmikmod_driverlist="esd $libmikmod_driverlist"
+	AC_DEFINE(DRV_ESD)
+fi
+
 if test $libmikmod_driver_sam9407 = yes
 then
 	libmikmod_driverlist="sam9407 $libmikmod_driverlist"
@@ -545,39 +504,6 @@ fi
 if test $libmikmod_dynload != no
 then
 	AC_DEFINE(MIKMOD_DYNAMIC)
-	if test "`uname`" = "Linux" && \
-	   test $libmikmod_cv_esd_needs_alsa = yes && \
-	   test $libmikmod_driver_esd = yes
-	then
-		# @#!$ libmikmod_dynloading esd requires us to define and make visible
-		# extra symbols (at least for 0.2.6 to 0.2.8)
-		AC_CACHE_CHECK([if libmikmod_dynloading esd causes problems],
-		libmikmod_cv_esd_broken,
-		libmikmod_oldlibs=$LIBS
-		LIBS="$LIBS $ESD_LIBS"
-		AC_TRY_LINK([
-#include <esd.h>
-extern void *handle;],
-		[void *p = handle;],
-		libmikmod_cv_esd_broken=no,libmikmod_cv_esd_broken=yes)
-		LIBS=$libmikmod_oldlibs)
-		if test $libmikmod_cv_esd_broken = yes
-		then
-			AC_DEFINE(MIKMOD_DYNAMIC_ESD_NEEDS_ALSA)
-			LIB_LDADD="$LIB_LDADD -rdynamic"
-			echo "
-*** This means your version of EsounD defines some public symbols,
-*** which should have been defined static or weak. The compiled
-*** libmikmod will try to take care of the problem, but you'll have to
-*** use the '-rdynamic' option when linking an application which uses
-*** libmikmod with gcc. If you use another compiler, or -rdynamic is not
-*** available on your platform, you might not be able to use libmikmod's
-*** EsounD driver.
-*** In this case, consider recompiling libmikmod specifying
-*** '--disable-dl' in the configure script options.
-" >&2
-		fi
-	fi
 
 	LIBRARY_LIB="$libmikmod_dllib $LIBRARY_LIB"
 	if test $libmikmod_dynload = sun
