From 66de1ffa3817ecc98617671b829f031c83bf740b Mon Sep 17 00:00:00 2001
From: Joshua Kwan <joshk@triplehelix.org>
Date: Thu, 24 Sep 2009 14:21:19 -0700
Subject: Define _GNU_SOURCE for RTLD_NEXT.

---
 sys/unix/unixres.c |    1 +
 1 file changed, 1 insertion(+)

diff --git a/sys/unix/unixres.c b/sys/unix/unixres.c
index 3e7f989..a6ac959 100644
--- a/sys/unix/unixres.c
+++ b/sys/unix/unixres.c
@@ -21,6 +21,7 @@
 # if defined(LINUX)
 
 /* requires dynamic linking with libc */
+#define _GNU_SOURCE
 #include <dlfcn.h>
 
 static int
