From 77d8972a0dfdd46bd7ad3661e0d0a2f39b8a253f Mon Sep 17 00:00:00 2001
From: Joshua Kwan <joshk@triplehelix.org>
Date: Thu, 24 Sep 2009 14:21:19 -0700
Subject: Use /usr/share/pixmaps/nethack/ for pretty pictures.

---
 win/X11/winX.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/win/X11/winX.c b/win/X11/winX.c
index 14f0e90..1c5d49b 100644
--- a/win/X11/winX.c
+++ b/win/X11/winX.c
@@ -919,12 +919,12 @@ static XtResource resources[] = {
     { "message_lines", "Message_lines", XtRInt, sizeof(int),
       XtOffset(AppResources *,message_lines), XtRString, "12" },
     { "pet_mark_bitmap", "Pet_mark_bitmap", XtRString, sizeof(String),
-      XtOffset(AppResources *,pet_mark_bitmap), XtRString, "pet_mark.xbm" },
+      XtOffset(AppResources *,pet_mark_bitmap), XtRString, PIXMAPDIR "pet_mark.xbm" },
     { "pet_mark_color", "Pet_mark_color", XtRPixel, sizeof(XtRPixel),
       XtOffset(AppResources *,pet_mark_color), XtRString, "Red" },
 #ifdef GRAPHIC_TOMBSTONE
     { "tombstone", "Tombstone", XtRString, sizeof(String),
-      XtOffset(AppResources *,tombstone), XtRString, "rip.xpm" },
+      XtOffset(AppResources *,tombstone), XtRString, PIXMAPDIR "rip.xpm" },
     { "tombtext_x", "Tombtext_x", XtRInt, sizeof(int),
       XtOffset(AppResources *,tombtext_x), XtRString, "155" },
     { "tombtext_y", "Tombtext_y", XtRInt, sizeof(int),
