From 49e68ac683cbe6ab2d2fa4db925938860a9b2a27 Mon Sep 17 00:00:00 2001
From: Joshua Kwan <joshk@triplehelix.org>
Date: Thu, 24 Sep 2009 14:21:19 -0700
Subject: Better way to do it.

---
 src/windows.c |    8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/src/windows.c b/src/windows.c
index d65710c..8953269 100644
--- a/src/windows.c
+++ b/src/windows.c
@@ -103,10 +103,16 @@ void
 choose_windows(s)
 const char *s;
 {
+    char *ow; const char *wt;
     register int i;
+    
+    if (!strcmp(s, DEFAULT_WINDOW_SYS) && (ow = getenv("OVERRIDEWIN")))
+      wt = ow;
+    else
+      wt = s;
 
     for(i=0; winchoices[i].procs; i++)
-	if (!strcmpi(s, winchoices[i].procs->name)) {
+	if (!strcmpi(wt, winchoices[i].procs->name)) {
 	    windowprocs = *winchoices[i].procs;
 	    if (winchoices[i].ini_routine) (*winchoices[i].ini_routine)();
 	    return;
