From: Jakob Haufe <sur5r@sur5r.net>
Date: Mon, 9 Apr 2012 18:40:04 +0200
Subject: Ignore license.page
Forwarded: not-needed

---
 help/Makefile.am |    1 -
 1 file changed, 1 deletion(-)

diff --git a/help/Makefile.am b/help/Makefile.am
index 1185a90..1427128 100644
--- a/help/Makefile.am
+++ b/help/Makefile.am
@@ -34,7 +34,6 @@ DOC_PAGES = \
 	editprop.page		\
 	index.page		\
 	labelprop.page		\
-	license.page		\
 	mancreate.page		\
 	merge-ex1.page		\
 	merge-ex2.page		\
