Index: xutils-dev/xorg-cf-files/bsdLib.rules
===================================================================
--- xutils-dev.orig/xorg-cf-files/bsdLib.rules
+++ xutils-dev/xorg-cf-files/bsdLib.rules
@@ -13,6 +13,12 @@ XCOMM $XFree86: xc/config/cf/bsdLib.rule
 #define UseElfFormat NO
 #endif
 
+#if NetBSDDistribution == NetBSDDebian
+# ifndef SharedOldX
+#  define SharedOldX NO
+# endif
+#endif /* Debian */
+
 #if !UseElfFormat
 /* a.out shared libs */
 #ifndef HasSharedLibraries
@@ -91,6 +97,34 @@ clean::									@@\
 
 #endif /* SharedDepLibraryTarget */
 
+/*
+ * SharedDepCplusplusLibraryTarget - generate rules to create a shared library.
+ */
+#ifndef SharedDepCplusplusLibraryTarget
+#define SharedDepCplusplusLibraryTarget(libname,rev,deplist,solist,down,up)	@@\
+AllTarget(Concat(lib,libname.so.rev))					@@\
+									@@\
+Concat(lib,libname.so.rev):  deplist					@@\
+	$(RM) $@~							@@\
+	@SONAME=`echo $@ | sed 's/\.[^\.]*$$//'`; set -x; \		@@\
+		(cd down; $(CXX) -o up/$@~ $(SHLIBLDFLAGS) -Wl,-soname,$$SONAME solist $(REQUIREDLIBS) BaseShLibReqs); \ @@\
+		$(RM) $$SONAME; $(LN) $@ $$SONAME; \			@@\
+		LinkBuildSonameLibrary($$SONAME)			@@\
+	$(RM) $@							@@\
+	$(MV) $@~ $@							@@\
+	$(RM) Concat(lib,libname.so)					@@\
+	$(LN) $@ Concat(lib,libname.so)					@@\
+	LinkBuildLibrary($@)						@@\
+	LinkBuildLibrary(Concat(lib,libname.so))			@@\
+									@@\
+clean::									@@\
+	@SONAME=`echo Concat(lib,libname.so.rev) | sed 's/\.[^\.]*$$//'`; \ @@\
+	set -x; $(RM) $$SONAME						@@\
+	$(RM) Concat(lib,libname.so)					@@\
+	$(RM) Concat(lib,libname.so.rev)
+
+#endif /* SharedDepCplusplusLibraryTarget */
+
 #ifndef SharedDepModuleTarget
 #define SharedDepModuleTarget(name,deps,solist)				@@\
 AllTarget(name)								@@\
Index: xutils-dev/xorg-cf-files/gnu.cf
===================================================================
--- xutils-dev.orig/xorg-cf-files/gnu.cf
+++ xutils-dev/xorg-cf-files/gnu.cf
@@ -17,16 +17,231 @@ XCOMM $XdotOrg: xc/config/cf/gnu.cf,v 1.
 #ifndef OSTeenyVersion
 #define OSTeenyVersion		DefaultOSTeenyVersion
 #endif
-XCOMM operating system:  OSName (OSMajorVersion./**/OSMinorVersion./**/OSTeenyVersion)
 
-#define GNUSourceDefines      -D_POSIX_C_SOURCE=199309L \
-                              -D_POSIX_SOURCE -D_XOPEN_SOURCE \
-                              -D_BSD_SOURCE -D_SVID_SOURCE
+#ifndef HasGhostScript
+# define HasGhostScript YES
+#endif
+#ifndef BuildPDFdocs
+# define BuildPDFdocs NO
+#endif
+
+#ifndef GnuBinUtilsMajorVersion
+# define GnuBinUtilsMajorVersion	DefaultGnuBinUtilsMajorVersion
+#endif
 
-XCOMM XXXMB: What about _GNU_SOURCE, see Linux/UseInstalled?
+XCOMM operating system:  OSName (OSMajorVersion./**/OSMinorVersion./**/OSTeenyVersion)
+XCOMM libc:	(GnuCLibMajorVersion./**/GnuCLibMinorVersion./**/GnuCLibTeenyVersion)
+XCOMM binutils:	(GnuBinUtilsMajorVersion)
 
-#define XawI18nDefines        -DHAS_WCHAR_H -DHAS_WCTYPE_H -DNO_WIDEC_H
+#define HasDevRandom		NO
+#define PollDevRandom		NO
 
+/* Path setup. */
+# ifndef NothingOutsideProjectRoot
+#  define NothingOutsideProjectRoot	NO
+# endif
+# if !defined(NothingOutsideProjectRoot) || !NothingOutsideProjectRoot
+#  define SystemManDirectory	/usr/share/man
+# endif
+
+/* Default GCC options. */
+# ifndef DefaultGcc2OptimizeOpt
+#  define DefaultGcc2OptimizeOpt	-O2
+# endif
+
+# define HasPam			YES
+
+# if !defined(NothingOutsideProjectRoot) || !NothingOutsideProjectRoot
+#  define XAppLoadDir		EtcX11Directory/app-defaults
+# define XFileSearchPathDefault	Concat4(EtcX11Directory/%L/%T/%N%C,%S:EtcX11Directory/%l/%T/%N%C,%S:EtcX11Directory/%T/%N%C,%S:EtcX11Directory/%L/%T/%N%S:EtcX11Directory/%l/%T/%N%S:EtcX11Directory/%T/%N%S):Concat4($(LIBDIR)/%L/%T/%N%C,%S:$(LIBDIR)/%l/%T/%N%C,%S:$(LIBDIR)/%T/%N%C,%S:$(LIBDIR)/%L/%T/%N%S:$(LIBDIR)/%l/%T/%N%S:$(LIBDIR)/%T/%N%S)
+/* the relative symlink created by this rule causes problems for us */
+# define InstallAppDefFiles	YES
+# ifdef InstallAppDefFiles
+#  if InstallAppDefFiles
+#   define InstallAppDefaultsLong(file,class)				@@\
+InstallNamedTarget(install,file.ad,$(INSTAPPFLAGS),$(XAPPLOADDIR),class)
+#  else
+#   define InstallAppDefaultsLong(file,class)				@@\
+InstallNamedTargetNoClobber(install,file.ad,$(INSTAPPFLAGS),$(XAPPLOADDIR),class)
+#  endif
+# endif /* InstallAppDefFiles */
+# endif /* !defined(NothingOutsideProjectRoot) || !NothingOutsideProjectRoot */
+
+# define SharedLibXdmGreet	NO
+# define LinkGLToUsrInclude	NO
+# define LinkGLToUsrLib		NO
+# define FSUseSyslog		YES
+
+# define DriverManSuffix       4x
+# define DriverManDir          $(MANSOURCEPATH)4
+
+# define MiscManSuffix         7x
+# define MiscManDir            $(MANSOURCEPATH)7
+
+# define ProjectManSuffix	x
+
+# define DebianMaintainer      YES
+
+# ifdef DebianMaintainer
+
+#  ifndef XorgCustomVersion
+#   define XorgCustomVersion		"Debian"
+#  endif
+#  ifndef BuilderEMailAddr
+#   define BuilderEMailAddr		"debian-x@lists.debian.org"
+#  endif
+
+#  define XFree86Devel			YES
+#  define InstallXinitConfig		YES
+#  define InstallXdmConfig		YES
+#  define InstallFSConfig		YES
+#  define DebuggableLibraries		YES
+#  define ForceNormalLib		YES
+#  define SharedLibFont			NO
+#  define SharedLibGlu			YES
+#  define NormalLibGlu			YES
+#  define BuildFontCache		YES
+
+#  define BuildAllSpecsDocs		NO
+#  ifndef BuildSpecsDocs
+#   define BuildSpecsDocs		YES
+#  endif
+#  define SpecsDocDirs			BDF CTEXT FSProtocol GL ICCCM ICE PM \
+					Render SM X11 XDMCP XIM XLFD XProtocol \
+					Xaw Xext Xi Xmu Xserver Xt Xv XvMC \
+					i18n rstart xfs xterm xtrans
+#  define XkbServerDefines		-DNOISY
+#  define BuildHtmlManPages		NO
+
+#  define DontPreprocessManPages	NO
+
+/* We have a local patch that defines PicLibraryTarget(), instead of upstream's
+ * method. */
+#  define StaticNeedsPicForShared	NO
+
+/* Some libraries changed compilation method from static to shared and they do not
+ * ship header files.
+ * Temporary revert the build to static until all the changes will be investigated
+ * properly
+ */
+#  define SharedLibXau                  YES
+#  define SharedLibXdmcp                YES
+#  define NormalOldX                    NO
+#  define DebugOldX                     NO
+#  define SharedOldX                    NO
+#  define SharedLibDmx                  YES
+#  define SharedLibGLw                  NO
+#  define SharedLibFont                 NO
+#  define SharedLibFontEnc              NO
+#  define SharedLibXfontcache           NO
+#  define BuildFreetypeDlModule         NO
+
+#  define KernelVersionInBanner		YES
+
+/* We don't want XRX. */
+#  define BuildPlugin			NO
+
+/* The TT/Meltho fonts are all under non-free licences, except for BitStream
+ * Vera, which is provided by an external package anyway. */
+#  define BuildTrueTypeFonts		NO
+#  define BuildBethMarduthoFonts	NO
+
+#  define MakeDllModules		YES
+
+/* All the following are fulfilled by Build-Depends. */
+#  define BuildRman			NO
+#  define RmanCmd			/usr/bin/rman
+#  define RmanOptions			/**/
+#  define HasTk				YES
+#  define TkLibDir			/usr/lib
+#  define TkIncDir			/usr/include
+#  define TkLibName			tk8.4
+#  define XF86SetupUsesStaticTk		NO
+#  define HasTcl			YES
+#  define TclLibDir			/usr/lib
+#  define TclIncDir			/usr/include
+#  define TclLibName			tcl8.4
+#  define XF86SetupUsesStaticTcl	NO
+#  define BuildExpatLibrary		NO
+#  define HasExpat			YES
+#  define ExpatIncDirStandard		YES
+#  define BuildFontconfigLibrary	NO
+#  define HasFontconfig			YES
+#  define UseFontconfig			YES
+#  define BuildRenderLibrary		NO
+#  define HasRenderLibrary		YES
+#  define BuildXcursorLibrary		NO
+#  define HasXcursorLibrary		YES
+/* Xlib needs to know the available version of Xcursor. */
+#  define SharedXcursorRev		1
+#  define BuildXftLibrary		NO
+#  define HasXftLibrary			YES
+#  define BuildFreetype2Library		NO
+#  define HasFreetype2			YES
+#  define FontLibSharedFreeType		NO
+#  define HasLatex			YES
+#  define HasGroffHtml			YES
+
+/*
+ * The XFree86 Debian package no longer builds the Xprt server, in favor
+ * of a separate xprint package.  We do still build the library,
+ * but only when we're not building servers only, so it gets built in the
+ * normal build, but not -dbg (as it requires libXt headers).
+ */
+#  define XprtServer			NO
+#  define BuildXprintLib		!BuildServersOnly
+
+#  ifdef UseInstalled
+#   define RenderLibraryDir		/usr
+#   define XftLibraryDir		/usr
+#   define XcursorLibraryDir		/usr
+#  else /* We have a special hack to find these. */
+#   define RenderLibraryDir		$(TOP)/../../imports
+#   define XftLibraryDir		$(TOP)/../../imports
+#   define XcursorLibraryDir		$(TOP)/../../imports
+#  endif
+
+/* Wraphelp.c magic is supplied via a custom patch. */
+#  define HasXdmAuth			YES
+
+#  define HasGlide2			NO
+#  define HasGlide3			NO
+
+/* Enable extended instruction set support. */
+#  ifdef i386Architecture
+#   define HasX86Support		YES
+#   define HasMMXSupport		YES
+#   define Has3DNowSupport		YES
+#   define HasSSESupport		YES
+#  endif /* i386Architecture */
+# endif /* DebianMaintainer */
+
+#ifndef BinUtilsMajorVersion
+# define BinUtilsMajorVersion	GnuBinUtilsMajorVersion
+#endif
+
+#ifndef GnuLocaleDefines
+# define GnuLocaleDefines	/**/
+#endif
+
+#ifndef  GnuAdditionalIncludes
+# define GnuAdditionalIncludes	/**/
+#endif
+
+#ifndef GnuGnuSourceDefines
+# ifdef UseInstalled
+#  define GnuGnuSourceDefines	/**/
+# else
+#  define GnuGnuSourceDefines	-D_GNU_SOURCE
+# endif
+#endif
+
+# define GnuSourceDefines	-D_POSIX_C_SOURCE=199309L \
+				-D_POSIX_SOURCE -D_XOPEN_SOURCE \
+				-D_BSD_SOURCE -D_SVID_SOURCE \
+				GnuAdditionalIncludes GnuGnuSourceDefines \
+				GnuLocaleDefines
+# define XawI18nDefines		-DHAS_WCHAR_H -DHAS_WCTYPE_H -DNO_WIDEC_H
 #define HasPosixThreads         YES
 #define ThreadedX               YES
 #define HasThreadSafeAPI        YES
@@ -36,35 +251,132 @@ XCOMM XXXMB: What about _GNU_SOURCE, see
 #ifndef HasLibCrypt
 #define HasLibCrypt             YES
 #endif
+/* netscape wraps select but not poll as of communicator 4.72 */
+# ifndef HasPoll
+#  define HasPoll		NO
+# endif
+
+#ifndef InstallXloadSetGID
+#define InstallXloadSetGID	NO
+#endif
+
+#ifndef BuildIPv6
+#define BuildIPv6		NO
+#endif
+
+#ifndef HasLinuxInput
+# define HasLinuxInput NO
+#endif
+
+#ifndef JoystickSupport
+# define JoystickSupport	NO
+#endif
+#ifndef HasLinuxJoystick
+# define HasLinuxJoystick	NO
+#endif
+
+#define LibtoolMinorVersions	YES
+
+#ifndef HasMTRRSupport
+# define HasMTRRSupport	NO
+#endif
+
+#ifndef XF86INT10_BUILD
+# define XF86INT10_BUILD	X86EMU_GENERIC
+#endif
+
+#ifndef BuildXF86DRI
+#  define BuildXF86DRI	NO
+#endif
+
+#ifndef HasAgpGart
+# define HasAgpGart	NO
+#endif
+
+#ifndef HasSSESupport
+#  if defined(i386Architecture)
+#    define HasSSESupport		YES
+#  endif
+#endif
+
+#ifndef HasMMXSupport
+#  if defined(i386Architecture)
+#    define HasMMXSupport		YES
+#  endif
+#endif
+
+#ifndef Has3DNowSupport
+#  if defined(i386Architecture)
+#    define Has3DNowSupport		YES
+#  endif
+#endif
 
 #ifndef BuildXF86RushExt
 # define BuildXF86RushExt	YES
 #endif
 
+#ifndef LinuxFBDevSupport
+# define LinuxFBDevSupport	NO
+#endif
+
+#define SourceDefines		GnuSourceDefines
+
 #define BuildLibPathVar		LD_LIBRARY_PATH
 #define GccUsesGas		YES
 #define UseGas			YES
 #define GnuCpp			YES
+#ifndef HasDlopen
 #define HasDlopen		YES
+#endif
+#define HasWeakSymbols		YES
 #ifndef HasShadowPasswd
 #  define HasShadowPasswd	YES
 #endif
 #define HasPutenv		YES
-
-XCOMM Not implemented and will always fail.
 #ifndef HasShm
-# define HasShm			YES
+#define HasShm			YES
+#endif
+
+#ifndef HasSecureRPC
+# define HasSecureRPC		YES
 #endif
 
 #define HasBSD44Sockets		YES
 #define HasSockets		YES
+#ifndef HasSnprintf
 #define HasSnprintf		YES
+#endif
 #define HasMkstemp		YES
+#ifndef HasGetresuid
+#define HasGetresuid		YES
+#endif
+#ifndef HasUsableFileMmap
 #define HasUsableFileMmap	YES
+#endif
 #ifndef HasNCurses
 #define HasNCurses		YES
 #endif
 
+#ifndef HasGroff
+#define HasGroff		YES
+#endif
+
+#ifndef HasZlib
+#define HasZlib			YES
+#endif
+
+#ifndef HasLibpng
+#define HasLibpng		YES
+#endif
+
+#ifndef HasExpat
+# define HasExpat		YES
+#endif
+
+#ifndef HasFreetype2
+# define HasFreetype2		YES
+#endif
+
 #define AvoidNullMakeCommand	YES
 #ifndef DebuggableLibraries
 #define DebuggableLibraries	NO
@@ -84,32 +396,64 @@ XCOMM Not implemented and will always fa
 #define UseStaticTermcapLib	NO
 #endif
 
+#ifndef HasCookieMaker
+#define HasCookieMaker		YES
+#define MkCookieCmd		mcookie
+#endif
+
+#ifndef BourneShell
+/*
+ * This will cause builds/installs to terminate on errors, as on other
+ * platforms.
+ */
+#define BourneShell		/bin/sh -e
+#endif
+
 #define MkdirHierCmd		mkdir -p
+
+#ifndef DoLoadableServer
+# define DoLoadableServer	YES
+#endif
+
+
 #ifndef CcCmd
 #define CcCmd			gcc
+#   define CplusplusCmd		c++
 #endif
 #ifndef AsCmd
-#define AsCmd			as
+#   define AsCmd		CcCmd -c -x assembler
 #endif
+#  define AsmDefines		-D__ELF__
 #ifndef LdCmd
-#define LdCmd			ld
+#   define LdCmd		CcCmd -nostdlib
+#  endif
+#  ifndef GnuUsesNcurses
+#    define GnuUsesNcurses	YES
 #endif
-
-#define AsmDefines		-D__ELF__
-#define CplusplusCmd		c++
 #ifndef TermcapLibrary
 #if UseStaticTermcapLib
+#    if GnuUsesNcurses
 #define TermcapLibrary		StaticLibrary(/usr/lib,ncurses)
 #else
+#     define TermcapLibrary	StaticLibrary(/usr/lib/termcap,termcap)
+#    endif
+#   else
+#    if GnuUsesNcurses
 #define TermcapLibrary		-lncurses
+#    else
+#     define TermcapLibrary	-ltermcap
 #endif
 #endif
-#ifndef DoLoadableServer
-#define DoLoadableServer	YES
 #endif
+
+/*
+ * Some older distros still need /lib/cpp.  There's probably a better
+ * test, but this handles the older Red Hat releases at least.
+ */
 #ifndef CppCmd
-#define CppCmd			/lib/cpp
+# define CppCmd		cpp
 #endif
+
 #define YaccCmd			bison -y
 #define LexCmd			flex -l
 #define HasFlex			YES
@@ -125,50 +469,68 @@ XCOMM Not implemented and will always fa
 #define StandardCppDefines	StandardDefines
 
 #define HasVarRun		YES
+#ifndef VarDbDirectory
 #define VarDbDirectory		$(VARDIR)/lib
+#endif
+
+#ifndef OSXInputDrivers
+# define OSXInputDrivers1	/**/
+# define OSXInputDrivers2	/**/
+# define OSXInputDrivers	OSXInputDrivers1 OSXInputDrivers2
+#endif
 
-XCOMM i386Architecture
-#define OptimizedCDebugFlags	DefaultGcc2i386Opt
-#define GNUMachineDefines	-D__i386__
+#define HasPlugin			YES
+#define VendorHasX11R6_3libXext	YES	/* XC or XFree86 >= 3.3.1 */
+
+#ifdef i386Architecture
+# ifndef OptimizedCDebugFlags
+#  define OptimizedCDebugFlags	DefaultGcc2DebugOpt DefaultGcc2OptimizeOpt GccAliasingArgs
+# endif
+# define GnuMachineDefines	-D__i386__
 #define ServerOSDefines		XFree86ServerOSDefines -DDDXTIME
 #define ServerExtraDefines	-DGCCUSESGAS XFree86ServerDefines
-#define HasPlugin		YES
-#define VendorHasX11R6_3libXext	YES
+#endif /* i386Architecture */
 
 #ifndef StandardDefines
-#define StandardDefines		GNUMachineDefines GNUSourceDefines
+# define StandardDefines	-DGnu GnuMachineDefines GnuSourceDefines
 #endif
 
+#ifndef DlLibrary
 #define DlLibrary		-rdynamic -ldl
+#endif
 
 #define ServerExtraSysLibs	/**/
 
-#define ConnectionFlags		-DUNIXCONN -DTCPCONN
-
-/* Some of these man page defaults are overriden in the above OS sections */
-#ifndef ManSuffix
-# define ManSuffix	1x
+#ifndef PamLibraries
+#define PamLibraries		-lpam DlLibrary
 #endif
-#ifndef ManDir
-# define ManDir		$(MANSOURCEPATH)1
-#endif
-#ifndef LibManSuffix
-# define LibManSuffix	3x
-#endif
-#ifndef LibmanDir
-# define LibmanDir	$(MANSOURCEPATH)3
+
+#ifndef PamMiscLibraries
+#define PamMiscLibraries	-lpam_misc
 #endif
-#ifndef FileManSuffix
-# define FileManSuffix	5x
+
+#ifndef HasDECnet
+#define HasDECnet		NO
 #endif
-#ifndef FileManDir
-# define FileManDir	$(MANSOURCEPATH)5
+#if HasDECnet
+# define ConnectionFlags	-DUNIXCONN -DTCPCONN -DDNETCONN
+# define ExtraLibraries		-ldnet
+#else
+#define ConnectionFlags		-DUNIXCONN -DTCPCONN
 #endif
 
 #ifndef StaticLibrary
 #define StaticLibrary(libpath,libname) -Wl,-Bstatic Concat(-L,libpath) Concat(-l,libname) -Wl,-Bdynamic
 #endif
 
+#ifndef LinkGLToUsrLib
+# define LinkGLToUsrLib		YES
+#endif
+# ifndef LinkGLToUsrInclude
+# define LinkGLToUsrInclude	YES
+#endif
+
+
 #define HasGnuMake	YES
 
 #define MakeNamedTargetSubdir(dir,flags,subname)\
@@ -176,12 +538,16 @@ XCOMM i386Architecture
 
 #define ArchitectureDefines	-DGNU_ARCHITECTURE
 
-#define XserverNeedsSetUID	YES
+#define TtClientLibs $(TTLIB) $(XTOOLLIB) $(XLIB)
+#define TtClientDepLibs $(DEPTTLIB) $(DEPXTOOLLIB) $(DEPXLIB)
 
-#include <gnuLib.rules>
-
-XCOMM XXX Might need this if they are not careful with slashes.
-XCOMM #define DirFailPrefix -
+#ifndef BuildDmx
+#define BuildDmx	YES
+#endif
 
+#include <gnuLib.rules>
 #include <xorg.cf>
 
+#ifndef XFree86ServerOSDefines
+# define XFree86ServerOSDefines
+#endif
Index: xutils-dev/xorg-cf-files/gnuLib.rules
===================================================================
--- xutils-dev.orig/xorg-cf-files/gnuLib.rules
+++ xutils-dev/xorg-cf-files/gnuLib.rules
@@ -17,7 +17,6 @@
 #define ForceNormalLib NO
 #endif
 
-XCOMM XXX To rpath or not to rpath...
 #ifndef UseRpath
 #define UseRpath NO
 #endif
@@ -26,10 +25,13 @@ XCOMM XXX To rpath or not to rpath...
 #define SharedOldX NO
 #endif
 
-#undef SpecialMalloc
+#ifndef SpecialMalloc
 #define SpecialMalloc NO
+#endif
 
+# ifndef BaseShLibReqs
 #define BaseShLibReqs   -lc
+# endif
 
 #ifndef SharedDataSeparation
 #define SharedDataSeparation NO
@@ -45,7 +47,7 @@ XCOMM XXX To rpath or not to rpath...
 #endif
 #ifndef RpathLoadFlags
 #if UseRpath
-#define RpathLoadFlags -Wl,-rpath=$(USRLIBDIRPATH)
+#define RpathLoadFlags -Wl,-rpath,$(USRLIBDIRPATH)
 #else
 #define RpathLoadFlags /**/
 #endif
@@ -54,7 +56,7 @@ XCOMM XXX To rpath or not to rpath...
 #define LibraryRpathLoadFlags RpathLoadFlags
 #endif
 #ifndef SharedLibraryLoadFlags
-#define SharedLibraryLoadFlags -shared LibraryRpathLoadFlags
+#  define SharedLibraryLoadFlags -shared
 #endif
 #ifndef PositionIndependentCFlags
 #define PositionIndependentCFlags -fPIC
@@ -63,12 +65,13 @@ XCOMM XXX To rpath or not to rpath...
 #define PositionIndependentCplusplusFlags -fPIC
 #endif
 #ifndef ExtraLoadFlags
-#ifdef UseInstalled
-XCOMM XXX Maybe superfluous.
-#define ExtraLoadFlags RpathLoadFlags -Wl,-rpath-link=$(USRLIBDIRPATH)
-#else
-#define ExtraLoadFlags RpathLoadFlags -Wl,-rpath-link=$(BUILDLIBDIR)
+#  define ExtraLoadFlags RpathLoadFlags
+# endif
+# ifndef HardCodeLibdirFlag
+#  define HardCodeLibdirFlag RpathLoadFlags
 #endif
+# if !defined(ShlibGlobalsFlags)
+#  define ShlibGlobalsFlags -Wl,-Bsymbolic
 #endif
 
 /*
@@ -87,6 +90,26 @@ install:: Concat(lib,libname.so.rev)
 	  $(LN) Concat(lib,libname.so.rev) Concat($(DESTDIR)dest/lib,libname.so)); fi
 #endif /* InstallSharedLibrary */
 
+# ifndef InstallSharedLibtoolLibrary
+
+#  define SetRevisions(rev) V=`expr rev : '\([^:]*\)'`; \		@@\
+		R=`expr rev : '.*:\([^:]*\):'`; \			@@\
+		A=`expr rev : '.*:\([^:]*\)'`; \			@@\
+		MAJ=`expr $$V - $$A`; \					@@\
+		MIN=$$A.$$R
+
+#  define	InstallSharedLibtoolLibrary(libname,rev,dest)		@@\
+install:: Concat(lib,libname.so) 					@@\
+	MakeDir($(DESTDIR)dest)						@@\
+	@set +e; SetRevisions(rev); \					@@\
+		set -xe; \						@@\
+		$(INSTALL) $(INSTALLFLAGS) $(INSTBINFLAGS) Concat(lib,libname.so.$$MAJ.$$MIN) $(DESTDIR)dest; \ @@\
+		$(RM) Concat($(DESTDIR)dest/lib,libname.so.$$MAJ); \	@@\
+		$(LN) Concat(lib,libname.so.$$MAJ.$$MIN) Concat($(DESTDIR)dest/lib,libname.so.$$MAJ); \ @@\
+		$(RM) Concat($(DESTDIR)dest/lib,libname.so); \		@@\
+		$(LN) Concat(lib,libname.so.$$MAJ.$$MIN) Concat($(DESTDIR)dest/lib,libname.so)
+# endif /* InstallSharedLibrary */
+
 /*
  * InstallSharedLibraryData - generate rules to install the shared library data
  */
@@ -106,9 +129,9 @@ AllTarget(Concat(lib,libname.so.rev))
 									@@\
 Concat(lib,libname.so.rev):  solist $(EXTRALIBRARYDEPS)			@@\
 	$(RM) $@~							@@\
-	@SONAME=`echo $@ | sed 's/\(lib[^\.]*\.so\.[0-9]*\)\(\.[0-9]*\)\{1,2\}/\1/'`; set -x; \		@@\
-		(cd down; $(CC) -o up/$@~ $(SHLIBLDFLAGS) -Wl,-soname,$$SONAME solist $(REQUIREDLIBS) BaseShLibReqs); \ @@\
-		$(RM) $$SONAME; $(LN) $@ $$SONAME; \			@@\
+	@SONAME=`echo $@ | sed 's/\(lib[^\.]*\.so\.[0-9]*\)\(\.[0-9]*\)\{1,2\}/\1/'`; (set -x; \	@@\
+		cd down; $(CC) -o up/$@~ $(SHLIBLDFLAGS) -Wl,-soname,$$SONAME solist $(REQUIREDLIBS) BaseShLibReqs) || exit 1; \ @@\
+		(set -x; $(RM) $$SONAME; $(LN) $@ $$SONAME); \		@@\
 		LinkBuildSonameLibrary($$SONAME)			@@\
 	$(RM) $@							@@\
 	$(MV) $@~ $@							@@\
@@ -125,6 +148,38 @@ clean::									@@\
 
 #endif /* SharedLibraryTarget */
 
+
+# ifndef SharedLibtoolLibraryTarget
+#  define SharedLibtoolLibraryTarget(libname,rev,solist,down,up)	@@\
+AllTarget(Concat(lib,libname.so))					@@\
+									@@\
+Concat(lib,libname.so):  solist $(EXTRALIBRARYDEPS)			@@\
+	$(RM) $@~							@@\
+	@set +e; SetRevisions(rev); set -e; \				@@\
+		SONAME=$@.$$MAJ; \					@@\
+		(set -x; \						@@\
+		$(RM) $@.$$MAJ.$$MIN~; \				@@\
+		cd down; $(CC) -o up/$@.$$MAJ.$$MIN~ $(SHLIBLDFLAGS) -Wl,-soname,$$SONAME solist $(REQUIREDLIBS) BaseShLibReqs) || exit 1; \ @@\
+		(set -x; \						@@\
+		$(RM) $$SONAME; $(LN) $@.$$MAJ.$$MIN $$SONAME); \	@@\
+		LinkBuildSonameLibrary($$SONAME); \			@@\
+		(set -x; \						@@\
+		$(RM) $@.$$MAJ.$$MIN; \					@@\
+		$(MV) $@.$$MAJ.$$MIN~ $@.$$MAJ.$$MIN; \			@@\
+	  	$(RM) $@; \						@@\
+	  	$(LN) $@.$$MAJ.$$MIN $@); \				@@\
+		LinkBuildLibraryInline($@.$$MAJ.$$MIN); \		@@\
+		LinkBuildLibraryInline($@)				@@\
+									@@\
+clean::									@@\
+	@set +e; SetRevisions(rev); \					@@\
+		set -xe; \						@@\
+		$(RM) Concat(lib,libname.so.$$MAJ); \			@@\
+		$(RM) Concat(lib,libname.so.$$MAJ.$$MIN)		@@\
+	$(RM) Concat(lib,libname.so)
+
+# endif /* SharedLibtoolLibraryTarget */
+
 /*
  * SharedDepLibraryTarget - generate rules to create a shared library.
  */
@@ -153,6 +208,34 @@ clean::									@@\
 
 #endif /* SharedDepLibraryTarget */
 
+/*
+ * SharedDepCplusplusLibraryTarget - generate rules to create a shared library.
+ */
+#ifndef SharedDepCplusplusLibraryTarget
+#define SharedDepCplusplusLibraryTarget(libname,rev,deplist,solist,down,up) @@\
+AllTarget(Concat(lib,libname.so.rev))					@@\
+									@@\
+Concat(lib,libname.so.rev):  deplist $(EXTRALIBRARYDEPS)		@@\
+	$(RM) $@~							@@\
+	@SONAME=`echo $@ | sed 's/\(lib[^\.]*\.so\.[0-9]*\)\(\.[0-9]*\)\{1,2\}/\1/'`; set -x; \		@@\
+		(cd down; $(CXX) -o up/$@~ $(SHLIBLDFLAGS) -Wl,-soname,$$SONAME solist $(REQUIREDLIBS) BaseShLibReqs); \ @@\
+		$(RM) $$SONAME; $(LN) $@ $$SONAME; \			@@\
+		LinkBuildSonameLibrary($$SONAME)			@@\
+	$(RM) $@ 							@@\
+	$(MV) $@~ $@							@@\
+	@if $(SOSYMLINK); then (set -x; \				@@\
+	  $(RM) Concat(lib,libname.so); \				@@\
+	  $(LN) $@ Concat(lib,libname.so)); fi				@@\
+	LinkBuildLibrary($@)						@@\
+	LinkBuildLibraryMaybe(Concat(lib,libname.so),$(SOSYMLINK))	@@\
+									@@\
+clean::									@@\
+	@MAJREV=`echo rev | sed 's/\([0-9]*\)\(\.[0-9]*\)\{1,2\}/\1/'`; \	@@\
+	set -x; $(RM) Concat(lib,libname.so.$$MAJREV)			@@\
+	$(RM) Concat(lib,libname.so.rev) Concat(lib,libname.so)
+
+#endif /* SharedDepCplusplusLibraryTarget */
+
 #ifndef SharedDepModuleTarget
 #define SharedDepModuleTarget(name,deps,solist)				@@\
 AllTarget(name)								@@\
@@ -168,6 +251,23 @@ clean::									@@\
 
 #endif /* SharedDepModuleTarget */
 
+# ifndef SharedDriModuleTarget
+#  define SharedDriModuleTarget(name,deps,solist)			@@\
+AllTarget(name)								@@\
+									@@\
+name: deps								@@\
+	$(RM) $@~ $@.map						@@\
+	@(echo 'DRI_MODULE { global: __dri*; local: *; };' > $@.map)	@@\
+	$(CC) -o $@~ -Wl,--version-script=$@.map $(SHLIBLDFLAGS) solist $(REQUIREDLIBS) BaseShLibReqs @@\
+	$(RM) $@ $@.map							@@\
+	$(MV) $@~ $@							@@\
+									@@\
+clean::									@@\
+	$(RM) name							@@\
+	$(RM) name.map
+
+# endif /* SharedDriModuleTarget */
+
 /*
  * SharedLibraryDataTarget - generate rules to create shlib data file;
  */
Index: xutils-dev/xorg-cf-files/gnuLib.tmpl
===================================================================
--- xutils-dev.orig/xorg-cf-files/gnuLib.tmpl
+++ xutils-dev/xorg-cf-files/gnuLib.tmpl
@@ -1,8 +1,9 @@
-XCOMM
-XCOMM  GNU Hurd shared library template
-XCOMM
-XCOMM $XFree86: xc/config/cf/gnuLib.tmpl,v 1.5 2003/05/29 21:56:57 herrb Exp $
-XCOMM
+XCOMM $Xorg: lnxLib.tmpl,v 1.3 2000/08/17 19:41:47 cpqbld Exp $
+XCOMM $XFree86: xc/config/cf/lnxLib.tmpl,v 3.19 2003/10/15 22:47:48 herrb Exp $
+
+/*
+ * GNU/Hurd shared library template
+ */
 
 #ifndef FixupLibReferences
 #define FixupLibReferences()						@@\
@@ -14,15 +15,71 @@ XMULIB = $(XMULIBONLY) $(XTOOLLIB) $(XLI
 #define XawClientLibs $(XAWLIB) $(XMULIBONLY) $(XTOOLLIB) $(XPMLIB) $(EXTRAXAWCLIENTLIBS) $(XLIB)
 #endif
 
-#define SharedX11Reqs
+#if ThreadedX
+# ifndef SharedThreadReqs
+#   define SharedThreadReqs -lpthread
+# endif
+#else
+# ifndef SharedThreadReqs
+#   define SharedThreadReqs
+# endif
+#endif
+
+#define SharedX11Reqs	  SharedThreadReqs
 #define SharedXlibi18nReqs $(LDPRELIB) $(XONLYLIB)
 #define SharedOldXReqs	$(LDPRELIB) $(XONLYLIB)
-#define SharedXtReqs	$(LDPRELIB) $(XONLYLIB) $(SMLIB) $(ICELIB)
+#define SharedXtReqs	  $(LDPRELIB) $(XONLYLIB) $(SMLIB) $(ICELIB) SharedThreadReqs
 #define SharedXaw6Reqs	$(LDPRELIB) $(XMULIB) $(XTOOLLIB) $(XLIB)
 #define SharedXaw7Reqs	$(LDPRELIB) $(XMULIB) $(XTOOLLIB) $(XPMLIB) $(XLIB)
 #define SharedXawReqs	$(LDPRELIB) $(XMULIB) $(XTOOLLIB) $(XPMLIB) $(EXTRAXAWREQS) $(XLIB)
 #define SharedXmuReqs	$(LDPRELIB) $(XTOOLLIB) $(XLIB)
+#define SharedXmuuReqs	  $(LDPRELIB) $(XONLYLIB)
 #define SharedXextReqs	$(LDPRELIB) $(XONLYLIB)
 #define SharedXiReqs	$(LDPRELIB) $(XLIB)
+#define SharedXpReqs	  $(LDPRELIB) $(XLIB)
+#define SharedXpmReqs	  $(LDPRELIB) $(XLIB)
+#define SharedXrandrReqs  $(LDPRELIB) $(XRENDERLIB) $(XLIB)
+#define SharedXrenderReqs $(LDPRELIB) $(XLIB)
 #define SharedXtstReqs	$(LDPRELIB) $(XLIB)
+#define SharedXvReqs	  $(LDPRELIB) $(XLIB)
 #define SharedSMReqs	$(LDPRELIB) $(ICELIB)
+#define SharedXcursorReqs $(LDPRELIB) $(XRENDERLIB) $(XLIB)
+#define SharedFontencReqs $(LDPRELIB) GzipLibrary
+
+#if GlxUseBuiltInDRIDriver
+#define ExtraSharedGLReqs /**/
+#else
+#define ExtraSharedGLReqs -ldl
+#endif
+#define SharedGLReqs	  $(LDPRELIB) $(XLIB) ExtraSharedGLReqs
+#ifndef SharedGLUReqs
+#define SharedGLUReqs	  $(LDPRELIB) $(XLIB) $(GLXLIB)
+#endif
+#define SharedDPSReqs	  $(LDPRELIB) $(SMLIB) $(ICELIB) $(XTOOLLIB) $(XLIB)
+#define SharedDPSTKReqs	  $(LDPRELIB) $(SMLIB) $(ICELIB) $(XTOOLLIB) $(DPSLIB) $(XLIB)
+#define SharedpsresReqs	  $(LDPRELIB) $(SMLIB) $(ICELIB) $(XTOOLLIB) $(XLIB)
+
+#define SharedXResReqs	  $(LDPRELIB) $(XLIB)
+#define SharedXfontcacheReqs	$(LDPRELIB) $(XLIB)
+#define SharedXineramaReqs	$(LDPRELIB) $(XLIB)
+#define SharedXssReqs	$(LDPRELIB) $(XLIB)
+#define SharedXvMCReqs	$(LDPRELIB) $(XLIB)
+#define SharedXxf86dgaReqs	$(LDPRELIB) $(XLIB)
+#define SharedXxf86miscReqs	$(LDPRELIB) $(XLIB)
+#define SharedXxf86vmReqs	$(LDPRELIB) $(XLIB)
+#define SharedxkbfileReqs	$(LDPRELIB) $(XONLYLIB)
+#define SharedxkbuiReqs $(LDPRELIB) $(XKBFILELIB) $(XONLYLIB)
+
+#ifndef SharedXReqs
+# define SharedXReqs $(XTOOLLIB) $(XPLIB) $(XLIB) $(LDPOSTLIBS) SharedThreadReqs
+#endif
+
+#ifndef SharedXmReqs
+# define SharedXmReqs $(LDPRELIBS) SharedXReqs -lc
+#endif
+
+#ifndef SharedTtReqs
+# define SharedTtReqs $(LDPRELIBS) SharedXReqs $(CXXLIB) SharedThreadReqs
+#endif
+
+#define NoMessageCatalog
Index: xutils-dev/xorg-cf-files/Imake.rules
===================================================================
--- xutils-dev.orig/xorg-cf-files/Imake.rules
+++ xutils-dev/xorg-cf-files/Imake.rules
@@ -2722,6 +2722,20 @@ LibraryTargetName(libname): objlist1 obj
 
 
 /*
+ * Rule to build libXXX_pic.a from PIC objects for a library
+ */
+#ifndef PicLibraryTarget
+#define	PicLibraryTarget(libname,objlist)				@@\
+AllTarget(LibraryTargetNameSuffix(libname,_pic))			@@\
+									@@\
+LibraryTargetNameSuffix(libname,_pic): objlist $(EXTRALIBRARYDEPS)	@@\
+	RemoveFile($@)							@@\
+	MakeLibrary($@,objlist)						@@\
+	RanLibrary($@)							@@\
+	_LinkBuildLibrary($@)
+#endif /* PicLibraryTarget */
+
+/*
  * SubdirLibraryRule -
  */
 #ifndef SubdirLibraryRule
Index: xutils-dev/xorg-cf-files/Imake.tmpl
===================================================================
--- xutils-dev.orig/xorg-cf-files/Imake.tmpl
+++ xutils-dev/xorg-cf-files/Imake.tmpl
@@ -902,37 +902,82 @@ TCLIBDIR = TclLibDir
 #define ManSourcePath $(MANPATH)/man
 #endif
 #ifndef ManDir
-#define ManDir $(MANSOURCEPATH)$(MANSUFFIX)
+#define ManDir $(MANSOURCEPATH)$(MANSECT)
 #endif
-#ifndef LibmanDir
-#define LibmanDir $(MANSOURCEPATH)$(LIBMANSUFFIX)
+#ifndef ManSuffix
+#define ManSuffix $(MANSECT)$(PROJECTMANSUFFIX)
+#endif
+#ifndef SyscallManDir
+#define SyscallManDir $(MANSOURCEPATH)$(SYSCALLMANSECT)
+#endif
+#ifndef SyscallManSuffix
+#define SyscallManSuffix $(SYSCALLMANSECT)$(PROJECTMANSUFFIX)
+#endif
+#ifndef LibManDir
+#define LibManDir $(MANSOURCEPATH)$(LIBMANSECT)
+#endif
+#ifndef LibManSuffix
+#define LibManSuffix $(LIBMANSECT)$(PROJECTMANSUFFIX)
+#endif
+#ifndef DriverManDir
+#define DriverManDir $(MANSOURCEPATH)$(DRIVERMANSECT)
+#endif
+#ifndef DriverManSuffix
+#define DriverManSuffix $(DRIVERMANSECT)$(PROJECTMANSUFFIX)
 #endif
 #ifndef FileManDir
-#define FileManDir $(MANSOURCEPATH)$(FILEMANSUFFIX)
+#define FileManDir $(MANSOURCEPATH)$(FILEMANSECT)
+#endif
+#ifndef FileManSuffix
+#define FileManSuffix $(FILEMANSECT)$(PROJECTMANSUFFIX)
+#endif
+#ifndef GameManDir
+#define GameManDir $(MANSOURCEPATH)$(GAMEMANSECT)
+#endif
+#ifndef GameManSuffix
+#define GameManSuffix $(GAMEMANSECT)$(PROJECTMANSUFFIX)
 #endif
 #ifndef MiscManDir
-#define MiscManDir $(MANSOURCEPATH)$(MISCMANSUFFIX)
+#define MiscManDir $(MANSOURCEPATH)$(MISCMANSECT)
 #endif
-#ifndef DriverManDir
-#define DriverManDir $(MANSOURCEPATH)$(DRIVERMANSUFFIX)
+#ifndef MiscManSuffix
+#define MiscManSuffix $(MISCMANSECT)$(PROJECTMANSUFFIX)
 #endif
-#ifndef AppLoadDefs
-#define AppLoadDefs -D__apploaddir__=$(XAPPLOADDIR)
+#ifndef AdmManDir
+#define AdmManDir $(MANSOURCEPATH)$(ADMMANSECT)
+#endif
+#ifndef AdmManSuffix
+#define AdmManSuffix $(ADMMANSECT)$(PROJECTMANSUFFIX)
 #endif
 #ifndef AppManDefs
-#define AppManDefs -D__appmansuffix__=$(MANSUFFIX)
+#define AppManDefs -D__appmansuffix__=$(MANSECT)$(PROJECTMANSUFFIX)
+#endif
+#ifndef ManDefs
+#define ManDefs -D__mansuffix__=$(MANSECT)$(PROJECTMANSUFFIX) -D__osmansuffix__=$(MANSECT)
+#endif
+#ifndef SyscallManDefs
+#define SyscallManDefs -D__syscallmansuffix__=$(SYSCALLMANSECT)$(PROJECTMANSUFFIX) -D__ossysmansuffix__=$(SYSCALLMANSECT)
 #endif
 #ifndef LibManDefs
-#define LibManDefs -D__libmansuffix__=$(LIBMANSUFFIX)
+#define LibManDefs -D__libmansuffix__=$(LIBMANSECT)$(PROJECTMANSUFFIX) -D__oslibmansuffix__=$(LIBMANSECT)
+#endif
+#ifndef DriverManDefs
+#define DriverManDefs -D__drivermansuffix__=$(DRIVERMANSECT)$(PROJECTMANSUFFIX) -D__osdrivermansuffix__=$(DRIVERMANSECT)
 #endif
 #ifndef FileManDefs
-#define FileManDefs -D__filemansuffix__=$(FILEMANSUFFIX)
+#define FileManDefs -D__filemansuffix__=$(FILEMANSECT)$(PROJECTMANSUFFIX) -D__osfilemansuffix__=$(FILEMANSECT)
+#endif
+#ifndef GameManDefs
+#define GameManDefs -D__gamemansuffix__=$(GAMEMANSECT)$(PROJECTMANSUFFIX) -D__osgamemansuffix__=$(GAMEMANSECT)
 #endif
 #ifndef MiscManDefs
-#define MiscManDefs -D__miscmansuffix__=$(MISCMANSUFFIX)
+#define MiscManDefs -D__miscmansuffix__=$(MISCMANSECT)$(PROJECTMANSUFFIX) -D__osmiscmansuffix__=$(MISCMANSECT)
 #endif
-#ifndef DriverManDefs
-#define DriverManDefs -D__drivermansuffix__=$(DRIVERMANSUFFIX)
+#ifndef AdmManDefs
+#define AdmManDefs -D__admmansuffix__=$(ADMMANSECT)$(PROJECTMANSUFFIX) -D__osadmmansuffix__=$(ADMMANSECT)
+#endif
+#ifndef AppLoadDefs
+#define AppLoadDefs -D__apploaddir__=$(XAPPLOADDIR)
 #endif
 #ifndef AdminManDefs
 #define AdminManDefs -D__adminmansuffix__=$(ADMINMANSUFFIX)
@@ -963,6 +1008,21 @@ TCLIBDIR = TclLibDir
 #endif
 #endif
 
+#ifndef IconDir
+# if !NothingOutsideProjectRoot
+#  define IconDir       "/usr/share/icons"
+#  ifndef XcursorPath
+#   define XcursorPath  "~/.icons:/usr/share/icons:/usr/share/pixmaps"
+#  endif
+# else
+#  define IconDir       Concat(ProjectRoot,/lib/X11/icons)
+# endif
+#endif
+
+#ifndef XcursorPath
+# define XcursorPath    Concat(~/.icons:/usr/share/icons:/usr/share/pixmaps:,IconDir)
+#endif
+
 #ifndef LogDirectory
 #if HasVarDirectory
 #define LogDirectory $(VARDIR)/log
@@ -1609,32 +1669,50 @@ TCLIBDIR = TclLibDir
 #ifndef SHsuf
 #define SHsuf sh
 #endif
-#ifndef ManSuffix
-#define ManSuffix	n		/* use just one tab or cpp will die */
+#ifndef ManSection
+#define ManSection	1		/* use just one tab or cpp will die */
 #endif
-#ifndef LibManSuffix
-#define LibManSuffix	3		/* use just one tab or cpp will die */
+#ifndef SyscallManSection
+#define SyscallManSection	2	/* use just one tab or cpp will die */
 #endif
-#ifndef FileManSuffix
+#ifndef LibManSection
+#define LibManSection	3		/* use just one tab or cpp will die */
+#endif
+#ifndef DriverManSection
 #if SystemV || SystemV4 || defined(OSF1Architecture)
-#define FileManSuffix	4		/* use just one tab or cpp will die */
+#define DriverManSection	7	/* use just one tab or cpp will die */
 #else
-#define FileManSuffix	5		/* use just one tab or cpp will die */
+#define DriverManSection	4	/* use just one tab or cpp will die */
 #endif
 #endif
-#ifndef MiscManSuffix
+#ifndef FileManSection
 #if SystemV || SystemV4 || defined(OSF1Architecture)
-#define MiscManSuffix	5		/* use just one tab or cpp will die */
+#define FileManSection	4		/* use just one tab or cpp will die */
 #else
-#define MiscManSuffix	7		/* use just one tab or cpp will die */
+#define FileManSection	5		/* use just one tab or cpp will die */
 #endif
 #endif
-#ifndef DriverManSuffix
+#ifndef GameManSection
+#define GameManSection	6		/* use just one tab or cpp will die */
+#endif
+#ifndef MiscManSection
 #if SystemV || SystemV4 || defined(OSF1Architecture)
-#define DriverManSuffix	7		/* use just one tab or cpp will die */
+#define MiscManSection	5		/* use just one tab or cpp will die */
 #else
-#define DriverManSuffix	4		/* use just one tab or cpp will die */
+#define MiscManSection	7		/* use just one tab or cpp will die */
+#endif
+#endif
+#ifndef AdmManSection
+#define AdmManSection	8		/* use just one tab or cpp will die */
 #endif
+#ifndef ManSrcSection
+#define ManSrcSection	s		/* use just one tab or cpp will die */
+#endif
+#ifndef ManNewSection
+#define ManNewSection	n		/* use just one tab or cpp will die */
+#endif
+#ifndef ProjectManSuffix
+#define ProjectManSuffix	X11	/* use just one tab or cpp will die */
 #endif
 #ifndef AdminManSuffix
 #if SystemV || SystemV4
@@ -1763,9 +1841,15 @@ TCLIBDIR = TclLibDir
           MANPATH = ManPath		/* top of manual page tree */
     MANSOURCEPATH = ManSourcePath	/* prefix for man page sources */
            MANDIR = ManDir		/* man pages for commands */
-        LIBMANDIR = LibmanDir		/* man pages for library routines */
+    SYSCALLMANDIR = SyscallManDir   	/* man pages for system calls */
+        LIBMANDIR = LibManDir		/* man pages for library routines */
+     DRIVERMANDIR = DriverManDir	/* man pages for drivers */
        FILEMANDIR = FileManDir		/* man pages for config files */
+       GAMEMANDIR = GameManDir		/* man pages for game commands */
        MISCMANDIR = MiscManDir		/* man pages for miscellaneous files */
+        ADMMANDIR = AdmManDir		/* man pages for administrative commands */
+	  ICONDIR = IconDir		/* Xcursor cursors/icon directory */
+      XCURSORPATH = XcursorPath		/* Xcursor cursors path */
      DRIVERMANDIR = DriverManDir	/* man pages for drivers */
      LOGDIRECTORY = LogDirectory	/* OS location of log files */
 #ifdef VarRunDirectory
@@ -1872,15 +1956,30 @@ CXXEXTRA_INCLUDES = CplusplusExtraInclud
              PERL = PerlCmd
          PERLOPTS = PerlOptions
      PERLENVSETUP = PerlEnvSetup
+
+          MANSECT = ManSection		/* section for command man pages */
+   SYSCALLMANSECT = SyscallManSection	/* section for system call man pages */
+       LIBMANSECT = LibManSection	/* section for library man pages */
+    DRIVERMANSECT = DriverManSection	/* section for driver man pages */
+      FILEMANSECT = FileManSection	/* section for file format man pages */
+      GAMEMANSECT = GameManSection	/* section for game command man pages */
+      MISCMANSECT = MiscManSection	/* section for misc man pages */
+       ADMMANSECT = AdmManSection	/* section for administrative command man pages */
+       MANSRCSECT = ManSrcSection	/* section for man page source */
+       MANNEWSECT = ManNewSection	/* section for preprocessed man source */
+ PROJECTMANSUFFIX = ProjectManSuffix	/* man page suffix for the X11 distribution */
         MANSUFFIX = ManSuffix		/* suffix for command man pages */
+ SYSCALLMANSUFFIX = SyscallManSuffix	/* suffix for system call man pages */
      LIBMANSUFFIX = LibManSuffix	/* suffix for library man pages */
+  DRIVERMANSUFFIX = DriverManSuffix	/* suffix for driver man pages */
     FILEMANSUFFIX = FileManSuffix	/* suffix for file format man pages */
+    GAMEMANSUFFIX = GameManSuffix	/* suffix for game command man pages */
     MISCMANSUFFIX = MiscManSuffix	/* suffix for misc man pages */
-  DRIVERMANSUFFIX = DriverManSuffix	/* suffix for driver man pages */
+     ADMMANSUFFIX = AdmManSuffix	/* suffix for administrative command man pages */
    ADMINMANSUFFIX = AdminManSuffix	/* suffix for admin command man pages */
      MANSRCSUFFIX = ManSrcSuffix	/* suffix for man page source */
      MANNEWSUFFIX = ManNewSuffix	/* suffix for preprocessed man source */
-          MANDEFS = AppLoadDefs AppManDefs FileManDefs LibManDefs MiscManDefs DriverManDefs AdminManDefs ProjectManDefs XConfigFileManDefs XConfigDirManDefs XLogFileManDefs XServerManDefs $(XORGMANDEFS) $(VENDORMANDEFS)
+          MANDEFS = AppLoadDefs FileManDefs LibManDefs ManDefs SyscallManDefs GameManDefs MiscManDefs AdmManDefs MiscManDefs DriverManDefs AdminManDefs ProjectManDefs XConfigFileManDefs XConfigDirManDefs XLogFileManDefs XServerManDefs AppManDefs $(XORGMANDEFS) $(VENDORMANDEFS)
 
    COMPRESSMANCMD = CompressManCmd
 
Index: xutils-dev/xorg-cf-files/Library.tmpl
===================================================================
--- xutils-dev.orig/xorg-cf-files/Library.tmpl
+++ xutils-dev/xorg-cf-files/Library.tmpl
@@ -37,6 +37,14 @@ XCOMM $XFree86: xc/config/cf/Library.tmp
 #define DoExtraLib NO
 #endif
 
+#ifndef DoPicLib
+# if DoNormalLib && !DoSharedLib && !StaticNeedsPicForShared
+#  define DoPicLib YES
+# else
+#  define DoPicLib NO
+# endif
+#endif
+
 #ifndef HasSharedData
 #define HasSharedData NO
 #endif
@@ -53,7 +61,7 @@ SHLIBGLOBALSFLAGS = ShlibGlobalsFlags
 #endif
 
 #ifndef LibraryCplusplusOptions
-# if DoSharedLib && defined(SharedLibraryCplusplusOptions)
+# if (DoSharedLib || DoPicLib) && defined(SharedLibraryCplusplusOptions)
 #  define LibraryCplusplusOptions SharedLibraryCplusplusOptions
 # else
 #  define LibraryCplusplusOptions DefaultCplusplusOptions
@@ -85,14 +93,14 @@ SHLIBGLOBALSFLAGS = ShlibGlobalsFlags
 
 #ifndef CplusplusSource
 # ifndef LibraryCcCmd
-#  if DoSharedLib && defined(SharedLibraryCcCmd)
+#  if (DoSharedLib || DoPicLib) && defined(SharedLibraryCcCmd)
 #   define LibraryCcCmd SharedLibraryCcCmd
 #  else
 #   define LibraryCcCmd CcCmd
 #  endif
 # endif
 # ifndef LibraryCCOptions
-#  if DoSharedLib && defined(SharedLibraryCCOptions)
+#  if (DoSharedLib || DoPicLib) && defined(SharedLibraryCCOptions)
 #   define LibraryCCOptions SharedLibraryCCOptions
 #  else
 #   define LibraryCCOptions DefaultCCOptions
@@ -121,7 +129,7 @@ CLIBDEBUGFLAGS = LibraryDebugOpt
 #else
 # ifndef CrossCompileDir
 #  ifndef LibraryCplusplusCmd
-#   if DoSharedLib && defined(SharedLibraryCplusplusCmd)
+#   if (DoSharedLib || DoPicLib) && defined(SharedLibraryCplusplusCmd)
 #    define LibraryCplusplusCmd SharedLibraryCplusplusCmd
 #   else
 #    define LibraryCplusplusCmd CplusplusCmd
@@ -129,7 +137,7 @@ CLIBDEBUGFLAGS = LibraryDebugOpt
 #  endif
 # else
 #  ifndef CrossLibraryCplusplusCmd
-#   if DoSharedLib && defined(SharedLibraryCplusplusCmd)
+#   if (DoSharedLib || DoPicLib) && defined(SharedLibraryCplusplusCmd)
 #    define CrossLibraryCplusplusCmd Concat3(CrossCompileDir,/,StripPath(SharedLibraryCplusplusCmd))
 #   else
 #    define CrossLibraryCplusplusCmd Concat3(CrossCompileDir,/,StripPath(CplusplusCmd))
@@ -138,7 +146,7 @@ CLIBDEBUGFLAGS = LibraryDebugOpt
 # endif
 
 # ifndef LibraryCplusplusOptions
-#  if DoSharedLib && defined(SharedLibraryCplusplusOptions)
+#  if (DoSharedLib || DoPicLib) && defined(SharedLibraryCplusplusOptions)
 #   define LibraryCplusplusOptions SharedLibraryCplusplusOptions
 #  else
 #   define LibraryCplusplusOptions DefaultCplusplusOptions
@@ -204,14 +212,14 @@ LIB_MT_DEFINES = LibraryMTDefines
 # define _NormalObjCplusplusCompile(options) $(_NULLCMD_)
 # define _NormalCleanDir() $(_NULLCMD_)
 #else
-# if DoSharedLib && SeparateSharedCompile
+# if (DoSharedLib || DoPicLib) && SeparateSharedCompile
 #  define _NormalLibMkdir() _LibMkdir(unshared)
 #  define _NormalObjCompile(options) UnsharedLibObjCompile(options)
 #  define _NormalObjCplusplusCompile(options) UnsharedLibObjCplusplusCompile(options)
 #  define _NormalCleanDir() LibCleanDir(unshared)
 # else
 #  define _NormalLibMkdir() $(_NULLCMD_)
-#  if !DoSharedLib &&  defined(IncludeSharedObjectInNormalLib)
+#  if (!DoSharedLib && !DoPicLib) &&  defined(IncludeSharedObjectInNormalLib)
 #   define _NormalObjCompile(options) NormalRelocLibObjCompile(options)
 #  else
 #   define _NormalObjCompile(options) NormalLibObjCompile(options)
@@ -221,7 +229,7 @@ LIB_MT_DEFINES = LibraryMTDefines
 # endif
 #endif
 
-#if !DoSharedLib || (DoNormalLib && !SeparateSharedCompile)
+#if !DoSharedLib && !DoPicLib || (DoNormalLib && !SeparateSharedCompile)
 # define _SharedObjCompile(options) $(_NULLCMD_)
 # define _SharedObjCplusplusCompile(options) $(_NULLCMD_)
 #else
@@ -517,6 +525,12 @@ DebuggedLibraryTarget($(LIBNAME),$(OBJS)
 InstallLibrary($(LIBNAME)_d,$(USRLIBDIR))
 #   endif
 #  endif
+#  if DoPicLib
+PicLibraryTarget($(LIBNAME),$(OBJS))
+#   if !defined(LibInstall) || LibInstall
+InstallLibrary($(LIBNAME)_pic,$(USRLIBDIR))
+#   endif
+#  endif
 #  if DoExtraLib && defined(ExtraLibRules)
 ExtraLibraryTarget($(LIBNAME),$(SoRev),$(OBJS) $(EXTRAUNSHAREDOBJS))
 #   if !defined(LibInstall) || LibInstall
Index: xutils-dev/xorg-cf-files/linux.cf
===================================================================
--- xutils-dev.orig/xorg-cf-files/linux.cf
+++ xutils-dev/xorg-cf-files/linux.cf
@@ -107,31 +107,48 @@ XCOMM operating system:  OSName (OSMajor
 XCOMM libc:	(LinuxCLibMajorVersion./**/LinuxCLibMinorVersion./**/LinuxCLibTeenyVersion)
 XCOMM binutils:	(LinuxBinUtilsMajorVersion)
 
+/* Linux >= 2.0 supports /dev/random, which is poll()able */
+#if OSMajorVersion >= 2
+# define HasDevRandom		YES
+# define PollDevRandom		YES
+#endif
+
 #if LinuxDistribution == LinuxDebian
+
+/* Path setup. */
+# ifndef NothingOutsideProjectRoot
+#  define NothingOutsideProjectRoot	NO
+# endif
 # if !defined(NothingOutsideProjectRoot) || !NothingOutsideProjectRoot
 #  define SystemManDirectory	/usr/share/man
 # endif
+
+/* Default GCC options. */
+# ifndef DefaultGcc2OptimizeOpt
+#  define DefaultGcc2OptimizeOpt	-O2
+# endif
+
 # define HasPam			YES
-/* un-comment this when it is un-broken */
-/* # define JoystickSupport YES */
+
 # if !defined(NothingOutsideProjectRoot) || !NothingOutsideProjectRoot
 #  define XAppLoadDir		EtcX11Directory/app-defaults
 # define XFileSearchPathDefault	Concat4(EtcX11Directory/%L/%T/%N%C,%S:EtcX11Directory/%l/%T/%N%C,%S:EtcX11Directory/%T/%N%C,%S:EtcX11Directory/%L/%T/%N%S:EtcX11Directory/%l/%T/%N%S:EtcX11Directory/%T/%N%S):Concat4($(LIBDIR)/%L/%T/%N%C,%S:$(LIBDIR)/%l/%T/%N%C,%S:$(LIBDIR)/%T/%N%C,%S:$(LIBDIR)/%L/%T/%N%S:$(LIBDIR)/%l/%T/%N%S:$(LIBDIR)/%T/%N%S)
 /* the relative symlink created by this rule causes problems for us */
-# if defined(InstallAppDefFiles) && InstallAppDefFiles
-#  define InstallAppDefaultsLong(file,class)				@@\
+# define InstallAppDefFiles	YES
+# ifdef InstallAppDefFiles
+#  if InstallAppDefFiles
+#   define InstallAppDefaultsLong(file,class)				@@\
 InstallNamedTarget(install,file.ad,$(INSTAPPFLAGS),$(XAPPLOADDIR),class)
-# else
-#  define InstallAppDefaultsLong(file,class)				@@\
+#  else
+#   define InstallAppDefaultsLong(file,class)				@@\
 InstallNamedTargetNoClobber(install,file.ad,$(INSTAPPFLAGS),$(XAPPLOADDIR),class)
+#  endif
 # endif /* InstallAppDefFiles */
 # endif /* !defined(NothingOutsideProjectRoot) || !NothingOutsideProjectRoot */
+
 # define SharedLibXdmGreet	NO
 # define LinkGLToUsrInclude	NO
 # define LinkGLToUsrLib		NO
-# define SharedLibFont		NO
-# define SharedLibGlu		YES
-# define NormalLibGlu		YES
 # define FSUseSyslog		YES
 
 # define BuildLibGlxWithoutPIC  NO
@@ -142,15 +159,12 @@ InstallNamedTargetNoClobber(install,file
 # define MiscManSuffix         7x
 # define MiscManDir            $(MANSOURCEPATH)7
 
-/*
- *
- *
+# define ProjectManSuffix	x
+
 # define DebianMaintainer	YES
- *
- *
- */
 
 # ifdef DebianMaintainer
+
 #  ifndef XorgCustomVersion
 #   define XorgCustomVersion		"Debian"
 #  endif
@@ -158,47 +172,138 @@ InstallNamedTargetNoClobber(install,file
 #   define BuilderEMailAddr		"debian-x@lists.debian.org"
 #  endif
 #  define XFree86Devel			YES
-#  define BuildAllSpecsDocs		YES
 #  define InstallXinitConfig		YES
 #  define InstallXdmConfig		YES
 #  define InstallFSConfig		YES
 #  define DebuggableLibraries		YES
 #  define ForceNormalLib		YES
-#  define BuildSpecsDocs		YES
-#  define SpecsDocDirs			CTEXT GL ICCCM X11 Xext Xv i18n xterm
-#  define BuildRman			NO
+#  define SharedLibFont			NO
+#  define SharedLibGlu			YES
+#  define NormalLibGlu			YES
+#  define BuildFontCache		YES
+
+#  define BuildAllSpecsDocs		NO
+#  ifndef BuildSpecsDocs
+#   define BuildSpecsDocs		YES
+#  endif
+#  define SpecsDocDirs			BDF CTEXT FSProtocol GL ICCCM ICE PM \
+				Render SM X11 XDMCP XIM XLFD XProtocol \
+				Xaw Xext Xi Xmu Xserver Xt Xv XvMC \
+				i18n rstart xfs xterm xtrans
+#  define XkbServerDefines		-DNOISY
 #  define BuildHtmlManPages		NO
+#  define DontPreprocessManPages	NO
 
-/* we build-depend on libfreetype6-dev (FreeType 2.x) */
-#   define BuildFreetype2Library	NO
-#   define HasFreetype2			YES
-/* m68k has no 2.4 kernel yet */
-#   ifndef Mc68020Architecture
-#    define HasLinuxInput		YES
-#   endif
-#   define HasXdmAuth			YES
-#   define HasLatex			YES
-/* at present, glide2 only works on i386 */
-#   if defined(__linux__) && defined(i386Architecture)
-#    define HasGlide2			YES
-#    define Glide2IncDir		/usr/include/glide
-#   endif /* __linux__ && i386Architecture */
-/* at present, glide3 only works on alpha, amd64, i386 and ia64 */
-#   if defined(__linux__) && (defined(i386Architecture) || \
-       defined(AlphaArchitecture) || defined(ia64Architecture) || \
-       defined(AMD64Architecture))
-#    define HasGlide3			YES
-#    define Glide3IncDir		/usr/include/glide3
-#   endif /* __linux__ && (i386Architecture || AlphaArchitecture || \
-             ia64Architecture || AMD64Architecture) */
-/* extended instruction set support */
-#   ifdef i386Architecture
+/* We have a local patch that defines PicLibraryTarget(), instead of
+ * upstream's method. */
+#  define StaticNeedsPicForShared	NO
+
+/* Some libraries changed compilation method from static to shared and they do not
+ * ship header files.
+ * Temporary revert the build to static until all the changes will be investigated
+ * properly
+ */
+#  define SharedLibXau			YES
+#  define SharedLibXdmcp		YES
+#  define NormalOldX			NO
+#  define DebugOldX			NO
+#  define SharedOldX			NO
+#  define SharedLibDmx			YES
+#  define SharedLibGLw			NO
+#  define SharedLibFont			NO
+#  define SharedLibFontEnc		NO
+#  define SharedLibXfontcache		NO
+#  define BuildFreetypeDlModule		NO
+
+#  define KernelVersionInBanner		YES
+
+/* We don't want XRX. */
+#  define BuildPlugin		NO
+
+/* The TT/Meltho fonts are all under non-free licences, except for BitStream
+ * Vera, which is provided by an external package anyway. */
+#  define BuildTrueTypeFonts		NO
+#  define BuildBethMarduthoFonts	NO
+
+#  define MakeDllModules		YES
+
+/* All the following are fulfilled by Build-Depends. */
+#  define BuildRman			NO
+#  define RmanCmd			/usr/bin/rman
+#  define RmanOptions			/**/
+#  define HasTk				YES
+#  define TkLibDir			/usr/lib
+#  define TkIncDir			/usr/include
+#  define TkLibName			tk8.4
+#  define XF86SetupUsesStaticTk		NO
+#  define HasTcl			YES
+#  define TclLibDir			/usr/lib
+#  define TclIncDir			/usr/include
+#  define TclLibName			tcl8.4
+#  define XF86SetupUsesStaticTcl	NO
+#  define BuildExpatLibrary		NO
+#  define HasExpat			YES
+#  define ExpatIncDirStandard		YES
+#  define BuildFontconfigLibrary	NO
+#  define HasFontconfig			YES
+#  define UseFontconfig			YES
+#  define BuildRenderLibrary		NO
+#  define HasRenderLibrary		YES
+#  define BuildXcursorLibrary		NO
+#  define HasXcursorLibrary		YES
+/* Xlib needs to know the available version of Xcursor. */
+#  define SharedXcursorRev		1
+#  define BuildXftLibrary		NO
+#  define HasXftLibrary			YES
+#  define BuildFreetype2Library		NO
+#  define HasFreetype2			YES
+#  define FontLibSharedFreeType		NO
+#  define HasLatex			YES
+
+/*
+ * The XFree86 Debian package no longer builds the Xprt server, in favor
+ * of a separate xprint package.  We do still build the library,
+ * but only when we're not building servers only, so it gets built in the
+ * normal build, but not -dbg (as it requires libXt headers).
+ */
+#  define XprtServer			NO
+#  define BuildXprintLib		!BuildServersOnly
+
+#  ifdef UseInstalled
+#   define RenderLibraryDir		/usr
+#   define XftLibraryDir		/usr
+#   define XcursorLibraryDir		/usr
+#  else /* We have a special hack to find these. */
+#   define RenderLibraryDir		$(TOP)/../../imports
+#   define XftLibraryDir		$(TOP)/../../imports
+#   define XcursorLibraryDir		$(TOP)/../../imports
+#  endif
+
+/* Wraphelp.c magic is supplied via a custom patch. */
+#  define HasXdmAuth			YES
+
+#  ifdef LinuxArchitecture
+/* Glide2 only works on i386. */
+#  ifdef i386Architecture
+#   define HasGlide2			YES
+#   define Glide2IncDir		/usr/include/glide
+#  endif /* i386Architecture */
+/* Glide3 only works on alpha, amd64, i386 and ia64. */
+#  if defined(i386Architecture) || defined(AlphaArchitecture) || \
+      defined(ia64Architecture) || defined(AMD64Architecture)
+#   define HasGlide3			YES
+#   define Glide3IncDir		/usr/include/glide3
+#  endif /* i386Architecture || AlphaArchitecture || ia64Architecture ||
+            AMD64Architecture */
+#  endif /* LinuxArchitecture */
+
+/* Enable extended instruction set support. */
+#  ifdef i386Architecture
 #   define HasX86Support		YES
-#    define HasMMXSupport		YES
-#    define Has3DNowSupport		YES
-/* 2.4 is not yet the official (or predominant) kernel in unstable */
-#    define HasSSESupport		NO
-#   endif /* i386Architecture */
+#   define HasMMXSupport       YES
+#   define Has3DNowSupport     YES
+#   define HasSSESupport       YES
+#  endif /* i386Architecture */
 # endif /* DebianMaintainer */
 #endif /* LinuxDebian */
 
@@ -591,11 +696,7 @@ InstallNamedTargetNoClobber(install,file
 #define MkdirHierCmd		mkdir -p
 
 #ifndef HaveLib64
-# if defined (AMD64Architecture) || defined (s390xArchitecture) || defined (Ppc64Architecture)
-#  define HaveLib64	YES
-# else
 #  define HaveLib64	NO
-# endif
 #endif
 
 #if UseElfFormat
@@ -725,10 +826,10 @@ InstallNamedTargetNoClobber(install,file
 # if UseElfFormat
 #  ifdef AlphaArchitecture
 #   define CcCmd		gcc -b alpha-linux
+#   define CplusplusCmd		c++ -b alpha-linux
 #   define AsCmd		CcCmd -c -x assembler
 #   define LdCmd		CcCmd -nostdlib -Wl,"-m elf64alpha"
 #   define AsmDefines		-D__ELF__
-#   define CplusplusCmd		c++ -b alpha-linux
 #  endif /* AlphaArchitecture */
 #  ifdef HPArchitecture
 #   define CcCmd		gcc
@@ -740,10 +841,10 @@ InstallNamedTargetNoClobber(install,file
 #  endif /* HPArchitecture */
 #  ifdef i386Architecture
 #   define CcCmd		gcc -b i486-linux
+#   define CplusplusCmd		c++ -b i486-linux
 #   define AsCmd		CcCmd -c -x assembler
 #   define LdCmd		CcCmd -nosdlib -Wl,"-m elf_i386"
 #   define AsmDefines		-D__ELF__
-#   define CplusplusCmd		c++ -b i486-linux
 #  endif /* i386Architecture */
 #  ifdef ia64Architecture
 #   define CcCmd		gcc
@@ -754,10 +855,10 @@ InstallNamedTargetNoClobber(install,file
 #  endif /* ia64Architecture */
 #  ifdef Mc68020Architecture
 #   define CcCmd		gcc -b m68k-linux
+#   define CplusplusCmd		c++ -b m68k-linux
 #   define AsCmd		CcCmd -c -x assembler
 #   define LdCmd		CcCmd -nostdlib -W,l"-m m68kelf"
 #   define AsmDefines		-D__ELF__
-#   define CplusplusCmd		c++ -b m68k-linux
 #  endif /* Mc68020Architecture */
 #  ifdef AMD64Architecture
 #   define CcCmd                 gcc
@@ -770,7 +871,7 @@ InstallNamedTargetNoClobber(install,file
 #  define CcCmd			gcc
 #  define AsCmd			CcCmd -c -x assembler
 #  define LdCmd			CcCmd -nostdlib
-#  define AsmDefines		-DUSE_GAS
+#  define AsmDefines		-DUSE_GAS -D__ELF__
 # endif
 #endif
 
@@ -837,7 +938,7 @@ InstallNamedTargetNoClobber(install,file
 
 #ifdef AlphaArchitecture
 # ifndef OptimizedCDebugFlags
-#  define OptimizedCDebugFlags	DefaultGcc2AxpOpt
+#  define OptimizedCDebugFlags	DefaultGcc2DebugOpt DefaultGcc2OptimizeOpt GccAliasingArgs
 # endif
 # define LinuxMachineDefines	-D__alpha__
 # define ServerOSDefines	XFree86ServerOSDefines -DDDXTIME
@@ -850,7 +951,7 @@ InstallNamedTargetNoClobber(install,file
 #ifdef Arm32Architecture
 # define DefaultCCOptions
 # ifndef OptimizedCDebugFlags
-#  define OptimizedCDebugFlags	-O3
+#  define OptimizedCDebugFlags	DefaultGcc2DebugOpt DefaultGcc2OptimizeOpt GccAliasingArgs
 # endif
 # define LinuxMachineDefines	-D__arm__ -D__arm32__ -U__arm -Uarm
 # define ServerOSDefines	XFree86ServerOSDefines -DDDXTIME
@@ -858,7 +959,9 @@ InstallNamedTargetNoClobber(install,file
 #endif /* Arm32Achitecture */
 
 #ifdef HPArchitecture
-# define OptimizedCDebugFlags	-O2  GccAliasingArgs
+# ifndef OptimizedCDebugFlags
+#  define OptimizedCDebugFlags	DefaultGcc2DebugOpt DefaultGcc2OptimizeOpt GccAliasingArgs
+# endif
 # define LinuxMachineDefines	-D__hppa__
 # define ServerOSDefines	XFree86ServerOSDefines -DDDXTIME
 # define ServerExtraDefines	-DGCCUSESGAS XFree86ServerDefines
@@ -866,7 +969,7 @@ InstallNamedTargetNoClobber(install,file
 
 #ifdef i386Architecture
 # ifndef OptimizedCDebugFlags
-#  define OptimizedCDebugFlags	DefaultGcc2i386Opt
+#  define OptimizedCDebugFlags	DefaultGcc2DebugOpt DefaultGcc2OptimizeOpt GccAliasingArgs
 # endif
 # define LinuxMachineDefines	-D__i386__
 # define ServerOSDefines	XFree86ServerOSDefines -DDDXTIME
@@ -875,7 +978,7 @@ InstallNamedTargetNoClobber(install,file
 
 #ifdef ia64Architecture
 # ifndef OptimizedCDebugFlags
-#  define OptimizedCDebugFlags	-O2  GccAliasingArgs
+#  define OptimizedCDebugFlags	DefaultGcc2DebugOpt DefaultGcc2OptimizeOpt GccAliasingArgs
 # endif
 # define LinuxMachineDefines	-D__ia64__
 # define ServerOSDefines	XFree86ServerOSDefines -DDDXTIME
@@ -884,7 +987,7 @@ InstallNamedTargetNoClobber(install,file
 
 #ifdef Mc68020Architecture
 # ifndef OptimizedCDebugFlags
-#  define OptimizedCDebugFlags	-O2  GccAliasingArgs
+#  define OptimizedCDebugFlags	DefaultGcc2DebugOpt DefaultGcc2OptimizeOpt GccAliasingArgs
 # endif
 # define LinuxMachineDefines	-D__mc68000__
 # define ServerOSDefines	XFree86ServerOSDefines -DDDXTIME
@@ -903,7 +1006,7 @@ InstallNamedTargetNoClobber(install,file
 #ifdef Ppc64Architecture
 # define DefaultCCOptions	-mminimal-toc
 # ifndef OptimizedCDebugFlags
-#  define OptimizedCDebugFlags	DefaultGcc2Ppc64Opt
+#  define OptimizedCDebugFlags	DefaultGcc2Ppc64Opt GccAliasingArgs
 # endif
 # define LinuxMachineDefines	-D__powerpc64__ -D__powerpc__
 # define ServerOSDefines	XFree86ServerOSDefines -DDDXTIME -DPART_NET
@@ -913,7 +1016,7 @@ InstallNamedTargetNoClobber(install,file
 #ifdef PpcArchitecture
 # define DefaultCCOptions	
 # ifndef OptimizedCDebugFlags
-#  define OptimizedCDebugFlags	DefaultGcc2PpcOpt
+#  define OptimizedCDebugFlags DefaultGcc2DebugOpt DefaultGcc2OptimizeOpt GccAliasingArgs
 # endif
 # define LinuxMachineDefines	-D__powerpc__
 # define ServerOSDefines	XFree86ServerOSDefines -DDDXTIME
@@ -922,7 +1025,7 @@ InstallNamedTargetNoClobber(install,file
 
 #ifdef s390Architecture
 # ifndef OptimizedCDebugFlags
-#  define OptimizedCDebugFlags	-O2 -fomit-frame-pointer GccAliasingArgs
+#  define OptimizedCDebugFlags DefaultGcc2DebugOpt DefaultGcc2OptimizeOpt GccAliasingArgs
 # endif
 # define LinuxMachineDefines	-D__s390__
 # define ServerOSDefines	XFree86ServerOSDefines -DDDXTIME
@@ -930,16 +1033,17 @@ InstallNamedTargetNoClobber(install,file
 #endif /* s390Architecture */
 
 #ifdef s390xArchitecture
-/*#define DefaultCCOptions	-fsigned-char */
-#define OptimizedCDebugFlags	-O3 -fomit-frame-pointer
+# ifndef OptimizedCDebugFlags
+#  define OptimizedCDebugFlags DefaultGcc2DebugOpt DefaultGcc2OptimizeOpt GccAliasingArgs
+# endif
 #define LinuxMachineDefines	-D__s390x__
-#define ServerOSDefines		XFree86ServerOSDefines -DDDXTIME
+#define ServerOSDefines		XFree86ServerOSDefines -DDDXTIME -DPART_NET
 #define ServerExtraDefines	-DGCCUSESGAS XFree86ServerDefines -D_XSERVER64
 #endif /* s390xArchitecture */
 
 #ifdef SparcArchitecture
 # ifndef OptimizedCDebugFlags
-#  define OptimizedCDebugFlags	-O2  GccAliasingArgs
+#  define OptimizedCDebugFlags	DefaultGcc2DebugOpt DefaultGcc2OptimizeOpt GccAliasingArgs
 # endif
 # define LinuxMachineDefines	-D__sparc__
 # define ServerOSDefines	XFree86ServerOSDefines -DDDXTIME
@@ -970,7 +1074,7 @@ InstallNamedTargetNoClobber(install,file
 #  endif
 # endif
 # ifndef OptimizedCDebugFlags
-#  define OptimizedCDebugFlags	-O2 SuperHArchOptFlags SuperHEndianFlags  GccAliasingArgs
+#  define OptimizedCDebugFlags	DefaultGcc2DebugOpt DefaultGcc2OptimizeOpt SuperHArchOptFlags SuperHEndianFlags GccAliasingArgs
 # endif
 # define LinuxMachineDefines	-D__sh__
 # define ServerOSDefines	XFree86ServerOSDefines -DDDXTIME
@@ -982,7 +1086,7 @@ InstallNamedTargetNoClobber(install,file
 
 #ifdef AMD64Architecture
 # ifndef OptimizedCDebugFlags
-#  define OptimizedCDebugFlags	DefaultGcc2AMD64Opt
+#  define OptimizedCDebugFlags	DefaultGcc2DebugOpt DefaultGcc2OptimizeOpt GccAliasingArgs
 # endif
 # define LinuxMachineDefines   -D__amd64__
 # define ServerOSDefines	XFree86ServerOSDefines -DDDXTIME
@@ -1023,26 +1127,6 @@ InstallNamedTargetNoClobber(install,file
 # define ConnectionFlags	-DUNIXCONN -DTCPCONN
 #endif
 
-/* Some of these man page defaults are overriden in the above OS sections */
-#ifndef ManSuffix
-# define ManSuffix	1x
-#endif
-#ifndef ManDir
-# define ManDir		$(MANSOURCEPATH)1
-#endif
-#ifndef LibManSuffix
-# define LibManSuffix	3x
-#endif
-#ifndef LibmanDir
-# define LibmanDir	$(MANSOURCEPATH)3
-#endif
-#ifndef FileManSuffix
-# define FileManSuffix	5x
-#endif
-#ifndef FileManDir
-# define FileManDir	$(MANSOURCEPATH)5
-#endif
-
 #ifndef StaticLibrary
 # define StaticLibrary(libpath,libname) -Wl,-Bstatic Concat(-L,libpath) Concat(-l,libname) -Wl,-Bdynamic
 #endif
@@ -1087,3 +1171,59 @@ InstallNamedTargetNoClobber(install,file
 #ifndef XFree86ServerOSDefines
 # define XFree86ServerOSDefines
 #endif
+
+/*
+ *     SELinux support
+ */
+#if defined(__linux__)
+# ifndef HasSELinux
+#  define HasSELinux YES
+# endif
+#endif
+
+#ifndef SELinuxDefines
+# if HasSELinux
+#  define SELinuxDefines -DHAVE_SELINUX
+# else
+#  define SELinuxDefines /**/
+# endif
+#endif
+
+#ifndef SELinuxIncludeFlags
+# if HasSELinux
+#   define SELinuxIncludeFlags -I/usr/include/selinux
+# else
+#   define SELinuxIncludeFlags /**/
+# endif
+#endif
+
+#ifndef SELinuxCompileFlags
+# define SELinuxCompileFlags /**/
+#endif
+
+#ifndef SELinuxLoadFlags
+# define SELinuxLoadFlags SELinuxCompileFlags
+#endif
+
+#ifndef SELinuxLibraries
+# if HasSELinux
+#  define SELinuxLibraries -lselinux
+# else
+#  define SELinuxLibraries /**/
+# endif
+#endif
+
+#if HasSELinux
+# ifndef SELINUX_LDFLAGS
+   SELINUX_LDFLAGS = SELinuxLoadFlags
+# endif
+# ifndef SELINUX_INCLUDES
+   SELINUX_INCLUDES = SELinuxIncludeFlags
+# endif
+# ifndef SELINUX_CFLAGS
+   SELINUX_CFLAGS = SELinuxCompileFlags SELinuxDefines
+# endif
+# ifndef SELINUX_LIBS
+   SELINUX_LIBS = SELinuxLibraries
+# endif
+#endif
Index: xutils-dev/xorg-cf-files/NetBSD.cf
===================================================================
--- xutils-dev.orig/xorg-cf-files/NetBSD.cf
+++ xutils-dev/xorg-cf-files/NetBSD.cf
@@ -18,12 +18,178 @@ XCOMM platform:  $XFree86: xc/config/cf/
 #define OSTeenyVersion		DefaultOSTeenyVersion
 #endif
 
+#ifndef NetBSDDistribution
+#define NetBSDDistribution	DefaultNetBSDDistribution
+/*
+  Add "#define NetBSDDistribution NetBSD<mumble>" to your site.def or host.def.
+  Currently only NetBSDDebian will be autodetected.
+  Valid values are:
+    NetBSDUnknown    (0)
+    NetBSDNative     (1)
+    NetBSDDebian     (2)
+*/
+#endif
+
+#ifndef NetBSDDistName
+# define NetBSDDistName		DefaultNetBSDDistName
+#endif
 
 #ifndef OSVendor
-#define	OSVendor		The NetBSD Foundation, Inc.
+#define OSVendor		The NetBSD Foundation, Inc.
 #endif
+
+#ifndef NetBSDBinUtilsMajorVersion
+# define NetBSDBinUtilsMajorVersion	DefaultNetBSDBinUtilsMajorVersion
+#endif
+
 XCOMM operating system:  OSName (OSMajorVersion./**/OSMinorVersion./**/OSTeenyVersion)
 
+/* Defines for Debian GNU/NetBSD */
+
+#if NetBSDDistribution == NetBSDDebian
+# ifndef DefaultGcc2OptimizeOpt
+#  define DefaultGcc2OptimizeOpt	-O2
+# endif
+# define DefaultGcc2AxpOpt	DefaultGcc2OptimizeOpt GccAliasingArgs
+# define DefaultGcc2i386Opt	DefaultGcc2OptimizeOpt GccAliasingArgs
+# define DefaultGcc2PpcOpt	DefaultGcc2OptimizeOpt GccAliasingArgs
+# define SystemManDirectory	/usr/share/man
+# define HasPam			YES
+# define HasTk			YES
+# define TkLibDir		/usr/lib
+# define TkIncDir		/usr/include
+# define TkLibName		tk8.4
+# define XF86SetupUsesStaticTk	NO
+# define HasTcl			YES
+# define TclLibDir		/usr/lib
+# define TclIncDir		/usr/include
+# define TclLibName		tcl8.4
+# define XF86SetupUsesStaticTcl	NO
+/* un-comment this when it is un-broken */
+/* # define JoystickSupport YES */
+# define XAppLoadDir		EtcX11Directory/app-defaults
+# define XFileSearchPathDefault	Concat4(EtcX11Directory/%L/%T/%N%C,%S:EtcX11Directory/%l/%T/%N%C,%S:EtcX11Directory/%T/%N%C,%S:EtcX11Directory/%L/%T/%N%S:EtcX11Directory/%l/%T/%N%S:EtcX11Directory/%T/%N%S):Concat4($(LIBDIR)/%L/%T/%N%C,%S:$(LIBDIR)/%l/%T/%N%C,%S:$(LIBDIR)/%T/%N%C,%S:$(LIBDIR)/%L/%T/%N%S:$(LIBDIR)/%l/%T/%N%S:$(LIBDIR)/%T/%N%S)
+# define InstallAppDefFiles	YES
+/* the relative symlink created by this rule causes problems for us */
+# ifdef InstallAppDefFiles
+#  if InstallAppDefFiles
+#   define InstallAppDefaultsLong(file,class)				@@\
+InstallNamedTarget(install,file.ad,$(INSTAPPFLAGS),$(XAPPLOADDIR),class)
+#  else
+#   define InstallAppDefaultsLong(file,class)				@@\
+InstallNamedTargetNoClobber(install,file.ad,$(INSTAPPFLAGS),$(XAPPLOADDIR),class)
+#  endif
+# endif /* InstallAppDefFiles */
+# define SharedLibXdmGreet	NO
+# define LinkGLToUsrInclude	NO
+# define LinkGLToUsrLib		NO
+# define SharedLibFont		NO
+# define HasZlib		YES
+# define SharedLibGlu		YES
+# define NormalLibGlu		YES
+# define FSUseSyslog		YES
+
+# define DebianMaintainer	YES
+
+# ifdef DebianMaintainer
+#  ifndef XFree86CustomVersion
+#    define XFree86CustomVersion	"Debian"
+#  endif
+#  ifndef BuilderEMailAddr
+#    define BuilderEMailAddr		"debian-x@lists.debian.org"
+#  endif
+#  define XFree86Devel			YES
+#  define InstallXinitConfig		YES
+#  define InstallXdmConfig		YES
+#  define InstallFSConfig		YES
+#  define DebuggableLibraries		YES
+#  define ForceNormalLib		YES
+#  define BuildRman			NO
+#  define BuildHtmlManPages		NO
+/*
+ * Debian doesn't use certain parts of the tree when doing
+ * architecture-specific package builds.
+ */
+#  ifndef BuildFonts
+#   define BuildFonts			YES
+#  endif
+#  ifndef BuildSpecsDocs
+#   define BuildSpecsDocs		YES
+#  endif
+#  if BuildSpecsDocs
+#   define BuildAllSpecsDocs		YES
+#  endif
+/*
+ * The XFree86 Debian package uses a local patch that defines a
+ * PicLibraryTarget() Imake rule, instead of using upstream's method.
+ */
+#  define StaticNeedsPicForShared	NO
+#  define KernelVersionInBanner		YES
+#  define ProjectManSuffix		x
+/* The TT/Meltho fonts are all under non-free licenses. */
+#  define BuildTrueTypeFonts		NO
+#  define BuildBethMarduthoFonts	NO
+/*
+ * The XFree86 Debian package build-depends on external Expat, Fontconfig,
+ * Freetype, Render, Xcursor, and Xft libraries.
+ */
+#  define BuildExpatLibrary		NO
+#  define HasExpat			YES
+#  define ExpatIncDirStandard		YES
+#  define BuildFontconfigLibrary	NO
+#  define HasFontconfig			YES
+#  define UseFontconfig			YES
+#  define FontconfigIncDir		/usr/include/fontconfig
+#  define BuildFreetype2Library		NO
+#  define HasFreetype2			YES
+#  define BuildRenderLibrary		NO
+#  define HasRenderLibrary		YES
+#  define BuildXcursorLibrary		NO
+#  define HasXcursorLibrary		YES
+/* Xlib needs to know the available version of Xcursor. */
+#  define SharedXcursorRev		1
+#  define BuildXftLibrary		NO
+#  define HasXftLibrary			YES
+
+#  ifdef UseInstalled
+#    define RenderLibraryDir		/usr
+#    define XftLibraryDir		/usr
+#    define XcursorLibraryDir		/usr
+#  else
+/* The XFree86 Debian package uses a special hack to find these libraries. */
+#    define RenderLibraryDir		$(TOP)/../../imports
+#    define XftLibraryDir		$(TOP)/../../imports
+#    define XcursorLibraryDir		$(TOP)/../../imports
+#  endif
+
+/*
+ * We want to be sure that the normal XFree86 X server and the debugging X
+ * server use the same FreeType2 library.  We'd *like* it if we could
+ * achieve this by both packages dynamically linking against the system's
+ * FreeType2 library; however, the normal X server package
+ * (xserver-xorg) *cannot* be built dynamically linked against the
+ * FreeType2 library when the module loader is enabled because of
+ * code/design issues.  Therefore, we encapsulate XFree86's internal "fork"
+ * of the FreeType2 library into *both* xserver-xorg and
+ * xserver-xorg-dbg.  When it becomes possible to build the
+ * module-loading server against an external FreeType2 shared library, we
+ * can drop this define:
+ */
+#   define FontLibSharedFreeType	NO
+
+#   define HasLinuxInput		YES
+#   define HasXdmAuth			YES
+#   define HasLatex			YES
+/* Enable extended instruction set support. */
+#   ifdef i386Architecture
+#    define HasX86Support		YES
+#    define HasMMXSupport		YES
+#    define Has3DNowSupport		YES
+#    define HasSSESupport		YES
+#   endif /* i386Architecture */
+# endif /* DebianMaintainer */
+#endif /* NetBSDDebian */
+
 /*
  * C library features
  */
@@ -93,8 +259,12 @@ XCOMM operating system:  OSName (OSMajor
 
 #define HasUsableFileMmap	YES
 
-#define HasVarRun		YES
+#if NetBSDDistribution == NetBSDDebian
+#define VarDbDirectory		$(VARDIR)/lib
+#else /* Debian */
 #define HasVarDb		YES
+#endif /* Debian */
+#define HasVarRun		YES
 
 #define HasShadowPasswd		NO
 
@@ -207,6 +377,8 @@ XCOMM operating system:  OSName (OSMajor
 
 #if UseElfFormat
 #define DlLibrary		-rdynamic /* no -ldl yet */
+#define HasPlugin		YES
+#define VendorHasX11R6_3libXext	YES
 #else
 #define DlLibrary		/**/
 #endif
@@ -336,7 +508,11 @@ XCOMM operating system:  OSName (OSMajor
 /* NetBSD doesn't have gperf in the base install */
 #define DontRebuildGperfOutput	YES
 
-/* 
+#if NetBSDDistribution == NetBSDDebian
+#define HasPerl			YES
+#endif
+
+/*
  * NetBSD doesn't have perl in default system
  * you may override this in host.def if you installed perl from the packages
  */
@@ -355,7 +531,13 @@ XCOMM operating system:  OSName (OSMajor
 /*
  * Make & install Features
  */
+
+#if NetBSDDistribution == NetBSDDebian
+#define HasBsdMake		NO
+#else
 #define HasBsdMake		YES
+#endif
+
 #define CompressAllFonts	YES
 #define GzipFontCompression	YES
 #define DefaultUserPath		/bin:/usr/bin:/usr/pkg/bin:/usr/local/bin:$(BINDIR)
@@ -463,6 +645,8 @@ dirs:	.EXEC								@@\
 									@@\
 .EXEC:
 
+#if NetBSDDistribution != NetBSDDebian /* Debian uses ProjectManSuffix */
+
 /*
  * Man pages need to be formatted when installed, so override the default
  * imake rules.
@@ -552,6 +736,8 @@ InstallHtmlManPageAliases(file,aliases,s
 									@@\
 InstallManPageAliasesBase(file,destdir,aliases)
 
+#endif /* NetBSDDistribution != NetBSDDebian */
+
 /*
  * Shared libraries supported starting in 1.0 on a.out platforms,
  * and on all ELF platforms.
Index: xutils-dev/xorg-cf-files/sco5.cf
===================================================================
--- xutils-dev.orig/xorg-cf-files/sco5.cf
+++ xutils-dev/xorg-cf-files/sco5.cf
@@ -69,7 +69,7 @@ XCOMM $XdotOrg: xc/config/cf/sco5.cf,v 1
 #endif
 
 #ifndef UseExportLists
-# define UseExportLists		YES
+# define UseExportLists		NO
 #endif
 
 #define StandardDefines		-DSYSV -Di386 -DSCO325 -D__SCO__
Index: xutils-dev/xorg-cf-files/usl.cf
===================================================================
--- xutils-dev.orig/xorg-cf-files/usl.cf
+++ xutils-dev/xorg-cf-files/usl.cf
@@ -159,7 +159,7 @@ XCOMM operating system: OSName (OSMajorV
 #endif
 
 #ifndef UseExportLists
-# define UseExportLists		YES
+# define UseExportLists		NO
 #endif
 
 #define GencatFlags		-m
Index: xutils-dev/xorg-cf-files/X11.tmpl
===================================================================
--- xutils-dev.orig/xorg-cf-files/X11.tmpl
+++ xutils-dev/xorg-cf-files/X11.tmpl
@@ -614,23 +614,29 @@ XORGRELSTRING = XorgManVersionString
 #ifndef BuildRender
 #define BuildRender		YES
 #endif
+#ifndef HasRenderLibrary
+#define HasRenderLibrary	NO
+#endif
 #ifndef BuildRenderLibrary
-#define BuildRenderLibrary	(XdmxServer || !BuildServersOnly)
+#define BuildRenderLibrary	(XdmxServer || (!HasRenderLibrary && BuildLibraries))
+#endif
+#ifndef HasXftLibrary
+#define HasXftLibrary		NO
 #endif
 #ifndef BuildXftLibrary
-#define BuildXftLibrary		BuildRenderLibrary
+#define BuildXftLibrary		((BuildRenderLibrary || HasRenderLibrary) && BuildLibraries)
 #endif
 #ifndef BuildXft1Library
 #define BuildXft1Library	BuildRenderLibrary
 #endif
 #ifndef UseFontconfig
-#define UseFontconfig		BuildXftLibrary
+#define UseFontconfig		(BuildXftLibrary || HasXftLibrary)
 #endif
 #ifndef BuildFontconfigLibrary
 #define BuildFontconfigLibrary	(UseFontconfig && !HasFontconfig)
 #endif
 #ifndef UseFreetype2
-#define UseFreetype2		(BuildXftLibrary || BuildXft1Library || \
+#define UseFreetype2		(BuildXftLibrary || HasXftLibrary || BuildXft1Library || \
 				 UseFontconfig || BuildFreeType)
 #endif
 #ifndef InstallFontconfigLibrary
@@ -700,8 +706,12 @@ XORGRELSTRING = XorgManVersionString
 #define BuildXkbcomp		BuildClients
 #endif
 
+#ifndef HasXcursorLibrary
+#define HasXcursorLibrary	NO
+#endif
+
 #ifndef BuildXcursorLibrary
-#define BuildXcursorLibrary	BuildRenderLibrary
+#define BuildXcursorLibrary	((BuildRenderLibrary || HasRenderLibrary) && !HasXcursorLibrary && BuildLibraries)
 #endif
 
 #ifndef BuildXcursorgen
@@ -1170,7 +1180,7 @@ FCHOWN_DEFINES = -DHAS_FCHOWN
 #endif
 
 #ifndef RenderDefines
-#if BuildRender
+#if BuildRender || HasRenderLibrary
 #define RenderDefines -DRENDER
 #else
 #define RenderDefines /**/
@@ -2669,6 +2679,7 @@ SharedDSLibReferences(GLW,GLw,$(GLWIDGET
 ProjectUnsharedLibReferences(GLW,GLw,$(GLWIDGETSRC),XBuildLibDir)
 #endif
 
+#if BuildRenderLibrary
     XRENDERLIBSRC = $(LIBSRC)/Xrender
 #if SharedLibXrender
 #ifndef SharedXrenderRev
@@ -2679,6 +2690,42 @@ SharedLibReferences(XRENDER,Xrender,$(XR
 ProjectUnsharedLibReferences(XRENDER,Xrender,$(XRENDERLIBSRC),XBuildLibDir)
 #endif
 
+#ifdef UseInstalled
+XRENDERINCDIR=$(INCDIR)
+#else
+XRENDERINCDIR=$(BUILDINCDIR)
+#endif
+
+XRENDERINCLUDES = -I$(XRENDERINCDIR)
+
+#else /* BuildRenderLibrary */
+
+#if HasRenderLibrary
+
+#ifndef RenderLibraryDir
+#define RenderLibraryDir /usr/X11R6
+#endif
+
+#ifndef RenderLibraryLibDir
+#define RenderLibraryLibDir RenderLibraryDir/LibDirName
+#endif
+
+#ifndef RenderLibraryIncDir
+#define RenderLibraryIncDir RenderLibraryDir/include
+#endif
+
+XRENDERDIR = RenderLibraryDir
+XRENDERLIBDIR = RenderLibraryLibDir
+XRENDERINCDIR = RenderLibraryIncDir
+XRENDERLIB = -L$(XRENDERLIBDIR) -lXrender
+DEPXRENDERLIB =
+
+XRENDERINCLUDES = -I$(XRENDERINCDIR)
+
+#endif /* HasRenderLibrary */
+
+#endif /* else BuildRenderLibrary */
+
     XRANDRLIBSRC = $(LIBSRC)/Xrandr
 #if SharedLibXrandr
 #ifndef SharedXrandrRev
@@ -2729,6 +2776,7 @@ SharedLibReferences(XEVIE,Xevie,$(XEVIEL
 ProjectUnsharedLibReferences(XEVIE,Xevie,$(XEVIELIBSRC),XBuildLibDir)
 #endif
 
+#if BuildXcursorLibrary
    XCURSORLIBSRC = $(LIBSRC)/Xcursor
 #if SharedLibXcursor
 #ifndef SharedXcursorRev
@@ -2739,6 +2787,41 @@ SharedLibReferences(XCURSOR,Xcursor,$(XC
 ProjectUnsharedLibReferences(XCURSOR,Xcursor,$(XCURSORLIBSRC),XBuildLibDir)
 #endif
 
+#ifdef UseInstalled
+XCURSORINCDIR=$(INCDIR)
+#else
+XCURSORINCDIR=$(BUILDINCDIR)
+#endif
+
+XCURSORINCLUDES=-I$(XCURSORINCDIR) $(XRENDERINCLUDES)
+
+#else /* BuildXcursorLibrary */
+
+#if HasXcursorLibrary
+
+#ifndef XcursorLibraryDir
+#define XcursorLibraryDir /usr/X11R6
+#endif
+
+#ifndef XcursorLibraryLibDir
+#define XcursorLibraryLibDir XcursorLibraryDir/LibDirName
+#endif
+
+#ifndef XcursorLibraryIncDir
+#define XcursorLibraryIncDir XcursorLibraryDir/include
+#endif
+
+XCURSORDIR = XcursorLibraryDir
+XCURSORLIBDIR = XcursorLibraryLibDir
+XCURSORINCDIR = XcursorLibraryIncDir
+XCURSORLIB = -L$(XCURSORLIBDIR) -lXcursor
+
+XCURSORINCLUDES=-I$(XCURSORINCDIR) $(XRENDERINCLUDES)
+
+#endif /* HasXcursorLibrary */
+
+#endif /* else BuildXcursorLibrary */
+
    APPLEWMLIBSRC = $(LIBSRC)/apple
 #if SharedLibAppleWM
 #ifndef SharedAppleWMRev
@@ -3544,6 +3627,7 @@ EXPATDEFINES = -DEXPAT
 
 #endif /* UseExpat */
 
+#if BuildXftLibrary
 #ifndef SharedLibXft1
 #define SharedLibXft1 HasSharedLibraries
 #endif
@@ -3556,6 +3640,18 @@ EXPATDEFINES = -DEXPAT
 #ifndef ProfileLibXft1
 #define ProfileLibXft1		NO
 #endif
+#else
+#undef  SharedLibXft
+#define SharedLibXft		NO
+#undef  NormalLibXft
+#define NormalLibXft		NO
+#undef  DebugLibXft
+#define DebugLibXft		NO
+#undef  ProfileLibXft
+#define ProfileLibXft		NO
+#endif
+
+#if BuildXftLibrary
           XFT1LIBSRC = $(LIBSRC)/Xft1
 #if SharedLibXft1
 #ifndef SharedXft1Rev
@@ -3610,6 +3706,42 @@ ProjectUnsharedLibReferences(XFT,Xft,$(X
 #define XftClientLibs $(XFTLIB) FontconfigClientLibs $(XRENDERLIB)
 #endif
 
+XFTINCLUDES=$(FONTCONFIGINCLUDES) $(FREETYPE2INCLUDES) $(XRENDERINCLUDES)
+
+#else /* BuildXftLibrary */
+
+#if HasXftLibrary
+
+#ifndef XftLibraryDir
+#define XftLibraryDir /usr/X11R6
+#endif
+
+#ifndef XftLibraryLibDir
+#define XftLibraryLibDir XftLibraryDir/LibDirName
+#endif
+
+#ifndef XftLibraryIncDir
+#define XftLibraryIncDir XftLibraryDir/include
+#endif
+
+XFTDIR = XftLibraryDir
+XFTLIBDIR = XftLibraryLibDir
+XFTINCDIR = XftLibraryIncDir
+XFTLIB = -L$(XFTLIBDIR) -lXft
+
+#ifndef XftClientDepLibs
+#define XftClientDepLibs FontconfigClientDepLibs $(DEPXRENDERLIB)
+#endif
+#ifndef XftClientLibs
+#define XftClientLibs $(XFTLIB) FontconfigClientLibs $(XRENDERLIB)
+#endif
+
+XFTINCLUDES= -I$(XFTINCDIR) $(FONTCONFIGINCLUDES) $(FREETYPE2INCLUDES) $(XRENDERINCLUDES)
+
+#endif /* HasXftLibrary */
+
+#endif /* else BuildXftLibrary */
+
 #if UseFontconfig
 
 #if BuildFontconfigLibrary
@@ -3725,8 +3857,6 @@ FONTCONFIGDEFINES = -DFONTCONFIG
 
 #endif /* UseFontconfig */
 
-XFTINCLUDES=$(FONTCONFIGINCLUDES) $(FREETYPE2INCLUDES)
-
 #ifdef HasLibpng
 
 /*
Index: xutils-dev/xorg-cf-files/xfree86.cf
===================================================================
--- xutils-dev.orig/xorg-cf-files/xfree86.cf
+++ xutils-dev/xorg-cf-files/xfree86.cf
@@ -100,6 +100,13 @@ XCOMM $Xorg: xfree86.cf,v 1.4 2000/08/17
 #define XVendorManNameString XFree86
 #endif
 
+VENDORMANNAME = XVendorManNameString
+VENDORMANVERSION = XVendorManVersionString
+
+#ifndef VendorManDefs
+#define VendorManDefs -D__vendorversion__="\"Version $(VENDORMANVERSION)\" $(VENDORMANNAME)"
+#endif
+
 /*
  * This enables some settings for developers.
  */
@@ -116,6 +123,15 @@ XCOMM $Xorg: xfree86.cf,v 1.4 2000/08/17
 # endif
 #endif
 
+/*
+ * No XF86Server for the S/390 platform.
+ */
+#ifdef s390Architecture
+# ifndef XF86Server
+#  define XF86Server		NO
+# endif
+#endif
+
 #ifndef BuildDebug
 # define BuildDebug 		YES
 #endif
@@ -233,6 +249,11 @@ XCOMM $Xorg: xfree86.cf,v 1.4 2000/08/17
 #define GlideDriver /**/
 #endif
 
+/* The tdfx DRI driver can only be built if glide3 is available. */
+#ifndef HasGlide3
+#define HasGlide3 NO
+#endif
+
 /*
  * There are three parameters that determine where and how the Matrox HAL
  * library is used:
@@ -372,15 +393,17 @@ XCOMM $Xorg: xfree86.cf,v 1.4 2000/08/17
 #  /* Don't build the nsc driver on IA64 */
 #  ifndef ia64Architecture
 #   define NscDriver		nsc
+#   define ViaDriver		via
 #  else
 #   define NscDriver		/**/
+#   define ViaDriver		/**/
 #  endif
 #  define XF86CardDrivers	mga glint nv tga s3 s3virge sis rendition \
 				neomagic i740 tdfx savage \
 				cirrus vmware tseng trident chips apm \
 				GlideDriver i128 NscDriver \
 				ati i810 AgpGartDrivers DevelDrivers ark \
-				cyrix siliconmotion via \
+				cyrix siliconmotion ViaDriver \
 				fbdev vesa vga \
 				dummy XF86OSCardDrivers XF86ExtraCardDrivers
 # endif
@@ -485,8 +508,11 @@ XCOMM $Xorg: xfree86.cf,v 1.4 2000/08/17
 				vga dummy fbdev vesa
 # endif
 
-/* DRI tdfx driver needs Glide, which is not available for AMD64 */
-# define TdfxDriDriver		/**/
+# if HasGlide3
+#  define TdfxDriDriver		tdfx
+# else
+#  define TdfxDriDriver		/**/
+#endif
 
 # define DevelDRIDrivers	/**/
 
@@ -579,9 +605,9 @@ XCOMM $Xorg: xfree86.cf,v 1.4 2000/08/17
     * require changes (or even removal) for things like endianness, etc.  Move
     * those known to work into the main XF86CardDrivers #define.
     */
-#   define DevelDrivers		apm ark chips cirrus i128 i740 i810 imstt \
-				neomagic newport nv rendition s3virge savage \
-				siliconmotion tga trident vmware dummy
+#   define DevelDrivers		apm ark chips cirrus i128 i740 imstt \
+				neomagic nv rendition s3virge savage \
+				siliconmotion trident dummy
 #  else
 #   define DevelDrivers		/* */
 #  endif
@@ -872,8 +898,14 @@ XCOMM $Xorg: xfree86.cf,v 1.4 2000/08/17
 				savage nv cirrus DevelDrivers siliconmotion  \
 				vga XF86OSCardDrivers XF86ExtraCardDrivers
 # endif
+# if HasGlide3
+#  define TdfxDriDriver		tdfx
+# else
+#  define TdfxDriDriver		/**/
+# endif
 # ifndef DriDrivers
-#  define DriDrivers		gamma tdfx mga r128 radeon r200 /*sis*/
+#  define DriDrivers		gamma tdfx mga r128 radeon r200 /*sis*/ \
+				TdfxDriDriver
 # endif
 #endif
 
@@ -968,8 +1000,13 @@ XCOMM $Xorg: xfree86.cf,v 1.4 2000/08/17
 				DevelDrivers vga \
 				XF86OSCardDrivers XF86ExtraCardDrivers
 # endif
+# if HasGlide3
+#  define TdfxDriDriver		tdfx
+# else
+#  define TdfxDriDriver		/**/
+# endif
 # ifndef DriDrivers
-#  define DriDrivers		gamma tdfx mga r128 radeon r200
+#  define DriDrivers		gamma TdfxDriDriver mga r128 radeon r200
 # endif
 #endif
 
@@ -1066,7 +1103,7 @@ XCOMM $Xorg: xfree86.cf,v 1.4 2000/08/17
 #  define TdfxDriDriver		/**/
 # endif
 # ifndef DriDrivers
-#  define DriDrivers		gamma tdfx mga r128 radeon r200
+#  define DriDrivers		gamma TdfxDriDriver mga r128 radeon r200
 # endif
 #endif
 
@@ -1745,7 +1782,7 @@ IPLAN2P8_DEFS = -DUSE_IPLAN2P8
 #endif
 #ifndef BuildScanpci
 # if SystemV || SystemV4 || \
-    (defined(LinuxArchitecture) && !defined(Mc68020Architecture)) || \
+    defined(LinuxArchitecture) || \
     defined(i386BsdArchitecture) || defined(LynxOSArchitecture) || \
     defined(OS2Architecture) || defined(GNUMachArchitecture) || \
     (defined(KFreeBSDArchitecture) && defined(i386Architecture))
@@ -1943,10 +1980,6 @@ XFREE86JAPANESEDOCDIR = $(DOCDIR)/Japane
 # define UseRgbTxt	YES
 #endif
 
-#ifndef ManSuffix
-# define ManSuffix	1
-#endif
-
 /*
  * GNU Compiler stuff
  */
@@ -2010,33 +2043,39 @@ XFREE86JAPANESEDOCDIR = $(DOCDIR)/Japane
 # define GccAliasingArgs      /* */
 #endif
 
+#if HasGcc2
+# ifndef DefaultGcc2OptimizeOpt
+#  define DefaultGcc2OptimizeOpt -O2
+# endif
+#endif
+
 #if HasGcc2 && defined(i386Architecture)
 # ifndef DefaultGcc2i386Opt
-#  define DefaultGcc2i386Opt -O2 -fno-strength-reduce GccAliasingArgs
+#  define DefaultGcc2i386Opt DefaultGcc2OptimizeOpt GccAliasingArgs
 # endif
 #endif
 
 #if HasGcc2 && defined(AMD64Architecture)
 # ifndef DefaultGcc2AMD64Opt
-#  define DefaultGcc2AMD64Opt -O2 -fno-strength-reduce GccAliasingArgs
+#  define DefaultGcc2AMD64Opt DefaultGcc2OptimizeOpt GccAliasingArgs
 # endif
 #endif
 
 #if HasGcc2 && defined(AlphaArchitecture)
 # ifndef DefaultGcc2AxpOpt
-#  define DefaultGcc2AxpOpt -O2 GccAliasingArgs
+#  define DefaultGcc2AxpOpt DefaultGcc2OptimizeOpt GccAliasingArgs
 # endif
 #endif
 
 #if HasGcc2 && defined(Ppc64Architecture)
 # ifndef DefaultGcc2Ppc64Opt
-#  define DefaultGcc2Ppc64Opt -O2 GccAliasingArgs
+#  define DefaultGcc2Ppc64Opt DefaultGcc2OptimizeOpt GccAliasingArgs
 # endif
 #endif
 
 #if HasGcc2 && defined(PpcArchitecture)
 # ifndef DefaultGcc2PpcOpt
-#  define DefaultGcc2PpcOpt -O2 GccAliasingArgs
+#  define DefaultGcc2PpcOpt DefaultGcc2OptimizeOpt GccAliasingArgs
 # endif
 #endif
 
Index: xutils-dev/xorg-cf-files/xorg.cf
===================================================================
--- xutils-dev.orig/xorg-cf-files/xorg.cf
+++ xutils-dev/xorg-cf-files/xorg.cf
@@ -262,6 +262,11 @@ RELEASE_VERSION = ReleaseVersion
 #define GlideDriver /**/
 #endif
 
+/* The tdfx DRI driver can only be built if glide3 is available. */
+#ifndef HasGlide3
+#define HasGlide3 NO
+#endif
+
 /*
  * There are three parameters that determine where and how the Matrox HAL
  * library is used:
@@ -425,7 +430,13 @@ RELEASE_VERSION = ReleaseVersion
    /* SiS is 64-bit unclean as of this writing. */
 #   define i386DRIDrivers /**/
 #  endif
-#  define DriDrivers		i386DRIDrivers mga r128 radeon r200 tdfx
+#  if HasGlide3
+#   define TdfxDriDriver   tdfx
+#  else
+#   define TdfxDriDriver   /**/
+#  endif
+#  define DriDrivers       i386DRIDrivers mga r128 radeon r200 \
+               TdfxDriDriver
 # endif
 #endif /* i386Architecture || ia64Architecture */
 
@@ -461,8 +472,13 @@ RELEASE_VERSION = ReleaseVersion
 # endif
 
 /* SiS is 64-bit unclean as of this writing. */
+#  if HasGlide3
+#   define TdfxDriDriver   tdfx
+#  else
+#   define TdfxDriDriver   /**/
+#  endif
 # ifndef DriDrivers
-#  define DriDrivers		i915 mga r128 radeon r200 tdfx
+#  define DriDrivers       i915 mga r128 radeon r200 TdfxDriDriver
 # endif
 #endif /*AMD64Arcitecture*/
 
@@ -602,8 +618,13 @@ RELEASE_VERSION = ReleaseVersion
  *  Glide is available for Alpha, therefore build tdfx DRM module.
  *  SiS is 64-bit unclean as of this writing. 
  */  
+#  if HasGlide3
+#   define TdfxDriDriver   tdfx
+#  else
+#   define TdfxDriDriver   /**/
+#  endif
 # ifndef DriDrivers
-#  define DriDrivers		tdfx mga r128 radeon r200
+#  define DriDrivers       TdfxDriDriver mga r128 radeon r200
 # endif
 #endif /* AlphaArchitecture */
 
@@ -767,7 +788,7 @@ RELEASE_VERSION = ReleaseVersion
 # endif
 # define TdfxDriDriver		/**/
 # ifndef DriDrivers
-#  define DriDrivers		r128
+#  define DriDrivers		r128 TdfxDriDriver
 # endif
 #endif /* Ppc64Architecture */
 
@@ -1480,10 +1501,6 @@ XFREE86JAPANESEDOCDIR = $(DOCDIR)/Japane
 # define UseRgbTxt	YES
 #endif
 
-#ifndef ManSuffix
-# define ManSuffix	1
-#endif
-
 /*
  * GNU Compiler stuff
  */
@@ -1557,37 +1574,37 @@ XFREE86JAPANESEDOCDIR = $(DOCDIR)/Japane
 
 #if HasGcc2 && defined(i386Architecture)
 # ifndef DefaultGcc2i386Opt
-#  define DefaultGcc2i386Opt -O2 -fno-strength-reduce GccAliasingArgs
+#  define DefaultGcc2i386Opt DefaultGcc2OptimizeOpt GccAliasingArgs
 # endif
 #endif
 
 #if HasGcc2 && defined(AMD64Architecture)
 # ifndef DefaultGcc2AMD64Opt
-#  define DefaultGcc2AMD64Opt -O2 -fno-strength-reduce GccAliasingArgs
+#  define DefaultGcc2AMD64Opt DefaultGcc2OptimizeOpt GccAliasingArgs
 # endif
 #endif
 
 #if HasGcc2 && defined(AlphaArchitecture)
 # ifndef DefaultGcc2AxpOpt
-#  define DefaultGcc2AxpOpt -O2 GccAliasingArgs
+#  define DefaultGcc2AxpOpt DefaultGcc2OptimizeOpt GccAliasingArgs
 # endif
 #endif
 
 #if HasGcc2 && defined(Ppc64Architecture)
 # ifndef DefaultGcc2Ppc64Opt
-#  define DefaultGcc2Ppc64Opt -O2 GccAliasingArgs
+#  define DefaultGcc2Ppc64Opt DefaultGcc2OptimizeOpt GccAliasingArgs
 # endif
 #endif
 
 #if HasGcc2 && defined(MipsArchitecture)
 # ifndef DefaultGcc2MipsOpt
-#  define DefaultGcc2MipsOpt -O2 GccAliasingArgs
+#  define DefaultGcc2MipsOpt DefaultGcc2OptimizeOpt GccAliasingArgs
 # endif
 #endif
 
 #if HasGcc2 && defined(PpcArchitecture)
 # ifndef DefaultGcc2PpcOpt
-#  define DefaultGcc2PpcOpt -O2 GccAliasingArgs
+#  define DefaultGcc2PpcOpt DefaultGcc2OptimizeOpt GccAliasingArgs
 # endif
 #endif
 
