From: Evgeni Golov <evgeni@debian.org>
Subject: Use the public suffix list provided by the publicsuffix package
Forwarded: not-needed

diff --git a/publicsuffix.py b/publicsuffix.py
index 5488ab2..18ba329 100644
--- a/publicsuffix.py
+++ b/publicsuffix.py
@@ -17,7 +17,7 @@ class PublicSuffixList(object):
 		"""
 
 		if input_file is None:
-			input_path = os.path.join(os.path.dirname(__file__), 'publicsuffix.txt')
+			input_path = '/usr/share/publicsuffix/effective_tld_names.dat'
 			input_file = codecs.open(input_path, "r", "utf8")
 
 		root = self._build_structure(input_file)
