--- a/dkms	2012-10-01 15:44:50.000000000 -0700
+++ b/dkms	2012-10-01 15:44:25.981366657 -0700
@@ -19,6 +19,7 @@
 #    Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
 #
 
+shopt -s extglob
 
 # All of the variables we will accept from dkms.conf.
 # Does not include directives
@@ -1827,7 +1831,7 @@
 # we do not get word splitting where it would be inconvienent.
 module_status_built() {
     local ret=1 directory ka k a state oifs="$IFS" IFS=''
-    for directory in "$dkms_tree/$1/$2/"${3:-[0-9].*}/${4:-*}; do
+    for directory in "$dkms_tree/$1/$2/"${3:-+([0-9]).*}/${4:-*}; do
 	IFS="$oifs"
 	ka="${directory#$dkms_tree/$1/$2/}"
 	k="${ka%/*}"
