Index: electricsheep-2.7~b12+svn20091224/electricsheep.c
===================================================================
--- electricsheep-2.7~b12+svn20091224.orig/electricsheep.c	2011-07-14 06:24:14.981991391 +0200
+++ electricsheep-2.7~b12+svn20091224/electricsheep.c	2011-07-14 06:30:50.861991426 +0200
@@ -670,7 +670,7 @@
     input_stream_index = -1;
     for (j = 0; j < ictx->nb_streams; j++) {
 	AVCodecContext *enc = ictx->streams[j]->codec;
-	if (CODEC_TYPE_VIDEO == enc->codec_type) {
+	if (AVMEDIA_TYPE_VIDEO == enc->codec_type) {
 	    input_stream_index = j;
 	    break;
 	}
@@ -696,7 +696,7 @@
 
 	output_ctx = avformat_alloc_context();
 
-	ofmt = guess_format(NULL, fname, NULL);
+	ofmt = av_guess_format(NULL, fname, NULL);
 	if (!ofmt) {
 	    fprintf(logout, "could not determine format from %s.\n", fname);
 	    exit(1);
@@ -738,7 +738,7 @@
 	av_init_packet(&opkt);
 	if (av_parser_change(ictx->streams[input_stream_index]->parser, output_ctx->streams[0]->codec,
 			     &opkt.data, &opkt.size, ipkt.data, ipkt.size,
-			     ipkt.flags & PKT_FLAG_KEY))
+			     ipkt.flags & AV_PKT_FLAG_KEY))
 	    opkt.destruct= av_destruct_packet;
 		
 	if (-1 == av_interleaved_write_frame(output_ctx, &opkt)) {
