Index: electricsheep-2.7~b12+svn20091224/Makefile.am
===================================================================
--- electricsheep-2.7~b12+svn20091224.orig/Makefile.am	2011-07-14 06:47:50.981991518 +0200
+++ electricsheep-2.7~b12+svn20091224/Makefile.am	2011-07-14 06:51:35.741991539 +0200
@@ -9,7 +9,8 @@
 pkgdata_DATA = electricsheep-wait.avi electricsheep-preferences.glade
 
 electricsheep_preferences_CFLAGS = @libglade_CFLAGS@ -DPACKAGE_DATA_DIR=\"$(pkgdatadir)\"
-electricsheep_preferences_LDFLAGS = @libglade_LIBS@ -export-dynamic
+electricsheep_preferences_LDFLAGS = -export-dynamic
+electricsheep_preferences_LDADD = @libglade_LIBS@
 electricsheep_preferences_SOURCES = electricsheep-preferences.c utils.c md5.c
 
 
Index: electricsheep-2.7~b12+svn20091224/Makefile.in
===================================================================
--- electricsheep-2.7~b12+svn20091224.orig/Makefile.in	2011-07-14 06:53:33.251991549 +0200
+++ electricsheep-2.7~b12+svn20091224/Makefile.in	2011-07-14 06:53:31.631991549 +0200
@@ -64,7 +64,7 @@
 	electricsheep_preferences-md5.$(OBJEXT)
 electricsheep_preferences_OBJECTS =  \
 	$(am_electricsheep_preferences_OBJECTS)
-electricsheep_preferences_LDADD = $(LDADD)
+electricsheep_preferences_LDADD = $(LDADD) @libglade_LIBS@
 electricsheep_preferences_LINK = $(CCLD) \
 	$(electricsheep_preferences_CFLAGS) $(CFLAGS) \
 	$(electricsheep_preferences_LDFLAGS) $(LDFLAGS) -o $@
@@ -202,7 +202,7 @@
 electricsheep_LDADD = -lavformat -lavcodec -lavutil -lm -lz
 pkgdata_DATA = electricsheep-wait.avi electricsheep-preferences.glade
 electricsheep_preferences_CFLAGS = @libglade_CFLAGS@ -DPACKAGE_DATA_DIR=\"$(pkgdatadir)\"
-electricsheep_preferences_LDFLAGS = @libglade_LIBS@ -export-dynamic
+electricsheep_preferences_LDFLAGS = -export-dynamic
 electricsheep_preferences_SOURCES = electricsheep-preferences.c utils.c md5.c
 electricsheep_SOURCES = electricsheep.c getdate.c utils.c md5.c
 man1_MANS = electricsheep.man
