Properly escape the use of hyphens in upstream's manpage.
--- trunk.orig/electricsheep.man.in
+++ trunk/electricsheep.man.in
@@ -1,39 +1,39 @@
 .TH ElectricSheep 1 "" "X Version 11"
 .SH NAME
-electricsheep - a distributed screen-saver (version @VERSION@)
+electricsheep \- a distributed screen-saver (version @VERSION@)
 
-electricsheep-preferences - GUI to edit defaults for electricsheep
+electricsheep-preferences \- GUI to edit defaults for electricsheep
 .SH SYNOPSIS
 .B electricsheep
-[\--preferences \fIfile\fP]
-[\--nick \fIname\fP]
-[\--password \fIstring\fP]
-[\--url \fIurl\fP]
-[\--nrepeats \fIN\fP]
-[\--max-repeats \fIN\fP]
-[\--frame-rate \fIN\fP]
-[\--nthreads \fIN\fP]
-[\--server \fIhost/path\fP]
-[\--display-anim \fI0/1\fP]
-[\--show-errors \fI0/1\fP]
-[\--standalone \fI0/1\fP]
-[\--save-dir \fIpath\fP]
-[\--reset-fuse \fIN\fP]
-[\--nice \fIN\fP]
-[\--proxy \fIURL\fP]
-[\--proxy-user \fIuser:password\fP]
-[\--timeout \fIN\fP]
-[\--start-sheep \fIN\fP]
-[\--debug \fI0/1\fP]
+[\-\-preferences \fIfile\fP]
+[\-\-nick \fIname\fP]
+[\-\-password \fIstring\fP]
+[\-\-url \fIurl\fP]
+[\-\-nrepeats \fIN\fP]
+[\-\-max-repeats \fIN\fP]
+[\-\-frame-rate \fIN\fP]
+[\-\-nthreads \fIN\fP]
+[\-\-server \fIhost/path\fP]
+[\-\-display-anim \fI0/1\fP]
+[\-\-show-errors \fI0/1\fP]
+[\-\-standalone \fI0/1\fP]
+[\-\-save-dir \fIpath\fP]
+[\-\-reset-fuse \fIN\fP]
+[\-\-nice \fIN\fP]
+[\-\-proxy \fIURL\fP]
+[\-\-proxy-user \fIuser:password\fP]
+[\-\-timeout \fIN\fP]
+[\-\-start-sheep \fIN\fP]
+[\-\-debug \fI0/1\fP]
 [\-window-id \fIid\fP]
-[\--bracket-begin \fIdate/id\fP]
-[\--bracket-end \fIdate/id\fP]
-[\--data-dir \fIdir\fP]
-[\--logfile \fIfile\fP]
-[\--video-driver \fIname\fP]
+[\-\-bracket-begin \fIdate/id\fP]
+[\-\-bracket-end \fIdate/id\fP]
+[\-\-data-dir \fIdir\fP]
+[\-\-logfile \fIfile\fP]
+[\-\-video-driver \fIname\fP]
 
 .B electricsheep-preferences
-[\--preferences \fI\fP]
+[\-\-preferences \fI\fP]
 
 .SH DESCRIPTION
 
@@ -52,111 +52,111 @@
 
 .SH OPTIONS
 .TP 8
-.B \--preferences \fIfile\fP
+.B \-\-preferences \fIfile\fP
 Give the location for the preferences.  The default is ~/.electricsheep/preferences.xml.
 .TP 8
-.B \--bracket-begin \fIdate/id\fP
+.B \-\-bracket-begin \fIdate/id\fP
 Play no sheep before this one or this time.
 .TP 8
-.B \--bracket-end \fIdate/id\fP
+.B \-\-bracket-end \fIdate/id\fP
 Play no sheep after this one or this time.
 .TP 8
-.B \--data-dir \fIdir\fP
+.B \-\-data-dir \fIdir\fP
 Set the directory to find splash images and other data files.
 .TP 8
-.B \--debug \fI0/1\fP
+.B \-\-debug \fI0/1\fP
 If 1 then print copious debug information.  The default is 0.
 .TP 8
-.B \--display-anim \fI0/1\fP
+.B \-\-display-anim \fI0/1\fP
 If 1 then display the animated sheep, if 0 then do not.  Not
 displaying the sheep allows one to contribute rendering more rendering
 cycles because no CPU time is spent on display.  It also allows one to
 run on a computer without an X display at all.  The default is 1.
 .TP 8
-.B \--frame-rate \fIN\fP
+.B \-\-frame-rate \fIN\fP
 Specify the frame-rate for sheep display in frames per second.  The
 default is 23.  If your client is having trouble completing frames
 because it is spending all its CPU time in the display process then
 decreasing this might help.  Or increase it if you have extra CPU for
 smoother display.
 .TP 8
-.B \--logfile \fIfile\fP
+.B \-\-logfile \fIfile\fP
 Write errors and diagnostics to this file instead of stderr and stdout.
 .TP 8
-.B \--max-megabytes \fIN\fP
+.B \-\-max-megabytes \fIN\fP
 Specify the maximum number of megabytes of disk storage to use to
-store sheep (in the directory specified with --save-dir).  The default
+store sheep (in the directory specified with \-\-save-dir).  The default
 is 2000.  Zero (0) means there is no maximum.
 .TP 8
-.B \--max-repeats \fIN\fP
+.B \-\-max-repeats \fIN\fP
 Maximum number of times to repeat any sheep.  The default is 3.
 .TP 8
-.B \--min-megabytes \fIN\fP
+.B \-\-min-megabytes \fIN\fP
 Specify the minimum number of megabytes of disk storage to leave free.
 The default is 2000. Zero (0) means there is no minimum.
 .TP 8
-.B \--nice \fIN\fP
+.B \-\-nice \fIN\fP
 Specify the priority adjustment for render process and all non-display
 processes.  The default is 10.
 .TP 8
-.B \--nick \fIname\fP
+.B \-\-nick \fIname\fP
 Specify a nickname.  The server keeps credits the frames according to
 nickname and ranks nicknames according to who contributes the most.
 .TP 8
-.B \--password \fIstring\fP
+.B \-\-password \fIstring\fP
 Authenticate to the server using this password.  Note: the password is
 not encrypted!
 .TP 8
-.B \--nrepeats \fIN\fP
+.B \-\-nrepeats \fIN\fP
 Number of times to repeat each sheep.  The default is 1.  Transitions
 between sheep are not repeated.
 .TP 8
-.B \--nthreads \fIN\fP
+.B \-\-nthreads \fIN\fP
 Specify the number of rendering threads.  By default there is one (1).
 .TP 8
-.B \--proxy \fIURL\fP
+.B \-\-proxy \fIURL\fP
 Specify a proxy as per curl(1).
 .TP 8
-.B \--proxy-user \fIuser:password\fP
+.B \-\-proxy-user \fIuser:password\fP
 Specify a proxy user and password as per curl(1).
 .TP 8
-.B \--save-dir \fIpath\fP
+.B \-\-save-dir \fIpath\fP
 Specifies a directory to save the sheep in.  The default is ~/.electricsheep.
 .TP 8
-.B \--show-errors \fI0/1\fP
+.B \-\-show-errors \fI0/1\fP
 If 0 then do not report problems connecting to server.  The default is 1.
 .TP 8
-.B \--standalone \fI0/1\fP
+.B \-\-standalone \fI0/1\fP
 If 1 then run without connecting to the internet at all, just displays
 the sheep already downloaded and do no rendering.  The default is 0.
 .TP 8
-.B \--start-sheep \fIN\fP
+.B \-\-start-sheep \fIN\fP
 Specify the ID of the sheep to display first.  The default is to
 display a random sheep.
 .TP 8
-.B \--timeout \fIN\fP
+.B \-\-timeout \fIN\fP
 Specify the timeout in seconds for server operations.  The default is
 401.
 .TP 8
-.B \--tryagain \fIN\fP
+.B \-\-tryagain \fIN\fP
 Specify the time in seconds to wait before retrying to contact the
 server.  The default is 696.
 .TP 8
-.B \--url \fIname\fP
+.B \-\-url \fIname\fP
 Specify a vanity URL to go with with the nickname.
 .TP 8
-.B \--video-driver \fIname\fP
-Passed on to mplayer -vo, try "gl" and "x11" and "xv" or leave it
+.B \-\-video-driver \fIname\fP
+Passed on to mplayer \-vo, try "gl" and "x11" and "xv" or leave it
 blank for the default.
 .TP 8
-.B \--voting \fI0/1\fP
+.B \-\-voting \fI0/1\fP
 If 1 then enable the voting interface, which is broken in this beta.
 .TP 8
 .B \-window-id \fIX\fP
 Specify in hex the window ID to draw into.  Note the single leading
 dash (this option's syntax is required by xscreensaver).  The default
-is to display in a new window. If both --root and -window-id are
-specified then -window-id takes precedence.
+is to display in a new window. If both \-\-root and \-window\-id are
+specified then \-window-id takes precedence.
 .SH ENVIRONMENT
 .PP
 .TP 8
