Index: libwnck-2.30.0/ltmain.sh
===================================================================
--- libwnck-2.30.0.orig/ltmain.sh	2010-04-23 19:41:31.894859918 +0200
+++ libwnck-2.30.0/ltmain.sh	2010-04-23 19:41:34.679353830 +0200
@@ -4716,6 +4716,11 @@ func_mode_link ()
 	arg=$func_stripname_result
 	;;
 
+      -Wl,--as-needed)
+	deplibs="$deplibs $arg"
+	continue
+	;;
+      
       -Wl,*)
 	func_stripname '-Wl,' '' "$arg"
 	args=$func_stripname_result
