Description: Fix spelling error in manpage
Origin: vendor
Bug: https://rt.cpan.org/Public/Bug/Display.html?id=79400
Forwarded: yes
Author: Salvatore Bonaccorso <carnil@debian.org>
Last-Update: 2012-09-04

--- a/lib/Text/Format.pm
+++ b/lib/Text/Format.pm
@@ -78,7 +78,7 @@
 
 =item B<format> @ARRAY || \@ARRAY || [<FILEHANDLE>] || NOTHING
 
-Allows to do some advanced formatting of text into a paragraph, with
+Allows one to do some advanced formatting of text into a paragraph, with
 indent for first line and body set separately.  Can specify total width
 of text, right fill with spaces or right align or justify (align to both
 margins), right margin and left margin, non-breaking space, two spaces
