diff --git a/xmu.pc.in b/xmu.pc.in
index a086f83..15dabbf 100644
--- a/xmu.pc.in
+++ b/xmu.pc.in
@@ -6,7 +6,7 @@ includedir=@includedir@
 Name: Xmu
 Description: Xmu Library
 Version: @PACKAGE_VERSION@
-Requires: xproto x11 xt
+Requires: xproto xt
 Requires.private: x11 xt xext
 Cflags: -I${includedir}
 Libs: -L${libdir} -lXmu
diff --git a/xmuu.pc.in b/xmuu.pc.in
index 1e91ac4..72fa5f9 100644
--- a/xmuu.pc.in
+++ b/xmuu.pc.in
@@ -6,7 +6,7 @@ includedir=@includedir@
 Name: Xmuu
 Description: Mini Xmu Library
 Version: @PACKAGE_VERSION@
-Requires: xproto x11
+Requires: xproto
 Requires.private: x11
 Cflags: -I${includedir}
 Libs: -L${libdir} -lXmuu
