Description: Fix spelling error in manpage of Algorithm::Diff
Origin: vendor
Bug: http://rt.cpan.org/Ticket/Modify.html?id=59789
Forwarded: http://rt.cpan.org/Ticket/Modify.html?id=59789
Author: Salvatore Bonaccorso <salvatore.bonaccorso@gmail.com>
Last-Update: 2010-07-28

--- a/lib/Algorithm/Diff.pm
+++ b/lib/Algorithm/Diff.pm
@@ -1580,7 +1580,7 @@
 with different order of events.
 
 C<traverse_balanced> might be a bit slower than C<traverse_sequences>,
-noticable only while processing huge amounts of data.
+noticeable only while processing huge amounts of data.
 
 The C<sdiff> function of this module 
 is implemented as call to C<traverse_balanced>.
