Description: fix pod errors
Author: Jonathan Yu <jawnsy@cpan.org>
Origin: vendor
Bug: https://rt.cpan.org/Ticket/Display.html?id=80563
Forwarded: https://rt.cpan.org/Ticket/Display.html?id=80563
Last-Update: 2012-11-02
--- a/lib/Algorithm/Diff.pm
+++ b/lib/Algorithm/Diff.pm
@@ -986,8 +986,6 @@
 a contiguous section of items which should be added, deleted, replaced,
 or left unchanged.
 
-=over 4
-
 The following summary of all of the methods looks a lot like Perl code
 but some of the symbols have different meanings:
 
@@ -1018,6 +1016,8 @@
 Passing in C<undef> for an optional argument is always treated the same
 as if no argument were passed in.
 
+=over
+
 =item C<Next>
 
     $pos = $diff->Next();    # Move forward 1 hunk
