Index: ft2demos-2.3.9/src/ftcommon.c
===================================================================
--- ft2demos-2.3.9.orig/src/ftcommon.c
+++ ft2demos-2.3.9/src/ftcommon.c
@@ -445,6 +445,7 @@
 
         if ( file == NULL )  /* shouldn't happen */
         {
+          free( (void *)font->filepathname );
           free( font );
           return FT_Err_Invalid_Argument;
         }
@@ -454,7 +455,15 @@
         fseek( file, 0, SEEK_SET );
 
         font->file_address = malloc( file_size );
-        fread( font->file_address, 1, file_size, file );
+
+        if (fread( font->file_address, 1, file_size, file ) < file_size)
+        {
+          fclose( file );
+          free( font->file_address );
+          free( (void *)font->filepathname );
+          free( font );
+          return FT_Err_Invalid_Argument;
+        }
 
         font->file_size    = file_size;
 
Index: ft2demos-2.3.9/src/ftdiff.c
===================================================================
--- ft2demos-2.3.9.orig/src/ftdiff.c
+++ ft2demos-2.3.9/src/ftdiff.c
@@ -1138,8 +1138,13 @@
 
         if ( text != NULL )
         {
-          fread( text, tsize, 1, tfile );
-          text[tsize] = 0;
+          if (fread( text, tsize, 1, tfile ) < 1) {
+            fprintf( stderr, "could not read '%s'\n", textfile );
+            free( text );
+            text = (unsigned char *)default_text;
+          } else {
+            text[tsize] = 0;
+          }
         }
         else
         {
