Index: graph/x11/rules.mk
===================================================================
--- graph/x11/rules.mk.orig	2003-06-16 01:37:10.000000000 -0700
+++ graph/x11/rules.mk	2006-03-05 11:57:36.000000000 -0800
@@ -67,9 +67,11 @@
   # The GRAPH_LINK variable is expanded each time an executable is linked
   # against the graphics library.
   #
-  ifeq ($(PLATFORM),unix)
-    GRAPH_LINK += $(X11_LIB:%=-R%)
-  endif
+  # No, we do not want rpath on Debian; reversing YAMANO-UCHI Hidetoshi's
+  # 2003-06-13 change.  (Anthony Fok, 2003-08-28)
+  #ifeq ($(PLATFORM),unix)
+  #  GRAPH_LINK += $(X11_LIB:%=-R%)
+  #endif
   GRAPH_LINK += $(X11_LIB:%=-L%) -lX11
 
   # Solaris needs a -lsocket in GRAPH_LINK.
