Description: Fix build failure with [-Werror=format-security]
Author: Michael Biebl <biebl@debian.org>
Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=643424
Bug: https://bugzilla.gnome.org/show_bug.cgi?id=662251
Index: libgnomecups-0.2.3/libgnomecups/gnome-cups-request.c
===================================================================
--- libgnomecups-0.2.3.orig/libgnomecups/gnome-cups-request.c	2011-10-20 00:46:59.000000000 +0200
+++ libgnomecups-0.2.3/libgnomecups/gnome-cups-request.c	2011-10-20 00:48:51.692022324 +0200
@@ -352,7 +352,7 @@
 		if (status > IPP_OK_CONFLICT && status != IPP_NOT_FOUND) {
 			g_warning ("IPP request failed with status %d", status);
 			if (request->error != NULL)
-				*(request->error) = g_error_new (GNOME_CUPS_ERROR, 
+				*(request->error) = g_error_new_literal (GNOME_CUPS_ERROR,
 								 status, ippErrorString (status));
 		} else if (request->response && _gnome_cups_debug) {
 			g_print ("response = \n");
@@ -363,7 +363,7 @@
 			request->path, request->output_fd);
 		request->response = NULL;
 		if (HTTP_OK != status && request->error != NULL)
-			*(request->error) = g_error_new (GNOME_CUPS_ERROR, 
+			*(request->error) = g_error_new_literal (GNOME_CUPS_ERROR,
 							 status, httpStatus (status));
 	} else {
 		g_warning ("Neither request nor output_fd set");
