Index: libgnomecups-0.2.3/libgnomecups/gnome-cups-i18n.c
===================================================================
--- libgnomecups-0.2.3.orig/libgnomecups/gnome-cups-i18n.c	2007-01-02 01:18:51.000000000 +0100
+++ libgnomecups-0.2.3/libgnomecups/gnome-cups-i18n.c	2012-03-25 15:26:31.962736258 +0200
@@ -1,5 +1,5 @@
 #include <config.h>
-#include <glib/gmacros.h>
+#include <glib.h>
 #include "gnome-cups-i18n.h"
 
 #ifdef ENABLE_NLS
Index: libgnomecups-0.2.3/libgnomecups/gnome-cups-init.h
===================================================================
--- libgnomecups-0.2.3.orig/libgnomecups/gnome-cups-init.h	2007-01-02 01:18:51.000000000 +0100
+++ libgnomecups-0.2.3/libgnomecups/gnome-cups-init.h	2012-03-25 15:26:31.938736257 +0200
@@ -1,8 +1,7 @@
 #ifndef GNOME_CUPS_INIT
 #define GNOME_CUPS_INIT
 
-#include <glib/gtypes.h>
-#include <glib/gmacros.h>
+#include <glib.h>
 
 G_BEGIN_DECLS
 
