Description: Fix a typo
 Fix a typo in a class member name.
Author: Julien BLACHE <jblache@debian.org>

Index: airport-utils-2/airport2config/airport/Airport2InfoElementVector.java
===================================================================
--- airport-utils-2.orig/airport2config/airport/Airport2InfoElementVector.java	2010-03-11 20:55:56.852753404 +0100
+++ airport-utils-2/airport2config/airport/Airport2InfoElementVector.java	2010-03-11 20:58:17.703753477 +0100
@@ -60,7 +60,7 @@
 			//read the type
 			byte[] typeBytes = new byte[4];
 			byteStream.read(typeBytes, 0, 4);
-			element.type = getIntegerValue(typeBytes);
+			element.dataType = getIntegerValue(typeBytes);
 			
 			//read the length
 			byte[] lengthBytes = new byte[4];
