From: David Paleino <dapal@debian.org>
Subject: prefer gksu over gksudo, fixes weird behaviour on some systems
Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=575403
Forwarded: no

---
 wicd/misc.py |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- wicd.orig/wicd/misc.py
+++ wicd/wicd/misc.py
@@ -525,7 +525,7 @@ def choose_sudo_prog(prog_num=0):
     if desktop_env == "kde":
         progs = ["kdesu", "kdesudo", "ktsuss"]
     else:
-        progs = ["gksudo", "gksu", "ktsuss"]
+        progs = ["gksu", "gksudo", "ktsuss"]
         
     for prog in progs:
         paths.extend([os.path.join(p, prog) for p in env_path])
