--- EDITME.exim4-light	2012-05-29 19:16:05.000000000 +0200
+++ EDITME.exim4-light	2012-05-29 19:17:05.000000000 +0200
@@ -697,13 +697,13 @@ SUPPORT_TLS=yes
 
 # Uncomment one of these settings if you are using OpenSSL; pkg-config vs not
 # USE_OPENSSL_PC=openssl
-# TLS_LIBS=-lssl -lcrypto
+TLS_LIBS=-lssl -lcrypto
 
 # Uncomment the first and either the second or the third of these if you
 # are using GnuTLS.  If you have pkg-config, then the second, else the third.
-USE_GNUTLS=yes
+# USE_GNUTLS=yes
 # USE_GNUTLS_PC=gnutls
-TLS_LIBS=-lgnutls
+# TLS_LIBS=-lgnutls
 
 # If you are running Exim as a server, note that just building it with TLS
 # support is not all you need to do. You also need to set up a suitable
