--- exim_monitor/EDITME	2012-05-18 05:04:36.000000000 +0200
+++ EDITME.eximon	2012-05-18 19:53:04.000000000 +0200
@@ -1,6 +1,7 @@
 ##################################################
 #                The Exim Monitor                #
 ##################################################
+# -*- makefile -*-
 
 # This is the template for the Exim monitor's main build-time configuration
 # file. It contains settings that are independent of any operating system. It
