--- EDITME.exim4-light	2012-05-18 20:11:24.000000000 +0200
+++ EDITME.exim4-heavy	2012-05-18 20:13:56.000000000 +0200
@@ -212,7 +212,7 @@ ROUTER_REDIRECT=yes
 
 # This one is very special-purpose, so is not included by default.
 
-# ROUTER_IPLOOKUP=yes
+ROUTER_IPLOOKUP=yes
 
 
 #------------------------------------------------------------------------------
@@ -244,7 +244,7 @@ TRANSPORT_LMTP=yes
 
 SUPPORT_MAILDIR=yes
 SUPPORT_MAILSTORE=yes
-# SUPPORT_MBX=yes
+SUPPORT_MBX=yes
 
 
 #------------------------------------------------------------------------------
@@ -300,14 +300,14 @@ LOOKUP_DNSDB=yes
 LOOKUP_CDB=yes
 LOOKUP_DSEARCH=yes
 # LOOKUP_IBASE=yes
-# LOOKUP_LDAP=yes
-# LOOKUP_MYSQL=yes
+LOOKUP_LDAP=yes
+LOOKUP_MYSQL=yes
 LOOKUP_NIS=yes
 # LOOKUP_NISPLUS=yes
 # LOOKUP_ORACLE=yes
 LOOKUP_PASSWD=yes
-# LOOKUP_PGSQL=yes
-# LOOKUP_SQLITE=yes
+LOOKUP_PGSQL=yes
+LOOKUP_SQLITE=yes
 # LOOKUP_SQLITE_PC=sqlite3
 # LOOKUP_WHOSON=yes
 
@@ -328,7 +328,7 @@ LOOKUP_PASSWD=yes
 # with Solaris 7 onwards. Uncomment whichever of these you are using.
 
 # LDAP_LIB_TYPE=OPENLDAP1
-# LDAP_LIB_TYPE=OPENLDAP2
+LDAP_LIB_TYPE=OPENLDAP2
 # LDAP_LIB_TYPE=NETSCAPE
 # LDAP_LIB_TYPE=SOLARIS
 
@@ -366,6 +366,9 @@ LOOKUP_PASSWD=yes
 # LOOKUP_LIBS=-L/usr/local/lib -lldap -llber -lmysqlclient -lpq -lgds -lsqlite3
 
 
+LOOKUP_INCLUDE=-I/usr/include/mysql -I`pg_config --includedir`
+LOOKUP_LIBS=-lldap -llber -lmysqlclient -lpq -lsqlite3
+
 #------------------------------------------------------------------------------
 # Compiling the Exim monitor: If you want to compile the Exim monitor, a
 # program that requires an X11 display, then EXIM_MONITOR should be set to the
@@ -374,7 +377,7 @@ LOOKUP_PASSWD=yes
 # files are defaulted in the OS/Makefile-Default file, but can be overridden in
 # local OS-specific make files.
 
-EXIM_MONITOR=eximon.bin
+# EXIM_MONITOR=eximon.bin
 
 
 #------------------------------------------------------------------------------
@@ -384,14 +387,14 @@ EXIM_MONITOR=eximon.bin
 # and the MIME ACL. Please read the documentation to learn more about these
 # features.
 
-# WITH_CONTENT_SCAN=yes
+WITH_CONTENT_SCAN=yes
 
 # If you want to use the deprecated "demime" condition in the DATA ACL,
 # uncomment the line below. Doing so will also explicitly turn on the
 # WITH_CONTENT_SCAN option. If possible, use the MIME ACL instead of
 # the "demime" condition.
 
-# WITH_OLD_DEMIME=yes
+WITH_OLD_DEMIME=yes
 
 # If you're using ClamAV and are backporting fixes to an old version, instead
 # of staying current (which is the more usual approach) then you may need to
@@ -578,14 +581,14 @@ WHITELIST_D_MACROS=OUTGOING
 # configuration to make use of the mechanism(s) selected.
 
 AUTH_CRAM_MD5=yes
-# AUTH_CYRUS_SASL=yes
-# AUTH_DOVECOT=yes
+AUTH_CYRUS_SASL=yes
+AUTH_DOVECOT=yes
 # AUTH_GSASL=yes
 # AUTH_GSASL_PC=libgsasl
 # AUTH_HEIMDAL_GSSAPI=yes
 # AUTH_HEIMDAL_GSSAPI_PC=heimdal-gssapi
 AUTH_PLAINTEXT=yes
-# AUTH_SPA=yes
+AUTH_SPA=yes
 
 
 #------------------------------------------------------------------------------
@@ -595,7 +598,7 @@ AUTH_PLAINTEXT=yes
 # Similarly for GNU SASL, unless pkg-config is used via AUTH_GSASL_PC.
 # Ditto for AUTH_HEIMDAL_GSSAPI(_PC).
 
-# AUTH_LIBS=-lsasl2
+AUTH_LIBS=-lsasl2
 # AUTH_LIBS=-lgsasl
 # AUTH_LIBS=-lgssapi -lheimntlm -lkrb5 -lhx509 -lcom_err -lhcrypto -lasn1 -lwind -lroken -lcrypt
 
@@ -830,7 +833,7 @@ ZCAT_COMMAND=/bin/zcat
 # (version 5.004 or later) installed, set EXIM_PERL to perl.o. Using embedded
 # Perl costs quite a lot of resources. Only do this if you really need it.
 
-# EXIM_PERL=perl.o
+EXIM_PERL=perl.o
 
 
 #------------------------------------------------------------------------------
@@ -840,7 +843,7 @@ ZCAT_COMMAND=/bin/zcat
 # that the local_scan API is made available by the linker. You may also need
 # to add -ldl to EXTRALIBS so that dlopen() is available to Exim.
 
-# EXPAND_DLFUNC=yes
+EXPAND_DLFUNC=yes
 
 
 #------------------------------------------------------------------------------
@@ -850,11 +853,11 @@ ZCAT_COMMAND=/bin/zcat
 # support, which is intended for use in conjunction with the SMTP AUTH
 # facilities, is included only when requested by the following setting:
 
-# SUPPORT_PAM=yes
+SUPPORT_PAM=yes
 
 # You probably need to add -lpam to EXTRALIBS, and in some releases of
 # GNU/Linux -ldl is also needed.
-EXTRALIBS=-ldl
+EXTRALIBS=-lpam -export-dynamic
 
 
 #------------------------------------------------------------------------------
@@ -1174,7 +1177,7 @@ TMPDIR="/tmp"
 # local part) can be increased by changing this value. It should be set to
 # a multiple of 16.
 
-# MAX_NAMED_LIST=16
+MAX_NAMED_LIST=32
 
 
 #------------------------------------------------------------------------------
