Index: b/src/lib/libcurses.c
===================================================================
--- a/src/lib/libcurses.c	2007-12-10 08:25:16.000000000 +0200
+++ b/src/lib/libcurses.c	2007-12-10 08:25:48.000000000 +0200
@@ -65,7 +65,8 @@ static Gpm_Event ev;
 	{
 	FD_ZERO(&selSet);
 	FD_SET(fd,&selSet);
-	FD_SET(gpm_fd,&selSet);
+	if (gpm_fd>-1)
+	  FD_SET(gpm_fd,&selSet);
 	gpm_timeout.tv_sec=SELECT_TIME;
 	flag=select(max+1,&selSet,(fd_set *)NULL,(fd_set *)NULL,&gpm_timeout);
 	}
