Index: b/src/prog/gpm-root.y
===================================================================
--- a/src/prog/gpm-root.y	2008-05-29 02:26:09.000000000 +0300
+++ b/src/prog/gpm-root.y	2008-06-11 06:21:41.000000000 +0300
@@ -44,7 +44,6 @@
 #include <sys/stat.h>       /* fstat() */
 #include <sys/utsname.h>    /* uname() */
 #include <termios.h>        /* winsize */
-#include <linux/limits.h>   /* OPEN_MAX */
 #include <linux/vt.h>       /* VT_ACTIVATE */
 #include <linux/keyboard.h> /* K_SHIFT */
 #include <utmp.h>         
Index: b/src/daemon/processspecial.c
===================================================================
--- a/src/daemon/processspecial.c	2008-06-11 06:22:28.000000000 +0300
+++ b/src/daemon/processspecial.c	2008-06-11 06:22:36.000000000 +0300
@@ -25,7 +25,6 @@
 
 /* This file is compiled conditionally, see the Makefile */
 
-#include <linux/limits.h> /* for OPEN_MAX */
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
