Description: Correctly check build architecture.
Author: Michael Meskes <meskes@debian.org>

diff -urNad virtualbox-ose-3.0.8-dfsg~/configure virtualbox-ose-3.0.8-dfsg/configure
--- virtualbox-ose-3.0.8-dfsg~/configure	2009-10-07 11:27:13.000000000 +0200
+++ virtualbox-ose-3.0.8-dfsg/configure	2009-10-07 13:41:49.012963906 +0200
@@ -314,7 +314,7 @@
 check_environment()
 {
   test_header environment
-  BUILD_CPU=`uname -m`
+  BUILD_CPU=`dpkg-architecture -qDEB_HOST_GNU_CPU`
   [ "$OS" = "solaris" ] && BUILD_CPU=`isainfo | cut -f 1 -d ' '`
   case "$BUILD_CPU" in
     i[3456789]86|x86|i86pc)
