Description: Don't install the selinux-fedora modules.
Author: Felix Geyer <debfx-pkg@fobos.de>

diff -Nur virtualbox-ose-4.0.0-dfsg.orig/src/VBox/Additions/linux/Makefile.kmk virtualbox-ose-4.0.0-dfsg/src/VBox/Additions/linux/Makefile.kmk
--- virtualbox-ose-4.0.0-dfsg.orig/src/VBox/Additions/linux/Makefile.kmk	2010-12-05 19:17:42.000000000 +0100
+++ virtualbox-ose-4.0.0-dfsg/src/VBox/Additions/linux/Makefile.kmk	2011-01-02 11:02:43.379712209 +0100
@@ -323,9 +323,7 @@
 lnx_add_inst-noexec_INSTTYPE = stage
 lnx_add_inst-noexec_SOURCES  = \
 	$(VBOX_REL_X11_ADD_INST)vboxclient.desktop \
-	$(VBOX_REL_X11_ADD_INST)vboxvideo.ids \
-	selinux-fedora/vbox_x11.pp \
-	selinux-fedora/vbox_accel.pp
+	$(VBOX_REL_X11_ADD_INST)vboxclient.desktop
 
 INSTALLS += lnx_add_inst-license
 lnx_add_inst-license_INST     = $(VBOX_LNX_ADD_INST_OUT_DIR)
