--- a/scan/sane/sclpml.c
+++ b/scan/sane/sclpml.c
@@ -1063,20 +1063,17 @@
     hpaio->option[OPTION_ADF_MODE].constraint_type = SANE_CONSTRAINT_STRING_LIST;
     hpaio->option[OPTION_ADF_MODE].constraint.string_list = hpaio->adfModeList;
 
-    // Duplex scanning is supported
-    if (hpaio->supportsDuplex  == 1)
-    {
-       hpaio->option[OPTION_DUPLEX].name = STR_NAME_DUPLEX;
-       hpaio->option[OPTION_DUPLEX].title = STR_TITLE_DUPLEX;
-       hpaio->option[OPTION_DUPLEX].desc = STR_DESC_DUPLEX;
-       hpaio->option[OPTION_DUPLEX].type = SANE_TYPE_BOOL;
-       hpaio->option[OPTION_DUPLEX].unit = SANE_UNIT_NONE;
-       hpaio->option[OPTION_DUPLEX].size = sizeof( SANE_Bool );
-       hpaio->option[OPTION_DUPLEX].cap = SANE_CAP_SOFT_SELECT |
+    hpaio->option[OPTION_DUPLEX].name = STR_NAME_DUPLEX;
+    hpaio->option[OPTION_DUPLEX].title = STR_TITLE_DUPLEX;
+    hpaio->option[OPTION_DUPLEX].desc = STR_DESC_DUPLEX;
+    hpaio->option[OPTION_DUPLEX].type = SANE_TYPE_BOOL;
+    hpaio->option[OPTION_DUPLEX].unit = SANE_UNIT_NONE;
+    hpaio->option[OPTION_DUPLEX].size = sizeof( SANE_Bool );
+    hpaio->option[OPTION_DUPLEX].cap = SANE_CAP_SOFT_SELECT |
                                        SANE_CAP_SOFT_DETECT |
                                        SANE_CAP_ADVANCED;
-       hpaio->option[OPTION_DUPLEX].constraint_type = SANE_CONSTRAINT_NONE;
-    }
+    hpaio->option[OPTION_DUPLEX].constraint_type = SANE_CONSTRAINT_NONE;
+
     hpaio->option[GROUP_GEOMETRY].title = STR_TITLE_GEOMETRY;
     hpaio->option[GROUP_GEOMETRY].type = SANE_TYPE_GROUP;
     hpaio->option[GROUP_GEOMETRY].cap = SANE_CAP_ADVANCED;
