Description: Fix broken test
Forwarded: not-needed
Author: Mathieu Malaterre <malat@debian.org>

Index: charls-1.0/defaulttraits.h
===================================================================
--- charls-1.0.orig/defaulttraits.h	2011-04-27 21:06:39.000000000 +0200
+++ charls-1.0/defaulttraits.h	2011-04-27 21:06:47.000000000 +0200
@@ -2,7 +2,7 @@
 // (C) Jan de Vaan 2007-2010, all rights reserved. See the accompanying "License.txt" for licensed use. 
 // 
 
-
+#include "header.h"
 #ifndef CHARLS_DEFAULTTRAITS
 #define CHARLS_DEFAULTTRAITS
 
