Description: Fix broken installation rules
Forwarded: not-needed
Author: Mathieu Malaterre <malat@debian.org>

Index: charls-1.0/CMakeLists.txt
===================================================================
--- charls-1.0.orig/CMakeLists.txt	2011-04-27 21:06:40.000000000 +0200
+++ charls-1.0/CMakeLists.txt	2011-04-27 21:06:42.000000000 +0200
@@ -12,6 +12,11 @@
 ENDIF(CMAKE_COMPILER_IS_GNUCC)
 ENDIF(NOT CMAKE_BUILD_TYPE)
 
+SET( charls_HEADERS "colortransform.h"  "context.h" "decoderstrategy.h"  "encoderstrategy.h"  "interface.h"
+    "losslesstraits.h"  "scan.h"    "streams.h" "config.h"          "contextrunmode.h"  "defaulttraits.h"
+    "header.h"           "lookuptable.h"  "processline.h"      "util.h" "publictypes.h"
+)
+
 OPTION(charls_BUILD_SHARED_LIBS "Build CharLS with shared libraries." OFF)
 SET(BUILD_SHARED_LIBS ${charls_BUILD_SHARED_LIBS})
 
@@ -21,3 +26,7 @@
 add_executable(charlstest test/main.cpp test/time.cpp test/util.cpp test/bitstreamdamage.cpp test/compliance.cpp test/performance.cpp test/dicomsamples.cpp)
 target_link_libraries (charlstest CharLS)
 
+# Installs the header files into the {build_dir}/include/libcharls directory
+install(FILES ${charls_HEADERS} DESTINATION include/CharLS)
+
+
