Description: This patch corrects some upstream spelling errors.
Author: Ignace Mouzannar <mouzannar@gmail.com>
Last-Update: 2011-04-22

--- a/src/kdocker.cpp
+++ b/src/kdocker.cpp
@@ -125,7 +125,7 @@
     out << "-h     \t" << tr("Display this help") << endl;
     out << "-i file\t" << tr("Set the tray icon to file") << endl;
     out << "-l     \t" << tr("Iconify when focus lost") << endl;
-    out << "-m     \t" << tr("Keep application window showing (dont hide on dock)") << endl;
+    out << "-m     \t" << tr("Keep application window showing (don't hide on dock)") << endl;
     out << "-n name\t" << tr("Match window based on window title") << endl;
     out << "-o     \t" << tr("Iconify when obscured") << endl;
     out << "-p secs\t" << tr("Set ballooning timeout (popup time)") << endl;
--- a/i18n/kdocker_it_IT.ts
+++ b/i18n/kdocker_it_IT.ts
@@ -86,7 +86,7 @@
     </message>
     <message>
         <location filename="../src/kdocker.cpp" line="128"/>
-        <source>Keep application window showing (dont hide on dock)</source>
+        <source>Keep application window showing (don't hide on dock)</source>
         <translation>Mantieni in primo piano la finestra (non nascondere quando agganciata)</translation>
     </message>
     <message>
--- a/i18n/base.ts
+++ b/i18n/base.ts
@@ -80,7 +80,7 @@
     </message>
     <message>
         <location filename="../src/kdocker.cpp" line="128"/>
-        <source>Keep application window showing (dont hide on dock)</source>
+        <source>Keep application window showing (don't hide on dock)</source>
         <translation type="unfinished"></translation>
     </message>
     <message>
--- a/INSTALL
+++ b/INSTALL
@@ -1,5 +1,5 @@
 Prerequisites
-1. Make sure you have Qt 4 installed and working. If you dont know what Qt is,
+1. Make sure you have Qt 4 installed and working. If you don't know what Qt is,
 go to http://qt.nokia.com/products/. You will need qmake to compile.
 
 Building
