Description: include missing header to avoid FTBFS with gcc 4.7
Origin: vendor
Bug-Debian: http://bugs.debian.org/667223
Forwarded: no
Author: gregor herrmann <gregoa@debian.org>
Last-Update: 2012-05-06

--- a/solutions/qtsingleapplication/qtlocalpeer.cpp
+++ b/solutions/qtsingleapplication/qtlocalpeer.cpp
@@ -48,6 +48,7 @@
 #include "qtlocalpeer.h"
 #include <QtCore/QCoreApplication>
 #include <QtCore/QTime>
+#include <unistd.h>
 
 #if defined(Q_OS_WIN)
 #include <QtCore/QLibrary>
