Index: libsecret-0.9/libsecret/tests/Makefile.am
===================================================================
--- libsecret-0.9.orig/libsecret/tests/Makefile.am	2012-08-16 11:30:14.000000000 +0200
+++ libsecret-0.9/libsecret/tests/Makefile.am	2012-08-20 17:35:10.183222354 +0200
@@ -135,7 +135,7 @@
 test-py:
 	@for py in $(PY_TESTS); do echo "TEST: $$py"; $(PY_ENV) python $(srcdir)/$$py; done
 
-test: test-c test-py test-js $(VALA_TEST_TARGET)
+test: test-c test-py $(VALA_TEST_TARGET)
 
 # ------------------------------------------------------------------
 # INTROSPECTION
