diff --git a/libsecret/Makefile.am b/libsecret/Makefile.am
index 46727d7..8436792 100644
--- a/libsecret/Makefile.am
+++ b/libsecret/Makefile.am
@@ -134,10 +134,10 @@ SecretUnstable_0_gir_CFLAGS = -I$(top_srcdir) -I$(top_builddir) -DSECRET_COMPILA
 SecretUnstable_0_gir_FILES = $(UNSTABLE_FILES)
 SecretUnstable_0_gir_SCANNERFLAGS = --c-include "libsecret/secret.h" --identifier-prefix=Secret --symbol-prefix=secret --include-uninstalled=$(builddir)/Secret-@SECRET_MAJOR@.gir
 
-girdir = $(datadir)/gir-1.0
+girdir = $(INTROSPECTION_GIRDIR)
 gir_DATA = $(INTROSPECTION_GIRS)
 
-typelibsdir = $(libdir)/girepository-1.0
+typelibsdir = $(INTROSPECTION_TYPELIBDIR)
 typelibs_DATA = $(INTROSPECTION_GIRS:.gir=.typelib)
 
 if ENABLE_VAPIGEN
