From 2c9a0e86055f1e86ca5cdde421f5f8c9a48b0194 Mon Sep 17 00:00:00 2001
From: Phil Pennock <pdp@exim.org>
Date: Wed, 6 Jun 2012 19:46:40 -0400
Subject: [PATCH] BUGFIX: forced-fail smtp option tls_sni would dereference
 NULL

---
 src/tls-openssl.c |    6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/src/tls-openssl.c b/src/tls-openssl.c
index 22c0730..17cc721 100644
--- a/src/tls-openssl.c
+++ b/src/tls-openssl.c
@@ -1289,7 +1289,11 @@ if (sni)
   {
   if (!expand_check(sni, US"tls_sni", &tls_sni))
     return FAIL;
-  if (!Ustrlen(tls_sni))
+  if (tls_sni == NULL)
+    {
+    DEBUG(D_tls) debug_printf("Setting TLS SNI forced to fail, not sending\n");
+    }
+  else if (!Ustrlen(tls_sni))
     tls_sni = NULL;
   else
     {
-- 
1.7.10

