Index: repo/src/Makefile.in
===================================================================
--- repo.orig/src/Makefile.in	2013-01-17 17:11:04.882322898 +0100
+++ repo/src/Makefile.in	2013-01-17 17:23:01.216239392 +0100
@@ -7,8 +7,8 @@
 MAGICFILES=$(wildcard $(MAGIC)/$(MAGIC).*)
 prefix=@prefix@
 exec_prefix=@exec_prefix@
-ETC=@sysconfdir@/binwalk
-BIN=@bindir@
+ETC=$(DESTDIR)@sysconfdir@/binwalk
+BIN=$(DESTDIR)@bindir@
 MAGICPATH=$(ETC)/$(MAGIC)
 EXTRACTPATH=$(ETC)/$(EXTRACT_FILE)
 
