Description: Handle pathnames with spaces in svn diff format
Author: Stuart Prescott <stuart+debian@nanonanonano.net>

--- a/diffstat.c
+++ b/diffstat.c
@@ -1444,6 +1444,10 @@
 			       &hour, &minute, &second) == 9
 			&& date_delims(yrmon, monday)
 			&& !version_num(b_fname))
+		    || (sscanf(buffer,
+			       "*** %[^\t]\t(%[^)])\t(%[^)])",
+			       b_fname, b_temp1, b_temp2) == 3
+			&& !version_num(b_fname))
 		    || sscanf(buffer,
 			      "*** %[^\t ]%[\t ]%[^ ] %[^ ] %d %d:%d:%d %d",
 			      b_fname,
