From: Etienne Millon <etienne.millon@gmail.com>
Date: Mon, 5 Sep 2011 13:35:10 +0200
Subject: Set minimal bound for libmpd dep to 0.20

---
 configure.ac |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index be9e883..c2721c4 100644
--- a/configure.ac
+++ b/configure.ac
@@ -217,7 +217,7 @@ AC_SUBST(gmodule_LIBS)
 AC_SUBST(gmodule_CFLAGS)
 
 # libmpd
-PKG_CHECK_MODULES([libmpd], libmpd >= 0.20.95)
+PKG_CHECK_MODULES([libmpd], libmpd >= 0.20.0)
 AC_SUBST(libmpd_LIBS)
 AC_SUBST(libmpd_CFLAGS)
 
