From: Qball Cow <qball@gmpclient.org>
Date: Wed, 7 Sep 2011 10:16:55 +0200
Subject: Small fixes to glyros and vapi

---
 src/vapi/gmpc.vapi |    6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/vapi/gmpc.vapi b/src/vapi/gmpc.vapi
index 39de1d0..4a38e60 100644
--- a/src/vapi/gmpc.vapi
+++ b/src/vapi/gmpc.vapi
@@ -282,9 +282,9 @@ namespace Gmpc {
         public bool have_appindicator_support();
     }
 
-   [CCode (cname = "config", cheader_filename="plugin.h")]
-    static Settings config;
-    [CCode (cheader_filename="config1.h")]
+	[CCode (cname = "config", cheader_filename="plugin.h")]
+	static Settings config;
+    [CCode (cheader_filename="config1.h",cname="config_obj")]
         [Compact]
         [Immutable]
     public class Settings {
