diff -ur gmpc_11.8.16/src/vala/gmpc-mpddata-treeview-tooltip.vala goodgmpc/src/vala/gmpc-mpddata-treeview-tooltip.vala
--- gmpc_11.8.16/src/vala/gmpc-mpddata-treeview-tooltip.vala	2011-08-16 23:09:55.000000000 -0400
+++ goodgmpc/src/vala/gmpc-mpddata-treeview-tooltip.vala	2012-12-29 08:59:19.636752348 -0500
@@ -43,7 +43,7 @@
             return false;
         }
 
-        if(!this.par_widget.get_tooltip_context(out x, out y,keyboard_tip, out model, out path, out iter)){
+        if(!this.par_widget.get_tooltip_context(ref x, ref y,keyboard_tip, out model, out path, out iter)){
             this.checksum = null;
             return false;
         }
