diff --git a/common/Makefile.in b/common/Makefile.in
index aff4c54..7f05e4b 100644
--- a/common/Makefile.in
+++ b/common/Makefile.in
@@ -45,6 +45,9 @@ CPPFLAGS = -I$(incdir) -I$(top_builddir) -I$(top_srcdir) @CPPFLAGS@
 # Usually -O and/or -g
 CFLAGS = @CFLAGS@
 
+# Linker flags
+LDFLAGS = @LDFLAGS@
+
 # OS dependent defines
 DEFS = @OSDEFS@ -D_PATH_SUDO_CONF=\"$(sysconfdir)/sudo.conf\"
 
@@ -67,7 +70,7 @@ Makefile: $(srcdir)/Makefile.in
 	$(LIBTOOL) --mode=compile $(CC) -c -o $@ $(CPPFLAGS) $(CFLAGS) $(DEFS) $<
 
 libcommon.la: $(LTOBJS)
-	$(LIBTOOL) --mode=link $(CC) -o $@ $(LTOBJS) -no-install
+	$(LIBTOOL) --mode=link $(CC) $(LDFLAGS) -o $@ $(LTOBJS) -no-install
 
 pre-install:
 
diff --git a/compat/Makefile.in b/compat/Makefile.in
index d0361ce..93672ac 100644
--- a/compat/Makefile.in
+++ b/compat/Makefile.in
@@ -45,6 +45,9 @@ CPPFLAGS = -I$(incdir) -I$(top_builddir) -I$(top_srcdir) @CPPFLAGS@
 # Usually -O and/or -g
 CFLAGS = @CFLAGS@
 
+# Linker flags
+LDFLAGS = @LDFLAGS@
+
 # OS dependent defines
 DEFS = @OSDEFS@
 
@@ -75,19 +78,19 @@ Makefile: $(srcdir)/Makefile.in
 	$(LIBTOOL) --mode=compile $(CC) -c -o $@ $(CPPFLAGS) $(CFLAGS) $(DEFS) $<
 
 libreplace.la: $(LTLIBOBJS)
-	$(LIBTOOL) --mode=link $(CC) -o $@ $(LTLIBOBJS) -no-install
+	$(LIBTOOL) --mode=link $(CC) $(LDFLAGS) -o $@ $(LTLIBOBJS) -no-install
 
 siglist.c: mksiglist
 	./mksiglist > $@
 
 mksiglist: $(srcdir)/mksiglist.c $(srcdir)/mksiglist.h $(incdir)/missing.h $(top_builddir)/config.h
-	$(CC) $(CPPFLAGS) $(CFLAGS) $(DEFS) $(srcdir)/mksiglist.c -o $@
+	$(CC) $(CPPFLAGS) $(CFLAGS) $(LDFLAGS) $(DEFS) $(srcdir)/mksiglist.c -o $@
 
 fnm_test: fnm_test.o libreplace.la
-	$(LIBTOOL) --mode=link $(CC) -o $@ fnm_test.o libreplace.la
+	$(LIBTOOL) --mode=link $(CC) $(LDFLAGS) -o $@ fnm_test.o libreplace.la
 
 globtest: globtest.o libreplace.la
-	$(LIBTOOL) --mode=link $(CC) -o $@ globtest.o libreplace.la
+	$(LIBTOOL) --mode=link $(CC) $(LDFLAGS) -o $@ globtest.o libreplace.la
 
 $(srcdir)/mksiglist.h: $(srcdir)/siglist.in
 	if [ -n "$(DEVEL)" ]; then \
diff --git a/plugins/sudoers/Makefile.in b/plugins/sudoers/Makefile.in
index 8769d38..0eb71e9 100644
--- a/plugins/sudoers/Makefile.in
+++ b/plugins/sudoers/Makefile.in
@@ -56,6 +56,9 @@ CPPFLAGS = -I$(incdir) -I$(top_builddir) -I$(devdir) -I$(srcdir) -I$(top_srcdir)
 # Usually -O and/or -g
 CFLAGS = @CFLAGS@
 
+# Linker flags
+LDFLAGS = @LDFLAGS@
+
 # Flags to pass to the link stage
 LDFLAGS = @LDFLAGS@
 SUDOERS_LDFLAGS = $(LDFLAGS) @SUDOERS_LDFLAGS@
@@ -163,7 +166,7 @@ Makefile: $(srcdir)/Makefile.in
 	(cd $(top_builddir) && ./config.status --file plugins/sudoers/Makefile)
 
 libparsesudoers.la: $(LIBPARSESUDOERS_OBJS)
-	$(LIBTOOL) --mode=link $(CC) -o $@ $(LIBPARSESUDOERS_OBJS) -no-install
+	$(LIBTOOL) --mode=link $(CC) $(LDFLAGS) -o $@ $(LIBPARSESUDOERS_OBJS) -no-install
 
 sudoers.la: $(SUDOERS_OBJS) $(LT_LIBS) libparsesudoers.la
 	$(LIBTOOL) @LT_STATIC@ --mode=link $(CC) $(SUDOERS_LDFLAGS) $(LTLDFLAGS) -o $@ $(SUDOERS_OBJS) libparsesudoers.la $(SUDOERS_LIBS) -module -export-symbols $(srcdir)/sudoers.sym -avoid-version -rpath $(plugindir)
