--- /home/bdale/Desktop/sudo-1.7.2p1/plugins/sudoers/ins_classic.h	2004-02-13 14:36:43.000000000 -0700
+++ sudo/plugins/sudoers/ins_classic.h	2009-11-20 07:31:58.000000000 -0700
@@ -32,7 +32,7 @@
     "Where did you learn to type?",
     "Are you on drugs?",
     "My pet ferret can type better than you!",
-    "You type like i drive.",
+    "You type like I drive.",
     "Do you think like you type?",
     "Your mind just hasn't been the same since the electro-shock, has it?",
 
