Use flock instead of lockf for visudo on hurd
Index: sudo-1.8.3p1/configure.in
===================================================================
--- sudo-1.8.3p1.orig/configure.in	2011-10-25 14:11:40.000000000 +0000
+++ sudo-1.8.3p1/configure.in	2012-01-08 04:05:23.000000000 +0000
@@ -1864,6 +1864,9 @@
 		;;
     *-gnu*)
 		OSDEFS="${OSDEFS} -D_GNU_SOURCE"
+		# lockf() isn't implemented on the Hurd -- use flock instead
+		ac_cv_func_lockf=no
+		ac_cv_func_flock=yes
 		;;
 esac
 
