Description: missing link to math lib
Author: Mathieu Malaterre <malat@debian.org>
Forwarded: http://code.google.com/p/openjpeg/source/detail?r=1628
Last-Update: 2012-04-16

Index: openjpeg/libopenjpeg/jpwl/CMakeLists.txt
===================================================================
--- openjpeg.orig/libopenjpeg/jpwl/CMakeLists.txt	2012-04-16 14:48:39.000000000 +0200
+++ openjpeg/libopenjpeg/jpwl/CMakeLists.txt	2012-04-16 14:49:01.000000000 +0200
@@ -23,6 +23,7 @@
   ENDIF(BUILD_SHARED_LIBS)
 ENDIF(WIN32)
 ADD_LIBRARY(${OPENJPEG_LIBRARY_NAME}_JPWL ${JPWL_SRCS} ${OPENJPEG_SRCS})
+target_link_libraries(${OPENJPEG_LIBRARY_NAME}_JPWL m)
 SET_TARGET_PROPERTIES(${OPENJPEG_LIBRARY_NAME}_JPWL 
   PROPERTIES ${OPENJPEG_LIBRARY_PROPERTIES})
 
