Description: missing rules for addXMLinJP2
Author: Mathieu Malaterre <malat@debian.org>
Forwarded: http://code.google.com/p/openjpeg/source/detail?r=1637
Last-Update: 2012-04-20

Index: openjpeg/applications/jpip/util/CMakeLists.txt
===================================================================
--- openjpeg.orig/applications/jpip/util/CMakeLists.txt	2012-04-20 13:13:26.905937928 +0200
+++ openjpeg/applications/jpip/util/CMakeLists.txt	2012-04-20 13:14:18.869452228 +0200
@@ -4,6 +4,14 @@
   ${FCGI_INCLUDE_DIRS}
 )
 
+# Tool to embed metadata into JP2 file
+ADD_EXECUTABLE(addXMLinJP2 addXMLinJP2.c)
+# Install exe
+INSTALL(TARGETS addXMLinJP2
+  EXPORT OpenJPEGTargets
+  DESTINATION ${OPENJPEG_INSTALL_BIN_DIR} COMPONENT Applications
+  )
+
 IF(BUILD_JPIP_SERVER)
   
   SET(OPJ_SERVER_SRCS
