Description: fix relative installation in multiarch system
Author: Mathieu Malaterre <malat@debian.org>
Forwarded: http://code.google.com/p/openjpeg/source/detail?r=1630
Last-Update: 2012-04-20

Index: openjpeg-1.5.0/CMake/OpenJPEGConfig.cmake.in
===================================================================
--- openjpeg-1.5.0.orig/CMake/OpenJPEGConfig.cmake.in	2012-04-19 12:38:27.933915198 +0200
+++ openjpeg-1.5.0/CMake/OpenJPEGConfig.cmake.in	2012-04-19 12:39:13.641453115 +0200
@@ -11,22 +11,28 @@
 SET(OPENJPEG_BUILD_VERSION "@OPENJPEG_VERSION_BUILD@")
 
 # The libraries.
-SET(OPENJPEG_LIBRARIES "@OPENJPEG_LIBRARIES@")
+SET(OPENJPEG_LIBRARIES "@OPENJPEG_LIBRARY_NAME@")
 
 # The CMake macros dir.
-SET(OPENJPEG_CMAKE_DIR "@OPENJPEG_CMAKE_DIR_CONFIG@")
+SET(OPENJPEG_CMAKE_DIR "@OPENJPEG_INSTALL_PACKAGE_DIR@")
 
 # The configuration options.
-SET(OPENJPEG_BUILD_SHARED_LIBS "@OPENJPEG_BUILD_SHARED_LIBS@")
+SET(OPENJPEG_BUILD_SHARED_LIBS "@BUILD_SHARED_LIBS@")
 
 # The "use" file.
 SET(OPENJPEG_USE_FILE "@OPENJPEG_USE_FILE_CONFIG@")
 
 get_filename_component(SELF_DIR "${CMAKE_CURRENT_LIST_FILE}" PATH)
+# The following is inspired from:
+# http://www.cmake.org/Wiki/CMake/Tutorials/Packaging#Packaging_and_Exporting
+# However the following is difficult to handle:
+# get_filename_component(myproj_INCLUDE_DIRS "${SELF_DIR}/../../include/myproj" ABSOLUTE)
+# it asssumes a non multi-arch system, where 'include' is located '../include' from lib
+# therefore we need to take into account the number of subdirs in OPENJPEG_INSTALL_LIB_DIR
 if(EXISTS ${SELF_DIR}/OpenJPEGTargets.cmake)
   # This is an install tree
   include(${SELF_DIR}/OpenJPEGTargets.cmake)
-  get_filename_component(OPENJPEG_INCLUDE_ROOT "${SELF_DIR}/../../@OPENJPEG_INSTALL_INCLUDE_DIR@" ABSOLUTE)
+  get_filename_component(OPENJPEG_INCLUDE_ROOT "${SELF_DIR}/@relative_parent@/@OPENJPEG_INSTALL_INCLUDE_DIR@" ABSOLUTE)
   set(OPENJPEG_INCLUDE_DIRS ${OPENJPEG_INCLUDE_ROOT})
 
 else(EXISTS ${SELF_DIR}/OpenJPEGTargets.cmake)
Index: openjpeg-1.5.0/CMakeLists.txt
===================================================================
--- openjpeg-1.5.0.orig/CMakeLists.txt	2012-04-19 12:39:27.165942506 +0200
+++ openjpeg-1.5.0/CMakeLists.txt	2012-04-19 12:39:41.069394025 +0200
@@ -103,6 +103,18 @@
   SET(OPENJPEG_INSTALL_LIB_DIR "lib")
 ENDIF(NOT OPENJPEG_INSTALL_LIB_DIR)
 
+# The following will compute the amount of parent dir to go
+# from include to lib. it works nicely with 
+# OPENJPEG_INSTALL_LIB_DIR=lib
+# OPENJPEG_INSTALL_LIB_DIR=lib/
+# OPENJPEG_INSTALL_LIB_DIR=/lib
+# OPENJPEG_INSTALL_LIB_DIR=lib/gnu-linux-x64
+STRING(REPLACE "/" ";" relative_to_lib ${OPENJPEG_INSTALL_LIB_DIR})
+set(relative_parent "..")
+foreach( elem ${relative_to_lib})
+  set( relative_parent "${relative_parent}/.." )
+endforeach()
+
 IF(NOT OPENJPEG_INSTALL_SHARE_DIR)
   SET(OPENJPEG_INSTALL_SHARE_DIR "share")
 ENDIF(NOT OPENJPEG_INSTALL_SHARE_DIR)
