Description: Detection of GEOM encryption devices (GELI)
Author: Robert Millan <rmh@debian.org>
Bug-Debian: http://bugs.debian.org/635393
Forwarded: no
Last-Update: 2011-07-30

Index: b/libparted/arch/freebsd.c
===================================================================
--- a/libparted/arch/freebsd.c
+++ b/libparted/arch/freebsd.c
@@ -1058,6 +1058,8 @@
 {
 	size_t listsize;
 	char *disklist, *pdisklist, *psave;
+	char buf[PATH_MAX];
+	struct stat st;
 
 	if (sysctlbyname("kern.disks", NULL, &listsize, NULL, 0) != 0) {
 		ped_exception_throw (
@@ -1091,6 +1093,10 @@
 		strncat (dev_name, token, sizeof(dev_name) - strlen(_PATH_DEV) - 1);
 		dev_name[sizeof(dev_name) - 1] = '\0';
 		_ped_device_probe (dev_name);
+
+		snprintf (buf, sizeof (buf), "%s.eli", dev_name);
+		if (stat (buf, &st) == 0)
+			_ped_device_probe (buf);
 	}
 
 	free(disklist);
