--- a/libparted/arch/freebsd.c
+++ b/libparted/arch/freebsd.c
@@ -1140,6 +1140,31 @@
 	return 1;
 }
 
+static int
+_probe_lvm_volumes ()
+{
+	DIR*	lvm_dir;
+	struct dirent*	lvm_dent;
+	char            buf[PATH_MAX];
+	struct stat     st;
+
+	lvm_dir = opendir ("/dev/linux_lvm");
+	if (!lvm_dir)
+		return 0;
+
+	while ((lvm_dent = readdir (lvm_dir))) {
+		if (strcmp (lvm_dent->d_name, ".")  == 0 || strcmp (lvm_dent->d_name, "..") == 0)
+			continue;
+		snprintf (buf, sizeof (buf), "/dev/linux_lvm/%s", lvm_dent->d_name);
+		if (stat (buf, &st) != 0)
+			continue;
+		_ped_device_probe (buf);
+	}
+	closedir (lvm_dir);
+
+	return 1;
+}
+
 static void
 freebsd_probe_all ()
 {
@@ -1148,6 +1173,8 @@
 	_probe_kern_disks ();
 
 	_probe_zfs_volumes ();
+
+	_probe_lvm_volumes ();
 }
 
 static char*
