Description: Accommodate two-component Linux version numbers like 3.0
Origin: upstream, http://git.debian.org/?p=parted/parted.git;a=commitdiff;h=bb4fbe2130cd85727bbaf6c71b2d27b6769aa4aa
Author: Jim Meyering <meyering@redhat.com>
Bug-Ubuntu: https://bugs.launchpad.net/bugs/796865
Forwarded: not-needed
Last-Update: 2011-06-15

Index: b/libparted/arch/linux.c
===================================================================
--- a/libparted/arch/linux.c
+++ b/libparted/arch/linux.c
@@ -600,7 +600,11 @@
 
         if (uname (&uts))
                 return kver = 0;
-        if (sscanf (uts.release, "%u.%u.%u", &major, &minor, &teeny) != 3)
+        if (sscanf (uts.release, "%u.%u.%u", &major, &minor, &teeny) == 3)
+                ; /* ok */
+        else if (sscanf (uts.release, "%u.%u", &major, &minor) == 2)
+                teeny = 0;
+        else
                 return kver = 0;
 
         return kver = KERNEL_VERSION (major, minor, teeny);
