Author: Daniel Baumann <daniel.baumann@progress-technologies.net>
Description: Avoid overwriting buildflags.

diff -Naurp gfxboot.orig/Makefile gfxboot/Makefile
--- gfxboot.orig/Makefile	2012-06-27 16:04:00.337448940 +0200
+++ gfxboot/Makefile	2012-06-29 18:17:12.147390138 +0200
@@ -1,5 +1,5 @@
 CC	 = gcc
-CFLAGS	 = -g -Wall -Wno-pointer-sign -O2 -fomit-frame-pointer
+#CFLAGS	 = -g -Wall -Wno-pointer-sign -O2 -fomit-frame-pointer
 
 GIT2LOG := $(shell if [ -x ./git2log ] ; then echo ./git2log --update ; else echo true ; fi)
 GITDEPS := $(shell [ -d .git ] && echo .git/HEAD .git/refs/heads .git/refs/tags)
