Author: Daniel Baumann <daniel.baumann@progress-technologies.net>
Description: Updating syslinux paths (Closes: #682974).

diff -Naurp gfxboot.orig/gfxboot gfxboot/gfxboot
--- gfxboot.orig/gfxboot	2012-06-27 16:04:00.357449136 +0200
+++ gfxboot/gfxboot	2012-07-27 17:41:56.811764900 +0200
@@ -989,9 +989,9 @@ my @vm_order = qw ( qemu64 qemu32 qemu q
 my %bl_list = (
   grub     => '/usr/sbin/grub',
   lilo     => '/sbin/lilo',
-  isolinux => '/usr/share/syslinux/isolinux.bin',
+  isolinux => '/usr/lib/syslinux/isolinux.bin',
   syslinux => '/usr/bin/syslinux',
-  pxelinux => '/usr/share/syslinux/pxelinux.0',
+  pxelinux => '/usr/lib/syslinux/pxelinux.0',
   bd       => '/usr/bin/bd',
   bochs    => '/usr/bin/bochs',
 );
@@ -1793,8 +1793,8 @@ sub prepare_isolinux
   $arch_dir = 'i386';
   $arch_dir = 'x86_64' if $opt_64 && !$opt_32;
 
-  $comboot = "$opt_syslinux/usr/share/syslinux/gfxboot.c32";
-  $comboot = "$opt_syslinux/usr/share/syslinux/gfxboot.com" unless -f $comboot;
+  $comboot = "$opt_syslinux/usr/lib/syslinux/gfxboot.c32";
+  $comboot = "$opt_syslinux/usr/lib/syslinux/gfxboot.com" unless -f $comboot;
   $comboot = 0 unless -f $comboot;
 
   my $menu = fake_menu 'install';
@@ -1864,7 +1864,7 @@ sub prepare_isolinux
     close F;
   }
 
-  system "cp $opt_syslinux/usr/share/syslinux/isolinux.bin $dst/$loader" and die "error: no isolinux\n";
+  system "cp $opt_syslinux/usr/lib/syslinux/isolinux.bin $dst/$loader" and die "error: no isolinux\n";
   system "cp $comboot $dst/$loader" if $comboot;
 
   for (@opt_test_addfiles) {
@@ -1908,8 +1908,8 @@ sub prepare_syslinux
 
   die "error: syslinux not found\n" unless -f "$opt_syslinux/$bl_list{syslinux}";
 
-  $comboot = "$opt_syslinux/usr/share/syslinux/gfxboot.c32";
-  $comboot = "$opt_syslinux/usr/share/syslinux/gfxboot.com" unless -f $comboot;
+  $comboot = "$opt_syslinux/usr/lib/syslinux/gfxboot.c32";
+  $comboot = "$opt_syslinux/usr/lib/syslinux/gfxboot.com" unless -f $comboot;
   $comboot = 0 unless -f $comboot;
 
   my $menu = fake_menu 'install';
@@ -2022,8 +2022,8 @@ sub prepare_pxelinux
   $arch_dir = 'i386';
   $arch_dir = 'x86_64' if $opt_64 && !$opt_32;
 
-  $comboot = "$opt_syslinux/usr/share/syslinux/gfxboot.c32";
-  $comboot = "$opt_syslinux/usr/share/syslinux/gfxboot.com" unless -f $comboot;
+  $comboot = "$opt_syslinux/usr/lib/syslinux/gfxboot.c32";
+  $comboot = "$opt_syslinux/usr/lib/syslinux/gfxboot.com" unless -f $comboot;
   $comboot = 0 unless -f $comboot;
 
   my $menu = fake_menu 'install';
@@ -2093,7 +2093,7 @@ sub prepare_pxelinux
     close F;
   }
 
-  system "cp $opt_syslinux/usr/share/syslinux/pxelinux.0 $dst/$loader" and die "error: no pxelinux\n";
+  system "cp $opt_syslinux/usr/lib/syslinux/pxelinux.0 $dst/$loader" and die "error: no pxelinux\n";
   system "cp $comboot $dst/$loader" if $comboot;
 
   for (@opt_test_addfiles) {
