Description: Update bug reporting address.
 * Reported by Faheem Mitha.
Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=542469
Author: Bob Proulx

--- time-1.7.orig/README
+++ time-1.7/README
@@ -8,6 +8,6 @@
 See the file INSTALL for compilation and installation instructions.
 
 Mail suggestions and bug reports for GNU time to
-bug-gnu-utils@prep.ai.mit.edu.  Please include the version of
+bug-gnu-utils@gnu.org.  Please include the version of
 `time', which you can get by running `time --version', and the
 operating system and C compiler you used.
--- time-1.7.orig/time.texi
+++ time-1.7/time.texi
@@ -137,7 +143,7 @@
 @end example
 
 Mail suggestions and bug reports for GNU @code{time} to
-@code{bug-gnu-utils@@prep.ai.mit.edu}.  Please include the version of
+@code{bug-gnu-utils@@gnu.org}.  Please include the version of
 @code{time}, which you can get by running @samp{time --version}, and the
 operating system and C compiler you used.
 
