Subject: Fix ru_maxrss reporting
Author: Richard Kettlewell
Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=649402

--- time-1.7.orig/time.c
+++ time-1.7/time.c
@@ -392,7 +398,7 @@
 		       ptok ((UL) resp->ru.ru_ixrss) / MSEC_TO_TICKS (v));
 	      break;
 	    case 'M':		/* Maximum resident set size.  */
-	      fprintf (fp, "%lu", ptok ((UL) resp->ru.ru_maxrss));
+	      fprintf (fp, "%lu", (UL) resp->ru.ru_maxrss);
 	      break;
 	    case 'O':		/* Outputs.  */
 	      fprintf (fp, "%ld", resp->ru.ru_oublock);
