 -
--- a/Makefile.PL
+++ b/Makefile.PL
@@ -28,10 +28,10 @@ WriteMakefile (
                        'URI::Escape'      => 0,
                      },
 	MAN1PODS      => { 
-                       'docs/razor-check.pod'    => '$(INST_MAN1DIR)/razor-check.1',
-                       'docs/razor-report.pod'   => '$(INST_MAN1DIR)/razor-report.1',
-                       'docs/razor-admin.pod'    => '$(INST_MAN1DIR)/razor-admin.1',
-                       'docs/razor-revoke.pod'   => '$(INST_MAN1DIR)/razor-revoke.1',
+                       'docs/razor-check.pod'    => '$(INST_MAN1DIR)/razor-check.1p',
+                       'docs/razor-report.pod'   => '$(INST_MAN1DIR)/razor-report.1p',
+                       'docs/razor-admin.pod'    => '$(INST_MAN1DIR)/razor-admin.1p',
+                       'docs/razor-revoke.pod'   => '$(INST_MAN1DIR)/razor-revoke.1p',
                      },
     dist          => {
                        'COMPRESS'   => 'bzip2',
@@ -118,7 +118,7 @@ sub MY::manifypods {
 manifypods-razor : docs/razor-agent.conf.pod \
 	docs/razor-agents.pod                \
 	docs/razor-whitelist.pod
-	$(POD2MAN) \
+	$(POD2MAN) --section=$(MAN5EXT) \
 	docs/razor-agent.conf.pod \
 	$(INST_MAN5DIR)/razor-agent.conf.$(MAN5EXT) \
 	docs/razor-agents.pod \
