Author: Daniel Baumann <daniel.baumann@progress-technologies.net>
Description: Add VERBOSE_MESSAGES to enable verbose messages :)

diff -Naurp syslinux.orig/core/tracers.inc syslinux/core/tracers.inc
--- syslinux.orig/core/tracers.inc	2011-10-12 19:51:48.224568516 +0200
+++ syslinux/core/tracers.inc	2011-10-12 21:11:48.896376484 +0200
@@ -22,6 +22,7 @@
 ; Note: The Makefile builds one version with DEBUG_MESSAGES automatically.
 ; %define DEBUG_TRACERS 1		; Uncomment to get debugging tracers
 ; %define DEBUG_MESSAGES		; Uncomment to get debugging messages
+; %define VERBOSE_MESSAGES 1		; Uncomment to get verbose messages
 
 %ifdef DEBUG_TRACERS
 
