Author: Salvatore Bonaccorso <carnil@debian.org>
Description: use Digest::SHA for utils/sha1pass (Closes: #625812).

diff -Naurp syslinux.orig/utils/sha1pass syslinux/utils/sha1pass
--- syslinux.orig/utils/sha1pass	2011-05-16 09:40:30.874806848 +0200
+++ syslinux/utils/sha1pass	2011-05-16 09:40:01.786652350 +0200
@@ -1,7 +1,7 @@
 #!/usr/bin/perl
 
 use bytes;
-use Digest::SHA1;
+use Digest::SHA;
 use MIME::Base64;
 
 sub random_bytes($) {
@@ -29,6 +29,6 @@ sub random_bytes($) {
 unless (defined($salt)) {
     $salt = MIME::Base64::encode(random_bytes(6), '');
 }
-$pass = Digest::SHA1::sha1_base64($salt, $pass);
+$pass = Digest::SHA::sha1_base64($salt, $pass);
 
 print '$4$', $salt, '$', $pass, "\$\n";
