Description: Fix typo writeable -> writable found by lintian
Origin: vendor
Forwarded: no
Author: Alessandro Ghedini <al3xbio@gmail.com>
Last-Update: 2011-08-07

--- a/xs/GdkColor.xs
+++ b/xs/GdkColor.xs
@@ -68,7 +68,7 @@
  ## deprecated
  ## gint gdk_colormap_get_system_size (void)
 
- ## gint gdk_colormap_alloc_colors (GdkColormap *colormap, GdkColor *colors, gint ncolors, gboolean writeable, gboolean best_match, gboolean *success)
+ ## gint gdk_colormap_alloc_colors (GdkColormap *colormap, GdkColor *colors, gint ncolors, gboolean writable, gboolean best_match, gboolean *success)
 ## success becomes an array of TRUE or FALSE corresponding to each input
 ## color, telling whether each one was successfully allocated.  the return
 ## value is the number that were NOT allocated.
@@ -78,9 +78,9 @@
 corresponding spot in I<...> could be allocated.
 =cut
 void
-gdk_colormap_alloc_colors (colormap, writeable, best_match, ...)
+gdk_colormap_alloc_colors (colormap, writable, best_match, ...)
 	GdkColormap *colormap
-	gboolean writeable
+	gboolean writable
 	gboolean best_match
     PREINIT:
 	gint ncolors, i;
@@ -99,7 +99,7 @@
 		argcolors[i] = SvGdkColor (ST (first + i));
 		colors[i] = *(argcolors[i]);
 	}
-	gdk_colormap_alloc_colors (colormap, colors, ncolors, writeable,
+	gdk_colormap_alloc_colors (colormap, colors, ncolors, writable,
 	                           best_match, success);
 	EXTEND (SP, ncolors);
 	for (i = 0 ; i < ncolors ; i++) {
@@ -111,12 +111,12 @@
 	g_free (success);
 #undef first
 
- ## gboolean gdk_colormap_alloc_color (GdkColormap *colormap, GdkColor *color, gboolean writeable, gboolean best_match)
+ ## gboolean gdk_colormap_alloc_color (GdkColormap *colormap, GdkColor *color, gboolean writable, gboolean best_match)
 gboolean
-gdk_colormap_alloc_color (colormap, color, writeable, best_match)
+gdk_colormap_alloc_color (colormap, color, writable, best_match)
 	GdkColormap *colormap
 	GdkColor *color
-	gboolean writeable
+	gboolean writable
 	gboolean best_match
 
  # this deallocates colors in the colormap, as allocated by alloc_color(s)
