Index: haskell-ami-0.1/Network/AMI.hs
===================================================================
--- haskell-ami-0.1.orig/Network/AMI.hs	2012-01-16 05:58:28.000000000 +0000
+++ haskell-ami-0.1/Network/AMI.hs	2012-10-20 19:43:18.499382941 +0000
@@ -32,7 +32,7 @@
 import Data.Digest.Pure.MD5
 import System.IO.Unsafe (unsafePerformIO)
 import Network
-import Network.Socket
+import Network.Socket hiding (close)
 import System.IO
 
 {- $usage
