# DP: avoid runtime library path for some extension modules.

diff -r 12a56a349af2 setup.py
--- a/setup.py	Sat Jun 30 19:28:16 2012 +0200
+++ b/setup.py	Sun Jul 01 10:25:14 2012 +0200
@@ -957,6 +957,15 @@
                 print("bsddb lib dir:", dblib_dir, " inc dir:", db_incdir)
             db_incs = [db_incdir]
             dblibs = [dblib]
+            # only add db_incdir/dblib_dir if not in the standard paths
+            # avoids a runtime library path for a system library dir
+            if db_incdir in inc_dirs:
+                db_incs = []
+            else:
+                db_incs = [db_incdir]
+            if dblib_dir[0] in lib_dirs:
+                dblib_dir = []
+            dblibs = [dblib]
         else:
             if db_setup_debug: print("db: no appropriate library found")
             db_incs = None
@@ -1065,6 +1074,9 @@
             # can end up with a bad search path order.
             if sqlite_incdir not in self.compiler.include_dirs:
                 include_dirs.append(sqlite_incdir)
+            # avoid a runtime library path for a system library dir
+            if sqlite_libdir and sqlite_libdir[0] in lib_dirs:
+                sqlite_libdir = None
             exts.append(Extension('_sqlite3', sqlite_srcs,
                                   define_macros=sqlite_defines,
                                   include_dirs=include_dirs,
