diff --git a/configure.ac b/configure.ac
index b00f31f..ae53499 100644
--- a/configure.ac
+++ b/configure.ac
@@ -21,9 +21,6 @@ AC_DEFINE_UNQUOTED(GETTEXT_PACKAGE,"$GETTEXT_PACKAGE",[The name of the gettext d
 IT_PROG_INTLTOOL(0.40.0)
 PKG_PROG_PKG_CONFIG([0.22])
 
-AM_GNU_GETTEXT([external])
-AM_GNU_GETTEXT_VERSION([0.17])
-
 GOBJECT_INTROSPECTION_REQUIRE([0.9.6])
 
 ## don't rerun to this point if we abort
