Author: Joshua Kwan <joshk@triplehelix.org>
Description: Exit with error if invalid option is passed. #329979
--- flac.orig/src/metaflac/main.c
+++ flac/src/metaflac/main.c
@@ -37,7 +37,7 @@ int main(int argc, char *argv[])
 	setlocale(LC_ALL, "");
 	init_options(&options);
 
-	if(parse_options(argc, argv, &options))
+	if ((ret = parse_options(argc, argv, &options)) == 0)
 		ret = !do_operations(&options);
 	else
 		ret = 1;
--- flac.orig/src/metaflac/options.c
+++ flac/src/metaflac/options.c
@@ -249,7 +249,7 @@ FLAC__bool parse_options(int argc, char
 		}
 	}
 
-	return !had_error;
+	return had_error;
 }
 
 void free_options(CommandLineOptions *options)
