Author: Cyril Brulebois <cyril.brulebois@enst-bretagne.fr>
Description: Fix gcc-4.3 FTBFS, patch by KiBi (Closes: #455304).
--- flac.orig/examples/cpp/encode/file/main.cpp
+++ flac/examples/cpp/encode/file/main.cpp
@@ -33,6 +33,8 @@
 #include "FLAC++/metadata.h"
 #include "FLAC++/encoder.h"
 
+#include <cstring>
+
 class OurEncoder: public FLAC::Encoder::File {
 public:
 	OurEncoder(): FLAC::Encoder::File() { }
