Author: Fabian Greffrath <fabian+debian@greffrath.com>
Subject: Fixes "undefined macro: AM_PATH_XMMS" and similar errors which cause
 autoreconf to fail.
Forwarded: https://sourceforge.net/tracker/?func=detail&aid=3027361&group_id=13478&atid=113478

--- flac.orig/Makefile.am
+++ flac/Makefile.am
@@ -30,6 +30,8 @@
 
 AUTOMAKE_OPTIONS = foreign 1.7
 
+ACLOCAL_AMFLAGS = -I m4
+
 SUBDIRS = doc include m4 man src examples test build obj
 
 DISTCLEANFILES = libtool-disable-static
--- flac.orig/configure.in
+++ flac/configure.in
@@ -33,6 +33,7 @@ sed -e 's/^build_old_libs=yes/build_old_
 chmod +x libtool-disable-static
 
 AC_SUBST(ACLOCAL_AMFLAGS, "-I m4")
+AC_CONFIG_MACRO_DIR([m4])
 
 AM_PROG_AS
 AC_PROG_CXX
