Index: libgnome-2.32.1/libgnome/gnome-program.c
===================================================================
--- libgnome-2.32.1.orig/libgnome/gnome-program.c	2010-02-07 19:04:26.000000000 +0100
+++ libgnome-2.32.1/libgnome/gnome-program.c	2011-05-29 09:48:34.451179579 +0200
@@ -64,6 +64,10 @@
 
 #define GNOME_PROGRAM_GET_PRIVATE(object)(G_TYPE_INSTANCE_GET_PRIVATE ((object), GNOME_TYPE_PROGRAM, GnomeProgramPrivate))
 
+#ifndef PATH_MAX
+#define PATH_MAX 4096
+#endif
+
 struct _GnomeProgramPrivate {
     enum {
 	APP_UNINIT=0,
