diff --git a/libgnome/gnome-program.c b/libgnome/gnome-program.c
index fa8e62d..3f32d39 100644
--- a/libgnome/gnome-program.c
+++ b/libgnome/gnome-program.c
@@ -1669,27 +1669,38 @@ gnome_program_parse_args (GnomeProgram *program)
 static char *
 find_accessibility_module (GnomeProgram *program, const char *libname)
 {
-	char *sub;
+	char *sub0, *sub1;
 	char *path;
 	char *fname;
 	char *retval;
 
 	fname = g_strconcat (libname, "." G_MODULE_SUFFIX, NULL);
-	sub = g_strconcat ("gtk-2.0/modules", G_DIR_SEPARATOR_S, fname, NULL);
+	sub0 = g_strconcat (MULTIARCH, G_DIR_SEPARATOR_S, "gtk-2.0/modules", G_DIR_SEPARATOR_S, fname, NULL);
+	sub1 = g_strconcat ("gtk-2.0/modules", G_DIR_SEPARATOR_S, fname, NULL);
 
 	path = gnome_program_locate_file (
-		program, GNOME_FILE_DOMAIN_LIBDIR, sub, TRUE, NULL);
+		program, GNOME_FILE_DOMAIN_LIBDIR, sub0, TRUE, NULL);
 
-	g_free (sub);
+	if (path) {
+		retval = path;
+		goto out;
+	}
+
+	path = gnome_program_locate_file (
+		program, GNOME_FILE_DOMAIN_LIBDIR, sub1, TRUE, NULL);
 
-	if (path)
+	if (path) {
 		retval = path;
-	else
-		retval = gnome_program_locate_file (
-			program, GNOME_FILE_DOMAIN_LIBDIR,
-			fname, TRUE, NULL);
+		goto out;
+	}
+
+	retval = gnome_program_locate_file (
+		program, GNOME_FILE_DOMAIN_LIBDIR, fname, TRUE, NULL);
 
+out:
 	g_free (fname);
+	g_free (sub0);
+	g_free (sub1);
 
 	return retval;
 }
