Description: Use system zlib
Author: Jérémy Lal <kapouer@melix.org>
Forwarded: not-needed
Last-Update: 2012-01-23

--- a/wscript
+++ b/wscript
@@ -296,9 +296,6 @@
   conf.env["USE_GDBJIT"] = o.use_gdbjit
   conf.env['USE_NPM'] = not o.without_npm
 
-  if not conf.env["USE_SHARED_ZLIB"] and not sys.platform.startswith("win32"):
-    conf.env.append_value("LINKFLAGS", "-lz")
-
   conf.check(lib='dl', uselib_store='DL')
   if not sys.platform.startswith("sunos") and not sys.platform.startswith("win32"):
     conf.env.append_value("CCFLAGS", "-rdynamic")
@@ -441,6 +438,9 @@
                             libpath=v8_libpath):
         conf.fatal("Cannot find v8_g")
 
+  if conf.env["USE_SHARED_ZLIB"] and not sys.platform.startswith("win32"):
+    conf.env.append_value("LINKFLAGS", "-lz")
+
   conf.define("HAVE_CONFIG_H", 1)
 
   if sys.platform.startswith("sunos"):
