Description: Fixes a typo.
Author: Niels Thykier <niels@thykier.net>

--- a/src/bsh/BshClassManager.java	2010-11-21 19:59:08.597086821 +0100
+++ b/src/bsh/BshClassManager.java	2010-11-21 20:01:06.128078201 +0100
@@ -493,7 +493,7 @@
 		if ( cur != null )
 			throw new InterpreterError("Defining class problem: "+className 
 				+": BeanShell cannot yet simultaneously define two or more "
-				+"dependant classes of the same name.  Attempt to define: "
+				+"dependent classes of the same name.  Attempt to define: "
 				+ className +" while defining: "+cur 
 			);
 		definingClasses.put( className, NOVALUE );
