Author: Damyan Ivanov
Description: avoid linkage to all libs perl is linked

--- a/Makefile.PL
+++ b/Makefile.PL
@@ -103,31 +103,6 @@
   }
 }
 
-if ($config{LIBS} !~ /\-lz(lib)?\b/ and !($is_win32 && $config{LIBS} !~ /\-lzlib\b/)) {
-    # note if libxml2 has not -lz within its cflags, we should not use
-    # it! We should trust libxml2 and assume libz is not available on the
-    # current system (this is ofcourse not true with win32 systems.
-    # $config{LIBS} .= $is_Win32 ? ' -lzlib' :' -lz';
-    if ( $config{DEBUG} ) {
-        warn "zlib was not configured\n";
-        warn "set zlib\n" if $is_Win32;
-    }
-  if ($is_Win32) {
-    if( $ENV{ACTIVEPERL_MINGW} ) {
-      $config{LIBS} .= '';
-    } else {
-      $config{LIBS} .= ' -lzlib';
-    }
-  } else {
-    $config{LIBS} .= ' -lz';
-  }
-}
-
-if ($config{LIBS} !~ /\-lm\b/) {
-    # math support is important, but is not available separately in W32
-    $config{LIBS} .= $is_Win32 ? '' :' -lm';
-}
-
 if (!have_library($is_Win32 ? "libxslt" : "xslt")) {
     print STDERR <<DEATH;
 libxslt not found
@@ -167,8 +142,6 @@
   $config{DEFINE} .= " -DHAVE_EXSLT"
 }
 
-$config{LIBS}.=' '.$Config{libs};
-
 if ($DEBUG) {
   print "LIBS: $config{LIBS}\n";
   print "INC: $config{INC}\n";
