---
 usb.ids |  772 +++++++++++++++++++++++++++++++++++++++++++++++++++++++---------
 1 file changed, 669 insertions(+), 103 deletions(-)

--- a/usb.ids
+++ b/usb.ids
@@ -9,8 +9,8 @@
 #	The latest version can be obtained from
 #		http://www.linux-usb.org/usb.ids
 #
-# Version: 2011.09.28
-# Date:    2011-09-28 20:34:04
+# Version: 2012.04.11
+# Date:    2012-04-11 20:34:03
 #
 
 # Vendors, devices and interfaces. Please keep sorted.
@@ -98,6 +98,7 @@
 	2105  AVRONE!
 	2106  STK600 development board
 	2107  AVR Dragon
+	2109  STK541 ZigBee Development Board
 	210d  XPLAIN evaluation kit (CDC ACM)
 	2122  XMEGA-A1 Explained evaluation kit
 	2310  EVK11xx evaluation board
@@ -109,7 +110,11 @@
 	4102  AirVast W-Buddie WN210
 	5601  at76c510 Prism-II 802.11b Access Point
 	5603  Cisco 7920 WiFi IP Phone
+	6119  AT91SAM CDC Demo Application
 	6124  at91sam SAMBA bootloader
+	6127  AT91SAM HID Keyboard Demo Application
+	6129  AT91SAM Mass Storage Demo Application
+	6200  AT91SAM HID Mouse Demo Application
 	7603  D-Link DWL-120 802.11b Wireless Adapter [Atmel at76c503a]
 	7604  at76c503a 802.11b Adapter
 	7605  at76c503a 802.11b Adapter
@@ -150,7 +155,7 @@
 	0111  G55xi Printer/Scanner/Copier
 	0117  LaserJet 3200
 	011c  hn210w 802.11b Adapter
-	011d  Integrated Bluetooth Module
+	011d  Bluetooth 1.2 Interface [Broadcom BCM2035]
 	0121  HP49g+ Calculator
 	0122  HID Internet Keyboard
 	0201  ScanJet 6200c
@@ -177,7 +182,7 @@
 	0411  OfficeJet G95
 	0412  Printing Support
 	0417  LaserJet 1200 series
-	0423  HS-COMBO
+	0423  HS-COMBO Cardreader
 	0504  DeskJet 885c
 	0505  ScanJet 2100c
 	0507  DVD+RW
@@ -185,7 +190,7 @@
 	0511  OfficeJet K60
 	0512  DeckJet 450
 	0517  LaserJet 1000
-	051d  Wireless (Bluetooth + WLAN) Interface [Integrated Module]
+	051d  Bluetooth Interface
 	0601  ScanJet 6300c
 	0604  DeskJet 840c
 	0605  ScanJet 2200c
@@ -271,7 +276,7 @@
 	1711  psc 780xi
 	1712  Printing Support
 	1717  LaserJet 3020
-	171d  Wireless (Bluetooth + WLAN) Interface [Integrated Module]
+	171d  Bluetooth 2.0 Interface [Broadcom BCM2045]
 	1801  Inkjet P-2000U
 	1802  PhotoSmart 470 series
 	1804  DeskJet 916C
@@ -279,7 +284,7 @@
 	1811  PSC 720
 	1812  OfficeJet Pro K550
 	1817  LaserJet 3030
-	181d  integrated module with Bluetooth 2.0 wireless technology.
+	181d  Bluetooth 2.0 Interface
 	1902  PhotoSmart A430 series
 	1904  DeskJet 3820
 	1911  OfficeJet V45
@@ -356,6 +361,7 @@
 	2c12  Officejet J4680
 	2c17  LaserJet 1022
 	2c24  Logitech M-UAL-96 Mouse
+	2d05  Scanjet 7000
 	2d11  OfficeJet 6110
 	2d17  Printing Support
 	2e11  PSC 1000
@@ -447,6 +453,7 @@
 	4d17  EWS UPD
 	4e11  PhotoSmart 2570 series
 	4f11  OfficeJet 5600 (USBHUB)
+	4f17  Color LaserJet CM1312 MFP
 	5004  DeskJet 995c
 	5011  PhotoSmart 3100 series
 	5017  EWS UPD
@@ -466,6 +473,7 @@
 	5a11  PhotoSmart C7100 series
 	5b11  OfficeJet J2100 series
 	5c11  PhotoSmart C4200 Printer series
+	5c17  LaserJet P2055 series
 	5d11  PhotoSmart C5200 series
 	5e11  PhotoSmart D7400 series
 	6004  DeskJet 5550
@@ -545,6 +553,7 @@
 	8504  DeskJet 6600 series
 	8604  DeskJet 5440
 	8704  DeskJet 5940
+	8711  Deskjet 2050 J510
 	8804  DeskJet 6980 series
 	8904  DeskJet 6940 series
 	8c07  Digital Stereo Headset
@@ -562,6 +571,7 @@
 	9b02  PhotoSmart M540 series
 	9c02  PhotoSmart M440 series
 	a004  DeskJet 5850c
+	a011  Deskjet 3050A
 	b002  PhotoSmart 7200 series
 	b102  PhotoSmart 7200 series
 	b116  Webcam
@@ -594,7 +604,7 @@
 03f1  Genoa Technology
 03f2  Oak Technology, Inc.
 03f3  Adaptec, Inc.
-	0020  AWN-8020 WLAN
+	0020  AWN-8020 WLAN [Intersil PRISM 2.5]
 	0080  AVC-1100 Audio Capture
 	0083  AVC-2200 Device
 	0087  AVC-2210 Loader
@@ -607,12 +617,13 @@
 	2001  USBXchange Adapter
 	2002  USB2-Xchange
 	2003  USB2-Xchange Adapter
+	4000  4-port hub
 	adcc  Composite Device Support
 03f4  Diebold, Inc.
 03f5  Siemens Electromechanical
 03f8  Epson Imaging Technology Center
 03f9  KeyTronic Corp.
-	0100  Keyboard
+	0100  KT-2001 Keyboard
 	0101  Keyboard
 	0102  Keyboard Mouse
 03fb  OPTi, Inc.
@@ -652,19 +663,38 @@
 	0232  Serial Converter
 	1060  JTAG adapter
 	6001  FT232 USB-Serial (UART) IC
+	6002  Lumel PD12
 	6007  Serial Converter
 	6008  Serial Converter
 	6009  Serial Converter
 	6010  FT2232C Dual USB-UART/FIFO IC
 	6011  FT4232H Quad HS USB-UART/FIFO IC
 	6014  FT232H Single HS USB-UART/FIFO IC
+	6015  Bridge(I2C/SPI/UART/FIFO)
 	8040  4 Port Hub
 	8070  7 Port Hub
 	8370  7 Port Hub
 	8371  PS/2 Keyboard And Mouse
 	8372  FT8U100AX Serial Port
+	8a28  Rainforest Automation ZigBee Controller
+	9e90  Marvell OpenRD Base/Client
+	9f80  Ewert Energy Systems CANdapter
+	a6d0  Texas Instruments XDS100v2 JTAG / BeagleBone A3
+	a951  HCP HIT GSM/GPRS modem [Cinterion MC55i]
+	abb8  Lego Mindstorms NXTCam
+	b810  US Interface Navigator (CAT and 2nd PTT lines)
+	b811  US Interface Navigator (WKEY and FSK lines)
+	b812  US Interface Navigator (RS232 and CONFIG lines)
+	baf8  Amontec JTAGkey - Open On-Chip Debugger
 	bcd8  Stellaris Development Board
 	bcd9  Stellaris Evaluation Board
+	bcda  Stellaris ICDI Board
+	bdc8  Egnite GmbH - JTAG/RS-232 adapter
+	bfd8  OpenDCC
+	bfd9  OpenDCC (Sniffer)
+	bfda  OpenDCC (Throttle)
+	bfdb  OpenDCC (Gateway)
+	bfdc  OpenDCC (GBM)
 	c630  lcd2usb interface
 	c631  i2c-tiny-usb interface
 	c632  xu1541 c64 floppy drive interface
@@ -678,20 +708,25 @@
 	c8bc  Alpermann+Velte Rubidium S1
 	c8bd  Alpermann+Velte Rubidium T1
 	c8be  Alpermann+Velte Rubidium D1
-	cc48  product FTDI TACTRIX_OPENPORT_13M 0xcc48 OpenPort 1.3 Mitsubishi
-	cc49  product FTDI TACTRIX_OPENPORT_13S 0xcc49 OpenPort 1.3 Subaru
-	cc4a  product FTDI TACTRIX_OPENPORT_13U 0xcc4a OpenPort 1.3 Universal
+	cc48  Tactrix OpenPort 1.3 Mitsubishi
+	cc49  Tactrix OpenPort 1.3 Subaru
+	cc4a  Tactrix OpenPort 1.3 Universal
+	cff8  Distortec JTAG-lock-pick
 	d010  SCS PTC-IIusb
 	d011  SCS Position-Tracker/TNC
 	d012  SCS DRAGON 1
 	d013  SCS DRAGON 1
+	d578  Accesio USB-COM-4SM
 	d6f8  UNI Black BOX
+	d738  Propox JTAGcable II
+	d739  Propox ISPcable III
 	d9a9  Actisense USG-1 NMEA Serial Gateway
 	d9aa  Actisense NGT-1 NMEA2000 PC Interface
 	e0d0  Total Phase Aardvark I2C/SPI Host Adapter
 	e521  EVER Sinline XL Series UPS
 	e6c8  PYRAMID Computer GmbH LCD
 	e700  Elster Unicom III Optical Probe
+	e729  Segway Robotic Mobility Platforms 200
 	e888  Expert ISDN Control USB
 	e889  USB-RS232 OptoBridge
 	e88a  Expert mouseCLOCK USB II
@@ -707,8 +742,11 @@
 	f1a0  Asix PRESTO Programmer
 	f208  Papenmeier Braille-Display
 	f3c0  4N-GALAXY Serial Converter
+	f608  CTI USB-485-Mini
+	f60b  CTI USB-Nano-485
 	f680  Suunto Sports Instrument
 	f758  GW Instek GDS-8x0 Oscilloscope
+	f7c0  ZeitControl Cardsystems TagTracer MIFARE
 	f850  USB-UIRT (Universal Infrared Receiver+Transmitter)
 	f918  Ant8 Logic Probe
 	fa00  Matrix Orbital USB Serial
@@ -725,12 +763,12 @@
 	fd48  ShipModul MiniPlex-4xUSB NMEA Multiplexer
 	fd49  ShipModul MiniPlex-4xUSB-AIS NMEA Multiplexer
 	ff08  ToolHouse LoopBack Adapter
-	ff18  Logbook Bus
+	ff18  ScienceScope Logbook ML
 	ff19  Logbook Bus
 	ff1a  Logbook Bus
 	ff1b  Logbook Bus
-	ff1c  Logbook Bus
-	ff1d  Logbook Bus
+	ff1c  ScienceScope Logbook LS
+	ff1d  ScienceScope Logbook HS
 	ff1e  Logbook Bus
 	ff1f  Logbook Bus
 0404  NCR Corp.
@@ -773,10 +811,13 @@
 	0040  Floppy
 	004e  SuperScript 1400 Series
 	004f  Wireless Keyboard with One-touch start buttons
+	0050  7-port hub
 	0058  HighSpeed Hub
 	0059  HighSpeed Hub
 	005a  HighSpeed Hub
 	006a  Conceptronic USB Harddisk Box
+	007d  MINICUBE2
+	007e  PG-FP5 Flash Memory Programmer
 	0081  SuperScript 1400 Series
 	0082  SuperScript 1400 Series
 	0094  Japanese Keyboard with One-touch start buttons
@@ -878,7 +919,7 @@
 	0595  Digital Camera
 	0596  Digital Camera
 	0597  Digital Camera
-	0598  Digital Camera
+	0598  EASYSHARE M1033 digital camera
 	0599  Digital Camera
 	059a  Digital Camera
 	059b  Digital Camera
@@ -929,6 +970,7 @@
 	05d4  EasyShare C180 Digital Camera
 	1001  EasyShare SV811 Digital Picture Frame
 	4000  InkJet Color Printer
+	4056  ESP 7200 Series AiO
 	4109  EasyShare Printer Dock Series 3
 	410d  EasyShare G600 Printer Dock
 	5010  Wireless Adapter
@@ -961,6 +1003,7 @@
 	0018  USB2-IDE Adapter
 	001c  USB-IDE Bridge: DUB-PxxG
 	0027  WLI-USB-KS11G 802.11b Adapter
+	002a  SMSC USB97C202 "HD-HB300V2-EU"
 	003d  LUA-U2-KTX Ethernet
 	0044  WLI-USB-KB11 Wireless LAN Adapter
 	004b  WLI-USB-G54 802.11g Adapter [Broadcom 4320 USB]
@@ -970,7 +1013,8 @@
 	0065  Python2 WDM Encoder
 	0066  WLI-U2-KG54 WLAN
 	0067  WLI-U2-KG54-AI WLAN
-	0089  Buffalo RUF-C/U2 Flash Drive
+	006e  LUA-U2-GT 10/100/1000 Ethernet Adapter
+	0089  RUF-C/U2 Flash Drive
 	008b  Nintendo Wi-Fi
 	0091  WLI-U2-KAMG54 Wireless LAN Adapter
 	0092  WLI-U2-KAMG54 Bootloader
@@ -984,6 +1028,7 @@
 	00d8  WLI-U2-SG54HP
 	00d9  WLI-U2-G54HP
 	00da  WLI-U2-KG54L 802.11bg [ZyDAS ZD1211B]
+	00db  External Hard Drive HD-PF32OU2 [Buffalo Ministation]
 	00e8  WLI-UC-G300N Wireless LAN Adapter [Ralink RT2870]
 	012e  WLI-UC-AG300N Wireless LAN Adapter
 	0148  WLI-UC-G300HP Wireless LAN Adapter
@@ -991,11 +1036,13 @@
 	0157  External Hard Drive HD-PEU2
 	0158  WLI-UC-GNHP Wireless LAN Adapter
 	015d  WLI-UC-GN Wireless LAN Adapter [Ralink RT3070]
-	016f  WLI-UC-G301N Wireless LAN Adapter
+	016f  WLI-UC-G301N Wireless LAN Adapter [Ralink RT3072]
 	017f  Sony UWA-BR100 802.11abgn Wireless Adapter [Atheros AR7010+AR9280]
 	019e  WLI-UC-GNP Wireless LAN Adapter
 	01a1  MiniStation Metro
-	01a2  WLI-UC-GNM Wireless LAN Adapter
+	01a2  WLI-UC-GNM Wireless LAN Adapter [Ralink RT8070]
+	01ee  WLI-UC-GNM2 Wireless LAN Adapter [Ralink RT3070]
+	01fd  WLI-UC-G450 Wireless LAN Adapter
 0412  Award Software International
 0413  Leadtek Research, Inc.
 	1310  WinFast TV - NTSC + FM
@@ -1035,6 +1082,7 @@
 	6029  WinFast DTV Dongle Gold
 	6125  WinFast DTV Dongle
 	6126  WinFast DTV Dongle BDA Driver
+	6a03  RTL2832 [WinFast DTV Dongle Mini]
 	6f00  WinFast DTV Dongle (STK7700P based)
 0414  Giga-Byte Technology Co., Ltd
 0416  Winbond Electronics Corp.
@@ -1098,6 +1146,7 @@
 	3061  SoundBlaster Audigy 2 ZS Video Editor
 	3090  Sound Blaster Digital Music SX
 	30d3  Sound Blaster Play!
+	3121  WoW tap chat
 	3f00  E-Mu Xboard 25 MIDI Controller
 	3f02  E-Mu 0202
 	3f04  E-Mu 0404
@@ -1158,6 +1207,7 @@
 	4068  Live! Cam Notebook [VF0470]
 	406c  Live! Cam Sync [VF0520]
 	4083  Live! Cam Socialize [VF0640]
+	4087  Live! Cam Socialize HD 1080 [VF0680]
 	4100  Nomad Jukebox 2
 	4101  Nomad Jukebox 3
 	4102  NOMAD MuVo^2
@@ -1196,6 +1246,7 @@
 	4151  Zen Vision:M (mtp)
 	4152  Zen V Plus
 	4153  Zen Vision W
+	4154  Zen Stone
 	4155  Zen Stone plus
 	4157  Zen (MTP)
 	500f  Broadband Blaster 8012U-V
@@ -1662,7 +1713,7 @@
 	0001  Mouse
 	0002  Genius NetMouse Pro
 	0003  Genius NetScroll+
-	0006  Easy Mouse+ USB(USB\Vid_0458&Pid;_0006) Mouse
+	0006  Easy Mouse+
 	000b  NetMouse Wheel(P+U)
 	000c  TACOMA Fingerprint V1.06.01
 	000e  VideoCAM Web
@@ -1680,6 +1731,7 @@
 	005c  Enhanced Laser Gaming Device
 	005d  Enhanced Device
 	0061  Bluetooth Dongle
+	0066  Genius Traveler 1000 Wireless Mouse
 	0072  Navigator 335
 	0083  Bluetooth Dongle
 	0087  Ergo 525V Laser Mouse
@@ -1720,7 +1772,7 @@
 	301a  MaxFire G-12U Vibration
 	301d  Genius MaxFire MiniPad
 	400f  Genius TVGo DVB-T02Q MCE
-	4012  TVGo DVB-T03
+	4012  TVGo DVB-T03 [AF9015]
 	5003  G-pen 560 Tablet
 	5004  G-pen Tablet
 	6001  GF3000F Ethernet Adapter
@@ -1746,6 +1798,7 @@
 	705a  Asus USB2.0 Webcam
 	705c  Genius iSlim 1300AF
 	7079  FaceCam 2025R
+	707f  TVGo DVB-T03 [RTL2832]
 0459  Adobe Systems, Inc.
 045a  SONICblue, Inc.
 	07da  Supra Express 56K modem
@@ -1773,6 +1826,7 @@
 	5220  Rio Nitrus MP3 Player
 	5221  Rio Eigen
 045b  Hitachi, Ltd
+	0053  RX610 RX-Stick
 045d  Nortel Networks, Ltd
 045e  Microsoft Corp.
 	0007  SideWinder Game Pad
@@ -1835,6 +1889,7 @@
 	009d  Wireless Optical Desktop 3.0
 	00a0  eHome Infrared Receiver
 	00b0  Digital Media Pro Keyboard
+	00b4  Digital Media Keyboard 1.0A
 	00b9  Wireless Optical Mouse 3.0
 	00bb  Fingerprint Reader
 	00bc  Fingerprint Reader
@@ -2016,13 +2071,17 @@
 	071f  Mouse/Keyboard 2.4GHz Transceiver V2.0
 	0721  LifeCam NX-3000 (UVC-compliant)
 	0723  LifeCam VX-7000 (UVC-compliant)
+	0724  SideWinder Mouse
+	0730  Digital Media Keyboard 3000
 	0734  Wireless Optical Desktop 700
 	0737  Compact Optical Mouse 500
 	0745  Nano Transceiver v1.0 for Bluetooth
 	0750  Wired Keyboard 600
+	0752  Wired Keyboard 400
 	075d  LifeCam Cinema
 	0768  Sidewinder X4
 	076d  LifeCam HD-5000
+	0772  LifeCam Studio
 	930a  ISOUSB.SYS Intel 82930 Isochronous IO Test Board
 	fff8  Keyboard
 	ffff  Windows CE Mass Storage
@@ -2031,6 +2090,7 @@
 	0006  LCD Tablet (12x9)
 	0008  Tablet (3x2.25)
 0461  Primax Electronics, Ltd
+	0010  HP Keyboard
 	0300  G2-300 Scanner
 	0301  G2E-300 Scanner
 	0302  G2-300 #2 Scanner
@@ -2075,6 +2135,7 @@
 	4d03  Kensington Mouse-in-a-box
 	4d04  Mouse
 	4d06  Balless Mouse (HID)
+	4d0f  HP Optical Mouse
 	4d15  Dell Optical Mouse
 	4d17  Optical Mouse
 	4d20  HP Optical Mouse
@@ -2113,6 +2174,7 @@
 	0101  PS/2 Keyboard, Mouse & Joystick Ports
 	0301  USB 1.0 Hub
 	0500  Serial & Parallel Ports
+	ff10  Virtual Keyboard and Mouse
 046c  Toshiba Corp., Digital Media Equipment
 046d  Logitech, Inc.
 	0082  Acer Aspire 5672 Webcam
@@ -2135,6 +2197,7 @@
 	0810  QuickCam Pro
 	0819  Webcam C210
 	081b  Webcam C310
+	081d  HD Webcam C510
 	0820  QuickCam VC
 	0821  HD Webcam C910
 	0825  Webcam C270
@@ -2294,14 +2357,18 @@
 	c050  RX 250 Optical Mouse
 	c051  G3 (MX518) Optical Mouse
 	c053  Laser Mouse
+	c054  Bluetooth mini-receiver
 	c058  M115 Mouse
 	c05a  Optical Mouse M90
+	c05b  M-U0004 810-001317 [B110 Optical USB Mouse]
 	c05d  Optical Mouse
 	c061  RX1500 Laser Mouse
 	c062  LS1 Laser Mouse, corded
 	c063  DELL Laser Mouse
 	c068  G500 Laser Mouse
+	c069  M500 Laser Mouse
 	c06b  G700 Wireless Gaming Mouse
+	c06c  Optical Mouse
 	c101  UltraX Media Remote
 	c110  Harmony 885 Remote
 	c111  Harmony 525 Remote
@@ -2328,6 +2395,7 @@
 	c21a  Precision Gamepad
 	c21c  G13 Advanced Gameboard
 	c21d  F310 Gamepad [XInput Mode]
+	c21e  F510 Gamepad [XInput Mode]
 	c21f  F710 Wireless Gamepad [XInput Mode]
 	c221  G11/G15 Keyboard / Keyboard
 	c222  G15 Keyboard / LCD
@@ -2335,8 +2403,11 @@
 	c225  G11/G15 Keyboard / G keys
 	c226  G15 Refresh Keyboard
 	c227  G15 Refresh Keyboard
+	c22a  Gaming Keyboard G110
+	c22b  Gaming Keyboard G110 G-keys
 	c22d  G510 Gaming Keyboard
 	c22e  G510 Gaming Keyboard onboard audio
+	c246  Gaming Mouse G300
 	c281  WingMan Force
 	c283  WingMan Force 3D
 	c285  WingMan Strike Force 3D
@@ -2546,6 +2617,7 @@
 	2076  GoGear Aria
 	2079  GoGear Opus
 	2088  MCE IR Receiver with ALS- Spinel plus for ASUS
+	209e  PTA01 Wireless Adapter
 	20b6  GoGear Vibe
 	20e3  GoGear Raga
 	262c  SPC230NC Webcam
@@ -2623,6 +2695,7 @@
 	102d  Pilot Optical
 	102e  Pilot Optical Pro
 	102f  Pilot Optical Pro Wireless
+	1042  Ci25m Notebook Optical Mouse [Diamond Eye Precision]
 	1043  Ci65m Wireless Notebook Optical Mouse
 	104a  PilotMouse Mini Retractable
 	105d  PocketMouse Bluetooth
@@ -2674,6 +2747,7 @@
 047f  Plantronics, Inc.
 	0101  Bulk Driver
 	0301  Bulk Driver
+	0411  Savi Office Base Station
 	0ca1  USB DSP v4 Audio Interface
 	4254  BUA-100 Bluetooth Adapter
 	ad01  GameCom 777 5.1 Headset
@@ -2682,6 +2756,7 @@
 	0004  InTouch Module
 	0011  InTouch Module
 	0014  InTouch Module
+	a007  External Disk USB 3.0
 0481  Zenith Data Systems
 0482  Kyocera Corp.
 	000e  FS-1020D Printer
@@ -2705,11 +2780,13 @@
 	2018  BioSimKey
 	2302  Portable Flash Device (PFD)
 	3744  STLINK Pseudo disk
+	3747  ST Micro Connect Lite
 	3748  ST-LINK/V2
 	4810  ISDN adapter
 	481d  BT Digital Access adapter
 	5000  ST Micro/Ergenic ERG BT-002 Bluetooth Adapter
 	5001  ST Micro Bluetooth Device
+	5710  Joystick in FS Mode
 	5721  Hantek DDS-3X25 Arbitrary Waveform Generator
 	7270  ST Micro Serial Bridge
 	7554  56k SoftModem
@@ -2728,12 +2805,15 @@
 	d00e  Rollei Compactline (Video Mode)
 	e000  T-Com TC 300
 	e003  Pirelli DP-L10
+	e00d  Broadcom Bluetooth 2.1 Device
 	e00f  Foxconn T77H114 BCM2070 [Single-Chip Bluetooth 2.1 + EDR Adapter]
 	e016  Ubee PXU1900 WiMAX Adapter [Beceem BCSM250]
 	e02c  Atheros AR5BBU12 Bluetooth Device
 048a  S-MOS Systems, Inc.
 048c  Alps Electric Ireland, Ltd
 048d  Integrated Technology Express, Inc.
+	1336  SD/MMC Cardreader
+	1345  Multi Cardreader
 	9009  Zolid HD DVD Maker
 	9135  Zolid Mini DVB-T Stick
 048f  Eicon Tech.
@@ -3103,6 +3183,7 @@
 	173e  MP560
 	173f  Pixma MP640 Multifunction device
 	1748  Pixma MG5150
+	174d  MX360 ser
 	1900  CanoScan LiDE 90
 	1901  CanoScan 8800F
 	1904  CanoScan LiDE 100
@@ -3223,6 +3304,7 @@
 	26b0  MF4600 series
 	26b4  MF4010 series
 	26b5  MF4200 series
+	2737  MF4410
 	3041  PowerShot S10
 	3042  CanoScan FS4000US Film Scanner
 	3043  PowerShot S20
@@ -3339,6 +3421,7 @@
 	312d  Elura 100
 	3138  PowerShot A710 IS
 	3147  EOS 1Ds Mark III
+	314f  Powershot SD1000
 	3155  PowerShot A450
 	315a  PowerShot G9
 	315d  PowerShot A720
@@ -3347,6 +3430,7 @@
 	3176  PowerShot A590
 	317a  Powershot A470
 	3184  Digital IXUS 80 IS (PTP mode)
+	3192  PowerShot SX110 IS
 	319a  EOS 7D
 	31bc  PowerShot D10
 	31bf  PowerShot A480
@@ -3356,6 +3440,9 @@
 	31ef  PowerShot A495
 	31f3  PowerShot Digital ELPH SD1400 IS
 	31ff  Digital IXUS 55
+	3210  Powershot SX30 IS
+	3214  SELPHY CP800
+	3218  EOS 600D / Rebel T3i (ptp)
 	3229  IXUS 220 HS
 04aa  DaeWoo Telecom, Ltd
 04ab  Chromatic Research
@@ -3441,6 +3528,7 @@
 	301b  SK-8815 Keyboard
 	301c  Enhanced Performance Keyboard
 	3020  Enhanced Performance Keyboard
+	3025  NetVista Full Width Keyboard
 	3100  NetVista Mouse
 	3103  ScrollPoint Pro Mouse
 	3104  ScrollPoint Wireless Mouse
@@ -3461,7 +3549,6 @@
 	4604  Keyboard w/ Card Reader
 	4671  4820 LCD w/ MSR/KB
 04b4  Cypress Semiconductor Corp.
-	0000  Dacal DC-101 CD Library
 	0001  Mouse
 	0002  CY7C63x0x Thermometer
 	0033  Mouse
@@ -3482,6 +3569,7 @@
 	5201  Combi Keyboard-Hub (Hub)
 	5202  Combi Keyboard-Hub (Keyboard)
 	5500  HID->COM RS232 Adapter
+	5a9b  Dacal CD/DVD Library D-101/DC-300/DC-016RW
 	6370  ViewMate Desktop Mouse CC2201
 	6560  CY7C65640 USB-2.0 "TetraHub"
 	6830  CY7C68300A EZ-USB AT2 USB 2.0 to ATA/ATAPI
@@ -3626,6 +3714,7 @@
 	0851  Stylus SX410
 	0856  Stylus SX515W
 	085c  Stylus SX125
+	087b  Stylus Photo PX730WD (Printer, Scanner, Copier)
 04b9  Rainbow Technologies, Inc.
 	0300  SafeNet USB SuperPro/UltraPro
 	1000  iKey 1000 Token
@@ -3835,6 +3924,8 @@
 	01f7  FinePix J250 (PTP)
 	01fd  A160
 	023e  FinePix AX300
+	0240  FinePix S2950 Digital Camera
+	0241  FinePix S3200 Digital Camera
 04cc  ST-Ericsson
 	1122  Hub
 	1520  USB 2.0 Hub (Avocent KVM)
@@ -3851,6 +3942,7 @@
 	04ce  SL11DEMO, VID: 0x4ce, PID: 0x4ce
 	07d1  SL11R, VID: 0x4ce, PID: 0x07D1
 04cf  Myson Century, Inc.
+	0022  OCZ Alchemy Series Elixir II Keyboard
 	0800  MTP800 Mass Storage Device
 	8810  CS8810 Mass Storage Device
 	8811  CS8811 Mass Storage Device
@@ -3901,7 +3993,9 @@
 	1400  PS/2 keyboard + mouse controller
 	1503  Shortboard Lefty
 	1603  Keyboard
+	2013  Keyboard [Das Keyboard]
 	2221  Keyboard
+	a055  Keyboard
 04da  Panasonic (Matsushita)
 	0901  LS-120 Camera
 	0912  SDR-S10
@@ -3969,6 +4063,7 @@
 	8005  Zaurus A-300
 	8006  Zaurus SL-B500/SL-5600 PDA
 	8007  Zaurus C-700 PDA
+	9009  AR-M160
 	9014  IM-DR80 Portable NetMD Player
 	9031  Zaurus C-750/C-760/C-860/SL-C3000 PDA
 	9032  Zaurus SL-6000
@@ -4203,7 +4298,10 @@
 	5f02  NEXiO Sync
 	5f03  NEXiO Sync
 	5f04  NEXiO Sync
+	5f05  STORY Station 1TB
 	6032  G2 Portable hard drive
+	60b3  M2 Portable Hard Drive
+	60c4  M2 Portable Hard Drive USB 3.0
 	6601  Mobile Phone
 	6602  Galaxy
 	6603  Galaxy
@@ -4240,7 +4338,7 @@
 	684e  Wave (GT-S8500)
 	685b  GT-I9100 Phone [Galaxy S II] (mass storage mode)
 	685e  GT-I9100 Phone [Galaxy S II] (USB Debugging mode)
-	6860  GT-I9100 Phone [Galaxy S II]
+	6860  GT-I9100 Phone [Galaxy S II], GT-P7500 [Galaxy Tab 10.1]
 	6875  GT-B3710 Standalone LTE device (Commercial)
 	6876  GT-B3710 LTE Modem
 	6877  Galaxy S
@@ -4284,6 +4382,7 @@
 	0111  KU-9908 Keyboard
 	0112  KU-8933 Keyboard with PS/2 Mouse port
 	0116  KU-2971/KU-0325 Keyboard
+	0201  Gaming Keyboard KPD0250
 	0220  Wireless HID Receiver
 	0402  Genius LuxeMate i200 Keyboard
 	0403  KU-0420 keyboard
@@ -4324,6 +4423,7 @@
 	b036  Asus Integrated 0.3M UVC Webcam
 	b044  Acer CrystalEye Webcam
 	b057  integrated USB webcam
+	b059  CKF7037 HP webcam
 	b071  2.0M UVC Webcam / CNF7129
 	b091  Webcam
 	b104  CNF7069 Webcam
@@ -4332,15 +4432,22 @@
 	b175  4-Port Hub
 	b1aa  Webcam-101
 	b1b4  Lenovo Integrated Camera
+	b1b9  Asus Integrated Webcam
 	b1cf  Lenovo Integrated Camera
+	b217  Lenovo Integrated Camera (0.3MP)
 	b221  integrated camera
 	b230  Integrated HP HD Webcam
+	b257  Lenovo Integrated Camera
+	b26b  Sony Visual Communication Camera
+	b272  Lenovo EasyCamera
+	b2b9  Lenovo Integrated Camera UVC
 04f3  Elan Microelectronics Corp.
 	01a4  Wireless Keyboard
 	0210  AM-400 Hama Optical Mouse
 	0212  Laser Mouse
 	0214  Lynx M9 Optical Mouse
 	0230  3D Optical Mouse
+	0232  Mouse
 	02f4  2.4G Cordless Mouse
 04f4  Harting Elektronik, Inc.
 04f5  Fujitsu-ICL Systems, Inc.
@@ -4385,6 +4492,7 @@
 	002b  HL-5250DN Printer
 	002c  Printer
 	002d  Printer
+	0039  HL-5340 series
 	0100  MFC8600/9650 series
 	0101  MFC9600/9870 series
 	0102  MFC9750/1200 series
@@ -4535,6 +4643,7 @@
 	01ea  DCP-7030
 	01eb  MFC-7320
 	01f4  MFC-5890CN
+	0223  DCP-365CN
 	1000  Printer
 	1002  Printer
 	2002  PTUSB Printing
@@ -4542,6 +4651,7 @@
 	2015  QL-500 P-touch label printer
 	2016  QL-550 P-touch label printer
 	201a  PT-18R P-touch label printer
+	201b  QL-650TD P-Touch Label Printer
 	2027  QL-560 P-Touch Label Printer
 	2100  Card Reader Writer
 04fa  Dallas Semiconductor
@@ -4549,7 +4659,7 @@
 	4201  DS4201 Audio DAC
 04fb  Biostar Microtech International Corp.
 04fc  Sunplus Technology Co., Ltd
-	0003  CM1092 Optical Scroller Mouse
+	0003  CM1092 / Wintech CM-5098 Optical Mouse
 	0005  USB OpticalWheel Mouse
 	0013  ViewMate Desktop Mouse CC2201
 	0015  ViewMate Desktop Mouse CC2201
@@ -4648,6 +4758,7 @@
 	0122  Ethernet Adapter
 	0131  Bluetooth Device with trace filter
 	016a  Bluetooth Mini Dongle
+	0200  Nostromo SpeedPad n52te Gaming Keyboard
 	0201  Peripheral Switch
 	0208  USBView II Video Adapter [nt1004]
 	0210  F5U228 Hi-Speed USB 2.0 DVD Creator
@@ -4659,6 +4770,7 @@
 	0249  USB 2 Flash Media Device
 	0257  F5U257 Serial
 	0304  FSU304 USB 2.0 - 4 Ports Hub
+	0307  USB 2.0 - 7 ports Hub [FSU307]
 	0409  F5U409 Serial
 	0551  F6C550-AVR UPS
 	0706  2-N-1 USB 2.0 7-Port Hub (Lower half)
@@ -4668,7 +4780,9 @@
 	0815  Nostromo n52 HID SpeedPad Mouse Wheel
 	0826  ErgoFit Wireless Optical Mouse (HID)
 	0980  HID UPS Battery
-	1102  F7D1102 N150/Surf Micro Wireless Adapter v1000 [Realtek RTL8188CE-VAU]
+	1004  F9L1004 802.11n Surf N300 XR Wireless Adapter [Realtek RTL8192CU]
+	1102  F7D1102 N150/Surf Micro Wireless Adapter v1000 [Realtek RTL8188CUS]
+	1103  F9L1103 N750 DB 802.11abgn 2x3:3 [Ralink RT3573]
 	1202  F5U120-PC Parallel Printer Port
 	1203  F5U120-PC Serial Port
 	2103  F7D2102 802.11n N300 Micro Wireless Adapter v3000 [Realtek RTL8192CU]
@@ -4677,6 +4791,7 @@
 	3201  F1DF102U/F1DG102U Flip KVM
 	4050  ZD1211B
 	5055  F5D5055 Gigabit Network Adapter [AX88xxx]
+	6050  F6D6050 802.11abgn Wireless Adapter [Broadcom BCM4323]
 	6051  F5D6051 802.11b Wireless Network Adapter [ZyDAS ZD1201]
 	615a  F7D4101 / F9L1101 802.11abgn Wireless Adapter [Broadcom BCM4323]
 	7050  F5D7050 Wireless G Adapter v1000/v2000 [Intersil ISL3887]
@@ -4692,13 +4807,15 @@
 	815c  F5D8053 N Wireless USB Adapter v3000 [Ralink RT2870]
 	815f  F5D8053 N Wireless USB Adapter v6000 [Realtek RTL8192SU]
 	825a  F5D8055 N+ Wireless Adapter v1000 [Ralink RT2870]
-	825b  F5D8055 N+ Wireless Adapter v2000 [Ralink RT3070]
+	825b  F5D8055 N+ Wireless Adapter v2000 [Ralink RT3072]
 	845a  F7D2101 802.11n Surf & Share Wireless Adapter v1000 [Realtek RTL8192SU]
 	905b  F5D9050 Wireless G+ MIMO Network Adapter v3000 [Ralink RT2573]
 	905c  F5D9050 Wireless G+ MIMO Network Adapter v4000 [Ralink RT2573]
-	935a  F6D4050 N150 Enhanced Wireless Network Adapter v1000 [Ralink RT2870]
+	935a  F6D4050 N150 Enhanced Wireless Network Adapter v1000 [Ralink RT3070]
 	935b  F6D4050 N150 Enhanced Wireless Network Adapter v2000 [Ralink RT3070]
-	945a  F7D1101 Basic Wireless USB Adapter v1000 [Realtek RTL8188SU]
+	945a  F7D1101 v1 Basic Wireless Adapter [Realtek RTL8188SU]
+	945b  F7D1101 v2 Basic Wireless Adapter [Ralink RT3370]
+	d321  Dynex DX-NUSB 802.11bgn Wireless Adapter
 050e  Neon Technology, Inc.
 050f  KC Technology, Inc.
 	0001  Hub
@@ -4743,7 +4860,9 @@
 0525  Netchip Technology, Inc.
 	100d  RFMD Bluetooth Device
 	1080  NET1080 USB-USB Bridge
+	1200  SSDC Adapter II
 	1265  File-backed Storage Gadget
+	a0f0  Cambridge Electronic Devices Power1401 mk 2
 	a140  USB Clik! 40
 	a141  (OME) PocketZip 40 MP3 Player Driver
 	a220  GVC Bluetooth Wireless Adapter
@@ -4873,6 +4992,7 @@
 0547  Anchor Chips, Inc.
 	0001  ICSI Bluetooth Device
 	1002  Python2 WDM Encoder
+	1006  Hantek DSO-2100 UF
 	2131  AN2131 EZUSB Microcontroller
 	2235  AN2235 EZUSB-FX Microcontroller
 	2710  EZ-Link Loader (EZLNKLDR.SYS)
@@ -5044,8 +5164,11 @@
 	033e  DSC-W120
 	0346  Handycam DCR-SR55E
 	0348  HandyCam HDR-TG3E
+	035b  Walkman NWZ-A828
 	035c  NWZ-A726/A728/A729
 	0382  Memory Stick PRO-HG Duo Adaptor (MSAC-UAH1)
+	0385  Walkman NWZ-E436F
+	0387  Sony IC Recorder (P)
 	03bc  Webbie HD - MHS-CM1
 	03d3  DR-BT100CX
 	03d5  PlayStation Move motion controller
@@ -5092,6 +5215,7 @@
 	2011  UC-2324 4xSerial Ports [mos7840]
 	2202  CS124U Miniview II KVM Switch
 	2213  CS682 2-Port USB 2.0 DVI KVM Switch
+	2221  Winbond Hermon
 	2404  4-port switch
 	2600  IDE Bridge
 	2701  CE700A KVM Extender
@@ -5286,6 +5410,7 @@
 	0003  Device Bay Controller
 056e  Elecom Co., Ltd
 	0002  29UO Mouse
+	0072  Mouse
 	200c  LD-USB/TX
 	4002  Laneed 100Mbps Ethernet LD-USB/TX [pegasus]
 	4005  LD-USBL/TX
@@ -5312,8 +5437,10 @@
 	1253  Zoom V.92 Faxmodem
 	1300  SoftK56 Data Fax Voice CARP
 	1301  Modem Enumerator
+	1328  TrendNet TFM-561 modem
 	2000  SoftGate 802.11 Adapter
 	2002  SoftGate 802.11 Adapter
+	262a  tm5600 Video & Audio Grabber Capture
 	8390  WinFast PalmTop/Novo TV Video
 	8392  WinFast PalmTop/Novo TV Video
 	cafe  AccessRunner ADSL Modem
@@ -5422,8 +5549,9 @@
 	5601  AVM Fritz!WLAN [Texas Instruments TNETW1450]
 	6201  AVM Fritz!WLAN v1.1 [Texas Instruments TNETW1450]
 	62ff  AVM Fritz!WLAN USB (in CD-ROM-mode)
-	8401  AVM Fritz!WLAN N [Atheros AR9001U]
-	8402  AVM Fritz!WLAN N 2.4 [Atheros AR9001U]
+	8401  Fritz!WLAN N [Atheros AR9001U]
+	8402  Fritz!WLAN N 2.4 [Atheros AR9001U]
+	8403  Fritz!WLAN N v2 [Atheros AR9271]
 	84ff  AVM Fritz!WLAN USB N (in CD-ROM-mode)
 057d  Shark Multimedia, Inc.
 057e  Nintendo Co., Ltd
@@ -5595,6 +5723,7 @@
 	0008  Fujifilm MemoryCard ReaderWriter
 	0220  U2SCX SCSI Converter
 	b000  REX-USB60
+	b020  REX-USB60F
 0585  FlashPoint Technology, Inc.
 	0001  Digital Camera
 	0002  Digital Camera
@@ -5624,9 +5753,12 @@
 	3400  ZyAIR B-220 IEEE 802.11b Adapter
 	3401  ZyAIR G-220 802.11bg
 	3402  ZyAIR G-220F 802.11bg
+	3403  AG-200 802.11abg Wireless Adapter [Atheros AR5523]
 	3407  G-200 v2 802.11bg
+	3408  G-260 802.11bg
 	3409  AG-225H 802.11bg
 	340a  M-202 802.11bg
+	340c  G-270S 802.11bg Wireless Adapter [Atheros AR5523]
 	340f  G-220 v2 802.11bg
 	3410  ZyAIR G-202 802.11bg
 	3412  802.11bg
@@ -5635,8 +5767,11 @@
 	3416  NWD-210N 802.11b/g/n-draft wireless adapter
 	3417  NWD271N 802.11n Wireless Adapter [Atheros AR9001U-(2)NG]
 	3418  NWD211AN 802.11abgn Wireless Adapter [Ralink RT2870]
+	3419  G-220 v3 802.11bg Wireless Adapter [ZyDAS ZD1211B]
 	341a  NWD-270N Wireless N-lite USB Adapter
+	341e  NWD2105 802.11bgn Wireless Adapter [Ralink RT3070]
 	341f  NWD2205 802.11n Wireless N Adapter [Realtek RTL8192CU]
+	343e  N220 802.11bgn Wireless Adapter
 0587  America Kotobuki Electronics Industries, Inc.
 0588  Sapien Design
 0589  Victron
@@ -5677,11 +5812,11 @@
 	6360  Multimedia Card Reader
 	6361  Multimedia Card Reader
 	6362  Flash Card Reader/Writer
-	6364  Hi-Speed 7-in-1 Flash Card Reader/Writer [Sabrent]
+	6364  AU6477 Card Reader Controller
 	6366  Multi Flash Reader
 	6377  Multimedia Card Reader
 	6386  Memory Card
-	6387  Transcend JetFlash Flash Drive
+	6387  Flash Drive
 	6390  USB 2.0-IDE bridge
 	9213  MacAlly Kbd Hub
 	9215  AU9814 Hub
@@ -5704,6 +5839,7 @@
 	9510  ChunghwaTL USB02 Smartcard Reader
 	9520  EMV Certified Smart Card Reader
 	9720  USB-Serial Adapter
+	a014  Asus Integrated Webcam
 0590  Omron Corp.
 	0004  Cable Modem
 	000b  MR56SVS
@@ -5745,6 +5881,7 @@
 	0061  Varo PocketZip 40 MP3 Player
 	006d  HipZip MP3 Player
 	007c  Ultra Max USB/1394
+	007d  HTC42606 0G9AT00 [Iomega HDD]
 	007e  Mini 256MB/512MB Flash Drive [IOM2D5]
 	00db  FotoShow Zip 250 Driver
 	0150  Mass Storage Device
@@ -5880,6 +6017,19 @@
 	0236  Internal Keyboard/Trackpad (ANSI)
 	0237  Internal Keyboard/Trackpad (ISO)
 	0238  Internal Keyboard/Trackpad (JIS)
+	023f  Internal Keyboard/Trackpad (ANSI)
+	0240  Internal Keyboard/Trackpad (ISO)
+	0241  Internal Keyboard/Trackpad (JIS)
+	0242  Internal Keyboard/Trackpad (ANSI)
+	0243  Internal Keyboard/Trackpad (ISO)
+	0244  Internal Keyboard/Trackpad (JIS)
+	0245  Internal Keyboard/Trackpad (ANSI)
+	0246  Internal Keyboard/Trackpad (ISO)
+	0247  Internal Keyboard/Trackpad (JIS)
+	0250  MP110Z/B
+	0252  Internal Keyboard/Trackpad (ANSI)
+	0253  Internal Keyboard/Trackpad (ISO)
+	0254  Internal Keyboard/Trackpad (JIS)
 	0301  USB Mouse [Mitsumi, M4848]
 	0302  Optical Mouse [Fujitsu]
 	0304  Optical USB Mouse [Mitsumi]
@@ -5893,6 +6043,7 @@
 	1003  Hub in Pro Keyboard [Mitsumi, A1048]
 	1006  Hub in Aluminum Keyboard
 	1101  Speakers
+	1107  Thunderbolt Display Audio
 	1201  3G iPod
 	1202  iPod 2G
 	1203  iPod 4.Gen Grayscale 40G
@@ -5932,11 +6083,14 @@
 	129a  iPad
 	129e  iPod Touch 4.Gen
 	129f  iPad 2
+	12a0  iPhone 4S
+	12a2  iPad 2 (3G; 64GB)
 	1300  iPod Shuffle
 	1301  iPod Shuffle 2.Gen
 	1302  iPod Shuffle 3.Gen
 	1303  iPod Shuffle 4.Gen
 	1401  Modem
+	1402  Ethernet Adapter [A1277]
 	8202  HCF V.90 Data/Fax Modem
 	8203  Bluetooth HCI
 	8204  Bluetooth HCI [Bluetooth 2.0 + EDR, built-in]
@@ -5947,6 +6101,7 @@
 	8216  Bluetooth USB Host Controller
 	8217  Bluetooth USB Host Controller
 	8218  Bluetooth Host Controller
+	821a  Bluetooth Host Controller
 	8240  IR Receiver [built-in]
 	8241  IR Receiver [built-in]
 	8242  IR Receiver [built-in]
@@ -5956,14 +6111,23 @@
 	8502  Built-in iSight
 	8505  Built-in iSight
 	8507  Built-in iSight
+	8509  FaceTime HD Camera
 	911c  Hub in A1082 [Cinema HD Display 23"]
 	912f  Hub in 30" Cinema Display
+	9215  Studio Display 15"
+	9217  Studio Display 17"
+	9218  Cinema Display 23"
+	9219  Cinema Display 20"
 	921c  A1082 [Cinema HD Display 23"]
+	921e  Cinema Display 24"
 	9221  30" Cinema Display
+	9227  Thunderbolt Display
+	9232  Cinema HD Display 30"
 	ffff  Bluetooth in DFU mode - Driver
 05ad  Y.C. Cable U.S.A., Inc.
 05ae  Synopsys, Inc.
 05af  Jing-Mold Enterprise Co., Ltd
+	0806  HP SK806A Keyboard
 	0809  Wireless Keyboard and Mouse
 	0821  IDE to
 	3062  Cordless Keyboard
@@ -6011,6 +6175,7 @@
 	9001  Gobi Wireless Modem
 	9002  Gobi Wireless Modem
 	9008  Gobi Wireless Modem (QDL mode)
+	9018  Qualcomm HSUSB Device
 	9201  Gobi Wireless Modem (QDL mode)
 	9202  Gobi Wireless Modem
 	9203  Gobi Wireless Modem
@@ -6039,6 +6204,8 @@
 	6001  Ten-Keypad
 05c8  Cheng Uei Precision Industry Co., Ltd (Foxlink)
 	0103  FO13FF-65 PC-CAM
+	021a  HP Webcam
+	0318  Webcam
 	0403  Webcam
 05c9  Semtech Corp.
 05ca  Ricoh Co., Ltd
@@ -6054,6 +6221,7 @@
 	1810  Pavilion Webcam [R5U870]
 	1812  Pavilion Webcam
 	1814  HD Webcam
+	1820  Integrated Webcam
 	1830  Visual Communication Camera VGP-VCC2 [R5U870]
 	1832  Visual Communication Camera VGP-VCC3 [R5U870]
 	1833  Visual Communication Camera VGP-VCC2 [R5U870]
@@ -6071,6 +6239,7 @@
 	18b0  Sony Vaio Integrated Webcam
 	18b1  Sony Vaio Integrated Webcam
 	18b3  Sony Vaio Integrated Webcam
+	18b5  Sony Vaio Integrated Webcam
 	2201  RDC-7 Camera
 	2202  Caplio RR30
 	2203  Caplio 300G
@@ -6317,7 +6486,7 @@
 05e0  Symbol Technologies
 	0700  Bar Code Scanner (CS1504)
 	0800  Spectrum24 Wireless LAN Adapter
-	1200  DS6608 Bar Code Scanner
+	1200  Bar Code Scanner
 	1900  SNAPI Imaging Device
 	2000  MC3090 Rugged Mobile Computer
 	200d  MC70 Rugged Mobile Computer
@@ -6378,7 +6547,9 @@
 	0716  USB 2.0 Multislot Card Reader/Writer
 	0717  All-in-1 Card Reader
 	0718  IDE/SATA Adapter
+	0719  SATA adapter
 	0723  GL827L SD/MMC/MS Flash Card Reader
+	0726  SD Card Reader
 	0727  microSD Reader/Writer
 	0760  USB 2.0 Card Reader/Writer
 	0761  Genesys Mass Storage Device
@@ -6481,11 +6652,13 @@
 	1006  Japanese Keyboard - 260U
 	2101  Keyboard
 	2231  KSK-6001 UELX Keyboard
+	2270  Gigabyte K8100 Aivia Gaming Keyboard
 	5811  ACK-571U Wireless Keyboard
 	5903  Japanese Keyboard - 595U
 	6001  SolidTek USB 2p HUB
 	6002  SolidTek USB Keyboard
 	6003  Japanese Keyboard - 600HM
+	8007  P-W1G1F12 VER:1 [Macally MegaCam]
 	a001  Maxwell Compact Pc PM3
 060c  EEH Datalink GmbH
 060d  Auctor Corp.
@@ -6516,6 +6689,7 @@
 061b  Promptus Communications, Inc.
 061c  Act Labs, Ltd
 061d  Quatech, Inc.
+	c020  SSU-100
 061e  Nissei Electric Co.
 	0001  nissei 128DE-USB -
 	0010  nissei 128DE-PNA -
@@ -6540,6 +6714,8 @@
 	0102  Wireless Keyboard/Mouse Combo [MK1152WC]
 	0201  Defender Office Keyboard (K7310) S Zodiak KM-9010
 	0252  Emerge Uni-retractable Laser Mouse
+	3286  Nano Receiver [Sandstrom Laser Mouse SMWLL11]
+	6301  Trust Wireless Optical Mouse MI-4150K
 	9003  VoIP Conference Hub (A16GH)
 	9004  USR9602 USB Internet Mini Phone
 062b  Greatlink Electronics Taiwan, Ltd
@@ -6556,14 +6732,16 @@
 	0003  Vivicam 35Xx
 0638  Avision, Inc.
 	0268  iVina 1200U Scanner
-	026a  Minolta Dimage Scan Dual II
+	026a  Minolta Dimage Scan Dual II AF-2820U (2886)
 	0a10  iVina FB1600/UMAX Astra 4500
 	0a13  AV600U
-	0a16  SC-215
+	0a15  Konica Minolta SC-110
+	0a16  Konica Minolta SC-215
 	0a30  UMAX Astra 6700 Scanner
 	0a41  Avision AM3000/MF3000 Series
 	0f01  fi-4010CU
-	4004  Minolta Dimage Scan Elite II
+# typo?
+	4004  Minolta Dimage Scan Elite II AF-2920 (2888)
 0639  Chrontel, Inc.
 063a  Techwin Corp.
 063b  Taugagreining HF
@@ -6877,6 +7055,8 @@
 	3c06  54g USB Network Adapter
 0682  Victor Company of Japan, Ltd
 0684  Actiontec Electronics, Inc.
+0685  ZD Incorporated
+	7000  HSDPA Modem
 0686  Minolta Co., Ltd
 	2001  PagePro 4110W
 	2004  PagePro 1200W
@@ -6891,7 +7071,7 @@
 	3034  Develop D 2050iD PCL
 	4001  Dimage 2300
 	4003  Dimage 2330 Zoom Camera
-	4004  Scan Elite II
+	4004  Dimage Scan Elite II AF-2920 (2888)
 	4005  Minolta DiMAGE E201 Mass Storage Device
 	4006  Dimage 7 Camera
 	4007  Dimage S304 Camera
@@ -6900,8 +7080,8 @@
 	400a  Dimage S404 Camera
 	400b  Dimage 7i Camera
 	400c  Dimage F100 Camera
-	400d  Scan Dual III
-	400e  Dimage 5400
+	400d  Dimage Scan Dual III AF-2840 (2889)
+	400e  Dimage Scan Elite 5400 (2890)
 	400f  Dimage 7Hi Camera
 	4010  Dimage Xi Camera
 	4011  Dimage F300 Camera
@@ -7012,7 +7192,10 @@
 	05d2  PC Dash 2
 	075c  X52 Flight Controller
 	0762  Saitek X52 Pro Flight Control System
+	0763  Pro Flight Rudder Pedals
 	0805  R440 Force Wheel
+	0bac  Pro Flight Yoke
+	0c2d  Pro Flight Quadrant
 	1003  GM2 Action Pad
 	1009  Action Pad
 	100a  SP550 Pad and Joystick Combo
@@ -7025,6 +7208,7 @@
 	3589  P3000 Wireless Pad
 	35be  Cyborg Evo
 	5509  P3000 Wireless Pad
+	712c  Pro Flight Yoke integrated hub
 	8000  Gamers' Keyboard
 	801e  Cyborg 3D Digital Stick II
 	8020  Eclipse Keyboard
@@ -7070,6 +7254,7 @@
 	0121  SpeedTouch 121g Wireless Dongle
 	2001  SPEED TOUCH Card
 	4061  SpeedTouch ISDN or ADSL Modem
+	4062  SpeedTouch ISDN or ADSL router
 	a5a5  DynaMiTe Modem
 06ba  Smooth Cord & Connector Co., Ltd
 06bb  EDA, Inc.
@@ -7134,17 +7319,17 @@
 	0009  Monitor Control
 06ca  Newer Technology, Inc.
 06cb  Synaptics, Inc.
-	0001  HID Device
-	0002  HID Device
-	0003  HID Device
+	0001  TouchPad
+	0002  Integrated TouchPad
+	0003  cPad
 	0005  Touchpad/FPS
-	0006  HID Device
-	0007  HID Device
-	0008  HID Device
+	0006  TouchScreen
+	0007  USB Styk
+	0008  WheelPad
 	0009  Composite TouchPad and TrackPoint
 	000e  HID Device
-	0010  Composite Human Interface Device
-	0013  Human Interface Device
+	0010  Wireless TouchPad
+	0013  DisplayPad
 06cc  Terayon Communication Systems
 	0101  Cable Modem
 	0102  Cable Modem
@@ -7186,6 +7371,8 @@
 	012a  USA-49Wlc serial adapter
 	0201  UIA-10 Digital Media Remote [Cypress AN2131SC]
 	0202  UIA-11 Digital Media Remote
+06ce  Contec
+	8311  COM-1(USB)H
 06cf  SpheronVR AG
 	1010  PanoCam 10
 	1012  PanoCam 12/12X
@@ -7247,6 +7434,7 @@
 	a191  Instant VideoXpress
 	a337  Mini DigitalTV
 	a701  DVD Xpress
+	a708  saa7114H video input card (Instant VideoMPX)
 	b337  Mini DigitalTV
 	b701  DVD Xpress B
 06e4  Alcatel Microelectronics
@@ -7507,6 +7695,7 @@
 	4588  XBox Device
 	8818  Street Fighter IV Arcade FightStick (PS3)
 073a  Chaplet Systems, Inc.
+	2230  infrared dongle for remote
 073b  Suncom Technologies
 073c  Industrial Electronic Engineers, Inc.
 	0305  Pole Display (PC305-3415  2 x 20 Line Display)
@@ -7596,6 +7785,7 @@
 	201a  M-Audio Micro
 	201b  M-Audio RunTime DFU
 	201d  M-Audio Producer
+	2024  M-Audio Fast Track MKII
 	2080  M-Audio RunTime DFU
 	2081  M-Audio RunTime DFU / Fast Track Ultra 8R
 	2803  M-Audio Audiophile DFU
@@ -7610,6 +7800,7 @@
 	0005  Cyber Power UPS
 	0501  CP1500 AVR UPS
 0765  X-Rite, Inc.
+	5001  Huey PRO Colorimeter
 	d094  X-Rite DTP94 [Quato Silver Haze Pro]
 0766  Jess-Link Products Co., Ltd
 	0204  TopSpeed Cyberlink Remote Control
@@ -7785,12 +7976,15 @@
 	0063  LDR Device
 	0064  LDR-R Device
 	00b3  DVD Multi-plus unit LDR-H443U2
+	0105  LAN-TX/U1H2 10/100 Ethernet Adapter [pegasus II]
 	010c  Realtek RTL8187 Wireless 802.11g 54Mbps Network Adapter
+	0160  LAN-GTJ/U2A
 	0162  LAN-WN22/U2 Wireless LAN Adapter
 	0163  LAN-WN12/U2 Wireless LAN Adapter
 	0164  LAN-W150/U2M Wireless LAN Adapter
 	0166  LAN-W300N/U2 Wireless LAN Adapter
 	0168  LAN-W150N/U2 Wireless LAN Adapter
+	0170  LAN-W300AN/U2 Wireless LAN Adapter
 078b  Happ Controls, Inc.
 	0010  Driving UGCI
 	0020  Flying UGCI
@@ -7858,7 +8052,9 @@
 	0018  Wireless LAN USB-11 mini 2
 	001a  ULUSB-11 Key
 	001c  CG-WLUSB2GTST 802.11g Wireless Adapter [Intersil ISL3887]
+	002e  CG-WLUSB2GPX [Ralink RT2571W]
 	002f  CG-WLUSB2GNL
+	0031  CG-WLUSB2GS 802.11bg [Atheros AR5523]
 	003c  CG-WLUSB2GNL
 	003f  CG-WLUSB300AGN
 	0041  CG-WLUSB300GNS
@@ -8019,6 +8215,8 @@
 07bd  Webgear, Inc.
 07be  Veridicom
 07c0  Code Mercenaries Hard- und Software GmbH
+	1113  JoyWarrior24F8
+	1116  JoyWarrior24F14
 	1121  The Claw
 	1500  IO-Warrior 40
 	1501  IO-Warrior 24
@@ -8187,28 +8385,30 @@
 	13ed  VvBus for Helium 2xx
 	13f1  DSL-302G Modem
 	13f2  DSL-502G Router
-	3300  DWA-130 802.11n Wireless N Adapter(rev.E) [Realtek RTL8192SU]
-	3302  DWA-130 802.11n Wireless N Adapter(rev.C2) [Realtek RTL8192SU]
+	3300  DWA-130 802.11n Wireless N Adapter(rev.E) [Realtek RTL8191SU]
+	3302  DWA-130 802.11n Wireless N Adapter(rev.C2) [Realtek RTL8191SU]
 	3303  DWA-131 802.11n Wireless N Nano Adapter(rev.A1) [Realtek RTL8192SU]
+	3304  FR-300USB 802.11bgn Wireless Adapter
 	3a07  WUA-2340 RangeBooster G Adapter(rev.A) [Atheros AR5523]
 	3a08  WUA-2340 RangeBooster G Adapter(rev.A) (no firmware) [Atheros AR5523]
 	3a09  DWA-160 802.11abgn Xtreme N Dual Band Adapter(rev.A2) [Atheros AR9170+AR9104]
 	3a0d  DWA-120 802.11g Wireless 108G Adapter [Atheros AR5523]
-	3a0f  DWA-130 802.11n Wireless N Adapter(rev.D) [Atheros AR9001U-(2)NG]
+	3a0f  DWA-130 802.11n Wireless N Adapter(rev.D) [Atheros AR9170+AR9102]
 	3a10  DWA-126 802.11n Wireless Adapter [Atheros AR9271]
 	3b01  AirPlus G DWL-G122 Wireless Adapter(rev.D) [Marvell 88W8338+88W8010]
-	3b10  RangeBooster N Adapter
-	3b11  DWA-130 802.11n Wireless N Adapter(rev.A1) [Marvell W8360USB]
-	3c03  AirPlus G DWL-G122 Wireless Adapter(rev.C1) [Ralink RT73]
+	3b10  DWA-142 RangeBooster N Adapter [Marvell 88W8362+88W8060]
+	3b11  DWA-130 802.11n Wireless N Adapter(rev.A1) [Marvell 88W8362+88W8060]
+	3c03  AirPlus G DWL-G122 Wireless Adapter(rev.C1) [Ralink RT2571W]
 	3c04  WUA-1340
 	3c05  EH103 Wireless G Adapter
-	3c07  Wireless G DWA-110 Adapter
+	3c06  DWA-111 802.11bg Wireless Adapter [Ralink RT2571W]
+	3c07  DWA-110 Wireless G Adapter(rev.A1) [Ralink RT2571W]
 	3c09  DWA-140 RangeBooster N Adapter(rev.B1) [Ralink RT2870]
-	3c0a  DWA-140 RangeBooster N Adapter(rev.B2) [Ralink RT2870]
-	3c0b  DWA-110 Wireless G USB Adapter(rev.B)
+	3c0a  DWA-140 RangeBooster N Adapter(rev.B2) [Ralink RT3072]
+	3c0b  DWA-110 Wireless G Adapter(rev.B) [Ralink RT2870]
 	3c0d  DWA-125 Wireless N 150 Adapter(rev.A1) [Ralink RT3070]
 	3c0e  WUA-2340 RangeBooster G Adapter(rev.B) [Ralink RT2070]
-	3c0f  AirPlus G DWL-G122 Wireless Adapter(rev.E) [Ralink RT2870]
+	3c0f  AirPlus G DWL-G122 Wireless Adapter(rev.E1) [Ralink RT2070]
 	3c10  DWA-160 802.11abgn Xtreme N Dual Band Adapter(rev.A1) [Atheros AR9170+AR9104]
 	3c11  DWA-160 Xtreme N Dual Band USB Adapter(rev.B) [Ralink RT2870]
 	3c13  DWA-130 802.11n Wireless N Adapter(rev.B) [Ralink RT2870]
@@ -8273,7 +8473,8 @@
 0802  Mako Technologies, LLC
 0803  Zoom Telephonics, Inc.
 	1300  V92 Faxmodem
-	4310  Wireless-G
+	4310  4410a Wireless-G Adapter [Intersil ISL3887]
+	4410  4410b Wireless-G Adapter [ZyDAS ZD1211B]
 	5241  Cable Modem
 	5551  DSL Modem
 	9700  2986L FaxModem
@@ -8419,7 +8620,7 @@
 	b522  SMCWUSBS-N2 EZ Connect N Wireless Adapter [Ralink RT2870]
 	bb01  BlueExpert Bluetooth Device
 	c003  802.11b Wireless Adapter
-	c501  Zoom Wireless-G
+	c501  Zoom 4410 Wireless-G [Intersil ISL3887]
 	c561  802.11a/g Wireless Adapter
 	d522  Speedport W 102 Stick IEEE 802.11n USB 2.0 Adapter
 	e501  ZD1211B
@@ -8457,6 +8658,7 @@
 	9001  WN111(v2) RangeMax Next Wireless [Atheros AR9170+AR9101]
 	9010  WNDA3100v1 802.11abgn [Atheros AR9170+AR9104]
 	9011  WNDA3100v2 802.11abgn [Broadcom BCM4323]
+	9012  WNDA4100 802.11abgn 3x3:3 [Ralink RT3573]
 	9018  WNDA3200 802.11abgn Wireless Adapter [Atheros AR7010+AR9280]
 	9020  WNA3100(v1) Wireless-N 300 [Broadcom BCM43231]
 	9030  WNA1100 Wireless-N 150 [Atheros AR9271]
@@ -8617,6 +8819,14 @@
 	2902  PCM2902 Audio Codec
 	2904  PCM2904 Audio Codec
 	2910  PCM2912 Audio Codec
+	29b0  PCM2900B Audio CODEC
+	29b2  PCM2902 Audio CODEC
+	29b3  PCM2903B Audio CODEC
+	29b6  PCM2906B Audio CODEC
+	29c0  PCM2900C Audio CODEC
+	29c2  PCM2902C Audio CODEC
+	29c3  PCM2903C Audio CODEC
+	29c6  PCM2906C Audio CODEC
 08bd  Citizen Watch Co., Ltd
 	0208  CLP-521 Label Printer
 	1100  X1-USB Floppy
@@ -8791,6 +9001,9 @@
 08ff  AuthenTec, Inc.
 	1600  AES1600
 	1610  AES1600
+	1660  AES1660 Fingerprint Sensor
+	1680  AES1660 Fingerprint Sensor
+	168f  AES1660 Fingerprint Sensor
 	2500  AES2501
 	2501  AES2501
 	2502  AES2501
@@ -8808,6 +9021,7 @@
 	250e  AES2501
 	250f  AES2501
 	2510  AES2510
+	2550  AES2550 Fingerprint Sensor
 	2580  AES2501 Fingerprint Sensor
 	2588  AES2501
 	2589  AES2501
@@ -8817,6 +9031,9 @@
 	258d  AES2501
 	258e  AES2501
 	258f  AES2501
+	2660  AES2660 Fingerprint Sensor
+	2680  AES2660 Fingerprint Sensor
+	268f  AES2660 Fingerprint Sensor
 	2810  AES2810
 	3400  AES3400 TruePrint Sensor
 	3401  AES3400 Sensor
@@ -8888,6 +9105,7 @@
 	337b  Silicon Motion Camera
 	3710  Silicon Motion Camera
 	3720  Silicon Motion Camera
+	37bc  HP Webcam-101 Integrated Camera
 	37c0  Silicon Motion Camera
 	6000  SD/SDHC Card Reader (SG365 / FlexiDrive XC+)
 	6200  microSD card reader
@@ -8946,6 +9164,7 @@
 	2295  Colorado 300
 	22b6  eTrex Vista HCx (Mass Storage mode)
 	2353  Nüvi 205T
+	2519  eTrex 30
 	2535  Edge 800
 0920  Echelon Co.
 	7500  Network Interface
@@ -9017,6 +9236,7 @@
 	130b  F3507g Mobile Broadband Module
 	130c  F3607gw Mobile Broadband Module
 	1311  F3607gw v2 Mobile Broadband Module
+	1400  Memory Stick 2GB
 	642f  TravelDrive
 	6506  TravelDrive 2C
 	6507  TravelDrive 2C
@@ -9073,6 +9293,7 @@
 	0030  Composite Device, Mass Storage Device (Flash Drive) amd HID
 	003c  Rhythmedics HID Bootloader
 0939  Lumberg, Inc.
+	0b15  Toshiba Stor.E Alu 2 1TB (PX1710E-1HJ0)
 093a  Pixart Imaging, Inc.
 	0007  CMOS 100K-R Rev. 1.90
 	010e  Digital camera, CD302N/Elta Medi@ digi-cam/HE-501A
@@ -9117,9 +9338,11 @@
 0944  KORG, Inc.
 	0001  PXR4 4-Track Digital Recorder
 	0020  KAOSS Pad KP3 Dynamic Effect/Sampler
+	0023  KAOSSILATOR PRO Dynamic Phrase Synthesizer
 	010d  nanoKEY MIDI keyboard
 	010e  nanoPAD pad controller
 	010f  nanoKONTROL studio controller
+	0117  nanoKONTROL2 MIDI Controller
 	0f03  K-Series K61P MIDI studio controller
 0945  Pasco Scientific
 0948  Kronauer music in digital
@@ -9148,7 +9371,9 @@
 	1624  DataTraveler G2 4GB Pen Drive
 	1625  DataTraveler 101 II
 	162a  DataTraveler 112 4GB Pen Drive
+	1642  DT101 G2
 	1643  DataTraveler G3 4GB
+	1653  Data Traveler 100 G2 8 GiB
 0954  RPM Systems Corp.
 0955  NVidia Corp.
 	7100  Notion Ink Adam
@@ -9156,6 +9381,8 @@
 0957  Agilent Technologies, Inc.
 	0200  E-Video DC-350 Camera
 	0202  E-Video DC-350 Camera
+	0518  82357B GPIB Interface
+	1745  Test and Measurement Device (IVI)
 0958  CompuLink Research, Inc.
 0959  Cologne Chip AG
 	2bd0  Intelligent ISDN (Ver. 3.60.04)
@@ -9190,6 +9417,8 @@
 097b  Knudsen Engineering, Ltd
 097c  Marunix Co., Ltd
 097d  Rosun Technologies, Inc.
+097e  Biopac Systems Inc.
+	0035  MP35 v1.0
 097f  Barun Electronics Co., Ltd
 0981  Oak Technology, Ltd
 0984  Apricorn
@@ -9234,9 +9463,21 @@
 	0001  MySmartPad
 09bf  Auerswald GmbH & Co. KG
 	00c0  COMpact 2104 ISDN PBX
-	00db  COMpact 4410/2206 ISDN ISDN
-	00f1  COMfort System Telephones
+	00db  COMpact 4410/2206 ISDN
+	00dc  COMpact 4406 DSL (PBX)
+	00dd  COMpact 2204 (PBX)
+	00de  COMpact 2104 (Rev.2 PBX)
+	00e0  COMmander Business (PBX)
+	00e2  COMmander Basic.2 (PBX)
+	00f1  COMfort 2000 (System telephone)
+	00f2  COMfort 1200 (System telephone)
+	00f5  COMfortel 2500 (System telephone)
+	8000  COMpact 2104 DSL (DSL modem)
+	8001  COMpact 4406 DSL (DSL modem)
+	8002  Analog/ISDN Converter (Line converter)
+	8005  WG-640 (Automatic event dialer)
 09c0  Genpix Electronics, LLC
+	0136  Axon CNS, MultiClamp 700B
 	0202  8PSK DVB-S tuner
 	0203  Skywalker-1 DVB-S tuner
 	0204  Skywalker-CW3K DVB-S tuner
@@ -9268,6 +9509,7 @@
 09da  A4 Tech Co., Ltd
 	0006  Optical Mouse WOP-35 / Trust 450L Optical Mouse
 	000a  Optical Mouse Opto 510D
+	000e  X-F710F Optical Mouse 3xFire Gaming Mouse
 	0018  Trust Human Interface Device
 	001a  Wireless Mouse & RXM-15 Receiver
 	002a  Wireless Optical Mouse NB-30
@@ -9477,7 +9719,7 @@
 	0201  iLine10(tm) Network Adapter
 	2000  Bluetooth Device
 	2001  Bluetooth Device
-	2009  Bluetooth Controller
+	2009  BCM2035 Bluetooth
 	200a  BCM2035 Bluetooth dongle
 	200f  Bluetooth Controller
 	201d  Bluetooth Device
@@ -9487,13 +9729,13 @@
 	2033  BCM2033 Bluetooth
 	2035  BCM2035 Bluetooth
 	2038  Blutonium Device
-	2039  Bluetooth Device
+	2039  BCM2045 Bluetooth
 	2045  Bluetooth Controller
 	2046  Bluetooth Device
 	2047  Bluetooth Device
 	205e  Bluetooth Device
 	2100  Bluetooth 2.0+eDR dongle
-	2101  Bluetooth Controller
+	2101  BCM2045 Bluetooth
 	2102  ANYCOM Blue USB-200/250
 	2110  Bluetooth Controller
 	2111  ANYCOM Blue USB-UHE 200/250
@@ -9525,7 +9767,8 @@
 	5802  BCM5880 Secure Applications Processor with fingerprint touch sensor
 	5803  BCM5880 Secure Applications Processor with secure keyboard
 	6300  Pirelli Remote NDIS Device
-	bd13  Arcadyan WN8522B [Broadcom BCM4323]
+	bd11  TiVo AG0100 802.11bg Wireless Adapter [Broadcom BCM4320]
+	bd13  BCM4323 802.11abgn Wireless Adapter
 	d11b  Eminent EM4045 [Broadcom 4320 USB]
 0a5d  Diatrend Corp.
 0a5f  Zebra
@@ -9648,6 +9891,26 @@
 0aab  Vision Shape Europe SA
 0aac  iCompression, Inc.
 0aad  Rohde & Schwarz GmbH & Co. KG
+	0003  NRP-Z21
+	000c  NRP-Z11
+	0013  NRP-Z22
+	0014  NRP-Z23
+	0015  NRP-Z24
+	0016  NRP-Z51
+	0017  NRP-Z52
+	0018  NRP-Z55
+	0019  NRP-Z56
+	0021  NRP-Z91
+	0023  NRP-Z81
+	002c  NRP-Z31
+	002d  NRP-Z37
+	002f  NRP-Z27
+	0051  NRP-Z28
+	0052  NRP-Z98
+	0062  NRP-Z92
+	0070  NRP-Z57
+	0083  NRP-Z85
+	0095  NRP-Z86
 0aae  NEC infrontia Corp. (Nitsuko)
 0aaf  Digitalway Co., Ltd
 0ab0  Arrow Strong Electronics Co., Ltd
@@ -9678,6 +9941,7 @@
 	307b  USB 1.1 Webcam
 	332d  Vega USB 2.0 Camera
 	3343  Sirius USB 2.0 Camera
+	3370  Traveler TV 6500 SF Dia-scanner
 	3420  Venus USB2.0 Camera
 	c001  Sony embedded vimicro Camera
 	c002  Visual Communication Camera VGP-VCC1
@@ -9697,8 +9961,9 @@
 	1060  OPEN NT1 Plus II
 0acc  Koga Electronics Co.
 0acd  ID Tech
-	0401  ID TECH Spectrum III Hybrid Smartcard Reader
-	0630  ID TECH Spectrum III Mag-Only Insert Reader (SPT3-355 Series) USB-CDC
+	0300  IDT1221U RS-232 Adapter
+	0401  Spectrum III Hybrid Smartcard Reader
+	0630  Spectrum III Mag-Only Insert Reader (SPT3-355 Series) USB-CDC
 0ace  ZyDAS
 	1201  ZD1201 802.11b
 	1211  ZD1211 802.11g
@@ -9759,6 +10024,7 @@
 	7501  Globetrotter HSUPA Modem (icon 411 aka "Vodafone K3760")
 	7601  Globetrotter MO40x 3G Modem (GTM 382)
 	7701  Globetrotter HSUPA Modem (aka icon 451)
+	d055  Globetrotter GI0505 [iCON 505]
 0af6  Silver I Co., Ltd
 0af7  B2C2, Inc.
 	0101  Digital TV USB Receiver (DVB-S/T/C / ATSC)
@@ -9800,13 +10066,18 @@
 	1751  BT-253 Bluetooth Adapter
 	175b  Laptop OLED Display
 	1760  802.11n Network Adapter
-	1761  802.11n Network Adapter
+	1761  USB-N11 802.11n Network Adapter [Ralink RT2870]
 	1774  Gobi Wireless Modem (QDL mode)
 	1776  Gobi Wireless Modem
 	1779  My Cinema U3100 Mini Plus [AF9035A]
-	1784  USB-N13 802.11n Network Adapter [Ralink RT3072]
-	1786  USB-N10 802.11n Network Adapter [Realtek RTL8192SU]
+	1784  USB-N13 802.11n Network Adapter (rev. A1) [Ralink RT3072]
+	1786  USB-N10 802.11n Network Adapter [Realtek RTL8188SU]
 	1791  WL-167G v3 802.11n Adapter [Realtek RTL8188SU]
+	179d  USB-N53 802.11abgn Network Adapter [Ralink RT3572]
+	179e  Eee Note EA800 (network mode)
+	179f  Eee Note EA800 (tablet mode)
+	17a1  Eee Note EA800 (mass storage mode)
+	17ab  USB-N13 802.11n Network Adapter (rev. B1) [Realtek RTL8192CU]
 	6101  Cable Modem
 	620a  Remote NDIS Device
 	b700  Broadcom Bluetooth 2.1
@@ -9821,6 +10092,7 @@
 0b0d  ProjectLab
 	0000  CenturyCD
 0b0e  GN Netcom
+	1022  Jabra PRO 9450, Type 9400BS (DECT Headset)
 0b0f  AVID Technology
 0b10  Pcally
 0b11  I Tech Solutions Co., Ltd
@@ -9857,7 +10129,7 @@
 	1602  ZyXEL ZyAIR B200 802.11b Adapter
 	1612  AIR.Mate 2@net 802.11b Adapter
 	1613  802.11b Wireless LAN Adapter
-	1620  Allnet USB 2.0 Wireless Network Adapter
+	1620  Allnet Wireless Network Adapter [Envara WiND512]
 	1630  QuickWLAN 802.11bg
 	5630  802.11bg
 	6630  ZD1211
@@ -9870,6 +10142,7 @@
 	0011  Crossam2+USB IR commander
 0b43  Play.com, Inc.
 	0003  PS2 Controller Converter
+	0005  GameCube Adaptor
 0b47  Sportbug.com, Inc.
 0b48  TechnoTrend AG
 	1003  Technotrend/Hauppauge USB-Nova
@@ -9965,6 +10238,8 @@
 	1720  10/100 Ethernet
 	1780  AX88178
 	7720  AX88772
+	772a  AX88772A Fast Ethernet
+	7e2b  AX88772B
 0b96  Sewon Telecom
 0b97  O2 Micro, Inc.
 	7732  Smart Card Reader
@@ -9993,7 +10268,7 @@
 	0111  USR5420 802.11g Adapter [Broadcom 4320 USB]
 	0118  U5 802.11g Adapter
 	011b  Wireless MAXg Adapter [Broadcom 4320]
-	0121  USR5423 802.11bg
+	0121  USR5423 802.11bg Wireless Adapter [ZyDAS ZD1211B]
 	6112  FaxModem Model 5633
 0bb0  Concord Camera Corp.
 	0100  Sound Vision Stream
@@ -10003,7 +10278,7 @@
 	0302  U10H010 802.11b Wireless Adapter [Intersil PRISM 3]
 	6098  USB Cable Modem
 0bb3  Ofuji Technology
-0bb4  High Tech Computer Corp.
+0bb4  HTC (High Tech Computer Corp.)
 	00ce  mmO2 XDA GSM/GPRS Pocket PC
 	00cf  SPV C500 Smart Phone
 	0a01  PocketPC Sync
@@ -10179,6 +10454,7 @@
 	0c13  Diamond
 	0c1f  Sony Ericsson XPERIA X1
 	0c5f  Snap
+	0c86  Sensation
 	0c87  Desire (debug)
 	0c8d  EVO 4G (debug)
 	0c91  Vision
@@ -10187,6 +10463,7 @@
 	0c99  Desire (debug)
 	0c9e  Incredible
 	0ca2  Desire HD (debug mode)
+	0ca5  Android Phone [Evo Shift 4G]
 	0ff8  Desire HD (Tethering Mode)
 	0ff9  Desire / Desire HD / Hero (Charge Mode)
 	0ffe  Desire HD (modem mode)
@@ -10205,8 +10482,11 @@
 0bc2  Seagate RSS LLC
 	2000  Storage Adapter V3 (TPP)
 	2200  FreeAgent Go FW
+	2300  Expansion Portable
+	3332  Expansion
 	5021  FreeAgent GoFlex USB 2.0
 	5031  FreeAgent GoFlex USB 3.0
+	50a5  FreeAgent GoFlex Desk USB 3.0
 0bc3  IPWireless, Inc.
 	0001  UMTS-TDD (TD-CDMA) modem
 0bc4  Microcube Corp.
@@ -10247,6 +10527,7 @@
 	0117  Mass Storage Device
 	0118  Mass Storage Device
 	0138  Card reader
+	0139  Card reader
 	0151  Mass Storage Device (Multicard Reader)
 	0152  Mass Storage Device
 	0153  Mass Storage Device
@@ -10260,16 +10541,22 @@
 	0171  Mass Storage Device
 	0176  Mass Storage Device
 	0178  Mass Storage Device
+	0186  Card Reader
 	2831  RTL2831U DVB-T
 	2832  RTL2832U DVB-T
+	2838  RTL2838 DVB-T
 	8150  RTL8150 Fast Ethernet Adapter
 	8151  RTL8151 Adapteon Business Mobile Networks BV
 	8171  RTL8188SU 802.11n WLAN Adapter
 	8172  RTL8191SU 802.11n WLAN Adapter
-	8176  RTL8188CUS 802.11n WLAN
+	8174  RTL8192SU 802.11n WLAN Adapter
+	8176  RTL8188CUS 802.11n WLAN Adapter
+	8178  RTL8192CU 802.11n WLAN Adapter
+	817f  RTL8188RU 802.11n WLAN Adapter
 	8187  RTL8187 Wireless Adapter
 	8189  RTL8187B Wireless 802.11g 54Mbps Network Adapter
 	8192  RTL8192U 802.11n Wireless Adapter
+	8193  RTL8192DU 802.11an WLAN Adapter
 	8197  RTL8187B Wireless Adapter
 	8198  RTL8187B Wireless Adapter
 0bdb  Ericsson Business Mobile Networks BV
@@ -10293,6 +10580,7 @@
 0be6  Dong Guan Humen Wonderful Wire Cable Factory
 0bed  Silicon Labs
 	1100  MEI (TM) Cashflow-SC Bill/Voucher Acceptor
+	1101  Series 2000 Combo Acceptor
 0bee  LTK Industries, Ltd
 0bef  Way2Call Communications
 0bf0  Pace Micro Technology PLC
@@ -10380,6 +10668,7 @@
 	0012  Bluetooth Device(BC04-External)
 	0018  Bluetooth Device(BC04-External)
 	0019  Bluetooth Device
+	0021  Bluetooth Device
 	0c24  Bluetooth Device(SAMPLE)
 	ffff  Bluetooth module with BlueCore in DFU mode
 0c25  Sampo Corp.
@@ -10533,9 +10822,12 @@
 	63f1  Integrated Webcam
 	63f8  Sonix Integrated Webcam
 	6409  Webcam
+	6413  Integrated Webcam
 	6417  Integrated Webcam
 	641d  1.3 MPixel Integrated Webcam
 	6480  Sonix 1.3 MP Laptop Integrated Webcam
+	64bd  Sony Visual Communication Camera
+	7402  TEMPerHUM Temperature & Humidity Sensor
 	8000  DC31VC
 	8006  Dual Mode Camera (8006 VGA)
 	800a  Vivitar Vivicam3350B
@@ -10550,7 +10842,57 @@
 0c4c  Needham's Electronics
 	0021  EMP-21 Universal Programmer
 0c52  Sealevel Systems, Inc.
-	2101  Serial Converter
+	2101  SeaLINK+232
+	2102  SeaLINK+485
+	2103  SeaLINK+232I
+	2104  SeaLINK+485I
+	2211  SeaPORT+2/232 (Port 1)
+	2212  SeaPORT+2/485 (Port 1)
+	2213  SeaPORT+2 (Port 1)
+	2221  SeaPORT+2/232 (Port 2)
+	2222  SeaPORT+2/485 (Port 2)
+	2223  SeaPORT+2 (Port 2)
+	2411  SeaPORT+4/232 (Port 1)
+	2412  SeaPORT+4/485 (Port 1)
+	2413  SeaPORT+4 (Port 1)
+	2421  SeaPORT+4/232 (Port 2)
+	2422  SeaPORT+4/485 (Port 2)
+	2423  SeaPORT+4 (Port 2)
+	2431  SeaPORT+4/232 (Port 3)
+	2432  SeaPORT+4/485 (Port 3)
+	2433  SeaPORT+4 (Port 3)
+	2441  SeaPORT+4/232 (Port 4)
+	2442  SeaPORT+4/485 (Port 4)
+	2443  SeaPORT+4 (Port 4)
+	2811  SeaLINK+8/232 (Port 1)
+	2812  SeaLINK+8/485 (Port 1)
+	2813  SeaLINK+8 (Port 1)
+	2821  SeaLINK+8/232 (Port 2)
+	2822  SeaLINK+8/485 (Port 2)
+	2823  SeaLINK+8 (Port 2)
+	2831  SeaLINK+8/232 (Port 3)
+	2832  SeaLINK+8/485 (Port 3)
+	2833  SeaLINK+8 (Port 3)
+	2841  SeaLINK+8/232 (Port 4)
+	2842  SeaLINK+8/485 (Port 4)
+	2843  SeaLINK+8 (Port 4)
+	2851  SeaLINK+8/232 (Port 5)
+	2852  SeaLINK+8/485 (Port 5)
+	2853  SeaLINK+8 (Port 5)
+	2861  SeaLINK+8/232 (Port 6)
+	2862  SeaLINK+8/485 (Port 6)
+	2863  SeaLINK+8 (Port 6)
+	2871  SeaLINK+8/232 (Port 7)
+	2872  SeaLINK+8/485 (Port 7)
+	2873  SeaLINK+8 (Port 7)
+	2881  SeaLINK+8/232 (Port 8)
+	2882  SeaLINK+8/485 (Port 8)
+	2883  SeaLINK+8 (Port 8)
+	9020  SeaLINK+422
+	a02a  SeaLINK+8 (Port 1+2)
+	a02b  SeaLINK+8 (Port 3+4)
+	a02c  SeaLINK+8 (Port 5+6)
+	a02d  SeaLINK+8 (Port 7+8)
 0c53  ViewPLUS, Inc.
 0c54  Glory, Ltd
 0c55  Spectrum Digital, Inc.
@@ -10571,10 +10913,13 @@
 0c65  Eminence Enterprise Co., Ltd
 0c66  Rexon Electronics Corp.
 0c67  Concept Telecom, Ltd
+0c6c  JETI Technische Instrumente GmbH
+	04b2  Specbos 1201
 0c70  MCT Elektronikladen
 	0000  USB08 Development board
 0c72  PEAK System
 	000c  PCAN-USB
+	000d  PCAN Pro
 0c74  Optronic Laboratories Inc.
 	0002  OL 700-30 Goniometer
 0c76  JMTek, LLC.
@@ -10697,6 +11042,7 @@
 	0077  Aureon Dual USB
 	0078  Cinergy T XXS
 	0086  Cinergy Hybrid XE
+	008e  Cinergy HTC XS
 	0097  Cinergy T RC MKII
 	0099  AfaTech 9015 [Cinergy T Stick Dual]
 	00a5  Cinergy Hybrid Stick
@@ -10734,6 +11080,7 @@
 	0023  UB81 802.11bgn
 	0025  802.11b/g/n USB Wireless Network Adapter
 	0026  UB82 802.11abgn
+	0027  Sphairon Homelink 1202 802.11n Wireless Adapter [Atheros AR9170]
 0ce5  Validation Technologies International
 	0003  Matrix
 0ce9  pico Technology
@@ -10756,6 +11103,7 @@
 	1010  3Com 3CRUSBN275 802.11abgn Wireless Adapter [Atheros AR9170]
 	20ff  Virtual CD-ROM
 	3000  AR3011 Bluetooth (no firmware)
+	3002  AR3011 Bluetooth
 	3005  AR3011 Bluetooth
 	7015  TP-Link TL-WN821N v3 802.11n [Atheros AR7010+AR9287]
 	9170  AR9170 802.11n
@@ -10866,7 +11214,7 @@
 0d61  Meilu Electronics (Shenzhen) Co., Ltd
 0d62  Darfon Electronics Corp.
 	0003  Smartcard Reader
-	0004  Filter Driver
+	0004  Keyboard
 	001c  Benq X120 Internet Keyboard Pro
 	0306  M530 Mouse
 	0800  Magic Wheel
@@ -10953,6 +11301,7 @@
 	0107  CM108 Audio Controller
 	010f  CM108 Audio Controller
 	0115  CM108 Audio Controller
+	013c  CM108 Audio Controller
 	0201  CM6501
 	5000  Mass Storage Controller
 	5200  Mass Storage Controller(0D8C,5200)
@@ -11017,6 +11366,7 @@
 	0101  SBIG Astronomy Camera (with firmware)
 0d98  Mars Semiconductor Corp.
 	0300  Avaya Wireless Card
+	1007  Discovery Kids Digital Camera
 0d99  Trazer Technologies, Inc.
 0d9a  RTX Telecom AS
 	0001  Bluetooth Device
@@ -11195,10 +11545,15 @@
 	0045  WL-353 Wireless USB Adapter 150N Nano
 	0047  WL-352v1 Wireless USB Adapter 300N 002
 	0048  WL-349v1 Wireless Adapter 150N 002 [Ralink RT3070]
+	0049  WL-356 Wireless Adapter 300N
 	004a  WL-358v1 Wireless Micro USB Adapter 300N X3 002
 	004b  WL-349v3 Wireless Micro Adapter 150N X1 [Realtek RTL8192SU]
 	004c  WL-352 802.11n Adapter [Realtek RTL8191SU]
 	0050  WL-349v4 Wireless Micro Adapter 150N X1 [Ralink RT3370]
+	0056  LN-031 10/100/1000 Ethernet Adapter
+	005d  WLA-2000 v1.001 WLAN [RTL8191SU]
+	0060  WLA-4000 802.11bgn [Ralink RT3072]
+	0062  WLA-5000 802.11abgn [Ralink RT3572]
 	061c  LN-028 Network USB 2.0 Adapter
 	21f4  44 St Bluetooth Device
 	2200  Sitecom bluetooth2.0 class 2 dongle CN-512
@@ -11287,6 +11642,7 @@
 0e3b  Mansella, Ltd
 0e41  Line6, Inc.
 	4147  TonePort GX
+	4156  POD HD Desktop
 	4250  BassPODxt
 	4252  BassPODxt Pro
 	4642  BassPODxt Live
@@ -11303,6 +11659,8 @@
 	1097  Gamester Controller
 	2390  Games Jtech Controller
 	7288  funkey reader
+0e50  TechnoData Interware
+	0002  Matrixlock Dongle (HID)
 0e55  Speed Dragon Multimedia, Ltd
 	110a  Tanic S110-SG1 + ISSC IS1002N [Slow Infra-Red (SIR) & Bluetooth 1.2 (Class 2) Adapter]
 	110b  MS3303H USB-to-Serial Bridge
@@ -11323,6 +11681,7 @@
 	0009  HWUN2 Hi-Gain Wireless-150N Adapter w/ Upgradable Antenna [Ralink RT2770]
 	000b  HWDN2 Hi-Gain Wireless-150N Dish Adapter [Ralink RT2770]
 	0013  HWUN3 Hi-Gain Wireless-N Adapter [Ralink RT3070]
+	0015  HWDN2 Rev. E Hi-Gain Wireless-150N Dish Adapter [Realtek RTL8191SU]
 	0017  HAWNU1 Hi-Gain Wireless-150N Network Adapter with Range Amplifier [Ralink RT3070]
 	0018  Wireless-N Network Adapter [Ralink RT2870]
 	400b  UF100 10/100 Network Adapter
@@ -11357,6 +11716,7 @@
 0e8d  MediaTek Inc.
 	0003  MT6227 phone
 	0004  MT6227 phone
+	1836  Samsung SE-S084 Super WriteMaster Slim External DVD writer
 0e8f  GreenAsia Inc.
 	0003  MaxFire Blaze2
 	0012  USB Wireless 2.4GHz Gamepad
@@ -11388,6 +11748,7 @@
 	6803  OTI-6803 Flash Disk
 	6808  OTI-6808 Flash Disk
 	6828  OTI-6828 Flash Disk
+	6858  OTi-6858 serial adapter
 0ea6  Nihon Computer Co., Ltd
 0ea7  MSL Enterprises Corp.
 0ea8  CenDyne, Inc.
@@ -11589,7 +11950,8 @@
 0fb1  Inclose Design, Inc.
 0fb2  Juan-Chern Industrial Co., Ltd
 0fb6  Heber Ltd
-	3fc4  Firefly X10i USB I/O Board
+	3fc3  Firefly X10i I/O Board (with firmware)
+	3fc4  Firefly X10i I/O Board (without firmware)
 0fb8  Wistron Corp.
 	0002  eHome Infrared Receiver
 0fb9  AACom Corp.
@@ -11614,6 +11976,7 @@
 	00d4  C902 [MTP]
 	00d9  C702 Phone
 	0112  W995 Walkman Phone
+	0166  Xperia Mini Pro
 	1010  WMC Modem
 	10af  V640i Phone [PictBridge]
 	10d4  C902 Phone [PictBridge]
@@ -11663,7 +12026,11 @@
 	e12e  X10i Phone
 	e133  Vivaz
 	e14f  Xperia Arc/X12
+	e161  Xperia Ray
 0fcf  Dynastream Innovations, Inc.
+	1003  ANT Development Board
+	1004  ANT2USB
+	1006  ANT Development Board
 0fd0  Tulip Computers B.V.
 0fd1  Giant Electronics Ltd.
 0fd4  Tenovis GmbH & Co., KG
@@ -11703,12 +12070,14 @@
 0fef  MJ Research, Inc.
 0ff6  Core Valley Co., Ltd
 0ff7  CHI SHING Computer Accessories Co., Ltd
+0ffc  Clavia DMI AB
+	0021  Nord Stage 2
 0fff  Aopen, Inc.
 1000  Speed Tech Corp.
 1001  Ritronics Components (S) Pte., Ltd
 1003  Sigma Corp.
 	0003  SD14
-	0100  Sigma SD10
+	0100  SD9/SD10
 1004  LG Electronics, Inc.
 	1fae  U8120 3G Cellphone
 	6000  KU330/KU990/VX4400/VX6000
@@ -11790,6 +12159,7 @@
 1038  Ideazon, Inc.
 	0100  Zboard
 1039  devolo AG
+	0824  1866 802.11bg [Texas Instruments TNETW1450]
 	2140  dsl+ 1100 duo
 103d  Stanton
 	0100  ScratchAmp
@@ -11852,13 +12222,17 @@
 	0704  Passport External HDD
 	070a  My Passport Essential SE
 	071a  My Passport 1TB
+	0740  My Passport 1TB
 	0742  My Passport Essential SE
 	0900  MyBook Essential External HDD
 	0901  MyBook External HDD
 	0903  My Book Premium Edition
 	0910  MyBook Essential External HDD
 	1001  External Hard Disk [Elements]
+	1003  Elements 1000 GB
+	1010  Elements External HDD
 	1021  Elements 2TB
+	1023  Elements SE
 	1103  My Book Studio
 	1104  MyBook Mirror Edition External HDD
 	1123  My Book 3.0
@@ -11914,6 +12288,7 @@
 	3701  Broadband Wireless modem
 	3702  Pantech PX-500
 	3714  PANTECH USB MODEM [UM175]
+	3716  UMW190 Modem
 	3eb4  Packet Service Diagnostic Serial Port (WDM)
 	4101  Packet Service Diagnostic Serial Port (WDM)
 	4102  Packet Service
@@ -11946,6 +12321,8 @@
 106d  San Chieh Manufacturing, Ltd
 106e  ConectL
 106f  Money Controls
+	0009  CT10x Coin Transaction
+	000a  CR10x Coin Recycler
 1076  GCT Semiconductor, Inc.
 	0031  Bluetooth Device
 	0032  Bluetooth Device
@@ -12015,10 +12392,20 @@
 	80a9  CP210x to UART Bridge Controller
 	80ca  ATM2400 Sensor Device
 	813f  tams EasyControl
+	8149  West Mountain Radio Computerized Battery Analyzer
+	814a  West Mountain Radio RIGblaster P&P
+	814b  West Mountain Radio RIGtalk
 	818a  Silicon Labs FM Radio Reference Design
+	81e8  Zephyr BioHarness
 	8460  Sangoma Wanpipe VoiceTime
 	8461  Sangoma U100
-	ea60  CP210x Composite Device
+	8477  Balluff RFID Reader
+	86bc  C8051F34x AudioDelay [AD-340]
+	8789  C8051F34x Extender & EDID MGR [EMX-DVI]
+	ea60  CP210x UART Bridge / myAVR mySmartUSB light
+	ea61  CP210x UART Bridge
+	ea70  CP210x UART Bridge
+	ea80  CP210x UART Bridge
 10c5  Sanei Electric, Inc.
 	819a  FM Radio
 10c6  Intec, Inc.
@@ -12217,6 +12604,7 @@
 	6870  MC8780 Device
 	6871  MC8781 Device
 	6893  MC8777 Device
+	68a3  MC8700 Modem
 	9000  Gobi 2000 Wireless Modem (QDL mode)
 	9001  Gobi 2000 Wireless Modem
 	9002  Gobi 2000 Wireless Modem
@@ -12277,7 +12665,8 @@
 1230  Chipidea-Microelectronica, S.A.
 1233  Denver Electronics
 	5677  FUSB200 mp3 player
-1234  Unknown
+1234  Brain Actuated Technologies
+	0000  Neural Impulse Actuator Prototype 1.0 [NIA]
 	ed02  Emotiv EPOC Developer Headset Wireless Dongle
 1235  Novation EMS
 	0001  ReMOTE Audio/XStation
@@ -12295,6 +12684,8 @@
 124a  AirVast
 	168b  PRISM3 WLAN Adapter
 	4017  PC-Chips 802.11b Adapter
+	4023  WM168g 802.11bg Wireless Adapter [Intersil ISL3886]
+	4025  IOGear GWU513 v2 802.11bg Wireless Adapter [Intersil ISL3887]
 124b  Nyko (Honey Bee)
 	4d01  Airflo EX Joystick
 124c  MXI - Memory Experts International, Inc.
@@ -12303,6 +12694,7 @@
 	0010  Alta series CCD
 125f  A-DATA Technology Co., Ltd.
 	312a  Superior S102
+	312b  Superior S102 Pro
 	a91a  Portable HDD CH91
 	c81a  Flash drive
 	c93a  4GB Pen Drive
@@ -12391,20 +12783,29 @@
 12d1  Huawei Technologies Co., Ltd.
 	1001  E169/E620/E800 HSDPA Modem
 	1003  E220 HSDPA Modem / E230/E270/E870 HSDPA/HSUPA Modem
+	1004  E220 (bis)
 	1009  U120
 	1010  ETS2252+ CDMA Fixed Wireless Terminal
 	1035  U8120
 	1037  Ideos
 	1038  Ideos (debug mode)
+	1039  Ideos (tethering mode)
 	1406  E1750
 	140b  EC1260 Wireless Data Modem HSD USB Card
-	1446  E1552/E1800 (HSPA modem)
+	1446  E1552/E1800/E173 (HSPA modem)
+	14c9  K3770 3G Modem
+	14d1  K3770 3G Modem (Mass Storage Mode)
+	14f1  Gobi 3000 HSPA+ Modem
 	1501  Pulse
 	1505  E398 LTE/UMTS/GSM Modem/Networkcard
 	1506  E398 LTE/UMTS/GSM Modem/Networkcard
 	150a  E398 LTE/UMTS/GSM Modem/Networkcard
 	1520  K3765 HSPA
 	1521  K4505 HSPA+
+	1805  AT&T Go Phone U2800A phone
+	1c05  E173s 3G broadband stick (modem on)
+	1c0b  E173s 3G broadband stick (modem off)
+	1d50  ET302s TD-SCDMA/TD-HSDPA Mobile Broadband
 	380b  WiMAX USB modem(s)
 12d2  LINE TECH INDUSTRIAL CO., LTD.
 12d6  EMS Dr. Thomas Wuensche
@@ -12426,8 +12827,9 @@
 1307  Transcend Information, Inc.
 	0163  256MB/512MB/1GB Flash Drive
 	0165  2GB/4GB Flash Drive
+	0310  SD/MicroSD CardReader [hama]
 	0330  63-in-1 Multi-Card Reader/Writer
-	0361  CR-75: 51-in-1 Card Reader/Writer
+	0361  CR-75: 51-in-1 Card Reader/Writer [Sakar]
 	1169  TS2GJF210 JetFlash 210 2GB
 	1171  Fingerprint Reader
 1308  Shuttle, Inc.
@@ -12448,12 +12850,12 @@
 	0006  Dimage Z2 Camera
 	0007  Minolta DiMAGE Z2 (PictBridge mode)
 	0008  Dimage X21 Camera
-	000a  Dimage Scan Dual IV
+	000a  Dimage Scan Dual IV AF-3200 (2891)
 	000b  Dimage Z10 Camera
 	000d  Dimage X50 Camera [storage?]
 	000f  Dimage X50 Camera [p2p?]
 	0010  Dimage G600 Camera
-	0012  Dimage Scan Elite5400 2
+	0012  Dimage Scan Elite 5400 II (2892)
 	0013  Dimage X31 Camera
 	0015  Dimage G530 Camera
 	0017  Dimage Z3 Camera
@@ -12485,7 +12887,7 @@
 	0401  G2CCD-S with Sony ICX285 CCD
 	0402  G2CCD2
 	0403  G2/G3CCD-I KAI CCD
-	0404  G2/G3CCD-F KAF CCD
+	0404  G2/G3/G4 CCD-F KAF CCD
 	0410  G1-0400 CCD
 	0411  G1-0800 CCD
 	0412  G1-0300 CCD
@@ -12535,7 +12937,7 @@
 138e  Jungo LTD
 	9000  Raisonance S.A. STM32 ARM evaluation board
 1390  TOMTOM B.V.
-	0001  GO 520 T
+	0001  GO 520 T/GO 630/ONE XL (v9)
 1391  IdealTEK, Inc.
 	1000  URTC-1000
 1395  Sennheiser Communications
@@ -12564,8 +12966,11 @@
 	0023  WUSB54GR
 	0024  WUSBF54G v1.1 802.11bg
 	0026  WUSB54GSC v1 802.11g Adapter [Broadcom 4320 USB]
+	0028  WUSB200 802.11g Adapter [Ralink RT2671]
 	002f  AE1000 v1 802.11n [Ralink RT3572]
 	0031  AM10 v1 802.11n [Ralink RT3072]
+	0039  AE1200 802.11bgn Wireless Adapter [Broadcom BCM43235]
+	003a  AE2500 802.11abgn Wireless Adapter [Broadcom BCM43236]
 	13b1  WUSB200: Wireless-G Business Network Adapter with Rangebooster
 13b2  Alesis
 	0030  Multimix 8
@@ -12635,6 +13040,7 @@
 13e1  Kaibo Wire & Cable (Shenzhen) Co., Ltd.
 13e5  Rane
 	0001  SL-1
+	0003  TTM 57SL
 13e6  TechnoScope Co., Ltd.
 13ea  Hengstler
 	0001  C-56 Thermal Printer
@@ -12656,6 +13062,7 @@
 	1f00  DataTraveler 2.0 4GB Flash Drive / Patriot Xporter 32GB (PEF32GUSB) Flash Drive
 	2240  microSD card reader
 	3100  2/4 GB stick
+	3800  Rage XT Flash Drive
 1400  Axxion Group Corp.
 1402  Bowe Bell & Howell
 1403  Sitronix
@@ -12724,6 +13131,8 @@
 	013f  Megapixel Auto Focus Webcam
 	0142  WB-6250X Webcam
 	015a  WB-8300X 2MP Webcam
+	0161  15901 802.11bg Wireless Adapter [Realtek RTL8187L]
+	0167  Widescreen 3MP Webcam
 1460  Tatung Co.
 	9150  eHome Infrared Transceiver
 1461  Staccato Communications
@@ -12738,12 +13147,16 @@
 	e016  eHome Infrared Receiver
 	e017  eHome Infrared Receiver
 	e018  eHome Infrared Receiver
+	e02c  Infrared Receiver
 	e03a  eHome Infrared Receiver
 	e03c  eHome Infrared Receiver
 	e03e  Infrared Receiver [IR605A/Q]
 147e  Upek
 	1000  Biometric Touchchip/Touchstrip Fingerprint Sensor
+	1001  TCS5B Fingerprint sensor
 	2016  Biometric Touchchip/Touchstrip Fingerprint Sensor
+	3000  TCS1C EIM/Cypress Fingerprint sensor
+	3001  TCS1C EIM/STM32 Fingerprint sensor
 147f  Hama GmbH & Co., KG
 1482  Vaillant
 	1005  VRD PC-Interface
@@ -12766,11 +13179,14 @@
 	3070  RT2870/RT3070 Wireless Adapter
 	3071  RT3071 Wireless Adapter
 	3072  RT3072 Wireless Adapter
+	3370  RT3370 Wireless Adapter
 	3572  RT3572 Wireless Adapter
 	3573  TEW-684UB
 	5370  RT5370 Wireless Adapter
 	9020  RT2500USB Wireless Adapter
 	9021  RT2501USB Wireless Adapter
+1491  Futronic Technology Co. Ltd.
+	0020  FS81 Fingerprint Scanner Module
 1497  Panstrong Company Ltd.
 1498  Microtek International Inc.
 	a090  DVB-T Tuner
@@ -12788,15 +13204,17 @@
 14af  ATP Electronics Inc.
 14b0  StarTech.com Ltd.
 14b2  Ralink Technology, Corp.
-	3a95  Toshiba WUS-G06G-JT 802.11bg Wireless Adapter
+	3a93  Topcom 802.11bg Wireless Adapter [Atheros AR5523]
+	3a95  Toshiba WUS-G06G-JT 802.11bg Wireless Adapter [Atheros AR5523]
 	3a98  Airlink101 AWLL4130 802.11bg Wireless Adapter [Atheros AR5523]
-	3c02  rt2500 802.11g WLAN
+	3c02  Conceptronic C54RU v2 802.11bg Wireless Adapter [Ralink RT2571]
 	3c05  rt2570 802.11g WLAN
 	3c06  Conceptronic C300RU v1 802.11bgn Wireless Adapter [Ralink RT2870]
 	3c07  802.11n adapter
 	3c09  802.11n adapter
-	3c22  Conceptronic C54RU 802.11bg Wireless Adapter [Ralink RT73]
+	3c22  Conceptronic C54RU v3 802.11bg Wireless Adapter [Ralink RT2571W]
 	3c23  Airlink101 AWLL6080 802.11bgn Wireless Adapter [Ralink RT2870]
+	3c24  NEC NP01LM 802.11abg Wireless Adapter [Ralink RT2571W]
 	3c25  DrayTek Vigor N61 802.11bgn Wireless Adapter [Ralink RT2870]
 	3c27  Airlink101 AWLL6070 802.11bgn Wireless Adapter [Ralink RT2770]
 	3c28  Conceptronic C300RU v2 802.11bgn Wireless Adapter [Ralink RT2770]
@@ -12818,6 +13236,7 @@
 	8125  SD MMC Reader
 14d8  JAMER INDUSTRIES CO., LTD.
 14dd  Raritan Computer, Inc.
+	1007  D2CIM-VUSB KVM connector
 14e1  Dialogue Technology Corp.
 	5000  PenMount 5000 Touch Controller
 14e5  SAIN Information & Communications Co., Ltd.
@@ -12872,6 +13291,7 @@
 	0001  RZ01-020300 Optical Mouse [Diamondback]
 	0003  Krait Mouse
 	0007  DeathAdder Mouse
+	0016  DeathAdder Mouse
 	0017  Imperator Mouse
 	001c  RZ01-0036 Optical Gaming Mouse [Abyssus]
 	0101  Copperhead Mouse
@@ -12898,7 +13318,7 @@
 1570  ALLTOP TECHNOLOGY CO., LTD.
 157b  Ketron SRL
 157e  TRENDnet
-	3006  TEW-444UB EU
+	3006  TEW-444UB EU [TRENDnet]
 	3007  TEW-444UB EU (no firmware)
 	300a  TEW-429UB 802.11bg
 	300b  TEW-429UB 802.11bg
@@ -12974,6 +13394,7 @@
 	0a37  Mouse
 	0a41  MI-2540D [Optical mouse]
 	0a4c  USB+PS/2 Optical Mouse
+	0a4d  Optical Mouse
 15dc  Hynix Semiconductor Inc.
 15e0  Seong Ji Industrial Co., Ltd.
 15e1  RSA
@@ -13249,6 +13670,22 @@
 	03f1  free for internal lab use 1009
 	0477  Teensy Rebootor
 	0478  Teensy Halfkay Bootloader
+	0479  Teensy Debug
+	047a  Teensy Serial
+	047b  Teensy Serial+Debug
+	047c  Teensy Keyboard
+	047d  Teensy Keyboard+Debug
+	047e  Teensy Mouse
+	047f  Teensy Mouse+Debug
+	0480  Teensy RawHID
+	0481  Teensy RawHID+Debug
+	0482  Teensyduino Keyboard+Mouse+Joystick
+	0483  Teensyduino Serial
+	0484  Teensyduino Disk
+	0485  Teensyduino MIDI
+	0486  Teensyduino RawHID
+	0487  Teensyduino Serial+Keyboard+Mouse+Joystick
+	0488  Teensyduino Flight Sim Controls
 	05dc  shared ID for use with libusb
 	05dd  BlackcatUSB2
 	05e1  CDC-ACM class devices (modems)
@@ -13295,6 +13732,7 @@
 	000b  VM
 16df  King Billion Electronics Co., Ltd.
 16f0  GN ReSound A/S
+	0001  Speedlink Programming Interface
 	0003  Airlink Wireless Programming Interface
 16f5  Futurelogic Inc.
 1706  BlueView Technologies, Inc.
@@ -13345,8 +13783,10 @@
 	9801  EUB9801 802.11abgn Wireless Adapter [Ralink RT3572]
 1743  General Atomics
 174c  ASMedia Technology Inc.
+	5106  Transcend StoreJet 25M3
 174f  Syntek
 	1105  SM-MS/Pro-MMC-XD Card Reader
+	110b  HP Webcam
 	1403  Integrated Webcam
 	1404  USB Camera device, 1.3 MPixel Web Cam
 	5212  USB 2.0 UVC PC Camera
@@ -13380,6 +13820,8 @@
 	083e  MetaGeek Wi-Spy
 	083f  MetaGeek Wi-Spy 2.4x
 	0938  Iguanaworks USB IR Transceiver
+	0c30  Telldus TellStick
+	0c31  Telldus TellStick Duo
 	0c9f  USBtiny
 1782  Spreadtrum Communications Inc.
 1784  TopSeed Technology Corp.
@@ -13416,9 +13858,10 @@
 	0004  Linux-USB Midi Gadget
 17ba  SAURIS GmbH
 	0001  SAU510-USB [no firmware]
-	0510  SAU510-USB and SAU510-USP plus JTAG Emulators
+	0510  SAU510-USB and SAU510-USB plus JTAG Emulators
 	0511  SAU510-USB Iso Plus JTAG Emulator
 	0520  SAU510-USB Nano JTAG Emulator
+	1511  Onboard Emulator on SAUModule development kit
 17c3  Singim International Corp.
 17cc  Native Instruments
 	041c  Audio 2 DJ
@@ -13438,9 +13881,15 @@
 17cf  Hip Hing Cable & Plug Mfy. Ltd.
 17d0  Sanford L.P.
 17d3  Korea Techtron Co., Ltd.
-17e9  Newnham Research
+17e9  DisplayLink
 	0051  USB VGA Adaptor
+	0377  Plugable UD-160-A (M)
+	0378  Plugable UGA-2K-A
+	0379  Plugable UGA-125
+	037a  Plugable UGA-165
+	037b  Plugable USB-VGA-165
 	037c  Plugable DC-125
+	037d  Plugable USB2-HDMI-165
 17eb  Cornice, Inc.
 17ef  Lenovo
 	1003  Integrated Smart Card Reader
@@ -13460,6 +13909,7 @@
 	4814  Integrated Webcam [R5U877]
 	4815  Integrated Webcam [R5U877]
 	481c  Integrated Webcam
+	481d  Integrated Webcam
 	6007  Smartcard Keyboard
 	6009  ThinkPad Keyboard with TrackPoint
 	6014  Mini Wireless Keyboard N5901
@@ -13515,6 +13965,7 @@
 	0216  External Hard Drive
 	0218  External Hard Drive
 	0227  Pocket Hard Drive
+	0237  Portable Harddrive (500 GB)
 18b1  Petalynx
 	0037  Maxter Remote Control
 18b4  e3C Technologies
@@ -13539,6 +13990,7 @@
 	4e21  Nexus S
 	4e22  Nexus S (debug)
 	4e24  Nexus S (tether)
+	7102  Toshiba Thrive tablet
 18d5  Starline International Group Limited
 18d9  Kaba
 	01a0  B-Net 91 07
@@ -13618,7 +14070,7 @@
 	a253  Model 2253 Audio/Video Codec Card (cold)
 1949  Lab126
 	0002  Amazon Kindle
-	0004  Amazon Kindle 3
+	0004  Amazon Kindle 3/4
 1951  Hyperstone AG
 1953  Ironkey Inc.
 1954  Radiient Technologies
@@ -13676,6 +14128,7 @@
 	0083  MF190
 	0103  MF112
 	0172  AX226 WIMAX MODEM (After Modeswitch)
+	1203  MF691 [ T-Mobile webConnect Rocket 2.0]
 	2000  MF627/MF628/MF628+/MF636+ HSDPA/HSUPA
 	fff2  Gobi Wireless Modem (QDL mode)
 	fff3  Gobi Wireless Modem
@@ -13693,13 +14146,16 @@
 1a0a  USB-IF non-workshop
 	badd  USB OTG Compliance test device
 1a12  KES Co., Ltd.
+1a1d  Veho
+	0407  Mimi WiFi speakers
 1a25  Amphenol East Asia Ltd.
 1a2a  Seagate Branded Solutions
 1a32  Quanta Microsystems, Inc.
 	0304  802.11n Wireless LAN Card
 1a36  Biwin Technology Ltd.
-1a40  TERMINUS TECHNOLOGY INC.
-	0101  USB-2.0 4-Port HUB
+1a40  Terminus Technology Inc.
+	0101  4-Port HUB
+	0201  FE 2.1 7-port Hub
 1a41  Action Electronics Co., Ltd.
 1a44  VASCO Data Security International
 	0001  Digipass 905 SmartCard Reader
@@ -13716,13 +14172,16 @@
 	0068  VerticalMouse 3
 	0168  VerticalMouse 3 Wireless
 	0191  VerticalMouse 4
+1a81  Holtek Semiconductor, Inc.
+	2203  Laser Gaming mouse
+	2204  Optical Mouse
 1a86  QinHeng Electronics
 	5523  CH341 in serial mode, usb to serial port converter
 	5584  CH341 in parallel mode, usb to printer port converter
 	7523  HL-340 USB-Serial adapter
 	752d  CH345 MIDI adapter
 	7584  CH340S
-	e008  HID-based USB-serial converter, full-speed, similar to HE2325U
+	e008  HID-based serial adapater
 1a89  Dynalith Systems Co., Ltd.
 1a8b  SGS Taiwan Ltd.
 1a8d  BandRich, Inc.
@@ -13733,8 +14192,13 @@
 1aa4  Data Drive Thru, Inc.
 1aa5  UBeacon Technologies, Inc.
 1aa6  eFortune Technology Corp.
+1ab1  Rigol Technologies
+	0588  DS1000 SERIES
 1acb  Salcomp Plc
 1ad1  Desay Wire Co., Ltd.
+1ad4  APS
+	0002  KM290-HRS
+1adb  SEL C662 Serial Cable
 1ae4  ic-design Reinhard Gottinger GmbH
 1ae7  X-TENSIONS
 	0381  VS-DVB-T 380U (af9015 based)
@@ -13818,6 +14282,12 @@
 	810b  ME-8100B
 	820a  ME-8200A
 	820b  ME-8200B
+1b0e  BLUTRONICS S.r.l.
+	1078  BLUDRIVE II CCID
+1b1c  Corsair
+	0890  Flash Padlock
+	0a00  SP2500 Speakers
+	1a90  Flash Voyager GT
 1b20  MStar Semiconductor, Inc.
 1b22  WiLinx Corp.
 1b26  Cellex Power Products, Inc.
@@ -13863,6 +14333,9 @@
 	1000  xHC1 Controller
 1b75  Ovislink Corp.
 	3072  AirLive WN-360USB adapter
+	8187  AirLive WL-1600USB 802.11g Adapter [Realtek RTL8187L]
+	9170  AirLive X.USB 802.11abgn [Atheros AR9170+AR9104]
+	a200  AirLive WN-200USB wireless 11b/g/n dongle
 1b76  Legend Silicon Corp.
 1b80  Afatech
 	c810  MC810 [af9015]
@@ -13871,9 +14344,11 @@
 	d397  DVB-T receiver [RTL2832U]
 	d398  DVB-T receiver [RTL2832U]
 	d700  FM Radio SnapMusic Mobile 700 (FM700)
+	e297  Conceptronic DVB-T CTVDIGRCU V3.0
 	e383  DVB-T UB383-T [af9015]
 	e385  DVB-T UB385-T [af9015]
 	e386  DVB-T UB385-T [af9015]
+	e399  DVB-T KWorld PlusTV 399U [af9015]
 	e39a  DVB-T395U [af9015]
 	e39b  DVB-T395U [af9015]
 	e409  IT9137FN Dual DVB-T [KWorld UB499-2T]
@@ -13904,9 +14379,12 @@
 1bce  Contac Cable Industrial Limited
 1bcf  Sunplus Innovation Technology Inc.
 	0007  Optical Mouse
+	053a  Targa Silvercrest OMC807-C optische Funkmaus
 	05c5  SPRF2413A [2.4GHz Wireless Keyboard/Mouse Receiver]
 	05cf  Micro keyboard & mouse receiver
+	0c31  SPIF30x Serial-ATA bridge
 1bd0  Hangzhou Riyue Electronic Co., Ltd.
+1bd5  BG Systems, Inc.
 1bde  P-TWO INDUSTRIES, INC.
 1bef  Shenzhen Tongyuan Network-Communication Cables Co., Ltd
 1bf0  RealVision Inc.
@@ -13940,11 +14418,19 @@
 1c37  Authorizer Technologies, Inc.
 1c3d  NONIN MEDICAL INC.
 1c3e  Wep Peripherals
+1c40  EZPrototypes
+	0533  TiltStick
+	0534  i2c-tiny-usb interface
+	0535  glcd2usb interface
+	0536  Swiss ColorPAL
 1c49  Cherng Weei Technology Corp.
 1c4f  SiGma Micro
+	0002  Keyboard TRACER Gamma Ivory
 	0003  HID controller
+	000e  Genius KB-120 Keyboard
 	3000  Micro USB Web Camera
 1c6b  Philips & Lite-ON Digital Solutions Corporation
+	a222  DVD Writer Slimtype eTAU108
 1c6c  Skydigital Inc.
 1c73  AMT
 	861f  Anysee E30 USB 2.0 DVB-T Receiver
@@ -13952,12 +14438,17 @@
 1c78  Datascope Corp.
 1c79  Unigen Corporation
 1c7a  LighTuning Technology Inc.
+	0801  Fingerprint Reader
 1c7b  LUXSHARE PRECISION INDUSTRY (SHENZHEN) CO., LTD.
 1c87  2N TELEKOMUNIKACE a.s.
 1c88  Somagic, Inc.
+	0007  SMI Grabber (EasyCAP DC60+ clone) (no firmware) [SMI-2021CBE]
+	003c  SMI Grabber (EasyCAP DC60+ clone) [SMI-2021CBE]
 1c89  HONGKONG WEIDIDA ELECTRON LIMITED
 1c8e  ASTRON INTERNATIONAL CORP.
 1c98  ALPINE ELECTRONICS, INC.
+1c9e  OMEGA TECHNOLOGY
+	6061  WL-72B 3.5G MODEM
 1ca0  ACCARIO Inc.
 1cac  Kinstone
 	a332  C8 Webcam
@@ -13978,6 +14469,8 @@
 1ce1  Amphenol KAE
 1cfc  ANDES TECHNOLOGY CORPORATION
 1cfd  Flextronics Digital Design Japan, LTD.
+1d03  iCON
+	0028  iCreativ MIDI Controller
 1d07  Solid-Motion
 1d08  NINGBO HENTEK DRAGON ELECTRONICS CO., LTD.
 1d09  TechFaith Wireless Technology Limited
@@ -14001,6 +14494,15 @@
 	000e  Ralink RT3070 802.11b/g/n Wireless Lan USB Device
 1d50  OpenMoko, Inc.
 	5119  GTA01/GTA02 U-Boot Bootloader
+1d57  Xenta
+	0005  Wireless Receiver (Keyboard and Mouse)
+	0006  Wireless Receiver (RC Laser Pointer)
+	000c  Optical Mouse
+	2400  Wireless Mouse Receiver
+	32da  2.4GHz Receiver (Keyboard and Mouse)
+	83d0  Click-mouse!
+	ac01  Wireless Receiver (Keyboard and Mouse)
+	af01  AUVIO Universal Remote Receiver for PlayStation 3
 1d5b  Smartronix, Inc.
 1d6b  Linux Foundation
 	0001  1.1 root hub
@@ -14012,11 +14514,15 @@
 	0103  NCM (Ethernet) Gadget
 	0104  Multifunction Composite Gadget
 	0105  FunctionFS Gadget
-	0106  Composite Gadget: ACM + Mass Storage
 	0200  Qemu Audio Device
+1de1  Actions Microelectronics Co.
+	1101  Generic Display Device (Mass storage mode)
+	c101  Generic Display Device
 1e0e  Qualcomm / Option
 1e10  Point Grey Research, Inc.
 	2004  Sony 1.3MP 1/3" ICX445 IIDC video camera [Chameleon]
+1e17  Mirion Technologies Dosimetry Services Division
+	0001  instadose dosimeter
 1e1d  Lumension Security
 	0165  Secure Pen drive
 1e1f  INVIA
@@ -14035,6 +14541,8 @@
 	040a  LPC2378 [Robotino Bootloader]
 	0501  CP2102 [CMSP]
 	0601  CMMP-AS
+1e3d  Chipsbank Microelectronics Co., Ltd
+	4082  CBM4082 SD Card Reader
 1e41  Cleverscope
 	0001  CS328A PC Oscilloscope
 1e54  TypeMatrix
@@ -14044,22 +14552,37 @@
 1e71  NZXT
 	0001  Avatar Optical Mouse
 1e74  Coby Electronics Corporation
+	2211  MP300
+	2659  Coby 4GB Go Video MP3 Player [MP620-4G]
+	4641  A8705 MP3/Video Player
 	6511  MP705-8G MP3 player
 	6512  MP705-4G
 1e7d  ROCCAT
 	2c24  Pyra Mouse (wired)
 	2ced  Kone Mouse
+	2cf6  Pyra Mouse (wireless)
+	2d50  Kova+ Mouse
 	2d51  Kone+ Mouse
 	30d4  Arvo Keyboard
 1ebb  NuCORE Technology, Inc.
 1eda  AirTies Wireless Networks
-	2310  802.11n USB Wireless LAN Card
+	2012  Air2210 54 Mbps Wireless Adapter
+	2210  Air2210 54 Mbps Wireless Adapter
+	2310  Air2310 150 Mbps Wireless Adapter
+	2410  Air2410 300 Mbps Wireless Adapter
+1edb  Blackmagic design
+	bd3b  Intensity Shuttle
 1ee8  ONDA COMMUNICATION S.p.a.
 	0014  MT833UP
 1ef6  EADS Deutschland GmbH
+	5064  FDR Interface
+	5648  RIU CSMU/BSD
+	564a  Cassidian RIU CSMU/BSD Simulator
 1f28  Cal-Comp
 	0020  CDMA USB Modem A600
 	0021  CD INSTALLER USB Device
+1f44  The Neat Company
+	0001  NM-1000 scanner
 1f4d  G-Tek Electronics Group
 	b803  Lifeview LV5TDLX DVB-T [RTL2832U]
 1f82  TANDBERG
@@ -14067,9 +14590,14 @@
 1f84  Alere, Inc.
 1f87  Stantum
 	0002  Multi-touch HID Controller
+1f9b  Ubiquiti Networks, Inc.
+	0241  AirView2-EXT
 1fbd  Delphin Technology AG
 	0001  Expert Key - Data aquisition system
 1fc9  NXP Semiconductors
+	010b  PR533
+1fde  ILX Lightwave Corporation
+	0001  UART Bridge
 1fe7  Vertex Wireless Co., Ltd.
 	1000  VW100 series CDMA EV-DO Rev.A modem
 2001  D-Link Corp.
@@ -14079,11 +14607,15 @@
 	200c  10/100 Ethernet
 	3200  DWL-120 802.11b Wireless Adapter(rev.E1) [Atmel at76c503a]
 	3301  DWA-130 802.11n Wireless N Adapter(rev.C1) [Realtek RTL8192U]
+	3306  DWL-G122 Wireless Adapter(rev.F1) [Realtek RTL8188SU]
+	3308  DWA-121 802.11n Wireless N 150 Pico Adapter [Realtek RTL8188CUS]
+	3309  DWA-135 802.11n Wireless N Adapter(rev.A1) [Realtek RTL8192CU]
+	330a  DWA-133 802.11n Wireless N Adapter [Realtek RTL8192CU]
 	3500  Elitegroup Computer Systems WLAN card WL-162
 	3700  DWL-122 802.11b [Intersil Prism 3]
 	3701  DWL-G120 Spinnaker 802.11g [Intersil ISL3886]
 	3702  DWL-120 802.11b Wireless Adapter(rev.F) [Intersil ISL3871]
-	3703  AirPlus G DWL-G122 Wireless Adapter [Intersil ISL3886]
+	3703  AirPlus G DWL-G122 Wireless Adapter(rev.A1) [Intersil ISL3880]
 	3704  AirPlus G DWL-G122 Wireless Adapter(rev.A2) [Intersil ISL3887]
 	3705  AirPlus G DWL-G120 Wireless Adapter(rev.C) [Intersil ISL3887]
 	3761  IEEE 802.11g USB2.0 Wireless Network Adapter-PN
@@ -14099,10 +14631,11 @@
 	3a83  predator Bootloader Download
 	3b00  AirPlus DWL-120+ Wireless Adapter [Texas Instruments ACX100USB]
 	3b01  WLAN Boot Device
-	3c00  AirPlus G DWL-G122 Wireless Adapter(rev.B1) [Ralink RT2500USB]
+	3c00  AirPlus G DWL-G122 Wireless Adapter(rev.B1) [Ralink RT2571]
 	3c01  AirPlus AG DWL-AG122 Wireless Adapter
 	3c02  AirPlus G DWL-G122 Wireless Adapter
 	3c05  DUB-E100 Fast Ethernet [asix]
+	3c19  DWA-125 Wireless N 150 Adapter(rev.A3) [Ralink RT5370]
 	4000  DSB-650C Ethernet [klsi]
 	4001  DSB-650TX Ethernet [pegasus]
 	4002  DSB-650TX Ethernet [pegasus]
@@ -14124,12 +14657,15 @@
 	f116  Formosa 2
 	f117  Formosa 3
 	f118  Formosa 4
-2013  Unknown (Pinnacle?)
+2002  DAP Technologies
+2013  PCTV Systems
 	0245  PCTV 73ESE
 	0246  PCTV 74E
 	0248  PCTV 282E
+	024f  nanoStick T2 290e
 2019  PLANEX
 	3220  GW-US11S WLAN [Atmel AT76C503A]
+	4903  GW-USFang300 802.11abgn Wireless Adapter [Realtek RTL8192DU]
 	5303  GW-US54GXS 802.11bg
 	5304  GWUS300 802.11n
 	ab01  GW-US54HP
@@ -14137,14 +14673,16 @@
 	ab25  GW-USMini2N 802.11n Wireless Adapter [Ralink RT2870]
 	ab28  GW-USNano
 	ab29  GW-USMicro300
+	ab2a  GW-USNano2 802.11n Wireless Adapter [Realtek RTL8188CUS]
+	ab2c  GW-USDual300 802.11abgn Wireless Adapter [Realtek RTL8192DU]
 	ab50  GW-US54Mini2
 	c002  GW-US54SG
 	c007  GW-US54GZL
 	ed02  GW-USMM
-	ed06  802.11n Wireless LAN Adapter
+	ed06  GW-US300MiniW 802.11bgn Wireless Adapter
 	ed10  GW-US300Mini2
 	ed14  GW-USMicroN
-	ed17  GW-USValue-EZ
+	ed17  GW-USValue-EZ 802.11n Wireless Adapter [Realtek RTL8188CUS]
 2040  Hauppauge
 	0c80  Windham
 	0c90  Windham
@@ -14160,6 +14698,7 @@
 	2400  WinTV PVR USB2 (Model 24019)
 	4700  WinTV Nova-S-USB2
 	4902  HD PVR
+	4903  HS PVR
 	4982  HD PVR
 	5500  Windham
 	5510  Windham
@@ -14190,12 +14729,18 @@
 2080  Barnes & Noble
 	0001  nook
 	0002  NOOKcolor
+	0003  NOOK Simple Touch
+	0004  NOOK Tablet
 2087  Cando
 	0a01  Multi Touch Panel
 	0a02  Multi Touch Panel
 	0b03  Multi Touch Panel
 20a0  Clay Logic
+	4123  IKALOGIC SCANALOGIC 2
 	414a  MDE SPI Interface
+	415a  OpenPilot
+	415b  CopterControl
+	415c  PipXtreme
 20b1  XMOS Ltd
 	10ad  XUSB Loader
 	f7d1  XTAG2 - JTAG Adapter
@@ -14210,6 +14755,8 @@
 	cb72  ben-wpan, cntr
 20df  Simtec Electronics
 	0001  Entropy Key [UDEKEY01]
+20f4  TRENDnet
+	648b  TEW-648UBM 802.11n 150Mbps Micro Wireless N Adapter [Realtek RTL8188CUS]
 2101  ActionStar
 	0201  SIIG 4-to-2 Printer Switch
 2162  Creative (?)
@@ -14375,6 +14922,15 @@
 	061e  PCTV Deluxe (PAL) Device
 2318  Shining Technologies, Inc. [hex]
 	0011  CitiDISK Jr. IDE Enclosure
+2341  Arduino SA
+	0001  Uno (CDC ACM)
+	0010  Mega 2560 (CDC ACM)
+	003b  Serial Adapter (CDC ACM)
+	003f  Mega ADK (CDC ACM)
+	0042  Mega 2560 R3 (CDC ACM)
+	0043  Uno R3 (CDC ACM)
+	0044  Mega ADK R3 (CDC ACM)
+	0045  Serial R3 (CDC ACM)
 2373  Pumatronix Ltda
 	0001  5 MegaPixel Digital Still Camera [DSC5M]
 2375  Digit@lway, Inc.
@@ -14386,6 +14942,16 @@
 2632  TwinMOS
 	3209  7-in-1 Card Reader
 2650  Electronics For Imaging, Inc. [hex]
+2659  Sundtek
+	1101  TNT DVB-T/DAB/DAB+/FM
+	1201  FM Transmitter/Receiver
+	1202  MediaTV Analog/FM/DVB-T
+	1203  MediaTV Analog/FM/DVB-T MiniPCIe
+	1204  MediaTV Analog/FM/ATSC
+	1205  SkyTV Ultimate V
+	1206  MediaTV DVB-T MiniPCIe
+	1207  Sundtek HD Capture
+	1208  Sundtek SkyTV Ultimate III
 2730  Citizen
 	200f  CT-S310 Label printer
 2735  DigitalWay
