Description: Dlopen libsndio.so.2 instead of libsndio.so.
Author: Andres Mejia <amejia@debian.org>
Forwarded: http://opensource.creative.com/pipermail/openal-devel/2012-May/005900.html

--- a/Alc/backends/sndio.c
+++ b/Alc/backends/sndio.c
@@ -79,7 +79,7 @@
 #ifdef HAVE_DYNLOAD
     if(!sndio_handle)
     {
-        sndio_handle = LoadLib("libsndio.so");
+        sndio_handle = LoadLib("libsndio.so.2");
         if(!sndio_handle)
             return ALC_FALSE;
 
