Index: pygobject-2.21.3/gobject/pygobject.c
===================================================================
--- pygobject-2.21.3.orig/gobject/pygobject.c	2010-08-16 18:53:01.622365078 +0200
+++ pygobject-2.21.3/gobject/pygobject.c	2010-08-16 18:53:09.678366668 +0200
@@ -176,9 +176,6 @@ pygobject_register_sinkfunc(GType type,
 {
     SinkFunc sf;
 
-    g_message ("pygobject_register_sinkfunc is deprecated (%s)",
-               g_type_name(type));
-
 #if 0
     g_return_if_fail(G_TYPE_IS_OBJECT(type));
 #endif
