Index: src/ftcommon.i
===================================================================
--- src/ftcommon.i.orig	2005-05-26 22:49:41.000000000 -0700
+++ src/ftcommon.i	2006-03-05 11:54:24.000000000 -0800
@@ -21,9 +21,6 @@
 #include FT_BITMAP_H
 #include FT_SYNTHESIS_H
 
-  /* the following header shouldn't be used in normal programs */
-#include FT_INTERNAL_DEBUG_H
-
 #include "common.h"
 
 #include <stdio.h>
Index: src/ftdump.c
===================================================================
--- src/ftdump.c.orig	2004-05-10 13:54:54.000000000 -0700
+++ src/ftdump.c	2006-03-05 11:54:24.000000000 -0800
@@ -13,7 +13,8 @@
 #include FT_SFNT_NAMES_H
 #include FT_TRUETYPE_IDS_H
 
-  /* the following header shouldn't be used in normal programs */
+  /* the following headers shouldn't be used in normal programs */
+#include "freetype/internal/internal.h"
 #include FT_INTERNAL_DEBUG_H
 
   /* showing driver name */
Index: src/ftvalid.c
===================================================================
--- src/ftvalid.c.orig	2005-05-23 13:42:39.000000000 -0700
+++ src/ftvalid.c	2006-03-05 11:54:24.000000000 -0800
@@ -17,6 +17,8 @@
 #include <ft2build.h>
 
 #include FT_FREETYPE_H
+/* XXX: these are internal headers and should never be used */
+#include "freetype/internal/internal.h"
 #include FT_INTERNAL_DEBUG_H
 #include FT_INTERNAL_VALIDATE_H
 #include FT_TRUETYPE_TABLES_H
