Index: ocaml-gstreamer/src/OCamlMakefile
===================================================================
--- ocaml-gstreamer.orig/src/OCamlMakefile	2011-10-16 18:44:32.604330145 +0200
+++ ocaml-gstreamer/src/OCamlMakefile	2011-10-16 18:45:41.507401419 +0200
@@ -601,9 +601,9 @@
 
 ifneq ($(strip $(OBJ_LINK)),)
   ifdef CREATE_LIB
-    OBJS_LIBS := -cclib -l$(CLIB_BASE) $(CLIBS_OPTS) $(MAYBE_IDL)
+    OBJS_LIBS := $(MAYBE_IDL) -cclib -l$(CLIB_BASE) $(CLIBS_OPTS)
   else
-    OBJS_LIBS := $(OBJ_LINK) $(CLIBS_OPTS) $(MAYBE_IDL)
+    OBJS_LIBS := $(MAYBE_IDL) $(OBJ_LINK) $(CLIBS_OPTS)
   endif
 else
   OBJS_LIBS := $(CLIBS_OPTS) $(MAYBE_IDL)
