Description: patch configure.ac to also handle kfreeBSD/Debian
Author: Petr Salinger <Petr.Salinger@seznam.cz>
Origin: http://bugs.debian.org/376570
Bug-Debian: http://bugs.debian.org/376570
Forwarded: not-needed



Index: libsigsegv/configure.ac
===================================================================
--- libsigsegv.orig/configure.ac	2011-07-26 15:13:10.000000000 +0200
+++ libsigsegv/configure.ac	2011-07-26 15:19:18.000000000 +0200
@@ -91,7 +91,7 @@
 dnl List of signals that are sent when an invalid virtual memory address
 dnl is accessed, or when the stack overflows.
 case "$host_os" in
-  sunos4* | freebsd* | dragonfly* | openbsd* | netbsd* | kfreebsd* | knetbsd*)
+  sunos4* | freebsd* | openbsd* | netbsd* | kfreebsd* | k*bsd*)
     CFG_SIGNALS=signals-bsd.h ;;
   hpux*)
     CFG_SIGNALS=signals-hpux.h ;;
@@ -133,7 +133,7 @@
 
 dnl FIXME: Put in some more known values into the third argument.
 SV_TRY_FAULT([POSIX], sv_cv_fault_posix,
-  [*-*-solaris2.[7-9] | i?86-*-linux2.[4-9]* | i?86-*-freebsd[4-9]* | i?86-*-mirbsd1[0-9] | alpha*-dec-osf[4-9]* | *-*-hpux11* | mips-sgi-irix6*],
+  [*-*-solaris2.[7-9] | i?86-*-linux2.[4-9]* | i?86-*-freebsd[4-9]* | i?86-*-kfreebsd[4-9]* | alpha*-dec-osf[4-9]* | *-*-hpux11* | mips-sgi-irix6*],
   [],
   [int sig, siginfo_t *sip, void *ucp],
   [sip->si_addr],
@@ -173,7 +173,7 @@
    action.sa_flags = SA_SIGINFO;])
 
 dnl FIXME: Put in some more known values into the third argument.
-SV_TRY_FAULT([BSD], sv_cv_fault_bsd, [i?86-*-freebsd[4-9]*],
+SV_TRY_FAULT([BSD], sv_cv_fault_bsd, [i?86-*-freebsd[4-9]* | i?86-*-kfreebsd[4-9]*],
   [],
   [int sig, int code, struct sigcontext *scp, void *addr],
   [addr])
@@ -672,7 +672,10 @@
   CFG_STACKVMA=stackvma-mquery.c
 fi
 if test -z "$CFG_STACKVMA" && test $ac_cv_func_mincore = yes; then
-  CFG_STACKVMA=stackvma-mincore.c
+  case "$host_os" in
+    kfreebsd*) ;; dnl mincore() does not work
+    *) CFG_STACKVMA=stackvma-mincore.c ;;
+  esac  
 fi
 if test -n "$CFG_STACKVMA"; then
   AC_DEFINE([HAVE_STACKVMA], [1],
@@ -716,7 +719,7 @@
 dnl FIXME: Put in some more known values into the third argument.
 SV_TRY_LEAVE_HANDLER_LONGJMP([ and sigaltstack],
   sv_cv_leave_handler_longjmp_sigaltstack,
-  [*-*-freebsd*],
+  [*-*-freebsd*|*-*-kfreebsd*],
   [
 #ifndef SS_ONSTACK
 #define SS_ONSTACK SA_ONSTACK
@@ -774,7 +777,7 @@
 dnl FIXME: Put in some more known values into the third argument.
 SV_TRY_LEAVE_HANDLER_SIGLONGJMP([ and sigaltstack],
   sv_cv_leave_handler_siglongjmp_sigaltstack,
-  [*-*-freebsd*],
+  [*-*-freebsd*|*-*-kfreebsd*],
   [
 #ifndef SS_ONSTACK
 #define SS_ONSTACK SA_ONSTACK
