Index: libsigsegv/configure.ac
===================================================================
--- libsigsegv.orig/configure.ac	2011-07-26 15:19:06.000000000 +0200
+++ libsigsegv/configure.ac	2011-07-26 15:19:06.000000000 +0200
@@ -878,6 +878,9 @@
         ;;
     esac
   fi
+  case "$host_cpu" in
+    *s390*) sv_cv_have_stack_overflow_recovery=no ;;
+  esac
   if test $sv_cv_have_stack_overflow_recovery = maybe; then
     if test -n "$CFG_FAULT"; then
       AC_EGREP_CPP([xyzzy], [
