Description: adjust ss.ss_size = SIGSTKSZ if smaller
Author: unknown
Origin: unknown
Forwarded: will do


Index: libsigsegv/src/handler-unix.c
===================================================================
--- libsigsegv.orig/src/handler-unix.c	2011-07-26 15:07:37.000000000 +0200
+++ libsigsegv/src/handler-unix.c	2011-07-26 15:19:06.000000000 +0200
@@ -499,6 +499,9 @@
     ss.ss_sp = extra_stack;
     ss.ss_size = extra_stack_size;
 #endif
+    if (ss.ss_size < SIGSTKSZ) 
+      ss.ss_size = SIGSTKSZ;
+
     ss.ss_flags = 0; /* no SS_DISABLE */
     if (sigaltstack (&ss, (stack_t*)0) < 0)
       return -1;
