From c1791e96182815449781fcda6007b69cd88cf7bf Mon Sep 17 00:00:00 2001
From: Colin Walters <walters@verbum.org>
Date: Mon, 04 Jun 2012 17:54:12 +0000
Subject: client: Fix namespacing of introspection annotations

---
Index: libsocialweb-0.25.20/libsocialweb-client/sw-client-contact-view.c
===================================================================
--- libsocialweb-0.25.20.orig/libsocialweb-client/sw-client-contact-view.c	2011-11-12 18:56:37.000000000 +0800
+++ libsocialweb-0.25.20/libsocialweb-client/sw-client-contact-view.c	2012-09-18 22:12:19.218891009 +0800
@@ -384,7 +384,7 @@
   /**
    * SwClientContactView::contacts-added:
    * @self:
-   * @contacts: (type GLib.List) (element-type Sw.Contact):
+   * @contacts: (type GLib.List) (element-type SocialWebClient.Contact):
    */
   signals[CONTACTS_ADDED_SIGNAL] =
     g_signal_new ("contacts-added",
@@ -401,7 +401,7 @@
   /**
    * SwClientContactView::contacts-removed:
    * @self:
-   * @contacts: (type GLib.List) (element-type Sw.Contact):
+   * @contacts: (type GLib.List) (element-type SocialWebClient.Contact):
    */
   signals[CONTACTS_REMOVED_SIGNAL] =
     g_signal_new ("contacts-removed",
@@ -418,7 +418,7 @@
   /**
    * SwClientContactView::contacts-changed:
    * @self:
-   * @contacts: (type GLib.List) (element-type Sw.Contact):
+   * @contacts: (type GLib.List) (element-type SocialWebClient.Contact):
    */
   signals[CONTACTS_CHANGED_SIGNAL] =
     g_signal_new ("contacts-changed",
Index: libsocialweb-0.25.20/libsocialweb-client/sw-client-item-view.c
===================================================================
--- libsocialweb-0.25.20.orig/libsocialweb-client/sw-client-item-view.c	2011-11-12 18:56:37.000000000 +0800
+++ libsocialweb-0.25.20/libsocialweb-client/sw-client-item-view.c	2012-09-18 22:12:19.222890935 +0800
@@ -389,7 +389,7 @@
   /**
    * SwClientItemView::items-added:
    * @self:
-   * @items: (type GLib.List) (element-type Sw.Item):
+   * @items: (type GLib.List) (element-type SocialWebClient.Item):
    */
   signals[ITEMS_ADDED_SIGNAL] =
     g_signal_new ("items-added",
@@ -406,7 +406,7 @@
   /**
    * SwClientItemView::items-removed:
    * @self:
-   * @items: (type GLib.List) (element-type Sw.Item):
+   * @items: (type GLib.List) (element-type SocialWebClient.Item):
    */
   signals[ITEMS_REMOVED_SIGNAL] =
     g_signal_new ("items-removed",
@@ -423,7 +423,7 @@
   /**
    * SwClientItemView::items-changed:
    * @self:
-   * @items: (type GLib.List) (element-type Sw.Item):
+   * @items: (type GLib.List) (element-type SocialWebClient.Item):
    */
   signals[ITEMS_CHANGED_SIGNAL] =
     g_signal_new ("items-changed",
Index: libsocialweb-0.25.20/libsocialweb-client/sw-contact.c
===================================================================
--- libsocialweb-0.25.20.orig/libsocialweb-client/sw-contact.c	2011-08-10 17:38:28.000000000 +0800
+++ libsocialweb-0.25.20/libsocialweb-client/sw-contact.c	2012-09-18 22:12:19.234890714 +0800
@@ -104,7 +104,6 @@
 
 /**
  * sw_contact_get_value_all: 
- *
  * @contact: :
  * @key: :
  *
