Author: Ondrej Vasik <ovasik@redhat.com>
Description: Fix warning about local builtin used.
Origin: https://fedorahosted.org/xmlto/changeset?reponame=&new=71%40xmlto.in&old=67%40xmlto.in
Bug-Debian: http://bugs.debian.org/652974

--- a/xmlto.in
+++ b/xmlto.in
@@ -95,6 +95,24 @@
   fi
 }
 
+compute_searchpath () {
+  local oldIFS="${IFS}"
+  IFS=":"
+  for asearchpath in "$1"; do
+    # wrangle relative paths into absolute ones so that the user
+    # isn't surprised if he does ``--searchpath .''
+    case "${asearchpath}" in
+     /*) ;;
+     *) asearchpath="${PWD}/${asearchpath}" ;;
+    esac
+    SEARCHPATH="${SEARCHPATH}${XML_SEARCHPATH_SEPARATOR}${asearchpath}"
+    # we only need a colon if more than one path is in the searchpath
+    # and only after the first iteration.
+    XML_SEARCHPATH_SEPARATOR=":"
+  done
+  IFS="${oldIFS}"
+}
+
 # Allow FORMAT_DIR to be over-ridden, so that we can be
 # run from the build directory.
 prefix=@prefix@
@@ -322,22 +340,7 @@
 	shift
 	;;
   --searchpath)
-	local oldIFS="${IFS}"
-	IFS=":"
-	for asearchpath in "$2"; do
-	  # wrangle relative paths into absolute ones so that the user
-	  # isn't surprised if he does ``--searchpath .''
-	  case "${asearchpath}" in
-	   /*) ;;
-	   *) asearchpath="${PWD}/${asearchpath}" ;;
-	  esac
-	  SEARCHPATH="${SEARCHPATH}${XML_SEARCHPATH_SEPARATOR}${asearchpath}"
-	  # we only need a colon if more than one path is in the searchpath
-	  # and only after the first iteration.
-	  XML_SEARCHPATH_SEPARATOR=":"
-	done
-	IFS="${oldIFS}"
-
+	compute_searchpath "$2"
 	# This is the cleanest method I can think of, but requires calls to
 	# xmllint and xsltproc to be run through eval --ohnobinki
 	[ -n "${SEARCHPATH}" ] && SEARCHPATH_FORMATTED="--path \"${SEARCHPATH}\""
