Author: Matthias Andree <mandree@freebsd.org>
Description: Fix the noextensions recognition by getopt.
Origin: https://fedorahosted.org/xmlto/changeset/72

--- a/xmlto.in
+++ b/xmlto.in
@@ -262,7 +262,7 @@
 
 # Process any options
 ARGS=$(${GETOPT} \
-	--longoptions=help,version,extensions,searchpath:,skip-validation,stringparam:,noclean,noautosize,with-fop,with-dblatex \
+	--longoptions=help,version,extensions,searchpath:,skip-validation,stringparam:,noclean,noautosize,noextensions,with-fop,with-dblatex \
 	-n xmlto -- x:m:o:p:v "$@")
 [ $? != 0 ] && { usage; exit 1; }
 eval set -- "$ARGS"
