--- a/aeson.cabal
+++ b/aeson.cabal
@@ -97,6 +97,10 @@
   description: operate in developer mode
   default: False
 
+flag th
+  description: build TH bits
+  default: True
+
 library
   exposed-modules:
     Data.Aeson
@@ -104,7 +108,8 @@
     Data.Aeson.Generic
     Data.Aeson.Parser
     Data.Aeson.Types
-    Data.Aeson.TH
+  if flag(th)
+      exposed-modules: Data.Aeson.TH
 
   other-modules:
     Data.Aeson.Functions
