--- a/aeson.cabal
+++ b/aeson.cabal
@@ -101,6 +101,10 @@
   description: build TH bits
   default: True
 
+flag threaded
+  description: build threaded
+  default: True
+
 library
   exposed-modules:
     Data.Aeson
@@ -152,7 +156,9 @@
   main-is:        Properties.hs
 
   ghc-options:
-    -Wall -threaded -rtsopts
+    -Wall -rtsopts
+  if flag(threaded)
+      ghc-options: -threaded
 
   build-depends:
     QuickCheck,
