Description: fix-readline-detection

--- linphone-3.5.2.orig/m4/readline.m4
+++ linphone-3.5.2/m4/readline.m4
@@ -21,8 +21,14 @@ if test "$readline_prefix" != "none"; th
 	LIBS="$LIBS $READLINE_LIBS"
 	AC_CHECK_HEADERS(readline.h readline/readline.h, readline_h_found=yes)
 	AC_CHECK_HEADERS(history.h readline/history.h)
-	
-	AC_CHECK_LIB(readline, readline, [readline_libs_found=yes],[],[-lncurses])
+
+  for termcap_lib in "" termcap curses ncurses; do
+    LIBS="$LIBS_save $READLINE_LIBS $termcap_lib"
+	   AC_CHECK_LIB(readline, readline, [readline_libs_found=yes])
+    if test "x$readline_libs_found" = "xyes" ; then
+      break
+    fi
+  done
 	
 	LIBS=$LIBS_save
 	CPPFLAGS=$CPPFLAGS_save
