commit 1e4a95361c71e8073f448012c3eeb6f09cc9a814
Author: Joseph Myers <joseph@codesourcery.com>
Date:   Wed Oct 10 22:07:36 2012 +0000

    Use $CXX not cc1plus to locate C++ headers.

diff --git a/configure b/configure
index a5e9316..74ff690 100755
--- a/configure
+++ b/configure
@@ -5342,10 +5342,9 @@ if test -n "$sysheaders"; then
 -isystem `echo $sysheaders | sed 's/:/ -isystem /g'`"
   if test -n "$CXX"; then
     CXX_SYSINCLUDES=
-    cxxversion=`$CXX -dumpversion 2>&5` &&
     cxxmachine=`$CXX -dumpmachine 2>&5` &&
-    cxxplus=`$CXX -print-prog-name=cc1plus`
-    cxxheaders=`$cxxplus -v /dev/null 2>&1 | sed -n '/ \//{p;q;}' | sed 's/ //'`
+    cxxheaders=`$CXX -v -S -x c++ /dev/null -o /dev/null 2>&1 \
+	| sed -n -e '1,/#include/d' -e '/^ \//{p;q;}' | sed 's/ //'`
     test "x$cxxheaders" != x && test "x$i" != "x$cxxheaders" &&
     CXX_SYSINCLUDES="$CXX_SYSINCLUDES -isystem $cxxheaders \
 -isystem $cxxheaders/$cxxmachine -isystem $cxxheaders/backward"
diff --git a/configure.in b/configure.in
index b9b5703..facde5e 100644
--- a/configure.in
+++ b/configure.in
@@ -973,10 +973,9 @@ if test -n "$sysheaders"; then
 -isystem `echo $sysheaders | sed 's/:/ -isystem /g'`"
   if test -n "$CXX"; then
     CXX_SYSINCLUDES=
-    cxxversion=`$CXX -dumpversion 2>&AS_MESSAGE_LOG_FD` &&
     cxxmachine=`$CXX -dumpmachine 2>&AS_MESSAGE_LOG_FD` &&
-    cxxplus=`$CXX -print-prog-name=cc1plus`
-    cxxheaders=`$cxxplus -v /dev/null 2>&1 | sed -n '/ \//{p;q;}' | sed 's/ //'`
+    cxxheaders=`$CXX -v -S -x c++ /dev/null -o /dev/null 2>&1 \
+	| sed -n -e '1,/#include/d' -e '/^ \//{p;q;}' | sed 's/ //'`
     test "x$cxxheaders" != x && test "x$i" != "x$cxxheaders" &&
     CXX_SYSINCLUDES="$CXX_SYSINCLUDES -isystem $cxxheaders \
 -isystem $cxxheaders/$cxxmachine -isystem $cxxheaders/backward"
