---
 include/stubs-prologue.h |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/include/stubs-prologue.h
+++ b/include/stubs-prologue.h
@@ -8,7 +8,7 @@
    every time called, usually setting errno to ENOSYS.  */
 
 #ifdef _LIBC
- #error Applications may not define the macro _LIBC
+# error Applications may not define the macro _LIBC
 #endif
 
 @ Placeholder line so we remember to keep the preceding blank line here.
