diff --git a/sysdeps/posix/getaddrinfo.c b/sysdeps/posix/getaddrinfo.c
index adb3c4f..d12835c 100644
--- a/sysdeps/posix/getaddrinfo.c
+++ b/sysdeps/posix/getaddrinfo.c
@@ -263,7 +263,7 @@ extern service_user *__nss_hosts_database attribute_hidden;
 static int
 gaih_inet (const char *name, const struct gaih_service *service,
 	   const struct addrinfo *req, struct addrinfo **pai,
-	   unsigned int *naddrs)
+	   unsigned int *naddrs, bool usable_ipv6)
 {
   const struct gaih_typeproto *tp = gaih_inet_typeproto;
   struct gaih_servtuple *st = (struct gaih_servtuple *) &nullserv;
@@ -706,7 +706,7 @@ gaih_inet (const char *name, const struct gaih_service *service,
 	      if (fct != NULL)
 		{
 		  if (req->ai_family == AF_INET6
-		      || req->ai_family == AF_UNSPEC)
+		      || (req->ai_family == AF_UNSPEC && usable_ipv6))
 		    {
 		      gethosts (AF_INET6, struct in6_addr);
 		      no_inet6_data = no_data;
@@ -1903,7 +1903,7 @@ getaddrinfo (const char *name, const char *service,
   if (hints->ai_family == AF_UNSPEC || hints->ai_family == AF_INET
       || hints->ai_family == AF_INET6)
     {
-      last_i = gaih_inet (name, pservice, hints, end, &naddrs);
+      last_i = gaih_inet (name, pservice, hints, end, &naddrs, seen_ipv6);
       if (last_i != 0)
 	{
 	  freeaddrinfo (p);
diff --git a/sysdeps/unix/sysv/linux/check_pf.c b/sysdeps/unix/sysv/linux/check_pf.c
index 46161a8..5287ed0 100644
--- a/sysdeps/unix/sysv/linux/check_pf.c
+++ b/sysdeps/unix/sysv/linux/check_pf.c
@@ -146,7 +146,10 @@ make_request (int fd, pid_t pid, bool *seen_ipv4, bool *seen_ipv6,
 		  *seen_ipv4 = true;
 		  break;
 		case AF_INET6:
-		  *seen_ipv6 = true;
+		  if (ifam->ifa_scope < RT_SCOPE_LINK)
+		    {
+		      *seen_ipv6 = true;
+		    }
 
 		  if (ifam->ifa_flags & (IFA_F_DEPRECATED
 					 | IFA_F_TEMPORARY
