---
 linuxthreads/Makefile |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/linuxthreads/Makefile
+++ b/linuxthreads/Makefile
@@ -111,7 +111,7 @@
 	ex17 ex18 tst-cancel tst-context bug-sleep \
 	tst-cancel1 tst-cancel2 tst-cancel3 tst-cancel4 tst-cancel5 \
 	tst-cancel6 tst-cancel7 tst-cancel8 tst-popen tst-popen2 tst-attr1 \
-	tst-stack1 tst-align tst-tsd1 tst-sighandler1 tst-sighandler2
+	tst-stack1 tst-align tst-tsd1
 test-srcs = tst-signal
 # These tests are linked with libc before libpthread
 tests-reverse += tst-cancel5
