---
 linuxthreads/pthread.c |    2 ++
 1 file changed, 2 insertions(+)

--- a/linuxthreads/pthread.c
+++ b/linuxthreads/pthread.c
@@ -429,6 +429,8 @@
 #ifdef FLOATING_STACKS
   if (limit.rlim_cur == RLIM_INFINITY)
     limit.rlim_cur = ARCH_STACK_MAX_SIZE;
+  if (limit.rlim_cur > 4 * ARCH_STACK_MAX_SIZE)
+    limit.rlim_cur = 4 * ARCH_STACK_MAX_SIZE;
 # ifdef NEED_SEPARATE_REGISTER_STACK
   max_stack = limit.rlim_cur / 2;
 # else
