---
 math/Makefile |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/math/Makefile
+++ b/math/Makefile
@@ -89,7 +89,7 @@
 # Rules for the test suite.
 tests = test-matherr test-fenv basic-test \
 	test-misc test-fpucw tst-definitions test-tgmath test-tgmath-ret \
-	bug-nextafter bug-nexttoward bug-tgmath1 test-tgmath-int test-tgmath2 \
+	bug-nextafter bug-nexttoward bug-tgmath1 test-tgmath-int \
 	test-dbl-wrap test-powl tst-CMPLX tst-CMPLX2
 tests-$(OPTION_EGLIBC_LIBM_BIG) += atest-exp atest-sincos atest-exp2
 # We do the `long double' tests only if this data type is available and
