---
 nscd/nscd.h |   10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

--- a/nscd/nscd.h
+++ b/nscd/nscd.h
@@ -112,11 +112,11 @@
 
 
 /* Paths of the file for the persistent storage.  */
-#define _PATH_NSCD_PASSWD_DB	"/var/db/nscd/passwd"
-#define _PATH_NSCD_GROUP_DB	"/var/db/nscd/group"
-#define _PATH_NSCD_HOSTS_DB	"/var/db/nscd/hosts"
-#define _PATH_NSCD_SERVICES_DB	"/var/db/nscd/services"
-#define _PATH_NSCD_NETGROUP_DB	"/var/db/nscd/netgroup"
+#define _PATH_NSCD_PASSWD_DB	"/var/cache/nscd/passwd"
+#define _PATH_NSCD_GROUP_DB	"/var/cache/nscd/group"
+#define _PATH_NSCD_HOSTS_DB	"/var/cache/nscd/hosts"
+#define _PATH_NSCD_SERVICES_DB	"/var/cache/nscd/services"
+#define _PATH_NSCD_NETGROUP_DB	"/var/cache/nscd/netgroup"
 
 /* Path used when not using persistent storage.  */
 #define _PATH_NSCD_XYZ_DB_TMP	"/var/run/nscd/dbXXXXXX"
