---
 posix/gai.conf              |    3 ---
 sysdeps/posix/getaddrinfo.c |    4 ----
 2 files changed, 7 deletions(-)

--- a/sysdeps/posix/getaddrinfo.c
+++ b/sysdeps/posix/getaddrinfo.c
@@ -1101,10 +1101,6 @@
     /* Link-local addresses: scope 2.  */
     { { { 169, 254, 0, 0 } }, htonl_c (0xffff0000), 2 },
     { { { 127, 0, 0, 0 } }, htonl_c (0xff000000), 2 },
-    /* Site-local addresses: scope 5.  */
-    { { { 10, 0, 0, 0 } }, htonl_c (0xff000000), 5 },
-    { { { 172, 16, 0, 0 } }, htonl_c (0xfff00000), 5 },
-    { { { 192, 168, 0, 0 } }, htonl_c (0xffff0000), 5 },
     /* Default: scope 14.  */
     { { { 0, 0, 0, 0 } }, htonl_c (0x00000000), 14 }
   };
--- a/posix/gai.conf
+++ b/posix/gai.conf
@@ -62,9 +62,6 @@
 #
 #scopev4 ::ffff:169.254.0.0/112  2
 #scopev4 ::ffff:127.0.0.0/104    2
-#scopev4 ::ffff:10.0.0.0/104     5
-#scopev4 ::ffff:172.16.0.0/108   5
-#scopev4 ::ffff:192.168.0.0/112  5
 #scopev4 ::ffff:0.0.0.0/96       14
 #
 #    For sites which use site-local IPv4 addresses behind NAT there is
