http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=524255

--- a/utils/exportfs/nfsd.man	2010-02-18 23:35:00.000000000 +1100
+++ b/utils/exportfs/nfsd.man	2010-04-06 19:03:32.000000000 +1000
@@ -196,6 +196,7 @@ classes of tracing to be enabled.  Consu
 find out what number correspond to what tracing.
 
 .SH SEE ALSO
+.BR nfsd (8),
 .BR rpc.nfsd (8),
 .BR exports (5),
 .BR nfsstat (8),
--- a/utils/nfsd/nfsd.man	2010-02-18 23:35:00.000000000 +1100
+++ b/utils/nfsd/nfsd.man	2010-04-06 19:03:32.000000000 +1000
@@ -94,6 +94,7 @@ address family combinations that are mar
 database.
 
 .SH SEE ALSO
+.BR nfsd (7),
 .BR rpc.mountd (8),
 .BR exports (5),
 .BR exportfs (8),
