From: Ben Hutchings <ben@decadent.org.uk>
Subject: [PATCH] mount.nfs.man, nfs.man: Update distinction between fstypes

This addresses Debian bug #575503.

--- nfs-utils.orig/utils/mount/mount.nfs.man
+++ nfs-utils/utils/mount/mount.nfs.man
@@ -15,16 +15,20 @@
 .BR mount (8)
 command for mounting NFS shares. This subcommand, however, can also be used as a standalone command with limited functionality.
 
-.BR mount.nfs4 
-is used for mounting NFSv4 file system, while 
-.BR mount.nfs 
-is used to mount NFS file systems versions 3 or 2.
 .I remotetarget 
 is a server share usually in the form of 
 .BR servername:/path/to/share.
 .I dir 
 is the directory on which the file system is to be mounted.
 
+Under Linux 2.6.32 and later kernel versions,
+.BR mount.nfs
+can mount all NFS file system versions.  Under earlier Linux kernel versions,
+.BR mount.nfs4
+must be used for mounting NFSv4 file systems while
+.BR mount.nfs
+must be used for NFSv3 and v2.
+
 .SH OPTIONS
 .TP
 .BI "\-r"
