--- smpeg-0.4.5+cvs20030824.orig/configure.in
+++ smpeg-0.4.5+cvs20030824/configure.in
@@ -93,9 +93,10 @@
 # Set runtime shared library paths as needed 
 
 case "$target" in
-    *-*-linux*)
-	SMPEG_RLD_FLAGS="-Wl,-rpath,\${exec_prefix}/lib"
-	;;
+# We don't want or need rpath in Debian
+#    *-*-linux*)
+#	SMPEG_RLD_FLAGS="-Wl,-rpath,\${exec_prefix}/lib"
+#	;;
     *-*-freebsd*)
 	SMPEG_RLD_FLAGS="-Wl,-rpath,\${exec_prefix}/lib"
 	;;
