Description: Fix a few comment typos in CMS-related makefiles
Author: Jonas Smedegaard <dr@jones.dk>
Last-Update: 2011-10-02

--- a/base/lcms2.mak
+++ b/base/lcms2.mak
@@ -14,7 +14,7 @@
 
 # makefile for the lcms library code.
 # Users of this makefile must define the following:
-#	SHARE_LCMS2 - whether to compile in or link to the library
+#	SHARE_LCMS - whether to compile in or link to the library
 #	LCMS2SRCDIR - the library source directory
 #
 # gs.mak and friends define the following:
--- a/base/lcms.mak
+++ b/base/lcms.mak
@@ -18,7 +18,7 @@
 #	LCMSSRCDIR - the library source directory
 #
 # gs.mak and friends define the following:
-#	LCSMOBJDIR - the output obj directory
+#	LCMSOBJDIR - the output obj directory
 #	LCMSGENDIR - generated (.dev) file directory
 #	LCMSI_ LCMS_CFLAGS - include and cflags for compiling the lib
 
