Index: libvpx/build/make/configure.sh
===================================================================
--- libvpx.orig/build/make/configure.sh	2012-05-14 10:02:11.000000000 +0200
+++ libvpx/build/make/configure.sh	2012-05-14 10:07:58.957598682 +0200
@@ -725,7 +725,6 @@
 
         case ${tgt_cc} in
         gcc)
-            CROSS=${CROSS:-arm-none-linux-gnueabi-}
             link_with_cc=gcc
             setup_gnu_toolchain
             arch_int=${tgt_isa##armv}
@@ -738,8 +737,8 @@
                     check_add_cflags -mfpu=neon #-ftree-vectorize
                     check_add_asflags -mfpu=neon
                 fi
-                check_add_cflags -march=armv7-a -mcpu=cortex-a8 -mfloat-abi=softfp
-                check_add_asflags -mcpu=cortex-a8 -mfloat-abi=softfp  #-march=armv7-a
+                check_add_cflags -march=armv7-a -mcpu=cortex-a8
+                check_add_asflags -mcpu=cortex-a8 #-march=armv7-a
             else
                 check_add_cflags -march=${tgt_isa}
                 check_add_asflags -march=${tgt_isa}
