Index: dcmtk-3.6.0/dcmwlm/docs/Makefile.in
===================================================================
--- dcmtk-3.6.0.orig/dcmwlm/docs/Makefile.in	2007-11-23 17:30:50.000000000 +0100
+++ dcmtk-3.6.0/dcmwlm/docs/Makefile.in	2011-01-28 09:58:14.000000000 +0100
@@ -11,7 +11,8 @@
 
 include $(configdir)/@common_makefile@
 
-files = wwwapp.txt
+## we use debhelper for this doc which goes to the dcmtk-www package
+# files = wwwapp.txt
 
 
 all:
Index: dcmtk-3.6.0/config/rootconf
===================================================================
--- dcmtk-3.6.0.orig/config/rootconf	2010-11-25 12:09:26.000000000 +0100
+++ dcmtk-3.6.0/config/rootconf	2011-01-28 09:58:53.000000000 +0100
@@ -67,7 +67,7 @@
 
 dcmtk-install-doc:
 	\$(configdir)/mkinstalldirs \$(DESTDIR)\$(docdir)
-	for file in ANNOUNCE.\$(PACKAGE_VERSION_NUMBER) CHANGES.\$(PACKAGE_VERSION_NUMBER) COPYRIGHT FAQ HISTORY VERSION; do \\
+	for file in ANNOUNCE.\$(PACKAGE_VERSION_NUMBER) CHANGES.\$(PACKAGE_VERSION_NUMBER) FAQ HISTORY VERSION; do \\
 		\$(INSTALL_DATA) \$\$file \$(DESTDIR)\$(docdir) ;\\
 	done
 
