Index: dcmtk-3.6.0/dcmjpls/Makefile.in
===================================================================
--- dcmtk-3.6.0.orig/dcmjpls/Makefile.in	2009-07-29 16:46:45.000000000 +0200
+++ dcmtk-3.6.0/dcmjpls/Makefile.in	2011-08-23 14:10:45.000000000 +0200
@@ -36,7 +36,6 @@
 	(cd include && $(MAKE) ARCH="$(ARCH)" all)
 
 libsrc-all: include-all
-	(cd libcharls && $(MAKE) ARCH="$(ARCH)" all)
 	(cd libsrc && $(MAKE) ARCH="$(ARCH)" all)
 
 apps-all: libsrc-all
@@ -50,7 +49,6 @@
 	(cd include && $(MAKE) ARCH="$(ARCH)" install)
 
 libsrc-install: libsrc-all
-	(cd libcharls && $(MAKE) ARCH="$(ARCH)" install)
 	(cd libsrc && $(MAKE) ARCH="$(ARCH)" install)
 
 apps-install: apps-all
@@ -68,7 +66,6 @@
 
 clean:
 	(cd include && $(MAKE) clean)
-	(cd libcharls && $(MAKE) clean)
 	(cd libsrc && $(MAKE) clean)
 	(cd apps && $(MAKE) clean)
 	(cd tests && $(MAKE) clean)
@@ -79,7 +76,6 @@
 
 distclean:
 	(cd include && $(MAKE) distclean)
-	(cd libcharls && $(MAKE) distclean)
 	(cd libsrc && $(MAKE) distclean)
 	(cd apps && $(MAKE) distclean)
 	(cd tests && $(MAKE) distclean)
@@ -89,7 +85,6 @@
 	rm -f $(DISTTRASH)
 
 dependencies:
-	(cd libcharls && touch $(DEP) && $(MAKE) dependencies)
 	(cd libsrc && touch $(DEP) && $(MAKE) dependencies)
 	(cd apps && touch $(DEP) && $(MAKE) dependencies)
 #	(cd tests && touch $(DEP) && $(MAKE) dependencies)
Index: dcmtk-3.6.0/dcmjpls/libsrc/Makefile.in
===================================================================
--- dcmtk-3.6.0.orig/dcmjpls/libsrc/Makefile.in	2009-10-07 15:16:47.000000000 +0200
+++ dcmtk-3.6.0/dcmjpls/libsrc/Makefile.in	2011-08-23 14:10:45.000000000 +0200
@@ -16,10 +16,9 @@
 oflogdir = $(top_srcdir)/../oflog
 dcmdatadir = $(top_srcdir)/../dcmdata
 dcmimgledir = $(top_srcdir)/../dcmimgle
-libcharlsdir = $(top_srcdir)/../dcmjpls/libcharls
 
 LOCALINCLUDES = -I$(ofstddir)/include -I$(oflogdir)/include -I$(dcmdatadir)/include \
-  -I$(dcmimgledir)/include -I$(libcharlsdir)
+  -I$(dcmimgledir)/include
 LOCALDEFS =
 
 objs = djcodecd.o djcodece.o djcparam.o djdecode.o djencode.o djrparam.o djutils.o
Index: dcmtk-3.6.0/dcmjpls/apps/Makefile.in
===================================================================
--- dcmtk-3.6.0.orig/dcmjpls/apps/Makefile.in	2011-08-23 14:10:41.000000000 +0200
+++ dcmtk-3.6.0/dcmjpls/apps/Makefile.in	2011-08-23 14:11:27.000000000 +0200
@@ -42,9 +42,7 @@
 dcmjplslibdir = -L$(dcmjplsdir)/libsrc
 dcmjplslib = -ldcmjpls
 
-libcharlsdir = $(dcmjplsdir)
-libcharlslibdir = -L$(dcmjplsdir)/libcharls
-libcharlslib = -lcharls
+libcharlslib = -lCharLS
 
 LOCALINCLUDES = $(dcmjplsinc) $(ofstdinc) $(ofloginc) $(dcmdatainc) $(dcmimageinc) $(dcmimgleinc)
 LIBDIRS = -L$(top_srcdir)/libsrc $(dcmjplslibdir) $(libcharlslibdir) $(dcmimagelibdir) \
Index: dcmtk-3.6.0/dcmjpls/libsrc/djcodecd.cc
===================================================================
--- dcmtk-3.6.0.orig/dcmjpls/libsrc/djcodecd.cc	2010-10-14 15:14:24.000000000 +0200
+++ dcmtk-3.6.0/dcmjpls/libsrc/djcodecd.cc	2011-08-23 14:10:45.000000000 +0200
@@ -44,7 +44,7 @@
 #include "djerror.h"                 /* for private class DJLSError */
 
 // JPEG-LS library (CharLS) includes
-#include "intrface.h"
+#include "CharLS/interface.h"
 
 E_TransferSyntax DJLSLosslessDecoder::supportedTransferSyntax() const
 {
Index: dcmtk-3.6.0/dcmjpls/libsrc/djcodece.cc
===================================================================
--- dcmtk-3.6.0.orig/dcmjpls/libsrc/djcodece.cc	2010-10-14 15:14:24.000000000 +0200
+++ dcmtk-3.6.0/dcmjpls/libsrc/djcodece.cc	2011-08-23 14:10:45.000000000 +0200
@@ -62,7 +62,7 @@
 #include "dcmtk/dcmimgle/dcmimage.h"  /* for class DicomImage */
 
 // JPEG-LS library (CharLS) includes
-#include "intrface.h"
+#include "CharLS/interface.h"
 
 BEGIN_EXTERN_C
 #ifdef HAVE_FCNTL_H
Index: dcmtk-3.6.0/dcmjpls/libsrc/djerror.h
===================================================================
--- dcmtk-3.6.0.orig/dcmjpls/libsrc/djerror.h	2010-10-14 15:20:24.000000000 +0200
+++ dcmtk-3.6.0/dcmjpls/libsrc/djerror.h	2011-08-23 14:10:45.000000000 +0200
@@ -31,7 +31,7 @@
 
 #include "dcmtk/config/osconfig.h"
 #include "dcmtk/dcmjpls/djlsutil.h" /* For the OFCondition codes */
-#include "intrface.h" /* CharLS include */
+#include "CharLS/interface.h" /* CharLS include */
 
 /** Helper class for converting between dcmjpls and CharLS error codes
  */
