Description: The calendar widget segfaults when you call activateCDKCalendar().
Bug: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=452401
Author: Ron Murray <rjmx@rjmx.net>

Index: libcdk5-5.0.20060507/calendar.c
===================================================================
--- libcdk5-5.0.20060507.orig/calendar.c	2006-05-04 20:27:45.000000000 -0400
+++ libcdk5-5.0.20060507/calendar.c	2010-08-11 23:35:22.000000000 -0400
@@ -459,7 +459,7 @@
    int day		= 1;
    int x, y;
    int save_y = -1, save_x = -1;
-   char temp[10];
+   char temp[20];
 
    for (x = 1; x <= 6; x++)
    {
