Index: libcdk5-5.0.20060507/Makefile.in
===================================================================
--- libcdk5-5.0.20060507.orig/Makefile.in	2011-02-19 23:59:24.665541599 +0000
+++ libcdk5-5.0.20060507/Makefile.in	2011-02-19 23:59:36.785541598 +0000
@@ -269,7 +269,7 @@
 	@$(RANLIB) $@
 
 cdkshlib $(CDKSHLIB) :: $(CDKSHOBJS)
-	gcc -shared -Wl,-soname=$(SONAME) $(LDFLAGS) -o $(CDKSHLIB) $(CDKSHOBJS)
+	gcc -shared -Wl,-soname=$(SONAME) $(LDFLAGS) -o $(CDKSHLIB) $(CDKSHOBJS) $(LIBS)
 
 #
 # Make the examples directory.
