Description: fixed missing headers for tests and missing examples
Bug: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=500161
Origin: https://bugs.launchpad.net/ubuntu/+source/libcdk5/+bug/565526

Index: libcdk5-5.0.20060507/demos/Makefile.in
===================================================================
--- libcdk5-5.0.20060507.orig/demos/Makefile.in	2003-11-27 07:03:51.000000000 -0500
+++ libcdk5-5.0.20060507/demos/Makefile.in	2010-08-11 23:28:14.000000000 -0400
@@ -24,7 +24,7 @@
 o		= .@OBJEXT@
 
 CFLAGS		= @CFLAGS@ @EXTRA_CFLAGS@
-CPPFLAGS	= @DEFS@ -I../include -I$(srcdir)/../include -I. @CPPFLAGS@
+CPPFLAGS	= @DEFS@ -I../include -I$(srcdir)/../include -I/usr/include/cdk -I. @CPPFLAGS@
 LDFLAGS		= -L.. @LDFLAGS@
 
 BINS		= \
Index: libcdk5-5.0.20060507/examples/Makefile.in
===================================================================
--- libcdk5-5.0.20060507.orig/examples/Makefile.in	2005-12-27 19:03:39.000000000 -0500
+++ libcdk5-5.0.20060507/examples/Makefile.in	2010-08-11 23:28:14.000000000 -0400
@@ -24,7 +24,7 @@
 o		= .@OBJEXT@
 
 CFLAGS		= @CFLAGS@ @EXTRA_CFLAGS@
-CPPFLAGS	= @DEFS@ -I../include -I$(srcdir)/../include -I. @CPPFLAGS@
+CPPFLAGS	= @DEFS@ -I../include -I$(srcdir)/../include -I/usr/include/cdk -I. @CPPFLAGS@
 LDFLAGS		= -L.. @LDFLAGS@
 
 BINS	= \
