Description: add support for Python 3
Author: Jakub Wilk <jwilk@debian.org>
Forwarded: no
Last-Update: 2012-05-05

--- a/setup.py
+++ b/setup.py
@@ -1,5 +1,10 @@
 from distutils.core import setup
 
+try:
+    from distutils.command.build_py import build_py_2to3 as build_py
+except ImportError:
+    from distutils.command.build_py import build_py
+
 setup (
     name='roman',
     version='1.4.0',
@@ -18,4 +23,5 @@
     url = 'http://pypi.python.org/pypi/roman',
     package_dir={"": "src"},
     py_modules=["roman"],
+    cmdclass=dict(build_py=build_py)
     )
