Description: don't use setuptools
Author: Jakub Wilk <jwilk@debian.org>
Forwarded: not-needed
Last-Update: 2012-05-05

--- a/setup.py
+++ b/setup.py
@@ -1,4 +1,4 @@
-from setuptools import setup
+from distutils.core import setup
 
 setup (
     name='roman',
@@ -18,6 +18,4 @@
     url = 'http://pypi.python.org/pypi/roman',
     package_dir={"": "src"},
     py_modules=["roman"],
-    include_package_data = True,
-    zip_safe = True,
     )
