From ba9976cd6e00f831e6bb104b275ae63998a6a6d0 Mon Sep 17 00:00:00 2001
From: Bernhard R. Link <brlink@debian.org>
Date: Sat, 12 Mar 2011 16:16:39 +0100
Subject: rename cl binary to cliquer

---
 Makefile |    8 ++++----
 1 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/Makefile b/Makefile
index da7fb79..8620a22 100644
--- a/Makefile
+++ b/Makefile
@@ -29,7 +29,7 @@ SHCFLAGS  = -fPIC -fvisibility=hidden
 
 ## Program options:
 
-# Enable long options for cl (eg. "cl --help"), comment out to disable.
+# Enable long options for cliquer (eg. "cliquer --help"), comment out to disable.
 # Requires getopt_long(3)  (a GNU extension)
 LONGOPTS = -DENABLE_LONG_OPTIONS
 
@@ -37,7 +37,7 @@ LONGOPTS = -DENABLE_LONG_OPTIONS
 ##### End of configurable options
 
 
-all: cl libcliquer.so
+all: cliquer libcliquer.so
 
 libcliquer_OBJECTS = cliquer.o graph.o reorder.o
 
@@ -48,7 +48,7 @@ testcases: testcases.o cliquer.o graph.o reorder.o
 libcliquer.so: $(libcliquer_OBJECTS) libcliquer.map
 	$(CC) $(LDFLAGS) $(SHCFLAGS) $(SHLDFLAGS) -o $@ $(libcliquer_OBJECTS)
 
-cl: cl.o libcliquer.so
+cliquer: cl.o libcliquer.so
 	$(CC) $(LDFLAGS) -o $@ cl.o -L. -lcliquer
 
 
@@ -61,7 +61,7 @@ $(libcliquer_OBJECTS): %.o: %.c
 	$(CC) $(CFLAGS) $(SHCFLAGS) $(LONGOPTS) -o $@ -c $<
 
 clean:
-	rm -f *.o *~ cl testcases *.so*
+	rm -f *.o *~ cliquer testcases *.so*
 
 backup:
 	mkdir "`date "+backup-%Y-%m-%d-%H-%M"`" 2>/dev/null || true
-- 
1.5.6.5

