From d62c750da74ed839b57719fdeb340ecf7bccec9f Mon Sep 17 00:00:00 2001
From: Bernhard R. Link <brlink@debian.org>
Date: Sat, 12 Mar 2011 17:11:13 +0100
Subject: add install target

---
 Makefile |   23 ++++++++++++++++++++---
 1 files changed, 20 insertions(+), 3 deletions(-)

diff --git a/Makefile b/Makefile
index 89ebbe8..b60e992 100644
--- a/Makefile
+++ b/Makefile
@@ -20,11 +20,11 @@ CFLAGS=-Wall -O3 -fomit-frame-pointer -funroll-loops
 #CFLAGS=-fast
 
 # GCC:
-SHLDFLAGS = --shared -Wl,--soname=libcliquer.so.1 -Wl,--version-script=libcliquer.map
+SHLDFLAGS = --shared -Wl,--soname=$(SONAME) -Wl,--version-script=libcliquer.map
 SHCFLAGS  = -fPIC -fvisibility=hidden
 
 # SunOS:
-#SHLDFLAGS = --shared -Wl,--soname=libcliquer.so.1 -Wl,--version-script=libcliquer.map
+#SHLDFLAGS = --shared -Wl,--soname=$(SONAME) -Wl,--version-script=libcliquer.map
 #SHCFLAGS  = -fPIC
 
 CPPFLAGS =
@@ -42,7 +42,7 @@ LONGOPTS = -DENABLE_LONG_OPTIONS
 all: cliquer libcliquer.so
 
 libcliquer_OBJECTS = cliquer.o graph.o reorder.o
-
+SONAME=libcliquer.so.1
 
 testcases: testcases.o cliquer.o graph.o reorder.o
 	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ testcases.o cliquer.o graph.o reorder.o
@@ -74,3 +74,20 @@ backup:
 
 test: testcases
 	./testcases
+
+DESTDIR=
+INSTALL=install
+INSTALL_DATA=$(INSTALL) -m644
+MKDIR_P=mkdir -p
+prefix=/usr/local
+
+install: cliquer libcliquer.so
+	$(MKDIR_P) $(DESTDIR)$(prefix)/bin
+	$(MKDIR_P) $(DESTDIR)$(prefix)/lib
+	$(MKDIR_P) $(DESTDIR)$(prefix)/include/cliquer
+	$(INSTALL) cliquer $(DESTDIR)$(prefix)/bin/
+	$(INSTALL) libcliquer.so $(DESTDIR)$(prefix)/lib/$(SONAME)
+	ln -s $(SONAME) $(DESTDIR)$(prefix)/lib/libcliquer.so
+	# users should #include <cliquer/cliquer.h>
+	$(INSTALL_DATA) cliquer.h set.h graph.h misc.h reorder.h \
+		cliquerconf.h $(DESTDIR)$(prefix)/include/cliquer/
-- 
1.5.6.5

