Description: Run exit hooks when "dhclient -1" fails (closes: #486520).
Author: Martin Pitt <martin.pitt@ubuntu.com>
Bug-Debian: http://bugs.debian.org/486520
--- isc-dhcp-4.2.4.orig/client/dhclient.c
+++ isc-dhcp-4.2.4/client/dhclient.c
@@ -2018,6 +2018,8 @@ void state_panic (cpp)
 		if (!quiet)
 			log_info ("Unable to obtain a lease on first try.%s",
 				  "  Exiting.");
+		script_init (client, "FAIL", (struct string_list *)0);
+		script_go (client);
 		exit (2);
 	}
 
