Index: b/util/getroot.c
===================================================================
--- a/util/getroot.c
+++ b/util/getroot.c
@@ -2144,7 +2144,7 @@
     }
   *ptr = 0;
   if (dos_part >= 0)
-    snprintf (ptr, end - ptr, ",%d", dos_part + 1);
+    snprintf (ptr, end - ptr, ",%d", dos_part + (getenv ("GRUB_LEGACY_0_BASED_PARTITIONS") ? 0 : 1));
   ptr += strlen (ptr);
   if (bsd_part >= 0)
     snprintf (ptr, end - ptr, ",%d", bsd_part + 1); 
@@ -2241,6 +2241,29 @@
 
 	if (start == part_start)
 	  {
+	    if (getenv ("GRUB_LEGACY_0_BASED_PARTITIONS"))
+	      {
+		int dos_part, bsd_part;
+		char *fullname, *comma;
+
+		if (partition->parent)
+		  {
+		    dos_part = partition->parent->number;
+		    bsd_part = partition->number;
+		  }
+		else
+		  {
+		    dos_part = partition->number;
+		    bsd_part = -1;
+		  }
+
+		fullname = make_device_name (drive, dos_part, bsd_part);
+		comma = strchr (fullname, ',');
+		partname = comma ? xstrdup (comma + 1) : NULL;
+		free (fullname);
+		return 1;
+	      }
+
 	    partname = grub_partition_get_name (partition);
 	    return 1;
 	  }
