Description: Detect Ubuntu's use of "recovery" rather than "single"
Author: Colin Watson <cjwatson@ubuntu.com>
Forwarded: no
Last-Update: 2012-09-18

Index: b/util/grub.d/30_os-prober.in
===================================================================
--- a/util/grub.d/30_os-prober.in
+++ b/util/grub.d/30_os-prober.in
@@ -170,7 +170,7 @@
 	fi
 
 	onstr="$(gettext_printf "(on %s)" "${DEVICE}")"
-	recovery_params="$(echo "${LPARAMS}" | grep single)" || true
+	recovery_params="$(echo "${LPARAMS}" | grep 'single\|recovery')" || true
 	counter=1
 	while echo "$used_osprober_linux_ids" | grep 'osprober-gnulinux-$LKERNEL-${recovery_params}-$counter-$boot_device_id' > /dev/null; do
 	    counter=$((counter+1));
