--- a/packages/jpl/Makefile.in
+++ b/packages/jpl/Makefile.in
@@ -26,6 +26,7 @@
 JUNIT=@JUNIT@
 JAVALIBS=@JAVALIBS@
 JAVA_PRELOAD=@JAVA_PRELOAD@
+JAVA_LIBRARY_PATH=@JAVA_LIBRARY_PATH@
 
 LIBPL=		jpl.pl
 LIBJPL=		@LIBJPL@.@SO@
@@ -53,7 +54,7 @@
 
 check_pl:	jpltest.jar
 		if [ -r jpltest.jar ]; then \
-		  $(PL) -q -f test_jpl.pl -g run_tests,halt -t 'halt(1)' ; \
+		  LD_LIBRARY_PATH=$(JAVA_LIBRARY_PATH) $(PL) -q -f test_jpl.pl -g run_tests,halt -t 'halt(1)' ; \
 		else \
 		  echo "No jpltest.jar; maybe junit is not installed?" ; \
 		fi
--- a/packages/jpl/configure.in
+++ b/packages/jpl/configure.in
@@ -11,6 +11,7 @@
 AC_SUBST(JPLLDFLAGS)
 AC_SUBST(LIBJPL)
 AC_SUBST(JAVA_PRELOAD)
+AC_SUBST(JAVA_LIBRARY_PATH)
 AC_SUBST(CMDEXT)
 
 m4_include([../ac_swi_c.m4])
@@ -97,6 +98,8 @@
 done
 JPLLDFLAGS="$(echo $JPLLDFLAGS | unescape_space)"
 
+JAVA_LIBRARY_PATH=$(echo $JNI_CLIENT_DIRS | escape_space | sed 's/ /:/' | unescape_space)
+
 for d in $(echo $JNI_CLIENT_DIRS | escape_space); do
   if test -f "$d/libjsig.$SO"; then
      JAVALIBS="-ljsig $JAVALIBS"
