--- a/packages/clib/maildrop/rfc2045/Makefile.in
+++ b/packages/clib/maildrop/rfc2045/Makefile.in
@@ -5,7 +5,7 @@
 ################################################################
 
 CC=@CC@
-CFLAGS=@CFLAGS@ @DEFS@
+CFLAGS=@CFLAGS@ @DEFS@ -fPIC
 LIBS=@LIBS@
 AR=@AR@
 RANLIB=@RANLIB@
--- a/packages/clib/maildrop/rfc822/Makefile.in
+++ b/packages/clib/maildrop/rfc822/Makefile.in
@@ -5,7 +5,7 @@
 ################################################################
 
 CC=@CC@
-CFLAGS=@CFLAGS@ @DEFS@
+CFLAGS=@CFLAGS@ @DEFS@ -fPIC
 LIBS=@LIBS@
 AR=@AR@
 RANLIB=@RANLIB@
