--- a/src/swipl.pc.in
+++ b/src/swipl.pc.in
@@ -3,17 +3,17 @@
 libdir=@libdir@
 includedir=@includedir@
 
-PLBASE=@PLBASE@
+PLBASE=/usr/lib/swi-prlog
 PLARCH=@PLARCH@
 INSTALL_PLARCH=@INSTALL_PLARCH@
 PLLIBS=@LNLIBS@
 PLLIB=@USEPLLIB@
-PLCFLAGS=@CMFLAGS@ @CIFLAGS@
-PLLDFLAGS=@LDFLAGS@
+PLCFLAGS=-fno-strict-aliasing -pthread
+PLLDFLAGS=-pthread
 PLSOEXT=@SO_EXT@
 
 Name: swipl
 Description: ISO/Edinburgh-style Prolog compiler
 Version: @PLVERSION@
-Libs: ${PLLDFLAGS}  -L@PLBASE@/lib/@INSTALL_PLARCH@ -lswipl
-Cflags: ${PLCFLAGS} -I@PLBASE@/include
+Libs: ${PLLDFLAGS}  -L${PLBASE}/lib/@INSTALL_PLARCH@ -lswipl
+Cflags: ${PLCFLAGS} -I${PLBASE}/include
