From: Youhei SASAKI <uwabami@gfd-dennou.org>
Date: Mon, 15 Oct 2012 00:17:36 +0900
Subject: Fix typo.

Signed-off-by: Youhei SASAKI <uwabami@gfd-dennou.org>
---
 System-Monitor/src/applet-init.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/System-Monitor/src/applet-init.c b/System-Monitor/src/applet-init.c
index 88cebcc..18603e7 100755
--- a/System-Monitor/src/applet-init.c
+++ b/System-Monitor/src/applet-init.c
@@ -37,7 +37,7 @@ CD_APPLET_DEFINITION (N_("System Monitor"),
 	CAIRO_DOCK_CATEGORY_APPLET_SYSTEM,
 	N_("This applet shows you the CPU load, RAM usage, graphic card temperature, etc.\n"
 	"Middle click on the icon to get some valuable info.\n"
-	"Left click on the icon to get a list of the most ressources using programs.\n"
+	"Left click on the icon to get a list of the most resources using programs.\n"
 	"You can instanciate this applet several times to show different values each time."),
 	"parAdOxxx_ZeRo and Fabounet")
 
