From: Eric Cooper <ecc@cmu.edu>
Date: Tue, 3 Mar 2009 16:28:31 -0500
Subject: use ocamlmakefile package


Signed-off-by: Eric Cooper <ecc@cmu.edu>
---
 Makefile |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/Makefile b/Makefile
index 61245a8..948bed5 100644
--- a/Makefile
+++ b/Makefile
@@ -1,4 +1,4 @@
--include Makefile.conf
+OCAMLMAKEFILE = /usr/share/ocamlmakefile/OCamlMakefile
 
 SOURCES=syslog.mli syslog.ml
 RESULT=syslog
@@ -13,4 +13,4 @@ install: libinstall
 uninstall: libuninstall
 doc: htdoc
 
--include OCamlMakefile
+include $(OCAMLMAKEFILE)
-- 
