Index: mpich2-1.4.1/Makefile.in
===================================================================
--- mpich2-1.4.1.orig/Makefile.in
+++ mpich2-1.4.1/Makefile.in
@@ -105,20 +105,20 @@
 C_COMPILE       = $(CC) $(DEFS) $(INCLUDES) $(CFLAGS) $(CPPFLAGS)
 # Build the shared library from the shared object files
 lib/lib${MPILIBNAME}.@SHLIB_EXT@: lib/lib${MPILIBNAME}.la
-	(cd lib && $(CREATESHLIB) --mode=link  -version-info "$(ABIVERSION)" -o lib${MPILIBNAME}.la  -rpath $(libdir) -L. @LDFLAGS_DEPS@ @LIB_DEPS@ -ldflags "$(LDFLAGS)" $(LIBS))
+	(cd lib && $(CREATESHLIB) --mode=link  -version-info "$(ABIVERSION)" -o lib${MPILIBNAME}.la  -rpath $(libdir) -L. @LDFLAGS_DEPS@ @LIB_DEPS@ -ldflags "$(LDFLAGS) -Wl,-z,defs" $(LIBS) -lmpl)
 	if [ -n "@NO_WEAK_SYM@" -a \
 		"${MPILIBNAME}" != "${PMPILIBNAME}" ] ; then \
-	(cd lib && $(CREATESHLIB) --mode=link  -version-info "$(ABIVERSION)" -o lib${PMPILIBNAME}.la  -rpath $(libdir) );\
+	(cd lib && $(CREATESHLIB) --mode=link  -version-info "$(ABIVERSION)" -o lib${PMPILIBNAME}.la  -rpath $(libdir) -ldflags "-Wl,-z,defs");\
 	fi
 # Build the shared library from the shared object files
 lib/libf${MPILIBNAME}.@SHLIB_EXT@: lib/libf${MPILIBNAME}.la
-	(cd lib && $(CREATESHLIB) --mode=link  -version-info "$(ABIVERSION)" -o libf${MPILIBNAME}.la  -rpath $(libdir) -L. @LDFLAGS_DEPS@ -l$(MPILIBNAME) @LIB_DEPS@ -ldflags "$(LDFLAGS)" $(LIBS))
+	(cd lib && $(CREATESHLIB) --mode=link  -version-info "$(ABIVERSION)" -o libf${MPILIBNAME}.la  -rpath $(libdir) -L. @LDFLAGS_DEPS@ -l$(MPILIBNAME) @LIB_DEPS@ -ldflags "$(LDFLAGS) -Wl,-z,defs" $(LIBS))
 # Build the shared library from the shared object files
 lib/lib${MPILIBNAME}f90.@SHLIB_EXT@: lib/lib${MPILIBNAME}f90.la
-	(cd lib && $(CREATESHLIB) --mode=link  -version-info "$(ABIVERSION)" -o lib${MPILIBNAME}f90.la  -rpath $(libdir) -L. @LDFLAGS_DEPS@ -l$(MPILIBNAME) @LIB_DEPS@ -ldflags "$(LDFLAGS)" $(LIBS))
+	(cd lib && $(CREATESHLIB) --mode=link  -version-info "$(ABIVERSION)" -o lib${MPILIBNAME}f90.la  -rpath $(libdir) -L. @LDFLAGS_DEPS@ -l$(MPILIBNAME) @LIB_DEPS@ -ldflags "$(LDFLAGS) -Wl,-z,defs" $(LIBS))
 # Build the shared library from the shared object files
 lib/lib${MPICXXLIBNAME}.@SHLIB_EXT@: lib/lib${MPICXXLIBNAME}.la
-	(cd lib && $(CREATESHLIB) --mode=link  -version-info "$(ABIVERSION)" -o lib${MPICXXLIBNAME}.la  -rpath $(libdir) -L. @LDFLAGS_DEPS@ -l$(MPILIBNAME) @LIB_DEPS@ -ldflags "$(LDFLAGS)" $(LIBS))
+	(cd lib && $(CREATESHLIB) --mode=link  -version-info "$(ABIVERSION)" -o lib${MPICXXLIBNAME}.la  -rpath $(libdir) -L. @LDFLAGS_DEPS@ -l$(MPILIBNAME) @LIB_DEPS@ -ldflags "$(LDFLAGS) -Wl,-z,defs" $(LIBS) -lstdc++)
 
 # --------------------------------------------------------------------------
 .libstamp0: 
