Index: pygtk-2.13.0/examples/pygtk-demo/pygtk-demo.in
===================================================================
--- pygtk-2.13.0.orig/examples/pygtk-demo/pygtk-demo.in	2008-04-27 11:51:14.000000000 +0200
+++ pygtk-2.13.0/examples/pygtk-demo/pygtk-demo.in	2008-11-12 17:59:10.695490976 +0100
@@ -1,7 +1,7 @@
-#! @PYTHON@
+#! /usr/bin/python
 
 import sys
 import os.path
-pygtklibdir = os.path.join("@libdir@", "pygtk", "2.0")
+pygtklibdir = os.path.join("/usr/share", "pygtk", "2.0")
 sys.path.insert(0, pygtklibdir)
 execfile(os.path.join(pygtklibdir, "pygtk-demo.py"))
