Index: pygtk-2.13.0/tests/runtests.py
===================================================================
--- pygtk-2.13.0.orig/tests/runtests.py	2008-11-26 00:13:20.451100628 +0100
+++ pygtk-2.13.0/tests/runtests.py	2008-11-26 00:13:34.107100762 +0100
@@ -21,7 +21,7 @@ else:
 common.importModules(buildDir=buildDir,
                      srcDir=srcDir)
 
-SKIP_FILES = ['common', 'runtests', 'testmodule']
+SKIP_FILES = ['common', 'runtests', 'testmodule', 'test_filechooserdialog']
 
 dir = os.path.split(os.path.abspath(__file__))[0]
 os.chdir(dir)
