Author: Lucas Nussbaum <lucas@lucas-nussbaum.net>
Description: run tests in verbose mode
--- a/common.mk
+++ b/common.mk
@@ -306,7 +306,7 @@ test: miniruby$(EXEEXT) $(RBCONFIG) $(PR
 	@$(MINIRUBY) $(srcdir)/rubytest.rb
 
 test-all:
-	$(RUNRUBY) "$(srcdir)/test/runner.rb" --basedir="$(TESTSDIR)" --runner=$(TESTUI) $(TESTS)
+	$(RUNRUBY) "$(srcdir)/test/runner.rb" --basedir="$(TESTSDIR)" --runner=$(TESTUI) -v $(TESTS)
 
 extconf:
 	$(MINIRUBY) -run -e mkdir -- -p "$(EXTCONFDIR)"
