Author: akira yamada <akira@debian.org>
Description: Debian specific patch
--- a/lib/rdoc/ri/ri_display.rb
+++ b/lib/rdoc/ri/ri_display.rb
@@ -210,7 +210,7 @@ class  DefaultDisplay
 
   def setup_pager
     unless @options.use_stdout
-      for pager in [ ENV['PAGER'], "less", "more", 'pager' ].compact.uniq
+      for pager in [ ENV['PAGER'], 'pager', "less", "more" ].compact.uniq
         return IO.popen(pager, "w") rescue nil
       end
       @options.use_stdout = true
