Index: bless/builder/ModuleBuilder.cs
===================================================================
--- bless.orig/builder/ModuleBuilder.cs	2011-05-25 21:04:08.826833195 +0200
+++ bless/builder/ModuleBuilder.cs	2011-05-25 21:04:28.776833188 +0200
@@ -119,7 +119,7 @@
 		//System.Console.WriteLine("gmcs {0}", sb.ToString());
 		System.Console.WriteLine(">> Building module {0}...", module.Name);
 
-		Process buildProcess = Process.Start("gmcs", sb.ToString());
+		Process buildProcess = Process.Start("mono-csc", sb.ToString());
 		buildProcess.WaitForExit();
 
 		if (buildProcess.ExitCode == 0) {
