Create the SystemV links, for compatibility with previous versions.

--- tzdata.orig/systemv	2006-04-10 11:36:52.455402059 +0200
+++ tzdata/systemv	2006-04-10 11:37:12.684500320 +0200
@@ -34,3 +34,16 @@
 ## Zone	SystemV/PST8	-8:00	-		PST
 ## Zone	SystemV/YST9	-9:00	-		YST
 ## Zone	SystemV/HST10	-10:00	-		HST
+Link    America/Halifax         SystemV/AST4ADT
+Link    America/New_York        SystemV/EST5EDT
+Link    America/Chicago         SystemV/CST6CDT
+Link    America/Denver          SystemV/MST7MDT
+Link    America/Los_Angeles     SystemV/PST8PDT
+Link    America/Anchorage       SystemV/YST9YDT
+Link    America/Puerto_Rico     SystemV/AST4
+Link    America/Panama          SystemV/EST5
+Link    America/Regina          SystemV/CST6
+Link    America/Phoenix         SystemV/MST7
+Link    Pacific/Pitcairn        SystemV/PST8
+Link    Pacific/Gambier         SystemV/YST9
+Link    Pacific/Honolulu        SystemV/HST10
