Description: add suexec-custom to the build system
Forwarded: not-needed
Author: Stefan Fritsch <sf@debian.org>
Last-Update: 2012-02-25
Index: apache2/Makefile.in
===================================================================
--- apache2.orig/Makefile.in
+++ apache2/Makefile.in
@@ -237,14 +237,16 @@
 	fi
 
 install-suexec:
-	@if test -f $(builddir)/support/suexec; then \
+	@if test -f $(builddir)/support/suexec-pristine && test -f $(builddir)/support/suexec-custom; then \
             test -d $(DESTDIR)$(sbindir) || $(MKINSTALLDIRS) $(DESTDIR)$(sbindir); \
-            $(INSTALL_PROGRAM) $(top_builddir)/support/suexec $(DESTDIR)$(sbindir); \
-            chmod 4755 $(DESTDIR)$(sbindir)/suexec; \
+            $(INSTALL_PROGRAM) $(top_builddir)/support/suexec-pristine $(DESTDIR)$(sbindir); \
+            chmod 4755 $(DESTDIR)$(sbindir)/suexec-pristine; \
+            $(INSTALL_PROGRAM) $(top_builddir)/support/suexec-custom $(DESTDIR)$(sbindir); \
+            chmod 4755 $(DESTDIR)$(sbindir)/suexec-custom; \
 	fi
 
 suexec:
-	cd support && $(MAKE) suexec
+	cd support && $(MAKE) suexec-pristine suexec-custom
 
 x-local-distclean:
 	@rm -rf autom4te.cache
Index: apache2/support/Makefile.in
===================================================================
--- apache2.orig/support/Makefile.in
+++ apache2/support/Makefile.in
@@ -1,7 +1,7 @@
 DISTCLEAN_TARGETS = apxs apachectl dbmmanage log_server_status \
 	logresolve.pl phf_abuse_log.cgi split-logfile envvars-std
 
-CLEAN_TARGETS = suexec
+CLEAN_TARGETS = suexec-pristine suexec-custom
 
 bin_PROGRAMS = htpasswd htdigest htdbm ab logresolve httxt2dbm
 sbin_PROGRAMS = htcacheclean rotatelogs $(NONPORTABLE_SUPPORT)
@@ -65,9 +65,13 @@
 checkgid: $(checkgid_OBJECTS)
 	$(LINK) $(checkgid_LTFLAGS) $(checkgid_OBJECTS) $(PROGRAM_LDADD)
 
-suexec_OBJECTS = suexec.lo
-suexec: $(suexec_OBJECTS)
-	$(LINK) $(suexec_OBJECTS)
+suexec-pristine_OBJECTS = suexec.lo
+suexec-pristine: $(suexec-pristine_OBJECTS)
+	$(LINK) $(suexec-pristine_OBJECTS)
+
+suexec-custom_OBJECTS = suexec-custom.lo
+suexec-custom: $(suexec-custom_OBJECTS)
+	$(LINK) $(suexec-custom_OBJECTS)
 
 htcacheclean_OBJECTS = htcacheclean.lo
 htcacheclean: $(htcacheclean_OBJECTS)
