Description: Remove LD_LIBRARY_PATH from envvars-std
Forwarded: no
Author: Adam Conrad <adconrad@0c3.net>
Last-Update: 2012-04-15
Index: apache2/support/envvars-std.in
===================================================================
--- apache2.orig/support/envvars-std.in
+++ apache2/support/envvars-std.in
@@ -18,11 +18,4 @@
 #
 # This file is generated from envvars-std.in
 #
-if test "x$@SHLIBPATH_VAR@" != "x" ; then
-  @SHLIBPATH_VAR@="@exp_libdir@:$@SHLIBPATH_VAR@"
-else
-  @SHLIBPATH_VAR@="@exp_libdir@"
-fi
-export @SHLIBPATH_VAR@
-#
 @OS_SPECIFIC_VARS@
