From: Kai Wasserbäch <curan@debian.org>
Subject: Make building of the demos optional.
Last-Update: 2012-03-14
Forwarded: not-needed
---
 CMakeLists.txt |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -211,4 +211,6 @@ configure_file( config.h.cmake ${CMAKE_B
 add_subdirectory( src )
 add_subdirectory( rc )
 add_subdirectory( special-icons )
-add_subdirectory( demo )
+if( NOT DEBIAN_NO_DEMOS )
+  add_subdirectory( demo )
+endif( NOT DEBIAN_NO_DEMOS )
