From: Cosme Domínguez <cosme.ddiaz@gmail.com>
Subject: we don't want Makefiles, in debian directory, linked to build system of upstream tarball
Forwarded: no

---
 Makefile.am  |    2 +-
 configure.in |    2 --
 2 files changed, 1 insertion(+), 3 deletions(-)

--- codeblocks.orig/Makefile.am
+++ codeblocks/Makefile.am
@@ -1,4 +1,4 @@
-SUBDIRS = src debian
+SUBDIRS = src
 
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = codeblocks.pc
--- codeblocks.orig/configure.in
+++ codeblocks/configure.in
@@ -484,8 +484,6 @@ AC_CONFIG_FILES([Makefile
 	codeblocks.pc
 	codeblocks.spec
 	codeblocks.plist
-	debian/Makefile
-	debian/lintian/Makefile
 ])
 
 AC_OUTPUT
