From 1fe9ccc25af441895e126a54af89f33f01948c91 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Nicolas=20L=C3=A9cureuil?= <neoclust.kde@free.fr>
Date: Thu, 28 Jun 2012 17:04:21 +0200
Subject: [PATCH] don't reset the default options in CXXFLAGS when building
 ffmpegthumbs

---
 CMakeLists.txt |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ffmpegthumbs/CMakeLists.txt b/ffmpegthumbs/CMakeLists.txt
index 4c4d67f..dd0974f 100644
--- a/ffmpegthumbs/CMakeLists.txt
+++ b/ffmpegthumbs/CMakeLists.txt
@@ -6,7 +6,7 @@ include_directories(
     )
 
 # Certain versions of FFMPEG need this to be defined
-SET( CMAKE_CXX_FLAGS "-D __STDC_CONSTANT_MACROS" )
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -D __STDC_CONSTANT_MACROS")
 
 set( ffmpegthumbs_PART_SRCS
   ffmpegthumbnailer.cpp
-- 
1.7.10

