Index: acpica-unix/source/compiler/dtio.c
===================================================================
--- acpica-unix.orig/source/compiler/dtio.c	2013-01-31 15:25:44.000000000 -0700
+++ acpica-unix/source/compiler/dtio.c	2013-01-31 15:25:44.000000000 -0700
@@ -426,7 +426,7 @@
     UINT32                  State = DT_NORMAL_TEXT;
     UINT32                  CurrentLineOffset;
     UINT32                  i;
-    char                    c;
+    int                     c;
 
 
     for (i = 0; ;)
@@ -440,7 +440,7 @@
             UtExpandLineBuffers ();
         }
 
-        c = (char) getc (Handle);
+        c = getc (Handle);
         if (c == EOF)
         {
             switch (State)
