Description: Add ./xaw3dxft directory to the include path.
Forwarded: http://sourceforge.net/tracker/?func=detail&aid=3029113&group_id=63290&atid=503458
Author: Josué Abarca <jmaslibre@debian.org.gt>
Last-Update: 2010-07-14

--- a/Imakefile
+++ b/Imakefile
@@ -8,6 +8,8 @@
 #include "./Local.config"
 #include "./Local.xawdefs"
 
+EXTRA_INCLUDES = -I./xaw3dxft
+
 DEFINES = $(ARCH_DEFINES) $(EXTRA_DEFINES) \
           $(XAWLIB_DEFINES) $(XPM_INCLUDE) $(XFT_INCLUDE) \
           "-DEDITOR=\"$(EDITOR)\"" \
