Description: Use sensible-editor instead of emacs

Forwarded: not-needed
Author: Josue Abarca <jmaslibre@debian.org.gt>
Last-Update: 2010-01-27

--- a/share/bin/xpaint_ocr
+++ b/share/bin/xpaint_ocr
@@ -18,19 +18,8 @@
 # adapt to whatever X pager or X editor is available
 # we strongly recommend good old 'xless' ...
 #
-if test `which xless` != ""
-then
-  if test `ps aux | grep -e [x]less | grep -c "xpaint_ocr_output.txt"` = "0"
-  then
-    xless -fn 10x20 -f $HOME/xpaint_ocr_output.txt &
-  fi
-elif test `which emacs` != ""
-then
-   emacs -fn 10x20 -f $HOME/xpaint_ocr_output.txt &
-elif test `which xterm` != ""
-then
-   xterm -e less $HOME/xpaint_ocr_output.txt &
-fi
+
+x-terminal-emulator -e sensible-editor xpaint_ocr_output.txt &
 
 rm -f $HOME/xpaint_output.ppm
 
