Author: Julien Lavergne <gilir@ubuntu.com>
Description: Use x-terminal-emulator for terminal by default.

diff -Naurp libfm.orig/data/libfm.conf libfm/data/libfm.conf
--- libfm.orig/data/libfm.conf	2012-04-16 07:18:19.385800480 +0000
+++ libfm/data/libfm.conf	2012-04-16 11:49:29.205006132 +0000
@@ -2,6 +2,7 @@
 single_click=0
 use_trash=1
 confirm_del=1
+terminal=x-terminal-emulator %s
 thumbnail_local=1
 thumbnail_max=2048
 
