diff --git a/editline.c b/editline.c
index c8e0a2b..25815e9 100644
--- a/editline.c
+++ b/editline.c
@@ -1024,9 +1024,11 @@ read_redirected()
 
     for (size = MEM_INC, p = line = NEW(char, size), end = p + size; ; p++) {
 	if (p == end) {
+	    int lengthSoFar = p - line;
 	    size += MEM_INC;
-	    p = line = realloc(line, size);
-	    end = p + size;
+	    line = realloc(line, size);
+	    p = line + lengthSoFar;
+	    end = line + size;
 	}
 	if (read(0, p, 1) <= 0) {
 	    /* Ignore "incomplete" lines at EOF, just like we do for a tty. */
