Index: smuxi/src/Engine/Config/Config.cs
===================================================================
--- smuxi.orig/src/Engine/Config/Config.cs	2012-06-30 16:06:14.000000000 +0200
+++ smuxi/src/Engine/Config/Config.cs	2012-06-30 16:13:36.000000000 +0200
@@ -325,6 +325,7 @@
             Get(prefix + "OnConnectCommands",
                 new string[] {
                     "/join #smuxi",
+                    "/join #debian",
                 }
             );
             
