Index: smuxi/src/Engine/Config/Config.cs
===================================================================
--- smuxi.orig/src/Engine/Config/Config.cs	2012-06-30 16:14:04.000000000 +0200
+++ smuxi/src/Engine/Config/Config.cs	2012-06-30 16:16:33.000000000 +0200
@@ -363,6 +363,12 @@
             Get(prefix + "Password", String.Empty);
             Get(prefix + "UseEncryption", false);
             Get(prefix + "ValidateServerCertificate", false);
+            Get(prefix + "OnStartupConnect", true);
+            Get(prefix + "OnConnectCommands",
+                new string[] {
+                    "/join #ubuntu",
+                }
+            );
             
             prefix = "Engine/Users/";
             Get(prefix+"Users", new string[] { "local" });
