Description: Add missing includes for std::vector
Author: Sven Eckelmann <sven@narfation.org>

---
diff --git a/codecs/pdf.hh b/codecs/pdf.hh
index 6beec621f75e18d4438866d80c8c41a0fd382f8f..0bf3f553494c5ae0ffcd1fe7ea4b7c6dc71a1145 100644
--- a/codecs/pdf.hh
+++ b/codecs/pdf.hh
@@ -16,6 +16,8 @@
  * copyright holder ExactCODE GmbH Germany.
  */
 
+#include <vector>
+
 #include "Codecs.hh"
 
 struct PDFContext; // fwd
diff --git a/codecs/xpm.cc b/codecs/xpm.cc
index 7736e3e70e1000a181c4e018e8595ebaf42e7cfa..0bf04b2ae2980acca0a5f9feb22d14fca37f8543 100644
--- a/codecs/xpm.cc
+++ b/codecs/xpm.cc
@@ -19,6 +19,7 @@
 #include <iostream>
 #include <string>
 #include <sstream>
+#include <vector>
 
 #include "xpm.hh"
 
diff --git a/lib/Colorspace.cc b/lib/Colorspace.cc
index c5ee5b303832d9b97b23cb72d6155b77e1cc9c59..ee517092ea6d93e90df8257133fda56bc6025b98 100644
--- a/lib/Colorspace.cc
+++ b/lib/Colorspace.cc
@@ -19,6 +19,7 @@
 
 #include <iostream>
 #include <map>
+#include <vector>
 
 #include "Image.hh"
 #include "ImageIterator2.hh"
