Description: Set PackageKit default config
Forwarded: not-needed
Author: Matthias Klumpp <matthias@tenstral.net>
Last-Update: 2011-07-14
--- a/etc/PackageKit.conf.in
+++ b/etc/PackageKit.conf.in
@@ -221,7 +221,7 @@
 # packages in one method.
 #
 # default=1200
-MaximumItemsToResolve=1200
+MaximumItemsToResolve=4000
 
 # The maximum number of packages that can be processed in one method
 #
@@ -229,7 +229,7 @@
 # cause errors if the desktop client is trying to do a large transaction.
 #
 # default=2500
-MaximumPackagesToProcess=2500
+MaximumPackagesToProcess=4000
 
 # How long the transaction is valid before it's destroyed, in seconds
 #
@@ -285,6 +285,8 @@
 # Separate entries can be separated with the ';' character.
 #
 # default=/usr/lib*/*/firefox
+# NOTE: On Debian, updating of Firefox/Iceweasel while running should work.
+#       If it does not, please file a bug report.
 NoUpdateProcessList=
 
 # If a custom root should be used when installing packages
