--- acpi-support-0.138/asus-touchpad.sh
+++ acpi-support-0.138/asus-touchpad.sh
@@ -1,14 +1,27 @@
 #!/bin/sh
-[ -f /usr/share/acpi-support/state-funcs ] || exit 0 
 
-. /usr/share/acpi-support/power-funcs
+set -e
+
+pff=/usr/share/acpi-support/power-funcs
+[ -f $pff ] || exit 0
+
+atp_error() {
+	logger -t${0##*/} -perr -- $*
+	exit 1
+}
+
+. $pff || atp_error "Sourcing $pff failed"
+
+[ -x /usr/bin/xinput ] || atp_error "Please install package xinput to enable toggling of touchpad devices."
 
 getXconsole
 
 getTouchDeviceId()
 {
     # extract the device id for the supplied touch device name
-    xinput list | sed -nr "s|.*$1.*id=([0-9]+).*|\1|p"
+    # XXX:	according to man page 'list', 'list-props' and 'set-int-prop' are
+    #	options, not arguments.
+    xinput --list | sed -nr "s|.*$1.*id=([0-9]+).*|\1|p"
 }
 
 ENABLEPROP="Synaptics Off"
@@ -21,16 +34,25 @@
 fi
 
 # if we failed to get an input, exit
-[ -z "$XINPUTNUM" ] && exit 1
+[ "$XINPUTNUM" ] || atp_error "Invalid TouchPad id '$XINPUTNUM'"
 
 # get the current state of the touchpad
-TPSTATUS=$(xinput list-props $XINPUTNUM | awk "/$ENABLEPROP/ { print \$NF }")
-
-# if getting the status failed, exit
-[ -z "$TPSTATUS" ] && exit 1
-
-if [ $TPSTATUS = 0 ]; then
-   xinput set-int-prop $XINPUTNUM "$ENABLEPROP" 8 1
+TPSTATUS=$(xinput --list-props $XINPUTNUM | awk "/$ENABLEPROP/ { print \$NF }")
+case $TPSTATUS in
+	[!01])
+		atp_error "Invalid TouchPad status '$TPSTATUS'"
+		;;
+esac
+
+# XXX: '--set-int-prop' deprecated
+xcmd="xinput --set-int-prop $XINPUTNUM '$ENABLEPROP' 8"
+ledf=/sys/class/leds/asus::touchpad/brightness
+if [ $TPSTATUS -eq 0 ]; then
+	eval $xcmd 1 || atp_error "Command '$xcmd 1' failed"
+	[ ! -w $ledf ] || echo 0 >$ledf ||
+		atp_error "Writing 0 to $ledf failed"
 else
-   xinput set-int-prop $XINPUTNUM "$ENABLEPROP" 8 0
+	eval $xcmd 0 || atp_error "Command '$xcmd 0' failed"
+	[ ! -w $ledf ] || echo 1 >$ledf ||
+		atp_error "Writing 0 to $ledf failed"
 fi
