--- acpi-support-0.138/asus-wireless.sh
+++ acpi-support-0.138/asus-wireless.sh
@@ -5,4 +5,24 @@
 
 . /usr/share/acpi-support/state-funcs
 
-toggleAllWirelessStates
+read vendor </sys/class/dmi/id/sys_vendor 2>/dev/null || exit 0
+case $vendor in
+	[As][Ss][Uu][Ss]*)
+		;;
+	*)
+		exit 0
+		;;
+esac
+
+if [ "$1" = "" ] ; then
+	toggleAllWirelessStates;
+elif isAnyWirelessPoweredOn; then
+	if [ "$1" = "off" ] ; then
+		toggleAllWirelessStates;
+	fi
+else
+	if [ "$1" = "on" ] ; then
+		toggleAllWirelessStates;
+	fi
+fi
+
