--- acpi-support/lib/device-funcs	2010-12-14 11:37:17.000000000 +0100
+++ acpi-support/lib/device-funcs	2010-12-14 21:24:06.000000000 +0100
@@ -1,10 +1,7 @@
 
 DeviceConfig() {
-
-    # Use a cached copy of the DMI data if it is available as we may
-    # not be running with privileges necessary to access /dev/mem
-    manufacturer=`cat /var/lib/acpi-support/system-manufacturer||dmidecode --string system-manufacturer|sed -e 's/ *$//'`
-    model=`cat /var/lib/acpi-support/system-product-name||dmidecode --string system-product-name|sed -e 's/ *$//'`
-    version=`cat /var/lib/acpi-support/system-version||dmidecode --string system-version|sed -e 's/ *$//'`
-    bios_version=`cat /var/lib/acpi-support/bios-version||dmidecode --string bios-version|sed -e 's/ *$//'`
+    read manufacturer </sys/class/dmi/id/sys_vendor
+    read model </sys/class/dmi/id/product_name
+    read version </sys/class/dmi/id/product_version
+    read bios_version </sys/class/dmi/id/bios_version
 }
--- acpi-support/thinkpad-stretchortouchpad.sh.orig	2010-12-19 23:32:58.000000000 +0100
+++ acpi-support/thinkpad-stretchortouchpad.sh	2010-12-25 18:23:56.000000000 +0100
@@ -17,10 +17,10 @@
 
 toggle_touchpad=0
 
-system_manufactuer=`dmidecode -s system-manufacturer`
+read system_manufactuer </sys/class/dmi/id/sys_vendor
 case "$system_manufactuer" in
     IBM*)
-    system_version=`dmidecode -s system-version`
+    read system_version </sys/class/dmi/id/product_version
     case "$system_version" in
 	ThinkPad\ [TXZ]60*)
 	toggle_touchpad=1
