--- acpi-support-0.140/lid.sh
+++ acpi-support-0.140/lid.sh
@@ -1,5 +1,4 @@
-#!/bin/bash
-# TODO:  Change the above to /bin/sh
+#!/bin/sh
 
 test -f /usr/share/acpi-support/state-funcs || exit 0
 
@@ -9,39 +8,46 @@
 
 [ -x /etc/acpi/local/lid.sh.pre ] && /etc/acpi/local/lid.sh.pre
 
-if [ `CheckPolicy` = 0 ]; then exit; fi
+CheckPolicy && exit
 
 grep -q closed /proc/acpi/button/lid/*/state
 if [ $? = 0 ]
 then
-    for x in /tmp/.X11-unix/*; do
-	displaynum=`echo $x | sed s#/tmp/.X11-unix/X##`
-	getXuser;
-	if [ x"$XAUTHORITY" != x"" ]; then
-	    export DISPLAY=":$displaynum"	    
-	    . /usr/share/acpi-support/screenblank
-	fi
-    done
+    . /usr/share/acpi-support/screenblank
+    if [ x$LID_SLEEP = xtrue ]; then
+	pm-suspend
+    fi
 else
-    for x in /tmp/.X11-unix/*; do
-	displaynum=`echo $x | sed s#/tmp/.X11-unix/X##`
+    d=/tmp/.X11-unix
+    for x in $d/X*; do
+	displaynum=${x#$d/X}
 	getXuser;
-	if [ x"$XAUTHORITY" != x"" ]; then
-	    export DISPLAY=":$displaynum"
-	    grep -q off-line /proc/acpi/ac_adapter/*/state
-	    if [ $? = 1 ]
-		then
-		if pidof xscreensaver > /dev/null; then 
-		    su $user -c "xscreensaver-command -unthrottle"
-		fi
-	    fi
+	if [ x$XAUTHORITY != x ]; then
+	    export DISPLAY=:$displaynum
 	    if [ x$RADEON_LIGHT = xtrue ]; then
 		[ -x /usr/sbin/radeontool ] && radeontool light on
 	    fi
-	    if [ `pidof xscreensaver` ]; then
-		su $user -c "xscreensaver-command -deactivate"
+	    case "$DISPLAY_DPMS" in
+		xset)
+			su $XUSER -s /bin/sh -c "xset dpms force on"
+			;;
+		xrandr)
+			su $XUSER -s /bin/sh -c "xrandr --output $XRANDR_OUTPUT --auto"
+			;;
+		vbetool)
+			/usr/sbin/vbetool dpms on
+			;;
+	    esac
+	    if pidof xscreensaver > /dev/null; then
+	        if on_ac_power; then 
+		    su $XUSER -s /bin/sh -c "xscreensaver-command -unthrottle"
+		fi
+		su $XUSER -s /bin/sh -c "xscreensaver-command -deactivate"
 	    fi
-	    su $user -c "xset dpms force on"
+	else
+	    if [ -x$DISPLAY_DPMS_NO_USER = xtrue ]; then
+            	[ -x /usr/sbin/vbetool ] && /usr/sbin/vbetool dpms on
+	    fi 
 	fi
     done
 fi
