--- acpi-support/rotatescreen.sh
+++ acpi-support/rotatescreen.sh
@@ -8,7 +8,7 @@
 . /usr/share/acpi-support/power-funcs
 
 if [ -f /var/lib/acpi-support/screen-rotation ] ; then
-  ROTATION=`cat /var/lib/acpi-support/screen-rotation`
+	read ROTATION </var/lib/acpi-support/screen-rotation
 fi
 
 case "$ROTATION" in
@@ -20,12 +20,27 @@
 	;;
 esac
 
-for x in /tmp/.X11-unix/*; do
-	displaynum=`echo $x | sed s#/tmp/.X11-unix/X##`
+d=/tmp/.X11-unix
+for x in $d/X*; do
+	displaynum=${x#$d/X}
 	getXuser;
-	if [ x"$XAUTHORITY" != x"" ]; then
-	    export DISPLAY=":$displaynum"
+	if [ x$XAUTHORITY != x ]; then
+	    export DISPLAY=:$displaynum
 	    /usr/bin/xrandr -o $NEW_ROTATION && echo $NEW_ROTATION > /var/lib/acpi-support/screen-rotation
+	    if [ -x /usr/bin/xsetwacom ]; then
+	        OIFS=$IFS
+	        IFS='
+'
+	        WACOMDEVICES=`xsetwacom --list | awk NF--`
+	        for device in $WACOMDEVICES; do
+	            if [ "$NEW_ROTATION" = "normal" ]; then
+	                xsetwacom set "$device" rotate NONE
+	            else
+	                xsetwacom set "$device" rotate CW
+	            fi
+	        done
+	        IFS=$OIFS
+	    fi
 	fi
 done
 
