--- acpi-support-0.138/screenblank.sh.orig	2011-01-08 00:14:47.000000000 +0100
+++ acpi-support-0.138/screenblank.sh	2011-01-08 01:23:15.000000000 +0100
@@ -4,12 +4,5 @@
 
 . /etc/default/acpi-support
 . /usr/share/acpi-support/power-funcs
 
-for x in /tmp/.X11-unix/*; do
-    displaynum=`echo $x | sed s#/tmp/.X11-unix/X##`
-    getXuser;
-    if [ x"$XAUTHORITY" != x"" ]; then
-        export DISPLAY=":$displaynum"
-	. /usr/share/acpi-support/screenblank
-    fi
-done
+. /usr/share/acpi-support/screenblank
