From a8e97f672325557b3fbc84b987299350ec5ac10b Mon Sep 17 00:00:00 2001
From: Rex Dieter <rdieter@fedoraproject.org>
Date: Mon, 9 Jan 2012 11:13:25 -0600
Subject: [PATCH] gcc47 fix, unistd.h header required unconditionally for
 'sleep'

---
 bin/daemon/eventlistener/eventlistenerqueue.cpp |    2 --
 1 file changed, 2 deletions(-)

diff --git a/strigidaemon/bin/daemon/eventlistener/eventlistenerqueue.cpp b/strigidaemon/bin/daemon/eventlistener/eventlistenerqueue.cpp
index 9fcdcb1..1e97599 100644
--- a/strigidaemon/bin/daemon/eventlistener/eventlistenerqueue.cpp
+++ b/strigidaemon/bin/daemon/eventlistener/eventlistenerqueue.cpp
@@ -27,9 +27,7 @@
 #include <iostream>
 #include <stdio.h>
 
-#if defined(__SUNPRO_CC)
 #include <unistd.h>
-#endif
 
 using namespace std;
 
-- 
1.7.10

