diff -c 'harker/checksendmail' 'checksendmail'
Index: ./harker/checksendmail
Prereq:  1.1.1.1 
*** ./harker/checksendmail	Mon Apr 14 15:44:43 2003
--- ./checksendmail	Mon Apr 14 15:57:56 2003
***************
*** 25,30 ****
--- 25,38 ----
  	$resolve  = "address.resolve";
  	$cffile   = "/etc/mail/sendmail.cf";
  
+ # Pre-create tmp files (so can use safely)
+ 	use Fcntl;
+ 	foreach ("/tmp/csm$$", "/tmp/csm.in$$") {
+ 		die "Can't pre-create $_" 
+ 			unless sysopen(T,$_,O_RDWR|O_CREAT|O_EXCL,0600)
+ 			and close(T);
+ 	};
+ 
  sub usage {
  	die "Usage: checksendmail [-C cffile.cf] [-b sendmail_binary] [-l log_file]\n\t\t\t\t[-a] [-d] [-r resolve_addrs_file || -T one\@test.addr]\n$errormsg\n";
  }
