Author: Rickard Närström <rickard.narstrom@gmail.com>
Subject: Hide symbols by default

Index: ffms2/Makefile.am
===================================================================
--- ffms2.orig/Makefile.am
+++ ffms2/Makefile.am
@@ -1,6 +1,9 @@
 ACLOCAL_AMFLAGS = -I m4
 AUTOMAKE_OPTIONS = foreign
 
+AM_CFLAGS=-fvisibility=hidden
+AM_CXXFLAGS=-fvisibility=hidden
+
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = ffms2.pc
 
