Author: Reinhard Tartler <siretart@tauware.de>
Description: Unbreak compilation against Libav 0.8
Forwarded-Upstream: not-needed

--- a/src/core/codectype.cpp
+++ b/src/core/codectype.cpp
@@ -96,7 +96,7 @@ typedef struct AVCodecTag {
 	unsigned int tag;
 } AVCodecTag;
 
-#if VERSION_CHECK(LIBAVFORMAT_VERSION_INT, <, 54, 1, 0, 53, 32, 100)
+#if 0
 static const AVCodecTag codec_bmp_tags[] = {
 	{ CODEC_ID_H264,         MKTAG('H', '2', '6', '4') },
 	{ CODEC_ID_H264,         MKTAG('h', '2', '6', '4') },
--- a/src/core/utils.cpp
+++ b/src/core/utils.cpp
@@ -566,7 +566,10 @@ void FlushBuffers(AVCodecContext *CodecC
 		// If the codec doesn't have flush(), it might not need it... or it
 		// might need it and just not implement it as in the case of VC-1, so
 		// close and reopen the codec
-		const AVCodec *codec = CodecContext->codec;
+#if (LIBAVCODEC_VERSION_INT) >= (AV_VERSION_INT(54, 0, 0))
+		const
+#endif
+		AVCodec *codec = CodecContext->codec;
 		avcodec_close(CodecContext);
 		avcodec_open2(CodecContext, codec, 0);
 	}
