GNOME #670493

Index: dconf-0.12.1/bin/dconf-update.vala
===================================================================
--- dconf-0.12.1.orig/bin/dconf-update.vala	2012-03-19 16:36:21.000000000 +0100
+++ dconf-0.12.1/bin/dconf-update.vala	2012-06-09 17:55:44.499132517 +0200
@@ -241,7 +241,16 @@ void update_all (string dirname) {
 }
 
 void dconf_update (string[] args) {
-	update_all ("/etc/dconf/db");
+	if (args[2] != null) {
+		update_all (args[2]);
+	} else {
+		var dirname = GLib.Environment.get_variable ("DCONF_DIR"
+);
+		if (dirname == null) {
+			dirname = "/etc/dconf";
+		}
+		update_all (Path.build_filename (dirname, "db"));
+	}
 }
 
 // vim:noet ts=4 sw=4
Index: dconf-0.12.1/bin/dconf.vala
===================================================================
--- dconf-0.12.1.orig/bin/dconf.vala	2012-04-10 14:36:04.000000000 +0200
+++ dconf-0.12.1/bin/dconf.vala	2012-06-09 17:48:31.661016471 +0200
@@ -56,7 +56,7 @@ void show_help (bool requested, string?
 
 		case "update":
 			description = "Update the system dconf databases";
-			synopsis = "";
+			synopsis = "[DIR]";
 			break;
 
 		case "watch":
@@ -119,7 +119,9 @@ Use 'dconf help COMMAND' to get detailed
 				str.append ("  KEY       A key path (starting, but not ending with '/')\n");
 			}
 
-			if ("PATH" in synopsis || "DIR" in synopsis) {
+			if ("[DIR]" in synopsis ) {
+				str.append ("  DIR       Directory containing the databases to update\n");
+			} else if ("PATH" in synopsis || "DIR" in synopsis) {
 				str.append ("  DIR       A directory path (starting and ending with '/')\n");
 			}
 
Index: dconf-0.12.1/engine/dconf-engine.c
===================================================================
--- dconf-0.12.1.orig/engine/dconf-engine.c	2012-04-12 17:24:29.000000000 +0200
+++ dconf-0.12.1/engine/dconf-engine.c	2012-06-09 18:01:24.848796399 +0200
@@ -181,7 +181,10 @@ dconf_engine_refresh_system (DConfEngine
 
   if (engine->gvdbs[i] == NULL)
     {
-      gchar *filename = g_build_filename ("/etc/dconf/db",
+      const gchar *dconf_dir = g_getenv ("DCONF_DIR");
+      if (dconf_dir == NULL)
+        dconf_dir = "/etc/dconf";
+      gchar *filename = g_build_filename (dconf_dir, "db",
                                           engine->names[i], NULL);
       engine->gvdbs[i] = gvdb_table_new (filename, TRUE, NULL);
       if (engine->gvdbs[i] == NULL)
@@ -247,7 +250,12 @@ dconf_engine_load_profile (const gchar
 
   /* DCONF_PROFILE starting with '/' gives an absolute path to a profile */
   if (profile[0] != '/')
-    filename = g_build_filename ("/etc/dconf/profile", profile, NULL);
+    {
+      const gchar *dconf_dir = g_getenv ("DCONF_DIR");
+      if (dconf_dir == NULL)
+        dconf_dir = "/etc/dconf";
+      filename = g_build_filename (dconf_dir, "profile", profile, NULL);
+    }
   else
     filename = g_strdup (profile);
 
