Description: fix spelling mistakes
Origin: vendor
Forwarded: no
Author: gregor herrmann <gregoa@debian.org>
Last-Update: 2011-02-14

--- a/Killall.pm
+++ b/Killall.pm
@@ -91,7 +91,7 @@
 
 C<killall> searches the process table and sends that signal to all processes
 which match the pattern. The return value is the number of processes that
-were succesfully signaled. If any kills failed, the C<$!> variable
+were successfully signaled. If any kills failed, the C<$!> variable
 will be set based on that last one that failed (even if a successful kill
 happened afterward).
 
--- a/ProcessTable.pm
+++ b/ProcessTable.pm
@@ -183,7 +183,7 @@
 systems, can be quite large and/or contain invalid device paths (for example,
 Solaris does not clean up invalid device entries when disks are swapped).  If
 this is specified with cache_ttys, a warning is generated and the cache_ttys
-is overriden to be false.
+is overridden to be false.
 
 cache_ttys -- causes the constructor to look for and use a file that
 caches a mapping of tty names to device numbers, and to create the
