Prevent searches with empty search string.

--- a/dist/mediawiki-extensions-ldapauth/usr/share/mediawiki-extensions/ldapauth/LdapAuthentication.php
+++ b/dist/mediawiki-extensions-ldapauth/usr/share/mediawiki-extensions/ldapauth/LdapAuthentication.php
@@ -497,7 +497,10 @@ class LdapAuthenticationPlugin extends A
 				$ret = true;
 			} else {
 				// Search for the entry.
+				$entry = false;
+			      if ($searchstring != '') {
 				$entry = LdapAuthenticationPlugin::ldap_read( $this->ldapconn, $searchstring, "objectclass=*" );
+			      }
 
 				if ( $entry ) {
 					$this->printDebug( "Found a matching user in LDAP", NONSENSITIVE );
