Description: remove encoding from desktop file
Author: Varun Hiremath <vh63@cornell.edu>
Date: Sun,  9 Aug 2009 19:30:19 -0400
Index: agave-0.4.7/data/agave.desktop.in
===================================================================
--- agave-0.4.7.orig/data/agave.desktop.in	2011-09-21 16:50:06.000000000 -0400
+++ agave-0.4.7/data/agave.desktop.in	2011-09-21 16:51:02.000000000 -0400
@@ -1,9 +1,8 @@
 [Desktop Entry]
 Version=1.0
-Encoding=UTF-8
 Type=Application
 _Name=Agave
 _Comment=Generate color schemes
 Icon=agave
 Exec=agave
-Categories=GNOME;Application;Graphics
+Categories=GNOME;Graphics;
Index: agave-0.4.7/data/agave.desktop.in.in
===================================================================
--- agave-0.4.7.orig/data/agave.desktop.in.in	2011-09-21 16:50:06.000000000 -0400
+++ agave-0.4.7/data/agave.desktop.in.in	2011-09-21 16:50:10.000000000 -0400
@@ -1,6 +1,5 @@
 [Desktop Entry]
 Version=1.0
-Encoding=UTF-8
 Type=Application
 _Name=Agave
 _Comment=Generate color schemes
