Index: vlc-2.0.1/modules/access/v4l2/v4l2.h
===================================================================
--- vlc-2.0.1.orig/modules/access/v4l2/v4l2.h	2012-02-28 23:44:03.000000000 +0100
+++ vlc-2.0.1/modules/access/v4l2/v4l2.h	2012-03-19 16:16:38.973394122 +0100
@@ -39,6 +39,10 @@
 #  define V4L2_CTRL_TYPE_BITMASK 8
 # endif
 #endif
+#if defined(__FreeBSD__) || defined(__FreeBSD_kernel__)
+/* FreeBSD 10 and earlier does not have V4L2_CTRL_TYPE_BITMASK */
+# define V4L2_CTRL_TYPE_BITMASK 8
+#endif
 #ifndef V4L2_CID_ILLUMINATORS_1 /* 2.6.37 */
 # define V4L2_CID_ILLUMINATORS_1 (V4L2_CID_BASE+38)
 # define V4L2_CID_ILLUMINATORS_2 (V4L2_CID_BASE+37)
