Index: vlc-2.0.0/bin/Makefile.am
===================================================================
--- vlc-2.0.0.orig/bin/Makefile.am	2012-03-04 12:07:13.332965394 +0100
+++ vlc-2.0.0/bin/Makefile.am	2012-03-04 12:42:29.421956212 +0100
@@ -16,7 +16,7 @@
 if !HAVE_WIN32
 if !HAVE_WINCE
 bin_PROGRAMS += vlc-wrapper
-vlc_SOURCES = vlc.c override.c
+vlc_SOURCES = vlc.c override.c dummy.cpp
 endif
 endif
 
@@ -59,7 +59,7 @@
 vlc_win32_rc.$(OBJEXT): vlc_win32_rc.rc
 	$(WINDRES) --include-dir $(top_srcdir)/share/icons -i $< -o $@
 
-vlc_cache_gen_SOURCES = cachegen.c
+vlc_cache_gen_SOURCES = cachegen.c dummy.cpp
 vlc_cache_gen_LDADD = \
 	$(GNUGETOPT_LIBS) \
 	../compat/libcompat.la \
Index: vlc-2.0.0/bin/dummy.cpp
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ vlc-2.0.0/bin/dummy.cpp	2012-03-04 12:42:49.002636671 +0100
@@ -0,0 +1 @@
+/* This file is only here to force cachegen to be linked against libstdc++ */
