Author: Andreas Beckmann <debian@abeckmann.de>
Description: fix a typo found by lintian
 s/ouput/output/

Index: b/nv-procfs.c
===================================================================
--- a/nv-procfs.c	2011-04-26 20:57:50.958352573 +0200
+++ b/nv-procfs.c	2011-04-26 20:58:01.450352581 +0200
@@ -405,7 +405,7 @@
         if (agp_config != NVOS_AGP_CONFIG_DISABLE_AGP && NV_AGP_FAILED(nv))
         {
             len += sprintf(page+len,
-                  "AGP initialization failed, please check the ouput  \n"
+                  "AGP initialization failed, please check the output \n"
                   "of the 'dmesg' command and/or your system log file \n"
                   "for additional information on this problem.        \n");
         }
