Author: Antonio Ospite <ospite@studenti.unina.it>
Author: Andreas Beckmann <debian@abeckmann.de>
Subject: Make the nvidia module build on linux-3.0.0
Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=628925

Index: nvidia-270.41.19/Makefile
===================================================================
--- nvidia-270.41.19.orig/Makefile	2011-06-02 09:58:55.000000000 +0200
+++ nvidia-270.41.19/Makefile	2011-06-02 09:59:17.000000000 +0200
@@ -158,7 +158,7 @@
 
 # Check for kernel versions that we don't support.
 
-BELOW26 := $(shell if [ $(VERSION) -lt 2 -o $(PATCHLEVEL) -lt 6 ]; then \
+BELOW26 := $(shell if [ $(VERSION) -lt 2 -o \( $(VERSION) = 2 -a $(PATCHLEVEL) -lt 6 \) ]; then \
 		echo y; fi)
 
 ifeq ($(BELOW26),y)
