--- a/NVIDIA-Linux-x86_64-310.14/kernel/Makefile.kbuild	2012-10-09 14:52:46.000000000 -0400
+++ b/Makefile.kbuild	2012-10-16 20:50:00.324404926 -0400
@@ -43,7 +43,7 @@
 # understands that we want a module.
 #
 
-CORE_OBJS := nv-kernel.o
+CORE_OBJS := nv-kernel.o$(NVARCH)
 RESMAN_GLUE_OBJS := $(patsubst %.c,%.o,nv.c nv-acpi.c nv-chrdev.c nv-cray.c nv-gvi.c nv-i2c.c nv-mempool.c nv-mlock.c nv-mmap.c nv-p2p.c nv-pat.c nv-procfs.c nv-usermap.c nv-vm.c nv-vtophys.c os-interface.c os-registry.c os-smp.c os-usermap.c)
 
 $(MODULE_NAME)-objs := $(CORE_OBJS) $(RESMAN_GLUE_OBJS)
