diff -aur fbi-2.07.orig/GNUmakefile fbi-2.07/GNUmakefile
--- fbi-2.07.orig/GNUmakefile	2011-12-28 21:25:29.000000000 +0100
+++ fbi-2.07/GNUmakefile	2011-12-28 21:26:37.000000000 +0100
@@ -58,6 +58,7 @@
   ida fbi : LDFLAGS  += -Wl,--wrap=fopen
 endif
 
+ida fbi : LDFLAGS  += `dpkg-buildflags --get LDFLAGS`
 
 ########################################################################
 # conditional stuff
Nur in fbi-2.07: GNUmakefile~.
