--- fbi-2.07.orig/mk/Variables.mk	2011-12-27 19:22:07.000000000 +0100
+++ fbi-2.07/mk/Variables.mk	2011-12-27 19:30:12.000000000 +0100
@@ -36,11 +36,12 @@
 INSTALL_DIR	:= $(INSTALL) -d
 
 # cflags
-CFLAGS		?= -g -O2
+CFLAGS  	?= $(shell dpkg-buildflags --get CFLAGS)
 CXXFLAGS	?= $(CFLAGS)
 CFLAGS		+= -Wall -Wmissing-prototypes -Wstrict-prototypes \
 		   -Wpointer-arith -Wunused
 CXXFLAGS	+= -Wall -Wpointer-arith -Wunused
+CFLAGS 	+= $(shell dpkg-buildflags --get CPPFLAGS)
 
 # add /usr/local to the search path if something is in there ...
 ifneq ($(wildcard /usr/local/include/*.h),)
@@ -48,8 +49,16 @@
   LDFLAGS += -L/usr/local/$(LIB)
 endif
 
+LDFLAGS 	?= $(shell dpkg-buildflags --get LDFLAGS)
+
 # fixup include path for $(srcdir) != "."
 ifneq ($(srcdir),.)
   CFLAGS  += -I. -I$(srcdir)
 endif
 
+
+
+
+
+
+
