diff -aur fbida-2.05.orig/fbgs.man fbida-2.05/fbgs.man
--- fbida-2.05.orig/fbgs.man	2004-11-08 14:10:49.000000000 +0100
+++ fbida-2.06/fbgs.man	2006-08-20 15:59:05.000000000 +0200
@@ -8,9 +8,9 @@
 .B fbgs
 is a simple wrapper script which takes a PostScript or pdf
 file as input, renders the pages using ghostscript into a
-temporary directory and finally calls fbi to display them.
+temporary directory and finally calls fbi to display them.
 .SH OPTIONS
-fbps understands all fbi options (they are passed through).
+fbgs understands all fbi options (they are passed through).
 Additionally you can specify -l, -xl or -xxl to get the pages
 rendered with 100, 120 or 150 dpi (default is 75). You can
 use option -p <password> if your PDF file requires password.
