diff -aur fbi-2.07.orig/thumbnail.cgi.c fbi-2.07/thumbnail.cgi.c
--- fbi-2.07.orig/thumbnail.cgi.c	2006-06-13 14:47:24.000000000 +0200
+++ fbi-2.07/thumbnail.cgi.c	2011-12-27 19:23:52.000000000 +0100
@@ -99,7 +99,7 @@
     char *path_info;
     
     if (NULL == getenv("GATEWAY_INTERFACE")) {
-	fprintf(stderr,shellhelp);
+	fprintf(stderr, "%s", shellhelp);
 	fprintf(stderr,description,"$DOCUMENT_ROOT");
 	exit(1);
     }
