Only show gnome-terminal in GNOME menus

Index: gnome-terminal-2.30.2/gnome-terminal.desktop.in.in
===================================================================
--- gnome-terminal-2.30.2.orig/gnome-terminal.desktop.in.in	2010-06-22 20:27:42.711402866 +0200
+++ gnome-terminal-2.30.2/gnome-terminal.desktop.in.in	2010-06-22 20:27:39.947410271 +0200
@@ -12,3 +12,4 @@
 X-GNOME-Bugzilla-Version=@VERSION@
 Categories=GNOME;GTK;Utility;TerminalEmulator;
 StartupNotify=true
+OnlyShowIn=GNOME;
