--- Raster3D-2.9-1.orig/examples/example1.script
+++ Raster3D-2.9-1/examples/example1.script
@@ -1,4 +1,4 @@
-#!/bin/csh
+#!/bin/bash
 # E A Merritt January 2003
 # Illustration of 
 #	material properties 
@@ -6,5 +6,5 @@
 #	R3D_LIB environmental variable
 #	gamma correction applied while rendering
 #
-setenv R3D_LIB ../materials
+export R3D_LIB="../materials"
 render -gamma 1.1 -png < example1.r3d | display png:-
