--- Raster3D-2.9-1.orig/examples/example8.script
+++ Raster3D-2.9-1/examples/example8.script
@@ -1,9 +1,14 @@
-#!/bin/csh
+#!/bin/bash
 #
 # Demonstrate stereo3d script by making stereo pair of
 # view down pore of LT B-pentamer
 #
-if (! $?TMPDIR) setenv TMPDIR /tmp ; endif
+if [ "$TMPDIR" ]; then
+       tmp=$TMPDIR
+else
+       tmp=/tmp
+fi
+export TMPDIR=$tmp
 #
 # First we make a VDW model of the B-pentamer itself
 #
