--- Raster3D-2.9-1.orig/Makefile.template
+++ Raster3D-2.9-1/Makefile.template
@@ -8,15 +8,15 @@
 #
 # Where to install the program, data files, and manuals
 #
-prefix  = /usr/local
+prefix  = /usr
 bindir  = $(prefix)/bin
-datadir = $(prefix)/share/Raster3D/materials
-mandir  = $(prefix)/man/manl
-htmldir = $(prefix)/share/Raster3D/html
-examdir = $(prefix)/share/Raster3D/examples
+datadir = $(prefix)/share/raster3d/materials
+mandir  = $(prefix)/share/man/man1
+htmldir = $(prefix)/share/raster3d/html
+examdir = $(prefix)/share/raster3d/examples
 
 INCDIRS  =	-I/usr/include -I/usr/local/include
-LIBDIRS  =	-L/usr/local/lib
+LIBDIRS  =	-L/usr/lib
 
 #
 # TIFF support (enabled by default)
