--- Raster3D_3.0-2.orig/Makefile
+++ Raster3D_3.0-2/Makefile
@@ -51,9 +51,9 @@
 	@cp Makefile.template Makefile.incl
 	@echo OS = linux                      >> Makefile.incl
 	@echo CC = gcc                        >> Makefile.incl
-	@echo CFLAGS = -g -Wall -Dgfortran    >> Makefile.incl
+	@echo CFLAGS = -g -O2 -Wall -Dgfortran    >> Makefile.incl
 	@echo FC = gfortran                   >> Makefile.incl
-	@echo FFLAGS = -g -w -O3 -Wtabs -ffixed-line-length-132 >> Makefile.incl
+	@echo FFLAGS = -g -w -O2 -Wtabs -ffixed-line-length-132 >> Makefile.incl
 	@echo RM = /bin/rm -f                 >> Makefile.incl
 	@echo OSDEFS =  -DLINUX -DNETWORKBYTEORDER       >> Makefile.incl
 	@echo include Makefile.package        >> Makefile.incl
@@ -238,12 +238,14 @@
 install:	all
 	if [ ! -e $(prefix)  ] ; then mkdir -p $(prefix) ; fi
 	if [ ! -e $(bindir)  ] ; then mkdir -p $(bindir) ; fi
-	chmod 755 $(PROGS);   cp $(PROGS) $(bindir)
+	chmod 755 $(PROGS);
+	for i in normal3d render; do cp $$i $(bindir)/$$i.real; done
+	cp avs2ps balls label3d rastep ribbon rings3d rods stereo3d $(bindir)
 	chmod 755 $(SCRIPTS); cp $(SCRIPTS) $(bindir)
 	if [ ! -e $(datadir) ] ; then mkdir -p $(datadir) ; fi
 	cp materials/* $(datadir)
 	if [ ! -e $(mandir)  ] ; then mkdir -p $(mandir) ; fi
-	cp doc/*.l $(mandir)
+	for i in doc/*.l; do j=`basename $$i .l`; cp $$i $(mandir)/$${j}.1; done
 	if [ ! -e $(htmldir) ] ; then mkdir -p $(htmldir) ; fi
 	cp html/* $(htmldir)
 	if [ ! -e $(examdir) ] ; then mkdir -p $(examdir) ; fi
