--- Raster3D-2.9-1.orig/doc/r3d_objects.l
+++ Raster3D-2.9-1/doc/r3d_objects.l
@@ -1,9 +1,11 @@
-.l
+.ad l
 .nh
 .TH R3D_OBJECTS 1 " 8 May 1999" "Raster3D"
 .SH NAME
-Object types used by the Raster3D package
 
+rd3_objects - object types used by the Raster3D package
+
+.SH DESCRIPTION
 These are the object descriptor types, and the required
 parameters, recognized by the \fBrender\fP program and
 other components of the Raster3D package.
