--- Raster3D-2.9-1.orig/doc/rods.l
+++ Raster3D-2.9-1/doc/rods.l
@@ -5,8 +5,8 @@
 
 rods - Raster3D preprocessor for ball-and-stick models 
 .SH SYNOPSIS
-
-.HB rods [\fI-h\fP] [\fI-b\fP] [\fI-radius R\fP] \fI[-Bcolor Bmin Bmax\fP]
+.PP
+.B rods [\fI-h\fP] [\fI-b\fP] [\fI-radius R\fP] \fI[-Bcolor Bmin Bmax\fP]
 .PP
 \fIRods\fP reads a file describing atom colours and/or a 
 PDB coordinate file and produces a file containing Raster3D descriptor
