Description: Fix spelling errors in the source code.
Author: Jakub Wilk <jwilk@debian.org>
Forwarded: no
Last-Update: 2011-03-13

--- a/nls/C/blackbox.m
+++ b/nls/C/blackbox.m
@@ -1,6 +1,6 @@
 $set 13 #blackbox
 
 $ #NoManagableScreens
-# Blackbox::Blackbox: no managable screens found, aborting\n
+# Blackbox::Blackbox: no manageable screens found, aborting\n
 $ #MapRequest
 # Blackbox::process_event: MapRequest for 0x%lx\n
--- a/nls/ko_KR/blackbox.m
+++ b/nls/ko_KR/blackbox.m
@@ -1,6 +1,6 @@
 $set 13 #blackbox
 
 $ #NoManagableScreens
-# Blackbox::Blackbox: no managable screens found, aborting\n
+# Blackbox::Blackbox: no maneagable screens found, aborting\n
 $ #MapRequest
 # Blackbox::process_event: MapRequest for 0x%lx\n
--- a/nls/zh_TW/blackbox.m
+++ b/nls/zh_TW/blackbox.m
@@ -1,6 +1,6 @@
 $set 13 #blackbox
 
 $ #NoManagableScreens
-# Blackbox::Blackbox: no managable screens found, aborting\n
+# Blackbox::Blackbox: no manageable screens found, aborting\n
 $ #MapRequest
 # Blackbox::process_event: MapRequest for 0x%lx\n
--- a/src/blackbox.cc
+++ b/src/blackbox.cc
@@ -420,7 +420,7 @@
   }
 
   if (managed == 0) {
-    fprintf(stderr, "%s: no managable screens found, exiting...\n",
+    fprintf(stderr, "%s: no manageable screens found, exiting...\n",
             applicationName().c_str());
     ::exit(3);
   }
