Description: Move XFT libraries from Requires to Requires.private.
Author: David Schmitt <david@schmitt.edv-bus.at>
Bug-Debian: http://bugs.debian.org/342035
Forwarded: no
Last-Update: 2011-03-13

--- a/lib/libbt.pc.in
+++ b/lib/libbt.pc.in
@@ -5,7 +5,7 @@
 
 Name: Blackbox Toolbox
 Description: Utility class library for writing small applications
-Requires: @XFT_PKGCONFIG@
 Version: @VERSION@
+Requires.private: @XFT_PKGCONFIG@
 Libs: -L${libdir} -lbt @LDFLAGS@ @ICONV@ @LOCALE@
 Cflags: -I${includedir}/bt
