http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=590240

--- acl-2.2.49.orig/getfacl/Makefile
+++ acl-2.2.49/getfacl/Makefile
@@ -32,5 +32,11 @@ include $(BUILDRULES)
 
 install: default
 	$(INSTALL) -m 755 -d $(PKG_BIN_DIR)
+ifneq ($(PKG_DISTRIBUTION), debian)
 	$(LTINSTALL) -m 755 $(LTCOMMAND) $(PKG_BIN_DIR)
+else
+	$(INSTALL) -m 755 -d $(PKG_SBIN_DIR)
+	$(LTINSTALL) -m 755 $(LTCOMMAND) $(PKG_SBIN_DIR)
+	$(LTINSTALL) -S $(PKG_SBIN_DIR)/$(LTCOMMAND) $(PKG_BIN_DIR)/$(LTCOMMAND)
+endif
 install-dev install-lib:
--- acl-2.2.49.orig/setfacl/Makefile
+++ acl-2.2.49/setfacl/Makefile
@@ -32,5 +32,11 @@ include $(BUILDRULES)
 
 install: default
 	$(INSTALL) -m 755 -d $(PKG_BIN_DIR)
+ifneq ($(PKG_DISTRIBUTION), debian)
 	$(LTINSTALL) -m 755 $(LTCOMMAND) $(PKG_BIN_DIR)
+else
+	$(INSTALL) -m 755 -d $(PKG_SBIN_DIR)
+	$(LTINSTALL) -m 755 $(LTCOMMAND) $(PKG_SBIN_DIR)
+	$(LTINSTALL) -S $(PKG_SBIN_DIR)/$(LTCOMMAND) $(PKG_BIN_DIR)/$(LTCOMMAND)
+endif
 install-dev install-lib:
--- acl-2.2.49.orig/chacl/Makefile
+++ acl-2.2.49/chacl/Makefile
@@ -31,5 +31,11 @@ include $(BUILDRULES)
 
 install: default
 	$(INSTALL) -m 755 -d $(PKG_BIN_DIR)
+ifneq ($(PKG_DISTRIBUTION), debian)
 	$(LTINSTALL) -m 755 $(LTCOMMAND) $(PKG_BIN_DIR)
+else
+	$(INSTALL) -m 755 -d $(PKG_SBIN_DIR)
+	$(LTINSTALL) -m 755 $(LTCOMMAND) $(PKG_SBIN_DIR)
+	$(LTINSTALL) -S $(PKG_SBIN_DIR)/$(LTCOMMAND) $(PKG_BIN_DIR)/$(LTCOMMAND)
+endif
 install-dev install-lib:
