Description: Fix build failure on Debian hurd-i386
Forwarded: not-needed
Origin: vendor, http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=647963#5
Bug-Debian: http://bugs.debian.org/647963
Author: Samuel Thibault <sthibault@debian.org>
Last-Update: 2011-11-08

diff -ur synaesthesia-2.4.orig/sound.cc synaesthesia-2.4/sound.cc
--- synaesthesia-2.4.orig/sound.cc	2005-12-29 03:52:19.000000000 +0100
+++ synaesthesia-2.4/sound.cc	2011-11-08 01:46:45.000000000 +0100
@@ -34,17 +34,22 @@
 #include <fcntl.h>
 #include <unistd.h>
 #include <signal.h>
-#if !defined (__FreeBSD__) && !defined(__FreeBSD_kernel__)
+#if defined(__linux__)
 #include <linux/soundcard.h>
 #include <linux/cdrom.h>
 //#include <linux/ucdrom.h>
-#else
+#endif
+#if defined (__FreeBSD__) || defined(__FreeBSD_kernel__)
 #include <sys/soundcard.h>
 #include <sys/cdio.h>
 #define CDROM_LEADOUT 0xAA
 #define CD_FRAMES 75 /* frames per second */
 #define CDROM_DATA_TRACK 0x4
 #endif
+#if defined (__GNU__)
+#include <sys/soundcard.h>
+#include <sys/cdrom.h>
+#endif
 #include <time.h>
 
 #include <stdlib.h>
