Index: lvm2_2.02.97-SprezzOS1/daemons/dmeventd/Makefile.in
===================================================================
--- lvm2_2.02.97-SprezzOS1.orig/daemons/dmeventd/Makefile.in	2012-08-07 17:05:15.000000000 -0400
+++ lvm2_2.02.97-SprezzOS1/daemons/dmeventd/Makefile.in	2012-09-27 22:43:34.312601996 -0400
@@ -58,6 +58,7 @@
 
 LIBS += -ldevmapper
 LVMLIBS += -ldevmapper-event $(PTHREAD_LIBS)
+CFLAGS += -DPLUGIN_PATH=\"$(libdir)/device-mapper/\"
 
 dmeventd: $(LIB_SHARED) dmeventd.o
 	$(CC) $(CFLAGS) $(LDFLAGS) $(ELDFLAGS) -L. -o $@ dmeventd.o \
Index: lvm2_2.02.97-SprezzOS1/daemons/dmeventd/dmeventd.c
===================================================================
--- lvm2_2.02.97-SprezzOS1.orig/daemons/dmeventd/dmeventd.c	2012-08-07 17:05:15.000000000 -0400
+++ lvm2_2.02.97-SprezzOS1/daemons/dmeventd/dmeventd.c	2012-09-27 22:43:34.312601996 -0400
@@ -913,8 +913,17 @@
 {
 	void *dl;
 	struct dso_data *ret = NULL;
+	char dso_name[PATH_MAX];
 
-	if (!(dl = dlopen(data->dso_name, RTLD_NOW))) {
+	if (strchr(data->dso_name, '/') == NULL) {
+		strcpy(dso_name, PLUGIN_PATH);
+		strncat(dso_name, data->dso_name, sizeof(dso_name));
+	} else {
+		strncpy(dso_name, data->dso_name, sizeof(dso_name));
+	}
+	dso_name[sizeof(dso_name) - 1] = 0;
+
+	if (!(dl = dlopen(dso_name, RTLD_NOW))) {
 		const char *dlerr = dlerror();
 		syslog(LOG_ERR, "dmeventd %s dlopen failed: %s", data->dso_name,
 		       dlerr);
Index: lvm2_2.02.97-SprezzOS1/doc/example.conf.in
===================================================================
--- lvm2_2.02.97-SprezzOS1.orig/doc/example.conf.in	2012-08-07 17:05:15.000000000 -0400
+++ lvm2_2.02.97-SprezzOS1/doc/example.conf.in	2012-09-27 22:43:34.309268663 -0400
@@ -89,7 +89,7 @@
     # N.B. If obtain_device_list_from_udev is set to 1 the list of
     # devices is instead obtained from udev and any existing .cache
     # file is removed.
-    cache_dir = "@DEFAULT_SYS_DIR@/@DEFAULT_CACHE_SUBDIR@"
+    cache_dir = "/run/lvm"
     cache_file_prefix = ""
 
     # You can turn off writing this cache file by setting this to 0.
@@ -406,7 +406,7 @@
     # Full pathnames can be given.
 
     # Search this directory first for shared libraries.
-    #   library_dir = "/lib"
+    #   library_dir = "/lib/lvm2"
 
     # The external locking library to load if locking_type is set to 2.
     #   locking_library = "liblvm2clusterlock.so"
Index: lvm2_2.02.97-SprezzOS1/lib/commands/toolcontext.c
===================================================================
--- lvm2_2.02.97-SprezzOS1.orig/lib/commands/toolcontext.c	2012-08-07 17:05:15.000000000 -0400
+++ lvm2_2.02.97-SprezzOS1/lib/commands/toolcontext.c	2012-09-27 22:43:34.309268663 -0400
@@ -828,10 +828,8 @@
 
 	if (cache_dir || cache_file_prefix) {
 		if (dm_snprintf(cache_file, sizeof(cache_file),
-		    "%s%s%s/%s.cache",
-		    cache_dir ? "" : cmd->system_dir,
-		    cache_dir ? "" : "/",
-		    cache_dir ? : DEFAULT_CACHE_SUBDIR,
+		    "%s/%s.cache",
+		    cache_dir ? : DEFAULT_RUN_DIR,
 		    cache_file_prefix ? : DEFAULT_CACHE_FILE_PREFIX) < 0) {
 			log_error("Persistent cache filename too long.");
 			f3->destroy(f3);
@@ -839,8 +837,8 @@
 		}
 	} else if (!(dev_cache = find_config_tree_str(cmd, "devices/cache", NULL)) &&
 		   (dm_snprintf(cache_file, sizeof(cache_file),
-				"%s/%s/%s.cache",
-				cmd->system_dir, DEFAULT_CACHE_SUBDIR,
+				"%s/%s.cache",
+				DEFAULT_RUN_DIR,
 				DEFAULT_CACHE_FILE_PREFIX) < 0)) {
 		log_error("Persistent cache filename too long.");
 		f3->destroy(f3);
