Index: lvm2_2.02.97-SprezzOS1/configure.in
===================================================================
--- lvm2_2.02.97-SprezzOS1.orig/configure.in	2012-09-27 22:40:18.000000000 -0400
+++ lvm2_2.02.97-SprezzOS1/configure.in	2012-09-27 22:44:29.582601000 -0400
@@ -1268,11 +1268,7 @@
 fi
 
 ################################################################################
-AC_PATH_PROG(MODPROBE_CMD, modprobe)
-
-if test x$MODPROBE_CMD != x; then
-	AC_DEFINE_UNQUOTED([MODPROBE_CMD], ["$MODPROBE_CMD"], [The path to 'modprobe', if available.])
-fi
+AC_DEFINE_UNQUOTED([MODPROBE_CMD], ["/sbin/modprobe"], [The path to 'modprobe', if available.])
 
 
 lvm_exec_prefix=$exec_prefix
