Index: lvm2_2.02.97-SprezzOS1/tools/lvm.c
===================================================================
--- lvm2_2.02.97-SprezzOS1.orig/tools/lvm.c	2012-08-07 17:05:15.000000000 -0400
+++ lvm2_2.02.97-SprezzOS1/tools/lvm.c	2012-09-27 22:44:57.445933832 -0400
@@ -25,9 +25,6 @@
 
 #  include <readline/readline.h>
 #  include <readline/history.h>
-#  ifndef HAVE_RL_COMPLETION_MATCHES
-#    define rl_completion_matches(a, b) completion_matches((char *)a, b)
-#  endif
 
 static struct cmdline_context *_cmdline;
 
