Index: file-roller-3.2.1/data/packages.match
===================================================================
--- file-roller-3.2.1.orig/data/packages.match	2011-11-13 22:16:29.224463195 +0100
+++ file-roller-3.2.1/data/packages.match	2011-11-13 22:29:40.556431917 +0100
@@ -15,7 +15,7 @@
 p7zip-full=
 p7zip-rar=
 rar=
-rpm=
+rpm=rpm2cpio
 rzip=
 tar=
 unace=
@@ -23,7 +23,7 @@
 unrar=
 unstaff=
 unzip=
-xz=
+xz=xz-utils
 zip=
 zoo=
 
