Bug-Debian: http://bugs.debian.org/347650
Last-Update: 2011-06-14

Index: file-roller-3.3.92/ltmain.sh
===================================================================
--- file-roller-3.3.92.orig/ltmain.sh	2012-03-20 00:21:26.000000000 +0100
+++ file-roller-3.3.92/ltmain.sh	2012-03-20 07:34:47.462360486 +0100
@@ -5800,6 +5800,11 @@
 	arg=$func_stripname_result
 	;;
 
+      -Wl,--as-needed|-Wl,--no-as-needed)
+	deplibs="$deplibs $arg"
+	continue
+	;;
+
       -Wl,*)
 	func_stripname '-Wl,' '' "$arg"
 	args=$func_stripname_result
@@ -6160,6 +6165,15 @@
 	lib=
 	found=no
 	case $deplib in
+	-Wl,--as-needed|-Wl,--no-as-needed)
+	  if test "$linkmode,$pass" = "prog,link"; then
+	    compile_deplibs="$deplib $compile_deplibs"
+	    finalize_deplibs="$deplib $finalize_deplibs"
+	  else
+	    deplibs="$deplib $deplibs"
+	  fi
+	  continue
+	  ;;
 	-mt|-mthreads|-kthread|-Kthread|-pthread|-pthreads|--thread-safe \
         |-threads|-fopenmp|-openmp|-mp|-xopenmp|-omp|-qsmp=*)
 	  if test "$linkmode,$pass" = "prog,link"; then
