From 4aa4059b40aed27ae76201baa48f419d1ba09576 Mon Sep 17 00:00:00 2001
From: Till Adam <till@kdab.com>
Date: Tue, 6 Sep 2011 09:35:06 +0200
Subject: [PATCH] Build with clang.

---
 libkleo/backends/qgpgme/threadedjobmixin.h |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libkleo/backends/qgpgme/threadedjobmixin.h b/libkleo/backends/qgpgme/threadedjobmixin.h
index 6d915e7..a1f85cc 100644
--- a/libkleo/backends/qgpgme/threadedjobmixin.h
+++ b/libkleo/backends/qgpgme/threadedjobmixin.h
@@ -143,7 +143,7 @@ namespace _detail {
 
     void lateInitialization() {
       assert( m_ctx );
-      connect( &m_thread, SIGNAL(finished()), this, SLOT(slotFinished()) );
+      QObject::connect( &m_thread, SIGNAL(finished()), this, SLOT(slotFinished()) );
       m_ctx->setProgressProvider( this );
     }
 
-- 
1.7.10

