commit c5ca46c2d1be78c651afb843cc834cf2b5b24953
Author: Julien BLACHE <jb@jblache.org>
Date:   Wed Feb 16 19:37:43 2011 +0100

    Fix v4l build with libv4l 0.8.3+

Index: sane-backends-1.0.22/backend/v4l.c
===================================================================
--- sane-backends-1.0.22.orig/backend/v4l.c	2011-02-16 19:20:51.193016002 +0100
+++ sane-backends-1.0.22/backend/v4l.c	2011-02-16 19:21:00.497016002 +0100
@@ -84,9 +84,8 @@
 #include "../include/sane/sanei_config.h"
 #define V4L_CONFIG_FILE "v4l.conf"
 
-#include "v4l.h"
-
 #include <libv4l1.h>
+#include "v4l.h"
 
 static const SANE_Device **devlist = NULL;
 static int num_devices;
Index: sane-backends-1.0.22/backend/v4l.h
===================================================================
--- sane-backends-1.0.22.orig/backend/v4l.h	2011-02-16 19:20:07.537016001 +0100
+++ sane-backends-1.0.22/backend/v4l.h	2011-02-16 19:20:41.889016001 +0100
@@ -29,6 +29,7 @@
 #ifndef v4l_h
 #define v4l_h
 
+#ifndef __LINUX_VIDEODEV_H
 /* Kernel interface */
 /* Only the stuff we need. For more features, more defines are needed */
 
@@ -165,7 +166,7 @@
 
 
 /* end of kernel interface */
-
+#endif /* !__LINUX_VIDEODEV_H */
 
 #include <../include/sane/sane.h>
 
