commit 56e39a90800611214385675aec0aa6539618dc28
Author: Pino Toscano <toscano.pino@tiscali.it>
Date:   Fri Mar 4 18:18:35 2011 +0100

    Define PATH_MAX if needed

diff --git a/frontend/saned.c b/frontend/saned.c
index c6695be..252b51e 100644
--- a/frontend/saned.c
+++ b/frontend/saned.c
@@ -209,6 +209,10 @@ static AvahiEntryGroup *avahi_group = NULL;
 # define MAXHOSTNAMELEN 120
 #endif
 
+#ifndef PATH_MAX
+# define PATH_MAX 1024
+#endif
+
 struct saned_child {
   pid_t pid;
   struct saned_child *next;
