description: Fix build failure on kfreebsd (closes: #680234)
author: Petr Salinger
Index: sane-backends-1.0.22/backend/umax_pp_low.c
===================================================================
--- sane-backends-1.0.22.orig/backend/umax_pp_low.c	2010-12-01 18:49:58.000000000 -0500
+++ sane-backends-1.0.22/backend/umax_pp_low.c	2012-07-15 17:10:47.000000000 -0400
@@ -73,8 +73,10 @@
 #endif
 
 #ifdef HAVE_MACHINE_CPUFUNC_H
+#ifndef __GLIBC__
 #include <machine/cpufunc.h>
 #endif
+#endif
 
 #ifdef HAVE_I386_SET_IOPERM
 #include <machine/sysarch.h>
