commit 22c7800b8dccfeb9090729e43f4bd40e3149b468
Author: Julien BLACHE <jb@jblache.org>
Date:   Sat Mar 12 11:44:15 2011 +0100

    batch-start defaults to 1, not 0

diff --git a/doc/scanimage.man b/doc/scanimage.man
index 121a1a5..0012f6e 100644
--- a/doc/scanimage.man
+++ b/doc/scanimage.man
@@ -171,7 +171,7 @@ is given as a printf style string with one integer parameter.
 .B \-\-batch\-start
 .I start
 selects the page number to start naming files with. If this option is not
-given, the counter will start at 0.
+given, the counter will start at 1.
 .B \-\-batch\-count
 .I count
 specifies the number of pages to attempt to scan.  If not given, 
