commit 90f15838eab8946a039394e0abaa70de25b5348c
Author: Alex Belkin <abc@telekom.ru>
Date:   Tue Mar 8 18:01:25 2011 +0300

    usb ids for Samsung SCX-4623FW

diff --git a/backend/xerox_mfp.conf.in b/backend/xerox_mfp.conf.in
index ffc4c4c..2a28004 100644
--- a/backend/xerox_mfp.conf.in
+++ b/backend/xerox_mfp.conf.in
@@ -39,6 +39,9 @@ usb 0x0924 0x4265
 #Samsung SCX-4623
 usb 0x04e8 0x3434
 
+#Samsung SCX-4623FW
+usb 0x04e8 0x3440
+
 # Samsung SCX-4825FN
 usb 0x04e8 0x343c
 
diff --git a/doc/descriptions/xerox_mfp.desc b/doc/descriptions/xerox_mfp.desc
index 9eee9ac..91d9be3 100644
--- a/doc/descriptions/xerox_mfp.desc
+++ b/doc/descriptions/xerox_mfp.desc
@@ -93,6 +93,11 @@
 :usbid "0x04e8" "0x3434"
 :status :good
 
+:model "SCX-4623FW"
+:interface "USB"
+:usbid "0x04e8" "0x3440"
+:status :good
+
 :model "SCX-4825FN"
 :interface "USB"
 :usbid "0x04e8" "0x343c"
