Index: xserver-xorg-video-glint/man/glint.man
===================================================================
--- xserver-xorg-video-glint.orig/man/glint.man	2007-04-22 05:18:54.000000000 +0200
+++ xserver-xorg-video-glint/man/glint.man	2007-04-22 05:19:09.000000000 +0200
@@ -106,5 +106,5 @@
 .SH "SEE ALSO"
 __xservername__(__appmansuffix__), __xconfigfile__(__filemansuffix__), xorgconfig(__appmansuffix__), Xserver(__appmansuffix__), X(__miscmansuffix__)
 .SH AUTHORS
-Authors include: Alan Hourihane, Dirk Hohndel, Stefan Dirsch, Michel Dänzer,
+Authors include: Alan Hourihane, Dirk Hohndel, Stefan Dirsch, Michel D\(:anzer,
 Sven Luther
