From: =?UTF-8?q?Guido=20G=C3=BCnther?= <agx@sigxcpu.org>
Date: Sat, 14 Jan 2012 18:09:01 +0100
Subject: Don't fail if we can't setup avahi

---
 src/rpc/virnetserver.c |    3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/src/rpc/virnetserver.c b/src/rpc/virnetserver.c
index cf15240..e534109 100644
--- a/src/rpc/virnetserver.c
+++ b/src/rpc/virnetserver.c
@@ -962,8 +962,7 @@ void virNetServerRun(virNetServerPtr srv)
 
     virNetServerLock(srv);
 
-    if (srv->mdns &&
-        virNetServerMDNSStart(srv->mdns) < 0)
+    if (srv->mdns)
         goto cleanup;
 
     srv->quit = 0;
