From abf9d0b9c431f56e3fe88cb5a22b2f5e7b41e92e Mon Sep 17 00:00:00 2001
From: Nicolas Bruguier <gandalfn@mithrandir.(none)>
Date: Tue, 12 Dec 2006 22:59:22 +0100
Subject: [PATCH] Fix no border window shadow
---
 libdecoration/decoration.c |    4 +++-
 1 files changed, 3 insertions(+), 1 deletions(-)

--- compiz.orig/libdecoration/decoration.c
+++ compiz/libdecoration/decoration.c
@@ -1243,7 +1243,9 @@ decor_shadow_create (Display		    *xdisp
     }
 
     /* create pixmap for temporary decorations */
-    d_pixmap = XCreatePixmap (xdisplay, xroot, d_width, d_height, 32);
+    d_pixmap = XCreatePixmap (xdisplay, xroot, 
+                              (d_width % 32) ? d_width : d_width + 1, 
+                              d_height, 32);
     if (!d_pixmap)
     {
 	XFreePixmap (xdisplay, pixmap);
