Index: compiz-0.9.2.1+glibmainloop2/plugins/fade/fade.xml.in
===================================================================
--- compiz-0.9.2.1+glibmainloop2.orig/plugins/fade/fade.xml.in	2010-12-06 12:23:04.705212002 +0100
+++ compiz-0.9.2.1+glibmainloop2/plugins/fade/fade.xml.in	2010-12-06 12:23:23.761212002 +0100
@@ -9,6 +9,7 @@
 	    </requirement>
 	    <relation type="after">
 		<plugin>decor</plugin>
+		<plugin>staticswitcher</plugin>
 	    </relation>
 	    <relation type="before">
 		<plugin>cube</plugin>
