Index: compiz-0.8.4/configure.ac
===================================================================
--- compiz-0.8.4.orig/configure.ac	2011-10-25 11:57:51.847926095 +0200
+++ compiz-0.8.4/configure.ac	2011-10-25 12:07:34.612599752 +0200
@@ -126,6 +126,8 @@
 		 xcursor    \
 		 ice	    \
 		 sm	    \
+		 x11	    \
+		 xext	    \
 		 libxml-2.0 \
 		 libxslt    \
 		 libstartup-notification-1.0 >= 0.7"
@@ -133,6 +135,8 @@
 PKG_CHECK_MODULES(COMPIZ, $COMPIZ_REQUIRES)
 AC_SUBST(COMPIZ_REQUIRES)
 
+AC_SEARCH_LIBS(dlopen, dl)
+
 DECORATION_REQUIRES="xrender"
 
 PKG_CHECK_MODULES(DECORATION, $DECORATION_REQUIRES)
@@ -300,6 +304,7 @@
 if test "x$use_gtk" = "xyes"; then
   PKG_CHECK_MODULES(GTK_WINDOW_DECORATOR,
 		    xrender >= 0.8.4  \
+		    xcursor	      \
 		    gtk+-2.0 >= 2.8.0 \
 		    libwnck-1.0	      \
 		    pangocairo,
@@ -514,11 +519,11 @@
 
   if test -d "$kde4libs"; then
     PKG_CHECK_MODULES(KDE4_WINDOW_DECORATOR,
-		      dbus-1 xdamage xcomposite QtCore >= 4.5.0 QtGui QtDBus,
+		      dbus-1 xdamage xcomposite x11 xext QtCore >= 4.5.0 QtGui QtDBus,
 			[use_kde4=yes], [use_kde4=no])
 
     KDE4_CFLAGS="-I$kde4incs"
-    KDE4_LIBS="-L$kde4libs -lkdecore -lkdecorations -lplasma"
+    KDE4_LIBS="-L$kde4libs -lkdecore -lkdecorations -lkdeui -lplasma"
 
     QT4_MOC=`pkg-config QtGui --variable=moc_location`
     QDBUSXML2CPP=`pkg-config QtGui --variable=exec_prefix`"/bin/qdbusxml2cpp"
Index: compiz-0.8.4/gtk/window-decorator/Makefile.am
===================================================================
--- compiz-0.8.4.orig/gtk/window-decorator/Makefile.am	2009-10-14 04:34:41.000000000 +0200
+++ compiz-0.8.4/gtk/window-decorator/Makefile.am	2011-10-25 12:04:58.614560913 +0200
@@ -11,7 +11,8 @@
 	@GTK_WINDOW_DECORATOR_LIBS@		       \
 	@GCONF_LIBS@				       \
 	@DBUS_GLIB_LIBS@			       \
-	$(metacitylibs)
+	$(metacitylibs)				       \
+	-lm
 
 gtk_window_decorator_SOURCES	  = \
 	gtk-window-decorator.c
