Index: compiz-0.9.4/plugins/grid/CMakeLists.txt
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ compiz-0.9.4/plugins/grid/CMakeLists.txt	2011-02-24 12:12:42.000000000 +0100
@@ -0,0 +1,5 @@
+find_package (Compiz REQUIRED)
+
+include (CompizPlugin)
+
+compiz_plugin(grid PLUGINDEPS composite opengl)
Index: compiz-0.9.4/plugins/grid/grid.xml.in
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ compiz-0.9.4/plugins/grid/grid.xml.in	2011-02-24 12:12:42.000000000 +0100
@@ -0,0 +1,565 @@
+<?xml version="1.0"?>
+<compiz>
+    <plugin name="grid" useBcop="true">
+	<_short>Grid</_short>
+	<_long>Warp and resize windows to fit an imaginary grid.</_long>
+	<category>Window Management</category>
+	<deps>
+	    <requirement>
+		<plugin>opengl</plugin>
+	    </requirement>
+	    <relation type="before">
+		<plugin>wobbly</plugin>
+	    </relation>
+	    <relation type="after">
+		<plugin>composite</plugin>
+		<plugin>opengl</plugin>
+		<plugin>decor</plugin>
+	    </relation>
+	</deps>
+	<options>
+	    <group>
+		<_short>Bindings</_short>
+		<option name="put_center_key" type="key">
+		    <_short>Put Center</_short>
+		    <_long>Move window to the center</_long>
+		    <default>&lt;Control&gt;&lt;Alt&gt;KP_5</default>
+		</option>
+		<option name="put_left_key" type="key">
+		    <_short>Put Left</_short>
+		    <_long>Move window to the left edge</_long>
+		    <default>&lt;Control&gt;&lt;Alt&gt;KP_4</default>
+		</option>
+		<option name="put_right_key" type="key">
+		    <_short>Put Right</_short>
+		    <_long>Move window to the right edge</_long>
+		    <default>&lt;Control&gt;&lt;Alt&gt;KP_6</default>
+		</option>
+		<option name="put_top_key" type="key">
+		    <_short>Put Top</_short>
+		    <_long>Move window to the top edge</_long>
+		    <default>&lt;Control&gt;&lt;Alt&gt;KP_8</default>
+		</option>
+		<option name="put_bottom_key" type="key">
+		    <_short>Put Bottom</_short>
+		    <_long>Move window to the bottom edge</_long>
+		    <default>&lt;Control&gt;&lt;Alt&gt;KP_2</default>
+		</option>
+		<option name="put_topleft_key" type="key">
+		    <_short>Put Top Left</_short>
+		    <_long>Move window to the top left corner</_long>
+		    <default>&lt;Control&gt;&lt;Alt&gt;KP_7</default>
+		</option>
+		<option name="put_topright_key" type="key">
+		    <_short>Put Top Right</_short>
+		    <_long>Move window to the top right corner</_long>
+		    <default>&lt;Control&gt;&lt;Alt&gt;KP_9</default>
+		</option>
+		<option name="put_bottomleft_key" type="key">
+		    <_short>Put Bottom Left</_short>
+		    <_long>Move window to the bottom left corner</_long>
+		    <default>&lt;Control&gt;&lt;Alt&gt;KP_1</default>
+		</option>
+		<option name="put_bottomright_key" type="key">
+		    <_short>Put Bottom Right</_short>
+		    <_long>Move window to the bottom right corner</_long>
+		    <default>&lt;Control&gt;&lt;Alt&gt;KP_3</default>
+		</option>
+		<option name="put_maximize_key" type="key">
+		    <_short>Maximize</_short>
+		    <_long>Maximize window</_long>
+		    <default>&lt;Control&gt;&lt;Alt&gt;KP_0</default>
+		</option>
+		<option name="put_restore_key" type="key">
+		    <_short>Restore</_short>
+		    <_long>Restore window to it's original size</_long>
+		    <default>&lt;Control&gt;&lt;Alt&gt;r</default>
+		</option>
+	    </group>
+	    <group>
+		<_short>Edges</_short>
+		<subgroup>
+			<_short>Resize Actions</_short>
+			<_long>Window resize action</_long>
+			<option name="top_left_corner_action" type="int">
+			    <_short>Upper Left Corner</_short>
+			    <_long>Action to be performed when window is dropped on the top left corner</_long>
+			    <default>7</default>
+			    <min>0</min>
+			    <max>10</max>
+			    <desc>
+				<value>0</value>
+				<_name>None</_name>
+			    </desc>
+			    <desc>
+				<value>1</value>
+				<_name>Bottom Left Corner</_name>
+			    </desc>
+			    <desc>
+				<value>2</value>
+				<_name>Bottom Half</_name>
+			    </desc>
+			    <desc>
+				<value>3</value>
+				<_name>Bottom Right Corner</_name>
+			    </desc>
+			    <desc>
+				<value>4</value>
+				<_name>Left Half</_name>
+			    </desc>
+			    <desc>
+				<value>5</value>
+				<_name>Fill Screen</_name>
+			    </desc>
+			    <desc>
+				<value>6</value>
+				<_name>Right Half</_name>
+			    </desc>
+			    <desc>
+				<value>7</value>
+				<_name>Top Left Corner</_name>
+			    </desc>
+			    <desc>
+				<value>8</value>
+				<_name>Top Half</_name>
+			    </desc>
+			    <desc>
+				<value>9</value>
+				<_name>Top Right Corner</_name>
+			    </desc>
+			    <desc>
+				<value>10</value>
+				<_name>Maximize</_name>
+			    </desc>
+			</option>
+			<option name="top_edge_action" type="int">
+			    <_short>Top Edge</_short>
+			    <_long>Action to be performed when window is dropped on the top edge</_long>
+			    <default>10</default>
+			    <min>0</min>
+			    <max>10</max>
+			    <desc>
+				<value>0</value>
+				<_name>None</_name>
+			    </desc>
+			    <desc>
+				<value>1</value>
+				<_name>Bottom Left Corner</_name>
+			    </desc>
+			    <desc>
+				<value>2</value>
+				<_name>Bottom Half</_name>
+			    </desc>
+			    <desc>
+				<value>3</value>
+				<_name>Bottom Right Corner</_name>
+			    </desc>
+			    <desc>
+				<value>4</value>
+				<_name>Left Half</_name>
+			    </desc>
+			    <desc>
+				<value>5</value>
+				<_name>Fill Screen</_name>
+			    </desc>
+			    <desc>
+				<value>6</value>
+				<_name>Right Half</_name>
+			    </desc>
+			    <desc>
+				<value>7</value>
+				<_name>Top Left Corner</_name>
+			    </desc>
+			    <desc>
+				<value>8</value>
+				<_name>Top Half</_name>
+			    </desc>
+			    <desc>
+				<value>9</value>
+				<_name>Top Right Corner</_name>
+			    </desc>
+			    <desc>
+				<value>10</value>
+				<_name>Maximize</_name>
+			    </desc>
+			</option>
+			<option name="top_right_corner_action" type="int">
+			    <_short>Upper Right Corner</_short>
+			    <_long>Action to be performed when window is dropped on the top right corner</_long>
+			    <default>9</default>
+			    <min>0</min>
+			    <max>10</max>
+			    <desc>
+				<value>0</value>
+				<_name>None</_name>
+			    </desc>
+			    <desc>
+				<value>1</value>
+				<_name>Bottom Left Corner</_name>
+			    </desc>
+			    <desc>
+				<value>2</value>
+				<_name>Bottom Half</_name>
+			    </desc>
+			    <desc>
+				<value>3</value>
+				<_name>Bottom Right Corner</_name>
+			    </desc>
+			    <desc>
+				<value>4</value>
+				<_name>Left Half</_name>
+			    </desc>
+			    <desc>
+				<value>5</value>
+				<_name>Fill Screen</_name>
+			    </desc>
+			    <desc>
+				<value>6</value>
+				<_name>Right Half</_name>
+			    </desc>
+			    <desc>
+				<value>7</value>
+				<_name>Top Left Corner</_name>
+			    </desc>
+			    <desc>
+				<value>8</value>
+				<_name>Top Half</_name>
+			    </desc>
+			    <desc>
+				<value>9</value>
+				<_name>Top Right Corner</_name>
+			    </desc>
+			    <desc>
+				<value>10</value>
+				<_name>Maximize</_name>
+			    </desc>
+			</option>
+			<option name="left_edge_action" type="int">
+			    <_short>Left Edge</_short>
+			    <_long>Action to be performed when window is dropped on the left edge</_long>
+			    <default>4</default>
+			    <min>0</min>
+			    <max>10</max>
+			    <desc>
+				<value>0</value>
+				<_name>None</_name>
+			    </desc>
+			    <desc>
+				<value>1</value>
+				<_name>Bottom Left Corner</_name>
+			    </desc>
+			    <desc>
+				<value>2</value>
+				<_name>Bottom Half</_name>
+			    </desc>
+			    <desc>
+				<value>3</value>
+				<_name>Bottom Right Corner</_name>
+			    </desc>
+			    <desc>
+				<value>4</value>
+				<_name>Left Half</_name>
+			    </desc>
+			    <desc>
+				<value>5</value>
+				<_name>Fill Screen</_name>
+			    </desc>
+			    <desc>
+				<value>6</value>
+				<_name>Right Half</_name>
+			    </desc>
+			    <desc>
+				<value>7</value>
+				<_name>Top Left Corner</_name>
+			    </desc>
+			    <desc>
+				<value>8</value>
+				<_name>Top Half</_name>
+			    </desc>
+			    <desc>
+				<value>9</value>
+				<_name>Top Right Corner</_name>
+			    </desc>
+			    <desc>
+				<value>10</value>
+				<_name>Maximize</_name>
+			    </desc>
+			</option>
+			<option name="right_edge_action" type="int">
+			    <_short>Right Edge</_short>
+			    <_long>Action to be performed when window is dropped on the right edge</_long>
+			    <default>6</default>
+			    <min>0</min>
+			    <max>10</max>
+			    <desc>
+				<value>0</value>
+				<_name>None</_name>
+			    </desc>
+			    <desc>
+				<value>1</value>
+				<_name>Bottom Left Corner</_name>
+			    </desc>
+			    <desc>
+				<value>2</value>
+				<_name>Bottom Half</_name>
+			    </desc>
+			    <desc>
+				<value>3</value>
+				<_name>Bottom Right Corner</_name>
+			    </desc>
+			    <desc>
+				<value>4</value>
+				<_name>Left Half</_name>
+			    </desc>
+			    <desc>
+				<value>5</value>
+				<_name>Fill Screen</_name>
+			    </desc>
+			    <desc>
+				<value>6</value>
+				<_name>Right Half</_name>
+			    </desc>
+			    <desc>
+				<value>7</value>
+				<_name>Top Left Corner</_name>
+			    </desc>
+			    <desc>
+				<value>8</value>
+				<_name>Top Half</_name>
+			    </desc>
+			    <desc>
+				<value>9</value>
+				<_name>Top Right Corner</_name>
+			    </desc>
+			    <desc>
+				<value>10</value>
+				<_name>Maximize</_name>
+			    </desc>
+			</option>
+			<option name="bottom_left_corner_action" type="int">
+			    <_short>Bottom Left Corner</_short>
+			    <_long>Action to be performed when window is dropped on the bottom left corner</_long>
+			    <default>1</default>
+			    <min>0</min>
+			    <max>10</max>
+			    <desc>
+				<value>0</value>
+				<_name>None</_name>
+			    </desc>
+			    <desc>
+				<value>1</value>
+				<_name>Bottom Left Corner</_name>
+			    </desc>
+			    <desc>
+				<value>2</value>
+				<_name>Bottom Half</_name>
+			    </desc>
+			    <desc>
+				<value>3</value>
+				<_name>Bottom Right Corner</_name>
+			    </desc>
+			    <desc>
+				<value>4</value>
+				<_name>Left Half</_name>
+			    </desc>
+			    <desc>
+				<value>5</value>
+				<_name>Fill Screen</_name>
+			    </desc>
+			    <desc>
+				<value>6</value>
+				<_name>Right Half</_name>
+			    </desc>
+			    <desc>
+				<value>7</value>
+				<_name>Top Left Corner</_name>
+			    </desc>
+			    <desc>
+				<value>8</value>
+				<_name>Top Half</_name>
+			    </desc>
+			    <desc>
+				<value>9</value>
+				<_name>Top Right Corner</_name>
+			    </desc>
+			    <desc>
+				<value>10</value>
+				<_name>Maximize</_name>
+			    </desc>
+			</option>
+			<option name="bottom_edge_action" type="int">
+			    <_short>Bottom Edge</_short>
+			    <_long>Action to be performed when window is dropped on the bottom edge</_long>
+			    <default>2</default>
+			    <min>0</min>
+			    <max>10</max>
+			    <desc>
+				<value>0</value>
+				<_name>None</_name>
+			    </desc>
+			    <desc>
+				<value>1</value>
+				<_name>Bottom Left Corner</_name>
+			    </desc>
+			    <desc>
+				<value>2</value>
+				<_name>Bottom Half</_name>
+			    </desc>
+			    <desc>
+				<value>3</value>
+				<_name>Bottom Right Corner</_name>
+			    </desc>
+			    <desc>
+				<value>4</value>
+				<_name>Left Half</_name>
+			    </desc>
+			    <desc>
+				<value>5</value>
+				<_name>Fill Screen</_name>
+			    </desc>
+			    <desc>
+				<value>6</value>
+				<_name>Right Half</_name>
+			    </desc>
+			    <desc>
+				<value>7</value>
+				<_name>Top Left Corner</_name>
+			    </desc>
+			    <desc>
+				<value>8</value>
+				<_name>Top Half</_name>
+			    </desc>
+			    <desc>
+				<value>9</value>
+				<_name>Top Right Corner</_name>
+			    </desc>
+			    <desc>
+				<value>10</value>
+				<_name>Maximize</_name>
+			    </desc>
+			</option>
+			<option name="bottom_right_corner_action" type="int">
+			    <_short>Bottom Right Corner</_short>
+			    <_long>Action to be performed when window is dropped on the bottom right corner</_long>
+			    <default>3</default>
+			    <min>0</min>
+			    <max>10</max>
+			    <desc>
+				<value>0</value>
+				<_name>None</_name>
+			    </desc>
+			    <desc>
+				<value>1</value>
+				<_name>Bottom Left Corner</_name>
+			    </desc>
+			    <desc>
+				<value>2</value>
+				<_name>Bottom Half</_name>
+			    </desc>
+			    <desc>
+				<value>3</value>
+				<_name>Bottom Right Corner</_name>
+			    </desc>
+			    <desc>
+				<value>4</value>
+				<_name>Left Half</_name>
+			    </desc>
+			    <desc>
+				<value>5</value>
+				<_name>Fill Screen</_name>
+			    </desc>
+			    <desc>
+				<value>6</value>
+				<_name>Right Half</_name>
+			    </desc>
+			    <desc>
+				<value>7</value>
+				<_name>Top Left Corner</_name>
+			    </desc>
+			    <desc>
+				<value>8</value>
+				<_name>Top Half</_name>
+			    </desc>
+			    <desc>
+				<value>9</value>
+				<_name>Top Right Corner</_name>
+			    </desc>
+			    <desc>
+				<value>10</value>
+				<_name>Maximize</_name>
+			    </desc>
+			</option>
+			<option name="snapoff_maximized" type="bool">
+				<_short>Snapoff maximized windows</_short>
+				<_long>Snapoff maximized windows when pulling to edge.</_long>
+				<default>false</default>
+			</option>
+			<option name="snapback_windows" type="bool">
+				<_short>Snap windows back to original size</_short>
+				<_long>Snaps windows back to their original size if dragged away from their gridded position.</_long>
+				<default>true</default>
+			</option>
+		</subgroup>
+		<subgroup>
+			<_short>Thresholds</_short>
+			<option name="left_edge_threshold" type="int">
+				<_short>Left Edge</_short>
+				<_long>Maximum number of pixels from the left edge a window can be dropped</_long>
+				<default>5</default>
+				<min>0</min>
+				<max>500</max>
+			</option>
+			<option name="right_edge_threshold" type="int">
+				<_short>Right Edge</_short>
+				<_long>Maximum number of pixels from the right edge a window can be dropped</_long>
+				<default>5</default>
+				<min>0</min>
+				<max>500</max>
+			</option>
+			<option name="top_edge_threshold" type="int">
+				<_short>Top Edge</_short>
+				<_long>Maximum number of pixels from the top edge a window can be dropped</_long>
+				<default>5</default>
+				<min>0</min>
+				<max>500</max>
+			</option>
+			<option name="bottom_edge_threshold" type="int">
+				<_short>Bottom Edge</_short>
+				<_long>Maximum number of pixels from the Bottom edge a window can be dropped</_long>
+				<default>5</default>
+				<min>0</min>
+				<max>500</max>
+			</option>
+		</subgroup>
+	    </group>
+	    <group>
+		<_short>Appearance</_short>
+		<option name="draw_indicator" type="bool">
+			<_short>Draw Indicator</_short>
+			<_long>Draw Window Resize Indicator</_long>
+			<default>true</default>
+		</option>
+		<option name="outline_color" type="color">
+			<_short>Outline Color</_short>
+			<_long>Color of the resize indicator outline</_long>
+			<default>
+			    <red>0x2f2f</red>
+			    <green>0x2f2f</green>
+			    <blue>0x4f4f</blue>
+			    <alpha>0x9f9f</alpha>
+			</default>
+		</option>
+		<option name="fill_color" type="color">
+			<_short>Fill Color</_short>
+			<_long>Fill color of the resize indicator</_long>
+			<default>
+			    <red>0x2f2f</red>
+			    <green>0x2f2f</green>
+			    <blue>0x4f4f</blue>
+			    <alpha>0x4f4f</alpha>
+			</default>
+		</option>
+	    </group>
+	</options>
+    </plugin>
+</compiz>
Index: compiz-0.9.4/plugins/grid/src/grid.cpp
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ compiz-0.9.4/plugins/grid/src/grid.cpp	2011-02-24 12:12:42.000000000 +0100
@@ -0,0 +1,710 @@
+/*
+ * Compiz Fusion Grid plugin
+ *
+ * Copyright (c) 2008 Stephen Kennedy <suasol@gmail.com>
+ * Copyright (c) 2010 Scott Moreau <oreaus@gmail.com>
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * Description:
+ *
+ * Plugin to act like winsplit revolution (http://www.winsplit-revolution.com/)
+ * use <Control><Alt>NUMPAD_KEY to move and tile your windows.
+ *
+ * Press the tiling keys several times to cycle through some tiling options.
+ */
+
+#include "grid.h"
+
+static const GridProps gridProps[] =
+{
+    {0,1, 1,1},
+
+    {0,1, 2,2},
+    {0,1, 1,2},
+    {1,1, 2,2},
+
+    {0,0, 2,1},
+    {0,0, 1,1},
+    {1,0, 2,1},
+
+    {0,0, 2,2},
+    {0,0, 1,2},
+    {1,0, 2,2},
+
+    {0,0, 1,1},
+};
+
+CompRect
+GridScreen::slotToRect (CompWindow      *w,
+			const CompRect& slot)
+{
+    return CompRect (slot.x () + w->input ().left,
+		     slot.y () + w->input ().top,
+		     slot.width () - (w->input ().left + w->input ().right),
+		     slot.height () - (w->input ().top + w->input ().bottom));
+}
+
+CompRect
+GridScreen::constrainSize (CompWindow      *w,
+			   const CompRect& slot)
+{
+    CompRect result;
+    int      cw, ch;
+
+    result = slotToRect (w, slot);
+
+    if (w->constrainNewWindowSize (result.width (), result.height (), &cw, &ch))
+    {
+	/* constrained size may put window offscreen, adjust for that case */
+	int dx = result.x () + cw - workarea.right () + w->input ().right;
+	int dy = result.y () + ch - workarea.bottom () + w->input ().bottom;
+
+	if (dx > 0)
+	    result.setX (result.x () - dx);
+	if (dy > 0)
+	    result.setY (result.y () - dy);
+
+	result.setWidth (cw);
+	result.setHeight (ch);
+    }
+
+    return result;
+}
+
+void
+GridScreen::getPaintRectangle (CompRect &cRect)
+{
+    if (edgeToGridType () != GridUnknown && optionGetDrawIndicator ())
+	cRect = desiredSlot;
+    else
+	cRect.setGeometry (0, 0, 0, 0);
+}
+
+bool
+GridScreen::initiateCommon (CompAction         *action,
+			    CompAction::State  state,
+			    CompOption::Vector &option,
+			    GridType           where,
+			    bool               resize)
+{
+    Window     xid;
+    CompWindow *cw = 0;
+
+    xid = CompOption::getIntOptionNamed (option, "window");
+    cw  = screen->findWindow (xid);
+
+    if (where == GridUnknown || screen->otherGrabExist ("move", NULL))
+	return false;
+
+    if (cw)
+    {
+	XWindowChanges xwc;
+
+	GRID_WINDOW (cw);
+
+	if (gw->lastTarget != where)
+	    gw->resizeCount = 0;
+
+	props = gridProps[where];
+
+	/* get current available area */
+	if (gw->grabIsMove)
+	    workarea = screen->getWorkareaForOutput
+			    (screen->outputDeviceForPoint (pointerX, pointerY));
+	else
+	{
+	    workarea = screen->getWorkareaForOutput (cw->outputDevice ());
+
+	    if (props.numCellsX == 1)
+		centerCheck = true;
+
+	    if (!gw->isGridResized)
+		/* Store size not including borders when using a keybinding */
+		gw->originalSize = slotToRect(cw, cw->serverInputRect ());
+	}
+
+	if ((cw->state () & MAXIMIZE_STATE) &&
+	    (resize || optionGetSnapoffMaximized ()))
+	{
+	    /* maximized state interferes with us, clear it */
+	    cw->maximize (0);
+	}
+
+	if (where == GridMaximize && resize)
+	{
+	    /* move the window to the correct output */
+	    if (GridWindow::get (cw)->grabIsMove)
+	    {
+		xwc.x = workarea.x () + 50;
+		xwc.y = workarea.y () + 50;
+		xwc.width = workarea.width ();
+		xwc.height = workarea.height ();
+		cw->configureXWindow (CWX | CWY, &xwc);
+	    }
+	    cw->maximize (MAXIMIZE_STATE);
+	    gw->isGridResized = true;
+	    gw->isGridMaximized = true;
+	    return true;
+	}
+
+	/* Convention:
+	 * xxxSlot include decorations (it's the screen area occupied)
+	 * xxxRect are undecorated (it's the constrained position
+	                            of the contents)
+	 */
+
+	/* slice and dice to get desired slot - including decorations */
+	desiredSlot.setY (workarea.y () + props.gravityDown *
+			  (workarea.height () / props.numCellsY));
+	desiredSlot.setHeight (workarea.height () / props.numCellsY);
+	desiredSlot.setX (workarea.x () + props.gravityRight *
+			  (workarea.width () / props.numCellsX));
+	desiredSlot.setWidth (workarea.width () / props.numCellsX);
+
+	/* Adjust for constraints and decorations */
+	desiredRect = constrainSize (cw, desiredSlot);
+	/* Get current rect not including decorations */
+	currentRect.setGeometry (cw->serverX (), cw->serverY (),
+				 cw->serverWidth (),
+				 cw->serverHeight ());
+
+	if (desiredRect.y () == currentRect.y () &&
+	    desiredRect.height () == currentRect.height () &&
+	    where != GridMaximize && gw->lastTarget == where)
+	{
+	    int slotWidth25  = workarea.width () / 4;
+	    int slotWidth33  = (workarea.width () / 3) + cw->input ().left;
+	    int slotWidth17  = slotWidth33 - slotWidth25;
+	    int slotWidth66  = workarea.width () - slotWidth33;
+	    int slotWidth75  = workarea.width () - slotWidth25;
+	    
+	    CompRect rect25, rect33, rect66, rect75,
+		     slot25, slot33, slot66, slot75;
+
+	    slot25 = desiredSlot;
+	    slot25.setX (workarea.x () +
+			 props.gravityRight * slotWidth75);
+	    slot25.setWidth (slotWidth25);
+	    rect25 = constrainSize (cw, slot25);
+
+	    slot33 = desiredSlot;
+	    slot33.setX (workarea.x () +
+			 props.gravityRight * slotWidth66);
+	    slot33.setWidth (slotWidth33);
+	    rect33 = constrainSize (cw, slot33);
+
+	    slot66 = desiredSlot;
+	    slot66.setX (workarea.x () +
+			 props.gravityRight * slotWidth33);
+	    slot66.setWidth (slotWidth66);
+	    rect66 = constrainSize (cw, slot66);
+
+	    slot75 = desiredSlot;
+	    slot75.setX (workarea.x () +
+			 props.gravityRight * slotWidth25);
+	    slot75.setWidth (slotWidth75);
+	    rect75 = constrainSize (cw, slot75);
+
+	    if (props.numCellsX == 2) /* keys (1, 4, 7, 3, 6, 9) */
+	    {
+		if ((currentRect.width () == desiredRect.width () &&
+		    currentRect.x () == desiredRect.x ()) ||
+		    (gw->resizeCount < 1) || (gw->resizeCount > 5))
+		    gw->resizeCount = 3;
+
+		/* tricky, have to allow for window constraints when
+		 * computing what the 33% and 66% offsets would be
+		 */
+		switch (gw->resizeCount)
+		{
+		    case 1:
+			desiredSlot.setWidth (slotWidth33);
+			desiredSlot.setX (workarea.x () +
+					  props.gravityRight * slotWidth66);
+			gw->resizeCount++;
+			break;
+		    case 2:
+			gw->resizeCount++;
+			break;
+		    case 3:
+			desiredSlot.setWidth (slotWidth66);
+			desiredSlot.setX (workarea.x () +
+					  props.gravityRight * slotWidth33);
+			gw->resizeCount++;
+			break;
+		    case 4:
+			desiredSlot.setWidth (slotWidth75);
+			desiredSlot.setX (workarea.x () +
+					  props.gravityRight * slotWidth25);
+			gw->resizeCount++;
+			break;
+		    case 5:
+			desiredSlot.setWidth (slotWidth25);
+			desiredSlot.setX (workarea.x () +
+					  props.gravityRight * slotWidth75);
+			gw->resizeCount++;
+			break;
+		    default:
+			break;
+		}
+	    }
+	    else /* keys (2, 5, 8) */
+	    {
+
+		if ((currentRect.width () == desiredRect.width () &&
+		    currentRect.x () == desiredRect.x ()) ||
+		    (gw->resizeCount < 1) || (gw->resizeCount > 5))
+		    gw->resizeCount = 1;
+	    
+		switch (gw->resizeCount)
+		{
+		    case 1:
+			desiredSlot.setWidth (slotWidth33 -
+			    (cw->input ().left + cw->input ().right));
+			desiredSlot.setX (workarea.x () + slotWidth33);
+			gw->resizeCount++;
+			break;
+		    case 2:
+			desiredSlot.setWidth ((slotWidth25 * 2));
+			desiredSlot.setX (workarea.x () + slotWidth25);
+			gw->resizeCount++;
+			break;
+		    case 3:
+			desiredSlot.setWidth ((slotWidth25 * 2) +
+					      (slotWidth17 * 2));
+			desiredSlot.setX (workarea.x () +
+					 (slotWidth25 - slotWidth17));
+			gw->resizeCount++;
+			break;
+		    case 4:
+			desiredSlot.setWidth (workarea.width () -
+					     (slotWidth17 * 2));
+			desiredSlot.setX (workarea.x () + slotWidth17);
+			gw->resizeCount++;
+			break;
+		    case 5:
+			gw->resizeCount++;
+			break;
+		    default:
+			break;
+		}
+	    }
+
+	    if (gw->resizeCount == 6)
+		gw->resizeCount = 1;
+
+	    desiredRect = constrainSize (cw, desiredSlot);
+	}
+
+	xwc.x = desiredRect.x ();
+	xwc.y = desiredRect.y ();
+	xwc.width  = desiredRect.width ();
+	xwc.height = desiredRect.height ();
+
+	/* Store a copy of xwc since configureXWindow changes it's values */
+	XWindowChanges wc = xwc;
+
+	if (cw->mapNum ())
+	    cw->sendSyncRequest ();
+
+	/* TODO: animate move+resize */
+	if (resize)
+	{
+	    cw->configureXWindow (CWX | CWY | CWWidth | CWHeight, &xwc);
+	    gw->isGridResized = true;
+	    gw->isGridMaximized = false;
+	}
+
+	/* This centers a window if it could not be resized to the desired
+	 * width. Without this, it can look buggy when desired width is
+	 * beyond the minimum or maximum width of the window.
+	 */
+	if (centerCheck)
+	{
+	    if ((cw->serverInputRect ().width () >
+		 desiredSlot.width ()) ||
+		 cw->serverInputRect ().width () <
+		 desiredSlot.width ())
+	    {
+		wc.x = (workarea.width () >> 1) -
+		      ((cw->serverInputRect ().width () >> 1) -
+			cw->input ().left);
+		cw->configureXWindow (CWX, &wc);
+	    }
+
+	    centerCheck = false;
+	}
+
+	gw->lastTarget = where;
+    }
+
+    return true;
+}
+
+void
+GridScreen::glPaintRectangle (const GLScreenPaintAttrib &sAttrib,
+			      const GLMatrix            &transform,
+			      CompOutput                *output)
+{
+    CompRect rect;
+    GLMatrix sTransform (transform);
+
+    getPaintRectangle (rect);
+
+    glPushMatrix ();
+
+    sTransform.toScreenSpace (output, -DEFAULT_Z_CAMERA);
+
+    glLoadMatrixf (sTransform.getMatrix ());
+
+    glDisableClientState (GL_TEXTURE_COORD_ARRAY);
+    glEnable (GL_BLEND);
+
+    /* fill rectangle */
+    glColor4usv (optionGetFillColor ());
+    glRecti (rect.x1 (), rect.y2 (), rect.x2 (), rect.y1 ());
+
+    /* Set outline rect smaller to avoid damage issues */
+    rect.setGeometry (rect.x () + 1, rect.y () + 1,
+		      rect.width () - 2, rect.height () - 2);
+
+    /* draw outline */
+    glColor4usv (optionGetOutlineColor ());
+    glLineWidth (2.0);
+    glBegin (GL_LINE_LOOP);
+    glVertex2i (rect.x1 (), rect.y1 ());
+    glVertex2i (rect.x2 (), rect.y1 ());
+    glVertex2i (rect.x2 (), rect.y2 ());
+    glVertex2i (rect.x1 (), rect.y2 ());
+    glEnd ();
+
+    /* clean up */
+    glColor4usv (defaultColor);
+    glDisable (GL_BLEND);
+    glEnableClientState (GL_TEXTURE_COORD_ARRAY);
+    glPopMatrix ();
+}
+
+bool
+GridScreen::glPaintOutput (const GLScreenPaintAttrib &attrib,
+			   const GLMatrix            &matrix,
+			   const CompRegion          &region,
+			   CompOutput                *output,
+			   unsigned int              mask)
+{
+    bool status;
+
+    status = glScreen->glPaintOutput (attrib, matrix, region, output, mask);
+
+    glPaintRectangle (attrib, matrix, output);
+
+    return status;
+}
+
+GridType
+GridScreen::edgeToGridType ()
+{
+    GridType ret;
+
+    switch (edge) {
+    case Left:
+	ret = (GridType) optionGetLeftEdgeAction ();
+	break;
+    case Right:
+	ret = (GridType) optionGetRightEdgeAction ();
+	break;
+    case Top:
+	ret = (GridType) optionGetTopEdgeAction ();
+	break;
+    case Bottom:
+	ret = (GridType) optionGetBottomEdgeAction ();
+	break;
+    case TopLeft:
+	ret = (GridType) optionGetTopLeftCornerAction ();
+	break;
+    case TopRight:
+	ret = (GridType) optionGetTopRightCornerAction ();
+	break;
+    case BottomLeft:
+	ret = (GridType) optionGetBottomLeftCornerAction ();
+	break;
+    case BottomRight:
+	ret = (GridType) optionGetBottomRightCornerAction ();
+	break;
+    case NoEdge:
+    default:
+	ret = GridUnknown;
+	break;
+    }
+
+    return ret;
+}
+
+void
+GridScreen::handleEvent (XEvent *event)
+{
+    screen->handleEvent (event);
+
+    if (event->type != MotionNotify)
+	return;
+
+    /* Detect corners first */
+    /* Bottom Left */
+    if (pointerY > (screen->height() - optionGetBottomEdgeThreshold()) &&
+	pointerX < optionGetLeftEdgeThreshold())
+	edge = BottomLeft;
+    /* Bottom Right */
+    else if (pointerY > (screen->height() - optionGetBottomEdgeThreshold()) &&
+	     pointerX > (screen->width() - optionGetRightEdgeThreshold()))
+	edge = BottomRight;
+    /* Top Left */
+    else if (pointerY < optionGetTopEdgeThreshold() &&
+	    pointerX < optionGetLeftEdgeThreshold())
+	edge = TopLeft;
+    /* Top Right */
+    else if (pointerY < optionGetTopEdgeThreshold() &&
+	     pointerX > (screen->width() - optionGetRightEdgeThreshold()))
+	edge = TopRight;
+    /* Left */
+    else if (pointerX < optionGetLeftEdgeThreshold())
+	edge = Left;
+    /* Right */
+    else if (pointerX > (screen->width() - optionGetRightEdgeThreshold()))
+	edge = Right;
+    /* Top */
+    else if (pointerY < optionGetTopEdgeThreshold())
+	edge = Top;
+    /* Bottom */
+    else if (pointerY > (screen->height() - optionGetBottomEdgeThreshold()))
+	edge = Bottom;
+    /* No Edge */
+    else
+	edge = NoEdge;
+
+    /* Detect when cursor enters another output */
+    currentWorkarea = screen->getWorkareaForOutput
+			    (screen->outputDeviceForPoint (pointerX, pointerY));
+    if (lastWorkarea != currentWorkarea)
+    {
+	lastWorkarea = currentWorkarea;
+
+	if (cScreen)
+	    cScreen->damageRegion (desiredSlot);
+
+	initiateCommon (0, 0, o, edgeToGridType (), false);
+
+	if (cScreen)
+	    cScreen->damageRegion (desiredSlot);
+    }
+
+    /* Detect edge region change */
+    if (lastEdge != edge)
+    {
+	lastEdge = edge;
+
+	if (cScreen)
+	    cScreen->damageRegion (desiredSlot);
+
+	initiateCommon (0, 0, o, edgeToGridType (), false);
+
+	if (cScreen)
+	    cScreen->damageRegion (desiredSlot);
+    }
+
+    GRID_WINDOW (screen->findWindow
+				(CompOption::getIntOptionNamed (o, "window")));
+
+    if ((gw->pointerBufDx > SNAPOFF_THRESHOLD ||
+	 gw->pointerBufDy > SNAPOFF_THRESHOLD ||
+	 gw->pointerBufDx < -SNAPOFF_THRESHOLD ||
+	 gw->pointerBufDy < -SNAPOFF_THRESHOLD) &&
+	 gw->isGridResized &&
+	 optionGetSnapbackWindows ())
+	    restoreWindow (0, 0, o);
+}
+
+void
+GridWindow::grabNotify (int          x,
+			int          y,
+			unsigned int state,
+			unsigned int mask)
+{
+    if (screen->grabExist ("move"))
+    {
+	gScreen->o.push_back (CompOption ("window", CompOption::TypeInt));
+	gScreen->o[0].value ().set ((int) window->id ());
+
+	screen->handleEventSetEnabled (gScreen, true);
+	gScreen->glScreen->glPaintOutputSetEnabled (gScreen, true);
+	grabIsMove = true;
+	pointerBufDx = pointerBufDy = 0;
+
+	if (!isGridResized && gScreen->optionGetSnapbackWindows ())
+	    /* Store size not including borders when grabbing with cursor */
+	    originalSize = gScreen->slotToRect(window,
+						    window->serverInputRect ());
+    }
+
+    if (screen->grabExist ("resize"))
+    {
+	isGridResized = false;
+	resizeCount = 0;
+    }
+}
+
+void
+GridWindow::ungrabNotify ()
+{
+    if (grabIsMove)
+    {
+	gScreen->initiateCommon
+			(0, 0, gScreen->o, gScreen->edgeToGridType (), true);
+
+	screen->handleEventSetEnabled (gScreen, false);
+	gScreen->glScreen->glPaintOutputSetEnabled (gScreen, false);
+	grabIsMove = false;
+	gScreen->cScreen->damageRegion (gScreen->desiredSlot);
+    }
+
+    gScreen->edge = NoEdge;
+}
+
+void
+GridWindow::moveNotify (int dx, int dy, bool immediate)
+{
+    window->moveNotify (dx, dy, immediate);
+
+    pointerBufDx += dx;
+    pointerBufDy += dy;
+}
+
+bool
+GridScreen::restoreWindow (CompAction         *action,
+			   CompAction::State  state,
+			   CompOption::Vector &option)
+{
+    XWindowChanges xwc;
+    CompWindow *cw = screen->findWindow (screen->activeWindow ());
+
+    if (!cw)
+	return false;
+
+    GRID_WINDOW (cw);
+
+    if (!gw->isGridResized)
+	return false;
+
+    if (gw->isGridMaximized & !(cw->state () & MAXIMIZE_STATE))
+	    gw->isGridMaximized = false;
+    else
+    {
+	if (gw->grabIsMove)
+	{
+	    xwc.x = pointerX - (gw->originalSize.width () >> 1);
+	    xwc.y = pointerY + (cw->input ().top >> 1);
+	}
+	else
+	{
+	    xwc.x = gw->originalSize.x ();
+	    xwc.y = gw->originalSize.y ();
+	}
+	xwc.width  = gw->originalSize.width ();
+	xwc.height = gw->originalSize.height ();
+	cw->maximize (0);
+	cw->configureXWindow (CWX | CWY | CWWidth | CWHeight, &xwc);
+	gw->pointerBufDx = 0;
+	gw->pointerBufDy = 0;
+    }
+    gw->isGridResized = false;
+    gw->resizeCount = 0;
+
+    return true;
+}
+
+void
+GridScreen::snapbackOptionChanged (CompOption *option,
+				    Options    num)
+{
+    GRID_WINDOW (screen->findWindow
+		    (CompOption::getIntOptionNamed (o, "window")));
+    gw->isGridResized = false;
+    gw->isGridMaximized = false;
+    gw->resizeCount = 0;
+}
+
+
+GridScreen::GridScreen (CompScreen *screen) :
+    PluginClassHandler<GridScreen, CompScreen> (screen),
+    cScreen (CompositeScreen::get (screen)),
+    glScreen (GLScreen::get (screen)),
+    centerCheck (false)
+{
+
+    ScreenInterface::setHandler (screen, false);
+    CompositeScreenInterface::setHandler (cScreen, false);
+    GLScreenInterface::setHandler (glScreen, false);
+
+    edge = lastEdge = NoEdge;
+    currentWorkarea = lastWorkarea = screen->getWorkareaForOutput
+			    (screen->outputDeviceForPoint (pointerX, pointerY));
+
+#define GRIDSET(opt,where,resize)					       \
+    optionSet##opt##Initiate (boost::bind (&GridScreen::initiateCommon, this,  \
+					   _1, _2, _3, where, resize))
+
+    GRIDSET (PutCenterKey, GridCenter, true);
+    GRIDSET (PutLeftKey, GridLeft, true);
+    GRIDSET (PutRightKey, GridRight, true);
+    GRIDSET (PutTopKey, GridTop, true);
+    GRIDSET (PutBottomKey, GridBottom, true);
+    GRIDSET (PutTopleftKey, GridTopLeft, true);
+    GRIDSET (PutToprightKey, GridTopRight, true);
+    GRIDSET (PutBottomleftKey, GridBottomLeft, true);
+    GRIDSET (PutBottomrightKey, GridBottomRight, true);
+    GRIDSET (PutMaximizeKey, GridMaximize, true);
+
+#undef GRIDSET
+
+    optionSetSnapbackWindowsNotify (boost::bind (&GridScreen::
+				    snapbackOptionChanged, this, _1, _2));
+
+    optionSetPutRestoreKeyInitiate (boost::bind (&GridScreen::
+					    restoreWindow, this, _1, _2, _3));
+
+}
+
+GridWindow::GridWindow (CompWindow *window) :
+    PluginClassHandler <GridWindow, CompWindow> (window),
+    window (window),
+    gScreen (GridScreen::get (screen)),
+    grabIsMove (false),
+    isGridResized (false),
+    isGridMaximized (false),
+    pointerBufDx (0),
+    pointerBufDy (0),
+    resizeCount (0),
+    lastTarget (GridUnknown)
+{
+    WindowInterface::setHandler (window);
+}
+
+/* Initial plugin init function called. Checks to see if we are ABI
+ * compatible with core, otherwise unload */
+
+bool
+GridPluginVTable::init ()
+{
+    if (!CompPlugin::checkPluginABI ("core", CORE_ABIVERSION))
+        return false;
+
+    return true;
+}
Index: compiz-0.9.4/plugins/grid/src/grid.h
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ compiz-0.9.4/plugins/grid/src/grid.h	2011-02-24 12:12:42.000000000 +0100
@@ -0,0 +1,161 @@
+/*
+ * Compiz Fusion Grid plugin
+ *
+ * Copyright (c) 2008 Stephen Kennedy <suasol@gmail.com>
+ * Copyright (c) 2010 Scott Moreau <oreaus@gmail.com>
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * Description:
+ *
+ * Plugin to act like winsplit revolution (http://www.winsplit-revolution.com/)
+ * use <Control><Alt>NUMPAD_KEY to move and tile your windows.
+ *
+ * Press the tiling keys several times to cycle through some tiling options.
+ */
+
+#include <core/core.h>
+#include <core/atoms.h>
+#include <core/pluginclasshandler.h>
+#include <composite/composite.h>
+#include <opengl/opengl.h>
+
+#include "grid_options.h"
+
+#define SNAPOFF_THRESHOLD 50
+
+typedef enum
+{
+    GridUnknown = 0,
+    GridBottomLeft = 1,
+    GridBottom = 2,
+    GridBottomRight = 3,
+    GridLeft = 4,
+    GridCenter = 5,
+    GridRight = 6,
+    GridTopLeft = 7,
+    GridTop = 8,
+    GridTopRight = 9,
+    GridMaximize = 10
+} GridType;
+
+typedef struct _GridProps
+{
+    int gravityRight;
+    int gravityDown;
+    int numCellsX;
+    int numCellsY;
+} GridProps;
+
+enum Edges
+{
+    NoEdge = 0,
+    BottomLeft,
+    Bottom,
+    BottomRight,
+    Left,
+    Right,
+    TopLeft,
+    Top,
+    TopRight
+};
+
+class GridScreen :
+    public ScreenInterface,
+    public CompositeScreenInterface,
+    public GLScreenInterface,
+    public PluginClassHandler <GridScreen, CompScreen>,
+    public GridOptions
+{
+    public:
+
+	GridScreen (CompScreen *);
+	CompositeScreen *cScreen;
+	GLScreen        *glScreen;
+
+	CompRect workarea, currentRect, desiredSlot,
+		 desiredRect, lastWorkarea, currentWorkarea;
+	GridProps props;
+	Edges edge, lastEdge;
+	CompOption::Vector o;
+	bool centerCheck;
+
+	void getPaintRectangle (CompRect&);
+
+	bool initiateCommon (CompAction*, CompAction::State,
+			     CompOption::Vector&, GridType, bool);
+
+	void glPaintRectangle (const GLScreenPaintAttrib&,
+			       const GLMatrix&, CompOutput *);
+
+	bool glPaintOutput (const GLScreenPaintAttrib &,
+			    const GLMatrix &, const CompRegion &,
+			    CompOutput *, unsigned int);
+
+	GridType edgeToGridType ();
+
+	void handleEvent (XEvent *event);
+
+	bool restoreWindow (CompAction*,
+			    CompAction::State,
+			    CompOption::Vector&);
+
+	void
+	snapbackOptionChanged (CompOption *option,
+				Options    num);
+
+	CompRect
+	slotToRect (CompWindow      *w,
+		    const CompRect& slot);
+	CompRect
+	constrainSize (CompWindow *w,
+		       const CompRect& slot);
+};
+
+class GridWindow :
+    public WindowInterface,
+    public PluginClassHandler <GridWindow, CompWindow>
+{
+    public:
+
+	GridWindow (CompWindow *);
+	CompWindow *window;
+	GridScreen *gScreen;
+
+	bool grabIsMove;
+	bool isGridResized;
+	bool isGridMaximized;
+	int pointerBufDx;
+	int pointerBufDy;
+	int resizeCount;
+	CompRect originalSize;
+	GridType lastTarget;
+
+	void grabNotify (int, int, unsigned int, unsigned int);
+
+	void ungrabNotify ();
+
+	void moveNotify (int, int, bool);
+};
+
+#define GRID_WINDOW(w) \
+    GridWindow *gw = GridWindow::get (w)
+
+class GridPluginVTable :
+    public CompPlugin::VTableForScreenAndWindow <GridScreen, GridWindow>
+{
+    public:
+
+	bool init ();
+};
+
+COMPIZ_PLUGIN_20090315 (grid, GridPluginVTable);
+
