Index: compiz-0.9.4/plugins/grid/grid.xml.in
===================================================================
--- compiz-0.9.4.orig/plugins/grid/grid.xml.in	2011-02-24 12:56:38.964211832 +0100
+++ compiz-0.9.4/plugins/grid/grid.xml.in	2011-02-24 13:06:00.650997089 +0100
@@ -84,7 +84,7 @@
 			<option name="top_left_corner_action" type="int">
 			    <_short>Upper Left Corner</_short>
 			    <_long>Action to be performed when window is dropped on the top left corner</_long>
-			    <default>7</default>
+			    <default>4</default>
 			    <min>0</min>
 			    <max>10</max>
 			    <desc>
@@ -186,7 +186,7 @@
 			<option name="top_right_corner_action" type="int">
 			    <_short>Upper Right Corner</_short>
 			    <_long>Action to be performed when window is dropped on the top right corner</_long>
-			    <default>9</default>
+			    <default>6</default>
 			    <min>0</min>
 			    <max>10</max>
 			    <desc>
@@ -339,7 +339,7 @@
 			<option name="bottom_left_corner_action" type="int">
 			    <_short>Bottom Left Corner</_short>
 			    <_long>Action to be performed when window is dropped on the bottom left corner</_long>
-			    <default>1</default>
+			    <default>4</default>
 			    <min>0</min>
 			    <max>10</max>
 			    <desc>
@@ -390,7 +390,7 @@
 			<option name="bottom_edge_action" type="int">
 			    <_short>Bottom Edge</_short>
 			    <_long>Action to be performed when window is dropped on the bottom edge</_long>
-			    <default>2</default>
+			    <default>0</default>
 			    <min>0</min>
 			    <max>10</max>
 			    <desc>
@@ -441,7 +441,7 @@
 			<option name="bottom_right_corner_action" type="int">
 			    <_short>Bottom Right Corner</_short>
 			    <_long>Action to be performed when window is dropped on the bottom right corner</_long>
-			    <default>3</default>
+			    <default>6</default>
 			    <min>0</min>
 			    <max>10</max>
 			    <desc>
