--- compiz-0.8.4.orig/kde/window-decorator/window.cpp
+++ compiz-0.8.4/kde/window-decorator/window.cpp
@@ -54,6 +54,8 @@
 #include <qcursor.h>
 #include <qpopupmenu.h>
 
+#include <unistd.h>
+
 KWD::Window::Window (WId  parentId,
 		     WId  clientId,
 		     WId  frame,
--- compiz-0.8.4.orig/kde/window-decorator-kde4/window.cpp
+++ compiz-0.8.4/kde/window-decorator-kde4/window.cpp
@@ -57,6 +57,8 @@
 #include <QStyle>
 #include <QPainter>
 
+#include <unistd.h>
+
 #include "paintredirector.h"
 
 KWD::Window::Window (WId  parentId,
