Index: compiz-0.9.8/compizconfig/ccsm/ccsm.desktop.in
===================================================================
--- compiz-0.9.8.orig/compizconfig/ccsm/ccsm.desktop.in	2012-05-29 12:09:56.000000000 +0200
+++ compiz-0.9.8/compizconfig/ccsm/ccsm.desktop.in	2012-05-29 14:54:50.181992801 +0200
@@ -1,10 +1,9 @@
 [Desktop Entry]
-Encoding=UTF-8
 _Name=CompizConfig Settings Manager
 _Comment=Configure Compiz with CompizConfig
 Icon=ccsm
 Exec=ccsm
 Terminal=false
 Type=Application
-Categories=Compiz;Settings;DesktopSettings;
+Categories=Settings;DesktopSettings;
 StartupNotify=true
