Index: compiz-0.9.8/src/main.cpp
===================================================================
--- compiz-0.9.8.orig/src/main.cpp	2012-05-29 12:09:56.000000000 +0200
+++ compiz-0.9.8/src/main.cpp	2012-05-30 16:29:24.940570174 +0200
@@ -146,23 +146,25 @@
 
     modHandler = new ModifierHandler ();
 
-    if (!initialPlugins.empty ())
-    {
-	CompOption::Value::Vector list;
-        CompOption::Value         value;
+    CompOption::Value::Vector list;
+    CompOption::Value         value;
 	CompOption                *o = screen->getOption ("active_plugins");
 
+    if (!initialPlugins.empty ())
+    {
 	foreach (CompString &str, initialPlugins)
 	{
 	    value.set (str);
 	    list.push_back (value);
 	}
+    }
+    else
+	list.push_back ("ccp");
 
 	value.set (CompOption::TypeString, list);
 
 	if (o)
 	    o->set (value);
-    }
 
     if (!screen->init (displayName))
 	return false;
