=== modified file 'ccm/Settings.py'
diff -Nur -x '*.orig' -x '*~' a/compizconfig/ccsm/ccm/Settings.py b/compizconfig/Settings.py
--- a/compizconfig/ccsm/ccm/Settings.py	2011-08-22 13:38:46.000000000 +0200
+++ b/compizconfig/ccsm/ccm/Settings.py	2012-02-01 12:44:46.254767412 +0100
@@ -441,16 +441,10 @@
         self.NoneValue = info[0]
         self.Adj = gtk.Adjustment(self.Get(), info[0], info[1], inc, inc*10)
         self.Spin = gtk.SpinButton(self.Adj)
-
-        self.Scale = gtk.HScale(self.Adj)
-
-        self.Scale.set_update_policy(gtk.UPDATE_DISCONTINUOUS)
-        self.Scale.connect("value-changed", self.Changed)
         self.Spin.connect("value-changed", self.Changed)
-        self.Widget = self.Scale
+        self.Widget = self.Spin
 
-        self.Box.pack_start(self.Scale, True, True)
-        self.Box.pack_start(self.Spin, False, False)
+        self.Box.pack_end(self.Spin, False, False)
 
     def _Read(self):
         self.Adj.set_value(self.Get())
@@ -464,7 +458,6 @@
         self.Inc = 1
         NumberSetting._Init(self)
         self.Spin.set_digits(0)
-        self.Scale.set_digits(0)
 
 class FloatSetting(NumberSetting):
 
@@ -474,7 +467,6 @@
         self.Inc = None
         NumberSetting._Init(self)
         self.Spin.set_digits(4)
-        self.Scale.set_digits(4)
 
 
 class ColorSetting(StockSetting):
