Index: compiz-0.9.8/compizconfig/libcompizconfig/src/bindings.c
===================================================================
--- compiz-0.9.8.orig/compizconfig/libcompizconfig/src/bindings.c	2012-05-29 12:09:56.000000000 +0200
+++ compiz-0.9.8/compizconfig/libcompizconfig/src/bindings.c	2012-05-29 16:22:32.382143341 +0200
@@ -65,6 +65,7 @@
 modifierList[] = {
     { "<Shift>",      ShiftMask		 },
     { "<Control>",    ControlMask	 },
+    { "<Primary>",    ControlMask	 },
     { "<Mod1>",	      Mod1Mask		 },
     { "<Mod2>",	      Mod2Mask		 },
     { "<Mod3>",	      Mod3Mask		 },
