=== modified file 'plugins/CMakeLists.txt'
--- a/plugins/CMakeLists.txt	2012-07-20 13:28:42 +0000
+++ b/plugins/CMakeLists.txt	2012-10-15 02:40:47 +0000
@@ -27,18 +27,16 @@
 
 # temporarily disable plugins that aren't ported yed
 set (COMPIZ_DISABLE_PLUGIN_ANIMATIONADDON ON)
-set (COMPIZ_DISABLE_PLUGIN_BICUBIC ON)
-set (COMPIZ_DISABLE_PLUGIN_BLUR ON)
-set (COMPIZ_DISABLE_PLUGIN_COLORFILTER ON)
-set (COMPIZ_DISABLE_PLUGIN_CUBEADDON ON)
 set (COMPIZ_DISABLE_PLUGIN_GEARS ON)
 set (COMPIZ_DISABLE_PLUGIN_GROUP ON)
+set (COMPIZ_DISABLE_PLUGIN_REFLEX ON)
 set (COMPIZ_DISABLE_PLUGIN_LOGINOUT ON)
-set (COMPIZ_DISABLE_PLUGIN_REFLEX ON)
-set (COMPIZ_DISABLE_PLUGIN_THUMBNAIL ON)
+set (COMPIZ_DISABLE_PLUGIN_BLUR ON)
+set (COMPIZ_DISABLE_PLUGIN_BICUBIC ON)
+set (COMPIZ_DISABLE_PLUGIN_COLORFILTER ON)
 set (COMPIZ_DISABLE_PLUGIN_STACKSWITCH ON)
-set (COMPIZ_DISABLE_PLUGIN_WALLPAPER ON)
 set (COMPIZ_DISABLE_PLUGIN_TRIP ON)
+set (COMPIZ_DISABLE_PLUGIN_CUBEADDON ON)
 
 # disable plugins which won't work on ES2 builds
 if (BUILD_GLES)

=== modified file 'plugins/wallpaper/CMakeLists.txt'
--- a/plugins/wallpaper/CMakeLists.txt	2012-05-16 17:44:47 +0000
+++ b/plugins/wallpaper/CMakeLists.txt	2012-10-15 02:45:38 +0000
@@ -2,4 +2,4 @@
 
 include (CompizPlugin)
 
-compiz_plugin (wallpaper PLUGINDEPS composite opengl)
+compiz_plugin (wallpaper PLUGINDEPS composite opengl compiztoolbox)

=== modified file 'plugins/wallpaper/src/wallpaper.cpp'
--- a/plugins/wallpaper/src/wallpaper.cpp	2012-07-31 13:49:40 +0000
+++ b/plugins/wallpaper/src/wallpaper.cpp	2012-10-15 02:46:08 +0000
@@ -416,7 +416,7 @@
 }
 
 void
-WallpaperWindow::drawBackgrounds (GLFragment::Attrib &attrib,
+WallpaperWindow::drawBackgrounds (const GLWindowPaintAttrib &attrib,
 				  const CompRegion &region,
 				  unsigned int mask,
 				  WallpaperBackgrounds& bg,
@@ -428,11 +428,11 @@
     GLTexture::Matrix   matrix;
     GLTexture::MatrixList tmpMatrixList;
     WallpaperBackground *back = ws->getBackgroundForViewport (bg);
-	GLFragment::Attrib tmpAttrib = attrib;
+	const GLWindowPaintAttrib tmpAttrib = attrib;
 
     tmpMatrixList.push_back (matrix);
 
-    gWindow->geometry().reset();
+    //gWindow->geometry().reset();
 
     tmpMatrixList[0] = back->fillTexMatrix[0];
 
@@ -451,17 +451,17 @@
 
     if (ws->optionGetCycleWallpapers ())
     {
-	if (fadingIn)
+	/*if (fadingIn)
 	    tmpAttrib.setOpacity ((OPAQUE * (1.0f - ws->alpha)) * (attrib.getOpacity () / (float)OPAQUE));
 	else
-	    tmpAttrib.setOpacity ((OPAQUE * ws->alpha) * (attrib.getOpacity () / (float)OPAQUE));
+	    tmpAttrib.setOpacity ((OPAQUE * ws->alpha) * (attrib.getOpacity () / (float)OPAQUE));*/
     }
 
-    if (tmpAttrib.getOpacity () != OPAQUE)
+    /*if (tmpAttrib.getOpacity () != OPAQUE)
 	mask |= PAINT_WINDOW_BLEND_MASK;
 
     if (gWindow->geometry ().vCount)
-	gWindow->glDrawTexture(back->fillTex[0], tmpAttrib, mask);
+	gWindow->glDrawTexture(back->fillTex[0], 0, tmpAttrib, mask);*/
 
     if (back->imgSize.width () && back->imgSize.height ())
     {
@@ -469,7 +469,7 @@
 	float  s1, s2;
 	int    x, y;
 
-	gWindow->geometry ().vCount = gWindow->geometry ().indexCount = 0;
+	//gWindow->geometry ().vCount = gWindow->geometry ().indexCount = 0;
 	tmpMatrixList[0] = back->imgTex[0]->matrix ();
 
 	if (back->imagePos == WallpaperOptions::BgImagePosScaleAndCrop)
@@ -558,14 +558,14 @@
 	    gWindow->glAddGeometry (tmpMatrixList, reg, region);
 	}
 
-	if (gWindow->geometry ().vCount)
-	    gWindow->glDrawTexture (back->imgTex[0], tmpAttrib, mask | PAINT_WINDOW_BLEND_MASK);
+	/*if (gWindow->geometry ().vCount)*/
+	    gWindow->glDrawTexture (back->imgTex[0], 0, tmpAttrib, mask | PAINT_WINDOW_BLEND_MASK);
     }
 }
 
 bool
 WallpaperWindow::glDraw (const GLMatrix &transform,
-			 GLFragment::Attrib &attrib,
+			 const GLWindowPaintAttrib &attrib,
 			 const CompRegion &region,
 			 unsigned int mask)
 {
@@ -596,7 +596,7 @@
 	ws->gScreen->setFilter (filterIdx, saveFilter);
 
 	ws->desktop = window;
-	attrib.setOpacity (OPAQUE);
+	//window->setOpacity (OPAQUE);
     }
 
     return gWindow->glDraw (transform, attrib, region, mask);

=== modified file 'plugins/wallpaper/src/wallpaper.h'
--- a/plugins/wallpaper/src/wallpaper.h	2010-02-12 19:03:03 +0000
+++ b/plugins/wallpaper/src/wallpaper.h	2012-10-15 02:35:01 +0000
@@ -132,11 +132,11 @@
         GLWindow        *gWindow;
 
 	void
-	drawBackgrounds (GLFragment::Attrib &,
+	drawBackgrounds (const GLWindowPaintAttrib &,
 			 const CompRegion &, unsigned int,
 			 WallpaperBackgrounds&, bool);
 
-	bool glDraw (const GLMatrix &, GLFragment::Attrib &,
+	bool glDraw (const GLMatrix &, const GLWindowPaintAttrib &,
 		     const CompRegion &, unsigned int);
 
 	bool damageRect (bool, const CompRect &);

