From d7a39b33407927005f65755b8a3a608394aaeccd Mon Sep 17 00:00:00 2001
From: Diego Biurrun <diego@biurrun.de>
Date: Mon, 22 Oct 2012 16:01:18 +0200
Subject: [PATCH 005/278] doxygen: Build Doxygen documentation in the doc/
 subdirectory

---
 .gitignore |    2 +-
 Doxyfile   |    2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/.gitignore b/.gitignore
index 8589127..3ed55b3 100644
--- a/.gitignore
+++ b/.gitignore
@@ -28,8 +28,8 @@
 /doc/*.pod
 /doc/avoptions_codec.texi
 /doc/avoptions_format.texi
+/doc/doxy/html/
 /doc/print_options
-/doxy/
 /libavcodec/*_tablegen
 /libavcodec/*_tables.c
 /libavcodec/*_tables.h
diff --git a/Doxyfile b/Doxyfile
index 8e0dcf3..1b4e7d5 100644
--- a/Doxyfile
+++ b/Doxyfile
@@ -44,7 +44,7 @@ PROJECT_LOGO           =
 # If a relative path is entered, it will be relative to the location
 # where doxygen was started. If left blank the current directory will be used.
 
-OUTPUT_DIRECTORY       = doxy
+OUTPUT_DIRECTORY       = doc/doxy
 
 # If the CREATE_SUBDIRS tag is set to YES, then doxygen will create
 # 4096 sub-directories (in 2 levels) under the output directory of each output
-- 
1.7.9.5

