From a5ef830b1226273c35d4baa1c2e4e7e1c9de7c7d Mon Sep 17 00:00:00 2001
From: Justin Ruggles <justin.ruggles@gmail.com>
Date: Tue, 23 Oct 2012 13:25:53 -0400
Subject: [PATCH 007/278] lavc: use the correct API version guard macro for
 avcodec_encode_audio()

---
 libavcodec/utils.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libavcodec/utils.c b/libavcodec/utils.c
index 5e22b9f..10230ca 100644
--- a/libavcodec/utils.c
+++ b/libavcodec/utils.c
@@ -1073,7 +1073,7 @@ end:
     return ret;
 }
 
-#if FF_API_OLD_DECODE_AUDIO
+#if FF_API_OLD_ENCODE_AUDIO
 int attribute_align_arg avcodec_encode_audio(AVCodecContext *avctx,
                                              uint8_t *buf, int buf_size,
                                              const short *samples)
-- 
1.7.9.5

