From c68317ebbe4915035df0b08c23eea7a0b80ab881 Mon Sep 17 00:00:00 2001
From: Justin Ruggles <justin.ruggles@gmail.com>
Date: Tue, 2 Oct 2012 11:38:34 -0400
Subject: [PATCH 009/278] lavc: fix documentation for AVCodecContext.delay

---
 libavcodec/avcodec.h |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libavcodec/avcodec.h b/libavcodec/avcodec.h
index 64d45db..c505a92 100644
--- a/libavcodec/avcodec.h
+++ b/libavcodec/avcodec.h
@@ -1442,7 +1442,7 @@ typedef struct AVCodecContext {
     int ticks_per_frame;
 
     /**
-     * Encoder delay.
+     * Codec delay.
      *
      * Video:
      *   Number of frames the decoded output will be delayed relative to the
-- 
1.7.9.5

