From 8e84f2055c4170e502d36db3539b5e243a5b6cca Mon Sep 17 00:00:00 2001
From: Anton Khirnov <anton@khirnov.net>
Date: Tue, 23 Oct 2012 08:32:56 +0200
Subject: [PATCH 011/278] avconv: remove now unneeded calls to
 avcodec_get_frame_defaults().

---
 avconv.c |    4 ----
 1 file changed, 4 deletions(-)

diff --git a/avconv.c b/avconv.c
index 6f67711..bf1e2fa 100644
--- a/avconv.c
+++ b/avconv.c
@@ -1079,8 +1079,6 @@ static int decode_audio(InputStream *ist, AVPacket *pkt, int *got_output)
 
     if (!ist->decoded_frame && !(ist->decoded_frame = avcodec_alloc_frame()))
         return AVERROR(ENOMEM);
-    else
-        avcodec_get_frame_defaults(ist->decoded_frame);
     decoded_frame = ist->decoded_frame;
 
     ret = avcodec_decode_audio4(avctx, decoded_frame, got_output, pkt);
@@ -1220,8 +1218,6 @@ static int decode_video(InputStream *ist, AVPacket *pkt, int *got_output)
 
     if (!ist->decoded_frame && !(ist->decoded_frame = avcodec_alloc_frame()))
         return AVERROR(ENOMEM);
-    else
-        avcodec_get_frame_defaults(ist->decoded_frame);
     decoded_frame = ist->decoded_frame;
 
     ret = avcodec_decode_video2(ist->st->codec,
-- 
1.7.9.5

