From 5a9567631a909a4c76dc678ebd603ffc4d57262d Mon Sep 17 00:00:00 2001
From: Anton Khirnov <anton@khirnov.net>
Date: Wed, 17 Oct 2012 12:16:00 +0200
Subject: [PATCH 012/278] g.723.1: add missing CODEC_CAP_DR1

---
 libavcodec/g723_1.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libavcodec/g723_1.c b/libavcodec/g723_1.c
index 3d4fa46..5c213da 100644
--- a/libavcodec/g723_1.c
+++ b/libavcodec/g723_1.c
@@ -1376,6 +1376,6 @@ AVCodec ff_g723_1_decoder = {
     .init           = g723_1_decode_init,
     .decode         = g723_1_decode_frame,
     .long_name      = NULL_IF_CONFIG_SMALL("G.723.1"),
-    .capabilities   = CODEC_CAP_SUBFRAMES,
+    .capabilities   = CODEC_CAP_SUBFRAMES | CODEC_CAP_DR1,
     .priv_class     = &g723_1dec_class,
 };
-- 
1.7.9.5

