From d4c99513f41272b9753e59642724717b834710a0 Mon Sep 17 00:00:00 2001
From: Mans Rullgard <mans@mansr.com>
Date: Tue, 23 Oct 2012 23:33:44 +0100
Subject: [PATCH 015/278] configure: generalise 64-bit test

Signed-off-by: Mans Rullgard <mans@mansr.com>
---
 configure |   11 +++++++++--
 1 file changed, 9 insertions(+), 2 deletions(-)

diff --git a/configure b/configure
index 37cdf1e..b1c6d45 100755
--- a/configure
+++ b/configure
@@ -2668,13 +2668,20 @@ EOF
 check_host_cflags -std=c99
 check_host_cflags -Wall
 
+check_64bit(){
+    arch32=$1
+    arch64=$2
+    expr=$3
+    check_code cc "" "int test[2*($expr) - 1]" &&
+        subarch=$arch64 || subarch=$arch32
+}
+
 case "$arch" in
     alpha|ia64|mips|parisc|sparc)
         spic=$shared
     ;;
     x86)
-        subarch="x86_32"
-        check_code cc "" "int test[(int)sizeof(char*) - 7]" && subarch="x86_64"
+        check_64bit x86_32 x86_64 'sizeof(void *) > 4'
         if test "$subarch" = "x86_64"; then
             spic=$shared
         fi
-- 
1.7.9.5

