From 2acda282eb8efac556cb351e91c557e580406294 Mon Sep 17 00:00:00 2001
From: Mans Rullgard <mans@mansr.com>
Date: Tue, 23 Oct 2012 23:34:35 +0100
Subject: [PATCH 016/278] configure: detect mips64 automatically

Signed-off-by: Mans Rullgard <mans@mansr.com>
---
 configure |   12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/configure b/configure
index b1c6d45..9f6eda5 100755
--- a/configure
+++ b/configure
@@ -2453,13 +2453,9 @@ case "$arch" in
     arm*)
         arch="arm"
     ;;
-    mips|mipsel|IP*)
+    mips*|IP*)
         arch="mips"
     ;;
-    mips64*)
-        arch="mips"
-        subarch="mips64"
-    ;;
     parisc|hppa)
         arch="parisc"
     ;;
@@ -2677,7 +2673,11 @@ check_64bit(){
 }
 
 case "$arch" in
-    alpha|ia64|mips|parisc|sparc)
+    alpha|ia64|parisc|sparc)
+        spic=$shared
+    ;;
+    mips)
+        check_64bit mips mips64 '_MIPS_SIM > 1'
         spic=$shared
     ;;
     x86)
-- 
1.7.9.5

