From 56203596aea365387917c40964c03b5156b15e22 Mon Sep 17 00:00:00 2001
From: Mans Rullgard <mans@mansr.com>
Date: Tue, 23 Oct 2012 23:39:40 +0100
Subject: [PATCH 017/278] configure: detect ppc64 automatically

Signed-off-by: Mans Rullgard <mans@mansr.com>
---
 configure |    9 ++++-----
 1 file changed, 4 insertions(+), 5 deletions(-)

diff --git a/configure b/configure
index 9f6eda5..1f48fbe 100755
--- a/configure
+++ b/configure
@@ -2463,13 +2463,9 @@ case "$arch" in
         arch="parisc"
         subarch="parisc64"
     ;;
-    "Power Macintosh"|ppc|powerpc)
+    "Power Macintosh"|ppc*|powerpc*)
         arch="ppc"
     ;;
-    ppc64|powerpc64)
-        arch="ppc"
-        subarch="ppc64"
-    ;;
     s390|s390x)
         arch="s390"
     ;;
@@ -2680,6 +2676,9 @@ case "$arch" in
         check_64bit mips mips64 '_MIPS_SIM > 1'
         spic=$shared
     ;;
+    ppc)
+        check_64bit ppc ppc64 'sizeof(void *) > 4'
+    ;;
     x86)
         check_64bit x86_32 x86_64 'sizeof(void *) > 4'
         if test "$subarch" = "x86_64"; then
-- 
1.7.9.5

