From a6e9d6497739b7b60e3284f03b27883514bbc94a Mon Sep 17 00:00:00 2001
From: Mans Rullgard <mans@mansr.com>
Date: Wed, 24 Oct 2012 00:21:15 +0100
Subject: [PATCH 018/278] configure: detect parisc64 automatically

Signed-off-by: Mans Rullgard <mans@mansr.com>
---
 configure |   12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/configure b/configure
index 1f48fbe..079e6db 100755
--- a/configure
+++ b/configure
@@ -2456,13 +2456,9 @@ case "$arch" in
     mips*|IP*)
         arch="mips"
     ;;
-    parisc|hppa)
+    parisc*|hppa*)
         arch="parisc"
     ;;
-    parisc64|hppa64)
-        arch="parisc"
-        subarch="parisc64"
-    ;;
     "Power Macintosh"|ppc*|powerpc*)
         arch="ppc"
     ;;
@@ -2669,13 +2665,17 @@ check_64bit(){
 }
 
 case "$arch" in
-    alpha|ia64|parisc|sparc)
+    alpha|ia64|sparc)
         spic=$shared
     ;;
     mips)
         check_64bit mips mips64 '_MIPS_SIM > 1'
         spic=$shared
     ;;
+    parisc)
+        check_64bit parisc parisc64 'sizeof(void *) > 4'
+        spic=$shared
+    ;;
     ppc)
         check_64bit ppc ppc64 'sizeof(void *) > 4'
     ;;
-- 
1.7.9.5

