From eadfb0560a2f194fbc453bcb22fea73f3c9a27ad Mon Sep 17 00:00:00 2001
From: Michael Kostylev <michael.kostylev@gmail.com>
Date: Thu, 25 Oct 2012 11:38:51 +0100
Subject: [PATCH 031/278] configure: recognise more sparc variants as --cpu
 argument

Signed-off-by: Mans Rullgard <mans@mansr.com>
---
 configure |    6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/configure b/configure
index df07a6f..a39cb77 100755
--- a/configure
+++ b/configure
@@ -2568,12 +2568,12 @@ elif enabled x86; then
 elif enabled sparc; then
 
     case $cpu in
-        niagara)
+        cypress|f93[04]|tsc701|sparcl*|supersparc|hypersparc|niagara|v[789])
             cpuflags="-mcpu=$cpu"
             disable vis
         ;;
-        sparc64)
-            cpuflags="-mcpu=v9"
+        ultrasparc*|niagara[234])
+            cpuflags="-mcpu=$cpu"
         ;;
     esac
 
-- 
1.7.9.5

