From 74e742d6ad09becc2d43e5382c141e0356ad6071 Mon Sep 17 00:00:00 2001
From: Diego Biurrun <diego@biurrun.de>
Date: Tue, 23 Oct 2012 21:12:48 +0200
Subject: [PATCH 034/278] doxygen: Drop some pointless entries from PREDEFINED
 macros list

---
 doc/Doxyfile |    6 ------
 1 file changed, 6 deletions(-)

diff --git a/doc/Doxyfile b/doc/Doxyfile
index 1b4e7d5..eb394f0 100644
--- a/doc/Doxyfile
+++ b/doc/Doxyfile
@@ -1374,12 +1374,6 @@ INCLUDE_FILE_PATTERNS  =
 # instead of the = operator.
 
 PREDEFINED             = "__attribute__(x)=" \
-                         "RENAME(x)=x ## _TMPL" \
-                         "DEF(x)=x ## _TMPL" \
-                         HAVE_AV_CONFIG_H \
-                         HAVE_MMX \
-                         HAVE_MMXEXT \
-                         HAVE_AMD3DNOW \
                          "DECLARE_ALIGNED(a,t,n)=t n" \
                          "offsetof(x,y)=0x42"
 
-- 
1.7.9.5

