From 8a61ba0e8194beffdfd9f843bdcf29dbbc974ca5 Mon Sep 17 00:00:00 2001
From: Diego Biurrun <diego@biurrun.de>
Date: Wed, 24 Oct 2012 16:03:31 +0200
Subject: [PATCH 038/278] cook: Remove senseless maybe_reformat_buffer32()
 function

---
 libavcodec/cook.c |    8 +-------
 1 file changed, 1 insertion(+), 7 deletions(-)

diff --git a/libavcodec/cook.c b/libavcodec/cook.c
index 05efa50..713d0b1 100644
--- a/libavcodec/cook.c
+++ b/libavcodec/cook.c
@@ -241,17 +241,11 @@ static av_cold int init_cook_mlt(COOKContext *q)
     return 0;
 }
 
-static const float *maybe_reformat_buffer32(COOKContext *q, const float *ptr, int n)
-{
-    if (1)
-        return ptr;
-}
-
 static av_cold void init_cplscales_table(COOKContext *q)
 {
     int i;
     for (i = 0; i < 5; i++)
-        q->cplscales[i] = maybe_reformat_buffer32(q, cplscales[i], (1 << (i + 2)) - 1);
+        q->cplscales[i] = cplscales[i];
 }
 
 /*************** init functions end ***********/
-- 
1.7.9.5

