From 6b776c61da26428ca67abec919c8a186a276ca66 Mon Sep 17 00:00:00 2001
From: Mans Rullgard <mans@mansr.com>
Date: Fri, 26 Oct 2012 17:58:19 +0100
Subject: [PATCH 043/278] buffersink: remove stray semicolon after function
 definition

Signed-off-by: Mans Rullgard <mans@mansr.com>
---
 libavfilter/buffersink.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libavfilter/buffersink.c b/libavfilter/buffersink.c
index 73af25d..c68b909 100644
--- a/libavfilter/buffersink.c
+++ b/libavfilter/buffersink.c
@@ -57,7 +57,7 @@ static int start_frame(AVFilterLink *link, AVFilterBufferRef *buf)
     link->cur_buf = NULL;
 
     return 0;
-};
+}
 
 int av_buffersink_read(AVFilterContext *ctx, AVFilterBufferRef **buf)
 {
-- 
1.7.9.5

