From 67a68dcec2271336c859d1fbd8f40c268f54ec8a Mon Sep 17 00:00:00 2001
From: Mans Rullgard <mans@mansr.com>
Date: Fri, 26 Oct 2012 22:25:37 +0100
Subject: [PATCH 044/278] af_channelmap: remove stray enum declaration

Signed-off-by: Mans Rullgard <mans@mansr.com>
---
 libavfilter/af_channelmap.c |    1 -
 1 file changed, 1 deletion(-)

diff --git a/libavfilter/af_channelmap.c b/libavfilter/af_channelmap.c
index 8c49d10..f70b481 100644
--- a/libavfilter/af_channelmap.c
+++ b/libavfilter/af_channelmap.c
@@ -124,7 +124,6 @@ static av_cold int channelmap_init(AVFilterContext *ctx, const char *args)
     ChannelMapContext *s = ctx->priv;
     int ret;
     char *mapping;
-    enum mode;
     int map_entries = 0;
     char buf[256];
     enum MappingMode mode;
-- 
1.7.9.5

