From 4521645b1aee9e9ad8f5cea7b2392cd5f6ffcd26 Mon Sep 17 00:00:00 2001
From: Mans Rullgard <mans@mansr.com>
Date: Fri, 26 Oct 2012 18:42:29 +0100
Subject: [PATCH 048/278] avio: fix pointer type mismatches in
 avio_enum_protocols()

Signed-off-by: Mans Rullgard <mans@mansr.com>
---
 libavformat/avio.c |   10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/libavformat/avio.c b/libavformat/avio.c
index 45ee866..a43b241 100644
--- a/libavformat/avio.c
+++ b/libavformat/avio.c
@@ -85,11 +85,11 @@ const AVClass ffurl_context_class = {
 
 const char *avio_enum_protocols(void **opaque, int output)
 {
-    URLProtocol **p = opaque;
-    *p = ffurl_protocol_next(*p);
-    if (!*p) return NULL;
-    if ((output && (*p)->url_write) || (!output && (*p)->url_read))
-        return (*p)->name;
+    URLProtocol *p;
+    *opaque = ffurl_protocol_next(*opaque);
+    if (!(p = *opaque)) return NULL;
+    if ((output && p->url_write) || (!output && p->url_read))
+        return p->name;
     return avio_enum_protocols(opaque, output);
 }
 
-- 
1.7.9.5

