From 4cd217c065e948a9cec38fef9c544d3c26ff8f37 Mon Sep 17 00:00:00 2001
From: Mans Rullgard <mans@mansr.com>
Date: Wed, 24 Oct 2012 18:34:21 +0100
Subject: [PATCH 052/278] configure: detect sparc64 automatically

Signed-off-by: Mans Rullgard <mans@mansr.com>
---
 configure |    9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/configure b/configure
index 81f945f..1a006be 100755
--- a/configure
+++ b/configure
@@ -2474,9 +2474,8 @@ case "$arch" in
     sh4|sh)
         arch="sh4"
     ;;
-    sun4u|sparc64)
+    sun4u|sparc*)
         arch="sparc"
-        subarch="sparc64"
     ;;
     tilegx|tile-gx)
         arch="tilegx"
@@ -2671,7 +2670,7 @@ check_64bit(){
 }
 
 case "$arch" in
-    alpha|ia64|sparc)
+    alpha|ia64)
         spic=$shared
     ;;
     mips)
@@ -2685,6 +2684,10 @@ case "$arch" in
     ppc)
         check_64bit ppc ppc64 'sizeof(void *) > 4'
     ;;
+    sparc)
+        check_64bit sparc sparc64 'sizeof(void *) > 4'
+        spic=$shared
+    ;;
     x86)
         check_64bit x86_32 x86_64 'sizeof(void *) > 4'
         if test "$subarch" = "x86_64"; then
-- 
1.7.9.5

