From 70478746a94eba6467c02f1c2ae8d8090ad9168e Mon Sep 17 00:00:00 2001
From: Diego Biurrun <diego@biurrun.de>
Date: Wed, 24 Oct 2012 19:18:12 +0200
Subject: [PATCH 063/278] avconv: Drop unused function argument from
 do_video_stats()

---
 avconv.c |    5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/avconv.c b/avconv.c
index 54a0e2a..d0902a6 100644
--- a/avconv.c
+++ b/avconv.c
@@ -621,8 +621,7 @@ static double psnr(double d)
     return -10.0 * log(d) / log(10.0);
 }
 
-static void do_video_stats(AVFormatContext *os, OutputStream *ost,
-                           int frame_size)
+static void do_video_stats(OutputStream *ost, int frame_size)
 {
     AVCodecContext *enc;
     int frame_number;
@@ -706,7 +705,7 @@ static int poll_filter(OutputStream *ost)
 
         do_video_out(of->ctx, ost, filtered_frame, &frame_size);
         if (vstats_filename && frame_size)
-            do_video_stats(of->ctx, ost, frame_size);
+            do_video_stats(ost, frame_size);
         break;
     case AVMEDIA_TYPE_AUDIO:
         do_audio_out(of->ctx, ost, filtered_frame);
-- 
1.7.9.5

