From bff5e5f8b3c5892a21a0df7b3352b31511e42a94 Mon Sep 17 00:00:00 2001
From: Anton Khirnov <anton@khirnov.net>
Date: Mon, 29 Oct 2012 12:16:05 +0100
Subject: [PATCH 067/278] indeo3: remove duplicate capabilities line.

---
 libavcodec/indeo3.c |    1 -
 1 file changed, 1 deletion(-)

diff --git a/libavcodec/indeo3.c b/libavcodec/indeo3.c
index 322d360..fc0d2bc 100644
--- a/libavcodec/indeo3.c
+++ b/libavcodec/indeo3.c
@@ -1114,5 +1114,4 @@ AVCodec ff_indeo3_decoder = {
     .decode         = decode_frame,
     .capabilities   = CODEC_CAP_DR1,
     .long_name      = NULL_IF_CONFIG_SMALL("Intel Indeo 3"),
-    .capabilities   = CODEC_CAP_DR1,
 };
-- 
1.7.9.5

