From 14f031d7ecfabba0ef02776d4516aa3dcb7c40d8 Mon Sep 17 00:00:00 2001
From: Anton Khirnov <anton@khirnov.net>
Date: Wed, 24 Oct 2012 09:06:32 +0200
Subject: [PATCH 074/278] dv: use AVStream.index instead of abusing
 AVStream.id

---
 libavformat/dv.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libavformat/dv.c b/libavformat/dv.c
index 70786b1..17c545c 100644
--- a/libavformat/dv.c
+++ b/libavformat/dv.c
@@ -372,7 +372,7 @@ int avpriv_dv_produce_packet(DVDemuxContext *c, AVPacket *pkt,
     pkt->data         = buf;
     pkt->size         = size;
     pkt->flags       |= AV_PKT_FLAG_KEY;
-    pkt->stream_index = c->vst->id;
+    pkt->stream_index = c->vst->index;
     pkt->pts          = c->frames;
 
     c->frames++;
-- 
1.7.9.5

