From 02e636425970fc9b9aebb31dcc7abbd36e54af2a Mon Sep 17 00:00:00 2001
From: Anton Khirnov <anton@khirnov.net>
Date: Mon, 29 Oct 2012 21:26:46 +0100
Subject: [PATCH 080/278] FATE: fix (AD)PCM test dependencies broken in
 e519990

Signed-off-by: Diego Biurrun <diego@biurrun.de>
---
 tests/fate/acodec.mak |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/tests/fate/acodec.mak b/tests/fate/acodec.mak
index 531f2f3..71521ab 100644
--- a/tests/fate/acodec.mak
+++ b/tests/fate/acodec.mak
@@ -19,7 +19,7 @@ FATE_ACODEC_PCM-$(call ENCDEC, PCM_F64BE, AU)  += f64be
 FATE_ACODEC_PCM-$(call ENCDEC, PCM_F64LE, WAV) += f64le
 
 FATE_ACODEC_PCM := $(FATE_ACODEC_PCM-yes:%=fate-acodec-pcm-%)
-FATE_AVCONV += $(FATE_ACODEC_PCM)
+FATE_ACODEC += $(FATE_ACODEC_PCM)
 fate-acodec-pcm: $(FATE_ACODEC_PCM)
 
 fate-acodec-pcm-%: FMT = wav
@@ -37,7 +37,7 @@ FATE_ACODEC_ADPCM-$(call ENCDEC, ADPCM_SWF,     FLV)  += swf
 FATE_ACODEC_ADPCM-$(call ENCDEC, ADPCM_YAMAHA,  WAV)  += yamaha
 
 FATE_ACODEC_ADPCM := $(FATE_ACODEC_ADPCM-yes:%=fate-acodec-adpcm-%)
-FATE_AVCONV += $(FATE_ACODEC_ADPCM)
+FATE_ACODEC += $(FATE_ACODEC_ADPCM)
 fate-acodec-adpcm: $(FATE_ACODEC_ADPCM)
 
 fate-acodec-adpcm-%: CODEC = adpcm_$(@:fate-acodec-adpcm-%=%)
-- 
1.7.9.5

