From ab9545a290d1f48eb4c361ecd0b5df0a966a9a79 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Martin=20Storsj=C3=B6?= <martin@martin.st>
Date: Tue, 30 Oct 2012 12:09:57 +0200
Subject: [PATCH 089/278] cng: Reindent some incorrectly indented lines
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Martin Storsjö <martin@martin.st>
---
 libavcodec/cngdec.c |    2 +-
 libavcodec/cngenc.c |    2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/libavcodec/cngdec.c b/libavcodec/cngdec.c
index 4fe7839..caaa9c1 100644
--- a/libavcodec/cngdec.c
+++ b/libavcodec/cngdec.c
@@ -102,7 +102,7 @@ static void cng_decode_flush(AVCodecContext *avctx)
 }
 
 static int cng_decode_frame(AVCodecContext *avctx, void *data,
-                              int *got_frame_ptr, AVPacket *avpkt)
+                            int *got_frame_ptr, AVPacket *avpkt)
 {
 
     CNGContext *p = avctx->priv_data;
diff --git a/libavcodec/cngenc.c b/libavcodec/cngenc.c
index 1e3f8f0..a553a3f 100644
--- a/libavcodec/cngenc.c
+++ b/libavcodec/cngenc.c
@@ -67,7 +67,7 @@ static av_cold int cng_encode_init(AVCodecContext *avctx)
 }
 
 static int cng_encode_frame(AVCodecContext *avctx, AVPacket *avpkt,
-                             const AVFrame *frame, int *got_packet_ptr)
+                            const AVFrame *frame, int *got_packet_ptr)
 {
     CNGContext *p = avctx->priv_data;
     int ret, i;
-- 
1.7.9.5

