From a4202003b21ee88c82eec909a0ad086b4c328903 Mon Sep 17 00:00:00 2001
From: Justin Ruggles <justin.ruggles@gmail.com>
Date: Sun, 7 Oct 2012 20:52:35 -0400
Subject: [PATCH 107/278] dca_parser: allow the parser to change the sample
 rate

---
 libavcodec/dca_parser.c |    3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/libavcodec/dca_parser.c b/libavcodec/dca_parser.c
index 7e65d0b..ab235cf 100644
--- a/libavcodec/dca_parser.c
+++ b/libavcodec/dca_parser.c
@@ -192,8 +192,7 @@ static int dca_parse(AVCodecParserContext * s,
     /* read the duration and sample rate from the frame header */
     if (!dca_parse_params(buf, buf_size, &duration, &sample_rate)) {
         s->duration = duration;
-        if (!avctx->sample_rate)
-            avctx->sample_rate = sample_rate;
+        avctx->sample_rate = sample_rate;
     } else
         s->duration = 0;
 
-- 
1.7.9.5

