From d21b2e4726822ec1c604e2b9010a1a2cc0d88aec Mon Sep 17 00:00:00 2001
From: Justin Ruggles <justin.ruggles@gmail.com>
Date: Sun, 21 Oct 2012 14:33:11 -0400
Subject: [PATCH 117/278] cook: reverse a condition so that the code makes
 more sense

---
 libavcodec/cook.c |    5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/libavcodec/cook.c b/libavcodec/cook.c
index f6d2e66..211fe12 100644
--- a/libavcodec/cook.c
+++ b/libavcodec/cook.c
@@ -1244,9 +1244,8 @@ static av_cold int cook_decode_init(AVCodecContext *avctx)
     }
 
     /* Try to catch some obviously faulty streams, othervise it might be exploitable */
-    if ((q->samples_per_channel == 256) || (q->samples_per_channel == 512)
-                || (q->samples_per_channel == 1024)) {
-    } else {
+    if (q->samples_per_channel != 256 && q->samples_per_channel != 512 &&
+        q->samples_per_channel != 1024) {
         av_log_ask_for_sample(avctx,
                               "unknown amount of samples_per_channel = %d\n",
                               q->samples_per_channel);
-- 
1.7.9.5

