From 99d868635725e3b85a5c549e6bb0e97e10cf5248 Mon Sep 17 00:00:00 2001
From: Justin Ruggles <justin.ruggles@gmail.com>
Date: Sun, 21 Oct 2012 17:04:58 -0400
Subject: [PATCH 124/278] flacdec: do not warn on sample rate change

---
 libavcodec/flacdec.c |    6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/libavcodec/flacdec.c b/libavcodec/flacdec.c
index 20af820..10e33f2 100644
--- a/libavcodec/flacdec.c
+++ b/libavcodec/flacdec.c
@@ -460,12 +460,8 @@ static int decode_frame(FLACContext *s)
                                         " or frame header\n");
         return -1;
     }
-    if (fi.samplerate == 0) {
+    if (fi.samplerate == 0)
         fi.samplerate = s->samplerate;
-    } else if (s->samplerate && fi.samplerate != s->samplerate) {
-        av_log(s->avctx, AV_LOG_WARNING, "sample rate changed from %d to %d\n",
-               s->samplerate, fi.samplerate);
-    }
     s->samplerate = s->avctx->sample_rate = fi.samplerate;
 
     if (!s->got_streaminfo) {
-- 
1.7.9.5

