From 30f8da29bf609d741bbebd33b2a5003c426ab919 Mon Sep 17 00:00:00 2001
From: Justin Ruggles <justin.ruggles@gmail.com>
Date: Mon, 22 Oct 2012 17:41:47 -0400
Subject: [PATCH 131/278] libilbc: set channel layout

---
 libavcodec/libilbc.c |    8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/libavcodec/libilbc.c b/libavcodec/libilbc.c
index a93285c..2812061 100644
--- a/libavcodec/libilbc.c
+++ b/libavcodec/libilbc.c
@@ -21,6 +21,7 @@
 
 #include <ilbc.h>
 
+#include "libavutil/audioconvert.h"
 #include "avcodec.h"
 #include "libavutil/common.h"
 #include "libavutil/opt.h"
@@ -68,9 +69,10 @@ static av_cold int ilbc_decode_init(AVCodecContext *avctx)
     avcodec_get_frame_defaults(&s->frame);
     avctx->coded_frame = &s->frame;
 
-    avctx->channels = 1;
-    avctx->sample_rate = 8000;
-    avctx->sample_fmt = AV_SAMPLE_FMT_S16;
+    avctx->channels       = 1;
+    avctx->channel_layout = AV_CH_LAYOUT_MONO;
+    avctx->sample_rate    = 8000;
+    avctx->sample_fmt     = AV_SAMPLE_FMT_S16;
 
     return 0;
 }
-- 
1.7.9.5

