From d26701ce2f3504e2ee341251448db3af6328a69b Mon Sep 17 00:00:00 2001
From: Justin Ruggles <justin.ruggles@gmail.com>
Date: Mon, 22 Oct 2012 17:58:24 -0400
Subject: [PATCH 133/278] nellymoserdec: set channels to 1

---
 libavcodec/nellymoserdec.c |    1 +
 1 file changed, 1 insertion(+)

diff --git a/libavcodec/nellymoserdec.c b/libavcodec/nellymoserdec.c
index fce184a..f17e9fd 100644
--- a/libavcodec/nellymoserdec.c
+++ b/libavcodec/nellymoserdec.c
@@ -129,6 +129,7 @@ static av_cold int decode_init(AVCodecContext * avctx) {
     if (!ff_sine_128[127])
         ff_init_ff_sine_windows(7);
 
+    avctx->channels       = 1;
     avctx->channel_layout = AV_CH_LAYOUT_MONO;
 
     avcodec_get_frame_defaults(&s->frame);
-- 
1.7.9.5

