From eb38d8fe926bdce8110fa4be4fddf6598a079a20 Mon Sep 17 00:00:00 2001
From: Justin Ruggles <justin.ruggles@gmail.com>
Date: Mon, 22 Oct 2012 18:54:29 -0400
Subject: [PATCH 136/278] qdm2: remove unneeded checks for channel count

---
 libavcodec/qdm2.c |    8 --------
 1 file changed, 8 deletions(-)

diff --git a/libavcodec/qdm2.c b/libavcodec/qdm2.c
index 8e93886..a094e6c 100644
--- a/libavcodec/qdm2.c
+++ b/libavcodec/qdm2.c
@@ -544,10 +544,6 @@ static void fill_tone_level_array (QDM2Context *q, int flag)
     int i, sb, ch, sb_used;
     int tmp, tab;
 
-    // This should never happen
-    if (q->nb_channels <= 0)
-        return;
-
     for (ch = 0; ch < q->nb_channels; ch++)
         for (sb = 0; sb < 30; sb++)
             for (i = 0; i < 8; i++) {
@@ -643,10 +639,6 @@ static void fill_coding_method_array (sb_int8_array tone_level_idx, sb_int8_arra
     int add1, add2, add3, add4;
     int64_t multres;
 
-    // This should never happen
-    if (nb_channels <= 0)
-        return;
-
     if (!superblocktype_2_3) {
         /* This case is untested, no samples available */
         SAMPLES_NEEDED
-- 
1.7.9.5

