From 523734eb6a904c43cf73d801f1c885829380f0de Mon Sep 17 00:00:00 2001
From: Justin Ruggles <justin.ruggles@gmail.com>
Date: Tue, 23 Oct 2012 00:51:26 -0400
Subject: [PATCH 140/278] sipr: set channel layout

---
 libavcodec/sipr.c |    5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/libavcodec/sipr.c b/libavcodec/sipr.c
index 1e9168a..9db0923 100644
--- a/libavcodec/sipr.c
+++ b/libavcodec/sipr.c
@@ -25,6 +25,7 @@
 #include <stdint.h>
 #include <string.h>
 
+#include "libavutil/audioconvert.h"
 #include "libavutil/mathematics.h"
 #include "avcodec.h"
 #define BITSTREAM_READER_LE
@@ -509,7 +510,9 @@ static av_cold int sipr_decoder_init(AVCodecContext * avctx)
     for (i = 0; i < 4; i++)
         ctx->energy_history[i] = -14;
 
-    avctx->sample_fmt = AV_SAMPLE_FMT_FLT;
+    avctx->channels       = 1;
+    avctx->channel_layout = AV_CH_LAYOUT_MONO;
+    avctx->sample_fmt     = AV_SAMPLE_FMT_FLT;
 
     avcodec_get_frame_defaults(&ctx->frame);
     avctx->coded_frame = &ctx->frame;
-- 
1.7.9.5

