From 335826cf5f7169199ed0a8d3a61f697ec0438d30 Mon Sep 17 00:00:00 2001
From: Justin Ruggles <justin.ruggles@gmail.com>
Date: Tue, 23 Oct 2012 13:17:17 -0400
Subject: [PATCH 143/278] twinvq: set channel layout

---
 libavcodec/twinvq.c |    4 ++++
 1 file changed, 4 insertions(+)

diff --git a/libavcodec/twinvq.c b/libavcodec/twinvq.c
index 50ee626..3159e49 100644
--- a/libavcodec/twinvq.c
+++ b/libavcodec/twinvq.c
@@ -19,6 +19,7 @@
  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
  */
 
+#include "libavutil/audioconvert.h"
 #include "libavutil/float_dsp.h"
 #include "avcodec.h"
 #include "get_bits.h"
@@ -1131,6 +1132,9 @@ static av_cold int twin_decode_init(AVCodecContext *avctx)
                avctx->channels);
         return -1;
     }
+    avctx->channel_layout = avctx->channels == 1 ? AV_CH_LAYOUT_MONO :
+                                                   AV_CH_LAYOUT_STEREO;
+
     ibps = avctx->bit_rate / (1000 * avctx->channels);
 
     switch ((isampf << 8) +  ibps) {
-- 
1.7.9.5

