From 8a58894fc63c9d367c4cd6a17e277d1a8608c2c0 Mon Sep 17 00:00:00 2001
From: Justin Ruggles <justin.ruggles@gmail.com>
Date: Mon, 5 Nov 2012 12:49:36 -0500
Subject: [PATCH 186/278] FATE: add a 24-bit FLAC encoding test

---
 tests/fate/flac.mak |    6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/tests/fate/flac.mak b/tests/fate/flac.mak
index f7ac342..92eb743 100644
--- a/tests/fate/flac.mak
+++ b/tests/fate/flac.mak
@@ -5,6 +5,7 @@ FATE_FLAC += fate-flac-16-chmode-indep                                  \
              fate-flac-16-fixed                                         \
              fate-flac-16-lpc-cholesky                                  \
              fate-flac-16-lpc-levinson                                  \
+             fate-flac-24-comp-8                                        \
 
 fate-flac-16-chmode-%: OPTS = -ch_mode $(@:fate-flac-16-chmode-%=%)
 fate-flac-16-fixed:    OPTS = -lpc_type fixed
@@ -13,6 +14,11 @@ fate-flac-16-lpc-%:    OPTS = -lpc_type $(@:fate-flac-16-lpc-%=%)
 fate-flac-16-%: REF = $(SAMPLES)/audio-reference/luckynight_2ch_44kHz_s16.wav
 fate-flac-16-%: CMD = enc_dec_pcm flac wav s16le $(REF) -c flac $(OPTS)
 
+fate-flac-24-comp-%: OPTS = -compression_level $(@:fate-flac-24-comp-%=%)
+
+fate-flac-24-%: REF = $(SAMPLES)/audio-reference/divertimenti_2ch_96kHz_s24.wav
+fate-flac-24-%: CMD = enc_dec_pcm flac wav s24le $(REF) -c flac $(OPTS)
+
 fate-flac-%: CMP = oneoff
 fate-flac-%: FUZZ = 0
 
-- 
1.7.9.5

