From faf340f60c18c0af282570eed156ec67c1d98f6f Mon Sep 17 00:00:00 2001
From: Justin Ruggles <justin.ruggles@gmail.com>
Date: Sat, 7 Apr 2012 12:10:59 -0400
Subject: [PATCH 210/278] binkaudio: set channel layout

---
 libavcodec/binkaudio.c |    3 +++
 1 file changed, 3 insertions(+)

diff --git a/libavcodec/binkaudio.c b/libavcodec/binkaudio.c
index af56526..3a138e5 100644
--- a/libavcodec/binkaudio.c
+++ b/libavcodec/binkaudio.c
@@ -28,6 +28,7 @@
  *  http://wiki.multimedia.cx/index.php?title=Bink_Audio
  */
 
+#include "libavutil/channel_layout.h"
 #include "avcodec.h"
 #define BITSTREAM_READER_LE
 #include "get_bits.h"
@@ -87,6 +88,8 @@ static av_cold int decode_init(AVCodecContext *avctx)
         av_log(avctx, AV_LOG_ERROR, "too many channels: %d\n", avctx->channels);
         return -1;
     }
+    avctx->channel_layout = avctx->channels == 1 ? AV_CH_LAYOUT_MONO :
+                                                   AV_CH_LAYOUT_STEREO;
 
     s->version_b = avctx->extradata && avctx->extradata[3] == 'b';
 
-- 
1.7.9.5

