From 5595368bcc6df7d307876bc051f41a8c72f38ce9 Mon Sep 17 00:00:00 2001
From: Justin Ruggles <justin.ruggles@gmail.com>
Date: Sat, 7 Apr 2012 11:07:16 -0400
Subject: [PATCH 214/278] amr: set channel_layout

---
 libavformat/amr.c |    3 +++
 1 file changed, 3 insertions(+)

diff --git a/libavformat/amr.c b/libavformat/amr.c
index cc4842e..ba91d7c 100644
--- a/libavformat/amr.c
+++ b/libavformat/amr.c
@@ -25,6 +25,8 @@ Write and read amr data according to RFC3267, http://www.ietf.org/rfc/rfc3267.tx
 Only mono files are supported.
 
 */
+
+#include "libavutil/channel_layout.h"
 #include "avformat.h"
 #include "internal.h"
 
@@ -97,6 +99,7 @@ static int amr_read_header(AVFormatContext *s)
         st->codec->sample_rate = 8000;
     }
     st->codec->channels   = 1;
+    st->codec->channel_layout = AV_CH_LAYOUT_MONO;
     st->codec->codec_type = AVMEDIA_TYPE_AUDIO;
     avpriv_set_pts_info(st, 64, 1, st->codec->sample_rate);
 
-- 
1.7.9.5

