From ff50d27a6391663be2332ec242de23131ac74034 Mon Sep 17 00:00:00 2001
From: Justin Ruggles <justin.ruggles@gmail.com>
Date: Sat, 7 Apr 2012 12:06:41 -0400
Subject: [PATCH 217/278] bfi: set channel layout

---
 libavformat/bfi.c |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/libavformat/bfi.c b/libavformat/bfi.c
index c520bea..e60bbf4 100644
--- a/libavformat/bfi.c
+++ b/libavformat/bfi.c
@@ -26,6 +26,7 @@
  * @see http://wiki.multimedia.cx/index.php?title=BFI
  */
 
+#include "libavutil/channel_layout.h"
 #include "libavutil/intreadwrite.h"
 #include "avformat.h"
 #include "internal.h"
@@ -96,6 +97,7 @@ static int bfi_read_header(AVFormatContext * s)
     astream->codec->codec_type      = AVMEDIA_TYPE_AUDIO;
     astream->codec->codec_id        = AV_CODEC_ID_PCM_U8;
     astream->codec->channels        = 1;
+    astream->codec->channel_layout  = AV_CH_LAYOUT_MONO;
     astream->codec->bits_per_coded_sample = 8;
     astream->codec->bit_rate        =
         astream->codec->sample_rate * astream->codec->bits_per_coded_sample;
-- 
1.7.9.5

