From e8088d6e4b1255a8a5969ef925e4eb3a01f0988b Mon Sep 17 00:00:00 2001
From: Justin Ruggles <justin.ruggles@gmail.com>
Date: Sat, 7 Apr 2012 12:12:35 -0400
Subject: [PATCH 219/278] bmv: set channel layout

---
 libavformat/bmv.c |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/libavformat/bmv.c b/libavformat/bmv.c
index fe5db3f..ce157e8 100644
--- a/libavformat/bmv.c
+++ b/libavformat/bmv.c
@@ -19,6 +19,7 @@
  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
  */
 
+#include "libavutil/channel_layout.h"
 #include "avformat.h"
 #include "internal.h"
 
@@ -58,6 +59,7 @@ static int bmv_read_header(AVFormatContext *s)
     ast->codec->codec_type      = AVMEDIA_TYPE_AUDIO;
     ast->codec->codec_id        = AV_CODEC_ID_BMV_AUDIO;
     ast->codec->channels        = 2;
+    ast->codec->channel_layout  = AV_CH_LAYOUT_STEREO;
     ast->codec->sample_rate     = 22050;
     avpriv_set_pts_info(ast, 16, 1, 22050);
 
-- 
1.7.9.5

