From 49e7af06f2fc68c9031cf7b67fe1cba066e2ace2 Mon Sep 17 00:00:00 2001
From: Justin Ruggles <justin.ruggles@gmail.com>
Date: Sat, 7 Apr 2012 12:56:51 -0400
Subject: [PATCH 222/278] dsicin: set channel layout

---
 libavformat/dsicin.c |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/libavformat/dsicin.c b/libavformat/dsicin.c
index 8cba461..ecc8c0b 100644
--- a/libavformat/dsicin.c
+++ b/libavformat/dsicin.c
@@ -24,6 +24,7 @@
  * Delphine Software International CIN file demuxer
  */
 
+#include "libavutil/channel_layout.h"
 #include "libavutil/intreadwrite.h"
 #include "avformat.h"
 #include "internal.h"
@@ -131,6 +132,7 @@ static int cin_read_header(AVFormatContext *s)
     st->codec->codec_id = AV_CODEC_ID_DSICINAUDIO;
     st->codec->codec_tag = 0;  /* no tag */
     st->codec->channels = 1;
+    st->codec->channel_layout = AV_CH_LAYOUT_MONO;
     st->codec->sample_rate = 22050;
     st->codec->bits_per_coded_sample = 8;
     st->codec->bit_rate = st->codec->sample_rate * st->codec->bits_per_coded_sample * st->codec->channels;
-- 
1.7.9.5

