From d4a105ae5c1b0c6118872c8b729c3ac751281ed0 Mon Sep 17 00:00:00 2001
From: Justin Ruggles <justin.ruggles@gmail.com>
Date: Sat, 7 Apr 2012 13:51:42 -0400
Subject: [PATCH 225/278] gsmdec: set channel layout

---
 libavformat/gsmdec.c |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/libavformat/gsmdec.c b/libavformat/gsmdec.c
index f0cb279..f9ecbaf 100644
--- a/libavformat/gsmdec.c
+++ b/libavformat/gsmdec.c
@@ -19,6 +19,7 @@
  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
  */
 
+#include "libavutil/channel_layout.h"
 #include "libavutil/mathematics.h"
 #include "libavutil/opt.h"
 #include "avformat.h"
@@ -64,6 +65,7 @@ static int gsm_read_header(AVFormatContext *s)
     st->codec->codec_type  = AVMEDIA_TYPE_AUDIO;
     st->codec->codec_id    = s->iformat->raw_codec_id;
     st->codec->channels    = 1;
+    st->codec->channel_layout = AV_CH_LAYOUT_MONO;
     st->codec->sample_rate = c->sample_rate;
     st->codec->bit_rate    = GSM_BLOCK_SIZE * 8 * c->sample_rate / GSM_BLOCK_SAMPLES;
 
-- 
1.7.9.5

