From 41a2d9590d89c3f240b1ef875649852197d945eb Mon Sep 17 00:00:00 2001
From: Justin Ruggles <justin.ruggles@gmail.com>
Date: Sat, 7 Apr 2012 16:44:46 -0400
Subject: [PATCH 229/278] ipmovie: set channel layout

---
 libavformat/ipmovie.c |    3 +++
 1 file changed, 3 insertions(+)

diff --git a/libavformat/ipmovie.c b/libavformat/ipmovie.c
index b567fb2..60ae939 100644
--- a/libavformat/ipmovie.c
+++ b/libavformat/ipmovie.c
@@ -32,6 +32,7 @@
  * up and sending out the chunks.
  */
 
+#include "libavutil/channel_layout.h"
 #include "libavutil/intreadwrite.h"
 #include "avformat.h"
 #include "internal.h"
@@ -600,6 +601,8 @@ static int ipmovie_read_header(AVFormatContext *s)
         st->codec->codec_id = ipmovie->audio_type;
         st->codec->codec_tag = 0;  /* no tag */
         st->codec->channels = ipmovie->audio_channels;
+        st->codec->channel_layout = st->codec->channels == 1 ? AV_CH_LAYOUT_MONO :
+                                                               AV_CH_LAYOUT_STEREO;
         st->codec->sample_rate = ipmovie->audio_sample_rate;
         st->codec->bits_per_coded_sample = ipmovie->audio_bits;
         st->codec->bit_rate = st->codec->channels * st->codec->sample_rate *
-- 
1.7.9.5

