From ef1b23ad21e3f12fc4ff2a73a6d4d4cd9d630c4b Mon Sep 17 00:00:00 2001
From: Justin Ruggles <justin.ruggles@gmail.com>
Date: Sat, 7 Apr 2012 16:47:15 -0400
Subject: [PATCH 231/278] jvdec: set channel layout

---
 libavformat/jvdec.c |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/libavformat/jvdec.c b/libavformat/jvdec.c
index 47a6142..cf9df8c 100644
--- a/libavformat/jvdec.c
+++ b/libavformat/jvdec.c
@@ -25,6 +25,7 @@
  * @author Peter Ross <pross@xvid.org>
  */
 
+#include "libavutil/channel_layout.h"
 #include "libavutil/intreadwrite.h"
 #include "avformat.h"
 #include "internal.h"
@@ -91,6 +92,7 @@ static int read_header(AVFormatContext *s)
     ast->codec->codec_tag   = 0; /* no fourcc */
     ast->codec->sample_rate = avio_rl16(pb);
     ast->codec->channels    = 1;
+    ast->codec->channel_layout = AV_CH_LAYOUT_MONO;
     avpriv_set_pts_info(ast, 64, 1, ast->codec->sample_rate);
 
     avio_skip(pb, 10);
-- 
1.7.9.5

