From 87199d34db5d79ad85c0801b29db0709c08f458c Mon Sep 17 00:00:00 2001
From: Justin Ruggles <justin.ruggles@gmail.com>
Date: Sat, 7 Apr 2012 16:52:57 -0400
Subject: [PATCH 232/278] mm: set channel layout

---
 libavformat/mm.c |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/libavformat/mm.c b/libavformat/mm.c
index 13f33a2..83539fa 100644
--- a/libavformat/mm.c
+++ b/libavformat/mm.c
@@ -31,6 +31,7 @@
  *  http://wiki.multimedia.cx/index.php?title=American_Laser_Games_MM
  */
 
+#include "libavutil/channel_layout.h"
 #include "libavutil/intreadwrite.h"
 #include "avformat.h"
 #include "internal.h"
@@ -124,6 +125,7 @@ static int read_header(AVFormatContext *s)
         st->codec->codec_tag = 0; /* no fourcc */
         st->codec->codec_id = AV_CODEC_ID_PCM_U8;
         st->codec->channels = 1;
+        st->codec->channel_layout = AV_CH_LAYOUT_MONO;
         st->codec->sample_rate = 8000;
         avpriv_set_pts_info(st, 64, 1, 8000); /* 8000 hz */
     }
-- 
1.7.9.5

