From f24b0b1b6cc3357637570d28f9a9fa285bec1770 Mon Sep 17 00:00:00 2001
From: Justin Ruggles <justin.ruggles@gmail.com>
Date: Sat, 7 Apr 2012 16:56:48 -0400
Subject: [PATCH 233/278] mmf: set channel layout

---
 libavformat/mmf.c |    3 +++
 1 file changed, 3 insertions(+)

diff --git a/libavformat/mmf.c b/libavformat/mmf.c
index 06d4a2b..20570f2 100644
--- a/libavformat/mmf.c
+++ b/libavformat/mmf.c
@@ -18,6 +18,8 @@
  * License along with Libav; if not, write to the Free Software
  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
  */
+
+#include "libavutil/channel_layout.h"
 #include "avformat.h"
 #include "internal.h"
 #include "avio_internal.h"
@@ -249,6 +251,7 @@ static int mmf_read_header(AVFormatContext *s)
     st->codec->codec_id = AV_CODEC_ID_ADPCM_YAMAHA;
     st->codec->sample_rate = rate;
     st->codec->channels = 1;
+    st->codec->channel_layout = AV_CH_LAYOUT_MONO;
     st->codec->bits_per_coded_sample = 4;
     st->codec->bit_rate = st->codec->sample_rate * st->codec->bits_per_coded_sample;
 
-- 
1.7.9.5

