From 66d7ceb4aa0a17c72de9a63748953d53a1e95a12 Mon Sep 17 00:00:00 2001
From: Justin Ruggles <justin.ruggles@gmail.com>
Date: Sat, 7 Apr 2012 17:17:19 -0400
Subject: [PATCH 235/278] mvi: set channel layout

---
 libavformat/mvi.c |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/libavformat/mvi.c b/libavformat/mvi.c
index c5aee5a..10ec8bb 100644
--- a/libavformat/mvi.c
+++ b/libavformat/mvi.c
@@ -19,6 +19,7 @@
  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
  */
 
+#include "libavutil/channel_layout.h"
 #include "avformat.h"
 #include "internal.h"
 
@@ -81,6 +82,7 @@ static int read_header(AVFormatContext *s)
     ast->codec->codec_type      = AVMEDIA_TYPE_AUDIO;
     ast->codec->codec_id        = AV_CODEC_ID_PCM_U8;
     ast->codec->channels        = 1;
+    ast->codec->channel_layout  = AV_CH_LAYOUT_MONO;
     ast->codec->bits_per_coded_sample = 8;
     ast->codec->bit_rate        = ast->codec->sample_rate * 8;
 
-- 
1.7.9.5

