From cc57228e31ace7ab112b7fd86b31cc11dfb405f0 Mon Sep 17 00:00:00 2001
From: Justin Ruggles <justin.ruggles@gmail.com>
Date: Sat, 7 Apr 2012 17:22:05 -0400
Subject: [PATCH 236/278] mxg: set channel layout

---
 libavformat/mxg.c |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/libavformat/mxg.c b/libavformat/mxg.c
index 09653ee..8959134 100644
--- a/libavformat/mxg.c
+++ b/libavformat/mxg.c
@@ -19,6 +19,7 @@
  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
  */
 
+#include "libavutil/channel_layout.h"
 #include "libavutil/intreadwrite.h"
 #include "libavcodec/mjpeg.h"
 #include "avformat.h"
@@ -56,6 +57,7 @@ static int mxg_read_header(AVFormatContext *s)
     audio_st->codec->codec_type = AVMEDIA_TYPE_AUDIO;
     audio_st->codec->codec_id = AV_CODEC_ID_PCM_ALAW;
     audio_st->codec->channels = 1;
+    audio_st->codec->channel_layout = AV_CH_LAYOUT_MONO;
     audio_st->codec->sample_rate = 8000;
     audio_st->codec->bits_per_coded_sample = 8;
     audio_st->codec->block_align = 1;
-- 
1.7.9.5

