From c9759eb426ea997822bf5e6cc43bb24b249e9e25 Mon Sep 17 00:00:00 2001
From: Justin Ruggles <justin.ruggles@gmail.com>
Date: Sat, 7 Apr 2012 17:53:39 -0400
Subject: [PATCH 239/278] omadec: set channel layout

---
 libavformat/omadec.c |    3 +++
 1 file changed, 3 insertions(+)

diff --git a/libavformat/omadec.c b/libavformat/omadec.c
index 022942d..9b22999 100644
--- a/libavformat/omadec.c
+++ b/libavformat/omadec.c
@@ -40,6 +40,7 @@
  * CODEC SUPPORT: Only ATRAC3 codec is currently supported!
  */
 
+#include "libavutil/channel_layout.h"
 #include "avformat.h"
 #include "internal.h"
 #include "libavutil/intreadwrite.h"
@@ -312,6 +313,7 @@ static int oma_read_header(AVFormatContext *s)
             framesize = (codec_params & 0x3FF) * 8;
             jsflag = (codec_params >> 17) & 1; /* get stereo coding mode, 1 for joint-stereo */
             st->codec->channels    = 2;
+            st->codec->channel_layout = AV_CH_LAYOUT_STEREO;
             st->codec->sample_rate = samplerate;
             st->codec->bit_rate    = st->codec->sample_rate * framesize * 8 / 1024;
 
@@ -346,6 +348,7 @@ static int oma_read_header(AVFormatContext *s)
         case OMA_CODECID_LPCM:
             /* PCM 44.1 kHz 16 bit stereo big-endian */
             st->codec->channels = 2;
+            st->codec->channel_layout = AV_CH_LAYOUT_STEREO;
             st->codec->sample_rate = 44100;
             framesize = 1024;
             /* bit rate = sample rate x PCM block align (= 4) x 8 */
-- 
1.7.9.5

