From 444b79c18acc057082a2fbdd8d0bedefafbea5fd Mon Sep 17 00:00:00 2001
From: Justin Ruggles <justin.ruggles@gmail.com>
Date: Sat, 7 Apr 2012 17:57:25 -0400
Subject: [PATCH 241/278] qcp: set channel layout

---
 libavformat/qcp.c |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/libavformat/qcp.c b/libavformat/qcp.c
index bde5990..30d362f 100644
--- a/libavformat/qcp.c
+++ b/libavformat/qcp.c
@@ -27,6 +27,7 @@
  *     http://tools.ietf.org/html/rfc3625
  */
 
+#include "libavutil/channel_layout.h"
 #include "libavutil/intreadwrite.h"
 #include "avformat.h"
 
@@ -96,6 +97,7 @@ static int qcp_read_header(AVFormatContext *s)
 
     st->codec->codec_type = AVMEDIA_TYPE_AUDIO;
     st->codec->channels   = 1;
+    st->codec->channel_layout = AV_CH_LAYOUT_MONO;
     avio_read(pb, buf, 16);
     if (is_qcelp_13k_guid(buf)) {
         st->codec->codec_id = AV_CODEC_ID_QCELP;
-- 
1.7.9.5

