From ce842029ce97ea3eb49854fc71ad288246eb76d1 Mon Sep 17 00:00:00 2001
From: Justin Ruggles <justin.ruggles@gmail.com>
Date: Sat, 7 Apr 2012 18:18:37 -0400
Subject: [PATCH 243/278] rsodec: set channel layout

---
 libavformat/rsodec.c |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/libavformat/rsodec.c b/libavformat/rsodec.c
index f5bcebb..c3016ca 100644
--- a/libavformat/rsodec.c
+++ b/libavformat/rsodec.c
@@ -20,6 +20,7 @@
  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
  */
 
+#include "libavutil/channel_layout.h"
 #include "libavutil/intreadwrite.h"
 #include "avformat.h"
 #include "internal.h"
@@ -63,6 +64,7 @@ static int rso_read_header(AVFormatContext *s)
     st->codec->codec_tag    = id;
     st->codec->codec_id     = codec;
     st->codec->channels     = 1;
+    st->codec->channel_layout = AV_CH_LAYOUT_MONO;
     st->codec->sample_rate  = rate;
 
     avpriv_set_pts_info(st, 64, 1, rate);
-- 
1.7.9.5

