From c1ac602d53ab250fd1ef37867e18191c55cc5e4f Mon Sep 17 00:00:00 2001
From: Justin Ruggles <justin.ruggles@gmail.com>
Date: Sat, 7 Apr 2012 18:24:45 -0400
Subject: [PATCH 244/278] rtpdec_amr: set channel layout

---
 libavformat/rtpdec_amr.c |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/libavformat/rtpdec_amr.c b/libavformat/rtpdec_amr.c
index cc43e31..1e60287 100644
--- a/libavformat/rtpdec_amr.c
+++ b/libavformat/rtpdec_amr.c
@@ -19,6 +19,7 @@
  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
  */
 
+#include "libavutil/channel_layout.h"
 #include "avformat.h"
 #include "rtpdec_formats.h"
 #include "libavutil/avstring.h"
@@ -77,6 +78,7 @@ static int amr_handle_packet(AVFormatContext *ctx,
         av_log(ctx, AV_LOG_ERROR, "Only mono AMR is supported\n");
         return AVERROR_INVALIDDATA;
     }
+    st->codec->channel_layout = AV_CH_LAYOUT_MONO;
 
     /* The AMR RTP packet consists of one header byte, followed
      * by one TOC byte for each AMR frame in the packet, followed
-- 
1.7.9.5

