From 57e590e4b8b94238800b36695f07d5cf4a4832ac Mon Sep 17 00:00:00 2001
From: Justin Ruggles <justin.ruggles@gmail.com>
Date: Sat, 7 Apr 2012 18:36:08 -0400
Subject: [PATCH 246/278] siff: set channel layout

---
 libavformat/siff.c |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/libavformat/siff.c b/libavformat/siff.c
index 02af80b..8ba7c60 100644
--- a/libavformat/siff.c
+++ b/libavformat/siff.c
@@ -19,6 +19,7 @@
  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
  */
 
+#include "libavutil/channel_layout.h"
 #include "libavutil/intreadwrite.h"
 #include "avformat.h"
 #include "internal.h"
@@ -78,6 +79,7 @@ static int create_audio_stream(AVFormatContext *s, SIFFContext *c)
     ast->codec->codec_type      = AVMEDIA_TYPE_AUDIO;
     ast->codec->codec_id        = AV_CODEC_ID_PCM_U8;
     ast->codec->channels        = 1;
+    ast->codec->channel_layout  = AV_CH_LAYOUT_MONO;
     ast->codec->bits_per_coded_sample = 8;
     ast->codec->sample_rate     = c->rate;
     avpriv_set_pts_info(ast, 16, 1, c->rate);
-- 
1.7.9.5

