From 1c7587728c81fcb8209ae079a1bc0ccc6b03b4c2 Mon Sep 17 00:00:00 2001
From: Justin Ruggles <justin.ruggles@gmail.com>
Date: Sat, 7 Apr 2012 18:38:29 -0400
Subject: [PATCH 248/278] sol: set channel layout

---
 libavformat/sol.c |    3 +++
 1 file changed, 3 insertions(+)

diff --git a/libavformat/sol.c b/libavformat/sol.c
index da84600..92599b1 100644
--- a/libavformat/sol.c
+++ b/libavformat/sol.c
@@ -23,6 +23,7 @@
  * Based on documents from Game Audio Player and own research
  */
 
+#include "libavutil/channel_layout.h"
 #include "libavutil/intreadwrite.h"
 #include "avformat.h"
 #include "internal.h"
@@ -116,6 +117,8 @@ static int sol_read_header(AVFormatContext *s)
     st->codec->codec_tag = id;
     st->codec->codec_id = codec;
     st->codec->channels = channels;
+    st->codec->channel_layout = channels == 1 ? AV_CH_LAYOUT_MONO :
+                                                AV_CH_LAYOUT_STEREO;
     st->codec->sample_rate = rate;
     avpriv_set_pts_info(st, 64, 1, rate);
     return 0;
-- 
1.7.9.5

