From 7f348bd764e1ff85054f1da5291f18c0e7f8f39b Mon Sep 17 00:00:00 2001
From: Justin Ruggles <justin.ruggles@gmail.com>
Date: Sat, 7 Apr 2012 18:44:20 -0400
Subject: [PATCH 250/278] tiertexseq: set channel layout

---
 libavformat/tiertexseq.c |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/libavformat/tiertexseq.c b/libavformat/tiertexseq.c
index 5582890..45300f9 100644
--- a/libavformat/tiertexseq.c
+++ b/libavformat/tiertexseq.c
@@ -24,6 +24,7 @@
  * Tiertex Limited SEQ file demuxer
  */
 
+#include "libavutil/channel_layout.h"
 #include "avformat.h"
 #include "internal.h"
 
@@ -231,6 +232,7 @@ static int seq_read_header(AVFormatContext *s)
     st->codec->codec_id = AV_CODEC_ID_PCM_S16BE;
     st->codec->codec_tag = 0;  /* no tag */
     st->codec->channels = 1;
+    st->codec->channel_layout = AV_CH_LAYOUT_MONO;
     st->codec->sample_rate = SEQ_SAMPLE_RATE;
     st->codec->bits_per_coded_sample = 16;
     st->codec->bit_rate = st->codec->sample_rate * st->codec->bits_per_coded_sample * st->codec->channels;
-- 
1.7.9.5

