From 2ce7f820d46aa268c3f903674c993362f875890d Mon Sep 17 00:00:00 2001
From: Justin Ruggles <justin.ruggles@gmail.com>
Date: Sat, 7 Apr 2012 18:49:09 -0400
Subject: [PATCH 252/278] wc3movie: set channel layout

---
 libavformat/wc3movie.c |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/libavformat/wc3movie.c b/libavformat/wc3movie.c
index 117e4ea..c894253 100644
--- a/libavformat/wc3movie.c
+++ b/libavformat/wc3movie.c
@@ -27,6 +27,7 @@
  *   http://www.pcisys.net/~melanson/codecs/
  */
 
+#include "libavutil/channel_layout.h"
 #include "libavutil/intreadwrite.h"
 #include "libavutil/dict.h"
 #include "avformat.h"
@@ -183,6 +184,7 @@ static int wc3_read_header(AVFormatContext *s)
     st->codec->codec_id = AV_CODEC_ID_PCM_S16LE;
     st->codec->codec_tag = 1;
     st->codec->channels = WC3_AUDIO_CHANNELS;
+    st->codec->channel_layout = AV_CH_LAYOUT_MONO;
     st->codec->bits_per_coded_sample = WC3_AUDIO_BITS;
     st->codec->sample_rate = WC3_SAMPLE_RATE;
     st->codec->bit_rate = st->codec->channels * st->codec->sample_rate *
-- 
1.7.9.5

