From 7b48d93e8abb8736f52548d02ca5d6137d441ba4 Mon Sep 17 00:00:00 2001
From: Justin Ruggles <justin.ruggles@gmail.com>
Date: Sat, 7 Apr 2012 18:50:41 -0400
Subject: [PATCH 253/278] westwood_aud: set channel layout

---
 libavformat/westwood_aud.c |    3 +++
 1 file changed, 3 insertions(+)

diff --git a/libavformat/westwood_aud.c b/libavformat/westwood_aud.c
index 806aea6..2a06c29 100644
--- a/libavformat/westwood_aud.c
+++ b/libavformat/westwood_aud.c
@@ -33,6 +33,7 @@
  * qualify a file. Refer to wsaud_probe() for the precise parameters.
  */
 
+#include "libavutil/channel_layout.h"
 #include "libavutil/intreadwrite.h"
 #include "avformat.h"
 #include "internal.h"
@@ -120,6 +121,8 @@ static int wsaud_read_header(AVFormatContext *s)
     avpriv_set_pts_info(st, 64, 1, sample_rate);
     st->codec->codec_type  = AVMEDIA_TYPE_AUDIO;
     st->codec->channels    = channels;
+    st->codec->channel_layout = channels == 1 ? AV_CH_LAYOUT_MONO :
+                                                AV_CH_LAYOUT_STEREO;
     st->codec->sample_rate = sample_rate;
 
     return 0;
-- 
1.7.9.5

