From b9629acb6b647b31b8b4e3932e836189bd03e5b9 Mon Sep 17 00:00:00 2001
From: Justin Ruggles <justin.ruggles@gmail.com>
Date: Sat, 7 Apr 2012 18:59:00 -0400
Subject: [PATCH 255/278] yop: set channel layout

---
 libavformat/yop.c |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/libavformat/yop.c b/libavformat/yop.c
index 51d088b..351938b 100644
--- a/libavformat/yop.c
+++ b/libavformat/yop.c
@@ -22,6 +22,7 @@
  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
  */
 
+#include "libavutil/channel_layout.h"
 #include "libavutil/intreadwrite.h"
 #include "avformat.h"
 #include "internal.h"
@@ -74,6 +75,7 @@ static int yop_read_header(AVFormatContext *s)
     audio_dec->codec_type   = AVMEDIA_TYPE_AUDIO;
     audio_dec->codec_id     = AV_CODEC_ID_ADPCM_IMA_APC;
     audio_dec->channels     = 1;
+    audio_dec->channels     = AV_CH_LAYOUT_MONO;
     audio_dec->sample_rate  = 22050;
 
     // Video
-- 
1.7.9.5

