From 3b4296f41473a5b39e84d7a49d480624c9c60040 Mon Sep 17 00:00:00 2001
From: Luca Barbato <lu_zero@gentoo.org>
Date: Sat, 10 Nov 2012 16:20:30 +0100
Subject: [PATCH 267/278] avformat: clarify stream id for muxing

---
 libavformat/avformat.h |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libavformat/avformat.h b/libavformat/avformat.h
index 3e10496..d1d4763 100644
--- a/libavformat/avformat.h
+++ b/libavformat/avformat.h
@@ -621,7 +621,7 @@ typedef struct AVStream {
     /**
      * Format-specific stream ID.
      * decoding: set by libavformat
-     * encoding: set by the user
+     * encoding: set by the user, replaced by libavformat if left unset
      */
     int id;
     /**
-- 
1.7.9.5

