From b74dbdd5e99aed7f3bf1ecb5da1b374e5ba35768 Mon Sep 17 00:00:00 2001
From: Xi Wang <xi.wang@gmail.com>
Date: Wed, 14 Nov 2012 13:45:24 -0500
Subject: [PATCH 269/278] bgmc: Fix av_malloc checks in ff_bgmc_init()

Signed-off-by: Xi Wang <xi.wang@gmail.com>
Signed-off-by: Diego Biurrun <diego@biurrun.de>
---
 libavcodec/bgmc.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libavcodec/bgmc.c b/libavcodec/bgmc.c
index 08b1337..ec8cf9b 100644
--- a/libavcodec/bgmc.c
+++ b/libavcodec/bgmc.c
@@ -461,7 +461,7 @@ int ff_bgmc_init(AVCodecContext *avctx, uint8_t **cf_lut, int **cf_lut_status)
     *cf_lut        = av_malloc(sizeof(*cf_lut)        * LUT_BUFF * 16 * LUT_SIZE);
     *cf_lut_status = av_malloc(sizeof(*cf_lut_status) * LUT_BUFF);
 
-    if (!cf_lut || !cf_lut_status) {
+    if (!*cf_lut || !*cf_lut_status) {
         ff_bgmc_end(cf_lut, cf_lut_status);
         av_log(avctx, AV_LOG_ERROR, "Allocating buffer memory failed.\n");
         return AVERROR(ENOMEM);
-- 
1.7.9.5

