From e28b7553aeab8aa0d590683a21a9a6ab113cb433 Mon Sep 17 00:00:00 2001
From: Diego Biurrun <diego@biurrun.de>
Date: Sat, 10 Nov 2012 18:10:59 +0100
Subject: [PATCH 270/278] configure: Add option to disable all command line
 programs

---
 configure |   10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/configure b/configure
index 779b733..261b0ac 100755
--- a/configure
+++ b/configure
@@ -101,12 +101,15 @@ Configuration options:
   --enable-gray            enable full grayscale support (slower color)
   --disable-swscale-alpha  disable alpha channel support in swscale
 
-Component options:
-  --disable-doc            do not build documentation
+Program options:
+  --disable-programs       do not build command line programs
   --disable-avconv         disable avconv build
   --disable-avplay         disable avplay build
   --disable-avprobe        disable avprobe build
   --disable-avserver       disable avserver build
+
+Component options:
+  --disable-doc            do not build documentation
   --disable-avdevice       disable libavdevice build
   --disable-avcodec        disable libavcodec build
   --disable-avformat       disable libavformat build
@@ -1945,6 +1948,9 @@ for opt do
     ;;
     --enable-debug=*) debuglevel="$optval"
     ;;
+    --disable-programs)
+    disable $PROGRAM_LIST
+    ;;
     --disable-everything)
     map 'eval unset \${$(toupper ${v%s})_LIST}' $COMPONENT_LIST
     ;;
-- 
1.7.9.5

