From 9eded0fe412e610ee8944681d5c554b723463e96 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Martin=20Storsj=C3=B6?= <martin@martin.st>
Date: Tue, 13 Nov 2012 19:01:51 +0200
Subject: [PATCH 272/278] configure: Check for -Werror parameters on clang
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Martin Storsjö <martin@martin.st>
---
 configure |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/configure b/configure
index 261b0ac..14a887e 100755
--- a/configure
+++ b/configure
@@ -3549,6 +3549,8 @@ elif enabled llvm_gcc; then
 elif enabled clang; then
     check_cflags -mllvm -stack-alignment=16
     check_cflags -Qunused-arguments
+    check_cflags -Werror=implicit-function-declaration
+    check_cflags -Werror=missing-prototypes
 elif enabled armcc; then
     # 2523: use of inline assembler is deprecated
     add_cflags -W${armcc_opt},--diag_suppress=2523
-- 
1.7.9.5

