From bf5f46b4cc47b7a4568119f224057d4ff91b6cdd Mon Sep 17 00:00:00 2001
From: Janne Grunau <janne-libav@jannau.net>
Date: Thu, 25 Oct 2012 16:26:06 +0200
Subject: [PATCH 276/278] APIChanges: add entry for av_read_packet deprecation

---
 doc/APIchanges |    4 ++++
 1 file changed, 4 insertions(+)

diff --git a/doc/APIchanges b/doc/APIchanges
index a0756da..1c6247e 100644
--- a/doc/APIchanges
+++ b/doc/APIchanges
@@ -182,6 +182,10 @@ API changes, most recent first:
 2012-04-05 - 5cc51a5 - lavu 51.26.0 - audioconvert.h
   Add av_get_default_channel_layout()
 
+2012-03-20 - 3c90cc2 - lavfo 54.2.0
+  Deprecate av_read_packet(), use av_read_frame() with
+  AVFMT_FLAG_NOPARSE | AVFMT_FLAG_NOFILLIN in AVFormatContext.flags
+
 2012-03-06 - 4d851f8 - lavu 51.25.0 - cpu.h
   Add av_set_cpu_flags_mask().
 
-- 
1.7.9.5

