From: Giovanni Mascellani <gio@debian.org>
Subject: Make getSourcePositionFromStack() pulic
Forwarded: no
Last-Update: 2011-12-08

This is needed by package geogebra.

Index: rhino-1.7R3/src/org/mozilla/javascript/Context.java
===================================================================
--- rhino-1.7R3.orig/src/org/mozilla/javascript/Context.java	2011-12-18 11:43:21.000000000 +0100
+++ rhino-1.7R3/src/org/mozilla/javascript/Context.java	2011-12-18 11:43:57.000000000 +0100
@@ -2481,7 +2481,7 @@
         return (Evaluator)Kit.newInstanceOrNull(interpreterClass);
     }
 
-    static String getSourcePositionFromStack(int[] linep)
+    public static String getSourcePositionFromStack(int[] linep)
     {
         Context cx = getCurrentContext();
         if (cx == null)
