From: Arnold Metselaar <arnold.metsel@gmail.com>
Subject: support 'now' 
Origin: vendor, http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=618651
Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=618651
Forwarded: no

---
 gphoto2/actions.c |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

--- gphoto2.orig/gphoto2/actions.c
+++ gphoto2/gphoto2/actions.c
@@ -1672,8 +1672,10 @@ set_config_action (GPParams *p, const ch
 		int	t = -1;
 		struct tm xtm;
 
+		if (!strcasecmp (value, "now")	|| !strcasecmp (value, _("now")))
+			t = time(NULL);
 #ifdef HAVE_STRPTIME
-		if (strptime (value, "%c", &xtm) || strptime (value, "%Ec", &xtm))
+		else if (strptime (value, "%c", &xtm) || strptime (value, "%Ec", &xtm))
 			t = mktime (&xtm);
 #endif
 		if (t == -1) {
