From 9ec177dd7907437bdf4af6510e8f51c36b19f6da Mon Sep 17 00:00:00 2001
From: arno <arenevier@fdn.fr>
Date: Tue, 25 Aug 2009 23:32:30 +0200
Subject: [PATCH 2/2] remove websites from default white list

---
 defaults/preferences/noscript.js |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

--- a/defaults/preferences/noscript.js
+++ b/defaults/preferences/noscript.js
@@ -29,7 +29,7 @@ pref("noscript.showExternalFilters", tru
 pref("noscript.showTempAllowPage", true);
 pref("noscript.showAllowPage", true);
 pref("noscript.mandatory", "chrome: blob: about: about:addons about:blocked about:crashes about:home about:config about:neterror about:certerror about:memory about:plugins about:privatebrowsing about:sessionrestore about:support resource:");
-pref("noscript.default", "about:blank addons.mozilla.org persona.org mozilla.net flashgot.net google.com gstatic.com googleapis.com paypal.com paypalobjects.com securecode.com securesuite.net firstdata.com firstdata.lv informaction.com yahoo.com yimg.com yahooapis.com youtube.com ytimg.com maone.net noscript.net hotmail.com msn.com passport.com passport.net passportimages.com live.com afx.ms gfx.ms wlxrs.com");
+pref("noscript.default", "about:blank");
 
 pref("noscript.eraseFloatingElements", true);
 
@@ -370,4 +370,4 @@ pref("noscript.ABE.rulesets.SYSTEM", "#
 pref("noscript.ABE.rulesets.USER", "# User-defined rules. Feel free to experiment here.\r\n");
 pref("noscript.ABE.migration", 0);
 
-pref("noscript.smartClickToPlay", true);
\ No newline at end of file
+pref("noscript.smartClickToPlay", true);
