--- libcrypt-openssl-random-perl.orig/Makefile.PL
+++ libcrypt-openssl-random-perl/Makefile.PL
@@ -5,7 +5,7 @@
     'NAME'		=> 'Crypt::OpenSSL::Random',
     'VERSION_FROM'	=> 'Random.pm', # finds $VERSION
     'PREREQ_PM'		=> {}, # e.g., Module::Name => 1.1
-    'LIBS'	        => ['-lssl -lcrypto'],   # e.g., '-lm' 
+    'LIBS'	        => ['-lcrypto'],   # e.g., '-lm' 
     'DEFINE'		=> '', # e.g., '-DHAVE_SOMETHING'
     'INC'		=> '', # e.g., '-I/usr/include/other'
 );
