From 15fc7175ab173b94dbd8f5f6e1300300e4b4433f Mon Sep 17 00:00:00 2001
From: Guus Sliepen <guus@sliepen.eu.org>
Date: Sun, 19 Feb 2012 19:02:40 +0100
Subject: [PATCH 1/2] Remove UTF-8 whitespace.

Found by cppcheck.
---
 src/readwrite.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/readwrite.c b/src/readwrite.c
index e86f70b..003d280 100644
--- a/src/readwrite.c
+++ b/src/readwrite.c
@@ -33,7 +33,7 @@
 #if HAVE_VALGRIND_VALGRIND_H
 # include <valgrind/memcheck.h>
 # ifndef VALGRIND_MAKE_MEM_DEFINED_IF_ADDRESSABLE
-#  define VALGRIND_MAKE_MEM_DEFINED_IF_ADDRESSABLE(m, n) do {} while(0)
+#  define VALGRIND_MAKE_MEM_DEFINED_IF_ADDRESSABLE(m, n) do {} while(0)
 # endif
 #else
 # define VALGRIND_MAKE_MEM_DEFINED_IF_ADDRESSABLE(m, n) do {} while(0)
-- 
1.7.9

