From 969657885f0505de28df3e49fa5463d585884387 Mon Sep 17 00:00:00 2001
From: Guus Sliepen <guus@sliepen.eu.org>
Date: Sun, 19 Feb 2012 19:04:16 +0100
Subject: [PATCH 2/2] Fix incorrect use of == instead of =.

---
 src/fw.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/fw.c b/src/fw.c
index 1131d23..03fae66 100644
--- a/src/fw.c
+++ b/src/fw.c
@@ -851,7 +851,7 @@ handle_arm_request(raw1394handle_t handle, struct address_closure *ac,
 
 	rrb = malloc(sizeof *rrb + in_length + response.length);
 	if (rrb == NULL) {
-		errno == ENOMEM;
+		errno = ENOMEM;
 		return -1;
 	}
 
-- 
1.7.9

