Index: rubygems/test/rubygems/test_gem_installer.rb
===================================================================
--- rubygems.orig/test/rubygems/test_gem_installer.rb	2012-06-09 22:22:15.000000000 +0900
+++ rubygems/test/rubygems/test_gem_installer.rb	2012-06-09 22:23:27.000000000 +0900
@@ -1217,7 +1217,7 @@
   def test_dir
     util_setup_install
 
-    assert_match @installer.dir, %r!/installer/gems/a-2$!
+    assert_match %r!/installer/gems/a-2$!, @installer.dir
   end
 
   def old_ruby_required
