Index: lmms-0.4.10/plugins/zynaddsubfx/LocalZynAddSubFx.cpp
===================================================================
--- lmms-0.4.10.orig/plugins/zynaddsubfx/LocalZynAddSubFx.cpp	2011-01-26 13:54:21.000000000 +0000
+++ lmms-0.4.10/plugins/zynaddsubfx/LocalZynAddSubFx.cpp	2012-04-17 12:08:49.451405933 +0000
@@ -22,6 +22,8 @@
  *
  */
 
+#include <unistd.h>
+
 #include <lmmsconfig.h>
 
 #include "LocalZynAddSubFx.h"
