From: Sune Vuorela <sune@debian.org>
From: Modestas Vainius <modax@debian.org>
Subject: define Qt/KDE definitions when building standalone
Forwarded: no
Last-Update: 2011-06-30
Origin: vendor

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -10,6 +10,8 @@ if(${CMAKE_SOURCE_DIR} STREQUAL ${CMAKE_
 
     include_directories (${KDE4_INCLUDES})
 
+    add_definitions(${QT_DEFINITIONS} ${KDE4_DEFINITIONS} -DQT3_SUPPORT)
+
 endif(${CMAKE_SOURCE_DIR} STREQUAL ${CMAKE_CURRENT_SOURCE_DIR})
 
 # This module can be built independant from the rest of kdebindings.
