From: Modestas Vainius <modax@debian.org>
Subject: Do not export private classes
Last-Update: 2012-06-16
Forwarded: no
Origin: vendor

--- a/okteta/libs/kasten/core/io/filesystem/abstractfilesystemconnectjob_p.h
+++ b/okteta/libs/kasten/core/io/filesystem/abstractfilesystemconnectjob_p.h
@@ -34,7 +34,7 @@
 namespace Kasten1
 {
 
-class KASTENCORE_EXPORT AbstractFileSystemConnectJobPrivate : public AbstractConnectJobPrivate
+class AbstractFileSystemConnectJobPrivate : public AbstractConnectJobPrivate
 {
   public:
     AbstractFileSystemConnectJobPrivate( AbstractFileSystemConnectJob* parent,
--- a/okteta/libs/kasten/core/io/filesystem/abstractfilesystemsyncfromremotejob_p.h
+++ b/okteta/libs/kasten/core/io/filesystem/abstractfilesystemsyncfromremotejob_p.h
@@ -36,7 +36,7 @@ namespace Kasten1
 class AbstractModelFileSystemSynchronizer;
 
 
-class KASTENCORE_EXPORT AbstractFileSystemSyncFromRemoteJobPrivate : public AbstractSyncFromRemoteJobPrivate
+class AbstractFileSystemSyncFromRemoteJobPrivate : public AbstractSyncFromRemoteJobPrivate
 {
   public:
     AbstractFileSystemSyncFromRemoteJobPrivate( AbstractFileSystemSyncFromRemoteJob* parent,
--- a/okteta/libs/kasten/core/io/filesystem/abstractfilesystemsyncwithremotejob_p.h
+++ b/okteta/libs/kasten/core/io/filesystem/abstractfilesystemsyncwithremotejob_p.h
@@ -33,7 +33,7 @@
 namespace Kasten1
 {
 
-class KASTENCORE_EXPORT AbstractFileSystemSyncWithRemoteJobPrivate : public AbstractSyncWithRemoteJobPrivate
+class AbstractFileSystemSyncWithRemoteJobPrivate : public AbstractSyncWithRemoteJobPrivate
 {
   public:
     AbstractFileSystemSyncWithRemoteJobPrivate( AbstractFileSystemSyncWithRemoteJob* parent,
