Author: 
Description: 
Index: openscenegraph/OpenSceneGraph/src/OpenThreads/CMakeLists.txt
===================================================================
--- openscenegraph.orig/OpenSceneGraph/src/OpenThreads/CMakeLists.txt	2011-04-19 13:40:22.000000000 +0200
+++ openscenegraph/OpenSceneGraph/src/OpenThreads/CMakeLists.txt	2011-05-06 20:43:55.000000000 +0200
@@ -5,7 +5,9 @@
 SET(OPENTHREADS_MAJOR_VERSION 2)
 SET(OPENTHREADS_MINOR_VERSION 6)
 SET(OPENTHREADS_PATCH_VERSION 0)
-SET(OPENTHREADS_SOVERSION 12)
+IF(NOT OPENTHREADS_SOVERSION)
+    SET(OPENTHREADS_SOVERSION 12)
+ENDIF (NOT OPENTHREADS_SOVERSION)
 
 SET(OPENTHREADS_VERSION ${OPENTHREADS_MAJOR_VERSION}.${OPENTHREADS_MINOR_VERSION}.${OPENTHREADS_PATCH_VERSION})
 
