My debian/ does not have Makefile.{am,in}.
--- a/Makefile.in
+++ b/Makefile.in
@@ -249,7 +249,7 @@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 AUTOMAKE_OPTIONS = foreign dist-zip
-SUBDIRS = po intl include share debian win32 @OPT_SUBDIRS@
+SUBDIRS = po intl include share win32 @OPT_SUBDIRS@
 DIST_SUBDIRS = po intl include share debian win32 ogg123 oggenc oggdec ogginfo \
 	vcut vorbiscomment m4
 
--- a/configure
+++ b/configure
@@ -31427,7 +31427,7 @@
 
 
 
-ac_config_files="$ac_config_files Makefile m4/Makefile po/Makefile.in intl/Makefile include/Makefile share/Makefile win32/Makefile oggdec/Makefile oggenc/Makefile oggenc/man/Makefile ogg123/Makefile vorbiscomment/Makefile vcut/Makefile ogginfo/Makefile debian/Makefile"
+ac_config_files="$ac_config_files Makefile m4/Makefile po/Makefile.in intl/Makefile include/Makefile share/Makefile win32/Makefile oggdec/Makefile oggenc/Makefile oggenc/man/Makefile ogg123/Makefile vorbiscomment/Makefile vcut/Makefile ogginfo/Makefile"
 
 cat >confcache <<\_ACEOF
 # This file is a shell script that caches the results of configure
@@ -32058,7 +32058,6 @@
     "vorbiscomment/Makefile") CONFIG_FILES="$CONFIG_FILES vorbiscomment/Makefile" ;;
     "vcut/Makefile") CONFIG_FILES="$CONFIG_FILES vcut/Makefile" ;;
     "ogginfo/Makefile") CONFIG_FILES="$CONFIG_FILES ogginfo/Makefile" ;;
-    "debian/Makefile") CONFIG_FILES="$CONFIG_FILES debian/Makefile" ;;
 
   *) { { echo "$as_me:$LINENO: error: invalid argument: $ac_config_target" >&5
 echo "$as_me: error: invalid argument: $ac_config_target" >&2;}
--- a/configure.ac
+++ b/configure.ac
@@ -333,7 +333,6 @@
 vorbiscomment/Makefile
 vcut/Makefile
 ogginfo/Makefile
-debian/Makefile
 ])
 
 if test "x$build_oggenc" = xyes -a "x$have_libFLAC" != xyes; then
