Subject: try to load native library from /usr/lib/jni if system
 property jna.boot.library.path is not set
Author: Jan Dittberner <jandd@debian.org>
--- a/src/com/sun/jna/Native.java
+++ b/src/com/sun/jna/Native.java
@@ -648,6 +648,9 @@
     private static void loadNativeLibrary() {
         String libName = "jnidispatch";
         String bootPath = System.getProperty("jna.boot.library.path");
+        if (bootPath == null) {
+            bootPath = "/usr/lib/jni";
+        }
         if (bootPath != null) {
             String[] dirs = bootPath.split(File.pathSeparator);
             for (int i = 0; i < dirs.length; ++i) {
