Author: Jan Dittberner <jandd@debian.org>
Subject: patch build.xml and native/Makefile to support Debian GNU/kFreeBSD

--- a/native/Makefile
+++ b/native/Makefile
@@ -22,6 +22,7 @@
 
 OS=$(shell uname | sed -e 's/\(CYGWIN\|MINGW32\).*/win32/g' \
                         -e 's/SunOS.*/solaris/g' \
+			-e 's/GNU\/kFreeBSD/linux/g' \
                         -e 's/FreeBSD.*/freebsd/g' \
                         -e 's/OpenBSD.*/openbsd/g' \
                         -e 's/Darwin.*/darwin/g' \
--- a/build.xml
+++ b/build.xml
@@ -129,6 +129,9 @@
     <condition property="os.prefix" value="freebsd-${os.arch}">
       <os name="FreeBSD"/>
     </condition>
+    <condition property="os.prefix" value="linux-${jre.arch}">
+      <os name="GNU/kFreeBSD"/>
+    </condition>
     <condition property="os.prefix" value="openbsd-${os.arch}">
       <os name="OpenBSD"/>
     </condition>
