Index: ATLAS/CONFIG/src/Makefile
===================================================================
--- ATLAS.orig/CONFIG/src/Makefile	2012-07-16 15:34:19.000000000 +0200
+++ ATLAS/CONFIG/src/Makefile	2012-07-16 15:34:21.000000000 +0200
@@ -367,6 +367,7 @@
 install : $(INCINSTdir) $(LIBINSTdir) $(INCINSTdir)/atlas
 	$(MAKE) -f Make.top install_inc INSTdir=$(INCINSTdir) 
 	$(MAKE) -f Make.top install_static_lib INSTdir=$(LIBINSTdir)
+	$(MAKE) -f Make.top install_shared_lib INSTdir=$(LIBINSTdir)
 install_headers : $(INCINSTdir) $(INCINSTdir)/atlas
 	$(MAKE) -f Make.top install_inc INSTdir=$(INCINSTdir) 
 install_static : $(LIBINSTdir) $(LIBINSTdir)/atlas
