Index: ATLAS/Make.top
===================================================================
--- ATLAS.orig/Make.top	2012-07-16 15:34:19.000000000 +0200
+++ ATLAS/Make.top	2012-07-16 15:35:25.000000000 +0200
@@ -298,6 +298,7 @@
 	- chmod 0644 $(INSTdir)/atlas/*
 
 install_static_lib:
+	mkdir -p $(INSTdir)/atlas
 	cp $(LIBdir)/libatlas.a $(INSTdir)/.
 	cp $(LIBdir)/libcblas.a $(INSTdir)/.
 	cp $(LIBdir)/liblapack_atlas.a $(INSTdir)/.
@@ -313,6 +314,7 @@
 	- chmod 0644 $(INSTdir)/libptcblas.a $(INSTdir)/libptf77blas.a
 
 install_shared_lib:
+	mkdir -p $(INSTdir)/atlas
 	for lib in atlas cblas f77blas lapack_atlas; do \
 		cp $(LIBdir)/lib$$lib.so.3.0 $(INSTdir)/. ; \
 		( cd $(INSTdir) && ln -s lib$$lib.so.3.0 lib$$lib.so.3); \
