Index: ATLAS/CONFIG/include/atlconf.h
===================================================================
--- ATLAS.orig/CONFIG/include/atlconf.h	2011-05-14 19:33:24.000000000 +0200
+++ ATLAS/CONFIG/include/atlconf.h	2012-07-16 15:39:21.000000000 +0200
@@ -51,6 +51,14 @@
 #endif
 #define MachIsPPC(mach_) \
    ( (mach_) >= PPCG4 && (mach_) <= PPCG5 )
+#ifdef __s390__
+#define MachIsS390(mach_) \
+  ( __s390__ )
+#else
+#define MachIsS390(mach_) \
+   ( 0 )
+#endif
+
 
 static char *f2c_namestr[5] = {"UNKNOWN","Add_", "Add__", "NoChange", "UpCase"};
 static char *f2c_intstr[5] =
Index: ATLAS/CONFIG/src/SpewMakeInc.c
===================================================================
--- ATLAS.orig/CONFIG/src/SpewMakeInc.c	2012-07-16 15:34:10.000000000 +0200
+++ ATLAS/CONFIG/src/SpewMakeInc.c	2012-07-16 15:39:21.000000000 +0200
@@ -340,6 +340,8 @@
 
    if (MachIsIA64(arch))
       return(sp);
+   if (MachIsS390(arch))
+      return(sp);
    if (MachIsMIPS(arch))
       return((ptrbits == 64) ? "-mabi=64" : "-mabi=n32");
    if (!CompIsGcc(comp))
Index: ATLAS/CONFIG/src/probe_comp.c
===================================================================
--- ATLAS.orig/CONFIG/src/probe_comp.c	2011-05-14 19:33:24.000000000 +0200
+++ ATLAS/CONFIG/src/probe_comp.c	2012-07-16 15:39:21.000000000 +0200
@@ -507,6 +507,8 @@
 
    if (MachIsIA64(arch))
       return(sp);
+   if (MachIsS390(arch))
+      return(sp);
    if (MachIsMIPS(arch))
       return((ptrbits == 64) ? "-mabi=64" : "-mabi=n32");
    if (!CompIsGcc(comp))
