Index: ATLAS/CONFIG/include/atlconf.h
===================================================================
--- ATLAS.orig/CONFIG/include/atlconf.h	2012-07-16 15:39:21.000000000 +0200
+++ ATLAS/CONFIG/include/atlconf.h	2012-07-16 15:39:32.000000000 +0200
@@ -58,6 +58,13 @@
 #define MachIsS390(mach_) \
    ( 0 )
 #endif
+#ifdef __hppa__
+#define MachIsHPPA(mach_) \
+   ( __hppa__ )
+#else
+#define MachIsHPPA(mach_) \
+   ( 0 )
+#endif
 
 
 static char *f2c_namestr[5] = {"UNKNOWN","Add_", "Add__", "NoChange", "UpCase"};
Index: ATLAS/CONFIG/src/SpewMakeInc.c
===================================================================
--- ATLAS.orig/CONFIG/src/SpewMakeInc.c	2012-07-16 15:39:21.000000000 +0200
+++ ATLAS/CONFIG/src/SpewMakeInc.c	2012-07-16 15:39:32.000000000 +0200
@@ -342,6 +342,8 @@
       return(sp);
    if (MachIsS390(arch))
       return(sp);
+   if (MachIsHPPA(arch))
+     return(sp);
    if (MachIsMIPS(arch))
       return((ptrbits == 64) ? "-mabi=64" : "-mabi=n32");
    if (!CompIsGcc(comp))
Index: ATLAS/CONFIG/src/probe_comp.c
===================================================================
--- ATLAS.orig/CONFIG/src/probe_comp.c	2012-07-16 15:39:21.000000000 +0200
+++ ATLAS/CONFIG/src/probe_comp.c	2012-07-16 15:39:32.000000000 +0200
@@ -509,6 +509,8 @@
       return(sp);
    if (MachIsS390(arch))
       return(sp);
+   if (MachIsHPPA(arch))
+     return(sp);
    if (MachIsMIPS(arch))
       return((ptrbits == 64) ? "-mabi=64" : "-mabi=n32");
    if (!CompIsGcc(comp))
