Index: ATLAS/CONFIG/include/atlconf.h
===================================================================
--- ATLAS.orig/CONFIG/include/atlconf.h	2012-07-16 15:39:32.000000000 +0200
+++ ATLAS/CONFIG/include/atlconf.h	2012-07-16 15:39:33.000000000 +0200
@@ -72,6 +72,14 @@
 #define MachIsALPHA(mach_) \
   ( 0 )
 #endif
+#ifdef __ARMEL__
+#define MachIsARMEL(mach_) \
+  ( __ARMEL__ )
+#else
+#define MachIsARMEL(mach_) \
+  ( 0 )
+#endif
+
 
 
 static char *f2c_namestr[5] = {"UNKNOWN","Add_", "Add__", "NoChange", "UpCase"};
Index: ATLAS/CONFIG/src/SpewMakeInc.c
===================================================================
--- ATLAS.orig/CONFIG/src/SpewMakeInc.c	2012-07-16 15:39:32.000000000 +0200
+++ ATLAS/CONFIG/src/SpewMakeInc.c	2012-07-16 15:39:33.000000000 +0200
@@ -346,6 +346,8 @@
      return(sp);
    if (MachIsALPHA(arch))
      return(sp);
+   if (MachIsARMEL(arch))
+     return(sp);
    if (MachIsMIPS(arch))
       return((ptrbits == 64) ? "-mabi=64" : "-mabi=n32");
    if (!CompIsGcc(comp))
Index: ATLAS/CONFIG/src/probe_comp.c
===================================================================
--- ATLAS.orig/CONFIG/src/probe_comp.c	2012-07-16 15:39:32.000000000 +0200
+++ ATLAS/CONFIG/src/probe_comp.c	2012-07-16 15:39:33.000000000 +0200
@@ -513,6 +513,8 @@
      return(sp);
    if (MachIsALPHA(arch))
      return(sp);
+   if (MachIsARMEL(arch))
+     return(sp);
    if (MachIsMIPS(arch))
       return((ptrbits == 64) ? "-mabi=64" : "-mabi=n32");
    if (!CompIsGcc(comp))
