Index: ATLAS/CONFIG/src/SpewMakeInc.c
===================================================================
--- ATLAS.orig/CONFIG/src/SpewMakeInc.c	2012-07-16 15:39:33.000000000 +0200
+++ ATLAS/CONFIG/src/SpewMakeInc.c	2012-07-16 15:39:34.000000000 +0200
@@ -349,7 +349,8 @@
    if (MachIsARMEL(arch))
      return(sp);
    if (MachIsMIPS(arch))
-      return((ptrbits == 64) ? "-mabi=64" : "-mabi=n32");
+	return(sp);
+//      return((ptrbits == 64) ? "-mabi=64" : "-mabi=n32");
    if (!CompIsGcc(comp))
    {
 /*
Index: ATLAS/CONFIG/src/probe_comp.c
===================================================================
--- ATLAS.orig/CONFIG/src/probe_comp.c	2012-07-16 15:39:33.000000000 +0200
+++ ATLAS/CONFIG/src/probe_comp.c	2012-07-16 15:39:34.000000000 +0200
@@ -516,7 +516,8 @@
    if (MachIsARMEL(arch))
      return(sp);
    if (MachIsMIPS(arch))
-      return((ptrbits == 64) ? "-mabi=64" : "-mabi=n32");
+	return(sp);
+//      return((ptrbits == 64) ? "-mabi=64" : "-mabi=n32");
    if (!CompIsGcc(comp))
    {
 /*
