Index: ATLAS/CONFIG/include/atlconf.h
===================================================================
--- ATLAS.orig/CONFIG/include/atlconf.h	2012-07-16 15:39:33.000000000 +0200
+++ ATLAS/CONFIG/include/atlconf.h	2012-07-16 15:39:35.000000000 +0200
@@ -79,8 +79,13 @@
 #define MachIsARMEL(mach_) \
   ( 0 )
 #endif
-
-
+#ifdef __sh__
+#define MachIsSH(mach_) \
+  ( __sh__ )
+#else
+#define MachIsSH(mach_) \
+  ( 0 )
+#endif
 
 static char *f2c_namestr[5] = {"UNKNOWN","Add_", "Add__", "NoChange", "UpCase"};
 static char *f2c_intstr[5] =
Index: ATLAS/CONFIG/src/SpewMakeInc.c
===================================================================
--- ATLAS.orig/CONFIG/src/SpewMakeInc.c	2012-07-16 15:39:34.000000000 +0200
+++ ATLAS/CONFIG/src/SpewMakeInc.c	2012-07-16 15:39:35.000000000 +0200
@@ -351,6 +351,8 @@
    if (MachIsMIPS(arch))
 	return(sp);
 //      return((ptrbits == 64) ? "-mabi=64" : "-mabi=n32");
+   if (MachIsSH(arch))
+     return(sp);
    if (!CompIsGcc(comp))
    {
 /*
Index: ATLAS/CONFIG/src/probe_comp.c
===================================================================
--- ATLAS.orig/CONFIG/src/probe_comp.c	2012-07-16 15:39:34.000000000 +0200
+++ ATLAS/CONFIG/src/probe_comp.c	2012-07-16 15:39:35.000000000 +0200
@@ -518,6 +518,8 @@
    if (MachIsMIPS(arch))
 	return(sp);
 //      return((ptrbits == 64) ? "-mabi=64" : "-mabi=n32");
+   if (MachIsSH(arch))
+     return(sp);
    if (!CompIsGcc(comp))
    {
 /*
