Index: ATLAS/makes/Make.lib
===================================================================
--- ATLAS.orig/makes/Make.lib	2012-07-16 15:34:19.000000000 +0200
+++ ATLAS/makes/Make.lib	2012-07-16 15:39:35.000000000 +0200
@@ -101,7 +101,7 @@
 
 libatlas.so.3.0 : libatlas.a
 	ld $(LDFLAGS) -shared -soname libatlas.so.3 -o $@ \
-        --whole-archive libatlas.a --no-whole-archive -lc $(LIBS)
+        --whole-archive libatlas.a --no-whole-archive -lc $(LIBS) $(F77SYSLIB)
 liblapack_atlas.so.3.0 : liblapack_atlas.a libatlas.so libcblas.so libf77blas.so
 	ld $(LDFLAGS) -shared -soname liblapack_atlas.so.3 -o $@ \
 	  --whole-archive liblapack_atlas.a  \
