Description: Fix detection of some Intel Core i1 CPUs
Origin: upstream, http://math-atlas.sourceforge.net/errata.html#i2i1
Index: ATLAS/CONFIG/src/backend/archinfo_x86.c
===================================================================
--- ATLAS.orig/CONFIG/src/backend/archinfo_x86.c	2011-05-14 19:33:24.000000000 +0200
+++ ATLAS/CONFIG/src/backend/archinfo_x86.c	2012-07-16 15:39:36.000000000 +0200
@@ -309,9 +309,9 @@
       case 0x1A:
       case 0x1E:
       case 0x1F:
+      case 0x25:
          iret = IntCorei1;
          break;
-      case 0x25:
       case 0x2A:
          iret = IntCorei2;
 	 break;
