Index: ATLAS/CONFIG/src/SpewMakeInc.c
===================================================================
--- ATLAS.orig/CONFIG/src/SpewMakeInc.c	2012-07-16 15:39:35.000000000 +0200
+++ ATLAS/CONFIG/src/SpewMakeInc.c	2012-07-16 15:39:37.000000000 +0200
@@ -682,8 +682,10 @@
          fprintf(fpout, " -melf_i386");
       else if (ptrbits == 64)
          fprintf(fpout, " -melf_x86_64");
-      if (OS == OSFreeBSD)
+#ifdef __FreeBSD_kernel__
+//      if (OS == OSFreeBSD)
          fprintf(fpout, "_fbsd");
+#endif
    }
    fprintf(fpout, "\n   F77SYSLIB = %s\n", f77lib ? f77lib : "");
    fprintf(fpout, "   BC = $(ICC)\n");
