Index: seahorse-3.2.2/configure.ac
===================================================================
--- seahorse-3.2.2.orig/configure.ac	2011-11-14 11:45:26.000000000 +0100
+++ seahorse-3.2.2/configure.ac	2011-11-18 00:10:20.011522368 +0100
@@ -412,16 +412,16 @@
     echo "disabling ssh support"
 else
 
-    AC_PATH_PROG(SSH_KEYGEN_PATH, ssh-keygen, "failed")
-    AC_PATH_PROG(SSH_PATH, ssh, "failed")
+#    AC_PATH_PROG(SSH_KEYGEN_PATH, ssh-keygen, "failed")
+#    AC_PATH_PROG(SSH_PATH, ssh, "failed")
     
-    if test $SSH_KEYGEN_PATH != "failed" && 
-       test $SSH_PATH != "failed"; then
+#    if test $SSH_KEYGEN_PATH != "failed" && i
+#       test $SSH_PATH != "failed"; then
         enable_ssh="yes"
         AC_DEFINE(WITH_SSH, 1, [With SSH Support])
-        AC_DEFINE_UNQUOTED(SSH_KEYGEN_PATH, "$SSH_KEYGEN_PATH", [Path to ssh-keygen])
-        AC_DEFINE_UNQUOTED(SSH_PATH, "$SSH_PATH", [Path to ssh])
-    fi
+        AC_DEFINE_UNQUOTED(SSH_KEYGEN_PATH, "/usr/bin/ssh-keygen", [Path to ssh-keygen])
+        AC_DEFINE_UNQUOTED(SSH_PATH, "/usr/bin/ssh", [Path to ssh])
+#    fi
     
 fi
 
