From 5286af2304b3a59a2deeb508433c13d96fe2287c Mon Sep 17 00:00:00 2001
From: Stef Walter <stefw@gnome.org>
Date: Mon, 30 Apr 2012 18:29:42 +0000
Subject: Fix typo in finalizing of GPGME keyring

---
diff --git a/pgp/seahorse-gpgme-keyring.c b/pgp/seahorse-gpgme-keyring.c
index 79ff39b..579ab09 100644
--- a/pgp/seahorse-gpgme-keyring.c
+++ b/pgp/seahorse-gpgme-keyring.c
@@ -859,7 +859,7 @@ seahorse_gpgme_keyring_finalize (GObject *object)
 {
 	SeahorseGpgmeKeyring *self = SEAHORSE_GPGME_KEYRING (object);
 
-	g_clear_object (self->pv->actions);
+	g_clear_object (&self->pv->actions);
 	g_hash_table_destroy (self->pv->keys);
 
 	/* All monitoring and scheduling should be done */
--
cgit v0.9.0.2
