Index: seahorse-2.30.1/src/seahorse.desktop.in.in
===================================================================
--- seahorse-2.30.1.orig/src/seahorse.desktop.in.in	2010-05-07 22:08:15.740896545 +0200
+++ seahorse-2.30.1/src/seahorse.desktop.in.in	2010-05-07 22:11:12.016895716 +0200
@@ -5,7 +5,7 @@ Exec=seahorse
 Terminal=false
 Type=Application
 Icon=seahorse
-Categories=GNOME;GTK;Security;Utility;
+Categories=GNOME;GTK;Security;Settings;
 StartupNotify=true
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=seahorse
