--- a/configure.ac
+++ b/configure.ac
@@ -59,7 +59,7 @@
     AC_TRY_CFLAGS([$TRY_CFLAGS $CFLAGS],[OPT_CFLAGS="$TRY_CFLAGS"])
 
     dnl arch-specific flags
-    arm_conditional=false
+    build_arm_opt=false
     case "$host" in
     i?86-* | k?-* | x86_64-* | amd64-*)
 	AC_DEFINE([ARCH_X86],,[x86 architecture])
@@ -102,8 +102,12 @@
     alpha*)
 	AC_DEFINE([ARCH_ALPHA],,[alpha architecture]);;
     arm*)
-	arm_conditional=:
-	AC_DEFINE([ARCH_ARM],,[ARM architecture]);;
+	AC_LANG(C)
+	AC_COMPILE_IFELSE(
+		[AC_LANG_SOURCE([[
+			void foo(void) { __asm__ volatile("pld [r1]"); }]])],
+		build_arm_opt=true; AC_DEFINE([ARCH_ARM],,[ARM architecture]),
+		build_arm_opt=false);;
     esac
 elif test x"$CC" = x"tendracc"; then
     dnl TenDRA portability checking compiler
@@ -123,7 +127,7 @@
     esac
 fi
 
-AM_CONDITIONAL(ARCH_ARM, ${arm_conditional})
+AM_CONDITIONAL(ARCH_ARM, test x$build_arm_opt = xtrue)
 
 dnl Checks for libtool - this must be done after we set cflags
 AC_LIBTOOL_WIN32_DLL
