From: Chow Loong Jin <hyperair@debian.org>
Subject: Include glib.h directly instead of glib/gtimer.h
Bug-Debian: http://bugs.debian.org/665546
Index: gtk2-engines-aurora/src/animation.c
===================================================================
--- gtk2-engines-aurora.orig/src/animation.c	2012-03-26 10:58:19.554931909 +0800
+++ gtk2-engines-aurora/src/animation.c	2012-03-26 10:58:39.868400287 +0800
@@ -7,7 +7,7 @@
 #include "animation.h"
 
 #ifdef HAVE_ANIMATION
-#include <glib/gtimer.h>
+#include <glib.h>
 
 struct _AnimationInfo {
 	GTimer *timer;
