Description: Fixes path for build on Debian
Forwarded: no
Author: Sandro Tosi <morph@debian.org>
Last-Update: 2010-02-06

Index: matplotlib-1.1.0~rc1/setupext.py
===================================================================
--- matplotlib-1.1.0~rc1.orig/setupext.py	2011-09-24 17:08:27.000000000 +0200
+++ matplotlib-1.1.0~rc1/setupext.py	2011-09-26 23:37:43.195196916 +0200
@@ -50,13 +50,13 @@
 
 basedir = {
     'win32'  : ['win32_static',],
-    'linux2-alpha' : ['/usr/local', '/usr'],
-    'linux2-hppa' : ['/usr/local', '/usr'],
-    'linux2-mips' : ['/usr/local', '/usr'],
-    'linux2-sparc' : ['/usr/local', '/usr'],
-    'linux2' : ['/usr/local', '/usr'],
-    'linux3' : ['/usr/local', '/usr'],
-    'linux'  : ['/usr/local', '/usr',],
+    'linux2-alpha' : ['/usr'],
+    'linux2-hppa' : ['/usr'],
+    'linux2-mips' : ['/usr'],
+    'linux2-sparc' : ['/usr'],
+    'linux2' : ['/usr'],
+    'linux3' : ['/usr'],
+    'linux'  : ['/usr',],
     'cygwin' : ['/usr/local', '/usr',],
     '_darwin' : ['/sw/lib/freetype2', '/sw/lib/freetype219', '/usr/local',
                 '/usr', '/sw'],
@@ -71,10 +71,10 @@
     'freebsd5' : ['/usr/local', '/usr'],
     'freebsd6' : ['/usr/local', '/usr'],
     'sunos5' : [os.getenv('MPLIB_BASE') or '/usr/local',],
-    'gnukfreebsd5' : ['/usr/local', '/usr'],
-    'gnukfreebsd6' : ['/usr/local', '/usr'],
-    'gnukfreebsd7' : ['/usr/local', '/usr'],
-    'gnukfreebsd8' : ['/usr/local', '/usr'],
+    'gnukfreebsd5' : ['/usr'],
+    'gnukfreebsd6' : ['/usr'],
+    'gnukfreebsd7' : ['/usr'],
+    'gnukfreebsd8' : ['/usr'],
     'gnu0' : ['/usr'],
     'aix5' : ['/usr/local'],
 }
