Description: Fix desktop file
Index: model-builder-0.4.1/model_builder/model-builder.desktop
===================================================================
--- model-builder-0.4.1.orig/model_builder/model-builder.desktop	2008-01-27 16:47:00.000000000 +0530
+++ model-builder-0.4.1/model_builder/model-builder.desktop	2008-01-27 16:50:31.000000000 +0530
@@ -1,5 +1,4 @@
 [Desktop Entry]
-Encoding=UTF-8
 Name=Model Builder
 Comment=Graphical ODE simulator
 GenericName=Graphical ODE simulator
@@ -8,4 +7,4 @@
 Type=Application
 StartupNotify=false
 Terminal=false
-Categories=Science;Math;Engineering;
+Categories=Education;Science;Math;Engineering;
