Description: Fix install directory
Index: model-builder-0.4.1/model_builder/wxFrame1.py
===================================================================
--- model-builder-0.4.1.orig/model_builder/wxFrame1.py	2008-01-22 23:59:46.000000000 +0530
+++ model-builder-0.4.1/model_builder/wxFrame1.py	2008-01-27 16:31:37.000000000 +0530
@@ -1419,7 +1419,7 @@
         return self.OnMenu1items1Menu(event)
 
     def OnMenu3MbhelpMenu(self, event):
-        helpviewer.main(['','MB.hhp'])
+        helpviewer.main(['','/usr/share/model-builder/help/MB.hhp'])
 
 class Future:
     """
