Description: Fix installation files
diff -urN Model-Builder-0.4.0.orig/setup.py Model-Builder-0.4.0/setup.py
--- Model-Builder-0.4.0.orig/setup.py	2007-04-10 20:57:25.000000000 +0530
+++ Model-Builder-0.4.0/setup.py	2007-04-10 21:06:09.000000000 +0530
@@ -33,5 +33,5 @@
       entry_points = {'gui_scripts':['PyMB = model_builder.PyMB:main',]},
       packages = ['model_builder','model_builder/Bayes'],
       package_data = {'':['MB*','*.pdf','*.desktop'],'model_builder':['Docs/*.pdf', 'model-builder.desktop']},
-      data_files=[('/usr/share/pixmaps',['model_builder/MB.ico']),('/usr/share/applications',['model_builder/model-builder.desktop']),('/usr/share/model-builder/examples',glob.glob("model_builder/Examples/*.ode")),('/usr/share/model-builder/docs',['model_builder/Docs/Quick-start.pdf'])],
+      data_files=[('/usr/share/pixmaps',['model_builder/MB.ico']),('/usr/share/applications',['model_builder/model-builder.desktop']),('/usr/share/doc/model-builder/examples',glob.glob("model_builder/Examples/*.ode")),('/usr/share/model-builder/help',glob.glob('model_builder/*.html')+glob.glob('model_builder/MB.*'))],
      )
