Description: Fix issues with displaying the PyMB window
Index: model-builder-0.4.1/model_builder/PyMB.py
===================================================================
--- model-builder-0.4.1.orig/model_builder/PyMB.py	2008-01-27 17:02:48.000000000 +0530
+++ model-builder-0.4.1/model_builder/PyMB.py	2008-01-27 17:02:55.000000000 +0530
@@ -56,7 +56,7 @@
         self.main = wxFrame1.create(None)
         # needed when running from Boa under Windows 9X
         self.SetTopWindow(self.main)
-        self.main.Show();self.main.Hide();self.main.Show()
+        self.main.Show();
         return True
 
 def main():
