--- xserver-xorg-video-fbdev.orig/man/fbdev.man
+++ xserver-xorg-video-fbdev/man/fbdev.man
@@ -38,8 +38,8 @@
 driver and will use it if there are no video modes configured.
 .PP
 For PCI boards you might have to add a BusID line to the Device
-section.  See above for a sample line.  You can use \*q\__xservername__
--scanpci\*q
+section.  See above for a sample line.  You can use \*q__xservername__
+\-scanpci\*q
 to figure out the correct values.
 .PP
 The following driver 
