From 7065f8986434e3dcea7c317e263f8b36a27024aa Mon Sep 17 00:00:00 2001
From: Cyril Brulebois <kibi@debian.org>
Date: Sun, 14 Feb 2010 04:46:12 +0000
Subject: [PATCH] Support building without XV.

---
 configure.ac |   11 +++++++++++
 src/fbdev.c  |    2 ++
 2 files changed, 13 insertions(+), 0 deletions(-)

diff --git a/configure.ac b/configure.ac
index f76cca0..8fa2204 100644
--- a/configure.ac
+++ b/configure.ac
@@ -50,6 +50,10 @@ AC_ARG_ENABLE(pciaccess,     AS_HELP_STRING([--enable-pciaccess],
                              [Enable use of libpciaccess (default: disabled)]),
 			     [PCIACCESS=$enableval], [PCIACCESS=no])
 
+AC_ARG_ENABLE(xv,            AS_HELP_STRING([--enable-xv],
+                             [Enable use of xv (default: enabled)]),
+                             [USE_XV=$enableval], [USE_XV=yes])
+
 # Checks for extensions
 XORG_DRIVER_CHECK_EXT(RANDR, randrproto)
 XORG_DRIVER_CHECK_EXT(RENDER, renderproto)
@@ -74,6 +78,13 @@ if test "x$PCIACCESS" = xyes; then
     XORG_CFLAGS="$XORG_CFLAGS $PCIACCESS_CFLAGS"
 fi
 
+AM_CONDITIONAL(USE_XV, [test "xUSE_XV" = xyes])
+if test "x$USE_XV" = xyes; then
+   AC_DEFINE(USE_XV, 1, [Use XV])
+else
+   AC_DEFINE(USE_XV, 0, [Do not use XV])
+fi
+
 # Checks for libraries.
 
 # Checks for header files.
diff --git a/src/fbdev.c b/src/fbdev.c
index 149bb24..a42f282 100644
--- a/src/fbdev.c
+++ b/src/fbdev.c
@@ -936,6 +936,7 @@ FBDevScreenInit(int scrnIndex, ScreenPtr pScreen, int argc, char **argv)
 	fPtr->CloseScreen = pScreen->CloseScreen;
 	pScreen->CloseScreen = FBDevCloseScreen;
 
+#if USE_XV
 	{
 	    XF86VideoAdaptorPtr *ptr;
 
@@ -944,6 +945,7 @@ FBDevScreenInit(int scrnIndex, ScreenPtr pScreen, int argc, char **argv)
 		xf86XVScreenInit(pScreen,ptr,n);
 	    }
 	}
+#endif
 
 	TRACE_EXIT("FBDevScreenInit");
 
-- 
1.6.6.2

