From 3a2ba2d0fbe78f4eacd438b708ceff6e96903d37 Mon Sep 17 00:00:00 2001
From: David Teigland <teigland@redhat.com>
Date: Wed, 1 Aug 2012 17:00:53 -0500
Subject: [PATCH] wdmd: use lockfile mode 644

Signed-off-by: David Teigland <teigland@redhat.com>
---
 wdmd/main.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/wdmd/main.c
+++ b/wdmd/main.c
@@ -819,7 +819,7 @@
 
 	sprintf(lockfile_path, "%s/wdmd.pid", WDMD_RUN_DIR);
 
-	fd = open(lockfile_path, O_CREAT|O_WRONLY|O_CLOEXEC, 0666);
+	fd = open(lockfile_path, O_CREAT|O_WRONLY|O_CLOEXEC, 0644);
 	if (fd < 0) {
 		log_error("lockfile open error %s: %s",
 			  lockfile_path, strerror(errno));
