From 1339694c3bad23055f896e90353c81fd65bd4a7e Mon Sep 17 00:00:00 2001
From: David Teigland <teigland@redhat.com>
Date: Thu, 2 Aug 2012 11:27:54 -0500
Subject: [PATCH] sanlock: use lockfile mode 644

Signed-off-by: David Teigland <teigland@redhat.com>
---
 src/lockfile.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/src/lockfile.c
+++ b/src/lockfile.c
@@ -47,7 +47,7 @@
 
 	snprintf(path, PATH_MAX, "%s/%s", dir, name);
 
-	fd = open(path, O_CREAT|O_WRONLY|O_CLOEXEC, 0666);
+	fd = open(path, O_CREAT|O_WRONLY|O_CLOEXEC, 0644);
 	if (fd < 0) {
 		log_error("lockfile open error %s: %s",
 			  path, strerror(errno));
