put the private shared libraries into /usr/lib/sanlock

diff --git a/src/Makefile b/src/Makefile
index d101f30..92d5d9f 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -8,6 +8,8 @@ CMD_TARGET = sanlock
 HEADER_TARGET = sanlock.h sanlock_rv.h sanlock_resource.h sanlock_admin.h sanlock_direct.h
 MAN_TARGET = sanlock.8
 
+LIBDIR_NAME = sanlock
+
 SOMAJOR=1
 SOMINOR=0
 
@@ -72,7 +74,7 @@ CFLAGS += -D_GNU_SOURCE -g \
 
 CMD_CFLAGS = $(CFLAGS) -fPIE -DPIE
 
-CMD_LDFLAGS += -Wl,-z,now -Wl,-z,relro -pie
+CMD_LDFLAGS += -Wl,-z,now -Wl,-z,relro -pie -Wl,-rpath=/usr/lib/sanlock
 CMD_LDADD += -lpthread -luuid -lrt -laio -lblkid -lsanlock -L../wdmd -lwdmd
 
 LIB_ENTIRE_LDFLAGS += -lpthread -lrt -laio -lblkid -L../wdmd -lwdmd
@@ -112,14 +114,15 @@ MANDIR=/usr/share/man
 install: all
 	$(INSTALL) -d $(DESTDIR)/$(BINDIR)
 	$(INSTALL) -d $(DESTDIR)/$(LIBDIR)
+	$(INSTALL) -d $(DESTDIR)/$(LIBDIR)/$(LIBDIR_NAME)
 	$(INSTALL) -d $(DESTDIR)/$(HEADIR)
 	$(INSTALL) -d $(DESTDIR)/$(MANDIR)/man8
 	$(INSTALL) -c -m 755 $(CMD_TARGET) $(DESTDIR)/$(BINDIR)
-	$(INSTALL) -c -m 755 $(LIBSO_ENTIRE_TARGET) $(DESTDIR)/$(LIBDIR)
+	$(INSTALL) -c -m 755 $(LIBSO_ENTIRE_TARGET) $(DESTDIR)/$(LIBDIR)/$(LIBDIR_NAME)
 	$(INSTALL) -c -m 755 $(LIBSO_CLIENT_TARGET) $(DESTDIR)/$(LIBDIR)
-	cp -a $(LIB_ENTIRE_TARGET).so $(DESTDIR)/$(LIBDIR)
+	cp -a $(LIB_ENTIRE_TARGET).so $(DESTDIR)/$(LIBDIR)/$(LIBDIR_NAME)
 	cp -a $(LIB_CLIENT_TARGET).so $(DESTDIR)/$(LIBDIR)
-	cp -a $(LIB_ENTIRE_TARGET).so.$(SOMAJOR) $(DESTDIR)/$(LIBDIR)
+	cp -a $(LIB_ENTIRE_TARGET).so.$(SOMAJOR) $(DESTDIR)/$(LIBDIR)/$(LIBDIR_NAME)
 	cp -a $(LIB_CLIENT_TARGET).so.$(SOMAJOR) $(DESTDIR)/$(LIBDIR)
 	$(INSTALL) -c -m 644 $(HEADER_TARGET) $(DESTDIR)/$(HEADIR)
 	$(INSTALL) -m 644 $(MAN_TARGET) $(DESTDIR)/$(MANDIR)/man8/
diff --git a/wdmd/Makefile b/wdmd/Makefile
index 205245b..4db2dc1 100644
--- a/wdmd/Makefile
+++ b/wdmd/Makefile
@@ -10,6 +10,8 @@ HEADER_TARGET = wdmd.h
 MAN_TARGET = wdmd.8
 TEST_TARGET = wdmd_client
 
+LIBDIR_NAME = sanlock
+
 SOMAJOR=1
 SOMINOR=0
 SHLIB_TARGET = $(LIB_TARGET).so.$(SOMAJOR).$(SOMINOR)
@@ -42,7 +44,7 @@ CFLAGS += -D_GNU_SOURCE -g \
 	-fdiagnostics-show-option \
 	-fPIE -DPIE
 
-CMD_LDFLAGS += -Wl,-z,now -Wl,-z,relro -pie
+CMD_LDFLAGS += -Wl,-z,now -Wl,-z,relro -pie -Wl,-rpath=/usr/lib/sanlock
 CMD_LDADD += -lwdmd -lrt
 
 LIB_LDFLAGS += -Wl,-z,relro -pie
@@ -79,11 +81,12 @@ MANDIR=/usr/share/man
 install: all
 	$(INSTALL) -d $(DESTDIR)/$(BINDIR)
 	$(INSTALL) -d $(DESTDIR)/$(LIBDIR)
+	$(INSTALL) -d $(DESTDIR)/$(LIBDIR)/$(LIBDIR_NAME)
 	$(INSTALL) -d $(DESTDIR)/$(HEADIR)
 	$(INSTALL) -d $(DESTDIR)/$(MANDIR)/man8
 	$(INSTALL) -c -m 755 $(CMD_TARGET) $(DESTDIR)/$(BINDIR)
-	$(INSTALL) -c -m 755 $(SHLIB_TARGET) $(DESTDIR)/$(LIBDIR)
-	cp -a $(LIB_TARGET).so $(DESTDIR)/$(LIBDIR)
-	cp -a $(LIB_TARGET).so.$(SOMAJOR) $(DESTDIR)/$(LIBDIR)
+	$(INSTALL) -c -m 755 $(SHLIB_TARGET) $(DESTDIR)/$(LIBDIR)/$(LIBDIR_NAME)
+	cp -a $(LIB_TARGET).so $(DESTDIR)/$(LIBDIR)/$(LIBDIR_NAME)
+	cp -a $(LIB_TARGET).so.$(SOMAJOR) $(DESTDIR)/$(LIBDIR)/$(LIBDIR_NAME)
 	$(INSTALL) -c -m 644 $(HEADER_TARGET) $(DESTDIR)/$(HEADIR)
 	$(INSTALL) -m 644 $(MAN_TARGET) $(DESTDIR)/$(MANDIR)/man8
