Index: sanlock-2.2/src/cmd.c
===================================================================
--- sanlock-2.2.orig/src/cmd.c	2012-05-07 17:43:52.000000000 +0200
+++ sanlock-2.2/src/cmd.c	2013-01-10 10:16:41.452028239 +0100
@@ -302,7 +302,7 @@
 		token->host_id = space.host_id;
 		token->host_generation = space.host_generation;
 		token->pid = cl_pid;
-		if (cl->restrict & SANLK_RESTRICT_SIGKILL)
+		if (cl->restricted & SANLK_RESTRICT_SIGKILL)
 			token->flags |= T_RESTRICT_SIGKILL;
 
 		/* save a record of what this token_id is for later debugging */
@@ -1278,7 +1278,7 @@
 		 "ci=%d "
 		 "fd=%d "
 		 "pid=%d "
-		 "restrict=%x "
+		 "restricted=%x "
 		 "cmd_active=%d "
 		 "cmd_last=%d "
 		 "pid_dead=%d "
@@ -1289,7 +1289,7 @@
 		 ci,
 		 cl->fd,
 		 cl->pid,
-		 cl->restrict,
+		 cl->restricted,
 		 cl->cmd_active,
 		 cl->cmd_last,
 		 cl->pid_dead,
@@ -1621,7 +1621,7 @@
 	log_debug("cmd_restrict ci %d fd %d pid %d flags %x",
 		  ci, fd, client[ci].pid, h_recv->cmd_flags);
 
-	client[ci].restrict = h_recv->cmd_flags;
+	client[ci].restricted = h_recv->cmd_flags;
 
 	send_result(fd, h_recv, 0);
 }
Index: sanlock-2.2/src/main.c
===================================================================
--- sanlock-2.2.orig/src/main.c	2013-01-10 09:52:54.000000000 +0100
+++ sanlock-2.2/src/main.c	2013-01-10 10:18:24.148033530 +0100
@@ -157,7 +157,7 @@
 	cl->need_free = 0;
 	cl->kill_count = 0;
 	cl->kill_last = 0;
-	cl->restrict = 0;
+	cl->restricted = 0;
 	memset(cl->owner_name, 0, sizeof(cl->owner_name));
 	cl->workfn = NULL;
 	cl->deadfn = NULL;
@@ -481,9 +481,9 @@
 		fd = cl->fd;
 		pid = cl->pid;
 
-		if (cl->restrict & SANLK_RESTRICT_SIGKILL)
+		if (cl->restricted & SANLK_RESTRICT_SIGKILL)
 			sig = SIGTERM;
-		else if (cl->restrict & SANLK_RESTRICT_SIGTERM)
+		else if (cl->restricted & SANLK_RESTRICT_SIGTERM)
 			sig = SIGKILL;
 		else if (cl->kill_count <= main_task.kill_count_term)
 			sig = SIGTERM;
@@ -990,7 +990,7 @@
 			  ci, rv, h.magic, SM_MAGIC);
 		goto dead;
 	}
-	if (client[ci].restrict & SANLK_RESTRICT_ALL) {
+	if (client[ci].restricted & SANLK_RESTRICT_ALL) {
 		log_error("ci %d fd %d pid %d cmd %d restrict all",
 			  ci, client[ci].fd, client[ci].pid, h.cmd);
 		goto dead;
Index: sanlock-2.2/src/sanlock_internal.h
===================================================================
--- sanlock-2.2.orig/src/sanlock_internal.h	2012-05-07 17:43:52.000000000 +0200
+++ sanlock-2.2/src/sanlock_internal.h	2013-01-10 10:18:58.000030946 +0100
@@ -67,7 +67,7 @@
  * 'struct resource' keeps track of resources globally, resources list
  */
 
-#define T_RESTRICT_SIGKILL	0x00000001 /* inherited from client->restrict */
+#define T_RESTRICT_SIGKILL	0x00000001 /* inherited from client->restricted */
 #define T_LS_DEAD		0x00000002 /* don't bother trying to release if ls is dead */
 
 struct token {
@@ -492,7 +492,7 @@
 	int suspend;
 	int need_free;
 	int kill_count;
-	uint32_t restrict;
+	uint32_t restricted;
 	uint64_t kill_last;
 	char owner_name[SANLK_NAME_LEN+1];
 	pthread_mutex_t mutex;
