Is needed to compile on some systems

diff --git a/src/main.c b/src/main.c
index 82c39af..898e2a8 100644
--- a/src/main.c
+++ b/src/main.c
@@ -52,6 +52,10 @@
 
 #define RELEASE_VERSION "2.2"
 
+#ifndef SCHED_RESET_ON_FORK
+#define SCHED_RESET_ON_FORK 0
+#endif
+
 struct thread_pool {
 	int num_workers;
 	int max_workers;
diff --git a/wdmd/main.c b/wdmd/main.c
index bd95f9d..0623b7c 100644
--- a/wdmd/main.c
+++ b/wdmd/main.c
@@ -43,6 +43,10 @@
 
 #define RELEASE_VERSION "2.2"
 
+#ifndef SCHED_RESET_ON_FORK
+#define SCHED_RESET_ON_FORK 0
+#endif
+
 #define DEFAULT_TEST_INTERVAL 10
 #define DEFAULT_FIRE_TIMEOUT 60
 #define DEFAULT_HIGH_PRIORITY 1
