Index: shtool-2.0.8/sh.mdate
Description: Fix manpage-has-errors-from-pod2man by William Vera <billy@billy.com.mx>
===================================================================
--- shtool-2.0.8.orig/sh.mdate	2010-04-21 01:19:07.000000000 -0500
+++ shtool-2.0.8/sh.mdate	2010-04-21 01:20:01.000000000 -0500
@@ -219,6 +219,9 @@
 element represented as a single character out of ``C<d>'', ``C<m>'' and
 ``C<y>''. The default for I<spec> is ``C<dmy>''.
 
+
+=back
+
 =head1 EXAMPLE
 
  #   shell script
Index: shtool-2.0.8/sh.mkdir
===================================================================
--- shtool-2.0.8.orig/sh.mkdir	2010-04-21 01:19:07.000000000 -0500
+++ shtool-2.0.8/sh.mkdir	2010-04-21 01:20:29.000000000 -0500
@@ -183,6 +183,9 @@
 which is usually based on the executing gid or the parent setgid
 directory.
 
+
+=back
+
 =head1 EXAMPLE
 
  #   Makefile
Index: shtool-2.0.8/sh.mkshadow
===================================================================
--- shtool-2.0.8.orig/sh.mkshadow	2010-04-21 01:19:07.000000000 -0500
+++ shtool-2.0.8/sh.mkshadow	2010-04-21 01:20:50.000000000 -0500
@@ -188,6 +188,9 @@
 Really shadow all files and directories in I<src-dir>. Default is to
 skip CVS related files and directories, backup files, object files, etc.
 
+
+=back
+
 =head1 EXAMPLE
 
  #   shell script
