Author: Raphael Geissert <geissert@debian.org>
Description: Fix the mkdir -p implementation, make parallel executions fail.
Index: shtool-2.0.8/sh.mkdir
===================================================================
--- shtool-2.0.8.orig/sh.mkdir	2010-07-25 20:10:47.000000000 -0500
+++ shtool-2.0.8/sh.mkdir	2010-07-25 20:14:54.000000000 -0500
@@ -86,7 +86,11 @@
                 if [ ".$opt_t" = .yes ]; then
                     echo "mkdir $pathcomp" 1>&2
                 fi
-                mkdir $pathcomp || errstatus=$?
+                mkdir $pathcomp || {
+                    _errstatus=$?
+                    [ -d "$pathcomp" ] || errstatus=${_errstatus}
+                    unset _errstatus
+                }
                 if [ ".$opt_o" != . ]; then
                     if [ ".$opt_t" = .yes ]; then
                         echo "chown $opt_o $pathcomp" 1>&2
