Index: shtool-2.0.8/sh.prop
Description: Fix spelling in a manpage by William Vera <billy@billy.com.mx>
===================================================================
--- shtool-2.0.8.orig/sh.prop	2010-04-20 07:18:46.000000000 -0500
+++ shtool-2.0.8/sh.prop	2010-04-20 07:19:11.000000000 -0500
@@ -85,7 +85,7 @@
 
 This command displays a processing indication though a running
 propeller. It is intended to be run at the end of a pipe ("C<|>")
-sequence where on C<stdin> logging/processing informations found. For
+sequence where on C<stdin> logging/processing information found. For
 every line on C<stdin> the propeller advances one step clock-wise.
 
 =head1 OPTIONS
