Index: shtool-2.0.8/sh.echo
Description: Fix manpage-has-bad-whatis-entry by William Vera <billy@billy.com.mx>
===================================================================
--- shtool-2.0.8.orig/sh.echo	2008-04-02 14:10:17.000000000 -0600
+++ shtool-2.0.8/sh.echo	2010-04-21 06:40:29.000000000 -0500
@@ -339,7 +339,7 @@
 
 =head1 NAME
 
-B<shtool echo> - B<GNU shtool> echo(1) extensional command
+B<shtool-echo> - B<GNU shtool> echo(1) extensional command
 
 =head1 SYNOPSIS
 
Index: shtool-2.0.8/sh.fixperm
===================================================================
--- shtool-2.0.8.orig/sh.fixperm	2008-04-02 14:10:17.000000000 -0600
+++ shtool-2.0.8/sh.fixperm	2010-04-21 06:40:29.000000000 -0500
@@ -86,7 +86,7 @@
 
 =head1 NAME
 
-B<shtool fixperm> - B<GNU shtool> file permission fixing command
+B<shtool-fixperm> - B<GNU shtool> file permission fixing command
 
 =head1 SYNOPSIS
 
Index: shtool-2.0.8/sh.install
===================================================================
--- shtool-2.0.8.orig/sh.install	2008-04-02 14:10:17.000000000 -0600
+++ shtool-2.0.8/sh.install	2010-04-21 06:40:29.000000000 -0500
@@ -200,7 +200,7 @@
 
 =head1 NAME
 
-B<shtool install> - B<GNU shtool> install(1) command
+B<shtool-install> - B<GNU shtool> install(1) command
 
 =head1 SYNOPSIS
 
Index: shtool-2.0.8/sh.mdate
===================================================================
--- shtool-2.0.8.orig/sh.mdate	2010-04-21 06:40:29.000000000 -0500
+++ shtool-2.0.8/sh.mdate	2010-04-21 06:40:29.000000000 -0500
@@ -162,7 +162,7 @@
 
 =head1 NAME
 
-B<shtool mdate> - B<GNU shtool> pretty-print last modification time
+B<shtool-mdate> - B<GNU shtool> pretty-print last modification time
 
 =head1 SYNOPSIS
 
Index: shtool-2.0.8/sh.mkdir
===================================================================
--- shtool-2.0.8.orig/sh.mkdir	2010-04-21 06:40:29.000000000 -0500
+++ shtool-2.0.8/sh.mkdir	2010-04-21 06:40:29.000000000 -0500
@@ -121,7 +121,7 @@
 
 =head1 NAME
 
-B<shtool mkdir> - B<GNU shtool> mkdir(1) style command
+B<shtool-mkdir> - B<GNU shtool> mkdir(1) style command
 
 =head1 SYNOPSIS
 
Index: shtool-2.0.8/sh.mkln
===================================================================
--- shtool-2.0.8.orig/sh.mkln	2008-06-06 07:09:35.000000000 -0500
+++ shtool-2.0.8/sh.mkln	2010-04-21 06:40:29.000000000 -0500
@@ -186,7 +186,7 @@
 
 =head1 NAME
 
-B<shtool mkln> - B<GNU shtool> enhanced ln(1) replacement
+B<shtool-mkln> - B<GNU shtool> enhanced ln(1) replacement
 
 =head1 SYNOPSIS
 
Index: shtool-2.0.8/sh.mkshadow
===================================================================
--- shtool-2.0.8.orig/sh.mkshadow	2010-04-21 06:40:29.000000000 -0500
+++ shtool-2.0.8/sh.mkshadow	2010-04-21 06:40:29.000000000 -0500
@@ -148,7 +148,7 @@
 
 =head1 NAME
 
-B<shtool mkshadow> - B<GNU shtool> create shadow tree using symlinks
+B<shtool-mkshadow> - B<GNU shtool> create shadow tree using symlinks
 
 =head1 SYNOPSIS
 
Index: shtool-2.0.8/sh.move
===================================================================
--- shtool-2.0.8.orig/sh.move	2008-04-02 14:10:17.000000000 -0600
+++ shtool-2.0.8/sh.move	2010-04-21 06:40:29.000000000 -0500
@@ -123,7 +123,7 @@
 
 =head1 NAME
 
-B<shtool move> - B<GNU shtool> enhanced mv(1) replacement
+B<shtool-move> - B<GNU shtool> enhanced mv(1) replacement
 
 =head1 SYNOPSIS
 
Index: shtool-2.0.8/sh.path
===================================================================
--- shtool-2.0.8.orig/sh.path	2008-04-02 14:10:17.000000000 -0600
+++ shtool-2.0.8/sh.path	2010-04-21 06:40:29.000000000 -0500
@@ -175,7 +175,7 @@
 
 =head1 NAME
 
-B<shtool path> - B<GNU shtool> command dealing with shell path variables
+B<shtool-path> - B<GNU shtool> command dealing with shell path variables
 
 =head1 SYNOPSIS
 
Index: shtool-2.0.8/sh.platform
===================================================================
--- shtool-2.0.8.orig/sh.platform	2008-04-02 14:10:17.000000000 -0600
+++ shtool-2.0.8/sh.platform	2010-04-21 06:40:29.000000000 -0500
@@ -667,7 +667,7 @@
 
 =head1 NAME
 
-B<shtool platform> - B<GNU shtool> Unix platform identification
+B<shtool-platform> - B<GNU shtool> Unix platform identification
 
 =head1 SYNOPSIS
 
Index: shtool-2.0.8/sh.prop
===================================================================
--- shtool-2.0.8.orig/sh.prop	2010-04-21 06:40:29.000000000 -0500
+++ shtool-2.0.8/sh.prop	2010-04-21 06:40:29.000000000 -0500
@@ -74,7 +74,7 @@
 
 =head1 NAME
 
-B<shtool prop> - B<GNU shtool> propeller processing indication
+B<shtool-prop> - B<GNU shtool> propeller processing indication
 
 =head1 SYNOPSIS
 
Index: shtool-2.0.8/sh.rotate
===================================================================
--- shtool-2.0.8.orig/sh.rotate	2008-04-02 14:10:17.000000000 -0600
+++ shtool-2.0.8/sh.rotate	2010-04-21 06:40:29.000000000 -0500
@@ -409,7 +409,7 @@
 
 =head1 NAME
 
-B<shtool rotate> - B<GNU shtool> log file rotation
+B<shtool-rotate> - B<GNU shtool> log file rotation
 
 =head1 SYNOPSIS
 
Index: shtool-2.0.8/sh.scpp
===================================================================
--- shtool-2.0.8.orig/sh.scpp	2008-04-02 14:10:17.000000000 -0600
+++ shtool-2.0.8/sh.scpp	2010-04-21 06:40:29.000000000 -0500
@@ -256,7 +256,7 @@
 
 =head1 NAME
 
-B<shtool scpp> - B<GNU shtool> C source file pre-processor
+B<shtool-scpp> - B<GNU shtool> C source file pre-processor
 
 =head1 SYNOPSIS
 
Index: shtool-2.0.8/sh.slo
===================================================================
--- shtool-2.0.8.orig/sh.slo	2008-04-02 14:10:17.000000000 -0600
+++ shtool-2.0.8/sh.slo	2010-04-21 06:40:29.000000000 -0500
@@ -187,7 +187,7 @@
 
 =head1 NAME
 
-B<shtool slo> - B<GNU shtool> separate linker options by library class
+B<shtool-slo> - B<GNU shtool> separate linker options by library class
 
 =head1 SYNOPSIS
 
Index: shtool-2.0.8/sh.subst
===================================================================
--- shtool-2.0.8.orig/sh.subst	2008-04-02 14:10:17.000000000 -0600
+++ shtool-2.0.8/sh.subst	2010-04-21 06:40:29.000000000 -0500
@@ -199,7 +199,7 @@
 
 =head1 NAME
 
-B<shtool subst> - B<GNU shtool> sed(1) substitution operations
+B<shtool-subst> - B<GNU shtool> sed(1) substitution operations
 
 =head1 SYNOPSIS
 
Index: shtool-2.0.8/sh.table
===================================================================
--- shtool-2.0.8.orig/sh.table	2008-04-02 14:10:17.000000000 -0600
+++ shtool-2.0.8/sh.table	2010-04-21 06:40:29.000000000 -0500
@@ -96,7 +96,7 @@
 
 =head1 NAME
 
-B<shtool table> - B<GNU shtool> pretty-print a field-separated list
+B<shtool-table> - B<GNU shtool> pretty-print a field-separated list
 
 =head1 SYNOPSIS
 
Index: shtool-2.0.8/sh.tarball
===================================================================
--- shtool-2.0.8.orig/sh.tarball	2008-04-02 14:10:17.000000000 -0600
+++ shtool-2.0.8/sh.tarball	2010-04-21 06:40:29.000000000 -0500
@@ -200,7 +200,7 @@
 
 =head1 NAME
 
-B<shtool tarball> - B<GNU shtool> command for rolling standardized tarballs
+B<shtool-tarball> - B<GNU shtool> command for rolling standardized tarballs
 
 =head1 SYNOPSIS
 
Index: shtool-2.0.8/sh.version
===================================================================
--- shtool-2.0.8.orig/sh.version	2008-04-02 14:10:17.000000000 -0600
+++ shtool-2.0.8/sh.version	2010-04-21 06:40:29.000000000 -0500
@@ -326,7 +326,7 @@
 
 =head1 NAME
 
-B<shtool version> - B<GNU shtool> maintain version information file
+B<shtool-version> - B<GNU shtool> maintain version information file
 
 =head1 SYNOPSIS
 
Index: shtool-2.0.8/shtool.pod
===================================================================
--- shtool-2.0.8.orig/shtool.pod	2008-04-02 14:10:17.000000000 -0600
+++ shtool-2.0.8/shtool.pod	2010-04-21 06:40:29.000000000 -0500
@@ -22,7 +22,7 @@
 
 =head1 NAME
 
-B<GNU shtool> - The GNU Portable Shell Tool
+B<shtool> - The GNU Portable Shell Tool
 
 =head1 VERSION
 
Index: shtool-2.0.8/sh.arx
===================================================================
--- shtool-2.0.8.orig/sh.arx	2010-04-21 06:42:21.000000000 -0500
+++ shtool-2.0.8/sh.arx	2010-04-21 06:42:39.000000000 -0500
@@ -108,7 +108,7 @@
 
 =head1 NAME
 
-B<shtool arx> - B<GNU shtool> ar(1) extensional command
+B<shtool-arx> - B<GNU shtool> ar(1) extensional command
 
 =head1 SYNOPSIS
 
