diff -r -u libextractor-0.6.3/src/main/extractor.c mine/src/main/extractor.c
--- libextractor-0.6.3/src/main/extractor.c	2011-11-28 17:52:27.000000000 +0800
+++ mine/src/main/extractor.c	2012-05-29 23:30:12.772916677 +0800
@@ -691,14 +691,14 @@
   const char *(*opt_fun)(void);
 
   if (NULL != options) *options = NULL;
-  sym_name = strstr (prefix, "_");
+  sym_name = strrchr (prefix, '_');
   if (sym_name == NULL)
     return NULL;
   sym_name++;
   sym = strdup (sym_name);
   if (sym == NULL)
     return NULL;
-  dot = strstr (sym, ".");
+  dot = strchr (sym, '.');
   if (dot != NULL)
     *dot = '\0';
   name = malloc(strlen(sym) + strlen(template) + 1);
