Author: Sune Vuorela <sune@debian.org>
Description: Version the cppparser library
Last-Update: 2010-03-01

--- a/parser/CMakeLists.txt
+++ b/parser/CMakeLists.txt
@@ -36,6 +36,7 @@
 
 add_definitions(-D__CPPPARSER_BUILDING)
 add_library(cppparser SHARED ${cppparser_LIB_SRCS})
+set_target_properties(cppparser PROPERTIES VERSION 0a.0.0 SOVERSION 0a)
 target_link_libraries(cppparser ${QT_QTCORE_LIBRARY})
 
 install(TARGETS cppparser LIBRARY DESTINATION ${LIB_INSTALL_DIR}
