--- a/configure.in
+++ b/configure.in
@@ -277,7 +277,7 @@ AC_ARG_WITH(pam, [  --with-pam=DIR
 if test "$with_pam" != no; then
   if test -d $with_pam; then
     CPPFLAGS="$CPPFLAGS -I${with_pam}/include"
-    LDFLAGS="$LDFLAGS -L${with_pam}/lib"
+    LDFLAGS="$LDFLAGS -L${with_pam}/$DEB_HOST_MULTIARCH/lib"
   fi
   AC_CHECK_HEADERS(security/pam_appl.h pam/pam_appl.h)
   cmu_save_LIBS="$LIBS"
--- a/saslauthd/configure.in
+++ b/saslauthd/configure.in
@@ -95,7 +95,7 @@ AC_ARG_WITH(pam, [  --with-pam=DIR
 if test "$with_pam" != no; then
   if test -d $with_pam; then
     CPPFLAGS="$CPPFLAGS -I${with_pam}/include"
-    LDFLAGS="$LDFLAGS -L${with_pam}/lib"
+    LDFLAGS="$LDFLAGS -L${with_pam}/$DEB_HOST_MULTIARCH/lib"
   fi
   cmu_save_LIBS="$LIBS"
   AC_CHECK_LIB(pam, pam_start, [
