--- a/src/celutil/util.cpp
+++ b/src/celutil/util.cpp
@@ -12,6 +12,7 @@
 #include "util.h"
 #include <sys/types.h>
 #include <sys/stat.h>
+#include <libintl.h>
 
 using namespace std;
 
--- a/src/celengine/overlay.h
+++ b/src/celengine/overlay.h
@@ -11,6 +11,7 @@
 #define _OVERLAY_H_
 
 #include <string>
+#include <cstdio>
 #include <iostream>
 #include <celtxf/texturefont.h>
 
--- a/src/celengine/asterism.cpp
+++ b/src/celengine/asterism.cpp
@@ -15,6 +15,7 @@
 #endif /* TARGET_OS_MAC */
 #endif /* _WIN32 */
 
+#include <libintl.h>
 #include <celutil/util.h>
 #include <celutil/debug.h>
 #include "parser.h"
--- a/src/celengine/astro.cpp
+++ b/src/celengine/astro.cpp
@@ -11,6 +11,7 @@
 #include <cmath>
 #include <iomanip>
 #include <cstdio>
+#include <cstring>
 #include <time.h>
 #include <celutil/basictypes.h>
 #include <celmath/mathlib.h>
--- a/src/celengine/dsodb.cpp
+++ b/src/celengine/dsodb.cpp
@@ -14,6 +14,7 @@
 #include <cstdlib>
 #include <cstdio>
 #include <cassert>
+#include <libintl.h>
 #include <algorithm>
 #include <celmath/mathlib.h>
 #include <celmath/plane.h>
--- a/src/celengine/fragmentprog.cpp
+++ b/src/celengine/fragmentprog.cpp
@@ -7,6 +7,7 @@
 // as published by the Free Software Foundation; either version 2
 // of the License, or (at your option) any later version.
 
+#include <libintl.h>
 #include <iostream>
 #include <fstream>
 #include <string>
--- a/src/celengine/galaxy.cpp
+++ b/src/celengine/galaxy.cpp
@@ -12,6 +12,8 @@
 #include <algorithm>
 #include <cstdio>
 #include <cassert>
+#include <cstring>
+#include <libintl.h>
 #include "celestia.h"
 #include <celmath/mathlib.h>
 #include <celmath/perlin.h>
--- a/src/celengine/image.cpp
+++ b/src/celengine/image.cpp
@@ -63,6 +63,7 @@
 #include "celestia.h"
 
 #include <cassert>
+#include <libintl.h>
 #include <iostream>
 #include <algorithm>
 #include <cmath>
--- a/src/celengine/location.cpp
+++ b/src/celengine/location.cpp
@@ -7,6 +7,7 @@
 // as published by the Free Software Foundation; either version 2
 // of the License, or (at your option) any later version.
 
+#include <libintl.h>
 #include <map>
 #include <celengine/location.h>
 #include <celengine/body.h>
--- a/src/celengine/meshmanager.cpp
+++ b/src/celengine/meshmanager.cpp
@@ -13,6 +13,7 @@
 #include <iostream>
 #include <fstream>
 #include <cassert>
+#include <libintl.h>
 
 #include "celestia.h"
 #include <celutil/debug.h>
--- a/src/celengine/nebula.cpp
+++ b/src/celengine/nebula.cpp
@@ -8,7 +8,8 @@
 // of the License, or (at your option) any later version.
 
 #include <algorithm>
-#include <stdio.h>
+#include <cstdio>
+#include <libintl.h>
 #include "celestia.h"
 #include <celmath/mathlib.h>
 #include <celutil/util.h>
--- a/src/celengine/opencluster.cpp
+++ b/src/celengine/opencluster.cpp
@@ -8,7 +8,8 @@
 // of the License, or (at your option) any later version.
 
 #include <algorithm>
-#include <stdio.h>
+#include <cstdio>
+#include <libintl.h>
 #include "celestia.h"
 #include <celmath/mathlib.h>
 #include <celutil/util.h>
--- a/src/celengine/overlay.cpp
+++ b/src/celengine/overlay.cpp
@@ -10,6 +10,7 @@
 #include <cstring>
 #include <cstdarg>
 #include <cstdio>
+#include <cstring>
 #include <celutil/utf8.h>
 #include "gl.h"
 #include "vecgl.h"
--- a/src/celengine/solarsys.cpp
+++ b/src/celengine/solarsys.cpp
@@ -19,6 +19,7 @@
 #endif /* ! TARGET_OS_MAC */
 #endif /* ! _WIN32 */
 
+#include <libintl.h>
 #include <celutil/debug.h>
 #include <celmath/mathlib.h>
 #include <celutil/util.h>
--- a/src/celengine/star.cpp
+++ b/src/celengine/star.cpp
@@ -9,6 +9,7 @@
 
 #include <celmath/mathlib.h>
 #include <cstring>
+#include <cstdio>
 #include <cassert>
 #include "celestia.h"
 #include "astro.h"
--- a/src/celengine/stardb.cpp
+++ b/src/celengine/stardb.cpp
@@ -11,7 +11,9 @@
 #include <cmath>
 #include <cstdlib>
 #include <cstdio>
+#include <cstring>
 #include <cassert>
+#include <libintl.h>
 #include <algorithm>
 #include <celmath/mathlib.h>
 #include <celmath/plane.h>
--- a/src/celengine/texture.cpp
+++ b/src/celengine/texture.cpp
@@ -22,6 +22,7 @@
 #endif
 
 #include <cmath>
+#include <libintl.h>
 #include <algorithm>
 #include <iostream>
 #include <fstream>
--- a/src/celengine/vertexprog.cpp
+++ b/src/celengine/vertexprog.cpp
@@ -7,6 +7,7 @@
 // as published by the Free Software Foundation; either version 2
 // of the License, or (at your option) any later version.
 
+#include <libintl.h>
 #include <iostream>
 #include <fstream>
 #include <string>
--- a/src/celestia/celestiacore.cpp
+++ b/src/celestia/celestiacore.cpp
@@ -23,6 +23,7 @@
 #include <cstring>
 #include <cassert>
 #include <ctime>
+#include <libintl.h>
 #include <celengine/gl.h>
 #include <celmath/vecmath.h>
 #include <celmath/quaternion.h>
--- a/src/celestia/glutmain.cpp
+++ b/src/celestia/glutmain.cpp
@@ -15,6 +15,7 @@
 #include <cstdlib>
 #include <cctype>
 #include <cstring>
+#include <libintl.h>
 #include <time.h>
 #include <unistd.h>
 #include <celengine/gl.h>
--- a/src/celestia/oggtheoracapture.cpp
+++ b/src/celestia/oggtheoracapture.cpp
@@ -62,6 +62,9 @@
 #include <cstdlib>
 #include <cstdio>
 #include <cmath>
+#include <cstring>
+#include <ctime>
+#include <libintl.h>
 #include <celutil/debug.h>
 #include <celutil/util.h>
 #include "../celengine/gl.h"
--- a/src/celestia/kde/kdeapp.cpp
+++ b/src/celestia/kde/kdeapp.cpp
@@ -19,6 +19,7 @@
 #include <libintl.h>
 #include <fstream>
 #include <sstream>
+#include <libintl.h>
 
 #include <qpushbutton.h>
 #include <qslider.h>
--- a/src/celestia/kde/kdeuniquecelestia.cpp
+++ b/src/celestia/kde/kdeuniquecelestia.cpp
@@ -20,6 +20,7 @@
 #include <qdir.h>
 #include <string>
 #include <vector>
+#include <libintl.h>
 #include <klocale.h>
 #include <libintl.h>
 
--- a/src/celestia/kde/selectionpopup.cpp
+++ b/src/celestia/kde/selectionpopup.cpp
@@ -20,6 +20,7 @@
 #include <sstream>
 #include <fstream>
 #include <algorithm>
+#include <libintl.h>
 
 #include "selectionpopup.h"
 
