Author: Peter Green
Subject: fix compilation with GCC 4.5.

--- a/src/celengine/mesh.cpp
+++ b/src/celengine/mesh.cpp
@@ -7,13 +7,13 @@
 // as published by the Free Software Foundation; either version 2
 // of the License, or (at your option) any later version.
 
+#include <cassert>
+#include <iostream>
+#include <algorithm>
 #include "mesh.h"
 #include "rendcontext.h"
 #include "gl.h"
 #include "glext.h"
-#include <cassert>
-#include <iostream>
-#include <algorithm>
 
 using namespace std;
 
--- a/src/celengine/model.cpp
+++ b/src/celengine/model.cpp
@@ -7,11 +7,11 @@
 // as published by the Free Software Foundation; either version 2
 // of the License, or (at your option) any later version.
 
-#include "model.h"
-#include "rendcontext.h"
 #include <cassert>
 #include <functional>
 #include <algorithm>
+#include "model.h"
+#include "rendcontext.h"
 
 using namespace std;
 
--- a/src/celestia/celx_object.cpp
+++ b/src/celestia/celx_object.cpp
@@ -773,7 +773,7 @@
         markAlpha = 1.0f;
     
     Color markColorAlpha(0.0f, 1.0f, 0.0f, 0.9f);
-    markColorAlpha = Color::Color(markColor, markAlpha);
+    markColorAlpha = Color(markColor, markAlpha);
     
     const char* markLabel = celx.safeGetString(6, WrongType, "Fifth argument to object:mark must be a string");
     if (markLabel == NULL)
