--- a/src/celestia/kde/data/celestia.desktop
+++ b/src/celestia/kde/data/celestia.desktop
@@ -1,8 +1,12 @@
 [Desktop Entry]
-Name=Celestia
-Comment=Celestia
-Exec=celestia %f
+Encoding=UTF-8
+Version=1.0
+Name=Celestia (KDE)
+GenericName=Space Simulator
+Comment=Explore the Universe in this detailed space simulation
+Type=Application
+Categories=Astronomy;Education;Science
+Icon=celestia
+Exec=celestia-kde %f
 Terminal=false
-Icon=celestia.png
 MimeType=application/x-celestia-script
-Type=Application
--- a/src/celestia/gtk/data/celestia.desktop
+++ b/src/celestia/gtk/data/celestia.desktop
@@ -1,12 +1,13 @@
 [Desktop Entry]
 Encoding=UTF-8
-Version=1.4.0
+Version=1.0
 Type=Application
-Exec=celestia
+Exec=celestia-gnome
 Icon=celestia
 X-GNOME-DocPath=
 Terminal=false
-Name=Celestia
-Comment=Space Simulator
+Name=Celestia (GNOME)
+GenericName=Space Simulator
+Comment=Explore the Universe in this detailed space simulation
 StartupNotify=true
-Categories=Astronomy;Science;Application;
+Categories=Astronomy;Education;Application;
