Bug: https://bugzilla.gnome.org/show_bug.cgi?id=608518
Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=559291
Author: Petr Salinger <Petr.Salinger@seznam.cz>

Fix a couple of checks for the kFreeBSD kernel.

Index: libgtop2/sysdeps/freebsd/procmap.c
===================================================================
--- libgtop2.orig/sysdeps/freebsd/procmap.c	2011-07-24 21:13:56.000000000 +0200
+++ libgtop2/sysdeps/freebsd/procmap.c	2011-09-06 14:03:04.705871157 +0200
@@ -42,7 +42,7 @@
 #include <sys/pipe.h>
 #include <sys/conf.h>
 #undef _KERNEL
-#if __FreeBSD_version >= 800038
+#if (__FreeBSD_version >= 800038) || (__FreeBSD_kernel_version >= 800038)
 #define _WANT_FILE
 #include <sys/file.h>
 #undef _WANT_FILE
@@ -113,7 +113,7 @@
                 return;
         }
 
-#if __FreeBSD_version >= 800039
+#if (__FreeBSD_version >= 800039) || (__FreeBSD_kernel_version >= 800039)
         if (kvm_read (server->machine.kd, (gulong) cdev2priv(inode.i_dev), (char *) &priv,
 		      sizeof (priv))
 #else
