Fix installation paths

--- a/Makefile	2008-05-19 19:28:59.000000000 +1000
+++ b/Makefile	2012-06-05 16:27:24.000000000 +1000
@@ -1,6 +1,7 @@
 OBJECTS = x86-common.o
 CFLAGS ?= -O2 -Wall -DDEBUG -g
 LIBDIR ?= /usr/lib
+LIBEXECDIR ?= /lib
 
 ifeq ($(BACKEND),x86emu)
 	OBJECTS += thunk.o x86emu/decode.o x86emu/debug.o x86emu/fpu.o \
@@ -22,7 +23,7 @@ static: $(OBJECTS)
 	$(AR) cru libx86.a $(OBJECTS)
 
 shared: $(OBJECTS)
-	$(CC) $(CFLAGS) -o libx86.so.1 -shared -Wl,-soname,libx86.so.1 $(OBJECTS)
+	$(CC) $(CFLAGS) $(LDFLAGS) -o libx86.so.1 -shared -Wl,-soname,libx86.so.1 $(OBJECTS)
 
 objclean:
 	$(MAKE) -C x86emu clean
@@ -32,7 +33,7 @@ clean: objclean
 	rm -f *.so.1 *.a
 
 install: libx86.so.1
-	install -D libx86.so.1 $(DESTDIR)$(LIBDIR)/libx86.so.1
+	install -D libx86.so.1 $(DESTDIR)$(LIBEXECDIR)/libx86.so.1
 	install -D libx86.a $(DESTDIR)$(LIBDIR)/libx86.a
-	ln -sf libx86.so.1 $(DESTDIR)$(LIBDIR)/libx86.so
+	ln -sf $(LIBEXECDIR)/libx86.so.1 $(DESTDIR)$(LIBDIR)/libx86.so
 	install -p -m 0644 -D lrmi.h $(DESTDIR)/usr/include/libx86.h
