Description: Use format string argument with fprintf.
Author: Luca Falavigna <dktrkranz@debian.org>

Index: libvncserver/test/encodingstest.c
===================================================================
--- libvncserver.orig/test/encodingstest.c	2012-05-05 13:52:52.457034521 +0200
+++ libvncserver/test/encodingstest.c	2012-05-05 23:11:51.650769263 +0200
@@ -256,7 +256,7 @@
 
 	time(&log_clock);
 	strftime(buf, 255, "%d/%m/%Y %X (client) ", localtime(&log_clock));
-	fprintf(stderr,buf);
+	fprintf(stderr,"%s",buf);
 
 	vfprintf(stderr, format, args);
 	fflush(stderr);
