Description: Do not consider webclients directory during build phase
Author: Luca Falavigna <dktrkranz@debian.org>

Index: libvncserver/Makefile.am
===================================================================
--- libvncserver.orig/Makefile.am	2012-05-05 12:59:59.840937327 +0200
+++ libvncserver/Makefile.am	2012-05-05 13:43:43.681017709 +0200
@@ -2,8 +2,8 @@
 X11VNC=x11vnc
 endif
 
-SUBDIRS=libvncserver examples libvncclient vncterm webclients client_examples test $(X11VNC)
-DIST_SUBDIRS=libvncserver examples libvncclient vncterm webclients client_examples test
+SUBDIRS=libvncserver examples libvncclient vncterm client_examples test $(X11VNC)
+DIST_SUBDIRS=libvncserver examples libvncclient vncterm client_examples test
 EXTRA_DIST = CMakeLists.txt rfb/rfbint.h.cmake rfb/rfbconfig.h.cmake
 
 bin_SCRIPTS = libvncserver-config
Index: libvncserver/configure.ac
===================================================================
--- libvncserver.orig/configure.ac	2012-05-05 12:59:59.856937327 +0200
+++ libvncserver/configure.ac	2012-05-05 13:43:13.749016792 +0200
@@ -970,9 +970,6 @@
 	examples/Makefile
 	examples/android/Makefile
 	vncterm/Makefile
-	webclients/Makefile
-	webclients/java-applet/Makefile
-	webclients/java-applet/ssl/Makefile
 	libvncclient/Makefile
 	client_examples/Makefile
 	test/Makefile
