From: Stefano Zacchiroli <zack@debian.org>
Date: Tue, 6 Oct 2009 15:36:30 +0200
Subject: Create installation directories

...since ocamlfind doesn't

Signed-off-by: Stefano Zacchiroli <zack@debian.org>
---
 Makefile |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/Makefile b/Makefile
index ce83c7b..e47c8f8 100644
--- a/Makefile
+++ b/Makefile
@@ -46,6 +46,7 @@ doc:
 # The following PHONY rule is important for Cygwin:
 .PHONY: install
 install:
+	mkdir -p $(DESTDIR) $(OCAMLFIND_DESTDIR) $(OCAMLFIND_DESTDIR)/stublibs
 	for pkg in $(PKGLIST); do \
 		( cd src/$$pkg && $(MAKE) -f Makefile.pre install ) || exit; \
 	done
-- 
