From: Stefano Zacchiroli <zack@debian.org>
Date: Tue, 6 Oct 2009 15:32:48 +0200
Subject: Link against the PIC version of libcamlrun

Signed-off-by: Stefano Zacchiroli <zack@debian.org>
---
 src/netcgi2-apache/Makefile.def |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/netcgi2-apache/Makefile.def b/src/netcgi2-apache/Makefile.def
index f191946..7b1bb61 100644
--- a/src/netcgi2-apache/Makefile.def
+++ b/src/netcgi2-apache/Makefile.def
@@ -50,7 +50,7 @@ endif
 ### Only works with OCaml >= 3.11 on plafforms where PIC differs from non-PIC
 ### See http://caml.inria.fr/mantis/view.php?id=3866
 mod_netcgi_apache.so: $(MOD_OBJECTS)
-	$(APXS) -c -o $@ $^ -L$(APACHE_OCAMLLIBDIR) $(APACHE_OCAMLLIBS)
+	$(APXS) -c -o $@ $^ -L$(APACHE_OCAMLLIBDIR) $(patsubst -lcamlrun,-lcamlrun_shared,$(APACHE_OCAMLLIBS))
 	test -f .libs/$@ && cp .libs/$@ .
 
 netcgi_apache_mod.lo: netcgi_apache_mod.o
-- 
