--- a/configure.in-old	2012-03-02 21:48:47.000000000 +0000
+++ b/configure.in	2012-03-02 21:51:29.000000000 +0000
@@ -9671,6 +9671,17 @@
 fi
 AC_SUBST(CDEFAULTOPT)
 
+if test -x /usr/bin/dpkg-buildflags; then
+	HARDEN_CFLAGS=`dpkg-buildflags --get CFLAGS 2>/dev/null | sed -e s/-g// | sed -e s/-O[[0-3]]//g`
+	HARDEN_CXXFLAGS=`dpkg-buildflags --get CXXFLAGS 2>/dev/null | sed -e s/-g// | sed -e s/-O[[0-3]]//g`
+	HARDEN_CPPFLAGS=`dpkg-buildflags --get CPPFLAGS 2>/dev/null`
+	HARDEN_LDFLAGS=`dpkg-buildflags --get LDFLAGS 2>/dev/null`
+fi
+AC_SUBST(HARDEN_CFLAGS)
+AC_SUBST(HARDEN_CXXFLAGS)
+AC_SUBST(HARDEN_CPPFLAGS)
+AC_SUBST(HARDEN_LDFLAGS)
+
 dnl ===================================================================
 dnl Setting up the environment.
 dnl ===================================================================
--- a/config_host.mk.in-old	2012-04-06 14:17:55.164239390 +0200
+++ b/config_host.mk.in	2012-04-06 14:18:55.440239065 +0200
@@ -56,6 +56,10 @@
 export CPU=@CPU@
 export CPUNAME=@CPUNAME@
 export CDEFAULTOPT=@CDEFAULTOPT@
+export HARDEN_CFLAGS=@HARDEN_CFLAGS@
+export HARDEN_CXXFLAGS=@HARDEN_CXXFLAGS@
+export HARDEN_CPPFLAGS=@HARDEN_CPPFLAGS@
+export HARDEN_LDFLAGS=@HARDEN_LDFLAGS@
 export CPUNAME_FOR_BUILD=@CPUNAME_FOR_BUILD@
 export CPU_FOR_BUILD=@CPU_FOR_BUILD@
 export CROSS_COMPILING=@CROSS_COMPILING@
diff --git a/solenv/gbuild/LinkTarget.mk b/solenv/gbuild/LinkTarget.mk
index c094bb4..a0608f4 100644
--- a/solenv/gbuild/LinkTarget.mk
+++ b/solenv/gbuild/LinkTarget.mk
@@ -54,6 +54,8 @@
 gb_LinkTarget__get_cxxflags=$(if $(CXXFLAGS),$(CXXFLAGS),$(call gb_LinkTarget__get_debugcxxflags,$(1)))
 gb_LinkTarget__get_objcxxflags=$(if $(OBJCXXFLAGS),$(OBJCXXFLAGS),$(call gb_LinkTarget__get_debugcxxflags,$(1)))
 
+LDFLAGS += $(HARDEN_LDFLAGS)
+
 # Overview of dependencies and tasks of LinkTarget
 #
 # target                      task                         depends on
diff --git a/solenv/gbuild/gbuild.mk b/solenv/gbuild/gbuild.mk
index e85ed83..1665642 100644
--- a/solenv/gbuild/gbuild.mk
+++ b/solenv/gbuild/gbuild.mk
@@ -145,6 +145,9 @@
 gb__ENV_CXXFLAGS := $(ENVCFLAGSCXX)
 endif
 
+gb__ENV_CXXFLAGS += $(HARDEN_CXXFLAGS)
+gb__ENV_CFLAGS += $(HARDEN_CFLAGS)
+
 include $(GBUILDDIR)/Helper.mk
 include $(GBUILDDIR)/TargetLocations.mk
 
@@ -240,6 +243,8 @@ ifeq ($(HAVE_THREADSAFE_STATICS),TRUE)
 gb_GLOBALDEFS += -DHAVE_THREADSAFE_STATICS
 endif

+gb_GLOBALDEFS += $(HARDEN_CPPFLAGS)
+
 gb_GLOBALDEFS := $(sort $(gb_GLOBALDEFS))

 include $(GBUILDDIR)/Deliver.mk
diff --git a/solenv/inc/unxgcc.mk b/solenv/inc/unxgcc.mk
index a3231a6..b97f354 100644
--- a/solenv/inc/unxgcc.mk
+++ b/solenv/inc/unxgcc.mk
@@ -233,6 +233,10 @@ LINKFLAGS += -Wl,--hash-style=$(WITH_LINKER_HASH_STYLE)
 LINKFLAGS += -Wl,-zdynsort
 .ENDIF

+LINKFLAGS += $(HARDEN_LDFLAGS)
+CDEFS += $(HARDEN_CPPFLAGS)
+CFLAGS += $(HARDEN_CFLAGS)
+
 # libraries for linking applications
 STDLIBGUIMT+=-Wl,--as-needed $(DL_LIB) $(PTHREAD_LIBS) -lm -Wl,--no-as-needed
 STDLIBCUIMT+=-Wl,--as-needed $(DL_LIB) $(PTHREAD_LIBS) -lm -Wl,--no-as-needed
