--- a/configure.in-old	2012-04-07 00:05:49.607740045 +0200
+++ b/configure.in	2012-04-07 00:06:53.899999955 +0200
@@ -5135,7 +5135,8 @@
     if test "$JDK" = "gcj"; then
         save_CFLAGS=$CFLAGS
         save_LDFLAGS=$LDFLAGS
-        CFLAGS="$CFLAGS -I$JAVA_HOME/include"
+        GCJ_JNI_INCLUDE=`-I$JAVA_HOME/bin/gcj --print-file-name include/jni.h | $SED -e s,\/jni.h,,g`
+        CFLAGS="$CFLAGS -I$JAVA_HOME/include $GCJ_JNI_INCLUDE"
         LDFLAGS="$LDFLAGS -L$JAVA_HOME/lib -lgcj"
         exec 6>/dev/null # no output
         AC_CHECK_HEADER(jni.h, [],
@@ -5469,6 +5470,7 @@
 AC_SUBST(XCLASSPATH)
 AC_SUBST(x_JAVALIB)
 AC_SUBST(x_JDKLIB)
+AC_SUBST(GCJ_JNI_INCLUDE)
 
 dnl ===================================================================
 dnl Checks for specific files.
--- a/config_host.mk.in-old	2012-04-07 02:57:37.475759985 +0200
+++ b/config_host.mk.in	2012-04-07 02:58:17.976557350 +0200
@@ -430,8 +430,8 @@
 export SIZEOF_POINTER=@SIZEOF_POINTER@
 export SIZEOF_SHORT=@SIZEOF_SHORT@
 export SOLARENV=@SRC_ROOT@/solenv
-export SOLARINC=@SOLARINC@
-export SOLARINC_FOR_BUILD=@SOLARINC_FOR_BUILD@
+export SOLARINC=@SOLARINC@ @GCJ_JNI_INCLUDE@
+export SOLARINC_FOR_BUILD=@SOLARINC_FOR_BUILD@ @GCJ_JNI_INCLUDE@
 export SOLARLIB=@SOLARLIB@
 export SOLARLIB_FOR_BUILD=@SOLARLIB_FOR_BUILD@
 export SOLARVER=@SOLARVER@
