Index: logit/sfx2/source/appl/sfxhelp.cxx
===================================================================
--- logit.orig/sfx2/source/appl/sfxhelp.cxx	2011-11-22 20:57:46.000000000 +0000
+++ logit/sfx2/source/appl/sfxhelp.cxx	2011-11-24 16:03:16.084114498 +0000
@@ -104,7 +104,7 @@
 
 NoHelpErrorBox::NoHelpErrorBox( Window* _pParent ) :
 
-    ErrorBox( _pParent, WB_OK, String( SfxResId( RID_STR_HLPFILENOTEXIST ) ) )
+    ErrorBox( _pParent, WB_OK, String( SfxResId( RID_STR_HLPFILENOTEXIST ) ).Append(String(DEFINE_CONST_UNICODE(".\nPlease install the libreoffice-help-en-us package or the locale specific help package libreoffice-help-<language-code>."))) )
 {
     // Error message: "No help available"
 }
