diff --git a/scp2/source/base/module_base.scp b/scp2/source/base/module_base.scp
index 1a245a6..dc9dbf0 100644
--- a/scp2/source/base/module_base.scp
+++ b/scp2/source/base/module_base.scp
@@ -51,6 +51,20 @@ Module gid_Module_Prg_Base_Bin
         ,gid_File_Lib_Rpt
         ,gid_File_Lib_Rptui
         ,gid_File_Lib_Rptxml
+        ,gid_File_Lib_Dba
+        ,gid_File_Lib_Dbase2
+        ,gid_File_Lib_Dbaxml
+        ,gid_File_Lib_Odbcbase
+        ,gid_File_Lib_Dbpool
+        ,gid_File_Lib_Odbc
+        ,gid_File_Lib_Sdbc_2
+        ,gid_File_Lib_Mysql
+#ifdef SOLAR_JAVA
+        ,gid_File_Lib_Jdbc
+        ,gid_File_Lib_Hsqldb_2
+#endif
+        ,gid_File_Lib_Flat
+        ,gid_File_Lib_Calc
     );
 End
 
diff --git a/scp2/source/calc/module_calc.scp b/scp2/source/calc/module_calc.scp
index 0e5e790..e5ba644 100644
--- a/scp2/source/calc/module_calc.scp
+++ b/scp2/source/calc/module_calc.scp
@@ -42,7 +42,7 @@
     Minimal = YES;
     Default = YES;
     Styles = (HIDDEN_ROOT);
-    Files = (gid_File_Share_Registry_Calc_Xcd,gid_File_Lib_Calc,gid_File_Lib_Sc,gid_File_Lib_Scui,gid_File_Lib_Scfilt,gid_File_Lib_Scd,gid_File_Extra_Urldesktop_Calc,gid_File_Extra_Urlnew_Calc,gid_File_Extra_Urlstart_Calc,gid_File_Extra_Urltasks_Calc,gid_File_Oo_Office_Filter_Calc_Pkg,gid_File_Share_Config_Sofficecfg_Calc_Menubar_Xml,
+    Files = (gid_File_Share_Registry_Calc_Xcd,gid_File_Lib_Sc,gid_File_Lib_Scui,gid_File_Lib_Scfilt,gid_File_Lib_Scd,gid_File_Extra_Urldesktop_Calc,gid_File_Extra_Urlnew_Calc,gid_File_Extra_Urlstart_Calc,gid_File_Extra_Urltasks_Calc,gid_File_Oo_Office_Filter_Calc_Pkg,gid_File_Share_Config_Sofficecfg_Calc_Menubar_Xml,
     gid_File_Lib_Solver,gid_File_Lib_Lpsolve,
     gid_File_Lib_Vbaobj,
     gid_File_Tmp_Userinstall_Calc_Inf);
diff --git a/scp2/source/draw/module_draw.scp b/scp2/source/draw/module_draw.scp
index b5bec12..920fc4e 100644
--- a/scp2/source/draw/module_draw.scp
+++ b/scp2/source/draw/module_draw.scp
@@ -43,6 +43,11 @@ Module gid_Module_Prg_Draw_Bin
     Default = YES;
     Styles = (HIDDEN_ROOT);
     Files = (gid_File_Extra_Urldesktop_Draw,gid_File_Extra_Urlnew_Draw,gid_File_Extra_Urlstart_Draw,gid_File_Extra_Urltasks_Draw,gid_File_Share_Config_Sofficecfg_Draw_Menubar_Xml,
+    gid_File_Filter_Sdfilt,
+    gid_File_Lib_Sd,
+    gid_File_Lib_Sdd,
+    gid_File_Lib_Sdui,
+    gid_File_Filter_Sdfilt,
     gid_File_Share_Registry_Draw_Xcd,
     gid_File_Tmp_Userinstall_Draw_Inf);
 End
diff --git a/scp2/source/impress/module_impress.scp b/scp2/source/impress/module_impress.scp
index e3313c8..3f5e257 100644
--- a/scp2/source/impress/module_impress.scp
+++ b/scp2/source/impress/module_impress.scp
@@ -43,6 +43,7 @@ Module gid_Module_Prg_Impress_Bin
     Default = YES;
     Styles = (HIDDEN_ROOT);
     Files = (gid_File_Extra_Urldesktop_Impress,gid_File_Extra_Urlnew_Impress,gid_File_Extra_Urlstart_Impress,gid_File_Extra_Urltasks_Impress,gid_File_Lib_Placeware,
+    gid_File_Lib_Slideshow,
     gid_File_Share_Registry_Impress_Xcd,
     gid_File_Lib_Animcore,gid_File_Share_Config_Sofficecfg_Impress_Effects_Xml, gid_File_Share_Config_Sofficecfg_Impress_Transitions_Xml,
     gid_File_Tmp_Userinstall_Impress_Inf);
diff --git a/scp2/source/kde/module_kde.scp b/scp2/source/kde/module_kde.scp
index 55180c5..742cb6e 100644
--- a/scp2/source/kde/module_kde.scp
+++ b/scp2/source/kde/module_kde.scp
@@ -35,15 +35,25 @@ Module gid_Module_Optional_Kde
     MOD_NAME_DESC(MODULE_OPTIONAL_KDE);
     Styles = (DONTSHOWINUSERINSTALL);
     Files = (
+    gid_File_Bin_Kde_Open_Url,
 #ifdef ENABLE_KDE
-    gid_File_Lib_Kdebe
+    gid_File_Lib_Kdebe,
+    gid_File_Lib_Vclplug_Kde,
+    gid_File_Lib_Fps_Kde
 #ifdef ENABLE_KDE4
     ,
 #endif // ENABLE_KDE4
 #endif // ENABLE_KDE
 #ifdef ENABLE_KDE4
-    gid_File_Lib_Kde4be
+    gid_File_Lib_Kde4be,
+    gid_File_Lib_Vclplug_Kde4,
+    gid_File_Lib_Fps_Kde4
 #endif // ENABLE_KDE4
+#ifdef ENABLE_KAB
+    ,
+    gid_File_Lib_Kab_1,
+    gid_File_Lib_Kab_Drv_1
+#endif
     );
 End
 #endif
diff --git a/scp2/source/ooo/module_hidden_ooo.scp b/scp2/source/ooo/module_hidden_ooo.scp
index 024ec68..a46e972 100644
--- a/scp2/source/ooo/module_hidden_ooo.scp
+++ b/scp2/source/ooo/module_hidden_ooo.scp
@@ -84,7 +84,6 @@
 	Styles = (HIDDEN_ROOT);
 	Files = (gid_File_Bin_Cde_Open_Url,
 	gid_File_Bin_Tde_Open_Url,
-	gid_File_Bin_Kde_Open_Url,
 	gid_File_Bin_Open_Url,
 	gid_File_Bin_Pagein,
 	gid_File_Bin_Pluginapp,
@@ -177,7 +176,6 @@ Module gid_Module_Root_Files_4
 	Styles = (HIDDEN_ROOT);
 	Files = (gid_File_Dat_Root4,
 	gid_File_Filter_Eme,
-	gid_File_Filter_Sdfilt,
 	gid_File_Filter_Icg,
 	gid_File_Filter_Egi,
 	gid_File_Filter_Epb,
@@ -208,9 +206,6 @@ Module gid_Module_Root_Files_4
 	gid_File_Lib_Cui,
     gid_File_Lib_Sdbt,
     gid_File_Lib_Dbmm,
-	gid_File_Lib_Dba,
-	gid_File_Lib_Dbase2,
-	gid_File_Lib_Dbaxml,
 	gid_File_Lib_Dbt,
 	gid_File_Lib_Dbfile,
 	gid_File_Lib_Drawinglayer,
@@ -240,9 +235,6 @@ Module gid_Module_Root_Files_4
 	gid_File_Lib_Sax,
 	gid_File_Lib_Sb,
 	gid_File_Lib_Scn,
-	gid_File_Lib_Sd,
-	gid_File_Lib_Sdd,
-	gid_File_Lib_Sdui,
 	gid_File_Lib_Sfx,
 	gid_File_Lib_Sot,
 	gid_File_Lib_Spa,
@@ -275,8 +274,6 @@
 	gid_File_Lib_Zip,
 	gid_File_Lib_Vclplug_Gen,
 	gid_File_Lib_Vclplug_Tde,
-	gid_File_Lib_Vclplug_Kde,
-	gid_File_Lib_Vclplug_Kde4,
 	gid_File_Lib_Vclplug_Svp,
 	gid_File_Lib_Basctl,
 	gid_File_Lib_CanvasTools,
@@ -354,7 +354,6 @@
 	gid_File_Lib_Evtatt,
 	gid_File_Lib_Fileacc,
 	gid_File_Lib_Filterconfig1,
-	gid_File_Lib_Dbodbcbase,
 	gid_File_Lib_I18npool,
 	gid_File_Lib_I18nregexp,
 	gid_File_Lib_I18nsearch,
@@ -385,7 +374,6 @@ Module gid_Module_Root_Files_5
 	gid_File_Lib_DeploymentGui,
 	gid_File_Lib_Expwrap_Uno,
 	gid_File_Lib_Fastsax_Uno,
-	gid_File_Lib_Slideshow,
 	gid_File_Lib_Recentfile,
 	gid_File_Lib_Scriptframe,
 	gid_File_Lib_Sdbc_2,
diff --git a/scp2/source/writer/module_writer.scp b/scp2/source/writer/module_writer.scp
index e66214b..44107d8 100644
--- a/scp2/source/writer/module_writer.scp
+++ b/scp2/source/writer/module_writer.scp
@@ -45,6 +45,7 @@
     Files = (gid_File_Bin_Cde_Sagenda,gid_File_Bin_Cde_Sfax,gid_File_Bin_Cde_Slabel,
     gid_File_Bin_Cde_Sletter,gid_File_Bin_Cde_Smemo,gid_File_Bin_Cde_Svcard,
     gid_File_Lib_Hwpreader,
+    gid_File_Lib_Sw,
     gid_File_Lib_Swd,
     gid_File_Lib_Swui,
     gid_File_Lib_Msword,
