diff --git a/swext/mediawiki/src/registry/data/org/openoffice/Office/Custom/WikiExtension.xcu b/swext/mediawiki/src/registry/data/org/openoffice/Office/Custom/WikiExtension.xcu
index 6888448..c41aee6 100644
--- a/swext/mediawiki/src/registry/data/org/openoffice/Office/Custom/WikiExtension.xcu
+++ b/swext/mediawiki/src/registry/data/org/openoffice/Office/Custom/WikiExtension.xcu
@@ -27,12 +27,12 @@
  ************************************************************************ -->
 <oor:component-data xmlns:oor="http://openoffice.org/2001/registry" xmlns:xs="http://www.w3.org/2001/XMLSchema" oor:name="WikiExtension" oor:package="org.openoffice.Office.Custom">
     <node oor:name="SpecialData">
-        <node oor:name="so-wiki.germany.sun.com" oor:op="fuse">
+        <node oor:name="so-wiki.germany.sun.com" oor:op="replace">
             <prop oor:name="AllowUnknownCertificate">
                 <value>true</value>
             </prop>
             <node oor:name="AdditionalLoginArguments">
-                <node oor:name="wpDomain" oor:op="fuse">
+                <node oor:name="wpDomain" oor:op="replace">
                     <prop oor:name="Value">
                         <value>sun</value>
                     </prop>
diff --git a/swext/mediawiki/src/registry/data/org/openoffice/Office/OptionsDialog.xcu b/swext/mediawiki/src/registry/data/org/openoffice/Office/OptionsDialog.xcu
index 5ae52ce..ecc89d4 100644
--- a/swext/mediawiki/src/registry/data/org/openoffice/Office/OptionsDialog.xcu
+++ b/swext/mediawiki/src/registry/data/org/openoffice/Office/OptionsDialog.xcu
@@ -27,9 +27,9 @@
  ************************************************************************ -->
 <oor:component-data oor:name="OptionsDialog" oor:package="org.openoffice.Office" xmlns:oor="http://openoffice.org/2001/registry">
     <node oor:name="Nodes">
-        <node oor:name="Internet" oor:op="fuse">
+        <node oor:name="Internet" oor:op="replace">
             <node oor:name="Leaves">
-                <node oor:name="com.sun.star.wiki.options" oor:op="fuse">
+                <node oor:name="com.sun.star.wiki.options" oor:op="replace">
                     <prop oor:name="Id">
                         <value>com.sun.wiki-publisher</value>
                     </prop>
diff --git a/swext/mediawiki/src/registry/data/org/openoffice/Office/Paths.xcu b/swext/mediawiki/src/registry/data/org/openoffice/Office/Paths.xcu
index e39bbc7..ab10cd7 100644
--- a/swext/mediawiki/src/registry/data/org/openoffice/Office/Paths.xcu
+++ b/swext/mediawiki/src/registry/data/org/openoffice/Office/Paths.xcu
@@ -27,9 +27,9 @@
  ************************************************************************ -->
 <oor:component-data xmlns:oor="http://openoffice.org/2001/registry" xmlns:xs="http://www.w3.org/2001/XMLSchema" oor:name="Paths" oor:package="org.openoffice.Office">
     <node oor:name="Paths">
-        <node oor:name="Template" oor:op="fuse">
+        <node oor:name="Template" oor:op="replace">
             <node oor:name="InternalPaths">
-                <node oor:name="%origin%/templates" oor:op="fuse"/>
+                <node oor:name="%origin%/templates" oor:op="replace"/>
             </node>
         </node>
     </node>
diff --git a/swext/mediawiki/src/registry/data/org/openoffice/TypeDetection/Filter.xcu b/swext/mediawiki/src/registry/data/org/openoffice/TypeDetection/Filter.xcu
index d0ddad5..9c8b6b3 100644
--- a/swext/mediawiki/src/registry/data/org/openoffice/TypeDetection/Filter.xcu
+++ b/swext/mediawiki/src/registry/data/org/openoffice/TypeDetection/Filter.xcu
@@ -28,7 +28,7 @@
 <!DOCTYPE oor:component-data SYSTEM "../../../../component-update.dtd">
 <oor:component-data xmlns:oor="http://openoffice.org/2001/registry" xmlns:xs="http://www.w3.org/2001/XMLSchema" oor:package="org.openoffice.TypeDetection" oor:name="Filter">
     <node oor:name="Filters">
-        <node oor:name="MediaWiki" oor:op="fuse">
+        <node oor:name="MediaWiki" oor:op="replace">
             <prop oor:name="FileFormatVersion"><value>0</value></prop>
                 <prop oor:name="Type"><value>MediaWiki</value></prop>
             <prop oor:name="DocumentService"><value>com.sun.star.text.TextDocument</value></prop>
@@ -40,7 +40,7 @@
             </prop>
             <prop oor:name="Flags"><value>EXPORT ALIEN 3RDPARTYFILTER</value></prop>
         </node>
-        <node oor:name="MediaWiki_Web" oor:op="fuse">
+        <node oor:name="MediaWiki_Web" oor:op="replace">
             <prop oor:name="FileFormatVersion"><value>0</value></prop>
                 <prop oor:name="Type"><value>MediaWiki</value></prop>
                 <prop oor:name="DocumentService"><value>com.sun.star.text.WebDocument</value></prop>
diff --git a/swext/mediawiki/src/registry/data/org/openoffice/TypeDetection/Types.xcu b/swext/mediawiki/src/registry/data/org/openoffice/TypeDetection/Types.xcu
index 64031b8..e6739c9 100644
--- a/swext/mediawiki/src/registry/data/org/openoffice/TypeDetection/Types.xcu
+++ b/swext/mediawiki/src/registry/data/org/openoffice/TypeDetection/Types.xcu
@@ -28,7 +28,7 @@
 <!DOCTYPE oor:component-data SYSTEM "../../../../component-update.dtd">
 <oor:component-data xmlns:oor="http://openoffice.org/2001/registry" xmlns:xs="http://www.w3.org/2001/XMLSchema" oor:package="org.openoffice.TypeDetection" oor:name="Types">
     <node oor:name="Types">
-        <node oor:name="MediaWiki" oor:op="fuse">
+        <node oor:name="MediaWiki" oor:op="replace">
             <prop oor:name="DetectService"><value>com.sun.star.comp.filters.XMLFilterDetect</value></prop>
             <prop oor:name="URLPattern"/>
             <prop oor:name="Extensions"><value>txt</value></prop>
