From 6b1d19a3242353d84afeec1f0a798a95c851459d Mon Sep 17 00:00:00 2001
From: Bjoern Michaelsen <bjoern.michaelsen@canonical.com>
Date: Tue, 21 Feb 2012 13:43:04 +0100
Subject: [PATCH] pythonloader.uno.so, libpyuno and pyuno are duplicates for
 linkoo too

---
 solenv/bin/linkoo |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/solenv/bin/linkoo b/solenv/bin/linkoo
index 06a103d..450da5a 100755
--- a/solenv/bin/linkoo
+++ b/solenv/bin/linkoo
@@ -137,7 +137,7 @@
 
 my @search_dirs = ( 'lib', 'bin', 'class' );
 
-my @known_duplicates = ( 'db.jar', 'libi18n', 'libnssckbi', 'libnssdbm', 'libsqlite3', 'libnssutil3', 'pythonloader.uno', 'pyuno', 'libpyuno' );
+my @known_duplicates = ( 'db.jar', 'libi18n', 'libnssckbi', 'libnssdbm', 'libsqlite3', 'libnssutil3', 'pythonloader.uno', 'pyuno', 'libpyuno', 'pythonloader.uno', 'pyuno', 'libpyuno' );
 
 sub sniff_target($)
 {
