diff --git a/dbaccess/source/ui/misc/dbumiscres.src b/dbaccess/source/ui/misc/dbumiscres.src
index 974b36e..2f42c27 100644
--- a/dbaccess/source/ui/misc/dbumiscres.src
+++ b/dbaccess/source/ui/misc/dbumiscres.src
@@ -96,5 +96,5 @@ String STR_NAMED_OBJECT_ALREADY_EXISTS
 String RID_STR_EXTENSION_NOT_PRESENT
 {
     // #i96130# use hard coded name
-    Text [ en-US ] = "The report, \"$file$\", requires the extension Oracle Report Builder.";
+    Text [ en-US ] = "The report, \"$file$\", requires the extension Oracle Report Builder. For this, install the package 'libreoffice-report-builder'.";
 };
