Index: logit/shell/source/unix/misc/gnome-open-url.sh
===================================================================
--- logit.orig/shell/source/unix/misc/gnome-open-url.sh	2011-11-22 20:57:46.000000000 +0000
+++ logit/shell/source/unix/misc/gnome-open-url.sh	2011-11-24 16:47:51.199972037 +0000
@@ -1,6 +1,6 @@
 #!/bin/sh
 
 # use xdg-open or gnome-open if available
-xdg-open "$1" 2>/dev/null || gnome-open "$1" 2>/dev/null || "$0.bin" $1
+xdg-open "$1" 2>/dev/null || gnome-open "$1" 2>/dev/null || sensible-browser "$1" || "$0.bin" $1
 
 exit 0
Index: logit/shell/source/unix/misc/kde-open-url.sh
===================================================================
--- logit.orig/shell/source/unix/misc/kde-open-url.sh	2011-11-22 20:57:46.000000000 +0000
+++ logit/shell/source/unix/misc/kde-open-url.sh	2011-11-24 16:47:51.199972037 +0000
@@ -2,9 +2,13 @@
 
 # special handling for mailto: uris
 if echo $1 | grep '^mailto:' > /dev/null; then
-  kmailservice "$1" &
+  if which kde-open; then
+    kde-open "$1" &
+  else
+    mailservice "$1" &
+  fi
 else
-  kfmclient openURL "$1" &
+  sensible-browser "$1" &
 fi
 
 exit 0
