Index: libreoffice-3.6.2~rc1/officecfg/registry/data/org/openoffice/Office/Common.xcu
===================================================================
--- libreoffice-3.6.2~rc1.orig/officecfg/registry/data/org/openoffice/Office/Common.xcu	2012-10-07 21:03:17.662850386 -0400
+++ libreoffice-3.6.2~rc1/officecfg/registry/data/org/openoffice/Office/Common.xcu	2012-10-07 21:03:57.174850591 -0400
@@ -665,8 +665,7 @@
   </node>
   <node oor:name="ExternalMailer" install:module="unixdesktop">
     <prop oor:name="Program">
-      <value oor:external=
-          "com.sun.star.configuration.backend.DesktopBackend ExternalMailer"/>
+      <value>sensible-lomua</value>
     </prop>
   </node>
   <node oor:name="Font">
Index: libreoffice-3.6.2~rc1/shell/source/unix/misc/senddoc.sh
===================================================================
--- libreoffice-3.6.2~rc1.orig/shell/source/unix/misc/senddoc.sh	2012-10-07 21:03:17.662850386 -0400
+++ libreoffice-3.6.2~rc1/shell/source/unix/misc/senddoc.sh	2012-10-07 21:29:13.074858447 -0400
@@ -59,6 +59,30 @@
     shift
 fi
 
+case `basename "$MAILER"` in
+    sensible-lomua)
+        if [ -n "$KDE_FULL_SESSION" -a -x /usr/bin/kde-open ] \
+           || [ -x /usr/bin/gnome-open ] \
+           || [ -x /usr/bin/xdg-open ]; then
+            # use an undefined mailer, to trigger the default handling
+            MAILER=undefined
+        elif [ -n "$GNOME_DESKTOP_SESSION_ID" -a -x /usr/bin/evolution ]; then
+            MAILER=/usr/bin/evolution
+        elif [ -n "$KDE_FULL_SESSION" -a -x /usr/bin/kmail ]; then
+            MAILER=/usr/bin/kmail
+        elif [ -x /usr/bin/evolution ]; then
+            # default
+            MAILER=/usr/bin/evolution
+        elif [ -x /usr/bin/icedove ]; then
+            # fallback
+            MAILER=/usr/bin/icedove
+        elif [ -x /usr/bin/thunderbird ]; then
+            # fallback
+            MAILER=/usr/bin/thunderbird
+        fi
+        ;;
+esac
+
 # autodetect mail client from executable name
 case `basename "$MAILER" | sed 's/-.*$//'` in
 
