--- libreoffice-3.5.0/postprocess/packcomponents/makefile.mk-old	2011-06-13 23:35:26.000000000 +0000
+++ libreoffice-3.5.0/postprocess/packcomponents/makefile.mk	2011-06-13 23:35:58.000000000 +0000
@@ -269,12 +265,6 @@
     component/wizards/com/sun/star/wizards/web/web
 .END
 
-.IF "$(WITH_BINFILTER)" != "NO"
-my_components += \
-    bf_migratefilter \
-    bindet
-.END
-
 .IF "$(WITH_LDAP)" == "YES"
 my_components += component/extensions/source/config/ldap/ldapbe2
 .END
@@ -394,10 +394,6 @@
 my_components += component/shell/source/cmdmail/cmdmail
 .END
 
-.IF "$(OS)" != "WNT" && "$(ENABLE_EVOAB2)" != ""
-my_components += component/connectivity/source/drivers/evoab2/evoab
-.END
-
 .IF "$(OS)" != "WNT" && "$(ENABLE_GSTREAMER)" != ""
 my_components += component/avmedia/source/gstreamer/avmediagstreamer
 .END
@@ -394,3 +404,50 @@
         '<list><filename>component/scripting/java/ScriptProviderForJavaScript.component</filename></list>' \
         > $@
 .END
+
+.IF "$(WITH_BINFILTER)" == "YES"
+ALLTAR: $(MISC)/bf_migratefilter.rdb $(MISC)/bindet.rdb
+
+$(MISC)/bf_migratefilter.rdb .ERRREMOVE : \
+        $(SOLARENV)/bin/packcomponents.xslt \
+        $(MISC)/bf_migratefilter.input \
+        $(SOLARXMLDIR)/bf_migratefilter.component
+    $(XSLTPROC) --nonet --stringparam prefix $(SOLARXMLDIR)/ -o $@ \
+        $(SOLARENV)/bin/packcomponents.xslt \
+        $(MISC)/bf_migratefilter.input
+
+$(MISC)/bindet.rdb .ERRREMOVE : \
+        $(SOLARENV)/bin/packcomponents.xslt \
+        $(MISC)/bindet.input \
+        $(SOLARXMLDIR)/bindet.component
+    $(XSLTPROC) --nonet --stringparam prefix $(SOLARXMLDIR)/ -o $@ \
+        $(SOLARENV)/bin/packcomponents.xslt \
+        $(MISC)/bindet.input
+
+$(MISC)/bf_migratefilter.input : makefile.mk
+    echo \
+        '<list><filename>bf_migratefilter.component</filename></list>' \
+        > $@
+
+$(MISC)/bindet.input : makefile.mk
+    echo \
+        '<list><filename>bindet.component</filename></list>' \
+        > $@
+.END
+
+.IF "$(ENABLE_EVOAB2)" == "YES"
+ALLTAR : $(MISC)/evoab.rdb
+
+$(MISC)/evoab.rdb .ERRREMOVE : \
+        $(SOLARENV)/bin/packcomponents.xslt \
+        $(MISC)/evoab.input \
+        $(SOLARXMLDIR)/evoab.component
+    $(XSLTPROC) --nonet --stringparam prefix $(SOLARXMLDIR)/ -o $@ \
+        $(SOLARENV)/bin/packcomponents.xslt \
+        $(MISC)/evoab.input
+
+$(MISC)/evoab.input : makefile.mk
+    echo \
+        '<list><filename>evoab.component</filename></list>' \
+        > $@
+.END
--- libreoffice-3.5.0/postprocess/packregistry/makefile.mk-old	2011-12-20 04:49:53.000000000 +0100
+++ libreoffice-3.5.0/postprocess/packregistry/makefile.mk	2011-12-20 04:52:13.000000000 +0100
@@ -40,6 +40,7 @@
     $(MISC)/ctl.xcd \
     $(MISC)/ctlseqcheck.xcd \
     $(MISC)/draw.xcd \
+    $(MISC)/evoab.xcd \
     $(MISC)/graphicfilter.xcd \
     $(MISC)/impress.xcd \
     $(MISC)/korea.xcd \
@@ -329,8 +329,8 @@
 ERROR : unknown-GUIBASE
 .END
 .IF "$(ENABLE_EVOAB2)" == "TRUE"
-MY_FILES_main += $(MY_MOD)/org/openoffice/Office/DataAccess/Drivers-evoab2.xcu
-MY_FILES_main += $(MY_MOD)/org/openoffice/Office/DataAccess-evoab2.xcu
+MY_FILES_evoab += $(MY_MOD)/org/openoffice/Office/DataAccess/Drivers-evoab2.xcu
+MY_FILES_evoab += $(MY_MOD)/org/openoffice/Office/DataAccess-evoab2.xcu
 MY_DRIVERS += evoab
 .END
 .IF "$(SOLAR_JAVA)" == "TRUE"
--- libreoffice-3.5.0/postprocess/prj/d.lst-old   2011-12-20 11:39:36.000000000 +0000
+++ libreoffice-3.5.0/postprocess/prj/d.lst       2011-12-20 11:40:07.000000000 +0000
@@ -4,4 +4,7 @@
 ..\%__SRC%\misc\lang\*.xcd %_DEST%\xml\
 ..\%__SRC%\misc\scriptproviderforbeanshell.rdb %_DEST%\xml\scriptproviderforbeanshell.rdb
 ..\%__SRC%\misc\scriptproviderforjavascript.rdb %_DEST%\xml\scriptproviderforjavascript.rdb
+..\%__SRC%\misc\bf_migratefilter.rdb %_DEST%\xml\bf_migratefilter.rdb
+..\%__SRC%\misc\bindet.rdb %_DEST%\xml\bindet.rdb
+..\%__SRC%\misc\evoab.rdb %_DEST%\xml\evoab.rdb
 ..\%__SRC%\misc\services.rdb %_DEST%\xml\services.rdb
--- libreoffice-3.5.0/scp2/source/ooo/file_library_ooo.scp-old	2011-12-20 05:11:44.000000000 +0100
+++ libreoffice-3.5.0/scp2/source/ooo/file_library_ooo.scp	2011-12-20 05:12:07.000000000 +0100
@@ -446,6 +446,13 @@
 
 STD_LIB_FILE( gid_File_Lib_Evoab , evoab)
 
+File gid_File_Share_Registry_Evoab_Xcd
+    TXT_FILE_BODY;
+    Styles = (PACKED);
+    Dir = gid_Brand_Dir_Share_Registry;
+    Name = "evoab.xcd";
+End
+
 #endif  // #ifdef ENABLE_EVOAB2
 
 #endif  // #ifdef UNX
--- libreoffice-3.5.0/scp2/source/gnome/module_gnome.scp-old	2011-12-20 05:15:16.000000000 +0100
+++ libreoffice-3.5.0/scp2/source/gnome/module_gnome.scp	2011-12-20 05:15:30.000000000 +0100
@@ -43,6 +43,7 @@ Module gid_Module_Optional_Gnome
     Files = (
 #ifdef ENABLE_EVOAB2
       gid_File_Lib_Evoab,
+      gid_File_Share_Registry_Evoab_Xcd,
 #endif
 #ifdef ENABLE_GTK
       gid_File_Lib_Fps_Gnome,
diff -u binfilter-old/file_binfilter.scp binfilter/file_binfilter.scp
--- libreoffice-3.5.0/scp2/source/binfilter/file_binfilter.scp	2011-12-06 14:02:07.000000000 +0000
+++ libreoffice-3.5.0/scp2/source/binfilter/file_binfilter.scp	2011-12-20 10:16:46.000000000 +0000
@@ -42,7 +42,19 @@
     Styles = (PACKED);
 End
 
+File gid_File_Rdb_Bf_Migratefilter
+    TXT_FILE_BODY;
+    Name = "bf_migratefilter.rdb";
+    Dir = gid_Brand_Dir_Program_Services;
+    Styles = (PACKED);
+End
 
+File gid_File_Rdb_Bindet
+    TXT_FILE_BODY;
+    Name = "bindet.rdb";
+    Dir = gid_Brand_Dir_Program_Services;
+    Styles = (PACKED);
+End
 
 STD_LIB_FILE( gid_File_Lib_Bf_Go, bf_go)
 STD_LIB_FILE( gid_File_Lib_Bf_Sch, bf_sch)
diff -u binfilter-old/module_binfilter.scp binfilter/module_binfilter.scp
--- libreoffice-3.5.0/scp2/source/binfilter/module_binfilter.scp	2011-12-06 14:02:07.000000000 +0000
+++ libreoffice-3.5.0/scp2/source/binfilter/module_binfilter.scp	2011-12-20 10:17:56.000000000 +0000
@@ -53,7 +53,9 @@
     gid_File_Lib_Bf_Wrapper,
     gid_File_Lib_Legacy_Binfilters,
     gid_File_Lib_Bf_Migratefilter,
+    gid_File_Rdb_Bf_Migratefilter,
     gid_File_Lib_Bindet,
+    gid_File_Rdb_Bindet,
     gid_File_Res_Bf_Sch,
     gid_File_Res_Bf_Svx,
     gid_File_Res_Bf_Svt,
