diff --git a/ure/source/jvmfwk3rc b/ure/source/jvmfwk3rc
index d1f79c5..c002f0e 100644
--- a/ure/source/jvmfwk3rc
+++ b/ure/source/jvmfwk3rc
@@ -1,5 +1,5 @@
 [Bootstrap]
 UNO_JAVA_JFW_VENDOR_SETTINGS=${ORIGIN}/../share/misc/javavendors.xml
-UNO_JAVA_JFW_SHARED_DATA=${URE_OVERRIDE_JAVA_JFW_SHARED_DATA} file:///etc/opt/ure/javasettings_${_OS}_${_ARCH}.xml
+UNO_JAVA_JFW_SHARED_DATA=${URE_OVERRIDE_JAVA_JFW_SHARED_DATA}
 UNO_JAVA_JFW_USER_DATA=${URE_OVERRIDE_JAVA_JFW_USER_DATA} ${SYSUSERHOME}/.ure/javasettings_${_OS}_${_ARCH}.xml
 UNO_JAVA_JFW_CLASSPATH_URLS=${URE_MORE_JAVA_CLASSPATH_URLS}
diff --git a/ure/source/unorc b/ure/source/unorc
index 7936432..577669c 100644
--- a/ure/source/unorc
+++ b/ure/source/unorc
@@ -19,5 +19,5 @@
 URE_INTERNAL_LIB_DIR=${ORIGIN}
 URE_INTERNAL_JAVA_DIR=${ORIGIN}/../share/java
 URE_INTERNAL_JAVA_CLASSPATH=${URE_MORE_JAVA_TYPES}
-UNO_TYPES=${ORIGIN}/../share/misc/types.rdb ?file:///etc/opt/ure/types.rdb ?${SYSUSERHOME}/.ure/types.rdb ${URE_MORE_TYPES}
-UNO_SERVICES=${ORIGIN}/../share/misc/services.rdb ?file:///etc/opt/ure/services.rdb ?${SYSUSERHOME}/.ure/services.rdb ${URE_MORE_SERVICES}
+UNO_TYPES=${ORIGIN}/../share/misc/types.rdb ?${SYSUSERHOME}/.ure/types.rdb ${URE_MORE_TYPES}
+UNO_SERVICES=${ORIGIN}/../share/misc/services.rdb ?${SYSUSERHOME}/.ure/services.rdb ${URE_MORE_SERVICES}
