Subject: Fix bashism
From: Michal Čihař <nijel@debian.org>
Forwarded: https://sourceforge.net/tracker/?func=detail&aid=2999814&group_id=222125&atid=1054682
--- a/plugins/rotate/geeqie-rotate
+++ b/plugins/rotate/geeqie-rotate
@@ -8,7 +8,7 @@
 rotate()
 {
     ext=`echo "${1##*.}" |tr "[:upper:]" "[:lower:]"`
-    [ "x$ext" == "x" ] && return 1 #no extension
+    [ "x$ext" = "x" ] && return 1 #no extension
 
     gq_metadata="$GQ_METADATA_DIR/$1.gq.xmp"
     if [ -f "$gq_metadata" ]; then
@@ -48,7 +48,7 @@
 }
 
 get_sidecars=
-if [ "x$1" == "x-g" ] ; then
+if [ "x$1" = "x-g" ] ; then
     get_sidecars=yes
     shift
 fi
