From 902157eeea55c4fe7e626977f8d2b161352b63dc Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Michal=20=C4=8Ciha=C5=99?= <mcihar@suse.cz>
Date: Tue, 21 Aug 2012 14:53:05 +0200
Subject: [PATCH] Fix big endian support in tiff loader

---
 src/image_load_tiff.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/image_load_tiff.c b/src/image_load_tiff.c
index d32899c..71a33b8 100644
--- a/src/image_load_tiff.c
+++ b/src/image_load_tiff.c
@@ -326,7 +326,7 @@ static gboolean image_loader_tiff_load (gpointer loader, const guchar *buf, gsiz
 		/* Turns out that the packing used by TIFFRGBAImage depends on 
         	 * the host byte order... 
 	         */ 
-		while (pixels < pixbuf->pixels + bytes) 
+		while (pixels < lt->pixbuf->pixels + bytes) 
 			{
 			uint32 pixel = *(uint32 *)pixels;
 			int r = TIFFGetR(pixel);
@@ -419,4 +419,4 @@ void image_loader_backend_set_tiff(ImageLoaderBackend *funcs)
 
 
 
-#endif
\ No newline at end of file
+#endif
-- 
1.7.10.4

