--- a/docs/Docs.mk
+++ b/docs/Docs.mk
@@ -1,7 +1,5 @@
-PS2PDF		:= ps2pdf
-DVIPS		:= dvips
-LATEX		:= latex
-FIG2DEV		:= fig2dev
+PDFLATEX	:= pdflatex
+EPSTOPDF	:= epstopdf
 LATEX2HTML	:= latex2html
 DOXYGEN		:= doxygen
 POD2MAN		:= pod2man
--- a/docs/xen-api/Makefile
+++ b/docs/xen-api/Makefile
@@ -6,35 +6,30 @@
 
 
 TEX := $(wildcard *.tex)
-EPS := $(wildcard *.eps)
-EPSDOT := $(patsubst %.dot,%.eps,$(wildcard *.dot))
+PDFEPS := $(patsubst %.eps,%.pdf,$(wildcard *.eps))
+PDFDOT := $(patsubst %.dot,%.pdf,$(wildcard *.dot))
 
 .PHONY: all
 all: build
 
 .PHONY: build
-build: xenapi.pdf xenapi.ps
+build: xenapi.pdf
 
 install:
-	$(INSTALL_DIR) $(DESTDIR)$(DOCDIR)/ps
-	$(INSTALL_DIR) $(DESTDIR)$(DOCDIR)/pdf
+	$(INSTALL_DIR) $(DESTDIR)$(DOCDIR)
 
-	[ -e xenapi.ps ] && cp xenapi.ps $(DESTDIR)$(DOCDIR)/ps || true
-	[ -e xenapi.pdf ] && cp xenapi.pdf $(DESTDIR)$(DOCDIR)/pdf || true
+	[ -e xenapi.pdf ] && cp xenapi.pdf $(DESTDIR)$(DOCDIR) || true
 
-xenapi.dvi: $(TEX) $(EPS) $(EPSDOT)
-	$(LATEX) xenapi.tex
-	$(LATEX) xenapi.tex
+xenapi.pdf: $(TEX) $(PDFEPS) $(PDFDOT)
+	$(PDFLATEX) xenapi.tex
+	$(PDFLATEX) xenapi.tex
 	rm -f *.aux *.log
 
-%.pdf: %.ps
-	$(PS2PDF) $< $@
+%.pdf: %.eps
+	$(EPSTOPDF) $<
 
-%.ps: %.dvi
-	$(DVIPS) $< -o $@
-
-%.eps: %.dot
-	$(DOT) -Tps $< >$@
+%.pdf: %.dot
+	$(DOT) -Tpdf $< >$@
 
 xenapi-datamodel-graph.eps: xenapi-datamodel-graph.dot
 	$(NEATO) -Goverlap=false -Tps $< >$@
--- a/docs/xen-api/xenapi-coversheet.tex
+++ b/docs/xen-api/xenapi-coversheet.tex
@@ -14,7 +14,7 @@
 %% Document title
 \newcommand{\doctitle}{Xen Management API}
 
-\newcommand{\coversheetlogo}{xen.eps}
+\newcommand{\coversheetlogo}{xen}
 
 %% Document date
 \newcommand{\datestring}{10th January 2010}
