--- a/tools/firmware/Makefile
+++ b/tools/firmware/Makefile
@@ -7,7 +7,6 @@
 
 SUBDIRS-y :=
 SUBDIRS-$(CONFIG_OVMF) += ovmf
-SUBDIRS-$(CONFIG_SEABIOS) += seabios-dir
 SUBDIRS-$(CONFIG_ROMBIOS) += rombios
 SUBDIRS-$(CONFIG_ROMBIOS) += vgabios
 SUBDIRS-y += hvmloader
@@ -48,9 +47,6 @@
 subdir-distclean-ovmf: .phony
 	rm -rf ovmf ovmf-remote
 
-subdir-distclean-seabios-dir: .phony
-	rm -rf seabios-dir seabios-dir-remote
-
 .PHONY: ovmf-find
 ovmf-find:
 	if test -d $(OVMF_UPSTREAM_URL) ; then \
--- a/tools/firmware/hvmloader/Makefile
+++ b/tools/firmware/hvmloader/Makefile
@@ -39,7 +39,7 @@
 
 OVMF_DIR := ../ovmf
 ROMBIOS_DIR := ../rombios
-SEABIOS_DIR := ../seabios-dir
+SEABIOS_DIR := /usr/share/seabios
 
 ifeq ($(CONFIG_ROMBIOS),y)
 STDVGA_ROM    := ../vgabios/VGABIOS-lgpl-latest.bin
@@ -70,7 +70,7 @@
 ifeq ($(CONFIG_SEABIOS),y)
 OBJS += seabios.o
 CFLAGS += -DENABLE_SEABIOS
-SEABIOS_ROM := $(SEABIOS_DIR)/out/bios.bin
+SEABIOS_ROM := $(SEABIOS_DIR)/bios.bin
 ROMS += $(SEABIOS_ROM)
 endif
 
