--- a/tools/examples/xend-config.sxp
+++ b/tools/examples/xend-config.sxp
@@ -132,6 +132,11 @@
 #(console-limit 1024)
 
 ##
+# NOTE:
+# Please read /usr/share/doc/xen-utils-common/README.Debian for Debian specific
+# informations about the network setup.
+
+##
 # To bridge network traffic, like this:
 #
 # dom0: ----------------- bridge -> real eth0 -> the network
@@ -170,7 +175,6 @@
 # two fake interfaces per guest domain.  To do things like this, write
 # yourself a wrapper script, and call network-bridge from it, as appropriate.
 #
-(network-script network-bridge)
 
 # The script used to control virtual interfaces.  This can be overridden on a
 # per-vif basis when creating a domain or a configuring a new vif.  The
