--- a/tools/hotplug/common/Makefile
+++ b/tools/hotplug/common/Makefile
@@ -6,8 +6,8 @@
 # OS-independent hotplug scripts go in this directory
 
 # Xen scripts to go there.
-XEN_SCRIPTS = $(HOTPLUGPATH)
-XEN_SCRIPT_DATA =
+XEN_SCRIPTS =
+XEN_SCRIPT_DATA = $(HOTPLUGPATH)
 
 genpath-target = $(call buildmakevars2file,$(HOTPLUGPATH))
 $(eval $(genpath-target))
--- a/tools/hotplug/Linux/vtpm-impl
+++ b/tools/hotplug/Linux/vtpm-impl
@@ -1,4 +1,3 @@
-#!/bin/bash
 # ===================================================================
 # 
 # Copyright (c) 2005, Intel Corp.
