--- a/tools/examples/xend-config.sxp
+++ b/tools/examples/xend-config.sxp
@@ -62,7 +62,6 @@
 #(xend-tcp-xmlrpc-server no)
 #(xend-unix-xmlrpc-server yes)
 #(xend-relocation-server no)
-(xend-relocation-server yes)
 #(xend-relocation-ssl-server no)
 #(xend-udev-event-server no)
 
@@ -126,7 +125,6 @@
 #  (xend-relocation-hosts-allow '^localhost$ ^.*\\.example\\.org$')
 #
 #(xend-relocation-hosts-allow '')
-(xend-relocation-hosts-allow '^localhost$ ^localhost\\.localdomain$')
 
 # The limit (in kilobytes) on the size of the console buffer
 #(console-limit 1024)
