--- a/tools/flask/libflask/Makefile
+++ b/tools/flask/libflask/Makefile
@@ -29,9 +29,10 @@
 
 .PHONY: install
 install: build
+	$(INSTALL_DIR) $(DESTDIR)$(PRIVATE_LIBDIR)
 	$(INSTALL_DIR) $(DESTDIR)$(LIBDIR)
 	$(INSTALL_DIR) $(DESTDIR)$(INCLUDEDIR)
-	$(INSTALL_PROG) libflask.so $(DESTDIR)$(LIBDIR)
+	$(INSTALL_PROG) libflask.so $(DESTDIR)$(PRIVATE_LIBDIR)
 	$(INSTALL_DATA) libflask.a $(DESTDIR)$(LIBDIR)
 	$(INSTALL_DATA) include/libflask.h $(DESTDIR)$(INCLUDEDIR)/xen/xsm
 
