--- a/tools/misc/Makefile
+++ b/tools/misc/Makefile
@@ -42,12 +42,8 @@
 
 .PHONY: install
 install: build
-	$(INSTALL_DIR) $(DESTDIR)$(BINDIR)
-	$(INSTALL_DIR) $(DESTDIR)$(SBINDIR)
 	$(INSTALL_DIR) $(DESTDIR)$(PRIVATE_BINDIR)
-	$(INSTALL_PYTHON_PROG) $(INSTALL_BIN) $(DESTDIR)$(BINDIR)
-	$(INSTALL_PYTHON_PROG) $(INSTALL_SBIN) $(DESTDIR)$(SBINDIR)
-	$(INSTALL_PYTHON_PROG) $(INSTALL_PRIVBIN) $(DESTDIR)$(PRIVATE_BINDIR)
+	$(INSTALL_PYTHON_PROG) $(INSTALL_BIN) $(INSTALL_SBIN) $(INSTALL_PRIVBIN) $(DESTDIR)$(PRIVATE_BINDIR)
 	set -e; for d in $(SUBDIRS); do $(MAKE) -C $$d install-recurse; done
 
 .PHONY: clean
--- a/tools/misc/lomount/Makefile
+++ b/tools/misc/lomount/Makefile
@@ -11,7 +11,7 @@
 
 .PHONY: install
 install install-recurse: build
-	$(INSTALL_PROG) lomount $(SCRIPTS) $(DESTDIR)$(BINDIR)
+	$(INSTALL_PROG) lomount $(SCRIPTS) $(DESTDIR)$(PRIVATE_BINDIR)
 
 .PHONY: clean
 clean:
@@ -20,4 +20,4 @@
 lomount: lomount.o
 	$(CC) $(CFLAGS) -o $@ $< 
 
--include $(DEPS)
\ No newline at end of file
+-include $(DEPS)
--- a/tools/misc/xend
+++ b/tools/misc/xend
@@ -33,6 +33,8 @@
 import time
 import commands
 
+sys.path.insert(1, sys.path[0] + '/../lib/python')
+
 from xen.xend.server import SrvDaemon
 
 class CheckError(ValueError):
--- a/tools/misc/xm
+++ b/tools/misc/xm
@@ -1,6 +1,7 @@
 #!/usr/bin/env python
 #  -*- mode: python; -*-
 import sys
+sys.path.insert(1, sys.path[0] + '/../lib/python')
 
 from xen.xm import main
 
