--- a/tools/python/xen/util/bugtool.py
+++ b/tools/python/xen/util/bugtool.py
@@ -1,5 +1,3 @@
-#!/usr/bin/env python
-
 # This library is free software; you can redistribute it and/or
 # modify it under the terms of version 2.1 of the GNU Lesser General Public
 # License as published by the Free Software Foundation.
--- a/tools/python/xen/util/pci.py
+++ b/tools/python/xen/util/pci.py
@@ -1,5 +1,3 @@
-#!/usr/bin/env python
-#
 # PCI Device Information Class
 # - Helps obtain information about which I/O resources a PCI device needs
 #
--- a/tools/python/xen/util/vscsi_util.py
+++ b/tools/python/xen/util/vscsi_util.py
@@ -1,4 +1,3 @@
-#!/usr/bin/env python
 #  -*- mode: python; -*-
 
 #============================================================================
--- a/tools/python/xen/xend/XendBase.py
+++ b/tools/python/xen/xend/XendBase.py
@@ -1,4 +1,3 @@
-#!/usr/bin/python
 #============================================================================
 # This library is free software; you can redistribute it and/or
 # modify it under the terms of version 2.1 of the GNU Lesser General Public
--- a/tools/python/xen/xend/XendClient.py
+++ b/tools/python/xen/xend/XendClient.py
@@ -1,4 +1,3 @@
-#!/usr/bin/env python
 #============================================================================
 # This library is free software; you can redistribute it and/or
 # modify it under the terms of version 2.1 of the GNU Lesser General Public
--- a/tools/python/xen/xend/XendLocalStorageRepo.py
+++ b/tools/python/xen/xend/XendLocalStorageRepo.py
@@ -1,4 +1,3 @@
-#!/usr/bin/python
 #============================================================================
 # This library is free software; you can redistribute it and/or
 # modify it under the terms of version 2.1 of the GNU Lesser General Public
--- a/tools/python/xen/xend/XendQCoWStorageRepo.py
+++ b/tools/python/xen/xend/XendQCoWStorageRepo.py
@@ -1,4 +1,3 @@
-#!/usr/bin/python
 #============================================================================
 # This library is free software; you can redistribute it and/or
 # modify it under the terms of version 2.1 of the GNU Lesser General Public
--- a/tools/python/xen/xend/XendStorageRepository.py
+++ b/tools/python/xen/xend/XendStorageRepository.py
@@ -1,4 +1,3 @@
-#!/usr/bin/python
 #============================================================================
 # This library is free software; you can redistribute it and/or
 # modify it under the terms of version 2.1 of the GNU Lesser General Public
--- a/tools/python/xen/xend/XendVDI.py
+++ b/tools/python/xen/xend/XendVDI.py
@@ -1,4 +1,3 @@
-#!/usr/bin/python
 #============================================================================
 # This library is free software; you can redistribute it and/or
 # modify it under the terms of version 2.1 of the GNU Lesser General Public
--- a/tools/python/xen/xend/arch.py
+++ b/tools/python/xen/xend/arch.py
@@ -1,5 +1,3 @@
-#!/usr/bin/env python
-#
 # This library is free software; you can redistribute it and/or
 # modify it under the terms of version 2.1 of the GNU Lesser General Public
 # License as published by the Free Software Foundation.
--- a/tools/python/xen/xend/osdep.py
+++ b/tools/python/xen/xend/osdep.py
@@ -1,5 +1,3 @@
-#!/usr/bin/env python
-#
 # This library is free software; you can redistribute it and/or
 # modify it under the terms of version 2.1 of the GNU Lesser General Public
 # License as published by the Free Software Foundation.
--- a/tools/python/xen/xend/sxp.py
+++ b/tools/python/xen/xend/sxp.py
@@ -1,4 +1,3 @@
-#!/usr/bin/env python
 #============================================================================
 # This library is free software; you can redistribute it and/or
 # modify it under the terms of version 2.1 of the GNU Lesser General Public
--- a/tools/python/xen/xm/xenapi_create.py
+++ b/tools/python/xen/xm/xenapi_create.py
@@ -1,4 +1,3 @@
-#!/usr/bin/python
 #============================================================================
 # This library is free software; you can redistribute it and/or
 # modify it under the terms of version 2.1 of the GNU Lesser General Public
--- a/tools/python/xen/remus/save.py
+++ b/tools/python/xen/remus/save.py
@@ -1,5 +1,3 @@
-#!/usr/bin/env python
-
 import os, select, socket, threading, time, signal, xmlrpclib
 
 from xen.xend.XendClient import server
--- a/tools/python/xen/remus/vm.py
+++ b/tools/python/xen/remus/vm.py
@@ -1,5 +1,3 @@
-#!/usr/bin/env python
-
 import xmlrpclib
 
 from xen.xend.XendClient import server
--- a/tools/python/xen/xend/XendSXPDev.py
+++ b/tools/python/xen/xend/XendSXPDev.py
@@ -1,5 +1,3 @@
-#!/usr/bin/env python
-#
 # Helper functions for dealing with the sxp representation of devices
 
 import types
