--- a/tools/Makefile
+++ b/tools/Makefile
@@ -1,10 +1,6 @@
 XEN_ROOT = $(CURDIR)/..
 include $(XEN_ROOT)/tools/Rules.mk
 
-ifneq ($(CONFIG_SYSTEM_LIBAIO),y)
-SUBDIRS-libaio := libaio
-endif
-
 SUBDIRS-y :=
 SUBDIRS-y += include
 SUBDIRS-y += libxc
@@ -23,7 +19,6 @@
 SUBDIRS-y += xenstat
 SUBDIRS-$(CONFIG_Linux) += $(SUBDIRS-libaio)
 SUBDIRS-$(CONFIG_Linux) += memshr 
-SUBDIRS-$(CONFIG_Linux) += blktap
 SUBDIRS-$(CONFIG_Linux) += blktap2
 SUBDIRS-$(CONFIG_NetBSD) += $(SUBDIRS-libaio)
 SUBDIRS-$(CONFIG_NetBSD) += blktap2
@@ -34,9 +29,12 @@
 
 # do not recurse in to a dir we are about to delete
 ifneq "$(MAKECMDGOALS)" "distclean"
+CONFIG_QEMU = $(CURDIR)/../qemu
+ifneq ($(wildcard $(CONFIG_QEMU)),)
 SUBDIRS-$(CONFIG_IOEMU) += qemu-xen-traditional-dir
 SUBDIRS-$(CONFIG_IOEMU) += qemu-xen-dir
 endif
+endif
 
 SUBDIRS-y += xenpmd
 SUBDIRS-y += libxl
