diff -NurbBp libgcrypt-1.4.5.orig/src/libgcrypt-config.in libgcrypt-1.4.5/src/libgcrypt-config.in
--- libgcrypt-1.4.5.orig/src/libgcrypt-config.in	2009-04-02 11:25:32.000000000 +0200
+++ libgcrypt-1.4.5/src/libgcrypt-config.in	2009-12-12 09:37:05.000000000 +0100
@@ -151,7 +151,8 @@ if test "$echo_libs" = "yes"; then
     fi
 
     # Set up `libs_final'.
-    libs_final="$libs_final $gpg_error_libs"
+    #libs_final="$libs_final $gpg_error_libs"
+    libs_final="-lgcrypt"
 
     tmp=""
     for i in $libdirs $libs_final; do
