Description: Disable dead code that is causing compilation errors.
Author: Jakub Wilk <jwilk@debian.org>
Bug-Debian: http://bugs.debian.org/583281
Last-Update: 2011-02-02

--- a/lib/ImageIterator2.hh
+++ b/lib/ImageIterator2.hh
@@ -42,12 +42,14 @@
       return a;
     }
 
+#if 0
     accu& abs() {
       v[0] = std::abs(v[0]);
       v[1] = std::abs(v[1]);
       v[2] = std::abs(v[2]);
       return *this;
     }
+#endif
     
     void saturate () {
       v[0] = std::min (std::max (v[0], (vtype)0), (vtype)0xff);
@@ -194,6 +196,7 @@
       return a;
     }
     
+#if 0
     accu& abs() {
       v[0] = std::abs(v[0]);
       v[1] = std::abs(v[1]);
@@ -201,6 +204,7 @@
       v[3] = std::abs(v[3]);
       return *this;
     }
+#endif
 
     void saturate () {
       v[0] = std::min (std::max (v[0], (vtype)0), (vtype)0xff);
@@ -357,12 +361,14 @@
       return a;
     }
     
+#if 0
     accu& abs() {
       v[0] = std::abs(v[0]);
       v[1] = std::abs(v[1]);
       v[2] = std::abs(v[2]);
       return *this;
     }
+#endif
 
     void saturate () {
       v[0] = std::min (std::max (v[0], (vtype)0), (vtype)0xffff);
@@ -509,10 +515,12 @@
       return a;
     }
 
+#if 0
     accu& abs() {
       v[0] = std::abs(v[0]);
       return *this;
     }
+#endif
 
     void saturate () {
       v[0] = std::min (std::max (v[0], (vtype)0), (vtype)0xff);
@@ -638,10 +646,12 @@
       return a;
     }
     
+#if 0
     accu& abs() {
       v[0] = std::abs(v[0]);
       return *this;
     }
+#endif
 
     void saturate () {
       v[0] = std::min (std::max (v[0], (vtype)0), (vtype)0xffff);
