Description: Fix denoise crash on 64-bit systems
Author: René Rebe <rene@exactcode.de>

---
diff --git a/lib/Colorspace.cc b/lib/Colorspace.cc
index ee517092ea6d93e90df8257133fda56bc6025b98..0b7e7cbefdb76503570e5761eb7e099ae7d75717 100644
--- a/lib/Colorspace.cc
+++ b/lib/Colorspace.cc
@@ -198,7 +198,7 @@ void colorspace_gray8_denoise_neighbours (Image &image, bool gross)
   struct compare_and_set
   {
     const Image& image;
-    const unsigned int stride;
+    const long stride;
     compare_and_set (const Image& _image)
       : image(_image), stride (image.stride())
     {
