Index: flexdock-1.0/src/java/demo/org/flexdock/demos/AllDemos.java
===================================================================
--- flexdock-1.0.orig/src/java/demo/org/flexdock/demos/AllDemos.java	2011-11-24 16:13:54.000000000 +0100
+++ flexdock-1.0/src/java/demo/org/flexdock/demos/AllDemos.java	2011-12-09 11:56:35.840465806 +0100
@@ -33,7 +33,6 @@
         "org.flexdock.demos.raw.border.BorderDemo",
         "org.flexdock.demos.raw.CompoundDemo",
         "org.flexdock.demos.raw.elegant.ElegantDemo",
-        "org.flexdock.demos.raw.jmf.JMFDemo",
         "org.flexdock.demos.raw.SimpleDemo",
         "org.flexdock.demos.raw.SplitPaneDemo",
         "org.flexdock.demos.raw.TabbedPaneDemo",
