From: Debian Java Maintainers <pkg-java-maintainers@lists.alioth.debian.org>
Date: Thu, 11 Mar 2010 21:50:06 +0100
Subject: [PATCH] add Xerces from Debian's path

---
 src/script/ant |    7 +++++++
 1 files changed, 7 insertions(+), 0 deletions(-)

diff --git a/src/script/ant b/src/script/ant
index 0dc84e0..7415eeb 100644
--- a/src/script/ant
+++ b/src/script/ant
@@ -155,6 +155,13 @@ if [ ! -x "$JAVACMD" ] ; then
   exit 1
 fi
 
+# Add the Xerces 2 XML parser in the Debian version
+if [ -z "$LOCALCLASSPATH" ] ; then
+  LOCALCLASSPATH="/usr/share/java/xmlParserAPIs.jar:/usr/share/java/xercesImpl.jar"
+else
+  LOCALCLASSPATH="/usr/share/java/xmlParserAPIs.jar:/usr/share/java/xercesImpl.jar:$LOCALCLASSPATH"
+fi
+
 # Build local classpath using just the launcher in non-rpm mode or
 # use the Jpackage helper in rpm mode with basic and default jars
 # specified in the ant.conf configuration. Because the launcher is
-- 
