From: Debian Java Maintainers <pkg-java-maintainers@lists.alioth.debian.org>
Date: Thu, 11 Mar 2010 21:50:12 +0100
Subject: [PATCH] fix $ANT_HOME path

---
 src/script/ant |   22 +++-------------------
 1 files changed, 3 insertions(+), 19 deletions(-)

diff --git a/src/script/ant b/src/script/ant
index 7415eeb..f2ba9de 100644
--- a/src/script/ant
+++ b/src/script/ant
@@ -92,25 +92,9 @@ case "`uname`" in
 esac
 
 if [ -z "$ANT_HOME" -o ! -d "$ANT_HOME" ] ; then
-  ## resolve links - $0 may be a link to ant's home
-  PRG="$0"
-  progname=`basename "$0"`
-
-  # need this for relative symlinks
-  while [ -h "$PRG" ] ; do
-    ls=`ls -ld "$PRG"`
-    link=`expr "$ls" : '.*-> \(.*\)$'`
-    if expr "$link" : '/.*' > /dev/null; then
-    PRG="$link"
-    else
-    PRG=`dirname "$PRG"`"/$link"
-    fi
-  done
-
-  ANT_HOME=`dirname "$PRG"`/..
-
-  # make it fully qualified
-  ANT_HOME=`cd "$ANT_HOME" > /dev/null && pwd`
+  if [ -d /usr/share/ant ] ; then
+     ANT_HOME=/usr/share/ant
+  fi
 fi
 
 # For Cygwin and Mingw, ensure paths are in UNIX format before
-- 
