---
 wscript |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- slv2.orig/wscript
+++ slv2/wscript
@@ -48,7 +48,7 @@ def set_options(opt):
 def configure(conf):
 	autowaf.configure(conf)
 	autowaf.check_tool(conf, 'compiler_cc')
-	autowaf.check_pkg(conf, 'lv2core', uselib_store='LV2CORE', atleast_version='1.0', mandatory=True)
+	autowaf.check_pkg(conf, 'lv2', uselib_store='LV2', atleast_version='1.0', mandatory=True)
 	autowaf.check_pkg(conf, 'redland', uselib_store='REDLAND', atleast_version='1.0.6', mandatory=True)
 	autowaf.check_pkg(conf, 'jack', uselib_store='JACK', atleast_version='0.107.0', mandatory=False)
 	conf.env.append_value('CCFLAGS', '-std=c99')
@@ -97,7 +97,7 @@ def build(bld):
 	obj.lib          = 'raptor2'
 	obj.vnum         = SLV2_LIB_VERSION
 	obj.install_path = '${LIBDIR}'
-	autowaf.use_lib(bld, obj, 'REDLAND LV2CORE')
+	autowaf.use_lib(bld, obj, 'REDLAND LV2')
 
 	# Static library (for unit test code coverage)
 	if bld.env['BUILD_TESTS']:
