diff -ur systemd-197/rules/99-systemd.rules.in s596/rules/99-systemd.rules.in
--- systemd-197/rules/99-systemd.rules.in	2012-10-11 11:47:13.134673514 -0400
+++ s596/rules/99-systemd.rules.in	2013-02-07 02:55:17.415682011 -0500
@@ -49,7 +49,7 @@
 
 # Apply sysctl variables to network devices (and only to those) as they appear.
 
-SUBSYSTEM=="net", KERNEL!="lo", RUN+="@rootlibexecdir@/systemd-sysctl --prefix=/proc/sys/net/ipv4/conf/$name --prefix=/proc/sys/net/ipv4/neigh/$name --prefix=/proc/sys/net/ipv6/conf/$name --prefix=/proc/sys/net/ipv6/neigh/$name"
+SUBSYSTEM=="net", KERNEL!="lo", RUN+="@rootlibexecdir@/systemd-sysctl --prefix=/proc/sys/net/ipv4/conf/$name --prefix=/proc/sys/net/ipv4/neigh/$name --prefix=/proc/sys/net/ipv6/conf/$name --prefix=/proc/sys/net/ipv6/neigh/$name", ENV{SYSTEMD_WANTS}="ifup@$name.service"
 
 # Asynchronously mount file systems implemented by these modules as
 # soon as they are loaded.
