Author: Daniel Baumann <daniel.baumann@progress-technologies.net>
Description: Print information about the CSS README.

diff -Naurp libdvdread.orig/src/dvd_input.c libdvdread/src/dvd_input.c
--- libdvdread.orig/src/dvd_input.c	2011-01-06 13:08:49.000000000 +0000
+++ libdvdread/src/dvd_input.c	2011-01-06 13:10:44.000000000 +0000
@@ -351,7 +351,14 @@ int dvdinput_setup(void)
     return 1;
 
   } else {
-    fprintf(stderr, "libdvdread: Encrypted DVD support unavailable.\n");
+    fprintf(stderr, "libdvdread: Encrypted DVD support unavailable.\n"
+	    "************************************************\n"
+	    "**                                            **\n"
+	    "**  No css library available. See             **\n"
+	    "**  /usr/share/doc/libdvdread4/README.css     **\n"
+	    "**  for more information.                     **\n"
+	    "**                                            **\n"
+	    "************************************************\n");
 
     /* libdvdcss replacement functions */
     dvdinput_open  = file_open;
