Author: Daniel Baumann <daniel.baumann@progress-technologies.net>
Description:
 Fixing symbol conflicts with other MD5 implementations (Closes: #522512).

diff -Naurp libdvdread.orig/src/md5.h libdvdread/src/md5.h
--- libdvdread.orig/src/md5.h	2011-01-06 13:08:49.000000000 +0000
+++ libdvdread/src/md5.h	2011-01-06 13:12:25.000000000 +0000
@@ -21,6 +21,15 @@
 #ifndef LIBDVDREAD_MD5_H
 #define LIBDVDREAD_MD5_H
 
+/* Change function names to prevent symbol conflicts */
+#define md5_init_ctx _DVD_md5_init_ctx
+#define md5_process_block _DVD_md5_process_block
+#define md5_process_bytes _DVD_md5_process_bytes
+#define md5_finish_ctx _DVD_md5_finish_ctx
+#define md5_read_ctx _DVD_md5_read_ctx
+#define md5_stream _DVD_md5_stream
+#define md5_buffer _DVD_md5_buffer
+
 #include <stdio.h>
 
 #if defined HAVE_LIMITS_H || _LIBC
