Author: Lorenz Minder <lminder@gmx.net>
Description: Use correct endianess on big-endian machines (Closes: #531621).

diff -Naurp libdvdread.orig/src/bswap.h libdvdread/src/bswap.h
--- libdvdread.orig/src/bswap.h	2011-01-06 13:08:49.000000000 +0000
+++ libdvdread/src/bswap.h	2011-01-06 13:14:17.000000000 +0000
@@ -22,6 +22,11 @@
 #ifndef LIBDVDREAD_BSWAP_H
 #define LIBDVDREAD_BSWAP_H
 
+#include <endian.h>
+#if BYTE_ORDER == BIG_ENDIAN
+#define WORDS_BIGENDIAN
+#endif
+
 #include <config.h>
 
 #if defined(WORDS_BIGENDIAN)
