Author: Daniel Baumann <daniel.baumann@progress-technologies.net>
Description: Bumping version to match original upstream (Closes: #532438).

diff -Naurp libdvdread.orig/src/dvdread/dvd_reader.h libdvdread/src/dvdread/dvd_reader.h
--- libdvdread.orig/src/dvdread/dvd_reader.h	2011-01-06 14:03:02.398951217 +0100
+++ libdvdread/src/dvdread/dvd_reader.h	2011-01-06 14:07:36.202950962 +0100
@@ -43,7 +43,7 @@
 /**
  * The current version.
  */
-#define DVDREAD_VERSION 904
+#define DVDREAD_VERSION 907
 
 /**
  * The length of one Logical Block of a DVD.
