Author: Peter Van Eynde <pvaneynd@debian.org>
Description: Fix miscompilation leading to a segfault (Closes: #688574).

diff -Naurp libdvdread.orig/src/dvd_reader.c libdvdread/src/dvd_reader.c
--- libdvdread.orig/src/dvd_reader.c	2012-09-24 16:55:49.681245164 +0200
+++ libdvdread/src/dvd_reader.c	2012-09-24 16:56:06.845356530 +0200
@@ -30,6 +30,7 @@
 #include <string.h>
 #include <ctype.h>
 #define _GNU_SOURCE
+#define __USE_GNU
 #include <unistd.h>
 #include <limits.h>
 #include <dirent.h>
