Fix for ImplicitPointerConversions in /demos/programs/TabStack/main.c

--- openmotif-2.3.0/demos/programs/TabStack/main.c.orig	2010-06-28 11:04:52.077200853 -0600
+++ openmotif-2.3.0/demos/programs/TabStack/main.c	2010-06-28 11:07:01.068047399 -0600
@@ -46,6 +46,8 @@ extern void MENU_POST(Widget, XtPointer,
 extern Pixmap XPM_PIXMAP(Widget, char**);
 extern void SET_BACKGROUND_COLOR(Widget, ArgList, Cardinal*, Pixel);
 
+Widget Createform(Widget parent);
+
 /* Begin user code block <globals> */
 /* End user code block <globals> */
 
@@ -110,7 +112,7 @@ int main( int argc, char **argv)
         parent,
         args, 
         ac);
-    form = (Widget)Createform(topLevelShell);
+    form = Createform(topLevelShell);
     XtManageChild(form);
     XtPopup(XtParent(form), XtGrabNone);
     
