0002-add-debian-menu-structure-to-system.mwmrc-menu.patch
===================================================================
--- openmotif-2.3.0.orig/clients/mwm/system.mwmrc	2010-06-24 13:32:07.000000000 +0200
+++ openmotif-2.3.0/clients/mwm/system.mwmrc	2010-06-24 13:32:26.000000000 +0200
@@ -154,3 +154,7 @@
 !!
 !!  END OF mwm RESOURCE DESCRIPTION FILE
 !!
+
+! debian menu
+!include-menu-defs
+include-menu-defs
