Index: webkitgtk+-1.4.2/Source/WebKit/gtk/webkit/webkitwebview.cpp
===================================================================
--- webkitgtk+-1.4.2.orig/Source/WebKit/gtk/webkit/webkitwebview.cpp	2011-09-08 15:58:40.196491146 +0200
+++ webkitgtk+-1.4.2/Source/WebKit/gtk/webkit/webkitwebview.cpp	2011-09-08 16:43:29.337637688 +0200
@@ -933,6 +933,12 @@ static gboolean webkit_web_view_scroll_e
     if (!frame->view())
         return FALSE;
 
+#ifndef GTK_API_VERSION_2
+    /* Ignore Alt-scroll events in GTK3 so that GtkNotebook can use it to switch tabs */
+    if (event->state & GDK_MOD1_MASK)
+        return FALSE;
+#endif
+
     PlatformWheelEvent wheelEvent(event);
     return frame->eventHandler()->handleWheelEvent(wheelEvent);
 }
