Encoding in desktop file is deprecated.
Index: gconf-cleaner/src/gconf-cleaner.desktop.in.in
===================================================================
--- gconf-cleaner.orig/src/gconf-cleaner.desktop.in.in	2009-03-29 17:27:53.000000000 -0700
+++ gconf-cleaner/src/gconf-cleaner.desktop.in.in	2009-03-29 17:41:56.000000000 -0700
@@ -1,9 +1,8 @@
 [Desktop Entry]
-Encoding=UTF-8
 _Name=GConf Cleaner
 _Comment=A Cleaning tool for GConf
 Exec=gconf-cleaner
-Icon=
+Icon=gconf-cleaner.png
 Terminal=false
 Type=Application
 StartupNotify=true
