Author: Ludovico Cavedon <cavedon@debian.org>
Bug-Debian: http://bugs.debian.org/643380
Index: gconf-cleaner/src/main.c
===================================================================
--- gconf-cleaner.orig/src/main.c	2011-09-28 00:32:04.000000000 -0700
+++ gconf-cleaner/src/main.c	2011-09-28 00:43:55.906028777 -0700
@@ -81,9 +81,11 @@
 							       GTK_DIALOG_MODAL,
 							       GTK_MESSAGE_ERROR,
 							       GTK_BUTTONS_OK,
+							       "%s",
 							       primary_text);
 
 	gtk_message_dialog_format_secondary_text(GTK_MESSAGE_DIALOG (dialog),
+						 "%s",
 						 secondary_text);
 	g_signal_connect(dialog, "response",
 			 G_CALLBACK (gtk_widget_destroy), NULL);
@@ -120,10 +122,12 @@
 							       GTK_DIALOG_MODAL,
 							       GTK_MESSAGE_WARNING,
 							       GTK_BUTTONS_YES_NO,
+							       "%s",
 							       primary_text);
 	gboolean retval;
 
 	gtk_message_dialog_format_secondary_text(GTK_MESSAGE_DIALOG (dialog),
+						 "%s",
 						 secondary_text);
 	g_signal_connect(dialog, "response",
 			 G_CALLBACK (_gconf_cleaner_question_dialog_on_response), NULL);
