Description: Disable Unity for Debian
Author: Kartik Mistry <kartik@debian.org>
Last-Update: 2012-07-23

--- gwibber-3.5.2.orig/Makefile.am
+++ gwibber-3.5.2/Makefile.am
@@ -1,10 +1,5 @@
 SUBDIRS = man bin gwibber data libgwibber libgwibber-gtk vapi po client examples
 
-if HAVE_UNITY
-SUBDIRS += lens
-endif
-
-
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = \
 	gwibber-@GWIBBER_API_VERSION@.pc \
@@ -19,7 +14,7 @@ check-report:
 update-pot:
 	$(MAKE) -C po "$(GETTEXT_PACKAGE).pot"
 
-DISTCHECK_CONFIGURE_FLAGS = --enable-localinstall --enable-introspection --enable-maintainer-flags=no --enable-unity
+DISTCHECK_CONFIGURE_FLAGS = --enable-localinstall --enable-introspection --enable-maintainer-flags=no
 
 DIST_SUBDIRS = ${SUBDIRS} tests
 
--- gwibber-3.5.2.orig/configure.ac
+++ gwibber-3.5.2/configure.ac
@@ -199,29 +199,6 @@ AC_ARG_ENABLE(localinstall,
 
 AM_CONDITIONAL([HAVE_LOCALINSTALL], [test "x$with_localinstall" = "xyes"])
 
-with_unity="yes"
-AC_ARG_ENABLE(unity,
-              AS_HELP_STRING([--enable-unity],
-                             [Enable building of the Unity lens]),
-              with_unity=$enableval,
-              with_unity=yes)
-
-AM_CONDITIONAL([HAVE_UNITY], [test "x$with_unity" = "xyes"])
-
-if test "x$with_unity" = "xyes"; then
-	PKG_CHECK_MODULES(UNITY,
-                  dbusmenu-glib-0.4 >= $DBUSMENU_REQUIRED
-                  unity >= $UNITY_REQUIRED)
-	AC_SUBST(UNITY_CFLAGS)
-	AC_SUBST(UNITY_LIBS)
-	if test "x$with_localinstall" = "xyes"; then
-        	LENSESDIR="${datadir}/unity/lenses"
-	else
-        	LENSESDIR=`$PKG_CONFIG --variable=lensesdir unity`
-	fi
-	AC_SUBST(LENSESDIR)
-fi
-
 dnl Create the Makefiles
 AC_CONFIG_FILES([
   Makefile
