diff -ur gwibber_3.6.0/configure.ac g/configure.ac
--- gwibber_3.6.0/configure.ac	2012-10-02 20:09:08.000000000 -0400
+++ g/configure.ac	2012-12-31 19:49:47.126265655 -0500
@@ -138,7 +138,7 @@
                   gdk-3.0 >= $GDK_REQUIRED
                   gio-2.0 
                   gio-unix-2.0 >= $GLIB_REQUIRED
-                  gee-1.0
+                  gee-0.8
                   libaccounts-glib
                   libnotify >= $LIBNOTIFY_REQUIRED
                   libsignon-glib
diff -ur gwibber-3.6.0/client/Makefile.am h/client/Makefile.am
--- gwibber-3.6.0/client/Makefile.am	2012-12-31 19:53:26.296261702 -0500
+++ h/client/Makefile.am	2012-12-31 19:51:38.822930306 -0500
@@ -38,7 +38,7 @@
 	--pkg accounts \
 	--pkg config \
         --pkg json-glib-1.0 \
-        --pkg gee-1.0 \
+        --pkg gee-0.8 \
         --pkg gtk+-3.0 \
 	--pkg gdk-3.0 \
         --pkg atk \
@@ -103,7 +103,7 @@
 	--pkg gdk-3.0 \
         --pkg atk \
         --pkg glib-2.0 \
-        --pkg gee-1.0 \
+        --pkg gee-0.8 \
         --pkg dee-1.0 \
         --pkg pango \
         --basedir ./ \
diff -ur gwibber-3.6.0/client/Makefile.in h/client/Makefile.in
--- gwibber-3.6.0/client/Makefile.in	2012-12-31 19:53:17.672928525 -0500
+++ h/client/Makefile.in	2012-12-31 19:51:38.822930306 -0500
@@ -382,7 +375,7 @@
 	--pkg accounts \
 	--pkg config \
         --pkg json-glib-1.0 \
-        --pkg gee-1.0 \
+        --pkg gee-0.8 \
         --pkg gtk+-3.0 \
 	--pkg gdk-3.0 \
         --pkg atk \
@@ -447,7 +440,7 @@
 	--pkg gdk-3.0 \
         --pkg atk \
         --pkg glib-2.0 \
-        --pkg gee-1.0 \
+        --pkg gee-0.8 \
         --pkg dee-1.0 \
         --pkg pango \
         --basedir ./ \
diff -ur gwibber-3.6.0/examples/Makefile.am h/examples/Makefile.am
--- gwibber-3.6.0/examples/Makefile.am	2012-12-31 19:53:26.239595037 -0500
+++ h/examples/Makefile.am	2012-12-31 19:51:38.812930308 -0500
@@ -17,7 +17,7 @@
 	--pkg accounts \
 	--pkg dbus-glib-1 \
         --pkg dee-1.0 \
-        --pkg gee-1.0 \
+        --pkg gee-0.8 \
 	--basedir ./
 
 example_vala_LDADD = \
@@ -45,7 +45,7 @@
         --pkg json-glib-1.0 \
         --pkg gtk+-3.0 \
         --pkg dee-1.0 \
-        --pkg gee-1.0 \
+        --pkg gee-0.8 \
 	--pkg pango \
         --basedir ./
 
diff -ur gwibber-3.6.0/examples/Makefile.in h/examples/Makefile.in
--- gwibber-3.6.0/examples/Makefile.in	2012-12-31 19:53:17.526261861 -0500
+++ h/examples/Makefile.in	2012-12-31 19:51:38.812930308 -0500
@@ -367,7 +360,7 @@
 	--pkg accounts \
 	--pkg dbus-glib-1 \
         --pkg dee-1.0 \
-        --pkg gee-1.0 \
+        --pkg gee-0.8 \
 	--basedir ./
 
 example_vala_LDADD = \
@@ -395,7 +388,7 @@
         --pkg json-glib-1.0 \
         --pkg gtk+-3.0 \
         --pkg dee-1.0 \
-        --pkg gee-1.0 \
+        --pkg gee-0.8 \
 	--pkg pango \
         --basedir ./
 
diff -ur gwibber-3.6.0/lens/src/Makefile.am h/lens/src/Makefile.am
--- gwibber-3.6.0/lens/src/Makefile.am	2012-12-31 19:53:26.302928369 -0500
+++ h/lens/src/Makefile.am	2012-12-31 19:51:38.829596973 -0500
@@ -32,7 +32,7 @@
   --pkg dee-1.0 \
   --pkg unity \
   --pkg gtk+-3.0 \
-  --pkg gee-1.0 \
+  --pkg gee-0.8 \
   --pkg gio-2.0 \
   --pkg gio-unix-2.0 \
   --pkg json-glib-1.0 \
diff -ur gwibber-3.6.0/lens/src/Makefile.in h/lens/src/Makefile.in
--- gwibber-3.6.0/lens/src/Makefile.in	2012-12-31 19:53:17.702928524 -0500
+++ h/lens/src/Makefile.in	2012-12-31 19:51:38.829596973 -0500
@@ -343,7 +336,7 @@
   --pkg dee-1.0 \
   --pkg unity \
   --pkg gtk+-3.0 \
-  --pkg gee-1.0 \
+  --pkg gee-0.8 \
   --pkg gio-2.0 \
   --pkg gio-unix-2.0 \
   --pkg json-glib-1.0 \
diff -ur gwibber-3.6.0/libgwibber/gwibber-0.3.deps h/libgwibber/gwibber-0.3.deps
--- gwibber-3.6.0/libgwibber/gwibber-0.3.deps	2012-12-31 19:53:29.582928309 -0500
+++ h/libgwibber/gwibber-0.3.deps	2012-12-31 19:51:38.829596973 -0500
@@ -2,6 +2,6 @@
 gobject-2.0
 glib-2.0
 dee-1.0
-gee-1.0
+gee-0.8
 json-glib-1.0
 gio-2.0
--- gwibber-3.6.0/libgwibber/Makefile.am	2012-12-31 19:53:26.306261702 -0500
+++ h/libgwibber/Makefile.am	2012-12-31 19:51:38.829596973 -0500
@@ -38,7 +38,7 @@
 	--pkg config \
 	--pkg gobject-2.0 \
 	--pkg dee-1.0 \
-	--pkg gee-1.0 \
+	--pkg gee-0.8 \
 	--pkg json-glib-1.0 \
 	--pkg gio-2.0 \
 	$(MAINTAINER_VALAFLAGS) \
diff -ur gwibber-3.6.0/libgwibber/Makefile.in h/libgwibber/Makefile.in
--- gwibber-3.6.0/libgwibber/Makefile.in	2012-12-31 19:53:17.712928524 -0500
+++ h/libgwibber/Makefile.in	2012-12-31 19:51:38.829596973 -0500
@@ -399,7 +392,7 @@
 	--pkg config \
 	--pkg gobject-2.0 \
 	--pkg dee-1.0 \
-	--pkg gee-1.0 \
+	--pkg gee-0.8 \
 	--pkg json-glib-1.0 \
 	--pkg gio-2.0 \
 	$(MAINTAINER_VALAFLAGS) \
diff -ur gwibber-3.6.0/libgwibber-gtk/Makefile.am h/libgwibber-gtk/Makefile.am
--- gwibber-3.6.0/libgwibber-gtk/Makefile.am	2012-12-31 19:53:26.309595035 -0500
+++ h/libgwibber-gtk/Makefile.am	2012-12-31 19:51:38.829596973 -0500
@@ -48,7 +48,7 @@
 	--pkg atk \
 	--pkg pangocairo \
 	--pkg dee-1.0 \
-	--pkg gee-1.0 \
+	--pkg gee-0.8 \
 	--pkg libsoup-2.4 \
 	--pkg libnotify \
 	--pkg json-glib-1.0 \
diff -ur gwibber-3.6.0/libgwibber-gtk/Makefile.in h/libgwibber-gtk/Makefile.in
--- gwibber-3.6.0/libgwibber-gtk/Makefile.in	2012-12-31 19:53:17.719595190 -0500
+++ h/libgwibber-gtk/Makefile.in	2012-12-31 19:51:38.829596973 -0500
@@ -406,7 +399,7 @@
 	--pkg atk \
 	--pkg pangocairo \
 	--pkg dee-1.0 \
-	--pkg gee-1.0 \
+	--pkg gee-0.8 \
 	--pkg libsoup-2.4 \
 	--pkg libnotify \
 	--pkg json-glib-1.0 \
Only in h: libtool
diff -ur gwibber-3.6.0/tests/vala/Makefile.am h/tests/vala/Makefile.am
--- gwibber-3.6.0/tests/vala/Makefile.am	2012-12-31 19:53:26.352928368 -0500
+++ h/tests/vala/Makefile.am	2012-12-31 19:51:38.832930307 -0500
@@ -27,7 +27,7 @@
 	--pkg gwibber-0.3 \
 	--pkg gobject-2.0 \
 	--pkg dee-1.0 \
-	--pkg gee-1.0 \
+	--pkg gee-0.8 \
 	--pkg json-glib-1.0 \
 	--pkg gio-2.0 \
 	--pkg gtk+-3.0 \
diff -ur gwibber-3.6.0/tests/vala/Makefile.in h/tests/vala/Makefile.in
--- gwibber-3.6.0/tests/vala/Makefile.in	2012-12-31 19:53:17.832928521 -0500
+++ h/tests/vala/Makefile.in	2012-12-31 19:51:38.832930307 -0500
@@ -343,7 +336,7 @@
 	--pkg gwibber-0.3 \
 	--pkg gobject-2.0 \
 	--pkg dee-1.0 \
-	--pkg gee-1.0 \
+	--pkg gee-0.8 \
 	--pkg json-glib-1.0 \
 	--pkg gio-2.0 \
 	--pkg gtk+-3.0 \
