index 39e9bf2..1b774bb 100644
--- a/plugins/avxffms2/src/avssources_avx.cpp
+++ b/plugins/avxffms2/src/avssources_avx.cpp
@@ -42,7 +42,7 @@ AvisynthVideoSource::AvisynthVideoSource(const char *SourceFile, int Track, FFMS
        if (!V)
                Env->ThrowError("FFVideoSource: %s", E.Buffer);
 
-   if (PP && FFMS_SetPP(V, PP, &E)) {
+ if (PP) {
                FFMS_DestroyVideoSource(V);
                Env->ThrowError("FFVideoSource: %s", E.Buffer);
        }
