Index: calibre-0.8.60+dfsg/recipes/economist_free.recipe
===================================================================
--- calibre-0.8.60+dfsg.orig/recipes/economist_free.recipe	2012-07-20 06:26:28.000000000 +0200
+++ calibre-0.8.60+dfsg/recipes/economist_free.recipe	2012-07-21 17:53:59.096926047 +0200
@@ -8,9 +8,13 @@
 from calibre.web.feeds.news import BasicNewsRecipe
 from calibre.ebooks.BeautifulSoup import Tag, NavigableString
 from collections import OrderedDict
+from feedparser import parse
+from functools import partial
 
 import time, re
 
+parse = partial(parse, agent='Mozilla/5.0 (Windows; U; Windows NT 6.0; en-US; rv:1.9.2.11) Gecko/20101012 Firefox/3.6.11')
+
 class Economist(BasicNewsRecipe):
 
     title = 'The Economist'
@@ -185,7 +189,6 @@
         lambda x:'</html>')]
 
     def parse_index(self):
-        from calibre.web.feeds.feedparser import parse
         if self.test:
             self.oldest_article = 14.0
         raw = self.index_to_soup(
Index: calibre-0.8.60+dfsg/recipes/lenta_ru.recipe
===================================================================
--- calibre-0.8.60+dfsg.orig/recipes/lenta_ru.recipe	2012-07-20 06:26:28.000000000 +0200
+++ calibre-0.8.60+dfsg/recipes/lenta_ru.recipe	2012-07-21 17:53:59.100926047 +0200
@@ -4,11 +4,14 @@
 Lenta.ru
 '''
 
-from calibre.web.feeds.feedparser import parse
 from calibre.ebooks.BeautifulSoup import Tag
 from calibre.web.feeds.news import BasicNewsRecipe
+from feedparser import parse
+from functools import partial
 import re
 
+parse = partial(parse, agent='Mozilla/5.0 (Windows; U; Windows NT 6.0; en-US; rv:1.9.2.11) Gecko/20101012 Firefox/3.6.11')
+
 class LentaRURecipe(BasicNewsRecipe):
     title = u'Lenta.ru: \u041d\u043e\u0432\u043e\u0441\u0442\u0438'
     __author__ = 'Nikolai Kotchetkov'
Index: calibre-0.8.60+dfsg/src/calibre/web/feeds/__init__.py
===================================================================
--- calibre-0.8.60+dfsg.orig/src/calibre/web/feeds/__init__.py	2012-07-20 06:26:28.000000000 +0200
+++ calibre-0.8.60+dfsg/src/calibre/web/feeds/__init__.py	2012-07-21 17:53:59.100926047 +0200
@@ -11,6 +11,10 @@
 from calibre import entity_to_unicode, strftime
 from calibre.utils.date import dt_factory, utcnow, local_tz
 from calibre.utils.cleantext import clean_ascii_chars
+from feedparser import parse
+from functools import partial
+
+parse = partial(parse, agent='Mozilla/5.0 (Windows; U; Windows NT 6.0; en-US; rv:1.9.2.11) Gecko/20101012 Firefox/3.6.11')
 
 class Article(object):
 
@@ -318,7 +322,6 @@
                   max_articles_per_feed=100,
                   get_article_url=lambda item: item.get('link', None),
                   log=default_log):
-    from calibre.web.feeds.feedparser import parse
     # Handle unclosed escaped entities. They trip up feedparser and HBR for one
     # generates them
     raw_xml = re.sub(r'(&amp;#\d+)([^0-9;])', r'\1;\2', raw_xml)
