Index: libtool-2.2.10/tests/deplibs-ident.at
===================================================================
--- libtool-2.2.10.orig/tests/deplibs-ident.at	2010-06-17 18:36:59.000000000 +0000
+++ libtool-2.2.10/tests/deplibs-ident.at	2010-06-17 18:39:32.000000000 +0000
@@ -66,10 +66,10 @@
   AT_CHECK([$LIBTOOL --mode=link $CC $CFLAGS $LDFLAGS -o b$EXEEXT b.$OBJEXT ../liba1.la ../liba2.la ../liba3.la ../../c/libcee.la -rpath /nowhere],
 	   [0],[stdout],[ignore])
   AT_CHECK([$EGREP 'cee.*cee' stdout], 1, [ignore], [ignore])
-  AT_XFAIL_IF([case $host in
-                 *-*-aix*|hppa*-*-hpux*|*-*-interix*|*-*-openbsd*) false;;
-                 *):;;
-               esac])
+#  AT_XFAIL_IF([case $host in
+#                 *-*-aix*|hppa*-*-hpux*|*-*-interix*|*-*-openbsd*) false;;
+#                 *):;;
+#               esac])
   dnl This is currently broken in libtool
 )
 
