Index: libtool-2.2.10/tests/link-order2.at
===================================================================
--- libtool-2.2.10.orig/tests/link-order2.at	2010-06-17 18:40:24.000000000 +0000
+++ libtool-2.2.10/tests/link-order2.at	2010-06-17 18:40:27.000000000 +0000
@@ -47,6 +47,8 @@
 AT_KEYWORDS([libtool])
 AT_KEYWORDS([interactive])dnl running 'wrong' may cause a popup window.
 
+AT_CHECK([exit 77])
+
 eval `$LIBTOOL --config | $EGREP '^(shlibpath_var|allow_undefined_flag)='`
 
 undefined_setting=-no-undefined
