Index: libtool-2.4/libltdl/config/ltmain.m4sh
===================================================================
--- libtool-2.4.orig/libltdl/config/ltmain.m4sh	2011-08-24 16:13:30.000000000 +0200
+++ libtool-2.4/libltdl/config/ltmain.m4sh	2011-08-24 16:28:49.000000000 +0200
@@ -6548,6 +6548,9 @@
 	    revision="$number_minor"
 	    lt_irix_increment=no
 	    ;;
+	  *)
+	    func_fatal_configuration "$modename: unknown library version type \`$version_type'"
+	    ;;
 	  esac
 	  ;;
 	no)
