Description: fixes man page typos spotted by Lintian
Author: Martin-Éric Racine <martin-eric.racine@iki.fi>
--- pulseaudio-2.0.orig/man/pulse-daemon.conf.5.xml.in
+++ pulseaudio-2.0/man/pulse-daemon.conf.5.xml.in
@@ -267,7 +267,7 @@ USA.
     <option>
       <p><opt>dl-search-path=</opt> The path were to look for dynamic
       shared objects (DSOs/plugins). You may specify more than one
-      path seperated by colons. The default path depends on compile
+      path separated by colons. The default path depends on compile
       time settings. The <opt>--dl-search-path</opt> command line
       option takes precedence. </p>
     </option>
@@ -410,7 +410,7 @@ USA.
       <opt>s16be</opt>, <opt>s24le</opt>, <opt>s24be</opt>,
       <opt>s24-32le</opt>, <opt>s24-32be</opt>, <opt>s32le</opt>,
       <opt>s32be</opt> <opt>float32le</opt>, <opt>float32be</opt>,
-      <opt>ulaw</opt>, <opt>alaw</opt>. Depending on the endianess of
+      <opt>ulaw</opt>, <opt>alaw</opt>. Depending on the endianness of
       the CPU the formats <opt>s16ne</opt>, <opt>s16re</opt>,
       <opt>s24ne</opt>, <opt>s24re</opt>, <opt>s24-32ne</opt>,
       <opt>s24-32re</opt>, <opt>s32ne</opt>, <opt>s32re</opt>,
--- pulseaudio-2.0.orig/man/pacat.1.xml.in
+++ pulseaudio-2.0/man/pacat.1.xml.in
@@ -116,7 +116,7 @@ USA.
       <opt>float32le</opt>, <opt>float32be</opt>, <opt>ulaw</opt>,
       <opt>alaw</opt>, <opt>s32le</opt>, <opt>s32be</opt>,
       <opt>s24le</opt>, <opt>s24be</opt>, <opt>s24-32le</opt>,
-      <opt>s24-32be</opt>. Depending on the endianess of the CPU the
+      <opt>s24-32be</opt>. Depending on the endianness of the CPU the
       formats <opt>s16ne</opt>, <opt>s16re</opt>, <opt>s32ne</opt>,
       <opt>s32re</opt>, <opt>float32ne</opt>, <opt>float32re</opt>,
       <opt>s32ne</opt>, <opt>s32re</opt>, <opt>s24ne</opt>,
