From: Clint Byrum <clint@ubuntu.com>
Subject: Do not setup the terminal if output is not a tty
Bug: https://sourceforge.net/tracker/index.php?func=detail&aid=3044367&group_id=18314&atid=118314
Bug-Ubuntu: https://launchpad.net/bugs/322214
Forwarded: yes, http://mail-index.netbsd.org/tech-userlevel/2011/01/26/msg004486.html

=== modified file 'libedit/tty.c'
--- a/libedit/tty.c	2008-06-14 17:13:08 +0000
+++ b/libedit/tty.c	2011-01-26 21:03:59 +0000
@@ -472,6 +472,9 @@
 	if (el->el_flags & EDIT_DISABLED)
 		return (0);
 
+	if (!isatty(fileno(el->el_outfile)))
+		return (-1);
+
 	if (tty_getty(el, &el->el_tty.t_ed) == -1) {
 #ifdef DEBUG_TTY
 		(void) fprintf(el->el_errfile,

