Description: fix warnings about deprecated headers
Author: Steve Langasek <steve.langasek@canonical.com>
Bug-Debian: http://bugs.debian.org/634324

Index: libedit-2.11-20080614/libedit/Makefile
===================================================================
--- libedit-2.11-20080614.orig/libedit/Makefile
+++ libedit-2.11-20080614/libedit/Makefile
@@ -4,7 +4,7 @@
 OS!=	sh -c "case `uname -s` in GNU|GNU/*) echo GNU ;; *) uname -s ;; esac"
 
 .if ${OS} == GNU || ${OS} == Linux
-CFLAGS += -include bsd/bsd.h
+CFLAGS += -include bsd/sys/cdefs.h -include bsd/stdio.h
 MKLINT=no
 .endif
 
Index: libedit-2.11-20080614/libedit/readline.c
===================================================================
--- libedit-2.11-20080614.orig/libedit/readline.c
+++ libedit-2.11-20080614/libedit/readline.c
@@ -48,7 +48,7 @@
 #include <fcntl.h>
 #include <setjmp.h>
 #ifdef HAVE_VIS_H
-#include <vis.h>
+#include <bsd/vis.h>
 #else
 #include "np/vis.h"
 #endif
Index: libedit-2.11-20080614/libedit/filecomplete.c
===================================================================
--- libedit-2.11-20080614.orig/libedit/filecomplete.c
+++ libedit-2.11-20080614/libedit/filecomplete.c
@@ -47,7 +47,7 @@
 #include <errno.h>
 #include <fcntl.h>
 #ifdef HAVE_VIS_H
-#include <vis.h>
+#include <bsd/vis.h>
 #else
 #include "np/vis.h"
 #endif
Index: libedit-2.11-20080614/libedit/history.c
===================================================================
--- libedit-2.11-20080614.orig/libedit/history.c
+++ libedit-2.11-20080614/libedit/history.c
@@ -48,7 +48,7 @@
 #include <stdlib.h>
 #include <stdarg.h>
 #ifdef HAVE_VIS_H
-#include <vis.h>
+#include <bsd/vis.h>
 #else
 #include "np/vis.h"
 #endif
