Index: lame-3.99/Makefile.in
===================================================================
--- lame-3.99.orig/Makefile.in	2011-04-20 14:53:26.000000000 +0200
+++ lame-3.99/Makefile.in	2011-10-15 17:31:16.314062386 +0200
@@ -249,7 +249,7 @@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 AUTOMAKE_OPTIONS = 1.11 foreign $(top_srcdir)/ansi2knr
-SUBDIRS = mpglib libmp3lame frontend Dll debian doc include misc dshow ACM \
+SUBDIRS = mpglib libmp3lame frontend Dll doc include misc dshow ACM \
 	mac macosx vc_solution
 
 CLEANFILES = testcase.new.mp3
Index: lame-3.99/configure
===================================================================
--- lame-3.99.orig/configure	2011-04-20 14:53:26.000000000 +0200
+++ lame-3.99/configure	2011-10-15 17:30:39.059596480 +0200
@@ -16284,7 +16284,7 @@
 
 
 
-ac_config_files="$ac_config_files Makefile libmp3lame/Makefile libmp3lame/i386/Makefile libmp3lame/vector/Makefile frontend/Makefile mpglib/Makefile doc/Makefile doc/html/Makefile doc/man/Makefile include/Makefile Dll/Makefile misc/Makefile debian/Makefile dshow/Makefile ACM/Makefile ACM/ADbg/Makefile ACM/ddk/Makefile ACM/tinyxml/Makefile lame.spec mac/Makefile macosx/Makefile macosx/English.lproj/Makefile macosx/LAME.xcodeproj/Makefile vc_solution/Makefile"
+ac_config_files="$ac_config_files Makefile libmp3lame/Makefile libmp3lame/i386/Makefile libmp3lame/vector/Makefile frontend/Makefile mpglib/Makefile doc/Makefile doc/html/Makefile doc/man/Makefile include/Makefile Dll/Makefile misc/Makefile dshow/Makefile ACM/Makefile ACM/ADbg/Makefile ACM/ddk/Makefile ACM/tinyxml/Makefile lame.spec mac/Makefile macosx/Makefile macosx/English.lproj/Makefile macosx/LAME.xcodeproj/Makefile vc_solution/Makefile"
 
 
 cat >confcache <<\_ACEOF
@@ -17315,7 +17315,6 @@
     "include/Makefile") CONFIG_FILES="$CONFIG_FILES include/Makefile" ;;
     "Dll/Makefile") CONFIG_FILES="$CONFIG_FILES Dll/Makefile" ;;
     "misc/Makefile") CONFIG_FILES="$CONFIG_FILES misc/Makefile" ;;
-    "debian/Makefile") CONFIG_FILES="$CONFIG_FILES debian/Makefile" ;;
     "dshow/Makefile") CONFIG_FILES="$CONFIG_FILES dshow/Makefile" ;;
     "ACM/Makefile") CONFIG_FILES="$CONFIG_FILES ACM/Makefile" ;;
     "ACM/ADbg/Makefile") CONFIG_FILES="$CONFIG_FILES ACM/ADbg/Makefile" ;;
