Index: lame-3.99/libmp3lame/i386/nasm.h
===================================================================
--- lame-3.99.orig/libmp3lame/i386/nasm.h	2011-05-07 17:35:27.000000000 +0200
+++ lame-3.99/libmp3lame/i386/nasm.h	2011-10-15 17:31:39.393111989 +0200
@@ -20,6 +20,8 @@
 
 section .note.GNU-stack noalloc noexec nowrite progbits
 
+section .note.GNU-stack noalloc noexec nowrite progbits
+
 %ifdef YASM
 	%define segment_code segment .text align=16 use32
 	%define segment_data segment .data align=16 use32
