Fixes misspelling of "Windows" in codepage list (LP: #649796)

Index: links2/Unicode/cp1250.cp
===================================================================
--- links2.orig/Unicode/cp1250.cp	2011-08-19 00:25:46.000000000 +0200
+++ links2/Unicode/cp1250.cp	2011-08-19 00:26:14.000000000 +0200
@@ -1,4 +1,4 @@
-Window$ 1250 (Central European)
+Windows 1250 (Central European)
 "windows-1250", "1250"
 #
 #    Name:     cp1250 to Unicode table
Index: links2/Unicode/cp1251.cp
===================================================================
--- links2.orig/Unicode/cp1251.cp	2011-08-19 00:25:46.000000000 +0200
+++ links2/Unicode/cp1251.cp	2011-08-19 00:26:14.000000000 +0200
@@ -1,4 +1,4 @@
-Window$ 1251 (Cyrillic)
+Windows 1251 (Cyrillic)
 "windows-1251", "1251", "cp1251"
 
 0x20	0x0020	# SPACE
Index: links2/Unicode/cp1252.cp
===================================================================
--- links2.orig/Unicode/cp1252.cp	2011-08-19 00:25:46.000000000 +0200
+++ links2/Unicode/cp1252.cp	2011-08-19 00:26:14.000000000 +0200
@@ -1,4 +1,4 @@
-Window$ 1252 (Western European)
+Windows 1252 (Western European)
 "windows-1252", "1252"
 
 0x00	0x0000	#NULL
Index: links2/Unicode/cp1256.cp
===================================================================
--- links2.orig/Unicode/cp1256.cp	2011-08-19 00:25:46.000000000 +0200
+++ links2/Unicode/cp1256.cp	2011-08-19 00:26:14.000000000 +0200
@@ -1,4 +1,4 @@
-Window$ 1256 (Arabic)
+Windows 1256 (Arabic)
 "windows-1256", "1256", "cp1256"
 #
 #    Name:     cp1256 to Unicode table
Index: links2/Unicode/cp1257.cp
===================================================================
--- links2.orig/Unicode/cp1257.cp	2011-08-19 00:25:46.000000000 +0200
+++ links2/Unicode/cp1257.cp	2011-08-19 00:26:21.000000000 +0200
@@ -1,4 +1,4 @@
-Window$ 1257 (Baltic)
+Windows 1257 (Baltic)
 "windows-1257", "1257", "cp1257"
 #
 #    Name:     cp1257 to Unicode table
Index: links2/codepage.inc
===================================================================
--- links2.orig/codepage.inc	2011-08-19 00:25:46.000000000 +0200
+++ links2/codepage.inc	2011-08-19 00:26:14.000000000 +0200
@@ -4287,11 +4287,11 @@
 	{ "ISO 8859-14 (Celtic)", aliases_8859_14, table_8859_14 },
 	{ "ISO 8859-15 (Western European)", aliases_8859_15, table_8859_15 },
 	{ "ISO 8859-16 (Romainian)", aliases_8859_16, table_8859_16 },
-	{ "Window$ 1250 (Central European)", aliases_cp1250, table_cp1250 },
-	{ "Window$ 1251 (Cyrillic)", aliases_cp1251, table_cp1251 },
-	{ "Window$ 1252 (Western European)", aliases_cp1252, table_cp1252 },
-	{ "Window$ 1256 (Arabic)", aliases_cp1256, table_cp1256 },
-	{ "Window$ 1257 (Baltic)", aliases_cp1257, table_cp1257 },
+	{ "Windows 1250 (Central European)", aliases_cp1250, table_cp1250 },
+	{ "Windows 1251 (Cyrillic)", aliases_cp1251, table_cp1251 },
+	{ "Windows 1252 (Western European)", aliases_cp1252, table_cp1252 },
+	{ "Windows 1256 (Arabic)", aliases_cp1256, table_cp1256 },
+	{ "Windows 1257 (Baltic)", aliases_cp1257, table_cp1257 },
 	{ "CP 437 (US)", aliases_cp437, table_cp437 },
 	{ "CP 737 (Greek)", aliases_cp737, table_cp737 },
 	{ "CP 850 (Western European)", aliases_cp850, table_cp850 },
