Use x-terminal-emulator instead of xterm.

Index: links2/os_dep.c
===================================================================
--- links2.orig/os_dep.c	2012-06-26 14:40:07.000000000 +0200
+++ links2/os_dep.c	2012-06-26 14:41:18.000000000 +0200
@@ -2289,7 +2289,7 @@
 static int open_in_new_xterm(struct terminal *term, unsigned char *exe, unsigned char *param)
 {
 	unsigned char *xterm;
-	if (!(xterm = cast_uchar getenv("LINKS_XTERM"))) xterm = cast_uchar "xterm -e";
+	if (!(xterm = cast_uchar getenv("LINKS_XTERM"))) xterm = cast_uchar "x-terminal-emulator -e";
 	exec_new_links(term, xterm, exe, param);
 	return 0;
 }
Index: links2/x.c
===================================================================
--- links2.orig/x.c	2012-06-26 14:40:07.000000000 +0200
+++ links2/x.c	2012-06-26 14:41:18.000000000 +0200
@@ -2352,7 +2352,7 @@
 		return retval;
 	}
 
-	run=subst_file(*x_driver.shell?x_driver.shell:(unsigned char *)"xterm -e %",command, 0);
+	run=subst_file(*x_driver.shell?x_driver.shell:(unsigned char *)"x-terminal-emulator -e %",command, 0);
 	errno = 0;
 	EINTRLOOP(retval, system(cast_const_char run));
 	mem_free(run);
