Subject: Attemp to adapt build flags to built package on all supported 
 architectures.
Origin: Debian Med Packaging Team <debian-med-packaging@lists.alioth.debian.org>
Date: 2012-05-09
--- bowtie2.orig/Makefile
+++ bowtie2/Makefile
@@ -108,20 +108,29 @@
 VERSION = $(shell cat VERSION)
 EXTRA_FLAGS =
 
-# Convert BITS=?? to a -m flag
-BITS=32
-ifeq (x86_64,$(shell uname -m))
-BITS=64
-endif
-BITS_FLAG =
-ifeq (32,$(BITS))
-BITS_FLAG = -m32
-endif
-ifeq (64,$(BITS))
-BITS_FLAG = -m64
-endif
+ifeq (1,$(ENABLE_SSE))
+ ARCH := $(shell getconf LONG_BIT)
+ UNAME := $(shell uname -m )
+ # Convert BITS=?? to a -m flag
+ BITS=32
+ BITS_FLAG = -m32
+ ifeq (64,$(ARCH))
+	BITS=64
+	BITS_FLAG = -m64
+ endif
+ ifeq ($(UNAME),'Power Macintosh')
+	ifeq (32,$(BITS))
+		BITS_FLAG = -m32
+		SSE_FLAG = 
+	endif
+	ifeq (64,$(BITS))
+		BITS_FLAG = -m64
+		SSE_FLAG =
+	endif
+ endif
 
-SSE_FLAG=-msse2
+ SSE_FLAG = -msse2
+endif
 
 DEBUG_FLAGS    = -O0 -g3 $(BITS_FLAG) $(SSE_FLAG)
 DEBUG_DEFS     = -DCOMPILER_OPTIONS="\"$(DEBUG_FLAGS) $(EXTRA_FLAGS)\""
