--- a/bw-simulation/BW-Film-Simulation-1.1.scm
+++ b/bw-simulation/BW-Film-Simulation-1.1.scm
@@ -155,7 +155,7 @@
 		; Kodak HIE
 		((= film 15)  
 		 (set! chan-name (string-append chan-name " Kodak HIE"))
-		 (plug-in-colors-channel-mixer 1 img bw-layer TRUE 1.0 1.0 -1.0 0.0 1.0 1.0 -1.0 0.0 1.0 1.0 -1.0 0.0 ))
+		 (plug-in-colors-channel-mixer 1 img bw-layer TRUE 1.0 1.0 -1.0 0.0 1.0 1.0 -1.0 0.0 1.0))
 		; Normal Contrast
 		((= film 16)  
 		 (set! chan-name (string-append chan-name " Normal Contrast"))
@@ -171,7 +171,7 @@
 		; 50-50
 		((= film 19)  
 		 (set! chan-name (string-append chan-name " 50-50"))
-		 (plug-in-colors-channel-mixer 1 img bw-layer TRUE 0.5 0.5 0.00 0.0 0.5 0.5 0.00 0.0 0.5 0.5 0.00 0.0 ))
+		 (plug-in-colors-channel-mixer 1 img bw-layer TRUE 0.5 0.5 0.00 0.0 0.5 0.5 0.00 0.0 0.5 ))
 		)    
 
 
