--- a/fix-ca/fix-ca.c
+++ b/fix-ca/fix-ca.c
@@ -138,13 +138,7 @@ void	query (void)
 				G_N_ELEMENTS (args), 0,
 				args, 0);
 
-#if 0
-				/* Need to decide about menu location */
-	if (GIMP_CHECK_VERSION(2, 4, 0))
 		gimp_plugin_menu_register (PROCEDURE_NAME, "<Image>/Colors");
-	else
-#endif
-		gimp_plugin_menu_register (PROCEDURE_NAME, "<Image>/Filters/Colors");
 }
 
 void	run (const gchar *name, gint nparams,
