diff -urNad gimp-plugin-registry~/bw-simulation/BW-Film-Simulation-1.1.scm gimp-plugin-registry/bw-simulation/BW-Film-Simulation-1.1.scm
--- gimp-plugin-registry~/bw-simulation/BW-Film-Simulation-1.1.scm	2007-08-17 15:21:20.000000000 +0200
+++ gimp-plugin-registry/bw-simulation/BW-Film-Simulation-1.1.scm	2007-09-08 00:21:13.000000000 +0200
@@ -192,7 +192,7 @@
       
       "script-fu-bw-film"
  
-      "<Image>/Script-Fu/BW Film Simulation" 
+      "<Image>/Colors/BW Film Simulation" 
       "Black and White Film Simulation"
       "Serge Mankovski (sevato@mankvoski.com)"
       "2007, Serge Mankovski, Toronto, ON, Canada"
