diff -urNad gimp-plugin-registry~/refocus/src/src/gimppreview.c gimp-plugin-registry/refocus/src/src/gimppreview.c
--- gimp-plugin-registry~/refocus/src/src/gimppreview.c	2006-12-16 23:28:49.000000000 +0100
+++ gimp-plugin-registry/refocus/src/src/gimppreview.c	2007-10-15 15:04:05.000000000 +0200
@@ -98,7 +98,7 @@
  * data!
  */
 #define PREVIEW_DATA(preview) \
-        ((RefocusPreviewData*)(GIMP_PREVIEW (preview)->private_data))
+        ((RefocusPreviewData*)(RF_GIMP_PREVIEW (preview)->private_data))
 
 typedef struct
 {
@@ -273,7 +273,7 @@
 {
   gchar buffer[10];
 
-  g_assert (GIMP_IS_PREVIEW (preview));
+  g_assert (RF_GIMP_IS_PREVIEW (preview));
   GTK_WIDGET_SET_FLAGS (preview, GTK_NO_WINDOW);
   GTK_CONTAINER (preview)->resize_mode = GTK_RESIZE_IMMEDIATE;
 
@@ -337,7 +337,7 @@
 
 
   /* Now allocate the actual preview window. */
-  preview = GIMP_PREVIEW (g_object_new (gimp_preview_get_type (), NULL));
+  preview = RF_GIMP_PREVIEW (g_object_new (gimp_preview_get_type (), NULL));
 
   /* Set the scale amount. */
   gimp_preview_set_scale_amount(preview, cb_scale_amount);
@@ -352,8 +352,8 @@
       preview_width = i2p (drawable->width, preview->scale);
       preview_height = i2p (drawable->height, preview->scale);
 
-      GIMP_PREVIEW (preview)->width = preview_width;
-      GIMP_PREVIEW (preview)->height = preview_height;
+      RF_GIMP_PREVIEW (preview)->width = preview_width;
+      RF_GIMP_PREVIEW (preview)->height = preview_height;
     }
   else
     {
@@ -454,7 +454,7 @@
 static void
 gimp_preview_size_request (GtkWidget * widget, GtkRequisition * requisition)
 {
-  RefocusPreview *preview = GIMP_PREVIEW (widget);
+  RefocusPreview *preview = RF_GIMP_PREVIEW (widget);
   GtkRequisition resize_box_requisition;
 
 #ifdef PREVIEW_DEBUG
@@ -478,7 +478,7 @@
 static void
 gimp_preview_size_allocate (GtkWidget * widget, GtkAllocation * allocation)
 {
-  RefocusPreview *preview = GIMP_PREVIEW (widget);
+  RefocusPreview *preview = RF_GIMP_PREVIEW (widget);
   GtkAllocation resize_box_allocation, progress_bar_allocation,
     event_box_allocation;
   GtkRequisition resize_box_requisition;
@@ -543,7 +543,7 @@
                      gboolean include_internals,
                      GtkCallback callback, gpointer callback_data)
 {
-  RefocusPreview *preview = GIMP_PREVIEW (container);
+  RefocusPreview *preview = RF_GIMP_PREVIEW (container);
 
   if (PREVIEW_DATA (preview)->resize_box)
     {
@@ -583,7 +583,7 @@
   gchar buffer[10];
   gdouble new_scale;
 
-  preview = GIMP_PREVIEW (data);
+  preview = RF_GIMP_PREVIEW (data);
   if (PREVIEW_DATA (preview)->scale_n == PREVIEW_SCALE_LAST)
     return;
 
@@ -620,7 +620,7 @@
   gchar buffer[10];
   gdouble new_scale;
 
-  preview = GIMP_PREVIEW (data);
+  preview = RF_GIMP_PREVIEW (data);
   if (PREVIEW_DATA (preview)->scale_n == 0)
     return;
 
@@ -656,7 +656,7 @@
   gint x, y;
   gint dx, dy;
 
-  preview = GIMP_PREVIEW (data);
+  preview = RF_GIMP_PREVIEW (data);
   button_event = (GdkEventButton *) event;
 
   switch (event->type)
@@ -1022,7 +1022,7 @@
 gboolean
 gimp_preview_update_preview_idle_fun (gpointer data)
 {
-  RefocusPreview *preview = GIMP_PREVIEW (data);
+  RefocusPreview *preview = RF_GIMP_PREVIEW (data);
   gint event_id = PREVIEW_DATA (preview)->current_event_id;
 
 #ifdef PREVIEW_DEBUG
diff -urNad gimp-plugin-registry~/refocus/src/src/gimppreview.h gimp-plugin-registry/refocus/src/src/gimppreview.h
--- gimp-plugin-registry~/refocus/src/src/gimppreview.h	2006-12-16 23:00:31.000000000 +0100
+++ gimp-plugin-registry/refocus/src/src/gimppreview.h	2007-10-15 15:04:05.000000000 +0200
@@ -20,8 +20,8 @@
  * Version $Id: gimppreview.h,v 1.1.1.1 2003/01/30 21:30:18 ernstl Exp $
  */
 
-#ifndef GIMP_PREVIEW_H
-#define GIMP_PREVIEW_H
+#ifndef RF_GIMP_PREVIEW_H
+#define RF_GIMP_PREVIEW_H
 
 #include <gtk/gtk.h>
 #include <gdk-pixbuf/gdk-pixbuf.h>
@@ -33,10 +33,10 @@
 #define PREVIEW_FIXED_SIZE           0
 #define PREVIEW_DEFAULT_SIZE         -1
 #define GIMP_TYPE_PREVIEW            (gimp_preview_get_type ())
-#define GIMP_PREVIEW(obj)            (GTK_CHECK_CAST ((obj), GIMP_TYPE_PREVIEW, RefocusPreview))
-#define GIMP_PREVIEW_CLASS(klass)    (GTK_CHECK_CLASS_CAST ((klass), GIMP_TYPE_PREVIEW, RefocusPreviewClass))
-#define GIMP_IS_PREVIEW(obj)         (GTK_CHECK_TYPE ((obj), GIMP_TYPE_PREVIEW))
-#define GIMP_IS_PREVIEW_CLASS(klass) (GTK_CHECK_CLASS_TYPE ((klass), GIMP_TYPE_PREVIEW))
+#define RF_GIMP_PREVIEW(obj)            (GTK_CHECK_CAST ((obj), GIMP_TYPE_PREVIEW, RefocusPreview))
+#define RF_GIMP_PREVIEW_CLASS(klass)    (GTK_CHECK_CLASS_CAST ((klass), GIMP_TYPE_PREVIEW, RefocusPreviewClass))
+#define RF_GIMP_IS_PREVIEW(obj)         (GTK_CHECK_TYPE ((obj), GIMP_TYPE_PREVIEW))
+#define RF_GIMP_IS_PREVIEW_CLASS(klass) (GTK_CHECK_CLASS_TYPE ((klass), GIMP_TYPE_PREVIEW))
 typedef struct _RefocusPreview RefocusPreview;
 typedef struct _RefocusPreviewClass RefocusPreviewClass;
 typedef struct _RefocusPreviewEvent RefocusPreviewEvent;
@@ -131,4 +131,4 @@
                                              double fraction);
 
 G_END_DECLS
-#endif /* __GIMP_PREVIEW_H__ */
+#endif /* __RF_GIMP_PREVIEW_H__ */
diff -urNad gimp-plugin-registry~/refocus/src/src/refocus.c gimp-plugin-registry/refocus/src/src/refocus.c
--- gimp-plugin-registry~/refocus/src/src/refocus.c	2006-12-16 23:28:13.000000000 +0100
+++ gimp-plugin-registry/refocus/src/src/refocus.c	2007-10-15 15:04:05.000000000 +0200
@@ -337,7 +337,7 @@
   (void) data;
   set_busy_cursor (widget, TRUE);
   update_matrix ();
-  gimp_preview_update (GIMP_PREVIEW (my_widgets.preview));
+  gimp_preview_update (RF_GIMP_PREVIEW (my_widgets.preview));
   set_busy_cursor (widget, FALSE);
 }
 
@@ -370,7 +370,7 @@
 {
   gint event_id = GPOINTER_TO_INT (data);
   return (gimp_preview_progress_set_fraction
-          (GIMP_PREVIEW (my_widgets.preview), event_id, arg));
+          (RF_GIMP_PREVIEW (my_widgets.preview), event_id, arg));
 }
 
 static void
@@ -394,7 +394,7 @@
                                       im_width, im_height);
       tile_sink_init_for_preview (&sink, drawable, image_x, image_y,
                                   im_width, im_height);
-      gimp_preview_progress_set_fraction (GIMP_PREVIEW (my_widgets.preview),
+      gimp_preview_progress_set_fraction (RF_GIMP_PREVIEW (my_widgets.preview),
                                           event->event_id, 0);
       bd_closure_init (&update_progress_closure,
                        preview_progress_update_fun,
@@ -410,7 +410,7 @@
         {
           tile_sink_get_row (&sink, buf, image_x, image_y + row, im_width);
           event_is_current =
-            gimp_preview_draw_unscaled_row (GIMP_PREVIEW (my_widgets.preview),
+            gimp_preview_draw_unscaled_row (RF_GIMP_PREVIEW (my_widgets.preview),
                                             event->event_id,
                                             gimp_drawable_type (drawable->drawable_id),
                                             row, buf);
@@ -584,7 +584,7 @@
 
   gtk_widget_show (dlg);
   redraw_all ();
-  gimp_preview_update (GIMP_PREVIEW (preview));
+  gimp_preview_update (RF_GIMP_PREVIEW (preview));
   gtk_main ();
   gdk_flush ();
   return run_flag;
