diff -urNad gimp-plugin-registry~/refocus/src/src/fwlapack.c gimp-plugin-registry/refocus/src/src/fwlapack.c
--- gimp-plugin-registry~/refocus/src/src/fwlapack.c	2003-01-30 22:30:18.000000000 +0100
+++ gimp-plugin-registry/refocus/src/src/fwlapack.c	2007-10-15 16:06:45.000000000 +0200
@@ -28,36 +28,17 @@
 static char vcid[] GCC_UNUSED = "$Id: fwlapack.c,v 1.1.1.1 2003/01/30 21:30:18 ernstl Exp $";
 #endif /* lint */
 
-#ifdef HAVE_ATLAS
-/* Fix stupid bugs in the atlas header file that use ATLAS_* instead of CBLAS_ */
-#define ATLAS_ORDER CBLAS_ORDER
-#define ATLAS_UPLO CBLAS_UPLO
-#define ATLAS_DIAG CBLAS_DIAG
-
-#include "clapack.h"
-
-#else /* Not HAVE_ATLAS */
-
-#include "f2c.h"
-#include "clapack.h"
-#endif
 
 int
 dgesv (const int N, const int NRHS, double *A, const int lda,
        double *B, const int ldb)
 {
   int result = 0;
-#ifdef HAVE_ATLAS
-  int *ipiv = g_new (int, N);
-  result = clapack_dgesv (CblasColMajor, N, NRHS, A, lda, ipiv, B, ldb);
-  g_free (ipiv);
-#else
   integer i_N = N, i_NHRS = NRHS, i_lda = lda, i_ldb = ldb, info;
   integer *ipiv = g_new (integer, N);
 
   dgesv_ (&i_N, &i_NHRS, A, &i_lda, ipiv, B, &i_ldb, &info);
   g_free (ipiv);
   result = info;
-#endif
   return (result);
 }
diff -urNad gimp-plugin-registry~/refocus/src/src/fwlapack.h gimp-plugin-registry/refocus/src/src/fwlapack.h
--- gimp-plugin-registry~/refocus/src/src/fwlapack.h	2003-01-30 22:30:18.000000000 +0100
+++ gimp-plugin-registry/refocus/src/src/fwlapack.h	2007-10-15 16:07:40.000000000 +0200
@@ -20,12 +20,17 @@
 #ifndef FWLAPACK_H_INCLUDED
 #include <glib.h>
 
+#include "f2c.h"
+
 G_BEGIN_DECLS
 
 /* Wrapper function for lapack dgesv function */
 int dgesv (const int N, const int NRHS, double *A, const int lda,
            double *B, const int ldb);
 
+extern int dgesv_(integer *n, integer *nrhs, doublereal *a, integer 
+	*lda, integer *ipiv, doublereal *b, integer *ldb, integer *info);
+
 G_END_DECLS
 #define FWLAPACK_H_INCLUDED (1)
 #endif
