From: Michael Casadevall <mcasadevall@debian.org>
Subject: Fix FTBFS on armel
Bug-Debian: http://bugs.debian.org/563972
Index: fontmatrix-0.6.0+svn20100107/src/fminfodisplay.cpp
===================================================================
--- fontmatrix-0.6.0+svn20100107.orig/src/fminfodisplay.cpp	2010-01-07 11:16:21.000000000 -0600
+++ fontmatrix-0.6.0+svn20100107/src/fminfodisplay.cpp	2010-01-07 11:16:50.000000000 -0600
@@ -151,7 +151,7 @@
 				GlyphToSVGHelper gtsh ( gpi->path(), tf );
 				svg += gtsh.getSVGPath() + "\n";
 				horOffset += gpi->data(GLYPH_DATA_HADVANCE).toDouble() * scaleFactor;
-				maxHeight = qMax ( gtsh.getRect().height(), maxHeight );
+				maxHeight = qMax<qreal> ( gtsh.getRect().height(), maxHeight );
 				tf.translate( gpi->data(GLYPH_DATA_HADVANCE).toDouble()  * scaleFactor,0 );
 				delete gpi;
 			}
