Author: William Vera <billy@billy.com.mx>
Description: Fix spelling error on source.
--- a/arp-scan.c	2011-07-04 20:15:38.967999123 -0500
+++ b/arp-scan.c	2011-07-04 20:16:52.859999102 -0500
@@ -947,7 +947,7 @@
       fprintf(stderr, "<s> A character string, e.g. --file=hostlist.txt.\n");
       fprintf(stderr, "\n");
       fprintf(stderr, "<i> An integer, which can be specified as a decimal number or as a hexadecimal\n");
-      fprintf(stderr, "    number if preceeded with 0x, e.g. --arppro=2048 or --arpro=0x0800.\n");
+      fprintf(stderr, "    number if preceded with 0x, e.g. --arppro=2048 or --arpro=0x0800.\n");
       fprintf(stderr, "\n");
       fprintf(stderr, "<f> A floating point decimal number, e.g. --backoff=1.5.\n");
       fprintf(stderr, "\n");
