Author: Mats Erik Andersson <mats.andersson@gisladisker.se>
Description: Fix FTBFS on GNU/kFreeBSD.
--- arp-scan-1.8.1.debian/arp-scan.h	2011-02-25 22:29:48.000000000 +0100
+++ arp-scan-1.8.1/arp-scan.h	2011-09-14 21:18:08.000000000 +0200
@@ -120,7 +120,7 @@
  *
  * 5370 is the value of __APPLE_CC__ for Xcode 2.5 on Tiger with GCC 4.0.1
  */
-#if defined(__APPLE_CC__) && (__APPLE_CC__ >= 5370)
+#if defined(__APPLE_CC__) && (__APPLE_CC__ >= 5370) || defined __FreeBSD_kernel__
 #include <net/bpf.h>
 #endif
 #include <pcap.h>
