# DP: ARM hard/soft float multilib support

Index: b/src/gcc/config/arm/t-linux-eabi
===================================================================
--- a/src/gcc/config/arm/t-linux-eabi
+++ b/src/gcc/config/arm/t-linux-eabi
@@ -21,6 +21,20 @@
 MULTILIB_OPTIONS	=
 MULTILIB_DIRNAMES	=
 
+ifeq ($(with_float),hard)
+MULTILIB_OPTIONS	= mfloat-abi=soft/mfloat-abi=hard
+MULTILIB_DIRNAMES	= sf hf
+MULTILIB_EXCEPTIONS	=
+MULTILIB_MATCHES	= mfloat-abi?hard=mhard-float mfloat-abi?soft=msoft-float mfloat-abi?soft=mfloat-abi?softfp
+MULTILIB_OSDIRNAMES	= arm-linux-gnueabi:arm-linux-gnueabi ../lib:arm-linux-gnueabihf
+else
+MULTILIB_OPTIONS	= mfloat-abi=soft/mfloat-abi=hard
+MULTILIB_DIRNAMES	= sf hf
+MULTILIB_EXCEPTIONS	=
+MULTILIB_MATCHES	= mfloat-abi?hard=mhard-float mfloat-abi?soft=msoft-float mfloat-abi?soft=mfloat-abi?softfp
+MULTILIB_OSDIRNAMES	= ../lib:arm-linux-gnueabi arm-linux-gnueabihf:arm-linux-gnueabihf
+endif
+
 #MULTILIB_OPTIONS     += mcpu=fa606te/mcpu=fa626te/mcpu=fmp626/mcpu=fa726te
 #MULTILIB_DIRNAMES    += fa606te fa626te fmp626 fa726te
 #MULTILIB_EXCEPTIONS  += *mthumb/*mcpu=fa606te *mthumb/*mcpu=fa626te *mthumb/*mcpu=fmp626 *mthumb/*mcpu=fa726te*
