--- a/src/libgcc/config/rs6000/crtsavfpr.S
+++ b/src/libgcc/config/rs6000/crtsavfpr.S
@@ -33,6 +33,7 @@
 
 /* On PowerPC64 Linux, these functions are provided by the linker.  */
 #ifndef __powerpc64__
+#ifndef __NO_FPRS__
 
 /* Routines for saving floating point registers, called by the compiler.  */
 /* Called with r11 pointing to the stack header word of the caller of the */
@@ -79,3 +80,4 @@
 CFI_ENDPROC
 
 #endif
+#endif
--- a/src/libgcc/config/rs6000/crtresfpr.S
+++ b/src/libgcc/config/rs6000/crtresfpr.S
@@ -33,6 +33,7 @@
 
 /* On PowerPC64 Linux, these functions are provided by the linker.  */
 #ifndef __powerpc64__
+#ifndef __NO_FPRS__
 
 /* Routines for restoring floating point registers, called by the compiler.  */
 /* Called with r11 pointing to the stack header word of the caller of the */
@@ -79,3 +80,4 @@
 CFI_ENDPROC
 
 #endif
+#endif
--- a/src/libgcc/config/rs6000/crtresxfpr.S
+++ b/src/libgcc/config/rs6000/crtresxfpr.S
@@ -33,6 +33,7 @@
 
 /* On PowerPC64 Linux, these functions are provided by the linker.  */
 #ifndef __powerpc64__
+#ifndef __NO_FPRS__
 
 /* Routines for restoring floating point registers, called by the compiler.  */
 /* Called with r11 pointing to the stack header word of the caller of the */
@@ -124,3 +125,4 @@
 CFI_ENDPROC
 
 #endif
+#endif
