Index: icewm-1.3.7/src/yarray.h
===================================================================
--- icewm-1.3.7.orig/src/yarray.h	2010-10-31 14:09:36.000000000 +0000
+++ icewm-1.3.7/src/yarray.h	2012-05-21 01:46:52.970218411 +0000
@@ -152,7 +152,7 @@
     }
 
     virtual void remove(const typename YArray<DataType *>::SizeType index) {
-        if (index < YArray<DataType *>::getCount()) delete getItem(index);
+        if (index < YArray<DataType *>::getCount()) delete this->getItem(index);
         YArray<DataType *>::remove(index);
     }
     
