--- itk3-3.3.orig/configure.in
+++ itk3-3.3/config/configure.in
@@ -86,7 +86,7 @@
 TEA_ADD_INCLUDES([${ITCL_INCLUDES} -I\"`${CYGPATH} ${srcdir}/generic`\"])
 TEA_ADD_LIBS([])
 TEA_ADD_CFLAGS([])
-#TEA_ADD_STUB_SOURCES([itkStubLib.c])
+TEA_ADD_STUB_SOURCES([itkStubLib.c])
 TEA_ADD_TCL_SOURCES([library/Archetype.itk
 		library/Toplevel.itk
 		library/Widget.itk
--- itk3-3.3.orig/configure
+++ itk3-3.3/config/configure.
@@ -6472,7 +6472,31 @@
     PKG_CFLAGS="$PKG_CFLAGS "
 
 
-#TEA_ADD_STUB_SOURCES([itkStubLib.c])
+
+    vars="itkStubLib.c"
+    for i in $vars; do
+	# check for existence - allows for generic/win/unix VPATH
+	if test ! -f "${srcdir}/$i" -a ! -f "${srcdir}/generic/$i" \
+	    -a ! -f "${srcdir}/win/$i" -a ! -f "${srcdir}/unix/$i" \
+	    ; then
+	    { { echo "$as_me:$LINENO: error: could not find stub source file '$i'" >&5
+echo "$as_me: error: could not find stub source file '$i'" >&2;}
+   { (exit 1); exit 1; }; }
+	fi
+	PKG_STUB_SOURCES="$PKG_STUB_SOURCES $i"
+	# this assumes it is in a VPATH dir
+	i=`basename $i`
+	# handle user calling this before or after TEA_SETUP_COMPILER
+	if test x"${OBJEXT}" != x ; then
+	    j="`echo $i | sed -e 's/\.[^.]*$//'`.${OBJEXT}"
+	else
+	    j="`echo $i | sed -e 's/\.[^.]*$//'`.\${OBJEXT}"
+	fi
+	PKG_STUB_OBJECTS="$PKG_STUB_OBJECTS $j"
+    done
+
+
+
 
     vars="library/Archetype.itk
 		library/Toplevel.itk
--- itk3-3.3.orig/Makefile.in
+++ itk3-3.3/config/Makefile.in
@@ -77,7 +77,7 @@
 PKG_LIB_FILE	= @PKG_LIB_FILE@
 PKG_STUB_LIB_FILE = @PKG_STUB_LIB_FILE@
 
-lib_BINARIES	= $(PKG_LIB_FILE)
+lib_BINARIES	= $(PKG_LIB_FILE) $(PKG_STUB_LIB_FILE)
 BINARIES	= $(lib_BINARIES)
 
 SHELL		= @SHELL@
