Index: pitivi/bin/pitivi.in
===================================================================
--- pitivi.orig/bin/pitivi.in	2011-09-28 18:57:27.773150417 +0200
+++ pitivi/bin/pitivi.in	2011-09-28 18:57:42.765224752 +0200
@@ -38,13 +38,6 @@
 # Add the path of pitivi stuff
 # TODO : change it when it's finally in cvs
 
-def _get_root_dir():
-    return '/'.join(os.path.dirname(os.path.abspath(__file__)).split('/')[:-1])
-
-def _in_devel():
-    rd = _get_root_dir()
-    return os.path.exists(os.path.join(rd, '.git'))
-
 def _prepend_env_path(name, value):
     os.environ[name] = os.pathsep.join(value +
             os.environ.get(name, "").split(os.pathsep))
@@ -56,13 +49,8 @@
 def _add_pitivi_path():
     global localedir
     dir = os.path.dirname(os.path.abspath(__file__))
-    root = None
-    if _in_devel():
-        root = os.path.split(dir)[0]
-        localedir = os.path.join(os.path.split(dir)[0], 'locale')
-    else:
-        root = os.path.join(LIBDIR, 'pitivi', 'python')
-        localedir = "@DATADIR@/locale"
+    root = os.path.join(LIBDIR, 'pitivi', 'python')
+    localedir = "@DATADIR@/locale"
 
     if not root in sys.path:
         sys.path.insert(0, root)
