--- src/th-app-window.c.old	2006-04-30 10:23:31.709865344 +0200
+++ src/th-app-window.c	2006-04-30 10:24:43.997875896 +0200
@@ -676,7 +676,7 @@
 	{
 		gsize llen;
 		g_string_truncate (fn, strstr (fn->str, "thoggen") - fn->str);
-		g_string_append (fn, "doc/thoggen/COPYING");
+		g_string_append (fn, "common-licenses/GPL");
 		if (!g_file_get_contents (fn->str, &license, &llen, NULL))
 		{
 			gchar *cmd, *stderr = NULL;
